<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Session 6 – Basic Statistics in R | Bioinformatics Workshop Gitbook</title>
  <meta name="description" content="none" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Session 6 – Basic Statistics in R | Bioinformatics Workshop Gitbook" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="none" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Session 6 – Basic Statistics in R | Bioinformatics Workshop Gitbook" />
  
  <meta name="twitter:description" content="none" />
  

<meta name="author" content="Juan C. Santos &amp; Randy Ortiz" />


<meta name="date" content="2023-02-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="getting-biological-data-from-public-repositories.html"/>
<link rel="next" href="regressions-and-anova-in-r.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics Workshop Gitbook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html"><i class="fa fa-check"></i><b>1</b> – Getting Started with R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#why-do-we-use-r"><i class="fa fa-check"></i><b>1.1</b> Why do we use R?</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#get-a-good-text-editor"><i class="fa fa-check"></i><b>1.2</b> Get a good text editor</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-r"><i class="fa fa-check"></i><b>1.4</b> Installing R</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#annotating-code"><i class="fa fa-check"></i><b>1.5</b> Annotating code</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#defining-a-working-directory-in-r"><i class="fa fa-check"></i><b>1.6</b> Defining a working directory in R</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html"><i class="fa fa-check"></i><b>2</b> – Using R, Installing Packages and Importing/Exporting Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#your-first-session"><i class="fa fa-check"></i><b>2.1</b> Your first session</a></li>
<li class="chapter" data-level="2.2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#doing-some-base-functions-and-getting-help"><i class="fa fa-check"></i><b>2.2</b> Doing some base functions and getting help</a></li>
<li class="chapter" data-level="2.3" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#functions-between-two-or-more-vectors"><i class="fa fa-check"></i><b>2.3</b> Functions between two or more vectors</a></li>
<li class="chapter" data-level="2.4" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#about-r-packages"><i class="fa fa-check"></i><b>2.4</b> About R-packages</a></li>
<li class="chapter" data-level="2.5" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#installing-packages"><i class="fa fa-check"></i><b>2.5</b> Installing packages</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html"><i class="fa fa-check"></i><b>3</b> – Basic Data Structures in R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#scalars-and-vectors"><i class="fa fa-check"></i><b>3.1</b> Scalars and vectors</a></li>
<li class="chapter" data-level="3.2" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#arrays"><i class="fa fa-check"></i><b>3.3</b> Arrays</a></li>
<li class="chapter" data-level="3.4" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a></li>
<li class="chapter" data-level="3.5" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tables"><i class="fa fa-check"></i><b>3.5</b> Tables</a></li>
<li class="chapter" data-level="3.6" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#lists"><i class="fa fa-check"></i><b>3.6</b> Lists</a></li>
<li class="chapter" data-level="3.7" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-tables"><i class="fa fa-check"></i><b>3.7</b> Data Tables</a></li>
<li class="chapter" data-level="3.8" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tibbles"><i class="fa fa-check"></i><b>3.8</b> Tibbles</a></li>
<li class="chapter" data-level="3.9" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#dnastringsets-rnastringsets-and-aastringsets"><i class="fa fa-check"></i><b>3.9</b> DNAStringSets, RNAStringSets and AAStringSets</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html"><i class="fa fa-check"></i><b>4</b> – Text Editing and Data Transformations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#working-with-character-strings"><i class="fa fa-check"></i><b>4.1</b> Working with character strings</a></li>
<li class="chapter" data-level="4.2" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#regular-expressions-regex"><i class="fa fa-check"></i><b>4.2</b> Regular expressions (Regex)</a></li>
<li class="chapter" data-level="4.3" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#comparing-ordering-and-simplifying-vectors"><i class="fa fa-check"></i><b>4.3</b> Comparing, ordering and simplifying vectors</a></li>
<li class="chapter" data-level="4.4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#useful-functions-of-stringr"><i class="fa fa-check"></i><b>4.4</b> Useful functions of <code>stringr</code></a></li>
<li class="chapter" data-level="4.5" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#data-transformations"><i class="fa fa-check"></i><b>4.5</b> Data transformations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html"><i class="fa fa-check"></i><b>5</b> – Getting Biological Data from Public Repositories</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#ncbi-databases-genbank-and-others"><i class="fa fa-check"></i><b>5.1</b> NCBI databases: GenBank and others</a></li>
<li class="chapter" data-level="5.2" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#accessing-ncbi-within-r-rentrez"><i class="fa fa-check"></i><b>5.2</b> Accessing NCBI within R: <code>rentrez</code></a></li>
<li class="chapter" data-level="5.3" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#retrieving-pubmed-references"><i class="fa fa-check"></i><b>5.3</b> Retrieving PubMed references</a></li>
<li class="chapter" data-level="5.4" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#comparing-sequences-blast"><i class="fa fa-check"></i><b>5.4</b> Comparing sequences: BLAST</a></li>
<li class="chapter" data-level="5.5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-uniprot"><i class="fa fa-check"></i><b>5.5</b> Other databases: UniProt</a></li>
<li class="chapter" data-level="5.6" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-ensembl"><i class="fa fa-check"></i><b>5.6</b> Other databases: Ensembl</a></li>
<li class="chapter" data-level="5.7" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-taxonomy-phylogenetic-trees-and-species-distributions"><i class="fa fa-check"></i><b>5.7</b> Other databases: Taxonomy, phylogenetic trees and species distributions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html"><i class="fa fa-check"></i><b>6</b> – Basic Statistics in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#exploring-your-data"><i class="fa fa-check"></i><b>6.1</b> Exploring your data</a></li>
<li class="chapter" data-level="6.2" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#sub-setting-your-dataset"><i class="fa fa-check"></i><b>6.2</b> Sub-setting your dataset</a></li>
<li class="chapter" data-level="6.3" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#transposing-rotating-your-dataset"><i class="fa fa-check"></i><b>6.3</b> Transposing (rotating) your dataset</a></li>
<li class="chapter" data-level="6.4" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#dealing-with-missing-na-values"><i class="fa fa-check"></i><b>6.4</b> Dealing with missing (NA) values</a></li>
<li class="chapter" data-level="6.5" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#about-p-values"><i class="fa fa-check"></i><b>6.5</b> About p-values</a></li>
<li class="chapter" data-level="6.6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#how-to-read-p-values"><i class="fa fa-check"></i><b>6.6</b> How to read p-values</a></li>
<li class="chapter" data-level="6.7" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#multiple-testing-corrections"><i class="fa fa-check"></i><b>6.7</b> Multiple testing corrections</a></li>
<li class="chapter" data-level="6.8" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#normalizing-your-data"><i class="fa fa-check"></i><b>6.8</b> Normalizing your data</a></li>
<li class="chapter" data-level="6.9" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#what-test-should-i-use"><i class="fa fa-check"></i><b>6.9</b> What test should I use?</a></li>
<li class="chapter" data-level="6.10" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#type-of-variables-in-dataset"><i class="fa fa-check"></i><b>6.10</b> Type of variables in dataset</a></li>
<li class="chapter" data-level="6.11" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-means-t-test"><i class="fa fa-check"></i><b>6.11</b> Comparing means: t-test</a></li>
<li class="chapter" data-level="6.12" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#two-sample-t-test"><i class="fa fa-check"></i><b>6.12</b> Two sample t-test</a></li>
<li class="chapter" data-level="6.13" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-wilcoxon-signed-rank-test"><i class="fa fa-check"></i><b>6.13</b> Beyond t-test: Wilcoxon signed rank test</a></li>
<li class="chapter" data-level="6.14" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-binomial-test"><i class="fa fa-check"></i><b>6.14</b> Beyond t-test: Binomial test</a></li>
<li class="chapter" data-level="6.15" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-chi-square-goodness-of-fit"><i class="fa fa-check"></i><b>6.15</b> Beyond t-test: Chi-square goodness of fit</a></li>
<li class="chapter" data-level="6.16" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-mcnemars-test"><i class="fa fa-check"></i><b>6.16</b> Beyond t-test: McNemar’s test</a></li>
<li class="chapter" data-level="6.17" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-fishers-exact-test"><i class="fa fa-check"></i><b>6.17</b> Beyond t-test: Fisher’s exact test</a></li>
<li class="chapter" data-level="6.18" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#correlations"><i class="fa fa-check"></i><b>6.18</b> Correlations</a></li>
<li class="chapter" data-level="6.19" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-distributions"><i class="fa fa-check"></i><b>6.19</b> Comparing distributions</a></li>
<li class="chapter" data-level="6.20" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#bootstraps"><i class="fa fa-check"></i><b>6.20</b> Bootstraps</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html"><i class="fa fa-check"></i><b>7</b> – Regressions and ANOVA in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#linear-regressions"><i class="fa fa-check"></i><b>7.1</b> Linear regressions</a></li>
<li class="chapter" data-level="7.2" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anatomy-of-a-linear-regression"><i class="fa fa-check"></i><b>7.2</b> Anatomy of a linear regression</a></li>
<li class="chapter" data-level="7.3" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#multiple-linear-regressions"><i class="fa fa-check"></i><b>7.3</b> Multiple linear regressions</a></li>
<li class="chapter" data-level="7.4" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#diagnostic-plots-for-regressions"><i class="fa fa-check"></i><b>7.4</b> Diagnostic plots for regressions</a></li>
<li class="chapter" data-level="7.5" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-influential-measurements"><i class="fa fa-check"></i><b>7.5</b> Regression influential measurements</a></li>
<li class="chapter" data-level="7.6" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-anova-coefficients-ci"><i class="fa fa-check"></i><b>7.6</b> Regression (ANOVA, coefficients, CI)</a></li>
<li class="chapter" data-level="7.7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#selecting-predictors"><i class="fa fa-check"></i><b>7.7</b> Selecting predictors</a></li>
<li class="chapter" data-level="7.8" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-predictions"><i class="fa fa-check"></i><b>7.8</b> Regression predictions</a></li>
<li class="chapter" data-level="7.9" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions"><i class="fa fa-check"></i><b>7.9</b> Logistic regressions</a></li>
<li class="chapter" data-level="7.10" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions-requirements"><i class="fa fa-check"></i><b>7.10</b> Logistic regressions requirements</a></li>
<li class="chapter" data-level="7.11" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#non-linear-regressions"><i class="fa fa-check"></i><b>7.11</b> Non-linear regressions</a></li>
<li class="chapter" data-level="7.12" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-regressions"><i class="fa fa-check"></i><b>7.12</b> Toy datasets for regressions</a></li>
<li class="chapter" data-level="7.13" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova"><i class="fa fa-check"></i><b>7.13</b> ANOVA</a></li>
<li class="chapter" data-level="7.14" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#one-way-anova"><i class="fa fa-check"></i><b>7.14</b> One-way ANOVA</a></li>
<li class="chapter" data-level="7.15" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#two-way-anova"><i class="fa fa-check"></i><b>7.15</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="7.16" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova-post-hoc-tests"><i class="fa fa-check"></i><b>7.16</b> ANOVA Post-hoc tests</a></li>
<li class="chapter" data-level="7.17" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#beyond-anova-kruskal-wallis-test"><i class="fa fa-check"></i><b>7.17</b> Beyond ANOVA: Kruskal-Wallis Test</a></li>
<li class="chapter" data-level="7.18" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-anova"><i class="fa fa-check"></i><b>7.18</b> Toy datasets for ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html"><i class="fa fa-check"></i><b>8</b> – Multidimensional Data in R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#principal-component-analysis"><i class="fa fa-check"></i><b>8.1</b> Principal Component Analysis</a></li>
<li class="chapter" data-level="8.2" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#multidimensional-scaling"><i class="fa fa-check"></i><b>8.2</b> Multidimensional scaling</a></li>
<li class="chapter" data-level="8.3" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-dfa"><i class="fa fa-check"></i><b>8.3</b> Cluster analysis: DFA</a></li>
<li class="chapter" data-level="8.4" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-k-means-unsupervised"><i class="fa fa-check"></i><b>8.4</b> Cluster analysis: <em>k</em>-means (unsupervised)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html"><i class="fa fa-check"></i><b>9</b> – Graphics and Colors</a>
<ul>
<li class="chapter" data-level="9.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#avoid-bad-graphics"><i class="fa fa-check"></i><b>9.1</b> Avoid bad graphics</a></li>
<li class="chapter" data-level="9.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#graphics-in-r-ggplot2"><i class="fa fa-check"></i><b>9.2</b> Graphics in R – ggplot2</a></li>
<li class="chapter" data-level="9.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#preparing-your-dataset"><i class="fa fa-check"></i><b>9.3</b> Preparing your dataset</a></li>
<li class="chapter" data-level="9.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#color-palettes-online"><i class="fa fa-check"></i><b>9.4</b> Color palettes online</a></li>
<li class="chapter" data-level="9.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggplot2-default-palette"><i class="fa fa-check"></i><b>9.5</b> <code>ggplot2</code> default palette</a></li>
<li class="chapter" data-level="9.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#heatmaps-or-tiled-charts"><i class="fa fa-check"></i><b>9.6</b> Heatmaps or tiled charts</a></li>
<li class="chapter" data-level="9.7" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#r-packages-about-color"><i class="fa fa-check"></i><b>9.7</b> R-packages about color</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#viridis-package"><i class="fa fa-check"></i><b>9.7.1</b> <code>viridis</code> package</a></li>
<li class="chapter" data-level="9.7.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcolorbrewer-package"><i class="fa fa-check"></i><b>9.7.2</b> <code>RColorBrewer</code> package</a></li>
<li class="chapter" data-level="9.7.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggsci-package"><i class="fa fa-check"></i><b>9.7.3</b> <code>ggsci</code> package</a></li>
<li class="chapter" data-level="9.7.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#wesanderson-package"><i class="fa fa-check"></i><b>9.7.4</b> <code>wesanderson</code> package</a></li>
<li class="chapter" data-level="9.7.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcartocolor-package"><i class="fa fa-check"></i><b>9.7.5</b> <code>rcartocolor</code> package</a></li>
<li class="chapter" data-level="9.7.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#colorspace-package"><i class="fa fa-check"></i><b>9.7.6</b> <code>colorspace</code> package</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#making-your-own-palette"><i class="fa fa-check"></i><b>9.8</b> Making your own palette</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="basic-graphics.html"><a href="basic-graphics.html"><i class="fa fa-check"></i><b>10</b> – Basic Graphics</a>
<ul>
<li class="chapter" data-level="10.1" data-path="basic-graphics.html"><a href="basic-graphics.html#what-would-you-like-to-show"><i class="fa fa-check"></i><b>10.1</b> What would you like to show?</a></li>
<li class="chapter" data-level="10.2" data-path="basic-graphics.html"><a href="basic-graphics.html#bar-charts"><i class="fa fa-check"></i><b>10.2</b> Bar charts</a></li>
<li class="chapter" data-level="10.3" data-path="basic-graphics.html"><a href="basic-graphics.html#stacked-bar-charts"><i class="fa fa-check"></i><b>10.3</b> Stacked bar charts</a></li>
<li class="chapter" data-level="10.4" data-path="basic-graphics.html"><a href="basic-graphics.html#histograms"><i class="fa fa-check"></i><b>10.4</b> Histograms</a></li>
<li class="chapter" data-level="10.5" data-path="basic-graphics.html"><a href="basic-graphics.html#multiple-histograms"><i class="fa fa-check"></i><b>10.5</b> Multiple histograms</a></li>
<li class="chapter" data-level="10.6" data-path="basic-graphics.html"><a href="basic-graphics.html#density-plots"><i class="fa fa-check"></i><b>10.6</b> Density plots</a></li>
<li class="chapter" data-level="10.7" data-path="basic-graphics.html"><a href="basic-graphics.html#box-plots"><i class="fa fa-check"></i><b>10.7</b> Box plots</a></li>
<li class="chapter" data-level="10.8" data-path="basic-graphics.html"><a href="basic-graphics.html#violin-plots"><i class="fa fa-check"></i><b>10.8</b> Violin plots</a></li>
<li class="chapter" data-level="10.9" data-path="basic-graphics.html"><a href="basic-graphics.html#scatterplots"><i class="fa fa-check"></i><b>10.9</b> Scatterplots</a></li>
<li class="chapter" data-level="10.10" data-path="basic-graphics.html"><a href="basic-graphics.html#lines-and-scatterplots"><i class="fa fa-check"></i><b>10.10</b> Lines and Scatterplots</a></li>
<li class="chapter" data-level="10.11" data-path="basic-graphics.html"><a href="basic-graphics.html#save-your-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>10.11</b> Save your plots as PDF or PNG</a></li>
<li class="chapter" data-level="10.12" data-path="basic-graphics.html"><a href="basic-graphics.html#ggplot2-add-on-packages-treemapify"><i class="fa fa-check"></i><b>10.12</b> ggplot2 add-on packages: <code>treemapify</code></a></li>
<li class="chapter" data-level="10.13" data-path="basic-graphics.html"><a href="basic-graphics.html#other-resources"><i class="fa fa-check"></i><b>10.13</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="sequence-alignments.html"><a href="sequence-alignments.html"><i class="fa fa-check"></i><b>11</b> – Sequence Alignments</a>
<ul>
<li class="chapter" data-level="11.1" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-using-rentrez"><i class="fa fa-check"></i><b>11.1</b> Retrieving sequences using ‘rentrez’</a></li>
<li class="chapter" data-level="11.2" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-manually-from-ncbi"><i class="fa fa-check"></i><b>11.2</b> Retrieving sequences manually from NCBI</a></li>
<li class="chapter" data-level="11.3" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-accession-numbers"><i class="fa fa-check"></i><b>11.3</b> Retrieving accession numbers</a></li>
<li class="chapter" data-level="11.4" data-path="sequence-alignments.html"><a href="sequence-alignments.html#alignment-with-decipher"><i class="fa fa-check"></i><b>11.4</b> Alignment with <code>DECIPHER</code></a></li>
<li class="chapter" data-level="11.5" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualize-alignment"><i class="fa fa-check"></i><b>11.5</b> Visualize alignment</a></li>
<li class="chapter" data-level="11.6" data-path="sequence-alignments.html"><a href="sequence-alignments.html#save-alignment-to-file"><i class="fa fa-check"></i><b>11.6</b> Save alignment to file</a></li>
<li class="chapter" data-level="11.7" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-and-align-several-markers"><i class="fa fa-check"></i><b>11.7</b> Concatenate and align several markers</a></li>
<li class="chapter" data-level="11.8" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-data.table"><i class="fa fa-check"></i><b>11.8</b> Concatenate with <code>data.table</code></a></li>
<li class="chapter" data-level="11.9" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-biostrings"><i class="fa fa-check"></i><b>11.9</b> Concatenate with <code>Biostrings</code></a></li>
<li class="chapter" data-level="11.10" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualizing-alignments-in-r"><i class="fa fa-check"></i><b>11.10</b> Visualizing alignments in R</a></li>
<li class="chapter" data-level="11.11" data-path="sequence-alignments.html"><a href="sequence-alignments.html#other-sequence-alignment-software"><i class="fa fa-check"></i><b>11.11</b> Other sequence alignment software</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html"><i class="fa fa-check"></i><b>12</b> – Estimating Phylogenetic Trees</a>
<ul>
<li class="chapter" data-level="12.1" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#phylogeny-inference-with-iqtree2"><i class="fa fa-check"></i><b>12.1</b> Phylogeny inference with <code>iqtree2</code></a></li>
<li class="chapter" data-level="12.2" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-iqtree2-in-macos"><i class="fa fa-check"></i><b>12.2</b> Installing <code>iqtree2</code> in macOS</a></li>
<li class="chapter" data-level="12.3" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-and-running-iqtree2-in-a-pc"><i class="fa fa-check"></i><b>12.3</b> Installing and running <code>iqtree2</code> in a PC</a></li>
<li class="chapter" data-level="12.4" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#loading-the-iqtree_runner_r-function"><i class="fa fa-check"></i><b>12.4</b> Loading the <code>iqtree_runner_R</code> function</a></li>
<li class="chapter" data-level="12.5" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#running-iqtree2-from-r"><i class="fa fa-check"></i><b>12.5</b> Running <code>iqtree2</code> from R</a></li>
<li class="chapter" data-level="12.6" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-no-partitions"><i class="fa fa-check"></i><b>12.6</b> One marker and no partitions</a></li>
<li class="chapter" data-level="12.7" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-with-partitions"><i class="fa fa-check"></i><b>12.7</b> One marker and with partitions</a></li>
<li class="chapter" data-level="12.8" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#bootstrap-support"><i class="fa fa-check"></i><b>12.8</b> Bootstrap support</a></li>
<li class="chapter" data-level="12.9" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#many-markers-and-with-partitions"><i class="fa fa-check"></i><b>12.9</b> Many markers and with partitions</a></li>
<li class="chapter" data-level="12.10" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#other-sequence-phylogenetic-tree-estimation-software"><i class="fa fa-check"></i><b>12.10</b> Other sequence phylogenetic tree estimation software</a></li>
<li class="chapter" data-level="12.11" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#drawing-phylogenetic-trees-with-ggtree"><i class="fa fa-check"></i><b>12.11</b> Drawing phylogenetic trees with <code>ggtree</code></a></li>
<li class="chapter" data-level="12.12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#save-ggtree-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>12.12</b> Save ggtree plots as PDF or PNG</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html"><i class="fa fa-check"></i><b>13</b> – Mining from a reference dataset</a>
<ul>
<li class="chapter" data-level="13.1" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#a-worked-example-using-the-mtcars-dataset"><i class="fa fa-check"></i><b>13.1</b> A worked example using the <code>mtcars</code> dataset</a></li>
<li class="chapter" data-level="13.2" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-reference-dataset"><i class="fa fa-check"></i><b>13.2</b> Build a reference dataset</a></li>
<li class="chapter" data-level="13.3" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-sampling-dataset"><i class="fa fa-check"></i><b>13.3</b> Build a sampling dataset</a></li>
<li class="chapter" data-level="13.4" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#extracting-from-reference-dataset"><i class="fa fa-check"></i><b>13.4</b> Extracting from reference dataset</a></li>
<li class="chapter" data-level="13.5" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#plots-of-the-extracted-data"><i class="fa fa-check"></i><b>13.5</b> Plots of the extracted data</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="introduction-to-python.html"><a href="introduction-to-python.html"><i class="fa fa-check"></i><b>14</b> – Introduction to Python</a>
<ul>
<li class="chapter" data-level="14.1" data-path="introduction-to-python.html"><a href="introduction-to-python.html#welcome-to-python-why-python"><i class="fa fa-check"></i><b>14.1</b> Welcome to Python! Why Python?</a></li>
<li class="chapter" data-level="14.2" data-path="introduction-to-python.html"><a href="introduction-to-python.html#which-python-should-i-install"><i class="fa fa-check"></i><b>14.2</b> Which Python should I install?</a></li>
<li class="chapter" data-level="14.3" data-path="introduction-to-python.html"><a href="introduction-to-python.html#mac-os-x-python-installations"><i class="fa fa-check"></i><b>14.3</b> Mac OS X Python Installations</a></li>
<li class="chapter" data-level="14.4" data-path="introduction-to-python.html"><a href="introduction-to-python.html#windows-python-installations"><i class="fa fa-check"></i><b>14.4</b> Windows Python Installations</a></li>
<li class="chapter" data-level="14.5" data-path="introduction-to-python.html"><a href="introduction-to-python.html#install-an-ide-or-integrated-development-environment"><i class="fa fa-check"></i><b>14.5</b> Install an IDE, or integrated development environment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html"><i class="fa fa-check"></i><b>15</b> – Python: Keywords and Identifiers</a>
<ul>
<li class="chapter" data-level="15.1" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#keywords-in-python"><i class="fa fa-check"></i><b>15.1</b> Keywords in Python</a></li>
<li class="chapter" data-level="15.2" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#identifiers-in-python"><i class="fa fa-check"></i><b>15.2</b> Identifiers in Python</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html"><i class="fa fa-check"></i><b>16</b> – Python: Statements and Expressions</a>
<ul>
<li class="chapter" data-level="16.1" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#what-are-statements"><i class="fa fa-check"></i><b>16.1</b> What are statements?</a></li>
<li class="chapter" data-level="16.2" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#expressions-in-python"><i class="fa fa-check"></i><b>16.2</b> Expressions in Python</a></li>
<li class="chapter" data-level="16.3" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#simple-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.3</b> Simple Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.4" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#augmented-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.4</b> Augmented Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.5" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#explicit-and-implicit-line-continuation"><i class="fa fa-check"></i><b>16.5</b> Explicit and Implicit Line Continuation</a></li>
<li class="chapter" data-level="16.6" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#indentation-in-python-is-crucial-to-proper-code-function"><i class="fa fa-check"></i><b>16.6</b> Indentation in Python is Crucial to Proper Code function</a></li>
<li class="chapter" data-level="16.7" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#comments"><i class="fa fa-check"></i><b>16.7</b> Comments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="python-data-types.html"><a href="python-data-types.html"><i class="fa fa-check"></i><b>17</b> – Python: Data Types</a>
<ul>
<li class="chapter" data-level="17.1" data-path="python-data-types.html"><a href="python-data-types.html#booleans"><i class="fa fa-check"></i><b>17.1</b> Booleans</a></li>
<li class="chapter" data-level="17.2" data-path="python-data-types.html"><a href="python-data-types.html#numbers-int-float-and-complex"><i class="fa fa-check"></i><b>17.2</b> Numbers: int, float and complex</a></li>
<li class="chapter" data-level="17.3" data-path="python-data-types.html"><a href="python-data-types.html#strings"><i class="fa fa-check"></i><b>17.3</b> Strings</a></li>
<li class="chapter" data-level="17.4" data-path="python-data-types.html"><a href="python-data-types.html#bytes"><i class="fa fa-check"></i><b>17.4</b> Bytes</a></li>
<li class="chapter" data-level="17.5" data-path="python-data-types.html"><a href="python-data-types.html#lists-1"><i class="fa fa-check"></i><b>17.5</b> Lists</a></li>
<li class="chapter" data-level="17.6" data-path="python-data-types.html"><a href="python-data-types.html#tuples"><i class="fa fa-check"></i><b>17.6</b> Tuples</a></li>
<li class="chapter" data-level="17.7" data-path="python-data-types.html"><a href="python-data-types.html#sets"><i class="fa fa-check"></i><b>17.7</b> Sets</a></li>
<li class="chapter" data-level="17.8" data-path="python-data-types.html"><a href="python-data-types.html#dictionary"><i class="fa fa-check"></i><b>17.8</b> Dictionary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="python-strings.html"><a href="python-strings.html"><i class="fa fa-check"></i><b>18</b> – Python: Strings</a>
<ul>
<li class="chapter" data-level="18.1" data-path="python-strings.html"><a href="python-strings.html#string-operators"><i class="fa fa-check"></i><b>18.1</b> String Operators</a></li>
<li class="chapter" data-level="18.2" data-path="python-strings.html"><a href="python-strings.html#other-string-modifiers"><i class="fa fa-check"></i><b>18.2</b> Other String Modifiers</a></li>
<li class="chapter" data-level="18.3" data-path="python-strings.html"><a href="python-strings.html#format-characters-in-strings"><i class="fa fa-check"></i><b>18.3</b> Format-Characters in Strings</a></li>
<li class="chapter" data-level="18.4" data-path="python-strings.html"><a href="python-strings.html#comparisons-with-strings"><i class="fa fa-check"></i><b>18.4</b> Comparisons with Strings</a></li>
<li class="chapter" data-level="18.5" data-path="python-strings.html"><a href="python-strings.html#padding-strings"><i class="fa fa-check"></i><b>18.5</b> Padding strings</a></li>
<li class="chapter" data-level="18.6" data-path="python-strings.html"><a href="python-strings.html#search-for-words-in-string-and-output-location"><i class="fa fa-check"></i><b>18.6</b> Search for words in string and output location</a></li>
<li class="chapter" data-level="18.7" data-path="python-strings.html"><a href="python-strings.html#seek-and-replace-in-strings"><i class="fa fa-check"></i><b>18.7</b> Seek and replace in strings</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="python-operators.html"><a href="python-operators.html"><i class="fa fa-check"></i><b>19</b> – Python: Operators</a>
<ul>
<li class="chapter" data-level="19.1" data-path="python-operators.html"><a href="python-operators.html#many-types-of-operators"><i class="fa fa-check"></i><b>19.1</b> Many types of operators</a></li>
<li class="chapter" data-level="19.2" data-path="python-operators.html"><a href="python-operators.html#arithmetic-operators"><i class="fa fa-check"></i><b>19.2</b> Arithmetic operators</a></li>
<li class="chapter" data-level="19.3" data-path="python-operators.html"><a href="python-operators.html#comparison-operators"><i class="fa fa-check"></i><b>19.3</b> Comparison operators</a></li>
<li class="chapter" data-level="19.4" data-path="python-operators.html"><a href="python-operators.html#logical-operators"><i class="fa fa-check"></i><b>19.4</b> Logical operators</a></li>
<li class="chapter" data-level="19.5" data-path="python-operators.html"><a href="python-operators.html#bitwise-operators"><i class="fa fa-check"></i><b>19.5</b> Bitwise operators</a></li>
<li class="chapter" data-level="19.6" data-path="python-operators.html"><a href="python-operators.html#identity-operators"><i class="fa fa-check"></i><b>19.6</b> Identity Operators</a></li>
<li class="chapter" data-level="19.7" data-path="python-operators.html"><a href="python-operators.html#membership-operators"><i class="fa fa-check"></i><b>19.7</b> Membership operators</a></li>
<li class="chapter" data-level="19.8" data-path="python-operators.html"><a href="python-operators.html#what-is-the-order-of-operations-with-operators"><i class="fa fa-check"></i><b>19.8</b> What is the order of operations with operators?</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="biopython.html"><a href="biopython.html"><i class="fa fa-check"></i><b>20</b> – Biopython</a>
<ul>
<li class="chapter" data-level="20.1" data-path="biopython.html"><a href="biopython.html#introduction-to-biopython"><i class="fa fa-check"></i><b>20.1</b> Introduction to Biopython</a></li>
<li class="chapter" data-level="20.2" data-path="biopython.html"><a href="biopython.html#biopython-installation"><i class="fa fa-check"></i><b>20.2</b> Biopython Installation</a></li>
<li class="chapter" data-level="20.3" data-path="biopython.html"><a href="biopython.html#getting-started"><i class="fa fa-check"></i><b>20.3</b> Getting Started</a></li>
<li class="chapter" data-level="20.4" data-path="biopython.html"><a href="biopython.html#parsing-my-fasta"><i class="fa fa-check"></i><b>20.4</b> Parsing my FASTA</a></li>
<li class="chapter" data-level="20.5" data-path="biopython.html"><a href="biopython.html#parsing-my-genbank-or-genpept"><i class="fa fa-check"></i><b>20.5</b> Parsing my GenBank or GenPept</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="biopython-sequences.html"><a href="biopython-sequences.html"><i class="fa fa-check"></i><b>21</b> – Biopython: Sequences</a>
<ul>
<li class="chapter" data-level="21.1" data-path="biopython-sequences.html"><a href="biopython-sequences.html#sequence-objects"><i class="fa fa-check"></i><b>21.1</b> Sequence Objects</a></li>
<li class="chapter" data-level="21.2" data-path="biopython-sequences.html"><a href="biopython-sequences.html#concatenation-with-sequences"><i class="fa fa-check"></i><b>21.2</b> Concatenation with Sequences</a></li>
<li class="chapter" data-level="21.3" data-path="biopython-sequences.html"><a href="biopython-sequences.html#transcription-dna-to-mrna"><i class="fa fa-check"></i><b>21.3</b> Transcription: DNA to mRNA</a></li>
<li class="chapter" data-level="21.4" data-path="biopython-sequences.html"><a href="biopython-sequences.html#translation-rna-to-prot"><i class="fa fa-check"></i><b>21.4</b> Translation: RNA to PROT</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html"><i class="fa fa-check"></i><b>22</b> – Biopython: Mulitple Sequence Alignments</a>
<ul>
<li class="chapter" data-level="22.1" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwise2"><i class="fa fa-check"></i><b>22.1</b> pairwise2</a></li>
<li class="chapter" data-level="22.2" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwisealigner"><i class="fa fa-check"></i><b>22.2</b> PairwiseAligner</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html"><i class="fa fa-check"></i><b>23</b> – Biopython: GenomeDiagram</a>
<ul>
<li class="chapter" data-level="23.1" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#linearcircular-gene-diagrams"><i class="fa fa-check"></i><b>23.1</b> Linear/Circular Gene Diagrams</a></li>
<li class="chapter" data-level="23.2" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#diagram-readability"><i class="fa fa-check"></i><b>23.2</b> Diagram Readability</a></li>
<li class="chapter" data-level="23.3" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#box-shape"><i class="fa fa-check"></i><b>23.3</b> Box Shape</a></li>
<li class="chapter" data-level="23.4" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#edit-box-shape-arrows"><i class="fa fa-check"></i><b>23.4</b> Edit Box Shape: Arrows</a></li>
<li class="chapter" data-level="23.5" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#restriction-map"><i class="fa fa-check"></i><b>23.5</b> Restriction Map</a></li>
<li class="chapter" data-level="23.6" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#tracks-on-tracks"><i class="fa fa-check"></i><b>23.6</b> Tracks on Tracks</a></li>
<li class="chapter" data-level="23.7" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#cross-link-data"><i class="fa fa-check"></i><b>23.7</b> Cross-link Data</a></li>
<li class="chapter" data-level="23.8" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#clean-cross-link"><i class="fa fa-check"></i><b>23.8</b> Clean Cross-Link</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html"><i class="fa fa-check"></i><b>24</b> – Biopython: Chromosome Modeling</a>
<ul>
<li class="chapter" data-level="24.1" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html#gene-feature-modeling"><i class="fa fa-check"></i><b>24.1</b> Gene Feature Modeling</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html"><i class="fa fa-check"></i><b>25</b> – Biopython: Sequence Parsing with Plots</a>
<ul>
<li class="chapter" data-level="25.1" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#histograms-1"><i class="fa fa-check"></i><b>25.1</b> Histograms</a></li>
<li class="chapter" data-level="25.2" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#gc-content"><i class="fa fa-check"></i><b>25.2</b> GC Content</a></li>
<li class="chapter" data-level="25.3" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#nt-alignment-dot-plot"><i class="fa fa-check"></i><b>25.3</b> Nt Alignment Dot Plot</a></li>
<li class="chapter" data-level="25.4" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#bigger-dots-for-the-dot-plot"><i class="fa fa-check"></i><b>25.4</b> Bigger Dots for the Dot Plot</a></li>
<li class="chapter" data-level="25.5" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#fastq-we-did-not-forget-you"><i class="fa fa-check"></i><b>25.5</b> FASTQ we did not forget you</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html"><i class="fa fa-check"></i><b>26</b> – Python: Web Scraping w/ BeautifulSoup</a>
<ul>
<li class="chapter" data-level="26.1" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#introduction"><i class="fa fa-check"></i><b>26.1</b> Introduction</a></li>
<li class="chapter" data-level="26.2" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#installation"><i class="fa fa-check"></i><b>26.2</b> Installation</a></li>
<li class="chapter" data-level="26.3" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#getting-started-1"><i class="fa fa-check"></i><b>26.3</b> Getting started</a></li>
<li class="chapter" data-level="26.4" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-all-matching-tags"><i class="fa fa-check"></i><b>26.4</b> Mining all matching tags</a></li>
<li class="chapter" data-level="26.5" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-tags-by-class-or-id"><i class="fa fa-check"></i><b>26.5</b> Mining tags by class or id</a></li>
<li class="chapter" data-level="26.6" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#use-css-selectors-for-positional-mining"><i class="fa fa-check"></i><b>26.6</b> Use CSS Selectors for positional mining</a></li>
<li class="chapter" data-level="26.7" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#weather-page-example"><i class="fa fa-check"></i><b>26.7</b> Weather Page Example</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="python-subprocess.html"><a href="python-subprocess.html"><i class="fa fa-check"></i><b>27</b> – Python Subprocess</a>
<ul>
<li class="chapter" data-level="27.1" data-path="python-subprocess.html"><a href="python-subprocess.html#introduction-and-installation"><i class="fa fa-check"></i><b>27.1</b> Introduction and Installation</a></li>
<li class="chapter" data-level="27.2" data-path="python-subprocess.html"><a href="python-subprocess.html#creating-your-first-python-script-w-bbtools"><i class="fa fa-check"></i><b>27.2</b> Creating your first python script w/ BBTools</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html"><i class="fa fa-check"></i><b>28</b> – Control Structures &amp; Loops in R</a>
<ul>
<li class="chapter" data-level="28.1" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#controling-the-flow"><i class="fa fa-check"></i><b>28.1</b> Controling the flow</a></li>
<li class="chapter" data-level="28.2" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if"><i class="fa fa-check"></i><b>28.2</b> Conditional <code>if</code></a></li>
<li class="chapter" data-level="28.3" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if-and-else"><i class="fa fa-check"></i><b>28.3</b> Conditional <code>if</code> and <code>else</code></a></li>
<li class="chapter" data-level="28.4" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#nested-conditionals-if-else-and-if-else"><i class="fa fa-check"></i><b>28.4</b> Nested conditionals <code>if</code>, <code>else</code>, and <code>if else</code></a></li>
<li class="chapter" data-level="28.5" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-ifelse"><i class="fa fa-check"></i><b>28.5</b> Conditional <code>ifelse</code></a></li>
<li class="chapter" data-level="28.6" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#loops-that-control-iterations-and-repeations"><i class="fa fa-check"></i><b>28.6</b> Loops that control iterations and repeations</a></li>
<li class="chapter" data-level="28.7" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-for-loop"><i class="fa fa-check"></i><b>28.7</b> The <code>for</code> Loop</a></li>
<li class="chapter" data-level="28.8" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-while-loop"><i class="fa fa-check"></i><b>28.8</b> The <code>while</code> Loop</a></li>
<li class="chapter" data-level="28.9" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-repeat-loop"><i class="fa fa-check"></i><b>28.9</b> The <code>repeat</code> Loop</a></li>
<li class="chapter" data-level="28.10" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-next-and-break-statements"><i class="fa fa-check"></i><b>28.10</b> The <code>next</code> and <code>break</code> statements</a></li>
<li class="chapter" data-level="28.11" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functions"><i class="fa fa-check"></i><b>28.11</b> Other loop functions</a></li>
<li class="chapter" data-level="28.12" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#lapply-functions"><i class="fa fa-check"></i><b>28.12</b> <code>lapply</code> functions</a></li>
<li class="chapter" data-level="28.13" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsapply"><i class="fa fa-check"></i><b>28.13</b> Other loop functions:<code>apply</code></a></li>
<li class="chapter" data-level="28.14" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsmapply"><i class="fa fa-check"></i><b>28.14</b> Other loop functions:<code>mapply</code></a></li>
<li class="chapter" data-level="28.15" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionstapply"><i class="fa fa-check"></i><b>28.15</b> Other loop functions:<code>tapply</code></a></li>
<li class="chapter" data-level="28.16" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionssplit"><i class="fa fa-check"></i><b>28.16</b> Other loop functions:<code>split</code></a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html"><i class="fa fa-check"></i><b>29</b> – Bioinformatics Literature</a>
<ul>
<li class="chapter" data-level="29.1" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#peer-reviewed-bioinformatics-journals"><i class="fa fa-check"></i><b>29.1</b> Peer-reviewed bioinformatics journals</a></li>
<li class="chapter" data-level="29.2" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#access-to-journals-articles"><i class="fa fa-check"></i><b>29.2</b> Access to journals articles</a></li>
<li class="chapter" data-level="29.3" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#informal-access-to-code-preprints-and-vignettes"><i class="fa fa-check"></i><b>29.3</b> Informal access to code: Preprints and vignettes</a></li>
<li class="chapter" data-level="29.4" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#online-bioinformatics-courses"><i class="fa fa-check"></i><b>29.4</b> Online bioinformatics courses</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html"><i class="fa fa-check"></i><b>30</b> – Importing/Exporting Data into R</a>
<ul>
<li class="chapter" data-level="30.1" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#typing-directly"><i class="fa fa-check"></i><b>30.1</b> Typing directly</a></li>
<li class="chapter" data-level="30.2" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#summary-versus-raw-data"><i class="fa fa-check"></i><b>30.2</b> Summary versus raw data</a></li>
<li class="chapter" data-level="30.3" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#importing-data-from-a-file"><i class="fa fa-check"></i><b>30.3</b> Importing data from a file</a></li>
<li class="chapter" data-level="30.4" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-to-a-file"><i class="fa fa-check"></i><b>30.4</b> Saving data to a file</a></li>
<li class="chapter" data-level="30.5" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-as-rdata-.rdata-.rda"><i class="fa fa-check"></i><b>30.5</b> Saving data as Rdata (.rdata, .rda)</a></li>
<li class="chapter" data-level="30.6" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#import-excel-formated-data"><i class="fa fa-check"></i><b>30.6</b> Import Excel formated data</a></li>
<li class="chapter" data-level="30.7" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#save-data-as-excel-formated-file"><i class="fa fa-check"></i><b>30.7</b> Save data as Excel formated file</a></li>
<li class="chapter" data-level="30.8" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#downloading-data-files-github"><i class="fa fa-check"></i><b>30.8</b> Downloading data files (GitHub)</a></li>
<li class="chapter" data-level="30.9" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#accessing-online-data"><i class="fa fa-check"></i><b>30.9</b> Accessing online data</a></li>
<li class="chapter" data-level="30.10" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#fixing-problems-while-importing-datasets"><i class="fa fa-check"></i><b>30.10</b> Fixing problems while importing datasets</a></li>
<li class="chapter" data-level="30.11" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#data-from-clinicaltrials.gov"><i class="fa fa-check"></i><b>30.11</b> Data from clinicaltrials.gov</a></li>
<li class="chapter" data-level="30.12" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#data-scrapping-from-a-website"><i class="fa fa-check"></i><b>30.12</b> Data scrapping from a website</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="miscellaneous.html"><a href="miscellaneous.html"><i class="fa fa-check"></i><b>31</b> – Miscellaneous</a>
<ul>
<li class="chapter" data-level="31.1" data-path="miscellaneous.html"><a href="miscellaneous.html#sampling-from-a-vector-of-names"><i class="fa fa-check"></i><b>31.1</b> Sampling from a vector of names</a></li>
<li class="chapter" data-level="31.2" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-an-article-pdf"><i class="fa fa-check"></i><b>31.2</b> Read a text from an article pdf</a></li>
<li class="chapter" data-level="31.3" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-website"><i class="fa fa-check"></i><b>31.3</b> Read a text from a website</a></li>
<li class="chapter" data-level="31.4" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-file"><i class="fa fa-check"></i><b>31.4</b> Read a text from a file</a></li>
<li class="chapter" data-level="31.5" data-path="miscellaneous.html"><a href="miscellaneous.html#generate-the-word-cloud"><i class="fa fa-check"></i><b>31.5</b> Generate the word cloud</a></li>
<li class="chapter" data-level="31.6" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-pdf"><i class="fa fa-check"></i><b>31.6</b> worldcloud – PDF</a></li>
<li class="chapter" data-level="31.7" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-website"><i class="fa fa-check"></i><b>31.7</b> worldcloud – website</a></li>
<li class="chapter" data-level="31.8" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-text"><i class="fa fa-check"></i><b>31.8</b> worldcloud – text</a></li>
<li class="chapter" data-level="31.9" data-path="miscellaneous.html"><a href="miscellaneous.html#miscellaneous-1"><i class="fa fa-check"></i><b>31.9</b> Miscellaneous</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics Workshop Gitbook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="basic-statistics-in-r" class="section level1" number="6">
<h1><span class="header-section-number">Session 6</span> – Basic Statistics in R</h1>
<p>This workshop is <strong>not designed</strong> to give you an extensive overview on all statistical methods that can be performed in R. Yet, this computing environment was born to provide a comprehensive platform for the implementation of statistical methods. We will overview of some basic and advanced methods with some illustrative graphs (these will be expanded in the session devoted to graphics). However, you can ask me if you want to explore a specific method that you would me to add to this gitbook.</p>
<div id="exploring-your-data" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Exploring your data</h2>
<p><strong>1)</strong> To start, we need to import a dataset that we want to explore and analyze.
As graduate students, you are encouraged to use your own datasets (i.e., this is ideal as you will know more about its contents, metrics and intended use). For this exercise, I will use two previously show datasets: (1) the small preloaded <code>mtcars</code> dataset and (2) the large <code>airway_scaledcounts.csv</code> dataset. These datasets are in our class <a href="https://github.com/corytophanes/BIO_BIT_Bioinformatics_209/tree/main/ref_files_pdfs" target="_blank"><strong>GitHub repository</strong></a> and you can download those to your computer.</p>
<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb245-1"><a href="basic-statistics-in-r.html#cb245-1" aria-hidden="true" tabindex="-1"></a><span class="do">## </span><span class="al">NOTE</span><span class="do">: remember to update the path to files with the datasets where you downloaded in your computer -- THIS IS EXCLUSIVE TO YOUR COMPUTER AND IT IS NOT THE PATH SHOWN BELOW</span></span>
<span id="cb245-2"><a href="basic-statistics-in-r.html#cb245-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb245-3"><a href="basic-statistics-in-r.html#cb245-3" aria-hidden="true" tabindex="-1"></a><span class="co">#define a working directory</span></span>
<span id="cb245-4"><a href="basic-statistics-in-r.html#cb245-4" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/my_working_directory&quot;</span>)</span>
<span id="cb245-5"><a href="basic-statistics-in-r.html#cb245-5" aria-hidden="true" tabindex="-1"></a><span class="co">#load get mtcars dataset -- this a preloaded dataset in R</span></span>
<span id="cb245-6"><a href="basic-statistics-in-r.html#cb245-6" aria-hidden="true" tabindex="-1"></a>cars_data <span class="ot">&lt;-</span> mtcars</span>
<span id="cb245-7"><a href="basic-statistics-in-r.html#cb245-7" aria-hidden="true" tabindex="-1"></a>cars_data</span>
<span id="cb245-8"><a href="basic-statistics-in-r.html#cb245-8" aria-hidden="true" tabindex="-1"></a><span class="co">#                     mpg cyl  disp  hp drat    wt  qsec vs am gear carb</span></span>
<span id="cb245-9"><a href="basic-statistics-in-r.html#cb245-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Mazda RX4           21.0   6 160.0 110 3.90 2.620 16.46  0  1    4    4</span></span>
<span id="cb245-10"><a href="basic-statistics-in-r.html#cb245-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Mazda RX4 Wag       21.0   6 160.0 110 3.90 2.875 17.02  0  1    4    4</span></span>
<span id="cb245-11"><a href="basic-statistics-in-r.html#cb245-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Datsun 710          22.8   4 108.0  93 3.85 2.320 18.61  1  1    4    1</span></span>
<span id="cb245-12"><a href="basic-statistics-in-r.html#cb245-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Hornet 4 Drive      21.4   6 258.0 110 3.08 3.215 19.44  1  0    3    1</span></span>
<span id="cb245-13"><a href="basic-statistics-in-r.html#cb245-13" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span>
<span id="cb245-14"><a href="basic-statistics-in-r.html#cb245-14" aria-hidden="true" tabindex="-1"></a><span class="co">#load get &#39;airway_scaledcounts.csv&#39; dataset </span></span>
<span id="cb245-15"><a href="basic-statistics-in-r.html#cb245-15" aria-hidden="true" tabindex="-1"></a>airway_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/my_working_directory/airway_scaledcounts.csv&quot;</span>,</span>
<span id="cb245-16"><a href="basic-statistics-in-r.html#cb245-16" aria-hidden="true" tabindex="-1"></a>               <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb245-17"><a href="basic-statistics-in-r.html#cb245-17" aria-hidden="true" tabindex="-1"></a><span class="co">#          ensgene SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb245-18"><a href="basic-statistics-in-r.html#cb245-18" aria-hidden="true" tabindex="-1"></a><span class="co">#1 ENSG00000000003        723        486        904        445       1170       1097        806        604</span></span>
<span id="cb245-19"><a href="basic-statistics-in-r.html#cb245-19" aria-hidden="true" tabindex="-1"></a><span class="co">#2 ENSG00000000005          0          0          0          0          0          0          0          0</span></span>
<span id="cb245-20"><a href="basic-statistics-in-r.html#cb245-20" aria-hidden="true" tabindex="-1"></a><span class="co">#3 ENSG00000000419        467        523        616        371        582        781        417        509</span></span>
<span id="cb245-21"><a href="basic-statistics-in-r.html#cb245-21" aria-hidden="true" tabindex="-1"></a><span class="co">#4 ENSG00000000457        347        258        364        237        318        447        330        324</span></span>
<span id="cb245-22"><a href="basic-statistics-in-r.html#cb245-22" aria-hidden="true" tabindex="-1"></a><span class="co">#5 ENSG00000000460         96         81         73         66        118         94        102         74</span></span>
<span id="cb245-23"><a href="basic-statistics-in-r.html#cb245-23" aria-hidden="true" tabindex="-1"></a><span class="co">#6 ENSG00000000938          0          0          1          0          2          0          0          0</span></span>
<span id="cb245-24"><a href="basic-statistics-in-r.html#cb245-24" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span></code></pre></div>
<p><strong>2)</strong> We can get some basic descriptive parameters of each data set including number of entries, number and names of the columns (i.e., variables), type of variables (e.g., numbers or characters), number of <code>NA</code> values.</p>
<p>For this purpose, the easiest approach is to use the function <code>summary()</code></p>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="basic-statistics-in-r.html#cb246-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(cars_data)</span>
<span id="cb246-2"><a href="basic-statistics-in-r.html#cb246-2" aria-hidden="true" tabindex="-1"></a><span class="co">#     mpg             cyl             disp             hp             drat             wt             qsec             vs        </span></span>
<span id="cb246-3"><a href="basic-statistics-in-r.html#cb246-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Min.   :10.40   Min.   :4.000   Min.   : 71.1   Min.   : 52.0   Min.   :2.760   Min.   :1.513   Min.   :14.50   Min.   :0.0000  </span></span>
<span id="cb246-4"><a href="basic-statistics-in-r.html#cb246-4" aria-hidden="true" tabindex="-1"></a><span class="co"># 1st Qu.:15.43   1st Qu.:4.000   1st Qu.:120.8   1st Qu.: 96.5   1st Qu.:3.080   1st Qu.:2.581   1st Qu.:16.89   1st Qu.:0.0000  </span></span>
<span id="cb246-5"><a href="basic-statistics-in-r.html#cb246-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Median :19.20   Median :6.000   Median :196.3   Median :123.0   Median :3.695   Median :3.325   Median :17.71   Median :0.0000  </span></span>
<span id="cb246-6"><a href="basic-statistics-in-r.html#cb246-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Mean   :20.09   Mean   :6.188   Mean   :230.7   Mean   :146.7   Mean   :3.597   Mean   :3.217   Mean   :17.85   Mean   :0.4375  </span></span>
<span id="cb246-7"><a href="basic-statistics-in-r.html#cb246-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 3rd Qu.:22.80   3rd Qu.:8.000   3rd Qu.:326.0   3rd Qu.:180.0   3rd Qu.:3.920   3rd Qu.:3.610   3rd Qu.:18.90   3rd Qu.:1.0000  </span></span>
<span id="cb246-8"><a href="basic-statistics-in-r.html#cb246-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Max.   :33.90   Max.   :8.000   Max.   :472.0   Max.   :335.0   Max.   :4.930   Max.   :5.424   Max.   :22.90   Max.   :1.0000  </span></span>
<span id="cb246-9"><a href="basic-statistics-in-r.html#cb246-9" aria-hidden="true" tabindex="-1"></a><span class="co">#       am              gear            carb      </span></span>
<span id="cb246-10"><a href="basic-statistics-in-r.html#cb246-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Min.   :0.0000   Min.   :3.000   Min.   :1.000  </span></span>
<span id="cb246-11"><a href="basic-statistics-in-r.html#cb246-11" aria-hidden="true" tabindex="-1"></a><span class="co"># 1st Qu.:0.0000   1st Qu.:3.000   1st Qu.:2.000  </span></span>
<span id="cb246-12"><a href="basic-statistics-in-r.html#cb246-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Median :0.0000   Median :4.000   Median :2.000  </span></span>
<span id="cb246-13"><a href="basic-statistics-in-r.html#cb246-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Mean   :0.4062   Mean   :3.688   Mean   :2.812  </span></span>
<span id="cb246-14"><a href="basic-statistics-in-r.html#cb246-14" aria-hidden="true" tabindex="-1"></a><span class="co"># 3rd Qu.:1.0000   3rd Qu.:4.000   3rd Qu.:4.000  </span></span>
<span id="cb246-15"><a href="basic-statistics-in-r.html#cb246-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Max.   :1.0000   Max.   :5.000   Max.   :8.000  </span></span>
<span id="cb246-16"><a href="basic-statistics-in-r.html#cb246-16" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(airway_data)</span>
<span id="cb246-17"><a href="basic-statistics-in-r.html#cb246-17" aria-hidden="true" tabindex="-1"></a><span class="co">#   ensgene            SRR1039508         SRR1039509         SRR1039512         SRR1039513         SRR1039516    </span></span>
<span id="cb246-18"><a href="basic-statistics-in-r.html#cb246-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Length:38694       Min.   :     0.0   Min.   :     0.0   Min.   :     0.0   Min.   :     0.0   Min.   :     0  </span></span>
<span id="cb246-19"><a href="basic-statistics-in-r.html#cb246-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Class :character   1st Qu.:     0.0   1st Qu.:     0.0   1st Qu.:     0.0   1st Qu.:     0.0   1st Qu.:     0  </span></span>
<span id="cb246-20"><a href="basic-statistics-in-r.html#cb246-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Mode  :character   Median :     1.0   Median :     1.0   Median :     1.0   Median :     0.0   Median :     1  </span></span>
<span id="cb246-21"><a href="basic-statistics-in-r.html#cb246-21" aria-hidden="true" tabindex="-1"></a><span class="co">#                    Mean   :   546.5   Mean   :   501.1   Mean   :   673.6   Mean   :   405.3   Mean   :   649  </span></span>
<span id="cb246-22"><a href="basic-statistics-in-r.html#cb246-22" aria-hidden="true" tabindex="-1"></a><span class="co">#                    3rd Qu.:   203.0   3rd Qu.:   172.0   3rd Qu.:   237.0   3rd Qu.:   128.0   3rd Qu.:   228  </span></span>
<span id="cb246-23"><a href="basic-statistics-in-r.html#cb246-23" aria-hidden="true" tabindex="-1"></a><span class="co">#                    Max.   :287372.0   Max.   :244921.0   Max.   :510107.0   Max.   :304818.0   Max.   :410979  </span></span>
<span id="cb246-24"><a href="basic-statistics-in-r.html#cb246-24" aria-hidden="true" tabindex="-1"></a><span class="co">#   SRR1039517         SRR1039520         SRR1039521      </span></span>
<span id="cb246-25"><a href="basic-statistics-in-r.html#cb246-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Min.   :     0.0   Min.   :     0.0   Min.   :     0.0  </span></span>
<span id="cb246-26"><a href="basic-statistics-in-r.html#cb246-26" aria-hidden="true" tabindex="-1"></a><span class="co"># 1st Qu.:     0.0   1st Qu.:     0.0   1st Qu.:     0.0  </span></span>
<span id="cb246-27"><a href="basic-statistics-in-r.html#cb246-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Median :     1.0   Median :     1.0   Median :     1.0  </span></span>
<span id="cb246-28"><a href="basic-statistics-in-r.html#cb246-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Mean   :   822.5   Mean   :   509.0   Mean   :   565.7  </span></span>
<span id="cb246-29"><a href="basic-statistics-in-r.html#cb246-29" aria-hidden="true" tabindex="-1"></a><span class="co"># 3rd Qu.:   261.0   3rd Qu.:   185.8   3rd Qu.:   183.0  </span></span>
<span id="cb246-30"><a href="basic-statistics-in-r.html#cb246-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Max.   :416837.0   Max.   :378108.0   Max.   :372973.0  </span></span></code></pre></div>
<p>These summary output provide a general descriptive and summary statistics of data distribution of the values on each column (variable). For continuous variables, minimum and maximum values are provided. Other statistics include mean and median that provide a general measurement of central location (i.e., average). The 1st quartile (Q1) is the middle value between the minimum and the median; and it also represents measurement that marks where 25% of the data is below this point. The 3rd quartile (Q3) is the middle value between the median and the maximum; and also represents 75% of the data is below this point. Notice the column named <code>ensgene</code> is a discrete (character) variable and only provides the number of elements in that column.</p>
<p>We can visualize these summaries using boxplots. The base package has the function <code>boxplot()</code>.</p>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="basic-statistics-in-r.html#cb247-1" aria-hidden="true" tabindex="-1"></a><span class="co">#car data boxplot</span></span>
<span id="cb247-2"><a href="basic-statistics-in-r.html#cb247-2" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(mpg<span class="sc">~</span>cyl, <span class="at">data=</span>cars_data, <span class="at">col=</span>(<span class="fu">c</span>(<span class="st">&quot;#DA291CFF&quot;</span>, <span class="st">&quot;#56A8CBFF&quot;</span>, <span class="st">&quot;#53A567FF&quot;</span>)), <span class="at">main=</span><span class="st">&quot;car data boxplots&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;cylinders&quot;</span>) </span></code></pre></div>
<p><img src="figures/session_6_figs/1_1_boxplot_cars.png" width="100%" /></p>
<p>You can also visualize these summaries using boxplots for all variables using the R-package <a href="https://cran.r-project.org/web/packages/reshape2/index.html" target="_blank"><strong>reshape2</strong></a> and its function <code>melt()</code>. This will reshape the data frame to be more amenable for boxplots.</p>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="basic-statistics-in-r.html#cb248-1" aria-hidden="true" tabindex="-1"></a><span class="co"># you might need to install R-package &#39;reshape2&#39; </span></span>
<span id="cb248-2"><a href="basic-statistics-in-r.html#cb248-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;reshape2&quot;</span>)</span>
<span id="cb248-3"><a href="basic-statistics-in-r.html#cb248-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb248-4"><a href="basic-statistics-in-r.html#cb248-4" aria-hidden="true" tabindex="-1"></a><span class="co"># reduce the &quot;airway_data&quot; to only the continuous variables.</span></span>
<span id="cb248-5"><a href="basic-statistics-in-r.html#cb248-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-6"><a href="basic-statistics-in-r.html#cb248-6" aria-hidden="true" tabindex="-1"></a>airway_data_very_high_low <span class="ot">&lt;-</span> airway_data</span>
<span id="cb248-7"><a href="basic-statistics-in-r.html#cb248-7" aria-hidden="true" tabindex="-1"></a>airway_data_reduced2 <span class="ot">&lt;-</span> airway_data[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(airway_data_very_high_low)]</span>
<span id="cb248-8"><a href="basic-statistics-in-r.html#cb248-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data_reduced2)</span>
<span id="cb248-9"><a href="basic-statistics-in-r.html#cb248-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb248-10"><a href="basic-statistics-in-r.html#cb248-10" aria-hidden="true" tabindex="-1"></a><span class="co">#1        723        486        904        445       1170       1097        806        604</span></span>
<span id="cb248-11"><a href="basic-statistics-in-r.html#cb248-11" aria-hidden="true" tabindex="-1"></a><span class="co">#2          0          0          0          0          0          0          0          0</span></span>
<span id="cb248-12"><a href="basic-statistics-in-r.html#cb248-12" aria-hidden="true" tabindex="-1"></a><span class="co">#3        467        523        616        371        582        781        417        509</span></span>
<span id="cb248-13"><a href="basic-statistics-in-r.html#cb248-13" aria-hidden="true" tabindex="-1"></a><span class="co">#4        347        258        364        237        318        447        330        324</span></span>
<span id="cb248-14"><a href="basic-statistics-in-r.html#cb248-14" aria-hidden="true" tabindex="-1"></a><span class="co">#5         96         81         73         66        118         94        102         74</span></span>
<span id="cb248-15"><a href="basic-statistics-in-r.html#cb248-15" aria-hidden="true" tabindex="-1"></a><span class="co">#6          0          0          1          0          2          0          0          0</span></span>
<span id="cb248-16"><a href="basic-statistics-in-r.html#cb248-16" aria-hidden="true" tabindex="-1"></a>airway_data_reduced2_melted <span class="ot">&lt;-</span> <span class="fu">melt</span>(airway_data_reduced2)</span>
<span id="cb248-17"><a href="basic-statistics-in-r.html#cb248-17" aria-hidden="true" tabindex="-1"></a><span class="co">#No id variables; using all as measure variables</span></span>
<span id="cb248-18"><a href="basic-statistics-in-r.html#cb248-18" aria-hidden="true" tabindex="-1"></a><span class="co"># we will change the names of the columns to reflect what they represent.</span></span>
<span id="cb248-19"><a href="basic-statistics-in-r.html#cb248-19" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(airway_data_reduced2_melted) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;sample&quot;</span>, <span class="st">&quot;expression&quot;</span>)</span>
<span id="cb248-20"><a href="basic-statistics-in-r.html#cb248-20" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data_reduced2_melted)</span>
<span id="cb248-21"><a href="basic-statistics-in-r.html#cb248-21" aria-hidden="true" tabindex="-1"></a><span class="co">#    sample expression</span></span>
<span id="cb248-22"><a href="basic-statistics-in-r.html#cb248-22" aria-hidden="true" tabindex="-1"></a><span class="co">#1 SRR1039508   723</span></span>
<span id="cb248-23"><a href="basic-statistics-in-r.html#cb248-23" aria-hidden="true" tabindex="-1"></a><span class="co">#2 SRR1039508     0</span></span>
<span id="cb248-24"><a href="basic-statistics-in-r.html#cb248-24" aria-hidden="true" tabindex="-1"></a><span class="co">#3 SRR1039508   467</span></span>
<span id="cb248-25"><a href="basic-statistics-in-r.html#cb248-25" aria-hidden="true" tabindex="-1"></a><span class="co">#4 SRR1039508   347</span></span>
<span id="cb248-26"><a href="basic-statistics-in-r.html#cb248-26" aria-hidden="true" tabindex="-1"></a><span class="co">#5 SRR1039508    96</span></span>
<span id="cb248-27"><a href="basic-statistics-in-r.html#cb248-27" aria-hidden="true" tabindex="-1"></a><span class="co">#6 SRR1039508     0</span></span>
<span id="cb248-28"><a href="basic-statistics-in-r.html#cb248-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb248-29"><a href="basic-statistics-in-r.html#cb248-29" aria-hidden="true" tabindex="-1"></a><span class="co">#airway_data boxplots</span></span>
<span id="cb248-30"><a href="basic-statistics-in-r.html#cb248-30" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(expression<span class="sc">~</span>sample, <span class="at">data=</span>airway_data_reduced2_melted, <span class="at">main=</span><span class="st">&quot;airway_data boxplots&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;samples&quot;</span>) </span></code></pre></div>
<p><img src="figures/session_6_figs/1_2_boxplot_airway_sample.png" width="100%" /></p>
<p>This figure is not very elegant (we will work with graph in that section). However, these boxplots are extremely skewed to low expression values.</p>
<p><strong>3)</strong> We can discretize our data by creating groups based on criterium (e.g., threshold value). Such discrete groups can help us to define if a treatment or condition is reflected in our data or experimental groups. We will create a binary data frame for <code>airway_data</code> for the columns (variables) that represent the normalized gene expression for the samples that start with<code>SRR</code>. For this example, we will consider an arbitrary scale to divide into groups as: <strong>very high</strong> (x =&gt; 10000), <strong>high</strong> (10000 &gt; x =&gt; 100) and <strong>low</strong> (x &lt; 100) expression levels.</p>
<p>This task can be perform using a specialized R-package: <a href="https://purrr.tidyverse.org/" target="_blank"><strong>purrr</strong></a> and its function <code>modify()</code> with a conditional <code>ifelse()</code>. We already introduce <strong>purrr</strong> on previous sessions. Notice that when we close some of the commands we use complex punctuation (i.e., <code>(</code>, <code>)</code>, <code>{</code>, <code>}</code>) and many mistakes can happen if you forget to close such punctuation pairs (i.e., missing one is a very common error)..</p>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="basic-statistics-in-r.html#cb249-1" aria-hidden="true" tabindex="-1"></a><span class="co"># we already installed R-package &#39;purrr&#39; </span></span>
<span id="cb249-2"><a href="basic-statistics-in-r.html#cb249-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb249-3"><a href="basic-statistics-in-r.html#cb249-3" aria-hidden="true" tabindex="-1"></a><span class="co"># we create an alternative object, so we original raw data airway_data</span></span>
<span id="cb249-4"><a href="basic-statistics-in-r.html#cb249-4" aria-hidden="true" tabindex="-1"></a>airway_data_very_high_low <span class="ot">&lt;-</span> airway_data</span>
<span id="cb249-5"><a href="basic-statistics-in-r.html#cb249-5" aria-hidden="true" tabindex="-1"></a><span class="co"># We will apply a the conditional</span></span>
<span id="cb249-6"><a href="basic-statistics-in-r.html#cb249-6" aria-hidden="true" tabindex="-1"></a>airway_data_very_high_low <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">modify</span>(airway_data_very_high_low,<span class="cf">function</span>(x) {<span class="fu">ifelse</span>(x <span class="sc">&gt;=</span> <span class="dv">10000</span>, <span class="st">&quot;very_high&quot;</span>,</span>
<span id="cb249-7"><a href="basic-statistics-in-r.html#cb249-7" aria-hidden="true" tabindex="-1"></a>                                                                                     <span class="fu">ifelse</span>( x <span class="sc">&gt;=</span> <span class="dv">1000</span>, <span class="st">&quot;high&quot;</span>, <span class="st">&quot;low&quot;</span>))})</span>
<span id="cb249-8"><a href="basic-statistics-in-r.html#cb249-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data_very_high_low)</span>
<span id="cb249-9"><a href="basic-statistics-in-r.html#cb249-9" aria-hidden="true" tabindex="-1"></a><span class="co">#    ensgene SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb249-10"><a href="basic-statistics-in-r.html#cb249-10" aria-hidden="true" tabindex="-1"></a><span class="co">#1 very_high        low        low        low        low       high       high        low        low</span></span>
<span id="cb249-11"><a href="basic-statistics-in-r.html#cb249-11" aria-hidden="true" tabindex="-1"></a><span class="co">#2 very_high        low        low        low        low        low        low        low        low</span></span>
<span id="cb249-12"><a href="basic-statistics-in-r.html#cb249-12" aria-hidden="true" tabindex="-1"></a><span class="co">#3 very_high        low        low        low        low        low        low        low        low</span></span>
<span id="cb249-13"><a href="basic-statistics-in-r.html#cb249-13" aria-hidden="true" tabindex="-1"></a><span class="co">#4 very_high        low        low        low        low        low        low        low        low</span></span>
<span id="cb249-14"><a href="basic-statistics-in-r.html#cb249-14" aria-hidden="true" tabindex="-1"></a><span class="co">#5 very_high        low        low        low        low        low        low        low        low</span></span>
<span id="cb249-15"><a href="basic-statistics-in-r.html#cb249-15" aria-hidden="true" tabindex="-1"></a><span class="co">#6 very_high        low        low        low        low        low        low        low        low</span></span>
<span id="cb249-16"><a href="basic-statistics-in-r.html#cb249-16" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span></code></pre></div>
<p>During the process above, we have “lost” the values of our variable <code>ensgene</code> that contains the names of the loci. We can add it back from our original data frame <code>airway_data</code> as indicated next.</p>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="basic-statistics-in-r.html#cb250-1" aria-hidden="true" tabindex="-1"></a>airway_data_very_high_low<span class="sc">$</span>ensgene <span class="ot">&lt;-</span> airway_data<span class="sc">$</span>ensgene</span>
<span id="cb250-2"><a href="basic-statistics-in-r.html#cb250-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data_very_high_low)</span>
<span id="cb250-3"><a href="basic-statistics-in-r.html#cb250-3" aria-hidden="true" tabindex="-1"></a><span class="co">#          ensgene SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb250-4"><a href="basic-statistics-in-r.html#cb250-4" aria-hidden="true" tabindex="-1"></a><span class="co">#1 ENSG00000000003        low        low        low        low       high       high        low        low</span></span>
<span id="cb250-5"><a href="basic-statistics-in-r.html#cb250-5" aria-hidden="true" tabindex="-1"></a><span class="co">#2 ENSG00000000005        low        low        low        low        low        low        low        low</span></span>
<span id="cb250-6"><a href="basic-statistics-in-r.html#cb250-6" aria-hidden="true" tabindex="-1"></a><span class="co">#3 ENSG00000000419        low        low        low        low        low        low        low        low</span></span>
<span id="cb250-7"><a href="basic-statistics-in-r.html#cb250-7" aria-hidden="true" tabindex="-1"></a><span class="co">#4 ENSG00000000457        low        low        low        low        low        low        low        low</span></span>
<span id="cb250-8"><a href="basic-statistics-in-r.html#cb250-8" aria-hidden="true" tabindex="-1"></a><span class="co">#5 ENSG00000000460        low        low        low        low        low        low        low        low</span></span>
<span id="cb250-9"><a href="basic-statistics-in-r.html#cb250-9" aria-hidden="true" tabindex="-1"></a><span class="co">#6 ENSG00000000938        low        low        low        low        low        low        low        low</span></span>
<span id="cb250-10"><a href="basic-statistics-in-r.html#cb250-10" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span></code></pre></div>
<p><strong>4)</strong> With this discretized data frame <code>airway_data_very_high_low</code>, we can perform contingency tables. These are a type of table in a matrix format that displays the frequency distribution of the variables tested. We will exclude the first column <code>ensgene</code> because it has unique values.</p>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="basic-statistics-in-r.html#cb251-1" aria-hidden="true" tabindex="-1"></a>airway_data_reduced <span class="ot">&lt;-</span> airway_data_very_high_low[,<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(airway_data_very_high_low)]</span>
<span id="cb251-2"><a href="basic-statistics-in-r.html#cb251-2" aria-hidden="true" tabindex="-1"></a><span class="co"># collect frequencies on samples</span></span>
<span id="cb251-3"><a href="basic-statistics-in-r.html#cb251-3" aria-hidden="true" tabindex="-1"></a>summary_frequencies_airway_data <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb251-4"><a href="basic-statistics-in-r.html#cb251-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(airway_data_reduced)) {</span>
<span id="cb251-5"><a href="basic-statistics-in-r.html#cb251-5" aria-hidden="true" tabindex="-1"></a>                    sample_name <span class="ot">&lt;-</span> <span class="fu">names</span>(airway_data_reduced[i])</span>
<span id="cb251-6"><a href="basic-statistics-in-r.html#cb251-6" aria-hidden="true" tabindex="-1"></a>                      one_table <span class="ot">&lt;-</span> <span class="fu">table</span>(airway_data_reduced[i])</span>
<span id="cb251-7"><a href="basic-statistics-in-r.html#cb251-7" aria-hidden="true" tabindex="-1"></a>                      <span class="co"># from table to vector</span></span>
<span id="cb251-8"><a href="basic-statistics-in-r.html#cb251-8" aria-hidden="true" tabindex="-1"></a>          one_contigency_vector <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(one_table)</span>
<span id="cb251-9"><a href="basic-statistics-in-r.html#cb251-9" aria-hidden="true" tabindex="-1"></a>                      <span class="co"># add names of categories</span></span>
<span id="cb251-10"><a href="basic-statistics-in-r.html#cb251-10" aria-hidden="true" tabindex="-1"></a>   <span class="fu">names</span>(one_contigency_vector) <span class="ot">&lt;-</span> <span class="fu">names</span>(one_table)</span>
<span id="cb251-11"><a href="basic-statistics-in-r.html#cb251-11" aria-hidden="true" tabindex="-1"></a>                 one_data_frame <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(one_contigency_vector)</span>
<span id="cb251-12"><a href="basic-statistics-in-r.html#cb251-12" aria-hidden="true" tabindex="-1"></a>                     <span class="co"># add name of sample</span></span>
<span id="cb251-13"><a href="basic-statistics-in-r.html#cb251-13" aria-hidden="true" tabindex="-1"></a>          <span class="fu">names</span>(one_data_frame) <span class="ot">&lt;-</span> sample_name</span>
<span id="cb251-14"><a href="basic-statistics-in-r.html#cb251-14" aria-hidden="true" tabindex="-1"></a>                     <span class="co"># collect results in list</span></span>
<span id="cb251-15"><a href="basic-statistics-in-r.html#cb251-15" aria-hidden="true" tabindex="-1"></a>    summary_frequencies_airway_data[[i]] <span class="ot">&lt;-</span> one_data_frame     </span>
<span id="cb251-16"><a href="basic-statistics-in-r.html#cb251-16" aria-hidden="true" tabindex="-1"></a>                                }</span>
<span id="cb251-17"><a href="basic-statistics-in-r.html#cb251-17" aria-hidden="true" tabindex="-1"></a>summary_frequencies_airway_data</span>
<span id="cb251-18"><a href="basic-statistics-in-r.html#cb251-18" aria-hidden="true" tabindex="-1"></a><span class="co">#[[1]]</span></span>
<span id="cb251-19"><a href="basic-statistics-in-r.html#cb251-19" aria-hidden="true" tabindex="-1"></a><span class="co">#          SRR1039508</span></span>
<span id="cb251-20"><a href="basic-statistics-in-r.html#cb251-20" aria-hidden="true" tabindex="-1"></a><span class="co">#high            3732</span></span>
<span id="cb251-21"><a href="basic-statistics-in-r.html#cb251-21" aria-hidden="true" tabindex="-1"></a><span class="co">#low            34691</span></span>
<span id="cb251-22"><a href="basic-statistics-in-r.html#cb251-22" aria-hidden="true" tabindex="-1"></a><span class="co">#very_high        271</span></span>
<span id="cb251-23"><a href="basic-statistics-in-r.html#cb251-23" aria-hidden="true" tabindex="-1"></a><span class="co">#[[2]]</span></span>
<span id="cb251-24"><a href="basic-statistics-in-r.html#cb251-24" aria-hidden="true" tabindex="-1"></a><span class="co">#          SRR1039509</span></span>
<span id="cb251-25"><a href="basic-statistics-in-r.html#cb251-25" aria-hidden="true" tabindex="-1"></a><span class="co">#high            3444</span></span>
<span id="cb251-26"><a href="basic-statistics-in-r.html#cb251-26" aria-hidden="true" tabindex="-1"></a><span class="co">#low            34986</span></span>
<span id="cb251-27"><a href="basic-statistics-in-r.html#cb251-27" aria-hidden="true" tabindex="-1"></a><span class="co">#very_high        264</span></span>
<span id="cb251-28"><a href="basic-statistics-in-r.html#cb251-28" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span></code></pre></div>
<p>With the list <code>summary_frequencies_airway_data</code>, we can cover it to a data frame by binding each vector as a column using two functions <code>do.call()</code> and <code>cbind()</code>.</p>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="basic-statistics-in-r.html#cb252-1" aria-hidden="true" tabindex="-1"></a>final_summary_all_samples_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, summary_frequencies_airway_data)</span>
<span id="cb252-2"><a href="basic-statistics-in-r.html#cb252-2" aria-hidden="true" tabindex="-1"></a>final_summary_all_samples_df</span>
<span id="cb252-3"><a href="basic-statistics-in-r.html#cb252-3" aria-hidden="true" tabindex="-1"></a><span class="co">#          SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb252-4"><a href="basic-statistics-in-r.html#cb252-4" aria-hidden="true" tabindex="-1"></a><span class="co">#high            3732       3444       4118       2607       4155       4703       3432       3683</span></span>
<span id="cb252-5"><a href="basic-statistics-in-r.html#cb252-5" aria-hidden="true" tabindex="-1"></a><span class="co">#low            34691      34986      34216      35902      34196      33500      35024      34725</span></span>
<span id="cb252-6"><a href="basic-statistics-in-r.html#cb252-6" aria-hidden="true" tabindex="-1"></a><span class="co">#very_high        271        264        360        185        343        491        238        286</span></span></code></pre></div>
<p><strong>5)</strong> With this data frame, we can test these discretized variables for independence using a chi-squared test with the function <code>summary()</code>.</p>
<p>For example, we can compare two of the <code>airway_data_reduced</code> samples (i.e., columns). Notice that I use a <code>;</code> command at the end of one line of commands, this allows you to paste several commands next to each other as if they were in new lines.</p>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="basic-statistics-in-r.html#cb253-1" aria-hidden="true" tabindex="-1"></a>sample_names <span class="ot">&lt;-</span> <span class="fu">names</span>(airway_data_reduced)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb253-2"><a href="basic-statistics-in-r.html#cb253-2" aria-hidden="true" tabindex="-1"></a>title_test <span class="ot">&lt;-</span> <span class="fu">paste0</span>(sample_names, <span class="at">collapse =</span><span class="st">&quot;..vs..&quot;</span>)</span>
<span id="cb253-3"><a href="basic-statistics-in-r.html#cb253-3" aria-hidden="true" tabindex="-1"></a>contigency_result <span class="ot">&lt;-</span> <span class="fu">summary</span>(<span class="fu">table</span>(airway_data_reduced[,<span class="dv">1</span>], airway_data_reduced[,<span class="dv">2</span>]))</span>
<span id="cb253-4"><a href="basic-statistics-in-r.html#cb253-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(title_test, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>); <span class="fu">print</span>(contigency_result)</span>
<span id="cb253-5"><a href="basic-statistics-in-r.html#cb253-5" aria-hidden="true" tabindex="-1"></a><span class="co">#SRR1039508..vs..SRR1039509 </span></span>
<span id="cb253-6"><a href="basic-statistics-in-r.html#cb253-6" aria-hidden="true" tabindex="-1"></a><span class="co">#ENumber of cases in table: 38694 </span></span>
<span id="cb253-7"><a href="basic-statistics-in-r.html#cb253-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Number of factors: 2 </span></span>
<span id="cb253-8"><a href="basic-statistics-in-r.html#cb253-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Test for independence of all factors:</span></span>
<span id="cb253-9"><a href="basic-statistics-in-r.html#cb253-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   Chisq = 58193, df = 4, p-value = 0</span></span>
<span id="cb253-10"><a href="basic-statistics-in-r.html#cb253-10" aria-hidden="true" tabindex="-1"></a><span class="co">#   Chi-squared approximation may be incorrect</span></span></code></pre></div>
<p>These results indicate that we can conclude that these samples are likely independent, but the large number of cases <strong>38694</strong> might have inflated the Chisq statistic.</p>
<p>Next we have an interesting loop for you to dissect, it will run all pairwise contingency tests, report the process, and add names of the pairs being compared. Notice the iterative variable <code>counter</code> and function <code>break</code> that breaks the loop when it has finished.</p>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="basic-statistics-in-r.html#cb254-1" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb254-2"><a href="basic-statistics-in-r.html#cb254-2" aria-hidden="true" tabindex="-1"></a>collect_test_independence <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb254-3"><a href="basic-statistics-in-r.html#cb254-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb254-4"><a href="basic-statistics-in-r.html#cb254-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(airway_data_reduced)) {</span>
<span id="cb254-5"><a href="basic-statistics-in-r.html#cb254-5" aria-hidden="true" tabindex="-1"></a>               <span class="co"># i &lt;- 1</span></span>
<span id="cb254-6"><a href="basic-statistics-in-r.html#cb254-6" aria-hidden="true" tabindex="-1"></a>               names_sample_i <span class="ot">&lt;-</span> <span class="fu">names</span>(airway_data_reduced)[i]</span>
<span id="cb254-7"><a href="basic-statistics-in-r.html#cb254-7" aria-hidden="true" tabindex="-1"></a>                n <span class="ot">&lt;-</span> i <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb254-8"><a href="basic-statistics-in-r.html#cb254-8" aria-hidden="true" tabindex="-1"></a>      upper_limit <span class="ot">&lt;-</span> <span class="fu">ncol</span>(airway_data_reduced)<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb254-9"><a href="basic-statistics-in-r.html#cb254-9" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span>(n <span class="sc">&lt;</span> upper_limit) {</span>
<span id="cb254-10"><a href="basic-statistics-in-r.html#cb254-10" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">cat</span>(<span class="st">&quot;sample:&quot;</span>,names_sample_i, <span class="st">&quot;..vs..&quot;</span>)</span>
<span id="cb254-11"><a href="basic-statistics-in-r.html#cb254-11" aria-hidden="true" tabindex="-1"></a>                      <span class="cf">for</span>(j <span class="cf">in</span> n<span class="sc">:</span><span class="fu">ncol</span>(airway_data_reduced)) {</span>
<span id="cb254-12"><a href="basic-statistics-in-r.html#cb254-12" aria-hidden="true" tabindex="-1"></a>                                    counter <span class="ot">&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb254-13"><a href="basic-statistics-in-r.html#cb254-13" aria-hidden="true" tabindex="-1"></a>                                    names_sample_j <span class="ot">&lt;-</span> <span class="fu">names</span>(airway_data_reduced)[j]</span>
<span id="cb254-14"><a href="basic-statistics-in-r.html#cb254-14" aria-hidden="true" tabindex="-1"></a>                                    <span class="fu">cat</span>(names_sample_j, <span class="st">&quot;....&quot;</span>)</span>
<span id="cb254-15"><a href="basic-statistics-in-r.html#cb254-15" aria-hidden="true" tabindex="-1"></a>                                  list_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(names_sample_i,<span class="st">&quot;..vs..&quot;</span>, names_sample_j)</span>
<span id="cb254-16"><a href="basic-statistics-in-r.html#cb254-16" aria-hidden="true" tabindex="-1"></a>       collect_test_independence[[counter]] <span class="ot">&lt;-</span> <span class="fu">summary</span>(<span class="fu">table</span>(airway_data_reduced[,i], airway_data_reduced[,j]))</span>
<span id="cb254-17"><a href="basic-statistics-in-r.html#cb254-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(collect_test_independence)[counter] <span class="ot">&lt;-</span> list_name</span>
<span id="cb254-18"><a href="basic-statistics-in-r.html#cb254-18" aria-hidden="true" tabindex="-1"></a>                                                             }</span>
<span id="cb254-19"><a href="basic-statistics-in-r.html#cb254-19" aria-hidden="true" tabindex="-1"></a>                                                           <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)                                             </span>
<span id="cb254-20"><a href="basic-statistics-in-r.html#cb254-20" aria-hidden="true" tabindex="-1"></a>                                   } <span class="cf">else</span> {<span class="cf">break</span>}</span>
<span id="cb254-21"><a href="basic-statistics-in-r.html#cb254-21" aria-hidden="true" tabindex="-1"></a>                                        }</span>
<span id="cb254-22"><a href="basic-statistics-in-r.html#cb254-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb254-23"><a href="basic-statistics-in-r.html#cb254-23" aria-hidden="true" tabindex="-1"></a><span class="co">#sample: SRR1039508 ..vs..SRR1039509 ....SRR1039512 ....SRR1039513 ....SRR1039516 ....SRR1039517 ....SRR1039520 ....SRR1039521 ....</span></span>
<span id="cb254-24"><a href="basic-statistics-in-r.html#cb254-24" aria-hidden="true" tabindex="-1"></a><span class="co">#sample: SRR1039509 ..vs..SRR1039512 ....SRR1039513 ....SRR1039516 ....SRR1039517 ....SRR1039520 ....SRR1039521 ....</span></span>
<span id="cb254-25"><a href="basic-statistics-in-r.html#cb254-25" aria-hidden="true" tabindex="-1"></a><span class="co">#sample: SRR1039512 ..vs..SRR1039513 ....SRR1039516 ....SRR1039517 ....SRR1039520 ....SRR1039521 ....</span></span>
<span id="cb254-26"><a href="basic-statistics-in-r.html#cb254-26" aria-hidden="true" tabindex="-1"></a><span class="co">#sample: SRR1039513 ..vs..SRR1039516 ....SRR1039517 ....SRR1039520 ....SRR1039521 ....</span></span>
<span id="cb254-27"><a href="basic-statistics-in-r.html#cb254-27" aria-hidden="true" tabindex="-1"></a><span class="co">#sample: SRR1039516 ..vs..SRR1039517 ....SRR1039520 ....SRR1039521 ....</span></span>
<span id="cb254-28"><a href="basic-statistics-in-r.html#cb254-28" aria-hidden="true" tabindex="-1"></a><span class="co">#sample: SRR1039517 ..vs..SRR1039520 ....SRR1039521 ....</span></span>
<span id="cb254-29"><a href="basic-statistics-in-r.html#cb254-29" aria-hidden="true" tabindex="-1"></a><span class="co">#sample: SRR1039520 ..vs..SRR1039521 ....</span></span>
<span id="cb254-30"><a href="basic-statistics-in-r.html#cb254-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb254-31"><a href="basic-statistics-in-r.html#cb254-31" aria-hidden="true" tabindex="-1"></a><span class="co"># if you call the collect_test_independence, you will get the statistics for independence</span></span>
<span id="cb254-32"><a href="basic-statistics-in-r.html#cb254-32" aria-hidden="true" tabindex="-1"></a>collect_test_independence</span>
<span id="cb254-33"><a href="basic-statistics-in-r.html#cb254-33" aria-hidden="true" tabindex="-1"></a><span class="co">#$SRR1039508..vs..SRR1039509</span></span>
<span id="cb254-34"><a href="basic-statistics-in-r.html#cb254-34" aria-hidden="true" tabindex="-1"></a><span class="co">#Number of cases in table: 38694 </span></span>
<span id="cb254-35"><a href="basic-statistics-in-r.html#cb254-35" aria-hidden="true" tabindex="-1"></a><span class="co">#Number of factors: 2 </span></span>
<span id="cb254-36"><a href="basic-statistics-in-r.html#cb254-36" aria-hidden="true" tabindex="-1"></a><span class="co">#Test for independence of all factors:</span></span>
<span id="cb254-37"><a href="basic-statistics-in-r.html#cb254-37" aria-hidden="true" tabindex="-1"></a><span class="co">#   Chisq = 58193, df = 4, p-value = 0</span></span>
<span id="cb254-38"><a href="basic-statistics-in-r.html#cb254-38" aria-hidden="true" tabindex="-1"></a><span class="co">#   Chi-squared approximation may be incorrect</span></span>
<span id="cb254-39"><a href="basic-statistics-in-r.html#cb254-39" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb254-40"><a href="basic-statistics-in-r.html#cb254-40" aria-hidden="true" tabindex="-1"></a><span class="co">#$SRR1039508..vs..SRR1039512</span></span>
<span id="cb254-41"><a href="basic-statistics-in-r.html#cb254-41" aria-hidden="true" tabindex="-1"></a><span class="co">#Number of cases in table: 38694 </span></span>
<span id="cb254-42"><a href="basic-statistics-in-r.html#cb254-42" aria-hidden="true" tabindex="-1"></a><span class="co">#Number of factors: 2 </span></span>
<span id="cb254-43"><a href="basic-statistics-in-r.html#cb254-43" aria-hidden="true" tabindex="-1"></a><span class="co">#Test for independence of all factors:</span></span>
<span id="cb254-44"><a href="basic-statistics-in-r.html#cb254-44" aria-hidden="true" tabindex="-1"></a><span class="co">#   Chisq = 58274, df = 4, p-value = 0</span></span>
<span id="cb254-45"><a href="basic-statistics-in-r.html#cb254-45" aria-hidden="true" tabindex="-1"></a><span class="co">#   Chi-squared approximation may be incorrect</span></span></code></pre></div>
</div>
<div id="sub-setting-your-dataset" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Sub-setting your dataset</h2>
<p>It is often the case that you do not want to run some analysis with some part and not all your data. This will require that your subset (i.e., select some parts) your data.</p>
<p>We will explore some different forms to subset your data. For that purpose, we will use <code>Cars93</code> dataset from the R-package <code>MASS</code>.</p>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="basic-statistics-in-r.html#cb255-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we can install MASS and load the dataset for this example</span></span>
<span id="cb255-2"><a href="basic-statistics-in-r.html#cb255-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;MASS&quot;</span>)</span>
<span id="cb255-3"><a href="basic-statistics-in-r.html#cb255-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb255-4"><a href="basic-statistics-in-r.html#cb255-4" aria-hidden="true" tabindex="-1"></a><span class="do">## We assign Cars93 dataset to an object</span></span>
<span id="cb255-5"><a href="basic-statistics-in-r.html#cb255-5" aria-hidden="true" tabindex="-1"></a>my_cars <span class="ot">&lt;-</span> Cars93</span>
<span id="cb255-6"><a href="basic-statistics-in-r.html#cb255-6" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(my_cars)</span>
<span id="cb255-7"><a href="basic-statistics-in-r.html#cb255-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  93 obs. of  27 variables:</span></span>
<span id="cb255-8"><a href="basic-statistics-in-r.html#cb255-8" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Manufacturer      : Factor w/ 32 levels &quot;Acura&quot;,&quot;Audi&quot;,..: 1 1 2 2 3 4 4 4 4 5 ...</span></span>
<span id="cb255-9"><a href="basic-statistics-in-r.html#cb255-9" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Model             : Factor w/ 93 levels &quot;100&quot;,&quot;190E&quot;,&quot;240&quot;,..: 49 56 9 1 6 24 54 74 73 35 ...</span></span>
<span id="cb255-10"><a href="basic-statistics-in-r.html#cb255-10" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Type              : Factor w/ 6 levels &quot;Compact&quot;,&quot;Large&quot;,..: 4 3 1 3 3 3 2 2 3 2 ...</span></span>
<span id="cb255-11"><a href="basic-statistics-in-r.html#cb255-11" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Min.Price         : num  12.9 29.2 25.9 30.8 23.7 14.2 19.9 22.6 26.3 33 ...</span></span>
<span id="cb255-12"><a href="basic-statistics-in-r.html#cb255-12" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Price             : num  15.9 33.9 29.1 37.7 30 15.7 20.8 23.7 26.3 34.7 ...</span></span>
<span id="cb255-13"><a href="basic-statistics-in-r.html#cb255-13" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Max.Price         : num  18.8 38.7 32.3 44.6 36.2 17.3 21.7 24.9 26.3 36.3 ...</span></span>
<span id="cb255-14"><a href="basic-statistics-in-r.html#cb255-14" aria-hidden="true" tabindex="-1"></a><span class="co"># $ MPG.city          : int  25 18 20 19 22 22 19 16 19 16 ...</span></span>
<span id="cb255-15"><a href="basic-statistics-in-r.html#cb255-15" aria-hidden="true" tabindex="-1"></a><span class="co"># $ MPG.highway       : int  31 25 26 26 30 31 28 25 27 25 ...</span></span>
<span id="cb255-16"><a href="basic-statistics-in-r.html#cb255-16" aria-hidden="true" tabindex="-1"></a><span class="co"># $ AirBags           : Factor w/ 3 levels &quot;Driver &amp; Passenger&quot;,..: 3 1 2 1 2 2 2 2 2 2 ...</span></span>
<span id="cb255-17"><a href="basic-statistics-in-r.html#cb255-17" aria-hidden="true" tabindex="-1"></a><span class="co"># $ DriveTrain        : Factor w/ 3 levels &quot;4WD&quot;,&quot;Front&quot;,..: 2 2 2 2 3 2 2 3 2 2 ...</span></span>
<span id="cb255-18"><a href="basic-statistics-in-r.html#cb255-18" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Cylinders         : Factor w/ 6 levels &quot;3&quot;,&quot;4&quot;,&quot;5&quot;,&quot;6&quot;,..: 2 4 4 4 2 2 4 4 4 5 ...</span></span>
<span id="cb255-19"><a href="basic-statistics-in-r.html#cb255-19" aria-hidden="true" tabindex="-1"></a><span class="co"># $ EngineSize        : num  1.8 3.2 2.8 2.8 3.5 2.2 3.8 5.7 3.8 4.9 ...</span></span>
<span id="cb255-20"><a href="basic-statistics-in-r.html#cb255-20" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Horsepower        : int  140 200 172 172 208 110 170 180 170 200 ...</span></span>
<span id="cb255-21"><a href="basic-statistics-in-r.html#cb255-21" aria-hidden="true" tabindex="-1"></a><span class="co"># $ RPM               : int  6300 5500 5500 5500 5700 5200 4800 4000 4800 4100 ...</span></span>
<span id="cb255-22"><a href="basic-statistics-in-r.html#cb255-22" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Rev.per.mile      : int  2890 2335 2280 2535 2545 2565 1570 1320 1690 1510 ...</span></span>
<span id="cb255-23"><a href="basic-statistics-in-r.html#cb255-23" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Man.trans.avail   : Factor w/ 2 levels &quot;No&quot;,&quot;Yes&quot;: 2 2 2 2 2 1 1 1 1 1 ...</span></span>
<span id="cb255-24"><a href="basic-statistics-in-r.html#cb255-24" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Fuel.tank.capacity: num  13.2 18 16.9 21.1 21.1 16.4 18 23 18.8 18 ...</span></span>
<span id="cb255-25"><a href="basic-statistics-in-r.html#cb255-25" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Passengers        : int  5 5 5 6 4 6 6 6 5 6 ...</span></span>
<span id="cb255-26"><a href="basic-statistics-in-r.html#cb255-26" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Length            : int  177 195 180 193 186 189 200 216 198 206 ...</span></span>
<span id="cb255-27"><a href="basic-statistics-in-r.html#cb255-27" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Wheelbase         : int  102 115 102 106 109 105 111 116 108 114 ...</span></span>
<span id="cb255-28"><a href="basic-statistics-in-r.html#cb255-28" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Width             : int  68 71 67 70 69 69 74 78 73 73 ...</span></span>
<span id="cb255-29"><a href="basic-statistics-in-r.html#cb255-29" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Turn.circle       : int  37 38 37 37 39 41 42 45 41 43 ...</span></span>
<span id="cb255-30"><a href="basic-statistics-in-r.html#cb255-30" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Rear.seat.room    : num  26.5 30 28 31 27 28 30.5 30.5 26.5 35 ...</span></span>
<span id="cb255-31"><a href="basic-statistics-in-r.html#cb255-31" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Luggage.room      : int  11 15 14 17 13 16 17 21 14 18 ...</span></span>
<span id="cb255-32"><a href="basic-statistics-in-r.html#cb255-32" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Weight            : int  2705 3560 3375 3405 3640 2880 3470 4105 3495 3620 ...</span></span>
<span id="cb255-33"><a href="basic-statistics-in-r.html#cb255-33" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Origin            : Factor w/ 2 levels &quot;USA&quot;,&quot;non-USA&quot;: 2 2 2 2 2 1 1 1 1 1 ...</span></span>
<span id="cb255-34"><a href="basic-statistics-in-r.html#cb255-34" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Make              : Factor w/ 93 levels &quot;Acura Integra&quot;,..: 1 2 4 3 5 6 7 9 8 10 ...</span></span></code></pre></div>
<p>For example, you might want to include only some variables. We will select <code>Origin</code>, <code>Manufacturer</code>,<code>Price</code>, <code>Passengers</code>, <code>MPG.city</code>, and <code>EngineSize</code>.
For that purpose, we use the function <code>subset()</code> and notice it syntax.</p>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="basic-statistics-in-r.html#cb256-1" aria-hidden="true" tabindex="-1"></a>subset_my_cars <span class="ot">&lt;-</span> <span class="fu">subset</span>(my_cars, <span class="at">select =</span> <span class="fu">c</span>(Origin, Manufacturer, Price, Passengers, MPG.city, EngineSize))</span>
<span id="cb256-2"><a href="basic-statistics-in-r.html#cb256-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(subset_my_cars)</span>
<span id="cb256-3"><a href="basic-statistics-in-r.html#cb256-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  Origin Manufacturer Price Passengers MPG.city EngineSize</span></span>
<span id="cb256-4"><a href="basic-statistics-in-r.html#cb256-4" aria-hidden="true" tabindex="-1"></a><span class="co">#1 non-USA        Acura  15.9          5       25        1.8</span></span>
<span id="cb256-5"><a href="basic-statistics-in-r.html#cb256-5" aria-hidden="true" tabindex="-1"></a><span class="co">#2 non-USA        Acura  33.9          5       18        3.2</span></span>
<span id="cb256-6"><a href="basic-statistics-in-r.html#cb256-6" aria-hidden="true" tabindex="-1"></a><span class="co">#3 non-USA         Audi  29.1          5       20        2.8</span></span>
<span id="cb256-7"><a href="basic-statistics-in-r.html#cb256-7" aria-hidden="true" tabindex="-1"></a><span class="co">#4 non-USA         Audi  37.7          6       19        2.8</span></span>
<span id="cb256-8"><a href="basic-statistics-in-r.html#cb256-8" aria-hidden="true" tabindex="-1"></a><span class="co">#5 non-USA          BMW  30.0          4       22        3.5</span></span>
<span id="cb256-9"><a href="basic-statistics-in-r.html#cb256-9" aria-hidden="true" tabindex="-1"></a><span class="co">#6     USA        Buick  15.7          6       22        2.2</span></span></code></pre></div>
<p>You might want to include groups based on a character name (i.e., equal to some term). We will select only cars with <code>Origin</code> from the USA.</p>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb257-1"><a href="basic-statistics-in-r.html#cb257-1" aria-hidden="true" tabindex="-1"></a>only_usa_subset_my_cars <span class="ot">&lt;-</span> <span class="fu">subset</span>(subset_my_cars, Origin <span class="sc">==</span> <span class="st">&quot;USA&quot;</span>)</span>
<span id="cb257-2"><a href="basic-statistics-in-r.html#cb257-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(only_usa_subset_my_cars)</span>
<span id="cb257-3"><a href="basic-statistics-in-r.html#cb257-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   Origin Manufacturer Price Passengers MPG.city EngineSize</span></span>
<span id="cb257-4"><a href="basic-statistics-in-r.html#cb257-4" aria-hidden="true" tabindex="-1"></a><span class="co">#6     USA        Buick  15.7          6       22        2.2</span></span>
<span id="cb257-5"><a href="basic-statistics-in-r.html#cb257-5" aria-hidden="true" tabindex="-1"></a><span class="co">#7     USA        Buick  20.8          6       19        3.8</span></span>
<span id="cb257-6"><a href="basic-statistics-in-r.html#cb257-6" aria-hidden="true" tabindex="-1"></a><span class="co">#8     USA        Buick  23.7          6       16        5.7</span></span>
<span id="cb257-7"><a href="basic-statistics-in-r.html#cb257-7" aria-hidden="true" tabindex="-1"></a><span class="co">#9     USA        Buick  26.3          5       19        3.8</span></span>
<span id="cb257-8"><a href="basic-statistics-in-r.html#cb257-8" aria-hidden="true" tabindex="-1"></a><span class="co">#10    USA     Cadillac  34.7          6       16        4.9</span></span>
<span id="cb257-9"><a href="basic-statistics-in-r.html#cb257-9" aria-hidden="true" tabindex="-1"></a><span class="co">#11    USA     Cadillac  40.1          5       16        4.6</span></span></code></pre></div>
<p>You might want to include groups based on a value. We will select only cars with an <code>EngineSize</code> more or equal to 3.</p>
<div class="sourceCode" id="cb258"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb258-1"><a href="basic-statistics-in-r.html#cb258-1" aria-hidden="true" tabindex="-1"></a>only_large_engine_subset_my_cars <span class="ot">&lt;-</span> <span class="fu">subset</span>(subset_my_cars, EngineSize <span class="sc">&gt;=</span> <span class="fl">3.0</span>)</span>
<span id="cb258-2"><a href="basic-statistics-in-r.html#cb258-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(only_large_engine_subset_my_cars)</span>
<span id="cb258-3"><a href="basic-statistics-in-r.html#cb258-3" aria-hidden="true" tabindex="-1"></a><span class="co">#       Origin Manufacturer Price Passengers MPG.city EngineSize</span></span>
<span id="cb258-4"><a href="basic-statistics-in-r.html#cb258-4" aria-hidden="true" tabindex="-1"></a><span class="co">#2  non-USA        Acura  33.9          5       18        3.2</span></span>
<span id="cb258-5"><a href="basic-statistics-in-r.html#cb258-5" aria-hidden="true" tabindex="-1"></a><span class="co">#5  non-USA          BMW  30.0          4       22        3.5</span></span>
<span id="cb258-6"><a href="basic-statistics-in-r.html#cb258-6" aria-hidden="true" tabindex="-1"></a><span class="co">#7      USA        Buick  20.8          6       19        3.8</span></span>
<span id="cb258-7"><a href="basic-statistics-in-r.html#cb258-7" aria-hidden="true" tabindex="-1"></a><span class="co">#8      USA        Buick  23.7          6       16        5.7</span></span>
<span id="cb258-8"><a href="basic-statistics-in-r.html#cb258-8" aria-hidden="true" tabindex="-1"></a><span class="co">#9      USA        Buick  26.3          5       19        3.8</span></span>
<span id="cb258-9"><a href="basic-statistics-in-r.html#cb258-9" aria-hidden="true" tabindex="-1"></a><span class="co">#10     USA     Cadillac  34.7          6       16        4.9</span></span></code></pre></div>
<p>You might want to combine several considerations. We will select only cars not made in the USA with an <code>EngineSize</code> less or equal to 4.</p>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="basic-statistics-in-r.html#cb259-1" aria-hidden="true" tabindex="-1"></a>some_of_my_cars <span class="ot">&lt;-</span> <span class="fu">subset</span>(subset_my_cars, Origin <span class="sc">==</span> <span class="st">&quot;non-USA&quot;</span> <span class="sc">&amp;</span> EngineSize <span class="sc">&lt;=</span> <span class="fl">4.0</span>)</span>
<span id="cb259-2"><a href="basic-statistics-in-r.html#cb259-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(some_of_my_cars)</span>
<span id="cb259-3"><a href="basic-statistics-in-r.html#cb259-3" aria-hidden="true" tabindex="-1"></a><span class="co">#    Origin Manufacturer Price Passengers MPG.city EngineSize</span></span>
<span id="cb259-4"><a href="basic-statistics-in-r.html#cb259-4" aria-hidden="true" tabindex="-1"></a><span class="co">#1  non-USA        Acura  15.9          5       25        1.8</span></span>
<span id="cb259-5"><a href="basic-statistics-in-r.html#cb259-5" aria-hidden="true" tabindex="-1"></a><span class="co">#2  non-USA        Acura  33.9          5       18        3.2</span></span>
<span id="cb259-6"><a href="basic-statistics-in-r.html#cb259-6" aria-hidden="true" tabindex="-1"></a><span class="co">#3  non-USA         Audi  29.1          5       20        2.8</span></span>
<span id="cb259-7"><a href="basic-statistics-in-r.html#cb259-7" aria-hidden="true" tabindex="-1"></a><span class="co">#4  non-USA         Audi  37.7          6       19        2.8</span></span>
<span id="cb259-8"><a href="basic-statistics-in-r.html#cb259-8" aria-hidden="true" tabindex="-1"></a><span class="co">#5  non-USA          BMW  30.0          4       22        3.5</span></span>
<span id="cb259-9"><a href="basic-statistics-in-r.html#cb259-9" aria-hidden="true" tabindex="-1"></a><span class="co">#39 non-USA          Geo   8.4          4       46        1.0</span></span></code></pre></div>
</div>
<div id="transposing-rotating-your-dataset" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Transposing (rotating) your dataset</h2>
<p>Some datasets might have the “wrong” orientation (e.g., individuals as columns and variables as rows). This usually depends on the source data and for most purposes, you migth want your individuals as rows and each column with the variable measurements. We can correct this “wrong” orientation by transposing your data frame or matrix.</p>
<p>Here is an example using <code>deaths</code> dataset from the R-package <code>MASS</code>. This is a time series giving the monthly deaths from bronchitis, emphysema and asthma in the UK, 1974-1979, both sexes (deaths). I retyped the data into a data frame to make it easy to transpose (rotate) this dataset.</p>
<div class="sourceCode" id="cb260"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb260-1"><a href="basic-statistics-in-r.html#cb260-1" aria-hidden="true" tabindex="-1"></a><span class="do">## this is time series dataset as a data frame</span></span>
<span id="cb260-2"><a href="basic-statistics-in-r.html#cb260-2" aria-hidden="true" tabindex="-1"></a>UK_deaths <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">year_1974=</span><span class="fu">c</span>(<span class="dv">3035</span>,<span class="dv">2552</span>,<span class="dv">2704</span>,<span class="dv">2554</span>,<span class="dv">2014</span>,<span class="dv">1655</span>,<span class="dv">1721</span>,<span class="dv">1524</span>,<span class="dv">1596</span>,<span class="dv">2074</span>,<span class="dv">2199</span>,<span class="dv">2512</span>),</span>
<span id="cb260-3"><a href="basic-statistics-in-r.html#cb260-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">year_1975 =</span> <span class="fu">c</span>(<span class="dv">2933</span>,<span class="dv">2889</span>,<span class="dv">2938</span>,<span class="dv">2497</span>,<span class="dv">1870</span>,<span class="dv">1726</span>,<span class="dv">1607</span>,<span class="dv">1545</span>,<span class="dv">1396</span>,<span class="dv">1787</span>,<span class="dv">2076</span>,<span class="dv">2837</span>),</span>
<span id="cb260-4"><a href="basic-statistics-in-r.html#cb260-4" aria-hidden="true" tabindex="-1"></a>                        <span class="at">year_1976 =</span> <span class="fu">c</span>(<span class="dv">2787</span>,<span class="dv">3891</span>,<span class="dv">3179</span>,<span class="dv">2011</span>,<span class="dv">1636</span>,<span class="dv">1580</span>,<span class="dv">1489</span>,<span class="dv">1300</span>,<span class="dv">1356</span>,<span class="dv">1653</span>,<span class="dv">2013</span>,<span class="dv">2823</span>),</span>
<span id="cb260-5"><a href="basic-statistics-in-r.html#cb260-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">year_1977 =</span> <span class="fu">c</span>(<span class="dv">3102</span>,<span class="dv">2294</span>,<span class="dv">2385</span>,<span class="dv">2444</span>,<span class="dv">1748</span>,<span class="dv">1554</span>,<span class="dv">1498</span>,<span class="dv">1361</span>,<span class="dv">1346</span>,<span class="dv">1564</span>,<span class="dv">1640</span>,<span class="dv">2293</span>),</span>
<span id="cb260-6"><a href="basic-statistics-in-r.html#cb260-6" aria-hidden="true" tabindex="-1"></a>                        <span class="at">year_1978 =</span> <span class="fu">c</span>(<span class="dv">2815</span>,<span class="dv">3137</span>,<span class="dv">2679</span>,<span class="dv">1969</span>,<span class="dv">1870</span>,<span class="dv">1633</span>,<span class="dv">1529</span>,<span class="dv">1366</span>,<span class="dv">1357</span>,<span class="dv">1570</span>,<span class="dv">1535</span>,<span class="dv">2491</span>),</span>
<span id="cb260-7"><a href="basic-statistics-in-r.html#cb260-7" aria-hidden="true" tabindex="-1"></a>                        <span class="at">year_1979 =</span> <span class="fu">c</span>(<span class="dv">3084</span>,<span class="dv">2605</span>,<span class="dv">2573</span>,<span class="dv">2143</span>,<span class="dv">1693</span>,<span class="dv">1504</span>,<span class="dv">1461</span>,<span class="dv">1354</span>,<span class="dv">1333</span>,<span class="dv">1492</span>,<span class="dv">1781</span>,<span class="dv">1915</span>), <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb260-8"><a href="basic-statistics-in-r.html#cb260-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(UK_deaths) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Jan&quot;</span>,<span class="st">&quot;Feb&quot;</span>,<span class="st">&quot;Mar&quot;</span>,<span class="st">&quot;Apr&quot;</span>,<span class="st">&quot;May&quot;</span>,<span class="st">&quot;Jun&quot;</span>,<span class="st">&quot;Jul&quot;</span>,<span class="st">&quot;Aug&quot;</span>,<span class="st">&quot;Sep&quot;</span>,<span class="st">&quot;Oct&quot;</span>,<span class="st">&quot;Nov&quot;</span>,<span class="st">&quot;Dec&quot;</span>)</span>
<span id="cb260-9"><a href="basic-statistics-in-r.html#cb260-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb260-10"><a href="basic-statistics-in-r.html#cb260-10" aria-hidden="true" tabindex="-1"></a><span class="do">## this dataset has months as rows and years as columns</span></span>
<span id="cb260-11"><a href="basic-statistics-in-r.html#cb260-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb260-12"><a href="basic-statistics-in-r.html#cb260-12" aria-hidden="true" tabindex="-1"></a>UK_deaths</span>
<span id="cb260-13"><a href="basic-statistics-in-r.html#cb260-13" aria-hidden="true" tabindex="-1"></a><span class="co">#    year_1974 year_1975 year_1976 year_1977 year_1978 year_1979</span></span>
<span id="cb260-14"><a href="basic-statistics-in-r.html#cb260-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Jan      3035      2933      2787      3102      2815      3084</span></span>
<span id="cb260-15"><a href="basic-statistics-in-r.html#cb260-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Feb      2552      2889      3891      2294      3137      2605</span></span>
<span id="cb260-16"><a href="basic-statistics-in-r.html#cb260-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Mar      2704      2938      3179      2385      2679      2573</span></span>
<span id="cb260-17"><a href="basic-statistics-in-r.html#cb260-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Apr      2554      2497      2011      2444      1969      2143</span></span>
<span id="cb260-18"><a href="basic-statistics-in-r.html#cb260-18" aria-hidden="true" tabindex="-1"></a><span class="co"># May      2014      1870      1636      1748      1870      1693</span></span>
<span id="cb260-19"><a href="basic-statistics-in-r.html#cb260-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Jun      1655      1726      1580      1554      1633      1504</span></span>
<span id="cb260-20"><a href="basic-statistics-in-r.html#cb260-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Jul      1721      1607      1489      1498      1529      1461</span></span>
<span id="cb260-21"><a href="basic-statistics-in-r.html#cb260-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Aug      1524      1545      1300      1361      1366      1354</span></span>
<span id="cb260-22"><a href="basic-statistics-in-r.html#cb260-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Sep      1596      1396      1356      1346      1357      1333</span></span>
<span id="cb260-23"><a href="basic-statistics-in-r.html#cb260-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Oct      2074      1787      1653      1564      1570      1492</span></span>
<span id="cb260-24"><a href="basic-statistics-in-r.html#cb260-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Nov      2199      2076      2013      1640      1535      1781</span></span>
<span id="cb260-25"><a href="basic-statistics-in-r.html#cb260-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Dec      2512      2837      2823      2293      2491      1915</span></span>
<span id="cb260-26"><a href="basic-statistics-in-r.html#cb260-26" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(UK_deaths)</span>
<span id="cb260-27"><a href="basic-statistics-in-r.html#cb260-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  12 obs. of  6 variables:</span></span>
<span id="cb260-28"><a href="basic-statistics-in-r.html#cb260-28" aria-hidden="true" tabindex="-1"></a><span class="co"># $ year_1974: num  3035 2552 2704 2554 2014 ...</span></span>
<span id="cb260-29"><a href="basic-statistics-in-r.html#cb260-29" aria-hidden="true" tabindex="-1"></a><span class="co"># $ year_1975: num  2933 2889 2938 2497 1870 ...</span></span>
<span id="cb260-30"><a href="basic-statistics-in-r.html#cb260-30" aria-hidden="true" tabindex="-1"></a><span class="co"># $ year_1976: num  2787 3891 3179 2011 1636 ...</span></span>
<span id="cb260-31"><a href="basic-statistics-in-r.html#cb260-31" aria-hidden="true" tabindex="-1"></a><span class="co"># $ year_1977: num  3102 2294 2385 2444 1748 ...</span></span>
<span id="cb260-32"><a href="basic-statistics-in-r.html#cb260-32" aria-hidden="true" tabindex="-1"></a><span class="co"># $ year_1978: num  2815 3137 2679 1969 1870 ...</span></span>
<span id="cb260-33"><a href="basic-statistics-in-r.html#cb260-33" aria-hidden="true" tabindex="-1"></a><span class="co"># $ year_1979: num  3084 2605 2573 2143 1693 ..</span></span>
<span id="cb260-34"><a href="basic-statistics-in-r.html#cb260-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb260-35"><a href="basic-statistics-in-r.html#cb260-35" aria-hidden="true" tabindex="-1"></a><span class="do">## we can transpose this UK_deaths to have months as columns</span></span>
<span id="cb260-36"><a href="basic-statistics-in-r.html#cb260-36" aria-hidden="true" tabindex="-1"></a>UK_deaths_t <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(UK_deaths),<span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb260-37"><a href="basic-statistics-in-r.html#cb260-37" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(UK_deaths_t)</span>
<span id="cb260-38"><a href="basic-statistics-in-r.html#cb260-38" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;data.frame&quot;</span></span>
<span id="cb260-39"><a href="basic-statistics-in-r.html#cb260-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb260-40"><a href="basic-statistics-in-r.html#cb260-40" aria-hidden="true" tabindex="-1"></a>UK_deaths_t</span>
<span id="cb260-41"><a href="basic-statistics-in-r.html#cb260-41" aria-hidden="true" tabindex="-1"></a><span class="co">#           Jan  Feb  Mar  Apr  May  Jun  Jul  Aug  Sep  Oct  Nov  Dec</span></span>
<span id="cb260-42"><a href="basic-statistics-in-r.html#cb260-42" aria-hidden="true" tabindex="-1"></a><span class="co"># year_1974 3035 2552 2704 2554 2014 1655 1721 1524 1596 2074 2199 2512</span></span>
<span id="cb260-43"><a href="basic-statistics-in-r.html#cb260-43" aria-hidden="true" tabindex="-1"></a><span class="co"># year_1975 2933 2889 2938 2497 1870 1726 1607 1545 1396 1787 2076 2837</span></span>
<span id="cb260-44"><a href="basic-statistics-in-r.html#cb260-44" aria-hidden="true" tabindex="-1"></a><span class="co"># year_1976 2787 3891 3179 2011 1636 1580 1489 1300 1356 1653 2013 2823</span></span>
<span id="cb260-45"><a href="basic-statistics-in-r.html#cb260-45" aria-hidden="true" tabindex="-1"></a><span class="co"># year_1977 3102 2294 2385 2444 1748 1554 1498 1361 1346 1564 1640 2293</span></span>
<span id="cb260-46"><a href="basic-statistics-in-r.html#cb260-46" aria-hidden="true" tabindex="-1"></a><span class="co"># year_1978 2815 3137 2679 1969 1870 1633 1529 1366 1357 1570 1535 2491</span></span>
<span id="cb260-47"><a href="basic-statistics-in-r.html#cb260-47" aria-hidden="true" tabindex="-1"></a><span class="co"># year_1979 3084 2605 2573 2143 1693 1504 1461 1354 1333 1492 1781 1915</span></span></code></pre></div>
</div>
<div id="dealing-with-missing-na-values" class="section level2" number="6.4">
<h2><span class="header-section-number">6.4</span> Dealing with missing (NA) values</h2>
<p>Most datasets would have some missing values usually coded as <strong>NA</strong>, <strong>&lt;NaN</strong> or <strong><NA></strong>. Such missing
values tend to be problematic for many statistical analyses and how are they treated will influence the results and inferences.</p>
<p>We will work with some build and complete datasets. Then, we will introduce missing values.</p>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="basic-statistics-in-r.html#cb261-1" aria-hidden="true" tabindex="-1"></a><span class="do">## the mtcars is comparisons between 32 automobiles (1973–74 models)</span></span>
<span id="cb261-2"><a href="basic-statistics-in-r.html#cb261-2" aria-hidden="true" tabindex="-1"></a>my_mtcars <span class="ot">&lt;-</span> mtcars</span>
<span id="cb261-3"><a href="basic-statistics-in-r.html#cb261-3" aria-hidden="true" tabindex="-1"></a>my_mtcars<span class="sc">$</span>car_name <span class="ot">&lt;-</span> <span class="fu">rownames</span>(my_mtcars)</span>
<span id="cb261-4"><a href="basic-statistics-in-r.html#cb261-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(my_mtcars) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb261-5"><a href="basic-statistics-in-r.html#cb261-5" aria-hidden="true" tabindex="-1"></a>my_mtcars</span>
<span id="cb261-6"><a href="basic-statistics-in-r.html#cb261-6" aria-hidden="true" tabindex="-1"></a><span class="co">#    mpg cyl  disp  hp drat    wt  qsec vs am gear carb            car_name</span></span>
<span id="cb261-7"><a href="basic-statistics-in-r.html#cb261-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 1  21.0   6 160.0 110 3.90 2.620 16.46  0  1    4    4           Mazda RX4</span></span>
<span id="cb261-8"><a href="basic-statistics-in-r.html#cb261-8" aria-hidden="true" tabindex="-1"></a><span class="co"># 2  21.0   6 160.0 110 3.90 2.875 17.02  0  1    4    4       Mazda RX4 Wag</span></span>
<span id="cb261-9"><a href="basic-statistics-in-r.html#cb261-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 3  22.8   4 108.0  93 3.85 2.320 18.61  1  1    4    1          Datsun 710</span></span>
<span id="cb261-10"><a href="basic-statistics-in-r.html#cb261-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 4  21.4   6 258.0 110 3.08 3.215 19.44  1  0    3    1      Hornet 4 Drive</span></span>
<span id="cb261-11"><a href="basic-statistics-in-r.html#cb261-11" aria-hidden="true" tabindex="-1"></a><span class="co"># 5  18.7   8 360.0 175 3.15 3.440 17.02  0  0    3    2   Hornet Sportabout</span></span>
<span id="cb261-12"><a href="basic-statistics-in-r.html#cb261-12" aria-hidden="true" tabindex="-1"></a><span class="co"># 6  18.1   6 225.0 105 2.76 3.460 20.22  1  0    3    1             Valiant</span></span>
<span id="cb261-13"><a href="basic-statistics-in-r.html#cb261-13" aria-hidden="true" tabindex="-1"></a><span class="co"># 7  14.3   8 360.0 245 3.21 3.570 15.84  0  0    3    4          Duster 360</span></span>
<span id="cb261-14"><a href="basic-statistics-in-r.html#cb261-14" aria-hidden="true" tabindex="-1"></a><span class="co"># 8  24.4   4 146.7  62 3.69 3.190 20.00  1  0    4    2           Merc 240D</span></span>
<span id="cb261-15"><a href="basic-statistics-in-r.html#cb261-15" aria-hidden="true" tabindex="-1"></a><span class="co"># 9  22.8   4 140.8  95 3.92 3.150 22.90  1  0    4    2            Merc 230</span></span>
<span id="cb261-16"><a href="basic-statistics-in-r.html#cb261-16" aria-hidden="true" tabindex="-1"></a><span class="co"># 10 19.2   6 167.6 123 3.92 3.440 18.30  1  0    4    4            Merc 280</span></span>
<span id="cb261-17"><a href="basic-statistics-in-r.html#cb261-17" aria-hidden="true" tabindex="-1"></a><span class="co"># 11 17.8   6 167.6 123 3.92 3.440 18.90  1  0    4    4           Merc 280C</span></span>
<span id="cb261-18"><a href="basic-statistics-in-r.html#cb261-18" aria-hidden="true" tabindex="-1"></a><span class="co"># 12 16.4   8 275.8 180 3.07 4.070 17.40  0  0    3    3          Merc 450SE</span></span>
<span id="cb261-19"><a href="basic-statistics-in-r.html#cb261-19" aria-hidden="true" tabindex="-1"></a><span class="co"># 13 17.3   8 275.8 180 3.07 3.730 17.60  0  0    3    3          Merc 450SL</span></span>
<span id="cb261-20"><a href="basic-statistics-in-r.html#cb261-20" aria-hidden="true" tabindex="-1"></a><span class="co"># 14 15.2   8 275.8 180 3.07 3.780 18.00  0  0    3    3         Merc 450SLC</span></span>
<span id="cb261-21"><a href="basic-statistics-in-r.html#cb261-21" aria-hidden="true" tabindex="-1"></a><span class="co"># 15 10.4   8 472.0 205 2.93 5.250 17.98  0  0    3    4  Cadillac Fleetwood</span></span>
<span id="cb261-22"><a href="basic-statistics-in-r.html#cb261-22" aria-hidden="true" tabindex="-1"></a><span class="co"># 16 10.4   8 460.0 215 3.00 5.424 17.82  0  0    3    4 Lincoln Continental</span></span>
<span id="cb261-23"><a href="basic-statistics-in-r.html#cb261-23" aria-hidden="true" tabindex="-1"></a><span class="co"># 17 14.7   8 440.0 230 3.23 5.345 17.42  0  0    3    4   Chrysler Imperial</span></span>
<span id="cb261-24"><a href="basic-statistics-in-r.html#cb261-24" aria-hidden="true" tabindex="-1"></a><span class="co"># 18 32.4   4  78.7  66 4.08 2.200 19.47  1  1    4    1            Fiat 128</span></span>
<span id="cb261-25"><a href="basic-statistics-in-r.html#cb261-25" aria-hidden="true" tabindex="-1"></a><span class="co"># 19 30.4   4  75.7  52 4.93 1.615 18.52  1  1    4    2         Honda Civic</span></span>
<span id="cb261-26"><a href="basic-statistics-in-r.html#cb261-26" aria-hidden="true" tabindex="-1"></a><span class="co"># 20 33.9   4  71.1  65 4.22 1.835 19.90  1  1    4    1      Toyota Corolla</span></span>
<span id="cb261-27"><a href="basic-statistics-in-r.html#cb261-27" aria-hidden="true" tabindex="-1"></a><span class="co"># 21 21.5   4 120.1  97 3.70 2.465 20.01  1  0    3    1       Toyota Corona</span></span>
<span id="cb261-28"><a href="basic-statistics-in-r.html#cb261-28" aria-hidden="true" tabindex="-1"></a><span class="co"># 22 15.5   8 318.0 150 2.76 3.520 16.87  0  0    3    2    Dodge Challenger</span></span>
<span id="cb261-29"><a href="basic-statistics-in-r.html#cb261-29" aria-hidden="true" tabindex="-1"></a><span class="co"># 23 15.2   8 304.0 150 3.15 3.435 17.30  0  0    3    2         AMC Javelin</span></span>
<span id="cb261-30"><a href="basic-statistics-in-r.html#cb261-30" aria-hidden="true" tabindex="-1"></a><span class="co"># 24 13.3   8 350.0 245 3.73 3.840 15.41  0  0    3    4          Camaro Z28</span></span>
<span id="cb261-31"><a href="basic-statistics-in-r.html#cb261-31" aria-hidden="true" tabindex="-1"></a><span class="co"># 25 19.2   8 400.0 175 3.08 3.845 17.05  0  0    3    2    Pontiac Firebird</span></span>
<span id="cb261-32"><a href="basic-statistics-in-r.html#cb261-32" aria-hidden="true" tabindex="-1"></a><span class="co"># 26 27.3   4  79.0  66 4.08 1.935 18.90  1  1    4    1           Fiat X1-9</span></span>
<span id="cb261-33"><a href="basic-statistics-in-r.html#cb261-33" aria-hidden="true" tabindex="-1"></a><span class="co"># 27 26.0   4 120.3  91 4.43 2.140 16.70  0  1    5    2       Porsche 914-2</span></span>
<span id="cb261-34"><a href="basic-statistics-in-r.html#cb261-34" aria-hidden="true" tabindex="-1"></a><span class="co"># 28 30.4   4  95.1 113 3.77 1.513 16.90  1  1    5    2        Lotus Europa</span></span>
<span id="cb261-35"><a href="basic-statistics-in-r.html#cb261-35" aria-hidden="true" tabindex="-1"></a><span class="co"># 29 15.8   8 351.0 264 4.22 3.170 14.50  0  1    5    4      Ford Pantera L</span></span>
<span id="cb261-36"><a href="basic-statistics-in-r.html#cb261-36" aria-hidden="true" tabindex="-1"></a><span class="co"># 30 19.7   6 145.0 175 3.62 2.770 15.50  0  1    5    6        Ferrari Dino</span></span>
<span id="cb261-37"><a href="basic-statistics-in-r.html#cb261-37" aria-hidden="true" tabindex="-1"></a><span class="co"># 31 15.0   8 301.0 335 3.54 3.570 14.60  0  1    5    8       Maserati Bora</span></span>
<span id="cb261-38"><a href="basic-statistics-in-r.html#cb261-38" aria-hidden="true" tabindex="-1"></a><span class="co"># 32 21.4   4 121.0 109 4.11 2.780 18.60  1  1    4    2          Volvo 142E</span></span>
<span id="cb261-39"><a href="basic-statistics-in-r.html#cb261-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb261-40"><a href="basic-statistics-in-r.html#cb261-40" aria-hidden="true" tabindex="-1"></a><span class="do">## Lets introduce some NA values randomly</span></span>
<span id="cb261-41"><a href="basic-statistics-in-r.html#cb261-41" aria-hidden="true" tabindex="-1"></a>my_mtcars_with_NA <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">lapply</span>(my_mtcars, <span class="cf">function</span>(x_df) x_df[ <span class="fu">sample</span>(<span class="fu">c</span>(<span class="cn">TRUE</span>, <span class="cn">NA</span>), <span class="at">prob =</span> <span class="fu">c</span>(<span class="fl">0.85</span>, <span class="fl">0.15</span>), <span class="at">size =</span> <span class="fu">length</span>(x_df), <span class="at">replace =</span> <span class="cn">TRUE</span>) ]))</span>
<span id="cb261-42"><a href="basic-statistics-in-r.html#cb261-42" aria-hidden="true" tabindex="-1"></a>my_mtcars_with_NA</span>
<span id="cb261-43"><a href="basic-statistics-in-r.html#cb261-43" aria-hidden="true" tabindex="-1"></a><span class="co">#    mpg cyl  disp  hp drat    wt  qsec vs am gear carb            car_name</span></span>
<span id="cb261-44"><a href="basic-statistics-in-r.html#cb261-44" aria-hidden="true" tabindex="-1"></a><span class="co"># 1  21.0   6 160.0 110 3.90 2.620 16.46  0  1    4    4                &lt;NA&gt;</span></span>
<span id="cb261-45"><a href="basic-statistics-in-r.html#cb261-45" aria-hidden="true" tabindex="-1"></a><span class="co"># 2  21.0   6 160.0  NA 3.90 2.875 17.02  0 NA    4   NA                &lt;NA&gt;</span></span>
<span id="cb261-46"><a href="basic-statistics-in-r.html#cb261-46" aria-hidden="true" tabindex="-1"></a><span class="co"># 3  22.8   4 108.0  93 3.85 2.320    NA NA  1    4    1          Datsun 710</span></span>
<span id="cb261-47"><a href="basic-statistics-in-r.html#cb261-47" aria-hidden="true" tabindex="-1"></a><span class="co"># 4    NA   6 258.0 110 3.08 3.215    NA  1  0    3    1      Hornet 4 Drive</span></span>
<span id="cb261-48"><a href="basic-statistics-in-r.html#cb261-48" aria-hidden="true" tabindex="-1"></a><span class="co"># 5  18.7   8 360.0 175 3.15 3.440 17.02  0  0    3    2   Hornet Sportabout</span></span>
<span id="cb261-49"><a href="basic-statistics-in-r.html#cb261-49" aria-hidden="true" tabindex="-1"></a><span class="co"># 6  18.1   6 225.0 105 2.76 3.460 20.22  1 NA    3   NA             Valiant</span></span>
<span id="cb261-50"><a href="basic-statistics-in-r.html#cb261-50" aria-hidden="true" tabindex="-1"></a><span class="co"># 7  14.3   8 360.0  NA 3.21 3.570    NA  0  0    3   NA          Duster 360</span></span>
<span id="cb261-51"><a href="basic-statistics-in-r.html#cb261-51" aria-hidden="true" tabindex="-1"></a><span class="co"># 8  24.4   4 146.7  62   NA 3.190 20.00  1  0    4    2           Merc 240D</span></span>
<span id="cb261-52"><a href="basic-statistics-in-r.html#cb261-52" aria-hidden="true" tabindex="-1"></a><span class="co"># 9  22.8   4    NA  95   NA 3.150 22.90  1  0    4   NA            Merc 230</span></span>
<span id="cb261-53"><a href="basic-statistics-in-r.html#cb261-53" aria-hidden="true" tabindex="-1"></a><span class="co"># 10 19.2   6 167.6 123 3.92 3.440 18.30  1  0   NA   NA                &lt;NA&gt;</span></span>
<span id="cb261-54"><a href="basic-statistics-in-r.html#cb261-54" aria-hidden="true" tabindex="-1"></a><span class="co"># 11 17.8   6 167.6 123 3.92 3.440 18.90  1  0    4   NA           Merc 280C</span></span>
<span id="cb261-55"><a href="basic-statistics-in-r.html#cb261-55" aria-hidden="true" tabindex="-1"></a><span class="co"># 12 16.4   8 275.8  NA 3.07 4.070 17.40  0  0    3    3          Merc 450SE</span></span>
<span id="cb261-56"><a href="basic-statistics-in-r.html#cb261-56" aria-hidden="true" tabindex="-1"></a><span class="co"># 13 17.3   8 275.8 180 3.07 3.730 17.60  0  0    3    3          Merc 450SL</span></span>
<span id="cb261-57"><a href="basic-statistics-in-r.html#cb261-57" aria-hidden="true" tabindex="-1"></a><span class="co"># 14 15.2   8    NA 180 3.07 3.780    NA  0  0    3    3                &lt;NA&gt;</span></span>
<span id="cb261-58"><a href="basic-statistics-in-r.html#cb261-58" aria-hidden="true" tabindex="-1"></a><span class="co"># 15 10.4  NA 472.0 205 2.93    NA 17.98  0  0    3    4  Cadillac Fleetwood</span></span>
<span id="cb261-59"><a href="basic-statistics-in-r.html#cb261-59" aria-hidden="true" tabindex="-1"></a><span class="co"># 16 10.4  NA 460.0 215 3.00 5.424 17.82  0 NA    3    4 Lincoln Continental</span></span>
<span id="cb261-60"><a href="basic-statistics-in-r.html#cb261-60" aria-hidden="true" tabindex="-1"></a><span class="co"># 17 14.7   8 440.0  NA 3.23 5.345    NA  0  0    3    4   Chrysler Imperial</span></span>
<span id="cb261-61"><a href="basic-statistics-in-r.html#cb261-61" aria-hidden="true" tabindex="-1"></a><span class="co"># 18 32.4   4  78.7  66   NA 2.200 19.47  1  1   NA    1            Fiat 128</span></span>
<span id="cb261-62"><a href="basic-statistics-in-r.html#cb261-62" aria-hidden="true" tabindex="-1"></a><span class="co"># 19 30.4   4  75.7  52 4.93 1.615 18.52  1  1    4    2         Honda Civic</span></span>
<span id="cb261-63"><a href="basic-statistics-in-r.html#cb261-63" aria-hidden="true" tabindex="-1"></a><span class="co"># 20 33.9   4  71.1  NA 4.22 1.835 19.90 NA NA    4    1      Toyota Corolla</span></span>
<span id="cb261-64"><a href="basic-statistics-in-r.html#cb261-64" aria-hidden="true" tabindex="-1"></a><span class="co"># 21 21.5   4 120.1  97 3.70 2.465 20.01  1  0    3    1       Toyota Corona</span></span>
<span id="cb261-65"><a href="basic-statistics-in-r.html#cb261-65" aria-hidden="true" tabindex="-1"></a><span class="co"># 22 15.5   8 318.0 150 2.76 3.520 16.87  0  0    3    2                &lt;NA&gt;</span></span>
<span id="cb261-66"><a href="basic-statistics-in-r.html#cb261-66" aria-hidden="true" tabindex="-1"></a><span class="co"># 23 15.2   8    NA 150 3.15    NA 17.30  0  0    3    2         AMC Javelin</span></span>
<span id="cb261-67"><a href="basic-statistics-in-r.html#cb261-67" aria-hidden="true" tabindex="-1"></a><span class="co"># 24   NA   8    NA 245 3.73    NA 15.41  0 NA    3    4          Camaro Z28</span></span>
<span id="cb261-68"><a href="basic-statistics-in-r.html#cb261-68" aria-hidden="true" tabindex="-1"></a><span class="co"># 25   NA   8 400.0 175   NA 3.845 17.05  0  0    3    2    Pontiac Firebird</span></span>
<span id="cb261-69"><a href="basic-statistics-in-r.html#cb261-69" aria-hidden="true" tabindex="-1"></a><span class="co"># 26   NA   4  79.0  66 4.08 1.935 18.90  1  1   NA    1           Fiat X1-9</span></span>
<span id="cb261-70"><a href="basic-statistics-in-r.html#cb261-70" aria-hidden="true" tabindex="-1"></a><span class="co"># 27 26.0   4 120.3  91 4.43    NA 16.70  0 NA    5   NA       Porsche 914-2</span></span>
<span id="cb261-71"><a href="basic-statistics-in-r.html#cb261-71" aria-hidden="true" tabindex="-1"></a><span class="co"># 28 30.4   4  95.1 113 3.77 1.513 16.90  1  1    5   NA        Lotus Europa</span></span>
<span id="cb261-72"><a href="basic-statistics-in-r.html#cb261-72" aria-hidden="true" tabindex="-1"></a><span class="co"># 29 15.8   8 351.0 264 4.22 3.170 14.50  0  1    5    4      Ford Pantera L</span></span>
<span id="cb261-73"><a href="basic-statistics-in-r.html#cb261-73" aria-hidden="true" tabindex="-1"></a><span class="co"># 30   NA   6 145.0 175 3.62    NA 15.50  0  1    5    6                &lt;NA&gt;</span></span>
<span id="cb261-74"><a href="basic-statistics-in-r.html#cb261-74" aria-hidden="true" tabindex="-1"></a><span class="co"># 31 15.0   8 301.0 335 3.54 3.570 14.60  0  1    5    8       Maserati Bora</span></span>
<span id="cb261-75"><a href="basic-statistics-in-r.html#cb261-75" aria-hidden="true" tabindex="-1"></a><span class="co"># 32 21.4   4 121.0 109 4.11 2.780 18.60  1  1   NA   NA          Volvo 142E</span></span></code></pre></div>
<p>You can compare these two datasets and notice several cells with values as <strong>NA</strong> (if variable is numeric) and <strong><NA></strong> (if variable is discrete).</p>
<p>We can remove all rows with at least one <strong>NA</strong> value using the function <code>complete.cases()</code>.</p>
<div class="sourceCode" id="cb262"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb262-1"><a href="basic-statistics-in-r.html#cb262-1" aria-hidden="true" tabindex="-1"></a><span class="do">## remove all rows with at least on NA</span></span>
<span id="cb262-2"><a href="basic-statistics-in-r.html#cb262-2" aria-hidden="true" tabindex="-1"></a>my_mtcars_complete_rows <span class="ot">&lt;-</span> my_mtcars_with_NA[<span class="fu">complete.cases</span>(my_mtcars_with_NA), ]</span>
<span id="cb262-3"><a href="basic-statistics-in-r.html#cb262-3" aria-hidden="true" tabindex="-1"></a>my_mtcars_complete_rows</span>
<span id="cb262-4"><a href="basic-statistics-in-r.html#cb262-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     mpg cyl  disp  hp drat    wt  qsec vs am gear carb          car_name</span></span>
<span id="cb262-5"><a href="basic-statistics-in-r.html#cb262-5" aria-hidden="true" tabindex="-1"></a><span class="co"># 5  18.7   8 360.0 175 3.15 3.440 17.02  0  0    3    2 Hornet Sportabout</span></span>
<span id="cb262-6"><a href="basic-statistics-in-r.html#cb262-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 13 17.3   8 275.8 180 3.07 3.730 17.60  0  0    3    3        Merc 450SL</span></span>
<span id="cb262-7"><a href="basic-statistics-in-r.html#cb262-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 19 30.4   4  75.7  52 4.93 1.615 18.52  1  1    4    2       Honda Civic</span></span>
<span id="cb262-8"><a href="basic-statistics-in-r.html#cb262-8" aria-hidden="true" tabindex="-1"></a><span class="co"># 21 21.5   4 120.1  97 3.70 2.465 20.01  1  0    3    1     Toyota Corona</span></span>
<span id="cb262-9"><a href="basic-statistics-in-r.html#cb262-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 29 15.8   8 351.0 264 4.22 3.170 14.50  0  1    5    4    Ford Pantera L</span></span>
<span id="cb262-10"><a href="basic-statistics-in-r.html#cb262-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 31 15.0   8 301.0 335 3.54 3.570 14.60  0  1    5    8     Maserati Bora</span></span></code></pre></div>
<p>We can remove rows with at least one <strong>NA</strong> value on specific columns using the function <code>complete.cases()</code>.</p>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="basic-statistics-in-r.html#cb263-1" aria-hidden="true" tabindex="-1"></a><span class="do">## remove rows with at least one NA in columns &quot;mpg&quot; and &quot;cyl&quot;</span></span>
<span id="cb263-2"><a href="basic-statistics-in-r.html#cb263-2" aria-hidden="true" tabindex="-1"></a>my_mtcars_complete_rows_mpg_cyl <span class="ot">&lt;-</span> my_mtcars_with_NA[<span class="fu">complete.cases</span>(my_mtcars_with_NA[,<span class="fu">c</span>(<span class="st">&quot;mpg&quot;</span>,<span class="st">&quot;cyl&quot;</span>)]), ]</span>
<span id="cb263-3"><a href="basic-statistics-in-r.html#cb263-3" aria-hidden="true" tabindex="-1"></a>my_mtcars_complete_rows_mpg_cyl</span>
<span id="cb263-4"><a href="basic-statistics-in-r.html#cb263-4" aria-hidden="true" tabindex="-1"></a><span class="co">#    mpg cyl  disp  hp drat    wt  qsec vs am gear carb          car_name</span></span>
<span id="cb263-5"><a href="basic-statistics-in-r.html#cb263-5" aria-hidden="true" tabindex="-1"></a><span class="co"># 1  21.0   6 160.0 110 3.90 2.620 16.46  0  1    4    4              &lt;NA&gt;</span></span>
<span id="cb263-6"><a href="basic-statistics-in-r.html#cb263-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 2  21.0   6 160.0  NA 3.90 2.875 17.02  0 NA    4   NA              &lt;NA&gt;</span></span>
<span id="cb263-7"><a href="basic-statistics-in-r.html#cb263-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 3  22.8   4 108.0  93 3.85 2.320    NA NA  1    4    1        Datsun 710</span></span>
<span id="cb263-8"><a href="basic-statistics-in-r.html#cb263-8" aria-hidden="true" tabindex="-1"></a><span class="co"># 5  18.7   8 360.0 175 3.15 3.440 17.02  0  0    3    2 Hornet Sportabout</span></span>
<span id="cb263-9"><a href="basic-statistics-in-r.html#cb263-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 6  18.1   6 225.0 105 2.76 3.460 20.22  1 NA    3   NA           Valiant</span></span>
<span id="cb263-10"><a href="basic-statistics-in-r.html#cb263-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 7  14.3   8 360.0  NA 3.21 3.570    NA  0  0    3   NA        Duster 360</span></span>
<span id="cb263-11"><a href="basic-statistics-in-r.html#cb263-11" aria-hidden="true" tabindex="-1"></a><span class="co"># 8  24.4   4 146.7  62   NA 3.190 20.00  1  0    4    2         Merc 240D</span></span>
<span id="cb263-12"><a href="basic-statistics-in-r.html#cb263-12" aria-hidden="true" tabindex="-1"></a><span class="co"># 9  22.8   4    NA  95   NA 3.150 22.90  1  0    4   NA          Merc 230</span></span>
<span id="cb263-13"><a href="basic-statistics-in-r.html#cb263-13" aria-hidden="true" tabindex="-1"></a><span class="co"># 10 19.2   6 167.6 123 3.92 3.440 18.30  1  0   NA   NA              &lt;NA&gt;</span></span>
<span id="cb263-14"><a href="basic-statistics-in-r.html#cb263-14" aria-hidden="true" tabindex="-1"></a><span class="co"># 11 17.8   6 167.6 123 3.92 3.440 18.90  1  0    4   NA         Merc 280C</span></span>
<span id="cb263-15"><a href="basic-statistics-in-r.html#cb263-15" aria-hidden="true" tabindex="-1"></a><span class="co"># 12 16.4   8 275.8  NA 3.07 4.070 17.40  0  0    3    3        Merc 450SE</span></span>
<span id="cb263-16"><a href="basic-statistics-in-r.html#cb263-16" aria-hidden="true" tabindex="-1"></a><span class="co"># 13 17.3   8 275.8 180 3.07 3.730 17.60  0  0    3    3        Merc 450SL</span></span>
<span id="cb263-17"><a href="basic-statistics-in-r.html#cb263-17" aria-hidden="true" tabindex="-1"></a><span class="co"># 14 15.2   8    NA 180 3.07 3.780    NA  0  0    3    3              &lt;NA&gt;</span></span>
<span id="cb263-18"><a href="basic-statistics-in-r.html#cb263-18" aria-hidden="true" tabindex="-1"></a><span class="co"># 17 14.7   8 440.0  NA 3.23 5.345    NA  0  0    3    4 Chrysler Imperial</span></span>
<span id="cb263-19"><a href="basic-statistics-in-r.html#cb263-19" aria-hidden="true" tabindex="-1"></a><span class="co"># 18 32.4   4  78.7  66   NA 2.200 19.47  1  1   NA    1          Fiat 128</span></span>
<span id="cb263-20"><a href="basic-statistics-in-r.html#cb263-20" aria-hidden="true" tabindex="-1"></a><span class="co"># 19 30.4   4  75.7  52 4.93 1.615 18.52  1  1    4    2       Honda Civic</span></span>
<span id="cb263-21"><a href="basic-statistics-in-r.html#cb263-21" aria-hidden="true" tabindex="-1"></a><span class="co"># 20 33.9   4  71.1  NA 4.22 1.835 19.90 NA NA    4    1    Toyota Corolla</span></span>
<span id="cb263-22"><a href="basic-statistics-in-r.html#cb263-22" aria-hidden="true" tabindex="-1"></a><span class="co"># 21 21.5   4 120.1  97 3.70 2.465 20.01  1  0    3    1     Toyota Corona</span></span>
<span id="cb263-23"><a href="basic-statistics-in-r.html#cb263-23" aria-hidden="true" tabindex="-1"></a><span class="co"># 22 15.5   8 318.0 150 2.76 3.520 16.87  0  0    3    2              &lt;NA&gt;</span></span>
<span id="cb263-24"><a href="basic-statistics-in-r.html#cb263-24" aria-hidden="true" tabindex="-1"></a><span class="co"># 23 15.2   8    NA 150 3.15    NA 17.30  0  0    3    2       AMC Javelin</span></span>
<span id="cb263-25"><a href="basic-statistics-in-r.html#cb263-25" aria-hidden="true" tabindex="-1"></a><span class="co"># 27 26.0   4 120.3  91 4.43    NA 16.70  0 NA    5   NA     Porsche 914-2</span></span>
<span id="cb263-26"><a href="basic-statistics-in-r.html#cb263-26" aria-hidden="true" tabindex="-1"></a><span class="co"># 28 30.4   4  95.1 113 3.77 1.513 16.90  1  1    5   NA      Lotus Europa</span></span>
<span id="cb263-27"><a href="basic-statistics-in-r.html#cb263-27" aria-hidden="true" tabindex="-1"></a><span class="co"># 29 15.8   8 351.0 264 4.22 3.170 14.50  0  1    5    4    Ford Pantera L</span></span>
<span id="cb263-28"><a href="basic-statistics-in-r.html#cb263-28" aria-hidden="true" tabindex="-1"></a><span class="co"># 31 15.0   8 301.0 335 3.54 3.570 14.60  0  1    5    8     Maserati Bora</span></span>
<span id="cb263-29"><a href="basic-statistics-in-r.html#cb263-29" aria-hidden="true" tabindex="-1"></a><span class="co"># 32 21.4   4 121.0 109 4.11 2.780 18.60  1  1   NA   NA        Volvo 142E</span></span></code></pre></div>
<p>Notice that columns called <code>mpg</code> and <code>cyl</code> have complete values a no missing <strong>NA</strong>.</p>
<p>You can also replace <strong>NA</strong> for specific values, for example as zero (0).</p>
<div class="sourceCode" id="cb264"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb264-1"><a href="basic-statistics-in-r.html#cb264-1" aria-hidden="true" tabindex="-1"></a><span class="do">## lets get the am variable as a vector</span></span>
<span id="cb264-2"><a href="basic-statistics-in-r.html#cb264-2" aria-hidden="true" tabindex="-1"></a>my_NA_am_variable <span class="ot">&lt;-</span> my_mtcars_with_NA<span class="sc">$</span>am</span>
<span id="cb264-3"><a href="basic-statistics-in-r.html#cb264-3" aria-hidden="true" tabindex="-1"></a>my_NA_am_variable</span>
<span id="cb264-4"><a href="basic-statistics-in-r.html#cb264-4" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 1 NA 1 0 NA 0 0 0 0 0 0 0 0 0 0 0 0 1 1 NA 0 0 0 0 0 1 1 NA 1 1 1 1</span></span>
<span id="cb264-5"><a href="basic-statistics-in-r.html#cb264-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb264-6"><a href="basic-statistics-in-r.html#cb264-6" aria-hidden="true" tabindex="-1"></a><span class="do">## lets replace NA in the variable am for 0</span></span>
<span id="cb264-7"><a href="basic-statistics-in-r.html#cb264-7" aria-hidden="true" tabindex="-1"></a>my_replaced_am_variable <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(my_NA_am_variable),<span class="dv">0</span>,my_NA_am_variable)</span>
<span id="cb264-8"><a href="basic-statistics-in-r.html#cb264-8" aria-hidden="true" tabindex="-1"></a>my_replaced_am_variable</span>
<span id="cb264-9"><a href="basic-statistics-in-r.html#cb264-9" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] 1 0 1 0 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 0 0 0 0 0 0 1 1 0 1 1 1 1</span></span>
<span id="cb264-10"><a href="basic-statistics-in-r.html#cb264-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb264-11"><a href="basic-statistics-in-r.html#cb264-11" aria-hidden="true" tabindex="-1"></a><span class="do">## lets put back the compled am variable in our data frame</span></span>
<span id="cb264-12"><a href="basic-statistics-in-r.html#cb264-12" aria-hidden="true" tabindex="-1"></a>my_mtcars_with_completed_am <span class="ot">&lt;-</span> my_mtcars_with_NA</span>
<span id="cb264-13"><a href="basic-statistics-in-r.html#cb264-13" aria-hidden="true" tabindex="-1"></a>my_mtcars_with_completed_am<span class="sc">$</span>am <span class="ot">&lt;-</span> my_replaced_am_variable</span>
<span id="cb264-14"><a href="basic-statistics-in-r.html#cb264-14" aria-hidden="true" tabindex="-1"></a>my_mtcars_with_completed_am</span>
<span id="cb264-15"><a href="basic-statistics-in-r.html#cb264-15" aria-hidden="true" tabindex="-1"></a><span class="co">#    mpg cyl  disp  hp drat    wt  qsec vs am gear carb            car_name</span></span>
<span id="cb264-16"><a href="basic-statistics-in-r.html#cb264-16" aria-hidden="true" tabindex="-1"></a><span class="co"># 1  21.0   6 160.0 110 3.90 2.620 16.46  0  1    4    4                &lt;NA&gt;</span></span>
<span id="cb264-17"><a href="basic-statistics-in-r.html#cb264-17" aria-hidden="true" tabindex="-1"></a><span class="co"># 2  21.0   6 160.0  NA 3.90 2.875 17.02  0  0    4   NA                &lt;NA&gt;</span></span>
<span id="cb264-18"><a href="basic-statistics-in-r.html#cb264-18" aria-hidden="true" tabindex="-1"></a><span class="co"># 3  22.8   4 108.0  93 3.85 2.320    NA NA  1    4    1          Datsun 710</span></span>
<span id="cb264-19"><a href="basic-statistics-in-r.html#cb264-19" aria-hidden="true" tabindex="-1"></a><span class="co"># 4    NA   6 258.0 110 3.08 3.215    NA  1  0    3    1      Hornet 4 Drive</span></span>
<span id="cb264-20"><a href="basic-statistics-in-r.html#cb264-20" aria-hidden="true" tabindex="-1"></a><span class="co"># 5  18.7   8 360.0 175 3.15 3.440 17.02  0  0    3    2   Hornet Sportabout</span></span>
<span id="cb264-21"><a href="basic-statistics-in-r.html#cb264-21" aria-hidden="true" tabindex="-1"></a><span class="co"># 6  18.1   6 225.0 105 2.76 3.460 20.22  1  0   3   NA             Valiant</span></span>
<span id="cb264-22"><a href="basic-statistics-in-r.html#cb264-22" aria-hidden="true" tabindex="-1"></a><span class="co"># 7  14.3   8 360.0  NA 3.21 3.570    NA  0  0    3   NA          Duster 360</span></span>
<span id="cb264-23"><a href="basic-statistics-in-r.html#cb264-23" aria-hidden="true" tabindex="-1"></a><span class="co"># 8  24.4   4 146.7  62   NA 3.190 20.00  1  0    4    2           Merc 240D</span></span>
<span id="cb264-24"><a href="basic-statistics-in-r.html#cb264-24" aria-hidden="true" tabindex="-1"></a><span class="co"># 9  22.8   4    NA  95   NA 3.150 22.90  1  0    4   NA            Merc 230</span></span>
<span id="cb264-25"><a href="basic-statistics-in-r.html#cb264-25" aria-hidden="true" tabindex="-1"></a><span class="co"># 10 19.2   6 167.6 123 3.92 3.440 18.30  1  0   NA   NA                &lt;NA&gt;</span></span>
<span id="cb264-26"><a href="basic-statistics-in-r.html#cb264-26" aria-hidden="true" tabindex="-1"></a><span class="co"># 11 17.8   6 167.6 123 3.92 3.440 18.90  1  0    4   NA           Merc 280C</span></span>
<span id="cb264-27"><a href="basic-statistics-in-r.html#cb264-27" aria-hidden="true" tabindex="-1"></a><span class="co"># 12 16.4   8 275.8  NA 3.07 4.070 17.40  0  0    3    3          Merc 450SE</span></span>
<span id="cb264-28"><a href="basic-statistics-in-r.html#cb264-28" aria-hidden="true" tabindex="-1"></a><span class="co"># 13 17.3   8 275.8 180 3.07 3.730 17.60  0  0    3    3          Merc 450SL</span></span>
<span id="cb264-29"><a href="basic-statistics-in-r.html#cb264-29" aria-hidden="true" tabindex="-1"></a><span class="co"># 14 15.2   8    NA 180 3.07 3.780    NA  0  0    3    3                &lt;NA&gt;</span></span>
<span id="cb264-30"><a href="basic-statistics-in-r.html#cb264-30" aria-hidden="true" tabindex="-1"></a><span class="co"># 15 10.4  NA 472.0 205 2.93    NA 17.98  0  0    3    4  Cadillac Fleetwood</span></span>
<span id="cb264-31"><a href="basic-statistics-in-r.html#cb264-31" aria-hidden="true" tabindex="-1"></a><span class="co"># 16 10.4  NA 460.0 215 3.00 5.424 17.82  0  0    3    4 Lincoln Continental</span></span>
<span id="cb264-32"><a href="basic-statistics-in-r.html#cb264-32" aria-hidden="true" tabindex="-1"></a><span class="co"># 17 14.7   8 440.0  NA 3.23 5.345    NA  0  0    3    4   Chrysler Imperial</span></span>
<span id="cb264-33"><a href="basic-statistics-in-r.html#cb264-33" aria-hidden="true" tabindex="-1"></a><span class="co"># 18 32.4   4  78.7  66   NA 2.200 19.47  1  1   NA    1            Fiat 128</span></span>
<span id="cb264-34"><a href="basic-statistics-in-r.html#cb264-34" aria-hidden="true" tabindex="-1"></a><span class="co"># 19 30.4   4  75.7  52 4.93 1.615 18.52  1  1    4    2         Honda Civic</span></span>
<span id="cb264-35"><a href="basic-statistics-in-r.html#cb264-35" aria-hidden="true" tabindex="-1"></a><span class="co"># 20 33.9   4  71.1  NA 4.22 1.835 19.90 NA  0    4    1      Toyota Corolla</span></span>
<span id="cb264-36"><a href="basic-statistics-in-r.html#cb264-36" aria-hidden="true" tabindex="-1"></a><span class="co"># 21 21.5   4 120.1  97 3.70 2.465 20.01  1  0    3    1       Toyota Corona</span></span>
<span id="cb264-37"><a href="basic-statistics-in-r.html#cb264-37" aria-hidden="true" tabindex="-1"></a><span class="co"># 22 15.5   8 318.0 150 2.76 3.520 16.87  0  0    3    2                &lt;NA&gt;</span></span>
<span id="cb264-38"><a href="basic-statistics-in-r.html#cb264-38" aria-hidden="true" tabindex="-1"></a><span class="co"># 23 15.2   8    NA 150 3.15    NA 17.30  0  0    3    2         AMC Javelin</span></span>
<span id="cb264-39"><a href="basic-statistics-in-r.html#cb264-39" aria-hidden="true" tabindex="-1"></a><span class="co"># 24   NA   8    NA 245 3.73    NA 15.41  0  0   3    4          Camaro Z28</span></span>
<span id="cb264-40"><a href="basic-statistics-in-r.html#cb264-40" aria-hidden="true" tabindex="-1"></a><span class="co"># 25   NA   8 400.0 175   NA 3.845 17.05  0  0    3    2    Pontiac Firebird</span></span>
<span id="cb264-41"><a href="basic-statistics-in-r.html#cb264-41" aria-hidden="true" tabindex="-1"></a><span class="co"># 26   NA   4  79.0  66 4.08 1.935 18.90  1  1   NA    1           Fiat X1-9</span></span>
<span id="cb264-42"><a href="basic-statistics-in-r.html#cb264-42" aria-hidden="true" tabindex="-1"></a><span class="co"># 27 26.0   4 120.3  91 4.43    NA 16.70  0  0    5   NA       Porsche 914-2</span></span>
<span id="cb264-43"><a href="basic-statistics-in-r.html#cb264-43" aria-hidden="true" tabindex="-1"></a><span class="co"># 28 30.4   4  95.1 113 3.77 1.513 16.90  1  1    5   NA        Lotus Europa</span></span>
<span id="cb264-44"><a href="basic-statistics-in-r.html#cb264-44" aria-hidden="true" tabindex="-1"></a><span class="co"># 29 15.8   8 351.0 264 4.22 3.170 14.50  0  1    5    4      Ford Pantera L</span></span>
<span id="cb264-45"><a href="basic-statistics-in-r.html#cb264-45" aria-hidden="true" tabindex="-1"></a><span class="co"># 30   NA   6 145.0 175 3.62    NA 15.50  0  1    5    6                &lt;NA&gt;</span></span>
<span id="cb264-46"><a href="basic-statistics-in-r.html#cb264-46" aria-hidden="true" tabindex="-1"></a><span class="co"># 31 15.0   8 301.0 335 3.54 3.570 14.60  0  1    5    8       Maserati Bora</span></span>
<span id="cb264-47"><a href="basic-statistics-in-r.html#cb264-47" aria-hidden="true" tabindex="-1"></a><span class="co"># 32 21.4   4 121.0 109 4.11 2.780 18.60  1  1   NA   NA          Volvo 142E</span></span></code></pre></div>
<p>Notice that columns called <code>am</code> is now complete with no missing <strong>NA</strong>.</p>
<p>In some tests or statistics, you can explicitly exclude <strong>NA</strong> in the calculations using <code>na.rm = TRUE</code>.</p>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb265-1"><a href="basic-statistics-in-r.html#cb265-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Lets get the mean value of mpg without removing NAs.</span></span>
<span id="cb265-2"><a href="basic-statistics-in-r.html#cb265-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(my_mtcars_with_NA<span class="sc">$</span>mpg)</span>
<span id="cb265-3"><a href="basic-statistics-in-r.html#cb265-3" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] NA</span></span>
<span id="cb265-4"><a href="basic-statistics-in-r.html#cb265-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb265-5"><a href="basic-statistics-in-r.html#cb265-5" aria-hidden="true" tabindex="-1"></a><span class="do">## we get an NA and a warning. Now, lets repat with the na.rm = TRUE</span></span>
<span id="cb265-6"><a href="basic-statistics-in-r.html#cb265-6" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(my_mtcars_with_NA<span class="sc">$</span>mpg,<span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb265-7"><a href="basic-statistics-in-r.html#cb265-7" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 20.07407</span></span></code></pre></div>
<p>You can get the mean of all columns in the data frame.</p>
<div class="sourceCode" id="cb266"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb266-1"><a href="basic-statistics-in-r.html#cb266-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We need to remove the discrete variable car_name</span></span>
<span id="cb266-2"><a href="basic-statistics-in-r.html#cb266-2" aria-hidden="true" tabindex="-1"></a>my_mtcars_with_NA_2 <span class="ot">&lt;-</span> <span class="fu">subset</span>(my_mtcars_with_NA, <span class="at">select =</span> <span class="sc">!</span><span class="fu">names</span>(my_mtcars_with_NA) <span class="sc">%in%</span> <span class="st">&quot;car_name&quot;</span>)</span>
<span id="cb266-3"><a href="basic-statistics-in-r.html#cb266-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(my_mtcars_with_NA_2)</span>
<span id="cb266-4"><a href="basic-statistics-in-r.html#cb266-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   mpg cyl disp  hp drat    wt  qsec vs am gear carb</span></span>
<span id="cb266-5"><a href="basic-statistics-in-r.html#cb266-5" aria-hidden="true" tabindex="-1"></a><span class="co"># 1 21.0   6  160 110 3.90 2.620 16.46  0  1    4    4</span></span>
<span id="cb266-6"><a href="basic-statistics-in-r.html#cb266-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 2 21.0   6  160  NA 3.90 2.875 17.02  0 NA    4   NA</span></span>
<span id="cb266-7"><a href="basic-statistics-in-r.html#cb266-7" aria-hidden="true" tabindex="-1"></a><span class="co"># 3 22.8   4  108  93 3.85 2.320    NA NA  1    4    1</span></span>
<span id="cb266-8"><a href="basic-statistics-in-r.html#cb266-8" aria-hidden="true" tabindex="-1"></a><span class="co"># 4   NA   6  258 110 3.08 3.215    NA  1  0    3    1</span></span>
<span id="cb266-9"><a href="basic-statistics-in-r.html#cb266-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 5 18.7   8  360 175 3.15 3.440 17.02  0  0    3    2</span></span>
<span id="cb266-10"><a href="basic-statistics-in-r.html#cb266-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 6 18.1   6  225 105 2.76 3.460 20.22  1 NA    3   NA</span></span>
<span id="cb266-11"><a href="basic-statistics-in-r.html#cb266-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb266-12"><a href="basic-statistics-in-r.html#cb266-12" aria-hidden="true" tabindex="-1"></a><span class="do">## We try to get the mean of all columns and fail</span></span>
<span id="cb266-13"><a href="basic-statistics-in-r.html#cb266-13" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(my_mtcars_with_NA_2)</span>
<span id="cb266-14"><a href="basic-statistics-in-r.html#cb266-14" aria-hidden="true" tabindex="-1"></a><span class="co"># mpg  cyl disp   hp drat   wt qsec   vs   am gear carb </span></span>
<span id="cb266-15"><a href="basic-statistics-in-r.html#cb266-15" aria-hidden="true" tabindex="-1"></a><span class="co">#   NA   NA   NA   NA   NA   NA   NA   NA   NA   NA   NA </span></span>
<span id="cb266-16"><a href="basic-statistics-in-r.html#cb266-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb266-17"><a href="basic-statistics-in-r.html#cb266-17" aria-hidden="true" tabindex="-1"></a><span class="do">## We try now by removing NA</span></span>
<span id="cb266-18"><a href="basic-statistics-in-r.html#cb266-18" aria-hidden="true" tabindex="-1"></a><span class="fu">colMeans</span>(my_mtcars_with_NA_2,<span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb266-19"><a href="basic-statistics-in-r.html#cb266-19" aria-hidden="true" tabindex="-1"></a><span class="co">#       mpg         cyl        disp          hp        drat          wt        qsec          vs          am        gear        carb </span></span>
<span id="cb266-20"><a href="basic-statistics-in-r.html#cb266-20" aria-hidden="true" tabindex="-1"></a><span class="co"># 20.0740741   6.0666667 225.4464286 143.1111111   3.5828571   3.1672963  17.8462963   0.4000000   0.3846154   3.6428571   2.8260870 </span></span></code></pre></div>
<p>You can do pair <code>t.test()</code> with cars with three classes of cylinders 4 and 6 with values on column <code>mpg</code></p>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb267-1"><a href="basic-statistics-in-r.html#cb267-1" aria-hidden="true" tabindex="-1"></a><span class="do">## group of cars with 4 cylinders</span></span>
<span id="cb267-2"><a href="basic-statistics-in-r.html#cb267-2" aria-hidden="true" tabindex="-1"></a>mpg_4 <span class="ot">&lt;-</span> <span class="fu">subset</span>(my_mtcars_with_NA, cyl <span class="sc">==</span> <span class="dv">4</span>, <span class="at">select =</span> mpg)</span>
<span id="cb267-3"><a href="basic-statistics-in-r.html#cb267-3" aria-hidden="true" tabindex="-1"></a>mpg_4</span>
<span id="cb267-4"><a href="basic-statistics-in-r.html#cb267-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     mpg</span></span>
<span id="cb267-5"><a href="basic-statistics-in-r.html#cb267-5" aria-hidden="true" tabindex="-1"></a><span class="co">#3  22.8</span></span>
<span id="cb267-6"><a href="basic-statistics-in-r.html#cb267-6" aria-hidden="true" tabindex="-1"></a><span class="co">#8  24.4</span></span>
<span id="cb267-7"><a href="basic-statistics-in-r.html#cb267-7" aria-hidden="true" tabindex="-1"></a><span class="co">#9  22.8</span></span>
<span id="cb267-8"><a href="basic-statistics-in-r.html#cb267-8" aria-hidden="true" tabindex="-1"></a><span class="co">#18 32.4</span></span>
<span id="cb267-9"><a href="basic-statistics-in-r.html#cb267-9" aria-hidden="true" tabindex="-1"></a><span class="co">#19 30.4</span></span>
<span id="cb267-10"><a href="basic-statistics-in-r.html#cb267-10" aria-hidden="true" tabindex="-1"></a><span class="co">#20 33.9</span></span>
<span id="cb267-11"><a href="basic-statistics-in-r.html#cb267-11" aria-hidden="true" tabindex="-1"></a><span class="co">#21 21.5</span></span>
<span id="cb267-12"><a href="basic-statistics-in-r.html#cb267-12" aria-hidden="true" tabindex="-1"></a><span class="co">#26   NA</span></span>
<span id="cb267-13"><a href="basic-statistics-in-r.html#cb267-13" aria-hidden="true" tabindex="-1"></a><span class="co">#27 26.0</span></span>
<span id="cb267-14"><a href="basic-statistics-in-r.html#cb267-14" aria-hidden="true" tabindex="-1"></a><span class="co">#28 30.4</span></span>
<span id="cb267-15"><a href="basic-statistics-in-r.html#cb267-15" aria-hidden="true" tabindex="-1"></a><span class="co">#32 21.4</span></span>
<span id="cb267-16"><a href="basic-statistics-in-r.html#cb267-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb267-17"><a href="basic-statistics-in-r.html#cb267-17" aria-hidden="true" tabindex="-1"></a><span class="do">## group of cars with 6 cylinders</span></span>
<span id="cb267-18"><a href="basic-statistics-in-r.html#cb267-18" aria-hidden="true" tabindex="-1"></a>mpg_6 <span class="ot">&lt;-</span> <span class="fu">subset</span>(my_mtcars_with_NA, cyl <span class="sc">==</span> <span class="dv">6</span>, <span class="at">select =</span> mpg)</span>
<span id="cb267-19"><a href="basic-statistics-in-r.html#cb267-19" aria-hidden="true" tabindex="-1"></a>mpg_6</span>
<span id="cb267-20"><a href="basic-statistics-in-r.html#cb267-20" aria-hidden="true" tabindex="-1"></a><span class="co">#    mpg</span></span>
<span id="cb267-21"><a href="basic-statistics-in-r.html#cb267-21" aria-hidden="true" tabindex="-1"></a><span class="co">#1  21.0</span></span>
<span id="cb267-22"><a href="basic-statistics-in-r.html#cb267-22" aria-hidden="true" tabindex="-1"></a><span class="co">#2  21.0</span></span>
<span id="cb267-23"><a href="basic-statistics-in-r.html#cb267-23" aria-hidden="true" tabindex="-1"></a><span class="co">#4    NA</span></span>
<span id="cb267-24"><a href="basic-statistics-in-r.html#cb267-24" aria-hidden="true" tabindex="-1"></a><span class="co">#6  18.1</span></span>
<span id="cb267-25"><a href="basic-statistics-in-r.html#cb267-25" aria-hidden="true" tabindex="-1"></a><span class="co">#10 19.2</span></span>
<span id="cb267-26"><a href="basic-statistics-in-r.html#cb267-26" aria-hidden="true" tabindex="-1"></a><span class="co">#11 17.8</span></span>
<span id="cb267-27"><a href="basic-statistics-in-r.html#cb267-27" aria-hidden="true" tabindex="-1"></a><span class="co">#30   NA</span></span>
<span id="cb267-28"><a href="basic-statistics-in-r.html#cb267-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb267-29"><a href="basic-statistics-in-r.html#cb267-29" aria-hidden="true" tabindex="-1"></a><span class="do">## We try to do a t.test for pair of groups</span></span>
<span id="cb267-30"><a href="basic-statistics-in-r.html#cb267-30" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(mpg_4, mpg_6, <span class="at">paired =</span> <span class="cn">FALSE</span>)</span>
<span id="cb267-31"><a href="basic-statistics-in-r.html#cb267-31" aria-hidden="true" tabindex="-1"></a><span class="co">#   Welch Two Sample t-test</span></span>
<span id="cb267-32"><a href="basic-statistics-in-r.html#cb267-32" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb267-33"><a href="basic-statistics-in-r.html#cb267-33" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  mpg_4 and mpg_6</span></span>
<span id="cb267-34"><a href="basic-statistics-in-r.html#cb267-34" aria-hidden="true" tabindex="-1"></a><span class="co">#t = 4.3493, df = 11.974, p-value = 0.0009507</span></span>
<span id="cb267-35"><a href="basic-statistics-in-r.html#cb267-35" aria-hidden="true" tabindex="-1"></a><span class="co">#alternative hypothesis: true difference in means is not equal to 0</span></span>
<span id="cb267-36"><a href="basic-statistics-in-r.html#cb267-36" aria-hidden="true" tabindex="-1"></a><span class="co">#95 percent confidence interval:</span></span>
<span id="cb267-37"><a href="basic-statistics-in-r.html#cb267-37" aria-hidden="true" tabindex="-1"></a><span class="co">#  3.582259 10.777741</span></span>
<span id="cb267-38"><a href="basic-statistics-in-r.html#cb267-38" aria-hidden="true" tabindex="-1"></a><span class="co">#sample estimates:</span></span>
<span id="cb267-39"><a href="basic-statistics-in-r.html#cb267-39" aria-hidden="true" tabindex="-1"></a><span class="co">#mean of x mean of y </span></span>
<span id="cb267-40"><a href="basic-statistics-in-r.html#cb267-40" aria-hidden="true" tabindex="-1"></a><span class="co">#    26.60     19.42 </span></span></code></pre></div>
<p>However, we try a similar <code>wilcox.test()</code> but we fail</p>
<div class="sourceCode" id="cb268"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb268-1"><a href="basic-statistics-in-r.html#cb268-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We try the wilcox.test and fail</span></span>
<span id="cb268-2"><a href="basic-statistics-in-r.html#cb268-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(mpg_4, mpg_6, <span class="at">paired =</span> <span class="cn">FALSE</span>)</span>
<span id="cb268-3"><a href="basic-statistics-in-r.html#cb268-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Error in wilcox.test.default(mpg_4, mpg_6, paired = FALSE) : </span></span>
<span id="cb268-4"><a href="basic-statistics-in-r.html#cb268-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  &#39;x&#39; must be numeric</span></span>
<span id="cb268-5"><a href="basic-statistics-in-r.html#cb268-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb268-6"><a href="basic-statistics-in-r.html#cb268-6" aria-hidden="true" tabindex="-1"></a><span class="do">## We need to remove NA</span></span>
<span id="cb268-7"><a href="basic-statistics-in-r.html#cb268-7" aria-hidden="true" tabindex="-1"></a>mpg_4_c <span class="ot">&lt;-</span> mpg_4[<span class="fu">complete.cases</span>(mpg_4),]</span>
<span id="cb268-8"><a href="basic-statistics-in-r.html#cb268-8" aria-hidden="true" tabindex="-1"></a>mpg_4_c</span>
<span id="cb268-9"><a href="basic-statistics-in-r.html#cb268-9" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 22.8 24.4 22.8 32.4 30.4 33.9 21.5 26.0 30.4 21.4</span></span>
<span id="cb268-10"><a href="basic-statistics-in-r.html#cb268-10" aria-hidden="true" tabindex="-1"></a>mpg_6_c <span class="ot">&lt;-</span> mpg_6[<span class="fu">complete.cases</span>(mpg_6),]</span>
<span id="cb268-11"><a href="basic-statistics-in-r.html#cb268-11" aria-hidden="true" tabindex="-1"></a>mpg_6_c</span>
<span id="cb268-12"><a href="basic-statistics-in-r.html#cb268-12" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 21.0 21.0 18.1 19.2 17.8</span></span>
<span id="cb268-13"><a href="basic-statistics-in-r.html#cb268-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb268-14"><a href="basic-statistics-in-r.html#cb268-14" aria-hidden="true" tabindex="-1"></a><span class="do">## We can try now again the wilcox.test</span></span>
<span id="cb268-15"><a href="basic-statistics-in-r.html#cb268-15" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(mpg_4_c, mpg_6_c, <span class="at">paired =</span> <span class="cn">FALSE</span>,)</span>
<span id="cb268-16"><a href="basic-statistics-in-r.html#cb268-16" aria-hidden="true" tabindex="-1"></a><span class="co">#   Wilcoxon rank sum test with continuity correction</span></span>
<span id="cb268-17"><a href="basic-statistics-in-r.html#cb268-17" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb268-18"><a href="basic-statistics-in-r.html#cb268-18" aria-hidden="true" tabindex="-1"></a><span class="co"># data:  mpg_4_c and mpg_6_c</span></span>
<span id="cb268-19"><a href="basic-statistics-in-r.html#cb268-19" aria-hidden="true" tabindex="-1"></a><span class="co"># W = 50, p-value = 0.002624</span></span>
<span id="cb268-20"><a href="basic-statistics-in-r.html#cb268-20" aria-hidden="true" tabindex="-1"></a><span class="co"># alternative hypothesis: true location shift is not equal to 0</span></span></code></pre></div>
</div>
<div id="about-p-values" class="section level2" number="6.5">
<h2><span class="header-section-number">6.5</span> About p-values</h2>
<p>In most of your statistical tests, you will pay attention to the number associated to a p-value.
The general interpretation of p-values is that they represent the probability of getting results similar or as extreme as those for the null hypothesis assuming that this one is correct (e.g., an expected mean value or similar mean values between two groups assuming that they are from the same population). Likewise, p-values measure the probability that an observed differences between groups could have occurred just by random chance.</p>
</div>
<div id="how-to-read-p-values" class="section level2" number="6.6">
<h2><span class="header-section-number">6.6</span> How to read p-values</h2>
<p>In most cases, small or tiny p-value are of interest as they represent stronger evidence to
reject the null hypothesis and opens the possibly for an alternative hypothesis or explanation.
In practice, the significance level is stated by the researcher and in advance (e.g., p &lt;0.05 is significant).
The research then determines if the observed p-value is safe to reject the null hypothesis. In contrast, a p-value greater
than 0.05 means that results do not deviate enough from those expected if the null hypothesis is true and these p-values are
considered to be not statistically significant and fail to reject the null hypothesis.</p>
<p>To explore this further, we will use the <code>mtcars</code> dataset.</p>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb269-1"><a href="basic-statistics-in-r.html#cb269-1" aria-hidden="true" tabindex="-1"></a><span class="do">## the mtcars is comparisons between 32 automobiles (1973–74 models)</span></span>
<span id="cb269-2"><a href="basic-statistics-in-r.html#cb269-2" aria-hidden="true" tabindex="-1"></a>my_mtcars <span class="ot">&lt;-</span> mtcars</span>
<span id="cb269-3"><a href="basic-statistics-in-r.html#cb269-3" aria-hidden="true" tabindex="-1"></a>my_mtcars<span class="sc">$</span>car_name <span class="ot">&lt;-</span> <span class="fu">rownames</span>(my_mtcars)</span>
<span id="cb269-4"><a href="basic-statistics-in-r.html#cb269-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(my_mtcars) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb269-5"><a href="basic-statistics-in-r.html#cb269-5" aria-hidden="true" tabindex="-1"></a><span class="do">## group of cars with 4 cylinders</span></span>
<span id="cb269-6"><a href="basic-statistics-in-r.html#cb269-6" aria-hidden="true" tabindex="-1"></a>mpg_4 <span class="ot">&lt;-</span> <span class="fu">subset</span>(my_mtcars, cyl <span class="sc">==</span> <span class="dv">4</span>, <span class="at">select =</span> mpg)</span>
<span id="cb269-7"><a href="basic-statistics-in-r.html#cb269-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb269-8"><a href="basic-statistics-in-r.html#cb269-8" aria-hidden="true" tabindex="-1"></a><span class="do">## group of 6 cylinders</span></span>
<span id="cb269-9"><a href="basic-statistics-in-r.html#cb269-9" aria-hidden="true" tabindex="-1"></a>mpg_6 <span class="ot">&lt;-</span> <span class="fu">subset</span>(my_mtcars, cyl <span class="sc">==</span> <span class="dv">6</span>, <span class="at">select =</span> mpg)</span>
<span id="cb269-10"><a href="basic-statistics-in-r.html#cb269-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb269-11"><a href="basic-statistics-in-r.html#cb269-11" aria-hidden="true" tabindex="-1"></a><span class="do">## We try to do a t.test for pair of groups</span></span>
<span id="cb269-12"><a href="basic-statistics-in-r.html#cb269-12" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(mpg_4, mpg_6, <span class="at">paired =</span> <span class="cn">FALSE</span>)</span>
<span id="cb269-13"><a href="basic-statistics-in-r.html#cb269-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   Welch Two Sample t-test</span></span>
<span id="cb269-14"><a href="basic-statistics-in-r.html#cb269-14" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb269-15"><a href="basic-statistics-in-r.html#cb269-15" aria-hidden="true" tabindex="-1"></a><span class="co"># data:  mpg_4 and mpg_6</span></span>
<span id="cb269-16"><a href="basic-statistics-in-r.html#cb269-16" aria-hidden="true" tabindex="-1"></a><span class="co"># t = 4.7191, df = 12.956, p-value = 0.0004048</span></span>
<span id="cb269-17"><a href="basic-statistics-in-r.html#cb269-17" aria-hidden="true" tabindex="-1"></a><span class="co"># alternative hypothesis: true difference in means is not equal to 0</span></span>
<span id="cb269-18"><a href="basic-statistics-in-r.html#cb269-18" aria-hidden="true" tabindex="-1"></a><span class="co"># 95 percent confidence interval:</span></span>
<span id="cb269-19"><a href="basic-statistics-in-r.html#cb269-19" aria-hidden="true" tabindex="-1"></a><span class="co">#  3.751376 10.090182</span></span>
<span id="cb269-20"><a href="basic-statistics-in-r.html#cb269-20" aria-hidden="true" tabindex="-1"></a><span class="co"># sample estimates:</span></span>
<span id="cb269-21"><a href="basic-statistics-in-r.html#cb269-21" aria-hidden="true" tabindex="-1"></a><span class="co"># mean of x mean of y </span></span>
<span id="cb269-22"><a href="basic-statistics-in-r.html#cb269-22" aria-hidden="true" tabindex="-1"></a><span class="co">#  26.66364  19.74286 </span></span></code></pre></div>
<p>In this exercise, what does a p-value = 0.0004048 mean? This value indicates that if the null hypothesis tested were indeed true (i.e., the mean mpg for cars with 4 and 6 cylinders is equal),
there would be a 4.048 in 10,000 chance of observing results at least as extreme were both groups will have the same mpg.
This leads the researcher to reject the null hypothesis because either an extreme rare data result has been observed,
or the null hypothesis is likely to be incorrect.</p>
</div>
<div id="multiple-testing-corrections" class="section level2" number="6.7">
<h2><span class="header-section-number">6.7</span> Multiple testing corrections</h2>
<p>Consider that in many instances, we can perform multiple tests and obtain a collection of p-values. We can apply a multiple testing correction to adjust p-values obtained to correct for occurrence of false positives. These false positives are significant p-values that should be considered not informative and due by chance (i.e., you could do 100 tests, and some might be significant just by the larger number of tests performed).</p>
<p>In bioinformatics, it is common to perform hundreds of statistical testes (e.g., t-tests or ANOVAs) when you are comparing genetic results such as to find differentially expressed genes
between a control and an experimental (study) group after the application of a treatment. The incidence of these genes falsely called as differentially expressed is proportional to the number of tests performed and the threshold p-value for significance (critical significance level).</p>
<p>We can illustrate this by assuming that we perform a t-test between two groups using the results derived from comparisons of gene expression between both. In this case, the probability by which the tests will be considered significantly (i.e., evidence of differential gene expression) is expressed by the p-value. We could use the standard threshold of p-value = 0.05, which signifies we have a 5% probability to find a significant differentially expressed gene by chance alone (i.e., not the result of treatment). Furthermore, if we tested 10,000 genes then 5% or 500 genes that might be called significant as differentially express by chance alone.</p>
<table>
<colgroup>
<col width="28%" />
<col width="20%" />
<col width="50%" />
</colgroup>
<thead>
<tr class="header">
<th>Number of genes tested</th>
<th>False positives</th>
<th>Probability of 1 or more false positives</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>5/100</td>
<td>5%</td>
</tr>
<tr class="even">
<td>2</td>
<td>10/100</td>
<td>9.75%</td>
</tr>
<tr class="odd">
<td>20</td>
<td>1</td>
<td>64.15%</td>
</tr>
<tr class="even">
<td>100</td>
<td>5</td>
<td>99.41%</td>
</tr>
<tr class="odd">
<td>1000</td>
<td>50</td>
<td>&gt;99.9%</td>
</tr>
<tr class="even">
<td>10000</td>
<td>500</td>
<td>&gt;99.9%</td>
</tr>
</tbody>
</table>
<p>If that is the case, it is advised to correct the p-values using a multiple testing correction such as the <a href="https://en.wikipedia.org/wiki/False_discovery_rate" target="_blank"><strong>false discovery rate (FDR)</strong></a> estimation.
This can be easily applied in R and usually require a vector containing the p-value and the function <code>p.adjust()</code>. I usually use these FDR corrections if &gt;5 tests are being calculated.</p>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb270-1"><a href="basic-statistics-in-r.html#cb270-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we create a vector of p-values</span></span>
<span id="cb270-2"><a href="basic-statistics-in-r.html#cb270-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">50</span>, <span class="at">mean =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="dv">0</span>, <span class="dv">25</span>), <span class="fu">rep</span>(<span class="dv">3</span>, <span class="dv">25</span>)))</span>
<span id="cb270-3"><a href="basic-statistics-in-r.html#cb270-3" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="dv">2</span><span class="sc">*</span><span class="fu">pnorm</span>(<span class="fu">sort</span>(<span class="sc">-</span><span class="fu">abs</span>(x)))</span>
<span id="cb270-4"><a href="basic-statistics-in-r.html#cb270-4" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">round</span>(p, <span class="dv">5</span>)</span>
<span id="cb270-5"><a href="basic-statistics-in-r.html#cb270-5" aria-hidden="true" tabindex="-1"></a>p</span>
<span id="cb270-6"><a href="basic-statistics-in-r.html#cb270-6" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 0.00000 0.00000 0.00002 0.00006 0.00019 0.00033 0.00033 0.00053 0.00131 0.00293 0.00348 0.00385 0.00412 0.01391 0.01479 0.01651 0.04093 0.04185</span></span>
<span id="cb270-7"><a href="basic-statistics-in-r.html#cb270-7" aria-hidden="true" tabindex="-1"></a><span class="co">#[19] 0.04283 0.06150 0.07324 0.07789 0.08126 0.09249 0.11938 0.13048 0.14542 0.15769 0.15990 0.19556 0.22030 0.22422 0.25180 0.25683 0.28846 0.31366</span></span>
<span id="cb270-8"><a href="basic-statistics-in-r.html#cb270-8" aria-hidden="true" tabindex="-1"></a><span class="co">#[37] 0.37433 0.40006 0.43389 0.51984 0.54879 0.55692 0.55747 0.66131 0.71076 0.71821 0.72748 0.88428 0.90186 0.92672</span></span>
<span id="cb270-9"><a href="basic-statistics-in-r.html#cb270-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb270-10"><a href="basic-statistics-in-r.html#cb270-10" aria-hidden="true" tabindex="-1"></a><span class="do">## how many are significant</span></span>
<span id="cb270-11"><a href="basic-statistics-in-r.html#cb270-11" aria-hidden="true" tabindex="-1"></a>p_less_0_05 <span class="ot">&lt;-</span> p[p<span class="sc">&lt;</span><span class="fl">0.05</span>]</span>
<span id="cb270-12"><a href="basic-statistics-in-r.html#cb270-12" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(p_less_0_05)</span>
<span id="cb270-13"><a href="basic-statistics-in-r.html#cb270-13" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 19</span></span>
<span id="cb270-14"><a href="basic-statistics-in-r.html#cb270-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb270-15"><a href="basic-statistics-in-r.html#cb270-15" aria-hidden="true" tabindex="-1"></a><span class="do">## we adjust these p-values </span></span>
<span id="cb270-16"><a href="basic-statistics-in-r.html#cb270-16" aria-hidden="true" tabindex="-1"></a>p.adjust<span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(p)</span>
<span id="cb270-17"><a href="basic-statistics-in-r.html#cb270-17" aria-hidden="true" tabindex="-1"></a>p.adjust</span>
<span id="cb270-18"><a href="basic-statistics-in-r.html#cb270-18" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] 0.00000 0.00000 0.00096 0.00282 0.00874 0.01485 0.01485 0.02279 0.05502 0.12013 0.13920 0.15015 0.15656 0.51467 0.53244 0.57785 1.00000 1.00000</span></span>
<span id="cb270-19"><a href="basic-statistics-in-r.html#cb270-19" aria-hidden="true" tabindex="-1"></a><span class="co">#[19] 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000</span></span>
<span id="cb270-20"><a href="basic-statistics-in-r.html#cb270-20" aria-hidden="true" tabindex="-1"></a><span class="co">#[37] 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000 1.00000</span></span>
<span id="cb270-21"><a href="basic-statistics-in-r.html#cb270-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb270-22"><a href="basic-statistics-in-r.html#cb270-22" aria-hidden="true" tabindex="-1"></a><span class="do">## how many are significant</span></span>
<span id="cb270-23"><a href="basic-statistics-in-r.html#cb270-23" aria-hidden="true" tabindex="-1"></a>p.adjust_less_0_05 <span class="ot">&lt;-</span> p.adjust[p.adjust<span class="sc">&lt;</span><span class="fl">0.05</span>]</span>
<span id="cb270-24"><a href="basic-statistics-in-r.html#cb270-24" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(p.adjust_less_0_05)</span>
<span id="cb270-25"><a href="basic-statistics-in-r.html#cb270-25" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 8</span></span></code></pre></div>
<p>Notice how the values in the <code>p.adjust</code> has increased. If not correction is applied 19 p-values of 50 are significant (~38% of all), but after correction only 8 p-values of 50 are significant (~16% of all).</p>
</div>
<div id="normalizing-your-data" class="section level2" number="6.8">
<h2><span class="header-section-number">6.8</span> Normalizing your data</h2>
<p><strong>6)</strong> Normalizing or scaling your data is sometimes useful when you are trying to compare datasets that might have extremely different units (e.g., tiny fractions and others in the hundreds of thousands). This can be done with the function <code>scale()</code>.</p>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="basic-statistics-in-r.html#cb271-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cars_data)</span>
<span id="cb271-2"><a href="basic-statistics-in-r.html#cb271-2" aria-hidden="true" tabindex="-1"></a><span class="co">#                   mpg cyl disp  hp drat    wt  qsec vs am gear carb</span></span>
<span id="cb271-3"><a href="basic-statistics-in-r.html#cb271-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Mazda RX4         21.0   6  160 110 3.90 2.620 16.46  0  1    4    4</span></span>
<span id="cb271-4"><a href="basic-statistics-in-r.html#cb271-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Mazda RX4 Wag     21.0   6  160 110 3.90 2.875 17.02  0  1    4    4</span></span>
<span id="cb271-5"><a href="basic-statistics-in-r.html#cb271-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Datsun 710        22.8   4  108  93 3.85 2.320 18.61  1  1    4    1</span></span>
<span id="cb271-6"><a href="basic-statistics-in-r.html#cb271-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Hornet 4 Drive    21.4   6  258 110 3.08 3.215 19.44  1  0    3    1</span></span>
<span id="cb271-7"><a href="basic-statistics-in-r.html#cb271-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Hornet Sportabout 18.7   8  360 175 3.15 3.440 17.02  0  0    3    2</span></span>
<span id="cb271-8"><a href="basic-statistics-in-r.html#cb271-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Valiant           18.1   6  225 105 2.76 3.460 20.22  1  0    3    1</span></span>
<span id="cb271-9"><a href="basic-statistics-in-r.html#cb271-9" aria-hidden="true" tabindex="-1"></a>cars_data_scaled <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">modify</span>(cars_data, scale)</span>
<span id="cb271-10"><a href="basic-statistics-in-r.html#cb271-10" aria-hidden="true" tabindex="-1"></a><span class="co"># head(cars_data_scaled)</span></span>
<span id="cb271-11"><a href="basic-statistics-in-r.html#cb271-11" aria-hidden="true" tabindex="-1"></a><span class="co">#                        mpg        cyl        disp         hp       drat           wt       qsec         vs         am       gear       carb</span></span>
<span id="cb271-12"><a href="basic-statistics-in-r.html#cb271-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Mazda RX4          0.1508848 -0.1049878 -0.57061982 -0.5350928  0.5675137 -0.610399567 -0.7771651 -0.8680278  1.1899014  0.4235542  0.7352031</span></span>
<span id="cb271-13"><a href="basic-statistics-in-r.html#cb271-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Mazda RX4 Wag      0.1508848 -0.1049878 -0.57061982 -0.5350928  0.5675137 -0.349785269 -0.4637808 -0.8680278  1.1899014  0.4235542  0.7352031</span></span>
<span id="cb271-14"><a href="basic-statistics-in-r.html#cb271-14" aria-hidden="true" tabindex="-1"></a><span class="co">#Datsun 710         0.4495434 -1.2248578 -0.99018209 -0.7830405  0.4739996 -0.917004624  0.4260068  1.1160357  1.1899014  0.4235542 -1.1221521</span></span>
<span id="cb271-15"><a href="basic-statistics-in-r.html#cb271-15" aria-hidden="true" tabindex="-1"></a><span class="co">#Hornet 4 Drive     0.2172534 -0.1049878  0.22009369 -0.5350928 -0.9661175 -0.002299538  0.8904872  1.1160357 -0.8141431 -0.9318192 -1.1221521</span></span>
<span id="cb271-16"><a href="basic-statistics-in-r.html#cb271-16" aria-hidden="true" tabindex="-1"></a><span class="co">#Hornet Sportabout -0.2307345  1.0148821  1.04308123  0.4129422 -0.8351978  0.227654255 -0.4637808 -0.8680278 -0.8141431 -0.9318192 -0.5030337</span></span>
<span id="cb271-17"><a href="basic-statistics-in-r.html#cb271-17" aria-hidden="true" tabindex="-1"></a><span class="co">#Valiant           -0.3302874 -0.1049878 -0.04616698 -0.6080186 -1.5646078  0.248094592  1.3269868  1.1160357 -0.8141431 -0.9318192 -1.1221521</span></span></code></pre></div>
<p><strong>NOTICE</strong> if you use <code>purrr::modify</code>, make sure that the data frame has <strong>ONLY</strong> continuous variables (no character variables, logicals, factors).</p>
<p><strong>7)</strong> Common statistical tests are parametric and most of them assume that your data has a normal (Gaussian) distribution, i.e., a typical bell-shaped curve. In other words, the distribution of values of variable that has a normal distribution will produce values (as they are being measured) with an equal chance to fall above and below the mean value of all measurements. The normal distribution is expected to have very close values for its mean (average of all values), median (mid-point value at the center of the distribution) and mode (the most frequent value). If a distribution is not normal, these three parameters will differ and the distribution is considered skewed (i.e., not normal).</p>
<div class="sourceCode" id="cb272"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb272-1"><a href="basic-statistics-in-r.html#cb272-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a sequence of numbers between 0 and 100 by 1</span></span>
<span id="cb272-2"><a href="basic-statistics-in-r.html#cb272-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">100</span>, <span class="at">by =</span> <span class="dv">1</span>)</span>
<span id="cb272-3"><a href="basic-statistics-in-r.html#cb272-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Choose the mean as 50 and standard deviation as 17.5</span></span>
<span id="cb272-4"><a href="basic-statistics-in-r.html#cb272-4" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">dnorm</span>(x, <span class="at">mean =</span> <span class="dv">50</span>, <span class="at">sd =</span> <span class="fl">17.5</span>)</span>
<span id="cb272-5"><a href="basic-statistics-in-r.html#cb272-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot this distribution</span></span>
<span id="cb272-6"><a href="basic-statistics-in-r.html#cb272-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(x,y)</span></code></pre></div>
<p><img src="figures/session_6_figs/2_normal_distribution.png" width="100%" /></p>
<p><strong>8)</strong> We can test for normal (Gaussian) distribution of our data using a normality test. We can use the Shapiro-Wilk’s test that determines if the null hypothesis (i.e., the sample has normal distribution) versus the alternative that it does not. If the test is significant, the distribution has a non-normal distribution. This test is sensitive to sample size, which means that large samples most often do not pass normality tests (i.e., show to be significant). Therefore, it’s important that you use visual inspection of your data distribution before you decide to transform your data.</p>
<p>The Shapiro-Wilk’s test is a build-in function and can be call as <code>shapiro.test()</code>. However, this test cannot be applied to large vectors (i.e., &gt; 5000 entries).</p>
<div class="sourceCode" id="cb273"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb273-1"><a href="basic-statistics-in-r.html#cb273-1" aria-hidden="true" tabindex="-1"></a><span class="co"># we can perfome in a variable of a data frame applying it as vector</span></span>
<span id="cb273-2"><a href="basic-statistics-in-r.html#cb273-2" aria-hidden="true" tabindex="-1"></a><span class="fu">shapiro.test</span>(cars_data<span class="sc">$</span>hp)</span>
<span id="cb273-3"><a href="basic-statistics-in-r.html#cb273-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   Shapiro-Wilk normality test</span></span>
<span id="cb273-4"><a href="basic-statistics-in-r.html#cb273-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb273-5"><a href="basic-statistics-in-r.html#cb273-5" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  cars_data$hp</span></span>
<span id="cb273-6"><a href="basic-statistics-in-r.html#cb273-6" aria-hidden="true" tabindex="-1"></a><span class="co">#W = 0.93342, p-value = 0.04881</span></span>
<span id="cb273-7"><a href="basic-statistics-in-r.html#cb273-7" aria-hidden="true" tabindex="-1"></a><span class="fu">shapiro.test</span>(cars_data<span class="sc">$</span>qsec)</span>
<span id="cb273-8"><a href="basic-statistics-in-r.html#cb273-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   Shapiro-Wilk normality test</span></span>
<span id="cb273-9"><a href="basic-statistics-in-r.html#cb273-9" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb273-10"><a href="basic-statistics-in-r.html#cb273-10" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  cars_data$qsec</span></span>
<span id="cb273-11"><a href="basic-statistics-in-r.html#cb273-11" aria-hidden="true" tabindex="-1"></a><span class="co">#W = 0.97325, p-value = 0.5935</span></span></code></pre></div>
<p>Based on these results, we can conclude that the variable <code>hp</code> has not normal distribution, while <code>qsec</code> has normal one. If you want to apply the <code>shapiro.test()</code> to all you variables in your data frame use <code>apply</code> with <code>2</code> argument for columns.</p>
<div class="sourceCode" id="cb274"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb274-1"><a href="basic-statistics-in-r.html#cb274-1" aria-hidden="true" tabindex="-1"></a><span class="fu">apply</span>(cars_data,<span class="dv">2</span>,shapiro.test)</span></code></pre></div>
<p><strong>NOTICE</strong> the Shapiro-Wilk’s test is ONLY to test if your data is normally distributed, nothing else!!!.</p>
<p><strong>9)</strong> In most cases, you might like to apply a transformation to your data before you apply a parametric statistical test. One of the most frequent transformations for continuous data is to <strong>normalize</strong> them, this means that change the distribution of the data to a quasinormal or normal one. Such transformations will make our data more amenable for parametric tests. However, the interpretation of the results has to be framed using the units of the transformed data (i.e., if log-transformed, you have to interpret in log-transformed unit).</p>
<p>To determine how if your data need to be normalized, it is a good idea to visualize its distribution. To exemplify this, we will describe this using the <code>hp</code> variable of the <code>car_data</code>.</p>
<div class="sourceCode" id="cb275"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb275-1"><a href="basic-statistics-in-r.html#cb275-1" aria-hidden="true" tabindex="-1"></a>hp_vector <span class="ot">&lt;-</span> cars_data<span class="sc">$</span>hp</span>
<span id="cb275-2"><a href="basic-statistics-in-r.html#cb275-2" aria-hidden="true" tabindex="-1"></a>hp_vector</span>
<span id="cb275-3"><a href="basic-statistics-in-r.html#cb275-3" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 110 110  93 110 175 105 245  62  95 123 123 180 180 180 205 215 230  66  52  65  97 150 150 245 175  66  91 113 264 175 335 109</span></span>
<span id="cb275-4"><a href="basic-statistics-in-r.html#cb275-4" aria-hidden="true" tabindex="-1"></a><span class="co"># we test for normality</span></span>
<span id="cb275-5"><a href="basic-statistics-in-r.html#cb275-5" aria-hidden="true" tabindex="-1"></a><span class="fu">shapiro.test</span>(hp_vector)</span>
<span id="cb275-6"><a href="basic-statistics-in-r.html#cb275-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   Shapiro-Wilk normality test</span></span>
<span id="cb275-7"><a href="basic-statistics-in-r.html#cb275-7" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb275-8"><a href="basic-statistics-in-r.html#cb275-8" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  cars_data$hp</span></span>
<span id="cb275-9"><a href="basic-statistics-in-r.html#cb275-9" aria-hidden="true" tabindex="-1"></a><span class="co">#W = 0.93342, p-value = 0.04881</span></span>
<span id="cb275-10"><a href="basic-statistics-in-r.html#cb275-10" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(hp_vector, <span class="at">xlab =</span> <span class="st">&quot;hp&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;values&quot;</span>)</span></code></pre></div>
<p><img src="figures/session_6_figs/3_hp_raw.png" width="100%" />
In this boxplot, we can see an outlier and it is skewed towards lower values of <code>hp</code>. Some of the most common transformations are to square root the data using the function <code>sqrt()</code> and to perform logarithms using <code>log10()</code> with base 10 or natural logarithms ‘log()’.</p>
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb276-1"><a href="basic-statistics-in-r.html#cb276-1" aria-hidden="true" tabindex="-1"></a>hp_vector_sqrt <span class="ot">&lt;-</span> <span class="fu">sqrt</span>(hp_vector)</span>
<span id="cb276-2"><a href="basic-statistics-in-r.html#cb276-2" aria-hidden="true" tabindex="-1"></a><span class="fu">shapiro.test</span>(hp_vector_sqrt)</span>
<span id="cb276-3"><a href="basic-statistics-in-r.html#cb276-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Shapiro-Wilk normality test</span></span>
<span id="cb276-4"><a href="basic-statistics-in-r.html#cb276-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb276-5"><a href="basic-statistics-in-r.html#cb276-5" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  hp_vector_sqrt</span></span>
<span id="cb276-6"><a href="basic-statistics-in-r.html#cb276-6" aria-hidden="true" tabindex="-1"></a><span class="co">#W = 0.96456, p-value = 0.3642</span></span>
<span id="cb276-7"><a href="basic-statistics-in-r.html#cb276-7" aria-hidden="true" tabindex="-1"></a>hp_vector_log10 <span class="ot">&lt;-</span> <span class="fu">log10</span>(hp_vector)</span>
<span id="cb276-8"><a href="basic-statistics-in-r.html#cb276-8" aria-hidden="true" tabindex="-1"></a><span class="fu">shapiro.test</span>(hp_vector_log10)</span>
<span id="cb276-9"><a href="basic-statistics-in-r.html#cb276-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   Shapiro-Wilk normality test</span></span>
<span id="cb276-10"><a href="basic-statistics-in-r.html#cb276-10" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb276-11"><a href="basic-statistics-in-r.html#cb276-11" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  hp_vector_log10</span></span>
<span id="cb276-12"><a href="basic-statistics-in-r.html#cb276-12" aria-hidden="true" tabindex="-1"></a><span class="co">#W = 0.97026, p-value = 0.5065</span></span>
<span id="cb276-13"><a href="basic-statistics-in-r.html#cb276-13" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(hp_vector_log10, <span class="at">xlab =</span> <span class="st">&quot;hp&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;values_transformed_by_log10&quot;</span>)</span></code></pre></div>
<p>Here is the resulting boxplot after<code>log10()</code> transformation.
<img src="figures/session_6_figs/4_hp_log10.png" width="100%" /></p>
<p>In some extreme cases, there are several R-packages that can help you to transform your data to have a normal distribution. One of these is <a href="https://cran.r-project.org/web/packages/bestNormalize/index.html" target="_blank"><strong>bestNormalize</strong></a> and its function <code>boxcox()</code>. This provides a Box-Cox power transformation that can be useful for extremely skewed distributions. Likewise, this function returns, by default, transformed values that are also centered and scaled.</p>
<div class="sourceCode" id="cb277"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb277-1"><a href="basic-statistics-in-r.html#cb277-1" aria-hidden="true" tabindex="-1"></a><span class="co"># if you need to install the R-package bestNormalize</span></span>
<span id="cb277-2"><a href="basic-statistics-in-r.html#cb277-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;bestNormalize&quot;</span>)</span>
<span id="cb277-3"><a href="basic-statistics-in-r.html#cb277-3" aria-hidden="true" tabindex="-1"></a><span class="co"># load library</span></span>
<span id="cb277-4"><a href="basic-statistics-in-r.html#cb277-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(bestNormalize)</span>
<span id="cb277-5"><a href="basic-statistics-in-r.html#cb277-5" aria-hidden="true" tabindex="-1"></a><span class="co"># perform Box-Cox power tranformation.</span></span>
<span id="cb277-6"><a href="basic-statistics-in-r.html#cb277-6" aria-hidden="true" tabindex="-1"></a>hp_vector_boxcox_list <span class="ot">&lt;-</span> bestNormalize<span class="sc">::</span><span class="fu">boxcox</span>(hp_vector)</span>
<span id="cb277-7"><a href="basic-statistics-in-r.html#cb277-7" aria-hidden="true" tabindex="-1"></a>hp_vector_boxcox_list</span>
<span id="cb277-8"><a href="basic-statistics-in-r.html#cb277-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Standardized Box Cox Transformation with 32 nonmissing obs.:</span></span>
<span id="cb277-9"><a href="basic-statistics-in-r.html#cb277-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimated statistics:</span></span>
<span id="cb277-10"><a href="basic-statistics-in-r.html#cb277-10" aria-hidden="true" tabindex="-1"></a><span class="co"># - lambda = 0.0940865 </span></span>
<span id="cb277-11"><a href="basic-statistics-in-r.html#cb277-11" aria-hidden="true" tabindex="-1"></a><span class="co"># - mean (before standardization) = 6.21272 </span></span>
<span id="cb277-12"><a href="basic-statistics-in-r.html#cb277-12" aria-hidden="true" tabindex="-1"></a><span class="co"># - sd (before standardization) = 0.7515468 </span></span>
<span id="cb277-13"><a href="basic-statistics-in-r.html#cb277-13" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(hp_vector_boxcox_list)</span>
<span id="cb277-14"><a href="basic-statistics-in-r.html#cb277-14" aria-hidden="true" tabindex="-1"></a><span class="co">#List of 8</span></span>
<span id="cb277-15"><a href="basic-statistics-in-r.html#cb277-15" aria-hidden="true" tabindex="-1"></a><span class="co"># $ x.t        : num [1:32] -0.401 -0.401 -0.745 -0.401 0.582 ...</span></span>
<span id="cb277-16"><a href="basic-statistics-in-r.html#cb277-16" aria-hidden="true" tabindex="-1"></a><span class="co">#$ x          : num [1:32] 110 110 93 110 175 105 245 62 95 123 ...</span></span>
<span id="cb277-17"><a href="basic-statistics-in-r.html#cb277-17" aria-hidden="true" tabindex="-1"></a><span class="co">#$ mean       : num 6.21</span></span>
<span id="cb277-18"><a href="basic-statistics-in-r.html#cb277-18" aria-hidden="true" tabindex="-1"></a><span class="co">#$ sd         : num 0.752</span></span>
<span id="cb277-19"><a href="basic-statistics-in-r.html#cb277-19" aria-hidden="true" tabindex="-1"></a><span class="co">#$ lambda     : num 0.0941</span></span>
<span id="cb277-20"><a href="basic-statistics-in-r.html#cb277-20" aria-hidden="true" tabindex="-1"></a><span class="co">#$ n          : int 32</span></span>
<span id="cb277-21"><a href="basic-statistics-in-r.html#cb277-21" aria-hidden="true" tabindex="-1"></a><span class="co">#$ norm_stat  : num 1.4</span></span>
<span id="cb277-22"><a href="basic-statistics-in-r.html#cb277-22" aria-hidden="true" tabindex="-1"></a><span class="co">#$ standardize: logi TRUE</span></span>
<span id="cb277-23"><a href="basic-statistics-in-r.html#cb277-23" aria-hidden="true" tabindex="-1"></a><span class="co">#- attr(*, &quot;class&quot;)= chr [1:2] &quot;boxcox&quot; &quot;list&quot;</span></span>
<span id="cb277-24"><a href="basic-statistics-in-r.html#cb277-24" aria-hidden="true" tabindex="-1"></a>hp_vector_boxcox <span class="ot">&lt;-</span> hp_vector_boxcox_list<span class="sc">$</span>x.t</span>
<span id="cb277-25"><a href="basic-statistics-in-r.html#cb277-25" aria-hidden="true" tabindex="-1"></a><span class="co">#(1] -0.4005863 -0.4005863 -0.7454816 -0.4005863  0.5821478 -0.4967035  1.3216259 -1.5563448 -0.7020701 -0.1680641 -0.1680641  0.6431660  0.6431660  0.6431660  0.9269679</span></span>
<span id="cb277-26"><a href="basic-statistics-in-r.html#cb277-26" aria-hidden="true" tabindex="-1"></a><span class="co">#[16]  1.0317736  1.1809841 -1.4333227 -1.8985905 -1.4634315 -0.6594788  0.2511055  0.2511055  1.3216259  0.5821478 -1.4333227 -0.7897473 -0.3447992  1.4889763  0.5821478</span></span>
<span id="cb277-27"><a href="basic-statistics-in-r.html#cb277-27" aria-hidden="true" tabindex="-1"></a><span class="co">#[31]  2.0305623 -0.4194885</span></span>
<span id="cb277-28"><a href="basic-statistics-in-r.html#cb277-28" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(hp_vector_boxcox, <span class="at">xlab =</span> <span class="st">&quot;hp&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;values_transformed_by_boxcox&quot;</span>)</span></code></pre></div>
<p><img src="figures/session_6_figs/5_hp_boxcox.png" width="100%" /></p>
</div>
<div id="what-test-should-i-use" class="section level2" number="6.9">
<h2><span class="header-section-number">6.9</span> What test should I use?</h2>
<p>The task of choosing the best statistical analyses given your dataset are one of the most important steps for any biological research. Your choice of methods depends on your understanding of your dataset and these include the following: (a) the number of dependent or outcome variables; (b) the number of your independent or predictor variables; (c) the characteristics of your dependent and independent variables, namely whether they are <strong>continuous (interval or ratio) variables</strong> (e.g., temperature, growth area, enzyme activity, dose amount, reaction rate, concentration, weight, length); (d) they are <strong>ordinal variables</strong> where order matters but differences between adjacent categories do not necessarily have the same meaning such as in treatment response (“low”, ”middle”, ”high”) or location (“understory”, “canopy”); (e) they are <strong>categorical variables</strong> if the categories that do not have a natural order or ranking such in colors (e.g., red, blue, green), populations (e.g., A, B, C, D); and (f) the type of distribution of the variables (it is normally distributed so parametric methods could be applied or not).</p>
<p>The table of statistical choices has been modified form <a href="https://stats.idre.ucla.edu/other/mult-pkg/whatstat/" target="_blank"><strong>here</strong></a>.</p>
<table>
<colgroup>
<col width="17%" />
<col width="15%" />
<col width="17%" />
<col width="15%" />
<col width="18%" />
<col width="14%" />
</colgroup>
<thead>
<tr class="header">
<th>Dependents (Number)</th>
<th>Dependents (Type)</th>
<th>Independents (Number)</th>
<th>Independents (Type)</th>
<th>Test</th>
<th>Section</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1 &amp; one population</td>
<td>continuous (normal)</td>
<td>N/A</td>
<td>N/A</td>
<td>one-sample t-test</td>
<td><strong>6.4</strong></td>
</tr>
<tr class="even">
<td>1 &amp; two populations</td>
<td>continuous (normal)</td>
<td>N/A</td>
<td>N/A</td>
<td>two-sample t-test</td>
<td><strong>6.4</strong></td>
</tr>
<tr class="odd">
<td>1 &amp; one population</td>
<td>continuous (non-normal); ordinal</td>
<td>N/A</td>
<td>N/A</td>
<td>one-sample median or Wilcoxon signed rank test</td>
<td><strong>6.5</strong></td>
</tr>
<tr class="even">
<td>1 &amp; two populations</td>
<td>continuous (non-normal); ordinal</td>
<td>N/A</td>
<td>N/A</td>
<td>two-sample median or Wilcoxon signed rank test</td>
<td><strong>6.5</strong></td>
</tr>
<tr class="odd">
<td>1 &amp; one population</td>
<td>categorical (2 categories)</td>
<td>N/A</td>
<td>N/A</td>
<td>binomial test</td>
<td><strong>6.6</strong></td>
</tr>
<tr class="even">
<td>1 &amp; one population</td>
<td>categorical (multiple categories)</td>
<td>N/A</td>
<td>N/A</td>
<td>Chi-square goodness-of-fit</td>
<td><strong>6.7</strong></td>
</tr>
<tr class="odd">
<td>1 &amp; two populations</td>
<td>categorical (multiple categories) compare agreements</td>
<td>N/A</td>
<td>N/A</td>
<td>Chi-square goodness-of-fit</td>
<td><strong>6.7</strong></td>
</tr>
<tr class="even">
<td>1 &amp; two populations</td>
<td>categorical (multiple categories)</td>
<td>N/A</td>
<td>N/A</td>
<td>Fisher’s exact test</td>
<td><strong>6.9</strong></td>
</tr>
<tr class="odd">
<td>1 &amp; one population</td>
<td>continuous (normal)</td>
<td>1</td>
<td>continuous (normal)</td>
<td>correlation</td>
<td><strong>6.10</strong></td>
</tr>
<tr class="even">
<td>bootstraps</td>
<td></td>
<td></td>
<td></td>
<td></td>
<td><strong>6.11</strong></td>
</tr>
<tr class="odd">
<td>1 &amp; one population</td>
<td>continuous (normal)</td>
<td>1</td>
<td>continuous (normal)</td>
<td>simple linear regression</td>
<td><strong>7.1</strong></td>
</tr>
<tr class="even">
<td>1 &amp; one population</td>
<td>continuous (normal)</td>
<td>many</td>
<td>continuous (normal)</td>
<td>multiple linear regression</td>
<td><strong>7.1</strong></td>
</tr>
<tr class="odd">
<td>1 &amp; one population</td>
<td>ordinal</td>
<td>1</td>
<td>continuous or ordinal</td>
<td>simple logistic regression</td>
<td><strong>7.3</strong></td>
</tr>
<tr class="even">
<td>1 &amp; one population</td>
<td>continuous</td>
<td>1</td>
<td>continuous or ordinal</td>
<td>non-parametric correlation</td>
<td><strong>7.4</strong></td>
</tr>
<tr class="odd">
<td>1 &amp; one population</td>
<td>continuous (normal)</td>
<td>1</td>
<td>categorical (2 or more categories)</td>
<td>one-way ANOVA</td>
<td><strong>7.7</strong></td>
</tr>
<tr class="even">
<td>1 &amp; one population</td>
<td>continuous (non-normal); ordinal</td>
<td>1</td>
<td>categorical (2 or more categories)</td>
<td>one-way ANOVA</td>
<td><strong>7.7</strong></td>
</tr>
<tr class="odd">
<td>1 &amp; one population</td>
<td>continuous (normal)</td>
<td>2</td>
<td>categorical (2 or more categories)</td>
<td>two-way ANOVA</td>
<td><strong>7.8</strong></td>
</tr>
</tbody>
</table>
<p>Here is another chart about selecting a statistial test for your project. The original source is in <a href="https://onishlab.colostate.edu/wp-content/uploads/2019/07/which_test_flowchart.png" target="_blank"><strong>here</strong></a>.</p>
<p><img src="figures/session_6_figs/6_which_test_flowchart.png" width="140%" /></p>
</div>
<div id="type-of-variables-in-dataset" class="section level2" number="6.10">
<h2><span class="header-section-number">6.10</span> Type of variables in dataset</h2>
<p>You need to determine what type of variables do you have in your dataset. This is crucial if you want to use the correct test in the table above. It is very common that students fail to explore their data and plan to
force a test on their data without knowing (i.e., exploring!!!) if the variables in your dataset are continuous, discrete,
categorical, and binomial. We can easily check variable type by checking the data structure with <code>str</code> function.</p>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="basic-statistics-in-r.html#cb278-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we can check the structure of car dataset by loading get mtcars dataset -- this a preloaded dataset in R</span></span>
<span id="cb278-2"><a href="basic-statistics-in-r.html#cb278-2" aria-hidden="true" tabindex="-1"></a>cars_data <span class="ot">&lt;-</span> mtcars</span>
<span id="cb278-3"><a href="basic-statistics-in-r.html#cb278-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb278-4"><a href="basic-statistics-in-r.html#cb278-4" aria-hidden="true" tabindex="-1"></a><span class="do">## we add the car brand as a variable brand.</span></span>
<span id="cb278-5"><a href="basic-statistics-in-r.html#cb278-5" aria-hidden="true" tabindex="-1"></a>cars_data<span class="sc">$</span>brand <span class="ot">&lt;-</span> <span class="fu">rownames</span>(cars_data)</span>
<span id="cb278-6"><a href="basic-statistics-in-r.html#cb278-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb278-7"><a href="basic-statistics-in-r.html#cb278-7" aria-hidden="true" tabindex="-1"></a><span class="do">## get the type of variable</span></span>
<span id="cb278-8"><a href="basic-statistics-in-r.html#cb278-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb278-9"><a href="basic-statistics-in-r.html#cb278-9" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(cars_data)</span>
<span id="cb278-10"><a href="basic-statistics-in-r.html#cb278-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  32 obs. of  12 variables:</span></span>
<span id="cb278-11"><a href="basic-statistics-in-r.html#cb278-11" aria-hidden="true" tabindex="-1"></a><span class="co"># $ mpg  : num  21 21 22.8 21.4 18.7 18.1 14.3 24.4 22.8 19.2 ...</span></span>
<span id="cb278-12"><a href="basic-statistics-in-r.html#cb278-12" aria-hidden="true" tabindex="-1"></a><span class="co"># $ cyl  : num  6 6 4 6 8 6 8 4 4 6 ...</span></span>
<span id="cb278-13"><a href="basic-statistics-in-r.html#cb278-13" aria-hidden="true" tabindex="-1"></a><span class="co"># $ disp : num  160 160 108 258 360 ...</span></span>
<span id="cb278-14"><a href="basic-statistics-in-r.html#cb278-14" aria-hidden="true" tabindex="-1"></a><span class="co"># $ hp   : num  110 110 93 110 175 105 245 62 95 123 ...</span></span>
<span id="cb278-15"><a href="basic-statistics-in-r.html#cb278-15" aria-hidden="true" tabindex="-1"></a><span class="co"># $ drat : num  3.9 3.9 3.85 3.08 3.15 2.76 3.21 3.69 3.92 3.92 ...</span></span>
<span id="cb278-16"><a href="basic-statistics-in-r.html#cb278-16" aria-hidden="true" tabindex="-1"></a><span class="co"># $ wt   : num  2.62 2.88 2.32 3.21 3.44 ...</span></span>
<span id="cb278-17"><a href="basic-statistics-in-r.html#cb278-17" aria-hidden="true" tabindex="-1"></a><span class="co"># $ qsec : num  16.5 17 18.6 19.4 17 ...</span></span>
<span id="cb278-18"><a href="basic-statistics-in-r.html#cb278-18" aria-hidden="true" tabindex="-1"></a><span class="co"># $ vs   : num  0 0 1 1 0 1 0 1 1 1 ...</span></span>
<span id="cb278-19"><a href="basic-statistics-in-r.html#cb278-19" aria-hidden="true" tabindex="-1"></a><span class="co"># $ am   : num  1 1 1 0 0 0 0 0 0 0 ...</span></span>
<span id="cb278-20"><a href="basic-statistics-in-r.html#cb278-20" aria-hidden="true" tabindex="-1"></a><span class="co"># $ gear : num  4 4 4 3 3 3 3 4 4 4 ...</span></span>
<span id="cb278-21"><a href="basic-statistics-in-r.html#cb278-21" aria-hidden="true" tabindex="-1"></a><span class="co"># $ carb : num  4 4 1 1 2 1 4 2 2 4 ...</span></span>
<span id="cb278-22"><a href="basic-statistics-in-r.html#cb278-22" aria-hidden="true" tabindex="-1"></a><span class="co"># $ brand: chr  &quot;Mazda RX4&quot; &quot;Mazda RX4 Wag&quot; &quot;Datsun 710&quot; &quot;Hornet 4 Drive&quot; ...</span></span>
<span id="cb278-23"><a href="basic-statistics-in-r.html#cb278-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb278-24"><a href="basic-statistics-in-r.html#cb278-24" aria-hidden="true" tabindex="-1"></a><span class="do">## we can a linear regreassion with mpg as dependent and disp, hp and wt as predictors</span></span>
<span id="cb278-25"><a href="basic-statistics-in-r.html#cb278-25" aria-hidden="true" tabindex="-1"></a>example_regression <span class="ot">&lt;-</span> <span class="fu">lm</span>(mpg <span class="sc">~</span> disp <span class="sc">+</span> hp <span class="sc">+</span> wt, <span class="at">data =</span> cars_data)</span>
<span id="cb278-26"><a href="basic-statistics-in-r.html#cb278-26" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(example_regression)</span>
<span id="cb278-27"><a href="basic-statistics-in-r.html#cb278-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Call:</span></span>
<span id="cb278-28"><a href="basic-statistics-in-r.html#cb278-28" aria-hidden="true" tabindex="-1"></a><span class="co"># lm(formula = mpg ~ disp + hp + wt, data = cars_data)</span></span>
<span id="cb278-29"><a href="basic-statistics-in-r.html#cb278-29" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb278-30"><a href="basic-statistics-in-r.html#cb278-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Residuals:</span></span>
<span id="cb278-31"><a href="basic-statistics-in-r.html#cb278-31" aria-hidden="true" tabindex="-1"></a><span class="co">#   Min     1Q Median     3Q    Max </span></span>
<span id="cb278-32"><a href="basic-statistics-in-r.html#cb278-32" aria-hidden="true" tabindex="-1"></a><span class="co"># -3.891 -1.640 -0.172  1.061  5.861 </span></span>
<span id="cb278-33"><a href="basic-statistics-in-r.html#cb278-33" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb278-34"><a href="basic-statistics-in-r.html#cb278-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Coefficients:</span></span>
<span id="cb278-35"><a href="basic-statistics-in-r.html#cb278-35" aria-hidden="true" tabindex="-1"></a><span class="co">#             Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span id="cb278-36"><a href="basic-statistics-in-r.html#cb278-36" aria-hidden="true" tabindex="-1"></a><span class="co"># (Intercept) 37.105505   2.110815  17.579  &lt; 2e-16 ***</span></span>
<span id="cb278-37"><a href="basic-statistics-in-r.html#cb278-37" aria-hidden="true" tabindex="-1"></a><span class="co"># disp        -0.000937   0.010350  -0.091  0.92851    </span></span>
<span id="cb278-38"><a href="basic-statistics-in-r.html#cb278-38" aria-hidden="true" tabindex="-1"></a><span class="co"># hp          -0.031157   0.011436  -2.724  0.01097 *  </span></span>
<span id="cb278-39"><a href="basic-statistics-in-r.html#cb278-39" aria-hidden="true" tabindex="-1"></a><span class="co"># wt          -3.800891   1.066191  -3.565  0.00133 ** </span></span>
<span id="cb278-40"><a href="basic-statistics-in-r.html#cb278-40" aria-hidden="true" tabindex="-1"></a><span class="co"># ---</span></span>
<span id="cb278-41"><a href="basic-statistics-in-r.html#cb278-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1</span></span>
<span id="cb278-42"><a href="basic-statistics-in-r.html#cb278-42" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb278-43"><a href="basic-statistics-in-r.html#cb278-43" aria-hidden="true" tabindex="-1"></a><span class="co"># Residual standard error: 2.639 on 28 degrees of freedom</span></span>
<span id="cb278-44"><a href="basic-statistics-in-r.html#cb278-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Multiple R-squared:  0.8268,  Adjusted R-squared:  0.8083 </span></span>
<span id="cb278-45"><a href="basic-statistics-in-r.html#cb278-45" aria-hidden="true" tabindex="-1"></a><span class="co"># F-statistic: 44.57 on 3 and 28 DF,  p-value: 8.65e-11</span></span>
<span id="cb278-46"><a href="basic-statistics-in-r.html#cb278-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb278-47"><a href="basic-statistics-in-r.html#cb278-47" aria-hidden="true" tabindex="-1"></a><span class="do">## we found taht hp and wt are good predictors of mpg. See regressions for more details on this test.</span></span></code></pre></div>
<p>We can see that this dataset has 12 variables, 11 of those are continuous as <code>num</code> and only one named <code>brand</code> is a discrete variable indicated as <code>chr</code>.</p>
<p>We can now test another type of datasets from the R-package <a href="https://cran.r-project.org/web/packages/MASS/index.html" target="_blank"><strong>MASS</strong></a>.</p>
<div class="sourceCode" id="cb279"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb279-1"><a href="basic-statistics-in-r.html#cb279-1" aria-hidden="true" tabindex="-1"></a><span class="co"># we can install MASS and load the dataset for this example</span></span>
<span id="cb279-2"><a href="basic-statistics-in-r.html#cb279-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;MASS&quot;</span>)</span>
<span id="cb279-3"><a href="basic-statistics-in-r.html#cb279-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb279-4"><a href="basic-statistics-in-r.html#cb279-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb279-5"><a href="basic-statistics-in-r.html#cb279-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Let&#39;s explore the Aids2 dataset </span></span>
<span id="cb279-6"><a href="basic-statistics-in-r.html#cb279-6" aria-hidden="true" tabindex="-1"></a>Aids2_raw <span class="ot">&lt;-</span> Aids2</span>
<span id="cb279-7"><a href="basic-statistics-in-r.html#cb279-7" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(Aids2_raw)</span>
<span id="cb279-8"><a href="basic-statistics-in-r.html#cb279-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  2843 obs. of  7 variables:</span></span>
<span id="cb279-9"><a href="basic-statistics-in-r.html#cb279-9" aria-hidden="true" tabindex="-1"></a><span class="co"># $ state  : Factor w/ 4 levels &quot;NSW&quot;,&quot;Other&quot;,..: 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb279-10"><a href="basic-statistics-in-r.html#cb279-10" aria-hidden="true" tabindex="-1"></a><span class="co"># $ sex    : Factor w/ 2 levels &quot;F&quot;,&quot;M&quot;: 2 2 2 2 2 2 2 2 2 2 ...</span></span>
<span id="cb279-11"><a href="basic-statistics-in-r.html#cb279-11" aria-hidden="true" tabindex="-1"></a><span class="co"># $ diag   : int  10905 11029 9551 9577 10015 9971 10746 10042 10464 10439 ...</span></span>
<span id="cb279-12"><a href="basic-statistics-in-r.html#cb279-12" aria-hidden="true" tabindex="-1"></a><span class="co"># $ death  : int  11081 11096 9983 9654 10290 10344 11135 11069 10956 10873 ...</span></span>
<span id="cb279-13"><a href="basic-statistics-in-r.html#cb279-13" aria-hidden="true" tabindex="-1"></a><span class="co"># $ status : Factor w/ 2 levels &quot;A&quot;,&quot;D&quot;: 2 2 2 2 2 2 2 2 2 2 ...</span></span>
<span id="cb279-14"><a href="basic-statistics-in-r.html#cb279-14" aria-hidden="true" tabindex="-1"></a><span class="co"># $ T.categ: Factor w/ 8 levels &quot;hs&quot;,&quot;hsid&quot;,&quot;id&quot;,..: 1 1 1 5 1 1 8 1 1 2 ...</span></span>
<span id="cb279-15"><a href="basic-statistics-in-r.html#cb279-15" aria-hidden="true" tabindex="-1"></a><span class="co"># $ age    : int  35 53 42 44 39 36 36 31 26 27 ..</span></span>
<span id="cb279-16"><a href="basic-statistics-in-r.html#cb279-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb279-17"><a href="basic-statistics-in-r.html#cb279-17" aria-hidden="true" tabindex="-1"></a><span class="co"># we can get the corresponding information for each with ?</span></span>
<span id="cb279-18"><a href="basic-statistics-in-r.html#cb279-18" aria-hidden="true" tabindex="-1"></a>?Aids2</span></code></pre></div>
<p>We can see 7 variables, 3 are continuous as <code>int</code> (for integer) including two that are
not very useful as dates including <code>diag</code> and <code>death</code>. The variable <code>age</code> corresponds to age at diagnosis.
Two variables could be binary and also coded as factors with 2-levels including <code>sex</code> (i.e., male or female)
and <code>status</code> (i.e., alive or dead). Two are categorical such as <code>state</code> (i.e., four states in Australia) and
<code>T.categ</code> has 8 categories for reported transmission category.</p>
<p>We can do some exploration and even a t-test for the continuous variable</p>
<div class="sourceCode" id="cb280"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb280-1"><a href="basic-statistics-in-r.html#cb280-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we can get the summary of the age category</span></span>
<span id="cb280-2"><a href="basic-statistics-in-r.html#cb280-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(Aids2_raw<span class="sc">$</span>age)</span>
<span id="cb280-3"><a href="basic-statistics-in-r.html#cb280-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span id="cb280-4"><a href="basic-statistics-in-r.html#cb280-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   0.00   30.00   37.00   37.41   43.00   82.00</span></span>
<span id="cb280-5"><a href="basic-statistics-in-r.html#cb280-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb280-6"><a href="basic-statistics-in-r.html#cb280-6" aria-hidden="true" tabindex="-1"></a><span class="do">## we can test if 35 years is the mean of age of diagnosis of this population</span></span>
<span id="cb280-7"><a href="basic-statistics-in-r.html#cb280-7" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(Aids2_raw<span class="sc">$</span>age, <span class="at">mu =</span> <span class="dv">35</span>)</span>
<span id="cb280-8"><a href="basic-statistics-in-r.html#cb280-8" aria-hidden="true" tabindex="-1"></a><span class="co"># One Sample t-test</span></span>
<span id="cb280-9"><a href="basic-statistics-in-r.html#cb280-9" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb280-10"><a href="basic-statistics-in-r.html#cb280-10" aria-hidden="true" tabindex="-1"></a><span class="co"># data:  Aids2_raw$age</span></span>
<span id="cb280-11"><a href="basic-statistics-in-r.html#cb280-11" aria-hidden="true" tabindex="-1"></a><span class="co"># t = 12.764, df = 2842, p-value &lt; 2.2e-16</span></span>
<span id="cb280-12"><a href="basic-statistics-in-r.html#cb280-12" aria-hidden="true" tabindex="-1"></a><span class="co"># alternative hypothesis: true mean is not equal to 35</span></span>
<span id="cb280-13"><a href="basic-statistics-in-r.html#cb280-13" aria-hidden="true" tabindex="-1"></a><span class="co"># 95 percent confidence interval:</span></span>
<span id="cb280-14"><a href="basic-statistics-in-r.html#cb280-14" aria-hidden="true" tabindex="-1"></a><span class="co"># 37.03901 37.77914</span></span>
<span id="cb280-15"><a href="basic-statistics-in-r.html#cb280-15" aria-hidden="true" tabindex="-1"></a><span class="co"># sample estimates:</span></span>
<span id="cb280-16"><a href="basic-statistics-in-r.html#cb280-16" aria-hidden="true" tabindex="-1"></a><span class="co"># mean of x </span></span>
<span id="cb280-17"><a href="basic-statistics-in-r.html#cb280-17" aria-hidden="true" tabindex="-1"></a><span class="co"># 37.40907 </span></span>
<span id="cb280-18"><a href="basic-statistics-in-r.html#cb280-18" aria-hidden="true" tabindex="-1"></a><span class="do">## we rejected the hypothesis that the age mean value of diagnosis is 35. See t-test for more details on this test.</span></span></code></pre></div>
<p>We can continue our exploration and test if males and females are equally diagnosed with AIDS.</p>
<div class="sourceCode" id="cb281"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb281-1"><a href="basic-statistics-in-r.html#cb281-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we can get the summary of the sex category</span></span>
<span id="cb281-2"><a href="basic-statistics-in-r.html#cb281-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(Aids2_raw<span class="sc">$</span>sex)</span>
<span id="cb281-3"><a href="basic-statistics-in-r.html#cb281-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   F    M </span></span>
<span id="cb281-4"><a href="basic-statistics-in-r.html#cb281-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  89 2754 </span></span>
<span id="cb281-5"><a href="basic-statistics-in-r.html#cb281-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb281-6"><a href="basic-statistics-in-r.html#cb281-6" aria-hidden="true" tabindex="-1"></a><span class="do">## we need to convert it first to a numeric vector and then to a table</span></span>
<span id="cb281-7"><a href="basic-statistics-in-r.html#cb281-7" aria-hidden="true" tabindex="-1"></a>sex_as_numeric <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(Aids2_raw<span class="sc">$</span>sex)</span>
<span id="cb281-8"><a href="basic-statistics-in-r.html#cb281-8" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(sex_as_numeric)</span>
<span id="cb281-9"><a href="basic-statistics-in-r.html#cb281-9" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;numeric&quot;</span></span>
<span id="cb281-10"><a href="basic-statistics-in-r.html#cb281-10" aria-hidden="true" tabindex="-1"></a>sex_as_table <span class="ot">&lt;-</span> <span class="fu">table</span>(sex_as_numeric)</span>
<span id="cb281-11"><a href="basic-statistics-in-r.html#cb281-11" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sex_as_table) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;female&quot;</span>,<span class="st">&quot;male&quot;</span>)</span>
<span id="cb281-12"><a href="basic-statistics-in-r.html#cb281-12" aria-hidden="true" tabindex="-1"></a>sex_as_table</span>
<span id="cb281-13"><a href="basic-statistics-in-r.html#cb281-13" aria-hidden="true" tabindex="-1"></a><span class="co">#female   male </span></span>
<span id="cb281-14"><a href="basic-statistics-in-r.html#cb281-14" aria-hidden="true" tabindex="-1"></a><span class="co">#    89   2754</span></span>
<span id="cb281-15"><a href="basic-statistics-in-r.html#cb281-15" aria-hidden="true" tabindex="-1"></a><span class="do">## we can test if males and females are equally infected (50/50).</span></span>
<span id="cb281-16"><a href="basic-statistics-in-r.html#cb281-16" aria-hidden="true" tabindex="-1"></a><span class="fu">chisq.test</span>(sex_as_table, <span class="at">p =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">2</span>))</span>
<span id="cb281-17"><a href="basic-statistics-in-r.html#cb281-17" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb281-18"><a href="basic-statistics-in-r.html#cb281-18" aria-hidden="true" tabindex="-1"></a><span class="co">#   Chi-squared test for given probabilities</span></span>
<span id="cb281-19"><a href="basic-statistics-in-r.html#cb281-19" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb281-20"><a href="basic-statistics-in-r.html#cb281-20" aria-hidden="true" tabindex="-1"></a><span class="co"># data:  sex_as_table</span></span>
<span id="cb281-21"><a href="basic-statistics-in-r.html#cb281-21" aria-hidden="true" tabindex="-1"></a><span class="co"># X-squared = 2498.1, df = 1, p-value &lt; 2.2e-16</span></span>
<span id="cb281-22"><a href="basic-statistics-in-r.html#cb281-22" aria-hidden="true" tabindex="-1"></a><span class="do">## we rejected the hypothesis that the both males and females are affected in the same proportion (i.e., 1/2 and 1/2). See Chi-squared test for more details.</span></span></code></pre></div>
</div>
<div id="comparing-means-t-test" class="section level2" number="6.11">
<h2><span class="header-section-number">6.11</span> Comparing means: t-test</h2>
<p><strong>9)</strong> We can perform one sample <a href="https://en.wikipedia.org/wiki/Student%27s_t-test" target="_blank"><strong>t-test</strong></a> on vectors using the function <code>t.test()</code>. For this test, you can test if its mean is statistically different from a given value specified by the <code>mu</code> argument equal such value.</p>
<div class="sourceCode" id="cb282"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb282-1"><a href="basic-statistics-in-r.html#cb282-1" aria-hidden="true" tabindex="-1"></a><span class="co"># we will test the mean of sample SRR1039508 equal to mu = 546.5 for </span></span>
<span id="cb282-2"><a href="basic-statistics-in-r.html#cb282-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(cars_data)[<span class="dv">7</span>]</span>
<span id="cb282-3"><a href="basic-statistics-in-r.html#cb282-3" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">&quot;qsec&quot;</span></span>
<span id="cb282-4"><a href="basic-statistics-in-r.html#cb282-4" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(cars_data<span class="sc">$</span>qsec, <span class="at">mu =</span> <span class="dv">11</span>)</span>
<span id="cb282-5"><a href="basic-statistics-in-r.html#cb282-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb282-6"><a href="basic-statistics-in-r.html#cb282-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   One Sample t-test</span></span>
<span id="cb282-7"><a href="basic-statistics-in-r.html#cb282-7" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb282-8"><a href="basic-statistics-in-r.html#cb282-8" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  cars_data$qsec</span></span>
<span id="cb282-9"><a href="basic-statistics-in-r.html#cb282-9" aria-hidden="true" tabindex="-1"></a><span class="co">#t = 21.681, df = 31, p-value &lt; 2.2e-16</span></span>
<span id="cb282-10"><a href="basic-statistics-in-r.html#cb282-10" aria-hidden="true" tabindex="-1"></a><span class="co">#alternative hypothesis: true mean is not equal to 11</span></span>
<span id="cb282-11"><a href="basic-statistics-in-r.html#cb282-11" aria-hidden="true" tabindex="-1"></a><span class="co">#95 percent confidence interval:</span></span>
<span id="cb282-12"><a href="basic-statistics-in-r.html#cb282-12" aria-hidden="true" tabindex="-1"></a><span class="co"># 17.20449 18.49301</span></span>
<span id="cb282-13"><a href="basic-statistics-in-r.html#cb282-13" aria-hidden="true" tabindex="-1"></a><span class="co">#sample estimates:</span></span>
<span id="cb282-14"><a href="basic-statistics-in-r.html#cb282-14" aria-hidden="true" tabindex="-1"></a><span class="co">#mean of x </span></span>
<span id="cb282-15"><a href="basic-statistics-in-r.html#cb282-15" aria-hidden="true" tabindex="-1"></a><span class="co">#17.84875 </span></span>
<span id="cb282-16"><a href="basic-statistics-in-r.html#cb282-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb282-17"><a href="basic-statistics-in-r.html#cb282-17" aria-hidden="true" tabindex="-1"></a><span class="co"># we will test the mean of sample SRR1039508 equal to mu = 546.5 for </span></span>
<span id="cb282-18"><a href="basic-statistics-in-r.html#cb282-18" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(airway_data)[<span class="dv">2</span>]</span>
<span id="cb282-19"><a href="basic-statistics-in-r.html#cb282-19" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;SRR1039508&quot;</span></span>
<span id="cb282-20"><a href="basic-statistics-in-r.html#cb282-20" aria-hidden="true" tabindex="-1"></a><span class="fu">t.test</span>(airway_data[,<span class="dv">2</span>], <span class="at">mu =</span> <span class="fl">546.5</span>)</span>
<span id="cb282-21"><a href="basic-statistics-in-r.html#cb282-21" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb282-22"><a href="basic-statistics-in-r.html#cb282-22" aria-hidden="true" tabindex="-1"></a><span class="co">#   One Sample t-test</span></span>
<span id="cb282-23"><a href="basic-statistics-in-r.html#cb282-23" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb282-24"><a href="basic-statistics-in-r.html#cb282-24" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  airway_data[, 2]</span></span>
<span id="cb282-25"><a href="basic-statistics-in-r.html#cb282-25" aria-hidden="true" tabindex="-1"></a><span class="co">#t = -0.00053579, df = 38693, p-value = 0.9996</span></span>
<span id="cb282-26"><a href="basic-statistics-in-r.html#cb282-26" aria-hidden="true" tabindex="-1"></a><span class="co">#alternative hypothesis: true mean is not equal to 546.5</span></span>
<span id="cb282-27"><a href="basic-statistics-in-r.html#cb282-27" aria-hidden="true" tabindex="-1"></a><span class="co">#95 percent confidence interval:</span></span>
<span id="cb282-28"><a href="basic-statistics-in-r.html#cb282-28" aria-hidden="true" tabindex="-1"></a><span class="co"># 501.7696 591.2060</span></span>
<span id="cb282-29"><a href="basic-statistics-in-r.html#cb282-29" aria-hidden="true" tabindex="-1"></a><span class="co">#sample estimates:</span></span>
<span id="cb282-30"><a href="basic-statistics-in-r.html#cb282-30" aria-hidden="true" tabindex="-1"></a><span class="co">#mean of x </span></span>
<span id="cb282-31"><a href="basic-statistics-in-r.html#cb282-31" aria-hidden="true" tabindex="-1"></a><span class="co"># 546.4878 </span></span></code></pre></div>
<p>These result indicate that the mean of <code>qsec</code> from the <code>cars_data</code> is statically different form 11, while the mean for sample <code>SRR1039508</code> from the <code>airway_data</code> is not different from the value <code>546.6</code> (this value was derived from the summary see <strong>2)</strong>). One sample t-test does not assume that the population is normally distributed. However, it is better to have a normalized distribution in our data when we apply parametric test like this one.</p>
<p>The output of the function <code>t.test()</code> also provides 95% confidence interval of the mean value. In these examples, <code>qsec</code> has a mean 17.84875 with 95CI (17.20449, 18.49301) and the <code>SRR1039508</code> sample mean is 546.4878 with 95CI (501.7696, 591.2060). Our interpretation is that the true value for the mean of the measurements of each of these variables is in the proposed range with a 95% confidence level.</p>
</div>
<div id="two-sample-t-test" class="section level2" number="6.12">
<h2><span class="header-section-number">6.12</span> Two sample t-test</h2>
<p><strong>10)</strong> For two samples, we can also perform t-tests with two vectors using the same function <code>t.test()</code>. In this case, you are testing if two samples each from different populations (e.g., control and experimental) have the same mean. If the test is significant, you can conclude that the difference might derive from a treatment (as long as you have controlled for other causal agents). These t-tests can be consider as paired (i.e., argument <code>paired = TRUE</code>) or unpaired (i.e., argument <code>paired = FALSE</code>). For paired, the t-test will compare the means of the same group under two separate conditions. In contrast, an unpaired t-test compares the means of two independent groups. In an unpaired t-test, the variance between groups is assumed to be equal.</p>
<p>We will test this for <code>airway_data</code> data that have several samples from the same group the argument <code>paired = TRUE</code>.</p>
<div class="sourceCode" id="cb283"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb283-1"><a href="basic-statistics-in-r.html#cb283-1" aria-hidden="true" tabindex="-1"></a>names_pairs <span class="ot">&lt;-</span> <span class="fu">names</span>(airway_data)[<span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb283-2"><a href="basic-statistics-in-r.html#cb283-2" aria-hidden="true" tabindex="-1"></a>names_pairs <span class="ot">&lt;-</span> <span class="fu">paste0</span>(names_pairs, <span class="at">collapse =</span> <span class="st">&quot;..vs..&quot;</span>)</span>
<span id="cb283-3"><a href="basic-statistics-in-r.html#cb283-3" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;SRR1039508..vs..SRR1039509&quot;</span></span>
<span id="cb283-4"><a href="basic-statistics-in-r.html#cb283-4" aria-hidden="true" tabindex="-1"></a>out_put_ttest_paired <span class="ot">&lt;-</span> <span class="fu">t.test</span>(airway_data[,<span class="dv">2</span>], airway_data[,<span class="dv">3</span>], <span class="at">paired =</span> <span class="cn">TRUE</span>)</span>
<span id="cb283-5"><a href="basic-statistics-in-r.html#cb283-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(names_pairs); <span class="fu">print</span>(out_put_ttest_paired)</span>
<span id="cb283-6"><a href="basic-statistics-in-r.html#cb283-6" aria-hidden="true" tabindex="-1"></a><span class="co">#SRR1039508..vs..SRR1039509</span></span>
<span id="cb283-7"><a href="basic-statistics-in-r.html#cb283-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   Paired t-test</span></span>
<span id="cb283-8"><a href="basic-statistics-in-r.html#cb283-8" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb283-9"><a href="basic-statistics-in-r.html#cb283-9" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  airway_data[, 2] and airway_data[, 3]</span></span>
<span id="cb283-10"><a href="basic-statistics-in-r.html#cb283-10" aria-hidden="true" tabindex="-1"></a><span class="co">#t = 6.9876, df = 38693, p-value = 2.842e-12</span></span>
<span id="cb283-11"><a href="basic-statistics-in-r.html#cb283-11" aria-hidden="true" tabindex="-1"></a><span class="co">#alternative hypothesis: true difference in means is not equal to 0</span></span>
<span id="cb283-12"><a href="basic-statistics-in-r.html#cb283-12" aria-hidden="true" tabindex="-1"></a><span class="co">#95 percent confidence interval:</span></span>
<span id="cb283-13"><a href="basic-statistics-in-r.html#cb283-13" aria-hidden="true" tabindex="-1"></a><span class="co"># 32.66123 58.12758</span></span>
<span id="cb283-14"><a href="basic-statistics-in-r.html#cb283-14" aria-hidden="true" tabindex="-1"></a><span class="co">#sample estimates:</span></span>
<span id="cb283-15"><a href="basic-statistics-in-r.html#cb283-15" aria-hidden="true" tabindex="-1"></a><span class="co">#mean of the differences </span></span>
<span id="cb283-16"><a href="basic-statistics-in-r.html#cb283-16" aria-hidden="true" tabindex="-1"></a><span class="co">#               45.3944 </span></span></code></pre></div>
<p>The above results suggest that there is strong difference between these two samples of the same group.</p>
<p>Now, we will also consider the case of unpaired for comparison.</p>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="basic-statistics-in-r.html#cb284-1" aria-hidden="true" tabindex="-1"></a>out_put_ttest_unpaired <span class="ot">&lt;-</span> <span class="fu">t.test</span>(airway_data[,<span class="dv">2</span>], airway_data[,<span class="dv">3</span>], <span class="at">paired =</span> <span class="cn">FALSE</span>)</span>
<span id="cb284-2"><a href="basic-statistics-in-r.html#cb284-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(names_pairs); <span class="fu">print</span>(out_put_ttest_unpaired)</span>
<span id="cb284-3"><a href="basic-statistics-in-r.html#cb284-3" aria-hidden="true" tabindex="-1"></a><span class="co">#SRR1039508..vs..SRR1039509</span></span>
<span id="cb284-4"><a href="basic-statistics-in-r.html#cb284-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   Welch Two Sample t-test</span></span>
<span id="cb284-5"><a href="basic-statistics-in-r.html#cb284-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb284-6"><a href="basic-statistics-in-r.html#cb284-6" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  airway_data[, 2] and airway_data[, 3]</span></span>
<span id="cb284-7"><a href="basic-statistics-in-r.html#cb284-7" aria-hidden="true" tabindex="-1"></a><span class="co">#t = 1.5108, df = 75614, p-value = 0.1309</span></span>
<span id="cb284-8"><a href="basic-statistics-in-r.html#cb284-8" aria-hidden="true" tabindex="-1"></a><span class="co">#alternative hypothesis: true difference in means is not equal to 0</span></span>
<span id="cb284-9"><a href="basic-statistics-in-r.html#cb284-9" aria-hidden="true" tabindex="-1"></a><span class="co">#95 percent confidence interval:</span></span>
<span id="cb284-10"><a href="basic-statistics-in-r.html#cb284-10" aria-hidden="true" tabindex="-1"></a><span class="co"># -13.49803 104.28683</span></span>
<span id="cb284-11"><a href="basic-statistics-in-r.html#cb284-11" aria-hidden="true" tabindex="-1"></a><span class="co">#sample estimates:</span></span>
<span id="cb284-12"><a href="basic-statistics-in-r.html#cb284-12" aria-hidden="true" tabindex="-1"></a><span class="co">#mean of x mean of y </span></span>
<span id="cb284-13"><a href="basic-statistics-in-r.html#cb284-13" aria-hidden="true" tabindex="-1"></a><span class="co"># 546.4878  501.0934 </span></span></code></pre></div>
<p>In this case, the result is the opposite and there is no evidence of a difference in the mean values if these samples were considered of different groups.</p>
<p>Note that a careful determination of what parametrization of the t-test is determined by your experimental group and other considerations. Likewise, a t-test is invalid for small samples from non-normal distributions, but it is valid for large samples from non-normal distributions (like our example above with &gt;30,000 entries). These considerations are beyond this bioinformatics workshop and are the topic of a biostatistics course.</p>
</div>
<div id="beyond-t-test-wilcoxon-signed-rank-test" class="section level2" number="6.13">
<h2><span class="header-section-number">6.13</span> Beyond t-test: Wilcoxon signed rank test</h2>
<p><strong>11)</strong> Alternatives to the parametric t-test exist, one of the most common is the one/two-sample median test or the <a href="https://en.wikipedia.org/wiki/Wilcoxon_signed-rank_test" target="_blank"><strong>Wilcoxon signed rank test</strong></a>. This test is used to compare two unpaired and paired samples or repeated measurements on a single sample to assess whether their population mean ranks differ.</p>
<p>The Wilcoxon signed rank test is implemented with the function <code>wilcox.test()</code> in a similar fashion as is in a <code>t.test()</code>.</p>
<div class="sourceCode" id="cb285"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb285-1"><a href="basic-statistics-in-r.html#cb285-1" aria-hidden="true" tabindex="-1"></a><span class="co"># test for mean of sample</span></span>
<span id="cb285-2"><a href="basic-statistics-in-r.html#cb285-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(airway_data[,<span class="dv">2</span>], <span class="at">mu =</span> <span class="fl">546.5</span>)</span>
<span id="cb285-3"><a href="basic-statistics-in-r.html#cb285-3" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb285-4"><a href="basic-statistics-in-r.html#cb285-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   Wilcoxon signed rank test with continuity correction</span></span>
<span id="cb285-5"><a href="basic-statistics-in-r.html#cb285-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb285-6"><a href="basic-statistics-in-r.html#cb285-6" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  airway_data[, 2]</span></span>
<span id="cb285-7"><a href="basic-statistics-in-r.html#cb285-7" aria-hidden="true" tabindex="-1"></a><span class="co">#V = 145919557, p-value &lt; 2.2e-16</span></span>
<span id="cb285-8"><a href="basic-statistics-in-r.html#cb285-8" aria-hidden="true" tabindex="-1"></a><span class="co">#alternative hypothesis: true location is not equal to 546.5</span></span>
<span id="cb285-9"><a href="basic-statistics-in-r.html#cb285-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb285-10"><a href="basic-statistics-in-r.html#cb285-10" aria-hidden="true" tabindex="-1"></a><span class="co"># test for two samples of different group</span></span>
<span id="cb285-11"><a href="basic-statistics-in-r.html#cb285-11" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(airway_data[,<span class="dv">2</span>], airway_data[,<span class="dv">3</span>], <span class="at">paired =</span> <span class="cn">FALSE</span>)</span>
<span id="cb285-12"><a href="basic-statistics-in-r.html#cb285-12" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb285-13"><a href="basic-statistics-in-r.html#cb285-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   Wilcoxon rank sum test with continuity correction</span></span>
<span id="cb285-14"><a href="basic-statistics-in-r.html#cb285-14" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb285-15"><a href="basic-statistics-in-r.html#cb285-15" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  airway_data[, 2] and airway_data[, 3]</span></span>
<span id="cb285-16"><a href="basic-statistics-in-r.html#cb285-16" aria-hidden="true" tabindex="-1"></a><span class="co">#W = 753579582, p-value = 0.08979</span></span>
<span id="cb285-17"><a href="basic-statistics-in-r.html#cb285-17" aria-hidden="true" tabindex="-1"></a><span class="co">#alternative hypothesis: true location shift is not equal to 0</span></span>
<span id="cb285-18"><a href="basic-statistics-in-r.html#cb285-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb285-19"><a href="basic-statistics-in-r.html#cb285-19" aria-hidden="true" tabindex="-1"></a><span class="co"># test for two samples of same group, paired-test</span></span>
<span id="cb285-20"><a href="basic-statistics-in-r.html#cb285-20" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(airway_data[,<span class="dv">2</span>], airway_data[,<span class="dv">3</span>], <span class="at">paired =</span> <span class="cn">TRUE</span>)</span>
<span id="cb285-21"><a href="basic-statistics-in-r.html#cb285-21" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb285-22"><a href="basic-statistics-in-r.html#cb285-22" aria-hidden="true" tabindex="-1"></a><span class="co">#   Wilcoxon signed rank test with continuity correction</span></span>
<span id="cb285-23"><a href="basic-statistics-in-r.html#cb285-23" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb285-24"><a href="basic-statistics-in-r.html#cb285-24" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  airway_data[, 2] and airway_data[, 3]</span></span>
<span id="cb285-25"><a href="basic-statistics-in-r.html#cb285-25" aria-hidden="true" tabindex="-1"></a><span class="co">#V = 147963112, p-value &lt; 2.2e-16</span></span>
<span id="cb285-26"><a href="basic-statistics-in-r.html#cb285-26" aria-hidden="true" tabindex="-1"></a><span class="co">#alternative hypothesis: true location shift is not equal to 0</span></span></code></pre></div>
</div>
<div id="beyond-t-test-binomial-test" class="section level2" number="6.14">
<h2><span class="header-section-number">6.14</span> Beyond t-test: Binomial test</h2>
<p><strong>12)</strong> The <a href="https://en.wikipedia.org/wiki/Binomial_test" target="_blank"><strong>binomial test</strong></a> will determine the significance of deviations from expected distribution of observations into two groups of categorical variable. One common use of this test is when an experiment has two possible outcomes (i.e. success/failure) and you have an expectation (i.e., probability) success. For example, if you flip fair coin you have expectation of 50/50 chance of heads of tail every flip. You can apply a binomial test to a population with two categories using the function <code>prop.test()</code>.</p>
<p>For example, we can test if the number of females in a population is <code>p = 0.51</code> (i.e., the chance of being female is 51%). We will use function <code>rbinom()</code> to create such population of 1000 individuals.</p>
<div class="sourceCode" id="cb286"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb286-1"><a href="basic-statistics-in-r.html#cb286-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we create a binomial variable assigned to males of females.</span></span>
<span id="cb286-2"><a href="basic-statistics-in-r.html#cb286-2" aria-hidden="true" tabindex="-1"></a><span class="do">## we assume that if female is 1 and 0 if male</span></span>
<span id="cb286-3"><a href="basic-statistics-in-r.html#cb286-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb286-4"><a href="basic-statistics-in-r.html#cb286-4" aria-hidden="true" tabindex="-1"></a>one_population <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(<span class="at">n =</span> <span class="dv">1000</span>, <span class="at">size =</span> <span class="dv">1</span>, <span class="at">prob=</span> <span class="fl">0.51</span>)</span>
<span id="cb286-5"><a href="basic-statistics-in-r.html#cb286-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(one_population)</span>
<span id="cb286-6"><a href="basic-statistics-in-r.html#cb286-6" aria-hidden="true" tabindex="-1"></a><span class="co">#int [1:1000] 0 0 0 0 1 0 1 0 0 1 ...</span></span>
<span id="cb286-7"><a href="basic-statistics-in-r.html#cb286-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb286-8"><a href="basic-statistics-in-r.html#cb286-8" aria-hidden="true" tabindex="-1"></a><span class="do">## you can tabulate the data in your one_population to visualize that the data follows </span></span>
<span id="cb286-9"><a href="basic-statistics-in-r.html#cb286-9" aria-hidden="true" tabindex="-1"></a><span class="do">## roughly the predicted proportion of 0-state (0.49 or 49%) and 1-state (0.51 or 51%)</span></span>
<span id="cb286-10"><a href="basic-statistics-in-r.html#cb286-10" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(one_population)</span>
<span id="cb286-11"><a href="basic-statistics-in-r.html#cb286-11" aria-hidden="true" tabindex="-1"></a><span class="co">#one_population</span></span>
<span id="cb286-12"><a href="basic-statistics-in-r.html#cb286-12" aria-hidden="true" tabindex="-1"></a><span class="co">#  0   1 </span></span>
<span id="cb286-13"><a href="basic-statistics-in-r.html#cb286-13" aria-hidden="true" tabindex="-1"></a><span class="co">#487 513 </span></span>
<span id="cb286-14"><a href="basic-statistics-in-r.html#cb286-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb286-15"><a href="basic-statistics-in-r.html#cb286-15" aria-hidden="true" tabindex="-1"></a><span class="do">## we can perform the bionomial test</span></span>
<span id="cb286-16"><a href="basic-statistics-in-r.html#cb286-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb286-17"><a href="basic-statistics-in-r.html#cb286-17" aria-hidden="true" tabindex="-1"></a><span class="fu">prop.test</span>(<span class="fu">sum</span>(one_population), <span class="fu">length</span>(one_population), <span class="at">p =</span> <span class="fl">0.51</span>)</span>
<span id="cb286-18"><a href="basic-statistics-in-r.html#cb286-18" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb286-19"><a href="basic-statistics-in-r.html#cb286-19" aria-hidden="true" tabindex="-1"></a><span class="co">#   1-sample proportions test with continuity correction</span></span>
<span id="cb286-20"><a href="basic-statistics-in-r.html#cb286-20" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb286-21"><a href="basic-statistics-in-r.html#cb286-21" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  sum(one_population) out of length(one_population), null probability 0.51</span></span>
<span id="cb286-22"><a href="basic-statistics-in-r.html#cb286-22" aria-hidden="true" tabindex="-1"></a><span class="co">#X-squared = 0.04902, df = 1, p-value = 0.8248</span></span>
<span id="cb286-23"><a href="basic-statistics-in-r.html#cb286-23" aria-hidden="true" tabindex="-1"></a><span class="co">#alternative hypothesis: true p is not equal to 0.51</span></span>
<span id="cb286-24"><a href="basic-statistics-in-r.html#cb286-24" aria-hidden="true" tabindex="-1"></a><span class="co">#95 percent confidence interval:</span></span>
<span id="cb286-25"><a href="basic-statistics-in-r.html#cb286-25" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.4745504 0.5374029</span></span>
<span id="cb286-26"><a href="basic-statistics-in-r.html#cb286-26" aria-hidden="true" tabindex="-1"></a><span class="co">#sample estimates:</span></span>
<span id="cb286-27"><a href="basic-statistics-in-r.html#cb286-27" aria-hidden="true" tabindex="-1"></a><span class="co">#    p </span></span>
<span id="cb286-28"><a href="basic-statistics-in-r.html#cb286-28" aria-hidden="true" tabindex="-1"></a><span class="co">#0.506 </span></span></code></pre></div>
<p>In the case above, we found a non-significant result <code>p-value = 0.8248</code> and this population is likely to have 51% chance of females. However, less create a biased population towards males.</p>
<div class="sourceCode" id="cb287"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb287-1"><a href="basic-statistics-in-r.html#cb287-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we create a binomial variable assigned to males of females.</span></span>
<span id="cb287-2"><a href="basic-statistics-in-r.html#cb287-2" aria-hidden="true" tabindex="-1"></a><span class="do">## we assume that if female is 1 and 0 if male</span></span>
<span id="cb287-3"><a href="basic-statistics-in-r.html#cb287-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb287-4"><a href="basic-statistics-in-r.html#cb287-4" aria-hidden="true" tabindex="-1"></a>other_population <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(<span class="at">n =</span> <span class="dv">1000</span>, <span class="at">size =</span> <span class="dv">1</span>, <span class="at">prob=</span> <span class="fl">0.40</span>)</span>
<span id="cb287-5"><a href="basic-statistics-in-r.html#cb287-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(other_population)</span>
<span id="cb287-6"><a href="basic-statistics-in-r.html#cb287-6" aria-hidden="true" tabindex="-1"></a><span class="co">#int [1:1000] 0 0 0 0 1 1 0 1 0 0 ...</span></span>
<span id="cb287-7"><a href="basic-statistics-in-r.html#cb287-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb287-8"><a href="basic-statistics-in-r.html#cb287-8" aria-hidden="true" tabindex="-1"></a><span class="do">## you can tabulate the data in your one_population to visualize that the data follows </span></span>
<span id="cb287-9"><a href="basic-statistics-in-r.html#cb287-9" aria-hidden="true" tabindex="-1"></a><span class="do">## roughly the predicted proportion of 0-state (0.60 or 60%) and 1-state (0.40 or 40%)</span></span>
<span id="cb287-10"><a href="basic-statistics-in-r.html#cb287-10" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(other_population)</span>
<span id="cb287-11"><a href="basic-statistics-in-r.html#cb287-11" aria-hidden="true" tabindex="-1"></a><span class="co">#other_population</span></span>
<span id="cb287-12"><a href="basic-statistics-in-r.html#cb287-12" aria-hidden="true" tabindex="-1"></a><span class="co">#  0   1 </span></span>
<span id="cb287-13"><a href="basic-statistics-in-r.html#cb287-13" aria-hidden="true" tabindex="-1"></a><span class="co">#603 397 </span></span>
<span id="cb287-14"><a href="basic-statistics-in-r.html#cb287-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb287-15"><a href="basic-statistics-in-r.html#cb287-15" aria-hidden="true" tabindex="-1"></a><span class="do">## we can perform the binomial test</span></span>
<span id="cb287-16"><a href="basic-statistics-in-r.html#cb287-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb287-17"><a href="basic-statistics-in-r.html#cb287-17" aria-hidden="true" tabindex="-1"></a><span class="fu">prop.test</span>(<span class="fu">sum</span>(other_population), <span class="fu">length</span>(other_population), <span class="at">p =</span> <span class="fl">0.51</span>)</span>
<span id="cb287-18"><a href="basic-statistics-in-r.html#cb287-18" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb287-19"><a href="basic-statistics-in-r.html#cb287-19" aria-hidden="true" tabindex="-1"></a><span class="co">#   1-sample proportions test with continuity correction</span></span>
<span id="cb287-20"><a href="basic-statistics-in-r.html#cb287-20" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb287-21"><a href="basic-statistics-in-r.html#cb287-21" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  sum(other_population) out of length(other_population), null probability 0.51</span></span>
<span id="cb287-22"><a href="basic-statistics-in-r.html#cb287-22" aria-hidden="true" tabindex="-1"></a><span class="co">#X-squared = 45.387, df = 1, p-value = 1.617e-11</span></span>
<span id="cb287-23"><a href="basic-statistics-in-r.html#cb287-23" aria-hidden="true" tabindex="-1"></a><span class="co">#alternative hypothesis: true p is not equal to 0.51</span></span>
<span id="cb287-24"><a href="basic-statistics-in-r.html#cb287-24" aria-hidden="true" tabindex="-1"></a><span class="co">#95 percent confidence interval:</span></span>
<span id="cb287-25"><a href="basic-statistics-in-r.html#cb287-25" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.3725342 0.4342204</span></span>
<span id="cb287-26"><a href="basic-statistics-in-r.html#cb287-26" aria-hidden="true" tabindex="-1"></a><span class="co">#sample estimates:</span></span>
<span id="cb287-27"><a href="basic-statistics-in-r.html#cb287-27" aria-hidden="true" tabindex="-1"></a><span class="co">#    p </span></span>
<span id="cb287-28"><a href="basic-statistics-in-r.html#cb287-28" aria-hidden="true" tabindex="-1"></a><span class="co">#0.403</span></span></code></pre></div>
<p>In the case above, we found a significant result <code>p-value = 1.617e-11</code> and this population unlikely has 51% chance of females.</p>
</div>
<div id="beyond-t-test-chi-square-goodness-of-fit" class="section level2" number="6.15">
<h2><span class="header-section-number">6.15</span> Beyond t-test: Chi-square goodness of fit</h2>
<p><strong>13)</strong> The <a href="https://en.wikipedia.org/wiki/Chi-squared_test" target="_blank"><strong>Chi-square goodness of fit test</strong>*</a> will determine the significance of deviations from expected probabilities of multiple categories in a population. For example, you might want to compare the observed distribution of categories of a population to an expected distribution of such categories. We will perform this test using a table object with multiple categories with the function <code>chisq.test()</code> and an argument of probabilities for each category <code>p</code>.</p>
<p>For example, we can test if the proportion of nitrogen fixing bacteria in a population collected of dataset <code>rhizobium</code> from R package <code>ade4</code>.</p>
<div class="sourceCode" id="cb288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb288-1"><a href="basic-statistics-in-r.html#cb288-1" aria-hidden="true" tabindex="-1"></a><span class="do">## load rhizobium dataset from ade4 R package</span></span>
<span id="cb288-2"><a href="basic-statistics-in-r.html#cb288-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb288-3"><a href="basic-statistics-in-r.html#cb288-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(rhizobium)</span>
<span id="cb288-4"><a href="basic-statistics-in-r.html#cb288-4" aria-hidden="true" tabindex="-1"></a>rhizobium_pop <span class="ot">&lt;-</span> <span class="fu">as.character</span>(rhizobium<span class="sc">$</span>pop)</span>
<span id="cb288-5"><a href="basic-statistics-in-r.html#cb288-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(rhizobium_pop)</span>
<span id="cb288-6"><a href="basic-statistics-in-r.html#cb288-6" aria-hidden="true" tabindex="-1"></a><span class="co">#chr [1:215] &quot;FTmdc&quot; &quot;FTmdc&quot; &quot;FTmdc&quot; &quot;FTmdc&quot; &quot;FTmdc&quot; &quot;FTmdc&quot; &quot;FTmdc&quot; &quot;FTmdc&quot; &quot;FTmdc&quot; &quot;FTmdc&quot; &quot;FTmdc&quot; &quot;FTmdc&quot; &quot;FTmdc&quot; ...</span></span>
<span id="cb288-7"><a href="basic-statistics-in-r.html#cb288-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb288-8"><a href="basic-statistics-in-r.html#cb288-8" aria-hidden="true" tabindex="-1"></a><span class="do">## get a idea of the unique character elements in the rhizobium_pop object</span></span>
<span id="cb288-9"><a href="basic-statistics-in-r.html#cb288-9" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(rhizobium_pop)</span>
<span id="cb288-10"><a href="basic-statistics-in-r.html#cb288-10" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;FTmdc&quot;  &quot;FTmlt&quot;  &quot;TETmdc&quot; &quot;THLmlt&quot; &quot;TELmlt&quot; &quot;TETmlt&quot; &quot;THTmlt&quot;</span></span>
<span id="cb288-11"><a href="basic-statistics-in-r.html#cb288-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb288-12"><a href="basic-statistics-in-r.html#cb288-12" aria-hidden="true" tabindex="-1"></a><span class="co">#we can see 7 unique text elements</span></span>
<span id="cb288-13"><a href="basic-statistics-in-r.html#cb288-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb288-14"><a href="basic-statistics-in-r.html#cb288-14" aria-hidden="true" tabindex="-1"></a><span class="do">## create a table object</span></span>
<span id="cb288-15"><a href="basic-statistics-in-r.html#cb288-15" aria-hidden="true" tabindex="-1"></a>rhizobium_pop_table <span class="ot">&lt;-</span> <span class="fu">table</span>(rhizobium_pop)</span>
<span id="cb288-16"><a href="basic-statistics-in-r.html#cb288-16" aria-hidden="true" tabindex="-1"></a>rhizobium_pop_table</span>
<span id="cb288-17"><a href="basic-statistics-in-r.html#cb288-17" aria-hidden="true" tabindex="-1"></a><span class="co">#rhizobium_pop</span></span>
<span id="cb288-18"><a href="basic-statistics-in-r.html#cb288-18" aria-hidden="true" tabindex="-1"></a><span class="co"># FTmdc  FTmlt TELmlt TETmdc TETmlt THLmlt THTmlt </span></span>
<span id="cb288-19"><a href="basic-statistics-in-r.html#cb288-19" aria-hidden="true" tabindex="-1"></a><span class="co">#    46     43     20     20     24     20     42</span></span>
<span id="cb288-20"><a href="basic-statistics-in-r.html#cb288-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb288-21"><a href="basic-statistics-in-r.html#cb288-21" aria-hidden="true" tabindex="-1"></a><span class="do">## we have 7 categories. We will assume that they should be in equal proportion (i.e., 1/7 each).</span></span>
<span id="cb288-22"><a href="basic-statistics-in-r.html#cb288-22" aria-hidden="true" tabindex="-1"></a><span class="fu">chisq.test</span>(rhizobium_pop_table, <span class="at">p =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">/</span><span class="dv">7</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">7</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">7</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">7</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">7</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">7</span>, <span class="dv">1</span><span class="sc">/</span><span class="dv">7</span>))</span>
<span id="cb288-23"><a href="basic-statistics-in-r.html#cb288-23" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb288-24"><a href="basic-statistics-in-r.html#cb288-24" aria-hidden="true" tabindex="-1"></a><span class="co">#   Chi-squared test for given probabilities</span></span>
<span id="cb288-25"><a href="basic-statistics-in-r.html#cb288-25" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb288-26"><a href="basic-statistics-in-r.html#cb288-26" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  rhizobium_pop_table</span></span>
<span id="cb288-27"><a href="basic-statistics-in-r.html#cb288-27" aria-hidden="true" tabindex="-1"></a><span class="co">#X-squared = 29.349, df = 6, p-value = 5.225e-05</span></span></code></pre></div>
<p><strong>14)</strong> You can also cross tabulate results of two measurements of population that result in two or more categories for each and determine if they are related using a Chi-square goodness of fit test. The goal is to create a contingency table where your compare the outcomes for two different measurements of the same population and determine if they tend to be in agreement or not.</p>
<div class="sourceCode" id="cb289"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb289-1"><a href="basic-statistics-in-r.html#cb289-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we can have a population and determine a categorical variable that result in two category (0, 1) like for example: healthy and sick</span></span>
<span id="cb289-2"><a href="basic-statistics-in-r.html#cb289-2" aria-hidden="true" tabindex="-1"></a>sample_population <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(<span class="at">n =</span> <span class="dv">100</span>, <span class="at">size =</span> <span class="dv">1</span>, <span class="at">prob=</span> <span class="fl">0.40</span>)</span>
<span id="cb289-3"><a href="basic-statistics-in-r.html#cb289-3" aria-hidden="true" tabindex="-1"></a>population_states <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_population  <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;sick&quot;</span>, <span class="st">&quot;healthy&quot;</span>)</span>
<span id="cb289-4"><a href="basic-statistics-in-r.html#cb289-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sample_population) <span class="ot">&lt;-</span> population_states</span>
<span id="cb289-5"><a href="basic-statistics-in-r.html#cb289-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sample_population)</span>
<span id="cb289-6"><a href="basic-statistics-in-r.html#cb289-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  healthy   sick healthy healthy healthy healthy </span></span>
<span id="cb289-7"><a href="basic-statistics-in-r.html#cb289-7" aria-hidden="true" tabindex="-1"></a><span class="co">#      0       1       0       0       0       0 </span></span>
<span id="cb289-8"><a href="basic-statistics-in-r.html#cb289-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb289-9"><a href="basic-statistics-in-r.html#cb289-9" aria-hidden="true" tabindex="-1"></a><span class="do">## we can have the same population and determine a different categorical variable that result in two categories (0, 1) like for example: with_virus and no_infected</span></span>
<span id="cb289-10"><a href="basic-statistics-in-r.html#cb289-10" aria-hidden="true" tabindex="-1"></a>sample_virus <span class="ot">&lt;-</span> <span class="fu">rbinom</span>(<span class="at">n =</span> <span class="dv">100</span>, <span class="at">size =</span> <span class="dv">1</span>, <span class="at">prob=</span> <span class="fl">0.40</span>)</span>
<span id="cb289-11"><a href="basic-statistics-in-r.html#cb289-11" aria-hidden="true" tabindex="-1"></a>virus_names <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_virus  <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;with_virus&quot;</span>, <span class="st">&quot;no_infected&quot;</span>)</span>
<span id="cb289-12"><a href="basic-statistics-in-r.html#cb289-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sample_virus) <span class="ot">&lt;-</span> virus_names</span>
<span id="cb289-13"><a href="basic-statistics-in-r.html#cb289-13" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sample_virus)</span>
<span id="cb289-14"><a href="basic-statistics-in-r.html#cb289-14" aria-hidden="true" tabindex="-1"></a><span class="co">#    no_infected no_infected no_infected no_infected  with_virus no_infected </span></span>
<span id="cb289-15"><a href="basic-statistics-in-r.html#cb289-15" aria-hidden="true" tabindex="-1"></a><span class="co">#          0           0           0           0           1           0 </span></span></code></pre></div>
<p>We cross-tabulate these results and it is important to understand this table. The outcomes for <code>sample_population</code> and <code>sample_virus</code> are <code>0 1</code> as indicated in the first row and column. The agreements are when the outcomes of both experiments are the same as each experiment was executed <code>0-0</code> and <code>1-1</code>. All others are disagreements (e.g., <code>0-1</code>, <code>1-0</code>).</p>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="basic-statistics-in-r.html#cb290-1" aria-hidden="true" tabindex="-1"></a>cross_tab <span class="ot">&lt;-</span> <span class="fu">table</span>(sample_population, sample_virus)</span>
<span id="cb290-2"><a href="basic-statistics-in-r.html#cb290-2" aria-hidden="true" tabindex="-1"></a>cross_tab</span>
<span id="cb290-3"><a href="basic-statistics-in-r.html#cb290-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                  sample_virus</span></span>
<span id="cb290-4"><a href="basic-statistics-in-r.html#cb290-4" aria-hidden="true" tabindex="-1"></a><span class="co">#sample_population  0  1</span></span>
<span id="cb290-5"><a href="basic-statistics-in-r.html#cb290-5" aria-hidden="true" tabindex="-1"></a><span class="co">#                0 44 13</span></span>
<span id="cb290-6"><a href="basic-statistics-in-r.html#cb290-6" aria-hidden="true" tabindex="-1"></a><span class="co">#                1 35  8</span></span></code></pre></div>
<p>We can now estimate if such agreements and disagreements are different using Chi-squared Test for Count Data with the function <code>chisq.test()</code>.</p>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb291-1"><a href="basic-statistics-in-r.html#cb291-1" aria-hidden="true" tabindex="-1"></a><span class="fu">chisq.test</span>(cross_tab)</span>
<span id="cb291-2"><a href="basic-statistics-in-r.html#cb291-2" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb291-3"><a href="basic-statistics-in-r.html#cb291-3" aria-hidden="true" tabindex="-1"></a><span class="co">#       Pearson&#39;s Chi-squared test with Yates&#39; continuity correction</span></span>
<span id="cb291-4"><a href="basic-statistics-in-r.html#cb291-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb291-5"><a href="basic-statistics-in-r.html#cb291-5" aria-hidden="true" tabindex="-1"></a><span class="co"># data:  cross_tab</span></span>
<span id="cb291-6"><a href="basic-statistics-in-r.html#cb291-6" aria-hidden="true" tabindex="-1"></a><span class="co"># X-squared = 0.069082, df = 1, p-value = 0.7927</span></span></code></pre></div>
<p>In this case, we found that they were non-significant different. In this case, if the sick people might not have this made-up virus.</p>
<p>Here is another example, in this case we will make people that are sick more likely to have a virus.test</p>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="basic-statistics-in-r.html#cb292-1" aria-hidden="true" tabindex="-1"></a><span class="co">#$ we construct a data frame</span></span>
<span id="cb292-2"><a href="basic-statistics-in-r.html#cb292-2" aria-hidden="true" tabindex="-1"></a>sample_virus_2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span> (<span class="at">population_1_if_sick =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">1</span>),</span>
<span id="cb292-3"><a href="basic-statistics-in-r.html#cb292-3" aria-hidden="true" tabindex="-1"></a>                               <span class="at">virus_1_if_detected =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">1</span>), <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb292-4"><a href="basic-statistics-in-r.html#cb292-4" aria-hidden="true" tabindex="-1"></a>sample_virus_2</span>
<span id="cb292-5"><a href="basic-statistics-in-r.html#cb292-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   population_1_if_sick virus_1_if_detected</span></span>
<span id="cb292-6"><a href="basic-statistics-in-r.html#cb292-6" aria-hidden="true" tabindex="-1"></a><span class="co">#1                     0                   0</span></span>
<span id="cb292-7"><a href="basic-statistics-in-r.html#cb292-7" aria-hidden="true" tabindex="-1"></a><span class="co">#2                     0                   0</span></span>
<span id="cb292-8"><a href="basic-statistics-in-r.html#cb292-8" aria-hidden="true" tabindex="-1"></a><span class="co">#3                     0                   0</span></span>
<span id="cb292-9"><a href="basic-statistics-in-r.html#cb292-9" aria-hidden="true" tabindex="-1"></a><span class="co">#4                     1                   1</span></span>
<span id="cb292-10"><a href="basic-statistics-in-r.html#cb292-10" aria-hidden="true" tabindex="-1"></a><span class="co">#5                     1                   1</span></span>
<span id="cb292-11"><a href="basic-statistics-in-r.html#cb292-11" aria-hidden="true" tabindex="-1"></a><span class="co">#6                     1                   1</span></span>
<span id="cb292-12"><a href="basic-statistics-in-r.html#cb292-12" aria-hidden="true" tabindex="-1"></a><span class="co">#7                     1                   1</span></span>
<span id="cb292-13"><a href="basic-statistics-in-r.html#cb292-13" aria-hidden="true" tabindex="-1"></a><span class="co">#8                     1                   0</span></span>
<span id="cb292-14"><a href="basic-statistics-in-r.html#cb292-14" aria-hidden="true" tabindex="-1"></a><span class="co">#9                     0                   0</span></span>
<span id="cb292-15"><a href="basic-statistics-in-r.html#cb292-15" aria-hidden="true" tabindex="-1"></a><span class="co">#10                    1                   1</span></span>
<span id="cb292-16"><a href="basic-statistics-in-r.html#cb292-16" aria-hidden="true" tabindex="-1"></a><span class="co">#11                    1                   1</span></span>
<span id="cb292-17"><a href="basic-statistics-in-r.html#cb292-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb292-18"><a href="basic-statistics-in-r.html#cb292-18" aria-hidden="true" tabindex="-1"></a><span class="do">## we save to numeric vectors</span></span>
<span id="cb292-19"><a href="basic-statistics-in-r.html#cb292-19" aria-hidden="true" tabindex="-1"></a>population_1_if_sick <span class="ot">&lt;-</span> sample_virus_2<span class="sc">$</span>population_1_if_sick</span>
<span id="cb292-20"><a href="basic-statistics-in-r.html#cb292-20" aria-hidden="true" tabindex="-1"></a>virus_1_if_detected <span class="ot">&lt;-</span> sample_virus_2<span class="sc">$</span>virus_1_if_detected</span>
<span id="cb292-21"><a href="basic-statistics-in-r.html#cb292-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb292-22"><a href="basic-statistics-in-r.html#cb292-22" aria-hidden="true" tabindex="-1"></a><span class="do">## we do a crosstabulation</span></span>
<span id="cb292-23"><a href="basic-statistics-in-r.html#cb292-23" aria-hidden="true" tabindex="-1"></a>cross_tab2 <span class="ot">&lt;-</span> <span class="fu">table</span>(population_1_if_sick, virus_1_if_detected)</span>
<span id="cb292-24"><a href="basic-statistics-in-r.html#cb292-24" aria-hidden="true" tabindex="-1"></a>cross_tab2</span>
<span id="cb292-25"><a href="basic-statistics-in-r.html#cb292-25" aria-hidden="true" tabindex="-1"></a><span class="co">#                    virus_1_if_detected</span></span>
<span id="cb292-26"><a href="basic-statistics-in-r.html#cb292-26" aria-hidden="true" tabindex="-1"></a><span class="co">#population_1_if_sick 0 1</span></span>
<span id="cb292-27"><a href="basic-statistics-in-r.html#cb292-27" aria-hidden="true" tabindex="-1"></a><span class="co">#                   0 4 0</span></span>
<span id="cb292-28"><a href="basic-statistics-in-r.html#cb292-28" aria-hidden="true" tabindex="-1"></a><span class="co">#                   1 1 6</span></span>
<span id="cb292-29"><a href="basic-statistics-in-r.html#cb292-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb292-30"><a href="basic-statistics-in-r.html#cb292-30" aria-hidden="true" tabindex="-1"></a><span class="do">## we perform the chisq.test</span></span>
<span id="cb292-31"><a href="basic-statistics-in-r.html#cb292-31" aria-hidden="true" tabindex="-1"></a><span class="fu">chisq.test</span>(cross_tab2)</span>
<span id="cb292-32"><a href="basic-statistics-in-r.html#cb292-32" aria-hidden="true" tabindex="-1"></a><span class="co">#   Pearson&#39;s Chi-squared test with Yates&#39; continuity correction</span></span>
<span id="cb292-33"><a href="basic-statistics-in-r.html#cb292-33" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb292-34"><a href="basic-statistics-in-r.html#cb292-34" aria-hidden="true" tabindex="-1"></a><span class="co"># data:  cross_tab2</span></span>
<span id="cb292-35"><a href="basic-statistics-in-r.html#cb292-35" aria-hidden="true" tabindex="-1"></a><span class="co"># X-squared = 4.4818, df = 1, p-value = 0.03426</span></span>
<span id="cb292-36"><a href="basic-statistics-in-r.html#cb292-36" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb292-37"><a href="basic-statistics-in-r.html#cb292-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Warning message:</span></span>
<span id="cb292-38"><a href="basic-statistics-in-r.html#cb292-38" aria-hidden="true" tabindex="-1"></a><span class="co"># In chisq.test(cross_tab2) : Chi-squared approximation may be incorrect</span></span></code></pre></div>
<p>We found that people sick most likely have the virus and this is significant.</p>
</div>
<div id="beyond-t-test-mcnemars-test" class="section level2" number="6.16">
<h2><span class="header-section-number">6.16</span> Beyond t-test: McNemar’s test</h2>
<p><strong>15)</strong> This is a similar test to the Chi-square goodness of fit and it is used with paired nominal data in 2 × 2 contingency tables derived from dichotomous trait (i.e., YES or NO; Present or Absent), with matched pairs of subjects, to determine whether the row and column marginal frequencies are equal (see <a href="https://en.wikipedia.org/wiki/McNemar%27s_test" target="_blank"><strong>here</strong></a>).</p>
<div class="sourceCode" id="cb293"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb293-1"><a href="basic-statistics-in-r.html#cb293-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we can use the same cross tabulation example</span></span>
<span id="cb293-2"><a href="basic-statistics-in-r.html#cb293-2" aria-hidden="true" tabindex="-1"></a>cross_tab <span class="ot">&lt;-</span> <span class="fu">table</span>(sample_population, sample_virus)</span>
<span id="cb293-3"><a href="basic-statistics-in-r.html#cb293-3" aria-hidden="true" tabindex="-1"></a>cross_tab</span>
<span id="cb293-4"><a href="basic-statistics-in-r.html#cb293-4" aria-hidden="true" tabindex="-1"></a><span class="co">#                  sample_virus</span></span>
<span id="cb293-5"><a href="basic-statistics-in-r.html#cb293-5" aria-hidden="true" tabindex="-1"></a><span class="co">#sample_population  0  1</span></span>
<span id="cb293-6"><a href="basic-statistics-in-r.html#cb293-6" aria-hidden="true" tabindex="-1"></a><span class="co">#                0 44 13</span></span>
<span id="cb293-7"><a href="basic-statistics-in-r.html#cb293-7" aria-hidden="true" tabindex="-1"></a><span class="co">#                1 35  8</span></span>
<span id="cb293-8"><a href="basic-statistics-in-r.html#cb293-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb293-9"><a href="basic-statistics-in-r.html#cb293-9" aria-hidden="true" tabindex="-1"></a><span class="do">## we apply the McNemar&#39;s test</span></span>
<span id="cb293-10"><a href="basic-statistics-in-r.html#cb293-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb293-11"><a href="basic-statistics-in-r.html#cb293-11" aria-hidden="true" tabindex="-1"></a><span class="fu">mcnemar.test</span>(cross_tab)</span>
<span id="cb293-12"><a href="basic-statistics-in-r.html#cb293-12" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb293-13"><a href="basic-statistics-in-r.html#cb293-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   McNemar&#39;s Chi-squared test with continuity correction</span></span>
<span id="cb293-14"><a href="basic-statistics-in-r.html#cb293-14" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb293-15"><a href="basic-statistics-in-r.html#cb293-15" aria-hidden="true" tabindex="-1"></a><span class="co"># data:  cross_tab</span></span>
<span id="cb293-16"><a href="basic-statistics-in-r.html#cb293-16" aria-hidden="true" tabindex="-1"></a><span class="co"># McNemar&#39;s chi-squared = 9.1875, df = 1, p-value = 0.002437</span></span></code></pre></div>
<p>In this case, we found that the same result as the Chi-squared test and McNemar-s test was also non-significant. In this case, if the sick people might not have this made-up virus.</p>
</div>
<div id="beyond-t-test-fishers-exact-test" class="section level2" number="6.17">
<h2><span class="header-section-number">6.17</span> Beyond t-test: Fisher’s exact test</h2>
<p><strong>16)</strong> This is a similar test to the Chi-square goodness of fit and can be used with populations with small samples. The <a href="https://en.wikipedia.org/wiki/Fisher%27s_exact_test" target="_blank"><strong>Fisher’s exact test</strong></a> will examine the significance of the association (contingency) between the two kinds of classification. The null hypothesis is that both classifications are dependent. We can use the same dataset as above with the function <code>fisher.test()</code>.</p>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="basic-statistics-in-r.html#cb294-1" aria-hidden="true" tabindex="-1"></a><span class="fu">fisher.test</span>(cross_tab)</span>
<span id="cb294-2"><a href="basic-statistics-in-r.html#cb294-2" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb294-3"><a href="basic-statistics-in-r.html#cb294-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     Fisher&#39;s Exact Test for Count Data</span></span>
<span id="cb294-4"><a href="basic-statistics-in-r.html#cb294-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb294-5"><a href="basic-statistics-in-r.html#cb294-5" aria-hidden="true" tabindex="-1"></a><span class="co"># data:  cross_tab</span></span>
<span id="cb294-6"><a href="basic-statistics-in-r.html#cb294-6" aria-hidden="true" tabindex="-1"></a><span class="co"># p-value = 0.8047</span></span>
<span id="cb294-7"><a href="basic-statistics-in-r.html#cb294-7" aria-hidden="true" tabindex="-1"></a><span class="co"># alternative hypothesis: true odds ratio is not equal to 1</span></span>
<span id="cb294-8"><a href="basic-statistics-in-r.html#cb294-8" aria-hidden="true" tabindex="-1"></a><span class="co"># 95 percent confidence interval:</span></span>
<span id="cb294-9"><a href="basic-statistics-in-r.html#cb294-9" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.2488368 2.2877938</span></span>
<span id="cb294-10"><a href="basic-statistics-in-r.html#cb294-10" aria-hidden="true" tabindex="-1"></a><span class="co"># sample estimates:</span></span>
<span id="cb294-11"><a href="basic-statistics-in-r.html#cb294-11" aria-hidden="true" tabindex="-1"></a><span class="co"># odds ratio </span></span>
<span id="cb294-12"><a href="basic-statistics-in-r.html#cb294-12" aria-hidden="true" tabindex="-1"></a><span class="co">#  0.7755955 </span></span></code></pre></div>
<p>In this case, we found that they were barely significant different. In this case, if the sick people might not have this made-up virus.</p>
</div>
<div id="correlations" class="section level2" number="6.18">
<h2><span class="header-section-number">6.18</span> Correlations</h2>
<p><strong>17)</strong> We can determine correlation between two variables using <code>cor()</code> and test for its significance using <code>cor.test()</code>. Correlation is the dependence or association between variables, and it usually refers to the degree to which a pair of variables are linearly related. For practical uses, correlations can provide a predictive relationship that can derived from one variable on the other. However, the evidence of correlation does not mean a causal relationship (i.e., A does not necessarily cause B). Likewise, it is important to determine if such correlations are significant and these can be performed for normal or non-normal distributions (in this case use the use argument <code>method = "spearman"</code>).</p>
<p>We will determine a correlation and test for its significance between two samples of the <code>airway_data</code>.</p>
<div class="sourceCode" id="cb295"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb295-1"><a href="basic-statistics-in-r.html#cb295-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(airway_data[,<span class="dv">2</span>], airway_data[,<span class="dv">3</span>])</span>
<span id="cb295-2"><a href="basic-statistics-in-r.html#cb295-2" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 0.9646265</span></span>
<span id="cb295-3"><a href="basic-statistics-in-r.html#cb295-3" aria-hidden="true" tabindex="-1"></a><span class="co"># we will assume that these are not normal</span></span>
<span id="cb295-4"><a href="basic-statistics-in-r.html#cb295-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(airway_data[,<span class="dv">2</span>], airway_data[,<span class="dv">3</span>], <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>)</span>
<span id="cb295-5"><a href="basic-statistics-in-r.html#cb295-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb295-6"><a href="basic-statistics-in-r.html#cb295-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   Spearman&#39;s rank correlation rho</span></span>
<span id="cb295-7"><a href="basic-statistics-in-r.html#cb295-7" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb295-8"><a href="basic-statistics-in-r.html#cb295-8" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  airway_data[, 2] and airway_data[, 3]</span></span>
<span id="cb295-9"><a href="basic-statistics-in-r.html#cb295-9" aria-hidden="true" tabindex="-1"></a><span class="co">#S = 4.8345e+11, p-value &lt; 2.2e-16</span></span>
<span id="cb295-10"><a href="basic-statistics-in-r.html#cb295-10" aria-hidden="true" tabindex="-1"></a><span class="co">#alternative hypothesis: true rho is not equal to 0</span></span>
<span id="cb295-11"><a href="basic-statistics-in-r.html#cb295-11" aria-hidden="true" tabindex="-1"></a><span class="co">#sample estimates:</span></span>
<span id="cb295-12"><a href="basic-statistics-in-r.html#cb295-12" aria-hidden="true" tabindex="-1"></a><span class="co">#      rho </span></span>
<span id="cb295-13"><a href="basic-statistics-in-r.html#cb295-13" aria-hidden="true" tabindex="-1"></a><span class="co">#0.9499302 </span></span></code></pre></div>
</div>
<div id="comparing-distributions" class="section level2" number="6.19">
<h2><span class="header-section-number">6.19</span> Comparing distributions</h2>
<p><strong>18)</strong> We can also determine if two variables come from the same distribution using Kolmogorov-Smirnov test <code>ks.test()</code>. The data does not require to be normal.</p>
<p>We will determine if two samples of the <code>airway_data</code> have the same distribution.</p>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="basic-statistics-in-r.html#cb296-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ks.test</span>(airway_data[,<span class="dv">2</span>], airway_data[,<span class="dv">3</span>])</span>
<span id="cb296-2"><a href="basic-statistics-in-r.html#cb296-2" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb296-3"><a href="basic-statistics-in-r.html#cb296-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   Two-sample Kolmogorov-Smirnov test</span></span>
<span id="cb296-4"><a href="basic-statistics-in-r.html#cb296-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb296-5"><a href="basic-statistics-in-r.html#cb296-5" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  airway_data[, 2] and airway_data[, 3]</span></span>
<span id="cb296-6"><a href="basic-statistics-in-r.html#cb296-6" aria-hidden="true" tabindex="-1"></a><span class="co">#D = 0.013413, p-value = 0.001896</span></span>
<span id="cb296-7"><a href="basic-statistics-in-r.html#cb296-7" aria-hidden="true" tabindex="-1"></a><span class="co">#alternative hypothesis: two-sided</span></span>
<span id="cb296-8"><a href="basic-statistics-in-r.html#cb296-8" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb296-9"><a href="basic-statistics-in-r.html#cb296-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Warning message:</span></span>
<span id="cb296-10"><a href="basic-statistics-in-r.html#cb296-10" aria-hidden="true" tabindex="-1"></a><span class="co">#In ks.test(airway_data[, 2], airway_data[, 3]) :</span></span>
<span id="cb296-11"><a href="basic-statistics-in-r.html#cb296-11" aria-hidden="true" tabindex="-1"></a><span class="co">#  p-value will be approximate in the presence of ties</span></span></code></pre></div>
<p>We can conclude that these two samples do not have the same distribution (i.e., <code>p-value = 0.001896</code>).</p>
<p><strong>19)</strong> We can compare two groups that have some individuals in such groups being exposed treatment versus control using</p>
<p>from the R-package <a href="https://cran.r-project.org/web/packages/pROC/index.html" target="_blank"><strong>pROC</strong></a> using the functions <code>roc()</code> and <code>roc.test()</code>.
This test will compare the AUC (area under the curve) of two <a href="https://en.wikipedia.org/wiki/Receiver_operating_characteristic#Basic_concept" target="_blank">ROC</a> (receiver operating characteristic) curves.
These are graphical plots useful in the diagnostic of a binary classifier (e.g., “control-0” or “exposed-1”) as its discrimination threshold is varied (i.e., what is the cutoff to consider a measurement as “exposed-1”).</p>
<p>We will illustrate this test using two <code>airway_data</code> by assigning “control” and “exposed” to sample of rows.</p>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="basic-statistics-in-r.html#cb297-1" aria-hidden="true" tabindex="-1"></a><span class="co"># you need to instal the package &quot;pROC&quot;</span></span>
<span id="cb297-2"><a href="basic-statistics-in-r.html#cb297-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;pROC&quot;</span>)</span>
<span id="cb297-3"><a href="basic-statistics-in-r.html#cb297-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pROC)</span></code></pre></div>
<p>We can prepare two samples of <code>airway_data</code> and assign some of elements of each as “control” and “exposed”.</p>
<div class="sourceCode" id="cb298"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb298-1"><a href="basic-statistics-in-r.html#cb298-1" aria-hidden="true" tabindex="-1"></a><span class="do">## set two samples</span></span>
<span id="cb298-2"><a href="basic-statistics-in-r.html#cb298-2" aria-hidden="true" tabindex="-1"></a>sample_SRR1039508 <span class="ot">&lt;-</span> airway_data<span class="sc">$</span>SRR1039508 </span>
<span id="cb298-3"><a href="basic-statistics-in-r.html#cb298-3" aria-hidden="true" tabindex="-1"></a>sample_SRR1039509 <span class="ot">&lt;-</span> airway_data<span class="sc">$</span>SRR1039509</span>
<span id="cb298-4"><a href="basic-statistics-in-r.html#cb298-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb298-5"><a href="basic-statistics-in-r.html#cb298-5" aria-hidden="true" tabindex="-1"></a><span class="do">## get summaries of each</span></span>
<span id="cb298-6"><a href="basic-statistics-in-r.html#cb298-6" aria-hidden="true" tabindex="-1"></a>summary_SRR1039508 <span class="ot">&lt;-</span> <span class="fu">summary</span>(sample_SRR1039508)</span>
<span id="cb298-7"><a href="basic-statistics-in-r.html#cb298-7" aria-hidden="true" tabindex="-1"></a>summary_SRR1039508</span>
<span id="cb298-8"><a href="basic-statistics-in-r.html#cb298-8" aria-hidden="true" tabindex="-1"></a><span class="co">#    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. </span></span>
<span id="cb298-9"><a href="basic-statistics-in-r.html#cb298-9" aria-hidden="true" tabindex="-1"></a><span class="co">#     0.0      0.0      1.0    546.5    203.0 287372.0 </span></span>
<span id="cb298-10"><a href="basic-statistics-in-r.html#cb298-10" aria-hidden="true" tabindex="-1"></a>summary_SRR1039509 <span class="ot">&lt;-</span> <span class="fu">summary</span>(sample_SRR1039509)</span>
<span id="cb298-11"><a href="basic-statistics-in-r.html#cb298-11" aria-hidden="true" tabindex="-1"></a>summary_SRR1039509</span>
<span id="cb298-12"><a href="basic-statistics-in-r.html#cb298-12" aria-hidden="true" tabindex="-1"></a><span class="co">#    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. </span></span>
<span id="cb298-13"><a href="basic-statistics-in-r.html#cb298-13" aria-hidden="true" tabindex="-1"></a><span class="co">#     0.0      0.0      1.0    501.1    172.0 244921.0</span></span>
<span id="cb298-14"><a href="basic-statistics-in-r.html#cb298-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb298-15"><a href="basic-statistics-in-r.html#cb298-15" aria-hidden="true" tabindex="-1"></a><span class="do">## assign  &quot;control&quot; if less than 3rd quartile of each</span></span>
<span id="cb298-16"><a href="basic-statistics-in-r.html#cb298-16" aria-hidden="true" tabindex="-1"></a>SRR1039508_binary <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_SRR1039508 <span class="sc">&gt;</span> summary_SRR1039508[<span class="dv">5</span>], <span class="st">&quot;exposed&quot;</span>, <span class="st">&quot;control&quot;</span>)</span>
<span id="cb298-17"><a href="basic-statistics-in-r.html#cb298-17" aria-hidden="true" tabindex="-1"></a><span class="do">## assign  &quot;control&quot; if more than 3rd quartile of each -- a reverse from last sample</span></span>
<span id="cb298-18"><a href="basic-statistics-in-r.html#cb298-18" aria-hidden="true" tabindex="-1"></a>SRR1039509_binary <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sample_SRR1039509 <span class="sc">&gt;</span> summary_SRR1039509[<span class="dv">5</span>], <span class="st">&quot;control&quot;</span>, <span class="st">&quot;exposed&quot;</span>)</span>
<span id="cb298-19"><a href="basic-statistics-in-r.html#cb298-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb298-20"><a href="basic-statistics-in-r.html#cb298-20" aria-hidden="true" tabindex="-1"></a><span class="do">## create a roc object for each sample</span></span>
<span id="cb298-21"><a href="basic-statistics-in-r.html#cb298-21" aria-hidden="true" tabindex="-1"></a>roc_SRR1039508 <span class="ot">&lt;-</span> <span class="fu">roc</span>(SRR1039508_binary, sample_SRR1039508, <span class="at">direction =</span> <span class="st">&quot;&lt;&quot;</span>)</span>
<span id="cb298-22"><a href="basic-statistics-in-r.html#cb298-22" aria-hidden="true" tabindex="-1"></a><span class="co">#Setting levels: control = control, case = exposed</span></span>
<span id="cb298-23"><a href="basic-statistics-in-r.html#cb298-23" aria-hidden="true" tabindex="-1"></a>roc_SRR1039509 <span class="ot">&lt;-</span> <span class="fu">roc</span>(SRR1039509_binary, sample_SRR1039509, <span class="at">direction =</span> <span class="st">&quot;&lt;&quot;</span>)</span>
<span id="cb298-24"><a href="basic-statistics-in-r.html#cb298-24" aria-hidden="true" tabindex="-1"></a><span class="co">#Setting levels: control = control, case = exposed</span></span>
<span id="cb298-25"><a href="basic-statistics-in-r.html#cb298-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb298-26"><a href="basic-statistics-in-r.html#cb298-26" aria-hidden="true" tabindex="-1"></a><span class="do">## perform the roc.test and increase boot.n for a more precise p-value:</span></span>
<span id="cb298-27"><a href="basic-statistics-in-r.html#cb298-27" aria-hidden="true" tabindex="-1"></a><span class="fu">roc.test</span>(roc_SRR1039508, roc_SRR1039509, <span class="at">method=</span><span class="st">&quot;bootstrap&quot;</span>, <span class="at">boot.n=</span><span class="dv">1000</span>)</span>
<span id="cb298-28"><a href="basic-statistics-in-r.html#cb298-28" aria-hidden="true" tabindex="-1"></a><span class="co">#    |=========================================================================================================================================================================| 100%</span></span>
<span id="cb298-29"><a href="basic-statistics-in-r.html#cb298-29" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb298-30"><a href="basic-statistics-in-r.html#cb298-30" aria-hidden="true" tabindex="-1"></a><span class="co">#   Bootstrap test for two ROC curves</span></span>
<span id="cb298-31"><a href="basic-statistics-in-r.html#cb298-31" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb298-32"><a href="basic-statistics-in-r.html#cb298-32" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  roc_SRR1039508 and roc_SRR1039509</span></span>
<span id="cb298-33"><a href="basic-statistics-in-r.html#cb298-33" aria-hidden="true" tabindex="-1"></a><span class="co">#D = Inf, boot.n = 1000, boot.stratified = 1, p-value &lt; 2.2e-16</span></span>
<span id="cb298-34"><a href="basic-statistics-in-r.html#cb298-34" aria-hidden="true" tabindex="-1"></a><span class="co">#alternative hypothesis: true difference in AUC is not equal to 0</span></span>
<span id="cb298-35"><a href="basic-statistics-in-r.html#cb298-35" aria-hidden="true" tabindex="-1"></a><span class="co">#sample estimates:</span></span>
<span id="cb298-36"><a href="basic-statistics-in-r.html#cb298-36" aria-hidden="true" tabindex="-1"></a><span class="co">#AUC of roc1 AUC of roc2 </span></span>
<span id="cb298-37"><a href="basic-statistics-in-r.html#cb298-37" aria-hidden="true" tabindex="-1"></a><span class="co">#          1           0 </span></span></code></pre></div>
<p>We can conclude that for SRR1039508 and SRR1039509 are different in their distribution in reference to outcomes for “control” versus “exposed”.</p>
</div>
<div id="bootstraps" class="section level2" number="6.20">
<h2><span class="header-section-number">6.20</span> Bootstraps</h2>
<p>This is a non-parametric method of repeated resampling from collection of data with replacement.
With enough sampling repetitions (usually in thousands),
we can estimate the distribution of a parameter(s) of a population where an original data collection can be used a source
for resampling. The rational is that with enough repeat sampling from this population of data, we likely obtain a distribution of
values for a parameter (e.g., means, correlations, etc.) that
we want to estimate from the population being studied. Repeated sampling allows to account for the uncertainty that resulted from a
single or few samples. We calculate from this collection of samples a statistic of interest (e.g., mean) and with enough repetitions
the shape of the distribution of the estimates of the statistic would be “bell shaped” (i.e., have Gaussian distribution),
which makes possible to construct a confidence interval of our statistic.</p>
<p>Bootstrap can be done in R and it requires to have a source collection (usually large) of data from our population of interest that
will serve to draw the bootstrap samples. We will define how many samples with replacement will be drawn (e.g., &gt;1000), with a given
sample size (i.e., how many elements are drawn during each sample usually). From each sample, we will estimate a parameter of interest (e.g., a mean)
and collect those to calculate of distribution where its mean (expected value for the statistic) and standard deviation
(a measure of the uncertainty about that value of the statistic) can be used to estimate a confidence interval of such
statistic.</p>
<p>We need to install the R-package <a href="https://cran.r-project.org/web/packages/boot/index.html" target="_blank"><strong>boot</strong></a>.</p>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="basic-statistics-in-r.html#cb299-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&#39;boot&#39;</span>)</span>
<span id="cb299-2"><a href="basic-statistics-in-r.html#cb299-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(boot)</span></code></pre></div>
<p>We will use the <strong>mtcars</strong> to illustrate the estimate of mean and correlation between some of the variables.</p>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="basic-statistics-in-r.html#cb300-1" aria-hidden="true" tabindex="-1"></a><span class="co">#load get mtcars dataset -- this a preloaded dataset in R</span></span>
<span id="cb300-2"><a href="basic-statistics-in-r.html#cb300-2" aria-hidden="true" tabindex="-1"></a>cars_data <span class="ot">&lt;-</span> mtcars</span>
<span id="cb300-3"><a href="basic-statistics-in-r.html#cb300-3" aria-hidden="true" tabindex="-1"></a>cars_data</span>
<span id="cb300-4"><a href="basic-statistics-in-r.html#cb300-4" aria-hidden="true" tabindex="-1"></a><span class="co">#                     mpg cyl  disp  hp drat    wt  qsec vs am gear carb</span></span>
<span id="cb300-5"><a href="basic-statistics-in-r.html#cb300-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Mazda RX4           21.0   6 160.0 110 3.90 2.620 16.46  0  1    4    4</span></span>
<span id="cb300-6"><a href="basic-statistics-in-r.html#cb300-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Mazda RX4 Wag       21.0   6 160.0 110 3.90 2.875 17.02  0  1    4    4</span></span>
<span id="cb300-7"><a href="basic-statistics-in-r.html#cb300-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Datsun 710          22.8   4 108.0  93 3.85 2.320 18.61  1  1    4    1</span></span>
<span id="cb300-8"><a href="basic-statistics-in-r.html#cb300-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Hornet 4 Drive      21.4   6 258.0 110 3.08 3.215 19.44  1  0    3    1</span></span>
<span id="cb300-9"><a href="basic-statistics-in-r.html#cb300-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Hornet Sportabout   18.7   8 360.0 175 3.15 3.440 17.02  0  0    3    2</span></span>
<span id="cb300-10"><a href="basic-statistics-in-r.html#cb300-10" aria-hidden="true" tabindex="-1"></a><span class="co">#..</span></span></code></pre></div>
<p>We can estimate the mean of the <code>mpg</code> variable</p>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="basic-statistics-in-r.html#cb301-1" aria-hidden="true" tabindex="-1"></a>mpg_mean <span class="ot">&lt;-</span> <span class="fu">mean</span>(cars_data<span class="sc">$</span>mpg)</span>
<span id="cb301-2"><a href="basic-statistics-in-r.html#cb301-2" aria-hidden="true" tabindex="-1"></a>mpg_mean</span>
<span id="cb301-3"><a href="basic-statistics-in-r.html#cb301-3" aria-hidden="true" tabindex="-1"></a><span class="co">#1] 20.09062</span></span></code></pre></div>
<p>We can now use boot to calculate how confident we are on the mean estimates</p>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="basic-statistics-in-r.html#cb302-1" aria-hidden="true" tabindex="-1"></a><span class="do">## First, we need a function that calculate a mean for each sample</span></span>
<span id="cb302-2"><a href="basic-statistics-in-r.html#cb302-2" aria-hidden="true" tabindex="-1"></a>function_mean <span class="ot">&lt;-</span> <span class="cf">function</span>(data, i){</span>
<span id="cb302-3"><a href="basic-statistics-in-r.html#cb302-3" aria-hidden="true" tabindex="-1"></a>                              my_data2 <span class="ot">&lt;-</span> data[i,] </span>
<span id="cb302-4"><a href="basic-statistics-in-r.html#cb302-4" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">return</span>(<span class="fu">mean</span>(my_data2<span class="sc">$</span>mpg))</span>
<span id="cb302-5"><a href="basic-statistics-in-r.html#cb302-5" aria-hidden="true" tabindex="-1"></a>                                  }</span>
<span id="cb302-6"><a href="basic-statistics-in-r.html#cb302-6" aria-hidden="true" tabindex="-1"></a><span class="co"># The arguments of this function are data or the source dataset </span></span>
<span id="cb302-7"><a href="basic-statistics-in-r.html#cb302-7" aria-hidden="true" tabindex="-1"></a><span class="co"># and [i] is the vector index of which rows from the source dataset will be draw to build a bootstrap sample.</span></span>
<span id="cb302-8"><a href="basic-statistics-in-r.html#cb302-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb302-9"><a href="basic-statistics-in-r.html#cb302-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Second, we can run our bootstrap with 100000 bootstrap replicates</span></span>
<span id="cb302-10"><a href="basic-statistics-in-r.html#cb302-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb302-11"><a href="basic-statistics-in-r.html#cb302-11" aria-hidden="true" tabindex="-1"></a>bootstrap_mpg_mean <span class="ot">&lt;-</span> <span class="fu">boot</span>(<span class="at">data =</span> cars_data, <span class="at">statistic =</span> function_mean, <span class="at">R =</span> <span class="dv">10000</span>)</span>
<span id="cb302-12"><a href="basic-statistics-in-r.html#cb302-12" aria-hidden="true" tabindex="-1"></a>bootstrap_mpg_mean</span>
<span id="cb302-13"><a href="basic-statistics-in-r.html#cb302-13" aria-hidden="true" tabindex="-1"></a><span class="co">#ORDINARY NONPARAMETRIC BOOTSTRAP</span></span>
<span id="cb302-14"><a href="basic-statistics-in-r.html#cb302-14" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb302-15"><a href="basic-statistics-in-r.html#cb302-15" aria-hidden="true" tabindex="-1"></a><span class="co">#Call:</span></span>
<span id="cb302-16"><a href="basic-statistics-in-r.html#cb302-16" aria-hidden="true" tabindex="-1"></a><span class="co">#boot(data = cars_data, statistic = function_mean, R = 10000)</span></span>
<span id="cb302-17"><a href="basic-statistics-in-r.html#cb302-17" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb302-18"><a href="basic-statistics-in-r.html#cb302-18" aria-hidden="true" tabindex="-1"></a><span class="co">#Bootstrap Statistics :</span></span>
<span id="cb302-19"><a href="basic-statistics-in-r.html#cb302-19" aria-hidden="true" tabindex="-1"></a><span class="co">#    original      bias    std. error</span></span>
<span id="cb302-20"><a href="basic-statistics-in-r.html#cb302-20" aria-hidden="true" tabindex="-1"></a><span class="co">#t1* 20.09062 0.002242188    1.053944</span></span>
<span id="cb302-21"><a href="basic-statistics-in-r.html#cb302-21" aria-hidden="true" tabindex="-1"></a>bootstrap_mpg_mean<span class="sc">$</span>t0</span>
<span id="cb302-22"><a href="basic-statistics-in-r.html#cb302-22" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 20.09062</span></span>
<span id="cb302-23"><a href="basic-statistics-in-r.html#cb302-23" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(bootstrap_mpg_mean<span class="sc">$</span>t)</span>
<span id="cb302-24"><a href="basic-statistics-in-r.html#cb302-24" aria-hidden="true" tabindex="-1"></a><span class="co">#         [,1]</span></span>
<span id="cb302-25"><a href="basic-statistics-in-r.html#cb302-25" aria-hidden="true" tabindex="-1"></a><span class="co">#[1,] 20.16562</span></span>
<span id="cb302-26"><a href="basic-statistics-in-r.html#cb302-26" aria-hidden="true" tabindex="-1"></a><span class="co">#[2,] 20.05937</span></span>
<span id="cb302-27"><a href="basic-statistics-in-r.html#cb302-27" aria-hidden="true" tabindex="-1"></a><span class="co">#[3,] 20.42188</span></span>
<span id="cb302-28"><a href="basic-statistics-in-r.html#cb302-28" aria-hidden="true" tabindex="-1"></a><span class="co">#[4,] 20.45000</span></span>
<span id="cb302-29"><a href="basic-statistics-in-r.html#cb302-29" aria-hidden="true" tabindex="-1"></a><span class="co">#[5,] 20.48125</span></span>
<span id="cb302-30"><a href="basic-statistics-in-r.html#cb302-30" aria-hidden="true" tabindex="-1"></a><span class="co">#[6,] 21.81875</span></span>
<span id="cb302-31"><a href="basic-statistics-in-r.html#cb302-31" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span>
<span id="cb302-32"><a href="basic-statistics-in-r.html#cb302-32" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(bootstrap_mpg_mean<span class="sc">$</span>t)</span>
<span id="cb302-33"><a href="basic-statistics-in-r.html#cb302-33" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 10000     1</span></span>
<span id="cb302-34"><a href="basic-statistics-in-r.html#cb302-34" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(bootstrap_mpg_mean<span class="sc">$</span>t)</span>
<span id="cb302-35"><a href="basic-statistics-in-r.html#cb302-35" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 20.09287</span></span>
<span id="cb302-36"><a href="basic-statistics-in-r.html#cb302-36" aria-hidden="true" tabindex="-1"></a><span class="fu">range</span>(bootstrap_mpg_mean<span class="sc">$</span>t)</span>
<span id="cb302-37"><a href="basic-statistics-in-r.html#cb302-37" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 15.7375 24.3375</span></span>
<span id="cb302-38"><a href="basic-statistics-in-r.html#cb302-38" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span>(bootstrap_mpg_mean<span class="sc">$</span>t)</span>
<span id="cb302-39"><a href="basic-statistics-in-r.html#cb302-39" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 1.053944</span></span>
<span id="cb302-40"><a href="basic-statistics-in-r.html#cb302-40" aria-hidden="true" tabindex="-1"></a><span class="fu">boot.ci</span>(<span class="at">boot.out=</span>bootstrap_mpg_mean,<span class="at">type=</span><span class="st">&quot;norm&quot;</span>)</span>
<span id="cb302-41"><a href="basic-statistics-in-r.html#cb302-41" aria-hidden="true" tabindex="-1"></a><span class="co">#BOOTSTRAP CONFIDENCE INTERVAL CALCULATIONS</span></span>
<span id="cb302-42"><a href="basic-statistics-in-r.html#cb302-42" aria-hidden="true" tabindex="-1"></a><span class="co">#Based on 10000 bootstrap replicates</span></span>
<span id="cb302-43"><a href="basic-statistics-in-r.html#cb302-43" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb302-44"><a href="basic-statistics-in-r.html#cb302-44" aria-hidden="true" tabindex="-1"></a><span class="co">#CALL : </span></span>
<span id="cb302-45"><a href="basic-statistics-in-r.html#cb302-45" aria-hidden="true" tabindex="-1"></a><span class="co">#boot.ci(boot.out = bootstrap_mpg_mean, type = &quot;norm&quot;)</span></span>
<span id="cb302-46"><a href="basic-statistics-in-r.html#cb302-46" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb302-47"><a href="basic-statistics-in-r.html#cb302-47" aria-hidden="true" tabindex="-1"></a><span class="co">#Intervals : </span></span>
<span id="cb302-48"><a href="basic-statistics-in-r.html#cb302-48" aria-hidden="true" tabindex="-1"></a><span class="co">#Level      Normal        </span></span>
<span id="cb302-49"><a href="basic-statistics-in-r.html#cb302-49" aria-hidden="true" tabindex="-1"></a><span class="co">#95%   (18.02, 22.15 )  </span></span>
<span id="cb302-50"><a href="basic-statistics-in-r.html#cb302-50" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculations and Intervals on Original Scale</span></span>
<span id="cb302-51"><a href="basic-statistics-in-r.html#cb302-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb302-52"><a href="basic-statistics-in-r.html#cb302-52" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(bootstrap_mpg_mean<span class="sc">$</span>t, <span class="at">breaks=</span><span class="dv">100</span>, <span class="at">col=</span><span class="st">&quot;dodgerblue1&quot;</span>, <span class="at">main=</span><span class="st">&quot;Bootstrap Distribution of the Mean: mpg&quot;</span>)</span>
<span id="cb302-53"><a href="basic-statistics-in-r.html#cb302-53" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(<span class="at">x =</span> <span class="fu">density</span>(<span class="at">x =</span> bootstrap_mpg_mean<span class="sc">$</span>t))</span>
<span id="cb302-54"><a href="basic-statistics-in-r.html#cb302-54" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> <span class="fu">mean</span>(bootstrap_mpg_mean<span class="sc">$</span>t), <span class="at">col =</span> <span class="st">&quot;black&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="figures/session_6_figs/7_boostrap_mean.png" width="100%" /></p>
<p>The results are as follows: <strong>t0</strong> are the values of the statistic (i.e., mean) in original dataset,
<strong>t</strong> is a matrix of 10000 rows (i.e., number of bootstrap repetitions) that we use to draw our estimate of the mean.
The original mean value is <strong>20.09062</strong> and the bootstrap estimate of the mean is
very close at <strong>20.09287</strong>, with a 95% CI between <strong>18.02</strong> and <strong>22.15</strong>. Therefore, it is some variability
in the original sample for <code>mpg</code>.</p>
<p>Now, we can try another more complex analysis. For example, the correlation between <code>disp</code> and <code>wt</code>.</p>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="basic-statistics-in-r.html#cb303-1" aria-hidden="true" tabindex="-1"></a><span class="do">## null estimate of correlation</span></span>
<span id="cb303-2"><a href="basic-statistics-in-r.html#cb303-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(cars_data<span class="sc">$</span>disp, cars_data<span class="sc">$</span>wt)</span>
<span id="cb303-3"><a href="basic-statistics-in-r.html#cb303-3" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 0.8879799</span></span>
<span id="cb303-4"><a href="basic-statistics-in-r.html#cb303-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb303-5"><a href="basic-statistics-in-r.html#cb303-5" aria-hidden="true" tabindex="-1"></a><span class="do">## First, we need a function that calculate correlation for each sample</span></span>
<span id="cb303-6"><a href="basic-statistics-in-r.html#cb303-6" aria-hidden="true" tabindex="-1"></a>function_correlation <span class="ot">&lt;-</span> <span class="cf">function</span>(data, i){</span>
<span id="cb303-7"><a href="basic-statistics-in-r.html#cb303-7" aria-hidden="true" tabindex="-1"></a>                              my_data2 <span class="ot">&lt;-</span> data[i,] </span>
<span id="cb303-8"><a href="basic-statistics-in-r.html#cb303-8" aria-hidden="true" tabindex="-1"></a>                              <span class="fu">return</span>(<span class="fu">cor</span>(my_data2<span class="sc">$</span>disp, my_data2<span class="sc">$</span>wt))</span>
<span id="cb303-9"><a href="basic-statistics-in-r.html#cb303-9" aria-hidden="true" tabindex="-1"></a>                                  }</span>
<span id="cb303-10"><a href="basic-statistics-in-r.html#cb303-10" aria-hidden="true" tabindex="-1"></a><span class="do">## Second, we can run our bootstrap with 100000 bootstrap replicates</span></span>
<span id="cb303-11"><a href="basic-statistics-in-r.html#cb303-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb303-12"><a href="basic-statistics-in-r.html#cb303-12" aria-hidden="true" tabindex="-1"></a>bootstrap_cor_dis_and_wt <span class="ot">&lt;-</span> <span class="fu">boot</span>(<span class="at">data =</span> cars_data, <span class="at">statistic =</span> function_correlation, <span class="at">R =</span> <span class="dv">10000</span>)</span>
<span id="cb303-13"><a href="basic-statistics-in-r.html#cb303-13" aria-hidden="true" tabindex="-1"></a>bootstrap_cor_dis_and_wt</span>
<span id="cb303-14"><a href="basic-statistics-in-r.html#cb303-14" aria-hidden="true" tabindex="-1"></a><span class="co">#ORDINARY NONPARAMETRIC BOOTSTRAP</span></span>
<span id="cb303-15"><a href="basic-statistics-in-r.html#cb303-15" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb303-16"><a href="basic-statistics-in-r.html#cb303-16" aria-hidden="true" tabindex="-1"></a><span class="co">#Call:</span></span>
<span id="cb303-17"><a href="basic-statistics-in-r.html#cb303-17" aria-hidden="true" tabindex="-1"></a><span class="co">#boot(data = cars_data, statistic = function_correlation, R = 10000)</span></span>
<span id="cb303-18"><a href="basic-statistics-in-r.html#cb303-18" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb303-19"><a href="basic-statistics-in-r.html#cb303-19" aria-hidden="true" tabindex="-1"></a><span class="co">#Bootstrap Statistics :</span></span>
<span id="cb303-20"><a href="basic-statistics-in-r.html#cb303-20" aria-hidden="true" tabindex="-1"></a><span class="co">#     original       bias    std. error</span></span>
<span id="cb303-21"><a href="basic-statistics-in-r.html#cb303-21" aria-hidden="true" tabindex="-1"></a><span class="co">#t1* 0.8879799 -0.003775209  0.03749136</span></span>
<span id="cb303-22"><a href="basic-statistics-in-r.html#cb303-22" aria-hidden="true" tabindex="-1"></a>bootstrap_cor_dis_and_wt<span class="sc">$</span>t0</span>
<span id="cb303-23"><a href="basic-statistics-in-r.html#cb303-23" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 0.8879799</span></span>
<span id="cb303-24"><a href="basic-statistics-in-r.html#cb303-24" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(bootstrap_cor_dis_and_wt<span class="sc">$</span>t)</span>
<span id="cb303-25"><a href="basic-statistics-in-r.html#cb303-25" aria-hidden="true" tabindex="-1"></a><span class="co">#         [,1]</span></span>
<span id="cb303-26"><a href="basic-statistics-in-r.html#cb303-26" aria-hidden="true" tabindex="-1"></a><span class="co">#[1,] 0.8612517</span></span>
<span id="cb303-27"><a href="basic-statistics-in-r.html#cb303-27" aria-hidden="true" tabindex="-1"></a><span class="co">#[2,] 0.7704631</span></span>
<span id="cb303-28"><a href="basic-statistics-in-r.html#cb303-28" aria-hidden="true" tabindex="-1"></a><span class="co">#[3,] 0.9229136</span></span>
<span id="cb303-29"><a href="basic-statistics-in-r.html#cb303-29" aria-hidden="true" tabindex="-1"></a><span class="co">#[4,] 0.8536788</span></span>
<span id="cb303-30"><a href="basic-statistics-in-r.html#cb303-30" aria-hidden="true" tabindex="-1"></a><span class="co">#[5,] 0.8677477</span></span>
<span id="cb303-31"><a href="basic-statistics-in-r.html#cb303-31" aria-hidden="true" tabindex="-1"></a><span class="co">#[6,] 0.8298354</span></span>
<span id="cb303-32"><a href="basic-statistics-in-r.html#cb303-32" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span>
<span id="cb303-33"><a href="basic-statistics-in-r.html#cb303-33" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(bootstrap_cor_dis_and_wt<span class="sc">$</span>t)</span>
<span id="cb303-34"><a href="basic-statistics-in-r.html#cb303-34" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 10000     1</span></span>
<span id="cb303-35"><a href="basic-statistics-in-r.html#cb303-35" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(bootstrap_cor_dis_and_wt<span class="sc">$</span>t)</span>
<span id="cb303-36"><a href="basic-statistics-in-r.html#cb303-36" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 0.8842047</span></span>
<span id="cb303-37"><a href="basic-statistics-in-r.html#cb303-37" aria-hidden="true" tabindex="-1"></a><span class="fu">range</span>(bootstrap_cor_dis_and_wt<span class="sc">$</span>t)</span>
<span id="cb303-38"><a href="basic-statistics-in-r.html#cb303-38" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 0.6994762 0.9806787</span></span>
<span id="cb303-39"><a href="basic-statistics-in-r.html#cb303-39" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span>(bootstrap_cor_dis_and_wt<span class="sc">$</span>t)</span>
<span id="cb303-40"><a href="basic-statistics-in-r.html#cb303-40" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 0.03749136</span></span>
<span id="cb303-41"><a href="basic-statistics-in-r.html#cb303-41" aria-hidden="true" tabindex="-1"></a><span class="fu">boot.ci</span>(<span class="at">boot.out=</span>bootstrap_cor_dis_and_wt,<span class="at">type=</span><span class="st">&quot;norm&quot;</span>)</span>
<span id="cb303-42"><a href="basic-statistics-in-r.html#cb303-42" aria-hidden="true" tabindex="-1"></a><span class="co">#BOOTSTRAP CONFIDENCE INTERVAL CALCULATIONS</span></span>
<span id="cb303-43"><a href="basic-statistics-in-r.html#cb303-43" aria-hidden="true" tabindex="-1"></a><span class="co">#Based on 10000 bootstrap replicates</span></span>
<span id="cb303-44"><a href="basic-statistics-in-r.html#cb303-44" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb303-45"><a href="basic-statistics-in-r.html#cb303-45" aria-hidden="true" tabindex="-1"></a><span class="co">#CALL : </span></span>
<span id="cb303-46"><a href="basic-statistics-in-r.html#cb303-46" aria-hidden="true" tabindex="-1"></a><span class="co">#boot.ci(boot.out = bootstrap_cor_dis_and_wt, type = &quot;norm&quot;)</span></span>
<span id="cb303-47"><a href="basic-statistics-in-r.html#cb303-47" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb303-48"><a href="basic-statistics-in-r.html#cb303-48" aria-hidden="true" tabindex="-1"></a><span class="co">#Intervals : </span></span>
<span id="cb303-49"><a href="basic-statistics-in-r.html#cb303-49" aria-hidden="true" tabindex="-1"></a><span class="co">#Level      Normal        </span></span>
<span id="cb303-50"><a href="basic-statistics-in-r.html#cb303-50" aria-hidden="true" tabindex="-1"></a><span class="co">#95%   ( 0.8183,  0.9652 )  </span></span>
<span id="cb303-51"><a href="basic-statistics-in-r.html#cb303-51" aria-hidden="true" tabindex="-1"></a><span class="co">#Calculations and Intervals on Original Scale</span></span>
<span id="cb303-52"><a href="basic-statistics-in-r.html#cb303-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb303-53"><a href="basic-statistics-in-r.html#cb303-53" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(bootstrap_cor_dis_and_wt<span class="sc">$</span>t, <span class="at">breaks=</span><span class="dv">100</span>, <span class="at">col=</span><span class="st">&quot;firebrick2&quot;</span>, <span class="at">main=</span><span class="st">&quot;Bootstrap Distribution of the Correlation: dis vs wt&quot;</span>)</span>
<span id="cb303-54"><a href="basic-statistics-in-r.html#cb303-54" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(<span class="at">x =</span> <span class="fu">density</span>(<span class="at">x =</span> bootstrap_cor_dis_and_wt<span class="sc">$</span>t))</span>
<span id="cb303-55"><a href="basic-statistics-in-r.html#cb303-55" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> <span class="fu">mean</span>(bootstrap_cor_dis_and_wt<span class="sc">$</span>t), <span class="at">col =</span> <span class="st">&quot;black&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="figures/session_6_figs/8_boostrap_correlation.png" width="100%" /></p>
<p>The results are as follows: <strong>t0</strong> are the values of the statistic (i.e., correlation) in original dataset,
<strong>t</strong> is a matrix of 10000 rows (i.e., number of bootstrap repetitions) that we use to draw our estimate of this correlation.
The original mean value is <strong>0.8879799</strong> and the bootstrap estimate of the mean is
very close at <strong>0.8842047</strong>, with a 95% CI between <strong>0.8183</strong> and <strong>0.9652</strong>. Therefore,
<code>disp</code> and <code>wt</code> are strongly and positively correlated.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="getting-biological-data-from-public-repositories.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="regressions-and-anova-in-r.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/06-General-Statistics_v19.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
