<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Session 10 – Basic Graphics | Bioinformatics Workshop Gitbook</title>
  <meta name="description" content="none" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Session 10 – Basic Graphics | Bioinformatics Workshop Gitbook" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="none" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Session 10 – Basic Graphics | Bioinformatics Workshop Gitbook" />
  
  <meta name="twitter:description" content="none" />
  

<meta name="author" content="Juan C. Santos &amp; Randy Ortiz" />


<meta name="date" content="2024-03-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="graphics-and-colors.html"/>
<link rel="next" href="sequence-alignments.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics Workshop Gitbook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html"><i class="fa fa-check"></i><b>1</b> – Getting Started with R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#why-do-we-use-r"><i class="fa fa-check"></i><b>1.1</b> Why do we use R?</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#get-a-good-text-editor"><i class="fa fa-check"></i><b>1.2</b> Get a good text editor</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-r"><i class="fa fa-check"></i><b>1.4</b> Installing R</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#annotating-code"><i class="fa fa-check"></i><b>1.5</b> Annotating code</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#defining-a-working-directory-in-r"><i class="fa fa-check"></i><b>1.6</b> Defining a working directory in R</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html"><i class="fa fa-check"></i><b>2</b> – Using R, Installing Packages and Importing/Exporting Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#your-first-session"><i class="fa fa-check"></i><b>2.1</b> Your first session</a></li>
<li class="chapter" data-level="2.2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#doing-some-base-functions-and-getting-help"><i class="fa fa-check"></i><b>2.2</b> Doing some base functions and getting help</a></li>
<li class="chapter" data-level="2.3" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#functions-between-two-or-more-vectors"><i class="fa fa-check"></i><b>2.3</b> Functions between two or more vectors</a></li>
<li class="chapter" data-level="2.4" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#about-r-packages"><i class="fa fa-check"></i><b>2.4</b> About R-packages</a></li>
<li class="chapter" data-level="2.5" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#cran-task-views-expert-opinion-on-useful-packages"><i class="fa fa-check"></i><b>2.5</b> CRAN Task Views – Expert opinion on useful packages</a></li>
<li class="chapter" data-level="2.6" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#installing-packages"><i class="fa fa-check"></i><b>2.6</b> Installing packages</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html"><i class="fa fa-check"></i><b>3</b> – Basic Data Structures in R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#scalars-and-vectors"><i class="fa fa-check"></i><b>3.1</b> Scalars and vectors</a></li>
<li class="chapter" data-level="3.2" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#arrays"><i class="fa fa-check"></i><b>3.3</b> Arrays</a></li>
<li class="chapter" data-level="3.4" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a></li>
<li class="chapter" data-level="3.5" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tables"><i class="fa fa-check"></i><b>3.5</b> Tables</a></li>
<li class="chapter" data-level="3.6" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#lists"><i class="fa fa-check"></i><b>3.6</b> Lists</a></li>
<li class="chapter" data-level="3.7" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-tables"><i class="fa fa-check"></i><b>3.7</b> Data Tables</a></li>
<li class="chapter" data-level="3.8" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tibbles"><i class="fa fa-check"></i><b>3.8</b> Tibbles</a></li>
<li class="chapter" data-level="3.9" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#dnastringsets-rnastringsets-and-aastringsets"><i class="fa fa-check"></i><b>3.9</b> DNAStringSets, RNAStringSets and AAStringSets</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html"><i class="fa fa-check"></i><b>4</b> – Text Editing and Data Transformations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#working-with-character-strings"><i class="fa fa-check"></i><b>4.1</b> Working with character strings</a></li>
<li class="chapter" data-level="4.2" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#regular-expressions-regex"><i class="fa fa-check"></i><b>4.2</b> Regular expressions (Regex)</a></li>
<li class="chapter" data-level="4.3" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#comparing-ordering-and-simplifying-vectors"><i class="fa fa-check"></i><b>4.3</b> Comparing, ordering and simplifying vectors</a></li>
<li class="chapter" data-level="4.4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#useful-functions-of-stringr"><i class="fa fa-check"></i><b>4.4</b> Useful functions of <code>stringr</code></a></li>
<li class="chapter" data-level="4.5" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#data-transformations"><i class="fa fa-check"></i><b>4.5</b> Data transformations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html"><i class="fa fa-check"></i><b>5</b> – Getting Biological Data from Public Repositories</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#ncbi-databases-genbank-and-others"><i class="fa fa-check"></i><b>5.1</b> NCBI databases: GenBank and others</a></li>
<li class="chapter" data-level="5.2" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#accessing-ncbi-within-r-rentrez"><i class="fa fa-check"></i><b>5.2</b> Accessing NCBI within R: <code>rentrez</code></a></li>
<li class="chapter" data-level="5.3" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#retrieving-pubmed-references"><i class="fa fa-check"></i><b>5.3</b> Retrieving PubMed references</a></li>
<li class="chapter" data-level="5.4" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#comparing-sequences-blast"><i class="fa fa-check"></i><b>5.4</b> Comparing sequences: BLAST</a></li>
<li class="chapter" data-level="5.5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-uniprot"><i class="fa fa-check"></i><b>5.5</b> Other databases: UniProt</a></li>
<li class="chapter" data-level="5.6" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-ensembl"><i class="fa fa-check"></i><b>5.6</b> Other databases: Ensembl</a></li>
<li class="chapter" data-level="5.7" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-taxonomy-phylogenetic-trees-and-species-distributions"><i class="fa fa-check"></i><b>5.7</b> Other databases: Taxonomy, phylogenetic trees and species distributions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html"><i class="fa fa-check"></i><b>6</b> – Basic Statistics in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#exploring-your-data"><i class="fa fa-check"></i><b>6.1</b> Exploring your data</a></li>
<li class="chapter" data-level="6.2" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#sub-setting-your-dataset"><i class="fa fa-check"></i><b>6.2</b> Sub-setting your dataset</a></li>
<li class="chapter" data-level="6.3" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#transposing-rotating-your-dataset"><i class="fa fa-check"></i><b>6.3</b> Transposing (rotating) your dataset</a></li>
<li class="chapter" data-level="6.4" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#dealing-with-missing-na-values"><i class="fa fa-check"></i><b>6.4</b> Dealing with missing (NA) values</a></li>
<li class="chapter" data-level="6.5" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#about-p-values"><i class="fa fa-check"></i><b>6.5</b> About p-values</a></li>
<li class="chapter" data-level="6.6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#how-to-read-p-values"><i class="fa fa-check"></i><b>6.6</b> How to read p-values</a></li>
<li class="chapter" data-level="6.7" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#multiple-testing-corrections"><i class="fa fa-check"></i><b>6.7</b> Multiple testing corrections</a></li>
<li class="chapter" data-level="6.8" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#normalizing-your-data"><i class="fa fa-check"></i><b>6.8</b> Normalizing your data</a></li>
<li class="chapter" data-level="6.9" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#what-test-should-i-use"><i class="fa fa-check"></i><b>6.9</b> What test should I use?</a></li>
<li class="chapter" data-level="6.10" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#type-of-variables-in-dataset"><i class="fa fa-check"></i><b>6.10</b> Type of variables in dataset</a></li>
<li class="chapter" data-level="6.11" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-means-t-test"><i class="fa fa-check"></i><b>6.11</b> Comparing means: t-test</a></li>
<li class="chapter" data-level="6.12" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#two-sample-t-test"><i class="fa fa-check"></i><b>6.12</b> Two sample t-test</a></li>
<li class="chapter" data-level="6.13" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-wilcoxon-signed-rank-test"><i class="fa fa-check"></i><b>6.13</b> Beyond t-test: Wilcoxon signed rank test</a></li>
<li class="chapter" data-level="6.14" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#tabulation-and-cross-tabulation"><i class="fa fa-check"></i><b>6.14</b> Tabulation and Cross-Tabulation</a></li>
<li class="chapter" data-level="6.15" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-binomial-test"><i class="fa fa-check"></i><b>6.15</b> Beyond t-test: Binomial test</a></li>
<li class="chapter" data-level="6.16" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-chi-square-goodness-of-fit"><i class="fa fa-check"></i><b>6.16</b> Beyond t-test: Chi-square goodness of fit</a></li>
<li class="chapter" data-level="6.17" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-mcnemars-test"><i class="fa fa-check"></i><b>6.17</b> Beyond t-test: McNemar’s test</a></li>
<li class="chapter" data-level="6.18" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-fishers-exact-test"><i class="fa fa-check"></i><b>6.18</b> Beyond t-test: Fisher’s exact test</a></li>
<li class="chapter" data-level="6.19" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#correlations"><i class="fa fa-check"></i><b>6.19</b> Correlations</a></li>
<li class="chapter" data-level="6.20" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-distributions"><i class="fa fa-check"></i><b>6.20</b> Comparing distributions</a></li>
<li class="chapter" data-level="6.21" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#bootstraps"><i class="fa fa-check"></i><b>6.21</b> Bootstraps</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html"><i class="fa fa-check"></i><b>7</b> – Regressions and ANOVA in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#linear-regressions"><i class="fa fa-check"></i><b>7.1</b> Linear regressions</a></li>
<li class="chapter" data-level="7.2" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anatomy-of-a-linear-regression"><i class="fa fa-check"></i><b>7.2</b> Anatomy of a linear regression</a></li>
<li class="chapter" data-level="7.3" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#multiple-linear-regressions"><i class="fa fa-check"></i><b>7.3</b> Multiple linear regressions</a></li>
<li class="chapter" data-level="7.4" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#diagnostic-plots-for-regressions"><i class="fa fa-check"></i><b>7.4</b> Diagnostic plots for regressions</a></li>
<li class="chapter" data-level="7.5" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-influential-measurements"><i class="fa fa-check"></i><b>7.5</b> Regression influential measurements</a></li>
<li class="chapter" data-level="7.6" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-anova-coefficients-ci"><i class="fa fa-check"></i><b>7.6</b> Regression (ANOVA, coefficients, CI)</a></li>
<li class="chapter" data-level="7.7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#selecting-predictors"><i class="fa fa-check"></i><b>7.7</b> Selecting predictors</a></li>
<li class="chapter" data-level="7.8" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-predictions"><i class="fa fa-check"></i><b>7.8</b> Regression predictions</a></li>
<li class="chapter" data-level="7.9" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions"><i class="fa fa-check"></i><b>7.9</b> Logistic regressions</a></li>
<li class="chapter" data-level="7.10" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions-requirements"><i class="fa fa-check"></i><b>7.10</b> Logistic regressions requirements</a></li>
<li class="chapter" data-level="7.11" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#non-linear-regressions"><i class="fa fa-check"></i><b>7.11</b> Non-linear regressions</a></li>
<li class="chapter" data-level="7.12" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-regressions"><i class="fa fa-check"></i><b>7.12</b> Toy datasets for regressions</a></li>
<li class="chapter" data-level="7.13" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova"><i class="fa fa-check"></i><b>7.13</b> ANOVA</a></li>
<li class="chapter" data-level="7.14" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#one-way-anova"><i class="fa fa-check"></i><b>7.14</b> One-way ANOVA</a></li>
<li class="chapter" data-level="7.15" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#two-way-anova"><i class="fa fa-check"></i><b>7.15</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="7.16" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova-post-hoc-tests"><i class="fa fa-check"></i><b>7.16</b> ANOVA Post-hoc tests</a></li>
<li class="chapter" data-level="7.17" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#beyond-anova-kruskal-wallis-test"><i class="fa fa-check"></i><b>7.17</b> Beyond ANOVA: Kruskal-Wallis Test</a></li>
<li class="chapter" data-level="7.18" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-anova"><i class="fa fa-check"></i><b>7.18</b> Toy datasets for ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html"><i class="fa fa-check"></i><b>8</b> – Multidimensional Data in R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#principal-component-analysis"><i class="fa fa-check"></i><b>8.1</b> Principal Component Analysis</a></li>
<li class="chapter" data-level="8.2" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#multidimensional-scaling"><i class="fa fa-check"></i><b>8.2</b> Multidimensional scaling</a></li>
<li class="chapter" data-level="8.3" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-dfa"><i class="fa fa-check"></i><b>8.3</b> Cluster analysis: DFA</a></li>
<li class="chapter" data-level="8.4" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-k-means-unsupervised"><i class="fa fa-check"></i><b>8.4</b> Cluster analysis: <em>k</em>-means (unsupervised)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html"><i class="fa fa-check"></i><b>9</b> – Graphics and Colors</a>
<ul>
<li class="chapter" data-level="9.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#avoid-bad-graphics"><i class="fa fa-check"></i><b>9.1</b> Avoid bad graphics</a></li>
<li class="chapter" data-level="9.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#graphics-in-r-ggplot2"><i class="fa fa-check"></i><b>9.2</b> Graphics in R – ggplot2</a></li>
<li class="chapter" data-level="9.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#preparing-your-dataset"><i class="fa fa-check"></i><b>9.3</b> Preparing your dataset</a></li>
<li class="chapter" data-level="9.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#color-palettes-online"><i class="fa fa-check"></i><b>9.4</b> Color palettes online</a></li>
<li class="chapter" data-level="9.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggplot2-default-palette"><i class="fa fa-check"></i><b>9.5</b> <code>ggplot2</code> default palette</a></li>
<li class="chapter" data-level="9.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#heatmaps-or-tiled-charts"><i class="fa fa-check"></i><b>9.6</b> Heatmaps or tiled charts</a></li>
<li class="chapter" data-level="9.7" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#r-packages-about-color"><i class="fa fa-check"></i><b>9.7</b> R-packages about color</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#viridis-package"><i class="fa fa-check"></i><b>9.7.1</b> <code>viridis</code> package</a></li>
<li class="chapter" data-level="9.7.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcolorbrewer-package"><i class="fa fa-check"></i><b>9.7.2</b> <code>RColorBrewer</code> package</a></li>
<li class="chapter" data-level="9.7.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggsci-package"><i class="fa fa-check"></i><b>9.7.3</b> <code>ggsci</code> package</a></li>
<li class="chapter" data-level="9.7.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#wesanderson-package"><i class="fa fa-check"></i><b>9.7.4</b> <code>wesanderson</code> package</a></li>
<li class="chapter" data-level="9.7.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcartocolor-package"><i class="fa fa-check"></i><b>9.7.5</b> <code>rcartocolor</code> package</a></li>
<li class="chapter" data-level="9.7.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#colorspace-package"><i class="fa fa-check"></i><b>9.7.6</b> <code>colorspace</code> package</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#making-your-own-palette"><i class="fa fa-check"></i><b>9.8</b> Making your own palette</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="basic-graphics.html"><a href="basic-graphics.html"><i class="fa fa-check"></i><b>10</b> – Basic Graphics</a>
<ul>
<li class="chapter" data-level="10.1" data-path="basic-graphics.html"><a href="basic-graphics.html#what-would-you-like-to-show"><i class="fa fa-check"></i><b>10.1</b> What would you like to show?</a></li>
<li class="chapter" data-level="10.2" data-path="basic-graphics.html"><a href="basic-graphics.html#bar-charts"><i class="fa fa-check"></i><b>10.2</b> Bar charts</a></li>
<li class="chapter" data-level="10.3" data-path="basic-graphics.html"><a href="basic-graphics.html#stacked-bar-charts"><i class="fa fa-check"></i><b>10.3</b> Stacked bar charts</a></li>
<li class="chapter" data-level="10.4" data-path="basic-graphics.html"><a href="basic-graphics.html#histograms"><i class="fa fa-check"></i><b>10.4</b> Histograms</a></li>
<li class="chapter" data-level="10.5" data-path="basic-graphics.html"><a href="basic-graphics.html#multiple-histograms"><i class="fa fa-check"></i><b>10.5</b> Multiple histograms</a></li>
<li class="chapter" data-level="10.6" data-path="basic-graphics.html"><a href="basic-graphics.html#density-plots"><i class="fa fa-check"></i><b>10.6</b> Density plots</a></li>
<li class="chapter" data-level="10.7" data-path="basic-graphics.html"><a href="basic-graphics.html#box-plots"><i class="fa fa-check"></i><b>10.7</b> Box plots</a></li>
<li class="chapter" data-level="10.8" data-path="basic-graphics.html"><a href="basic-graphics.html#violin-plots"><i class="fa fa-check"></i><b>10.8</b> Violin plots</a></li>
<li class="chapter" data-level="10.9" data-path="basic-graphics.html"><a href="basic-graphics.html#scatterplots"><i class="fa fa-check"></i><b>10.9</b> Scatterplots</a></li>
<li class="chapter" data-level="10.10" data-path="basic-graphics.html"><a href="basic-graphics.html#lines-and-scatterplots"><i class="fa fa-check"></i><b>10.10</b> Lines and Scatterplots</a></li>
<li class="chapter" data-level="10.11" data-path="basic-graphics.html"><a href="basic-graphics.html#save-your-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>10.11</b> Save your plots as PDF or PNG</a></li>
<li class="chapter" data-level="10.12" data-path="basic-graphics.html"><a href="basic-graphics.html#ggplot2-add-on-packages-treemapify"><i class="fa fa-check"></i><b>10.12</b> ggplot2 add-on packages: <code>treemapify</code></a></li>
<li class="chapter" data-level="10.13" data-path="basic-graphics.html"><a href="basic-graphics.html#other-resources"><i class="fa fa-check"></i><b>10.13</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="sequence-alignments.html"><a href="sequence-alignments.html"><i class="fa fa-check"></i><b>11</b> – Sequence Alignments</a>
<ul>
<li class="chapter" data-level="11.1" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-using-rentrez"><i class="fa fa-check"></i><b>11.1</b> Retrieving sequences using ‘rentrez’</a></li>
<li class="chapter" data-level="11.2" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-manually-from-ncbi"><i class="fa fa-check"></i><b>11.2</b> Retrieving sequences manually from NCBI</a></li>
<li class="chapter" data-level="11.3" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-accession-numbers"><i class="fa fa-check"></i><b>11.3</b> Retrieving accession numbers</a></li>
<li class="chapter" data-level="11.4" data-path="sequence-alignments.html"><a href="sequence-alignments.html#alignment-with-decipher"><i class="fa fa-check"></i><b>11.4</b> Alignment with <code>DECIPHER</code></a></li>
<li class="chapter" data-level="11.5" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualize-alignment"><i class="fa fa-check"></i><b>11.5</b> Visualize alignment</a></li>
<li class="chapter" data-level="11.6" data-path="sequence-alignments.html"><a href="sequence-alignments.html#save-alignment-to-file"><i class="fa fa-check"></i><b>11.6</b> Save alignment to file</a></li>
<li class="chapter" data-level="11.7" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-and-align-several-markers"><i class="fa fa-check"></i><b>11.7</b> Concatenate and align several markers</a></li>
<li class="chapter" data-level="11.8" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-data.table"><i class="fa fa-check"></i><b>11.8</b> Concatenate with <code>data.table</code></a></li>
<li class="chapter" data-level="11.9" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-biostrings"><i class="fa fa-check"></i><b>11.9</b> Concatenate with <code>Biostrings</code></a></li>
<li class="chapter" data-level="11.10" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualizing-alignments-in-r"><i class="fa fa-check"></i><b>11.10</b> Visualizing alignments in R</a></li>
<li class="chapter" data-level="11.11" data-path="sequence-alignments.html"><a href="sequence-alignments.html#other-sequence-alignment-software"><i class="fa fa-check"></i><b>11.11</b> Other sequence alignment software</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html"><i class="fa fa-check"></i><b>12</b> – Estimating Phylogenetic Trees</a>
<ul>
<li class="chapter" data-level="12.1" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#phylogeny-inference-with-iqtree2"><i class="fa fa-check"></i><b>12.1</b> Phylogeny inference with <code>iqtree2</code></a></li>
<li class="chapter" data-level="12.2" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-iqtree2-in-macos"><i class="fa fa-check"></i><b>12.2</b> Installing <code>iqtree2</code> in macOS</a></li>
<li class="chapter" data-level="12.3" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-and-running-iqtree2-in-a-pc"><i class="fa fa-check"></i><b>12.3</b> Installing and running <code>iqtree2</code> in a PC</a></li>
<li class="chapter" data-level="12.4" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#loading-the-iqtree_runner_r-function"><i class="fa fa-check"></i><b>12.4</b> Loading the <code>iqtree_runner_R</code> function</a></li>
<li class="chapter" data-level="12.5" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#running-iqtree2-from-r"><i class="fa fa-check"></i><b>12.5</b> Running <code>iqtree2</code> from R</a></li>
<li class="chapter" data-level="12.6" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-no-partitions"><i class="fa fa-check"></i><b>12.6</b> One marker and no partitions</a></li>
<li class="chapter" data-level="12.7" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-with-partitions"><i class="fa fa-check"></i><b>12.7</b> One marker and with partitions</a></li>
<li class="chapter" data-level="12.8" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#bootstrap-support"><i class="fa fa-check"></i><b>12.8</b> Bootstrap support</a></li>
<li class="chapter" data-level="12.9" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#many-markers-and-with-partitions"><i class="fa fa-check"></i><b>12.9</b> Many markers and with partitions</a></li>
<li class="chapter" data-level="12.10" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#other-sequence-phylogenetic-tree-estimation-software"><i class="fa fa-check"></i><b>12.10</b> Other sequence phylogenetic tree estimation software</a></li>
<li class="chapter" data-level="12.11" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#drawing-phylogenetic-trees-with-ggtree"><i class="fa fa-check"></i><b>12.11</b> Drawing phylogenetic trees with <code>ggtree</code></a></li>
<li class="chapter" data-level="12.12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#save-ggtree-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>12.12</b> Save ggtree plots as PDF or PNG</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html"><i class="fa fa-check"></i><b>13</b> – Mining from a reference dataset</a>
<ul>
<li class="chapter" data-level="13.1" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#a-worked-example-using-the-mtcars-dataset"><i class="fa fa-check"></i><b>13.1</b> A worked example using the <code>mtcars</code> dataset</a></li>
<li class="chapter" data-level="13.2" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-reference-dataset"><i class="fa fa-check"></i><b>13.2</b> Build a reference dataset</a></li>
<li class="chapter" data-level="13.3" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-sampling-dataset"><i class="fa fa-check"></i><b>13.3</b> Build a sampling dataset</a></li>
<li class="chapter" data-level="13.4" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#extracting-from-reference-dataset"><i class="fa fa-check"></i><b>13.4</b> Extracting from reference dataset</a></li>
<li class="chapter" data-level="13.5" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#plots-of-the-extracted-data"><i class="fa fa-check"></i><b>13.5</b> Plots of the extracted data</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="introduction-to-python.html"><a href="introduction-to-python.html"><i class="fa fa-check"></i><b>14</b> – Introduction to Python</a>
<ul>
<li class="chapter" data-level="14.1" data-path="introduction-to-python.html"><a href="introduction-to-python.html#welcome-to-python-why-python"><i class="fa fa-check"></i><b>14.1</b> Welcome to Python! Why Python?</a></li>
<li class="chapter" data-level="14.2" data-path="introduction-to-python.html"><a href="introduction-to-python.html#which-python-should-i-install"><i class="fa fa-check"></i><b>14.2</b> Which Python should I install?</a></li>
<li class="chapter" data-level="14.3" data-path="introduction-to-python.html"><a href="introduction-to-python.html#mac-os-x-python-installations"><i class="fa fa-check"></i><b>14.3</b> Mac OS X Python Installations</a></li>
<li class="chapter" data-level="14.4" data-path="introduction-to-python.html"><a href="introduction-to-python.html#windows-python-installations"><i class="fa fa-check"></i><b>14.4</b> Windows Python Installations</a></li>
<li class="chapter" data-level="14.5" data-path="introduction-to-python.html"><a href="introduction-to-python.html#install-an-ide-or-integrated-development-environment"><i class="fa fa-check"></i><b>14.5</b> Install an IDE, or integrated development environment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html"><i class="fa fa-check"></i><b>15</b> – Python: Keywords and Identifiers</a>
<ul>
<li class="chapter" data-level="15.1" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#keywords-in-python"><i class="fa fa-check"></i><b>15.1</b> Keywords in Python</a></li>
<li class="chapter" data-level="15.2" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#identifiers-in-python"><i class="fa fa-check"></i><b>15.2</b> Identifiers in Python</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html"><i class="fa fa-check"></i><b>16</b> – Python: Statements and Expressions</a>
<ul>
<li class="chapter" data-level="16.1" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#what-are-statements"><i class="fa fa-check"></i><b>16.1</b> What are statements?</a></li>
<li class="chapter" data-level="16.2" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#expressions-in-python"><i class="fa fa-check"></i><b>16.2</b> Expressions in Python</a></li>
<li class="chapter" data-level="16.3" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#simple-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.3</b> Simple Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.4" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#augmented-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.4</b> Augmented Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.5" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#explicit-and-implicit-line-continuation"><i class="fa fa-check"></i><b>16.5</b> Explicit and Implicit Line Continuation</a></li>
<li class="chapter" data-level="16.6" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#indentation-in-python-is-crucial-to-proper-code-function"><i class="fa fa-check"></i><b>16.6</b> Indentation in Python is Crucial to Proper Code function</a></li>
<li class="chapter" data-level="16.7" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#comments"><i class="fa fa-check"></i><b>16.7</b> Comments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="python-data-types.html"><a href="python-data-types.html"><i class="fa fa-check"></i><b>17</b> – Python: Data Types</a>
<ul>
<li class="chapter" data-level="17.1" data-path="python-data-types.html"><a href="python-data-types.html#booleans"><i class="fa fa-check"></i><b>17.1</b> Booleans</a></li>
<li class="chapter" data-level="17.2" data-path="python-data-types.html"><a href="python-data-types.html#numbers-int-float-and-complex"><i class="fa fa-check"></i><b>17.2</b> Numbers: int, float and complex</a></li>
<li class="chapter" data-level="17.3" data-path="python-data-types.html"><a href="python-data-types.html#strings"><i class="fa fa-check"></i><b>17.3</b> Strings</a></li>
<li class="chapter" data-level="17.4" data-path="python-data-types.html"><a href="python-data-types.html#bytes"><i class="fa fa-check"></i><b>17.4</b> Bytes</a></li>
<li class="chapter" data-level="17.5" data-path="python-data-types.html"><a href="python-data-types.html#lists-1"><i class="fa fa-check"></i><b>17.5</b> Lists</a></li>
<li class="chapter" data-level="17.6" data-path="python-data-types.html"><a href="python-data-types.html#tuples"><i class="fa fa-check"></i><b>17.6</b> Tuples</a></li>
<li class="chapter" data-level="17.7" data-path="python-data-types.html"><a href="python-data-types.html#sets"><i class="fa fa-check"></i><b>17.7</b> Sets</a></li>
<li class="chapter" data-level="17.8" data-path="python-data-types.html"><a href="python-data-types.html#dictionary"><i class="fa fa-check"></i><b>17.8</b> Dictionary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="python-strings.html"><a href="python-strings.html"><i class="fa fa-check"></i><b>18</b> – Python: Strings</a>
<ul>
<li class="chapter" data-level="18.1" data-path="python-strings.html"><a href="python-strings.html#string-operators"><i class="fa fa-check"></i><b>18.1</b> String Operators</a></li>
<li class="chapter" data-level="18.2" data-path="python-strings.html"><a href="python-strings.html#other-string-modifiers"><i class="fa fa-check"></i><b>18.2</b> Other String Modifiers</a></li>
<li class="chapter" data-level="18.3" data-path="python-strings.html"><a href="python-strings.html#format-characters-in-strings"><i class="fa fa-check"></i><b>18.3</b> Format-Characters in Strings</a></li>
<li class="chapter" data-level="18.4" data-path="python-strings.html"><a href="python-strings.html#comparisons-with-strings"><i class="fa fa-check"></i><b>18.4</b> Comparisons with Strings</a></li>
<li class="chapter" data-level="18.5" data-path="python-strings.html"><a href="python-strings.html#padding-strings"><i class="fa fa-check"></i><b>18.5</b> Padding strings</a></li>
<li class="chapter" data-level="18.6" data-path="python-strings.html"><a href="python-strings.html#search-for-words-in-string-and-output-location"><i class="fa fa-check"></i><b>18.6</b> Search for words in string and output location</a></li>
<li class="chapter" data-level="18.7" data-path="python-strings.html"><a href="python-strings.html#seek-and-replace-in-strings"><i class="fa fa-check"></i><b>18.7</b> Seek and replace in strings</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="python-operators.html"><a href="python-operators.html"><i class="fa fa-check"></i><b>19</b> – Python: Operators</a>
<ul>
<li class="chapter" data-level="19.1" data-path="python-operators.html"><a href="python-operators.html#many-types-of-operators"><i class="fa fa-check"></i><b>19.1</b> Many types of operators</a></li>
<li class="chapter" data-level="19.2" data-path="python-operators.html"><a href="python-operators.html#arithmetic-operators"><i class="fa fa-check"></i><b>19.2</b> Arithmetic operators</a></li>
<li class="chapter" data-level="19.3" data-path="python-operators.html"><a href="python-operators.html#comparison-operators"><i class="fa fa-check"></i><b>19.3</b> Comparison operators</a></li>
<li class="chapter" data-level="19.4" data-path="python-operators.html"><a href="python-operators.html#logical-operators"><i class="fa fa-check"></i><b>19.4</b> Logical operators</a></li>
<li class="chapter" data-level="19.5" data-path="python-operators.html"><a href="python-operators.html#bitwise-operators"><i class="fa fa-check"></i><b>19.5</b> Bitwise operators</a></li>
<li class="chapter" data-level="19.6" data-path="python-operators.html"><a href="python-operators.html#identity-operators"><i class="fa fa-check"></i><b>19.6</b> Identity Operators</a></li>
<li class="chapter" data-level="19.7" data-path="python-operators.html"><a href="python-operators.html#membership-operators"><i class="fa fa-check"></i><b>19.7</b> Membership operators</a></li>
<li class="chapter" data-level="19.8" data-path="python-operators.html"><a href="python-operators.html#what-is-the-order-of-operations-with-operators"><i class="fa fa-check"></i><b>19.8</b> What is the order of operations with operators?</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="biopython.html"><a href="biopython.html"><i class="fa fa-check"></i><b>20</b> – Biopython</a>
<ul>
<li class="chapter" data-level="20.1" data-path="biopython.html"><a href="biopython.html#introduction-to-biopython"><i class="fa fa-check"></i><b>20.1</b> Introduction to Biopython</a></li>
<li class="chapter" data-level="20.2" data-path="biopython.html"><a href="biopython.html#biopython-installation"><i class="fa fa-check"></i><b>20.2</b> Biopython Installation</a></li>
<li class="chapter" data-level="20.3" data-path="biopython.html"><a href="biopython.html#getting-started"><i class="fa fa-check"></i><b>20.3</b> Getting Started</a></li>
<li class="chapter" data-level="20.4" data-path="biopython.html"><a href="biopython.html#parsing-my-fasta"><i class="fa fa-check"></i><b>20.4</b> Parsing my FASTA</a></li>
<li class="chapter" data-level="20.5" data-path="biopython.html"><a href="biopython.html#parsing-my-genbank-or-genpept"><i class="fa fa-check"></i><b>20.5</b> Parsing my GenBank or GenPept</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="biopython-sequences.html"><a href="biopython-sequences.html"><i class="fa fa-check"></i><b>21</b> – Biopython: Sequences</a>
<ul>
<li class="chapter" data-level="21.1" data-path="biopython-sequences.html"><a href="biopython-sequences.html#sequence-objects"><i class="fa fa-check"></i><b>21.1</b> Sequence Objects</a></li>
<li class="chapter" data-level="21.2" data-path="biopython-sequences.html"><a href="biopython-sequences.html#concatenation-with-sequences"><i class="fa fa-check"></i><b>21.2</b> Concatenation with Sequences</a></li>
<li class="chapter" data-level="21.3" data-path="biopython-sequences.html"><a href="biopython-sequences.html#transcription-dna-to-mrna"><i class="fa fa-check"></i><b>21.3</b> Transcription: DNA to mRNA</a></li>
<li class="chapter" data-level="21.4" data-path="biopython-sequences.html"><a href="biopython-sequences.html#translation-rna-to-prot"><i class="fa fa-check"></i><b>21.4</b> Translation: RNA to PROT</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html"><i class="fa fa-check"></i><b>22</b> – Biopython: Mulitple Sequence Alignments</a>
<ul>
<li class="chapter" data-level="22.1" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwise2"><i class="fa fa-check"></i><b>22.1</b> pairwise2</a></li>
<li class="chapter" data-level="22.2" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwisealigner"><i class="fa fa-check"></i><b>22.2</b> PairwiseAligner</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html"><i class="fa fa-check"></i><b>23</b> – Biopython: GenomeDiagram</a>
<ul>
<li class="chapter" data-level="23.1" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#linearcircular-gene-diagrams"><i class="fa fa-check"></i><b>23.1</b> Linear/Circular Gene Diagrams</a></li>
<li class="chapter" data-level="23.2" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#diagram-readability"><i class="fa fa-check"></i><b>23.2</b> Diagram Readability</a></li>
<li class="chapter" data-level="23.3" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#box-shape"><i class="fa fa-check"></i><b>23.3</b> Box Shape</a></li>
<li class="chapter" data-level="23.4" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#edit-box-shape-arrows"><i class="fa fa-check"></i><b>23.4</b> Edit Box Shape: Arrows</a></li>
<li class="chapter" data-level="23.5" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#restriction-map"><i class="fa fa-check"></i><b>23.5</b> Restriction Map</a></li>
<li class="chapter" data-level="23.6" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#tracks-on-tracks"><i class="fa fa-check"></i><b>23.6</b> Tracks on Tracks</a></li>
<li class="chapter" data-level="23.7" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#cross-link-data"><i class="fa fa-check"></i><b>23.7</b> Cross-link Data</a></li>
<li class="chapter" data-level="23.8" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#clean-cross-link"><i class="fa fa-check"></i><b>23.8</b> Clean Cross-Link</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html"><i class="fa fa-check"></i><b>24</b> – Biopython: Chromosome Modeling</a>
<ul>
<li class="chapter" data-level="24.1" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html#gene-feature-modeling"><i class="fa fa-check"></i><b>24.1</b> Gene Feature Modeling</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html"><i class="fa fa-check"></i><b>25</b> – Biopython: Sequence Parsing with Plots</a>
<ul>
<li class="chapter" data-level="25.1" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#histograms-1"><i class="fa fa-check"></i><b>25.1</b> Histograms</a></li>
<li class="chapter" data-level="25.2" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#gc-content"><i class="fa fa-check"></i><b>25.2</b> GC Content</a></li>
<li class="chapter" data-level="25.3" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#nt-alignment-dot-plot"><i class="fa fa-check"></i><b>25.3</b> Nt Alignment Dot Plot</a></li>
<li class="chapter" data-level="25.4" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#bigger-dots-for-the-dot-plot"><i class="fa fa-check"></i><b>25.4</b> Bigger Dots for the Dot Plot</a></li>
<li class="chapter" data-level="25.5" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#fastq-we-did-not-forget-you"><i class="fa fa-check"></i><b>25.5</b> FASTQ we did not forget you</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html"><i class="fa fa-check"></i><b>26</b> – Python: Web Scraping w/ BeautifulSoup</a>
<ul>
<li class="chapter" data-level="26.1" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#introduction"><i class="fa fa-check"></i><b>26.1</b> Introduction</a></li>
<li class="chapter" data-level="26.2" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#installation"><i class="fa fa-check"></i><b>26.2</b> Installation</a></li>
<li class="chapter" data-level="26.3" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#getting-started-1"><i class="fa fa-check"></i><b>26.3</b> Getting started</a></li>
<li class="chapter" data-level="26.4" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-all-matching-tags"><i class="fa fa-check"></i><b>26.4</b> Mining all matching tags</a></li>
<li class="chapter" data-level="26.5" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-tags-by-class-or-id"><i class="fa fa-check"></i><b>26.5</b> Mining tags by class or id</a></li>
<li class="chapter" data-level="26.6" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#use-css-selectors-for-positional-mining"><i class="fa fa-check"></i><b>26.6</b> Use CSS Selectors for positional mining</a></li>
<li class="chapter" data-level="26.7" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#weather-page-example"><i class="fa fa-check"></i><b>26.7</b> Weather Page Example</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="python-subprocess.html"><a href="python-subprocess.html"><i class="fa fa-check"></i><b>27</b> – Python Subprocess</a>
<ul>
<li class="chapter" data-level="27.1" data-path="python-subprocess.html"><a href="python-subprocess.html#introduction-and-installation"><i class="fa fa-check"></i><b>27.1</b> Introduction and Installation</a></li>
<li class="chapter" data-level="27.2" data-path="python-subprocess.html"><a href="python-subprocess.html#creating-your-first-python-script-w-bbtools"><i class="fa fa-check"></i><b>27.2</b> Creating your first python script w/ BBTools</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html"><i class="fa fa-check"></i><b>28</b> – Control Structures &amp; Loops in R</a>
<ul>
<li class="chapter" data-level="28.1" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#controling-the-flow"><i class="fa fa-check"></i><b>28.1</b> Controling the flow</a></li>
<li class="chapter" data-level="28.2" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if"><i class="fa fa-check"></i><b>28.2</b> Conditional <code>if</code></a></li>
<li class="chapter" data-level="28.3" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if-and-else"><i class="fa fa-check"></i><b>28.3</b> Conditional <code>if</code> and <code>else</code></a></li>
<li class="chapter" data-level="28.4" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#nested-conditionals-if-else-and-if-else"><i class="fa fa-check"></i><b>28.4</b> Nested conditionals <code>if</code>, <code>else</code>, and <code>if else</code></a></li>
<li class="chapter" data-level="28.5" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-ifelse"><i class="fa fa-check"></i><b>28.5</b> Conditional <code>ifelse</code></a></li>
<li class="chapter" data-level="28.6" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#loops-that-control-iterations-and-repeations"><i class="fa fa-check"></i><b>28.6</b> Loops that control iterations and repeations</a></li>
<li class="chapter" data-level="28.7" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-for-loop"><i class="fa fa-check"></i><b>28.7</b> The <code>for</code> Loop</a></li>
<li class="chapter" data-level="28.8" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-while-loop"><i class="fa fa-check"></i><b>28.8</b> The <code>while</code> Loop</a></li>
<li class="chapter" data-level="28.9" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-repeat-loop"><i class="fa fa-check"></i><b>28.9</b> The <code>repeat</code> Loop</a></li>
<li class="chapter" data-level="28.10" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-next-and-break-statements"><i class="fa fa-check"></i><b>28.10</b> The <code>next</code> and <code>break</code> statements</a></li>
<li class="chapter" data-level="28.11" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functions"><i class="fa fa-check"></i><b>28.11</b> Other loop functions</a></li>
<li class="chapter" data-level="28.12" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#lapply-functions"><i class="fa fa-check"></i><b>28.12</b> <code>lapply</code> functions</a></li>
<li class="chapter" data-level="28.13" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsapply"><i class="fa fa-check"></i><b>28.13</b> Other loop functions:<code>apply</code></a></li>
<li class="chapter" data-level="28.14" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsmapply"><i class="fa fa-check"></i><b>28.14</b> Other loop functions:<code>mapply</code></a></li>
<li class="chapter" data-level="28.15" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionstapply"><i class="fa fa-check"></i><b>28.15</b> Other loop functions:<code>tapply</code></a></li>
<li class="chapter" data-level="28.16" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionssplit"><i class="fa fa-check"></i><b>28.16</b> Other loop functions:<code>split</code></a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html"><i class="fa fa-check"></i><b>29</b> – Bioinformatics Literature</a>
<ul>
<li class="chapter" data-level="29.1" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#peer-reviewed-bioinformatics-journals"><i class="fa fa-check"></i><b>29.1</b> Peer-reviewed bioinformatics journals</a></li>
<li class="chapter" data-level="29.2" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#access-to-journals-articles"><i class="fa fa-check"></i><b>29.2</b> Access to journals articles</a></li>
<li class="chapter" data-level="29.3" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#informal-access-to-code-preprints-and-vignettes"><i class="fa fa-check"></i><b>29.3</b> Informal access to code: Preprints and vignettes</a></li>
<li class="chapter" data-level="29.4" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#online-bioinformatics-courses"><i class="fa fa-check"></i><b>29.4</b> Online bioinformatics courses</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html"><i class="fa fa-check"></i><b>30</b> – Importing/Exporting Data into R</a>
<ul>
<li class="chapter" data-level="30.1" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#typing-directly"><i class="fa fa-check"></i><b>30.1</b> Typing directly</a></li>
<li class="chapter" data-level="30.2" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#summary-versus-raw-data"><i class="fa fa-check"></i><b>30.2</b> Summary versus raw data</a></li>
<li class="chapter" data-level="30.3" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#downloading-data-files-github"><i class="fa fa-check"></i><b>30.3</b> Downloading data files (GitHub)</a></li>
<li class="chapter" data-level="30.4" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#importing-data-from-a-file"><i class="fa fa-check"></i><b>30.4</b> Importing data from a file</a></li>
<li class="chapter" data-level="30.5" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-to-a-file"><i class="fa fa-check"></i><b>30.5</b> Saving data to a file</a></li>
<li class="chapter" data-level="30.6" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-as-rdata-.rdata-.rda"><i class="fa fa-check"></i><b>30.6</b> Saving data as Rdata (.rdata, .rda)</a></li>
<li class="chapter" data-level="30.7" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#import-excel-formated-data"><i class="fa fa-check"></i><b>30.7</b> Import Excel formated data</a></li>
<li class="chapter" data-level="30.8" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#save-data-as-excel-formated-file"><i class="fa fa-check"></i><b>30.8</b> Save data as Excel formated file</a></li>
<li class="chapter" data-level="30.9" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#fixing-problems-while-importing-datasets"><i class="fa fa-check"></i><b>30.9</b> Fixing problems while importing datasets</a></li>
<li class="chapter" data-level="30.10" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#simplifying-embellished-excel-datasets"><i class="fa fa-check"></i><b>30.10</b> Simplifying embellished excel datasets</a></li>
<li class="chapter" data-level="30.11" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#accessing-online-data"><i class="fa fa-check"></i><b>30.11</b> Accessing online data</a></li>
<li class="chapter" data-level="30.12" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#data-from-clinicaltrials.gov"><i class="fa fa-check"></i><b>30.12</b> Data from clinicaltrials.gov</a></li>
<li class="chapter" data-level="30.13" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#data-scrapping-from-a-website"><i class="fa fa-check"></i><b>30.13</b> Data scrapping from a website</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="working-with-proteins.html"><a href="working-with-proteins.html"><i class="fa fa-check"></i><b>31</b> – Working with Proteins</a>
<ul>
<li class="chapter" data-level="31.1" data-path="working-with-proteins.html"><a href="working-with-proteins.html#sequences-in-fasta-format"><i class="fa fa-check"></i><b>31.1</b> Sequences in FASTA format</a></li>
<li class="chapter" data-level="31.2" data-path="working-with-proteins.html"><a href="working-with-proteins.html#fetching-aa-sequences-from-ncbi"><i class="fa fa-check"></i><b>31.2</b> Fetching AA sequences from NCBI</a></li>
<li class="chapter" data-level="31.3" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-aa-sequences-in-r"><i class="fa fa-check"></i><b>31.3</b> Importing AA sequences in R</a></li>
<li class="chapter" data-level="31.4" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-nucleotide-sequences-in-r"><i class="fa fa-check"></i><b>31.4</b> Importing nucleotide sequences in R</a></li>
<li class="chapter" data-level="31.5" data-path="working-with-proteins.html"><a href="working-with-proteins.html#translating-a-nucleotides-to-aa"><i class="fa fa-check"></i><b>31.5</b> Translating a nucleotides to AA</a></li>
<li class="chapter" data-level="31.6" data-path="working-with-proteins.html"><a href="working-with-proteins.html#finding-orfs-in-a-nucleotide-sequence"><i class="fa fa-check"></i><b>31.6</b> Finding ORFs in a nucleotide sequence</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="miscellaneous.html"><a href="miscellaneous.html"><i class="fa fa-check"></i><b>32</b> – Miscellaneous</a>
<ul>
<li class="chapter" data-level="32.1" data-path="miscellaneous.html"><a href="miscellaneous.html#sampling-from-a-vector-of-names"><i class="fa fa-check"></i><b>32.1</b> Sampling from a vector of names</a></li>
<li class="chapter" data-level="32.2" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-an-article-pdf"><i class="fa fa-check"></i><b>32.2</b> Read a text from an article pdf</a></li>
<li class="chapter" data-level="32.3" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-website"><i class="fa fa-check"></i><b>32.3</b> Read a text from a website</a></li>
<li class="chapter" data-level="32.4" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-file"><i class="fa fa-check"></i><b>32.4</b> Read a text from a file</a></li>
<li class="chapter" data-level="32.5" data-path="miscellaneous.html"><a href="miscellaneous.html#generate-the-word-cloud"><i class="fa fa-check"></i><b>32.5</b> Generate the word cloud</a></li>
<li class="chapter" data-level="32.6" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-pdf"><i class="fa fa-check"></i><b>32.6</b> worldcloud – PDF</a></li>
<li class="chapter" data-level="32.7" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-website"><i class="fa fa-check"></i><b>32.7</b> worldcloud – website</a></li>
<li class="chapter" data-level="32.8" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-text"><i class="fa fa-check"></i><b>32.8</b> worldcloud – text</a></li>
<li class="chapter" data-level="32.9" data-path="miscellaneous.html"><a href="miscellaneous.html#miscellaneous-1"><i class="fa fa-check"></i><b>32.9</b> Miscellaneous</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics Workshop Gitbook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="basic-graphics" class="section level1" number="10">
<h1><span class="header-section-number">Session 10</span> – Basic Graphics</h1>
<div id="what-would-you-like-to-show" class="section level2" number="10.1">
<h2><span class="header-section-number">10.1</span> What would you like to show?</h2>
<p>An important aspect of communicating science or research is to select the correct plot, chart, and graphics. Today there is a plethora of options for graphics that would help you to inform visually the results of experiments, test, and possible projections. However, you want to present graphics that are the easiest for your audience to read and visually pleasing. This means that it has to have the right amount of detail, minimal extraneous information (e.g., 3D or fancy charts), and highlight important relationships that you are trying to emphasize. In other words, the chart that you select must convey clearly and succinctly information about data in study.</p>
<p>I tried to find a guide chart for selecting graphics. I included an adapted version of <a href="https://extremepresentation.typepad.com/about.html" target="_blank"><strong>Andrew V. Abela</strong></a> chart on <strong>what would you like to show?</strong> modified and updated
<a href="https://www.crazyegg.com/blog/data-storytelling-5-steps-charts" target="_blank"><strong>here</strong></a>.</p>
<p><img src="figures/session_10_figs/71_what_to_show_chart1.jpg" width="100%" /></p>
<p>Some more questions that you can ask yourself to select graphics:</p>
<ol style="list-style-type: decimal">
<li><p>Distribution: If my data is continuous, do I want to know the shape of the distribution? Where do my all (or particular) values fall in it? Am I interested in finding outliers?</p></li>
<li><p>Relationship: Am I interested in finding how two or more variables are related to one another?</p></li>
<li><p>Composition: Am I interested in how different components made up the whole of the dataset?</p></li>
<li><p>Comparison: How are values similar or different given a group, category, or time?</p></li>
</ol>
<p>Here is another guide chart that might help you choose the correct graphics:</p>
<p><img src="figures/session_10_figs/72_show_me_the_numbers.png" width="100%" /></p>
</div>
<div id="bar-charts" class="section level2" number="10.2">
<h2><span class="header-section-number">10.2</span> Bar charts</h2>
<p>One of the most basic graphs is those that illustrate counts or mean values of discrete variables or ranges of values as groups in a continuous variable. For counts, the x-axis includes discrete variables and the y-axis are the counts assigned to each these variables (this is a bar graph). For a continuous variable, values on this variable can be divided into groups or ranges (e.g., 0-5, 6-10, 11-15, 16-20, etc) in the x-axis and then the number of times that individual values of the continuous variable fall in the ranges are represented in the y-axis (this is a histogram).</p>
<p><strong>17)</strong> To do a bar chart, we need a dataset of discrete variables that include count and in the correct format to <strong>ggplot2</strong>.</p>
<p>We will use the ‘ichtyo’ dataset of the package ‘ade4’.</p>
<div class="sourceCode" id="cb438"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb438-1"><a href="basic-graphics.html#cb438-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we will load &#39;ade4&#39; and the ‘ichtyo’ dataset</span></span>
<span id="cb438-2"><a href="basic-graphics.html#cb438-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb438-3"><a href="basic-graphics.html#cb438-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(ichtyo)</span>
<span id="cb438-4"><a href="basic-graphics.html#cb438-4" aria-hidden="true" tabindex="-1"></a>ichtyo_data <span class="ot">&lt;-</span> ichtyo<span class="sc">$</span>tab</span>
<span id="cb438-5"><a href="basic-graphics.html#cb438-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ichtyo_data)</span>
<span id="cb438-6"><a href="basic-graphics.html#cb438-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  HOT VAN CHE SPI GOU BAF GAR ABL PER</span></span>
<span id="cb438-7"><a href="basic-graphics.html#cb438-7" aria-hidden="true" tabindex="-1"></a><span class="co">#1  57  47  60  11  27  17  10   9  21</span></span>
<span id="cb438-8"><a href="basic-graphics.html#cb438-8" aria-hidden="true" tabindex="-1"></a><span class="co">#2   9  25  19   6   2   5   0   4   6</span></span>
<span id="cb438-9"><a href="basic-graphics.html#cb438-9" aria-hidden="true" tabindex="-1"></a><span class="co">#3  48  60  52  16  25  24  17   8  12</span></span>
<span id="cb438-10"><a href="basic-graphics.html#cb438-10" aria-hidden="true" tabindex="-1"></a><span class="co">#4  26  50  32   8   6  11   2  10   5</span></span>
<span id="cb438-11"><a href="basic-graphics.html#cb438-11" aria-hidden="true" tabindex="-1"></a><span class="co">#5  26  43  31  14   9  18  13  14   2</span></span>
<span id="cb438-12"><a href="basic-graphics.html#cb438-12" aria-hidden="true" tabindex="-1"></a><span class="co">#6  62  43  47  16  11   9  15  16   3</span></span>
<span id="cb438-13"><a href="basic-graphics.html#cb438-13" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(ichtyo_data)</span>
<span id="cb438-14"><a href="basic-graphics.html#cb438-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  32 obs. of  9 variables:</span></span>
<span id="cb438-15"><a href="basic-graphics.html#cb438-15" aria-hidden="true" tabindex="-1"></a><span class="co"># $ HOT: num  57 9 48 26 26 62 17 18 9 12 ...</span></span>
<span id="cb438-16"><a href="basic-graphics.html#cb438-16" aria-hidden="true" tabindex="-1"></a><span class="co"># $ VAN: num  47 25 60 50 43 43 34 22 14 9 ...</span></span>
<span id="cb438-17"><a href="basic-graphics.html#cb438-17" aria-hidden="true" tabindex="-1"></a><span class="co"># $ CHE: num  60 19 52 32 31 47 28 18 8 9 ...</span></span>
<span id="cb438-18"><a href="basic-graphics.html#cb438-18" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SPI: num  11 6 16 8 14 16 13 9 5 10 ...</span></span>
<span id="cb438-19"><a href="basic-graphics.html#cb438-19" aria-hidden="true" tabindex="-1"></a><span class="co"># $ GOU: num  27 2 25 6 9 11 8 3 8 2 ...</span></span>
<span id="cb438-20"><a href="basic-graphics.html#cb438-20" aria-hidden="true" tabindex="-1"></a><span class="co"># $ BAF: num  17 5 24 11 18 9 13 7 8 2 ...</span></span>
<span id="cb438-21"><a href="basic-graphics.html#cb438-21" aria-hidden="true" tabindex="-1"></a><span class="co"># $ GAR: num  10 0 17 2 13 15 14 6 1 10 ...</span></span>
<span id="cb438-22"><a href="basic-graphics.html#cb438-22" aria-hidden="true" tabindex="-1"></a><span class="co"># $ ABL: num  9 4 8 10 14 16 5 8 1 5 ...</span></span>
<span id="cb438-23"><a href="basic-graphics.html#cb438-23" aria-hidden="true" tabindex="-1"></a><span class="co"># $ PER: num  21 6 12 5 2 3 4 4 0 0 ...</span></span></code></pre></div>
<p>For a bar chart, we need total counts. Therefore, we need to get the total sums per each column using <code>colSums()</code> and as a data frame using <code>as.data.frame()</code>.</p>
<div class="sourceCode" id="cb439"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb439-1"><a href="basic-graphics.html#cb439-1" aria-hidden="true" tabindex="-1"></a><span class="do">## colum sum as data frame</span></span>
<span id="cb439-2"><a href="basic-graphics.html#cb439-2" aria-hidden="true" tabindex="-1"></a>ichtyo_sum_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colSums</span> (ichtyo_data, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb439-3"><a href="basic-graphics.html#cb439-3" aria-hidden="true" tabindex="-1"></a>ichtyo_sum_df</span>
<span id="cb439-4"><a href="basic-graphics.html#cb439-4" aria-hidden="true" tabindex="-1"></a><span class="co">#    colSums(ichtyo_data, na.rm = TRUE)</span></span>
<span id="cb439-5"><a href="basic-graphics.html#cb439-5" aria-hidden="true" tabindex="-1"></a><span class="co">#HOT                                936</span></span>
<span id="cb439-6"><a href="basic-graphics.html#cb439-6" aria-hidden="true" tabindex="-1"></a><span class="co">#VAN                                995</span></span>
<span id="cb439-7"><a href="basic-graphics.html#cb439-7" aria-hidden="true" tabindex="-1"></a><span class="co">#CHE                               1741</span></span>
<span id="cb439-8"><a href="basic-graphics.html#cb439-8" aria-hidden="true" tabindex="-1"></a><span class="co">#SPI                                491</span></span>
<span id="cb439-9"><a href="basic-graphics.html#cb439-9" aria-hidden="true" tabindex="-1"></a><span class="co">#GOU                                951</span></span>
<span id="cb439-10"><a href="basic-graphics.html#cb439-10" aria-hidden="true" tabindex="-1"></a><span class="co">#BAF                                658</span></span>
<span id="cb439-11"><a href="basic-graphics.html#cb439-11" aria-hidden="true" tabindex="-1"></a><span class="co">#GAR                               1007</span></span>
<span id="cb439-12"><a href="basic-graphics.html#cb439-12" aria-hidden="true" tabindex="-1"></a><span class="co">#ABL                                337</span></span>
<span id="cb439-13"><a href="basic-graphics.html#cb439-13" aria-hidden="true" tabindex="-1"></a><span class="co">#PER                                625</span></span>
<span id="cb439-14"><a href="basic-graphics.html#cb439-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb439-15"><a href="basic-graphics.html#cb439-15" aria-hidden="true" tabindex="-1"></a><span class="do">## build a nicer data frame and add column of site_names</span></span>
<span id="cb439-16"><a href="basic-graphics.html#cb439-16" aria-hidden="true" tabindex="-1"></a>ichtyo_sum_df<span class="sc">$</span>site_names <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ichtyo_sum_df)</span>
<span id="cb439-17"><a href="basic-graphics.html#cb439-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb439-18"><a href="basic-graphics.html#cb439-18" aria-hidden="true" tabindex="-1"></a><span class="do">## upgrade column names</span></span>
<span id="cb439-19"><a href="basic-graphics.html#cb439-19" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ichtyo_sum_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;counts&quot;</span>, <span class="st">&quot;site_names&quot;</span>)</span>
<span id="cb439-20"><a href="basic-graphics.html#cb439-20" aria-hidden="true" tabindex="-1"></a>ichtyo_sum_df</span>
<span id="cb439-21"><a href="basic-graphics.html#cb439-21" aria-hidden="true" tabindex="-1"></a><span class="co">#    counts site_names</span></span>
<span id="cb439-22"><a href="basic-graphics.html#cb439-22" aria-hidden="true" tabindex="-1"></a><span class="co">#HOT    936        HOT</span></span>
<span id="cb439-23"><a href="basic-graphics.html#cb439-23" aria-hidden="true" tabindex="-1"></a><span class="co">#VAN    995        VAN</span></span>
<span id="cb439-24"><a href="basic-graphics.html#cb439-24" aria-hidden="true" tabindex="-1"></a><span class="co">#CHE   1741        CHE</span></span>
<span id="cb439-25"><a href="basic-graphics.html#cb439-25" aria-hidden="true" tabindex="-1"></a><span class="co">#SPI    491        SPI</span></span>
<span id="cb439-26"><a href="basic-graphics.html#cb439-26" aria-hidden="true" tabindex="-1"></a><span class="co">#GOU    951        GOU</span></span>
<span id="cb439-27"><a href="basic-graphics.html#cb439-27" aria-hidden="true" tabindex="-1"></a><span class="co">#BAF    658        BAF</span></span>
<span id="cb439-28"><a href="basic-graphics.html#cb439-28" aria-hidden="true" tabindex="-1"></a><span class="co">#GAR   1007        GAR</span></span>
<span id="cb439-29"><a href="basic-graphics.html#cb439-29" aria-hidden="true" tabindex="-1"></a><span class="co">#ABL    337        ABL</span></span>
<span id="cb439-30"><a href="basic-graphics.html#cb439-30" aria-hidden="true" tabindex="-1"></a><span class="co">#PER    625        PER</span></span>
<span id="cb439-31"><a href="basic-graphics.html#cb439-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb439-32"><a href="basic-graphics.html#cb439-32" aria-hidden="true" tabindex="-1"></a><span class="do">## add count labels for later use</span></span>
<span id="cb439-33"><a href="basic-graphics.html#cb439-33" aria-hidden="true" tabindex="-1"></a>ichtyo_sum_df<span class="sc">$</span>count_label <span class="ot">&lt;-</span> <span class="fu">as.character</span>(ichtyo_sum_df<span class="sc">$</span>counts)</span>
<span id="cb439-34"><a href="basic-graphics.html#cb439-34" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(ichtyo_sum_df)</span>
<span id="cb439-35"><a href="basic-graphics.html#cb439-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  9 obs. of  3 variables:</span></span>
<span id="cb439-36"><a href="basic-graphics.html#cb439-36" aria-hidden="true" tabindex="-1"></a><span class="co"># $ counts     : num  936 995 1741 491 951 ...</span></span>
<span id="cb439-37"><a href="basic-graphics.html#cb439-37" aria-hidden="true" tabindex="-1"></a><span class="co"># $ site_names : chr  &quot;HOT&quot; &quot;VAN&quot; &quot;CHE&quot; &quot;SPI&quot; ...</span></span>
<span id="cb439-38"><a href="basic-graphics.html#cb439-38" aria-hidden="true" tabindex="-1"></a><span class="co"># $ count_label: chr  &quot;936&quot; &quot;995&quot; &quot;1741&quot; &quot;491&quot; ...</span></span></code></pre></div>
<p>We can now plot our bar chart with <code>site_names</code> in the x-axis and <code>counts</code> in the y-axis. A series the options to the ggplot are added: <code>ggplot()</code> is the main scaffold of the plot and includes the argument <code>data = ichtyo_sum_df</code> as source of data,<code>aes()</code> defines axis in the 2D ggplot, <code>x</code> is the x-axis and we defined <code>site_names</code> as the variable for that axis,<code>y</code> is the y-axis and we defined <code>counts</code> as the variable for that axis, <code>fill</code> is the variable that will define groups to color the bars in the plot.<code>geom_bar()</code> is the option for bar charts, the argument <code>colour = black</code> is for black countour of the bars, the argument <code>stat = "identity"</code> basically says that we want the values of the y-axis (i.e., counts) as the heights of the bars.<code>guides(fill = FALSE)</code> indicagtes that we do not want legends. <code>theme_minimal()</code> is a present style of the plot that is simple. <code>xlab()</code>, <code>ylab()</code> and <code>ggtitle()</code> are labels for the axis and title, you can put in there whatever you want.</p>
<div class="sourceCode" id="cb440"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb440-1"><a href="basic-graphics.html#cb440-1" aria-hidden="true" tabindex="-1"></a><span class="do">## checking if ggplot2  and ggsci are loaded</span></span>
<span id="cb440-2"><a href="basic-graphics.html#cb440-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggplot2)</span>
<span id="cb440-3"><a href="basic-graphics.html#cb440-3" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggsci)</span>
<span id="cb440-4"><a href="basic-graphics.html#cb440-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb440-5"><a href="basic-graphics.html#cb440-5" aria-hidden="true" tabindex="-1"></a><span class="co"># base plot</span></span>
<span id="cb440-6"><a href="basic-graphics.html#cb440-6" aria-hidden="true" tabindex="-1"></a>ichtyo_barchart <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>ichtyo_sum_df, <span class="fu">aes</span>(<span class="at">x=</span>site_names, <span class="at">y=</span>counts, <span class="at">fill =</span> site_names)) <span class="sc">+</span></span>
<span id="cb440-7"><a href="basic-graphics.html#cb440-7" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">geom_bar</span>(<span class="at">colour=</span><span class="st">&quot;black&quot;</span>, <span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb440-8"><a href="basic-graphics.html#cb440-8" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb440-9"><a href="basic-graphics.html#cb440-9" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">xlab</span>(<span class="st">&quot;Sites on a place&quot;</span>) <span class="sc">+</span> </span>
<span id="cb440-10"><a href="basic-graphics.html#cb440-10" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">ylab</span>(<span class="st">&quot;Total counts&quot;</span>) <span class="sc">+</span></span>
<span id="cb440-11"><a href="basic-graphics.html#cb440-11" aria-hidden="true" tabindex="-1"></a>                <span class="fu">ggtitle</span>(<span class="st">&quot;My First Bar Chart&quot;</span>)</span>
<span id="cb440-12"><a href="basic-graphics.html#cb440-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb440-13"><a href="basic-graphics.html#cb440-13" aria-hidden="true" tabindex="-1"></a>ichtyo_barchart</span></code></pre></div>
<p><img src="figures/session_9_figs/19_barchart_example_v1.png" width="90%" /></p>
<p>We can simplify this plot with adding the counts on top of the bars with the <code>count_label</code> of the <code>ichtyo_sum_df</code> and remove the background grids. A series the options to the ggplot are added: <code>geom_text()</code> will add the text of the <code>count_label</code> using the <code>site_names</code> and <code>counts</code> as x and y coordinates and the argument <code>vjust=-1</code> will move it a bit above the x/y coordinate.<code>theme()</code> will remove some unwanted elements by assigning them as <code>element_blank()</code> these include the y-axis text and all the grids.</p>
<div class="sourceCode" id="cb441"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb441-1"><a href="basic-graphics.html#cb441-1" aria-hidden="true" tabindex="-1"></a>ichtyo_barchart <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>ichtyo_sum_df, <span class="fu">aes</span>(<span class="at">x=</span>site_names, <span class="at">y=</span>counts, <span class="at">fill =</span> site_names)) <span class="sc">+</span></span>
<span id="cb441-2"><a href="basic-graphics.html#cb441-2" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">geom_bar</span>(<span class="at">colour=</span><span class="st">&quot;black&quot;</span>, <span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb441-3"><a href="basic-graphics.html#cb441-3" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb441-4"><a href="basic-graphics.html#cb441-4" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">xlab</span>(<span class="st">&quot;Sites on a place&quot;</span>) <span class="sc">+</span> </span>
<span id="cb441-5"><a href="basic-graphics.html#cb441-5" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">ylab</span>(<span class="st">&quot;Total counts&quot;</span>) <span class="sc">+</span></span>
<span id="cb441-6"><a href="basic-graphics.html#cb441-6" aria-hidden="true" tabindex="-1"></a>                <span class="fu">ggtitle</span>(<span class="st">&quot;My First Bar Chart&quot;</span>) <span class="sc">+</span></span>
<span id="cb441-7"><a href="basic-graphics.html#cb441-7" aria-hidden="true" tabindex="-1"></a>              <span class="fu">geom_text</span>(<span class="at">data=</span>ichtyo_sum_df,<span class="fu">aes</span>(<span class="at">x=</span>site_names,<span class="at">y=</span>counts,<span class="at">label=</span>count_label),<span class="at">vjust=</span><span class="sc">-</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb441-8"><a href="basic-graphics.html#cb441-8" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb441-9"><a href="basic-graphics.html#cb441-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb441-10"><a href="basic-graphics.html#cb441-10" aria-hidden="true" tabindex="-1"></a>                    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())</span>
<span id="cb441-11"><a href="basic-graphics.html#cb441-11" aria-hidden="true" tabindex="-1"></a>ichtyo_barchart</span></code></pre></div>
<p><img src="figures/session_9_figs/20_barchart_example_v2.png" width="90%" /></p>
<p>The base colors of this plot are becoming cliché as the enormous popularity of ggplot2, so we can change it to something more original using the color packages that we presented on previous section.</p>
<div class="sourceCode" id="cb442"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb442-1"><a href="basic-graphics.html#cb442-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot with futurama fill colors</span></span>
<span id="cb442-2"><a href="basic-graphics.html#cb442-2" aria-hidden="true" tabindex="-1"></a>ichtyo_barchart <span class="ot">&lt;-</span> ichtyo_barchart <span class="sc">+</span> <span class="fu">scale_fill_futurama</span>()</span>
<span id="cb442-3"><a href="basic-graphics.html#cb442-3" aria-hidden="true" tabindex="-1"></a>ichtyo_barchart</span></code></pre></div>
<p><img src="figures/session_9_figs/21_barchart_example_v3.png" width="90%" /></p>
</div>
<div id="stacked-bar-charts" class="section level2" number="10.3">
<h2><span class="header-section-number">10.3</span> Stacked bar charts</h2>
<p><strong>18)</strong> Some people argue against stacked bar charts, yet they are much better than pie charts to show relationships between proportions per sample.</p>
<p>We will use the ‘geom_col()‘ with the same dataset ‘ecomor’ of the package ‘ade4’. Notice that is a rather complex processing, yet you should be able to dissect the processing from input data to data frame ready for <strong>ggplot2</strong>.</p>
<div class="sourceCode" id="cb443"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb443-1"><a href="basic-graphics.html#cb443-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we will load &#39;ade4&#39; and the ‘ecomor’ dataset</span></span>
<span id="cb443-2"><a href="basic-graphics.html#cb443-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb443-3"><a href="basic-graphics.html#cb443-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(ecomor)</span>
<span id="cb443-4"><a href="basic-graphics.html#cb443-4" aria-hidden="true" tabindex="-1"></a><span class="co">#taxois a data frame with 129 species and 3 factors: Genus, Family and Order. It is a data frame of class&#39;taxo&#39;: the variables are factors giving nested classifications.</span></span>
<span id="cb443-5"><a href="basic-graphics.html#cb443-5" aria-hidden="true" tabindex="-1"></a>ecomor_data <span class="ot">&lt;-</span> ecomor<span class="sc">$</span>taxo</span>
<span id="cb443-6"><a href="basic-graphics.html#cb443-6" aria-hidden="true" tabindex="-1"></a><span class="co">#categ is a data frame with 129 species, 2 factors : ’forsub’ summarizing the feeding place and ’diet’the diet type.</span></span>
<span id="cb443-7"><a href="basic-graphics.html#cb443-7" aria-hidden="true" tabindex="-1"></a>ecomor_data <span class="ot">&lt;-</span> <span class="fu">cbind</span>(ecomor_data,ecomor<span class="sc">$</span>categ)</span>
<span id="cb443-8"><a href="basic-graphics.html#cb443-8" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(ecomor_data)</span>
<span id="cb443-9"><a href="basic-graphics.html#cb443-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  129 obs. of  5 variables:</span></span>
<span id="cb443-10"><a href="basic-graphics.html#cb443-10" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Genus : Factor w/ 86 levels &quot;Acanithis&quot;,&quot;Aegithalos&quot;,..: 11 12 12 49 70 21 21 73 85 85 ...</span></span>
<span id="cb443-11"><a href="basic-graphics.html#cb443-11" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Family: Factor w/ 35 levels &quot;Aegithalidae&quot;,..: 31 31 31 31 31 6 6 6 6 6 ...</span></span>
<span id="cb443-12"><a href="basic-graphics.html#cb443-12" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Ordre : Factor w/ 7 levels &quot;Apodiformes&quot;,..: 1 1 1 1 1 2 2 2 2 2 ...</span></span>
<span id="cb443-13"><a href="basic-graphics.html#cb443-13" aria-hidden="true" tabindex="-1"></a><span class="co"># $ forsub: Factor w/ 7 levels &quot;A&quot;,&quot;B&quot;,&quot;F&quot;,&quot;G&quot;,..: 5 5 5 1 3 5 5 4 4 4 ...</span></span>
<span id="cb443-14"><a href="basic-graphics.html#cb443-14" aria-hidden="true" tabindex="-1"></a><span class="co"># $ diet  : Factor w/ 8 levels &quot;F&quot;,&quot;G&quot;,&quot;H&quot;,&quot;I&quot;,..: 7 7 7 7 7 2 6 2 2 2 ...</span></span>
<span id="cb443-15"><a href="basic-graphics.html#cb443-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-16"><a href="basic-graphics.html#cb443-16" aria-hidden="true" tabindex="-1"></a><span class="do">## we subset to families and diet</span></span>
<span id="cb443-17"><a href="basic-graphics.html#cb443-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-18"><a href="basic-graphics.html#cb443-18" aria-hidden="true" tabindex="-1"></a>ecomor_data_set <span class="ot">&lt;-</span> <span class="fu">subset</span>(ecomor_data, <span class="at">select =</span> <span class="fu">c</span>(Family, diet)) </span>
<span id="cb443-19"><a href="basic-graphics.html#cb443-19" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ecomor_data_set)</span>
<span id="cb443-20"><a href="basic-graphics.html#cb443-20" aria-hidden="true" tabindex="-1"></a><span class="co">#          Family diet</span></span>
<span id="cb443-21"><a href="basic-graphics.html#cb443-21" aria-hidden="true" tabindex="-1"></a><span class="co">#E033 Trochilidae    N</span></span>
<span id="cb443-22"><a href="basic-graphics.html#cb443-22" aria-hidden="true" tabindex="-1"></a><span class="co">#E034 Trochilidae    N</span></span>
<span id="cb443-23"><a href="basic-graphics.html#cb443-23" aria-hidden="true" tabindex="-1"></a><span class="co">#E035 Trochilidae    N</span></span>
<span id="cb443-24"><a href="basic-graphics.html#cb443-24" aria-hidden="true" tabindex="-1"></a><span class="co">#E070 Trochilidae    N</span></span>
<span id="cb443-25"><a href="basic-graphics.html#cb443-25" aria-hidden="true" tabindex="-1"></a><span class="co">#E071 Trochilidae    N</span></span>
<span id="cb443-26"><a href="basic-graphics.html#cb443-26" aria-hidden="true" tabindex="-1"></a><span class="co">#E001  Columbidae    G</span></span>
<span id="cb443-27"><a href="basic-graphics.html#cb443-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-28"><a href="basic-graphics.html#cb443-28" aria-hidden="true" tabindex="-1"></a><span class="do">## we split by family</span></span>
<span id="cb443-29"><a href="basic-graphics.html#cb443-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-30"><a href="basic-graphics.html#cb443-30" aria-hidden="true" tabindex="-1"></a>list_of_families <span class="ot">&lt;-</span> <span class="fu">split</span>(ecomor_data_set, ecomor_data_set<span class="sc">$</span>Family)</span>
<span id="cb443-31"><a href="basic-graphics.html#cb443-31" aria-hidden="true" tabindex="-1"></a>list_of_families</span>
<span id="cb443-32"><a href="basic-graphics.html#cb443-32" aria-hidden="true" tabindex="-1"></a><span class="co">#$Aegithalidae</span></span>
<span id="cb443-33"><a href="basic-graphics.html#cb443-33" aria-hidden="true" tabindex="-1"></a><span class="co">#           Family diet</span></span>
<span id="cb443-34"><a href="basic-graphics.html#cb443-34" aria-hidden="true" tabindex="-1"></a><span class="co">#E121 Aegithalidae    I</span></span>
<span id="cb443-35"><a href="basic-graphics.html#cb443-35" aria-hidden="true" tabindex="-1"></a><span class="co">#E048 Aegithalidae    I</span></span>
<span id="cb443-36"><a href="basic-graphics.html#cb443-36" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb443-37"><a href="basic-graphics.html#cb443-37" aria-hidden="true" tabindex="-1"></a><span class="co">#$Alaudidae</span></span>
<span id="cb443-38"><a href="basic-graphics.html#cb443-38" aria-hidden="true" tabindex="-1"></a><span class="co">#        Family diet</span></span>
<span id="cb443-39"><a href="basic-graphics.html#cb443-39" aria-hidden="true" tabindex="-1"></a><span class="co">#E104 Alaudidae    H</span></span>
<span id="cb443-40"><a href="basic-graphics.html#cb443-40" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb443-41"><a href="basic-graphics.html#cb443-41" aria-hidden="true" tabindex="-1"></a><span class="co">#$Cardinalidae</span></span>
<span id="cb443-42"><a href="basic-graphics.html#cb443-42" aria-hidden="true" tabindex="-1"></a><span class="co">#           Family diet</span></span>
<span id="cb443-43"><a href="basic-graphics.html#cb443-43" aria-hidden="true" tabindex="-1"></a><span class="co">#E057 Cardinalidae    G</span></span>
<span id="cb443-44"><a href="basic-graphics.html#cb443-44" aria-hidden="true" tabindex="-1"></a><span class="co">#E056 Cardinalidae    G</span></span>
<span id="cb443-45"><a href="basic-graphics.html#cb443-45" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span>
<span id="cb443-46"><a href="basic-graphics.html#cb443-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-47"><a href="basic-graphics.html#cb443-47" aria-hidden="true" tabindex="-1"></a><span class="do">## we can calculate the frequency of diet types by each family of these birds</span></span>
<span id="cb443-48"><a href="basic-graphics.html#cb443-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-49"><a href="basic-graphics.html#cb443-49" aria-hidden="true" tabindex="-1"></a>collect_families_processed <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb443-50"><a href="basic-graphics.html#cb443-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-51"><a href="basic-graphics.html#cb443-51" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(list_of_families)) {</span>
<span id="cb443-52"><a href="basic-graphics.html#cb443-52" aria-hidden="true" tabindex="-1"></a>  <span class="co"># i &lt;- 1</span></span>
<span id="cb443-53"><a href="basic-graphics.html#cb443-53" aria-hidden="true" tabindex="-1"></a>              one_family <span class="ot">&lt;-</span> list_of_families[[i]]</span>
<span id="cb443-54"><a href="basic-graphics.html#cb443-54" aria-hidden="true" tabindex="-1"></a>             name_family <span class="ot">&lt;-</span> <span class="fu">unique</span>(one_family[,<span class="dv">1</span>])</span>
<span id="cb443-55"><a href="basic-graphics.html#cb443-55" aria-hidden="true" tabindex="-1"></a>           one_family_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">table</span>(one_family [,<span class="dv">2</span>]), <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb443-56"><a href="basic-graphics.html#cb443-56" aria-hidden="true" tabindex="-1"></a>       one_family_df_sum <span class="ot">&lt;-</span> <span class="fu">sum</span>(one_family_df<span class="sc">$</span>Freq)</span>
<span id="cb443-57"><a href="basic-graphics.html#cb443-57" aria-hidden="true" tabindex="-1"></a>      one_family_df<span class="sc">$</span>Freq <span class="ot">&lt;-</span> one_family_df<span class="sc">$</span>Freq<span class="sc">/</span>one_family_df_sum</span>
<span id="cb443-58"><a href="basic-graphics.html#cb443-58" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(one_family_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;type_diet&quot;</span>,<span class="st">&quot;Freq&quot;</span>)</span>
<span id="cb443-59"><a href="basic-graphics.html#cb443-59" aria-hidden="true" tabindex="-1"></a>    one_family_df<span class="sc">$</span>Family <span class="ot">&lt;-</span> name_family</span>
<span id="cb443-60"><a href="basic-graphics.html#cb443-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-61"><a href="basic-graphics.html#cb443-61" aria-hidden="true" tabindex="-1"></a>    collect_families_processed [[i]] <span class="ot">&lt;-</span> one_family_df</span>
<span id="cb443-62"><a href="basic-graphics.html#cb443-62" aria-hidden="true" tabindex="-1"></a>                       }</span>
<span id="cb443-63"><a href="basic-graphics.html#cb443-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-64"><a href="basic-graphics.html#cb443-64" aria-hidden="true" tabindex="-1"></a><span class="do">## we put all of these in a single data frame</span></span>
<span id="cb443-65"><a href="basic-graphics.html#cb443-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb443-66"><a href="basic-graphics.html#cb443-66" aria-hidden="true" tabindex="-1"></a>collect_families_processed_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, collect_families_processed)</span>
<span id="cb443-67"><a href="basic-graphics.html#cb443-67" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(collect_families_processed_df)</span>
<span id="cb443-68"><a href="basic-graphics.html#cb443-68" aria-hidden="true" tabindex="-1"></a><span class="co">#  type_diet Freq       Family</span></span>
<span id="cb443-69"><a href="basic-graphics.html#cb443-69" aria-hidden="true" tabindex="-1"></a><span class="co">#1         F    0 Aegithalidae</span></span>
<span id="cb443-70"><a href="basic-graphics.html#cb443-70" aria-hidden="true" tabindex="-1"></a><span class="co">#2         G    0 Aegithalidae</span></span>
<span id="cb443-71"><a href="basic-graphics.html#cb443-71" aria-hidden="true" tabindex="-1"></a><span class="co">#3         H    0 Aegithalidae</span></span>
<span id="cb443-72"><a href="basic-graphics.html#cb443-72" aria-hidden="true" tabindex="-1"></a><span class="co">#4         I    1 Aegithalidae</span></span>
<span id="cb443-73"><a href="basic-graphics.html#cb443-73" aria-hidden="true" tabindex="-1"></a><span class="co">#5         J    0 Aegithalidae</span></span>
<span id="cb443-74"><a href="basic-graphics.html#cb443-74" aria-hidden="true" tabindex="-1"></a><span class="co">#6         K    0 Aegithalidae</span></span>
<span id="cb443-75"><a href="basic-graphics.html#cb443-75" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(collect_families_processed_df)</span>
<span id="cb443-76"><a href="basic-graphics.html#cb443-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;: 280 obs. of  3 variables:</span></span>
<span id="cb443-77"><a href="basic-graphics.html#cb443-77" aria-hidden="true" tabindex="-1"></a><span class="co"># $ type_diet: chr  &quot;F&quot; &quot;G&quot; &quot;H&quot; &quot;I&quot; ...</span></span>
<span id="cb443-78"><a href="basic-graphics.html#cb443-78" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Freq     : num  0 0 0 1 0 0 0 0 0 0 ...</span></span>
<span id="cb443-79"><a href="basic-graphics.html#cb443-79" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Family   : Factor w/ 35 levels &quot;Aegithalidae&quot;,..: 1 1 1 1 1 1 1 1 2 2 ...</span></span></code></pre></div>
<p>We can now plot a stacked bar plot. Notice we need 3 variables (columns): 2 categorical and 1 numeric. In this case the two categorical are <code>type_of_diet</code> and <code>Family</code>, and the one continuous is <code>Freq</code>.</p>
<div class="sourceCode" id="cb444"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb444-1"><a href="basic-graphics.html#cb444-1" aria-hidden="true" tabindex="-1"></a><span class="do">## checking if ggplot2  and ggsci are loaded</span></span>
<span id="cb444-2"><a href="basic-graphics.html#cb444-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggplot2)</span>
<span id="cb444-3"><a href="basic-graphics.html#cb444-3" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggsci)</span>
<span id="cb444-4"><a href="basic-graphics.html#cb444-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb444-5"><a href="basic-graphics.html#cb444-5" aria-hidden="true" tabindex="-1"></a><span class="do">## the plot</span></span>
<span id="cb444-6"><a href="basic-graphics.html#cb444-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb444-7"><a href="basic-graphics.html#cb444-7" aria-hidden="true" tabindex="-1"></a>diet_family_stackedbar <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(collect_families_processed_df) <span class="sc">+</span></span>
<span id="cb444-8"><a href="basic-graphics.html#cb444-8" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">x =</span> Family, <span class="at">y =</span> Freq, <span class="at">fill =</span> type_diet), <span class="at">color =</span> <span class="st">&quot;black&quot;</span>)<span class="sc">+</span></span>
<span id="cb444-9"><a href="basic-graphics.html#cb444-9" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Stacked Bar Graph -- Diet and Families Birds&quot;</span>, <span class="at">x=</span> <span class="st">&quot;Family&quot;</span> , <span class="at">y=</span><span class="st">&quot;Frequency&quot;</span>) <span class="sc">+</span></span>
<span id="cb444-10"><a href="basic-graphics.html#cb444-10" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb444-11"><a href="basic-graphics.html#cb444-11" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb444-12"><a href="basic-graphics.html#cb444-12" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">0.7</span>),<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust=</span><span class="dv">1</span>))</span>
<span id="cb444-13"><a href="basic-graphics.html#cb444-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb444-14"><a href="basic-graphics.html#cb444-14" aria-hidden="true" tabindex="-1"></a>diet_family_stackedbar <span class="ot">&lt;-</span> diet_family_stackedbar <span class="sc">+</span> <span class="fu">scale_fill_nejm</span>()</span>
<span id="cb444-15"><a href="basic-graphics.html#cb444-15" aria-hidden="true" tabindex="-1"></a>diet_family_stackedbar</span></code></pre></div>
<p><img src="figures/session_10_figs/63_stackbar.png" width="90%" /></p>
</div>
<div id="histograms" class="section level2" number="10.4">
<h2><span class="header-section-number">10.4</span> Histograms</h2>
<p><strong>19)</strong> To do a histogram, we need a dataset with continuous variables and we will divide into ranges and format them into <strong>ggplot2</strong>.</p>
<p>We will use two samples of the <code>airway_scaledcounts.csv</code> dataset that indicates gene expression.</p>
<div class="sourceCode" id="cb445"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb445-1"><a href="basic-graphics.html#cb445-1" aria-hidden="true" tabindex="-1"></a><span class="do">## </span><span class="al">NOTE</span><span class="do">: remember to update the path to file with the dataset where you downloaded in your computer -- THIS IS EXCLUSIVE TO YOUR COMPUTER AND IT IS NOT THE PATH SHOWN BELOW</span></span>
<span id="cb445-2"><a href="basic-graphics.html#cb445-2" aria-hidden="true" tabindex="-1"></a><span class="do">## load get &#39;airway_scaledcounts.csv&#39; dataset </span></span>
<span id="cb445-3"><a href="basic-graphics.html#cb445-3" aria-hidden="true" tabindex="-1"></a>airway_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/ref_files_pdfs/airway_scaledcounts.csv&quot;</span>,</span>
<span id="cb445-4"><a href="basic-graphics.html#cb445-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb445-5"><a href="basic-graphics.html#cb445-5" aria-hidden="true" tabindex="-1"></a><span class="co"># remove &#39;ensgene&#39; variable</span></span>
<span id="cb445-6"><a href="basic-graphics.html#cb445-6" aria-hidden="true" tabindex="-1"></a>airway_data_red <span class="ot">&lt;-</span> airway_data[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb445-7"><a href="basic-graphics.html#cb445-7" aria-hidden="true" tabindex="-1"></a><span class="co"># select two sample: &quot;SRR1039509&quot; and &quot;SRR1039517&quot;</span></span>
<span id="cb445-8"><a href="basic-graphics.html#cb445-8" aria-hidden="true" tabindex="-1"></a>airway_data_red <span class="ot">&lt;-</span> <span class="fu">subset</span>(airway_data_red, <span class="at">select =</span> <span class="fu">c</span>(SRR1039509,SRR1039517))</span>
<span id="cb445-9"><a href="basic-graphics.html#cb445-9" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(airway_data_red)</span>
<span id="cb445-10"><a href="basic-graphics.html#cb445-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  38694 obs. of  2 variables:</span></span>
<span id="cb445-11"><a href="basic-graphics.html#cb445-11" aria-hidden="true" tabindex="-1"></a><span class="co">#$ SRR1039509: num  486 0 523 258 81 ...</span></span>
<span id="cb445-12"><a href="basic-graphics.html#cb445-12" aria-hidden="true" tabindex="-1"></a><span class="co">#$ SRR1039517: num  1097 0 781 447 94 ...</span></span></code></pre></div>
<p>We estimate a histogram for sample <code>SRR1039509</code>, for this we need to explore this with <code>summary()</code>.</p>
<div class="sourceCode" id="cb446"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb446-1"><a href="basic-graphics.html#cb446-1" aria-hidden="true" tabindex="-1"></a><span class="do">## summary</span></span>
<span id="cb446-2"><a href="basic-graphics.html#cb446-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(airway_data_red<span class="sc">$</span>SRR1039509)</span>
<span id="cb446-3"><a href="basic-graphics.html#cb446-3" aria-hidden="true" tabindex="-1"></a><span class="co">#    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. </span></span>
<span id="cb446-4"><a href="basic-graphics.html#cb446-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     0.0      0.0      1.0    501.1    172.0 244921.0 </span></span></code></pre></div>
<p>For a regular histogram in ggplot2 we want <code>geom_histogram()</code> with the <code>airway_data_red</code> data frame that includes sample <code>SRR1039509</code>. Next, we need to define some parameters: (1) how often we want the bins to be separated with the argument <code>breaks()</code>, it usually will be a sequence called by function <code>seq()</code> and in this case from a minimum (i.e., 0) and maximum (i.e., 244921 - 250000) and width argument <code>by=</code> for example 1000. (2) you can choose the color of the boundary of the histogram with <code>col=</code> like red, its fill color <code>fill=</code> like green and transparency <code>alpha</code> where 0 is transparent and 1 completely opaque.</p>
<div class="sourceCode" id="cb447"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb447-1"><a href="basic-graphics.html#cb447-1" aria-hidden="true" tabindex="-1"></a><span class="do">## checking if ggplot2 is loaded</span></span>
<span id="cb447-2"><a href="basic-graphics.html#cb447-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggplot2)</span>
<span id="cb447-3"><a href="basic-graphics.html#cb447-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb447-4"><a href="basic-graphics.html#cb447-4" aria-hidden="true" tabindex="-1"></a>SRR1039509_histogram <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>airway_data_red, <span class="fu">aes</span>(<span class="at">x=</span>SRR1039509)) <span class="sc">+</span> </span>
<span id="cb447-5"><a href="basic-graphics.html#cb447-5" aria-hidden="true" tabindex="-1"></a>     <span class="fu">geom_histogram</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">250000</span>, <span class="at">by=</span><span class="dv">1000</span>), </span>
<span id="cb447-6"><a href="basic-graphics.html#cb447-6" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col=</span><span class="st">&quot;red&quot;</span>, </span>
<span id="cb447-7"><a href="basic-graphics.html#cb447-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">fill=</span><span class="st">&quot;green&quot;</span>, </span>
<span id="cb447-8"><a href="basic-graphics.html#cb447-8" aria-hidden="true" tabindex="-1"></a>                    <span class="at">alpha =</span> .<span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb447-9"><a href="basic-graphics.html#cb447-9" aria-hidden="true" tabindex="-1"></a>                <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Histogram for SRR1039509&quot;</span>, <span class="at">x=</span><span class="st">&quot;Expression range&quot;</span>, <span class="at">y=</span><span class="st">&quot;Count&quot;</span>) <span class="sc">+</span> </span>
<span id="cb447-10"><a href="basic-graphics.html#cb447-10" aria-hidden="true" tabindex="-1"></a>                <span class="fu">xlim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">250000</span>))<span class="sc">+</span></span>
<span id="cb447-11"><a href="basic-graphics.html#cb447-11" aria-hidden="true" tabindex="-1"></a>                <span class="fu">theme_minimal</span>()</span>
<span id="cb447-12"><a href="basic-graphics.html#cb447-12" aria-hidden="true" tabindex="-1"></a>SRR1039509_histogram</span></code></pre></div>
<p><img src="figures/session_9_figs/22_histogram_raw_1.png" width="90%" /></p>
<p>This histogram is quite skewed to the right and we notice the maximum of the sample (i.e., 244921) is ~500 x mean value. Some genes are highly overexpressed and most of these are rare (low count). We can modify the <code>breaks()</code> argument to series of individual breaks every 100 (e.g., 100,200,..,2000), in this last break it we want to include all genes +2000 expression units.</p>
<div class="sourceCode" id="cb448"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb448-1"><a href="basic-graphics.html#cb448-1" aria-hidden="true" tabindex="-1"></a>SRR1039509_histogram <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>airway_data_red, <span class="fu">aes</span>(<span class="at">x=</span>SRR1039509)) <span class="sc">+</span> </span>
<span id="cb448-2"><a href="basic-graphics.html#cb448-2" aria-hidden="true" tabindex="-1"></a>     <span class="fu">geom_histogram</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">2000</span>, <span class="at">by=</span><span class="dv">100</span>), </span>
<span id="cb448-3"><a href="basic-graphics.html#cb448-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col=</span><span class="st">&quot;red&quot;</span>, </span>
<span id="cb448-4"><a href="basic-graphics.html#cb448-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">fill=</span><span class="st">&quot;green&quot;</span>, </span>
<span id="cb448-5"><a href="basic-graphics.html#cb448-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">alpha =</span> .<span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb448-6"><a href="basic-graphics.html#cb448-6" aria-hidden="true" tabindex="-1"></a>                <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Histogram for SRR1039509&quot;</span>, <span class="at">x=</span><span class="st">&quot;Expression range&quot;</span>, <span class="at">y=</span><span class="st">&quot;Count&quot;</span>) <span class="sc">+</span> </span>
<span id="cb448-7"><a href="basic-graphics.html#cb448-7" aria-hidden="true" tabindex="-1"></a>                <span class="fu">xlim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">2000</span>))<span class="sc">+</span></span>
<span id="cb448-8"><a href="basic-graphics.html#cb448-8" aria-hidden="true" tabindex="-1"></a>                <span class="fu">theme_minimal</span>()</span>
<span id="cb448-9"><a href="basic-graphics.html#cb448-9" aria-hidden="true" tabindex="-1"></a>SRR1039509_histogram</span></code></pre></div>
<p><img src="figures/session_9_figs/23_histogram_ver_2.png" width="90%" /></p>
<p>This histogram is better, yet not ideal. First, it still is right skewed and count values &gt;2000 were excluded from the graph. This suggest that we need to simplify our data and bin all values more 2000 to that maximum. We can do this by creating an small function that we create and named <code>cap_2000_fun</code> that uses <code>ifelse()</code>.</p>
<div class="sourceCode" id="cb449"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb449-1"><a href="basic-graphics.html#cb449-1" aria-hidden="true" tabindex="-1"></a><span class="do">## create a small function to cap values to 2000</span></span>
<span id="cb449-2"><a href="basic-graphics.html#cb449-2" aria-hidden="true" tabindex="-1"></a>cap_2000_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {cap <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(x<span class="sc">&gt;</span><span class="dv">2000</span>,<span class="dv">2000</span>, x)</span>
<span id="cb449-3"><a href="basic-graphics.html#cb449-3" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">return</span>(cap)}</span>
<span id="cb449-4"><a href="basic-graphics.html#cb449-4" aria-hidden="true" tabindex="-1"></a><span class="do">## we can test our function</span></span>
<span id="cb449-5"><a href="basic-graphics.html#cb449-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cap_2000_fun</span>(<span class="dv">1</span>)</span>
<span id="cb449-6"><a href="basic-graphics.html#cb449-6" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 1</span></span>
<span id="cb449-7"><a href="basic-graphics.html#cb449-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cap_2000_fun</span>(<span class="dv">10000</span>)</span>
<span id="cb449-8"><a href="basic-graphics.html#cb449-8" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 2000</span></span></code></pre></div>
<p>Now, we can apply this function to <code>airway_data_red</code> using the <code>modify()</code> from the package <code>purrr</code>.</p>
<div class="sourceCode" id="cb450"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb450-1"><a href="basic-graphics.html#cb450-1" aria-hidden="true" tabindex="-1"></a><span class="do">## modify values with function so all values &gt;2000 will be cap to 2000</span></span>
<span id="cb450-2"><a href="basic-graphics.html#cb450-2" aria-hidden="true" tabindex="-1"></a>airway_data_red_capped <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">modify</span>(airway_data_red, cap_2000_fun)</span>
<span id="cb450-3"><a href="basic-graphics.html#cb450-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb450-4"><a href="basic-graphics.html#cb450-4" aria-hidden="true" tabindex="-1"></a><span class="do">## plot histogram</span></span>
<span id="cb450-5"><a href="basic-graphics.html#cb450-5" aria-hidden="true" tabindex="-1"></a>SRR1039509_histogram_capped <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>airway_data_red_capped, <span class="fu">aes</span>(<span class="at">x=</span>SRR1039509)) <span class="sc">+</span> </span>
<span id="cb450-6"><a href="basic-graphics.html#cb450-6" aria-hidden="true" tabindex="-1"></a>     <span class="fu">geom_histogram</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">2000</span>, <span class="at">by=</span><span class="dv">100</span>), </span>
<span id="cb450-7"><a href="basic-graphics.html#cb450-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col=</span><span class="st">&quot;red&quot;</span>, </span>
<span id="cb450-8"><a href="basic-graphics.html#cb450-8" aria-hidden="true" tabindex="-1"></a>                      <span class="at">fill=</span><span class="st">&quot;green&quot;</span>, </span>
<span id="cb450-9"><a href="basic-graphics.html#cb450-9" aria-hidden="true" tabindex="-1"></a>                    <span class="at">alpha =</span> .<span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb450-10"><a href="basic-graphics.html#cb450-10" aria-hidden="true" tabindex="-1"></a>                <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Histogram for SRR1039509&quot;</span>, <span class="at">x=</span><span class="st">&quot;Expression range with capped values to 2000 if more than this value&quot;</span>, <span class="at">y=</span><span class="st">&quot;Count&quot;</span>) <span class="sc">+</span> </span>
<span id="cb450-11"><a href="basic-graphics.html#cb450-11" aria-hidden="true" tabindex="-1"></a>                <span class="fu">xlim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">2000</span>))<span class="sc">+</span></span>
<span id="cb450-12"><a href="basic-graphics.html#cb450-12" aria-hidden="true" tabindex="-1"></a>                <span class="fu">theme_minimal</span>()</span>
<span id="cb450-13"><a href="basic-graphics.html#cb450-13" aria-hidden="true" tabindex="-1"></a>SRR1039509_histogram_capped</span></code></pre></div>
<p><img src="figures/session_9_figs/24_histogram_ver_3.png" width="90%" /></p>
<p>This histogram still looks skewed to the right. We can try bin all values above 500 by creating a new function and use shorter breaks every 10 on ggplot2.</p>
<div class="sourceCode" id="cb451"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb451-1"><a href="basic-graphics.html#cb451-1" aria-hidden="true" tabindex="-1"></a><span class="do">## create a small function to cap values to 500</span></span>
<span id="cb451-2"><a href="basic-graphics.html#cb451-2" aria-hidden="true" tabindex="-1"></a>cap_500_fun <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {cap <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(x<span class="sc">&gt;</span><span class="dv">500</span>,<span class="dv">500</span>, x)</span>
<span id="cb451-3"><a href="basic-graphics.html#cb451-3" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">return</span>(cap)}</span>
<span id="cb451-4"><a href="basic-graphics.html#cb451-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb451-5"><a href="basic-graphics.html#cb451-5" aria-hidden="true" tabindex="-1"></a><span class="do">## modify values with function so all values &gt;500 will be cap to 500 </span></span>
<span id="cb451-6"><a href="basic-graphics.html#cb451-6" aria-hidden="true" tabindex="-1"></a>airway_data_red_capped <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">modify</span>(airway_data_red, cap_500_fun)</span>
<span id="cb451-7"><a href="basic-graphics.html#cb451-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb451-8"><a href="basic-graphics.html#cb451-8" aria-hidden="true" tabindex="-1"></a><span class="do">## plot histogram and modify breaks</span></span>
<span id="cb451-9"><a href="basic-graphics.html#cb451-9" aria-hidden="true" tabindex="-1"></a>SRR1039509_histogram_capped <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>airway_data_red_capped, <span class="fu">aes</span>(<span class="at">x=</span>SRR1039509)) <span class="sc">+</span> </span>
<span id="cb451-10"><a href="basic-graphics.html#cb451-10" aria-hidden="true" tabindex="-1"></a>     <span class="fu">geom_histogram</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">500</span>, <span class="at">by=</span><span class="dv">10</span>), </span>
<span id="cb451-11"><a href="basic-graphics.html#cb451-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col=</span><span class="st">&quot;red&quot;</span>, </span>
<span id="cb451-12"><a href="basic-graphics.html#cb451-12" aria-hidden="true" tabindex="-1"></a>                      <span class="at">fill=</span><span class="st">&quot;green&quot;</span>, </span>
<span id="cb451-13"><a href="basic-graphics.html#cb451-13" aria-hidden="true" tabindex="-1"></a>                    <span class="at">alpha =</span> .<span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb451-14"><a href="basic-graphics.html#cb451-14" aria-hidden="true" tabindex="-1"></a>                <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Histogram for SRR1039509&quot;</span>, <span class="at">x=</span><span class="st">&quot;Expression range with capped&quot;</span>, <span class="at">y=</span><span class="st">&quot;Count&quot;</span>) <span class="sc">+</span> </span>
<span id="cb451-15"><a href="basic-graphics.html#cb451-15" aria-hidden="true" tabindex="-1"></a>                <span class="fu">xlim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">500</span>))<span class="sc">+</span></span>
<span id="cb451-16"><a href="basic-graphics.html#cb451-16" aria-hidden="true" tabindex="-1"></a>                <span class="fu">theme_minimal</span>()</span>
<span id="cb451-17"><a href="basic-graphics.html#cb451-17" aria-hidden="true" tabindex="-1"></a>SRR1039509_histogram_capped</span></code></pre></div>
<p><img src="figures/session_9_figs/25_histogram_ver_4.png" width="90%" />
We can also remove some low expressed genes (e.g., values less than 100) and over expressed genes (e.g., values more than 500). This can be done just by defining the argument <code>xlim()</code> between 100 and 400.</p>
<div class="sourceCode" id="cb452"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb452-1"><a href="basic-graphics.html#cb452-1" aria-hidden="true" tabindex="-1"></a><span class="do">## plot histogram and modify breaks</span></span>
<span id="cb452-2"><a href="basic-graphics.html#cb452-2" aria-hidden="true" tabindex="-1"></a>SRR1039509_histogram_capped <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>airway_data_red_capped, <span class="fu">aes</span>(<span class="at">x=</span>SRR1039509)) <span class="sc">+</span> </span>
<span id="cb452-3"><a href="basic-graphics.html#cb452-3" aria-hidden="true" tabindex="-1"></a>     <span class="fu">geom_histogram</span>(<span class="at">breaks=</span><span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">500</span>, <span class="at">by=</span><span class="dv">10</span>), </span>
<span id="cb452-4"><a href="basic-graphics.html#cb452-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col=</span><span class="st">&quot;red&quot;</span>, </span>
<span id="cb452-5"><a href="basic-graphics.html#cb452-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">fill=</span><span class="st">&quot;green&quot;</span>, </span>
<span id="cb452-6"><a href="basic-graphics.html#cb452-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">alpha =</span> .<span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb452-7"><a href="basic-graphics.html#cb452-7" aria-hidden="true" tabindex="-1"></a>                <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Histogram for SRR1039509&quot;</span>, <span class="at">x=</span><span class="st">&quot;Expression range with capped -- yet be dropped less 100 and more 400 counts&quot;</span>, <span class="at">y=</span><span class="st">&quot;Count&quot;</span>) <span class="sc">+</span> </span>
<span id="cb452-8"><a href="basic-graphics.html#cb452-8" aria-hidden="true" tabindex="-1"></a>                <span class="fu">xlim</span>(<span class="fu">c</span>(<span class="dv">100</span>,<span class="dv">400</span>))<span class="sc">+</span></span>
<span id="cb452-9"><a href="basic-graphics.html#cb452-9" aria-hidden="true" tabindex="-1"></a>                <span class="fu">theme_minimal</span>()</span>
<span id="cb452-10"><a href="basic-graphics.html#cb452-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb452-11"><a href="basic-graphics.html#cb452-11" aria-hidden="true" tabindex="-1"></a><span class="do">## </span><span class="al">NOTICE</span><span class="do">: the warning about dropping rows outside the xlim().</span></span>
<span id="cb452-12"><a href="basic-graphics.html#cb452-12" aria-hidden="true" tabindex="-1"></a>SRR1039509_histogram_capped</span>
<span id="cb452-13"><a href="basic-graphics.html#cb452-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Warning messages:</span></span>
<span id="cb452-14"><a href="basic-graphics.html#cb452-14" aria-hidden="true" tabindex="-1"></a><span class="co">#1: Removed 34710 rows containing non-finite values (stat_bin). </span></span>
<span id="cb452-15"><a href="basic-graphics.html#cb452-15" aria-hidden="true" tabindex="-1"></a><span class="co">#2: Removed 20 rows containing missing values (geom_bar). </span></span></code></pre></div>
<p><img src="figures/session_9_figs/26_histogram_ver_5.png" width="90%" /></p>
</div>
<div id="multiple-histograms" class="section level2" number="10.5">
<h2><span class="header-section-number">10.5</span> Multiple histograms</h2>
<p><strong>20)</strong> To plot multiple histogram, we need two or more samples or populations with measurements on the same variable.</p>
<p>For this example, we will use the <code>crabs</code> database of the package <code>MASS</code>. We will compare the two species based on their <code>CL</code> carapace length (mm).</p>
<div class="sourceCode" id="cb453"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb453-1"><a href="basic-graphics.html#cb453-1" aria-hidden="true" tabindex="-1"></a><span class="do">## check is MASS is loaded and get &#39;crabs&#39; database</span></span>
<span id="cb453-2"><a href="basic-graphics.html#cb453-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(MASS)</span>
<span id="cb453-3"><a href="basic-graphics.html#cb453-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(crabs)</span>
<span id="cb453-4"><a href="basic-graphics.html#cb453-4" aria-hidden="true" tabindex="-1"></a>crabs_data <span class="ot">&lt;-</span> crabs</span>
<span id="cb453-5"><a href="basic-graphics.html#cb453-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(crabs_data)</span>
<span id="cb453-6"><a href="basic-graphics.html#cb453-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  200 obs. of  8 variables:</span></span>
<span id="cb453-7"><a href="basic-graphics.html#cb453-7" aria-hidden="true" tabindex="-1"></a><span class="co"># $ sp   : Factor w/ 2 levels &quot;B&quot;,&quot;O&quot;: 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb453-8"><a href="basic-graphics.html#cb453-8" aria-hidden="true" tabindex="-1"></a><span class="co"># $ sex  : Factor w/ 2 levels &quot;F&quot;,&quot;M&quot;: 2 2 2 2 2 2 2 2 2 2 ...</span></span>
<span id="cb453-9"><a href="basic-graphics.html#cb453-9" aria-hidden="true" tabindex="-1"></a><span class="co"># $ index: int  1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span id="cb453-10"><a href="basic-graphics.html#cb453-10" aria-hidden="true" tabindex="-1"></a><span class="co"># $ FL   : num  8.1 8.8 9.2 9.6 9.8 10.8 11.1 11.6 11.8 11.8 ...</span></span>
<span id="cb453-11"><a href="basic-graphics.html#cb453-11" aria-hidden="true" tabindex="-1"></a><span class="co"># $ RW   : num  6.7 7.7 7.8 7.9 8 9 9.9 9.1 9.6 10.5 ...</span></span>
<span id="cb453-12"><a href="basic-graphics.html#cb453-12" aria-hidden="true" tabindex="-1"></a><span class="co"># $ CL   : num  16.1 18.1 19 20.1 20.3 23 23.8 24.5 24.2 25.2 ...</span></span>
<span id="cb453-13"><a href="basic-graphics.html#cb453-13" aria-hidden="true" tabindex="-1"></a><span class="co"># $ CW   : num  19 20.8 22.4 23.1 23 26.5 27.1 28.4 27.8 29.3 ...</span></span>
<span id="cb453-14"><a href="basic-graphics.html#cb453-14" aria-hidden="true" tabindex="-1"></a><span class="co"># $ BD   : num  7 7.4 7.7 8.2 8.2 9.8 9.8 10.4 9.7 10.3 ...</span></span></code></pre></div>
<p>We can plot overlaid histograms with its x-axis as <code>x = CL</code> and grouped by species as <code>fill = sp</code> and using <code>geom_histogram()</code>. You can control the width of the bins of the histogram with the argument <code>binwidth</code> and we selected 2.5.</p>
<div class="sourceCode" id="cb454"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb454-1"><a href="basic-graphics.html#cb454-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Overlaid histograms</span></span>
<span id="cb454-2"><a href="basic-graphics.html#cb454-2" aria-hidden="true" tabindex="-1"></a>crabs_sp_CL_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(crabs_data, <span class="fu">aes</span>(<span class="at">x=</span>CL, <span class="at">fill=</span>sp)) <span class="sc">+</span></span>
<span id="cb454-3"><a href="basic-graphics.html#cb454-3" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">geom_histogram</span>(<span class="at">binwidth=</span><span class="dv">1</span>, <span class="at">alpha=</span>.<span class="dv">4</span>, <span class="at">position=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb454-4"><a href="basic-graphics.html#cb454-4" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">ggtitle</span>(<span class="st">&quot;Crab species CL overlap -- histograms&quot;</span>) <span class="sc">+</span></span>
<span id="cb454-5"><a href="basic-graphics.html#cb454-5" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme_minimal</span>()</span>
<span id="cb454-6"><a href="basic-graphics.html#cb454-6" aria-hidden="true" tabindex="-1"></a>crabs_sp_CL_plot</span></code></pre></div>
<p><img src="figures/session_9_figs/27_two_his_ver_1.png" width="90%" /></p>
<p>You can add a line to indicate the mean of each species, but you have to determine such values in a separated data frame. We use function <code>split()</code> and indicate that is by species with <code>crabs_data_CL$sp</code>. This will create a list with two data frames each one per species.</p>
<div class="sourceCode" id="cb455"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb455-1"><a href="basic-graphics.html#cb455-1" aria-hidden="true" tabindex="-1"></a><span class="do">## get mean per species</span></span>
<span id="cb455-2"><a href="basic-graphics.html#cb455-2" aria-hidden="true" tabindex="-1"></a>crabs_data_CL <span class="ot">&lt;-</span> <span class="fu">subset</span>(crabs_data, <span class="at">select =</span> <span class="fu">c</span>(sp, CL))</span>
<span id="cb455-3"><a href="basic-graphics.html#cb455-3" aria-hidden="true" tabindex="-1"></a>crabs_data_CL_sp_list <span class="ot">&lt;-</span><span class="fu">split</span>(crabs_data_CL, crabs_data_CL<span class="sc">$</span>sp)</span>
<span id="cb455-4"><a href="basic-graphics.html#cb455-4" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(crabs_data_CL_sp_list)</span>
<span id="cb455-5"><a href="basic-graphics.html#cb455-5" aria-hidden="true" tabindex="-1"></a><span class="co">#List of 2</span></span>
<span id="cb455-6"><a href="basic-graphics.html#cb455-6" aria-hidden="true" tabindex="-1"></a><span class="co"># $ B:&#39;data.frame&#39;: 100 obs. of  2 variables:</span></span>
<span id="cb455-7"><a href="basic-graphics.html#cb455-7" aria-hidden="true" tabindex="-1"></a><span class="co">#  ..$ sp: Factor w/ 2 levels &quot;B&quot;,&quot;O&quot;: 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb455-8"><a href="basic-graphics.html#cb455-8" aria-hidden="true" tabindex="-1"></a><span class="co">#  ..$ CL: num [1:100] 16.1 18.1 19 20.1 20.3 23 23.8 24.5 24.2 25.2 ...</span></span>
<span id="cb455-9"><a href="basic-graphics.html#cb455-9" aria-hidden="true" tabindex="-1"></a><span class="co"># $ O:&#39;data.frame&#39;: 100 obs. of  2 variables:</span></span>
<span id="cb455-10"><a href="basic-graphics.html#cb455-10" aria-hidden="true" tabindex="-1"></a><span class="co">#  ..$ sp: Factor w/ 2 levels &quot;B&quot;,&quot;O&quot;: 2 2 2 2 2 2 2 2 2 2 ...</span></span>
<span id="cb455-11"><a href="basic-graphics.html#cb455-11" aria-hidden="true" tabindex="-1"></a><span class="co">#  ..$ CL: num [1:100] 16.7 20.2 20.7 22.7 23.2 24.2 26 27.1 26.6 27.5 ...</span></span></code></pre></div>
<p>Here are two forms that we get the mean of each species.</p>
<div class="sourceCode" id="cb456"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb456-1"><a href="basic-graphics.html#cb456-1" aria-hidden="true" tabindex="-1"></a><span class="do">## first form</span></span>
<span id="cb456-2"><a href="basic-graphics.html#cb456-2" aria-hidden="true" tabindex="-1"></a>means_out_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb456-3"><a href="basic-graphics.html#cb456-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(crabs_data_CL_sp_list)) {</span>
<span id="cb456-4"><a href="basic-graphics.html#cb456-4" aria-hidden="true" tabindex="-1"></a>                        one_species_name <span class="ot">&lt;-</span> <span class="fu">names</span>(crabs_data_CL_sp_list[i])</span>
<span id="cb456-5"><a href="basic-graphics.html#cb456-5" aria-hidden="true" tabindex="-1"></a>                      one_species_mean <span class="ot">&lt;-</span> <span class="fu">mean</span>(crabs_data_CL_sp_list[[i]][,<span class="dv">2</span>], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb456-6"><a href="basic-graphics.html#cb456-6" aria-hidden="true" tabindex="-1"></a>                            one_sp_out <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sp =</span> one_species_name, <span class="at">sp_mean =</span> one_species_mean, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb456-7"><a href="basic-graphics.html#cb456-7" aria-hidden="true" tabindex="-1"></a>                     means_out_list[[i]] <span class="ot">&lt;-</span> one_sp_out</span>
<span id="cb456-8"><a href="basic-graphics.html#cb456-8" aria-hidden="true" tabindex="-1"></a>                                        }</span>
<span id="cb456-9"><a href="basic-graphics.html#cb456-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb456-10"><a href="basic-graphics.html#cb456-10" aria-hidden="true" tabindex="-1"></a>means_out_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, means_out_list)</span>
<span id="cb456-11"><a href="basic-graphics.html#cb456-11" aria-hidden="true" tabindex="-1"></a>means_out_df</span>
<span id="cb456-12"><a href="basic-graphics.html#cb456-12" aria-hidden="true" tabindex="-1"></a><span class="co">#  sp sp_mean</span></span>
<span id="cb456-13"><a href="basic-graphics.html#cb456-13" aria-hidden="true" tabindex="-1"></a><span class="co">#1  B  30.058</span></span>
<span id="cb456-14"><a href="basic-graphics.html#cb456-14" aria-hidden="true" tabindex="-1"></a><span class="co">#2  O  34.153</span></span>
<span id="cb456-15"><a href="basic-graphics.html#cb456-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb456-16"><a href="basic-graphics.html#cb456-16" aria-hidden="true" tabindex="-1"></a><span class="do">## second form</span></span>
<span id="cb456-17"><a href="basic-graphics.html#cb456-17" aria-hidden="true" tabindex="-1"></a>crab_removed_sp_column_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(crabs_data_CL_sp_list, <span class="cf">function</span>(x) { x[,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span>; x })</span>
<span id="cb456-18"><a href="basic-graphics.html#cb456-18" aria-hidden="true" tabindex="-1"></a>crab_sp_means <span class="ot">&lt;-</span> <span class="fu">lapply</span>(crab_removed_sp_column_list, colMeans, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb456-19"><a href="basic-graphics.html#cb456-19" aria-hidden="true" tabindex="-1"></a>means_out_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, crab_sp_means)</span>
<span id="cb456-20"><a href="basic-graphics.html#cb456-20" aria-hidden="true" tabindex="-1"></a>means_out_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(means_out_df)</span>
<span id="cb456-21"><a href="basic-graphics.html#cb456-21" aria-hidden="true" tabindex="-1"></a>means_out_df<span class="sc">$</span>sp <span class="ot">&lt;-</span> <span class="fu">rownames</span>(means_out_df)</span>
<span id="cb456-22"><a href="basic-graphics.html#cb456-22" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(means_out_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;sp_mean&quot;</span>, <span class="st">&quot;sp&quot;</span>)</span>
<span id="cb456-23"><a href="basic-graphics.html#cb456-23" aria-hidden="true" tabindex="-1"></a>means_out_df</span>
<span id="cb456-24"><a href="basic-graphics.html#cb456-24" aria-hidden="true" tabindex="-1"></a><span class="co">#  sp_mean sp</span></span>
<span id="cb456-25"><a href="basic-graphics.html#cb456-25" aria-hidden="true" tabindex="-1"></a><span class="co">#B  30.058  B</span></span>
<span id="cb456-26"><a href="basic-graphics.html#cb456-26" aria-hidden="true" tabindex="-1"></a><span class="co">#O  34.153  O</span></span></code></pre></div>
<p>We can add a line to indicate the location of the mean of the histogram of each species with <code>geom_vline()</code>.</p>
<div class="sourceCode" id="cb457"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb457-1"><a href="basic-graphics.html#cb457-1" aria-hidden="true" tabindex="-1"></a><span class="do">## add mean line</span></span>
<span id="cb457-2"><a href="basic-graphics.html#cb457-2" aria-hidden="true" tabindex="-1"></a>crabs_sp_CL_plot <span class="ot">&lt;-</span> crabs_sp_CL_plot <span class="sc">+</span> <span class="fu">geom_vline</span>(<span class="at">data=</span>means_out_df, </span>
<span id="cb457-3"><a href="basic-graphics.html#cb457-3" aria-hidden="true" tabindex="-1"></a>                                            <span class="fu">aes</span>(<span class="at">xintercept=</span>sp_mean,  <span class="at">colour=</span>sp),</span>
<span id="cb457-4"><a href="basic-graphics.html#cb457-4" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">size=</span><span class="fl">0.5</span>)</span>
<span id="cb457-5"><a href="basic-graphics.html#cb457-5" aria-hidden="true" tabindex="-1"></a>crabs_sp_CL_plot</span></code></pre></div>
<p><img src="figures/session_9_figs/28_two_his_ver_2.png" width="90%" /></p>
</div>
<div id="density-plots" class="section level2" number="10.6">
<h2><span class="header-section-number">10.6</span> Density plots</h2>
<p><strong>21)</strong> These graphics are similar to histograms, but they are called <code>geom_density()</code>. We will use the same crab dataset to illustrate this example.</p>
<div class="sourceCode" id="cb458"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb458-1"><a href="basic-graphics.html#cb458-1" aria-hidden="true" tabindex="-1"></a><span class="do">## check is MASS is loaded and get &#39;crabs&#39; database</span></span>
<span id="cb458-2"><a href="basic-graphics.html#cb458-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(MASS)</span>
<span id="cb458-3"><a href="basic-graphics.html#cb458-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(crabs)</span>
<span id="cb458-4"><a href="basic-graphics.html#cb458-4" aria-hidden="true" tabindex="-1"></a>crabs_data <span class="ot">&lt;-</span> crabs</span>
<span id="cb458-5"><a href="basic-graphics.html#cb458-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(crabs_data)</span>
<span id="cb458-6"><a href="basic-graphics.html#cb458-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  200 obs. of  8 variables:</span></span>
<span id="cb458-7"><a href="basic-graphics.html#cb458-7" aria-hidden="true" tabindex="-1"></a><span class="co"># $ sp   : Factor w/ 2 levels &quot;B&quot;,&quot;O&quot;: 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb458-8"><a href="basic-graphics.html#cb458-8" aria-hidden="true" tabindex="-1"></a><span class="co"># $ sex  : Factor w/ 2 levels &quot;F&quot;,&quot;M&quot;: 2 2 2 2 2 2 2 2 2 2 ...</span></span>
<span id="cb458-9"><a href="basic-graphics.html#cb458-9" aria-hidden="true" tabindex="-1"></a><span class="co"># $ index: int  1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span id="cb458-10"><a href="basic-graphics.html#cb458-10" aria-hidden="true" tabindex="-1"></a><span class="co"># $ FL   : num  8.1 8.8 9.2 9.6 9.8 10.8 11.1 11.6 11.8 11.8 ...</span></span>
<span id="cb458-11"><a href="basic-graphics.html#cb458-11" aria-hidden="true" tabindex="-1"></a><span class="co"># $ RW   : num  6.7 7.7 7.8 7.9 8 9 9.9 9.1 9.6 10.5 ...</span></span>
<span id="cb458-12"><a href="basic-graphics.html#cb458-12" aria-hidden="true" tabindex="-1"></a><span class="co"># $ CL   : num  16.1 18.1 19 20.1 20.3 23 23.8 24.5 24.2 25.2 ...</span></span>
<span id="cb458-13"><a href="basic-graphics.html#cb458-13" aria-hidden="true" tabindex="-1"></a><span class="co"># $ CW   : num  19 20.8 22.4 23.1 23 26.5 27.1 28.4 27.8 29.3 ...</span></span>
<span id="cb458-14"><a href="basic-graphics.html#cb458-14" aria-hidden="true" tabindex="-1"></a><span class="co"># $ BD   : num  7 7.4 7.7 8.2 8.2 9.8 9.8 10.4 9.7 10.3 ...</span></span></code></pre></div>
<p>We can plot overlaid densities with its x-axis as <code>x = CL</code> and grouped by species as <code>fill = sp</code>.</p>
<div class="sourceCode" id="cb459"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb459-1"><a href="basic-graphics.html#cb459-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Overlaid densities</span></span>
<span id="cb459-2"><a href="basic-graphics.html#cb459-2" aria-hidden="true" tabindex="-1"></a>crabs_sp_CL_plot_den <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(crabs_data, <span class="fu">aes</span>(<span class="at">x=</span>CL, <span class="at">fill=</span>sp)) <span class="sc">+</span></span>
<span id="cb459-3"><a href="basic-graphics.html#cb459-3" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">geom_density</span>(<span class="at">alpha=</span>.<span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb459-4"><a href="basic-graphics.html#cb459-4" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">ggtitle</span>(<span class="st">&quot;Crab species CL overlap -- density plots&quot;</span>) <span class="sc">+</span></span>
<span id="cb459-5"><a href="basic-graphics.html#cb459-5" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme_minimal</span>()</span>
<span id="cb459-6"><a href="basic-graphics.html#cb459-6" aria-hidden="true" tabindex="-1"></a>crabs_sp_CL_plot_den</span></code></pre></div>
<p><img src="figures/session_9_figs/29_two_density_plots_1.png" width="90%" /></p>
<p><strong>22)</strong> The flexibility of <strong>ggplot2</strong> can be evidenced by combining histograms and density plots with the mean values.</p>
<div class="sourceCode" id="cb460"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb460-1"><a href="basic-graphics.html#cb460-1" aria-hidden="true" tabindex="-1"></a>crabs_sp_CL_plot_his_den <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(crabs_data, <span class="fu">aes</span>(<span class="at">x=</span>CL, <span class="at">fill=</span>sp)) <span class="sc">+</span> </span>
<span id="cb460-2"><a href="basic-graphics.html#cb460-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">y=</span>..density.., <span class="at">fill=</span>sp), </span>
<span id="cb460-3"><a href="basic-graphics.html#cb460-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">binwidth =</span> <span class="dv">1</span>,</span>
<span id="cb460-4"><a href="basic-graphics.html#cb460-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">alpha =</span> .<span class="dv">6</span>, </span>
<span id="cb460-5"><a href="basic-graphics.html#cb460-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">colour =</span> <span class="st">&quot;black&quot;</span>, </span>
<span id="cb460-6"><a href="basic-graphics.html#cb460-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">position =</span> <span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb460-7"><a href="basic-graphics.html#cb460-7" aria-hidden="true" tabindex="-1"></a>   <span class="fu">geom_density</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x=</span>CL, <span class="at">fill=</span>sp),</span>
<span id="cb460-8"><a href="basic-graphics.html#cb460-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">alpha =</span> .<span class="dv">4</span>,</span>
<span id="cb460-9"><a href="basic-graphics.html#cb460-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">position =</span> <span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb460-10"><a href="basic-graphics.html#cb460-10" aria-hidden="true" tabindex="-1"></a>   <span class="fu">ggtitle</span>(<span class="st">&quot;Crab species CL overlap -- histograms and density plots&quot;</span>) <span class="sc">+</span></span>
<span id="cb460-11"><a href="basic-graphics.html#cb460-11" aria-hidden="true" tabindex="-1"></a>          <span class="fu">theme_minimal</span>()</span>
<span id="cb460-12"><a href="basic-graphics.html#cb460-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb460-13"><a href="basic-graphics.html#cb460-13" aria-hidden="true" tabindex="-1"></a><span class="co"># change some colors</span></span>
<span id="cb460-14"><a href="basic-graphics.html#cb460-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb460-15"><a href="basic-graphics.html#cb460-15" aria-hidden="true" tabindex="-1"></a>crabs_sp_CL_plot_his_den <span class="ot">&lt;-</span> crabs_sp_CL_plot_his_den <span class="sc">+</span></span>
<span id="cb460-16"><a href="basic-graphics.html#cb460-16" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">scale_color_brewer</span>(<span class="at">palette=</span><span class="st">&quot;Set3&quot;</span>)<span class="sc">+</span></span>
<span id="cb460-17"><a href="basic-graphics.html#cb460-17" aria-hidden="true" tabindex="-1"></a>                            <span class="fu">scale_fill_brewer</span>(<span class="at">palette=</span><span class="st">&quot;Set3&quot;</span>)</span>
<span id="cb460-18"><a href="basic-graphics.html#cb460-18" aria-hidden="true" tabindex="-1"></a>crabs_sp_CL_plot_his_den</span></code></pre></div>
<p><img src="figures/session_9_figs/30_two_histo_denstity.png" width="90%" /></p>
</div>
<div id="box-plots" class="section level2" number="10.7">
<h2><span class="header-section-number">10.7</span> Box plots</h2>
<p>These graphics are similar to histograms and density plots that summarizes a distribution. In these plots, the graphics show the following summaries: (1) First quartile (Q1) or 25th Percentile, which is the middle value between the smallest value and the median of the dataset; (2) Median (Q2) or 50th Percentile, which is the middle value of the dataset; (3) Third quartile (Q3) or 75th Percentile, which is the middle value between the median and the highest value of the dataset; (4) A “box” named the interquartile range (IQR) that is bounded between the 25th and 75th percentiles; (5) Line or whisker that extend to a “minimum” equal to Q1 - 1.5xIQR, (6) Line or whisker that extend to a “maximum” equal to Q3 + 1.5xIQR; and (7) Outliers or values that fall below the “minimum” and above the “maximum”.</p>
<p><img src="figures/session_9_figs/31_boxplot_exaplained.png" width="90%" /></p>
<p>Source: <a href="https://towardsdatascience.com/understanding-boxplots-5e2df7bcbd51" target="_blank">Understanding Boxplots</a> by Michael Galarnyk.</p>
<p><strong>23)</strong> We can use the same <code>crabs</code> dataset to revise how to construct boxplots in <strong>ggplot2</strong>.</p>
<div class="sourceCode" id="cb461"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb461-1"><a href="basic-graphics.html#cb461-1" aria-hidden="true" tabindex="-1"></a><span class="do">## check is MASS is loaded and get &#39;crabs&#39; database</span></span>
<span id="cb461-2"><a href="basic-graphics.html#cb461-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(MASS)</span>
<span id="cb461-3"><a href="basic-graphics.html#cb461-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(crabs)</span>
<span id="cb461-4"><a href="basic-graphics.html#cb461-4" aria-hidden="true" tabindex="-1"></a>crabs_data <span class="ot">&lt;-</span> crabs</span>
<span id="cb461-5"><a href="basic-graphics.html#cb461-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(crabs_data)</span>
<span id="cb461-6"><a href="basic-graphics.html#cb461-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  200 obs. of  8 variables:</span></span>
<span id="cb461-7"><a href="basic-graphics.html#cb461-7" aria-hidden="true" tabindex="-1"></a><span class="co"># $ sp   : Factor w/ 2 levels &quot;B&quot;,&quot;O&quot;: 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb461-8"><a href="basic-graphics.html#cb461-8" aria-hidden="true" tabindex="-1"></a><span class="co"># $ sex  : Factor w/ 2 levels &quot;F&quot;,&quot;M&quot;: 2 2 2 2 2 2 2 2 2 2 ...</span></span>
<span id="cb461-9"><a href="basic-graphics.html#cb461-9" aria-hidden="true" tabindex="-1"></a><span class="co"># $ index: int  1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span id="cb461-10"><a href="basic-graphics.html#cb461-10" aria-hidden="true" tabindex="-1"></a><span class="co"># $ FL   : num  8.1 8.8 9.2 9.6 9.8 10.8 11.1 11.6 11.8 11.8 ...</span></span>
<span id="cb461-11"><a href="basic-graphics.html#cb461-11" aria-hidden="true" tabindex="-1"></a><span class="co"># $ RW   : num  6.7 7.7 7.8 7.9 8 9 9.9 9.1 9.6 10.5 ...</span></span>
<span id="cb461-12"><a href="basic-graphics.html#cb461-12" aria-hidden="true" tabindex="-1"></a><span class="co"># $ CL   : num  16.1 18.1 19 20.1 20.3 23 23.8 24.5 24.2 25.2 ...</span></span>
<span id="cb461-13"><a href="basic-graphics.html#cb461-13" aria-hidden="true" tabindex="-1"></a><span class="co"># $ CW   : num  19 20.8 22.4 23.1 23 26.5 27.1 28.4 27.8 29.3 ...</span></span>
<span id="cb461-14"><a href="basic-graphics.html#cb461-14" aria-hidden="true" tabindex="-1"></a><span class="co"># $ BD   : num  7 7.4 7.7 8.2 8.2 9.8 9.8 10.4 9.7 10.3 ...</span></span></code></pre></div>
<p>We can present boxplots with its x-axis by species <code>x = sp</code>, y-axis by the CL measurement <code>y = CL</code> and box color as <code>fill = sp</code>.</p>
<div class="sourceCode" id="cb462"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb462-1"><a href="basic-graphics.html#cb462-1" aria-hidden="true" tabindex="-1"></a><span class="do">## check ig ggplot2 is loaded</span></span>
<span id="cb462-2"><a href="basic-graphics.html#cb462-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggplot2)</span>
<span id="cb462-3"><a href="basic-graphics.html#cb462-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb462-4"><a href="basic-graphics.html#cb462-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Boxplots with standard color without legend -- guides(fill=FALSE)</span></span>
<span id="cb462-5"><a href="basic-graphics.html#cb462-5" aria-hidden="true" tabindex="-1"></a>crabs_sp_CL_boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(crabs_data, <span class="fu">aes</span>(<span class="at">x=</span>sp, <span class="at">y=</span>CL, <span class="at">fill=</span>sp)) <span class="sc">+</span> </span>
<span id="cb462-6"><a href="basic-graphics.html#cb462-6" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">geom_boxplot</span>(<span class="at">width=</span><span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb462-7"><a href="basic-graphics.html#cb462-7" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Crab species CL overlap -- boxplots&quot;</span>, <span class="at">x=</span><span class="st">&quot;Species&quot;</span>, <span class="at">y=</span><span class="st">&quot;CL - carapace length (mm)&quot;</span>) <span class="sc">+</span></span>
<span id="cb462-8"><a href="basic-graphics.html#cb462-8" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme_minimal</span>()</span>
<span id="cb462-9"><a href="basic-graphics.html#cb462-9" aria-hidden="true" tabindex="-1"></a>crabs_sp_CL_boxplot_v1 <span class="ot">&lt;-</span> crabs_sp_CL_boxplot <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span><span class="fu">c</span>(<span class="st">&quot;#C7A76C&quot;</span>, <span class="st">&quot;#99B56B&quot;</span>))</span>
<span id="cb462-10"><a href="basic-graphics.html#cb462-10" aria-hidden="true" tabindex="-1"></a>crabs_sp_CL_boxplot_v1</span></code></pre></div>
<p><img src="figures/session_9_figs/40_boxplot_base.png" width="70%" /></p>
<p>You can flip coordinates with <code>coord_flip()</code> and changed the colors of the plots using <code>scale_fill_manual()</code>.</p>
<div class="sourceCode" id="cb463"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb463-1"><a href="basic-graphics.html#cb463-1" aria-hidden="true" tabindex="-1"></a><span class="do">## flip coordinates</span></span>
<span id="cb463-2"><a href="basic-graphics.html#cb463-2" aria-hidden="true" tabindex="-1"></a>crabs_sp_CL_boxplot_v2 <span class="ot">&lt;-</span> crabs_sp_CL_boxplot <span class="sc">+</span> <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">scale_fill_manual</span>(<span class="at">values =</span><span class="fu">c</span>(<span class="st">&quot;#8C241B&quot;</span>, <span class="st">&quot;#29809E&quot;</span>))</span>
<span id="cb463-3"><a href="basic-graphics.html#cb463-3" aria-hidden="true" tabindex="-1"></a>crabs_sp_CL_boxplot_v2</span></code></pre></div>
<p><img src="figures/session_9_figs/41_boxplot_flip.png" width="70%" /></p>
</div>
<div id="violin-plots" class="section level2" number="10.8">
<h2><span class="header-section-number">10.8</span> Violin plots</h2>
<p><strong>24)</strong> We can use again the <code>crabs</code> dataset for violin plots containing its corresponding boxplot and <code>mean</code> value marker. Notice that we can stack different plots on a ggplot object. For violin plots, we selected a white color with <code>fill= "white"</code> and thinner silhouette with <code>width = 0.5</code>. For box plots, we selected a <code>width = 0.3</code> and also added a diamond symbol for the mean value with <code>stat_summar()</code>.</p>
<div class="sourceCode" id="cb464"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb464-1"><a href="basic-graphics.html#cb464-1" aria-hidden="true" tabindex="-1"></a><span class="do">## check is MASS is loaded and get &#39;crabs&#39; database</span></span>
<span id="cb464-2"><a href="basic-graphics.html#cb464-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(MASS)</span>
<span id="cb464-3"><a href="basic-graphics.html#cb464-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(crabs)</span>
<span id="cb464-4"><a href="basic-graphics.html#cb464-4" aria-hidden="true" tabindex="-1"></a>crabs_data <span class="ot">&lt;-</span> crabs</span>
<span id="cb464-5"><a href="basic-graphics.html#cb464-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(crabs_data)</span>
<span id="cb464-6"><a href="basic-graphics.html#cb464-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  200 obs. of  8 variables:</span></span>
<span id="cb464-7"><a href="basic-graphics.html#cb464-7" aria-hidden="true" tabindex="-1"></a><span class="co"># $ sp   : Factor w/ 2 levels &quot;B&quot;,&quot;O&quot;: 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb464-8"><a href="basic-graphics.html#cb464-8" aria-hidden="true" tabindex="-1"></a><span class="co"># $ sex  : Factor w/ 2 levels &quot;F&quot;,&quot;M&quot;: 2 2 2 2 2 2 2 2 2 2 ...</span></span>
<span id="cb464-9"><a href="basic-graphics.html#cb464-9" aria-hidden="true" tabindex="-1"></a><span class="co"># $ index: int  1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span id="cb464-10"><a href="basic-graphics.html#cb464-10" aria-hidden="true" tabindex="-1"></a><span class="co"># $ FL   : num  8.1 8.8 9.2 9.6 9.8 10.8 11.1 11.6 11.8 11.8 ...</span></span>
<span id="cb464-11"><a href="basic-graphics.html#cb464-11" aria-hidden="true" tabindex="-1"></a><span class="co"># $ RW   : num  6.7 7.7 7.8 7.9 8 9 9.9 9.1 9.6 10.5 ...</span></span>
<span id="cb464-12"><a href="basic-graphics.html#cb464-12" aria-hidden="true" tabindex="-1"></a><span class="co"># $ CL   : num  16.1 18.1 19 20.1 20.3 23 23.8 24.5 24.2 25.2 ...</span></span>
<span id="cb464-13"><a href="basic-graphics.html#cb464-13" aria-hidden="true" tabindex="-1"></a><span class="co"># $ CW   : num  19 20.8 22.4 23.1 23 26.5 27.1 28.4 27.8 29.3 ...</span></span>
<span id="cb464-14"><a href="basic-graphics.html#cb464-14" aria-hidden="true" tabindex="-1"></a><span class="co"># $ BD   : num  7 7.4 7.7 8.2 8.2 9.8 9.8 10.4 9.7 10.3 ...</span></span>
<span id="cb464-15"><a href="basic-graphics.html#cb464-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb464-16"><a href="basic-graphics.html#cb464-16" aria-hidden="true" tabindex="-1"></a><span class="do">## violin and boxplots with mean indicated</span></span>
<span id="cb464-17"><a href="basic-graphics.html#cb464-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb464-18"><a href="basic-graphics.html#cb464-18" aria-hidden="true" tabindex="-1"></a>crabs_sp_CL_violin <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(crabs_data, <span class="fu">aes</span>(<span class="at">x=</span>sp, <span class="at">y=</span>CL, <span class="at">fill=</span>sp)) <span class="sc">+</span> </span>
<span id="cb464-19"><a href="basic-graphics.html#cb464-19" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">geom_violin</span>(<span class="at">trim=</span><span class="cn">FALSE</span>, <span class="at">width=</span><span class="fl">0.5</span>, <span class="at">fill=</span> <span class="st">&quot;white&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>) <span class="sc">+</span></span>
<span id="cb464-20"><a href="basic-graphics.html#cb464-20" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">geom_boxplot</span>(<span class="at">width=</span><span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb464-21"><a href="basic-graphics.html#cb464-21" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">stat_summary</span>(<span class="at">fun =</span> mean, <span class="at">geom=</span><span class="st">&quot;point&quot;</span>, <span class="at">shape=</span><span class="dv">5</span>, <span class="at">size=</span><span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb464-22"><a href="basic-graphics.html#cb464-22" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Crab species CL overlap -- violin and box plots&quot;</span>, <span class="at">x=</span><span class="st">&quot;Species&quot;</span>, <span class="at">y=</span><span class="st">&quot;CL - carapace length (mm)&quot;</span>) <span class="sc">+</span></span>
<span id="cb464-23"><a href="basic-graphics.html#cb464-23" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">scale_fill_manual</span>(<span class="at">values =</span><span class="fu">c</span>(<span class="st">&quot;#8C241B&quot;</span>, <span class="st">&quot;#29809E&quot;</span>)) <span class="sc">+</span></span>
<span id="cb464-24"><a href="basic-graphics.html#cb464-24" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme_minimal</span>()</span>
<span id="cb464-25"><a href="basic-graphics.html#cb464-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb464-26"><a href="basic-graphics.html#cb464-26" aria-hidden="true" tabindex="-1"></a>crabs_sp_CL_violin</span></code></pre></div>
<p><img src="figures/session_9_figs/42_violin_boxplot.png" width="70%" /></p>
</div>
<div id="scatterplots" class="section level2" number="10.9">
<h2><span class="header-section-number">10.9</span> Scatterplots</h2>
<p>These are graphics where the relationship of two variables presented in cartesian (x versus y) coordinates to display values. In <strong>ggplot2</strong>, such data points are presented with <code>geom_point()</code> and, in most cases, tendency and regression lines are added with <code>geom_smooth()</code>.</p>
<p>As in the case of choices of colors, there are several types of point shapes (markers) and lines. These can be explored with options of the R package <a href="https://cran.r-project.org/web/packages/ggpubr/" target="_blank"><strong>ggpubr</strong></a>. This package provides some easy-to-use functions for creating and customizing <strong>ggplot2</strong>- based publication ready plots.</p>
<div class="sourceCode" id="cb465"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb465-1"><a href="basic-graphics.html#cb465-1" aria-hidden="true" tabindex="-1"></a><span class="do">## intall ggpubr</span></span>
<span id="cb465-2"><a href="basic-graphics.html#cb465-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggpubr&quot;</span>)</span>
<span id="cb465-3"><a href="basic-graphics.html#cb465-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb465-4"><a href="basic-graphics.html#cb465-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb465-5"><a href="basic-graphics.html#cb465-5" aria-hidden="true" tabindex="-1"></a><span class="do">## we can select from these types of point shapes; the number indicates the shape code.</span></span>
<span id="cb465-6"><a href="basic-graphics.html#cb465-6" aria-hidden="true" tabindex="-1"></a><span class="fu">show_point_shapes</span>()<span class="sc">+</span><span class="fu">theme_minimal</span>()<span class="sc">+</span><span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;&quot;</span>)<span class="sc">+</span><span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), <span class="at">axis.text.y =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="figures/session_10_figs/43_shape_forms.png" width="90%" /></p>
<p>We can control the line type with <code>geom_line()</code> and its argument <code>linetype</code>. Lines can also be modified manually with <code>scale_linetype_manual()</code> and it argument <code>values</code>.</p>
<div class="sourceCode" id="cb466"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb466-1"><a href="basic-graphics.html#cb466-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we can select from these types of lines, the number indicates the line code.</span></span>
<span id="cb466-2"><a href="basic-graphics.html#cb466-2" aria-hidden="true" tabindex="-1"></a><span class="fu">show_line_types</span>()<span class="sc">+</span><span class="fu">theme_minimal</span>()</span></code></pre></div>
<p><img src="figures/session_10_figs/44_line_forms.png" width="90%" /></p>
<p>Notice that you can also determine the type of line with argument <code>lty</code> that can be used in special type of <strong>ggplot</strong> function like <code>geom_smooth</code>. These codes are <code>lty = 1</code> for “solid”, <code>lty = 2</code> “dashed”, <code>lty = 3</code> “dotted”, <code>lty = 4</code> “dotdash”, <code>lty = 5</code> “longdash” and <code>lty = 6</code> “twodash”.</p>
<p><strong>25)</strong> We will use the <code>oribatid</code> dataset from the R package <code>ade4</code>. This data set contains information about environmental control and spatial structure in ecological communities of Oribatid mites. The environmental variables include <code>substrate</code> a factor with seven levels that describes the nature of the substratum, <code>shrubs</code> a factor with three levels that describes the absence/presence of shrubs, <code>topo</code> a factor with two levels that describes the microtopography, <code>density</code> substratum density (g.L−1) and <code>water</code> content of the substratum (g.L−1). We will merge in a single data frame the environmental data and fauna information of these 70 sites to explore several types of scatterplots.</p>
<p>We can prepare <code>oribatid</code> dataset for the scatterplot.</p>
<div class="sourceCode" id="cb467"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb467-1"><a href="basic-graphics.html#cb467-1" aria-hidden="true" tabindex="-1"></a><span class="do">## check is ade4 is loaded and get &#39;oribatid&#39; database</span></span>
<span id="cb467-2"><a href="basic-graphics.html#cb467-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ade4)</span>
<span id="cb467-3"><a href="basic-graphics.html#cb467-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(oribatid)</span>
<span id="cb467-4"><a href="basic-graphics.html#cb467-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-5"><a href="basic-graphics.html#cb467-5" aria-hidden="true" tabindex="-1"></a><span class="do">## environmental and genetic data</span></span>
<span id="cb467-6"><a href="basic-graphics.html#cb467-6" aria-hidden="true" tabindex="-1"></a>envir_data <span class="ot">&lt;-</span> oribatid<span class="sc">$</span>envir</span>
<span id="cb467-7"><a href="basic-graphics.html#cb467-7" aria-hidden="true" tabindex="-1"></a>fauna_data <span class="ot">&lt;-</span> oribatid<span class="sc">$</span>fau</span>
<span id="cb467-8"><a href="basic-graphics.html#cb467-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-9"><a href="basic-graphics.html#cb467-9" aria-hidden="true" tabindex="-1"></a><span class="do">## total number of oribatid mites</span></span>
<span id="cb467-10"><a href="basic-graphics.html#cb467-10" aria-hidden="true" tabindex="-1"></a>fauna_data<span class="sc">$</span>total_mites <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(fauna_data)</span>
<span id="cb467-11"><a href="basic-graphics.html#cb467-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-12"><a href="basic-graphics.html#cb467-12" aria-hidden="true" tabindex="-1"></a><span class="do">## we add a common column in both data frames, in this case the &#39;site&#39;</span></span>
<span id="cb467-13"><a href="basic-graphics.html#cb467-13" aria-hidden="true" tabindex="-1"></a>envir_data<span class="sc">$</span>site <span class="ot">&lt;-</span> <span class="fu">rownames</span>(envir_data)</span>
<span id="cb467-14"><a href="basic-graphics.html#cb467-14" aria-hidden="true" tabindex="-1"></a>fauna_data<span class="sc">$</span>site <span class="ot">&lt;-</span> <span class="fu">rownames</span>(fauna_data)</span>
<span id="cb467-15"><a href="basic-graphics.html#cb467-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-16"><a href="basic-graphics.html#cb467-16" aria-hidden="true" tabindex="-1"></a><span class="do">## merge both data frames using &#39;site&#39;</span></span>
<span id="cb467-17"><a href="basic-graphics.html#cb467-17" aria-hidden="true" tabindex="-1"></a>oribatid_data <span class="ot">&lt;-</span> <span class="fu">merge</span>(envir_data, fauna_data, <span class="at">by =</span><span class="st">&quot;site&quot;</span>)</span>
<span id="cb467-18"><a href="basic-graphics.html#cb467-18" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(oribatid_data)</span>
<span id="cb467-19"><a href="basic-graphics.html#cb467-19" aria-hidden="true" tabindex="-1"></a><span class="co"># you can explore this data frame content</span></span>
<span id="cb467-20"><a href="basic-graphics.html#cb467-20" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(oribatid_data)</span>
<span id="cb467-21"><a href="basic-graphics.html#cb467-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  70 obs. of  41 variables:</span></span>
<span id="cb467-22"><a href="basic-graphics.html#cb467-22" aria-hidden="true" tabindex="-1"></a><span class="co"># $ site         : chr  &quot;1&quot; &quot;10&quot; &quot;11&quot; &quot;12&quot; ...</span></span>
<span id="cb467-23"><a href="basic-graphics.html#cb467-23" aria-hidden="true" tabindex="-1"></a><span class="co"># $ substrate    : Factor w/ 7 levels &quot;inter&quot;,&quot;litter&quot;,..: 4 4 6 4 4 1 1 1 1 4 ...</span></span>
<span id="cb467-24"><a href="basic-graphics.html#cb467-24" aria-hidden="true" tabindex="-1"></a><span class="co"># $ shrubs       : Factor w/ 3 levels &quot;few&quot;,&quot;many&quot;,&quot;none&quot;: 1 2 2 1 2 2 2 1 2 2 ...</span></span>
<span id="cb467-25"><a href="basic-graphics.html#cb467-25" aria-hidden="true" tabindex="-1"></a><span class="co"># $ topo         : Factor w/ 2 levels &quot;blanket&quot;,&quot;hummock&quot;: 2 2 1 2 2 1 1 2 2 2 ...</span></span>
<span id="cb467-26"><a href="basic-graphics.html#cb467-26" aria-hidden="true" tabindex="-1"></a><span class="co"># $ density      : num  39.2 32.1 35.6 46.8 28 ...</span></span>
<span id="cb467-27"><a href="basic-graphics.html#cb467-27" aria-hidden="true" tabindex="-1"></a><span class="co"># $ water        : num  350 221 134 406 244 ...</span></span>
<span id="cb467-28"><a href="basic-graphics.html#cb467-28" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Brachy       : int  17 22 36 28 3 41 6 7 9 19 ...</span></span>
<span id="cb467-29"><a href="basic-graphics.html#cb467-29" aria-hidden="true" tabindex="-1"></a><span class="co"># $ PHTH         : int  5 4 7 2 2 5 0 2 0 3 ...</span></span>
<span id="cb467-30"><a href="basic-graphics.html#cb467-30" aria-hidden="true" tabindex="-1"></a><span class="co"># $ HPAV         : int  5 5 35 12 4 12 6 3 1 7 ...</span></span>
<span id="cb467-31"><a href="basic-graphics.html#cb467-31" aria-hidden="true" tabindex="-1"></a><span class="co"># $ RARD         : int  3 3 9 13 12 0 0 2 2 0 ...</span></span>
<span id="cb467-32"><a href="basic-graphics.html#cb467-32" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SSTR         : int  2 0 0 0 0 2 0 0 0 0 ...</span></span>
<span id="cb467-33"><a href="basic-graphics.html#cb467-33" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Protopl      : int  1 0 2 0 0 0 0 0 0 0 ...</span></span>
<span id="cb467-34"><a href="basic-graphics.html#cb467-34" aria-hidden="true" tabindex="-1"></a><span class="co"># ... </span></span></code></pre></div>
<p>We can explore the relationship between <code>density</code> and <code>total_mites</code> with a simple scatterplot.</p>
<div class="sourceCode" id="cb468"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb468-1"><a href="basic-graphics.html#cb468-1" aria-hidden="true" tabindex="-1"></a><span class="do">## make sure that ggplot2 is loaded</span></span>
<span id="cb468-2"><a href="basic-graphics.html#cb468-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggplot2)</span>
<span id="cb468-3"><a href="basic-graphics.html#cb468-3" aria-hidden="true" tabindex="-1"></a><span class="do">## basic scatterplot</span></span>
<span id="cb468-4"><a href="basic-graphics.html#cb468-4" aria-hidden="true" tabindex="-1"></a>my_scatterplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> oribatid_data, <span class="fu">aes</span>(<span class="at">x =</span> water, <span class="at">y =</span> total_mites)) <span class="sc">+</span></span>
<span id="cb468-5"><a href="basic-graphics.html#cb468-5" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb468-6"><a href="basic-graphics.html#cb468-6" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Oribatid data -- simple scatterplot&quot;</span>, <span class="at">x=</span><span class="st">&quot;water content of the substratum (g/L)&quot;</span>, <span class="at">y=</span><span class="st">&quot;total mites&quot;</span>) <span class="sc">+</span></span>
<span id="cb468-7"><a href="basic-graphics.html#cb468-7" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">theme_minimal</span>()</span>
<span id="cb468-8"><a href="basic-graphics.html#cb468-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb468-9"><a href="basic-graphics.html#cb468-9" aria-hidden="true" tabindex="-1"></a>my_scatterplot</span></code></pre></div>
<p><img src="figures/session_10_figs/45_scatterplot_v1.png" width="90%" /></p>
<p>We see and outlier, so we can label the point by the sites that each point represents by adding argument <code>label</code> and <code>geom_text</code></p>
<div class="sourceCode" id="cb469"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb469-1"><a href="basic-graphics.html#cb469-1" aria-hidden="true" tabindex="-1"></a><span class="do">## labeled scatterplot</span></span>
<span id="cb469-2"><a href="basic-graphics.html#cb469-2" aria-hidden="true" tabindex="-1"></a>my_scatterplot_l <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> oribatid_data, <span class="fu">aes</span>(<span class="at">x =</span> water, <span class="at">y =</span> total_mites, <span class="at">label=</span>site)) <span class="sc">+</span></span>
<span id="cb469-3"><a href="basic-graphics.html#cb469-3" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb469-4"><a href="basic-graphics.html#cb469-4" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Oribatid data -- simple scatterplot labeled&quot;</span>, <span class="at">x=</span><span class="st">&quot;water content of the substratum (g/L)&quot;</span>, <span class="at">y=</span><span class="st">&quot;total mites&quot;</span>) <span class="sc">+</span></span>
<span id="cb469-5"><a href="basic-graphics.html#cb469-5" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label=</span>site),<span class="at">hjust=</span><span class="dv">0</span>, <span class="at">vjust=</span><span class="sc">-</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb469-6"><a href="basic-graphics.html#cb469-6" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">theme_minimal</span>()</span>
<span id="cb469-7"><a href="basic-graphics.html#cb469-7" aria-hidden="true" tabindex="-1"></a>my_scatterplot_l</span></code></pre></div>
<p><img src="figures/session_10_figs/46_scatterplot_labeled.png" width="90%" /></p>
<p>We identify that point or site <code>67</code> as an outlier. We can determine why this site is an outlier, yet we can exclude it in our dataset.</p>
<div class="sourceCode" id="cb470"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb470-1"><a href="basic-graphics.html#cb470-1" aria-hidden="true" tabindex="-1"></a><span class="do">## search the site &quot;67&quot;</span></span>
<span id="cb470-2"><a href="basic-graphics.html#cb470-2" aria-hidden="true" tabindex="-1"></a>oribatid_data[oribatid_data<span class="sc">$</span>site <span class="sc">==</span> <span class="st">&quot;67&quot;</span>, ]</span>
<span id="cb470-3"><a href="basic-graphics.html#cb470-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   site substrate shrubs    topo density   water Brachy PHTH HPAV RARD SSTR Protopl MEGR MPRO TVIE HMIN HMIN2 NPRA TVEL ONOV SUCT LCIL</span></span>
<span id="cb470-4"><a href="basic-graphics.html#cb470-4" aria-hidden="true" tabindex="-1"></a><span class="co">#64   67      sph1   none blanket  52.123 826.958      4    0    3    0    0       0    0    0    0    0     0    0    0    0    0  723</span></span>
<span id="cb470-5"><a href="basic-graphics.html#cb470-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   Oribatul1 Ceratoz1 PWIL Galumna1 Steganacarus2 HRUF Trhypochth1 PPEL NCOR SLAT FSET Lepidozetes Eupelops Minigalumna LRUG PLAG2 Ceratoz3</span></span>
<span id="cb470-6"><a href="basic-graphics.html#cb470-6" aria-hidden="true" tabindex="-1"></a><span class="co">#64         0        0    0        0             0    0           7    0    0    0    0           0        0           0   11     0        0</span></span>
<span id="cb470-7"><a href="basic-graphics.html#cb470-7" aria-hidden="true" tabindex="-1"></a><span class="co">#   Oppia.minus Trimalaco2 total_mites</span></span>
<span id="cb470-8"><a href="basic-graphics.html#cb470-8" aria-hidden="true" tabindex="-1"></a><span class="co">#64           0         33         781</span></span>
<span id="cb470-9"><a href="basic-graphics.html#cb470-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb470-10"><a href="basic-graphics.html#cb470-10" aria-hidden="true" tabindex="-1"></a><span class="do">## this site has an unusual high numbers of LCIL mites. We can delete or remove this row by its number indicated by 64 at the row number</span></span>
<span id="cb470-11"><a href="basic-graphics.html#cb470-11" aria-hidden="true" tabindex="-1"></a>oribatid_data_2 <span class="ot">&lt;-</span> oribatid_data[<span class="sc">-</span><span class="dv">64</span>,]</span>
<span id="cb470-12"><a href="basic-graphics.html#cb470-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb470-13"><a href="basic-graphics.html#cb470-13" aria-hidden="true" tabindex="-1"></a><span class="do">## labeled scatterplot</span></span>
<span id="cb470-14"><a href="basic-graphics.html#cb470-14" aria-hidden="true" tabindex="-1"></a>my_scatterplot_l2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> oribatid_data_2, <span class="fu">aes</span>(<span class="at">x =</span> water, <span class="at">y =</span> total_mites, <span class="at">label=</span>site)) <span class="sc">+</span></span>
<span id="cb470-15"><a href="basic-graphics.html#cb470-15" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb470-16"><a href="basic-graphics.html#cb470-16" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Oribatid data -- simple scatterplot labeled&quot;</span>, <span class="at">x=</span><span class="st">&quot;water content of the substratum (g/L)&quot;</span>, <span class="at">y=</span><span class="st">&quot;total mites&quot;</span>) <span class="sc">+</span></span>
<span id="cb470-17"><a href="basic-graphics.html#cb470-17" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label=</span>site),<span class="at">hjust=</span><span class="dv">0</span>, <span class="at">vjust=</span><span class="sc">-</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb470-18"><a href="basic-graphics.html#cb470-18" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">theme_minimal</span>()</span>
<span id="cb470-19"><a href="basic-graphics.html#cb470-19" aria-hidden="true" tabindex="-1"></a>my_scatterplot_l2</span></code></pre></div>
<p><img src="figures/session_10_figs/47_scatterplot_labeled_v2.png" width="90%" /></p>
<p><strong>26)</strong> We can modify this scatterplot with different shapes, colors and sizes. These increase the amount of information provided by the plot.</p>
<p>We can change the shape and color of the points in the scatterplot.</p>
<div class="sourceCode" id="cb471"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb471-1"><a href="basic-graphics.html#cb471-1" aria-hidden="true" tabindex="-1"></a><span class="do">## load color library &#39;ggsci&#39;</span></span>
<span id="cb471-2"><a href="basic-graphics.html#cb471-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggsci)</span>
<span id="cb471-3"><a href="basic-graphics.html#cb471-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb471-4"><a href="basic-graphics.html#cb471-4" aria-hidden="true" tabindex="-1"></a><span class="do">## scatterplot with different shape and color palette</span></span>
<span id="cb471-5"><a href="basic-graphics.html#cb471-5" aria-hidden="true" tabindex="-1"></a>my_scatterplot_l3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> oribatid_data_2, <span class="fu">aes</span>(<span class="at">x =</span> water, <span class="at">y =</span> total_mites, <span class="at">label=</span>site)) <span class="sc">+</span></span>
<span id="cb471-6"><a href="basic-graphics.html#cb471-6" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> substrate), <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">size =</span> <span class="dv">5</span>, <span class="at">shape =</span> <span class="dv">18</span>) <span class="sc">+</span></span>
<span id="cb471-7"><a href="basic-graphics.html#cb471-7" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Oribatid data -- scatterplot by substrate -- jco palette &quot;</span>, <span class="at">x=</span><span class="st">&quot;water content of the substratum (g/L)&quot;</span>, <span class="at">y=</span><span class="st">&quot;total mites&quot;</span>) <span class="sc">+</span></span>
<span id="cb471-8"><a href="basic-graphics.html#cb471-8" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">scale_color_jco</span>() <span class="sc">+</span></span>
<span id="cb471-9"><a href="basic-graphics.html#cb471-9" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">theme_minimal</span>()</span>
<span id="cb471-10"><a href="basic-graphics.html#cb471-10" aria-hidden="true" tabindex="-1"></a>my_scatterplot_l3</span></code></pre></div>
<p><img src="figures/session_10_figs/48_scatterplot_color_v2.png" width="90%" /></p>
<p>We can change the size of points in the scatterplot based on another continuous variable. For example, number of <code>LCIL</code> mite species.</p>
<div class="sourceCode" id="cb472"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb472-1"><a href="basic-graphics.html#cb472-1" aria-hidden="true" tabindex="-1"></a><span class="do">## scatterplot with size of points by LCIL numbers </span></span>
<span id="cb472-2"><a href="basic-graphics.html#cb472-2" aria-hidden="true" tabindex="-1"></a>my_scatterplot_l4 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> oribatid_data_2, <span class="fu">aes</span>(<span class="at">x =</span> water, <span class="at">y =</span> total_mites, <span class="at">label=</span>site)) <span class="sc">+</span></span>
<span id="cb472-3"><a href="basic-graphics.html#cb472-3" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> substrate, <span class="at">size =</span> LCIL), <span class="at">alpha =</span> <span class="fl">0.9</span>, <span class="at">shape =</span> <span class="dv">18</span>) <span class="sc">+</span></span>
<span id="cb472-4"><a href="basic-graphics.html#cb472-4" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Oribatid data -- scatterplot by substrate and size by LCIL mite species -- npg palette &quot;</span>, <span class="at">x=</span><span class="st">&quot;water content of the substratum (g/L)&quot;</span>, <span class="at">y=</span><span class="st">&quot;total mites&quot;</span>) <span class="sc">+</span></span>
<span id="cb472-5"><a href="basic-graphics.html#cb472-5" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">scale_color_npg</span>() <span class="sc">+</span></span>
<span id="cb472-6"><a href="basic-graphics.html#cb472-6" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">theme_minimal</span>()</span>
<span id="cb472-7"><a href="basic-graphics.html#cb472-7" aria-hidden="true" tabindex="-1"></a>my_scatterplot_l4</span></code></pre></div>
<p><img src="figures/session_10_figs/49_scatterplot_color_size_v3.png" width="90%" /></p>
<p><strong>27)</strong> We might be interested in separating a group from the rest of data points in this scatterplot. We will try the <code>sph1</code> different from the others. We can change manually the points colors with <code>scale_colour_manual()</code> and shapes with <code>scale_colour_manual()</code>.</p>
<div class="sourceCode" id="cb473"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb473-1"><a href="basic-graphics.html#cb473-1" aria-hidden="true" tabindex="-1"></a><span class="do">## create a vector for manual colors bases on substrate</span></span>
<span id="cb473-2"><a href="basic-graphics.html#cb473-2" aria-hidden="true" tabindex="-1"></a>names_substrates <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">unique</span>(oribatid_data_2<span class="sc">$</span>substrate))</span>
<span id="cb473-3"><a href="basic-graphics.html#cb473-3" aria-hidden="true" tabindex="-1"></a>names_substrates</span>
<span id="cb473-4"><a href="basic-graphics.html#cb473-4" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;sph1&quot;   &quot;sph3&quot;   &quot;inter&quot;  &quot;litter&quot; &quot;sph4&quot;   &quot;sph2&quot;   &quot;peat&quot;</span></span>
<span id="cb473-5"><a href="basic-graphics.html#cb473-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb473-6"><a href="basic-graphics.html#cb473-6" aria-hidden="true" tabindex="-1"></a><span class="do">## create a vector for colors by substrate type, but binary</span></span>
<span id="cb473-7"><a href="basic-graphics.html#cb473-7" aria-hidden="true" tabindex="-1"></a>colors_substrate <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(names_substrates <span class="sc">==</span> <span class="st">&quot;sph1&quot;</span>, <span class="st">&quot;firebrick1&quot;</span>, <span class="st">&quot;darkgrey&quot;</span>)</span>
<span id="cb473-8"><a href="basic-graphics.html#cb473-8" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(colors_substrate) <span class="ot">&lt;-</span> names_substrates</span>
<span id="cb473-9"><a href="basic-graphics.html#cb473-9" aria-hidden="true" tabindex="-1"></a>colors_substrate</span>
<span id="cb473-10"><a href="basic-graphics.html#cb473-10" aria-hidden="true" tabindex="-1"></a><span class="co">#      sph1       sph3      inter     litter       sph4       sph2       peat </span></span>
<span id="cb473-11"><a href="basic-graphics.html#cb473-11" aria-hidden="true" tabindex="-1"></a><span class="co"># &quot;darkred&quot; &quot;darkgrey&quot; &quot;darkgrey&quot; &quot;darkgrey&quot; &quot;darkgrey&quot; &quot;darkgrey&quot; &quot;darkgrey&quot; </span></span>
<span id="cb473-12"><a href="basic-graphics.html#cb473-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb473-13"><a href="basic-graphics.html#cb473-13" aria-hidden="true" tabindex="-1"></a><span class="do">## scatterplot for &#39;sph1&#39; substrate only color</span></span>
<span id="cb473-14"><a href="basic-graphics.html#cb473-14" aria-hidden="true" tabindex="-1"></a>my_scatterplot_l4 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> oribatid_data_2, <span class="fu">aes</span>(<span class="at">x =</span> water, <span class="at">y =</span> total_mites)) <span class="sc">+</span></span>
<span id="cb473-15"><a href="basic-graphics.html#cb473-15" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> substrate), <span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">shape =</span> <span class="dv">18</span>, <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb473-16"><a href="basic-graphics.html#cb473-16" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Oribatid data -- scatterplot with emphasis on &#39;sph1&#39; substrate &quot;</span>, <span class="at">x=</span><span class="st">&quot;water content of the substratum (g/L)&quot;</span>, <span class="at">y=</span><span class="st">&quot;total mites&quot;</span>) <span class="sc">+</span></span>
<span id="cb473-17"><a href="basic-graphics.html#cb473-17" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> colors_substrate) <span class="sc">+</span></span>
<span id="cb473-18"><a href="basic-graphics.html#cb473-18" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">theme_minimal</span>()</span>
<span id="cb473-19"><a href="basic-graphics.html#cb473-19" aria-hidden="true" tabindex="-1"></a>my_scatterplot_l4</span></code></pre></div>
<p><img src="figures/session_10_figs/50_scatterplot_color_substrate_v4.png" width="90%" /></p>
<div class="sourceCode" id="cb474"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb474-1"><a href="basic-graphics.html#cb474-1" aria-hidden="true" tabindex="-1"></a><span class="do">## create a vector for shapes by substrate type, but binary</span></span>
<span id="cb474-2"><a href="basic-graphics.html#cb474-2" aria-hidden="true" tabindex="-1"></a>shape_substrate <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(names_substrates <span class="sc">==</span> <span class="st">&quot;sph1&quot;</span>, <span class="dv">16</span>, <span class="dv">18</span>)</span>
<span id="cb474-3"><a href="basic-graphics.html#cb474-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(shape_substrate) <span class="ot">&lt;-</span> names_substrates</span>
<span id="cb474-4"><a href="basic-graphics.html#cb474-4" aria-hidden="true" tabindex="-1"></a>shape_substrate</span>
<span id="cb474-5"><a href="basic-graphics.html#cb474-5" aria-hidden="true" tabindex="-1"></a><span class="co">#   sph1   sph3  inter litter   sph4   sph2   peat </span></span>
<span id="cb474-6"><a href="basic-graphics.html#cb474-6" aria-hidden="true" tabindex="-1"></a><span class="co">#    16     18     18     18     18     18     18</span></span>
<span id="cb474-7"><a href="basic-graphics.html#cb474-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-8"><a href="basic-graphics.html#cb474-8" aria-hidden="true" tabindex="-1"></a><span class="do">## scatterplot for &#39;sph1&#39; substrate with color and shape</span></span>
<span id="cb474-9"><a href="basic-graphics.html#cb474-9" aria-hidden="true" tabindex="-1"></a>my_scatterplot_l5 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> oribatid_data_2, <span class="fu">aes</span>(<span class="at">x =</span> water, <span class="at">y =</span> total_mites)) <span class="sc">+</span></span>
<span id="cb474-10"><a href="basic-graphics.html#cb474-10" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> substrate , <span class="at">shape =</span> substrate), <span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb474-11"><a href="basic-graphics.html#cb474-11" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Oribatid data -- scatterplot with emphasis on &#39;sph1&#39; substrate &quot;</span>, <span class="at">x=</span><span class="st">&quot;water content of the substratum (g/L)&quot;</span>, <span class="at">y=</span><span class="st">&quot;total mites&quot;</span>) <span class="sc">+</span></span>
<span id="cb474-12"><a href="basic-graphics.html#cb474-12" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> colors_substrate) <span class="sc">+</span></span>
<span id="cb474-13"><a href="basic-graphics.html#cb474-13" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">scale_shape_manual</span>(<span class="at">values =</span> shape_substrate) <span class="sc">+</span> </span>
<span id="cb474-14"><a href="basic-graphics.html#cb474-14" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">theme_minimal</span>()</span>
<span id="cb474-15"><a href="basic-graphics.html#cb474-15" aria-hidden="true" tabindex="-1"></a>my_scatterplot_l5</span></code></pre></div>
<p><img src="figures/session_10_figs/51_scatterplot_color_shape_substrate_v5.png" width="90%" /></p>
</div>
<div id="lines-and-scatterplots" class="section level2" number="10.10">
<h2><span class="header-section-number">10.10</span> Lines and Scatterplots</h2>
<p>In most cases, you would be interested on plotting trends on your scatterplots and these can be easily added with functions like <code>geom_line()</code> and <code>geom_sooth()</code>.</p>
<p><strong>28)</strong> For lines, we will use again the <code>oribatid</code> dataset. As mentioned, this data set contains information about environment and diversity in communities of Oribatid mites.</p>
<p>We can prepare <code>oribatid</code> dataset for the scatterplot.</p>
<div class="sourceCode" id="cb475"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb475-1"><a href="basic-graphics.html#cb475-1" aria-hidden="true" tabindex="-1"></a><span class="do">## check is ade4 is loaded and get &#39;oribatid&#39; database</span></span>
<span id="cb475-2"><a href="basic-graphics.html#cb475-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ade4)</span>
<span id="cb475-3"><a href="basic-graphics.html#cb475-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(oribatid)</span>
<span id="cb475-4"><a href="basic-graphics.html#cb475-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb475-5"><a href="basic-graphics.html#cb475-5" aria-hidden="true" tabindex="-1"></a><span class="do">## enviromental and genetic data</span></span>
<span id="cb475-6"><a href="basic-graphics.html#cb475-6" aria-hidden="true" tabindex="-1"></a>envir_data <span class="ot">&lt;-</span> oribatid<span class="sc">$</span>envir</span>
<span id="cb475-7"><a href="basic-graphics.html#cb475-7" aria-hidden="true" tabindex="-1"></a>fauna_data <span class="ot">&lt;-</span> oribatid<span class="sc">$</span>fau</span>
<span id="cb475-8"><a href="basic-graphics.html#cb475-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb475-9"><a href="basic-graphics.html#cb475-9" aria-hidden="true" tabindex="-1"></a><span class="do">## total number of oribatid mites</span></span>
<span id="cb475-10"><a href="basic-graphics.html#cb475-10" aria-hidden="true" tabindex="-1"></a>fauna_data<span class="sc">$</span>total_mites <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(fauna_data)</span>
<span id="cb475-11"><a href="basic-graphics.html#cb475-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb475-12"><a href="basic-graphics.html#cb475-12" aria-hidden="true" tabindex="-1"></a><span class="do">## we add a common column in both data frames, in this case the &#39;site&#39;</span></span>
<span id="cb475-13"><a href="basic-graphics.html#cb475-13" aria-hidden="true" tabindex="-1"></a>envir_data<span class="sc">$</span>site <span class="ot">&lt;-</span> <span class="fu">rownames</span>(envir_data)</span>
<span id="cb475-14"><a href="basic-graphics.html#cb475-14" aria-hidden="true" tabindex="-1"></a>fauna_data<span class="sc">$</span>site <span class="ot">&lt;-</span> <span class="fu">rownames</span>(fauna_data)</span>
<span id="cb475-15"><a href="basic-graphics.html#cb475-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb475-16"><a href="basic-graphics.html#cb475-16" aria-hidden="true" tabindex="-1"></a><span class="do">## merge both data frames using &#39;site&#39;</span></span>
<span id="cb475-17"><a href="basic-graphics.html#cb475-17" aria-hidden="true" tabindex="-1"></a>oribatid_data <span class="ot">&lt;-</span> <span class="fu">merge</span>(envir_data, fauna_data, <span class="at">by =</span><span class="st">&quot;site&quot;</span>)</span>
<span id="cb475-18"><a href="basic-graphics.html#cb475-18" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(oribatid_data)</span>
<span id="cb475-19"><a href="basic-graphics.html#cb475-19" aria-hidden="true" tabindex="-1"></a><span class="co"># you can explore this data frame content</span></span>
<span id="cb475-20"><a href="basic-graphics.html#cb475-20" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(oribatid_data)</span>
<span id="cb475-21"><a href="basic-graphics.html#cb475-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  70 obs. of  41 variables:</span></span>
<span id="cb475-22"><a href="basic-graphics.html#cb475-22" aria-hidden="true" tabindex="-1"></a><span class="co"># $ site         : chr  &quot;1&quot; &quot;10&quot; &quot;11&quot; &quot;12&quot; ...</span></span>
<span id="cb475-23"><a href="basic-graphics.html#cb475-23" aria-hidden="true" tabindex="-1"></a><span class="co"># $ substrate    : Factor w/ 7 levels &quot;inter&quot;,&quot;litter&quot;,..: 4 4 6 4 4 1 1 1 1 4 ...</span></span>
<span id="cb475-24"><a href="basic-graphics.html#cb475-24" aria-hidden="true" tabindex="-1"></a><span class="co"># $ shrubs       : Factor w/ 3 levels &quot;few&quot;,&quot;many&quot;,&quot;none&quot;: 1 2 2 1 2 2 2 1 2 2 ...</span></span>
<span id="cb475-25"><a href="basic-graphics.html#cb475-25" aria-hidden="true" tabindex="-1"></a><span class="co"># $ topo         : Factor w/ 2 levels &quot;blanket&quot;,&quot;hummock&quot;: 2 2 1 2 2 1 1 2 2 2 ...</span></span>
<span id="cb475-26"><a href="basic-graphics.html#cb475-26" aria-hidden="true" tabindex="-1"></a><span class="co"># $ density      : num  39.2 32.1 35.6 46.8 28 ...</span></span>
<span id="cb475-27"><a href="basic-graphics.html#cb475-27" aria-hidden="true" tabindex="-1"></a><span class="co"># $ water        : num  350 221 134 406 244 ...</span></span>
<span id="cb475-28"><a href="basic-graphics.html#cb475-28" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Brachy       : int  17 22 36 28 3 41 6 7 9 19 ...</span></span>
<span id="cb475-29"><a href="basic-graphics.html#cb475-29" aria-hidden="true" tabindex="-1"></a><span class="co"># $ PHTH         : int  5 4 7 2 2 5 0 2 0 3 ...</span></span>
<span id="cb475-30"><a href="basic-graphics.html#cb475-30" aria-hidden="true" tabindex="-1"></a><span class="co"># $ HPAV         : int  5 5 35 12 4 12 6 3 1 7 ...</span></span>
<span id="cb475-31"><a href="basic-graphics.html#cb475-31" aria-hidden="true" tabindex="-1"></a><span class="co"># $ RARD         : int  3 3 9 13 12 0 0 2 2 0 ...</span></span>
<span id="cb475-32"><a href="basic-graphics.html#cb475-32" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SSTR         : int  2 0 0 0 0 2 0 0 0 0 ...</span></span>
<span id="cb475-33"><a href="basic-graphics.html#cb475-33" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Protopl      : int  1 0 2 0 0 0 0 0 0 0 ...</span></span>
<span id="cb475-34"><a href="basic-graphics.html#cb475-34" aria-hidden="true" tabindex="-1"></a><span class="co"># ... </span></span>
<span id="cb475-35"><a href="basic-graphics.html#cb475-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb475-36"><a href="basic-graphics.html#cb475-36" aria-hidden="true" tabindex="-1"></a><span class="do">## as mentioned before, we will remove unusual site 67</span></span>
<span id="cb475-37"><a href="basic-graphics.html#cb475-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb475-38"><a href="basic-graphics.html#cb475-38" aria-hidden="true" tabindex="-1"></a>oribatid_data_2 <span class="ot">&lt;-</span> oribatid_data[<span class="sc">-</span><span class="dv">64</span>,]</span></code></pre></div>
<p>We can explore the relationship between <code>water</code> and <code>total_mites</code> by <code>shrubs</code> groups with <code>geom_line()</code>. Notice that you can change groups based on the argument <code>group</code>.</p>
<div class="sourceCode" id="cb476"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb476-1"><a href="basic-graphics.html#cb476-1" aria-hidden="true" tabindex="-1"></a><span class="do">## make sure that ggplot2 is loaded</span></span>
<span id="cb476-2"><a href="basic-graphics.html#cb476-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggplot2)</span>
<span id="cb476-3"><a href="basic-graphics.html#cb476-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-4"><a href="basic-graphics.html#cb476-4" aria-hidden="true" tabindex="-1"></a><span class="do">## add a line for each shrubs group</span></span>
<span id="cb476-5"><a href="basic-graphics.html#cb476-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-6"><a href="basic-graphics.html#cb476-6" aria-hidden="true" tabindex="-1"></a>substrate_plots <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> oribatid_data_2, <span class="fu">aes</span>(<span class="at">x =</span> water, <span class="at">y =</span> total_mites, <span class="at">group =</span> shrubs)) <span class="sc">+</span></span>
<span id="cb476-7"><a href="basic-graphics.html#cb476-7" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">color =</span> shrubs))<span class="sc">+</span></span>
<span id="cb476-8"><a href="basic-graphics.html#cb476-8" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> shrubs), <span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb476-9"><a href="basic-graphics.html#cb476-9" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Oribatid data -- scatterplot with lines by shrub density&quot;</span>, <span class="at">x=</span><span class="st">&quot;water content of the substratum (g/L)&quot;</span>, <span class="at">y=</span><span class="st">&quot;total mites&quot;</span>) <span class="sc">+</span></span>
<span id="cb476-10"><a href="basic-graphics.html#cb476-10" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">theme_minimal</span>()</span>
<span id="cb476-11"><a href="basic-graphics.html#cb476-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-12"><a href="basic-graphics.html#cb476-12" aria-hidden="true" tabindex="-1"></a>substrate_plots</span></code></pre></div>
<p><img src="figures/session_10_figs/56_scatterplot_shurb_v1.png" width="90%" /></p>
<p>You can also change the type of line and color manually as long as you provide a vector with the same number of groups using <code>scale_linetype_manual()</code> and <code>scale_color_manual()</code>. For our example, <code>group = shrubs</code> has to have vector of 3 elements.</p>
<div class="sourceCode" id="cb477"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb477-1"><a href="basic-graphics.html#cb477-1" aria-hidden="true" tabindex="-1"></a><span class="do">## change line type manually for each shrubs group</span></span>
<span id="cb477-2"><a href="basic-graphics.html#cb477-2" aria-hidden="true" tabindex="-1"></a>substrate_plots <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> oribatid_data_2, <span class="fu">aes</span>(<span class="at">x =</span> water, <span class="at">y =</span> total_mites, <span class="at">group =</span> shrubs)) <span class="sc">+</span></span>
<span id="cb477-3"><a href="basic-graphics.html#cb477-3" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">color =</span> shrubs, <span class="at">linetype =</span> shrubs))<span class="sc">+</span></span>
<span id="cb477-4"><a href="basic-graphics.html#cb477-4" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> shrubs), <span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb477-5"><a href="basic-graphics.html#cb477-5" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Oribatid data -- scatterplot with lines by shrub density&quot;</span>, <span class="at">x=</span><span class="st">&quot;water content of the substratum (g/L)&quot;</span>, <span class="at">y=</span><span class="st">&quot;total mites&quot;</span>) <span class="sc">+</span></span>
<span id="cb477-6"><a href="basic-graphics.html#cb477-6" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">theme_minimal</span>()</span>
<span id="cb477-7"><a href="basic-graphics.html#cb477-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb477-8"><a href="basic-graphics.html#cb477-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb477-9"><a href="basic-graphics.html#cb477-9" aria-hidden="true" tabindex="-1"></a>substrate_plots <span class="ot">&lt;-</span> substrate_plots <span class="sc">+</span> <span class="fu">scale_linetype_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">&quot;solid&quot;</span>, <span class="st">&quot;dashed&quot;</span>, <span class="st">&quot;solid&quot;</span>))</span>
<span id="cb477-10"><a href="basic-graphics.html#cb477-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb477-11"><a href="basic-graphics.html#cb477-11" aria-hidden="true" tabindex="-1"></a>substrate_plots</span></code></pre></div>
<p><img src="figures/session_10_figs/57_scatterplot_shurb_v2.png" width="90%" />
<strong>29)</strong> We an explore a line and adding a confidence region to a line. For this example, we will use the NYC Central Park temperature record from <a href="https://www.weather.gov/okx/CentralParkHistorical" target="_blank"><strong>here</strong></a>. This <code>nyc_temps.txt</code> dataset is also in our class <a href="https://github.com/corytophanes/BIO_BIT_Bioinformatics_209/tree/main/ref_files_pdfs" target="_blank"><strong>GitHub repository</strong></a>.</p>
<div class="sourceCode" id="cb478"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb478-1"><a href="basic-graphics.html#cb478-1" aria-hidden="true" tabindex="-1"></a><span class="do">## </span><span class="al">NOTE</span><span class="do">: remember to update the path to file with the dataset where you downloaded in your computer -- THIS IS EXCLUSIVE TO YOUR COMPUTER AND IT IS NOT THE PATH SHOWN BELOW</span></span>
<span id="cb478-2"><a href="basic-graphics.html#cb478-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-3"><a href="basic-graphics.html#cb478-3" aria-hidden="true" tabindex="-1"></a><span class="do">## load get &#39;nyc_temps.txt&#39; dataset </span></span>
<span id="cb478-4"><a href="basic-graphics.html#cb478-4" aria-hidden="true" tabindex="-1"></a>nyc_temps <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/ref_files_pdfs/nyc_temps.txt&quot;</span>,</span>
<span id="cb478-5"><a href="basic-graphics.html#cb478-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb478-6"><a href="basic-graphics.html#cb478-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(nyc_temps)</span>
<span id="cb478-7"><a href="basic-graphics.html#cb478-7" aria-hidden="true" tabindex="-1"></a><span class="co">#  YEAR  JAN  FEB  MAR  APR  MAY  JUN  JUL  AUG  SEP  OCT  NOV  DEC ANNUAL</span></span>
<span id="cb478-8"><a href="basic-graphics.html#cb478-8" aria-hidden="true" tabindex="-1"></a><span class="co">#1 1869 35.1 34.5 34.8 49.2 57.7 69.3 72.8 71.8 65.6 50.9 40.3 34.7   51.4</span></span>
<span id="cb478-9"><a href="basic-graphics.html#cb478-9" aria-hidden="true" tabindex="-1"></a><span class="co">#2 1870 37.5 31.3 34.1 50.7 60.9 72.9 76.6 75.3 67.6 56.7 45.5 34.1   53.6</span></span>
<span id="cb478-10"><a href="basic-graphics.html#cb478-10" aria-hidden="true" tabindex="-1"></a><span class="co">#3 1871 28.3 30.2 44.2 52.0 60.4 68.2 72.3 73.6 60.8 55.6 38.8 29.2   51.1</span></span>
<span id="cb478-11"><a href="basic-graphics.html#cb478-11" aria-hidden="true" tabindex="-1"></a><span class="co">#4 1872 28.8 29.9 30.5 49.4 61.5 71.2 77.5 75.6 66.4 53.2 41.0 26.7   51.0</span></span>
<span id="cb478-12"><a href="basic-graphics.html#cb478-12" aria-hidden="true" tabindex="-1"></a><span class="co">#5 1873 28.6 29.5 35.7 46.7 58.8 70.3 75.4 72.0 65.4 55.8 37.0 36.5   51.0</span></span>
<span id="cb478-13"><a href="basic-graphics.html#cb478-13" aria-hidden="true" tabindex="-1"></a><span class="co">#6 1874 34.2 31.3 37.1 41.1 58.8 70.1 73.9 70.3 67.0 55.1 43.4 33.8   51.3</span></span>
<span id="cb478-14"><a href="basic-graphics.html#cb478-14" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(nyc_temps)</span>
<span id="cb478-15"><a href="basic-graphics.html#cb478-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;: 151 obs. of  14 variables:</span></span>
<span id="cb478-16"><a href="basic-graphics.html#cb478-16" aria-hidden="true" tabindex="-1"></a><span class="co"># $ YEAR  : int  1869 1870 1871 1872 1873 1874 1875 1876 1877 1878 ...</span></span>
<span id="cb478-17"><a href="basic-graphics.html#cb478-17" aria-hidden="true" tabindex="-1"></a><span class="co"># $ JAN   : num  35.1 37.5 28.3 28.8 28.6 34.2 23.8 36.6 27.7 30.3 ...</span></span>
<span id="cb478-18"><a href="basic-graphics.html#cb478-18" aria-hidden="true" tabindex="-1"></a><span class="co"># $ FEB   : num  34.5 31.3 30.2 29.9 29.5 31.3 25.2 31.8 37 32.2 ...</span></span>
<span id="cb478-19"><a href="basic-graphics.html#cb478-19" aria-hidden="true" tabindex="-1"></a><span class="co"># $ MAR   : num  34.8 34.1 44.2 30.5 35.7 37.1 34.1 34.4 35.8 44.1 ...</span></span>
<span id="cb478-20"><a href="basic-graphics.html#cb478-20" aria-hidden="true" tabindex="-1"></a><span class="co"># $ APR   : num  49.2 50.7 52 49.4 46.7 41.1 43.1 47 47.7 53.3 ...</span></span>
<span id="cb478-21"><a href="basic-graphics.html#cb478-21" aria-hidden="true" tabindex="-1"></a><span class="co"># $ MAY   : num  57.7 60.9 60.4 61.5 58.8 58.8 60.1 60.2 59.6 59.4 ...</span></span>
<span id="cb478-22"><a href="basic-graphics.html#cb478-22" aria-hidden="true" tabindex="-1"></a><span class="co"># $ JUN   : num  69.3 72.9 68.2 71.2 70.3 70.1 69.2 73.5 70.2 67.7 ...</span></span>
<span id="cb478-23"><a href="basic-graphics.html#cb478-23" aria-hidden="true" tabindex="-1"></a><span class="co"># $ JUL   : num  72.8 76.6 72.3 77.5 75.4 73.9 74 79.4 75 77.8 ...</span></span>
<span id="cb478-24"><a href="basic-graphics.html#cb478-24" aria-hidden="true" tabindex="-1"></a><span class="co"># $ AUG   : num  71.8 75.3 73.6 75.6 72 70.3 72.9 75.2 75.4 74.2 ...</span></span>
<span id="cb478-25"><a href="basic-graphics.html#cb478-25" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SEP   : num  65.6 67.6 60.8 66.4 65.4 67 64 63.7 66.9 68.3 ...</span></span>
<span id="cb478-26"><a href="basic-graphics.html#cb478-26" aria-hidden="true" tabindex="-1"></a><span class="co"># $ OCT   : num  50.9 56.7 55.6 53.2 55.8 55.1 53.6 50.6 55.8 58.7 ...</span></span>
<span id="cb478-27"><a href="basic-graphics.html#cb478-27" aria-hidden="true" tabindex="-1"></a><span class="co"># $ NOV   : num  40.3 45.5 38.8 41 37 43.4 39.3 45.2 44.5 43.8 ...</span></span>
<span id="cb478-28"><a href="basic-graphics.html#cb478-28" aria-hidden="true" tabindex="-1"></a><span class="co"># $ DEC   : num  34.7 34.1 29.2 26.7 36.5 33.8 33.9 24.9 37.4 32.8 ...</span></span>
<span id="cb478-29"><a href="basic-graphics.html#cb478-29" aria-hidden="true" tabindex="-1"></a><span class="co"># $ ANNUAL: num  51.4 53.6 51.1 51 51 51.3 49.4 51.9 52.8 53.6 ...</span></span>
<span id="cb478-30"><a href="basic-graphics.html#cb478-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-31"><a href="basic-graphics.html#cb478-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-32"><a href="basic-graphics.html#cb478-32" aria-hidden="true" tabindex="-1"></a><span class="do">## we need to create a confidence level by determining the max and min temperature values per year</span></span>
<span id="cb478-33"><a href="basic-graphics.html#cb478-33" aria-hidden="true" tabindex="-1"></a><span class="do">## </span><span class="al">NOTICE</span><span class="do"> that we need to exclude the YEAR and ANNUAL columns</span></span>
<span id="cb478-34"><a href="basic-graphics.html#cb478-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-35"><a href="basic-graphics.html#cb478-35" aria-hidden="true" tabindex="-1"></a><span class="do">## max</span></span>
<span id="cb478-36"><a href="basic-graphics.html#cb478-36" aria-hidden="true" tabindex="-1"></a>nyc_temps[, <span class="st">&quot;max_temp&quot;</span>] <span class="ot">&lt;-</span> <span class="fu">apply</span>(nyc_temps[, <span class="dv">2</span><span class="sc">:</span><span class="dv">13</span>], <span class="dv">1</span>, max)</span>
<span id="cb478-37"><a href="basic-graphics.html#cb478-37" aria-hidden="true" tabindex="-1"></a><span class="do">## min</span></span>
<span id="cb478-38"><a href="basic-graphics.html#cb478-38" aria-hidden="true" tabindex="-1"></a>nyc_temps[, <span class="st">&quot;min_temp&quot;</span>] <span class="ot">&lt;-</span> <span class="fu">apply</span>(nyc_temps[, <span class="dv">2</span><span class="sc">:</span><span class="dv">13</span>], <span class="dv">1</span>, min)</span>
<span id="cb478-39"><a href="basic-graphics.html#cb478-39" aria-hidden="true" tabindex="-1"></a><span class="do">## standard deviation</span></span>
<span id="cb478-40"><a href="basic-graphics.html#cb478-40" aria-hidden="true" tabindex="-1"></a>nyc_temps[, <span class="st">&quot;sd_temp&quot;</span>] <span class="ot">&lt;-</span> <span class="fu">apply</span>(nyc_temps[, <span class="dv">2</span><span class="sc">:</span><span class="dv">13</span>], <span class="dv">1</span>, sd)</span>
<span id="cb478-41"><a href="basic-graphics.html#cb478-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb478-42"><a href="basic-graphics.html#cb478-42" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(nyc_temps)</span>
<span id="cb478-43"><a href="basic-graphics.html#cb478-43" aria-hidden="true" tabindex="-1"></a><span class="co">#  YEAR  JAN  FEB  MAR  APR  MAY  JUN  JUL  AUG  SEP  OCT  NOV  DEC ANNUAL max_temp min_temp  sd_temp</span></span>
<span id="cb478-44"><a href="basic-graphics.html#cb478-44" aria-hidden="true" tabindex="-1"></a><span class="co">#1 1869 35.1 34.5 34.8 49.2 57.7 69.3 72.8 71.8 65.6 50.9 40.3 34.7   51.4     72.8     34.5 15.57944</span></span>
<span id="cb478-45"><a href="basic-graphics.html#cb478-45" aria-hidden="true" tabindex="-1"></a><span class="co">#2 1870 37.5 31.3 34.1 50.7 60.9 72.9 76.6 75.3 67.6 56.7 45.5 34.1   53.6     76.6     31.3 17.11538</span></span>
<span id="cb478-46"><a href="basic-graphics.html#cb478-46" aria-hidden="true" tabindex="-1"></a><span class="co">#3 1871 28.3 30.2 44.2 52.0 60.4 68.2 72.3 73.6 60.8 55.6 38.8 29.2   51.1     73.6     28.3 16.74647</span></span>
<span id="cb478-47"><a href="basic-graphics.html#cb478-47" aria-hidden="true" tabindex="-1"></a><span class="co">#4 1872 28.8 29.9 30.5 49.4 61.5 71.2 77.5 75.6 66.4 53.2 41.0 26.7   51.0     77.5     26.7 19.35093</span></span>
<span id="cb478-48"><a href="basic-graphics.html#cb478-48" aria-hidden="true" tabindex="-1"></a><span class="co">#5 1873 28.6 29.5 35.7 46.7 58.8 70.3 75.4 72.0 65.4 55.8 37.0 36.5   51.0     75.4     28.6 17.38056</span></span>
<span id="cb478-49"><a href="basic-graphics.html#cb478-49" aria-hidden="true" tabindex="-1"></a><span class="co">#6 1874 34.2 31.3 37.1 41.1 58.8 70.1 73.9 70.3 67.0 55.1 43.4 33.8   51.3     73.9     31.3 16.26236</span></span></code></pre></div>
<p>We now can plot the temperature trends in NYC until 2019. For the confidence plot, we use the function <code>geom_ribbon()</code>.</p>
<div class="sourceCode" id="cb479"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb479-1"><a href="basic-graphics.html#cb479-1" aria-hidden="true" tabindex="-1"></a><span class="do">## NYC temperatures plot</span></span>
<span id="cb479-2"><a href="basic-graphics.html#cb479-2" aria-hidden="true" tabindex="-1"></a>nyc_temp_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(nyc_temps, <span class="fu">aes</span>(<span class="at">x =</span> YEAR, <span class="at">y =</span> ANNUAL)) <span class="sc">+</span></span>
<span id="cb479-3"><a href="basic-graphics.html#cb479-3" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">geom_ribbon</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> ANNUAL <span class="sc">-</span> sd_temp, <span class="at">ymax =</span> ANNUAL <span class="sc">+</span> sd_temp), <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb479-4"><a href="basic-graphics.html#cb479-4" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> YEAR, <span class="at">y =</span> max_temp), <span class="at">colour =</span> <span class="st">&quot;red&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dotted&quot;</span>) <span class="sc">+</span></span>
<span id="cb479-5"><a href="basic-graphics.html#cb479-5" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> YEAR, <span class="at">y =</span> min_temp), <span class="at">colour =</span> <span class="st">&quot;blue&quot;</span>, <span class="at">linetype =</span> <span class="st">&quot;dotted&quot;</span>) <span class="sc">+</span></span>
<span id="cb479-6"><a href="basic-graphics.html#cb479-6" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">geom_line</span>() <span class="sc">+</span></span>
<span id="cb479-7"><a href="basic-graphics.html#cb479-7" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;NYC temperature Central Park&quot;</span>, <span class="at">x=</span><span class="st">&quot;Year&quot;</span>, <span class="at">y=</span><span class="st">&quot;Temp (F)&quot;</span>) <span class="sc">+</span></span>
<span id="cb479-8"><a href="basic-graphics.html#cb479-8" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">90</span>) <span class="sc">+</span></span>
<span id="cb479-9"><a href="basic-graphics.html#cb479-9" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">theme_minimal</span>()</span>
<span id="cb479-10"><a href="basic-graphics.html#cb479-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb479-11"><a href="basic-graphics.html#cb479-11" aria-hidden="true" tabindex="-1"></a>nyc_temp_plot</span></code></pre></div>
<p><img src="figures/session_10_figs/58_scatterplot_nyc.png" width="90%" /></p>
<p><strong>30)</strong> We can explore by fitting a line to the scatterplot using <code>geom_smooth()</code> to our NYC climate data.</p>
<div class="sourceCode" id="cb480"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb480-1"><a href="basic-graphics.html#cb480-1" aria-hidden="true" tabindex="-1"></a><span class="do">## NYC temperatures plot and fit line loess</span></span>
<span id="cb480-2"><a href="basic-graphics.html#cb480-2" aria-hidden="true" tabindex="-1"></a>nyc_temp_plot_smooth <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(nyc_temps, <span class="fu">aes</span>(<span class="at">x =</span> YEAR, <span class="at">y =</span> ANNUAL)) <span class="sc">+</span></span>
<span id="cb480-3"><a href="basic-graphics.html#cb480-3" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb480-4"><a href="basic-graphics.html#cb480-4" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">geom_smooth</span>() <span class="sc">+</span></span>
<span id="cb480-5"><a href="basic-graphics.html#cb480-5" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;NYC temperature Central Park&quot;</span>, <span class="at">x=</span><span class="st">&quot;Year&quot;</span>, <span class="at">y=</span><span class="st">&quot;Temp (F)&quot;</span>) <span class="sc">+</span></span>
<span id="cb480-6"><a href="basic-graphics.html#cb480-6" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="dv">35</span>, <span class="dv">65</span>) <span class="sc">+</span></span>
<span id="cb480-7"><a href="basic-graphics.html#cb480-7" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme_minimal</span>()</span>
<span id="cb480-8"><a href="basic-graphics.html#cb480-8" aria-hidden="true" tabindex="-1"></a>nyc_temp_plot_smooth</span>
<span id="cb480-9"><a href="basic-graphics.html#cb480-9" aria-hidden="true" tabindex="-1"></a><span class="co">#`geom_smooth()` using method = &#39;loess&#39; and formula &#39;y ~ x&#39;</span></span></code></pre></div>
<p><img src="figures/session_10_figs/59_scatterplot_nyc_loess.png" width="90%" /></p>
<p>Notice the upward trend in the temperature. We can plot using a curve with other methods like <code>lm</code>.</p>
<div class="sourceCode" id="cb481"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb481-1"><a href="basic-graphics.html#cb481-1" aria-hidden="true" tabindex="-1"></a><span class="do">## NYC temperatures plot and fit line with lm</span></span>
<span id="cb481-2"><a href="basic-graphics.html#cb481-2" aria-hidden="true" tabindex="-1"></a>nyc_temp_plot_smooth_lm <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(nyc_temps, <span class="fu">aes</span>(<span class="at">x =</span> YEAR, <span class="at">y =</span> ANNUAL)) <span class="sc">+</span></span>
<span id="cb481-3"><a href="basic-graphics.html#cb481-3" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb481-4"><a href="basic-graphics.html#cb481-4" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">geom_smooth</span>(<span class="at">method=</span>lm, <span class="at">se=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb481-5"><a href="basic-graphics.html#cb481-5" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;NYC temperature Central Park - lm line and without se&quot;</span>, <span class="at">x=</span><span class="st">&quot;Year&quot;</span>, <span class="at">y=</span><span class="st">&quot;Temp (F)&quot;</span>) <span class="sc">+</span></span>
<span id="cb481-6"><a href="basic-graphics.html#cb481-6" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="dv">35</span>, <span class="dv">65</span>) <span class="sc">+</span></span>
<span id="cb481-7"><a href="basic-graphics.html#cb481-7" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme_minimal</span>()</span>
<span id="cb481-8"><a href="basic-graphics.html#cb481-8" aria-hidden="true" tabindex="-1"></a><span class="do">## I removed the confidence line change by setting in geom_smooth() --&gt; se=FALSE</span></span>
<span id="cb481-9"><a href="basic-graphics.html#cb481-9" aria-hidden="true" tabindex="-1"></a>nyc_temp_plot_smooth_lm</span>
<span id="cb481-10"><a href="basic-graphics.html#cb481-10" aria-hidden="true" tabindex="-1"></a><span class="co">#`geom_smooth()` using formula &#39;y ~ x&#39;</span></span></code></pre></div>
<p><img src="figures/session_10_figs/60_scatterplot_nyc_lm.png" width="90%" /></p>
<p><strong>31)</strong> We can also plot multiple lines when we provide groupings. We need to prepare our data.</p>
<div class="sourceCode" id="cb482"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb482-1"><a href="basic-graphics.html#cb482-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we need to prepare our data</span></span>
<span id="cb482-2"><a href="basic-graphics.html#cb482-2" aria-hidden="true" tabindex="-1"></a>nyc_average <span class="ot">&lt;-</span> <span class="fu">subset</span>(nyc_temps, <span class="at">select =</span> <span class="fu">c</span>(<span class="st">&quot;YEAR&quot;</span>, <span class="st">&quot;ANNUAL&quot;</span>))</span>
<span id="cb482-3"><a href="basic-graphics.html#cb482-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(nyc_average) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;YEAR&quot;</span>, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb482-4"><a href="basic-graphics.html#cb482-4" aria-hidden="true" tabindex="-1"></a>nyc_average<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="st">&quot;average&quot;</span></span>
<span id="cb482-5"><a href="basic-graphics.html#cb482-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-6"><a href="basic-graphics.html#cb482-6" aria-hidden="true" tabindex="-1"></a>nyc_max <span class="ot">&lt;-</span> <span class="fu">subset</span>(nyc_temps, <span class="at">select =</span> <span class="fu">c</span>(<span class="st">&quot;YEAR&quot;</span>, <span class="st">&quot;max_temp&quot;</span>))</span>
<span id="cb482-7"><a href="basic-graphics.html#cb482-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(nyc_max) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;YEAR&quot;</span>, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb482-8"><a href="basic-graphics.html#cb482-8" aria-hidden="true" tabindex="-1"></a>nyc_max<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="st">&quot;max&quot;</span></span>
<span id="cb482-9"><a href="basic-graphics.html#cb482-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-10"><a href="basic-graphics.html#cb482-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-11"><a href="basic-graphics.html#cb482-11" aria-hidden="true" tabindex="-1"></a>nyc_min <span class="ot">&lt;-</span> <span class="fu">subset</span>(nyc_temps, <span class="at">select =</span> <span class="fu">c</span>(<span class="st">&quot;YEAR&quot;</span>, <span class="st">&quot;min_temp&quot;</span>))</span>
<span id="cb482-12"><a href="basic-graphics.html#cb482-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(nyc_min) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;YEAR&quot;</span>, <span class="st">&quot;TEMP&quot;</span>)</span>
<span id="cb482-13"><a href="basic-graphics.html#cb482-13" aria-hidden="true" tabindex="-1"></a>nyc_min<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="st">&quot;min&quot;</span></span>
<span id="cb482-14"><a href="basic-graphics.html#cb482-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-15"><a href="basic-graphics.html#cb482-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb482-16"><a href="basic-graphics.html#cb482-16" aria-hidden="true" tabindex="-1"></a><span class="do">## all data together</span></span>
<span id="cb482-17"><a href="basic-graphics.html#cb482-17" aria-hidden="true" tabindex="-1"></a>nyc_all <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, <span class="fu">list</span>(nyc_average,nyc_max, nyc_min))</span>
<span id="cb482-18"><a href="basic-graphics.html#cb482-18" aria-hidden="true" tabindex="-1"></a>nyc_all<span class="sc">$</span>group2 <span class="ot">&lt;-</span> nyc_all<span class="sc">$</span>group</span>
<span id="cb482-19"><a href="basic-graphics.html#cb482-19" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(nyc_all)</span>
<span id="cb482-20"><a href="basic-graphics.html#cb482-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  453 obs. of  4 variables:</span></span>
<span id="cb482-21"><a href="basic-graphics.html#cb482-21" aria-hidden="true" tabindex="-1"></a><span class="co"># $ YEAR  : int  1869 1870 1871 1872 1873 1874 1875 1876 1877 1878 ...</span></span>
<span id="cb482-22"><a href="basic-graphics.html#cb482-22" aria-hidden="true" tabindex="-1"></a><span class="co"># $ TEMP  : num  51.4 53.6 51.1 51 51 51.3 49.4 51.9 52.8 53.6 ...</span></span>
<span id="cb482-23"><a href="basic-graphics.html#cb482-23" aria-hidden="true" tabindex="-1"></a><span class="co"># $ group : chr  &quot;average&quot; &quot;average&quot; &quot;average&quot; &quot;average&quot; ...</span></span>
<span id="cb482-24"><a href="basic-graphics.html#cb482-24" aria-hidden="true" tabindex="-1"></a><span class="co"># $ group2: chr  &quot;average&quot; &quot;average&quot; &quot;average&quot; &quot;average&quot; ...</span></span></code></pre></div>
<p>We now have the data ready and now we can plot the trends on average, maximum and minimum temperatures in NYC.</p>
<div class="sourceCode" id="cb483"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb483-1"><a href="basic-graphics.html#cb483-1" aria-hidden="true" tabindex="-1"></a><span class="do">## NYC temperatures plot and fit line loess</span></span>
<span id="cb483-2"><a href="basic-graphics.html#cb483-2" aria-hidden="true" tabindex="-1"></a>nyc_temp_plot_smooth_3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(nyc_all, <span class="fu">aes</span>(<span class="at">x =</span> YEAR, <span class="at">y =</span> TEMP, <span class="at">color =</span> group)) <span class="sc">+</span></span>
<span id="cb483-3"><a href="basic-graphics.html#cb483-3" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> group), <span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb483-4"><a href="basic-graphics.html#cb483-4" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">geom_smooth</span>(<span class="fu">aes</span>(<span class="at">color =</span> group2), <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb483-5"><a href="basic-graphics.html#cb483-5" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;NYC temperature Central Park&quot;</span>, <span class="at">x=</span><span class="st">&quot;Year&quot;</span>, <span class="at">y=</span><span class="st">&quot;Temp (F)&quot;</span>) <span class="sc">+</span></span>
<span id="cb483-6"><a href="basic-graphics.html#cb483-6" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="dv">15</span>, <span class="dv">90</span>) <span class="sc">+</span></span>
<span id="cb483-7"><a href="basic-graphics.html#cb483-7" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme_minimal</span>()</span>
<span id="cb483-8"><a href="basic-graphics.html#cb483-8" aria-hidden="true" tabindex="-1"></a>nyc_temp_plot_smooth_3</span>
<span id="cb483-9"><a href="basic-graphics.html#cb483-9" aria-hidden="true" tabindex="-1"></a><span class="co">#`geom_smooth()` using formula &#39;y ~ x&#39;</span></span></code></pre></div>
<p><img src="figures/session_10_figs/61_scatterplot_nyc_3lines.png" width="90%" /></p>
<p>A more elaborated plot of the same temperature trends.</p>
<div class="sourceCode" id="cb484"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb484-1"><a href="basic-graphics.html#cb484-1" aria-hidden="true" tabindex="-1"></a><span class="do">## NYC temperatures plot and fit line loess</span></span>
<span id="cb484-2"><a href="basic-graphics.html#cb484-2" aria-hidden="true" tabindex="-1"></a>color_points <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;green&quot;</span>,<span class="st">&quot;blue&quot;</span>)</span>
<span id="cb484-3"><a href="basic-graphics.html#cb484-3" aria-hidden="true" tabindex="-1"></a>color_lines <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;black&quot;</span>,<span class="st">&quot;black&quot;</span>,<span class="st">&quot;black&quot;</span>)</span>
<span id="cb484-4"><a href="basic-graphics.html#cb484-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb484-5"><a href="basic-graphics.html#cb484-5" aria-hidden="true" tabindex="-1"></a>nyc_temp_plot_smooth_4 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(nyc_all, <span class="fu">aes</span>(<span class="at">x =</span> YEAR, <span class="at">y =</span> TEMP)) <span class="sc">+</span></span>
<span id="cb484-6"><a href="basic-graphics.html#cb484-6" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">fill=</span><span class="fu">factor</span>(group)),<span class="at">size=</span><span class="dv">4</span>, <span class="at">shape=</span><span class="dv">21</span>, <span class="at">stroke=</span><span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb484-7"><a href="basic-graphics.html#cb484-7" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">geom_smooth</span>(<span class="fu">aes</span>(<span class="at">color =</span> group2), <span class="at">alpha =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb484-8"><a href="basic-graphics.html#cb484-8" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;NYC temperature Central Park&quot;</span>, <span class="at">x=</span><span class="st">&quot;Year&quot;</span>, <span class="at">y=</span><span class="st">&quot;Temp (F)&quot;</span>) <span class="sc">+</span></span>
<span id="cb484-9"><a href="basic-graphics.html#cb484-9" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>color_points) <span class="sc">+</span></span>
<span id="cb484-10"><a href="basic-graphics.html#cb484-10" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">scale_colour_manual</span>(<span class="at">values=</span>color_lines) <span class="sc">+</span></span>
<span id="cb484-11"><a href="basic-graphics.html#cb484-11" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">ylim</span>(<span class="dv">15</span>, <span class="dv">90</span>) <span class="sc">+</span></span>
<span id="cb484-12"><a href="basic-graphics.html#cb484-12" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">theme_minimal</span>()</span>
<span id="cb484-13"><a href="basic-graphics.html#cb484-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb484-14"><a href="basic-graphics.html#cb484-14" aria-hidden="true" tabindex="-1"></a>nyc_temp_plot_smooth_4 </span>
<span id="cb484-15"><a href="basic-graphics.html#cb484-15" aria-hidden="true" tabindex="-1"></a><span class="co">#`geom_smooth()` using method = &#39;loess&#39; and formula &#39;y ~ x&#39;</span></span></code></pre></div>
<p><img src="figures/session_10_figs/62_scatterplot_nyc_3lines_r.png" width="90%" />
## Multiple graphs at once with <code>cowplot</code></p>
<p>In most cases, you might want multiple plots at once before we print our graphics. We will use the R package <a href="https://cran.r-project.org/web/packages/cowplot/index.html" target="_blank"><strong>cowplot</strong></a> that provides various features that help with creating publication-quality figures with ‘ggplot2’, such as a set of themes, functions to align plots and arrange them into complex compound figures. This R package has a <a href="https://cran.r-project.org/web/packages/cowplot/vignettes/introduction.html" target="_blank"><strong>vignette</strong></a>.</p>
<p><strong>32)</strong> We need to install <strong>cowplot</strong> and you need two or more ggplot2 objects to make use of the features of this package. For this purpose, we will create four plots using the dataset <code>fruits</code> of the R package <code>ade4</code>.</p>
<div class="sourceCode" id="cb485"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb485-1"><a href="basic-graphics.html#cb485-1" aria-hidden="true" tabindex="-1"></a><span class="do">## install cowplot</span></span>
<span id="cb485-2"><a href="basic-graphics.html#cb485-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;cowplot&quot;</span>)</span>
<span id="cb485-3"><a href="basic-graphics.html#cb485-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb485-4"><a href="basic-graphics.html#cb485-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb485-5"><a href="basic-graphics.html#cb485-5" aria-hidden="true" tabindex="-1"></a><span class="do">## make sure that ggplot2 is loaded</span></span>
<span id="cb485-6"><a href="basic-graphics.html#cb485-6" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggplot2)</span>
<span id="cb485-7"><a href="basic-graphics.html#cb485-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb485-8"><a href="basic-graphics.html#cb485-8" aria-hidden="true" tabindex="-1"></a><span class="do">## “fruits” is dataset about batches of fruits -two types- are judged by two different ways.</span></span>
<span id="cb485-9"><a href="basic-graphics.html#cb485-9" aria-hidden="true" tabindex="-1"></a><span class="do">## They are classified in order of preference by 16 individuals (J1-J16)</span></span>
<span id="cb485-10"><a href="basic-graphics.html#cb485-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb485-11"><a href="basic-graphics.html#cb485-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb485-12"><a href="basic-graphics.html#cb485-12" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(fruits)</span>
<span id="cb485-13"><a href="basic-graphics.html#cb485-13" aria-hidden="true" tabindex="-1"></a>fruits_dataset <span class="ot">&lt;-</span> fruits</span>
<span id="cb485-14"><a href="basic-graphics.html#cb485-14" aria-hidden="true" tabindex="-1"></a>jug <span class="ot">&lt;-</span> fruits_dataset<span class="sc">$</span>jug</span>
<span id="cb485-15"><a href="basic-graphics.html#cb485-15" aria-hidden="true" tabindex="-1"></a>jug<span class="sc">$</span>fruit <span class="ot">&lt;-</span> <span class="fu">as.character</span>(fruits_dataset<span class="sc">$</span>type)</span>
<span id="cb485-16"><a href="basic-graphics.html#cb485-16" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(jug) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb485-17"><a href="basic-graphics.html#cb485-17" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(jug)</span>
<span id="cb485-18"><a href="basic-graphics.html#cb485-18" aria-hidden="true" tabindex="-1"></a><span class="co">#  J1 J2 J3 J4 J5 J6 J7 J8 J9 J10 J11 J12 J13 J14 J15 J16 fruit</span></span>
<span id="cb485-19"><a href="basic-graphics.html#cb485-19" aria-hidden="true" tabindex="-1"></a><span class="co">#1 10  5  8  3  1 18  5 17  3   4   1   2   5   3   1   1 necta</span></span>
<span id="cb485-20"><a href="basic-graphics.html#cb485-20" aria-hidden="true" tabindex="-1"></a><span class="co">#2  3  1  9  8  6 16  8 10  2   1   8   8   9   5   6   4 necta</span></span>
<span id="cb485-21"><a href="basic-graphics.html#cb485-21" aria-hidden="true" tabindex="-1"></a><span class="co">#3  5 11  5  2  8  8 18  3  4  15  14   4   7   1   3  13 peche</span></span>
<span id="cb485-22"><a href="basic-graphics.html#cb485-22" aria-hidden="true" tabindex="-1"></a><span class="co">#4  6 12  3  4  4  7 17  2  1  16  13   7   3   8   4  14 peche</span></span>
<span id="cb485-23"><a href="basic-graphics.html#cb485-23" aria-hidden="true" tabindex="-1"></a><span class="co">#5  4  2  4 14 17 10 16  1  5  19  21  13   6   2   5  15 peche</span></span>
<span id="cb485-24"><a href="basic-graphics.html#cb485-24" aria-hidden="true" tabindex="-1"></a><span class="co">#6  2  6 16 10 13  2 11  5 13   8  10  14  10   9  15   8 necta</span></span>
<span id="cb485-25"><a href="basic-graphics.html#cb485-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb485-26"><a href="basic-graphics.html#cb485-26" aria-hidden="true" tabindex="-1"></a><span class="do">## first plot -- individual J1 versus J2</span></span>
<span id="cb485-27"><a href="basic-graphics.html#cb485-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb485-28"><a href="basic-graphics.html#cb485-28" aria-hidden="true" tabindex="-1"></a>plot_J1_vs_J2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(jug, <span class="fu">aes</span>(<span class="at">x =</span> J1, <span class="at">y =</span> J2, <span class="at">color =</span> fruit)) <span class="sc">+</span> </span>
<span id="cb485-29"><a href="basic-graphics.html#cb485-29" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb485-30"><a href="basic-graphics.html#cb485-30" aria-hidden="true" tabindex="-1"></a>          <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;J1 versus J2 on fruits&quot;</span>) <span class="sc">+</span></span>
<span id="cb485-31"><a href="basic-graphics.html#cb485-31" aria-hidden="true" tabindex="-1"></a>          <span class="fu">theme_minimal</span>()</span>
<span id="cb485-32"><a href="basic-graphics.html#cb485-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb485-33"><a href="basic-graphics.html#cb485-33" aria-hidden="true" tabindex="-1"></a>plot_J1_vs_J2 </span></code></pre></div>
<p><img src="figures/session_10_figs/64_J1_J2.png" width="90%" /></p>
<div class="sourceCode" id="cb486"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb486-1"><a href="basic-graphics.html#cb486-1" aria-hidden="true" tabindex="-1"></a><span class="do">## second plot -- individual J1 versus J3</span></span>
<span id="cb486-2"><a href="basic-graphics.html#cb486-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb486-3"><a href="basic-graphics.html#cb486-3" aria-hidden="true" tabindex="-1"></a>plot_J1_vs_J3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(jug, <span class="fu">aes</span>(<span class="at">x =</span> J1, <span class="at">y =</span> J3, <span class="at">color =</span> fruit)) <span class="sc">+</span> </span>
<span id="cb486-4"><a href="basic-graphics.html#cb486-4" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb486-5"><a href="basic-graphics.html#cb486-5" aria-hidden="true" tabindex="-1"></a>          <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;J1 versus J3 on fruits&quot;</span>) <span class="sc">+</span></span>
<span id="cb486-6"><a href="basic-graphics.html#cb486-6" aria-hidden="true" tabindex="-1"></a>          <span class="fu">theme_minimal</span>()</span>
<span id="cb486-7"><a href="basic-graphics.html#cb486-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb486-8"><a href="basic-graphics.html#cb486-8" aria-hidden="true" tabindex="-1"></a>plot_J1_vs_J3 </span></code></pre></div>
<p><img src="figures/session_10_figs/65_J1_J3.png" width="90%" /></p>
<div class="sourceCode" id="cb487"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb487-1"><a href="basic-graphics.html#cb487-1" aria-hidden="true" tabindex="-1"></a><span class="do">## load reshape2 to make more amenable</span></span>
<span id="cb487-2"><a href="basic-graphics.html#cb487-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb487-3"><a href="basic-graphics.html#cb487-3" aria-hidden="true" tabindex="-1"></a>jug_reshape <span class="ot">&lt;-</span> <span class="fu">melt</span>(jug)</span>
<span id="cb487-4"><a href="basic-graphics.html#cb487-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(jug_reshape) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;fruit&quot;</span>, <span class="st">&quot;judge&quot;</span>, <span class="st">&quot;score&quot;</span>)</span>
<span id="cb487-5"><a href="basic-graphics.html#cb487-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(jug_reshape)</span>
<span id="cb487-6"><a href="basic-graphics.html#cb487-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  fruit judge score</span></span>
<span id="cb487-7"><a href="basic-graphics.html#cb487-7" aria-hidden="true" tabindex="-1"></a><span class="co">#1 necta    J1    10</span></span>
<span id="cb487-8"><a href="basic-graphics.html#cb487-8" aria-hidden="true" tabindex="-1"></a><span class="co">#2 necta    J1     3</span></span>
<span id="cb487-9"><a href="basic-graphics.html#cb487-9" aria-hidden="true" tabindex="-1"></a><span class="co">#3 peche    J1     5</span></span>
<span id="cb487-10"><a href="basic-graphics.html#cb487-10" aria-hidden="true" tabindex="-1"></a><span class="co">#4 peche    J1     6</span></span>
<span id="cb487-11"><a href="basic-graphics.html#cb487-11" aria-hidden="true" tabindex="-1"></a><span class="co">#5 peche    J1     4</span></span>
<span id="cb487-12"><a href="basic-graphics.html#cb487-12" aria-hidden="true" tabindex="-1"></a><span class="co">#6 necta    J1     2</span></span>
<span id="cb487-13"><a href="basic-graphics.html#cb487-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb487-14"><a href="basic-graphics.html#cb487-14" aria-hidden="true" tabindex="-1"></a><span class="do">## third plot -- all judges</span></span>
<span id="cb487-15"><a href="basic-graphics.html#cb487-15" aria-hidden="true" tabindex="-1"></a>plot_all_judges <span class="ot">&lt;-</span><span class="fu">ggplot</span>(jug_reshape, <span class="fu">aes</span>(<span class="at">x =</span> judge, <span class="at">y =</span> score, <span class="at">color =</span> fruit)) <span class="sc">+</span> </span>
<span id="cb487-16"><a href="basic-graphics.html#cb487-16" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb487-17"><a href="basic-graphics.html#cb487-17" aria-hidden="true" tabindex="-1"></a>          <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;all judges&quot;</span>) <span class="sc">+</span></span>
<span id="cb487-18"><a href="basic-graphics.html#cb487-18" aria-hidden="true" tabindex="-1"></a>          <span class="fu">theme_minimal</span>()</span>
<span id="cb487-19"><a href="basic-graphics.html#cb487-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb487-20"><a href="basic-graphics.html#cb487-20" aria-hidden="true" tabindex="-1"></a>plot_all_judges</span></code></pre></div>
<p><img src="figures/session_10_figs/66_boxplot_1.png" width="90%" /></p>
<div class="sourceCode" id="cb488"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb488-1"><a href="basic-graphics.html#cb488-1" aria-hidden="true" tabindex="-1"></a><span class="do">## fourth plot -- all fruits</span></span>
<span id="cb488-2"><a href="basic-graphics.html#cb488-2" aria-hidden="true" tabindex="-1"></a>plot_all_fruits <span class="ot">&lt;-</span><span class="fu">ggplot</span>(jug_reshape, <span class="fu">aes</span>(<span class="at">x =</span> fruit, <span class="at">y =</span> score, <span class="at">color =</span> judge)) <span class="sc">+</span> </span>
<span id="cb488-3"><a href="basic-graphics.html#cb488-3" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb488-4"><a href="basic-graphics.html#cb488-4" aria-hidden="true" tabindex="-1"></a>          <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;fruits&quot;</span>) <span class="sc">+</span></span>
<span id="cb488-5"><a href="basic-graphics.html#cb488-5" aria-hidden="true" tabindex="-1"></a>          <span class="fu">theme_minimal</span>()</span>
<span id="cb488-6"><a href="basic-graphics.html#cb488-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb488-7"><a href="basic-graphics.html#cb488-7" aria-hidden="true" tabindex="-1"></a>plot_all_fruits</span></code></pre></div>
<p><img src="figures/session_10_figs/67_boxplot_2.png" width="90%" /></p>
<p><strong>32)</strong> With this four plots, we can make use of several of the options of <strong>cowplot</strong>.</p>
<p>We can arrange two plots side by side in a row using the function <code>plot_grid()</code>. We will label them <code>A</code> and <code>B</code>.</p>
<div class="sourceCode" id="cb489"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb489-1"><a href="basic-graphics.html#cb489-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(plot_J1_vs_J2, plot_J1_vs_J3, </span>
<span id="cb489-2"><a href="basic-graphics.html#cb489-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&#39;A&#39;</span>, <span class="st">&#39;B&#39;</span>), <span class="at">label_size =</span> <span class="dv">12</span>)</span></code></pre></div>
<p><img src="figures/session_10_figs/68_two_plots_side.png" width="90%" /></p>
<p>We can arrange two plots one on top of the other with <code>plot_grid()</code> and the arguments for one column <code>ncol = 1</code> and align them vertically <code>align = "v"</code>.</p>
<div class="sourceCode" id="cb490"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb490-1"><a href="basic-graphics.html#cb490-1" aria-hidden="true" tabindex="-1"></a>plot_all_fruits <span class="ot">&lt;-</span> plot_all_fruits <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb490-2"><a href="basic-graphics.html#cb490-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(plot_all_judges, plot_all_fruits, </span>
<span id="cb490-3"><a href="basic-graphics.html#cb490-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">ncol =</span> <span class="dv">1</span>, <span class="at">align =</span> <span class="st">&quot;v&quot;</span>)</span></code></pre></div>
<p><img src="figures/session_10_figs/69_two_pots_column.png" width="90%" /></p>
<p>All four plots in a single one.</p>
<div class="sourceCode" id="cb491"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb491-1"><a href="basic-graphics.html#cb491-1" aria-hidden="true" tabindex="-1"></a><span class="do">## combine all into one plot</span></span>
<span id="cb491-2"><a href="basic-graphics.html#cb491-2" aria-hidden="true" tabindex="-1"></a>all_four_plots <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(plot_J1_vs_J2, plot_J1_vs_J3,  </span>
<span id="cb491-3"><a href="basic-graphics.html#cb491-3" aria-hidden="true" tabindex="-1"></a>        plot_all_judges, plot_all_fruits, </span>
<span id="cb491-4"><a href="basic-graphics.html#cb491-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">rel_heights =</span> <span class="fu">c</span>(.<span class="dv">6</span>, <span class="dv">1</span>), <span class="at">labels =</span> <span class="st">&quot;auto&quot;</span>)</span>
<span id="cb491-5"><a href="basic-graphics.html#cb491-5" aria-hidden="true" tabindex="-1"></a>all_four_plots</span></code></pre></div>
<p><img src="figures/session_10_figs/70_all_four_plots.png" width="90%" /></p>
</div>
<div id="save-your-plots-as-pdf-or-png" class="section level2" number="10.11">
<h2><span class="header-section-number">10.11</span> Save your plots as PDF or PNG</h2>
<p><strong>34)</strong> You can save your plots using the function <code>ggsave()</code>.</p>
<div class="sourceCode" id="cb492"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb492-1"><a href="basic-graphics.html#cb492-1" aria-hidden="true" tabindex="-1"></a><span class="do">## save &#39;all_plot&#39; graph as pdf</span></span>
<span id="cb492-2"><a href="basic-graphics.html#cb492-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;all_plots.pdf&quot;</span>)</span>
<span id="cb492-3"><a href="basic-graphics.html#cb492-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb492-4"><a href="basic-graphics.html#cb492-4" aria-hidden="true" tabindex="-1"></a><span class="do">## save &#39;all_plot&#39; graph as png</span></span>
<span id="cb492-5"><a href="basic-graphics.html#cb492-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb492-6"><a href="basic-graphics.html#cb492-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;all_plots.png&quot;</span>)</span>
<span id="cb492-7"><a href="basic-graphics.html#cb492-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb492-8"><a href="basic-graphics.html#cb492-8" aria-hidden="true" tabindex="-1"></a><span class="do">## save &#39;all_plot&#39; graph as png with some size defintions</span></span>
<span id="cb492-9"><a href="basic-graphics.html#cb492-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;all_plots2.png&quot;</span>, <span class="at">width =</span> <span class="dv">4</span>, <span class="at">height =</span> <span class="dv">4</span>)</span>
<span id="cb492-10"><a href="basic-graphics.html#cb492-10" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;all_plots3.png&quot;</span>, <span class="at">width =</span> <span class="dv">20</span>, <span class="at">height =</span> <span class="dv">20</span>, <span class="at">units =</span> <span class="st">&quot;cm&quot;</span>)</span></code></pre></div>
<p>Note that it has to be last plot you have in environment.</p>
</div>
<div id="ggplot2-add-on-packages-treemapify" class="section level2" number="10.12">
<h2><span class="header-section-number">10.12</span> ggplot2 add-on packages: <code>treemapify</code></h2>
<p>Treemaps are an alternative to piecharts as they can serve to display quantities
of a whole by category represented by rectagles with a given area size.
Each category is assigned a rectangle area with their subcategory rectangles nested inside of it.</p>
<div class="sourceCode" id="cb493"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb493-1"><a href="basic-graphics.html#cb493-1" aria-hidden="true" tabindex="-1"></a><span class="do">## install package if needed</span></span>
<span id="cb493-2"><a href="basic-graphics.html#cb493-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;treemapify&quot;</span>)</span>
<span id="cb493-3"><a href="basic-graphics.html#cb493-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb493-4"><a href="basic-graphics.html#cb493-4" aria-hidden="true" tabindex="-1"></a><span class="do">## load libraries</span></span>
<span id="cb493-5"><a href="basic-graphics.html#cb493-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb493-6"><a href="basic-graphics.html#cb493-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(treemapify)</span>
<span id="cb493-7"><a href="basic-graphics.html#cb493-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb493-8"><a href="basic-graphics.html#cb493-8" aria-hidden="true" tabindex="-1"></a><span class="do">## we will use the &#39;ichtyo’ dataset</span></span>
<span id="cb493-9"><a href="basic-graphics.html#cb493-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb493-10"><a href="basic-graphics.html#cb493-10" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(ichtyo)</span>
<span id="cb493-11"><a href="basic-graphics.html#cb493-11" aria-hidden="true" tabindex="-1"></a>ichtyo_data <span class="ot">&lt;-</span> ichtyo<span class="sc">$</span>tab</span>
<span id="cb493-12"><a href="basic-graphics.html#cb493-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb493-13"><a href="basic-graphics.html#cb493-13" aria-hidden="true" tabindex="-1"></a><span class="do">## column sum as data frame</span></span>
<span id="cb493-14"><a href="basic-graphics.html#cb493-14" aria-hidden="true" tabindex="-1"></a>ichtyo_sum_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">colSums</span> (ichtyo_data, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb493-15"><a href="basic-graphics.html#cb493-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb493-16"><a href="basic-graphics.html#cb493-16" aria-hidden="true" tabindex="-1"></a><span class="do">## build a nicer data frame and add column of site_names</span></span>
<span id="cb493-17"><a href="basic-graphics.html#cb493-17" aria-hidden="true" tabindex="-1"></a>ichtyo_sum_df<span class="sc">$</span>site_names <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ichtyo_sum_df)</span>
<span id="cb493-18"><a href="basic-graphics.html#cb493-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb493-19"><a href="basic-graphics.html#cb493-19" aria-hidden="true" tabindex="-1"></a><span class="do">## upgrade column names</span></span>
<span id="cb493-20"><a href="basic-graphics.html#cb493-20" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ichtyo_sum_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;counts&quot;</span>, <span class="st">&quot;site_names&quot;</span>)</span>
<span id="cb493-21"><a href="basic-graphics.html#cb493-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb493-22"><a href="basic-graphics.html#cb493-22" aria-hidden="true" tabindex="-1"></a><span class="do">## add count labels for later use</span></span>
<span id="cb493-23"><a href="basic-graphics.html#cb493-23" aria-hidden="true" tabindex="-1"></a>ichtyo_sum_df<span class="sc">$</span>count_label <span class="ot">&lt;-</span> <span class="fu">as.character</span>(ichtyo_sum_df<span class="sc">$</span>counts)</span>
<span id="cb493-24"><a href="basic-graphics.html#cb493-24" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(ichtyo_sum_df)</span>
<span id="cb493-25"><a href="basic-graphics.html#cb493-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  9 obs. of  3 variables:</span></span>
<span id="cb493-26"><a href="basic-graphics.html#cb493-26" aria-hidden="true" tabindex="-1"></a><span class="co"># $ counts     : num  936 995 1741 491 951 ...</span></span>
<span id="cb493-27"><a href="basic-graphics.html#cb493-27" aria-hidden="true" tabindex="-1"></a><span class="co"># $ site_names : chr  &quot;HOT&quot; &quot;VAN&quot; &quot;CHE&quot; &quot;SPI&quot; ...</span></span>
<span id="cb493-28"><a href="basic-graphics.html#cb493-28" aria-hidden="true" tabindex="-1"></a><span class="co"># $ count_label: chr  &quot;936&quot; &quot;995&quot; &quot;1741&quot; &quot;491&quot; ...</span></span>
<span id="cb493-29"><a href="basic-graphics.html#cb493-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb493-30"><a href="basic-graphics.html#cb493-30" aria-hidden="true" tabindex="-1"></a><span class="do">## Drawing a simple treemap</span></span>
<span id="cb493-31"><a href="basic-graphics.html#cb493-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb493-32"><a href="basic-graphics.html#cb493-32" aria-hidden="true" tabindex="-1"></a>ichtyo_treemap <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>ichtyo_sum_df, <span class="fu">aes</span>(<span class="at">label=</span>site_names, <span class="at">area=</span>counts, <span class="at">fill =</span> site_names, <span class="at">subgroup =</span> count_label)) <span class="sc">+</span></span>
<span id="cb493-33"><a href="basic-graphics.html#cb493-33" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_treemap</span>() <span class="sc">+</span></span>
<span id="cb493-34"><a href="basic-graphics.html#cb493-34" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_treemap_subgroup_text</span>(<span class="at">place =</span> <span class="st">&quot;centre&quot;</span>, <span class="at">grow =</span> <span class="cn">FALSE</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>, </span>
<span id="cb493-35"><a href="basic-graphics.html#cb493-35" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">colour =</span><span class="st">&quot;black&quot;</span>, <span class="at">fontface =</span> <span class="st">&quot;italic&quot;</span>, <span class="at">min.size =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb493-36"><a href="basic-graphics.html#cb493-36" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">geom_treemap_text</span>(<span class="at">colour =</span> <span class="st">&quot;white&quot;</span>, <span class="at">place =</span> <span class="st">&quot;topleft&quot;</span>, <span class="at">reflow =</span> <span class="cn">TRUE</span>, <span class="at">grow =</span> <span class="cn">FALSE</span>)</span>
<span id="cb493-37"><a href="basic-graphics.html#cb493-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb493-38"><a href="basic-graphics.html#cb493-38" aria-hidden="true" tabindex="-1"></a>ichtyo_treemap</span></code></pre></div>
<p><img src="figures/session_10_figs/73_treemap_example.png" width="90%" /></p>
<p>Notice that this plot is similar to the barchart, but presented as a treemap.</p>
</div>
<div id="other-resources" class="section level2" number="10.13">
<h2><span class="header-section-number">10.13</span> Other resources</h2>
<p><strong>33)</strong> Online resources:</p>
<p><strong>a) ggplot Wizardry</strong> is a nice page with some examples see <a href="https://github.com/Z3tt/OutlierConf2021/" target="_blank"><strong>here</strong></a></p>
<p><strong>b) ggplot tutorial</strong> is another page from the same author see <a href="https://www.cedricscherer.com/2019/08/05/a-ggplot2-tutorial-for-beautiful-plotting-in-r/" target="_blank"><strong>here</strong></a></p>
<p><strong>c) tidytuesady</strong> is a page of datasets see <a href="https://github.com/rfordatascience/tidytuesday" target="_blank"><strong>here</strong></a></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="graphics-and-colors.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="sequence-alignments.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/10-Basic_Graphics_v8.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
