<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Session 29 – Bioinformatics Literature | Bioinformatics Workshop Gitbook</title>
  <meta name="description" content="none" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Session 29 – Bioinformatics Literature | Bioinformatics Workshop Gitbook" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="none" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Session 29 – Bioinformatics Literature | Bioinformatics Workshop Gitbook" />
  
  <meta name="twitter:description" content="none" />
  

<meta name="author" content="Juan C. Santos &amp; Randy Ortiz" />


<meta name="date" content="2022-02-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="control-structures-loops-in-r.html"/>
<link rel="next" href="miscellaneous.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics Workshop Gitbook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html"><i class="fa fa-check"></i><b>1</b> – Getting Started with R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#why-do-we-use-r"><i class="fa fa-check"></i><b>1.1</b> Why do we use R?</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#get-a-good-text-editor"><i class="fa fa-check"></i><b>1.2</b> Get a good text editor</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-r"><i class="fa fa-check"></i><b>1.4</b> Installing R</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#annotating-code"><i class="fa fa-check"></i><b>1.5</b> Annotating code</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#defining-a-working-directory-in-r"><i class="fa fa-check"></i><b>1.6</b> Defining a working directory in R</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html"><i class="fa fa-check"></i><b>2</b> – Using R, Installing Packages and Importing/Exporting Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#your-first-session"><i class="fa fa-check"></i><b>2.1</b> Your first session</a></li>
<li class="chapter" data-level="2.2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#doing-some-base-functions-and-getting-help"><i class="fa fa-check"></i><b>2.2</b> Doing some base functions and getting help</a></li>
<li class="chapter" data-level="2.3" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#about-r-packages"><i class="fa fa-check"></i><b>2.3</b> About R-packages</a></li>
<li class="chapter" data-level="2.4" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#installing-packages"><i class="fa fa-check"></i><b>2.4</b> Installing packages</a></li>
<li class="chapter" data-level="2.5" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#downloading-data-files-github"><i class="fa fa-check"></i><b>2.5</b> Downloading data files (GitHub)</a></li>
<li class="chapter" data-level="2.6" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#importing-and-exporting-data"><i class="fa fa-check"></i><b>2.6</b> Importing and exporting data</a></li>
<li class="chapter" data-level="2.7" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#saving-data-as-rdata-.rdata-.rda"><i class="fa fa-check"></i><b>2.7</b> Saving data as Rdata (.rdata, .rda)</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html"><i class="fa fa-check"></i><b>3</b> – Basic Data Structures in R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#scalars-and-vectors"><i class="fa fa-check"></i><b>3.1</b> Scalars and vectors</a></li>
<li class="chapter" data-level="3.2" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#arrays"><i class="fa fa-check"></i><b>3.3</b> Arrays</a></li>
<li class="chapter" data-level="3.4" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a></li>
<li class="chapter" data-level="3.5" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tables"><i class="fa fa-check"></i><b>3.5</b> Tables</a></li>
<li class="chapter" data-level="3.6" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#lists"><i class="fa fa-check"></i><b>3.6</b> Lists</a></li>
<li class="chapter" data-level="3.7" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-tables"><i class="fa fa-check"></i><b>3.7</b> Data Tables</a></li>
<li class="chapter" data-level="3.8" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tibbles"><i class="fa fa-check"></i><b>3.8</b> Tibbles</a></li>
<li class="chapter" data-level="3.9" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#dnastringsets-rnastringsets-and-aastringsets"><i class="fa fa-check"></i><b>3.9</b> DNAStringSets, RNAStringSets and AAStringSets</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html"><i class="fa fa-check"></i><b>4</b> – Text Editing and Data Transformations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#working-with-character-strings"><i class="fa fa-check"></i><b>4.1</b> Working with character strings</a></li>
<li class="chapter" data-level="4.2" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#regular-expressions-regex"><i class="fa fa-check"></i><b>4.2</b> Regular expressions (Regex)</a></li>
<li class="chapter" data-level="4.3" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#comparing-ordering-and-simplifying-vectors"><i class="fa fa-check"></i><b>4.3</b> Comparing, ordering and simplifying vectors</a></li>
<li class="chapter" data-level="4.4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#useful-functions-of-stringr"><i class="fa fa-check"></i><b>4.4</b> Useful functions of <code>stringr</code></a></li>
<li class="chapter" data-level="4.5" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#data-transformations"><i class="fa fa-check"></i><b>4.5</b> Data transformations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html"><i class="fa fa-check"></i><b>5</b> – Getting Biological Data from Public Repositories</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#ncbi-databases-genbank-and-others"><i class="fa fa-check"></i><b>5.1</b> NCBI databases: GenBank and others</a></li>
<li class="chapter" data-level="5.2" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#accessing-ncbi-within-r-rentrez"><i class="fa fa-check"></i><b>5.2</b> Accessing NCBI within R: <code>rentrez</code></a></li>
<li class="chapter" data-level="5.3" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#retrieving-pubmed-references"><i class="fa fa-check"></i><b>5.3</b> Retrieving PubMed references</a></li>
<li class="chapter" data-level="5.4" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#comparing-sequences-blast"><i class="fa fa-check"></i><b>5.4</b> Comparing sequences: BLAST</a></li>
<li class="chapter" data-level="5.5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-uniprot"><i class="fa fa-check"></i><b>5.5</b> Other databases: UniProt</a></li>
<li class="chapter" data-level="5.6" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-ensembl"><i class="fa fa-check"></i><b>5.6</b> Other databases: Ensembl</a></li>
<li class="chapter" data-level="5.7" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-taxonomy-phylogenetic-trees-and-species-distributions"><i class="fa fa-check"></i><b>5.7</b> Other databases: Taxonomy, phylogenetic trees and species distributions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html"><i class="fa fa-check"></i><b>6</b> – Basic Statistics in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#exploring-your-data"><i class="fa fa-check"></i><b>6.1</b> Exploring your data</a></li>
<li class="chapter" data-level="6.2" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#normalizing-your-data"><i class="fa fa-check"></i><b>6.2</b> Normalizing your data</a></li>
<li class="chapter" data-level="6.3" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#what-test-should-i-use"><i class="fa fa-check"></i><b>6.3</b> What test should I use?</a></li>
<li class="chapter" data-level="6.4" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-means-t-test"><i class="fa fa-check"></i><b>6.4</b> Comparing means: t-test</a></li>
<li class="chapter" data-level="6.5" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-wilcoxon-signed-rank-test"><i class="fa fa-check"></i><b>6.5</b> Beyond t-test: Wilcoxon signed rank test</a></li>
<li class="chapter" data-level="6.6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-binomial-test"><i class="fa fa-check"></i><b>6.6</b> Beyond t-test: Binomial test</a></li>
<li class="chapter" data-level="6.7" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-chi-square-goodness-of-fit"><i class="fa fa-check"></i><b>6.7</b> Beyond t-test: Chi-square goodness of fit</a></li>
<li class="chapter" data-level="6.8" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-mcnemars-test"><i class="fa fa-check"></i><b>6.8</b> Beyond t-test: McNemar’s test</a></li>
<li class="chapter" data-level="6.9" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-fishers-exact-test"><i class="fa fa-check"></i><b>6.9</b> Beyond t-test: Fisher’s exact test</a></li>
<li class="chapter" data-level="6.10" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#correlations"><i class="fa fa-check"></i><b>6.10</b> Correlations</a></li>
<li class="chapter" data-level="6.11" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-distributions"><i class="fa fa-check"></i><b>6.11</b> Comparing distributions</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html"><i class="fa fa-check"></i><b>7</b> – Regressions and ANOVA in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#linear-regressions"><i class="fa fa-check"></i><b>7.1</b> Linear regressions</a></li>
<li class="chapter" data-level="7.2" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#selecting-predictors"><i class="fa fa-check"></i><b>7.2</b> Selecting predictors</a></li>
<li class="chapter" data-level="7.3" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions"><i class="fa fa-check"></i><b>7.3</b> Logistic regressions</a></li>
<li class="chapter" data-level="7.4" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#non-linear-regressions"><i class="fa fa-check"></i><b>7.4</b> Non-linear regressions</a></li>
<li class="chapter" data-level="7.5" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-regressions"><i class="fa fa-check"></i><b>7.5</b> Toy datasets for regressions</a></li>
<li class="chapter" data-level="7.6" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova"><i class="fa fa-check"></i><b>7.6</b> ANOVA</a></li>
<li class="chapter" data-level="7.7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#one-way-anova"><i class="fa fa-check"></i><b>7.7</b> One-way ANOVA</a></li>
<li class="chapter" data-level="7.8" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#two-way-anova"><i class="fa fa-check"></i><b>7.8</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="7.9" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#post-hoc-tests"><i class="fa fa-check"></i><b>7.9</b> Post-hoc tests</a></li>
<li class="chapter" data-level="7.10" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#beyond-anova-kruskal-wallis-test"><i class="fa fa-check"></i><b>7.10</b> Beyond ANOVA: Kruskal-Wallis Test</a></li>
<li class="chapter" data-level="7.11" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-anova"><i class="fa fa-check"></i><b>7.11</b> Toy datasets for ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html"><i class="fa fa-check"></i><b>8</b> – Exploring Multidimensional Data in R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html#principal-component-analysis"><i class="fa fa-check"></i><b>8.1</b> Principal Component Analysis</a></li>
<li class="chapter" data-level="8.2" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html#multidimensional-scaling"><i class="fa fa-check"></i><b>8.2</b> Multidimensional scaling</a></li>
<li class="chapter" data-level="8.3" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html#cluster-analysis-dfa"><i class="fa fa-check"></i><b>8.3</b> Cluster analysis: DFA</a></li>
<li class="chapter" data-level="8.4" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html#cluster-analysis-k-means-unsupervised"><i class="fa fa-check"></i><b>8.4</b> Cluster analysis: <em>k</em>-means (unsupervised)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html"><i class="fa fa-check"></i><b>9</b> – Graphics and Colors</a>
<ul>
<li class="chapter" data-level="9.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#avoid-bad-graphics"><i class="fa fa-check"></i><b>9.1</b> Avoid bad graphics</a></li>
<li class="chapter" data-level="9.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#graphics-in-r-ggplot2"><i class="fa fa-check"></i><b>9.2</b> Graphics in R – ggplot2</a></li>
<li class="chapter" data-level="9.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#preparing-your-dataset"><i class="fa fa-check"></i><b>9.3</b> Preparing your dataset</a></li>
<li class="chapter" data-level="9.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#color-palettes-online"><i class="fa fa-check"></i><b>9.4</b> Color palettes online</a></li>
<li class="chapter" data-level="9.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggplot2-default-palette"><i class="fa fa-check"></i><b>9.5</b> <code>ggplot2</code> default palette</a></li>
<li class="chapter" data-level="9.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#heatmaps-or-tiled-charts"><i class="fa fa-check"></i><b>9.6</b> Heatmaps or tiled charts</a></li>
<li class="chapter" data-level="9.7" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#r-packages-about-color"><i class="fa fa-check"></i><b>9.7</b> R-packages about color</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#viridis-package"><i class="fa fa-check"></i><b>9.7.1</b> <code>viridis</code> package</a></li>
<li class="chapter" data-level="9.7.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcolorbrewer-package"><i class="fa fa-check"></i><b>9.7.2</b> <code>RColorBrewer</code> package</a></li>
<li class="chapter" data-level="9.7.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggsci-package"><i class="fa fa-check"></i><b>9.7.3</b> <code>ggsci</code> package</a></li>
<li class="chapter" data-level="9.7.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#wesanderson-package"><i class="fa fa-check"></i><b>9.7.4</b> <code>wesanderson</code> package</a></li>
<li class="chapter" data-level="9.7.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcartocolor-package"><i class="fa fa-check"></i><b>9.7.5</b> <code>rcartocolor</code> package</a></li>
<li class="chapter" data-level="9.7.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#colorspace-package"><i class="fa fa-check"></i><b>9.7.6</b> <code>colorspace</code> package</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#making-your-own-palette"><i class="fa fa-check"></i><b>9.8</b> Making your own palette</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="basic-graphics.html"><a href="basic-graphics.html"><i class="fa fa-check"></i><b>10</b> – Basic Graphics</a>
<ul>
<li class="chapter" data-level="10.1" data-path="basic-graphics.html"><a href="basic-graphics.html#bar-charts"><i class="fa fa-check"></i><b>10.1</b> Bar charts</a></li>
<li class="chapter" data-level="10.2" data-path="basic-graphics.html"><a href="basic-graphics.html#stacked-bar-charts"><i class="fa fa-check"></i><b>10.2</b> Stacked bar charts</a></li>
<li class="chapter" data-level="10.3" data-path="basic-graphics.html"><a href="basic-graphics.html#histograms"><i class="fa fa-check"></i><b>10.3</b> Histograms</a></li>
<li class="chapter" data-level="10.4" data-path="basic-graphics.html"><a href="basic-graphics.html#multiple-histograms"><i class="fa fa-check"></i><b>10.4</b> Multiple histograms</a></li>
<li class="chapter" data-level="10.5" data-path="basic-graphics.html"><a href="basic-graphics.html#density-plots"><i class="fa fa-check"></i><b>10.5</b> Density plots</a></li>
<li class="chapter" data-level="10.6" data-path="basic-graphics.html"><a href="basic-graphics.html#box-plots"><i class="fa fa-check"></i><b>10.6</b> Box plots</a></li>
<li class="chapter" data-level="10.7" data-path="basic-graphics.html"><a href="basic-graphics.html#violin-plots"><i class="fa fa-check"></i><b>10.7</b> Violin plots</a></li>
<li class="chapter" data-level="10.8" data-path="basic-graphics.html"><a href="basic-graphics.html#scatterplots"><i class="fa fa-check"></i><b>10.8</b> Scatterplots</a></li>
<li class="chapter" data-level="10.9" data-path="basic-graphics.html"><a href="basic-graphics.html#lines-and-scatterplots"><i class="fa fa-check"></i><b>10.9</b> Lines and Scatterplots</a></li>
<li class="chapter" data-level="10.10" data-path="basic-graphics.html"><a href="basic-graphics.html#multiple-graphs-at-once-with-cowplot"><i class="fa fa-check"></i><b>10.10</b> Multiple graphs at once with <code>cowplot</code></a></li>
<li class="chapter" data-level="10.11" data-path="basic-graphics.html"><a href="basic-graphics.html#save-your-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>10.11</b> Save your plots as PDF or PNG</a></li>
<li class="chapter" data-level="10.12" data-path="basic-graphics.html"><a href="basic-graphics.html#other-resources"><i class="fa fa-check"></i><b>10.12</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="sequence-alignments.html"><a href="sequence-alignments.html"><i class="fa fa-check"></i><b>11</b> – Sequence Alignments</a>
<ul>
<li class="chapter" data-level="11.1" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-using-rentrez"><i class="fa fa-check"></i><b>11.1</b> Retrieving sequences using ‘rentrez’</a></li>
<li class="chapter" data-level="11.2" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-manually-from-ncbi"><i class="fa fa-check"></i><b>11.2</b> Retrieving sequences manually from NCBI</a></li>
<li class="chapter" data-level="11.3" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-accession-numbers"><i class="fa fa-check"></i><b>11.3</b> Retrieving accession numbers</a></li>
<li class="chapter" data-level="11.4" data-path="sequence-alignments.html"><a href="sequence-alignments.html#alignment-with-decipher"><i class="fa fa-check"></i><b>11.4</b> Alignment with <code>DECIPHER</code></a></li>
<li class="chapter" data-level="11.5" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualize-alignment"><i class="fa fa-check"></i><b>11.5</b> Visualize alignment</a></li>
<li class="chapter" data-level="11.6" data-path="sequence-alignments.html"><a href="sequence-alignments.html#save-alignment-to-file"><i class="fa fa-check"></i><b>11.6</b> Save alignment to file</a></li>
<li class="chapter" data-level="11.7" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-and-align-several-markers"><i class="fa fa-check"></i><b>11.7</b> Concatenate and align several markers</a></li>
<li class="chapter" data-level="11.8" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-data.table"><i class="fa fa-check"></i><b>11.8</b> Concatenate with <code>data.table</code></a></li>
<li class="chapter" data-level="11.9" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-biostrings"><i class="fa fa-check"></i><b>11.9</b> Concatenate with <code>Biostrings</code></a></li>
<li class="chapter" data-level="11.10" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualizing-alignments-in-r"><i class="fa fa-check"></i><b>11.10</b> Visualizing alignments in R</a></li>
<li class="chapter" data-level="11.11" data-path="sequence-alignments.html"><a href="sequence-alignments.html#other-sequence-alignment-software"><i class="fa fa-check"></i><b>11.11</b> Other sequence alignment software</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html"><i class="fa fa-check"></i><b>12</b> – Estimating Phylogenetic Trees</a>
<ul>
<li class="chapter" data-level="12.1" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#phylogeny-inference-with-iqtree2"><i class="fa fa-check"></i><b>12.1</b> Phylogeny inference with <code>iqtree2</code></a></li>
<li class="chapter" data-level="12.2" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-iqtree2-in-macos"><i class="fa fa-check"></i><b>12.2</b> Installing <code>iqtree2</code> in macOS</a></li>
<li class="chapter" data-level="12.3" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-and-running-iqtree2-in-a-pc"><i class="fa fa-check"></i><b>12.3</b> Installing and running <code>iqtree2</code> in a PC</a></li>
<li class="chapter" data-level="12.4" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#loading-the-iqtree_runner_r-function"><i class="fa fa-check"></i><b>12.4</b> Loading the <code>iqtree_runner_R</code> function</a></li>
<li class="chapter" data-level="12.5" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#running-iqtree2-from-r"><i class="fa fa-check"></i><b>12.5</b> Running <code>iqtree2</code> from R</a></li>
<li class="chapter" data-level="12.6" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-no-partitions"><i class="fa fa-check"></i><b>12.6</b> One marker and no partitions</a></li>
<li class="chapter" data-level="12.7" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-with-partitions"><i class="fa fa-check"></i><b>12.7</b> One marker and with partitions</a></li>
<li class="chapter" data-level="12.8" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#bootstrap-support"><i class="fa fa-check"></i><b>12.8</b> Bootstrap support</a></li>
<li class="chapter" data-level="12.9" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#many-markers-and-with-partitions"><i class="fa fa-check"></i><b>12.9</b> Many markers and with partitions</a></li>
<li class="chapter" data-level="12.10" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#other-sequence-phylogenetic-tree-estimation-software"><i class="fa fa-check"></i><b>12.10</b> Other sequence phylogenetic tree estimation software</a></li>
<li class="chapter" data-level="12.11" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#drawing-phylogenetic-trees-with-ggtree"><i class="fa fa-check"></i><b>12.11</b> Drawing phylogenetic trees with <code>ggtree</code></a></li>
<li class="chapter" data-level="12.12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#save-ggtree-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>12.12</b> Save ggtree plots as PDF or PNG</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html"><i class="fa fa-check"></i><b>13</b> – Mining from a reference dataset</a>
<ul>
<li class="chapter" data-level="13.1" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#a-worked-example-using-the-mtcars-dataset"><i class="fa fa-check"></i><b>13.1</b> A worked example using the <code>mtcars</code> dataset</a></li>
<li class="chapter" data-level="13.2" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-reference-dataset"><i class="fa fa-check"></i><b>13.2</b> Build a reference dataset</a></li>
<li class="chapter" data-level="13.3" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-sampling-dataset"><i class="fa fa-check"></i><b>13.3</b> Build a sampling dataset</a></li>
<li class="chapter" data-level="13.4" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#extracting-from-reference-dataset"><i class="fa fa-check"></i><b>13.4</b> Extracting from reference dataset</a></li>
<li class="chapter" data-level="13.5" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#plots-of-the-extracted-data"><i class="fa fa-check"></i><b>13.5</b> Plots of the extracted data</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="introduction-to-python.html"><a href="introduction-to-python.html"><i class="fa fa-check"></i><b>14</b> – Introduction to Python</a>
<ul>
<li class="chapter" data-level="14.1" data-path="introduction-to-python.html"><a href="introduction-to-python.html#welcome-to-python-why-python"><i class="fa fa-check"></i><b>14.1</b> Welcome to Python! Why Python?</a></li>
<li class="chapter" data-level="14.2" data-path="introduction-to-python.html"><a href="introduction-to-python.html#which-python-should-i-install"><i class="fa fa-check"></i><b>14.2</b> Which Python should I install?</a></li>
<li class="chapter" data-level="14.3" data-path="introduction-to-python.html"><a href="introduction-to-python.html#mac-os-x-python-installations"><i class="fa fa-check"></i><b>14.3</b> Mac OS X Python Installations</a></li>
<li class="chapter" data-level="14.4" data-path="introduction-to-python.html"><a href="introduction-to-python.html#windows-python-installations"><i class="fa fa-check"></i><b>14.4</b> Windows Python Installations</a></li>
<li class="chapter" data-level="14.5" data-path="introduction-to-python.html"><a href="introduction-to-python.html#install-an-ide-or-integrated-development-environment"><i class="fa fa-check"></i><b>14.5</b> Install an IDE, or integrated development environment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html"><i class="fa fa-check"></i><b>15</b> – Python: Keywords and Identifiers</a>
<ul>
<li class="chapter" data-level="15.1" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#keywords-in-python"><i class="fa fa-check"></i><b>15.1</b> Keywords in Python</a></li>
<li class="chapter" data-level="15.2" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#identifiers-in-python"><i class="fa fa-check"></i><b>15.2</b> Identifiers in Python</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html"><i class="fa fa-check"></i><b>16</b> – Python: Statements and Expressions</a>
<ul>
<li class="chapter" data-level="16.1" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#what-are-statements"><i class="fa fa-check"></i><b>16.1</b> What are statements?</a></li>
<li class="chapter" data-level="16.2" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#expressions-in-python"><i class="fa fa-check"></i><b>16.2</b> Expressions in Python</a></li>
<li class="chapter" data-level="16.3" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#simple-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.3</b> Simple Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.4" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#augmented-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.4</b> Augmented Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.5" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#explicit-and-implicit-line-continuation"><i class="fa fa-check"></i><b>16.5</b> Explicit and Implicit Line Continuation</a></li>
<li class="chapter" data-level="16.6" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#indentation-in-python-is-crucial-to-proper-code-function"><i class="fa fa-check"></i><b>16.6</b> Indentation in Python is Crucial to Proper Code function</a></li>
<li class="chapter" data-level="16.7" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#comments"><i class="fa fa-check"></i><b>16.7</b> Comments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="python-data-types.html"><a href="python-data-types.html"><i class="fa fa-check"></i><b>17</b> – Python: Data Types</a>
<ul>
<li class="chapter" data-level="17.1" data-path="python-data-types.html"><a href="python-data-types.html#booleans"><i class="fa fa-check"></i><b>17.1</b> Booleans</a></li>
<li class="chapter" data-level="17.2" data-path="python-data-types.html"><a href="python-data-types.html#numbers-int-float-and-complex"><i class="fa fa-check"></i><b>17.2</b> Numbers: int, float and complex</a></li>
<li class="chapter" data-level="17.3" data-path="python-data-types.html"><a href="python-data-types.html#strings"><i class="fa fa-check"></i><b>17.3</b> Strings</a></li>
<li class="chapter" data-level="17.4" data-path="python-data-types.html"><a href="python-data-types.html#bytes"><i class="fa fa-check"></i><b>17.4</b> Bytes</a></li>
<li class="chapter" data-level="17.5" data-path="python-data-types.html"><a href="python-data-types.html#lists-1"><i class="fa fa-check"></i><b>17.5</b> Lists</a></li>
<li class="chapter" data-level="17.6" data-path="python-data-types.html"><a href="python-data-types.html#tuples"><i class="fa fa-check"></i><b>17.6</b> Tuples</a></li>
<li class="chapter" data-level="17.7" data-path="python-data-types.html"><a href="python-data-types.html#sets"><i class="fa fa-check"></i><b>17.7</b> Sets</a></li>
<li class="chapter" data-level="17.8" data-path="python-data-types.html"><a href="python-data-types.html#dictionary"><i class="fa fa-check"></i><b>17.8</b> Dictionary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="python-strings.html"><a href="python-strings.html"><i class="fa fa-check"></i><b>18</b> – Python: Strings</a>
<ul>
<li class="chapter" data-level="18.1" data-path="python-strings.html"><a href="python-strings.html#string-operators"><i class="fa fa-check"></i><b>18.1</b> String Operators</a></li>
<li class="chapter" data-level="18.2" data-path="python-strings.html"><a href="python-strings.html#other-string-modifiers"><i class="fa fa-check"></i><b>18.2</b> Other String Modifiers</a></li>
<li class="chapter" data-level="18.3" data-path="python-strings.html"><a href="python-strings.html#format-characters-in-strings"><i class="fa fa-check"></i><b>18.3</b> Format-Characters in Strings</a></li>
<li class="chapter" data-level="18.4" data-path="python-strings.html"><a href="python-strings.html#comparisons-with-strings"><i class="fa fa-check"></i><b>18.4</b> Comparisons with Strings</a></li>
<li class="chapter" data-level="18.5" data-path="python-strings.html"><a href="python-strings.html#padding-strings"><i class="fa fa-check"></i><b>18.5</b> Padding strings</a></li>
<li class="chapter" data-level="18.6" data-path="python-strings.html"><a href="python-strings.html#search-for-words-in-string-and-output-location"><i class="fa fa-check"></i><b>18.6</b> Search for words in string and output location</a></li>
<li class="chapter" data-level="18.7" data-path="python-strings.html"><a href="python-strings.html#seek-and-replace-in-strings"><i class="fa fa-check"></i><b>18.7</b> Seek and replace in strings</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="python-operators.html"><a href="python-operators.html"><i class="fa fa-check"></i><b>19</b> – Python: Operators</a>
<ul>
<li class="chapter" data-level="19.1" data-path="python-operators.html"><a href="python-operators.html#many-types-of-operators"><i class="fa fa-check"></i><b>19.1</b> Many types of operators</a></li>
<li class="chapter" data-level="19.2" data-path="python-operators.html"><a href="python-operators.html#arithmetic-operators"><i class="fa fa-check"></i><b>19.2</b> Arithmetic operators</a></li>
<li class="chapter" data-level="19.3" data-path="python-operators.html"><a href="python-operators.html#comparison-operators"><i class="fa fa-check"></i><b>19.3</b> Comparison operators</a></li>
<li class="chapter" data-level="19.4" data-path="python-operators.html"><a href="python-operators.html#logical-operators"><i class="fa fa-check"></i><b>19.4</b> Logical operators</a></li>
<li class="chapter" data-level="19.5" data-path="python-operators.html"><a href="python-operators.html#bitwise-operators"><i class="fa fa-check"></i><b>19.5</b> Bitwise operators</a></li>
<li class="chapter" data-level="19.6" data-path="python-operators.html"><a href="python-operators.html#identity-operators"><i class="fa fa-check"></i><b>19.6</b> Identity Operators</a></li>
<li class="chapter" data-level="19.7" data-path="python-operators.html"><a href="python-operators.html#membership-operators"><i class="fa fa-check"></i><b>19.7</b> Membership operators</a></li>
<li class="chapter" data-level="19.8" data-path="python-operators.html"><a href="python-operators.html#what-is-the-order-of-operations-with-operators"><i class="fa fa-check"></i><b>19.8</b> What is the order of operations with operators?</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="biopython.html"><a href="biopython.html"><i class="fa fa-check"></i><b>20</b> – Biopython</a>
<ul>
<li class="chapter" data-level="20.1" data-path="biopython.html"><a href="biopython.html#introduction-to-biopython"><i class="fa fa-check"></i><b>20.1</b> Introduction to Biopython</a></li>
<li class="chapter" data-level="20.2" data-path="biopython.html"><a href="biopython.html#biopython-installation"><i class="fa fa-check"></i><b>20.2</b> Biopython Installation</a></li>
<li class="chapter" data-level="20.3" data-path="biopython.html"><a href="biopython.html#getting-started"><i class="fa fa-check"></i><b>20.3</b> Getting Started</a></li>
<li class="chapter" data-level="20.4" data-path="biopython.html"><a href="biopython.html#parsing-my-fasta"><i class="fa fa-check"></i><b>20.4</b> Parsing my FASTA</a></li>
<li class="chapter" data-level="20.5" data-path="biopython.html"><a href="biopython.html#parsing-my-genbank-or-genpept"><i class="fa fa-check"></i><b>20.5</b> Parsing my GenBank or GenPept</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="biopython-sequences.html"><a href="biopython-sequences.html"><i class="fa fa-check"></i><b>21</b> – Biopython: Sequences</a>
<ul>
<li class="chapter" data-level="21.1" data-path="biopython-sequences.html"><a href="biopython-sequences.html#sequence-objects"><i class="fa fa-check"></i><b>21.1</b> Sequence Objects</a></li>
<li class="chapter" data-level="21.2" data-path="biopython-sequences.html"><a href="biopython-sequences.html#concatenation-with-sequences"><i class="fa fa-check"></i><b>21.2</b> Concatenation with Sequences</a></li>
<li class="chapter" data-level="21.3" data-path="biopython-sequences.html"><a href="biopython-sequences.html#transcription-dna-to-mrna"><i class="fa fa-check"></i><b>21.3</b> Transcription: DNA to mRNA</a></li>
<li class="chapter" data-level="21.4" data-path="biopython-sequences.html"><a href="biopython-sequences.html#translation-rna-to-prot"><i class="fa fa-check"></i><b>21.4</b> Translation: RNA to PROT</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html"><i class="fa fa-check"></i><b>22</b> – Biopython: Mulitple Sequence Alignments</a>
<ul>
<li class="chapter" data-level="22.1" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwise2"><i class="fa fa-check"></i><b>22.1</b> pairwise2</a></li>
<li class="chapter" data-level="22.2" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwisealigner"><i class="fa fa-check"></i><b>22.2</b> PairwiseAligner</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html"><i class="fa fa-check"></i><b>23</b> – Biopython: GenomeDiagram</a>
<ul>
<li class="chapter" data-level="23.1" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#linearcircular-gene-diagrams"><i class="fa fa-check"></i><b>23.1</b> Linear/Circular Gene Diagrams</a></li>
<li class="chapter" data-level="23.2" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#diagram-readability"><i class="fa fa-check"></i><b>23.2</b> Diagram Readability</a></li>
<li class="chapter" data-level="23.3" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#box-shape"><i class="fa fa-check"></i><b>23.3</b> Box Shape</a></li>
<li class="chapter" data-level="23.4" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#edit-box-shape-arrows"><i class="fa fa-check"></i><b>23.4</b> Edit Box Shape: Arrows</a></li>
<li class="chapter" data-level="23.5" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#restriction-map"><i class="fa fa-check"></i><b>23.5</b> Restriction Map</a></li>
<li class="chapter" data-level="23.6" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#tracks-on-tracks"><i class="fa fa-check"></i><b>23.6</b> Tracks on Tracks</a></li>
<li class="chapter" data-level="23.7" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#cross-link-data"><i class="fa fa-check"></i><b>23.7</b> Cross-link Data</a></li>
<li class="chapter" data-level="23.8" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#clean-cross-link"><i class="fa fa-check"></i><b>23.8</b> Clean Cross-Link</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html"><i class="fa fa-check"></i><b>24</b> – Biopython: Chromosome Modeling</a>
<ul>
<li class="chapter" data-level="24.1" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html#gene-feature-modeling"><i class="fa fa-check"></i><b>24.1</b> Gene Feature Modeling</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html"><i class="fa fa-check"></i><b>25</b> – Biopython: Sequence Parsing with Plots</a>
<ul>
<li class="chapter" data-level="25.1" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#histograms-1"><i class="fa fa-check"></i><b>25.1</b> Histograms</a></li>
<li class="chapter" data-level="25.2" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#gc-content"><i class="fa fa-check"></i><b>25.2</b> GC Content</a></li>
<li class="chapter" data-level="25.3" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#nt-alignment-dot-plot"><i class="fa fa-check"></i><b>25.3</b> Nt Alignment Dot Plot</a></li>
<li class="chapter" data-level="25.4" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#bigger-dots-for-the-dot-plot"><i class="fa fa-check"></i><b>25.4</b> Bigger Dots for the Dot Plot</a></li>
<li class="chapter" data-level="25.5" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#fastq-we-did-not-forget-you"><i class="fa fa-check"></i><b>25.5</b> FASTQ we did not forget you</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html"><i class="fa fa-check"></i><b>26</b> – Python: Web Scraping w/ BeautifulSoup</a>
<ul>
<li class="chapter" data-level="26.1" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#introduction"><i class="fa fa-check"></i><b>26.1</b> Introduction</a></li>
<li class="chapter" data-level="26.2" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#installation"><i class="fa fa-check"></i><b>26.2</b> Installation</a></li>
<li class="chapter" data-level="26.3" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#getting-started-1"><i class="fa fa-check"></i><b>26.3</b> Getting started</a></li>
<li class="chapter" data-level="26.4" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-all-matching-tags"><i class="fa fa-check"></i><b>26.4</b> Mining all matching tags</a></li>
<li class="chapter" data-level="26.5" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-tags-by-class-or-id"><i class="fa fa-check"></i><b>26.5</b> Mining tags by class or id</a></li>
<li class="chapter" data-level="26.6" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#use-css-selectors-for-positional-mining"><i class="fa fa-check"></i><b>26.6</b> Use CSS Selectors for positional mining</a></li>
<li class="chapter" data-level="26.7" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#weather-page-example"><i class="fa fa-check"></i><b>26.7</b> Weather Page Example</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="python-subprocess.html"><a href="python-subprocess.html"><i class="fa fa-check"></i><b>27</b> – Python Subprocess</a>
<ul>
<li class="chapter" data-level="27.1" data-path="python-subprocess.html"><a href="python-subprocess.html#introduction-and-installation"><i class="fa fa-check"></i><b>27.1</b> Introduction and Installation</a></li>
<li class="chapter" data-level="27.2" data-path="python-subprocess.html"><a href="python-subprocess.html#creating-your-first-python-script-w-bbtools"><i class="fa fa-check"></i><b>27.2</b> Creating your first python script w/ BBTools</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html"><i class="fa fa-check"></i><b>28</b> – Control Structures &amp; Loops in R</a>
<ul>
<li class="chapter" data-level="28.1" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#controling-the-flow"><i class="fa fa-check"></i><b>28.1</b> Controling the flow</a></li>
<li class="chapter" data-level="28.2" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if"><i class="fa fa-check"></i><b>28.2</b> Conditional <code>if</code></a></li>
<li class="chapter" data-level="28.3" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if-and-else"><i class="fa fa-check"></i><b>28.3</b> Conditional <code>if</code> and <code>else</code></a></li>
<li class="chapter" data-level="28.4" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#nested-conditionals-if-if-else-and-else"><i class="fa fa-check"></i><b>28.4</b> Nested conditionals <code>if</code>, <code>if else</code> and <code>else</code></a></li>
<li class="chapter" data-level="28.5" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-ifelse"><i class="fa fa-check"></i><b>28.5</b> Conditional <code>ifelse</code></a></li>
<li class="chapter" data-level="28.6" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#loops-that-control-iterations-and-repeations"><i class="fa fa-check"></i><b>28.6</b> Loops that control iterations and repeations</a></li>
<li class="chapter" data-level="28.7" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-for-loop"><i class="fa fa-check"></i><b>28.7</b> The <code>for</code> Loop</a></li>
<li class="chapter" data-level="28.8" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-while-loop"><i class="fa fa-check"></i><b>28.8</b> The <code>while</code> Loop</a></li>
<li class="chapter" data-level="28.9" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-repeat-loop"><i class="fa fa-check"></i><b>28.9</b> The <code>repeat</code> Loop</a></li>
<li class="chapter" data-level="28.10" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-next-and-break-statements"><i class="fa fa-check"></i><b>28.10</b> The <code>next</code> and <code>break</code> statements</a></li>
<li class="chapter" data-level="28.11" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionslapply"><i class="fa fa-check"></i><b>28.11</b> Other loop functions:<code>lapply</code></a></li>
<li class="chapter" data-level="28.12" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsapply"><i class="fa fa-check"></i><b>28.12</b> Other loop functions:<code>apply</code></a></li>
<li class="chapter" data-level="28.13" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsmapply"><i class="fa fa-check"></i><b>28.13</b> Other loop functions:<code>mapply</code></a></li>
<li class="chapter" data-level="28.14" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionstapply"><i class="fa fa-check"></i><b>28.14</b> Other loop functions:<code>tapply</code></a></li>
<li class="chapter" data-level="28.15" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionssplit"><i class="fa fa-check"></i><b>28.15</b> Other loop functions:<code>split</code></a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html"><i class="fa fa-check"></i><b>29</b> – Bioinformatics Literature</a>
<ul>
<li class="chapter" data-level="29.1" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#peer-reviewed-bioinformatics-journals"><i class="fa fa-check"></i><b>29.1</b> Peer-reviewed bioinformatics journals</a></li>
<li class="chapter" data-level="29.2" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#access-to-journals-articles"><i class="fa fa-check"></i><b>29.2</b> Access to journals articles</a></li>
<li class="chapter" data-level="29.3" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#informal-access-to-code-preprints-and-vignettes"><i class="fa fa-check"></i><b>29.3</b> Informal access to code: Preprints and vignettes</a></li>
<li class="chapter" data-level="29.4" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#accessing-online-data-dryad"><i class="fa fa-check"></i><b>29.4</b> Accessing online data (DRYAD)</a></li>
<li class="chapter" data-level="29.5" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#online-bioinformatics-courses"><i class="fa fa-check"></i><b>29.5</b> Online bioinformatics courses</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="miscellaneous.html"><a href="miscellaneous.html"><i class="fa fa-check"></i><b>30</b> – Miscellaneous</a>
<ul>
<li class="chapter" data-level="30.1" data-path="miscellaneous.html"><a href="miscellaneous.html#sampling-from-a-vector-of-names"><i class="fa fa-check"></i><b>30.1</b> Sampling from a vector of names</a></li>
<li class="chapter" data-level="30.2" data-path="miscellaneous.html"><a href="miscellaneous.html#miscellaneous-1"><i class="fa fa-check"></i><b>30.2</b> Miscellaneous</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics Workshop Gitbook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bioinformatics-literature" class="section level1" number="29">
<h1><span class="header-section-number">Session 29</span> – Bioinformatics Literature</h1>
<div id="peer-reviewed-bioinformatics-journals" class="section level2" number="29.1">
<h2><span class="header-section-number">29.1</span> Peer-reviewed bioinformatics journals</h2>
<p>More than +25 journals exist that are focused on bioinformatics research and tools.
Many of those report the development of new tools and software with applications in computational medicine and biology. Most articles in these journals undergo peer-review evaluation, which is when one or more researchers with similar competencies as the authors (i.e., peers) determine if the presented research or paper is valid, reproducible, and the interpretation of results are accurate and consistent with the current knowledge on the subject.
This process allows to improve the quality of science published and prevents flawed or misinterpreted research to become public. Most scientists, industry, and policymakers play most attention to these articles.</p>
<p>Here is a list of some of the most widely cited journals in bioinformatics or with a strong emphasis on research derived from bioinformatics. All these journals publish articles after peer-reviewed evaluations:</p>
<ol style="list-style-type: decimal">
<li><a href="https://academic.oup.com/bioinformatics" target="_blank"><strong>Bioinformatics</strong></a> (2019, impact factor IF: 5.610). Mostly brief notes on new software and some reviews.</li>
<li><a href="https://academic.oup.com/bib" target="_blank"><strong>Briefings in Bioinformatics</strong></a> (2019, impact factor IF: 8.990). This journal publishes reviews for the users of databases and analytical tools of contemporary genetics, molecular and systems biology and is unique in providing practical help and guidance to the non-specialist in computerized methodology.</li>
<li><a href="https://journals.plos.org/ploscompbiol/" target="_blank"><strong>PLOS Computational Biology</strong></a> (does not provide impact factor). This journal publishes articles focused on the application of computational methods from molecules to ecosystems.</li>
<li><a href="https://bmcbioinformatics.biomedcentral.com/" target="_blank"><strong>BMC Bioinformatics</strong></a> (2019, impact factor IF: 3.169). This is an open access journal that considers articles describing novel computational algorithms and software, models and tools, including statistical methods, machine learning and artificial intelligence, as well as systems biology.</li>
<li><a href="https://academic.oup.com/gigascience" target="_blank"><strong>GigaScience</strong></a> (2020, impact factor IF: 6.524). This is an open access journal focusing on ‘big data’ research from the life and biomedical sciences.</li>
<li><a href="https://academic.oup.com/mbe" target="_blank"><strong>Molecular Biology and Evolution</strong></a> (2020, impact factor IF: 16.240). This is an open access journal focusing on patterns and processes that impact the evolution of life at molecular, it usually includes new and novel applications of bioinformatic tools.</li>
<li><a href="https://academic.oup.com/gbe" target="_blank"><strong>Genome Biology and Evolution</strong></a> (2020, impact factor IF: 3.416). This journal focusses on research at the interface between evolutionary biology and genomics.</li>
<li><a href="https://academic.oup.com/nar" target="_blank"><strong>Nucleic Acids Research</strong></a> (2020, impact factor IF: 16.971). This journal publishes research into physical, chemical, biochemical and biological aspects of nucleic acids and proteins involved in nucleic acid metabolism and/or interactions.</li>
<li><a href="https://besjournals.onlinelibrary.wiley.com/journal/2041210X" target="_blank"><strong>Methods in Ecology and Evolution</strong></a> (2020, impact factor IF: 7.780). This journal publishes articles that promote development of new methods (usually computational) in ecology &amp; evolution.</li>
<li><a href="https://www.nature.com/nmeth/" target="_blank"><strong>Nature Methods</strong></a> (2020, impact factor IF: 28.550). This journal publishes detailed protocols for diverse fields of the biological, chemical, and clinical sciences.</li>
<li><a href="https://www.embopress.org/journal/17444292" target="_blank"><strong>Molecular Systems Biology</strong></a> (2020, impact factor IF: 11.400). This is an open access journal focusing on the fields of systems biology, synthetic biology and systems medicine.</li>
<li><a href="https://genome.cshlp.org/" target="_blank"><strong>Genome Research</strong></a> (2020, impact factor IF: 9.043). This journal publishes studies on the structure, function, biology, and evolution of genomes.</li>
<li><a href="https://genomebiology.biomedcentral.com/" target="_blank"><strong>Genome Biology</strong></a> (2020, impact factor IF: 13.583). This journal publishes research in all areas of biology and biomedicine from a genomic and post-genomic perspective.</li>
<li><a href="https://academic.oup.com/sysbio" target="_blank"><strong>Systematic Biology</strong></a> (2020, impact factor IF: 15.683). This journal publishes research about the theory, principles, and methods of systematics as well as phylogeny, evolution, morphology, biogeography, paleontology, genetics, and the classification of all living things.</li>
</ol>
<p>Note: Impact factor or <strong>IF/JIF</strong> is a relative index of the overall quality of journal and it is calculated as the yearly mean number of citations of articles published
in the last two years. <strong>IF</strong> is considered as a proxy for the importance and prestige of a journal within its field of study. <strong>Low IF</strong> journals (e.g., those with less than 2.0)
are considered as less impactful than those with <strong>higher IF</strong>. However, this index is <a href="https://www.nature.com/articles/nature.2016.20224" target="_blank"><strong>not without controversy</strong></a>, here is <a href="https://www.biomedcentral.com/p/bmc-impact" target="_blank"><strong>other source</strong></a> on this controversy.</p>
</div>
<div id="access-to-journals-articles" class="section level2" number="29.2">
<h2><span class="header-section-number">29.2</span> Access to journals articles</h2>
<p>Most published research is now accessible through online sources (i.e., print forms are becoming obsolete). However, to access such articles, many of the journals above require an institutional or personal subscription that can be costly. However, many of these papers are provided to readers for free through different databases that are public.</p>
<p>Here is a list of some of the most comprehensive collections of free papers that accessible to the public:</p>
<ol style="list-style-type: decimal">
<li><a href="https://pubmed.ncbi.nlm.nih.gov/" target="_blank"><strong>PubMed</strong></a>. PubMed is a free search engine accessing primarily the MEDLINE database of references, abstracts and online books on life sciences and biomedical topics. Citations may include links to full text content from PubMed Central and publisher web sites.</li>
<li><a href="https://zenodo.org/" target="_blank"><strong>Zenodo</strong></a>. This open-access repository that allows researchers to deposit research papers, data sets, research software, reports, and any other research related digital artefacts.</li>
<li><a href="https://www.biorxiv.org/" target="_blank"><strong>bioRxiv</strong></a>. This is an open access preprint repository for the biological sciences. It provides preprints and papers that are not peer-reviewed.</li>
<li><a href="https://www.researchgate.net/" target="_blank"><strong>ResearchGate</strong></a>. This is is a social networking site for scientists and researchers to share papers, ask and answer questions, and find collaborators.</li>
<li><a href="https://www.academia.edu" target="_blank"><strong>Academia</strong></a>. This is for-profit social networking website for academics. You might find full papers, yet it requires a bit navigating or searching through browsers.</li>
</ol>
<p>Unfortunately, many recently published papers are not available on the sources above and you might get stuck trying to get a PDF of a paper of interest. However, most universities can provide you a PDF copy of a paper of interest through the <strong>Interlibrary Loan Service</strong> or ILS where a researcher of an institution that does not have access to article, book, or other library resource can receive a PDF or the actual document from another library that has access to it. The process requires that the user makes a request through ILS with their home library. Your institution library will act as an intermediary, identifies libraries with the desired item, places the request, receives the item, makes it available to the user, as well as arranges for its return (if necessary). For books, the lending library usually sets a due date and overdue fees of the material borrowed.</p>
</div>
<div id="informal-access-to-code-preprints-and-vignettes" class="section level2" number="29.3">
<h2><span class="header-section-number">29.3</span> Informal access to code: Preprints and vignettes</h2>
<p>The top bioinformatic journals (and <a href="https://en.wikipedia.org/wiki/List_of_bioinformatics_journals" target="_blank"><strong>other publications</strong></a>) provide a formal outlet about bioinformatics research, yet most research in biology now involves some level of development and implementation of bioinformatic tools.
This trend is consequence of the rapid expansion, generation, and accessibility of big datasets that require some level of automation and complex computation analyses.
In most multidisciplinary research articles, it has become customary that bioinformatics code is provided as annexes, appendices, supplementary materials, or hosted in dedicated online databases
such as <a href="https://github.com/" target="_blank"><strong>GitHub</strong></a>. Likewise, many authors prefer to publish manuals and description of their bioinformatic methods as a <a href="https://en.wikipedia.org/wiki/Preprint" target="_blank"><strong>preprints</strong></a>, which are
draft versions of a research article that precedes formal peer-review process and its publication in a journal. These preprints are usually hosted in specialized
databases such as <a href="https://www.biorxiv.org/" target="_blank"><strong>bioRxiv</strong></a>. For introductory, manuals, application examples, and guidelines on use of specific bioinformatic tools
(like those written in R language), you might find vignettes associated with a specific R package in the <a href="https://cran.r-project.org/" target="_blank"><strong>CRAN</strong></a> or <a href="https://www.bioconductor.org/" target="_blank"><strong>Bioconductor</strong></a> repositories.</p>
<p>Here is an example of how to access the <a href="https://www.bioconductor.org/packages/release/bioc/html/DESeq2.html" target="_blank"><strong><code>DESeq2</code></strong></a> R-package vignette, which is a package for the analysis of differential gene expression analysis.</p>
<div class="sourceCode" id="cb595"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb595-1"><a href="bioinformatics-literature.html#cb595-1" aria-hidden="true" tabindex="-1"></a><span class="do">## If you need to install this package in your computer:</span></span>
<span id="cb595-2"><a href="bioinformatics-literature.html#cb595-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb595-3"><a href="bioinformatics-literature.html#cb595-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb595-4"><a href="bioinformatics-literature.html#cb595-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb595-5"><a href="bioinformatics-literature.html#cb595-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb595-6"><a href="bioinformatics-literature.html#cb595-6" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;DESeq2&quot;</span>)</span>
<span id="cb595-7"><a href="bioinformatics-literature.html#cb595-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb595-8"><a href="bioinformatics-literature.html#cb595-8" aria-hidden="true" tabindex="-1"></a><span class="do">## If you already have installed the DESeq2 package, load it in R space:</span></span>
<span id="cb595-9"><a href="bioinformatics-literature.html#cb595-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb595-10"><a href="bioinformatics-literature.html#cb595-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb595-11"><a href="bioinformatics-literature.html#cb595-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb595-12"><a href="bioinformatics-literature.html#cb595-12" aria-hidden="true" tabindex="-1"></a><span class="do">## To view documentation and vignette associated with DESeq2:</span></span>
<span id="cb595-13"><a href="bioinformatics-literature.html#cb595-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb595-14"><a href="bioinformatics-literature.html#cb595-14" aria-hidden="true" tabindex="-1"></a><span class="fu">browseVignettes</span>(<span class="st">&quot;DESeq2&quot;</span>)</span></code></pre></div>
</div>
<div id="accessing-online-data-dryad" class="section level2" number="29.4">
<h2><span class="header-section-number">29.4</span> Accessing online data (DRYAD)</h2>
<p>You might want to access data for your own research, data comparisons with previous published studies, and/or complement your own datasets and projects. This can be done using and accessing several online databases. One of the most important is <a href="https://datadryad.org/stash" target="_blank"><strong>DRYAD</strong></a>. From their website, Dryad Digital Repository mission is to serve as an indefinity repository of curated (i.e., revised) digital resources that makes published data discoverable, freely reusable, and citable. Therefore, these data can then be re-used to create new knowledge. Dryad is an initiative &gt;300 journals and scientific societies that adopted a joint data archiving policy (JDAP) for their publications, and their recognition that open data, easy-to-use, not-for-profit, community-governed data infrastructure is fundamental to maintain such digital resources. Dryad stores mostly biological and medical data used in peer-reviewed journals.</p>
<p>To access Dryad, you can first determine if the article of interest has archived data in this repository. We will use this article that you can access from <strong>PubMed</strong>:</p>
<p><strong>Allen, R. C., Angst, D. C., &amp; Hall, A. R. (2019). Resistance Gene Carriage Predicts Growth of Natural and Clinical Escherichia coli Isolates in the Absence of Antibiotics. Applied and environmental microbiology, 85(4), e02111-18.</strong> <a href="https://doi.org/10.1128/AEM.02111-18" target="_blank">https://doi.org/10.1128/AEM.02111-18</a></p>
<p><strong>1)</strong> For this example, we find the digital article by searching <a href="https://pubmed.ncbi.nlm.nih.gov" target="_blank"><strong>PubMed</strong></a>. The results show that it can be accessed by a <a href="https://pubmed.ncbi.nlm.nih.gov/30530714/" target="_blank"><strong>PubMed link</strong></a> or by a <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6365833/" target="_blank"><strong>PMC link</strong></a>.</p>
<p><img src="figures/session_14_figs/1_article_in_PMC.png" width="100%" /></p>
<p><strong>2)</strong> Most times, the authors will report that they have a supplementary material in their methods or in special section at the end of the article (e.g., footnotes, supplementary material, etc). In this case, it provides only a PDF and a link to journal digital article.</p>
<p><img src="figures/session_14_figs/2_article_SM_PMC.png" width="100%" /></p>
<p><strong>3)</strong> Like in many cases, there is no mention of a Dryad submission or DOI in the text of the paper. However, you can try to determine if such digital repository exists by searching the title of the article directly in <a href="https://datadryad.org/stash" target="_blank"><strong>DRYAD</strong></a>. I suggest doing this and not use the Dryad DOI (sometimes this is a broken link). To search in the Dryad website, you will paste the title of the article on the <strong>Search tab</strong> in the top right side.
<img src="figures/session_14_figs/3_dryad_intro.png" width="100%" /></p>
<p><strong>4)</strong> If associated data files exist for this paper, you will see how many entries are associated to the paper title. In this case, only one exists and you can click on such link.</p>
<p><img src="figures/session_14_figs/4_dryad_example_1.png" width="100%" /></p>
<p><strong>5)</strong> In the webpage associated with the data, you will see a title like <strong>Data from:…</strong>, a summary/abstract of the paper, and a link to the <strong>Data Files</strong> with a tab to <strong>Download dataset</strong>. You can click on the latter to get the associated zip file.</p>
<p><img src="figures/session_14_figs/5_dryad_example_2.png" width="100%" /></p>
<p><strong>6)</strong> The associated data is in a compressed file called <code>doi_10.5061_dryad.d0q331t__v1.zip</code>. After uncompressing this file, you can identify several archives and folders that include a <code>Readme.txt</code> with explanation of the contents of each data file with a <code>*.csv</code> extension.</p>
<p><img src="figures/session_14_figs/6_dryad_example_3.png" width="100%" /></p>
<p>I added these files to the <a href="https://github.com/corytophanes/BIO_BIT_Bioinformatics_209/tree/main/ref_files_pdfs" target="_blank"><strong>LINK TO FILES of BIO/BIT 209</strong></a>.</p>
<p><strong>7)</strong> After downloading, you can explore those files in a text editor and they can be easily imported into R or excel. For reference, we can import two of these datasets (<code>ARG_Raw.csv</code> and <code>Growth_Curves.csv</code>) as explained in section <strong>2.6 Importing and exporting data</strong>.</p>
<p>We can start with <code>ARG_Raw.csv</code></p>
<div class="sourceCode" id="cb596"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb596-1"><a href="bioinformatics-literature.html#cb596-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We start by readinf the metadata about such file in the text of README_for_Data_Files.txt. </span></span>
<span id="cb596-2"><a href="bioinformatics-literature.html#cb596-2" aria-hidden="true" tabindex="-1"></a><span class="do">## We found the following:</span></span>
<span id="cb596-3"><a href="bioinformatics-literature.html#cb596-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-4"><a href="bioinformatics-literature.html#cb596-4" aria-hidden="true" tabindex="-1"></a>ARG_Raw.csv</span>
<span id="cb596-5"><a href="bioinformatics-literature.html#cb596-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-6"><a href="bioinformatics-literature.html#cb596-6" aria-hidden="true" tabindex="-1"></a>This is a data table of the output of running BLASTn with the contigs of the draft </span>
<span id="cb596-7"><a href="bioinformatics-literature.html#cb596-7" aria-hidden="true" tabindex="-1"></a>genome assemblies of the isolates against the ResFinder <span class="fu">database</span> (nucleotide). The </span>
<span id="cb596-8"><a href="bioinformatics-literature.html#cb596-8" aria-hidden="true" tabindex="-1"></a>column headings mostly correspond to standard blast outputs but are outlined below<span class="sc">:</span></span>
<span id="cb596-9"><a href="bioinformatics-literature.html#cb596-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-10"><a href="bioinformatics-literature.html#cb596-10" aria-hidden="true" tabindex="-1"></a>   Strain<span class="sc">-</span>Which strain genome the hit was.</span>
<span id="cb596-11"><a href="bioinformatics-literature.html#cb596-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-12"><a href="bioinformatics-literature.html#cb596-12" aria-hidden="true" tabindex="-1"></a>   Contig<span class="sc">-</span>Most genomes are draft assemblies so there are multiple <span class="fu">segments</span> (here </span>
<span id="cb596-13"><a href="bioinformatics-literature.html#cb596-13" aria-hidden="true" tabindex="-1"></a>   called nodes), this shows which node the hit occurred in.</span>
<span id="cb596-14"><a href="bioinformatics-literature.html#cb596-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-15"><a href="bioinformatics-literature.html#cb596-15" aria-hidden="true" tabindex="-1"></a>   ARG<span class="sc">-</span> which antibiotic resistance gene <span class="cf">in</span> the query matched the sequence.</span>
<span id="cb596-16"><a href="bioinformatics-literature.html#cb596-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-17"><a href="bioinformatics-literature.html#cb596-17" aria-hidden="true" tabindex="-1"></a>   Per.ID<span class="sc">-</span> Percentage identity.</span>
<span id="cb596-18"><a href="bioinformatics-literature.html#cb596-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-19"><a href="bioinformatics-literature.html#cb596-19" aria-hidden="true" tabindex="-1"></a>   Length<span class="sc">-</span> Length of the matching region</span>
<span id="cb596-20"><a href="bioinformatics-literature.html#cb596-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-21"><a href="bioinformatics-literature.html#cb596-21" aria-hidden="true" tabindex="-1"></a>   MiMa<span class="sc">-</span> Number of mismatches </span>
<span id="cb596-22"><a href="bioinformatics-literature.html#cb596-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-23"><a href="bioinformatics-literature.html#cb596-23" aria-hidden="true" tabindex="-1"></a>   Gap<span class="sc">-</span> Total length of gaps</span>
<span id="cb596-24"><a href="bioinformatics-literature.html#cb596-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-25"><a href="bioinformatics-literature.html#cb596-25" aria-hidden="true" tabindex="-1"></a>   C.Start <span class="sc">&amp;</span> C.End<span class="sc">-</span> Start and end of the matching region on the contig</span>
<span id="cb596-26"><a href="bioinformatics-literature.html#cb596-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-27"><a href="bioinformatics-literature.html#cb596-27" aria-hidden="true" tabindex="-1"></a>   A.Start <span class="sc">&amp;</span> A.End<span class="sc">-</span> Start and end of the matching region on the query</span>
<span id="cb596-28"><a href="bioinformatics-literature.html#cb596-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-29"><a href="bioinformatics-literature.html#cb596-29" aria-hidden="true" tabindex="-1"></a>   E. value<span class="sc">-</span> Significance value <span class="cf">for</span> the hit, calculated within strain.</span>
<span id="cb596-30"><a href="bioinformatics-literature.html#cb596-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-31"><a href="bioinformatics-literature.html#cb596-31" aria-hidden="true" tabindex="-1"></a>   Bit.Score<span class="sc">-</span> overall score <span class="cf">for</span> the strength of the Hit. For overlapping</span>
<span id="cb596-32"><a href="bioinformatics-literature.html#cb596-32" aria-hidden="true" tabindex="-1"></a>   hits, only the highest bit score was retained.</span>
<span id="cb596-33"><a href="bioinformatics-literature.html#cb596-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-34"><a href="bioinformatics-literature.html#cb596-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-35"><a href="bioinformatics-literature.html#cb596-35" aria-hidden="true" tabindex="-1"></a><span class="do">## We need to find the path to the corresponding file by &quot;drag-and-drop&quot; in </span></span>
<span id="cb596-36"><a href="bioinformatics-literature.html#cb596-36" aria-hidden="true" tabindex="-1"></a><span class="do">## MACs or finding file properties in PCs. </span></span>
<span id="cb596-37"><a href="bioinformatics-literature.html#cb596-37" aria-hidden="true" tabindex="-1"></a><span class="do">## In my computer (MAC), they look like this:</span></span>
<span id="cb596-38"><a href="bioinformatics-literature.html#cb596-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-39"><a href="bioinformatics-literature.html#cb596-39" aria-hidden="true" tabindex="-1"></a><span class="sc">~</span><span class="er">/</span>Desktop<span class="sc">/</span>Teach_R<span class="sc">/</span>class_datasets<span class="sc">/</span>Dryad_example_for_Data_Files<span class="sc">/</span>ARG_Raw.csv</span>
<span id="cb596-40"><a href="bioinformatics-literature.html#cb596-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-41"><a href="bioinformatics-literature.html#cb596-41" aria-hidden="true" tabindex="-1"></a><span class="do">## Both files are *.csv (i.e., comma delimited files), so they can be easily </span></span>
<span id="cb596-42"><a href="bioinformatics-literature.html#cb596-42" aria-hidden="true" tabindex="-1"></a><span class="do">## imported with the read.table function. </span></span>
<span id="cb596-43"><a href="bioinformatics-literature.html#cb596-43" aria-hidden="true" tabindex="-1"></a><span class="do">## These files have data as comma delimited and we want to preserve headers</span></span>
<span id="cb596-44"><a href="bioinformatics-literature.html#cb596-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-45"><a href="bioinformatics-literature.html#cb596-45" aria-hidden="true" tabindex="-1"></a>ARG_Raw_Allen_etal_2019_df <span class="ot">&lt;-</span> <span class="fu">read.table</span> (<span class="at">file =</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets/Dryad_example_for_Data_Files/ARG_Raw.csv&quot;</span>, </span>
<span id="cb596-46"><a href="bioinformatics-literature.html#cb596-46" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb596-47"><a href="bioinformatics-literature.html#cb596-47" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>,</span>
<span id="cb596-48"><a href="bioinformatics-literature.html#cb596-48" aria-hidden="true" tabindex="-1"></a>                          <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb596-49"><a href="bioinformatics-literature.html#cb596-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-50"><a href="bioinformatics-literature.html#cb596-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-51"><a href="bioinformatics-literature.html#cb596-51" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ARG_Raw_Allen_etal_2019_df)</span>
<span id="cb596-52"><a href="bioinformatics-literature.html#cb596-52" aria-hidden="true" tabindex="-1"></a><span class="co">#    X Strain  Contig                  ARG  Per.ID Length MiMa Gap C.Start  C.End A.Start A.End E.val Bit.Score</span></span>
<span id="cb596-53"><a href="bioinformatics-literature.html#cb596-53" aria-hidden="true" tabindex="-1"></a><span class="co">#1  2 705963  NODE_6      mdf(A)_1_Y08743  97.891   1233   26   0   40970  42202       1  1233     0      2134</span></span>
<span id="cb596-54"><a href="bioinformatics-literature.html#cb596-54" aria-hidden="true" tabindex="-1"></a><span class="co">#2  1 706090  NODE_4      mdf(A)_1_Y08743  97.891   1233   26   0   19449  20681       1  1233     0      2134</span></span>
<span id="cb596-55"><a href="bioinformatics-literature.html#cb596-55" aria-hidden="true" tabindex="-1"></a><span class="co">#3 50 707404 NODE_12      mdf(A)_1_Y08743  97.567   1233   30   0  113267 114499    1233     1     0      2111</span></span>
<span id="cb596-56"><a href="bioinformatics-literature.html#cb596-56" aria-hidden="true" tabindex="-1"></a><span class="co">#4  6 707404 NODE_23      sul2_2_AY034138 100.000    816    0   0    8928   9743       1   816     0      1507</span></span>
<span id="cb596-57"><a href="bioinformatics-literature.html#cb596-57" aria-hidden="true" tabindex="-1"></a><span class="co">#5 28 707404 NODE_23 aph(3)-Ib_5_AF321551 100.000    529    0   0    9804  10332       1   529     0       977</span></span>
<span id="cb596-58"><a href="bioinformatics-literature.html#cb596-58" aria-hidden="true" tabindex="-1"></a><span class="co">#6 38 707404 NODE_23    dfrA14_5_DQ388123  99.793    483    1   0   10343  10825       1   483     0       887</span></span>
<span id="cb596-59"><a href="bioinformatics-literature.html#cb596-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-60"><a href="bioinformatics-literature.html#cb596-60" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(ARG_Raw_Allen_etal_2019_df)</span>
<span id="cb596-61"><a href="bioinformatics-literature.html#cb596-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  36 obs. of  14 variables:</span></span>
<span id="cb596-62"><a href="bioinformatics-literature.html#cb596-62" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X        : int  2 1 50 6 28 38 11 5 21 61 ...</span></span>
<span id="cb596-63"><a href="bioinformatics-literature.html#cb596-63" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Strain   : chr  &quot;705963&quot; &quot;706090&quot; &quot;707404&quot; &quot;707404&quot; ...</span></span>
<span id="cb596-64"><a href="bioinformatics-literature.html#cb596-64" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Contig   : chr  &quot;NODE_6&quot; &quot;NODE_4&quot; &quot;NODE_12&quot; &quot;NODE_23&quot; ...</span></span>
<span id="cb596-65"><a href="bioinformatics-literature.html#cb596-65" aria-hidden="true" tabindex="-1"></a><span class="co"># $ ARG      : chr  &quot;mdf(A)_1_Y08743&quot; &quot;mdf(A)_1_Y08743&quot; &quot;mdf(A)_1_Y08743&quot; &quot;sul2_2_AY034138&quot; ...</span></span>
<span id="cb596-66"><a href="bioinformatics-literature.html#cb596-66" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Per.ID   : num  97.9 97.9 97.6 100 100 ...</span></span>
<span id="cb596-67"><a href="bioinformatics-literature.html#cb596-67" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Length   : int  1233 1233 1233 816 529 483 837 1233 1206 861 ...</span></span>
<span id="cb596-68"><a href="bioinformatics-literature.html#cb596-68" aria-hidden="true" tabindex="-1"></a><span class="co"># $ MiMa     : int  26 26 30 0 0 1 0 26 0 0 ...</span></span>
<span id="cb596-69"><a href="bioinformatics-literature.html#cb596-69" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Gap      : int  0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb596-70"><a href="bioinformatics-literature.html#cb596-70" aria-hidden="true" tabindex="-1"></a><span class="co"># $ C.Start  : int  40970 19449 113267 8928 9804 10343 11175 28147 2128 4849 ...</span></span>
<span id="cb596-71"><a href="bioinformatics-literature.html#cb596-71" aria-hidden="true" tabindex="-1"></a><span class="co"># $ C.End    : int  42202 20681 114499 9743 10332 10825 12011 29379 3333 5709 ...</span></span>
<span id="cb596-72"><a href="bioinformatics-literature.html#cb596-72" aria-hidden="true" tabindex="-1"></a><span class="co"># $ A.Start  : int  1 1 1233 1 1 1 1 1233 1 1 ...</span></span>
<span id="cb596-73"><a href="bioinformatics-literature.html#cb596-73" aria-hidden="true" tabindex="-1"></a><span class="co"># $ A.End    : int  1233 1233 1 816 529 483 837 1 1206 861 ...</span></span>
<span id="cb596-74"><a href="bioinformatics-literature.html#cb596-74" aria-hidden="true" tabindex="-1"></a><span class="co"># $ E.val    : num  0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb596-75"><a href="bioinformatics-literature.html#cb596-75" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Bit.Score: int  2134 2134 2111 1507 977 887 1546 2134 2228 1591 ...</span></span>
<span id="cb596-76"><a href="bioinformatics-literature.html#cb596-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-77"><a href="bioinformatics-literature.html#cb596-77" aria-hidden="true" tabindex="-1"></a><span class="do">## We notice this data frame has LESS rows (only 36) than the original dataset </span></span>
<span id="cb596-78"><a href="bioinformatics-literature.html#cb596-78" aria-hidden="true" tabindex="-1"></a><span class="do">## that has 230 rows (you can check this by opening the ARG_Raw.csv file in a text editor). </span></span>
<span id="cb596-79"><a href="bioinformatics-literature.html#cb596-79" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ARG_Raw_Allen_etal_2019_df)</span>
<span id="cb596-80"><a href="bioinformatics-literature.html#cb596-80" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 36 14</span></span>
<span id="cb596-81"><a href="bioinformatics-literature.html#cb596-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-82"><a href="bioinformatics-literature.html#cb596-82" aria-hidden="true" tabindex="-1"></a><span class="do">## If we explore the original file: ARG_Raw.csv, this has special characters (&#39;). </span></span>
<span id="cb596-83"><a href="bioinformatics-literature.html#cb596-83" aria-hidden="true" tabindex="-1"></a><span class="do">## This character can be problematic, so we find a replace it  with &quot;_&quot; or removing </span></span>
<span id="cb596-84"><a href="bioinformatics-literature.html#cb596-84" aria-hidden="true" tabindex="-1"></a><span class="do">## them completely using a text editor. After removing this special character, we </span></span>
<span id="cb596-85"><a href="bioinformatics-literature.html#cb596-85" aria-hidden="true" tabindex="-1"></a><span class="do">## save the new modified file as ARG_Raw_modified.csv and we can import it to R.</span></span>
<span id="cb596-86"><a href="bioinformatics-literature.html#cb596-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-87"><a href="bioinformatics-literature.html#cb596-87" aria-hidden="true" tabindex="-1"></a>ARG_Raw_modified_Allen_etal_2019_df <span class="ot">&lt;-</span> <span class="fu">read.table</span> (<span class="at">file =</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets/Dryad_example_for_Data_Files/ARG_Raw_modified.csv&quot;</span>, </span>
<span id="cb596-88"><a href="bioinformatics-literature.html#cb596-88" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb596-89"><a href="bioinformatics-literature.html#cb596-89" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>,</span>
<span id="cb596-90"><a href="bioinformatics-literature.html#cb596-90" aria-hidden="true" tabindex="-1"></a>                          <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb596-91"><a href="bioinformatics-literature.html#cb596-91" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ARG_Raw_modified_Allen_etal_2019_df)</span>
<span id="cb596-92"><a href="bioinformatics-literature.html#cb596-92" aria-hidden="true" tabindex="-1"></a><span class="co">#    X Strain  Contig                    ARG  Per.ID Length MiMa Gap C.Start  C.End A.Start A.End E.val Bit.Score</span></span>
<span id="cb596-93"><a href="bioinformatics-literature.html#cb596-93" aria-hidden="true" tabindex="-1"></a><span class="co">#1  2 705963  NODE_6        mdf(A)_1_Y08743  97.891   1233   26   0   40970  42202       1  1233     0      2134</span></span>
<span id="cb596-94"><a href="bioinformatics-literature.html#cb596-94" aria-hidden="true" tabindex="-1"></a><span class="co">#2  1 706090  NODE_4        mdf(A)_1_Y08743  97.891   1233   26   0   19449  20681       1  1233     0      2134</span></span>
<span id="cb596-95"><a href="bioinformatics-literature.html#cb596-95" aria-hidden="true" tabindex="-1"></a><span class="co">#3 50 707404 NODE_12        mdf(A)_1_Y08743  97.567   1233   30   0  113267 114499    1233     1     0      2111</span></span>
<span id="cb596-96"><a href="bioinformatics-literature.html#cb596-96" aria-hidden="true" tabindex="-1"></a><span class="co">#4  6 707404 NODE_23        sul2_2_AY034138 100.000    816    0   0    8928   9743       1   816     0      1507</span></span>
<span id="cb596-97"><a href="bioinformatics-literature.html#cb596-97" aria-hidden="true" tabindex="-1"></a><span class="co">#5 28 707404 NODE_23 aph(3__)-Ib_5_AF321551 100.000    529    0   0    9804  10332       1   529     0       977</span></span>
<span id="cb596-98"><a href="bioinformatics-literature.html#cb596-98" aria-hidden="true" tabindex="-1"></a><span class="co">#6 38 707404 NODE_23      dfrA14_5_DQ388123  99.793    483    1   0   10343  10825       1   483     0       887</span></span>
<span id="cb596-99"><a href="bioinformatics-literature.html#cb596-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-100"><a href="bioinformatics-literature.html#cb596-100" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(ARG_Raw_modified_Allen_etal_2019_df)</span>
<span id="cb596-101"><a href="bioinformatics-literature.html#cb596-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;&#39;data.frame&#39;: 230 obs. of  14 variables:</span></span>
<span id="cb596-102"><a href="bioinformatics-literature.html#cb596-102" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X        : int  2 1 50 6 28 38 11 5 21 61 ...</span></span>
<span id="cb596-103"><a href="bioinformatics-literature.html#cb596-103" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Strain   : chr  &quot;705963&quot; &quot;706090&quot; &quot;707404&quot; &quot;707404&quot; ...</span></span>
<span id="cb596-104"><a href="bioinformatics-literature.html#cb596-104" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Contig   : chr  &quot;NODE_6&quot; &quot;NODE_4&quot; &quot;NODE_12&quot; &quot;NODE_23&quot; ...</span></span>
<span id="cb596-105"><a href="bioinformatics-literature.html#cb596-105" aria-hidden="true" tabindex="-1"></a><span class="co"># $ ARG      : chr  &quot;mdf(A)_1_Y08743&quot; &quot;mdf(A)_1_Y08743&quot; &quot;mdf(A)_1_Y08743&quot; &quot;sul2_2_AY034138&quot; ...</span></span>
<span id="cb596-106"><a href="bioinformatics-literature.html#cb596-106" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Per.ID   : num  97.9 97.9 97.6 100 100 ...</span></span>
<span id="cb596-107"><a href="bioinformatics-literature.html#cb596-107" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Length   : int  1233 1233 1233 816 529 483 837 1233 1206 861 ...</span></span>
<span id="cb596-108"><a href="bioinformatics-literature.html#cb596-108" aria-hidden="true" tabindex="-1"></a><span class="co"># $ MiMa     : int  26 26 30 0 0 1 0 26 0 0 ...</span></span>
<span id="cb596-109"><a href="bioinformatics-literature.html#cb596-109" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Gap      : int  0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb596-110"><a href="bioinformatics-literature.html#cb596-110" aria-hidden="true" tabindex="-1"></a><span class="co"># $ C.Start  : int  40970 19449 113267 8928 9804 10343 11175 28147 2128 4849 ...</span></span>
<span id="cb596-111"><a href="bioinformatics-literature.html#cb596-111" aria-hidden="true" tabindex="-1"></a><span class="co"># $ C.End    : int  42202 20681 114499 9743 10332 10825 12011 29379 3333 5709 ...</span></span>
<span id="cb596-112"><a href="bioinformatics-literature.html#cb596-112" aria-hidden="true" tabindex="-1"></a><span class="co"># $ A.Start  : int  1 1 1233 1 1 1 1 1233 1 1 ...</span></span>
<span id="cb596-113"><a href="bioinformatics-literature.html#cb596-113" aria-hidden="true" tabindex="-1"></a><span class="co"># $ A.End    : int  1233 1233 1 816 529 483 837 1 1206 861 ...</span></span>
<span id="cb596-114"><a href="bioinformatics-literature.html#cb596-114" aria-hidden="true" tabindex="-1"></a><span class="co"># $ E.val    : num  0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb596-115"><a href="bioinformatics-literature.html#cb596-115" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Bit.Score: int  2134 2134 2111 1507 977 887 1546 2134 2228 1591 ...</span></span>
<span id="cb596-116"><a href="bioinformatics-literature.html#cb596-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-117"><a href="bioinformatics-literature.html#cb596-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb596-118"><a href="bioinformatics-literature.html#cb596-118" aria-hidden="true" tabindex="-1"></a><span class="do">## We notice that have imported dataset has the corrected number of rows (230) </span></span>
<span id="cb596-119"><a href="bioinformatics-literature.html#cb596-119" aria-hidden="true" tabindex="-1"></a><span class="do">## and it is ready for our use in your analyses</span></span>
<span id="cb596-120"><a href="bioinformatics-literature.html#cb596-120" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ARG_Raw_modified_Allen_etal_2019_df)</span>
<span id="cb596-121"><a href="bioinformatics-literature.html#cb596-121" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 230  14</span></span></code></pre></div>
<p>Now, we import the next file <code>Growth_Curves.csv</code>. We follow the similar procedure as explained in section <strong>2.6 Importing and exporting data</strong>.</p>
<div class="sourceCode" id="cb597"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb597-1"><a href="bioinformatics-literature.html#cb597-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We start by readinf the metadata about such file in the text of README_for_Data_Files.txt. </span></span>
<span id="cb597-2"><a href="bioinformatics-literature.html#cb597-2" aria-hidden="true" tabindex="-1"></a><span class="do">## We found the following:</span></span>
<span id="cb597-3"><a href="bioinformatics-literature.html#cb597-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-4"><a href="bioinformatics-literature.html#cb597-4" aria-hidden="true" tabindex="-1"></a>Growth_Curves.csv</span>
<span id="cb597-5"><a href="bioinformatics-literature.html#cb597-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-6"><a href="bioinformatics-literature.html#cb597-6" aria-hidden="true" tabindex="-1"></a>This file contains the raw growth data which was used to fit the Gompertz growth curves. </span>
<span id="cb597-7"><a href="bioinformatics-literature.html#cb597-7" aria-hidden="true" tabindex="-1"></a>The column headings are<span class="sc">:</span></span>
<span id="cb597-8"><a href="bioinformatics-literature.html#cb597-8" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb597-9"><a href="bioinformatics-literature.html#cb597-9" aria-hidden="true" tabindex="-1"></a>   Master<span class="sc">-</span>Which of the two frozen master plates was used to inoculate the </span>
<span id="cb597-10"><a href="bioinformatics-literature.html#cb597-10" aria-hidden="true" tabindex="-1"></a>   overnight culture which inoculated this assay plate.</span>
<span id="cb597-11"><a href="bioinformatics-literature.html#cb597-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-12"><a href="bioinformatics-literature.html#cb597-12" aria-hidden="true" tabindex="-1"></a>   Sub.plate<span class="sc">-</span>Which of the <span class="dv">2</span> plates inoculated by the master plate this is. Thus </span>
<span id="cb597-13"><a href="bioinformatics-literature.html#cb597-13" aria-hidden="true" tabindex="-1"></a>   there are <span class="dv">4</span> replicates.</span>
<span id="cb597-14"><a href="bioinformatics-literature.html#cb597-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-15"><a href="bioinformatics-literature.html#cb597-15" aria-hidden="true" tabindex="-1"></a>   Treatment<span class="sc">-</span>Which of the <span class="dv">3</span> environmental conditions this growth was for.</span>
<span id="cb597-16"><a href="bioinformatics-literature.html#cb597-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-17"><a href="bioinformatics-literature.html#cb597-17" aria-hidden="true" tabindex="-1"></a>   Strain<span class="sc">-</span>Which strain the measurement came from. Those marked as blank had </span>
<span id="cb597-18"><a href="bioinformatics-literature.html#cb597-18" aria-hidden="true" tabindex="-1"></a>   no strain inoculated. Blanks have not been subtracted,there was a single </span>
<span id="cb597-19"><a href="bioinformatics-literature.html#cb597-19" aria-hidden="true" tabindex="-1"></a>   blank on each plate.</span>
<span id="cb597-20"><a href="bioinformatics-literature.html#cb597-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-21"><a href="bioinformatics-literature.html#cb597-21" aria-hidden="true" tabindex="-1"></a>   OD<span class="sc">-</span>Optical density measured at 595nm.</span>
<span id="cb597-22"><a href="bioinformatics-literature.html#cb597-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-23"><a href="bioinformatics-literature.html#cb597-23" aria-hidden="true" tabindex="-1"></a>   Read<span class="sc">-</span>An integer value <span class="cf">for</span> which read this is, takes values of <span class="dv">1</span><span class="fl">-41.</span></span>
<span id="cb597-24"><a href="bioinformatics-literature.html#cb597-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-25"><a href="bioinformatics-literature.html#cb597-25" aria-hidden="true" tabindex="-1"></a>   Time<span class="sc">-</span>The exact <span class="fu">time</span> (relative to the start of the growth experiment) </span>
<span id="cb597-26"><a href="bioinformatics-literature.html#cb597-26" aria-hidden="true" tabindex="-1"></a>   that the read was taken <span class="cf">in</span> hours, shown as a <span class="fu">fraction</span> (e.g. <span class="fl">4.5</span> is 4Hrs30Mins).</span>
<span id="cb597-27"><a href="bioinformatics-literature.html#cb597-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-28"><a href="bioinformatics-literature.html#cb597-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-29"><a href="bioinformatics-literature.html#cb597-29" aria-hidden="true" tabindex="-1"></a><span class="do">## We need to find the path to the corresponding file by &quot;drag-and_drop&quot; in MACs </span></span>
<span id="cb597-30"><a href="bioinformatics-literature.html#cb597-30" aria-hidden="true" tabindex="-1"></a><span class="do">## or finding file properties in PCs. </span></span>
<span id="cb597-31"><a href="bioinformatics-literature.html#cb597-31" aria-hidden="true" tabindex="-1"></a><span class="do">## In my computer (MAC), they look like this:</span></span>
<span id="cb597-32"><a href="bioinformatics-literature.html#cb597-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-33"><a href="bioinformatics-literature.html#cb597-33" aria-hidden="true" tabindex="-1"></a><span class="sc">~</span><span class="er">/</span>Desktop<span class="sc">/</span>Teach_R<span class="sc">/</span>class_datasets<span class="sc">/</span>Dryad_example_for_Data_Files<span class="sc">/</span>Growth_Curves.csv</span>
<span id="cb597-34"><a href="bioinformatics-literature.html#cb597-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-35"><a href="bioinformatics-literature.html#cb597-35" aria-hidden="true" tabindex="-1"></a><span class="do">## Both files are *.csv (i.e., comma delimited files), so they can be easily </span></span>
<span id="cb597-36"><a href="bioinformatics-literature.html#cb597-36" aria-hidden="true" tabindex="-1"></a><span class="do">## imported with the read.table function. These files have data as comma </span></span>
<span id="cb597-37"><a href="bioinformatics-literature.html#cb597-37" aria-hidden="true" tabindex="-1"></a><span class="do">## delimited and we want to preserve headers</span></span>
<span id="cb597-38"><a href="bioinformatics-literature.html#cb597-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-39"><a href="bioinformatics-literature.html#cb597-39" aria-hidden="true" tabindex="-1"></a>Growth_Curves_Allen_etal_2019_df <span class="ot">&lt;-</span> <span class="fu">read.table</span> (<span class="at">file =</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets/Dryad_example_for_Data_Files/Growth_Curves.csv&quot;</span>, </span>
<span id="cb597-40"><a href="bioinformatics-literature.html#cb597-40" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb597-41"><a href="bioinformatics-literature.html#cb597-41" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>,</span>
<span id="cb597-42"><a href="bioinformatics-literature.html#cb597-42" aria-hidden="true" tabindex="-1"></a>                          <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb597-43"><a href="bioinformatics-literature.html#cb597-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-44"><a href="bioinformatics-literature.html#cb597-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-45"><a href="bioinformatics-literature.html#cb597-45" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Growth_Curves_Allen_etal_2019_df)</span>
<span id="cb597-46"><a href="bioinformatics-literature.html#cb597-46" aria-hidden="true" tabindex="-1"></a><span class="co">#      X Master Sub.plate Treatment Strain     OD Read      Time</span></span>
<span id="cb597-47"><a href="bioinformatics-literature.html#cb597-47" aria-hidden="true" tabindex="-1"></a><span class="co">#1 1      2         1     Basal ECOR31 0.3877   38 44.488156</span></span>
<span id="cb597-48"><a href="bioinformatics-literature.html#cb597-48" aria-hidden="true" tabindex="-1"></a><span class="co">#2 2      2         1     Basal ECOR31 0.5019   12 13.760846</span></span>
<span id="cb597-49"><a href="bioinformatics-literature.html#cb597-49" aria-hidden="true" tabindex="-1"></a><span class="co">#3 3      2         1     Basal ECOR31 0.4839   11 12.588198</span></span>
<span id="cb597-50"><a href="bioinformatics-literature.html#cb597-50" aria-hidden="true" tabindex="-1"></a><span class="co">#4 4      2         1     Basal ECOR31 0.5245   20 23.211779</span></span>
<span id="cb597-51"><a href="bioinformatics-literature.html#cb597-51" aria-hidden="true" tabindex="-1"></a><span class="co">#5 5      2         1     Basal ECOR31 0.4581   29 33.816143</span></span>
<span id="cb597-52"><a href="bioinformatics-literature.html#cb597-52" aria-hidden="true" tabindex="-1"></a><span class="co">#6 6      2         1     Basal ECOR31 0.0590    2  1.977896</span></span>
<span id="cb597-53"><a href="bioinformatics-literature.html#cb597-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-54"><a href="bioinformatics-literature.html#cb597-54" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(Growth_Curves_Allen_etal_2019_df)</span>
<span id="cb597-55"><a href="bioinformatics-literature.html#cb597-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;&#39;data.frame&#39;: 45756 obs. of  8 variables:</span></span>
<span id="cb597-56"><a href="bioinformatics-literature.html#cb597-56" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X        : int  1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span id="cb597-57"><a href="bioinformatics-literature.html#cb597-57" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Master   : int  2 2 2 2 2 2 2 2 2 2 ...</span></span>
<span id="cb597-58"><a href="bioinformatics-literature.html#cb597-58" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Sub.plate: int  1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb597-59"><a href="bioinformatics-literature.html#cb597-59" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Treatment: chr  &quot;Basal&quot; &quot;Basal&quot; &quot;Basal&quot; &quot;Basal&quot; ...</span></span>
<span id="cb597-60"><a href="bioinformatics-literature.html#cb597-60" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Strain   : chr  &quot;ECOR31&quot; &quot;ECOR31&quot; &quot;ECOR31&quot; &quot;ECOR31&quot; ...</span></span>
<span id="cb597-61"><a href="bioinformatics-literature.html#cb597-61" aria-hidden="true" tabindex="-1"></a><span class="co"># $ OD       : num  0.388 0.502 0.484 0.524 0.458 ...</span></span>
<span id="cb597-62"><a href="bioinformatics-literature.html#cb597-62" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Read     : int  38 12 11 20 29 2 21 39 18 4 ...</span></span>
<span id="cb597-63"><a href="bioinformatics-literature.html#cb597-63" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Time     : num  44.5 13.8 12.6 23.2 33.8 ...</span></span>
<span id="cb597-64"><a href="bioinformatics-literature.html#cb597-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb597-65"><a href="bioinformatics-literature.html#cb597-65" aria-hidden="true" tabindex="-1"></a><span class="do">## We notice this data frame, this has EXACT number of rows than the original </span></span>
<span id="cb597-66"><a href="bioinformatics-literature.html#cb597-66" aria-hidden="true" tabindex="-1"></a><span class="do">## that the original files with 45756 rows. </span></span>
<span id="cb597-67"><a href="bioinformatics-literature.html#cb597-67" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(Growth_Curves_Allen_etal_2019_df)</span>
<span id="cb597-68"><a href="bioinformatics-literature.html#cb597-68" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 45756     8</span></span></code></pre></div>
</div>
<div id="online-bioinformatics-courses" class="section level2" number="29.5">
<h2><span class="header-section-number">29.5</span> Online bioinformatics courses</h2>
<p>Now days most students and researchers can find free bioinformatics classes that range in depth, quality, specificity, and usefulness.
It all depends on your needs and time that you are willing to dedicate to such courses. Some are short (few hours) and others are semester/months long.
In my experience, those that are short are not really useful unless the topic is something very specific that enhances or expands your previous knowledge
on the topic (i.e., data mining for genomics, if you already know basic data mining). Those courses that more useful for students starting in bioinformatics are those classes that
take more time, require you to do exercises, and do lots of coding exercises and provides you with worked examples. Nevertheless, the regular practice and goal-oriented implementation
of your basic coding skills are, in my opinion, the best form to improve your ability to do coding in a meaningful way.</p>
<p>Here is a list of some of <strong>free</strong> online courses that you can access, yet I have not tried or revised most of them.</p>
<ol style="list-style-type: decimal">
<li><a href="https://www.coursera.org/learn/r-programming" target="_blank">https://www.coursera.org/learn/r-programming</a>. This is a very popular online class for beginners focused on how to program in R and how to use it for effective data analysis. The course covers practical issues that include programming, reading data, accessing R packages, writing R functions, debugging, profiling R code, and organizing and commenting R code. It includes examples and exercises.</li>
<li><a href="https://www.coursera.org/learn/data-analysis-r" target="_blank">https://www.coursera.org/learn/data-analysis-r</a>. This course introduces data analysis using R. It provides an overview of RStudio withR packages relevant to organize, analyze, visualize, and report data.</li>
<li><a href="https://www.coursera.org/learn/introducton-r-programming-data-science" target="_blank">https://www.coursera.org/learn/introducton-r-programming-data-science</a>. This is a basic course in R programming for data analysis.</li>
</ol>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="control-structures-loops-in-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="miscellaneous.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/29-Bioinformatics_literature_v5.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
