<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Session 23 – Biopython: GenomeDiagram | Bioinformatics Workshop Gitbook</title>
  <meta name="description" content="none" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Session 23 – Biopython: GenomeDiagram | Bioinformatics Workshop Gitbook" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="none" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Session 23 – Biopython: GenomeDiagram | Bioinformatics Workshop Gitbook" />
  
  <meta name="twitter:description" content="none" />
  

<meta name="author" content="Juan C. Santos &amp; Randy Ortiz" />


<meta name="date" content="2024-02-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="biopython-mulitple-sequence-alignments.html"/>
<link rel="next" href="biopython-chromosome-modeling.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics Workshop Gitbook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html"><i class="fa fa-check"></i><b>1</b> – Getting Started with R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#why-do-we-use-r"><i class="fa fa-check"></i><b>1.1</b> Why do we use R?</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#get-a-good-text-editor"><i class="fa fa-check"></i><b>1.2</b> Get a good text editor</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-r"><i class="fa fa-check"></i><b>1.4</b> Installing R</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#annotating-code"><i class="fa fa-check"></i><b>1.5</b> Annotating code</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#defining-a-working-directory-in-r"><i class="fa fa-check"></i><b>1.6</b> Defining a working directory in R</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html"><i class="fa fa-check"></i><b>2</b> – Using R, Installing Packages and Importing/Exporting Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#your-first-session"><i class="fa fa-check"></i><b>2.1</b> Your first session</a></li>
<li class="chapter" data-level="2.2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#doing-some-base-functions-and-getting-help"><i class="fa fa-check"></i><b>2.2</b> Doing some base functions and getting help</a></li>
<li class="chapter" data-level="2.3" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#functions-between-two-or-more-vectors"><i class="fa fa-check"></i><b>2.3</b> Functions between two or more vectors</a></li>
<li class="chapter" data-level="2.4" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#about-r-packages"><i class="fa fa-check"></i><b>2.4</b> About R-packages</a></li>
<li class="chapter" data-level="2.5" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#cran-task-views-expert-opinion-on-useful-packages"><i class="fa fa-check"></i><b>2.5</b> CRAN Task Views – Expert opinion on useful packages</a></li>
<li class="chapter" data-level="2.6" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#installing-packages"><i class="fa fa-check"></i><b>2.6</b> Installing packages</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html"><i class="fa fa-check"></i><b>3</b> – Basic Data Structures in R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#scalars-and-vectors"><i class="fa fa-check"></i><b>3.1</b> Scalars and vectors</a></li>
<li class="chapter" data-level="3.2" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#arrays"><i class="fa fa-check"></i><b>3.3</b> Arrays</a></li>
<li class="chapter" data-level="3.4" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a></li>
<li class="chapter" data-level="3.5" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tables"><i class="fa fa-check"></i><b>3.5</b> Tables</a></li>
<li class="chapter" data-level="3.6" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#lists"><i class="fa fa-check"></i><b>3.6</b> Lists</a></li>
<li class="chapter" data-level="3.7" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-tables"><i class="fa fa-check"></i><b>3.7</b> Data Tables</a></li>
<li class="chapter" data-level="3.8" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tibbles"><i class="fa fa-check"></i><b>3.8</b> Tibbles</a></li>
<li class="chapter" data-level="3.9" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#dnastringsets-rnastringsets-and-aastringsets"><i class="fa fa-check"></i><b>3.9</b> DNAStringSets, RNAStringSets and AAStringSets</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html"><i class="fa fa-check"></i><b>4</b> – Text Editing and Data Transformations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#working-with-character-strings"><i class="fa fa-check"></i><b>4.1</b> Working with character strings</a></li>
<li class="chapter" data-level="4.2" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#regular-expressions-regex"><i class="fa fa-check"></i><b>4.2</b> Regular expressions (Regex)</a></li>
<li class="chapter" data-level="4.3" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#comparing-ordering-and-simplifying-vectors"><i class="fa fa-check"></i><b>4.3</b> Comparing, ordering and simplifying vectors</a></li>
<li class="chapter" data-level="4.4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#useful-functions-of-stringr"><i class="fa fa-check"></i><b>4.4</b> Useful functions of <code>stringr</code></a></li>
<li class="chapter" data-level="4.5" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#data-transformations"><i class="fa fa-check"></i><b>4.5</b> Data transformations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html"><i class="fa fa-check"></i><b>5</b> – Getting Biological Data from Public Repositories</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#ncbi-databases-genbank-and-others"><i class="fa fa-check"></i><b>5.1</b> NCBI databases: GenBank and others</a></li>
<li class="chapter" data-level="5.2" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#accessing-ncbi-within-r-rentrez"><i class="fa fa-check"></i><b>5.2</b> Accessing NCBI within R: <code>rentrez</code></a></li>
<li class="chapter" data-level="5.3" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#retrieving-pubmed-references"><i class="fa fa-check"></i><b>5.3</b> Retrieving PubMed references</a></li>
<li class="chapter" data-level="5.4" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#comparing-sequences-blast"><i class="fa fa-check"></i><b>5.4</b> Comparing sequences: BLAST</a></li>
<li class="chapter" data-level="5.5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-uniprot"><i class="fa fa-check"></i><b>5.5</b> Other databases: UniProt</a></li>
<li class="chapter" data-level="5.6" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-ensembl"><i class="fa fa-check"></i><b>5.6</b> Other databases: Ensembl</a></li>
<li class="chapter" data-level="5.7" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-taxonomy-phylogenetic-trees-and-species-distributions"><i class="fa fa-check"></i><b>5.7</b> Other databases: Taxonomy, phylogenetic trees and species distributions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html"><i class="fa fa-check"></i><b>6</b> – Basic Statistics in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#exploring-your-data"><i class="fa fa-check"></i><b>6.1</b> Exploring your data</a></li>
<li class="chapter" data-level="6.2" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#sub-setting-your-dataset"><i class="fa fa-check"></i><b>6.2</b> Sub-setting your dataset</a></li>
<li class="chapter" data-level="6.3" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#transposing-rotating-your-dataset"><i class="fa fa-check"></i><b>6.3</b> Transposing (rotating) your dataset</a></li>
<li class="chapter" data-level="6.4" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#dealing-with-missing-na-values"><i class="fa fa-check"></i><b>6.4</b> Dealing with missing (NA) values</a></li>
<li class="chapter" data-level="6.5" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#about-p-values"><i class="fa fa-check"></i><b>6.5</b> About p-values</a></li>
<li class="chapter" data-level="6.6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#how-to-read-p-values"><i class="fa fa-check"></i><b>6.6</b> How to read p-values</a></li>
<li class="chapter" data-level="6.7" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#multiple-testing-corrections"><i class="fa fa-check"></i><b>6.7</b> Multiple testing corrections</a></li>
<li class="chapter" data-level="6.8" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#normalizing-your-data"><i class="fa fa-check"></i><b>6.8</b> Normalizing your data</a></li>
<li class="chapter" data-level="6.9" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#what-test-should-i-use"><i class="fa fa-check"></i><b>6.9</b> What test should I use?</a></li>
<li class="chapter" data-level="6.10" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#type-of-variables-in-dataset"><i class="fa fa-check"></i><b>6.10</b> Type of variables in dataset</a></li>
<li class="chapter" data-level="6.11" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-means-t-test"><i class="fa fa-check"></i><b>6.11</b> Comparing means: t-test</a></li>
<li class="chapter" data-level="6.12" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#two-sample-t-test"><i class="fa fa-check"></i><b>6.12</b> Two sample t-test</a></li>
<li class="chapter" data-level="6.13" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-wilcoxon-signed-rank-test"><i class="fa fa-check"></i><b>6.13</b> Beyond t-test: Wilcoxon signed rank test</a></li>
<li class="chapter" data-level="6.14" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#tabulation-and-cross-tabulation"><i class="fa fa-check"></i><b>6.14</b> Tabulation and Cross-Tabulation</a></li>
<li class="chapter" data-level="6.15" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-binomial-test"><i class="fa fa-check"></i><b>6.15</b> Beyond t-test: Binomial test</a></li>
<li class="chapter" data-level="6.16" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-chi-square-goodness-of-fit"><i class="fa fa-check"></i><b>6.16</b> Beyond t-test: Chi-square goodness of fit</a></li>
<li class="chapter" data-level="6.17" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-mcnemars-test"><i class="fa fa-check"></i><b>6.17</b> Beyond t-test: McNemar’s test</a></li>
<li class="chapter" data-level="6.18" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-fishers-exact-test"><i class="fa fa-check"></i><b>6.18</b> Beyond t-test: Fisher’s exact test</a></li>
<li class="chapter" data-level="6.19" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#correlations"><i class="fa fa-check"></i><b>6.19</b> Correlations</a></li>
<li class="chapter" data-level="6.20" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-distributions"><i class="fa fa-check"></i><b>6.20</b> Comparing distributions</a></li>
<li class="chapter" data-level="6.21" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#bootstraps"><i class="fa fa-check"></i><b>6.21</b> Bootstraps</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html"><i class="fa fa-check"></i><b>7</b> – Regressions and ANOVA in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#linear-regressions"><i class="fa fa-check"></i><b>7.1</b> Linear regressions</a></li>
<li class="chapter" data-level="7.2" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anatomy-of-a-linear-regression"><i class="fa fa-check"></i><b>7.2</b> Anatomy of a linear regression</a></li>
<li class="chapter" data-level="7.3" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#multiple-linear-regressions"><i class="fa fa-check"></i><b>7.3</b> Multiple linear regressions</a></li>
<li class="chapter" data-level="7.4" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#diagnostic-plots-for-regressions"><i class="fa fa-check"></i><b>7.4</b> Diagnostic plots for regressions</a></li>
<li class="chapter" data-level="7.5" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-influential-measurements"><i class="fa fa-check"></i><b>7.5</b> Regression influential measurements</a></li>
<li class="chapter" data-level="7.6" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-anova-coefficients-ci"><i class="fa fa-check"></i><b>7.6</b> Regression (ANOVA, coefficients, CI)</a></li>
<li class="chapter" data-level="7.7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#selecting-predictors"><i class="fa fa-check"></i><b>7.7</b> Selecting predictors</a></li>
<li class="chapter" data-level="7.8" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-predictions"><i class="fa fa-check"></i><b>7.8</b> Regression predictions</a></li>
<li class="chapter" data-level="7.9" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions"><i class="fa fa-check"></i><b>7.9</b> Logistic regressions</a></li>
<li class="chapter" data-level="7.10" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions-requirements"><i class="fa fa-check"></i><b>7.10</b> Logistic regressions requirements</a></li>
<li class="chapter" data-level="7.11" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#non-linear-regressions"><i class="fa fa-check"></i><b>7.11</b> Non-linear regressions</a></li>
<li class="chapter" data-level="7.12" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-regressions"><i class="fa fa-check"></i><b>7.12</b> Toy datasets for regressions</a></li>
<li class="chapter" data-level="7.13" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova"><i class="fa fa-check"></i><b>7.13</b> ANOVA</a></li>
<li class="chapter" data-level="7.14" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#one-way-anova"><i class="fa fa-check"></i><b>7.14</b> One-way ANOVA</a></li>
<li class="chapter" data-level="7.15" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#two-way-anova"><i class="fa fa-check"></i><b>7.15</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="7.16" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova-post-hoc-tests"><i class="fa fa-check"></i><b>7.16</b> ANOVA Post-hoc tests</a></li>
<li class="chapter" data-level="7.17" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#beyond-anova-kruskal-wallis-test"><i class="fa fa-check"></i><b>7.17</b> Beyond ANOVA: Kruskal-Wallis Test</a></li>
<li class="chapter" data-level="7.18" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-anova"><i class="fa fa-check"></i><b>7.18</b> Toy datasets for ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html"><i class="fa fa-check"></i><b>8</b> – Multidimensional Data in R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#principal-component-analysis"><i class="fa fa-check"></i><b>8.1</b> Principal Component Analysis</a></li>
<li class="chapter" data-level="8.2" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#multidimensional-scaling"><i class="fa fa-check"></i><b>8.2</b> Multidimensional scaling</a></li>
<li class="chapter" data-level="8.3" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-dfa"><i class="fa fa-check"></i><b>8.3</b> Cluster analysis: DFA</a></li>
<li class="chapter" data-level="8.4" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-k-means-unsupervised"><i class="fa fa-check"></i><b>8.4</b> Cluster analysis: <em>k</em>-means (unsupervised)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html"><i class="fa fa-check"></i><b>9</b> – Graphics and Colors</a>
<ul>
<li class="chapter" data-level="9.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#avoid-bad-graphics"><i class="fa fa-check"></i><b>9.1</b> Avoid bad graphics</a></li>
<li class="chapter" data-level="9.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#graphics-in-r-ggplot2"><i class="fa fa-check"></i><b>9.2</b> Graphics in R – ggplot2</a></li>
<li class="chapter" data-level="9.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#preparing-your-dataset"><i class="fa fa-check"></i><b>9.3</b> Preparing your dataset</a></li>
<li class="chapter" data-level="9.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#color-palettes-online"><i class="fa fa-check"></i><b>9.4</b> Color palettes online</a></li>
<li class="chapter" data-level="9.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggplot2-default-palette"><i class="fa fa-check"></i><b>9.5</b> <code>ggplot2</code> default palette</a></li>
<li class="chapter" data-level="9.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#heatmaps-or-tiled-charts"><i class="fa fa-check"></i><b>9.6</b> Heatmaps or tiled charts</a></li>
<li class="chapter" data-level="9.7" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#r-packages-about-color"><i class="fa fa-check"></i><b>9.7</b> R-packages about color</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#viridis-package"><i class="fa fa-check"></i><b>9.7.1</b> <code>viridis</code> package</a></li>
<li class="chapter" data-level="9.7.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcolorbrewer-package"><i class="fa fa-check"></i><b>9.7.2</b> <code>RColorBrewer</code> package</a></li>
<li class="chapter" data-level="9.7.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggsci-package"><i class="fa fa-check"></i><b>9.7.3</b> <code>ggsci</code> package</a></li>
<li class="chapter" data-level="9.7.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#wesanderson-package"><i class="fa fa-check"></i><b>9.7.4</b> <code>wesanderson</code> package</a></li>
<li class="chapter" data-level="9.7.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcartocolor-package"><i class="fa fa-check"></i><b>9.7.5</b> <code>rcartocolor</code> package</a></li>
<li class="chapter" data-level="9.7.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#colorspace-package"><i class="fa fa-check"></i><b>9.7.6</b> <code>colorspace</code> package</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#making-your-own-palette"><i class="fa fa-check"></i><b>9.8</b> Making your own palette</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="basic-graphics.html"><a href="basic-graphics.html"><i class="fa fa-check"></i><b>10</b> – Basic Graphics</a>
<ul>
<li class="chapter" data-level="10.1" data-path="basic-graphics.html"><a href="basic-graphics.html#what-would-you-like-to-show"><i class="fa fa-check"></i><b>10.1</b> What would you like to show?</a></li>
<li class="chapter" data-level="10.2" data-path="basic-graphics.html"><a href="basic-graphics.html#bar-charts"><i class="fa fa-check"></i><b>10.2</b> Bar charts</a></li>
<li class="chapter" data-level="10.3" data-path="basic-graphics.html"><a href="basic-graphics.html#stacked-bar-charts"><i class="fa fa-check"></i><b>10.3</b> Stacked bar charts</a></li>
<li class="chapter" data-level="10.4" data-path="basic-graphics.html"><a href="basic-graphics.html#histograms"><i class="fa fa-check"></i><b>10.4</b> Histograms</a></li>
<li class="chapter" data-level="10.5" data-path="basic-graphics.html"><a href="basic-graphics.html#multiple-histograms"><i class="fa fa-check"></i><b>10.5</b> Multiple histograms</a></li>
<li class="chapter" data-level="10.6" data-path="basic-graphics.html"><a href="basic-graphics.html#density-plots"><i class="fa fa-check"></i><b>10.6</b> Density plots</a></li>
<li class="chapter" data-level="10.7" data-path="basic-graphics.html"><a href="basic-graphics.html#box-plots"><i class="fa fa-check"></i><b>10.7</b> Box plots</a></li>
<li class="chapter" data-level="10.8" data-path="basic-graphics.html"><a href="basic-graphics.html#violin-plots"><i class="fa fa-check"></i><b>10.8</b> Violin plots</a></li>
<li class="chapter" data-level="10.9" data-path="basic-graphics.html"><a href="basic-graphics.html#scatterplots"><i class="fa fa-check"></i><b>10.9</b> Scatterplots</a></li>
<li class="chapter" data-level="10.10" data-path="basic-graphics.html"><a href="basic-graphics.html#lines-and-scatterplots"><i class="fa fa-check"></i><b>10.10</b> Lines and Scatterplots</a></li>
<li class="chapter" data-level="10.11" data-path="basic-graphics.html"><a href="basic-graphics.html#save-your-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>10.11</b> Save your plots as PDF or PNG</a></li>
<li class="chapter" data-level="10.12" data-path="basic-graphics.html"><a href="basic-graphics.html#ggplot2-add-on-packages-treemapify"><i class="fa fa-check"></i><b>10.12</b> ggplot2 add-on packages: <code>treemapify</code></a></li>
<li class="chapter" data-level="10.13" data-path="basic-graphics.html"><a href="basic-graphics.html#other-resources"><i class="fa fa-check"></i><b>10.13</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="sequence-alignments.html"><a href="sequence-alignments.html"><i class="fa fa-check"></i><b>11</b> – Sequence Alignments</a>
<ul>
<li class="chapter" data-level="11.1" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-using-rentrez"><i class="fa fa-check"></i><b>11.1</b> Retrieving sequences using ‘rentrez’</a></li>
<li class="chapter" data-level="11.2" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-manually-from-ncbi"><i class="fa fa-check"></i><b>11.2</b> Retrieving sequences manually from NCBI</a></li>
<li class="chapter" data-level="11.3" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-accession-numbers"><i class="fa fa-check"></i><b>11.3</b> Retrieving accession numbers</a></li>
<li class="chapter" data-level="11.4" data-path="sequence-alignments.html"><a href="sequence-alignments.html#alignment-with-decipher"><i class="fa fa-check"></i><b>11.4</b> Alignment with <code>DECIPHER</code></a></li>
<li class="chapter" data-level="11.5" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualize-alignment"><i class="fa fa-check"></i><b>11.5</b> Visualize alignment</a></li>
<li class="chapter" data-level="11.6" data-path="sequence-alignments.html"><a href="sequence-alignments.html#save-alignment-to-file"><i class="fa fa-check"></i><b>11.6</b> Save alignment to file</a></li>
<li class="chapter" data-level="11.7" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-and-align-several-markers"><i class="fa fa-check"></i><b>11.7</b> Concatenate and align several markers</a></li>
<li class="chapter" data-level="11.8" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-data.table"><i class="fa fa-check"></i><b>11.8</b> Concatenate with <code>data.table</code></a></li>
<li class="chapter" data-level="11.9" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-biostrings"><i class="fa fa-check"></i><b>11.9</b> Concatenate with <code>Biostrings</code></a></li>
<li class="chapter" data-level="11.10" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualizing-alignments-in-r"><i class="fa fa-check"></i><b>11.10</b> Visualizing alignments in R</a></li>
<li class="chapter" data-level="11.11" data-path="sequence-alignments.html"><a href="sequence-alignments.html#other-sequence-alignment-software"><i class="fa fa-check"></i><b>11.11</b> Other sequence alignment software</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html"><i class="fa fa-check"></i><b>12</b> – Estimating Phylogenetic Trees</a>
<ul>
<li class="chapter" data-level="12.1" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#phylogeny-inference-with-iqtree2"><i class="fa fa-check"></i><b>12.1</b> Phylogeny inference with <code>iqtree2</code></a></li>
<li class="chapter" data-level="12.2" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-iqtree2-in-macos"><i class="fa fa-check"></i><b>12.2</b> Installing <code>iqtree2</code> in macOS</a></li>
<li class="chapter" data-level="12.3" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-and-running-iqtree2-in-a-pc"><i class="fa fa-check"></i><b>12.3</b> Installing and running <code>iqtree2</code> in a PC</a></li>
<li class="chapter" data-level="12.4" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#loading-the-iqtree_runner_r-function"><i class="fa fa-check"></i><b>12.4</b> Loading the <code>iqtree_runner_R</code> function</a></li>
<li class="chapter" data-level="12.5" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#running-iqtree2-from-r"><i class="fa fa-check"></i><b>12.5</b> Running <code>iqtree2</code> from R</a></li>
<li class="chapter" data-level="12.6" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-no-partitions"><i class="fa fa-check"></i><b>12.6</b> One marker and no partitions</a></li>
<li class="chapter" data-level="12.7" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-with-partitions"><i class="fa fa-check"></i><b>12.7</b> One marker and with partitions</a></li>
<li class="chapter" data-level="12.8" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#bootstrap-support"><i class="fa fa-check"></i><b>12.8</b> Bootstrap support</a></li>
<li class="chapter" data-level="12.9" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#many-markers-and-with-partitions"><i class="fa fa-check"></i><b>12.9</b> Many markers and with partitions</a></li>
<li class="chapter" data-level="12.10" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#other-sequence-phylogenetic-tree-estimation-software"><i class="fa fa-check"></i><b>12.10</b> Other sequence phylogenetic tree estimation software</a></li>
<li class="chapter" data-level="12.11" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#drawing-phylogenetic-trees-with-ggtree"><i class="fa fa-check"></i><b>12.11</b> Drawing phylogenetic trees with <code>ggtree</code></a></li>
<li class="chapter" data-level="12.12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#save-ggtree-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>12.12</b> Save ggtree plots as PDF or PNG</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html"><i class="fa fa-check"></i><b>13</b> – Mining from a reference dataset</a>
<ul>
<li class="chapter" data-level="13.1" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#a-worked-example-using-the-mtcars-dataset"><i class="fa fa-check"></i><b>13.1</b> A worked example using the <code>mtcars</code> dataset</a></li>
<li class="chapter" data-level="13.2" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-reference-dataset"><i class="fa fa-check"></i><b>13.2</b> Build a reference dataset</a></li>
<li class="chapter" data-level="13.3" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-sampling-dataset"><i class="fa fa-check"></i><b>13.3</b> Build a sampling dataset</a></li>
<li class="chapter" data-level="13.4" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#extracting-from-reference-dataset"><i class="fa fa-check"></i><b>13.4</b> Extracting from reference dataset</a></li>
<li class="chapter" data-level="13.5" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#plots-of-the-extracted-data"><i class="fa fa-check"></i><b>13.5</b> Plots of the extracted data</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="introduction-to-python.html"><a href="introduction-to-python.html"><i class="fa fa-check"></i><b>14</b> – Introduction to Python</a>
<ul>
<li class="chapter" data-level="14.1" data-path="introduction-to-python.html"><a href="introduction-to-python.html#welcome-to-python-why-python"><i class="fa fa-check"></i><b>14.1</b> Welcome to Python! Why Python?</a></li>
<li class="chapter" data-level="14.2" data-path="introduction-to-python.html"><a href="introduction-to-python.html#which-python-should-i-install"><i class="fa fa-check"></i><b>14.2</b> Which Python should I install?</a></li>
<li class="chapter" data-level="14.3" data-path="introduction-to-python.html"><a href="introduction-to-python.html#mac-os-x-python-installations"><i class="fa fa-check"></i><b>14.3</b> Mac OS X Python Installations</a></li>
<li class="chapter" data-level="14.4" data-path="introduction-to-python.html"><a href="introduction-to-python.html#windows-python-installations"><i class="fa fa-check"></i><b>14.4</b> Windows Python Installations</a></li>
<li class="chapter" data-level="14.5" data-path="introduction-to-python.html"><a href="introduction-to-python.html#install-an-ide-or-integrated-development-environment"><i class="fa fa-check"></i><b>14.5</b> Install an IDE, or integrated development environment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html"><i class="fa fa-check"></i><b>15</b> – Python: Keywords and Identifiers</a>
<ul>
<li class="chapter" data-level="15.1" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#keywords-in-python"><i class="fa fa-check"></i><b>15.1</b> Keywords in Python</a></li>
<li class="chapter" data-level="15.2" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#identifiers-in-python"><i class="fa fa-check"></i><b>15.2</b> Identifiers in Python</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html"><i class="fa fa-check"></i><b>16</b> – Python: Statements and Expressions</a>
<ul>
<li class="chapter" data-level="16.1" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#what-are-statements"><i class="fa fa-check"></i><b>16.1</b> What are statements?</a></li>
<li class="chapter" data-level="16.2" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#expressions-in-python"><i class="fa fa-check"></i><b>16.2</b> Expressions in Python</a></li>
<li class="chapter" data-level="16.3" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#simple-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.3</b> Simple Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.4" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#augmented-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.4</b> Augmented Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.5" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#explicit-and-implicit-line-continuation"><i class="fa fa-check"></i><b>16.5</b> Explicit and Implicit Line Continuation</a></li>
<li class="chapter" data-level="16.6" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#indentation-in-python-is-crucial-to-proper-code-function"><i class="fa fa-check"></i><b>16.6</b> Indentation in Python is Crucial to Proper Code function</a></li>
<li class="chapter" data-level="16.7" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#comments"><i class="fa fa-check"></i><b>16.7</b> Comments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="python-data-types.html"><a href="python-data-types.html"><i class="fa fa-check"></i><b>17</b> – Python: Data Types</a>
<ul>
<li class="chapter" data-level="17.1" data-path="python-data-types.html"><a href="python-data-types.html#booleans"><i class="fa fa-check"></i><b>17.1</b> Booleans</a></li>
<li class="chapter" data-level="17.2" data-path="python-data-types.html"><a href="python-data-types.html#numbers-int-float-and-complex"><i class="fa fa-check"></i><b>17.2</b> Numbers: int, float and complex</a></li>
<li class="chapter" data-level="17.3" data-path="python-data-types.html"><a href="python-data-types.html#strings"><i class="fa fa-check"></i><b>17.3</b> Strings</a></li>
<li class="chapter" data-level="17.4" data-path="python-data-types.html"><a href="python-data-types.html#bytes"><i class="fa fa-check"></i><b>17.4</b> Bytes</a></li>
<li class="chapter" data-level="17.5" data-path="python-data-types.html"><a href="python-data-types.html#lists-1"><i class="fa fa-check"></i><b>17.5</b> Lists</a></li>
<li class="chapter" data-level="17.6" data-path="python-data-types.html"><a href="python-data-types.html#tuples"><i class="fa fa-check"></i><b>17.6</b> Tuples</a></li>
<li class="chapter" data-level="17.7" data-path="python-data-types.html"><a href="python-data-types.html#sets"><i class="fa fa-check"></i><b>17.7</b> Sets</a></li>
<li class="chapter" data-level="17.8" data-path="python-data-types.html"><a href="python-data-types.html#dictionary"><i class="fa fa-check"></i><b>17.8</b> Dictionary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="python-strings.html"><a href="python-strings.html"><i class="fa fa-check"></i><b>18</b> – Python: Strings</a>
<ul>
<li class="chapter" data-level="18.1" data-path="python-strings.html"><a href="python-strings.html#string-operators"><i class="fa fa-check"></i><b>18.1</b> String Operators</a></li>
<li class="chapter" data-level="18.2" data-path="python-strings.html"><a href="python-strings.html#other-string-modifiers"><i class="fa fa-check"></i><b>18.2</b> Other String Modifiers</a></li>
<li class="chapter" data-level="18.3" data-path="python-strings.html"><a href="python-strings.html#format-characters-in-strings"><i class="fa fa-check"></i><b>18.3</b> Format-Characters in Strings</a></li>
<li class="chapter" data-level="18.4" data-path="python-strings.html"><a href="python-strings.html#comparisons-with-strings"><i class="fa fa-check"></i><b>18.4</b> Comparisons with Strings</a></li>
<li class="chapter" data-level="18.5" data-path="python-strings.html"><a href="python-strings.html#padding-strings"><i class="fa fa-check"></i><b>18.5</b> Padding strings</a></li>
<li class="chapter" data-level="18.6" data-path="python-strings.html"><a href="python-strings.html#search-for-words-in-string-and-output-location"><i class="fa fa-check"></i><b>18.6</b> Search for words in string and output location</a></li>
<li class="chapter" data-level="18.7" data-path="python-strings.html"><a href="python-strings.html#seek-and-replace-in-strings"><i class="fa fa-check"></i><b>18.7</b> Seek and replace in strings</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="python-operators.html"><a href="python-operators.html"><i class="fa fa-check"></i><b>19</b> – Python: Operators</a>
<ul>
<li class="chapter" data-level="19.1" data-path="python-operators.html"><a href="python-operators.html#many-types-of-operators"><i class="fa fa-check"></i><b>19.1</b> Many types of operators</a></li>
<li class="chapter" data-level="19.2" data-path="python-operators.html"><a href="python-operators.html#arithmetic-operators"><i class="fa fa-check"></i><b>19.2</b> Arithmetic operators</a></li>
<li class="chapter" data-level="19.3" data-path="python-operators.html"><a href="python-operators.html#comparison-operators"><i class="fa fa-check"></i><b>19.3</b> Comparison operators</a></li>
<li class="chapter" data-level="19.4" data-path="python-operators.html"><a href="python-operators.html#logical-operators"><i class="fa fa-check"></i><b>19.4</b> Logical operators</a></li>
<li class="chapter" data-level="19.5" data-path="python-operators.html"><a href="python-operators.html#bitwise-operators"><i class="fa fa-check"></i><b>19.5</b> Bitwise operators</a></li>
<li class="chapter" data-level="19.6" data-path="python-operators.html"><a href="python-operators.html#identity-operators"><i class="fa fa-check"></i><b>19.6</b> Identity Operators</a></li>
<li class="chapter" data-level="19.7" data-path="python-operators.html"><a href="python-operators.html#membership-operators"><i class="fa fa-check"></i><b>19.7</b> Membership operators</a></li>
<li class="chapter" data-level="19.8" data-path="python-operators.html"><a href="python-operators.html#what-is-the-order-of-operations-with-operators"><i class="fa fa-check"></i><b>19.8</b> What is the order of operations with operators?</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="biopython.html"><a href="biopython.html"><i class="fa fa-check"></i><b>20</b> – Biopython</a>
<ul>
<li class="chapter" data-level="20.1" data-path="biopython.html"><a href="biopython.html#introduction-to-biopython"><i class="fa fa-check"></i><b>20.1</b> Introduction to Biopython</a></li>
<li class="chapter" data-level="20.2" data-path="biopython.html"><a href="biopython.html#biopython-installation"><i class="fa fa-check"></i><b>20.2</b> Biopython Installation</a></li>
<li class="chapter" data-level="20.3" data-path="biopython.html"><a href="biopython.html#getting-started"><i class="fa fa-check"></i><b>20.3</b> Getting Started</a></li>
<li class="chapter" data-level="20.4" data-path="biopython.html"><a href="biopython.html#parsing-my-fasta"><i class="fa fa-check"></i><b>20.4</b> Parsing my FASTA</a></li>
<li class="chapter" data-level="20.5" data-path="biopython.html"><a href="biopython.html#parsing-my-genbank-or-genpept"><i class="fa fa-check"></i><b>20.5</b> Parsing my GenBank or GenPept</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="biopython-sequences.html"><a href="biopython-sequences.html"><i class="fa fa-check"></i><b>21</b> – Biopython: Sequences</a>
<ul>
<li class="chapter" data-level="21.1" data-path="biopython-sequences.html"><a href="biopython-sequences.html#sequence-objects"><i class="fa fa-check"></i><b>21.1</b> Sequence Objects</a></li>
<li class="chapter" data-level="21.2" data-path="biopython-sequences.html"><a href="biopython-sequences.html#concatenation-with-sequences"><i class="fa fa-check"></i><b>21.2</b> Concatenation with Sequences</a></li>
<li class="chapter" data-level="21.3" data-path="biopython-sequences.html"><a href="biopython-sequences.html#transcription-dna-to-mrna"><i class="fa fa-check"></i><b>21.3</b> Transcription: DNA to mRNA</a></li>
<li class="chapter" data-level="21.4" data-path="biopython-sequences.html"><a href="biopython-sequences.html#translation-rna-to-prot"><i class="fa fa-check"></i><b>21.4</b> Translation: RNA to PROT</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html"><i class="fa fa-check"></i><b>22</b> – Biopython: Mulitple Sequence Alignments</a>
<ul>
<li class="chapter" data-level="22.1" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwise2"><i class="fa fa-check"></i><b>22.1</b> pairwise2</a></li>
<li class="chapter" data-level="22.2" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwisealigner"><i class="fa fa-check"></i><b>22.2</b> PairwiseAligner</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html"><i class="fa fa-check"></i><b>23</b> – Biopython: GenomeDiagram</a>
<ul>
<li class="chapter" data-level="23.1" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#linearcircular-gene-diagrams"><i class="fa fa-check"></i><b>23.1</b> Linear/Circular Gene Diagrams</a></li>
<li class="chapter" data-level="23.2" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#diagram-readability"><i class="fa fa-check"></i><b>23.2</b> Diagram Readability</a></li>
<li class="chapter" data-level="23.3" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#box-shape"><i class="fa fa-check"></i><b>23.3</b> Box Shape</a></li>
<li class="chapter" data-level="23.4" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#edit-box-shape-arrows"><i class="fa fa-check"></i><b>23.4</b> Edit Box Shape: Arrows</a></li>
<li class="chapter" data-level="23.5" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#restriction-map"><i class="fa fa-check"></i><b>23.5</b> Restriction Map</a></li>
<li class="chapter" data-level="23.6" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#tracks-on-tracks"><i class="fa fa-check"></i><b>23.6</b> Tracks on Tracks</a></li>
<li class="chapter" data-level="23.7" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#cross-link-data"><i class="fa fa-check"></i><b>23.7</b> Cross-link Data</a></li>
<li class="chapter" data-level="23.8" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#clean-cross-link"><i class="fa fa-check"></i><b>23.8</b> Clean Cross-Link</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html"><i class="fa fa-check"></i><b>24</b> – Biopython: Chromosome Modeling</a>
<ul>
<li class="chapter" data-level="24.1" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html#gene-feature-modeling"><i class="fa fa-check"></i><b>24.1</b> Gene Feature Modeling</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html"><i class="fa fa-check"></i><b>25</b> – Biopython: Sequence Parsing with Plots</a>
<ul>
<li class="chapter" data-level="25.1" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#histograms-1"><i class="fa fa-check"></i><b>25.1</b> Histograms</a></li>
<li class="chapter" data-level="25.2" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#gc-content"><i class="fa fa-check"></i><b>25.2</b> GC Content</a></li>
<li class="chapter" data-level="25.3" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#nt-alignment-dot-plot"><i class="fa fa-check"></i><b>25.3</b> Nt Alignment Dot Plot</a></li>
<li class="chapter" data-level="25.4" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#bigger-dots-for-the-dot-plot"><i class="fa fa-check"></i><b>25.4</b> Bigger Dots for the Dot Plot</a></li>
<li class="chapter" data-level="25.5" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#fastq-we-did-not-forget-you"><i class="fa fa-check"></i><b>25.5</b> FASTQ we did not forget you</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html"><i class="fa fa-check"></i><b>26</b> – Python: Web Scraping w/ BeautifulSoup</a>
<ul>
<li class="chapter" data-level="26.1" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#introduction"><i class="fa fa-check"></i><b>26.1</b> Introduction</a></li>
<li class="chapter" data-level="26.2" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#installation"><i class="fa fa-check"></i><b>26.2</b> Installation</a></li>
<li class="chapter" data-level="26.3" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#getting-started-1"><i class="fa fa-check"></i><b>26.3</b> Getting started</a></li>
<li class="chapter" data-level="26.4" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-all-matching-tags"><i class="fa fa-check"></i><b>26.4</b> Mining all matching tags</a></li>
<li class="chapter" data-level="26.5" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-tags-by-class-or-id"><i class="fa fa-check"></i><b>26.5</b> Mining tags by class or id</a></li>
<li class="chapter" data-level="26.6" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#use-css-selectors-for-positional-mining"><i class="fa fa-check"></i><b>26.6</b> Use CSS Selectors for positional mining</a></li>
<li class="chapter" data-level="26.7" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#weather-page-example"><i class="fa fa-check"></i><b>26.7</b> Weather Page Example</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="python-subprocess.html"><a href="python-subprocess.html"><i class="fa fa-check"></i><b>27</b> – Python Subprocess</a>
<ul>
<li class="chapter" data-level="27.1" data-path="python-subprocess.html"><a href="python-subprocess.html#introduction-and-installation"><i class="fa fa-check"></i><b>27.1</b> Introduction and Installation</a></li>
<li class="chapter" data-level="27.2" data-path="python-subprocess.html"><a href="python-subprocess.html#creating-your-first-python-script-w-bbtools"><i class="fa fa-check"></i><b>27.2</b> Creating your first python script w/ BBTools</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html"><i class="fa fa-check"></i><b>28</b> – Control Structures &amp; Loops in R</a>
<ul>
<li class="chapter" data-level="28.1" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#controling-the-flow"><i class="fa fa-check"></i><b>28.1</b> Controling the flow</a></li>
<li class="chapter" data-level="28.2" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if"><i class="fa fa-check"></i><b>28.2</b> Conditional <code>if</code></a></li>
<li class="chapter" data-level="28.3" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if-and-else"><i class="fa fa-check"></i><b>28.3</b> Conditional <code>if</code> and <code>else</code></a></li>
<li class="chapter" data-level="28.4" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#nested-conditionals-if-else-and-if-else"><i class="fa fa-check"></i><b>28.4</b> Nested conditionals <code>if</code>, <code>else</code>, and <code>if else</code></a></li>
<li class="chapter" data-level="28.5" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-ifelse"><i class="fa fa-check"></i><b>28.5</b> Conditional <code>ifelse</code></a></li>
<li class="chapter" data-level="28.6" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#loops-that-control-iterations-and-repeations"><i class="fa fa-check"></i><b>28.6</b> Loops that control iterations and repeations</a></li>
<li class="chapter" data-level="28.7" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-for-loop"><i class="fa fa-check"></i><b>28.7</b> The <code>for</code> Loop</a></li>
<li class="chapter" data-level="28.8" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-while-loop"><i class="fa fa-check"></i><b>28.8</b> The <code>while</code> Loop</a></li>
<li class="chapter" data-level="28.9" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-repeat-loop"><i class="fa fa-check"></i><b>28.9</b> The <code>repeat</code> Loop</a></li>
<li class="chapter" data-level="28.10" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-next-and-break-statements"><i class="fa fa-check"></i><b>28.10</b> The <code>next</code> and <code>break</code> statements</a></li>
<li class="chapter" data-level="28.11" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functions"><i class="fa fa-check"></i><b>28.11</b> Other loop functions</a></li>
<li class="chapter" data-level="28.12" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#lapply-functions"><i class="fa fa-check"></i><b>28.12</b> <code>lapply</code> functions</a></li>
<li class="chapter" data-level="28.13" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsapply"><i class="fa fa-check"></i><b>28.13</b> Other loop functions:<code>apply</code></a></li>
<li class="chapter" data-level="28.14" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsmapply"><i class="fa fa-check"></i><b>28.14</b> Other loop functions:<code>mapply</code></a></li>
<li class="chapter" data-level="28.15" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionstapply"><i class="fa fa-check"></i><b>28.15</b> Other loop functions:<code>tapply</code></a></li>
<li class="chapter" data-level="28.16" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionssplit"><i class="fa fa-check"></i><b>28.16</b> Other loop functions:<code>split</code></a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html"><i class="fa fa-check"></i><b>29</b> – Bioinformatics Literature</a>
<ul>
<li class="chapter" data-level="29.1" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#peer-reviewed-bioinformatics-journals"><i class="fa fa-check"></i><b>29.1</b> Peer-reviewed bioinformatics journals</a></li>
<li class="chapter" data-level="29.2" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#access-to-journals-articles"><i class="fa fa-check"></i><b>29.2</b> Access to journals articles</a></li>
<li class="chapter" data-level="29.3" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#informal-access-to-code-preprints-and-vignettes"><i class="fa fa-check"></i><b>29.3</b> Informal access to code: Preprints and vignettes</a></li>
<li class="chapter" data-level="29.4" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#online-bioinformatics-courses"><i class="fa fa-check"></i><b>29.4</b> Online bioinformatics courses</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html"><i class="fa fa-check"></i><b>30</b> – Importing/Exporting Data into R</a>
<ul>
<li class="chapter" data-level="30.1" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#typing-directly"><i class="fa fa-check"></i><b>30.1</b> Typing directly</a></li>
<li class="chapter" data-level="30.2" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#summary-versus-raw-data"><i class="fa fa-check"></i><b>30.2</b> Summary versus raw data</a></li>
<li class="chapter" data-level="30.3" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#downloading-data-files-github"><i class="fa fa-check"></i><b>30.3</b> Downloading data files (GitHub)</a></li>
<li class="chapter" data-level="30.4" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#importing-data-from-a-file"><i class="fa fa-check"></i><b>30.4</b> Importing data from a file</a></li>
<li class="chapter" data-level="30.5" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-to-a-file"><i class="fa fa-check"></i><b>30.5</b> Saving data to a file</a></li>
<li class="chapter" data-level="30.6" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-as-rdata-.rdata-.rda"><i class="fa fa-check"></i><b>30.6</b> Saving data as Rdata (.rdata, .rda)</a></li>
<li class="chapter" data-level="30.7" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#import-excel-formated-data"><i class="fa fa-check"></i><b>30.7</b> Import Excel formated data</a></li>
<li class="chapter" data-level="30.8" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#save-data-as-excel-formated-file"><i class="fa fa-check"></i><b>30.8</b> Save data as Excel formated file</a></li>
<li class="chapter" data-level="30.9" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#fixing-problems-while-importing-datasets"><i class="fa fa-check"></i><b>30.9</b> Fixing problems while importing datasets</a></li>
<li class="chapter" data-level="30.10" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#simplifying-embellished-excel-datasets"><i class="fa fa-check"></i><b>30.10</b> Simplifying embellished excel datasets</a></li>
<li class="chapter" data-level="30.11" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#accessing-online-data"><i class="fa fa-check"></i><b>30.11</b> Accessing online data</a></li>
<li class="chapter" data-level="30.12" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#data-from-clinicaltrials.gov"><i class="fa fa-check"></i><b>30.12</b> Data from clinicaltrials.gov</a></li>
<li class="chapter" data-level="30.13" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#data-scrapping-from-a-website"><i class="fa fa-check"></i><b>30.13</b> Data scrapping from a website</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="working-with-proteins.html"><a href="working-with-proteins.html"><i class="fa fa-check"></i><b>31</b> – Working with Proteins</a>
<ul>
<li class="chapter" data-level="31.1" data-path="working-with-proteins.html"><a href="working-with-proteins.html#sequences-in-fasta-format"><i class="fa fa-check"></i><b>31.1</b> Sequences in FASTA format</a></li>
<li class="chapter" data-level="31.2" data-path="working-with-proteins.html"><a href="working-with-proteins.html#fetching-aa-sequences-from-ncbi"><i class="fa fa-check"></i><b>31.2</b> Fetching AA sequences from NCBI</a></li>
<li class="chapter" data-level="31.3" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-aa-sequences-in-r"><i class="fa fa-check"></i><b>31.3</b> Importing AA sequences in R</a></li>
<li class="chapter" data-level="31.4" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-nucleotide-sequences-in-r"><i class="fa fa-check"></i><b>31.4</b> Importing nucleotide sequences in R</a></li>
<li class="chapter" data-level="31.5" data-path="working-with-proteins.html"><a href="working-with-proteins.html#translating-a-nucleotides-to-aa"><i class="fa fa-check"></i><b>31.5</b> Translating a nucleotides to AA</a></li>
<li class="chapter" data-level="31.6" data-path="working-with-proteins.html"><a href="working-with-proteins.html#finding-orfs-in-a-nucleotide-sequence"><i class="fa fa-check"></i><b>31.6</b> Finding ORFs in a nucleotide sequence</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="miscellaneous.html"><a href="miscellaneous.html"><i class="fa fa-check"></i><b>32</b> – Miscellaneous</a>
<ul>
<li class="chapter" data-level="32.1" data-path="miscellaneous.html"><a href="miscellaneous.html#sampling-from-a-vector-of-names"><i class="fa fa-check"></i><b>32.1</b> Sampling from a vector of names</a></li>
<li class="chapter" data-level="32.2" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-an-article-pdf"><i class="fa fa-check"></i><b>32.2</b> Read a text from an article pdf</a></li>
<li class="chapter" data-level="32.3" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-website"><i class="fa fa-check"></i><b>32.3</b> Read a text from a website</a></li>
<li class="chapter" data-level="32.4" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-file"><i class="fa fa-check"></i><b>32.4</b> Read a text from a file</a></li>
<li class="chapter" data-level="32.5" data-path="miscellaneous.html"><a href="miscellaneous.html#generate-the-word-cloud"><i class="fa fa-check"></i><b>32.5</b> Generate the word cloud</a></li>
<li class="chapter" data-level="32.6" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-pdf"><i class="fa fa-check"></i><b>32.6</b> worldcloud – PDF</a></li>
<li class="chapter" data-level="32.7" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-website"><i class="fa fa-check"></i><b>32.7</b> worldcloud – website</a></li>
<li class="chapter" data-level="32.8" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-text"><i class="fa fa-check"></i><b>32.8</b> worldcloud – text</a></li>
<li class="chapter" data-level="32.9" data-path="miscellaneous.html"><a href="miscellaneous.html#miscellaneous-1"><i class="fa fa-check"></i><b>32.9</b> Miscellaneous</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics Workshop Gitbook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="biopython-genomediagram" class="section level1" number="23">
<h1><span class="header-section-number">Session 23</span> – Biopython: GenomeDiagram</h1>
<div id="linearcircular-gene-diagrams" class="section level2" number="23.1">
<h2><span class="header-section-number">23.1</span> Linear/Circular Gene Diagrams</h2>
<p>First get a plasmid genbank file or even a mitochondrial dna genbank file. Feel free to use the example from this <a href="https://www.ncbi.nlm.nih.gov/nucleotide/NC_001573.1" target="_blank"><strong>link</strong></a>.</p>
<p>Notice: macOSX migth need to install <strong>reportlab</strong> and use <code>pip3 install reportlab</code>.</p>
<div class="sourceCode" id="cb583"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb583-1"><a href="biopython-genomediagram.html#cb583-1" aria-hidden="true" tabindex="-1"></a><span class="co"># import modules</span></span>
<span id="cb583-2"><a href="biopython-genomediagram.html#cb583-2" aria-hidden="true" tabindex="-1"></a>import os</span>
<span id="cb583-3"><a href="biopython-genomediagram.html#cb583-3" aria-hidden="true" tabindex="-1"></a>from reportlab.lib import colors</span>
<span id="cb583-4"><a href="biopython-genomediagram.html#cb583-4" aria-hidden="true" tabindex="-1"></a>from reportlab.lib.units import cm</span>
<span id="cb583-5"><a href="biopython-genomediagram.html#cb583-5" aria-hidden="true" tabindex="-1"></a>from Bio.Graphics import GenomeDiagram</span>
<span id="cb583-6"><a href="biopython-genomediagram.html#cb583-6" aria-hidden="true" tabindex="-1"></a>from Bio import SeqIO</span>
<span id="cb583-7"><a href="biopython-genomediagram.html#cb583-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb583-8"><a href="biopython-genomediagram.html#cb583-8" aria-hidden="true" tabindex="-1"></a><span class="co"># set wd</span></span>
<span id="cb583-9"><a href="biopython-genomediagram.html#cb583-9" aria-hidden="true" tabindex="-1"></a><span class="fu">os.chdir</span>(<span class="st">&quot;YOUR WORKING DIRECTORY WITH GENBANK FILE HERE&quot;</span>)</span>
<span id="cb583-10"><a href="biopython-genomediagram.html#cb583-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb583-11"><a href="biopython-genomediagram.html#cb583-11" aria-hidden="true" tabindex="-1"></a><span class="co"># load file as record</span></span>
<span id="cb583-12"><a href="biopython-genomediagram.html#cb583-12" aria-hidden="true" tabindex="-1"></a>record <span class="ot">=</span> <span class="fu">SeqIO.read</span>(<span class="st">&quot;GENBANK FILE HERE&quot;</span>, <span class="st">&quot;genbank&quot;</span>)</span>
<span id="cb583-13"><a href="biopython-genomediagram.html#cb583-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb583-14"><a href="biopython-genomediagram.html#cb583-14" aria-hidden="true" tabindex="-1"></a><span class="co"># set parameters, set empty diagram</span></span>
<span id="cb583-15"><a href="biopython-genomediagram.html#cb583-15" aria-hidden="true" tabindex="-1"></a>gen_diagram <span class="ot">=</span> <span class="fu">GenomeDiagram.Diagram</span>(<span class="st">&quot;TITLE HERE&quot;</span>)</span>
<span id="cb583-16"><a href="biopython-genomediagram.html#cb583-16" aria-hidden="true" tabindex="-1"></a>gen_track_for_features <span class="ot">=</span> <span class="fu">gen_diagram.new_track</span>(<span class="dv">1</span>, <span class="at">name=</span><span class="st">&quot;Annotated Features&quot;</span>)</span>
<span id="cb583-17"><a href="biopython-genomediagram.html#cb583-17" aria-hidden="true" tabindex="-1"></a>gen_feature_set <span class="ot">=</span> <span class="fu">gen_track_for_features.new_set</span>()</span>
<span id="cb583-18"><a href="biopython-genomediagram.html#cb583-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb583-19"><a href="biopython-genomediagram.html#cb583-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate diagram, alternate colors</span></span>
<span id="cb583-20"><a href="biopython-genomediagram.html#cb583-20" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> feature <span class="cf">in</span> record.features<span class="sc">:</span></span>
<span id="cb583-21"><a href="biopython-genomediagram.html#cb583-21" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> feature.type <span class="sc">!=</span> <span class="st">&quot;gene&quot;</span><span class="sc">:</span></span>
<span id="cb583-22"><a href="biopython-genomediagram.html#cb583-22" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Do Nothing</span></span>
<span id="cb583-23"><a href="biopython-genomediagram.html#cb583-23" aria-hidden="true" tabindex="-1"></a>        continue</span>
<span id="cb583-24"><a href="biopython-genomediagram.html#cb583-24" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="fu">len</span>(gen_feature_set) % <span class="dv">2</span> <span class="sc">==</span> <span class="dv">0</span><span class="sc">:</span></span>
<span id="cb583-25"><a href="biopython-genomediagram.html#cb583-25" aria-hidden="true" tabindex="-1"></a>        color <span class="ot">=</span> colors.blue</span>
<span id="cb583-26"><a href="biopython-genomediagram.html#cb583-26" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span><span class="sc">:</span></span>
<span id="cb583-27"><a href="biopython-genomediagram.html#cb583-27" aria-hidden="true" tabindex="-1"></a>        color <span class="ot">=</span> colors.lightblue</span>
<span id="cb583-28"><a href="biopython-genomediagram.html#cb583-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">gen_feature_set.add_feature</span>(feature, <span class="at">color=</span>color, <span class="at">label=</span>True, <span class="at">label_size=</span><span class="dv">10</span>)</span>
<span id="cb583-29"><a href="biopython-genomediagram.html#cb583-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb583-30"><a href="biopython-genomediagram.html#cb583-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Create output</span></span>
<span id="cb583-31"><a href="biopython-genomediagram.html#cb583-31" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_diagram.draw</span>(</span>
<span id="cb583-32"><a href="biopython-genomediagram.html#cb583-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">format=</span><span class="st">&quot;linear&quot;</span>,</span>
<span id="cb583-33"><a href="biopython-genomediagram.html#cb583-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">orientation=</span><span class="st">&quot;landscape&quot;</span>,</span>
<span id="cb583-34"><a href="biopython-genomediagram.html#cb583-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">pagesize=</span><span class="st">&quot;A4&quot;</span>,</span>
<span id="cb583-35"><a href="biopython-genomediagram.html#cb583-35" aria-hidden="true" tabindex="-1"></a>    <span class="at">fragments=</span><span class="dv">4</span>,</span>
<span id="cb583-36"><a href="biopython-genomediagram.html#cb583-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">start=</span><span class="dv">0</span>,</span>
<span id="cb583-37"><a href="biopython-genomediagram.html#cb583-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">end=</span><span class="fu">len</span>(record),</span>
<span id="cb583-38"><a href="biopython-genomediagram.html#cb583-38" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb583-39"><a href="biopython-genomediagram.html#cb583-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb583-40"><a href="biopython-genomediagram.html#cb583-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Save figure as ...</span></span>
<span id="cb583-41"><a href="biopython-genomediagram.html#cb583-41" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_diagram.write</span>(<span class="st">&quot;example1.pdf&quot;</span>, <span class="st">&quot;PDF&quot;</span>)</span>
<span id="cb583-42"><a href="biopython-genomediagram.html#cb583-42" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_diagram.write</span>(<span class="st">&quot;example1.eps&quot;</span>, <span class="st">&quot;EPS&quot;</span>)</span>
<span id="cb583-43"><a href="biopython-genomediagram.html#cb583-43" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_diagram.write</span>(<span class="st">&quot;example1.svg&quot;</span>, <span class="st">&quot;SVG&quot;</span>)</span>
<span id="cb583-44"><a href="biopython-genomediagram.html#cb583-44" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_diagram.write</span>(<span class="st">&quot;example1.png&quot;</span>, <span class="st">&quot;PNG&quot;</span>)</span>
<span id="cb583-45"><a href="biopython-genomediagram.html#cb583-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb583-46"><a href="biopython-genomediagram.html#cb583-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb583-47"><a href="biopython-genomediagram.html#cb583-47" aria-hidden="true" tabindex="-1"></a><span class="co"># Circular figure:</span></span>
<span id="cb583-48"><a href="biopython-genomediagram.html#cb583-48" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_diagram.draw</span>(</span>
<span id="cb583-49"><a href="biopython-genomediagram.html#cb583-49" aria-hidden="true" tabindex="-1"></a>    <span class="at">format=</span><span class="st">&quot;circular&quot;</span>,</span>
<span id="cb583-50"><a href="biopython-genomediagram.html#cb583-50" aria-hidden="true" tabindex="-1"></a>    <span class="at">circular=</span>True,</span>
<span id="cb583-51"><a href="biopython-genomediagram.html#cb583-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">pagesize=</span>(<span class="dv">20</span> <span class="sc">*</span> cm, <span class="dv">20</span> <span class="sc">*</span> cm),</span>
<span id="cb583-52"><a href="biopython-genomediagram.html#cb583-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">start=</span><span class="dv">0</span>,</span>
<span id="cb583-53"><a href="biopython-genomediagram.html#cb583-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">end=</span><span class="fu">len</span>(record),</span>
<span id="cb583-54"><a href="biopython-genomediagram.html#cb583-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">circle_core=</span><span class="fl">0.7</span>,</span>
<span id="cb583-55"><a href="biopython-genomediagram.html#cb583-55" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb583-56"><a href="biopython-genomediagram.html#cb583-56" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_diagram.write</span>(<span class="st">&quot;example1.pdf&quot;</span>, <span class="st">&quot;PDF&quot;</span>)</span></code></pre></div>
</div>
<div id="diagram-readability" class="section level2" number="23.2">
<h2><span class="header-section-number">23.2</span> Diagram Readability</h2>
<p>You can make several changes to the above code to improve readability of the diagram.</p>
<div class="sourceCode" id="cb584"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb584-1"><a href="biopython-genomediagram.html#cb584-1" aria-hidden="true" tabindex="-1"></a><span class="co"># edit the code above with the code below and see what changes</span></span>
<span id="cb584-2"><a href="biopython-genomediagram.html#cb584-2" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_feature_set.add_feature</span>(</span>
<span id="cb584-3"><a href="biopython-genomediagram.html#cb584-3" aria-hidden="true" tabindex="-1"></a>    feature,</span>
<span id="cb584-4"><a href="biopython-genomediagram.html#cb584-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">label=</span>True,</span>
<span id="cb584-5"><a href="biopython-genomediagram.html#cb584-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">color=</span><span class="st">&quot;purple&quot;</span>,</span>
<span id="cb584-6"><a href="biopython-genomediagram.html#cb584-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">label_position=</span><span class="st">&quot;end&quot;</span>,</span>
<span id="cb584-7"><a href="biopython-genomediagram.html#cb584-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">label_size=</span><span class="dv">20</span>,</span>
<span id="cb584-8"><a href="biopython-genomediagram.html#cb584-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">label_angle=</span><span class="dv">90</span>,</span>
<span id="cb584-9"><a href="biopython-genomediagram.html#cb584-9" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="box-shape" class="section level2" number="23.3">
<h2><span class="header-section-number">23.3</span> Box Shape</h2>
<p>You can change the default shape of the gene boxes.</p>
<div class="sourceCode" id="cb585"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb585-1"><a href="biopython-genomediagram.html#cb585-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Default</span></span>
<span id="cb585-2"><a href="biopython-genomediagram.html#cb585-2" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_feature_set.add_feature</span>(feature)</span>
<span id="cb585-3"><a href="biopython-genomediagram.html#cb585-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb585-4"><a href="biopython-genomediagram.html#cb585-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Box</span></span>
<span id="cb585-5"><a href="biopython-genomediagram.html#cb585-5" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_feature_set.add_feature</span>(feature, <span class="at">sigil=</span><span class="st">&quot;BOX&quot;</span>)</span>
<span id="cb585-6"><a href="biopython-genomediagram.html#cb585-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb585-7"><a href="biopython-genomediagram.html#cb585-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Arrow</span></span>
<span id="cb585-8"><a href="biopython-genomediagram.html#cb585-8" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_feature_set.add_feature</span>(feature, <span class="at">sigil=</span><span class="st">&quot;ARROW&quot;</span>)</span>
<span id="cb585-9"><a href="biopython-genomediagram.html#cb585-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb585-10"><a href="biopython-genomediagram.html#cb585-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Big Arrow</span></span>
<span id="cb585-11"><a href="biopython-genomediagram.html#cb585-11" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_feature_set.add_feature</span>(feature, <span class="at">sigil=</span><span class="st">&quot;BIGARROW&quot;</span>)</span>
<span id="cb585-12"><a href="biopython-genomediagram.html#cb585-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb585-13"><a href="biopython-genomediagram.html#cb585-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Octo-gon</span></span>
<span id="cb585-14"><a href="biopython-genomediagram.html#cb585-14" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_feature_set.add_feature</span>(feature, <span class="at">sigil=</span><span class="st">&quot;OCTO&quot;</span>)</span>
<span id="cb585-15"><a href="biopython-genomediagram.html#cb585-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb585-16"><a href="biopython-genomediagram.html#cb585-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Potter Zag</span></span>
<span id="cb585-17"><a href="biopython-genomediagram.html#cb585-17" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_feature_set.add_feature</span>(feature, <span class="at">sigil=</span><span class="st">&quot;JAGGY&quot;</span>)</span></code></pre></div>
</div>
<div id="edit-box-shape-arrows" class="section level2" number="23.4">
<h2><span class="header-section-number">23.4</span> Edit Box Shape: Arrows</h2>
<p>You can edit the shape of the new shape.</p>
<div class="sourceCode" id="cb586"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb586-1"><a href="biopython-genomediagram.html#cb586-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Tail-less Arrow</span></span>
<span id="cb586-2"><a href="biopython-genomediagram.html#cb586-2" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_feature_set.add_feature</span>(feature, <span class="at">sigil=</span><span class="st">&quot;ARROW&quot;</span>, <span class="at">color=</span><span class="st">&quot;brown&quot;</span>, <span class="at">arrowshaft_height=</span><span class="fl">1.0</span>)</span>
<span id="cb586-3"><a href="biopython-genomediagram.html#cb586-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb586-4"><a href="biopython-genomediagram.html#cb586-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Thin Arrow</span></span>
<span id="cb586-5"><a href="biopython-genomediagram.html#cb586-5" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_feature_set.add_feature</span>(feature, <span class="at">sigil=</span><span class="st">&quot;ARROW&quot;</span>, <span class="at">color=</span><span class="st">&quot;teal&quot;</span>, <span class="at">arrowshaft_height=</span><span class="fl">0.2</span>)</span>
<span id="cb586-6"><a href="biopython-genomediagram.html#cb586-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb586-7"><a href="biopython-genomediagram.html#cb586-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Very Thin Arrow</span></span>
<span id="cb586-8"><a href="biopython-genomediagram.html#cb586-8" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_feature_set.add_feature</span>(</span>
<span id="cb586-9"><a href="biopython-genomediagram.html#cb586-9" aria-hidden="true" tabindex="-1"></a>    feature, <span class="at">sigil=</span><span class="st">&quot;ARROW&quot;</span>, <span class="at">color=</span><span class="st">&quot;darkgreen&quot;</span>, <span class="at">arrowshaft_height=</span><span class="fl">0.1</span></span>
<span id="cb586-10"><a href="biopython-genomediagram.html#cb586-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb586-11"><a href="biopython-genomediagram.html#cb586-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb586-12"><a href="biopython-genomediagram.html#cb586-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Short Arrowhead</span></span>
<span id="cb586-13"><a href="biopython-genomediagram.html#cb586-13" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_feature_set.add_feature</span>(feature, <span class="at">sigil=</span><span class="st">&quot;ARROW&quot;</span>, <span class="at">color=</span><span class="st">&quot;blue&quot;</span>, <span class="at">arrowhead_length=</span><span class="fl">0.25</span>)</span>
<span id="cb586-14"><a href="biopython-genomediagram.html#cb586-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb586-15"><a href="biopython-genomediagram.html#cb586-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Long Arrowhead</span></span>
<span id="cb586-16"><a href="biopython-genomediagram.html#cb586-16" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_feature_set.add_feature</span>(feature, <span class="at">sigil=</span><span class="st">&quot;ARROW&quot;</span>, <span class="at">color=</span><span class="st">&quot;orange&quot;</span>, <span class="at">arrowhead_length=</span><span class="dv">1</span>)</span>
<span id="cb586-17"><a href="biopython-genomediagram.html#cb586-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb586-18"><a href="biopython-genomediagram.html#cb586-18" aria-hidden="true" tabindex="-1"></a><span class="co"># All Arrowhead</span></span>
<span id="cb586-19"><a href="biopython-genomediagram.html#cb586-19" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_feature_set.add_feature</span>(feature, <span class="at">sigil=</span><span class="st">&quot;ARROW&quot;</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>, <span class="at">arrowhead_length=</span><span class="dv">10000</span>)</span>
<span id="cb586-20"><a href="biopython-genomediagram.html#cb586-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb586-21"><a href="biopython-genomediagram.html#cb586-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Big Arrow</span></span>
<span id="cb586-22"><a href="biopython-genomediagram.html#cb586-22" aria-hidden="true" tabindex="-1"></a><span class="fu">gen_feature_set.add_feature</span>(feature, <span class="at">sigil=</span><span class="st">&quot;BIGARROW&quot;</span>)</span></code></pre></div>
</div>
<div id="restriction-map" class="section level2" number="23.5">
<h2><span class="header-section-number">23.5</span> Restriction Map</h2>
<p>Add restriction sites to your small genome</p>
<div class="sourceCode" id="cb587"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb587-1"><a href="biopython-genomediagram.html#cb587-1" aria-hidden="true" tabindex="-1"></a><span class="co"># old</span></span>
<span id="cb587-2"><a href="biopython-genomediagram.html#cb587-2" aria-hidden="true" tabindex="-1"></a>from reportlab.lib import colors</span>
<span id="cb587-3"><a href="biopython-genomediagram.html#cb587-3" aria-hidden="true" tabindex="-1"></a>from reportlab.lib.units import cm</span>
<span id="cb587-4"><a href="biopython-genomediagram.html#cb587-4" aria-hidden="true" tabindex="-1"></a>from Bio.Graphics import GenomeDiagram</span>
<span id="cb587-5"><a href="biopython-genomediagram.html#cb587-5" aria-hidden="true" tabindex="-1"></a>from Bio import SeqIO</span>
<span id="cb587-6"><a href="biopython-genomediagram.html#cb587-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb587-7"><a href="biopython-genomediagram.html#cb587-7" aria-hidden="true" tabindex="-1"></a><span class="co"># new</span></span>
<span id="cb587-8"><a href="biopython-genomediagram.html#cb587-8" aria-hidden="true" tabindex="-1"></a>from Bio.SeqFeature import SeqFeature, FeatureLocation</span>
<span id="cb587-9"><a href="biopython-genomediagram.html#cb587-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb587-10"><a href="biopython-genomediagram.html#cb587-10" aria-hidden="true" tabindex="-1"></a><span class="co"># old</span></span>
<span id="cb587-11"><a href="biopython-genomediagram.html#cb587-11" aria-hidden="true" tabindex="-1"></a>record <span class="ot">=</span> <span class="fu">SeqIO.read</span>(<span class="st">&quot;YOUR GB FILE HERE&quot;</span>, <span class="st">&quot;genbank&quot;</span>)</span>
<span id="cb587-12"><a href="biopython-genomediagram.html#cb587-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb587-13"><a href="biopython-genomediagram.html#cb587-13" aria-hidden="true" tabindex="-1"></a><span class="co"># old</span></span>
<span id="cb587-14"><a href="biopython-genomediagram.html#cb587-14" aria-hidden="true" tabindex="-1"></a>gd_diagram <span class="ot">=</span> <span class="fu">GenomeDiagram.Diagram</span>(record.id)</span>
<span id="cb587-15"><a href="biopython-genomediagram.html#cb587-15" aria-hidden="true" tabindex="-1"></a>gd_track_for_features <span class="ot">=</span> <span class="fu">gd_diagram.new_track</span>(<span class="dv">1</span>, <span class="at">name=</span><span class="st">&quot;Annotated Features&quot;</span>)</span>
<span id="cb587-16"><a href="biopython-genomediagram.html#cb587-16" aria-hidden="true" tabindex="-1"></a>gd_feature_set <span class="ot">=</span> <span class="fu">gd_track_for_features.new_set</span>()</span>
<span id="cb587-17"><a href="biopython-genomediagram.html#cb587-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb587-18"><a href="biopython-genomediagram.html#cb587-18" aria-hidden="true" tabindex="-1"></a><span class="co"># old</span></span>
<span id="cb587-19"><a href="biopython-genomediagram.html#cb587-19" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> feature <span class="cf">in</span> record.features<span class="sc">:</span></span>
<span id="cb587-20"><a href="biopython-genomediagram.html#cb587-20" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> feature.type <span class="sc">!=</span> <span class="st">&quot;gene&quot;</span><span class="sc">:</span></span>
<span id="cb587-21"><a href="biopython-genomediagram.html#cb587-21" aria-hidden="true" tabindex="-1"></a>        <span class="co"># Exclude this feature</span></span>
<span id="cb587-22"><a href="biopython-genomediagram.html#cb587-22" aria-hidden="true" tabindex="-1"></a>        continue</span>
<span id="cb587-23"><a href="biopython-genomediagram.html#cb587-23" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="fu">len</span>(gd_feature_set) % <span class="dv">2</span> <span class="sc">==</span> <span class="dv">0</span><span class="sc">:</span></span>
<span id="cb587-24"><a href="biopython-genomediagram.html#cb587-24" aria-hidden="true" tabindex="-1"></a>        color <span class="ot">=</span> colors.blue</span>
<span id="cb587-25"><a href="biopython-genomediagram.html#cb587-25" aria-hidden="true" tabindex="-1"></a>    <span class="cf">else</span><span class="sc">:</span></span>
<span id="cb587-26"><a href="biopython-genomediagram.html#cb587-26" aria-hidden="true" tabindex="-1"></a>        color <span class="ot">=</span> colors.lightblue</span>
<span id="cb587-27"><a href="biopython-genomediagram.html#cb587-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">gd_feature_set.add_feature</span>(</span>
<span id="cb587-28"><a href="biopython-genomediagram.html#cb587-28" aria-hidden="true" tabindex="-1"></a>        feature, <span class="at">sigil=</span><span class="st">&quot;ARROW&quot;</span>, <span class="at">color=</span>color, <span class="at">label=</span>True, <span class="at">label_size=</span><span class="dv">14</span>, <span class="at">label_angle=</span><span class="dv">0</span></span>
<span id="cb587-29"><a href="biopython-genomediagram.html#cb587-29" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb587-30"><a href="biopython-genomediagram.html#cb587-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb587-31"><a href="biopython-genomediagram.html#cb587-31" aria-hidden="true" tabindex="-1"></a><span class="co"># new: Add restriction enzymes and sites</span></span>
<span id="cb587-32"><a href="biopython-genomediagram.html#cb587-32" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> site, name, color <span class="cf">in</span> [</span>
<span id="cb587-33"><a href="biopython-genomediagram.html#cb587-33" aria-hidden="true" tabindex="-1"></a>    (<span class="st">&quot;GAATTC&quot;</span>, <span class="st">&quot;EcoRI&quot;</span>, colors.green),</span>
<span id="cb587-34"><a href="biopython-genomediagram.html#cb587-34" aria-hidden="true" tabindex="-1"></a>    (<span class="st">&quot;CCCGGG&quot;</span>, <span class="st">&quot;SmaI&quot;</span>, colors.orange),</span>
<span id="cb587-35"><a href="biopython-genomediagram.html#cb587-35" aria-hidden="true" tabindex="-1"></a>    (<span class="st">&quot;AAGCTT&quot;</span>, <span class="st">&quot;HindIII&quot;</span>, colors.red),</span>
<span id="cb587-36"><a href="biopython-genomediagram.html#cb587-36" aria-hidden="true" tabindex="-1"></a>    (<span class="st">&quot;GGATCC&quot;</span>, <span class="st">&quot;BamHI&quot;</span>, colors.purple),</span>
<span id="cb587-37"><a href="biopython-genomediagram.html#cb587-37" aria-hidden="true" tabindex="-1"></a>]<span class="sc">:</span></span>
<span id="cb587-38"><a href="biopython-genomediagram.html#cb587-38" aria-hidden="true" tabindex="-1"></a>    index <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb587-39"><a href="biopython-genomediagram.html#cb587-39" aria-hidden="true" tabindex="-1"></a>    <span class="cf">while</span> True<span class="sc">:</span></span>
<span id="cb587-40"><a href="biopython-genomediagram.html#cb587-40" aria-hidden="true" tabindex="-1"></a>        index <span class="ot">=</span> <span class="fu">record.seq.find</span>(site, <span class="at">start=</span>index)</span>
<span id="cb587-41"><a href="biopython-genomediagram.html#cb587-41" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> index <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span><span class="sc">:</span></span>
<span id="cb587-42"><a href="biopython-genomediagram.html#cb587-42" aria-hidden="true" tabindex="-1"></a>            <span class="cf">break</span></span>
<span id="cb587-43"><a href="biopython-genomediagram.html#cb587-43" aria-hidden="true" tabindex="-1"></a>        feature <span class="ot">=</span> <span class="fu">SeqFeature</span>(<span class="fu">FeatureLocation</span>(index, index <span class="sc">+</span> <span class="fu">len</span>(site)))</span>
<span id="cb587-44"><a href="biopython-genomediagram.html#cb587-44" aria-hidden="true" tabindex="-1"></a>        <span class="fu">gd_feature_set.add_feature</span>(</span>
<span id="cb587-45"><a href="biopython-genomediagram.html#cb587-45" aria-hidden="true" tabindex="-1"></a>            feature,</span>
<span id="cb587-46"><a href="biopython-genomediagram.html#cb587-46" aria-hidden="true" tabindex="-1"></a>            <span class="at">color=</span>color,</span>
<span id="cb587-47"><a href="biopython-genomediagram.html#cb587-47" aria-hidden="true" tabindex="-1"></a>            <span class="at">name=</span>name,</span>
<span id="cb587-48"><a href="biopython-genomediagram.html#cb587-48" aria-hidden="true" tabindex="-1"></a>            <span class="at">label=</span>True,</span>
<span id="cb587-49"><a href="biopython-genomediagram.html#cb587-49" aria-hidden="true" tabindex="-1"></a>            <span class="at">label_size=</span><span class="dv">10</span>,</span>
<span id="cb587-50"><a href="biopython-genomediagram.html#cb587-50" aria-hidden="true" tabindex="-1"></a>            <span class="at">label_color=</span>color,</span>
<span id="cb587-51"><a href="biopython-genomediagram.html#cb587-51" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb587-52"><a href="biopython-genomediagram.html#cb587-52" aria-hidden="true" tabindex="-1"></a>        index <span class="sc">+</span><span class="er">=</span> <span class="fu">len</span>(site)</span>
<span id="cb587-53"><a href="biopython-genomediagram.html#cb587-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb587-54"><a href="biopython-genomediagram.html#cb587-54" aria-hidden="true" tabindex="-1"></a><span class="co"># old, linear</span></span>
<span id="cb587-55"><a href="biopython-genomediagram.html#cb587-55" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.draw</span>(<span class="at">format=</span><span class="st">&quot;linear&quot;</span>, <span class="at">pagesize=</span><span class="st">&quot;A4&quot;</span>, <span class="at">fragments=</span><span class="dv">4</span>, <span class="at">start=</span><span class="dv">0</span>, <span class="at">end=</span><span class="fu">len</span>(record))</span>
<span id="cb587-56"><a href="biopython-genomediagram.html#cb587-56" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.write</span>(<span class="st">&quot;example2.pdf&quot;</span>, <span class="st">&quot;PDF&quot;</span>)</span>
<span id="cb587-57"><a href="biopython-genomediagram.html#cb587-57" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.write</span>(<span class="st">&quot;example2.eps&quot;</span>, <span class="st">&quot;EPS&quot;</span>)</span>
<span id="cb587-58"><a href="biopython-genomediagram.html#cb587-58" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.write</span>(<span class="st">&quot;example2.svg&quot;</span>, <span class="st">&quot;SVG&quot;</span>)</span>
<span id="cb587-59"><a href="biopython-genomediagram.html#cb587-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb587-60"><a href="biopython-genomediagram.html#cb587-60" aria-hidden="true" tabindex="-1"></a><span class="co"># old, circular</span></span>
<span id="cb587-61"><a href="biopython-genomediagram.html#cb587-61" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.draw</span>(</span>
<span id="cb587-62"><a href="biopython-genomediagram.html#cb587-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">format=</span><span class="st">&quot;circular&quot;</span>,</span>
<span id="cb587-63"><a href="biopython-genomediagram.html#cb587-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">circular=</span>True,</span>
<span id="cb587-64"><a href="biopython-genomediagram.html#cb587-64" aria-hidden="true" tabindex="-1"></a>    <span class="at">pagesize=</span>(<span class="dv">20</span> <span class="sc">*</span> cm, <span class="dv">20</span> <span class="sc">*</span> cm),</span>
<span id="cb587-65"><a href="biopython-genomediagram.html#cb587-65" aria-hidden="true" tabindex="-1"></a>    <span class="at">start=</span><span class="dv">0</span>,</span>
<span id="cb587-66"><a href="biopython-genomediagram.html#cb587-66" aria-hidden="true" tabindex="-1"></a>    <span class="at">end=</span><span class="fu">len</span>(record),</span>
<span id="cb587-67"><a href="biopython-genomediagram.html#cb587-67" aria-hidden="true" tabindex="-1"></a>    <span class="at">circle_core=</span><span class="fl">0.5</span>,</span>
<span id="cb587-68"><a href="biopython-genomediagram.html#cb587-68" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb587-69"><a href="biopython-genomediagram.html#cb587-69" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.write</span>(<span class="st">&quot;example2.pdf&quot;</span>, <span class="st">&quot;PDF&quot;</span>)</span>
<span id="cb587-70"><a href="biopython-genomediagram.html#cb587-70" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.write</span>(<span class="st">&quot;example2.eps&quot;</span>, <span class="st">&quot;EPS&quot;</span>)</span>
<span id="cb587-71"><a href="biopython-genomediagram.html#cb587-71" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.write</span>(<span class="st">&quot;example2.svg&quot;</span>, <span class="st">&quot;SVG&quot;</span>)</span></code></pre></div>
</div>
<div id="tracks-on-tracks" class="section level2" number="23.6">
<h2><span class="header-section-number">23.6</span> Tracks on Tracks</h2>
<p>I will use the following genbank file for my example of stacked tracks along with my previous file. You are again free to use your own genome/genome segment of choice.</p>
<p><a href="https://www.ncbi.nlm.nih.gov/nucleotide/NC_006839.1" target="_blank"><strong>genome_mt_1</strong></a></p>
<p><a href="https://www.ncbi.nlm.nih.gov/nucleotide/NC_026789.1" target="_blank"><strong>genome_mt_2</strong></a></p>
<div class="sourceCode" id="cb588"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb588-1"><a href="biopython-genomediagram.html#cb588-1" aria-hidden="true" tabindex="-1"></a>from Bio import SeqIO</span>
<span id="cb588-2"><a href="biopython-genomediagram.html#cb588-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb588-3"><a href="biopython-genomediagram.html#cb588-3" aria-hidden="true" tabindex="-1"></a>A_rec <span class="ot">=</span> <span class="fu">SeqIO.read</span>(<span class="st">&quot;XL.gb&quot;</span>, <span class="st">&quot;gb&quot;</span>)</span>
<span id="cb588-4"><a href="biopython-genomediagram.html#cb588-4" aria-hidden="true" tabindex="-1"></a>B_rec <span class="ot">=</span> <span class="fu">SeqIO.read</span>(<span class="st">&quot;XT.gb&quot;</span>, <span class="st">&quot;gb&quot;</span>)</span>
<span id="cb588-5"><a href="biopython-genomediagram.html#cb588-5" aria-hidden="true" tabindex="-1"></a>C_rec <span class="ot">=</span> <span class="fu">SeqIO.read</span>(<span class="st">&quot;NP.gb&quot;</span>, <span class="st">&quot;gb&quot;</span>)</span>
<span id="cb588-6"><a href="biopython-genomediagram.html#cb588-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb588-7"><a href="biopython-genomediagram.html#cb588-7" aria-hidden="true" tabindex="-1"></a>from reportlab.lib.colors <span class="fu">import</span> (</span>
<span id="cb588-8"><a href="biopython-genomediagram.html#cb588-8" aria-hidden="true" tabindex="-1"></a>    red,</span>
<span id="cb588-9"><a href="biopython-genomediagram.html#cb588-9" aria-hidden="true" tabindex="-1"></a>    grey,</span>
<span id="cb588-10"><a href="biopython-genomediagram.html#cb588-10" aria-hidden="true" tabindex="-1"></a>    orange,</span>
<span id="cb588-11"><a href="biopython-genomediagram.html#cb588-11" aria-hidden="true" tabindex="-1"></a>    green,</span>
<span id="cb588-12"><a href="biopython-genomediagram.html#cb588-12" aria-hidden="true" tabindex="-1"></a>    brown,</span>
<span id="cb588-13"><a href="biopython-genomediagram.html#cb588-13" aria-hidden="true" tabindex="-1"></a>    blue,</span>
<span id="cb588-14"><a href="biopython-genomediagram.html#cb588-14" aria-hidden="true" tabindex="-1"></a>    lightblue,</span>
<span id="cb588-15"><a href="biopython-genomediagram.html#cb588-15" aria-hidden="true" tabindex="-1"></a>    purple,</span>
<span id="cb588-16"><a href="biopython-genomediagram.html#cb588-16" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb588-17"><a href="biopython-genomediagram.html#cb588-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb588-18"><a href="biopython-genomediagram.html#cb588-18" aria-hidden="true" tabindex="-1"></a>A_colors <span class="ot">=</span> (</span>
<span id="cb588-19"><a href="biopython-genomediagram.html#cb588-19" aria-hidden="true" tabindex="-1"></a>    [red] <span class="sc">*</span> <span class="dv">5</span></span>
<span id="cb588-20"><a href="biopython-genomediagram.html#cb588-20" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey] <span class="sc">*</span> <span class="dv">7</span></span>
<span id="cb588-21"><a href="biopython-genomediagram.html#cb588-21" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [orange] <span class="sc">*</span> <span class="dv">2</span></span>
<span id="cb588-22"><a href="biopython-genomediagram.html#cb588-22" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey] <span class="sc">*</span> <span class="dv">2</span></span>
<span id="cb588-23"><a href="biopython-genomediagram.html#cb588-23" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [orange]</span>
<span id="cb588-24"><a href="biopython-genomediagram.html#cb588-24" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey] <span class="sc">*</span> <span class="dv">11</span></span>
<span id="cb588-25"><a href="biopython-genomediagram.html#cb588-25" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [green] <span class="sc">*</span> <span class="dv">4</span></span>
<span id="cb588-26"><a href="biopython-genomediagram.html#cb588-26" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey]</span>
<span id="cb588-27"><a href="biopython-genomediagram.html#cb588-27" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [green] <span class="sc">*</span> <span class="dv">2</span></span>
<span id="cb588-28"><a href="biopython-genomediagram.html#cb588-28" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey, green]</span>
<span id="cb588-29"><a href="biopython-genomediagram.html#cb588-29" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [brown] <span class="sc">*</span> <span class="dv">5</span></span>
<span id="cb588-30"><a href="biopython-genomediagram.html#cb588-30" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [blue] <span class="sc">*</span> <span class="dv">4</span></span>
<span id="cb588-31"><a href="biopython-genomediagram.html#cb588-31" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [lightblue] <span class="sc">*</span> <span class="dv">5</span></span>
<span id="cb588-32"><a href="biopython-genomediagram.html#cb588-32" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey, lightblue]</span>
<span id="cb588-33"><a href="biopython-genomediagram.html#cb588-33" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [purple] <span class="sc">*</span> <span class="dv">2</span></span>
<span id="cb588-34"><a href="biopython-genomediagram.html#cb588-34" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey]</span>
<span id="cb588-35"><a href="biopython-genomediagram.html#cb588-35" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb588-36"><a href="biopython-genomediagram.html#cb588-36" aria-hidden="true" tabindex="-1"></a>B_colors <span class="ot">=</span> (</span>
<span id="cb588-37"><a href="biopython-genomediagram.html#cb588-37" aria-hidden="true" tabindex="-1"></a>    [red] <span class="sc">*</span> <span class="dv">6</span></span>
<span id="cb588-38"><a href="biopython-genomediagram.html#cb588-38" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey] <span class="sc">*</span> <span class="dv">8</span></span>
<span id="cb588-39"><a href="biopython-genomediagram.html#cb588-39" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [orange] <span class="sc">*</span> <span class="dv">2</span></span>
<span id="cb588-40"><a href="biopython-genomediagram.html#cb588-40" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey]</span>
<span id="cb588-41"><a href="biopython-genomediagram.html#cb588-41" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [orange]</span>
<span id="cb588-42"><a href="biopython-genomediagram.html#cb588-42" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey] <span class="sc">*</span> <span class="dv">21</span></span>
<span id="cb588-43"><a href="biopython-genomediagram.html#cb588-43" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [green] <span class="sc">*</span> <span class="dv">5</span></span>
<span id="cb588-44"><a href="biopython-genomediagram.html#cb588-44" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey]</span>
<span id="cb588-45"><a href="biopython-genomediagram.html#cb588-45" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [brown] <span class="sc">*</span> <span class="dv">4</span></span>
<span id="cb588-46"><a href="biopython-genomediagram.html#cb588-46" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [blue] <span class="sc">*</span> <span class="dv">3</span></span>
<span id="cb588-47"><a href="biopython-genomediagram.html#cb588-47" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [lightblue] <span class="sc">*</span> <span class="dv">3</span></span>
<span id="cb588-48"><a href="biopython-genomediagram.html#cb588-48" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey] <span class="sc">*</span> <span class="dv">5</span></span>
<span id="cb588-49"><a href="biopython-genomediagram.html#cb588-49" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [purple] <span class="sc">*</span> <span class="dv">2</span></span>
<span id="cb588-50"><a href="biopython-genomediagram.html#cb588-50" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb588-51"><a href="biopython-genomediagram.html#cb588-51" aria-hidden="true" tabindex="-1"></a>C_colors <span class="ot">=</span> (</span>
<span id="cb588-52"><a href="biopython-genomediagram.html#cb588-52" aria-hidden="true" tabindex="-1"></a>    [grey] <span class="sc">*</span> <span class="dv">30</span></span>
<span id="cb588-53"><a href="biopython-genomediagram.html#cb588-53" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [green] <span class="sc">*</span> <span class="dv">5</span></span>
<span id="cb588-54"><a href="biopython-genomediagram.html#cb588-54" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [brown] <span class="sc">*</span> <span class="dv">4</span></span>
<span id="cb588-55"><a href="biopython-genomediagram.html#cb588-55" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [blue] <span class="sc">*</span> <span class="dv">2</span></span>
<span id="cb588-56"><a href="biopython-genomediagram.html#cb588-56" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey, blue]</span>
<span id="cb588-57"><a href="biopython-genomediagram.html#cb588-57" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [lightblue] <span class="sc">*</span> <span class="dv">2</span></span>
<span id="cb588-58"><a href="biopython-genomediagram.html#cb588-58" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey] <span class="sc">*</span> <span class="dv">5</span></span>
<span id="cb588-59"><a href="biopython-genomediagram.html#cb588-59" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb588-60"><a href="biopython-genomediagram.html#cb588-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb588-61"><a href="biopython-genomediagram.html#cb588-61" aria-hidden="true" tabindex="-1"></a>from Bio.Graphics import GenomeDiagram</span>
<span id="cb588-62"><a href="biopython-genomediagram.html#cb588-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb588-63"><a href="biopython-genomediagram.html#cb588-63" aria-hidden="true" tabindex="-1"></a>name <span class="ot">=</span> <span class="st">&quot;Stacked&quot;</span></span>
<span id="cb588-64"><a href="biopython-genomediagram.html#cb588-64" aria-hidden="true" tabindex="-1"></a>gd_diagram <span class="ot">=</span> <span class="fu">GenomeDiagram.Diagram</span>(name)</span>
<span id="cb588-65"><a href="biopython-genomediagram.html#cb588-65" aria-hidden="true" tabindex="-1"></a>max_len <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb588-66"><a href="biopython-genomediagram.html#cb588-66" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> record, gene_colors <span class="cf">in</span> <span class="fu">zip</span>([A_rec, B_rec, C_rec], [A_colors, B_colors, C_colors])<span class="sc">:</span></span>
<span id="cb588-67"><a href="biopython-genomediagram.html#cb588-67" aria-hidden="true" tabindex="-1"></a>    max_len <span class="ot">=</span> <span class="fu">max</span>(max_len, <span class="fu">len</span>(record))</span>
<span id="cb588-68"><a href="biopython-genomediagram.html#cb588-68" aria-hidden="true" tabindex="-1"></a>    gd_track_for_features <span class="ot">=</span> <span class="fu">gd_diagram.new_track</span>(</span>
<span id="cb588-69"><a href="biopython-genomediagram.html#cb588-69" aria-hidden="true" tabindex="-1"></a>        <span class="dv">1</span>, <span class="at">name=</span>record.name, <span class="at">greytrack=</span>True, <span class="at">start=</span><span class="dv">0</span>, <span class="at">end=</span><span class="fu">len</span>(record)</span>
<span id="cb588-70"><a href="biopython-genomediagram.html#cb588-70" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb588-71"><a href="biopython-genomediagram.html#cb588-71" aria-hidden="true" tabindex="-1"></a>    gd_feature_set <span class="ot">=</span> <span class="fu">gd_track_for_features.new_set</span>()</span>
<span id="cb588-72"><a href="biopython-genomediagram.html#cb588-72" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb588-73"><a href="biopython-genomediagram.html#cb588-73" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> feature <span class="cf">in</span> record.features<span class="sc">:</span></span>
<span id="cb588-74"><a href="biopython-genomediagram.html#cb588-74" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> feature.type <span class="sc">!=</span> <span class="st">&quot;gene&quot;</span><span class="sc">:</span></span>
<span id="cb588-75"><a href="biopython-genomediagram.html#cb588-75" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Exclude this feature</span></span>
<span id="cb588-76"><a href="biopython-genomediagram.html#cb588-76" aria-hidden="true" tabindex="-1"></a>            continue</span>
<span id="cb588-77"><a href="biopython-genomediagram.html#cb588-77" aria-hidden="true" tabindex="-1"></a>        <span class="fu">gd_feature_set.add_feature</span>(</span>
<span id="cb588-78"><a href="biopython-genomediagram.html#cb588-78" aria-hidden="true" tabindex="-1"></a>            feature,</span>
<span id="cb588-79"><a href="biopython-genomediagram.html#cb588-79" aria-hidden="true" tabindex="-1"></a>            <span class="at">sigil=</span><span class="st">&quot;ARROW&quot;</span>,</span>
<span id="cb588-80"><a href="biopython-genomediagram.html#cb588-80" aria-hidden="true" tabindex="-1"></a>            <span class="at">color=</span>gene_colors[i],</span>
<span id="cb588-81"><a href="biopython-genomediagram.html#cb588-81" aria-hidden="true" tabindex="-1"></a>            <span class="at">label=</span>True,</span>
<span id="cb588-82"><a href="biopython-genomediagram.html#cb588-82" aria-hidden="true" tabindex="-1"></a>            <span class="at">name=</span><span class="fu">str</span>(i <span class="sc">+</span> <span class="dv">1</span>),</span>
<span id="cb588-83"><a href="biopython-genomediagram.html#cb588-83" aria-hidden="true" tabindex="-1"></a>            <span class="at">label_position=</span><span class="st">&quot;start&quot;</span>,</span>
<span id="cb588-84"><a href="biopython-genomediagram.html#cb588-84" aria-hidden="true" tabindex="-1"></a>            <span class="at">label_size=</span><span class="dv">6</span>,</span>
<span id="cb588-85"><a href="biopython-genomediagram.html#cb588-85" aria-hidden="true" tabindex="-1"></a>            <span class="at">label_angle=</span><span class="dv">0</span>,</span>
<span id="cb588-86"><a href="biopython-genomediagram.html#cb588-86" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb588-87"><a href="biopython-genomediagram.html#cb588-87" aria-hidden="true" tabindex="-1"></a>        i <span class="sc">+</span><span class="er">=</span> <span class="dv">1</span></span>
<span id="cb588-88"><a href="biopython-genomediagram.html#cb588-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb588-89"><a href="biopython-genomediagram.html#cb588-89" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.draw</span>(<span class="at">format=</span><span class="st">&quot;linear&quot;</span>, <span class="at">pagesize=</span><span class="st">&quot;A4&quot;</span>, <span class="at">fragments=</span><span class="dv">1</span>, <span class="at">start=</span><span class="dv">0</span>, <span class="at">end=</span>max_len)</span>
<span id="cb588-90"><a href="biopython-genomediagram.html#cb588-90" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.write</span>(name <span class="sc">+</span> <span class="st">&quot;.pdf&quot;</span>, <span class="st">&quot;PDF&quot;</span>)</span>
<span id="cb588-91"><a href="biopython-genomediagram.html#cb588-91" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.write</span>(name <span class="sc">+</span> <span class="st">&quot;.eps&quot;</span>, <span class="st">&quot;EPS&quot;</span>)</span>
<span id="cb588-92"><a href="biopython-genomediagram.html#cb588-92" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.write</span>(name <span class="sc">+</span> <span class="st">&quot;.svg&quot;</span>, <span class="st">&quot;SVG&quot;</span>)</span></code></pre></div>
</div>
<div id="cross-link-data" class="section level2" number="23.7">
<h2><span class="header-section-number">23.7</span> Cross-link Data</h2>
<p>Time to find similarities and differences.</p>
<div class="sourceCode" id="cb589"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb589-1"><a href="biopython-genomediagram.html#cb589-1" aria-hidden="true" tabindex="-1"></a>A_vs_B <span class="ot">=</span> [</span>
<span id="cb589-2"><a href="biopython-genomediagram.html#cb589-2" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">99</span>, <span class="st">&quot;ND1&quot;</span>, <span class="st">&quot;ND1&quot;</span>),</span>
<span id="cb589-3"><a href="biopython-genomediagram.html#cb589-3" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">33</span>, <span class="st">&quot;ND2&quot;</span>, <span class="st">&quot;ND2&quot;</span>),</span>
<span id="cb589-4"><a href="biopython-genomediagram.html#cb589-4" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">94</span>, <span class="st">&quot;COX1&quot;</span>, <span class="st">&quot;COX1&quot;</span>),</span>
<span id="cb589-5"><a href="biopython-genomediagram.html#cb589-5" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">100</span>, <span class="st">&quot;COX2&quot;</span>, <span class="st">&quot;COX2&quot;</span>),</span>
<span id="cb589-6"><a href="biopython-genomediagram.html#cb589-6" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">97</span>, <span class="st">&quot;ATP8&quot;</span>, <span class="st">&quot;ATP8&quot;</span>),</span>
<span id="cb589-7"><a href="biopython-genomediagram.html#cb589-7" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">98</span>, <span class="st">&quot;ATP6&quot;</span>, <span class="st">&quot;ATP6&quot;</span>),</span>
<span id="cb589-8"><a href="biopython-genomediagram.html#cb589-8" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">98</span>, <span class="st">&quot;COX3&quot;</span>, <span class="st">&quot;COX3&quot;</span>),</span>
<span id="cb589-9"><a href="biopython-genomediagram.html#cb589-9" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">100</span>, <span class="st">&quot;ND3&quot;</span>, <span class="st">&quot;ND3&quot;</span>),</span>
<span id="cb589-10"><a href="biopython-genomediagram.html#cb589-10" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">100</span>, <span class="st">&quot;ND4L&quot;</span>, <span class="st">&quot;ND4L&quot;</span>),</span>
<span id="cb589-11"><a href="biopython-genomediagram.html#cb589-11" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">94</span>, <span class="st">&quot;ND4&quot;</span>, <span class="st">&quot;ND4&quot;</span>),</span>
<span id="cb589-12"><a href="biopython-genomediagram.html#cb589-12" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">87</span>, <span class="st">&quot;ND5&quot;</span>, <span class="st">&quot;ND5&quot;</span>),</span>
<span id="cb589-13"><a href="biopython-genomediagram.html#cb589-13" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">94</span>, <span class="st">&quot;ND6&quot;</span>, <span class="st">&quot;ND6&quot;</span>),</span>
<span id="cb589-14"><a href="biopython-genomediagram.html#cb589-14" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">94</span>, <span class="st">&quot;CYTB&quot;</span>, <span class="st">&quot;CYTB&quot;</span>),</span>
<span id="cb589-15"><a href="biopython-genomediagram.html#cb589-15" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb589-16"><a href="biopython-genomediagram.html#cb589-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb589-17"><a href="biopython-genomediagram.html#cb589-17" aria-hidden="true" tabindex="-1"></a>B_vs_C <span class="ot">=</span> [</span>
<span id="cb589-18"><a href="biopython-genomediagram.html#cb589-18" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">49</span>, <span class="st">&quot;ND1&quot;</span>, <span class="st">&quot;ND1&quot;</span>),</span>
<span id="cb589-19"><a href="biopython-genomediagram.html#cb589-19" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">53</span>, <span class="st">&quot;ND2&quot;</span>, <span class="st">&quot;ND2&quot;</span>),</span>
<span id="cb589-20"><a href="biopython-genomediagram.html#cb589-20" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">64</span>, <span class="st">&quot;COX1&quot;</span>, <span class="st">&quot;COX1&quot;</span>),</span>
<span id="cb589-21"><a href="biopython-genomediagram.html#cb589-21" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">30</span>, <span class="st">&quot;COX2&quot;</span>, <span class="st">&quot;COX2&quot;</span>),</span>
<span id="cb589-22"><a href="biopython-genomediagram.html#cb589-22" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">27</span>, <span class="st">&quot;ATP8&quot;</span>, <span class="st">&quot;ATP8&quot;</span>),</span>
<span id="cb589-23"><a href="biopython-genomediagram.html#cb589-23" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">88</span>, <span class="st">&quot;ATP6&quot;</span>, <span class="st">&quot;ATP6&quot;</span>),</span>
<span id="cb589-24"><a href="biopython-genomediagram.html#cb589-24" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">68</span>, <span class="st">&quot;COX3&quot;</span>, <span class="st">&quot;COX3&quot;</span>),</span>
<span id="cb589-25"><a href="biopython-genomediagram.html#cb589-25" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">30</span>, <span class="st">&quot;ND3&quot;</span>, <span class="st">&quot;ND3&quot;</span>),</span>
<span id="cb589-26"><a href="biopython-genomediagram.html#cb589-26" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">100</span>, <span class="st">&quot;ND4L&quot;</span>, <span class="st">&quot;ND4L&quot;</span>),</span>
<span id="cb589-27"><a href="biopython-genomediagram.html#cb589-27" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">74</span>, <span class="st">&quot;ND4&quot;</span>, <span class="st">&quot;ND4&quot;</span>),</span>
<span id="cb589-28"><a href="biopython-genomediagram.html#cb589-28" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">37</span>, <span class="st">&quot;ND5&quot;</span>, <span class="st">&quot;ND5&quot;</span>),</span>
<span id="cb589-29"><a href="biopython-genomediagram.html#cb589-29" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">74</span>, <span class="st">&quot;ND6&quot;</span>, <span class="st">&quot;ND6&quot;</span>),</span>
<span id="cb589-30"><a href="biopython-genomediagram.html#cb589-30" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">64</span>, <span class="st">&quot;CYTB&quot;</span>, <span class="st">&quot;CYTB&quot;</span>),</span>
<span id="cb589-31"><a href="biopython-genomediagram.html#cb589-31" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb589-32"><a href="biopython-genomediagram.html#cb589-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb589-33"><a href="biopython-genomediagram.html#cb589-33" aria-hidden="true" tabindex="-1"></a>def <span class="fu">get_feature</span>(features, id, <span class="at">tags=</span>[<span class="st">&quot;locus_tag&quot;</span>, <span class="st">&quot;gene&quot;</span>])<span class="sc">:</span></span>
<span id="cb589-34"><a href="biopython-genomediagram.html#cb589-34" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;&quot;&quot;Search list of SeqFeature objects for an identifier under the given tags.&quot;&quot;&quot;</span></span>
<span id="cb589-35"><a href="biopython-genomediagram.html#cb589-35" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> f <span class="cf">in</span> features<span class="sc">:</span></span>
<span id="cb589-36"><a href="biopython-genomediagram.html#cb589-36" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> key <span class="cf">in</span> tags<span class="sc">:</span></span>
<span id="cb589-37"><a href="biopython-genomediagram.html#cb589-37" aria-hidden="true" tabindex="-1"></a>            <span class="co"># tag may not be present in this feature</span></span>
<span id="cb589-38"><a href="biopython-genomediagram.html#cb589-38" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> x <span class="cf">in</span> <span class="fu">f.qualifiers.get</span>(key, [])<span class="sc">:</span></span>
<span id="cb589-39"><a href="biopython-genomediagram.html#cb589-39" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> x <span class="sc">==</span> id<span class="sc">:</span></span>
<span id="cb589-40"><a href="biopython-genomediagram.html#cb589-40" aria-hidden="true" tabindex="-1"></a>                    return f</span>
<span id="cb589-41"><a href="biopython-genomediagram.html#cb589-41" aria-hidden="true" tabindex="-1"></a>    raise <span class="fu">KeyError</span>(id)</span>
<span id="cb589-42"><a href="biopython-genomediagram.html#cb589-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb589-43"><a href="biopython-genomediagram.html#cb589-43" aria-hidden="true" tabindex="-1"></a>from Bio.Graphics.GenomeDiagram import CrossLink</span>
<span id="cb589-44"><a href="biopython-genomediagram.html#cb589-44" aria-hidden="true" tabindex="-1"></a>from reportlab.lib import colors</span>
<span id="cb589-45"><a href="biopython-genomediagram.html#cb589-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb589-46"><a href="biopython-genomediagram.html#cb589-46" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> rec_X, tn_X, rec_Y, tn_Y, X_vs_Y <span class="cf">in</span> [</span>
<span id="cb589-47"><a href="biopython-genomediagram.html#cb589-47" aria-hidden="true" tabindex="-1"></a>    (A_rec, <span class="dv">3</span>, B_rec, <span class="dv">2</span>, A_vs_B),</span>
<span id="cb589-48"><a href="biopython-genomediagram.html#cb589-48" aria-hidden="true" tabindex="-1"></a>    (B_rec, <span class="dv">2</span>, C_rec, <span class="dv">1</span>, B_vs_C),</span>
<span id="cb589-49"><a href="biopython-genomediagram.html#cb589-49" aria-hidden="true" tabindex="-1"></a>]<span class="sc">:</span></span>
<span id="cb589-50"><a href="biopython-genomediagram.html#cb589-50" aria-hidden="true" tabindex="-1"></a>    track_X <span class="ot">=</span> gd_diagram.tracks[tn_X]</span>
<span id="cb589-51"><a href="biopython-genomediagram.html#cb589-51" aria-hidden="true" tabindex="-1"></a>    track_Y <span class="ot">=</span> gd_diagram.tracks[tn_Y]</span>
<span id="cb589-52"><a href="biopython-genomediagram.html#cb589-52" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> score, id_X, id_Y <span class="cf">in</span> X_vs_Y<span class="sc">:</span></span>
<span id="cb589-53"><a href="biopython-genomediagram.html#cb589-53" aria-hidden="true" tabindex="-1"></a>        feature_X <span class="ot">=</span> <span class="fu">get_feature</span>(rec_X.features, id_X)</span>
<span id="cb589-54"><a href="biopython-genomediagram.html#cb589-54" aria-hidden="true" tabindex="-1"></a>        feature_Y <span class="ot">=</span> <span class="fu">get_feature</span>(rec_Y.features, id_Y)</span>
<span id="cb589-55"><a href="biopython-genomediagram.html#cb589-55" aria-hidden="true" tabindex="-1"></a>        color <span class="ot">=</span> <span class="fu">colors.linearlyInterpolatedColor</span>(</span>
<span id="cb589-56"><a href="biopython-genomediagram.html#cb589-56" aria-hidden="true" tabindex="-1"></a>            colors.white, colors.firebrick, <span class="dv">0</span>, <span class="dv">100</span>, score</span>
<span id="cb589-57"><a href="biopython-genomediagram.html#cb589-57" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb589-58"><a href="biopython-genomediagram.html#cb589-58" aria-hidden="true" tabindex="-1"></a>        link_xy <span class="ot">=</span> <span class="fu">CrossLink</span>(</span>
<span id="cb589-59"><a href="biopython-genomediagram.html#cb589-59" aria-hidden="true" tabindex="-1"></a>            (track_X, feature_X.location.start, feature_X.location.end),</span>
<span id="cb589-60"><a href="biopython-genomediagram.html#cb589-60" aria-hidden="true" tabindex="-1"></a>            (track_Y, feature_Y.location.start, feature_Y.location.end),</span>
<span id="cb589-61"><a href="biopython-genomediagram.html#cb589-61" aria-hidden="true" tabindex="-1"></a>            color,</span>
<span id="cb589-62"><a href="biopython-genomediagram.html#cb589-62" aria-hidden="true" tabindex="-1"></a>            colors.lightgrey,</span>
<span id="cb589-63"><a href="biopython-genomediagram.html#cb589-63" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb589-64"><a href="biopython-genomediagram.html#cb589-64" aria-hidden="true" tabindex="-1"></a>        <span class="fu">gd_diagram.cross_track_links.append</span>(link_xy)</span>
<span id="cb589-65"><a href="biopython-genomediagram.html#cb589-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb589-66"><a href="biopython-genomediagram.html#cb589-66" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.draw</span>(<span class="at">format=</span><span class="st">&quot;linear&quot;</span>, <span class="at">pagesize=</span><span class="st">&quot;A4&quot;</span>, <span class="at">fragments=</span><span class="dv">1</span>, <span class="at">start=</span><span class="dv">0</span>, <span class="at">end=</span>max_len)</span>
<span id="cb589-67"><a href="biopython-genomediagram.html#cb589-67" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.write</span>(name <span class="sc">+</span> <span class="st">&quot;.pdf&quot;</span>, <span class="st">&quot;PDF&quot;</span>)</span>
<span id="cb589-68"><a href="biopython-genomediagram.html#cb589-68" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.write</span>(name <span class="sc">+</span> <span class="st">&quot;.eps&quot;</span>, <span class="st">&quot;EPS&quot;</span>)</span>
<span id="cb589-69"><a href="biopython-genomediagram.html#cb589-69" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.write</span>(name <span class="sc">+</span> <span class="st">&quot;.svg&quot;</span>, <span class="st">&quot;SVG&quot;</span>)</span></code></pre></div>
</div>
<div id="clean-cross-link" class="section level2" number="23.8">
<h2><span class="header-section-number">23.8</span> Clean Cross-Link</h2>
<div class="sourceCode" id="cb590"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb590-1"><a href="biopython-genomediagram.html#cb590-1" aria-hidden="true" tabindex="-1"></a>from reportlab.lib import colors</span>
<span id="cb590-2"><a href="biopython-genomediagram.html#cb590-2" aria-hidden="true" tabindex="-1"></a>from reportlab.lib.colors import red, grey, orange, green, brown</span>
<span id="cb590-3"><a href="biopython-genomediagram.html#cb590-3" aria-hidden="true" tabindex="-1"></a>from reportlab.lib.colors import blue, lightblue, purple</span>
<span id="cb590-4"><a href="biopython-genomediagram.html#cb590-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-5"><a href="biopython-genomediagram.html#cb590-5" aria-hidden="true" tabindex="-1"></a>from Bio.Graphics import GenomeDiagram</span>
<span id="cb590-6"><a href="biopython-genomediagram.html#cb590-6" aria-hidden="true" tabindex="-1"></a>from Bio.Graphics.GenomeDiagram import CrossLink</span>
<span id="cb590-7"><a href="biopython-genomediagram.html#cb590-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-8"><a href="biopython-genomediagram.html#cb590-8" aria-hidden="true" tabindex="-1"></a>from Bio import SeqIO</span>
<span id="cb590-9"><a href="biopython-genomediagram.html#cb590-9" aria-hidden="true" tabindex="-1"></a>from Bio.SeqFeature import SeqFeature, FeatureLocation</span>
<span id="cb590-10"><a href="biopython-genomediagram.html#cb590-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-11"><a href="biopython-genomediagram.html#cb590-11" aria-hidden="true" tabindex="-1"></a>name <span class="ot">=</span> <span class="st">&quot;Example3&quot;</span></span>
<span id="cb590-12"><a href="biopython-genomediagram.html#cb590-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-13"><a href="biopython-genomediagram.html#cb590-13" aria-hidden="true" tabindex="-1"></a>A_rec <span class="ot">=</span> <span class="fu">SeqIO.read</span>(<span class="st">&quot;XL.gb&quot;</span>, <span class="st">&quot;gb&quot;</span>)</span>
<span id="cb590-14"><a href="biopython-genomediagram.html#cb590-14" aria-hidden="true" tabindex="-1"></a>B_rec <span class="ot">=</span> <span class="fu">SeqIO.read</span>(<span class="st">&quot;XT.gb&quot;</span>, <span class="st">&quot;gb&quot;</span>)</span>
<span id="cb590-15"><a href="biopython-genomediagram.html#cb590-15" aria-hidden="true" tabindex="-1"></a>C_rec <span class="ot">=</span> <span class="fu">SeqIO.read</span>(<span class="st">&quot;NP.gb&quot;</span>, <span class="st">&quot;gb&quot;</span>)</span>
<span id="cb590-16"><a href="biopython-genomediagram.html#cb590-16" aria-hidden="true" tabindex="-1"></a>records <span class="ot">=</span> {rec.name<span class="sc">:</span> rec <span class="cf">for</span> rec <span class="cf">in</span> [A_rec, B_rec, C_rec]}</span>
<span id="cb590-17"><a href="biopython-genomediagram.html#cb590-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-18"><a href="biopython-genomediagram.html#cb590-18" aria-hidden="true" tabindex="-1"></a>A_colors <span class="ot">=</span> (</span>
<span id="cb590-19"><a href="biopython-genomediagram.html#cb590-19" aria-hidden="true" tabindex="-1"></a>    [red]</span>
<span id="cb590-20"><a href="biopython-genomediagram.html#cb590-20" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey]</span>
<span id="cb590-21"><a href="biopython-genomediagram.html#cb590-21" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [orange]</span>
<span id="cb590-22"><a href="biopython-genomediagram.html#cb590-22" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey]</span>
<span id="cb590-23"><a href="biopython-genomediagram.html#cb590-23" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [orange]</span>
<span id="cb590-24"><a href="biopython-genomediagram.html#cb590-24" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey]</span>
<span id="cb590-25"><a href="biopython-genomediagram.html#cb590-25" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [green]</span>
<span id="cb590-26"><a href="biopython-genomediagram.html#cb590-26" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey]</span>
<span id="cb590-27"><a href="biopython-genomediagram.html#cb590-27" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [green]</span>
<span id="cb590-28"><a href="biopython-genomediagram.html#cb590-28" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey, green]</span>
<span id="cb590-29"><a href="biopython-genomediagram.html#cb590-29" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [brown]</span>
<span id="cb590-30"><a href="biopython-genomediagram.html#cb590-30" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [blue]</span>
<span id="cb590-31"><a href="biopython-genomediagram.html#cb590-31" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [lightblue]</span>
<span id="cb590-32"><a href="biopython-genomediagram.html#cb590-32" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey, lightblue]</span>
<span id="cb590-33"><a href="biopython-genomediagram.html#cb590-33" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [purple]</span>
<span id="cb590-34"><a href="biopython-genomediagram.html#cb590-34" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey]</span>
<span id="cb590-35"><a href="biopython-genomediagram.html#cb590-35" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb590-36"><a href="biopython-genomediagram.html#cb590-36" aria-hidden="true" tabindex="-1"></a>B_colors <span class="ot">=</span> (</span>
<span id="cb590-37"><a href="biopython-genomediagram.html#cb590-37" aria-hidden="true" tabindex="-1"></a>    [red]</span>
<span id="cb590-38"><a href="biopython-genomediagram.html#cb590-38" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey]</span>
<span id="cb590-39"><a href="biopython-genomediagram.html#cb590-39" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [orange]</span>
<span id="cb590-40"><a href="biopython-genomediagram.html#cb590-40" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey]</span>
<span id="cb590-41"><a href="biopython-genomediagram.html#cb590-41" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [orange]</span>
<span id="cb590-42"><a href="biopython-genomediagram.html#cb590-42" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey]</span>
<span id="cb590-43"><a href="biopython-genomediagram.html#cb590-43" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [green]</span>
<span id="cb590-44"><a href="biopython-genomediagram.html#cb590-44" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey]</span>
<span id="cb590-45"><a href="biopython-genomediagram.html#cb590-45" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [brown]</span>
<span id="cb590-46"><a href="biopython-genomediagram.html#cb590-46" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [blue]</span>
<span id="cb590-47"><a href="biopython-genomediagram.html#cb590-47" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [lightblue]</span>
<span id="cb590-48"><a href="biopython-genomediagram.html#cb590-48" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey]</span>
<span id="cb590-49"><a href="biopython-genomediagram.html#cb590-49" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [purple]</span>
<span id="cb590-50"><a href="biopython-genomediagram.html#cb590-50" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb590-51"><a href="biopython-genomediagram.html#cb590-51" aria-hidden="true" tabindex="-1"></a>C_colors <span class="ot">=</span> (</span>
<span id="cb590-52"><a href="biopython-genomediagram.html#cb590-52" aria-hidden="true" tabindex="-1"></a>    [grey]</span>
<span id="cb590-53"><a href="biopython-genomediagram.html#cb590-53" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [green]</span>
<span id="cb590-54"><a href="biopython-genomediagram.html#cb590-54" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [brown]</span>
<span id="cb590-55"><a href="biopython-genomediagram.html#cb590-55" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [blue]</span>
<span id="cb590-56"><a href="biopython-genomediagram.html#cb590-56" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey, blue]</span>
<span id="cb590-57"><a href="biopython-genomediagram.html#cb590-57" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [lightblue]</span>
<span id="cb590-58"><a href="biopython-genomediagram.html#cb590-58" aria-hidden="true" tabindex="-1"></a>    <span class="sc">+</span> [grey] <span class="sc">*</span> <span class="dv">8</span></span>
<span id="cb590-59"><a href="biopython-genomediagram.html#cb590-59" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb590-60"><a href="biopython-genomediagram.html#cb590-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-61"><a href="biopython-genomediagram.html#cb590-61" aria-hidden="true" tabindex="-1"></a>A_vs_B <span class="ot">=</span> [</span>
<span id="cb590-62"><a href="biopython-genomediagram.html#cb590-62" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">99</span>, <span class="st">&quot;ND1&quot;</span>, <span class="st">&quot;ND1&quot;</span>),</span>
<span id="cb590-63"><a href="biopython-genomediagram.html#cb590-63" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">33</span>, <span class="st">&quot;ND2&quot;</span>, <span class="st">&quot;ND2&quot;</span>),</span>
<span id="cb590-64"><a href="biopython-genomediagram.html#cb590-64" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">94</span>, <span class="st">&quot;COX1&quot;</span>, <span class="st">&quot;COX1&quot;</span>),</span>
<span id="cb590-65"><a href="biopython-genomediagram.html#cb590-65" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">100</span>, <span class="st">&quot;COX2&quot;</span>, <span class="st">&quot;COX2&quot;</span>),</span>
<span id="cb590-66"><a href="biopython-genomediagram.html#cb590-66" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">97</span>, <span class="st">&quot;ATP8&quot;</span>, <span class="st">&quot;ATP8&quot;</span>),</span>
<span id="cb590-67"><a href="biopython-genomediagram.html#cb590-67" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">98</span>, <span class="st">&quot;ATP6&quot;</span>, <span class="st">&quot;ATP6&quot;</span>),</span>
<span id="cb590-68"><a href="biopython-genomediagram.html#cb590-68" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">98</span>, <span class="st">&quot;COX3&quot;</span>, <span class="st">&quot;COX3&quot;</span>),</span>
<span id="cb590-69"><a href="biopython-genomediagram.html#cb590-69" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">100</span>, <span class="st">&quot;ND3&quot;</span>, <span class="st">&quot;ND3&quot;</span>),</span>
<span id="cb590-70"><a href="biopython-genomediagram.html#cb590-70" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">100</span>, <span class="st">&quot;ND4L&quot;</span>, <span class="st">&quot;ND4L&quot;</span>),</span>
<span id="cb590-71"><a href="biopython-genomediagram.html#cb590-71" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">94</span>, <span class="st">&quot;ND4&quot;</span>, <span class="st">&quot;ND4&quot;</span>),</span>
<span id="cb590-72"><a href="biopython-genomediagram.html#cb590-72" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">87</span>, <span class="st">&quot;ND5&quot;</span>, <span class="st">&quot;ND5&quot;</span>),</span>
<span id="cb590-73"><a href="biopython-genomediagram.html#cb590-73" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">94</span>, <span class="st">&quot;ND6&quot;</span>, <span class="st">&quot;ND6&quot;</span>),</span>
<span id="cb590-74"><a href="biopython-genomediagram.html#cb590-74" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">94</span>, <span class="st">&quot;CYTB&quot;</span>, <span class="st">&quot;CYTB&quot;</span>),</span>
<span id="cb590-75"><a href="biopython-genomediagram.html#cb590-75" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb590-76"><a href="biopython-genomediagram.html#cb590-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-77"><a href="biopython-genomediagram.html#cb590-77" aria-hidden="true" tabindex="-1"></a>B_vs_C <span class="ot">=</span> [</span>
<span id="cb590-78"><a href="biopython-genomediagram.html#cb590-78" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">49</span>, <span class="st">&quot;ND1&quot;</span>, <span class="st">&quot;ND1&quot;</span>),</span>
<span id="cb590-79"><a href="biopython-genomediagram.html#cb590-79" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">53</span>, <span class="st">&quot;ND2&quot;</span>, <span class="st">&quot;ND2&quot;</span>),</span>
<span id="cb590-80"><a href="biopython-genomediagram.html#cb590-80" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">64</span>, <span class="st">&quot;COX1&quot;</span>, <span class="st">&quot;COX1&quot;</span>),</span>
<span id="cb590-81"><a href="biopython-genomediagram.html#cb590-81" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">30</span>, <span class="st">&quot;COX2&quot;</span>, <span class="st">&quot;COX2&quot;</span>),</span>
<span id="cb590-82"><a href="biopython-genomediagram.html#cb590-82" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">27</span>, <span class="st">&quot;ATP8&quot;</span>, <span class="st">&quot;ATP8&quot;</span>),</span>
<span id="cb590-83"><a href="biopython-genomediagram.html#cb590-83" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">88</span>, <span class="st">&quot;ATP6&quot;</span>, <span class="st">&quot;ATP6&quot;</span>),</span>
<span id="cb590-84"><a href="biopython-genomediagram.html#cb590-84" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">68</span>, <span class="st">&quot;COX3&quot;</span>, <span class="st">&quot;COX3&quot;</span>),</span>
<span id="cb590-85"><a href="biopython-genomediagram.html#cb590-85" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">30</span>, <span class="st">&quot;ND3&quot;</span>, <span class="st">&quot;ND3&quot;</span>),</span>
<span id="cb590-86"><a href="biopython-genomediagram.html#cb590-86" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">100</span>, <span class="st">&quot;ND4L&quot;</span>, <span class="st">&quot;ND4L&quot;</span>),</span>
<span id="cb590-87"><a href="biopython-genomediagram.html#cb590-87" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">74</span>, <span class="st">&quot;ND4&quot;</span>, <span class="st">&quot;ND4&quot;</span>),</span>
<span id="cb590-88"><a href="biopython-genomediagram.html#cb590-88" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">37</span>, <span class="st">&quot;ND5&quot;</span>, <span class="st">&quot;ND5&quot;</span>),</span>
<span id="cb590-89"><a href="biopython-genomediagram.html#cb590-89" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">74</span>, <span class="st">&quot;ND6&quot;</span>, <span class="st">&quot;ND6&quot;</span>),</span>
<span id="cb590-90"><a href="biopython-genomediagram.html#cb590-90" aria-hidden="true" tabindex="-1"></a>    (<span class="dv">64</span>, <span class="st">&quot;CYTB&quot;</span>, <span class="st">&quot;CYTB&quot;</span>),</span>
<span id="cb590-91"><a href="biopython-genomediagram.html#cb590-91" aria-hidden="true" tabindex="-1"></a>]</span>
<span id="cb590-92"><a href="biopython-genomediagram.html#cb590-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-93"><a href="biopython-genomediagram.html#cb590-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-94"><a href="biopython-genomediagram.html#cb590-94" aria-hidden="true" tabindex="-1"></a>def <span class="fu">get_feature</span>(features, id, <span class="at">tags=</span>(<span class="st">&quot;locus_tag&quot;</span>, <span class="st">&quot;gene&quot;</span>, <span class="st">&quot;old_locus_tag&quot;</span>))<span class="sc">:</span></span>
<span id="cb590-95"><a href="biopython-genomediagram.html#cb590-95" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;&quot;&quot;Search list of SeqFeature objects for an identifier under the given tags.&quot;&quot;&quot;</span></span>
<span id="cb590-96"><a href="biopython-genomediagram.html#cb590-96" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> f <span class="cf">in</span> features<span class="sc">:</span></span>
<span id="cb590-97"><a href="biopython-genomediagram.html#cb590-97" aria-hidden="true" tabindex="-1"></a>        <span class="cf">for</span> key <span class="cf">in</span> tags<span class="sc">:</span></span>
<span id="cb590-98"><a href="biopython-genomediagram.html#cb590-98" aria-hidden="true" tabindex="-1"></a>            <span class="co"># tag may not be present in this feature</span></span>
<span id="cb590-99"><a href="biopython-genomediagram.html#cb590-99" aria-hidden="true" tabindex="-1"></a>            <span class="cf">for</span> x <span class="cf">in</span> <span class="fu">f.qualifiers.get</span>(key, [])<span class="sc">:</span></span>
<span id="cb590-100"><a href="biopython-genomediagram.html#cb590-100" aria-hidden="true" tabindex="-1"></a>                <span class="cf">if</span> x <span class="sc">==</span> id<span class="sc">:</span></span>
<span id="cb590-101"><a href="biopython-genomediagram.html#cb590-101" aria-hidden="true" tabindex="-1"></a>                    return f</span>
<span id="cb590-102"><a href="biopython-genomediagram.html#cb590-102" aria-hidden="true" tabindex="-1"></a>    raise <span class="fu">KeyError</span>(id)</span>
<span id="cb590-103"><a href="biopython-genomediagram.html#cb590-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-104"><a href="biopython-genomediagram.html#cb590-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-105"><a href="biopython-genomediagram.html#cb590-105" aria-hidden="true" tabindex="-1"></a>gd_diagram <span class="ot">=</span> <span class="fu">GenomeDiagram.Diagram</span>(name)</span>
<span id="cb590-106"><a href="biopython-genomediagram.html#cb590-106" aria-hidden="true" tabindex="-1"></a>feature_sets <span class="ot">=</span> {}</span>
<span id="cb590-107"><a href="biopython-genomediagram.html#cb590-107" aria-hidden="true" tabindex="-1"></a>max_len <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb590-108"><a href="biopython-genomediagram.html#cb590-108" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> i, record <span class="cf">in</span> <span class="fu">enumerate</span>([A_rec, B_rec, C_rec])<span class="sc">:</span></span>
<span id="cb590-109"><a href="biopython-genomediagram.html#cb590-109" aria-hidden="true" tabindex="-1"></a>    max_len <span class="ot">=</span> <span class="fu">max</span>(max_len, <span class="fu">len</span>(record))</span>
<span id="cb590-110"><a href="biopython-genomediagram.html#cb590-110" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Allocate tracks 5 (top), 3, 1 (bottom) for A, B, C</span></span>
<span id="cb590-111"><a href="biopython-genomediagram.html#cb590-111" aria-hidden="true" tabindex="-1"></a>    <span class="co"># (empty tracks 2 and 4 add useful white space to emphasise the cross links</span></span>
<span id="cb590-112"><a href="biopython-genomediagram.html#cb590-112" aria-hidden="true" tabindex="-1"></a>    <span class="co"># and also serve to make the tracks vertically more compressed)</span></span>
<span id="cb590-113"><a href="biopython-genomediagram.html#cb590-113" aria-hidden="true" tabindex="-1"></a>    gd_track_for_features <span class="ot">=</span> <span class="fu">gd_diagram.new_track</span>(</span>
<span id="cb590-114"><a href="biopython-genomediagram.html#cb590-114" aria-hidden="true" tabindex="-1"></a>        <span class="dv">5</span> <span class="sc">-</span> <span class="dv">2</span> <span class="sc">*</span> i,</span>
<span id="cb590-115"><a href="biopython-genomediagram.html#cb590-115" aria-hidden="true" tabindex="-1"></a>        <span class="at">name=</span>record.name,</span>
<span id="cb590-116"><a href="biopython-genomediagram.html#cb590-116" aria-hidden="true" tabindex="-1"></a>        <span class="at">greytrack=</span>True,</span>
<span id="cb590-117"><a href="biopython-genomediagram.html#cb590-117" aria-hidden="true" tabindex="-1"></a>        <span class="at">height=</span><span class="fl">0.5</span>,</span>
<span id="cb590-118"><a href="biopython-genomediagram.html#cb590-118" aria-hidden="true" tabindex="-1"></a>        <span class="at">start=</span><span class="dv">0</span>,</span>
<span id="cb590-119"><a href="biopython-genomediagram.html#cb590-119" aria-hidden="true" tabindex="-1"></a>        <span class="at">end=</span><span class="fu">len</span>(record),</span>
<span id="cb590-120"><a href="biopython-genomediagram.html#cb590-120" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb590-121"><a href="biopython-genomediagram.html#cb590-121" aria-hidden="true" tabindex="-1"></a>    assert record.name not <span class="cf">in</span> feature_sets</span>
<span id="cb590-122"><a href="biopython-genomediagram.html#cb590-122" aria-hidden="true" tabindex="-1"></a>    feature_sets[record.name] <span class="ot">=</span> <span class="fu">gd_track_for_features.new_set</span>()</span>
<span id="cb590-123"><a href="biopython-genomediagram.html#cb590-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-124"><a href="biopython-genomediagram.html#cb590-124" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> X, Y, X_vs_Y <span class="cf">in</span> [</span>
<span id="cb590-125"><a href="biopython-genomediagram.html#cb590-125" aria-hidden="true" tabindex="-1"></a>    (<span class="st">&quot;NC_001573&quot;</span>, <span class="st">&quot;NC_006839&quot;</span>, A_vs_B),</span>
<span id="cb590-126"><a href="biopython-genomediagram.html#cb590-126" aria-hidden="true" tabindex="-1"></a>    (<span class="st">&quot;NC_006839&quot;</span>, <span class="st">&quot;NC_026789&quot;</span>, B_vs_C),</span>
<span id="cb590-127"><a href="biopython-genomediagram.html#cb590-127" aria-hidden="true" tabindex="-1"></a>]<span class="sc">:</span></span>
<span id="cb590-128"><a href="biopython-genomediagram.html#cb590-128" aria-hidden="true" tabindex="-1"></a>    features_X <span class="ot">=</span> records[X].features</span>
<span id="cb590-129"><a href="biopython-genomediagram.html#cb590-129" aria-hidden="true" tabindex="-1"></a>    features_Y <span class="ot">=</span> records[Y].features</span>
<span id="cb590-130"><a href="biopython-genomediagram.html#cb590-130" aria-hidden="true" tabindex="-1"></a>    set_X <span class="ot">=</span> feature_sets[X]</span>
<span id="cb590-131"><a href="biopython-genomediagram.html#cb590-131" aria-hidden="true" tabindex="-1"></a>    set_Y <span class="ot">=</span> feature_sets[Y]</span>
<span id="cb590-132"><a href="biopython-genomediagram.html#cb590-132" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> score, x, y <span class="cf">in</span> X_vs_Y<span class="sc">:</span></span>
<span id="cb590-133"><a href="biopython-genomediagram.html#cb590-133" aria-hidden="true" tabindex="-1"></a>        color <span class="ot">=</span> <span class="fu">colors.linearlyInterpolatedColor</span>(</span>
<span id="cb590-134"><a href="biopython-genomediagram.html#cb590-134" aria-hidden="true" tabindex="-1"></a>            colors.white, colors.firebrick, <span class="dv">0</span>, <span class="dv">100</span>, score</span>
<span id="cb590-135"><a href="biopython-genomediagram.html#cb590-135" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb590-136"><a href="biopython-genomediagram.html#cb590-136" aria-hidden="true" tabindex="-1"></a>        border <span class="ot">=</span> colors.lightgrey</span>
<span id="cb590-137"><a href="biopython-genomediagram.html#cb590-137" aria-hidden="true" tabindex="-1"></a>        f_x <span class="ot">=</span> <span class="fu">get_feature</span>(features_X, x)</span>
<span id="cb590-138"><a href="biopython-genomediagram.html#cb590-138" aria-hidden="true" tabindex="-1"></a>        F_x <span class="ot">=</span> <span class="fu">set_X.add_feature</span>(</span>
<span id="cb590-139"><a href="biopython-genomediagram.html#cb590-139" aria-hidden="true" tabindex="-1"></a>            <span class="fu">SeqFeature</span>(<span class="fu">FeatureLocation</span>(f_x.location.start, f_x.location.end, <span class="at">strand=</span><span class="dv">0</span>)),</span>
<span id="cb590-140"><a href="biopython-genomediagram.html#cb590-140" aria-hidden="true" tabindex="-1"></a>            <span class="at">color=</span>color,</span>
<span id="cb590-141"><a href="biopython-genomediagram.html#cb590-141" aria-hidden="true" tabindex="-1"></a>            <span class="at">border=</span>border,</span>
<span id="cb590-142"><a href="biopython-genomediagram.html#cb590-142" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb590-143"><a href="biopython-genomediagram.html#cb590-143" aria-hidden="true" tabindex="-1"></a>        f_y <span class="ot">=</span> <span class="fu">get_feature</span>(features_Y, y)</span>
<span id="cb590-144"><a href="biopython-genomediagram.html#cb590-144" aria-hidden="true" tabindex="-1"></a>        F_y <span class="ot">=</span> <span class="fu">set_Y.add_feature</span>(</span>
<span id="cb590-145"><a href="biopython-genomediagram.html#cb590-145" aria-hidden="true" tabindex="-1"></a>            <span class="fu">SeqFeature</span>(<span class="fu">FeatureLocation</span>(f_y.location.start, f_y.location.end, <span class="at">strand=</span><span class="dv">0</span>)),</span>
<span id="cb590-146"><a href="biopython-genomediagram.html#cb590-146" aria-hidden="true" tabindex="-1"></a>            <span class="at">color=</span>color,</span>
<span id="cb590-147"><a href="biopython-genomediagram.html#cb590-147" aria-hidden="true" tabindex="-1"></a>            <span class="at">border=</span>border,</span>
<span id="cb590-148"><a href="biopython-genomediagram.html#cb590-148" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb590-149"><a href="biopython-genomediagram.html#cb590-149" aria-hidden="true" tabindex="-1"></a>        <span class="fu">gd_diagram.cross_track_links.append</span>(<span class="fu">CrossLink</span>(F_x, F_y, color, border))</span>
<span id="cb590-150"><a href="biopython-genomediagram.html#cb590-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-151"><a href="biopython-genomediagram.html#cb590-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-152"><a href="biopython-genomediagram.html#cb590-152" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> record, gene_colors <span class="cf">in</span> <span class="fu">zip</span>([A_rec, B_rec, C_rec], [A_colors, B_colors, C_colors])<span class="sc">:</span></span>
<span id="cb590-153"><a href="biopython-genomediagram.html#cb590-153" aria-hidden="true" tabindex="-1"></a>    gd_feature_set <span class="ot">=</span> feature_sets[record.name]</span>
<span id="cb590-154"><a href="biopython-genomediagram.html#cb590-154" aria-hidden="true" tabindex="-1"></a>    i <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb590-155"><a href="biopython-genomediagram.html#cb590-155" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> feature <span class="cf">in</span> record.features<span class="sc">:</span></span>
<span id="cb590-156"><a href="biopython-genomediagram.html#cb590-156" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span> feature.type <span class="sc">!=</span> <span class="st">&quot;gene&quot;</span><span class="sc">:</span></span>
<span id="cb590-157"><a href="biopython-genomediagram.html#cb590-157" aria-hidden="true" tabindex="-1"></a>            <span class="co"># Exclude this feature</span></span>
<span id="cb590-158"><a href="biopython-genomediagram.html#cb590-158" aria-hidden="true" tabindex="-1"></a>            continue</span>
<span id="cb590-159"><a href="biopython-genomediagram.html#cb590-159" aria-hidden="true" tabindex="-1"></a>        try<span class="sc">:</span></span>
<span id="cb590-160"><a href="biopython-genomediagram.html#cb590-160" aria-hidden="true" tabindex="-1"></a>            g_color <span class="ot">=</span> gene_colors[i]</span>
<span id="cb590-161"><a href="biopython-genomediagram.html#cb590-161" aria-hidden="true" tabindex="-1"></a>        except IndexError<span class="sc">:</span></span>
<span id="cb590-162"><a href="biopython-genomediagram.html#cb590-162" aria-hidden="true" tabindex="-1"></a>            <span class="fu">print</span>(<span class="st">&quot;Don&#39;t have color for %s gene %i&quot;</span> % (record.name, i))</span>
<span id="cb590-163"><a href="biopython-genomediagram.html#cb590-163" aria-hidden="true" tabindex="-1"></a>            g_color <span class="ot">=</span> grey</span>
<span id="cb590-164"><a href="biopython-genomediagram.html#cb590-164" aria-hidden="true" tabindex="-1"></a>        <span class="fu">gd_feature_set.add_feature</span>(</span>
<span id="cb590-165"><a href="biopython-genomediagram.html#cb590-165" aria-hidden="true" tabindex="-1"></a>            feature,</span>
<span id="cb590-166"><a href="biopython-genomediagram.html#cb590-166" aria-hidden="true" tabindex="-1"></a>            <span class="at">sigil=</span><span class="st">&quot;BIGARROW&quot;</span>,</span>
<span id="cb590-167"><a href="biopython-genomediagram.html#cb590-167" aria-hidden="true" tabindex="-1"></a>            <span class="at">color=</span>g_color,</span>
<span id="cb590-168"><a href="biopython-genomediagram.html#cb590-168" aria-hidden="true" tabindex="-1"></a>            <span class="at">label=</span>True,</span>
<span id="cb590-169"><a href="biopython-genomediagram.html#cb590-169" aria-hidden="true" tabindex="-1"></a>            <span class="at">name=</span><span class="fu">str</span>(i <span class="sc">+</span> <span class="dv">1</span>),</span>
<span id="cb590-170"><a href="biopython-genomediagram.html#cb590-170" aria-hidden="true" tabindex="-1"></a>            <span class="at">label_position=</span><span class="st">&quot;start&quot;</span>,</span>
<span id="cb590-171"><a href="biopython-genomediagram.html#cb590-171" aria-hidden="true" tabindex="-1"></a>            <span class="at">label_size=</span><span class="dv">6</span>,</span>
<span id="cb590-172"><a href="biopython-genomediagram.html#cb590-172" aria-hidden="true" tabindex="-1"></a>            <span class="at">label_angle=</span><span class="dv">0</span>,</span>
<span id="cb590-173"><a href="biopython-genomediagram.html#cb590-173" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb590-174"><a href="biopython-genomediagram.html#cb590-174" aria-hidden="true" tabindex="-1"></a>        i <span class="sc">+</span><span class="er">=</span> <span class="dv">1</span></span>
<span id="cb590-175"><a href="biopython-genomediagram.html#cb590-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb590-176"><a href="biopython-genomediagram.html#cb590-176" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.draw</span>(<span class="at">format=</span><span class="st">&quot;linear&quot;</span>, <span class="at">pagesize=</span><span class="st">&quot;A4&quot;</span>, <span class="at">fragments=</span><span class="dv">1</span>, <span class="at">start=</span><span class="dv">0</span>, <span class="at">end=</span>max_len)</span>
<span id="cb590-177"><a href="biopython-genomediagram.html#cb590-177" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.write</span>(name <span class="sc">+</span> <span class="st">&quot;.pdf&quot;</span>, <span class="st">&quot;PDF&quot;</span>)</span>
<span id="cb590-178"><a href="biopython-genomediagram.html#cb590-178" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.write</span>(name <span class="sc">+</span> <span class="st">&quot;.eps&quot;</span>, <span class="st">&quot;EPS&quot;</span>)</span>
<span id="cb590-179"><a href="biopython-genomediagram.html#cb590-179" aria-hidden="true" tabindex="-1"></a><span class="fu">gd_diagram.write</span>(name <span class="sc">+</span> <span class="st">&quot;.svg&quot;</span>, <span class="st">&quot;SVG&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="biopython-mulitple-sequence-alignments.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="biopython-chromosome-modeling.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/23-Biopython_GenomeDiagram.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
