<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Session 12 – Estimating Phylogenetic Trees | Bioinformatics Workshop Gitbook</title>
  <meta name="description" content="none" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Session 12 – Estimating Phylogenetic Trees | Bioinformatics Workshop Gitbook" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="none" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Session 12 – Estimating Phylogenetic Trees | Bioinformatics Workshop Gitbook" />
  
  <meta name="twitter:description" content="none" />
  

<meta name="author" content="Juan C. Santos &amp; Randy Ortiz" />


<meta name="date" content="2024-04-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="sequence-alignments.html"/>
<link rel="next" href="mining-from-a-reference-dataset.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics Workshop Gitbook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html"><i class="fa fa-check"></i><b>1</b> – Getting Started with R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#why-do-we-use-r"><i class="fa fa-check"></i><b>1.1</b> Why do we use R?</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#get-a-good-text-editor"><i class="fa fa-check"></i><b>1.2</b> Get a good text editor</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-r"><i class="fa fa-check"></i><b>1.4</b> Installing R</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#annotating-code"><i class="fa fa-check"></i><b>1.5</b> Annotating code</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#defining-a-working-directory-in-r"><i class="fa fa-check"></i><b>1.6</b> Defining a working directory in R</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html"><i class="fa fa-check"></i><b>2</b> – Using R, Installing Packages and Importing/Exporting Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#your-first-session"><i class="fa fa-check"></i><b>2.1</b> Your first session</a></li>
<li class="chapter" data-level="2.2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#doing-some-base-functions-and-getting-help"><i class="fa fa-check"></i><b>2.2</b> Doing some base functions and getting help</a></li>
<li class="chapter" data-level="2.3" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#functions-between-two-or-more-vectors"><i class="fa fa-check"></i><b>2.3</b> Functions between two or more vectors</a></li>
<li class="chapter" data-level="2.4" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#about-r-packages"><i class="fa fa-check"></i><b>2.4</b> About R-packages</a></li>
<li class="chapter" data-level="2.5" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#cran-task-views-expert-opinion-on-useful-packages"><i class="fa fa-check"></i><b>2.5</b> CRAN Task Views – Expert opinion on useful packages</a></li>
<li class="chapter" data-level="2.6" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#installing-packages"><i class="fa fa-check"></i><b>2.6</b> Installing packages</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html"><i class="fa fa-check"></i><b>3</b> – Basic Data Structures in R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#scalars-and-vectors"><i class="fa fa-check"></i><b>3.1</b> Scalars and vectors</a></li>
<li class="chapter" data-level="3.2" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#arrays"><i class="fa fa-check"></i><b>3.3</b> Arrays</a></li>
<li class="chapter" data-level="3.4" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a></li>
<li class="chapter" data-level="3.5" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tables"><i class="fa fa-check"></i><b>3.5</b> Tables</a></li>
<li class="chapter" data-level="3.6" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#lists"><i class="fa fa-check"></i><b>3.6</b> Lists</a></li>
<li class="chapter" data-level="3.7" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-tables"><i class="fa fa-check"></i><b>3.7</b> Data Tables</a></li>
<li class="chapter" data-level="3.8" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tibbles"><i class="fa fa-check"></i><b>3.8</b> Tibbles</a></li>
<li class="chapter" data-level="3.9" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#dnastringsets-rnastringsets-and-aastringsets"><i class="fa fa-check"></i><b>3.9</b> DNAStringSets, RNAStringSets and AAStringSets</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html"><i class="fa fa-check"></i><b>4</b> – Text Editing and Data Transformations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#working-with-character-strings"><i class="fa fa-check"></i><b>4.1</b> Working with character strings</a></li>
<li class="chapter" data-level="4.2" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#regular-expressions-regex"><i class="fa fa-check"></i><b>4.2</b> Regular expressions (Regex)</a></li>
<li class="chapter" data-level="4.3" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#comparing-ordering-and-simplifying-vectors"><i class="fa fa-check"></i><b>4.3</b> Comparing, ordering and simplifying vectors</a></li>
<li class="chapter" data-level="4.4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#useful-functions-of-stringr"><i class="fa fa-check"></i><b>4.4</b> Useful functions of <code>stringr</code></a></li>
<li class="chapter" data-level="4.5" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#data-transformations"><i class="fa fa-check"></i><b>4.5</b> Data transformations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html"><i class="fa fa-check"></i><b>5</b> – Getting Biological Data from Public Repositories</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#ncbi-databases-genbank-and-others"><i class="fa fa-check"></i><b>5.1</b> NCBI databases: GenBank and others</a></li>
<li class="chapter" data-level="5.2" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#accessing-ncbi-within-r-rentrez"><i class="fa fa-check"></i><b>5.2</b> Accessing NCBI within R: <code>rentrez</code></a></li>
<li class="chapter" data-level="5.3" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#retrieving-pubmed-references"><i class="fa fa-check"></i><b>5.3</b> Retrieving PubMed references</a></li>
<li class="chapter" data-level="5.4" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#comparing-sequences-blast"><i class="fa fa-check"></i><b>5.4</b> Comparing sequences: BLAST</a></li>
<li class="chapter" data-level="5.5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-uniprot"><i class="fa fa-check"></i><b>5.5</b> Other databases: UniProt</a></li>
<li class="chapter" data-level="5.6" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-ensembl"><i class="fa fa-check"></i><b>5.6</b> Other databases: Ensembl</a></li>
<li class="chapter" data-level="5.7" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-taxonomy-phylogenetic-trees-and-species-distributions"><i class="fa fa-check"></i><b>5.7</b> Other databases: Taxonomy, phylogenetic trees and species distributions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html"><i class="fa fa-check"></i><b>6</b> – Basic Statistics in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#exploring-your-data"><i class="fa fa-check"></i><b>6.1</b> Exploring your data</a></li>
<li class="chapter" data-level="6.2" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#sub-setting-your-dataset"><i class="fa fa-check"></i><b>6.2</b> Sub-setting your dataset</a></li>
<li class="chapter" data-level="6.3" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#transposing-rotating-your-dataset"><i class="fa fa-check"></i><b>6.3</b> Transposing (rotating) your dataset</a></li>
<li class="chapter" data-level="6.4" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#dealing-with-missing-na-values"><i class="fa fa-check"></i><b>6.4</b> Dealing with missing (NA) values</a></li>
<li class="chapter" data-level="6.5" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#about-p-values"><i class="fa fa-check"></i><b>6.5</b> About p-values</a></li>
<li class="chapter" data-level="6.6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#how-to-read-p-values"><i class="fa fa-check"></i><b>6.6</b> How to read p-values</a></li>
<li class="chapter" data-level="6.7" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#multiple-testing-corrections"><i class="fa fa-check"></i><b>6.7</b> Multiple testing corrections</a></li>
<li class="chapter" data-level="6.8" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#normalizing-your-data"><i class="fa fa-check"></i><b>6.8</b> Normalizing your data</a></li>
<li class="chapter" data-level="6.9" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#what-test-should-i-use"><i class="fa fa-check"></i><b>6.9</b> What test should I use?</a></li>
<li class="chapter" data-level="6.10" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#type-of-variables-in-dataset"><i class="fa fa-check"></i><b>6.10</b> Type of variables in dataset</a></li>
<li class="chapter" data-level="6.11" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-means-t-test"><i class="fa fa-check"></i><b>6.11</b> Comparing means: t-test</a></li>
<li class="chapter" data-level="6.12" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#two-sample-t-test"><i class="fa fa-check"></i><b>6.12</b> Two sample t-test</a></li>
<li class="chapter" data-level="6.13" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-wilcoxon-signed-rank-test"><i class="fa fa-check"></i><b>6.13</b> Beyond t-test: Wilcoxon signed rank test</a></li>
<li class="chapter" data-level="6.14" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#tabulation-and-cross-tabulation"><i class="fa fa-check"></i><b>6.14</b> Tabulation and Cross-Tabulation</a></li>
<li class="chapter" data-level="6.15" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-binomial-test"><i class="fa fa-check"></i><b>6.15</b> Beyond t-test: Binomial test</a></li>
<li class="chapter" data-level="6.16" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-chi-square-goodness-of-fit"><i class="fa fa-check"></i><b>6.16</b> Beyond t-test: Chi-square goodness of fit</a></li>
<li class="chapter" data-level="6.17" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-mcnemars-test"><i class="fa fa-check"></i><b>6.17</b> Beyond t-test: McNemar’s test</a></li>
<li class="chapter" data-level="6.18" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-fishers-exact-test"><i class="fa fa-check"></i><b>6.18</b> Beyond t-test: Fisher’s exact test</a></li>
<li class="chapter" data-level="6.19" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#correlations"><i class="fa fa-check"></i><b>6.19</b> Correlations</a></li>
<li class="chapter" data-level="6.20" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-distributions"><i class="fa fa-check"></i><b>6.20</b> Comparing distributions</a></li>
<li class="chapter" data-level="6.21" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#bootstraps"><i class="fa fa-check"></i><b>6.21</b> Bootstraps</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html"><i class="fa fa-check"></i><b>7</b> – Regressions and ANOVA in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#linear-regressions"><i class="fa fa-check"></i><b>7.1</b> Linear regressions</a></li>
<li class="chapter" data-level="7.2" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anatomy-of-a-linear-regression"><i class="fa fa-check"></i><b>7.2</b> Anatomy of a linear regression</a></li>
<li class="chapter" data-level="7.3" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#multiple-linear-regressions"><i class="fa fa-check"></i><b>7.3</b> Multiple linear regressions</a></li>
<li class="chapter" data-level="7.4" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#diagnostic-plots-for-regressions"><i class="fa fa-check"></i><b>7.4</b> Diagnostic plots for regressions</a></li>
<li class="chapter" data-level="7.5" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-influential-measurements"><i class="fa fa-check"></i><b>7.5</b> Regression influential measurements</a></li>
<li class="chapter" data-level="7.6" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-coefficients-and-their-95ci"><i class="fa fa-check"></i><b>7.6</b> Regression coefficients and their 95%CI</a></li>
<li class="chapter" data-level="7.7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#selecting-predictors"><i class="fa fa-check"></i><b>7.7</b> Selecting predictors</a></li>
<li class="chapter" data-level="7.8" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-predictions"><i class="fa fa-check"></i><b>7.8</b> Regression predictions</a></li>
<li class="chapter" data-level="7.9" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions"><i class="fa fa-check"></i><b>7.9</b> Logistic regressions</a></li>
<li class="chapter" data-level="7.10" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions-requirements"><i class="fa fa-check"></i><b>7.10</b> Logistic regressions requirements</a></li>
<li class="chapter" data-level="7.11" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#non-linear-regressions"><i class="fa fa-check"></i><b>7.11</b> Non-linear regressions</a></li>
<li class="chapter" data-level="7.12" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-regressions"><i class="fa fa-check"></i><b>7.12</b> Toy datasets for regressions</a></li>
<li class="chapter" data-level="7.13" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova"><i class="fa fa-check"></i><b>7.13</b> ANOVA</a></li>
<li class="chapter" data-level="7.14" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#one-way-anova"><i class="fa fa-check"></i><b>7.14</b> One-way ANOVA</a></li>
<li class="chapter" data-level="7.15" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#two-way-anova"><i class="fa fa-check"></i><b>7.15</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="7.16" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova-post-hoc-tests"><i class="fa fa-check"></i><b>7.16</b> ANOVA Post-hoc tests</a></li>
<li class="chapter" data-level="7.17" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#beyond-anova-kruskal-wallis-test"><i class="fa fa-check"></i><b>7.17</b> Beyond ANOVA: Kruskal-Wallis Test</a></li>
<li class="chapter" data-level="7.18" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-anova"><i class="fa fa-check"></i><b>7.18</b> Toy datasets for ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html"><i class="fa fa-check"></i><b>8</b> – Multidimensional Data in R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#principal-component-analysis"><i class="fa fa-check"></i><b>8.1</b> Principal Component Analysis</a></li>
<li class="chapter" data-level="8.2" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#multidimensional-scaling"><i class="fa fa-check"></i><b>8.2</b> Multidimensional scaling</a></li>
<li class="chapter" data-level="8.3" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-dfa"><i class="fa fa-check"></i><b>8.3</b> Cluster analysis: DFA</a></li>
<li class="chapter" data-level="8.4" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-k-means-unsupervised"><i class="fa fa-check"></i><b>8.4</b> Cluster analysis: <em>k</em>-means (unsupervised)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html"><i class="fa fa-check"></i><b>9</b> – Graphics and Colors</a>
<ul>
<li class="chapter" data-level="9.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#avoid-bad-graphics"><i class="fa fa-check"></i><b>9.1</b> Avoid bad graphics</a></li>
<li class="chapter" data-level="9.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#graphics-in-r-ggplot2"><i class="fa fa-check"></i><b>9.2</b> Graphics in R – ggplot2</a></li>
<li class="chapter" data-level="9.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#preparing-your-dataset"><i class="fa fa-check"></i><b>9.3</b> Preparing your dataset</a></li>
<li class="chapter" data-level="9.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#color-palettes-online"><i class="fa fa-check"></i><b>9.4</b> Color palettes online</a></li>
<li class="chapter" data-level="9.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggplot2-default-palette"><i class="fa fa-check"></i><b>9.5</b> <code>ggplot2</code> default palette</a></li>
<li class="chapter" data-level="9.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#heatmaps-or-tiled-charts"><i class="fa fa-check"></i><b>9.6</b> Heatmaps or tiled charts</a></li>
<li class="chapter" data-level="9.7" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#r-packages-about-color"><i class="fa fa-check"></i><b>9.7</b> R-packages about color</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#viridis-package"><i class="fa fa-check"></i><b>9.7.1</b> <code>viridis</code> package</a></li>
<li class="chapter" data-level="9.7.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcolorbrewer-package"><i class="fa fa-check"></i><b>9.7.2</b> <code>RColorBrewer</code> package</a></li>
<li class="chapter" data-level="9.7.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggsci-package"><i class="fa fa-check"></i><b>9.7.3</b> <code>ggsci</code> package</a></li>
<li class="chapter" data-level="9.7.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#wesanderson-package"><i class="fa fa-check"></i><b>9.7.4</b> <code>wesanderson</code> package</a></li>
<li class="chapter" data-level="9.7.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcartocolor-package"><i class="fa fa-check"></i><b>9.7.5</b> <code>rcartocolor</code> package</a></li>
<li class="chapter" data-level="9.7.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#colorspace-package"><i class="fa fa-check"></i><b>9.7.6</b> <code>colorspace</code> package</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#making-your-own-palette"><i class="fa fa-check"></i><b>9.8</b> Making your own palette</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="basic-graphics.html"><a href="basic-graphics.html"><i class="fa fa-check"></i><b>10</b> – Basic Graphics</a>
<ul>
<li class="chapter" data-level="10.1" data-path="basic-graphics.html"><a href="basic-graphics.html#what-would-you-like-to-show"><i class="fa fa-check"></i><b>10.1</b> What would you like to show?</a></li>
<li class="chapter" data-level="10.2" data-path="basic-graphics.html"><a href="basic-graphics.html#bar-charts"><i class="fa fa-check"></i><b>10.2</b> Bar charts</a></li>
<li class="chapter" data-level="10.3" data-path="basic-graphics.html"><a href="basic-graphics.html#stacked-bar-charts"><i class="fa fa-check"></i><b>10.3</b> Stacked bar charts</a></li>
<li class="chapter" data-level="10.4" data-path="basic-graphics.html"><a href="basic-graphics.html#histograms"><i class="fa fa-check"></i><b>10.4</b> Histograms</a></li>
<li class="chapter" data-level="10.5" data-path="basic-graphics.html"><a href="basic-graphics.html#multiple-histograms"><i class="fa fa-check"></i><b>10.5</b> Multiple histograms</a></li>
<li class="chapter" data-level="10.6" data-path="basic-graphics.html"><a href="basic-graphics.html#density-plots"><i class="fa fa-check"></i><b>10.6</b> Density plots</a></li>
<li class="chapter" data-level="10.7" data-path="basic-graphics.html"><a href="basic-graphics.html#box-plots"><i class="fa fa-check"></i><b>10.7</b> Box plots</a></li>
<li class="chapter" data-level="10.8" data-path="basic-graphics.html"><a href="basic-graphics.html#violin-plots"><i class="fa fa-check"></i><b>10.8</b> Violin plots</a></li>
<li class="chapter" data-level="10.9" data-path="basic-graphics.html"><a href="basic-graphics.html#scatterplots"><i class="fa fa-check"></i><b>10.9</b> Scatterplots</a></li>
<li class="chapter" data-level="10.10" data-path="basic-graphics.html"><a href="basic-graphics.html#lines-and-scatterplots"><i class="fa fa-check"></i><b>10.10</b> Lines and Scatterplots</a></li>
<li class="chapter" data-level="10.11" data-path="basic-graphics.html"><a href="basic-graphics.html#save-your-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>10.11</b> Save your plots as PDF or PNG</a></li>
<li class="chapter" data-level="10.12" data-path="basic-graphics.html"><a href="basic-graphics.html#ggplot2-add-on-packages-treemapify"><i class="fa fa-check"></i><b>10.12</b> ggplot2 add-on packages: <code>treemapify</code></a></li>
<li class="chapter" data-level="10.13" data-path="basic-graphics.html"><a href="basic-graphics.html#other-resources"><i class="fa fa-check"></i><b>10.13</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="sequence-alignments.html"><a href="sequence-alignments.html"><i class="fa fa-check"></i><b>11</b> – Sequence Alignments</a>
<ul>
<li class="chapter" data-level="11.1" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-using-rentrez"><i class="fa fa-check"></i><b>11.1</b> Retrieving sequences using ‘rentrez’</a></li>
<li class="chapter" data-level="11.2" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-manually-from-ncbi"><i class="fa fa-check"></i><b>11.2</b> Retrieving sequences manually from NCBI</a></li>
<li class="chapter" data-level="11.3" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-accession-numbers"><i class="fa fa-check"></i><b>11.3</b> Retrieving accession numbers</a></li>
<li class="chapter" data-level="11.4" data-path="sequence-alignments.html"><a href="sequence-alignments.html#alignment-with-decipher"><i class="fa fa-check"></i><b>11.4</b> Alignment with <code>DECIPHER</code></a></li>
<li class="chapter" data-level="11.5" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualize-alignment"><i class="fa fa-check"></i><b>11.5</b> Visualize alignment</a></li>
<li class="chapter" data-level="11.6" data-path="sequence-alignments.html"><a href="sequence-alignments.html#save-alignment-to-file"><i class="fa fa-check"></i><b>11.6</b> Save alignment to file</a></li>
<li class="chapter" data-level="11.7" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-and-align-several-markers"><i class="fa fa-check"></i><b>11.7</b> Concatenate and align several markers</a></li>
<li class="chapter" data-level="11.8" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-data.table"><i class="fa fa-check"></i><b>11.8</b> Concatenate with <code>data.table</code></a></li>
<li class="chapter" data-level="11.9" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-biostrings"><i class="fa fa-check"></i><b>11.9</b> Concatenate with <code>Biostrings</code></a></li>
<li class="chapter" data-level="11.10" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualizing-alignments-in-r"><i class="fa fa-check"></i><b>11.10</b> Visualizing alignments in R</a></li>
<li class="chapter" data-level="11.11" data-path="sequence-alignments.html"><a href="sequence-alignments.html#other-sequence-alignment-software"><i class="fa fa-check"></i><b>11.11</b> Other sequence alignment software</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html"><i class="fa fa-check"></i><b>12</b> – Estimating Phylogenetic Trees</a>
<ul>
<li class="chapter" data-level="12.1" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#phylogeny-inference-with-iqtree2"><i class="fa fa-check"></i><b>12.1</b> Phylogeny inference with <code>iqtree2</code></a></li>
<li class="chapter" data-level="12.2" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-iqtree2-in-macos"><i class="fa fa-check"></i><b>12.2</b> Installing <code>iqtree2</code> in macOS</a></li>
<li class="chapter" data-level="12.3" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-and-running-iqtree2-in-a-pc"><i class="fa fa-check"></i><b>12.3</b> Installing and running <code>iqtree2</code> in a PC</a></li>
<li class="chapter" data-level="12.4" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#loading-the-iqtree_runner_r-function"><i class="fa fa-check"></i><b>12.4</b> Loading the <code>iqtree_runner_R</code> function</a></li>
<li class="chapter" data-level="12.5" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#running-iqtree2-from-r"><i class="fa fa-check"></i><b>12.5</b> Running <code>iqtree2</code> from R</a></li>
<li class="chapter" data-level="12.6" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-no-partitions"><i class="fa fa-check"></i><b>12.6</b> One marker and no partitions</a></li>
<li class="chapter" data-level="12.7" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-with-partitions"><i class="fa fa-check"></i><b>12.7</b> One marker and with partitions</a></li>
<li class="chapter" data-level="12.8" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#bootstrap-support"><i class="fa fa-check"></i><b>12.8</b> Bootstrap support</a></li>
<li class="chapter" data-level="12.9" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#many-markers-and-with-partitions"><i class="fa fa-check"></i><b>12.9</b> Many markers and with partitions</a></li>
<li class="chapter" data-level="12.10" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#other-sequence-phylogenetic-tree-estimation-software"><i class="fa fa-check"></i><b>12.10</b> Other sequence phylogenetic tree estimation software</a></li>
<li class="chapter" data-level="12.11" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#drawing-phylogenetic-trees-with-ggtree"><i class="fa fa-check"></i><b>12.11</b> Drawing phylogenetic trees with <code>ggtree</code></a></li>
<li class="chapter" data-level="12.12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#save-ggtree-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>12.12</b> Save ggtree plots as PDF or PNG</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html"><i class="fa fa-check"></i><b>13</b> – Mining from a reference dataset</a>
<ul>
<li class="chapter" data-level="13.1" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#a-worked-example-using-the-mtcars-dataset"><i class="fa fa-check"></i><b>13.1</b> A worked example using the <code>mtcars</code> dataset</a></li>
<li class="chapter" data-level="13.2" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-reference-dataset"><i class="fa fa-check"></i><b>13.2</b> Build a reference dataset</a></li>
<li class="chapter" data-level="13.3" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-sampling-dataset"><i class="fa fa-check"></i><b>13.3</b> Build a sampling dataset</a></li>
<li class="chapter" data-level="13.4" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#extracting-from-reference-dataset"><i class="fa fa-check"></i><b>13.4</b> Extracting from reference dataset</a></li>
<li class="chapter" data-level="13.5" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#plots-of-the-extracted-data"><i class="fa fa-check"></i><b>13.5</b> Plots of the extracted data</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="introduction-to-python.html"><a href="introduction-to-python.html"><i class="fa fa-check"></i><b>14</b> – Introduction to Python</a>
<ul>
<li class="chapter" data-level="14.1" data-path="introduction-to-python.html"><a href="introduction-to-python.html#welcome-to-python-why-python"><i class="fa fa-check"></i><b>14.1</b> Welcome to Python! Why Python?</a></li>
<li class="chapter" data-level="14.2" data-path="introduction-to-python.html"><a href="introduction-to-python.html#which-python-should-i-install"><i class="fa fa-check"></i><b>14.2</b> Which Python should I install?</a></li>
<li class="chapter" data-level="14.3" data-path="introduction-to-python.html"><a href="introduction-to-python.html#mac-os-x-python-installations"><i class="fa fa-check"></i><b>14.3</b> Mac OS X Python Installations</a></li>
<li class="chapter" data-level="14.4" data-path="introduction-to-python.html"><a href="introduction-to-python.html#windows-python-installations"><i class="fa fa-check"></i><b>14.4</b> Windows Python Installations</a></li>
<li class="chapter" data-level="14.5" data-path="introduction-to-python.html"><a href="introduction-to-python.html#install-an-ide-or-integrated-development-environment"><i class="fa fa-check"></i><b>14.5</b> Install an IDE, or integrated development environment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html"><i class="fa fa-check"></i><b>15</b> – Python: Keywords and Identifiers</a>
<ul>
<li class="chapter" data-level="15.1" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#keywords-in-python"><i class="fa fa-check"></i><b>15.1</b> Keywords in Python</a></li>
<li class="chapter" data-level="15.2" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#identifiers-in-python"><i class="fa fa-check"></i><b>15.2</b> Identifiers in Python</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html"><i class="fa fa-check"></i><b>16</b> – Python: Statements and Expressions</a>
<ul>
<li class="chapter" data-level="16.1" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#what-are-statements"><i class="fa fa-check"></i><b>16.1</b> What are statements?</a></li>
<li class="chapter" data-level="16.2" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#expressions-in-python"><i class="fa fa-check"></i><b>16.2</b> Expressions in Python</a></li>
<li class="chapter" data-level="16.3" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#simple-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.3</b> Simple Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.4" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#augmented-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.4</b> Augmented Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.5" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#explicit-and-implicit-line-continuation"><i class="fa fa-check"></i><b>16.5</b> Explicit and Implicit Line Continuation</a></li>
<li class="chapter" data-level="16.6" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#indentation-in-python-is-crucial-to-proper-code-function"><i class="fa fa-check"></i><b>16.6</b> Indentation in Python is Crucial to Proper Code function</a></li>
<li class="chapter" data-level="16.7" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#comments"><i class="fa fa-check"></i><b>16.7</b> Comments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="python-data-types.html"><a href="python-data-types.html"><i class="fa fa-check"></i><b>17</b> – Python: Data Types</a>
<ul>
<li class="chapter" data-level="17.1" data-path="python-data-types.html"><a href="python-data-types.html#booleans"><i class="fa fa-check"></i><b>17.1</b> Booleans</a></li>
<li class="chapter" data-level="17.2" data-path="python-data-types.html"><a href="python-data-types.html#numbers-int-float-and-complex"><i class="fa fa-check"></i><b>17.2</b> Numbers: int, float and complex</a></li>
<li class="chapter" data-level="17.3" data-path="python-data-types.html"><a href="python-data-types.html#strings"><i class="fa fa-check"></i><b>17.3</b> Strings</a></li>
<li class="chapter" data-level="17.4" data-path="python-data-types.html"><a href="python-data-types.html#bytes"><i class="fa fa-check"></i><b>17.4</b> Bytes</a></li>
<li class="chapter" data-level="17.5" data-path="python-data-types.html"><a href="python-data-types.html#lists-1"><i class="fa fa-check"></i><b>17.5</b> Lists</a></li>
<li class="chapter" data-level="17.6" data-path="python-data-types.html"><a href="python-data-types.html#tuples"><i class="fa fa-check"></i><b>17.6</b> Tuples</a></li>
<li class="chapter" data-level="17.7" data-path="python-data-types.html"><a href="python-data-types.html#sets"><i class="fa fa-check"></i><b>17.7</b> Sets</a></li>
<li class="chapter" data-level="17.8" data-path="python-data-types.html"><a href="python-data-types.html#dictionary"><i class="fa fa-check"></i><b>17.8</b> Dictionary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="python-strings.html"><a href="python-strings.html"><i class="fa fa-check"></i><b>18</b> – Python: Strings</a>
<ul>
<li class="chapter" data-level="18.1" data-path="python-strings.html"><a href="python-strings.html#string-operators"><i class="fa fa-check"></i><b>18.1</b> String Operators</a></li>
<li class="chapter" data-level="18.2" data-path="python-strings.html"><a href="python-strings.html#other-string-modifiers"><i class="fa fa-check"></i><b>18.2</b> Other String Modifiers</a></li>
<li class="chapter" data-level="18.3" data-path="python-strings.html"><a href="python-strings.html#format-characters-in-strings"><i class="fa fa-check"></i><b>18.3</b> Format-Characters in Strings</a></li>
<li class="chapter" data-level="18.4" data-path="python-strings.html"><a href="python-strings.html#comparisons-with-strings"><i class="fa fa-check"></i><b>18.4</b> Comparisons with Strings</a></li>
<li class="chapter" data-level="18.5" data-path="python-strings.html"><a href="python-strings.html#padding-strings"><i class="fa fa-check"></i><b>18.5</b> Padding strings</a></li>
<li class="chapter" data-level="18.6" data-path="python-strings.html"><a href="python-strings.html#search-for-words-in-string-and-output-location"><i class="fa fa-check"></i><b>18.6</b> Search for words in string and output location</a></li>
<li class="chapter" data-level="18.7" data-path="python-strings.html"><a href="python-strings.html#seek-and-replace-in-strings"><i class="fa fa-check"></i><b>18.7</b> Seek and replace in strings</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="python-operators.html"><a href="python-operators.html"><i class="fa fa-check"></i><b>19</b> – Python: Operators</a>
<ul>
<li class="chapter" data-level="19.1" data-path="python-operators.html"><a href="python-operators.html#many-types-of-operators"><i class="fa fa-check"></i><b>19.1</b> Many types of operators</a></li>
<li class="chapter" data-level="19.2" data-path="python-operators.html"><a href="python-operators.html#arithmetic-operators"><i class="fa fa-check"></i><b>19.2</b> Arithmetic operators</a></li>
<li class="chapter" data-level="19.3" data-path="python-operators.html"><a href="python-operators.html#comparison-operators"><i class="fa fa-check"></i><b>19.3</b> Comparison operators</a></li>
<li class="chapter" data-level="19.4" data-path="python-operators.html"><a href="python-operators.html#logical-operators"><i class="fa fa-check"></i><b>19.4</b> Logical operators</a></li>
<li class="chapter" data-level="19.5" data-path="python-operators.html"><a href="python-operators.html#bitwise-operators"><i class="fa fa-check"></i><b>19.5</b> Bitwise operators</a></li>
<li class="chapter" data-level="19.6" data-path="python-operators.html"><a href="python-operators.html#identity-operators"><i class="fa fa-check"></i><b>19.6</b> Identity Operators</a></li>
<li class="chapter" data-level="19.7" data-path="python-operators.html"><a href="python-operators.html#membership-operators"><i class="fa fa-check"></i><b>19.7</b> Membership operators</a></li>
<li class="chapter" data-level="19.8" data-path="python-operators.html"><a href="python-operators.html#what-is-the-order-of-operations-with-operators"><i class="fa fa-check"></i><b>19.8</b> What is the order of operations with operators?</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="biopython.html"><a href="biopython.html"><i class="fa fa-check"></i><b>20</b> – Biopython</a>
<ul>
<li class="chapter" data-level="20.1" data-path="biopython.html"><a href="biopython.html#introduction-to-biopython"><i class="fa fa-check"></i><b>20.1</b> Introduction to Biopython</a></li>
<li class="chapter" data-level="20.2" data-path="biopython.html"><a href="biopython.html#biopython-installation"><i class="fa fa-check"></i><b>20.2</b> Biopython Installation</a></li>
<li class="chapter" data-level="20.3" data-path="biopython.html"><a href="biopython.html#getting-started"><i class="fa fa-check"></i><b>20.3</b> Getting Started</a></li>
<li class="chapter" data-level="20.4" data-path="biopython.html"><a href="biopython.html#parsing-my-fasta"><i class="fa fa-check"></i><b>20.4</b> Parsing my FASTA</a></li>
<li class="chapter" data-level="20.5" data-path="biopython.html"><a href="biopython.html#parsing-my-genbank-or-genpept"><i class="fa fa-check"></i><b>20.5</b> Parsing my GenBank or GenPept</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="biopython-sequences.html"><a href="biopython-sequences.html"><i class="fa fa-check"></i><b>21</b> – Biopython: Sequences</a>
<ul>
<li class="chapter" data-level="21.1" data-path="biopython-sequences.html"><a href="biopython-sequences.html#sequence-objects"><i class="fa fa-check"></i><b>21.1</b> Sequence Objects</a></li>
<li class="chapter" data-level="21.2" data-path="biopython-sequences.html"><a href="biopython-sequences.html#concatenation-with-sequences"><i class="fa fa-check"></i><b>21.2</b> Concatenation with Sequences</a></li>
<li class="chapter" data-level="21.3" data-path="biopython-sequences.html"><a href="biopython-sequences.html#transcription-dna-to-mrna"><i class="fa fa-check"></i><b>21.3</b> Transcription: DNA to mRNA</a></li>
<li class="chapter" data-level="21.4" data-path="biopython-sequences.html"><a href="biopython-sequences.html#translation-rna-to-prot"><i class="fa fa-check"></i><b>21.4</b> Translation: RNA to PROT</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html"><i class="fa fa-check"></i><b>22</b> – Biopython: Mulitple Sequence Alignments</a>
<ul>
<li class="chapter" data-level="22.1" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwise2"><i class="fa fa-check"></i><b>22.1</b> pairwise2</a></li>
<li class="chapter" data-level="22.2" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwisealigner"><i class="fa fa-check"></i><b>22.2</b> PairwiseAligner</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html"><i class="fa fa-check"></i><b>23</b> – Biopython: GenomeDiagram</a>
<ul>
<li class="chapter" data-level="23.1" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#linearcircular-gene-diagrams"><i class="fa fa-check"></i><b>23.1</b> Linear/Circular Gene Diagrams</a></li>
<li class="chapter" data-level="23.2" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#diagram-readability"><i class="fa fa-check"></i><b>23.2</b> Diagram Readability</a></li>
<li class="chapter" data-level="23.3" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#box-shape"><i class="fa fa-check"></i><b>23.3</b> Box Shape</a></li>
<li class="chapter" data-level="23.4" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#edit-box-shape-arrows"><i class="fa fa-check"></i><b>23.4</b> Edit Box Shape: Arrows</a></li>
<li class="chapter" data-level="23.5" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#restriction-map"><i class="fa fa-check"></i><b>23.5</b> Restriction Map</a></li>
<li class="chapter" data-level="23.6" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#tracks-on-tracks"><i class="fa fa-check"></i><b>23.6</b> Tracks on Tracks</a></li>
<li class="chapter" data-level="23.7" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#cross-link-data"><i class="fa fa-check"></i><b>23.7</b> Cross-link Data</a></li>
<li class="chapter" data-level="23.8" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#clean-cross-link"><i class="fa fa-check"></i><b>23.8</b> Clean Cross-Link</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html"><i class="fa fa-check"></i><b>24</b> – Biopython: Chromosome Modeling</a>
<ul>
<li class="chapter" data-level="24.1" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html#gene-feature-modeling"><i class="fa fa-check"></i><b>24.1</b> Gene Feature Modeling</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html"><i class="fa fa-check"></i><b>25</b> – Biopython: Sequence Parsing with Plots</a>
<ul>
<li class="chapter" data-level="25.1" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#histograms-1"><i class="fa fa-check"></i><b>25.1</b> Histograms</a></li>
<li class="chapter" data-level="25.2" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#gc-content"><i class="fa fa-check"></i><b>25.2</b> GC Content</a></li>
<li class="chapter" data-level="25.3" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#nt-alignment-dot-plot"><i class="fa fa-check"></i><b>25.3</b> Nt Alignment Dot Plot</a></li>
<li class="chapter" data-level="25.4" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#bigger-dots-for-the-dot-plot"><i class="fa fa-check"></i><b>25.4</b> Bigger Dots for the Dot Plot</a></li>
<li class="chapter" data-level="25.5" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#fastq-we-did-not-forget-you"><i class="fa fa-check"></i><b>25.5</b> FASTQ we did not forget you</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html"><i class="fa fa-check"></i><b>26</b> – Python: Web Scraping w/ BeautifulSoup</a>
<ul>
<li class="chapter" data-level="26.1" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#introduction"><i class="fa fa-check"></i><b>26.1</b> Introduction</a></li>
<li class="chapter" data-level="26.2" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#installation"><i class="fa fa-check"></i><b>26.2</b> Installation</a></li>
<li class="chapter" data-level="26.3" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#getting-started-1"><i class="fa fa-check"></i><b>26.3</b> Getting started</a></li>
<li class="chapter" data-level="26.4" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-all-matching-tags"><i class="fa fa-check"></i><b>26.4</b> Mining all matching tags</a></li>
<li class="chapter" data-level="26.5" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-tags-by-class-or-id"><i class="fa fa-check"></i><b>26.5</b> Mining tags by class or id</a></li>
<li class="chapter" data-level="26.6" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#use-css-selectors-for-positional-mining"><i class="fa fa-check"></i><b>26.6</b> Use CSS Selectors for positional mining</a></li>
<li class="chapter" data-level="26.7" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#weather-page-example"><i class="fa fa-check"></i><b>26.7</b> Weather Page Example</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="python-subprocess.html"><a href="python-subprocess.html"><i class="fa fa-check"></i><b>27</b> – Python Subprocess</a>
<ul>
<li class="chapter" data-level="27.1" data-path="python-subprocess.html"><a href="python-subprocess.html#introduction-and-installation"><i class="fa fa-check"></i><b>27.1</b> Introduction and Installation</a></li>
<li class="chapter" data-level="27.2" data-path="python-subprocess.html"><a href="python-subprocess.html#creating-your-first-python-script-w-bbtools"><i class="fa fa-check"></i><b>27.2</b> Creating your first python script w/ BBTools</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html"><i class="fa fa-check"></i><b>28</b> – Control Structures &amp; Loops in R</a>
<ul>
<li class="chapter" data-level="28.1" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#controling-the-flow"><i class="fa fa-check"></i><b>28.1</b> Controling the flow</a></li>
<li class="chapter" data-level="28.2" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if"><i class="fa fa-check"></i><b>28.2</b> Conditional <code>if</code></a></li>
<li class="chapter" data-level="28.3" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if-and-else"><i class="fa fa-check"></i><b>28.3</b> Conditional <code>if</code> and <code>else</code></a></li>
<li class="chapter" data-level="28.4" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#nested-conditionals-if-else-and-if-else"><i class="fa fa-check"></i><b>28.4</b> Nested conditionals <code>if</code>, <code>else</code>, and <code>if else</code></a></li>
<li class="chapter" data-level="28.5" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-ifelse"><i class="fa fa-check"></i><b>28.5</b> Conditional <code>ifelse</code></a></li>
<li class="chapter" data-level="28.6" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#loops-that-control-iterations-and-repeations"><i class="fa fa-check"></i><b>28.6</b> Loops that control iterations and repeations</a></li>
<li class="chapter" data-level="28.7" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-for-loop"><i class="fa fa-check"></i><b>28.7</b> The <code>for</code> Loop</a></li>
<li class="chapter" data-level="28.8" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-while-loop"><i class="fa fa-check"></i><b>28.8</b> The <code>while</code> Loop</a></li>
<li class="chapter" data-level="28.9" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-repeat-loop"><i class="fa fa-check"></i><b>28.9</b> The <code>repeat</code> Loop</a></li>
<li class="chapter" data-level="28.10" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-next-and-break-statements"><i class="fa fa-check"></i><b>28.10</b> The <code>next</code> and <code>break</code> statements</a></li>
<li class="chapter" data-level="28.11" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functions"><i class="fa fa-check"></i><b>28.11</b> Other loop functions</a></li>
<li class="chapter" data-level="28.12" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#lapply-functions"><i class="fa fa-check"></i><b>28.12</b> <code>lapply</code> functions</a></li>
<li class="chapter" data-level="28.13" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsapply"><i class="fa fa-check"></i><b>28.13</b> Other loop functions:<code>apply</code></a></li>
<li class="chapter" data-level="28.14" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsmapply"><i class="fa fa-check"></i><b>28.14</b> Other loop functions:<code>mapply</code></a></li>
<li class="chapter" data-level="28.15" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionstapply"><i class="fa fa-check"></i><b>28.15</b> Other loop functions:<code>tapply</code></a></li>
<li class="chapter" data-level="28.16" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionssplit"><i class="fa fa-check"></i><b>28.16</b> Other loop functions:<code>split</code></a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html"><i class="fa fa-check"></i><b>29</b> – Bioinformatics Literature</a>
<ul>
<li class="chapter" data-level="29.1" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#peer-reviewed-bioinformatics-journals"><i class="fa fa-check"></i><b>29.1</b> Peer-reviewed bioinformatics journals</a></li>
<li class="chapter" data-level="29.2" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#access-to-journals-articles"><i class="fa fa-check"></i><b>29.2</b> Access to journals articles</a></li>
<li class="chapter" data-level="29.3" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#informal-access-to-code-preprints-and-vignettes"><i class="fa fa-check"></i><b>29.3</b> Informal access to code: Preprints and vignettes</a></li>
<li class="chapter" data-level="29.4" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#online-bioinformatics-courses"><i class="fa fa-check"></i><b>29.4</b> Online bioinformatics courses</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html"><i class="fa fa-check"></i><b>30</b> – Importing/Exporting Data into R</a>
<ul>
<li class="chapter" data-level="30.1" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#typing-directly"><i class="fa fa-check"></i><b>30.1</b> Typing directly</a></li>
<li class="chapter" data-level="30.2" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#summary-versus-raw-data"><i class="fa fa-check"></i><b>30.2</b> Summary versus raw data</a></li>
<li class="chapter" data-level="30.3" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#downloading-data-files-github"><i class="fa fa-check"></i><b>30.3</b> Downloading data files (GitHub)</a></li>
<li class="chapter" data-level="30.4" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#importing-data-from-a-file"><i class="fa fa-check"></i><b>30.4</b> Importing data from a file</a></li>
<li class="chapter" data-level="30.5" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-to-a-file"><i class="fa fa-check"></i><b>30.5</b> Saving data to a file</a></li>
<li class="chapter" data-level="30.6" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-as-rdata-.rdata-.rda"><i class="fa fa-check"></i><b>30.6</b> Saving data as Rdata (.rdata, .rda)</a></li>
<li class="chapter" data-level="30.7" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#import-excel-formated-data"><i class="fa fa-check"></i><b>30.7</b> Import Excel formated data</a></li>
<li class="chapter" data-level="30.8" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#save-data-as-excel-formated-file"><i class="fa fa-check"></i><b>30.8</b> Save data as Excel formated file</a></li>
<li class="chapter" data-level="30.9" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#fixing-problems-while-importing-datasets"><i class="fa fa-check"></i><b>30.9</b> Fixing problems while importing datasets</a></li>
<li class="chapter" data-level="30.10" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#simplifying-embellished-excel-datasets"><i class="fa fa-check"></i><b>30.10</b> Simplifying embellished excel datasets</a></li>
<li class="chapter" data-level="30.11" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#accessing-online-data"><i class="fa fa-check"></i><b>30.11</b> Accessing online data</a></li>
<li class="chapter" data-level="30.12" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#data-from-clinicaltrials.gov"><i class="fa fa-check"></i><b>30.12</b> Data from clinicaltrials.gov</a></li>
<li class="chapter" data-level="30.13" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#data-scrapping-from-a-website"><i class="fa fa-check"></i><b>30.13</b> Data scrapping from a website</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="working-with-proteins.html"><a href="working-with-proteins.html"><i class="fa fa-check"></i><b>31</b> – Working with Proteins</a>
<ul>
<li class="chapter" data-level="31.1" data-path="working-with-proteins.html"><a href="working-with-proteins.html#sequences-in-fasta-format"><i class="fa fa-check"></i><b>31.1</b> Sequences in FASTA format</a></li>
<li class="chapter" data-level="31.2" data-path="working-with-proteins.html"><a href="working-with-proteins.html#fetching-aa-sequences-from-ncbi"><i class="fa fa-check"></i><b>31.2</b> Fetching AA sequences from NCBI</a></li>
<li class="chapter" data-level="31.3" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-aa-sequences-in-r"><i class="fa fa-check"></i><b>31.3</b> Importing AA sequences in R</a></li>
<li class="chapter" data-level="31.4" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-nucleotide-sequences-in-r"><i class="fa fa-check"></i><b>31.4</b> Importing nucleotide sequences in R</a></li>
<li class="chapter" data-level="31.5" data-path="working-with-proteins.html"><a href="working-with-proteins.html#translating-a-nucleotides-to-aa"><i class="fa fa-check"></i><b>31.5</b> Translating a nucleotides to AA</a></li>
<li class="chapter" data-level="31.6" data-path="working-with-proteins.html"><a href="working-with-proteins.html#finding-orfs-in-a-nucleotide-sequence"><i class="fa fa-check"></i><b>31.6</b> Finding ORFs in a nucleotide sequence</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="miscellaneous.html"><a href="miscellaneous.html"><i class="fa fa-check"></i><b>32</b> – Miscellaneous</a>
<ul>
<li class="chapter" data-level="32.1" data-path="miscellaneous.html"><a href="miscellaneous.html#sampling-from-a-vector-of-names"><i class="fa fa-check"></i><b>32.1</b> Sampling from a vector of names</a></li>
<li class="chapter" data-level="32.2" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-an-article-pdf"><i class="fa fa-check"></i><b>32.2</b> Read a text from an article pdf</a></li>
<li class="chapter" data-level="32.3" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-website"><i class="fa fa-check"></i><b>32.3</b> Read a text from a website</a></li>
<li class="chapter" data-level="32.4" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-file"><i class="fa fa-check"></i><b>32.4</b> Read a text from a file</a></li>
<li class="chapter" data-level="32.5" data-path="miscellaneous.html"><a href="miscellaneous.html#generate-the-word-cloud"><i class="fa fa-check"></i><b>32.5</b> Generate the word cloud</a></li>
<li class="chapter" data-level="32.6" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-pdf"><i class="fa fa-check"></i><b>32.6</b> worldcloud – PDF</a></li>
<li class="chapter" data-level="32.7" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-website"><i class="fa fa-check"></i><b>32.7</b> worldcloud – website</a></li>
<li class="chapter" data-level="32.8" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-text"><i class="fa fa-check"></i><b>32.8</b> worldcloud – text</a></li>
<li class="chapter" data-level="32.9" data-path="miscellaneous.html"><a href="miscellaneous.html#miscellaneous-1"><i class="fa fa-check"></i><b>32.9</b> Miscellaneous</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics Workshop Gitbook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="estimating-phylogenetic-trees" class="section level1" number="12">
<h1><span class="header-section-number">Session 12</span> – Estimating Phylogenetic Trees</h1>
<p>Most sequence alignments can be used to estimate phylogenies. These are hypotheses of evolutionary relationships between sequences and the species or populations that they represent. Phylogenies can be estimated from DNA, RNA and amino acid sequences (other features like morphology can also be used). The tree topology provides information about the observed organisms (i.e., source of sequences) at the tips or leaves of the tree, the internal vertices or nodes that represent intermediate ancestors (i.e., likely extinct), and the edges or connecting lines represent evolutionary ancestor-descendant relationships. The value of phylogenies is fundamental in the study biological diversity from taxonomy, epidemiology and population genetics. Data for molecular phylogenies are easy to collect, yet problems derived from homoplasy, reversals and incomplete lineage sorting might affect the reliability of phylogenetic hypotheses.</p>
<p><img src="figures/session_12_figs/1_Darwins_I_think.jpg" width="60%" /></p>
<p>One of <a href="https://en.wikipedia.org/wiki/Charles_Darwin" target="_blank"><strong>Charles Darwin</strong></a> greatest insights was phylogenetic trees that has shape modern biology.</p>
<div id="phylogeny-inference-with-iqtree2" class="section level2" number="12.1">
<h2><span class="header-section-number">12.1</span> Phylogeny inference with <code>iqtree2</code></h2>
<p>I do not use R or any R packages to estimate phylogenies, there are so many good programs on their own that can to this better and faster. We can devote many sessions to talk about these software, yet we will concentrate only in <a href="http://www.iqtree.org/" target="_blank"><strong>iqtree2</strong></a> that has emerged as a fast and reliable software that also includes molecular model testing and sequence partition. Likewise, <strong>iqtree2</strong> has an extensive manual <a href="http://www.iqtree.org/doc/Tutorial" target="_blank"><strong>here</strong></a> and it also can be easily manipulated within R. This will also illustrate how R can manage software outside its environment and bring back results to it.</p>
</div>
<div id="installing-iqtree2-in-macos" class="section level2" number="12.2">
<h2><span class="header-section-number">12.2</span> Installing <code>iqtree2</code> in macOS</h2>
<p><strong>1)</strong> We will start with installing <a href="http://www.iqtree.org/" target="_blank"><strong>iqtree2</strong></a> and locating the path of the binary file that will be used R to run phylogenetic estimations.</p>
<p><strong>a)</strong> You need to go first to <a href="http://www.iqtree.org/#download" target="_blank"><strong>iqtree2 download</strong></a> website and choose the 64-bit multicore version for your operative system (i.e., macOS, Windows or Linux). Make sure that it is version 2: <strong>COVID-19 release 2.1.2 (October 22, 2020)</strong>.</p>
<p><img src="figures/session_12_figs/2_iqtree_donwload.png" width="90%" /></p>
<p><strong>b)</strong> This is a zip file and you need to uncompress it and search for the folder with the precompiled version (in masOS is called <code>bin</code>) and copy that file to directly that you will like to store such software. In my case, I named <code>my_programs</code></p>
<p><img src="figures/session_12_figs/3_iqtree_unpack.png" width="90%" /></p>
<p><strong>c)</strong> Now, the <code>my_programs</code> folder includes the binary <strong>iqtree2</strong> that is the only thing that you might need to run this program.</p>
<p><strong>d) For macOS:</strong> You can test if this program is working by opening the <strong>Terminal</strong> on your computer. Most people do not have <strong>Terminal</strong> in their dock for easy access software. To open <strong>Terminal</strong>, you need to access your hard drive (click on it). Then, look for <strong>Applications</strong> folder (click on it). In this folder look for subfolder <strong>Utilities</strong>.</p>
<p><img src="figures/session_12_figs/4_iterminal.png" width="90%" /></p>
<p><strong>e) For macOS:</strong> the <strong>Terminal</strong> icon is in the <strong>Utilities</strong> folder and you can create a shortcut by draging <strong>Terminal</strong> to your dock. However, you do not have to do this to test <strong>iqtree2</strong>.</p>
<p><img src="figures/session_12_figs/5_find_terminal.png" width="90%" /></p>
<p><strong>f) For macOS:</strong> You can activate <strong>Terminal</strong> by double clicking on it and a new window will appear like the one below.</p>
<p><img src="figures/session_12_figs/6_open_terminal.png" width="90%" /></p>
<p><strong>g) For macOS:</strong> The working directory by default is root. You need to change working directory by typing <code>cd</code> and dragging and dropping the folder that contains <strong>iqtree2</strong> (i.e., in my case, the folder <code>my_programs</code>).</p>
<p><img src="figures/session_12_figs/7_change_to_folder.png" width="90%" /></p>
<p><strong>h) For macOS:</strong> After pressing enter, the working directory will be now <code>my_programs</code>. You can type <code>ls</code> and enter. You will get that <strong>iqtree2</strong> file is there.</p>
<p><img src="figures/session_12_figs/8_ls_folder.png" width="90%" /></p>
<p><strong>i) For macOS:</strong> To make sure that <strong>iqtree2</strong> will work, you will need to type <code>./iqtree2</code> and press enter.</p>
<p><img src="figures/session_12_figs/9_fot_iqtree.png" width="90%" /></p>
<p><strong>j) For macOS:</strong> After pressing enter, you should get the following message.</p>
<p><img src="figures/session_12_figs/10_test_for_iqtree2.png" width="90%" /></p>
<p><strong>NOTICE:</strong> For PCs, the process to run <strong>iqtree2</strong> outside R are provided in <a href="http://www.iqtree.org/doc/Quickstart#for-windows-users" target="_blank"><strong>here</strong></a>.</p>
</div>
<div id="installing-and-running-iqtree2-in-a-pc" class="section level2" number="12.3">
<h2><span class="header-section-number">12.3</span> Installing and running <code>iqtree2</code> in a PC</h2>
<p><strong>2)</strong> We will start with installing <a href="http://www.iqtree.org/" target="_blank"><strong>iqtree2</strong></a> and locating the path of the binary file that will be used R to run phylogenetic estimations.</p>
<p><strong>a)</strong> You need to go first to <a href="http://www.iqtree.org/#download" target="_blank"><strong>iqtree2 download</strong></a> website and choose the 64-bit multicore version for your operative system (i.e., macOS, Windows or Linux). Make sure that it is version 2: <strong>COVID-19 release 2.1.2 (October 22, 2020)</strong>.</p>
<p><img src="figures/session_12_figs/2_iqtree_donwload.png" width="90%" /></p>
<p><strong>b)</strong> This is a zip file and you need to uncompress it and search for the folder with the precompiled version (in PC is called <code>bin</code>) and you will see three files: <code>iqtree</code>, <code>iqtree-click</code> and <code>libiomp5md.dll</code>.</p>
<p><img src="figures/session_12_figs/22_pc_iqtree_pc3.jpg" width="100%" /></p>
<p><strong>c)</strong> The <code>iqtree_runner_R</code> function <strong>DOES NOT</strong> run in PCs, but it will give the commands that you need to run this software.</p>
<p><strong>d)</strong> On the folder that contains your alignment, you need to copy that files <code>iqtree-click</code> and <code>libiomp5md.dll</code> to be able to run the iqtree2 program. This folder should contain the aligned sequence (<strong>Session 12</strong>) or you can access some these files with aligned data and download them from the <a href="https://github.com/corytophanes/BIO_BIT_Bioinformatics_209/tree/main/ref_files_pdfs" target="_blank"><strong>GitHub repository</strong></a> for this class (e.g., <code>Ameerega_aligned_end.fasta</code>).</p>
<p><img src="figures/session_12_figs/23_pc_iqtree_pc2.jpg" width="100%" /></p>
<p><strong>e)</strong> To activate the interactive <code>iqtree</code> command building tool, you need to <strong>double click</strong> on the <code>iqtree-click</code> (Notice that windows might prevent you from running so, you will need to allow the program to run).
You will get the following window.</p>
<p><img src="figures/session_12_figs/24_pc_iqtree_pc4.jpg" width="100%" /></p>
<p><strong>f)</strong> Now, you will interact with this window bulding commands to run <code>iqtree</code> by presing <code>y</code>.</p>
<p><img src="figures/session_12_figs/25_pc_iqtree_pc5.jpg" width="100%" /></p>
<p><strong>g)</strong> The first command will be <code>-s</code> for the alignment input and then <code>e</code> to extend.</p>
<p><img src="figures/session_12_figs/26_pc_iqtree_pc6.jpg" width="100%" /></p>
<p><strong>h)</strong> We will add to the command line the name of the alignment in the folder <code>Ameerega_aligned_end.fasta</code> for the alignment input and then <code>e</code> to extend.</p>
<p><img src="figures/session_12_figs/27_pc_iqtree_pc8.jpg" width="100%" /></p>
<p><strong>h)</strong> We will continue adding to the command line the options for model <code>-m</code>, then <code>e</code> to extend, then <code>MFP+MERGE</code>.</p>
<p><img src="figures/session_12_figs/28_pc_iqtree_pc9.jpg" width="100%" /></p>
<p><strong>i)</strong> After we have included all commands to the command line, we will type <code>y</code> to confirm.</p>
<p><img src="figures/session_12_figs/29_pc_iqtree_pc10.jpg" width="100%" /></p>
<p><strong>j)</strong> If correct, the program will run and you will have to wait until it finish and then press <code>return</code> to exit.</p>
<p><img src="figures/session_12_figs/30_pc_iqtree_pc11.jpg" width="100%" /></p>
<p><strong>k)</strong> In the folder, you will see the output files. The file <code>*.treefile</code> will contain the resulting phylogeny.</p>
<p><img src="figures/session_12_figs/31_pc_iqtree_pc12.jpg" width="100%" /></p>
<p><strong>l)</strong> To run other commands in this format (e.g., bootstraps or with partitions), you need to run <code>iqtree_runner_R</code> first to get all the necessary input commands. They will look as follows on the first few lines of the <code>iqtree_runner_R</code> function output (even if it does not run):</p>
<div class="sourceCode" id="cb518"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb518-1"><a href="estimating-phylogenetic-trees.html#cb518-1" aria-hidden="true" tabindex="-1"></a><span class="do">## this is an example of the commands that you need to input in the format for PCs</span></span>
<span id="cb518-2"><a href="estimating-phylogenetic-trees.html#cb518-2" aria-hidden="true" tabindex="-1"></a>[<span class="dv">1</span>] <span class="st">&quot;iqtree2  -s Ameerega_aligned_end.fasta     -m  MFP+MERGE  &quot;</span></span></code></pre></div>
</div>
<div id="loading-the-iqtree_runner_r-function" class="section level2" number="12.4">
<h2><span class="header-section-number">12.4</span> Loading the <code>iqtree_runner_R</code> function</h2>
<p><strong>3)</strong> To run software outside R requires several components and scripting outside this workshop, I will not go in detail on these other than provide you with a function that will allow you to run our aligned sequence matrices (<strong>Session 12</strong>) or you can access some these files with aligned data on the <a href="https://github.com/corytophanes/BIO_BIT_Bioinformatics_209/tree/main/ref_files_pdfs" target="_blank"><strong>GitHub repository</strong></a> for this class (e.g., <code>Ameerega_aligned_end.fasta</code>). The function provided below allows to run <strong>iqtree2</strong> and importing back the tree and its nodal support after the analyses. However, it only works in macOS and for PC users please follow instructions on <strong>2)</strong>. If you have questions about the <code>iqtree_runner_R</code> function below, please contact me for more details.</p>
<p>You need to install two packages <code>doParallel</code> and <code>ape</code> before we run <code>iqtree_runner_R</code>. We will not get into details about <code>doParallel</code>, but we will use <code>ape</code> for other phylogenetic analyses.</p>
<div class="sourceCode" id="cb519"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb519-1"><a href="estimating-phylogenetic-trees.html#cb519-1" aria-hidden="true" tabindex="-1"></a><span class="do">## it requires to install these R packages</span></span>
<span id="cb519-2"><a href="estimating-phylogenetic-trees.html#cb519-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;doParallel&quot;</span>)</span>
<span id="cb519-3"><a href="estimating-phylogenetic-trees.html#cb519-3" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ape&quot;</span>)</span></code></pre></div>
<p>Please copy an paste the text below for <code>iqtree_runner_R</code> in R. This function is to run iqtree2 from your computer, but it has been used in macOS. We will need to test for PCs</p>
<div class="sourceCode" id="cb520"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb520-1"><a href="estimating-phylogenetic-trees.html#cb520-1" aria-hidden="true" tabindex="-1"></a><span class="do">## make sure that you have the required packages</span></span>
<span id="cb520-2"><a href="estimating-phylogenetic-trees.html#cb520-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(doParallel)</span>
<span id="cb520-3"><a href="estimating-phylogenetic-trees.html#cb520-3" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ape)</span>
<span id="cb520-4"><a href="estimating-phylogenetic-trees.html#cb520-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-5"><a href="estimating-phylogenetic-trees.html#cb520-5" aria-hidden="true" tabindex="-1"></a><span class="do">## iqtree_runner function copy text from here until </span><span class="re">END</span><span class="do"> OF FUNCTION (find this below)</span></span>
<span id="cb520-6"><a href="estimating-phylogenetic-trees.html#cb520-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-7"><a href="estimating-phylogenetic-trees.html#cb520-7" aria-hidden="true" tabindex="-1"></a>iqtree_runner_R <span class="ot">&lt;-</span> <span class="cf">function</span>(input_alignment_file_user,</span>
<span id="cb520-8"><a href="estimating-phylogenetic-trees.html#cb520-8" aria-hidden="true" tabindex="-1"></a>                           iqtree_bin_location_user,</span>
<span id="cb520-9"><a href="estimating-phylogenetic-trees.html#cb520-9" aria-hidden="true" tabindex="-1"></a>                           <span class="at">run_iqtree_call_user =</span> <span class="fu">c</span>(<span class="st">&quot;serial&quot;</span>,<span class="st">&quot;parallel&quot;</span>),</span>
<span id="cb520-10"><a href="estimating-phylogenetic-trees.html#cb520-10" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_quiet_user =</span> <span class="cn">FALSE</span>,</span>
<span id="cb520-11"><a href="estimating-phylogenetic-trees.html#cb520-11" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_redo_user =</span> <span class="cn">FALSE</span>,</span>
<span id="cb520-12"><a href="estimating-phylogenetic-trees.html#cb520-12" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_prefix_user =</span> <span class="cn">NULL</span>,</span>
<span id="cb520-13"><a href="estimating-phylogenetic-trees.html#cb520-13" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_input_model_m_user =</span> <span class="cn">NULL</span>,</span>
<span id="cb520-14"><a href="estimating-phylogenetic-trees.html#cb520-14" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_mtree_user =</span> <span class="cn">FALSE</span>,</span>
<span id="cb520-15"><a href="estimating-phylogenetic-trees.html#cb520-15" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_sequence_type_st_user =</span> <span class="cn">NULL</span>,</span>
<span id="cb520-16"><a href="estimating-phylogenetic-trees.html#cb520-16" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_UFBoot_B_user =</span> <span class="cn">NULL</span>,</span>
<span id="cb520-17"><a href="estimating-phylogenetic-trees.html#cb520-17" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_UFBoot_bnni_user =</span> <span class="cn">TRUE</span>,</span>
<span id="cb520-18"><a href="estimating-phylogenetic-trees.html#cb520-18" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_nonpara_boot_b_user =</span> <span class="cn">NULL</span>,</span>
<span id="cb520-19"><a href="estimating-phylogenetic-trees.html#cb520-19" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_SHaLRT_alrt_user =</span> <span class="cn">NULL</span>,</span>
<span id="cb520-20"><a href="estimating-phylogenetic-trees.html#cb520-20" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_multicore_T_user =</span> <span class="cn">NULL</span>,</span>
<span id="cb520-21"><a href="estimating-phylogenetic-trees.html#cb520-21" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_maxcores_ntmax_user =</span> <span class="cn">NULL</span>,</span>
<span id="cb520-22"><a href="estimating-phylogenetic-trees.html#cb520-22" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_partition_user =</span> <span class="cn">NULL</span>,</span>
<span id="cb520-23"><a href="estimating-phylogenetic-trees.html#cb520-23" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_partition_if_by_position_user =</span> <span class="cn">TRUE</span>,</span>
<span id="cb520-24"><a href="estimating-phylogenetic-trees.html#cb520-24" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_partition_file_user =</span> <span class="cn">NULL</span>,</span>
<span id="cb520-25"><a href="estimating-phylogenetic-trees.html#cb520-25" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_chronogram_from_tree_file_user =</span> <span class="cn">NULL</span>,</span>
<span id="cb520-26"><a href="estimating-phylogenetic-trees.html#cb520-26" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_chronogram_mrca_bounds_user =</span> <span class="cn">NULL</span>,</span>
<span id="cb520-27"><a href="estimating-phylogenetic-trees.html#cb520-27" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_date_file_user =</span> <span class="cn">NULL</span>,</span>
<span id="cb520-28"><a href="estimating-phylogenetic-trees.html#cb520-28" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_pass_parameters_user =</span> <span class="cn">NULL</span>){</span>
<span id="cb520-29"><a href="estimating-phylogenetic-trees.html#cb520-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-30"><a href="estimating-phylogenetic-trees.html#cb520-30" aria-hidden="true" tabindex="-1"></a><span class="do">## require libraries</span></span>
<span id="cb520-31"><a href="estimating-phylogenetic-trees.html#cb520-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-32"><a href="estimating-phylogenetic-trees.html#cb520-32" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(doParallel)</span>
<span id="cb520-33"><a href="estimating-phylogenetic-trees.html#cb520-33" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ape)</span>
<span id="cb520-34"><a href="estimating-phylogenetic-trees.html#cb520-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-35"><a href="estimating-phylogenetic-trees.html#cb520-35" aria-hidden="true" tabindex="-1"></a><span class="do">############# input from user</span></span>
<span id="cb520-36"><a href="estimating-phylogenetic-trees.html#cb520-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-37"><a href="estimating-phylogenetic-trees.html#cb520-37" aria-hidden="true" tabindex="-1"></a><span class="do">## sequence and tree arguments</span></span>
<span id="cb520-38"><a href="estimating-phylogenetic-trees.html#cb520-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-39"><a href="estimating-phylogenetic-trees.html#cb520-39" aria-hidden="true" tabindex="-1"></a>input_files <span class="ot">&lt;-</span> input_alignment_file_user </span>
<span id="cb520-40"><a href="estimating-phylogenetic-trees.html#cb520-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-41"><a href="estimating-phylogenetic-trees.html#cb520-41" aria-hidden="true" tabindex="-1"></a><span class="co"># iqtree location</span></span>
<span id="cb520-42"><a href="estimating-phylogenetic-trees.html#cb520-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-43"><a href="estimating-phylogenetic-trees.html#cb520-43" aria-hidden="true" tabindex="-1"></a>iqtree_bin_location <span class="ot">&lt;-</span> iqtree_bin_location_user</span>
<span id="cb520-44"><a href="estimating-phylogenetic-trees.html#cb520-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-45"><a href="estimating-phylogenetic-trees.html#cb520-45" aria-hidden="true" tabindex="-1"></a><span class="co"># serial or parallel calls</span></span>
<span id="cb520-46"><a href="estimating-phylogenetic-trees.html#cb520-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-47"><a href="estimating-phylogenetic-trees.html#cb520-47" aria-hidden="true" tabindex="-1"></a>run_iqtree <span class="ot">&lt;-</span> run_iqtree_call_user</span>
<span id="cb520-48"><a href="estimating-phylogenetic-trees.html#cb520-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-49"><a href="estimating-phylogenetic-trees.html#cb520-49" aria-hidden="true" tabindex="-1"></a>iqtree_quiet <span class="ot">&lt;-</span> iqtree_quiet_user</span>
<span id="cb520-50"><a href="estimating-phylogenetic-trees.html#cb520-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-51"><a href="estimating-phylogenetic-trees.html#cb520-51" aria-hidden="true" tabindex="-1"></a><span class="do">####### basic iqtree parameters</span></span>
<span id="cb520-52"><a href="estimating-phylogenetic-trees.html#cb520-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-53"><a href="estimating-phylogenetic-trees.html#cb520-53" aria-hidden="true" tabindex="-1"></a><span class="co"># overwrite all previous output: -redo</span></span>
<span id="cb520-54"><a href="estimating-phylogenetic-trees.html#cb520-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-55"><a href="estimating-phylogenetic-trees.html#cb520-55" aria-hidden="true" tabindex="-1"></a>iqtree_redo <span class="ot">&lt;-</span> iqtree_redo_user</span>
<span id="cb520-56"><a href="estimating-phylogenetic-trees.html#cb520-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-57"><a href="estimating-phylogenetic-trees.html#cb520-57" aria-hidden="true" tabindex="-1"></a><span class="co"># add prefi: --prefix</span></span>
<span id="cb520-58"><a href="estimating-phylogenetic-trees.html#cb520-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-59"><a href="estimating-phylogenetic-trees.html#cb520-59" aria-hidden="true" tabindex="-1"></a>iqtree_prefix <span class="ot">&lt;-</span> iqtree_prefix_user</span>
<span id="cb520-60"><a href="estimating-phylogenetic-trees.html#cb520-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-61"><a href="estimating-phylogenetic-trees.html#cb520-61" aria-hidden="true" tabindex="-1"></a><span class="co"># add model: -m</span></span>
<span id="cb520-62"><a href="estimating-phylogenetic-trees.html#cb520-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-63"><a href="estimating-phylogenetic-trees.html#cb520-63" aria-hidden="true" tabindex="-1"></a>iqtree_model <span class="ot">&lt;-</span> iqtree_input_model_m_user</span>
<span id="cb520-64"><a href="estimating-phylogenetic-trees.html#cb520-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-65"><a href="estimating-phylogenetic-trees.html#cb520-65" aria-hidden="true" tabindex="-1"></a><span class="co"># increase accuracy: -mtree</span></span>
<span id="cb520-66"><a href="estimating-phylogenetic-trees.html#cb520-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-67"><a href="estimating-phylogenetic-trees.html#cb520-67" aria-hidden="true" tabindex="-1"></a>iqtree_mtree <span class="ot">&lt;-</span> iqtree_mtree_user</span>
<span id="cb520-68"><a href="estimating-phylogenetic-trees.html#cb520-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-69"><a href="estimating-phylogenetic-trees.html#cb520-69" aria-hidden="true" tabindex="-1"></a><span class="co"># Specify sequence type as either of NULL (autodetect) DNA, AA, BIN, MORPH, CODON or NT2AA for DNA</span></span>
<span id="cb520-70"><a href="estimating-phylogenetic-trees.html#cb520-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-71"><a href="estimating-phylogenetic-trees.html#cb520-71" aria-hidden="true" tabindex="-1"></a>iqtree_st <span class="ot">&lt;-</span> iqtree_sequence_type_st_user</span>
<span id="cb520-72"><a href="estimating-phylogenetic-trees.html#cb520-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-73"><a href="estimating-phylogenetic-trees.html#cb520-73" aria-hidden="true" tabindex="-1"></a><span class="co"># UFBoot mode: -B 1000</span></span>
<span id="cb520-74"><a href="estimating-phylogenetic-trees.html#cb520-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-75"><a href="estimating-phylogenetic-trees.html#cb520-75" aria-hidden="true" tabindex="-1"></a>iqtree_UFBoot_B <span class="ot">&lt;-</span> iqtree_UFBoot_B_user</span>
<span id="cb520-76"><a href="estimating-phylogenetic-trees.html#cb520-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-77"><a href="estimating-phylogenetic-trees.html#cb520-77" aria-hidden="true" tabindex="-1"></a><span class="co"># Red_impact_UFBoot</span></span>
<span id="cb520-78"><a href="estimating-phylogenetic-trees.html#cb520-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-79"><a href="estimating-phylogenetic-trees.html#cb520-79" aria-hidden="true" tabindex="-1"></a>iqtree_UFBoot_bnni <span class="ot">&lt;-</span> iqtree_UFBoot_bnni_user</span>
<span id="cb520-80"><a href="estimating-phylogenetic-trees.html#cb520-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-81"><a href="estimating-phylogenetic-trees.html#cb520-81" aria-hidden="true" tabindex="-1"></a><span class="co"># nonparametric bootstrap -b 100</span></span>
<span id="cb520-82"><a href="estimating-phylogenetic-trees.html#cb520-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-83"><a href="estimating-phylogenetic-trees.html#cb520-83" aria-hidden="true" tabindex="-1"></a>iqtree_nonpara_b <span class="ot">&lt;-</span> iqtree_nonpara_boot_b_user</span>
<span id="cb520-84"><a href="estimating-phylogenetic-trees.html#cb520-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-85"><a href="estimating-phylogenetic-trees.html#cb520-85" aria-hidden="true" tabindex="-1"></a><span class="co"># Assessing branch supports with single branch tests SH-like approximate likelihood ratio test: -alrt 1000</span></span>
<span id="cb520-86"><a href="estimating-phylogenetic-trees.html#cb520-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-87"><a href="estimating-phylogenetic-trees.html#cb520-87" aria-hidden="true" tabindex="-1"></a>iqtree_alrt <span class="ot">&lt;-</span> iqtree_SHaLRT_alrt_user</span>
<span id="cb520-88"><a href="estimating-phylogenetic-trees.html#cb520-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-89"><a href="estimating-phylogenetic-trees.html#cb520-89" aria-hidden="true" tabindex="-1"></a><span class="co"># multicore -T AUTO -T 2</span></span>
<span id="cb520-90"><a href="estimating-phylogenetic-trees.html#cb520-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-91"><a href="estimating-phylogenetic-trees.html#cb520-91" aria-hidden="true" tabindex="-1"></a>iqtree_multicore_T <span class="ot">&lt;-</span> iqtree_multicore_T_user</span>
<span id="cb520-92"><a href="estimating-phylogenetic-trees.html#cb520-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-93"><a href="estimating-phylogenetic-trees.html#cb520-93" aria-hidden="true" tabindex="-1"></a><span class="co"># maxcores -ntmax </span></span>
<span id="cb520-94"><a href="estimating-phylogenetic-trees.html#cb520-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-95"><a href="estimating-phylogenetic-trees.html#cb520-95" aria-hidden="true" tabindex="-1"></a>iqtree_maxcores_ntmax <span class="ot">&lt;-</span> iqtree_maxcores_ntmax_user</span>
<span id="cb520-96"><a href="estimating-phylogenetic-trees.html#cb520-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-97"><a href="estimating-phylogenetic-trees.html#cb520-97" aria-hidden="true" tabindex="-1"></a><span class="do">##############       iqtree_partition_user</span></span>
<span id="cb520-98"><a href="estimating-phylogenetic-trees.html#cb520-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-99"><a href="estimating-phylogenetic-trees.html#cb520-99" aria-hidden="true" tabindex="-1"></a>iqtree_partition <span class="ot">&lt;-</span> iqtree_partition_user</span>
<span id="cb520-100"><a href="estimating-phylogenetic-trees.html#cb520-100" aria-hidden="true" tabindex="-1"></a>do_by_codon_pos <span class="ot">&lt;-</span> iqtree_partition_if_by_position_user</span>
<span id="cb520-101"><a href="estimating-phylogenetic-trees.html#cb520-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-102"><a href="estimating-phylogenetic-trees.html#cb520-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-103"><a href="estimating-phylogenetic-trees.html#cb520-103" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(iqtree_partition)) {</span>
<span id="cb520-104"><a href="estimating-phylogenetic-trees.html#cb520-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-105"><a href="estimating-phylogenetic-trees.html#cb520-105" aria-hidden="true" tabindex="-1"></a>      iqtree_partition_matrix <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind,iqtree_partition)</span>
<span id="cb520-106"><a href="estimating-phylogenetic-trees.html#cb520-106" aria-hidden="true" tabindex="-1"></a>             partitions_names <span class="ot">&lt;-</span> <span class="fu">rownames</span>(iqtree_partition_matrix)</span>
<span id="cb520-107"><a href="estimating-phylogenetic-trees.html#cb520-107" aria-hidden="true" tabindex="-1"></a>              type_partitions <span class="ot">&lt;-</span> iqtree_partition_matrix[,<span class="dv">1</span>]</span>
<span id="cb520-108"><a href="estimating-phylogenetic-trees.html#cb520-108" aria-hidden="true" tabindex="-1"></a>                   start_part <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(iqtree_partition_matrix[,<span class="dv">2</span>])</span>
<span id="cb520-109"><a href="estimating-phylogenetic-trees.html#cb520-109" aria-hidden="true" tabindex="-1"></a>                     end_part <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(iqtree_partition_matrix[,<span class="dv">3</span>])</span>
<span id="cb520-110"><a href="estimating-phylogenetic-trees.html#cb520-110" aria-hidden="true" tabindex="-1"></a>                   codon_part <span class="ot">&lt;-</span> iqtree_partition_matrix[,<span class="dv">4</span>]</span>
<span id="cb520-111"><a href="estimating-phylogenetic-trees.html#cb520-111" aria-hidden="true" tabindex="-1"></a>          iqtree_partition_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">partition_name =</span> partitions_names, <span class="at">type =</span> type_partitions, <span class="at">start =</span> start_part, <span class="at">end =</span> end_part, <span class="at">codon =</span> codon_part, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb520-112"><a href="estimating-phylogenetic-trees.html#cb520-112" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(iqtree_partition_df) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb520-113"><a href="estimating-phylogenetic-trees.html#cb520-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-114"><a href="estimating-phylogenetic-trees.html#cb520-114" aria-hidden="true" tabindex="-1"></a><span class="co"># output file name</span></span>
<span id="cb520-115"><a href="estimating-phylogenetic-trees.html#cb520-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-116"><a href="estimating-phylogenetic-trees.html#cb520-116" aria-hidden="true" tabindex="-1"></a>out_part_file <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;.*/&quot;</span>, <span class="st">&quot;&quot;</span>, input_files)</span>
<span id="cb520-117"><a href="estimating-phylogenetic-trees.html#cb520-117" aria-hidden="true" tabindex="-1"></a>out_part_file <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;[.]&quot;</span>, <span class="st">&quot;_&quot;</span>, out_part_file)</span>
<span id="cb520-118"><a href="estimating-phylogenetic-trees.html#cb520-118" aria-hidden="true" tabindex="-1"></a>out_part_file <span class="ot">&lt;-</span> <span class="fu">paste0</span>(out_part_file,<span class="st">&quot;_partition.txt&quot;</span>)</span>
<span id="cb520-119"><a href="estimating-phylogenetic-trees.html#cb520-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-120"><a href="estimating-phylogenetic-trees.html#cb520-120" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-121"><a href="estimating-phylogenetic-trees.html#cb520-121" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(iqtree_partition_df)) {</span>
<span id="cb520-122"><a href="estimating-phylogenetic-trees.html#cb520-122" aria-hidden="true" tabindex="-1"></a>  <span class="co"># i &lt;- 3</span></span>
<span id="cb520-123"><a href="estimating-phylogenetic-trees.html#cb520-123" aria-hidden="true" tabindex="-1"></a>   one_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(iqtree_partition_df<span class="sc">$</span>type[i], <span class="st">&quot;, &quot;</span>, iqtree_partition_df<span class="sc">$</span>partition_name[i], <span class="st">&quot; = &quot;</span>, </span>
<span id="cb520-124"><a href="estimating-phylogenetic-trees.html#cb520-124" aria-hidden="true" tabindex="-1"></a>                      iqtree_partition_df<span class="sc">$</span>start[i], <span class="st">&quot;-&quot;</span>, iqtree_partition_df<span class="sc">$</span>end[i])</span>
<span id="cb520-125"><a href="estimating-phylogenetic-trees.html#cb520-125" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-126"><a href="estimating-phylogenetic-trees.html#cb520-126" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.na</span>(iqtree_partition_df<span class="sc">$</span>codon[i])) {</span>
<span id="cb520-127"><a href="estimating-phylogenetic-trees.html#cb520-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-128"><a href="estimating-phylogenetic-trees.html#cb520-128" aria-hidden="true" tabindex="-1"></a><span class="co"># do_by_codon_pos</span></span>
<span id="cb520-129"><a href="estimating-phylogenetic-trees.html#cb520-129" aria-hidden="true" tabindex="-1"></a>   <span class="cf">if</span>(do_by_codon_pos) {</span>
<span id="cb520-130"><a href="estimating-phylogenetic-trees.html#cb520-130" aria-hidden="true" tabindex="-1"></a>              pos1_start <span class="ot">&lt;-</span> iqtree_partition_df<span class="sc">$</span>start[i]</span>
<span id="cb520-131"><a href="estimating-phylogenetic-trees.html#cb520-131" aria-hidden="true" tabindex="-1"></a>               pos1_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(iqtree_partition_df<span class="sc">$</span>partition_name[i],<span class="st">&quot;_pos1&quot;</span>)</span>
<span id="cb520-132"><a href="estimating-phylogenetic-trees.html#cb520-132" aria-hidden="true" tabindex="-1"></a>              pos2_start <span class="ot">&lt;-</span> iqtree_partition_df<span class="sc">$</span>start[i]<span class="sc">+</span><span class="dv">1</span></span>
<span id="cb520-133"><a href="estimating-phylogenetic-trees.html#cb520-133" aria-hidden="true" tabindex="-1"></a>               pos2_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(iqtree_partition_df<span class="sc">$</span>partition_name[i],<span class="st">&quot;_pos2&quot;</span>)</span>
<span id="cb520-134"><a href="estimating-phylogenetic-trees.html#cb520-134" aria-hidden="true" tabindex="-1"></a>              pos3_start <span class="ot">&lt;-</span> iqtree_partition_df<span class="sc">$</span>start[i]<span class="sc">+</span><span class="dv">2</span></span>
<span id="cb520-135"><a href="estimating-phylogenetic-trees.html#cb520-135" aria-hidden="true" tabindex="-1"></a>               pos3_name <span class="ot">&lt;-</span> <span class="fu">paste0</span>(iqtree_partition_df<span class="sc">$</span>partition_name[i],<span class="st">&quot;_pos3&quot;</span>)</span>
<span id="cb520-136"><a href="estimating-phylogenetic-trees.html#cb520-136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-137"><a href="estimating-phylogenetic-trees.html#cb520-137" aria-hidden="true" tabindex="-1"></a>   one_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(iqtree_partition_df<span class="sc">$</span>type[i], <span class="st">&quot;, &quot;</span>, pos1_name, <span class="st">&quot; = &quot;</span>, pos1_start, <span class="st">&quot;-&quot;</span>, iqtree_partition_df<span class="sc">$</span>end[i], <span class="st">&quot;</span><span class="sc">\\</span><span class="st">&quot;</span>, iqtree_partition_df<span class="sc">$</span>codon[i], <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb520-138"><a href="estimating-phylogenetic-trees.html#cb520-138" aria-hidden="true" tabindex="-1"></a>                      iqtree_partition_df<span class="sc">$</span>type[i], <span class="st">&quot;, &quot;</span>, pos2_name, <span class="st">&quot; = &quot;</span>, pos2_start, <span class="st">&quot;-&quot;</span>, iqtree_partition_df<span class="sc">$</span>end[i], <span class="st">&quot;</span><span class="sc">\\</span><span class="st">&quot;</span>, iqtree_partition_df<span class="sc">$</span>codon[i], <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb520-139"><a href="estimating-phylogenetic-trees.html#cb520-139" aria-hidden="true" tabindex="-1"></a>                      iqtree_partition_df<span class="sc">$</span>type[i], <span class="st">&quot;, &quot;</span>, pos3_name, <span class="st">&quot; = &quot;</span>, pos3_start, <span class="st">&quot;-&quot;</span>, iqtree_partition_df<span class="sc">$</span>end[i], <span class="st">&quot;</span><span class="sc">\\</span><span class="st">&quot;</span>, iqtree_partition_df<span class="sc">$</span>codon[i], <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb520-140"><a href="estimating-phylogenetic-trees.html#cb520-140" aria-hidden="true" tabindex="-1"></a>                       }</span>
<span id="cb520-141"><a href="estimating-phylogenetic-trees.html#cb520-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-142"><a href="estimating-phylogenetic-trees.html#cb520-142" aria-hidden="true" tabindex="-1"></a>                                        }</span>
<span id="cb520-143"><a href="estimating-phylogenetic-trees.html#cb520-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-144"><a href="estimating-phylogenetic-trees.html#cb520-144" aria-hidden="true" tabindex="-1"></a><span class="co">#     one_line &lt;- paste0(one_line, &quot;\\&quot;, iqtree_partition_df$codon[i]) }</span></span>
<span id="cb520-145"><a href="estimating-phylogenetic-trees.html#cb520-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-146"><a href="estimating-phylogenetic-trees.html#cb520-146" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(one_line,<span class="at">file=</span>out_part_file,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,<span class="at">append=</span><span class="cn">TRUE</span>)</span>
<span id="cb520-147"><a href="estimating-phylogenetic-trees.html#cb520-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-148"><a href="estimating-phylogenetic-trees.html#cb520-148" aria-hidden="true" tabindex="-1"></a>                                      }</span>
<span id="cb520-149"><a href="estimating-phylogenetic-trees.html#cb520-149" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">rm</span>(i)</span>
<span id="cb520-150"><a href="estimating-phylogenetic-trees.html#cb520-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-151"><a href="estimating-phylogenetic-trees.html#cb520-151" aria-hidden="true" tabindex="-1"></a>iqtree_partition_files <span class="ot">&lt;-</span> out_part_file </span>
<span id="cb520-152"><a href="estimating-phylogenetic-trees.html#cb520-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-153"><a href="estimating-phylogenetic-trees.html#cb520-153" aria-hidden="true" tabindex="-1"></a>                                         } <span class="cf">else</span> {</span>
<span id="cb520-154"><a href="estimating-phylogenetic-trees.html#cb520-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-155"><a href="estimating-phylogenetic-trees.html#cb520-155" aria-hidden="true" tabindex="-1"></a>iqtree_partition_files <span class="ot">&lt;-</span> iqtree_partition_file_user</span>
<span id="cb520-156"><a href="estimating-phylogenetic-trees.html#cb520-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-157"><a href="estimating-phylogenetic-trees.html#cb520-157" aria-hidden="true" tabindex="-1"></a>                                         }</span>
<span id="cb520-158"><a href="estimating-phylogenetic-trees.html#cb520-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-159"><a href="estimating-phylogenetic-trees.html#cb520-159" aria-hidden="true" tabindex="-1"></a><span class="do">##############       chronogram_functions</span></span>
<span id="cb520-160"><a href="estimating-phylogenetic-trees.html#cb520-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-161"><a href="estimating-phylogenetic-trees.html#cb520-161" aria-hidden="true" tabindex="-1"></a>iqtree_chronogram_from_tree <span class="ot">&lt;-</span> iqtree_chronogram_from_tree_file_user</span>
<span id="cb520-162"><a href="estimating-phylogenetic-trees.html#cb520-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-163"><a href="estimating-phylogenetic-trees.html#cb520-163" aria-hidden="true" tabindex="-1"></a>iqtree_mrca_bounds <span class="ot">&lt;-</span> iqtree_chronogram_mrca_bounds_user</span>
<span id="cb520-164"><a href="estimating-phylogenetic-trees.html#cb520-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-165"><a href="estimating-phylogenetic-trees.html#cb520-165" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(iqtree_mrca_bounds)) {</span>
<span id="cb520-166"><a href="estimating-phylogenetic-trees.html#cb520-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-167"><a href="estimating-phylogenetic-trees.html#cb520-167" aria-hidden="true" tabindex="-1"></a><span class="co"># name for DATE_FILE</span></span>
<span id="cb520-168"><a href="estimating-phylogenetic-trees.html#cb520-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-169"><a href="estimating-phylogenetic-trees.html#cb520-169" aria-hidden="true" tabindex="-1"></a>out_part_file <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;.*/&quot;</span>, <span class="st">&quot;&quot;</span>, input_files)</span>
<span id="cb520-170"><a href="estimating-phylogenetic-trees.html#cb520-170" aria-hidden="true" tabindex="-1"></a>out_part_file <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;[.]&quot;</span>, <span class="st">&quot;_&quot;</span>, out_part_file)</span>
<span id="cb520-171"><a href="estimating-phylogenetic-trees.html#cb520-171" aria-hidden="true" tabindex="-1"></a>out_date_file <span class="ot">&lt;-</span> <span class="fu">paste0</span>(out_part_file,<span class="st">&quot;_dates.txt&quot;</span>)</span>
<span id="cb520-172"><a href="estimating-phylogenetic-trees.html#cb520-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-173"><a href="estimating-phylogenetic-trees.html#cb520-173" aria-hidden="true" tabindex="-1"></a><span class="co"># build DATE_FILE</span></span>
<span id="cb520-174"><a href="estimating-phylogenetic-trees.html#cb520-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-175"><a href="estimating-phylogenetic-trees.html#cb520-175" aria-hidden="true" tabindex="-1"></a>n_total <span class="ot">&lt;-</span> <span class="fu">length</span>(iqtree_mrca_bounds)</span>
<span id="cb520-176"><a href="estimating-phylogenetic-trees.html#cb520-176" aria-hidden="true" tabindex="-1"></a>n_constraints <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">1</span>,<span class="at">by=</span><span class="dv">2</span>, <span class="at">to=</span>n_total)</span>
<span id="cb520-177"><a href="estimating-phylogenetic-trees.html#cb520-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-178"><a href="estimating-phylogenetic-trees.html#cb520-178" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> n_constraints) {</span>
<span id="cb520-179"><a href="estimating-phylogenetic-trees.html#cb520-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-180"><a href="estimating-phylogenetic-trees.html#cb520-180" aria-hidden="true" tabindex="-1"></a><span class="co"># i &lt;- 1</span></span>
<span id="cb520-181"><a href="estimating-phylogenetic-trees.html#cb520-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-182"><a href="estimating-phylogenetic-trees.html#cb520-182" aria-hidden="true" tabindex="-1"></a>   taxa_part <span class="ot">&lt;-</span> <span class="fu">paste0</span>(iqtree_mrca_bounds[[i]], <span class="at">collapse=</span><span class="st">&quot;,&quot;</span>)</span>
<span id="cb520-183"><a href="estimating-phylogenetic-trees.html#cb520-183" aria-hidden="true" tabindex="-1"></a>   date_part <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;-&quot;</span>, iqtree_mrca_bounds[[i<span class="sc">+</span><span class="dv">1</span>]])</span>
<span id="cb520-184"><a href="estimating-phylogenetic-trees.html#cb520-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-185"><a href="estimating-phylogenetic-trees.html#cb520-185" aria-hidden="true" tabindex="-1"></a>   one_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(taxa_part, <span class="st">&quot; &quot;</span>, date_part)</span>
<span id="cb520-186"><a href="estimating-phylogenetic-trees.html#cb520-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-187"><a href="estimating-phylogenetic-trees.html#cb520-187" aria-hidden="true" tabindex="-1"></a><span class="co">#     one_line &lt;- paste0(one_line, &quot;\\&quot;, iqtree_partition_df$codon[i]) }</span></span>
<span id="cb520-188"><a href="estimating-phylogenetic-trees.html#cb520-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-189"><a href="estimating-phylogenetic-trees.html#cb520-189" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(one_line,<span class="at">file=</span>out_date_file,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,<span class="at">append=</span><span class="cn">TRUE</span>)</span>
<span id="cb520-190"><a href="estimating-phylogenetic-trees.html#cb520-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-191"><a href="estimating-phylogenetic-trees.html#cb520-191" aria-hidden="true" tabindex="-1"></a>                                      }</span>
<span id="cb520-192"><a href="estimating-phylogenetic-trees.html#cb520-192" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">rm</span>(i)</span>
<span id="cb520-193"><a href="estimating-phylogenetic-trees.html#cb520-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-194"><a href="estimating-phylogenetic-trees.html#cb520-194" aria-hidden="true" tabindex="-1"></a>iqtree_date_files <span class="ot">&lt;-</span> out_date_file </span>
<span id="cb520-195"><a href="estimating-phylogenetic-trees.html#cb520-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-196"><a href="estimating-phylogenetic-trees.html#cb520-196" aria-hidden="true" tabindex="-1"></a>                                 } <span class="cf">else</span> {</span>
<span id="cb520-197"><a href="estimating-phylogenetic-trees.html#cb520-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-198"><a href="estimating-phylogenetic-trees.html#cb520-198" aria-hidden="true" tabindex="-1"></a>iqtree_date_files <span class="ot">&lt;-</span> iqtree_date_file_user</span>
<span id="cb520-199"><a href="estimating-phylogenetic-trees.html#cb520-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-200"><a href="estimating-phylogenetic-trees.html#cb520-200" aria-hidden="true" tabindex="-1"></a>                                         }</span>
<span id="cb520-201"><a href="estimating-phylogenetic-trees.html#cb520-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-202"><a href="estimating-phylogenetic-trees.html#cb520-202" aria-hidden="true" tabindex="-1"></a><span class="co"># iq pass paramters</span></span>
<span id="cb520-203"><a href="estimating-phylogenetic-trees.html#cb520-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-204"><a href="estimating-phylogenetic-trees.html#cb520-204" aria-hidden="true" tabindex="-1"></a>iqtree_pass <span class="ot">&lt;-</span> iqtree_pass_parameters_user</span>
<span id="cb520-205"><a href="estimating-phylogenetic-trees.html#cb520-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-206"><a href="estimating-phylogenetic-trees.html#cb520-206" aria-hidden="true" tabindex="-1"></a><span class="co"># get work directory to go back</span></span>
<span id="cb520-207"><a href="estimating-phylogenetic-trees.html#cb520-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-208"><a href="estimating-phylogenetic-trees.html#cb520-208" aria-hidden="true" tabindex="-1"></a>master_wd_directory <span class="ot">&lt;-</span> <span class="fu">getwd</span>()</span>
<span id="cb520-209"><a href="estimating-phylogenetic-trees.html#cb520-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-210"><a href="estimating-phylogenetic-trees.html#cb520-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-211"><a href="estimating-phylogenetic-trees.html#cb520-211" aria-hidden="true" tabindex="-1"></a><span class="do">## get iqtree path</span></span>
<span id="cb520-212"><a href="estimating-phylogenetic-trees.html#cb520-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-213"><a href="estimating-phylogenetic-trees.html#cb520-213" aria-hidden="true" tabindex="-1"></a>path_of_iqtree <span class="ot">&lt;-</span> <span class="fu">paste0</span>(iqtree_bin_location)</span>
<span id="cb520-214"><a href="estimating-phylogenetic-trees.html#cb520-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-215"><a href="estimating-phylogenetic-trees.html#cb520-215" aria-hidden="true" tabindex="-1"></a><span class="do">######     prepare iqtree commands</span></span>
<span id="cb520-216"><a href="estimating-phylogenetic-trees.html#cb520-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-217"><a href="estimating-phylogenetic-trees.html#cb520-217" aria-hidden="true" tabindex="-1"></a>list_iqtree_scripts <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb520-218"><a href="estimating-phylogenetic-trees.html#cb520-218" aria-hidden="true" tabindex="-1"></a>name_of_analyses_iqtree <span class="ot">&lt;-</span> <span class="fu">character</span>()</span>
<span id="cb520-219"><a href="estimating-phylogenetic-trees.html#cb520-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-220"><a href="estimating-phylogenetic-trees.html#cb520-220" aria-hidden="true" tabindex="-1"></a>counter <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb520-221"><a href="estimating-phylogenetic-trees.html#cb520-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-222"><a href="estimating-phylogenetic-trees.html#cb520-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-223"><a href="estimating-phylogenetic-trees.html#cb520-223" aria-hidden="true" tabindex="-1"></a>   <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(input_files)) {</span>
<span id="cb520-224"><a href="estimating-phylogenetic-trees.html#cb520-224" aria-hidden="true" tabindex="-1"></a>      <span class="co"># i &lt;- 1</span></span>
<span id="cb520-225"><a href="estimating-phylogenetic-trees.html#cb520-225" aria-hidden="true" tabindex="-1"></a>    one_alignment_file <span class="ot">&lt;-</span> input_files[i]</span>
<span id="cb520-226"><a href="estimating-phylogenetic-trees.html#cb520-226" aria-hidden="true" tabindex="-1"></a>    counter <span class="ot">&lt;-</span> counter <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-227"><a href="estimating-phylogenetic-trees.html#cb520-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-228"><a href="estimating-phylogenetic-trees.html#cb520-228" aria-hidden="true" tabindex="-1"></a><span class="co"># simplify names</span></span>
<span id="cb520-229"><a href="estimating-phylogenetic-trees.html#cb520-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-230"><a href="estimating-phylogenetic-trees.html#cb520-230" aria-hidden="true" tabindex="-1"></a>one_input_file <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">&quot;.*/&quot;</span>, <span class="st">&quot;&quot;</span>, one_alignment_file)</span>
<span id="cb520-231"><a href="estimating-phylogenetic-trees.html#cb520-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-232"><a href="estimating-phylogenetic-trees.html#cb520-232" aria-hidden="true" tabindex="-1"></a><span class="co"># prepare commands</span></span>
<span id="cb520-233"><a href="estimating-phylogenetic-trees.html#cb520-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-234"><a href="estimating-phylogenetic-trees.html#cb520-234" aria-hidden="true" tabindex="-1"></a>line_iqtree <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;iqtree2  -s &quot;</span>, one_input_file, <span class="st">&quot;  &quot;</span>)</span>
<span id="cb520-235"><a href="estimating-phylogenetic-trees.html#cb520-235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-236"><a href="estimating-phylogenetic-trees.html#cb520-236" aria-hidden="true" tabindex="-1"></a><span class="co"># list of iqtre parameters</span></span>
<span id="cb520-237"><a href="estimating-phylogenetic-trees.html#cb520-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-238"><a href="estimating-phylogenetic-trees.html#cb520-238" aria-hidden="true" tabindex="-1"></a>list_of_iqtree <span class="ot">&lt;-</span> <span class="fu">c</span>(line_iqtree)</span>
<span id="cb520-239"><a href="estimating-phylogenetic-trees.html#cb520-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-240"><a href="estimating-phylogenetic-trees.html#cb520-240" aria-hidden="true" tabindex="-1"></a>counter_2 <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb520-241"><a href="estimating-phylogenetic-trees.html#cb520-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-242"><a href="estimating-phylogenetic-trees.html#cb520-242" aria-hidden="true" tabindex="-1"></a><span class="co"># Specify sequence type as either of NULL (autodetect) DNA, AA, BIN, MORPH, CODON or NT2AA for DNA</span></span>
<span id="cb520-243"><a href="estimating-phylogenetic-trees.html#cb520-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-244"><a href="estimating-phylogenetic-trees.html#cb520-244" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(iqtree_st)) { counter_2 <span class="ot">&lt;-</span> counter_2 <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-245"><a href="estimating-phylogenetic-trees.html#cb520-245" aria-hidden="true" tabindex="-1"></a>                   out_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;  -st  &quot;</span>, iqtree_st, <span class="st">&quot;  &quot;</span>) </span>
<span id="cb520-246"><a href="estimating-phylogenetic-trees.html#cb520-246" aria-hidden="true" tabindex="-1"></a>   list_of_iqtree[counter_2] <span class="ot">&lt;-</span> out_line</span>
<span id="cb520-247"><a href="estimating-phylogenetic-trees.html#cb520-247" aria-hidden="true" tabindex="-1"></a>                     }</span>
<span id="cb520-248"><a href="estimating-phylogenetic-trees.html#cb520-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-249"><a href="estimating-phylogenetic-trees.html#cb520-249" aria-hidden="true" tabindex="-1"></a><span class="co"># add prefi: --prefix</span></span>
<span id="cb520-250"><a href="estimating-phylogenetic-trees.html#cb520-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-251"><a href="estimating-phylogenetic-trees.html#cb520-251" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(iqtree_prefix)) { counter_2 <span class="ot">&lt;-</span> counter_2 <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-252"><a href="estimating-phylogenetic-trees.html#cb520-252" aria-hidden="true" tabindex="-1"></a>                   out_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;  --prefix  &quot;</span>, iqtree_prefix, <span class="st">&quot;  &quot;</span>) </span>
<span id="cb520-253"><a href="estimating-phylogenetic-trees.html#cb520-253" aria-hidden="true" tabindex="-1"></a>   list_of_iqtree[counter_2] <span class="ot">&lt;-</span> out_line</span>
<span id="cb520-254"><a href="estimating-phylogenetic-trees.html#cb520-254" aria-hidden="true" tabindex="-1"></a>                     }</span>
<span id="cb520-255"><a href="estimating-phylogenetic-trees.html#cb520-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-256"><a href="estimating-phylogenetic-trees.html#cb520-256" aria-hidden="true" tabindex="-1"></a><span class="co"># add model: -m</span></span>
<span id="cb520-257"><a href="estimating-phylogenetic-trees.html#cb520-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-258"><a href="estimating-phylogenetic-trees.html#cb520-258" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(iqtree_model)) { counter_2 <span class="ot">&lt;-</span> counter_2 <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-259"><a href="estimating-phylogenetic-trees.html#cb520-259" aria-hidden="true" tabindex="-1"></a>                   out_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;  -m  &quot;</span>, iqtree_model, <span class="st">&quot;  &quot;</span>) </span>
<span id="cb520-260"><a href="estimating-phylogenetic-trees.html#cb520-260" aria-hidden="true" tabindex="-1"></a>   list_of_iqtree[counter_2] <span class="ot">&lt;-</span> out_line</span>
<span id="cb520-261"><a href="estimating-phylogenetic-trees.html#cb520-261" aria-hidden="true" tabindex="-1"></a>                     }</span>
<span id="cb520-262"><a href="estimating-phylogenetic-trees.html#cb520-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-263"><a href="estimating-phylogenetic-trees.html#cb520-263" aria-hidden="true" tabindex="-1"></a><span class="co"># increase accuracy: -mtree</span></span>
<span id="cb520-264"><a href="estimating-phylogenetic-trees.html#cb520-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-265"><a href="estimating-phylogenetic-trees.html#cb520-265" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(iqtree_mtree) { counter_2 <span class="ot">&lt;-</span> counter_2 <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-266"><a href="estimating-phylogenetic-trees.html#cb520-266" aria-hidden="true" tabindex="-1"></a>                   out_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;  -mtree  &quot;</span>) </span>
<span id="cb520-267"><a href="estimating-phylogenetic-trees.html#cb520-267" aria-hidden="true" tabindex="-1"></a>   list_of_iqtree[counter_2] <span class="ot">&lt;-</span> out_line</span>
<span id="cb520-268"><a href="estimating-phylogenetic-trees.html#cb520-268" aria-hidden="true" tabindex="-1"></a>                     }</span>
<span id="cb520-269"><a href="estimating-phylogenetic-trees.html#cb520-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-270"><a href="estimating-phylogenetic-trees.html#cb520-270" aria-hidden="true" tabindex="-1"></a><span class="co"># overwrite all previous output: -redo</span></span>
<span id="cb520-271"><a href="estimating-phylogenetic-trees.html#cb520-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-272"><a href="estimating-phylogenetic-trees.html#cb520-272" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(iqtree_redo) { counter_2 <span class="ot">&lt;-</span> counter_2 <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-273"><a href="estimating-phylogenetic-trees.html#cb520-273" aria-hidden="true" tabindex="-1"></a>                   out_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;  -redo  &quot;</span>) </span>
<span id="cb520-274"><a href="estimating-phylogenetic-trees.html#cb520-274" aria-hidden="true" tabindex="-1"></a>   list_of_iqtree[counter_2] <span class="ot">&lt;-</span> out_line</span>
<span id="cb520-275"><a href="estimating-phylogenetic-trees.html#cb520-275" aria-hidden="true" tabindex="-1"></a>                     }</span>
<span id="cb520-276"><a href="estimating-phylogenetic-trees.html#cb520-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-277"><a href="estimating-phylogenetic-trees.html#cb520-277" aria-hidden="true" tabindex="-1"></a><span class="co"># add partition</span></span>
<span id="cb520-278"><a href="estimating-phylogenetic-trees.html#cb520-278" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-279"><a href="estimating-phylogenetic-trees.html#cb520-279" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(iqtree_partition_files)) {counter_2 <span class="ot">&lt;-</span> counter_2 <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-280"><a href="estimating-phylogenetic-trees.html#cb520-280" aria-hidden="true" tabindex="-1"></a>                   out_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;  -p  &quot;</span>, iqtree_partition_files[i]) </span>
<span id="cb520-281"><a href="estimating-phylogenetic-trees.html#cb520-281" aria-hidden="true" tabindex="-1"></a>   list_of_iqtree[counter_2] <span class="ot">&lt;-</span> out_line</span>
<span id="cb520-282"><a href="estimating-phylogenetic-trees.html#cb520-282" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-283"><a href="estimating-phylogenetic-trees.html#cb520-283" aria-hidden="true" tabindex="-1"></a>                              }</span>
<span id="cb520-284"><a href="estimating-phylogenetic-trees.html#cb520-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-285"><a href="estimating-phylogenetic-trees.html#cb520-285" aria-hidden="true" tabindex="-1"></a><span class="co"># add DATE_FILE</span></span>
<span id="cb520-286"><a href="estimating-phylogenetic-trees.html#cb520-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-287"><a href="estimating-phylogenetic-trees.html#cb520-287" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">is.null</span>(iqtree_chronogram_from_tree)) {</span>
<span id="cb520-288"><a href="estimating-phylogenetic-trees.html#cb520-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-289"><a href="estimating-phylogenetic-trees.html#cb520-289" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(iqtree_date_files)) {counter_2 <span class="ot">&lt;-</span> counter_2 <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-290"><a href="estimating-phylogenetic-trees.html#cb520-290" aria-hidden="true" tabindex="-1"></a>                   out_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;  --date  &quot;</span>, iqtree_date_files[i], <span class="st">&#39; --date-tip 0&#39;</span>) </span>
<span id="cb520-291"><a href="estimating-phylogenetic-trees.html#cb520-291" aria-hidden="true" tabindex="-1"></a>   list_of_iqtree[counter_2] <span class="ot">&lt;-</span> out_line</span>
<span id="cb520-292"><a href="estimating-phylogenetic-trees.html#cb520-292" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-293"><a href="estimating-phylogenetic-trees.html#cb520-293" aria-hidden="true" tabindex="-1"></a>                              }</span>
<span id="cb520-294"><a href="estimating-phylogenetic-trees.html#cb520-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-295"><a href="estimating-phylogenetic-trees.html#cb520-295" aria-hidden="true" tabindex="-1"></a>                                        }</span>
<span id="cb520-296"><a href="estimating-phylogenetic-trees.html#cb520-296" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-297"><a href="estimating-phylogenetic-trees.html#cb520-297" aria-hidden="true" tabindex="-1"></a><span class="co"># make chronogram from tree</span></span>
<span id="cb520-298"><a href="estimating-phylogenetic-trees.html#cb520-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-299"><a href="estimating-phylogenetic-trees.html#cb520-299" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(iqtree_chronogram_from_tree)) {counter_2 <span class="ot">&lt;-</span> counter_2 <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-300"><a href="estimating-phylogenetic-trees.html#cb520-300" aria-hidden="true" tabindex="-1"></a>                   out_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;  --date  &quot;</span>, iqtree_date_files[i], <span class="st">&#39; --date-tip 0&#39;</span>, <span class="st">&quot; -te &quot;</span>, iqtree_chronogram_from_tree) </span>
<span id="cb520-301"><a href="estimating-phylogenetic-trees.html#cb520-301" aria-hidden="true" tabindex="-1"></a>   list_of_iqtree[counter_2] <span class="ot">&lt;-</span> out_line</span>
<span id="cb520-302"><a href="estimating-phylogenetic-trees.html#cb520-302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-303"><a href="estimating-phylogenetic-trees.html#cb520-303" aria-hidden="true" tabindex="-1"></a>                              }</span>
<span id="cb520-304"><a href="estimating-phylogenetic-trees.html#cb520-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-305"><a href="estimating-phylogenetic-trees.html#cb520-305" aria-hidden="true" tabindex="-1"></a><span class="do">##### ultra fast bootstrap</span></span>
<span id="cb520-306"><a href="estimating-phylogenetic-trees.html#cb520-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-307"><a href="estimating-phylogenetic-trees.html#cb520-307" aria-hidden="true" tabindex="-1"></a><span class="co"># UFBoot mode: -B 1000</span></span>
<span id="cb520-308"><a href="estimating-phylogenetic-trees.html#cb520-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-309"><a href="estimating-phylogenetic-trees.html#cb520-309" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(iqtree_UFBoot_B)) { counter_2 <span class="ot">&lt;-</span> counter_2 <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-310"><a href="estimating-phylogenetic-trees.html#cb520-310" aria-hidden="true" tabindex="-1"></a>                   out_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;  -B &quot;</span>, iqtree_UFBoot_B, <span class="st">&quot;  &quot;</span>) </span>
<span id="cb520-311"><a href="estimating-phylogenetic-trees.html#cb520-311" aria-hidden="true" tabindex="-1"></a>   list_of_iqtree[counter_2] <span class="ot">&lt;-</span> out_line</span>
<span id="cb520-312"><a href="estimating-phylogenetic-trees.html#cb520-312" aria-hidden="true" tabindex="-1"></a>                     }</span>
<span id="cb520-313"><a href="estimating-phylogenetic-trees.html#cb520-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-314"><a href="estimating-phylogenetic-trees.html#cb520-314" aria-hidden="true" tabindex="-1"></a><span class="co"># Red_impact_UFBoot</span></span>
<span id="cb520-315"><a href="estimating-phylogenetic-trees.html#cb520-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-316"><a href="estimating-phylogenetic-trees.html#cb520-316" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(iqtree_UFBoot_B)) {</span>
<span id="cb520-317"><a href="estimating-phylogenetic-trees.html#cb520-317" aria-hidden="true" tabindex="-1"></a>         <span class="cf">if</span>(iqtree_UFBoot_bnni) { counter_2 <span class="ot">&lt;-</span> counter_2 <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-318"><a href="estimating-phylogenetic-trees.html#cb520-318" aria-hidden="true" tabindex="-1"></a>                   out_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;  -bnni  &quot;</span>) </span>
<span id="cb520-319"><a href="estimating-phylogenetic-trees.html#cb520-319" aria-hidden="true" tabindex="-1"></a>   list_of_iqtree[counter_2] <span class="ot">&lt;-</span> out_line</span>
<span id="cb520-320"><a href="estimating-phylogenetic-trees.html#cb520-320" aria-hidden="true" tabindex="-1"></a>                                }</span>
<span id="cb520-321"><a href="estimating-phylogenetic-trees.html#cb520-321" aria-hidden="true" tabindex="-1"></a>                              }</span>
<span id="cb520-322"><a href="estimating-phylogenetic-trees.html#cb520-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-323"><a href="estimating-phylogenetic-trees.html#cb520-323" aria-hidden="true" tabindex="-1"></a><span class="do">##### non-parameteric bootstrap</span></span>
<span id="cb520-324"><a href="estimating-phylogenetic-trees.html#cb520-324" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-325"><a href="estimating-phylogenetic-trees.html#cb520-325" aria-hidden="true" tabindex="-1"></a><span class="co"># nonparametric bootstrap -b 100</span></span>
<span id="cb520-326"><a href="estimating-phylogenetic-trees.html#cb520-326" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-327"><a href="estimating-phylogenetic-trees.html#cb520-327" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(iqtree_nonpara_b)) { counter_2 <span class="ot">&lt;-</span> counter_2 <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-328"><a href="estimating-phylogenetic-trees.html#cb520-328" aria-hidden="true" tabindex="-1"></a>                   out_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;  -b &quot;</span>, iqtree_nonpara_b, <span class="st">&quot;  &quot;</span>) </span>
<span id="cb520-329"><a href="estimating-phylogenetic-trees.html#cb520-329" aria-hidden="true" tabindex="-1"></a>   list_of_iqtree[counter_2] <span class="ot">&lt;-</span> out_line</span>
<span id="cb520-330"><a href="estimating-phylogenetic-trees.html#cb520-330" aria-hidden="true" tabindex="-1"></a>                     }</span>
<span id="cb520-331"><a href="estimating-phylogenetic-trees.html#cb520-331" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-332"><a href="estimating-phylogenetic-trees.html#cb520-332" aria-hidden="true" tabindex="-1"></a><span class="do">##### SH-like approximate likelihood ratio test</span></span>
<span id="cb520-333"><a href="estimating-phylogenetic-trees.html#cb520-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-334"><a href="estimating-phylogenetic-trees.html#cb520-334" aria-hidden="true" tabindex="-1"></a><span class="co"># Assessing branch supports with single branch tests : -alrt 1000</span></span>
<span id="cb520-335"><a href="estimating-phylogenetic-trees.html#cb520-335" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-336"><a href="estimating-phylogenetic-trees.html#cb520-336" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(iqtree_alrt)) { counter_2 <span class="ot">&lt;-</span> counter_2 <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-337"><a href="estimating-phylogenetic-trees.html#cb520-337" aria-hidden="true" tabindex="-1"></a>                   out_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;  -alrt &quot;</span>, iqtree_alrt, <span class="st">&quot;  &quot;</span>) </span>
<span id="cb520-338"><a href="estimating-phylogenetic-trees.html#cb520-338" aria-hidden="true" tabindex="-1"></a>   list_of_iqtree[counter_2] <span class="ot">&lt;-</span> out_line</span>
<span id="cb520-339"><a href="estimating-phylogenetic-trees.html#cb520-339" aria-hidden="true" tabindex="-1"></a>                     }</span>
<span id="cb520-340"><a href="estimating-phylogenetic-trees.html#cb520-340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-341"><a href="estimating-phylogenetic-trees.html#cb520-341" aria-hidden="true" tabindex="-1"></a><span class="do">### multicore</span></span>
<span id="cb520-342"><a href="estimating-phylogenetic-trees.html#cb520-342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-343"><a href="estimating-phylogenetic-trees.html#cb520-343" aria-hidden="true" tabindex="-1"></a><span class="co"># multicore -T AUTO -T 2</span></span>
<span id="cb520-344"><a href="estimating-phylogenetic-trees.html#cb520-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-345"><a href="estimating-phylogenetic-trees.html#cb520-345" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(iqtree_multicore_T)) { counter_2 <span class="ot">&lt;-</span> counter_2 <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-346"><a href="estimating-phylogenetic-trees.html#cb520-346" aria-hidden="true" tabindex="-1"></a>                   out_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;  -T &quot;</span>, iqtree_multicore_T, <span class="st">&quot;  &quot;</span>) </span>
<span id="cb520-347"><a href="estimating-phylogenetic-trees.html#cb520-347" aria-hidden="true" tabindex="-1"></a>   list_of_iqtree[counter_2] <span class="ot">&lt;-</span> out_line</span>
<span id="cb520-348"><a href="estimating-phylogenetic-trees.html#cb520-348" aria-hidden="true" tabindex="-1"></a>                     }</span>
<span id="cb520-349"><a href="estimating-phylogenetic-trees.html#cb520-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-350"><a href="estimating-phylogenetic-trees.html#cb520-350" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(iqtree_multicore_T)) {</span>
<span id="cb520-351"><a href="estimating-phylogenetic-trees.html#cb520-351" aria-hidden="true" tabindex="-1"></a>          <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(iqtree_maxcores_ntmax)) {</span>
<span id="cb520-352"><a href="estimating-phylogenetic-trees.html#cb520-352" aria-hidden="true" tabindex="-1"></a>                                    counter_2 <span class="ot">&lt;-</span> counter_2 <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-353"><a href="estimating-phylogenetic-trees.html#cb520-353" aria-hidden="true" tabindex="-1"></a>                                     out_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;  -ntmax &quot;</span>, iqtree_maxcores_ntmax, <span class="st">&quot;  &quot;</span>) </span>
<span id="cb520-354"><a href="estimating-phylogenetic-trees.html#cb520-354" aria-hidden="true" tabindex="-1"></a>                    list_of_iqtree[counter_2] <span class="ot">&lt;-</span> out_line</span>
<span id="cb520-355"><a href="estimating-phylogenetic-trees.html#cb520-355" aria-hidden="true" tabindex="-1"></a>                                              }</span>
<span id="cb520-356"><a href="estimating-phylogenetic-trees.html#cb520-356" aria-hidden="true" tabindex="-1"></a>                                  }</span>
<span id="cb520-357"><a href="estimating-phylogenetic-trees.html#cb520-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-358"><a href="estimating-phylogenetic-trees.html#cb520-358" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-359"><a href="estimating-phylogenetic-trees.html#cb520-359" aria-hidden="true" tabindex="-1"></a><span class="do">### iqtree pass paramters</span></span>
<span id="cb520-360"><a href="estimating-phylogenetic-trees.html#cb520-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-361"><a href="estimating-phylogenetic-trees.html#cb520-361" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(iqtree_pass)) { counter_2 <span class="ot">&lt;-</span> counter_2 <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-362"><a href="estimating-phylogenetic-trees.html#cb520-362" aria-hidden="true" tabindex="-1"></a>                   out_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot; &quot;</span>, iqtree_pass, <span class="st">&quot;  &quot;</span>) </span>
<span id="cb520-363"><a href="estimating-phylogenetic-trees.html#cb520-363" aria-hidden="true" tabindex="-1"></a>   list_of_iqtree[counter_2] <span class="ot">&lt;-</span> out_line</span>
<span id="cb520-364"><a href="estimating-phylogenetic-trees.html#cb520-364" aria-hidden="true" tabindex="-1"></a>                          }</span>
<span id="cb520-365"><a href="estimating-phylogenetic-trees.html#cb520-365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-366"><a href="estimating-phylogenetic-trees.html#cb520-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-367"><a href="estimating-phylogenetic-trees.html#cb520-367" aria-hidden="true" tabindex="-1"></a><span class="co"># quiet</span></span>
<span id="cb520-368"><a href="estimating-phylogenetic-trees.html#cb520-368" aria-hidden="true" tabindex="-1"></a>                        </span>
<span id="cb520-369"><a href="estimating-phylogenetic-trees.html#cb520-369" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(iqtree_quiet) { counter_2 <span class="ot">&lt;-</span> counter_2 <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb520-370"><a href="estimating-phylogenetic-trees.html#cb520-370" aria-hidden="true" tabindex="-1"></a>                   out_line <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot; -quiet  &quot;</span>) </span>
<span id="cb520-371"><a href="estimating-phylogenetic-trees.html#cb520-371" aria-hidden="true" tabindex="-1"></a>   list_of_iqtree[counter_2] <span class="ot">&lt;-</span> out_line</span>
<span id="cb520-372"><a href="estimating-phylogenetic-trees.html#cb520-372" aria-hidden="true" tabindex="-1"></a>                          }</span>
<span id="cb520-373"><a href="estimating-phylogenetic-trees.html#cb520-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-374"><a href="estimating-phylogenetic-trees.html#cb520-374" aria-hidden="true" tabindex="-1"></a><span class="do">### all lines together</span></span>
<span id="cb520-375"><a href="estimating-phylogenetic-trees.html#cb520-375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-376"><a href="estimating-phylogenetic-trees.html#cb520-376" aria-hidden="true" tabindex="-1"></a>iqtree_out <span class="ot">&lt;-</span> <span class="fu">paste0</span>(list_of_iqtree, <span class="at">collapse =</span> <span class="st">&quot; &quot;</span>)</span>
<span id="cb520-377"><a href="estimating-phylogenetic-trees.html#cb520-377" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(iqtree_out)</span>
<span id="cb520-378"><a href="estimating-phylogenetic-trees.html#cb520-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-379"><a href="estimating-phylogenetic-trees.html#cb520-379" aria-hidden="true" tabindex="-1"></a>  list_iqtree_scripts[[counter]] <span class="ot">&lt;-</span> iqtree_out</span>
<span id="cb520-380"><a href="estimating-phylogenetic-trees.html#cb520-380" aria-hidden="true" tabindex="-1"></a>name_of_analyses_iqtree[counter] <span class="ot">&lt;-</span> one_input_file</span>
<span id="cb520-381"><a href="estimating-phylogenetic-trees.html#cb520-381" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-382"><a href="estimating-phylogenetic-trees.html#cb520-382" aria-hidden="true" tabindex="-1"></a>                                  }</span>
<span id="cb520-383"><a href="estimating-phylogenetic-trees.html#cb520-383" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">rm</span>(i)</span>
<span id="cb520-384"><a href="estimating-phylogenetic-trees.html#cb520-384" aria-hidden="true" tabindex="-1"></a>                       <span class="cf">if</span>(<span class="fu">exists</span>(<span class="st">&quot;counter&quot;</span>)){<span class="fu">rm</span>(counter)}</span>
<span id="cb520-385"><a href="estimating-phylogenetic-trees.html#cb520-385" aria-hidden="true" tabindex="-1"></a>                       <span class="cf">if</span>(<span class="fu">exists</span>(<span class="st">&quot;counter_2&quot;</span>)){<span class="fu">rm</span>(counter_2)}</span>
<span id="cb520-386"><a href="estimating-phylogenetic-trees.html#cb520-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-387"><a href="estimating-phylogenetic-trees.html#cb520-387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-388"><a href="estimating-phylogenetic-trees.html#cb520-388" aria-hidden="true" tabindex="-1"></a><span class="do">#####################################################################################</span></span>
<span id="cb520-389"><a href="estimating-phylogenetic-trees.html#cb520-389" aria-hidden="true" tabindex="-1"></a><span class="do">#### open and run iqtree from R</span></span>
<span id="cb520-390"><a href="estimating-phylogenetic-trees.html#cb520-390" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-391"><a href="estimating-phylogenetic-trees.html#cb520-391" aria-hidden="true" tabindex="-1"></a><span class="co"># create vector to export path -- export PATH=/opt/local/bin:/opt/local/sbin:$PATH</span></span>
<span id="cb520-392"><a href="estimating-phylogenetic-trees.html#cb520-392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-393"><a href="estimating-phylogenetic-trees.html#cb520-393" aria-hidden="true" tabindex="-1"></a>export_path_iqtree_bins <span class="ot">&lt;-</span> <span class="fu">normalizePath</span>(iqtree_bin_location)</span>
<span id="cb520-394"><a href="estimating-phylogenetic-trees.html#cb520-394" aria-hidden="true" tabindex="-1"></a>export_path_iqtree_vector <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;export PATH=&quot;</span>,export_path_iqtree_bins,<span class="st">&quot;:$PATH&quot;</span>)</span>
<span id="cb520-395"><a href="estimating-phylogenetic-trees.html#cb520-395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-396"><a href="estimating-phylogenetic-trees.html#cb520-396" aria-hidden="true" tabindex="-1"></a><span class="do">## iqtree execute function</span></span>
<span id="cb520-397"><a href="estimating-phylogenetic-trees.html#cb520-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-398"><a href="estimating-phylogenetic-trees.html#cb520-398" aria-hidden="true" tabindex="-1"></a>execute_iqtree_terminal <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">export_path_section =</span> export_path_iqtree_vector, </span>
<span id="cb520-399"><a href="estimating-phylogenetic-trees.html#cb520-399" aria-hidden="true" tabindex="-1"></a>                                   iqtree_command, </span>
<span id="cb520-400"><a href="estimating-phylogenetic-trees.html#cb520-400" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">intern =</span> <span class="cn">FALSE</span>, </span>
<span id="cb520-401"><a href="estimating-phylogenetic-trees.html#cb520-401" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">wait =</span> <span class="cn">FALSE</span>){</span>
<span id="cb520-402"><a href="estimating-phylogenetic-trees.html#cb520-402" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-403"><a href="estimating-phylogenetic-trees.html#cb520-403" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>(<span class="fu">paste0</span>(export_path_section, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,</span>
<span id="cb520-404"><a href="estimating-phylogenetic-trees.html#cb520-404" aria-hidden="true" tabindex="-1"></a>                    iqtree_command, <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>), </span>
<span id="cb520-405"><a href="estimating-phylogenetic-trees.html#cb520-405" aria-hidden="true" tabindex="-1"></a>                                <span class="at">intern =</span> intern, </span>
<span id="cb520-406"><a href="estimating-phylogenetic-trees.html#cb520-406" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">wait =</span> wait)</span>
<span id="cb520-407"><a href="estimating-phylogenetic-trees.html#cb520-407" aria-hidden="true" tabindex="-1"></a>                                                 }</span>
<span id="cb520-408"><a href="estimating-phylogenetic-trees.html#cb520-408" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-409"><a href="estimating-phylogenetic-trees.html#cb520-409" aria-hidden="true" tabindex="-1"></a><span class="do">## run list elements with iqtree commands</span></span>
<span id="cb520-410"><a href="estimating-phylogenetic-trees.html#cb520-410" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-411"><a href="estimating-phylogenetic-trees.html#cb520-411" aria-hidden="true" tabindex="-1"></a><span class="do">## serial</span></span>
<span id="cb520-412"><a href="estimating-phylogenetic-trees.html#cb520-412" aria-hidden="true" tabindex="-1"></a><span class="co"># run_iqtree &lt;- &quot;serial&quot;</span></span>
<span id="cb520-413"><a href="estimating-phylogenetic-trees.html#cb520-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-414"><a href="estimating-phylogenetic-trees.html#cb520-414" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(run_iqtree <span class="sc">==</span> <span class="st">&quot;serial&quot;</span>) {</span>
<span id="cb520-415"><a href="estimating-phylogenetic-trees.html#cb520-415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-416"><a href="estimating-phylogenetic-trees.html#cb520-416" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">****** started serial analisis of: &quot;</span>, <span class="fu">length</span>(list_iqtree_scripts), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span> )</span>
<span id="cb520-417"><a href="estimating-phylogenetic-trees.html#cb520-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-418"><a href="estimating-phylogenetic-trees.html#cb520-418" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(list_iqtree_scripts)) {</span>
<span id="cb520-419"><a href="estimating-phylogenetic-trees.html#cb520-419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-420"><a href="estimating-phylogenetic-trees.html#cb520-420" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;***** iqtree analysis name: &quot;</span>,name_of_analyses_iqtree[j], <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb520-421"><a href="estimating-phylogenetic-trees.html#cb520-421" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;***** iqtree commands: &quot;</span>, list_iqtree_scripts[[j]], <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb520-422"><a href="estimating-phylogenetic-trees.html#cb520-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-423"><a href="estimating-phylogenetic-trees.html#cb520-423" aria-hidden="true" tabindex="-1"></a><span class="fu">execute_iqtree_terminal</span>(<span class="at">export_path_section =</span> export_path_iqtree_vector, </span>
<span id="cb520-424"><a href="estimating-phylogenetic-trees.html#cb520-424" aria-hidden="true" tabindex="-1"></a>                             <span class="at">iqtree_command =</span>  list_iqtree_scripts[[j]], </span>
<span id="cb520-425"><a href="estimating-phylogenetic-trees.html#cb520-425" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">intern =</span> <span class="cn">FALSE</span>, </span>
<span id="cb520-426"><a href="estimating-phylogenetic-trees.html#cb520-426" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">wait =</span> <span class="cn">TRUE</span>)</span>
<span id="cb520-427"><a href="estimating-phylogenetic-trees.html#cb520-427" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;***** DONE ***** </span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb520-428"><a href="estimating-phylogenetic-trees.html#cb520-428" aria-hidden="true" tabindex="-1"></a>                                        }</span>
<span id="cb520-429"><a href="estimating-phylogenetic-trees.html#cb520-429" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">rm</span>(j)</span>
<span id="cb520-430"><a href="estimating-phylogenetic-trees.html#cb520-430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-431"><a href="estimating-phylogenetic-trees.html#cb520-431" aria-hidden="true" tabindex="-1"></a>                           }</span>
<span id="cb520-432"><a href="estimating-phylogenetic-trees.html#cb520-432" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-433"><a href="estimating-phylogenetic-trees.html#cb520-433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-434"><a href="estimating-phylogenetic-trees.html#cb520-434" aria-hidden="true" tabindex="-1"></a><span class="do">### parallel</span></span>
<span id="cb520-435"><a href="estimating-phylogenetic-trees.html#cb520-435" aria-hidden="true" tabindex="-1"></a><span class="co"># run_iqtree &lt;- &quot;parallel&quot;</span></span>
<span id="cb520-436"><a href="estimating-phylogenetic-trees.html#cb520-436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-437"><a href="estimating-phylogenetic-trees.html#cb520-437" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(run_iqtree <span class="sc">==</span> <span class="st">&quot;parallel&quot;</span>) {</span>
<span id="cb520-438"><a href="estimating-phylogenetic-trees.html#cb520-438" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-439"><a href="estimating-phylogenetic-trees.html#cb520-439" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">****** started parallel analisis of: &quot;</span>, <span class="fu">length</span>(list_iqtree_scripts), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span> )</span>
<span id="cb520-440"><a href="estimating-phylogenetic-trees.html#cb520-440" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(name_of_analyses_iqtree)</span>
<span id="cb520-441"><a href="estimating-phylogenetic-trees.html#cb520-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-442"><a href="estimating-phylogenetic-trees.html#cb520-442" aria-hidden="true" tabindex="-1"></a>ncores <span class="ot">&lt;-</span> parallel<span class="sc">::</span><span class="fu">detectCores</span>()</span>
<span id="cb520-443"><a href="estimating-phylogenetic-trees.html#cb520-443" aria-hidden="true" tabindex="-1"></a>cl <span class="ot">&lt;-</span> <span class="fu">makeCluster</span>(ncores)</span>
<span id="cb520-444"><a href="estimating-phylogenetic-trees.html#cb520-444" aria-hidden="true" tabindex="-1"></a><span class="fu">registerDoParallel</span>(cl)</span>
<span id="cb520-445"><a href="estimating-phylogenetic-trees.html#cb520-445" aria-hidden="true" tabindex="-1"></a><span class="fu">foreach</span>(<span class="at">iqtree_i=</span>list_iqtree_scripts, <span class="at">.export =</span> <span class="st">&quot;execute_iqtree_terminal&quot;</span>) <span class="sc">%dopar%</span> </span>
<span id="cb520-446"><a href="estimating-phylogenetic-trees.html#cb520-446" aria-hidden="true" tabindex="-1"></a>                               {<span class="fu">execute_iqtree_terminal</span>(<span class="at">iqtree_command=</span>iqtree_i)}</span>
<span id="cb520-447"><a href="estimating-phylogenetic-trees.html#cb520-447" aria-hidden="true" tabindex="-1"></a>parallel<span class="sc">::</span><span class="fu">stopCluster</span>(cl)</span>
<span id="cb520-448"><a href="estimating-phylogenetic-trees.html#cb520-448" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-449"><a href="estimating-phylogenetic-trees.html#cb520-449" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st"> ****** DONE ****** </span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb520-450"><a href="estimating-phylogenetic-trees.html#cb520-450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-451"><a href="estimating-phylogenetic-trees.html#cb520-451" aria-hidden="true" tabindex="-1"></a>                           }</span>
<span id="cb520-452"><a href="estimating-phylogenetic-trees.html#cb520-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-453"><a href="estimating-phylogenetic-trees.html#cb520-453" aria-hidden="true" tabindex="-1"></a><span class="do">### to return to user grab treefile</span></span>
<span id="cb520-454"><a href="estimating-phylogenetic-trees.html#cb520-454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-455"><a href="estimating-phylogenetic-trees.html#cb520-455" aria-hidden="true" tabindex="-1"></a>iqtrees_vector <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">pattern =</span> <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.treefile$&quot;</span>, <span class="at">ignore.case=</span><span class="cn">TRUE</span>)</span>
<span id="cb520-456"><a href="estimating-phylogenetic-trees.html#cb520-456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-457"><a href="estimating-phylogenetic-trees.html#cb520-457" aria-hidden="true" tabindex="-1"></a>list_of_trees <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb520-458"><a href="estimating-phylogenetic-trees.html#cb520-458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-459"><a href="estimating-phylogenetic-trees.html#cb520-459" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(iqtrees_vector)) {list_of_trees[[i]] <span class="ot">&lt;-</span> ape<span class="sc">::</span><span class="fu">read.tree</span>(iqtrees_vector[i])}</span>
<span id="cb520-460"><a href="estimating-phylogenetic-trees.html#cb520-460" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(i)</span>
<span id="cb520-461"><a href="estimating-phylogenetic-trees.html#cb520-461" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-462"><a href="estimating-phylogenetic-trees.html#cb520-462" aria-hidden="true" tabindex="-1"></a><span class="do">### if chronograms</span></span>
<span id="cb520-463"><a href="estimating-phylogenetic-trees.html#cb520-463" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-464"><a href="estimating-phylogenetic-trees.html#cb520-464" aria-hidden="true" tabindex="-1"></a>iqtree_chronograms <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">pattern =</span> <span class="st">&quot;</span><span class="sc">\\</span><span class="st">.timetree.nex$&quot;</span>, <span class="at">ignore.case=</span><span class="cn">TRUE</span>)</span>
<span id="cb520-465"><a href="estimating-phylogenetic-trees.html#cb520-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-466"><a href="estimating-phylogenetic-trees.html#cb520-466" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">length</span>(iqtree_chronograms) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb520-467"><a href="estimating-phylogenetic-trees.html#cb520-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-468"><a href="estimating-phylogenetic-trees.html#cb520-468" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(iqtree_chronograms)) {</span>
<span id="cb520-469"><a href="estimating-phylogenetic-trees.html#cb520-469" aria-hidden="true" tabindex="-1"></a>                               one_chronogram <span class="ot">&lt;-</span> ape<span class="sc">::</span><span class="fu">read.nexus</span>(iqtree_chronograms[i])</span>
<span id="cb520-470"><a href="estimating-phylogenetic-trees.html#cb520-470" aria-hidden="true" tabindex="-1"></a>                                      n_trees <span class="ot">&lt;-</span> <span class="fu">length</span>(list_of_trees)</span>
<span id="cb520-471"><a href="estimating-phylogenetic-trees.html#cb520-471" aria-hidden="true" tabindex="-1"></a>                   list_of_trees[[n_trees<span class="sc">+</span><span class="dv">1</span>]] <span class="ot">&lt;-</span> one_chronogram }</span>
<span id="cb520-472"><a href="estimating-phylogenetic-trees.html#cb520-472" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(i)</span>
<span id="cb520-473"><a href="estimating-phylogenetic-trees.html#cb520-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-474"><a href="estimating-phylogenetic-trees.html#cb520-474" aria-hidden="true" tabindex="-1"></a>                                   }</span>
<span id="cb520-475"><a href="estimating-phylogenetic-trees.html#cb520-475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-476"><a href="estimating-phylogenetic-trees.html#cb520-476" aria-hidden="true" tabindex="-1"></a><span class="do">############ open and run iqtree from R: DONE</span></span>
<span id="cb520-477"><a href="estimating-phylogenetic-trees.html#cb520-477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-478"><a href="estimating-phylogenetic-trees.html#cb520-478" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(master_wd_directory)</span>
<span id="cb520-479"><a href="estimating-phylogenetic-trees.html#cb520-479" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-480"><a href="estimating-phylogenetic-trees.html#cb520-480" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st"> *************         iqtree outfiles written to        ************* </span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb520-481"><a href="estimating-phylogenetic-trees.html#cb520-481" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(master_wd_directory)</span>
<span id="cb520-482"><a href="estimating-phylogenetic-trees.html#cb520-482" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-483"><a href="estimating-phylogenetic-trees.html#cb520-483" aria-hidden="true" tabindex="-1"></a><span class="fu">return</span>(list_of_trees)</span>
<span id="cb520-484"><a href="estimating-phylogenetic-trees.html#cb520-484" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb520-485"><a href="estimating-phylogenetic-trees.html#cb520-485" aria-hidden="true" tabindex="-1"></a>                                           }</span>
<span id="cb520-486"><a href="estimating-phylogenetic-trees.html#cb520-486" aria-hidden="true" tabindex="-1"></a><span class="do">##    </span><span class="re">END</span><span class="do"> OF FUNCTION   </span></span></code></pre></div>
<p>After you have copied and pasted the function <code>iqtree_runner_R()</code>, you will not get any result yet. However, the function will be in memory ready to run.</p>
</div>
<div id="running-iqtree2-from-r" class="section level2" number="12.5">
<h2><span class="header-section-number">12.5</span> Running <code>iqtree2</code> from R</h2>
<p>After you have loaded the function <code>iqtree_runner_R()</code>, we will perform several analyses. <strong>IMPORTANT</strong> Your alignment file (fasta or nexus) cannot have taxa, terminals or sequences with the same name; if you want to run including such sequences make sure that you label them accordingly.</p>
</div>
<div id="one-marker-and-no-partitions" class="section level2" number="12.6">
<h2><span class="header-section-number">12.6</span> One marker and no partitions</h2>
<p>In most cases, you just want a phylogenetic tree estimated from your sequences. This makes <strong>iqtree2</strong> extremely powerful as it can automatically select molecular model that best fits your data (i.e., <code>ModelFinder</code> section).</p>
<p><strong>4)</strong> We will start with the most simple using the COI aligned sequences in the file <code>Ameerega_aligned_end.fasta</code>.</p>
<p><strong>a)</strong> Set up your working directory (e.g., <code>my_phylogeny_1</code>) and get its path.</p>
<div class="sourceCode" id="cb521"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb521-1"><a href="estimating-phylogenetic-trees.html#cb521-1" aria-hidden="true" tabindex="-1"></a><span class="do">## this is exclusive to your OWN COMPUTER change it accordingly</span></span>
<span id="cb521-2"><a href="estimating-phylogenetic-trees.html#cb521-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/my_working_directory/my_phylogeny_1&quot;</span>)</span></code></pre></div>
<p><strong>b)</strong> find the file with aligned sequences: <code>Ameerega_aligned_end.fasta</code> and copy that file to your working directory (e.g., <code>my_phylogeny_1</code>)</p>
<div class="sourceCode" id="cb522"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb522-1"><a href="estimating-phylogenetic-trees.html#cb522-1" aria-hidden="true" tabindex="-1"></a><span class="do">## copy the file Ameerega_aligned_end.fasta to your working directory my_phylogeny_1</span></span>
<span id="cb522-2"><a href="estimating-phylogenetic-trees.html#cb522-2" aria-hidden="true" tabindex="-1"></a><span class="do">## this is the path to that file and exclusive to your OWN COMPUTER change it accordingly</span></span>
<span id="cb522-3"><a href="estimating-phylogenetic-trees.html#cb522-3" aria-hidden="true" tabindex="-1"></a>my_path_to_aligned_sequences <span class="ot">&lt;-</span> <span class="st">&quot;~/Desktop/Teach_R/my_working_directory/my_phylogeny_1/Ameerega_aligned_end.fasta&quot;</span></span></code></pre></div>
<p><img src="figures/session_12_figs/11_preparing_directory.png" width="90%" /></p>
<p><strong>c)</strong> find the path to the directory that contain the binary of <strong>iqtree2</strong>. Note that this is path to the directory, NOT the binary itself (check the syntax below).</p>
<div class="sourceCode" id="cb523"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb523-1"><a href="estimating-phylogenetic-trees.html#cb523-1" aria-hidden="true" tabindex="-1"></a><span class="do">## macOS: if you drag and drop the binary iqtree2</span></span>
<span id="cb523-2"><a href="estimating-phylogenetic-trees.html#cb523-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb523-3"><a href="estimating-phylogenetic-trees.html#cb523-3" aria-hidden="true" tabindex="-1"></a><span class="sc">~</span><span class="er">/</span>Desktop<span class="sc">/</span>Teach_R<span class="sc">/</span>class_pages_reference<span class="sc">/</span>bioinformatics_gitbook_1<span class="sc">/</span>my_programs<span class="sc">/</span>iqtree2</span>
<span id="cb523-4"><a href="estimating-phylogenetic-trees.html#cb523-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb523-5"><a href="estimating-phylogenetic-trees.html#cb523-5" aria-hidden="true" tabindex="-1"></a><span class="do">## this is exclusive to your OWN COMPUTER and we just need the path to the directory that contains iqtree2</span></span>
<span id="cb523-6"><a href="estimating-phylogenetic-trees.html#cb523-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Note: the diffecent from path above and the one below.</span></span>
<span id="cb523-7"><a href="estimating-phylogenetic-trees.html#cb523-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb523-8"><a href="estimating-phylogenetic-trees.html#cb523-8" aria-hidden="true" tabindex="-1"></a>my_path_to_iqtree2 <span class="ot">&lt;-</span> <span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/my_programs&quot;</span></span></code></pre></div>
<p><strong>d)</strong> Now, we can run the function <code>iqtree_runner_R()</code>.</p>
<div class="sourceCode" id="cb524"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb524-1"><a href="estimating-phylogenetic-trees.html#cb524-1" aria-hidden="true" tabindex="-1"></a><span class="do">## SIMPLE iqtree2 run </span></span>
<span id="cb524-2"><a href="estimating-phylogenetic-trees.html#cb524-2" aria-hidden="true" tabindex="-1"></a>COI_simple_phylogeny <span class="ot">&lt;-</span> <span class="fu">iqtree_runner_R</span> (<span class="at">input_alignment_file_user =</span> my_path_to_aligned_sequences,</span>
<span id="cb524-3"><a href="estimating-phylogenetic-trees.html#cb524-3" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_bin_location_user =</span> my_path_to_iqtree2,</span>
<span id="cb524-4"><a href="estimating-phylogenetic-trees.html#cb524-4" aria-hidden="true" tabindex="-1"></a>                           <span class="at">run_iqtree_call_user =</span> <span class="st">&quot;serial&quot;</span>,</span>
<span id="cb524-5"><a href="estimating-phylogenetic-trees.html#cb524-5" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_input_model_m_user =</span> <span class="st">&quot;MFP+MERGE&quot;</span>)</span>
<span id="cb524-6"><a href="estimating-phylogenetic-trees.html#cb524-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb524-7"><a href="estimating-phylogenetic-trees.html#cb524-7" aria-hidden="true" tabindex="-1"></a><span class="do">## you will get this output on the console</span></span>
<span id="cb524-8"><a href="estimating-phylogenetic-trees.html#cb524-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb524-9"><a href="estimating-phylogenetic-trees.html#cb524-9" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;iqtree2  -s Ameerega_aligned_end.fasta     -m  MFP+MERGE  &quot;</span></span>
<span id="cb524-10"><a href="estimating-phylogenetic-trees.html#cb524-10" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-11"><a href="estimating-phylogenetic-trees.html#cb524-11" aria-hidden="true" tabindex="-1"></a><span class="co"># ****** started serial analisis of:  1 </span></span>
<span id="cb524-12"><a href="estimating-phylogenetic-trees.html#cb524-12" aria-hidden="true" tabindex="-1"></a><span class="co"># ***** iqtree analysis name:  Ameerega_aligned_end.fasta </span></span>
<span id="cb524-13"><a href="estimating-phylogenetic-trees.html#cb524-13" aria-hidden="true" tabindex="-1"></a><span class="co"># ***** iqtree commands:  iqtree2  -s Ameerega_aligned_end.fasta     -m  MFP+MERGE   </span></span>
<span id="cb524-14"><a href="estimating-phylogenetic-trees.html#cb524-14" aria-hidden="true" tabindex="-1"></a><span class="co"># IQ-TREE multicore version 2.1.2 COVID-edition for Mac OS X 64-bit built Oct 22 2020</span></span>
<span id="cb524-15"><a href="estimating-phylogenetic-trees.html#cb524-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Developed by Bui Quang Minh, James Barbetti, Nguyen Lam Tung,</span></span>
<span id="cb524-16"><a href="estimating-phylogenetic-trees.html#cb524-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Olga Chernomor, Heiko Schmidt, Dominik Schrempf, Michael Woodhams.</span></span>
<span id="cb524-17"><a href="estimating-phylogenetic-trees.html#cb524-17" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-18"><a href="estimating-phylogenetic-trees.html#cb524-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Host:    LFQSANTOSJ2016 (AVX2, FMA3, 8 GB RAM)</span></span>
<span id="cb524-19"><a href="estimating-phylogenetic-trees.html#cb524-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Command: iqtree2 -s Ameerega_aligned_end.fasta -m MFP+MERGE</span></span>
<span id="cb524-20"><a href="estimating-phylogenetic-trees.html#cb524-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Seed:    487027 (Using SPRNG - Scalable Parallel Random Number Generator)</span></span>
<span id="cb524-21"><a href="estimating-phylogenetic-trees.html#cb524-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Time:    Tue Feb 23 17:46:25 2021</span></span>
<span id="cb524-22"><a href="estimating-phylogenetic-trees.html#cb524-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Kernel:  AVX+FMA - 1 threads (4 CPU cores detected)</span></span>
<span id="cb524-23"><a href="estimating-phylogenetic-trees.html#cb524-23" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-24"><a href="estimating-phylogenetic-trees.html#cb524-24" aria-hidden="true" tabindex="-1"></a><span class="co"># HINT: Use -nt option to specify number of threads because your CPU has 4 cores!</span></span>
<span id="cb524-25"><a href="estimating-phylogenetic-trees.html#cb524-25" aria-hidden="true" tabindex="-1"></a><span class="co"># HINT: -nt AUTO will automatically determine the best number of threads to use.</span></span>
<span id="cb524-26"><a href="estimating-phylogenetic-trees.html#cb524-26" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-27"><a href="estimating-phylogenetic-trees.html#cb524-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Reading alignment file Ameerega_aligned_end.fasta ... Fasta format detected</span></span>
<span id="cb524-28"><a href="estimating-phylogenetic-trees.html#cb524-28" aria-hidden="true" tabindex="-1"></a><span class="co"># Alignment most likely contains DNA/RNA sequences</span></span>
<span id="cb524-29"><a href="estimating-phylogenetic-trees.html#cb524-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Alignment has 24 sequences with 1539 columns, 452 distinct patterns</span></span>
<span id="cb524-30"><a href="estimating-phylogenetic-trees.html#cb524-30" aria-hidden="true" tabindex="-1"></a><span class="co"># 378 parsimony-informative, 140 singleton sites, 1021 constant sites</span></span>
<span id="cb524-31"><a href="estimating-phylogenetic-trees.html#cb524-31" aria-hidden="true" tabindex="-1"></a><span class="co">#             Gap/Ambiguity  Composition  p-value</span></span>
<span id="cb524-32"><a href="estimating-phylogenetic-trees.html#cb524-32" aria-hidden="true" tabindex="-1"></a><span class="co">#    1  MW042032.1    0.00%    passed     91.59%</span></span>
<span id="cb524-33"><a href="estimating-phylogenetic-trees.html#cb524-33" aria-hidden="true" tabindex="-1"></a><span class="co">#    2  MW042031.1    0.00%    passed     61.48%</span></span>
<span id="cb524-34"><a href="estimating-phylogenetic-trees.html#cb524-34" aria-hidden="true" tabindex="-1"></a><span class="co">#    3  MW042030.1    0.00%    passed     81.11%</span></span>
<span id="cb524-35"><a href="estimating-phylogenetic-trees.html#cb524-35" aria-hidden="true" tabindex="-1"></a><span class="co">#    4  KU494334.1   58.02%    passed     72.66%</span></span>
<span id="cb524-36"><a href="estimating-phylogenetic-trees.html#cb524-36" aria-hidden="true" tabindex="-1"></a><span class="co">#    5  KU494333.1   57.96%    passed     99.83%</span></span>
<span id="cb524-37"><a href="estimating-phylogenetic-trees.html#cb524-37" aria-hidden="true" tabindex="-1"></a><span class="co">#    6  KU494332.1   57.96%    passed     99.83%</span></span>
<span id="cb524-38"><a href="estimating-phylogenetic-trees.html#cb524-38" aria-hidden="true" tabindex="-1"></a><span class="co">#    7  AF097506.1   66.21%    passed     51.07%</span></span>
<span id="cb524-39"><a href="estimating-phylogenetic-trees.html#cb524-39" aria-hidden="true" tabindex="-1"></a><span class="co">#    8  DQ502932.1   57.24%    passed     90.55%</span></span>
<span id="cb524-40"><a href="estimating-phylogenetic-trees.html#cb524-40" aria-hidden="true" tabindex="-1"></a><span class="co">#    9  DQ502929.1   57.24%    passed     55.70%</span></span>
<span id="cb524-41"><a href="estimating-phylogenetic-trees.html#cb524-41" aria-hidden="true" tabindex="-1"></a><span class="co">#   10  DQ502922.1   57.24%    passed     84.11%</span></span>
<span id="cb524-42"><a href="estimating-phylogenetic-trees.html#cb524-42" aria-hidden="true" tabindex="-1"></a><span class="co">#   11  DQ502920.1   57.24%    passed     33.21%</span></span>
<span id="cb524-43"><a href="estimating-phylogenetic-trees.html#cb524-43" aria-hidden="true" tabindex="-1"></a><span class="co">#   12  DQ502903.1   57.24%    passed     51.16%</span></span>
<span id="cb524-44"><a href="estimating-phylogenetic-trees.html#cb524-44" aria-hidden="true" tabindex="-1"></a><span class="co">#   13  DQ502902.1   57.24%    passed     65.03%</span></span>
<span id="cb524-45"><a href="estimating-phylogenetic-trees.html#cb524-45" aria-hidden="true" tabindex="-1"></a><span class="co">#   14  DQ502901.1   57.24%    passed     48.46%</span></span>
<span id="cb524-46"><a href="estimating-phylogenetic-trees.html#cb524-46" aria-hidden="true" tabindex="-1"></a><span class="co">#   15  DQ502851.1   57.24%    passed     94.07%</span></span>
<span id="cb524-47"><a href="estimating-phylogenetic-trees.html#cb524-47" aria-hidden="true" tabindex="-1"></a><span class="co">#   16  DQ502849.1   57.24%    passed     33.21%</span></span>
<span id="cb524-48"><a href="estimating-phylogenetic-trees.html#cb524-48" aria-hidden="true" tabindex="-1"></a><span class="co">#   17  DQ502848.1   57.24%    passed     29.31%</span></span>
<span id="cb524-49"><a href="estimating-phylogenetic-trees.html#cb524-49" aria-hidden="true" tabindex="-1"></a><span class="co">#   18  DQ502832.1   57.31%    passed     91.59%</span></span>
<span id="cb524-50"><a href="estimating-phylogenetic-trees.html#cb524-50" aria-hidden="true" tabindex="-1"></a><span class="co">#   19  DQ502831.1   57.24%    passed     93.67%</span></span>
<span id="cb524-51"><a href="estimating-phylogenetic-trees.html#cb524-51" aria-hidden="true" tabindex="-1"></a><span class="co">#   20  DQ502825.1   57.24%    passed     60.02%</span></span>
<span id="cb524-52"><a href="estimating-phylogenetic-trees.html#cb524-52" aria-hidden="true" tabindex="-1"></a><span class="co">#   21  MW042037.1    0.00%    passed     77.97%</span></span>
<span id="cb524-53"><a href="estimating-phylogenetic-trees.html#cb524-53" aria-hidden="true" tabindex="-1"></a><span class="co">#   22  KR862889.1   57.50%    passed     41.02%</span></span>
<span id="cb524-54"><a href="estimating-phylogenetic-trees.html#cb524-54" aria-hidden="true" tabindex="-1"></a><span class="co">#   23  MW042036.1    0.00%    failed      0.13%</span></span>
<span id="cb524-55"><a href="estimating-phylogenetic-trees.html#cb524-55" aria-hidden="true" tabindex="-1"></a><span class="co">#   24  MW042039.1    0.00%    failed      0.57%</span></span>
<span id="cb524-56"><a href="estimating-phylogenetic-trees.html#cb524-56" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">WARNING</span><span class="co">: 18 sequences contain more than 50% gaps/ambiguity</span></span>
<span id="cb524-57"><a href="estimating-phylogenetic-trees.html#cb524-57" aria-hidden="true" tabindex="-1"></a><span class="co"># ****  TOTAL        43.41%  2 sequences failed composition chi2 test (p-value&lt;5%; df=3)</span></span>
<span id="cb524-58"><a href="estimating-phylogenetic-trees.html#cb524-58" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: KU494332.1 is identical to KU494333.1 but kept for subsequent analysis</span></span>
<span id="cb524-59"><a href="estimating-phylogenetic-trees.html#cb524-59" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-60"><a href="estimating-phylogenetic-trees.html#cb524-60" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-61"><a href="estimating-phylogenetic-trees.html#cb524-61" aria-hidden="true" tabindex="-1"></a><span class="co"># Create initial parsimony tree by phylogenetic likelihood library (PLL)... 0.003 seconds</span></span>
<span id="cb524-62"><a href="estimating-phylogenetic-trees.html#cb524-62" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform fast likelihood tree search using GTR+I+G model...</span></span>
<span id="cb524-63"><a href="estimating-phylogenetic-trees.html#cb524-63" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate model parameters (epsilon = 5.000)</span></span>
<span id="cb524-64"><a href="estimating-phylogenetic-trees.html#cb524-64" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-65"><a href="estimating-phylogenetic-trees.html#cb524-65" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">WARNING</span><span class="co">: Estimated model parameters are at boundary that can cause numerical instability!</span></span>
<span id="cb524-66"><a href="estimating-phylogenetic-trees.html#cb524-66" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-67"><a href="estimating-phylogenetic-trees.html#cb524-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform nearest neighbor interchange...</span></span>
<span id="cb524-68"><a href="estimating-phylogenetic-trees.html#cb524-68" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate model parameters (epsilon = 1.000)</span></span>
<span id="cb524-69"><a href="estimating-phylogenetic-trees.html#cb524-69" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. Initial log-likelihood: -7191.031</span></span>
<span id="cb524-70"><a href="estimating-phylogenetic-trees.html#cb524-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Optimal log-likelihood: -7189.967</span></span>
<span id="cb524-71"><a href="estimating-phylogenetic-trees.html#cb524-71" aria-hidden="true" tabindex="-1"></a><span class="co"># Rate parameters:  A-C: 7.98907  A-G: 64.12175  A-T: 11.66195  C-G: 1.43333  C-T: 100.00000  G-T: 1.00000</span></span>
<span id="cb524-72"><a href="estimating-phylogenetic-trees.html#cb524-72" aria-hidden="true" tabindex="-1"></a><span class="co"># Base frequencies:  A: 0.240  C: 0.266  G: 0.172  T: 0.321</span></span>
<span id="cb524-73"><a href="estimating-phylogenetic-trees.html#cb524-73" aria-hidden="true" tabindex="-1"></a><span class="co"># Proportion of invariable sites: 0.357</span></span>
<span id="cb524-74"><a href="estimating-phylogenetic-trees.html#cb524-74" aria-hidden="true" tabindex="-1"></a><span class="co"># Gamma shape alpha: 0.388</span></span>
<span id="cb524-75"><a href="estimating-phylogenetic-trees.html#cb524-75" aria-hidden="true" tabindex="-1"></a><span class="co"># Parameters optimization took 1 rounds (0.044 sec)</span></span>
<span id="cb524-76"><a href="estimating-phylogenetic-trees.html#cb524-76" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-77"><a href="estimating-phylogenetic-trees.html#cb524-77" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">WARNING</span><span class="co">: Estimated model parameters are at boundary that can cause numerical instability!</span></span>
<span id="cb524-78"><a href="estimating-phylogenetic-trees.html#cb524-78" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-79"><a href="estimating-phylogenetic-trees.html#cb524-79" aria-hidden="true" tabindex="-1"></a><span class="co"># Time for fast ML tree search: 0.322 seconds</span></span>
<span id="cb524-80"><a href="estimating-phylogenetic-trees.html#cb524-80" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-81"><a href="estimating-phylogenetic-trees.html#cb524-81" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: ModelFinder requires 3 MB RAM!</span></span>
<span id="cb524-82"><a href="estimating-phylogenetic-trees.html#cb524-82" aria-hidden="true" tabindex="-1"></a><span class="co"># ModelFinder will test up to 286 DNA models (sample size: 1539) ...</span></span>
<span id="cb524-83"><a href="estimating-phylogenetic-trees.html#cb524-83" aria-hidden="true" tabindex="-1"></a><span class="co">#  No. Model         -LnL         df  AIC          AICc         BIC</span></span>
<span id="cb524-84"><a href="estimating-phylogenetic-trees.html#cb524-84" aria-hidden="true" tabindex="-1"></a><span class="co">#   1  GTR+F         7870.984     53  15847.967    15851.822    16130.928</span></span>
<span id="cb524-85"><a href="estimating-phylogenetic-trees.html#cb524-85" aria-hidden="true" tabindex="-1"></a><span class="co">#   2  GTR+F+I       7216.150     54  14540.301    14544.303    14828.601</span></span>
<span id="cb524-86"><a href="estimating-phylogenetic-trees.html#cb524-86" aria-hidden="true" tabindex="-1"></a><span class="co">#   3  GTR+F+G4      7199.116     54  14506.232    14510.235    14794.532</span></span>
<span id="cb524-87"><a href="estimating-phylogenetic-trees.html#cb524-87" aria-hidden="true" tabindex="-1"></a><span class="co">#   4  GTR+F+I+G4    7176.558     55  14463.117    14467.270    14756.755</span></span>
<span id="cb524-88"><a href="estimating-phylogenetic-trees.html#cb524-88" aria-hidden="true" tabindex="-1"></a><span class="co">#   5  GTR+F+R2      7295.486     55  14700.971    14705.125    14994.610</span></span>
<span id="cb524-89"><a href="estimating-phylogenetic-trees.html#cb524-89" aria-hidden="true" tabindex="-1"></a><span class="co">#   6  GTR+F+R3      7242.022     57  14598.044    14602.509    14902.361</span></span>
<span id="cb524-90"><a href="estimating-phylogenetic-trees.html#cb524-90" aria-hidden="true" tabindex="-1"></a><span class="co">#   7  GTR+F+R4      7215.932     59  14549.863    14554.650    14864.858</span></span>
<span id="cb524-91"><a href="estimating-phylogenetic-trees.html#cb524-91" aria-hidden="true" tabindex="-1"></a><span class="co">#   8  GTR+F+R5      7202.474     61  14526.949    14532.070    14852.621</span></span>
<span id="cb524-92"><a href="estimating-phylogenetic-trees.html#cb524-92" aria-hidden="true" tabindex="-1"></a><span class="co">#   9  GTR+F+R6      7195.140     63  14516.279    14521.746    14852.629</span></span>
<span id="cb524-93"><a href="estimating-phylogenetic-trees.html#cb524-93" aria-hidden="true" tabindex="-1"></a><span class="co">#  17  SYM+I+G4      7230.681     52  14565.361    14569.071    14842.984</span></span>
<span id="cb524-94"><a href="estimating-phylogenetic-trees.html#cb524-94" aria-hidden="true" tabindex="-1"></a><span class="co">#  30  TVM+F+I+G4    7179.891     54  14467.782    14471.784    14756.081</span></span>
<span id="cb524-95"><a href="estimating-phylogenetic-trees.html#cb524-95" aria-hidden="true" tabindex="-1"></a><span class="co">#  43  TVMe+I+G4     7265.512     51  14633.024    14636.590    14905.307</span></span>
<span id="cb524-96"><a href="estimating-phylogenetic-trees.html#cb524-96" aria-hidden="true" tabindex="-1"></a><span class="co">#  56  TIM3+F+I+G4   7203.542     53  14513.085    14516.939    14796.046</span></span>
<span id="cb524-97"><a href="estimating-phylogenetic-trees.html#cb524-97" aria-hidden="true" tabindex="-1"></a><span class="co">#  69  TIM3e+I+G4    7284.220     50  14668.440    14671.867    14935.384</span></span>
<span id="cb524-98"><a href="estimating-phylogenetic-trees.html#cb524-98" aria-hidden="true" tabindex="-1"></a><span class="co">#  82  TIM2+F+I+G4   7177.515     53  14461.029    14464.884    14743.990</span></span>
<span id="cb524-99"><a href="estimating-phylogenetic-trees.html#cb524-99" aria-hidden="true" tabindex="-1"></a><span class="co">#  95  TIM2e+I+G4    7233.907     50  14567.815    14571.242    14834.759</span></span>
<span id="cb524-100"><a href="estimating-phylogenetic-trees.html#cb524-100" aria-hidden="true" tabindex="-1"></a><span class="co"># 108  TIM+F+I+G4    7202.565     53  14511.129    14514.984    14794.090</span></span>
<span id="cb524-101"><a href="estimating-phylogenetic-trees.html#cb524-101" aria-hidden="true" tabindex="-1"></a><span class="co"># 121  TIMe+I+G4     7283.344     50  14666.689    14670.116    14933.633</span></span>
<span id="cb524-102"><a href="estimating-phylogenetic-trees.html#cb524-102" aria-hidden="true" tabindex="-1"></a><span class="co"># 134  TPM3u+F+I+G4  7205.716     52  14515.433    14519.142    14793.055</span></span>
<span id="cb524-103"><a href="estimating-phylogenetic-trees.html#cb524-103" aria-hidden="true" tabindex="-1"></a><span class="co"># 147  TPM3+F+I+G4   7205.716     52  14515.433    14519.142    14793.055</span></span>
<span id="cb524-104"><a href="estimating-phylogenetic-trees.html#cb524-104" aria-hidden="true" tabindex="-1"></a><span class="co"># 160  TPM2u+F+I+G4  7181.001     52  14466.003    14469.712    14743.625</span></span>
<span id="cb524-105"><a href="estimating-phylogenetic-trees.html#cb524-105" aria-hidden="true" tabindex="-1"></a><span class="co"># 173  TPM2+F+I+G4   7180.996     52  14465.992    14469.702    14743.614</span></span>
<span id="cb524-106"><a href="estimating-phylogenetic-trees.html#cb524-106" aria-hidden="true" tabindex="-1"></a><span class="co"># 186  K3Pu+F+I+G4   7204.884     52  14513.768    14517.477    14791.390</span></span>
<span id="cb524-107"><a href="estimating-phylogenetic-trees.html#cb524-107" aria-hidden="true" tabindex="-1"></a><span class="co"># 199  K3P+I+G4      7315.342     49  14728.684    14731.975    14990.290</span></span>
<span id="cb524-108"><a href="estimating-phylogenetic-trees.html#cb524-108" aria-hidden="true" tabindex="-1"></a><span class="co"># 212  TN+F+I+G4     7204.352     52  14512.704    14516.413    14790.326</span></span>
<span id="cb524-109"><a href="estimating-phylogenetic-trees.html#cb524-109" aria-hidden="true" tabindex="-1"></a><span class="co"># 225  TNe+I+G4      7286.452     49  14670.904    14674.195    14932.510</span></span>
<span id="cb524-110"><a href="estimating-phylogenetic-trees.html#cb524-110" aria-hidden="true" tabindex="-1"></a><span class="co"># 238  HKY+F+I+G4    7206.672     51  14515.343    14518.910    14787.627</span></span>
<span id="cb524-111"><a href="estimating-phylogenetic-trees.html#cb524-111" aria-hidden="true" tabindex="-1"></a><span class="co"># 251  K2P+I+G4      7318.616     48  14733.231    14736.388    14989.498</span></span>
<span id="cb524-112"><a href="estimating-phylogenetic-trees.html#cb524-112" aria-hidden="true" tabindex="-1"></a><span class="co"># 264  F81+F+I+G4    7804.765     50  15709.531    15712.958    15976.475</span></span>
<span id="cb524-113"><a href="estimating-phylogenetic-trees.html#cb524-113" aria-hidden="true" tabindex="-1"></a><span class="co"># 277  JC+I+G4       7964.515     47  16023.030    16026.056    16273.957</span></span>
<span id="cb524-114"><a href="estimating-phylogenetic-trees.html#cb524-114" aria-hidden="true" tabindex="-1"></a><span class="co"># Akaike Information Criterion:           TIM2+F+I+G4</span></span>
<span id="cb524-115"><a href="estimating-phylogenetic-trees.html#cb524-115" aria-hidden="true" tabindex="-1"></a><span class="co"># Corrected Akaike Information Criterion: TIM2+F+I+G4</span></span>
<span id="cb524-116"><a href="estimating-phylogenetic-trees.html#cb524-116" aria-hidden="true" tabindex="-1"></a><span class="co"># Bayesian Information Criterion:         TPM2+F+I+G4</span></span>
<span id="cb524-117"><a href="estimating-phylogenetic-trees.html#cb524-117" aria-hidden="true" tabindex="-1"></a><span class="co"># Best-fit model: TPM2+F+I+G4 chosen according to BIC</span></span>
<span id="cb524-118"><a href="estimating-phylogenetic-trees.html#cb524-118" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-119"><a href="estimating-phylogenetic-trees.html#cb524-119" aria-hidden="true" tabindex="-1"></a><span class="co"># All model information printed to Ameerega_aligned_end.fasta.model.gz</span></span>
<span id="cb524-120"><a href="estimating-phylogenetic-trees.html#cb524-120" aria-hidden="true" tabindex="-1"></a><span class="co"># CPU time for ModelFinder: 6.355 seconds (0h:0m:6s)</span></span>
<span id="cb524-121"><a href="estimating-phylogenetic-trees.html#cb524-121" aria-hidden="true" tabindex="-1"></a><span class="co"># Wall-clock time for ModelFinder: 6.536 seconds (0h:0m:6s)</span></span>
<span id="cb524-122"><a href="estimating-phylogenetic-trees.html#cb524-122" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-123"><a href="estimating-phylogenetic-trees.html#cb524-123" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co">: 1 MB RAM (0 GB) is required!</span></span>
<span id="cb524-124"><a href="estimating-phylogenetic-trees.html#cb524-124" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate model parameters (epsilon = 0.100)</span></span>
<span id="cb524-125"><a href="estimating-phylogenetic-trees.html#cb524-125" aria-hidden="true" tabindex="-1"></a><span class="co"># Thoroughly optimizing +I+G parameters from 10 start values...</span></span>
<span id="cb524-126"><a href="estimating-phylogenetic-trees.html#cb524-126" aria-hidden="true" tabindex="-1"></a><span class="co"># Init pinv, alpha: 0.000, 1.370 / Estimate: 0.000, 0.166 / LogL: -7202.724</span></span>
<span id="cb524-127"><a href="estimating-phylogenetic-trees.html#cb524-127" aria-hidden="true" tabindex="-1"></a><span class="co"># Init pinv, alpha: 0.074, 1.370 / Estimate: 0.071, 0.192 / LogL: -7203.417</span></span>
<span id="cb524-128"><a href="estimating-phylogenetic-trees.html#cb524-128" aria-hidden="true" tabindex="-1"></a><span class="co"># Init pinv, alpha: 0.147, 1.370 / Estimate: 0.149, 0.227 / LogL: -7203.178</span></span>
<span id="cb524-129"><a href="estimating-phylogenetic-trees.html#cb524-129" aria-hidden="true" tabindex="-1"></a><span class="co"># Init pinv, alpha: 0.221, 1.370 / Estimate: 0.592, 1.298 / LogL: -7181.054</span></span>
<span id="cb524-130"><a href="estimating-phylogenetic-trees.html#cb524-130" aria-hidden="true" tabindex="-1"></a><span class="co"># Init pinv, alpha: 0.295, 1.370 / Estimate: 0.593, 1.312 / LogL: -7181.038</span></span>
<span id="cb524-131"><a href="estimating-phylogenetic-trees.html#cb524-131" aria-hidden="true" tabindex="-1"></a><span class="co"># Init pinv, alpha: 0.369, 1.370 / Estimate: 0.592, 1.300 / LogL: -7181.052</span></span>
<span id="cb524-132"><a href="estimating-phylogenetic-trees.html#cb524-132" aria-hidden="true" tabindex="-1"></a><span class="co"># Init pinv, alpha: 0.442, 1.370 / Estimate: 0.593, 1.311 / LogL: -7181.038</span></span>
<span id="cb524-133"><a href="estimating-phylogenetic-trees.html#cb524-133" aria-hidden="true" tabindex="-1"></a><span class="co"># Init pinv, alpha: 0.516, 1.370 / Estimate: 0.593, 1.314 / LogL: -7181.034</span></span>
<span id="cb524-134"><a href="estimating-phylogenetic-trees.html#cb524-134" aria-hidden="true" tabindex="-1"></a><span class="co"># Init pinv, alpha: 0.590, 1.370 / Estimate: 0.593, 1.324 / LogL: -7181.039</span></span>
<span id="cb524-135"><a href="estimating-phylogenetic-trees.html#cb524-135" aria-hidden="true" tabindex="-1"></a><span class="co"># Init pinv, alpha: 0.663, 1.370 / Estimate: 0.603, 1.476 / LogL: -7181.034</span></span>
<span id="cb524-136"><a href="estimating-phylogenetic-trees.html#cb524-136" aria-hidden="true" tabindex="-1"></a><span class="co"># Optimal pinv,alpha: 0.593, 1.314 / LogL: -7181.034</span></span>
<span id="cb524-137"><a href="estimating-phylogenetic-trees.html#cb524-137" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-138"><a href="estimating-phylogenetic-trees.html#cb524-138" aria-hidden="true" tabindex="-1"></a><span class="co"># Parameters optimization took 2.390 sec</span></span>
<span id="cb524-139"><a href="estimating-phylogenetic-trees.html#cb524-139" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-140"><a href="estimating-phylogenetic-trees.html#cb524-140" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">WARNING</span><span class="co">: Estimated model parameters are at boundary that can cause numerical instability!</span></span>
<span id="cb524-141"><a href="estimating-phylogenetic-trees.html#cb524-141" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-142"><a href="estimating-phylogenetic-trees.html#cb524-142" aria-hidden="true" tabindex="-1"></a><span class="co"># Computing ML distances based on estimated model parameters...</span></span>
<span id="cb524-143"><a href="estimating-phylogenetic-trees.html#cb524-143" aria-hidden="true" tabindex="-1"></a><span class="co"># Computing ML distances took 0.013737 sec (of wall-clock time) 0.013278 sec(of CPU time)</span></span>
<span id="cb524-144"><a href="estimating-phylogenetic-trees.html#cb524-144" aria-hidden="true" tabindex="-1"></a><span class="co"># Computing RapidNJ tree took 0.000465 sec (of wall-clock time) 0.000218 sec (of CPU time)</span></span>
<span id="cb524-145"><a href="estimating-phylogenetic-trees.html#cb524-145" aria-hidden="true" tabindex="-1"></a><span class="co"># Log-likelihood of RapidNJ tree: -7193.279</span></span>
<span id="cb524-146"><a href="estimating-phylogenetic-trees.html#cb524-146" aria-hidden="true" tabindex="-1"></a><span class="co"># --------------------------------------------------------------------</span></span>
<span id="cb524-147"><a href="estimating-phylogenetic-trees.html#cb524-147" aria-hidden="true" tabindex="-1"></a><span class="co"># |             INITIALIZING CANDIDATE TREE SET                      |</span></span>
<span id="cb524-148"><a href="estimating-phylogenetic-trees.html#cb524-148" aria-hidden="true" tabindex="-1"></a><span class="co"># --------------------------------------------------------------------</span></span>
<span id="cb524-149"><a href="estimating-phylogenetic-trees.html#cb524-149" aria-hidden="true" tabindex="-1"></a><span class="co"># Generating 98 parsimony trees... 0.182 second</span></span>
<span id="cb524-150"><a href="estimating-phylogenetic-trees.html#cb524-150" aria-hidden="true" tabindex="-1"></a><span class="co"># Computing log-likelihood of 98 initial trees ... 0.260 seconds</span></span>
<span id="cb524-151"><a href="estimating-phylogenetic-trees.html#cb524-151" aria-hidden="true" tabindex="-1"></a><span class="co"># Current best score: -7181.034</span></span>
<span id="cb524-152"><a href="estimating-phylogenetic-trees.html#cb524-152" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-153"><a href="estimating-phylogenetic-trees.html#cb524-153" aria-hidden="true" tabindex="-1"></a><span class="co"># Do NNI search on 20 best initial trees</span></span>
<span id="cb524-154"><a href="estimating-phylogenetic-trees.html#cb524-154" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate model parameters (epsilon = 0.100)</span></span>
<span id="cb524-155"><a href="estimating-phylogenetic-trees.html#cb524-155" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-156"><a href="estimating-phylogenetic-trees.html#cb524-156" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">WARNING</span><span class="co">: Estimated model parameters are at boundary that can cause numerical instability!</span></span>
<span id="cb524-157"><a href="estimating-phylogenetic-trees.html#cb524-157" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-158"><a href="estimating-phylogenetic-trees.html#cb524-158" aria-hidden="true" tabindex="-1"></a><span class="co"># BETTER TREE FOUND at iteration 1: -7181.009</span></span>
<span id="cb524-159"><a href="estimating-phylogenetic-trees.html#cb524-159" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate model parameters (epsilon = 0.100)</span></span>
<span id="cb524-160"><a href="estimating-phylogenetic-trees.html#cb524-160" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-161"><a href="estimating-phylogenetic-trees.html#cb524-161" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">WARNING</span><span class="co">: Estimated model parameters are at boundary that can cause numerical instability!</span></span>
<span id="cb524-162"><a href="estimating-phylogenetic-trees.html#cb524-162" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-163"><a href="estimating-phylogenetic-trees.html#cb524-163" aria-hidden="true" tabindex="-1"></a><span class="co"># BETTER TREE FOUND at iteration 3: -7179.089</span></span>
<span id="cb524-164"><a href="estimating-phylogenetic-trees.html#cb524-164" aria-hidden="true" tabindex="-1"></a><span class="co"># BETTER TREE FOUND at iteration 6: -7179.089</span></span>
<span id="cb524-165"><a href="estimating-phylogenetic-trees.html#cb524-165" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate model parameters (epsilon = 0.100)</span></span>
<span id="cb524-166"><a href="estimating-phylogenetic-trees.html#cb524-166" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-167"><a href="estimating-phylogenetic-trees.html#cb524-167" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">WARNING</span><span class="co">: Estimated model parameters are at boundary that can cause numerical instability!</span></span>
<span id="cb524-168"><a href="estimating-phylogenetic-trees.html#cb524-168" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-169"><a href="estimating-phylogenetic-trees.html#cb524-169" aria-hidden="true" tabindex="-1"></a><span class="co"># BETTER TREE FOUND at iteration 7: -7176.557</span></span>
<span id="cb524-170"><a href="estimating-phylogenetic-trees.html#cb524-170" aria-hidden="true" tabindex="-1"></a><span class="co"># Iteration 10 / LogL: -7178.385 / Time: 0h:0m:3s</span></span>
<span id="cb524-171"><a href="estimating-phylogenetic-trees.html#cb524-171" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate model parameters (epsilon = 0.100)</span></span>
<span id="cb524-172"><a href="estimating-phylogenetic-trees.html#cb524-172" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-173"><a href="estimating-phylogenetic-trees.html#cb524-173" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">WARNING</span><span class="co">: Estimated model parameters are at boundary that can cause numerical instability!</span></span>
<span id="cb524-174"><a href="estimating-phylogenetic-trees.html#cb524-174" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-175"><a href="estimating-phylogenetic-trees.html#cb524-175" aria-hidden="true" tabindex="-1"></a><span class="co"># BETTER TREE FOUND at iteration 11: -7176.259</span></span>
<span id="cb524-176"><a href="estimating-phylogenetic-trees.html#cb524-176" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate model parameters (epsilon = 0.100)</span></span>
<span id="cb524-177"><a href="estimating-phylogenetic-trees.html#cb524-177" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-178"><a href="estimating-phylogenetic-trees.html#cb524-178" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">WARNING</span><span class="co">: Estimated model parameters are at boundary that can cause numerical instability!</span></span>
<span id="cb524-179"><a href="estimating-phylogenetic-trees.html#cb524-179" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-180"><a href="estimating-phylogenetic-trees.html#cb524-180" aria-hidden="true" tabindex="-1"></a><span class="co"># BETTER TREE FOUND at iteration 13: -7175.959</span></span>
<span id="cb524-181"><a href="estimating-phylogenetic-trees.html#cb524-181" aria-hidden="true" tabindex="-1"></a><span class="co"># Iteration 20 / LogL: -7176.267 / Time: 0h:0m:3s</span></span>
<span id="cb524-182"><a href="estimating-phylogenetic-trees.html#cb524-182" aria-hidden="true" tabindex="-1"></a><span class="co"># Finish initializing candidate tree set (16)</span></span>
<span id="cb524-183"><a href="estimating-phylogenetic-trees.html#cb524-183" aria-hidden="true" tabindex="-1"></a><span class="co"># Current best tree score: -7175.959 / CPU time: 1.545</span></span>
<span id="cb524-184"><a href="estimating-phylogenetic-trees.html#cb524-184" aria-hidden="true" tabindex="-1"></a><span class="co"># Number of iterations: 20</span></span>
<span id="cb524-185"><a href="estimating-phylogenetic-trees.html#cb524-185" aria-hidden="true" tabindex="-1"></a><span class="co"># --------------------------------------------------------------------</span></span>
<span id="cb524-186"><a href="estimating-phylogenetic-trees.html#cb524-186" aria-hidden="true" tabindex="-1"></a><span class="co"># |               OPTIMIZING CANDIDATE TREE SET                      |</span></span>
<span id="cb524-187"><a href="estimating-phylogenetic-trees.html#cb524-187" aria-hidden="true" tabindex="-1"></a><span class="co"># --------------------------------------------------------------------</span></span>
<span id="cb524-188"><a href="estimating-phylogenetic-trees.html#cb524-188" aria-hidden="true" tabindex="-1"></a><span class="co"># UPDATE BEST LOG-LIKELIHOOD: -7175.952</span></span>
<span id="cb524-189"><a href="estimating-phylogenetic-trees.html#cb524-189" aria-hidden="true" tabindex="-1"></a><span class="co"># Iteration 30 / LogL: -7179.103 / Time: 0h:0m:4s (0h:0m:12s left)</span></span>
<span id="cb524-190"><a href="estimating-phylogenetic-trees.html#cb524-190" aria-hidden="true" tabindex="-1"></a><span class="co"># Iteration 40 / LogL: -7176.262 / Time: 0h:0m:5s (0h:0m:9s left)</span></span>
<span id="cb524-191"><a href="estimating-phylogenetic-trees.html#cb524-191" aria-hidden="true" tabindex="-1"></a><span class="co"># Iteration 50 / LogL: -7175.952 / Time: 0h:0m:5s (0h:0m:7s left)</span></span>
<span id="cb524-192"><a href="estimating-phylogenetic-trees.html#cb524-192" aria-hidden="true" tabindex="-1"></a><span class="co"># Iteration 60 / LogL: -7176.560 / Time: 0h:0m:6s (0h:0m:5s left)</span></span>
<span id="cb524-193"><a href="estimating-phylogenetic-trees.html#cb524-193" aria-hidden="true" tabindex="-1"></a><span class="co"># Iteration 70 / LogL: -7180.060 / Time: 0h:0m:6s (0h:0m:4s left)</span></span>
<span id="cb524-194"><a href="estimating-phylogenetic-trees.html#cb524-194" aria-hidden="true" tabindex="-1"></a><span class="co"># Iteration 80 / LogL: -7176.560 / Time: 0h:0m:7s (0h:0m:3s left)</span></span>
<span id="cb524-195"><a href="estimating-phylogenetic-trees.html#cb524-195" aria-hidden="true" tabindex="-1"></a><span class="co"># Iteration 90 / LogL: -7176.793 / Time: 0h:0m:7s (0h:0m:2s left)</span></span>
<span id="cb524-196"><a href="estimating-phylogenetic-trees.html#cb524-196" aria-hidden="true" tabindex="-1"></a><span class="co"># UPDATE BEST LOG-LIKELIHOOD: -7175.950</span></span>
<span id="cb524-197"><a href="estimating-phylogenetic-trees.html#cb524-197" aria-hidden="true" tabindex="-1"></a><span class="co"># UPDATE BEST LOG-LIKELIHOOD: -7175.950</span></span>
<span id="cb524-198"><a href="estimating-phylogenetic-trees.html#cb524-198" aria-hidden="true" tabindex="-1"></a><span class="co"># Iteration 100 / LogL: -7175.950 / Time: 0h:0m:8s (0h:0m:1s left)</span></span>
<span id="cb524-199"><a href="estimating-phylogenetic-trees.html#cb524-199" aria-hidden="true" tabindex="-1"></a><span class="co"># Iteration 110 / LogL: -7176.560 / Time: 0h:0m:8s (0h:0m:0s left)</span></span>
<span id="cb524-200"><a href="estimating-phylogenetic-trees.html#cb524-200" aria-hidden="true" tabindex="-1"></a><span class="co"># TREE SEARCH COMPLETED AFTER 114 ITERATIONS / Time: 0h:0m:9s</span></span>
<span id="cb524-201"><a href="estimating-phylogenetic-trees.html#cb524-201" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-202"><a href="estimating-phylogenetic-trees.html#cb524-202" aria-hidden="true" tabindex="-1"></a><span class="co"># --------------------------------------------------------------------</span></span>
<span id="cb524-203"><a href="estimating-phylogenetic-trees.html#cb524-203" aria-hidden="true" tabindex="-1"></a><span class="co"># |                    FINALIZING TREE SEARCH                        |</span></span>
<span id="cb524-204"><a href="estimating-phylogenetic-trees.html#cb524-204" aria-hidden="true" tabindex="-1"></a><span class="co"># --------------------------------------------------------------------</span></span>
<span id="cb524-205"><a href="estimating-phylogenetic-trees.html#cb524-205" aria-hidden="true" tabindex="-1"></a><span class="co"># Performs final model parameters optimization</span></span>
<span id="cb524-206"><a href="estimating-phylogenetic-trees.html#cb524-206" aria-hidden="true" tabindex="-1"></a><span class="co"># Estimate model parameters (epsilon = 0.010)</span></span>
<span id="cb524-207"><a href="estimating-phylogenetic-trees.html#cb524-207" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. Initial log-likelihood: -7175.950</span></span>
<span id="cb524-208"><a href="estimating-phylogenetic-trees.html#cb524-208" aria-hidden="true" tabindex="-1"></a><span class="co"># Optimal log-likelihood: -7175.948</span></span>
<span id="cb524-209"><a href="estimating-phylogenetic-trees.html#cb524-209" aria-hidden="true" tabindex="-1"></a><span class="co"># Rate parameters:  A-C: 11.77863  A-G: 100.00000  A-T: 11.77863  C-G: 1.00000  C-T: 100.00000  G-T: 1.00000</span></span>
<span id="cb524-210"><a href="estimating-phylogenetic-trees.html#cb524-210" aria-hidden="true" tabindex="-1"></a><span class="co"># Base frequencies:  A: 0.240  C: 0.266  G: 0.172  T: 0.321</span></span>
<span id="cb524-211"><a href="estimating-phylogenetic-trees.html#cb524-211" aria-hidden="true" tabindex="-1"></a><span class="co"># Proportion of invariable sites: 0.596</span></span>
<span id="cb524-212"><a href="estimating-phylogenetic-trees.html#cb524-212" aria-hidden="true" tabindex="-1"></a><span class="co"># Gamma shape alpha: 1.312</span></span>
<span id="cb524-213"><a href="estimating-phylogenetic-trees.html#cb524-213" aria-hidden="true" tabindex="-1"></a><span class="co"># Parameters optimization took 1 rounds (0.011 sec)</span></span>
<span id="cb524-214"><a href="estimating-phylogenetic-trees.html#cb524-214" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-215"><a href="estimating-phylogenetic-trees.html#cb524-215" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">WARNING</span><span class="co">: Estimated model parameters are at boundary that can cause numerical instability!</span></span>
<span id="cb524-216"><a href="estimating-phylogenetic-trees.html#cb524-216" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-217"><a href="estimating-phylogenetic-trees.html#cb524-217" aria-hidden="true" tabindex="-1"></a><span class="co"># BEST SCORE FOUND : -7175.948</span></span>
<span id="cb524-218"><a href="estimating-phylogenetic-trees.html#cb524-218" aria-hidden="true" tabindex="-1"></a><span class="co"># Total tree length: 2.452</span></span>
<span id="cb524-219"><a href="estimating-phylogenetic-trees.html#cb524-219" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-220"><a href="estimating-phylogenetic-trees.html#cb524-220" aria-hidden="true" tabindex="-1"></a><span class="co"># Total number of iterations: 114</span></span>
<span id="cb524-221"><a href="estimating-phylogenetic-trees.html#cb524-221" aria-hidden="true" tabindex="-1"></a><span class="co"># CPU time used for tree search: 6.683 sec (0h:0m:6s)</span></span>
<span id="cb524-222"><a href="estimating-phylogenetic-trees.html#cb524-222" aria-hidden="true" tabindex="-1"></a><span class="co"># Wall-clock time used for tree search: 6.730 sec (0h:0m:6s)</span></span>
<span id="cb524-223"><a href="estimating-phylogenetic-trees.html#cb524-223" aria-hidden="true" tabindex="-1"></a><span class="co"># Total CPU time used: 9.078 sec (0h:0m:9s)</span></span>
<span id="cb524-224"><a href="estimating-phylogenetic-trees.html#cb524-224" aria-hidden="true" tabindex="-1"></a><span class="co"># Total wall-clock time used: 9.182 sec (0h:0m:9s)</span></span>
<span id="cb524-225"><a href="estimating-phylogenetic-trees.html#cb524-225" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-226"><a href="estimating-phylogenetic-trees.html#cb524-226" aria-hidden="true" tabindex="-1"></a><span class="co"># Analysis results written to: </span></span>
<span id="cb524-227"><a href="estimating-phylogenetic-trees.html#cb524-227" aria-hidden="true" tabindex="-1"></a><span class="co">#   IQ-TREE report:                Ameerega_aligned_end.fasta.iqtree</span></span>
<span id="cb524-228"><a href="estimating-phylogenetic-trees.html#cb524-228" aria-hidden="true" tabindex="-1"></a><span class="co">#   Maximum-likelihood tree:       Ameerega_aligned_end.fasta.treefile</span></span>
<span id="cb524-229"><a href="estimating-phylogenetic-trees.html#cb524-229" aria-hidden="true" tabindex="-1"></a><span class="co">#   Likelihood distances:          Ameerega_aligned_end.fasta.mldist</span></span>
<span id="cb524-230"><a href="estimating-phylogenetic-trees.html#cb524-230" aria-hidden="true" tabindex="-1"></a><span class="co">#   Screen log file:               Ameerega_aligned_end.fasta.log</span></span>
<span id="cb524-231"><a href="estimating-phylogenetic-trees.html#cb524-231" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb524-232"><a href="estimating-phylogenetic-trees.html#cb524-232" aria-hidden="true" tabindex="-1"></a><span class="co"># Date and Time: Tue Feb 23 17:46:41 2021</span></span>
<span id="cb524-233"><a href="estimating-phylogenetic-trees.html#cb524-233" aria-hidden="true" tabindex="-1"></a><span class="co"># #***** DONE ***** </span></span>
<span id="cb524-234"><a href="estimating-phylogenetic-trees.html#cb524-234" aria-hidden="true" tabindex="-1"></a><span class="co"># #</span></span>
<span id="cb524-235"><a href="estimating-phylogenetic-trees.html#cb524-235" aria-hidden="true" tabindex="-1"></a><span class="co"># # *************         iqtree outfiles written to        ************* </span></span>
<span id="cb524-236"><a href="estimating-phylogenetic-trees.html#cb524-236" aria-hidden="true" tabindex="-1"></a><span class="co"># #[1] &quot;/Users/santosj/Desktop/Teach_R/my_working_directory/my_phylogeny_1&quot;</span></span></code></pre></div>
<p>The following files will be now present in your working directory.</p>
<p><img src="figures/session_12_figs/12_output_directory.png" width="90%" /></p>
<p><strong>e)</strong> You can visualize the resulting phylogenetic tree as follows.</p>
<div class="sourceCode" id="cb525"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb525-1"><a href="estimating-phylogenetic-trees.html#cb525-1" aria-hidden="true" tabindex="-1"></a><span class="do">## content of output object.</span></span>
<span id="cb525-2"><a href="estimating-phylogenetic-trees.html#cb525-2" aria-hidden="true" tabindex="-1"></a>COI_simple_phylogeny</span>
<span id="cb525-3"><a href="estimating-phylogenetic-trees.html#cb525-3" aria-hidden="true" tabindex="-1"></a><span class="co">#[[1]]</span></span>
<span id="cb525-4"><a href="estimating-phylogenetic-trees.html#cb525-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb525-5"><a href="estimating-phylogenetic-trees.html#cb525-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Phylogenetic tree with 24 tips and 22 internal nodes.</span></span>
<span id="cb525-6"><a href="estimating-phylogenetic-trees.html#cb525-6" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb525-7"><a href="estimating-phylogenetic-trees.html#cb525-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Tip labels:</span></span>
<span id="cb525-8"><a href="estimating-phylogenetic-trees.html#cb525-8" aria-hidden="true" tabindex="-1"></a><span class="co">#  MW042032.1, MW042031.1, DQ502932.1, DQ502902.1, DQ502922.1, DQ502929.1, ...</span></span>
<span id="cb525-9"><a href="estimating-phylogenetic-trees.html#cb525-9" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb525-10"><a href="estimating-phylogenetic-trees.html#cb525-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Unrooted; includes branch lengths.</span></span>
<span id="cb525-11"><a href="estimating-phylogenetic-trees.html#cb525-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb525-12"><a href="estimating-phylogenetic-trees.html#cb525-12" aria-hidden="true" tabindex="-1"></a>COI_simple_phylogeny[[<span class="dv">1</span>]]</span>
<span id="cb525-13"><a href="estimating-phylogenetic-trees.html#cb525-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Phylogenetic tree with 24 tips and 22 internal nodes.</span></span>
<span id="cb525-14"><a href="estimating-phylogenetic-trees.html#cb525-14" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb525-15"><a href="estimating-phylogenetic-trees.html#cb525-15" aria-hidden="true" tabindex="-1"></a><span class="co">#Tip labels:</span></span>
<span id="cb525-16"><a href="estimating-phylogenetic-trees.html#cb525-16" aria-hidden="true" tabindex="-1"></a><span class="co">#  MW042032.1, MW042031.1, DQ502932.1, DQ502902.1, DQ502922.1, DQ502929.1, ...</span></span>
<span id="cb525-17"><a href="estimating-phylogenetic-trees.html#cb525-17" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb525-18"><a href="estimating-phylogenetic-trees.html#cb525-18" aria-hidden="true" tabindex="-1"></a><span class="co">#Unrooted; includes branch lengths.</span></span>
<span id="cb525-19"><a href="estimating-phylogenetic-trees.html#cb525-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb525-20"><a href="estimating-phylogenetic-trees.html#cb525-20" aria-hidden="true" tabindex="-1"></a><span class="do">## you can plot this tree using ape</span></span>
<span id="cb525-21"><a href="estimating-phylogenetic-trees.html#cb525-21" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(COI_simple_phylogeny[[<span class="dv">1</span>]])</span></code></pre></div>
<p><img src="figures/session_12_figs/13_output_phylogeny.png" width="90%" />
The actual phylogenetic tree is in the file <code>Ameerega_aligned_end.fasta.treefile</code> in our working folder.</p>
</div>
<div id="one-marker-and-with-partitions" class="section level2" number="12.7">
<h2><span class="header-section-number">12.7</span> One marker and with partitions</h2>
<p>You provide a partition for your sequences based on information about it. For example, you can divide a protein coding gene in three sections based on the codon position. As it has been demonstrated, the codon code is redundant and variations in 3rd positions usually result in the same amino acid residue after translation, for more information see <a href="https://en.wikipedia.org/wiki/DNA_and_RNA_codon_tables" target="_blank"><strong>here</strong></a>. In our example, we can split our split our COI matrix by codon position.</p>
<p><strong>5)</strong> We will use the COI aligned sequences in the file <code>Ameerega_aligned_end.fasta</code>. Notice the argument <code>iqtree_partition_user = list(COI = c("DNA",1,1539,3))</code>. We define it a <code>DNA</code> segment, starting on position <code>1</code>, ending on position <code>1539</code> (this number can be find in <strong>Session 11.7</strong> or just by counting the number of bases on the aligned matrix) and <code>3</code> that means to divide it by 3 for 1st, 2nd, and 3rd codon positions.</p>
<div class="sourceCode" id="cb526"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb526-1"><a href="estimating-phylogenetic-trees.html#cb526-1" aria-hidden="true" tabindex="-1"></a><span class="do">## set up a new working directory</span></span>
<span id="cb526-2"><a href="estimating-phylogenetic-trees.html#cb526-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/my_working_directory/my_phylogeny_2&quot;</span>)</span>
<span id="cb526-3"><a href="estimating-phylogenetic-trees.html#cb526-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb526-4"><a href="estimating-phylogenetic-trees.html#cb526-4" aria-hidden="true" tabindex="-1"></a><span class="do">## copy as paste again the sequence data</span></span>
<span id="cb526-5"><a href="estimating-phylogenetic-trees.html#cb526-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb526-6"><a href="estimating-phylogenetic-trees.html#cb526-6" aria-hidden="true" tabindex="-1"></a><span class="do">## update the </span></span>
<span id="cb526-7"><a href="estimating-phylogenetic-trees.html#cb526-7" aria-hidden="true" tabindex="-1"></a>my_path_to_aligned_sequences2 <span class="ot">&lt;-</span> <span class="st">&quot;~/Desktop/Teach_R/my_working_directory/my_phylogeny_2/Ameerega_aligned_end.fasta&quot;</span></span>
<span id="cb526-8"><a href="estimating-phylogenetic-trees.html#cb526-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb526-9"><a href="estimating-phylogenetic-trees.html#cb526-9" aria-hidden="true" tabindex="-1"></a><span class="do">## iqtree2 run with defined codon positions</span></span>
<span id="cb526-10"><a href="estimating-phylogenetic-trees.html#cb526-10" aria-hidden="true" tabindex="-1"></a>COI_with_partitions_phylogeny <span class="ot">&lt;-</span> <span class="fu">iqtree_runner_R</span> (<span class="at">input_alignment_file_user =</span> my_path_to_aligned_sequences2,</span>
<span id="cb526-11"><a href="estimating-phylogenetic-trees.html#cb526-11" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_bin_location_user =</span> my_path_to_iqtree2,</span>
<span id="cb526-12"><a href="estimating-phylogenetic-trees.html#cb526-12" aria-hidden="true" tabindex="-1"></a>                           <span class="at">run_iqtree_call_user =</span> <span class="st">&quot;serial&quot;</span>,</span>
<span id="cb526-13"><a href="estimating-phylogenetic-trees.html#cb526-13" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_input_model_m_user =</span> <span class="st">&quot;MFP+MERGE&quot;</span>,</span>
<span id="cb526-14"><a href="estimating-phylogenetic-trees.html#cb526-14" aria-hidden="true" tabindex="-1"></a>             <span class="at">iqtree_partition_user =</span> <span class="fu">list</span>(<span class="at">COI =</span> <span class="fu">c</span>(<span class="st">&quot;DNA&quot;</span>,<span class="dv">1</span>,<span class="dv">1539</span>,<span class="dv">3</span>)),</span>
<span id="cb526-15"><a href="estimating-phylogenetic-trees.html#cb526-15" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iqtree_partition_if_by_position_user =</span> <span class="cn">TRUE</span>)</span>
<span id="cb526-16"><a href="estimating-phylogenetic-trees.html#cb526-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb526-17"><a href="estimating-phylogenetic-trees.html#cb526-17" aria-hidden="true" tabindex="-1"></a><span class="do">## see and plot the new tree</span></span>
<span id="cb526-18"><a href="estimating-phylogenetic-trees.html#cb526-18" aria-hidden="true" tabindex="-1"></a>COI_with_partitions_phylogeny[[<span class="dv">1</span>]]</span>
<span id="cb526-19"><a href="estimating-phylogenetic-trees.html#cb526-19" aria-hidden="true" tabindex="-1"></a><span class="co">#Phylogenetic tree with 24 tips and 22 internal nodes.</span></span>
<span id="cb526-20"><a href="estimating-phylogenetic-trees.html#cb526-20" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb526-21"><a href="estimating-phylogenetic-trees.html#cb526-21" aria-hidden="true" tabindex="-1"></a><span class="co">#Tip labels:</span></span>
<span id="cb526-22"><a href="estimating-phylogenetic-trees.html#cb526-22" aria-hidden="true" tabindex="-1"></a><span class="co">#  MW042032.1, MW042031.1, DQ502932.1, DQ502902.1, DQ502922.1, DQ502929.1, ...</span></span>
<span id="cb526-23"><a href="estimating-phylogenetic-trees.html#cb526-23" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb526-24"><a href="estimating-phylogenetic-trees.html#cb526-24" aria-hidden="true" tabindex="-1"></a><span class="co">#Unrooted; includes branch lengths.</span></span>
<span id="cb526-25"><a href="estimating-phylogenetic-trees.html#cb526-25" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(COI_with_partitions_phylogeny[[<span class="dv">1</span>]])</span></code></pre></div>
<p><img src="figures/session_12_figs/14_output_phylogeny2.png" width="90%" /></p>
<p>The actual phylogenetic tree is in the file <code>Ameerega_aligned_end_fasta_partition.txt.treefile</code> in our working folder.</p>
</div>
<div id="bootstrap-support" class="section level2" number="12.8">
<h2><span class="header-section-number">12.8</span> Bootstrap support</h2>
<p><strong>6)</strong> In most cases, you also want to know how reliable is your phylogenetic reconstruction. This is determined by the statistical support of the nodes (the internal junctions of lines). These are usually evaluated over 100% based on the type of the approach used by <strong>iqtree2</strong>, which means that such phylogenetic relationships are very reliable. If values are less than 75%, your confidence drops significantly and usually means conflicting hypotheses about the evolutionary relationships for that node. We will use the COI aligned sequences in the file <code>Ameerega_aligned_end.fasta</code>. Notice the arguments <code>iqtree_UFBoot_B_user = 1000</code> and <code>iqtree_UFBoot_bnni_user = TRUE</code>. This will ask for 1000 non-parametric bootstraps with corrections, for details see <a href="http://www.iqtree.org/doc/Tutorial#assessing-branch-supports-with--standard-nonparametric-bootstrap" target="_blank"><strong>here</strong></a>.</p>
<div class="sourceCode" id="cb527"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb527-1"><a href="estimating-phylogenetic-trees.html#cb527-1" aria-hidden="true" tabindex="-1"></a><span class="do">## set up a new working directory</span></span>
<span id="cb527-2"><a href="estimating-phylogenetic-trees.html#cb527-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/my_working_directory/my_phylogeny_3&quot;</span>)</span>
<span id="cb527-3"><a href="estimating-phylogenetic-trees.html#cb527-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb527-4"><a href="estimating-phylogenetic-trees.html#cb527-4" aria-hidden="true" tabindex="-1"></a><span class="do">## copy as paste again the sequence data</span></span>
<span id="cb527-5"><a href="estimating-phylogenetic-trees.html#cb527-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb527-6"><a href="estimating-phylogenetic-trees.html#cb527-6" aria-hidden="true" tabindex="-1"></a><span class="do">## update the </span></span>
<span id="cb527-7"><a href="estimating-phylogenetic-trees.html#cb527-7" aria-hidden="true" tabindex="-1"></a>my_path_to_aligned_sequences3 <span class="ot">&lt;-</span> <span class="st">&quot;~/Desktop/Teach_R/my_working_directory/my_phylogeny_3/Ameerega_aligned_end.fasta&quot;</span></span>
<span id="cb527-8"><a href="estimating-phylogenetic-trees.html#cb527-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb527-9"><a href="estimating-phylogenetic-trees.html#cb527-9" aria-hidden="true" tabindex="-1"></a><span class="do">## iqtree2 run with defined codon positions and get 1000 bootstraps</span></span>
<span id="cb527-10"><a href="estimating-phylogenetic-trees.html#cb527-10" aria-hidden="true" tabindex="-1"></a>COI_with_partitions_phylogeny_boot <span class="ot">&lt;-</span> <span class="fu">iqtree_runner_R</span> (<span class="at">input_alignment_file_user =</span> my_path_to_aligned_sequences3,</span>
<span id="cb527-11"><a href="estimating-phylogenetic-trees.html#cb527-11" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_bin_location_user =</span> my_path_to_iqtree2,</span>
<span id="cb527-12"><a href="estimating-phylogenetic-trees.html#cb527-12" aria-hidden="true" tabindex="-1"></a>                           <span class="at">run_iqtree_call_user =</span> <span class="st">&quot;serial&quot;</span>,</span>
<span id="cb527-13"><a href="estimating-phylogenetic-trees.html#cb527-13" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_input_model_m_user =</span> <span class="st">&quot;GTR+F+I+G4&quot;</span>,</span>
<span id="cb527-14"><a href="estimating-phylogenetic-trees.html#cb527-14" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_UFBoot_B_user =</span> <span class="dv">1000</span>,</span>
<span id="cb527-15"><a href="estimating-phylogenetic-trees.html#cb527-15" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_UFBoot_bnni_user =</span> <span class="cn">TRUE</span>,</span>
<span id="cb527-16"><a href="estimating-phylogenetic-trees.html#cb527-16" aria-hidden="true" tabindex="-1"></a>             <span class="at">iqtree_partition_user =</span> <span class="fu">list</span>(<span class="at">COI =</span> <span class="fu">c</span>(<span class="st">&quot;DNA&quot;</span>,<span class="dv">1</span>,<span class="dv">1539</span>,<span class="dv">3</span>)),</span>
<span id="cb527-17"><a href="estimating-phylogenetic-trees.html#cb527-17" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iqtree_partition_if_by_position_user =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>The actual collection of bootstrap trees is in the file <code>Ameerega_aligned_end_fasta_partition.txt.ufboot</code> and its consensus is in the file <code>Ameerega_aligned_end_fasta_partition.txt.treefile</code> in our working folder.</p>
<p><strong>7)</strong> I could not find an easy way to visualize these threes other than use the other program to visualize phylogenetic trees in general. You need to install <a href="http://tree.bio.ed.ac.uk/software/figtree/" target="_blank"><strong>figtree</strong></a>. To visualize your consensus tree after installing <strong>figtree</strong> you need open this program.</p>
<p><img src="figures/session_12_figs/15_open_figtree.png" width="90%" /></p>
<p>The <code>File&gt;Open...</code> find the bootstrap summary file <code>Ameerega_aligned_end_fasta_partition.txt.treefile</code> and allow to see labels.</p>
<p><img src="figures/session_12_figs/16_open_figtree_labels.png" width="90%" /></p>
<p>You will see your tree, but without the nodal support.</p>
<p><img src="figures/session_12_figs/17_open_figtree_no_support.png" width="90%" /></p>
<p>You will clikc on <code>Nodel Labels</code> and then on <code>label</code>.</p>
<p><img src="figures/session_12_figs/18_open_figtree_no_support2.png" width="90%" /></p>
<p>This will show you the bootstraps for your tree.</p>
<p><img src="figures/session_12_figs/19_open_figtree_with_support.png" width="90%" /></p>
</div>
<div id="many-markers-and-with-partitions" class="section level2" number="12.9">
<h2><span class="header-section-number">12.9</span> Many markers and with partitions</h2>
<p><strong>8)</strong> A phylogeny derived from two or more markers require some modifications to indicate the start and end of each marker.</p>
<p>We start by defining the output directory.</p>
<div class="sourceCode" id="cb528"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb528-1"><a href="estimating-phylogenetic-trees.html#cb528-1" aria-hidden="true" tabindex="-1"></a><span class="do">## set up a new working directory by creating one; this is exclusive to your OWN COMPUTER change it accordingly</span></span>
<span id="cb528-2"><a href="estimating-phylogenetic-trees.html#cb528-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/my_working_directory/my_phylogeny_4&quot;</span>)</span></code></pre></div>
<p>We find the file with aligned sequences: <code>my_collected_COI_CYTB_String_path_fasta</code> and copy that file to your working directory (e.g., <code>my_phylogeny_4</code>)</p>
<div class="sourceCode" id="cb529"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb529-1"><a href="estimating-phylogenetic-trees.html#cb529-1" aria-hidden="true" tabindex="-1"></a><span class="do">## copy the file my_collected_COI_CYTB_String_path_fasta to your working directory my_phylogeny_4</span></span>
<span id="cb529-2"><a href="estimating-phylogenetic-trees.html#cb529-2" aria-hidden="true" tabindex="-1"></a><span class="do">## this is the path to that file and exclusive to your OWN COMPUTER change it accordingly</span></span>
<span id="cb529-3"><a href="estimating-phylogenetic-trees.html#cb529-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb529-4"><a href="estimating-phylogenetic-trees.html#cb529-4" aria-hidden="true" tabindex="-1"></a>my_path_to_aligned_COI_CYTB_sequences <span class="ot">&lt;-</span> <span class="st">&quot;~/Desktop/Teach_R/my_working_directory/my_phylogeny_4/my_collected_COI_CYTB_String_path_fasta.txt&quot;</span></span></code></pre></div>
<p>We run <code>iqtree2</code> and notice the <code>iqtree_partition_user</code> argument that defines the <code>COI</code> and <code>CYTB</code> markers. We define both as <code>DNA</code> segments. For COI, we define the starting on position of COI at <code>1</code> and the ending on position at <code>1539</code> (this number can be find in <strong>Session 11.7</strong> or just by counting the number of bases on the aligned matrix). For CYTB, we check first the that the length of this maker is 944 (see <strong>Session 11.9</strong>) and we concatenated this maker after COI, then we add 1539 + 944 = 2483. However, the CYTB gene does not start on base 1539 but on 1540, so we add 1. Therefore, we define the starting on position of CYTB at <code>1540</code> and the ending on position at <code>2483</code>. Finally, the <code>3</code> on the <code>iqtree_partition_user</code> argument means to divide it by 3 for 1st, 2nd, and 3rd codon positions.</p>
<div class="sourceCode" id="cb530"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb530-1"><a href="estimating-phylogenetic-trees.html#cb530-1" aria-hidden="true" tabindex="-1"></a><span class="do">## this is exclusive to your OWN COMPUTER and we just need the path to the directory that contains iqtree2</span></span>
<span id="cb530-2"><a href="estimating-phylogenetic-trees.html#cb530-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Note: the diffecent from path above and the one below.</span></span>
<span id="cb530-3"><a href="estimating-phylogenetic-trees.html#cb530-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb530-4"><a href="estimating-phylogenetic-trees.html#cb530-4" aria-hidden="true" tabindex="-1"></a>my_path_to_iqtree2 <span class="ot">&lt;-</span> <span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/my_programs&quot;</span></span>
<span id="cb530-5"><a href="estimating-phylogenetic-trees.html#cb530-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb530-6"><a href="estimating-phylogenetic-trees.html#cb530-6" aria-hidden="true" tabindex="-1"></a><span class="do">## iqtree2 run with defined codon positions</span></span>
<span id="cb530-7"><a href="estimating-phylogenetic-trees.html#cb530-7" aria-hidden="true" tabindex="-1"></a>COI_CYTB_with_partitions_phylogeny <span class="ot">&lt;-</span> <span class="fu">iqtree_runner_R</span> (<span class="at">input_alignment_file_user =</span> my_path_to_aligned_COI_CYTB_sequences,</span>
<span id="cb530-8"><a href="estimating-phylogenetic-trees.html#cb530-8" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_bin_location_user =</span> my_path_to_iqtree2,</span>
<span id="cb530-9"><a href="estimating-phylogenetic-trees.html#cb530-9" aria-hidden="true" tabindex="-1"></a>                           <span class="at">run_iqtree_call_user =</span> <span class="st">&quot;serial&quot;</span>,</span>
<span id="cb530-10"><a href="estimating-phylogenetic-trees.html#cb530-10" aria-hidden="true" tabindex="-1"></a>                           <span class="at">iqtree_input_model_m_user =</span> <span class="st">&quot;MFP+MERGE&quot;</span>,</span>
<span id="cb530-11"><a href="estimating-phylogenetic-trees.html#cb530-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">iqtree_partition_user =</span> <span class="fu">list</span>(<span class="at">COI =</span> <span class="fu">c</span>(<span class="st">&quot;DNA&quot;</span>,<span class="dv">1</span>,<span class="dv">1539</span>,<span class="dv">3</span>),</span>
<span id="cb530-12"><a href="estimating-phylogenetic-trees.html#cb530-12" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">CYTB =</span> <span class="fu">c</span>(<span class="st">&quot;DNA&quot;</span>,<span class="dv">1540</span>,<span class="dv">2483</span>,<span class="dv">3</span>)),</span>
<span id="cb530-13"><a href="estimating-phylogenetic-trees.html#cb530-13" aria-hidden="true" tabindex="-1"></a>                       <span class="at">iqtree_partition_if_by_position_user =</span> <span class="cn">TRUE</span>)</span>
<span id="cb530-14"><a href="estimating-phylogenetic-trees.html#cb530-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb530-15"><a href="estimating-phylogenetic-trees.html#cb530-15" aria-hidden="true" tabindex="-1"></a><span class="do">## see and plot the new tree</span></span>
<span id="cb530-16"><a href="estimating-phylogenetic-trees.html#cb530-16" aria-hidden="true" tabindex="-1"></a>COI_CYTB_with_partitions_phylogeny[[<span class="dv">1</span>]]</span>
<span id="cb530-17"><a href="estimating-phylogenetic-trees.html#cb530-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb530-18"><a href="estimating-phylogenetic-trees.html#cb530-18" aria-hidden="true" tabindex="-1"></a><span class="co">#Phylogenetic tree with 12 tips and 10 internal nodes.</span></span>
<span id="cb530-19"><a href="estimating-phylogenetic-trees.html#cb530-19" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb530-20"><a href="estimating-phylogenetic-trees.html#cb530-20" aria-hidden="true" tabindex="-1"></a><span class="co">#Tip labels:</span></span>
<span id="cb530-21"><a href="estimating-phylogenetic-trees.html#cb530-21" aria-hidden="true" tabindex="-1"></a><span class="co">#  Ameerega_bilinguis, Ameerega_braccata, Ameerega_hahneli, Ameerega_trivittata, Ameerega_silverstonei, Ameerega_parvula, ...</span></span>
<span id="cb530-22"><a href="estimating-phylogenetic-trees.html#cb530-22" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb530-23"><a href="estimating-phylogenetic-trees.html#cb530-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Unrooted; includes branch lengths.</span></span>
<span id="cb530-24"><a href="estimating-phylogenetic-trees.html#cb530-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb530-25"><a href="estimating-phylogenetic-trees.html#cb530-25" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(COI_CYTB_with_partitions_phylogeny[[<span class="dv">1</span>]])</span></code></pre></div>
<p><img src="figures/session_12_figs/20_output_two_markers.png" width="90%" /></p>
<p>The actual phylogenetic tree is in the file <code>my_collected_COI_CYTB_String_path_fasta_txt_partition.txt.treefile</code> in our working folder.</p>
</div>
<div id="other-sequence-phylogenetic-tree-estimation-software" class="section level2" number="12.10">
<h2><span class="header-section-number">12.10</span> Other sequence phylogenetic tree estimation software</h2>
<p><strong>9)</strong> Other widely used programs that can do phylogenetic estimations are as follows. You need to follow their vignettes and installation instructions.</p>
<p><strong>a)</strong> <a href="https://github.com/stamatak/standard-RAxML" target="_blank"><strong>RAxML</strong></a>. A manual and tutorial is provided in <a href="https://cme.h-its.org/exelixis/resource/download/NewManual.pdf" target="_blank"><strong>here</strong></a>. General info, vignettes and a GUI interface are also provided <a href="https://cme.h-its.org/exelixis/web/software/raxml/" target="_blank"><strong>here</strong></a>. An online version is provided also <a href="https://raxml-ng.vital-it.ch/#/" target="_blank"><strong>here</strong></a>.</p>
<p><strong>b)</strong> <a href="http://www.atgc-montpellier.fr/phyml/" target="_blank"><strong>PhyML 3.0</strong></a>. A GitHub repository associated is <a href="https://github.com/stephaneguindon/phyml" target="_blank"><strong>here</strong></a>. This software manual is <a href="https://github.com/stephaneguindon/phyml/blob/master/doc/phyml-manual.pdf" target="_blank"><strong>here</strong></a>.</p>
</div>
<div id="drawing-phylogenetic-trees-with-ggtree" class="section level2" number="12.11">
<h2><span class="header-section-number">12.11</span> Drawing phylogenetic trees with <code>ggtree</code></h2>
<p>In most instances you want to draw a phylogenetic tree for your publications, presentations or to visualize taxon/sequence relationships. A phylogenetic tree is graph or diagram that illustrates such relationships usually among organisms (or sequences of nucleotides/proteins that represent them). Each of these diagrams represent hypotheses of relationship derived from the data used to construct them (i.e., it might change if more data are added). Using R, you can plot these phylogenetic trees and noticing the pattern of branching, grouping and the relationship of tips (sources of data) from the series of common ancestors in the nodes of these diagrams. You can read this friendly summary by <a href="https://evolution-outreach.biomedcentral.com/articles/10.1007/s12052-010-0273-6" target="_blank"><strong>McLennan 2010</strong></a>, which provides an easy guide of how to read a phylogenetic tree. You can plot trees using diverse R packages such as <a href="https://cran.r-project.org/web/packages/ape/index.html" target="_blank"><strong>ape</strong></a>, <a href="https://cran.r-project.org/web/packages/phylotools/index.html" target="_blank"><strong>phylotools</strong></a>, and <a href="https://bioconductor.org/packages/release/bioc/html/ggtree.html" target="_blank"><strong>ggtree</strong></a>. We will use <a href="https://bioconductor.org/packages/release/bioc/html/ggtree.html" target="_blank"><strong>ggtree</strong></a> as this one uses many of the strengths of <strong>ggplot</strong> for most of color and graph manipulations. Likewise, it has a extensive <a href="https://yulab-smu.top/treedata-book/" target="_blank"><strong>vignette</strong></a> that make tree plotting easy.</p>
<p><strong>10)</strong> You will need to install <strong>ggtree</strong> and the tree parsing package <a href="http://bioconductor.org/packages/release/bioc/html/treeio.html" target="_blank"><strong>treeio</strong></a> from <strong>Bioconductor</strong> and have a tree file to be plotted. The package <strong>treeio</strong> provides an extensive list of formats to <a href="https://yulab-smu.top/treedata-book/chapter1.html" target="_blank"><strong>import</strong></a>. We will use the functions <code>read.newick</code> or <code>read.tree</code> to import a newick tree (the format that we get from iqtree2). For trees with support values for iqtree we will use <code>read.iqtree</code> to allow parsing IQ-Tree newick string, with ability to parse SH-aLRT and UFBoot support values.</p>
<div class="sourceCode" id="cb531"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb531-1"><a href="estimating-phylogenetic-trees.html#cb531-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we will install ggtree</span></span>
<span id="cb531-2"><a href="estimating-phylogenetic-trees.html#cb531-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb531-3"><a href="estimating-phylogenetic-trees.html#cb531-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb531-4"><a href="estimating-phylogenetic-trees.html#cb531-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb531-5"><a href="estimating-phylogenetic-trees.html#cb531-5" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;ggtree&quot;</span>)</span>
<span id="cb531-6"><a href="estimating-phylogenetic-trees.html#cb531-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggtree)</span>
<span id="cb531-7"><a href="estimating-phylogenetic-trees.html#cb531-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb531-8"><a href="estimating-phylogenetic-trees.html#cb531-8" aria-hidden="true" tabindex="-1"></a><span class="do">## we also need treeio</span></span>
<span id="cb531-9"><a href="estimating-phylogenetic-trees.html#cb531-9" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb531-10"><a href="estimating-phylogenetic-trees.html#cb531-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb531-11"><a href="estimating-phylogenetic-trees.html#cb531-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb531-12"><a href="estimating-phylogenetic-trees.html#cb531-12" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;treeio&quot;</span>)</span>
<span id="cb531-13"><a href="estimating-phylogenetic-trees.html#cb531-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(treeio)</span>
<span id="cb531-14"><a href="estimating-phylogenetic-trees.html#cb531-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb531-15"><a href="estimating-phylogenetic-trees.html#cb531-15" aria-hidden="true" tabindex="-1"></a><span class="do">## we can import a regular iqtree file </span></span>
<span id="cb531-16"><a href="estimating-phylogenetic-trees.html#cb531-16" aria-hidden="true" tabindex="-1"></a><span class="do">## Notice: the extionsion is *.treefile</span></span>
<span id="cb531-17"><a href="estimating-phylogenetic-trees.html#cb531-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb531-18"><a href="estimating-phylogenetic-trees.html#cb531-18" aria-hidden="true" tabindex="-1"></a>my_tree_file_path <span class="ot">&lt;-</span> <span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/my_working_directory/my_phylogeny_1/Ameerega_aligned_end.fasta.treefile&quot;</span></span>
<span id="cb531-19"><a href="estimating-phylogenetic-trees.html#cb531-19" aria-hidden="true" tabindex="-1"></a>my_tree <span class="ot">&lt;-</span> <span class="fu">read.newick</span>(my_tree_file_path)</span>
<span id="cb531-20"><a href="estimating-phylogenetic-trees.html#cb531-20" aria-hidden="true" tabindex="-1"></a>my_tree</span>
<span id="cb531-21"><a href="estimating-phylogenetic-trees.html#cb531-21" aria-hidden="true" tabindex="-1"></a><span class="co">#Phylogenetic tree with 24 tips and 22 internal nodes.</span></span>
<span id="cb531-22"><a href="estimating-phylogenetic-trees.html#cb531-22" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb531-23"><a href="estimating-phylogenetic-trees.html#cb531-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Tip labels:</span></span>
<span id="cb531-24"><a href="estimating-phylogenetic-trees.html#cb531-24" aria-hidden="true" tabindex="-1"></a><span class="co">#  MW042032.1, MW042031.1, DQ502932.1, DQ502902.1, DQ502922.1, DQ502929.1, ...</span></span>
<span id="cb531-25"><a href="estimating-phylogenetic-trees.html#cb531-25" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb531-26"><a href="estimating-phylogenetic-trees.html#cb531-26" aria-hidden="true" tabindex="-1"></a><span class="co">#Unrooted; includes branch lengths.</span></span>
<span id="cb531-27"><a href="estimating-phylogenetic-trees.html#cb531-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb531-28"><a href="estimating-phylogenetic-trees.html#cb531-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb531-29"><a href="estimating-phylogenetic-trees.html#cb531-29" aria-hidden="true" tabindex="-1"></a><span class="do">## we can import the bootstrap summary of iqtree file </span></span>
<span id="cb531-30"><a href="estimating-phylogenetic-trees.html#cb531-30" aria-hidden="true" tabindex="-1"></a><span class="do">## Notice: the extionsion is *txt.treefile</span></span>
<span id="cb531-31"><a href="estimating-phylogenetic-trees.html#cb531-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb531-32"><a href="estimating-phylogenetic-trees.html#cb531-32" aria-hidden="true" tabindex="-1"></a>my_tree_boot_file_path <span class="ot">&lt;-</span> <span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/my_working_directory/my_phylogeny_3/Ameerega_aligned_end_fasta_partition.txt.treefile&quot;</span></span>
<span id="cb531-33"><a href="estimating-phylogenetic-trees.html#cb531-33" aria-hidden="true" tabindex="-1"></a>my_tree_boot <span class="ot">&lt;-</span> <span class="fu">read.iqtree</span>(my_tree_boot_file_path)</span>
<span id="cb531-34"><a href="estimating-phylogenetic-trees.html#cb531-34" aria-hidden="true" tabindex="-1"></a>my_tree_boot</span>
<span id="cb531-35"><a href="estimating-phylogenetic-trees.html#cb531-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;treedata&#39; S4 object that stored information of</span></span>
<span id="cb531-36"><a href="estimating-phylogenetic-trees.html#cb531-36" aria-hidden="true" tabindex="-1"></a><span class="co">#  &#39;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/my_working_directory/my_phylogeny_3/Ameerega_aligned_end_fasta_partition.txt.treefile&#39;.</span></span>
<span id="cb531-37"><a href="estimating-phylogenetic-trees.html#cb531-37" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb531-38"><a href="estimating-phylogenetic-trees.html#cb531-38" aria-hidden="true" tabindex="-1"></a><span class="co">#...@ phylo: </span></span>
<span id="cb531-39"><a href="estimating-phylogenetic-trees.html#cb531-39" aria-hidden="true" tabindex="-1"></a><span class="co">#Phylogenetic tree with 24 tips and 22 internal nodes.</span></span>
<span id="cb531-40"><a href="estimating-phylogenetic-trees.html#cb531-40" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb531-41"><a href="estimating-phylogenetic-trees.html#cb531-41" aria-hidden="true" tabindex="-1"></a><span class="co">#Tip labels:</span></span>
<span id="cb531-42"><a href="estimating-phylogenetic-trees.html#cb531-42" aria-hidden="true" tabindex="-1"></a><span class="co">#  MW042032.1, MW042031.1, DQ502932.1, DQ502902.1, DQ502922.1, DQ502929.1, ...</span></span>
<span id="cb531-43"><a href="estimating-phylogenetic-trees.html#cb531-43" aria-hidden="true" tabindex="-1"></a><span class="co">#Node labels:</span></span>
<span id="cb531-44"><a href="estimating-phylogenetic-trees.html#cb531-44" aria-hidden="true" tabindex="-1"></a><span class="co">#  , 69, 33, 48, 29, 94, ...</span></span>
<span id="cb531-45"><a href="estimating-phylogenetic-trees.html#cb531-45" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb531-46"><a href="estimating-phylogenetic-trees.html#cb531-46" aria-hidden="true" tabindex="-1"></a><span class="co">#Unrooted; includes branch lengths.</span></span>
<span id="cb531-47"><a href="estimating-phylogenetic-trees.html#cb531-47" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb531-48"><a href="estimating-phylogenetic-trees.html#cb531-48" aria-hidden="true" tabindex="-1"></a><span class="co">#with the following features available:</span></span>
<span id="cb531-49"><a href="estimating-phylogenetic-trees.html#cb531-49" aria-hidden="true" tabindex="-1"></a><span class="co">#  &#39;SH_aLRT&#39;,  &#39;UFboot&#39;.</span></span></code></pre></div>
<p><strong>11)</strong> Basic tree plots in <strong>ggtree</strong> are straight forward as if they were ggplot objects.</p>
<p>Here, we will plot both trees. Add the following names/labels to our tree. To add tip labels use <code>geom_tiplab()</code> and bootstrap values <code>geom_label2()</code>. In the latter, we indicate that we will use the node labels named <code>label</code>, as a numeric value and only show those that are more than 75. We also added a scale with <code>geom_treescale()</code>.</p>
<div class="sourceCode" id="cb532"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb532-1"><a href="estimating-phylogenetic-trees.html#cb532-1" aria-hidden="true" tabindex="-1"></a><span class="do">## make sure that we load cowplot</span></span>
<span id="cb532-2"><a href="estimating-phylogenetic-trees.html#cb532-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb532-3"><a href="estimating-phylogenetic-trees.html#cb532-3" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(cowplot)</span>
<span id="cb532-4"><a href="estimating-phylogenetic-trees.html#cb532-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb532-5"><a href="estimating-phylogenetic-trees.html#cb532-5" aria-hidden="true" tabindex="-1"></a><span class="do">## save to an object a both tree</span></span>
<span id="cb532-6"><a href="estimating-phylogenetic-trees.html#cb532-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb532-7"><a href="estimating-phylogenetic-trees.html#cb532-7" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(my_tree, <span class="at">color=</span><span class="st">&quot;blue&quot;</span>) <span class="sc">+</span> <span class="fu">geom_tiplab</span>(<span class="at">size=</span><span class="fl">1.7</span>, <span class="at">color=</span><span class="st">&#39;blue&#39;</span>) <span class="sc">+</span> <span class="fu">geom_treescale</span>(<span class="at">fontsize=</span><span class="dv">4</span>, <span class="at">linesize=</span><span class="dv">1</span>, <span class="at">offset=</span><span class="dv">1</span>, <span class="at">color=</span><span class="st">&#39;blue&#39;</span>)</span>
<span id="cb532-8"><a href="estimating-phylogenetic-trees.html#cb532-8" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggtree</span>(my_tree_boot) <span class="sc">+</span> <span class="fu">geom_tiplab</span>(<span class="at">size=</span><span class="fl">1.7</span>) <span class="sc">+</span> <span class="fu">geom_label2</span>(<span class="fu">aes</span>(<span class="at">label=</span>label, <span class="at">subset =</span> <span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">as.numeric</span>(label)) <span class="sc">&amp;</span> <span class="fu">as.numeric</span>(label) <span class="sc">&gt;</span> <span class="dv">75</span>), <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span> <span class="fu">geom_treescale</span>(<span class="at">fontsize=</span><span class="dv">4</span>, <span class="at">linesize=</span><span class="dv">1</span>, <span class="at">offset=</span><span class="dv">1</span>)</span>
<span id="cb532-9"><a href="estimating-phylogenetic-trees.html#cb532-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb532-10"><a href="estimating-phylogenetic-trees.html#cb532-10" aria-hidden="true" tabindex="-1"></a><span class="do">## you can call each individually as</span></span>
<span id="cb532-11"><a href="estimating-phylogenetic-trees.html#cb532-11" aria-hidden="true" tabindex="-1"></a>p1</span>
<span id="cb532-12"><a href="estimating-phylogenetic-trees.html#cb532-12" aria-hidden="true" tabindex="-1"></a>p2</span>
<span id="cb532-13"><a href="estimating-phylogenetic-trees.html#cb532-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb532-14"><a href="estimating-phylogenetic-trees.html#cb532-14" aria-hidden="true" tabindex="-1"></a><span class="do">## but you can use cowplot to print both side-by-side</span></span>
<span id="cb532-15"><a href="estimating-phylogenetic-trees.html#cb532-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb532-16"><a href="estimating-phylogenetic-trees.html#cb532-16" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p1, p2, <span class="at">ncol=</span><span class="dv">2</span>, <span class="at">labels =</span> LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span></code></pre></div>
<p><img src="figures/session_12_figs/21_ggtree_plot1.png" width="90%" /></p>
</div>
<div id="save-ggtree-plots-as-pdf-or-png" class="section level2" number="12.12">
<h2><span class="header-section-number">12.12</span> Save ggtree plots as PDF or PNG</h2>
<p><strong>12)</strong> You can save your ggtree plots using the function <code>ggsave()</code> and any other ggplot graph.</p>
<div class="sourceCode" id="cb533"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb533-1"><a href="estimating-phylogenetic-trees.html#cb533-1" aria-hidden="true" tabindex="-1"></a><span class="do">## save ggtree as pdf and png</span></span>
<span id="cb533-2"><a href="estimating-phylogenetic-trees.html#cb533-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;regular_bootstrap_ggtree.pdf&quot;</span>)</span>
<span id="cb533-3"><a href="estimating-phylogenetic-trees.html#cb533-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;regular_bootstrap_ggtree.png&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="sequence-alignments.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="mining-from-a-reference-dataset.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/12-Phylogenetic_trees_v7.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
