<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Session 5 – Getting Biological Data from Public Repositories | Bioinformatics Workshop Gitbook</title>
  <meta name="description" content="none" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Session 5 – Getting Biological Data from Public Repositories | Bioinformatics Workshop Gitbook" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="none" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Session 5 – Getting Biological Data from Public Repositories | Bioinformatics Workshop Gitbook" />
  
  <meta name="twitter:description" content="none" />
  

<meta name="author" content="Juan C. Santos &amp; Randy Ortiz" />


<meta name="date" content="2024-03-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="text-editing-and-data-transformations.html"/>
<link rel="next" href="basic-statistics-in-r.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics Workshop Gitbook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html"><i class="fa fa-check"></i><b>1</b> – Getting Started with R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#why-do-we-use-r"><i class="fa fa-check"></i><b>1.1</b> Why do we use R?</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#get-a-good-text-editor"><i class="fa fa-check"></i><b>1.2</b> Get a good text editor</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-r"><i class="fa fa-check"></i><b>1.4</b> Installing R</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#annotating-code"><i class="fa fa-check"></i><b>1.5</b> Annotating code</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#defining-a-working-directory-in-r"><i class="fa fa-check"></i><b>1.6</b> Defining a working directory in R</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html"><i class="fa fa-check"></i><b>2</b> – Using R, Installing Packages and Importing/Exporting Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#your-first-session"><i class="fa fa-check"></i><b>2.1</b> Your first session</a></li>
<li class="chapter" data-level="2.2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#doing-some-base-functions-and-getting-help"><i class="fa fa-check"></i><b>2.2</b> Doing some base functions and getting help</a></li>
<li class="chapter" data-level="2.3" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#functions-between-two-or-more-vectors"><i class="fa fa-check"></i><b>2.3</b> Functions between two or more vectors</a></li>
<li class="chapter" data-level="2.4" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#about-r-packages"><i class="fa fa-check"></i><b>2.4</b> About R-packages</a></li>
<li class="chapter" data-level="2.5" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#cran-task-views-expert-opinion-on-useful-packages"><i class="fa fa-check"></i><b>2.5</b> CRAN Task Views – Expert opinion on useful packages</a></li>
<li class="chapter" data-level="2.6" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#installing-packages"><i class="fa fa-check"></i><b>2.6</b> Installing packages</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html"><i class="fa fa-check"></i><b>3</b> – Basic Data Structures in R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#scalars-and-vectors"><i class="fa fa-check"></i><b>3.1</b> Scalars and vectors</a></li>
<li class="chapter" data-level="3.2" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#arrays"><i class="fa fa-check"></i><b>3.3</b> Arrays</a></li>
<li class="chapter" data-level="3.4" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a></li>
<li class="chapter" data-level="3.5" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tables"><i class="fa fa-check"></i><b>3.5</b> Tables</a></li>
<li class="chapter" data-level="3.6" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#lists"><i class="fa fa-check"></i><b>3.6</b> Lists</a></li>
<li class="chapter" data-level="3.7" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-tables"><i class="fa fa-check"></i><b>3.7</b> Data Tables</a></li>
<li class="chapter" data-level="3.8" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tibbles"><i class="fa fa-check"></i><b>3.8</b> Tibbles</a></li>
<li class="chapter" data-level="3.9" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#dnastringsets-rnastringsets-and-aastringsets"><i class="fa fa-check"></i><b>3.9</b> DNAStringSets, RNAStringSets and AAStringSets</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html"><i class="fa fa-check"></i><b>4</b> – Text Editing and Data Transformations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#working-with-character-strings"><i class="fa fa-check"></i><b>4.1</b> Working with character strings</a></li>
<li class="chapter" data-level="4.2" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#regular-expressions-regex"><i class="fa fa-check"></i><b>4.2</b> Regular expressions (Regex)</a></li>
<li class="chapter" data-level="4.3" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#comparing-ordering-and-simplifying-vectors"><i class="fa fa-check"></i><b>4.3</b> Comparing, ordering and simplifying vectors</a></li>
<li class="chapter" data-level="4.4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#useful-functions-of-stringr"><i class="fa fa-check"></i><b>4.4</b> Useful functions of <code>stringr</code></a></li>
<li class="chapter" data-level="4.5" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#data-transformations"><i class="fa fa-check"></i><b>4.5</b> Data transformations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html"><i class="fa fa-check"></i><b>5</b> – Getting Biological Data from Public Repositories</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#ncbi-databases-genbank-and-others"><i class="fa fa-check"></i><b>5.1</b> NCBI databases: GenBank and others</a></li>
<li class="chapter" data-level="5.2" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#accessing-ncbi-within-r-rentrez"><i class="fa fa-check"></i><b>5.2</b> Accessing NCBI within R: <code>rentrez</code></a></li>
<li class="chapter" data-level="5.3" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#retrieving-pubmed-references"><i class="fa fa-check"></i><b>5.3</b> Retrieving PubMed references</a></li>
<li class="chapter" data-level="5.4" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#comparing-sequences-blast"><i class="fa fa-check"></i><b>5.4</b> Comparing sequences: BLAST</a></li>
<li class="chapter" data-level="5.5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-uniprot"><i class="fa fa-check"></i><b>5.5</b> Other databases: UniProt</a></li>
<li class="chapter" data-level="5.6" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-ensembl"><i class="fa fa-check"></i><b>5.6</b> Other databases: Ensembl</a></li>
<li class="chapter" data-level="5.7" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-taxonomy-phylogenetic-trees-and-species-distributions"><i class="fa fa-check"></i><b>5.7</b> Other databases: Taxonomy, phylogenetic trees and species distributions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html"><i class="fa fa-check"></i><b>6</b> – Basic Statistics in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#exploring-your-data"><i class="fa fa-check"></i><b>6.1</b> Exploring your data</a></li>
<li class="chapter" data-level="6.2" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#sub-setting-your-dataset"><i class="fa fa-check"></i><b>6.2</b> Sub-setting your dataset</a></li>
<li class="chapter" data-level="6.3" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#transposing-rotating-your-dataset"><i class="fa fa-check"></i><b>6.3</b> Transposing (rotating) your dataset</a></li>
<li class="chapter" data-level="6.4" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#dealing-with-missing-na-values"><i class="fa fa-check"></i><b>6.4</b> Dealing with missing (NA) values</a></li>
<li class="chapter" data-level="6.5" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#about-p-values"><i class="fa fa-check"></i><b>6.5</b> About p-values</a></li>
<li class="chapter" data-level="6.6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#how-to-read-p-values"><i class="fa fa-check"></i><b>6.6</b> How to read p-values</a></li>
<li class="chapter" data-level="6.7" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#multiple-testing-corrections"><i class="fa fa-check"></i><b>6.7</b> Multiple testing corrections</a></li>
<li class="chapter" data-level="6.8" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#normalizing-your-data"><i class="fa fa-check"></i><b>6.8</b> Normalizing your data</a></li>
<li class="chapter" data-level="6.9" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#what-test-should-i-use"><i class="fa fa-check"></i><b>6.9</b> What test should I use?</a></li>
<li class="chapter" data-level="6.10" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#type-of-variables-in-dataset"><i class="fa fa-check"></i><b>6.10</b> Type of variables in dataset</a></li>
<li class="chapter" data-level="6.11" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-means-t-test"><i class="fa fa-check"></i><b>6.11</b> Comparing means: t-test</a></li>
<li class="chapter" data-level="6.12" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#two-sample-t-test"><i class="fa fa-check"></i><b>6.12</b> Two sample t-test</a></li>
<li class="chapter" data-level="6.13" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-wilcoxon-signed-rank-test"><i class="fa fa-check"></i><b>6.13</b> Beyond t-test: Wilcoxon signed rank test</a></li>
<li class="chapter" data-level="6.14" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#tabulation-and-cross-tabulation"><i class="fa fa-check"></i><b>6.14</b> Tabulation and Cross-Tabulation</a></li>
<li class="chapter" data-level="6.15" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-binomial-test"><i class="fa fa-check"></i><b>6.15</b> Beyond t-test: Binomial test</a></li>
<li class="chapter" data-level="6.16" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-chi-square-goodness-of-fit"><i class="fa fa-check"></i><b>6.16</b> Beyond t-test: Chi-square goodness of fit</a></li>
<li class="chapter" data-level="6.17" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-mcnemars-test"><i class="fa fa-check"></i><b>6.17</b> Beyond t-test: McNemar’s test</a></li>
<li class="chapter" data-level="6.18" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-fishers-exact-test"><i class="fa fa-check"></i><b>6.18</b> Beyond t-test: Fisher’s exact test</a></li>
<li class="chapter" data-level="6.19" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#correlations"><i class="fa fa-check"></i><b>6.19</b> Correlations</a></li>
<li class="chapter" data-level="6.20" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-distributions"><i class="fa fa-check"></i><b>6.20</b> Comparing distributions</a></li>
<li class="chapter" data-level="6.21" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#bootstraps"><i class="fa fa-check"></i><b>6.21</b> Bootstraps</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html"><i class="fa fa-check"></i><b>7</b> – Regressions and ANOVA in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#linear-regressions"><i class="fa fa-check"></i><b>7.1</b> Linear regressions</a></li>
<li class="chapter" data-level="7.2" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anatomy-of-a-linear-regression"><i class="fa fa-check"></i><b>7.2</b> Anatomy of a linear regression</a></li>
<li class="chapter" data-level="7.3" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#multiple-linear-regressions"><i class="fa fa-check"></i><b>7.3</b> Multiple linear regressions</a></li>
<li class="chapter" data-level="7.4" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#diagnostic-plots-for-regressions"><i class="fa fa-check"></i><b>7.4</b> Diagnostic plots for regressions</a></li>
<li class="chapter" data-level="7.5" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-influential-measurements"><i class="fa fa-check"></i><b>7.5</b> Regression influential measurements</a></li>
<li class="chapter" data-level="7.6" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-coefficients-and-their-95ci"><i class="fa fa-check"></i><b>7.6</b> Regression coefficients and their 95%CI</a></li>
<li class="chapter" data-level="7.7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#selecting-predictors"><i class="fa fa-check"></i><b>7.7</b> Selecting predictors</a></li>
<li class="chapter" data-level="7.8" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-predictions"><i class="fa fa-check"></i><b>7.8</b> Regression predictions</a></li>
<li class="chapter" data-level="7.9" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions"><i class="fa fa-check"></i><b>7.9</b> Logistic regressions</a></li>
<li class="chapter" data-level="7.10" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions-requirements"><i class="fa fa-check"></i><b>7.10</b> Logistic regressions requirements</a></li>
<li class="chapter" data-level="7.11" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#non-linear-regressions"><i class="fa fa-check"></i><b>7.11</b> Non-linear regressions</a></li>
<li class="chapter" data-level="7.12" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-regressions"><i class="fa fa-check"></i><b>7.12</b> Toy datasets for regressions</a></li>
<li class="chapter" data-level="7.13" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova"><i class="fa fa-check"></i><b>7.13</b> ANOVA</a></li>
<li class="chapter" data-level="7.14" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#one-way-anova"><i class="fa fa-check"></i><b>7.14</b> One-way ANOVA</a></li>
<li class="chapter" data-level="7.15" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#two-way-anova"><i class="fa fa-check"></i><b>7.15</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="7.16" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova-post-hoc-tests"><i class="fa fa-check"></i><b>7.16</b> ANOVA Post-hoc tests</a></li>
<li class="chapter" data-level="7.17" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#beyond-anova-kruskal-wallis-test"><i class="fa fa-check"></i><b>7.17</b> Beyond ANOVA: Kruskal-Wallis Test</a></li>
<li class="chapter" data-level="7.18" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-anova"><i class="fa fa-check"></i><b>7.18</b> Toy datasets for ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html"><i class="fa fa-check"></i><b>8</b> – Multidimensional Data in R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#principal-component-analysis"><i class="fa fa-check"></i><b>8.1</b> Principal Component Analysis</a></li>
<li class="chapter" data-level="8.2" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#multidimensional-scaling"><i class="fa fa-check"></i><b>8.2</b> Multidimensional scaling</a></li>
<li class="chapter" data-level="8.3" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-dfa"><i class="fa fa-check"></i><b>8.3</b> Cluster analysis: DFA</a></li>
<li class="chapter" data-level="8.4" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-k-means-unsupervised"><i class="fa fa-check"></i><b>8.4</b> Cluster analysis: <em>k</em>-means (unsupervised)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html"><i class="fa fa-check"></i><b>9</b> – Graphics and Colors</a>
<ul>
<li class="chapter" data-level="9.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#avoid-bad-graphics"><i class="fa fa-check"></i><b>9.1</b> Avoid bad graphics</a></li>
<li class="chapter" data-level="9.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#graphics-in-r-ggplot2"><i class="fa fa-check"></i><b>9.2</b> Graphics in R – ggplot2</a></li>
<li class="chapter" data-level="9.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#preparing-your-dataset"><i class="fa fa-check"></i><b>9.3</b> Preparing your dataset</a></li>
<li class="chapter" data-level="9.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#color-palettes-online"><i class="fa fa-check"></i><b>9.4</b> Color palettes online</a></li>
<li class="chapter" data-level="9.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggplot2-default-palette"><i class="fa fa-check"></i><b>9.5</b> <code>ggplot2</code> default palette</a></li>
<li class="chapter" data-level="9.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#heatmaps-or-tiled-charts"><i class="fa fa-check"></i><b>9.6</b> Heatmaps or tiled charts</a></li>
<li class="chapter" data-level="9.7" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#r-packages-about-color"><i class="fa fa-check"></i><b>9.7</b> R-packages about color</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#viridis-package"><i class="fa fa-check"></i><b>9.7.1</b> <code>viridis</code> package</a></li>
<li class="chapter" data-level="9.7.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcolorbrewer-package"><i class="fa fa-check"></i><b>9.7.2</b> <code>RColorBrewer</code> package</a></li>
<li class="chapter" data-level="9.7.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggsci-package"><i class="fa fa-check"></i><b>9.7.3</b> <code>ggsci</code> package</a></li>
<li class="chapter" data-level="9.7.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#wesanderson-package"><i class="fa fa-check"></i><b>9.7.4</b> <code>wesanderson</code> package</a></li>
<li class="chapter" data-level="9.7.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcartocolor-package"><i class="fa fa-check"></i><b>9.7.5</b> <code>rcartocolor</code> package</a></li>
<li class="chapter" data-level="9.7.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#colorspace-package"><i class="fa fa-check"></i><b>9.7.6</b> <code>colorspace</code> package</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#making-your-own-palette"><i class="fa fa-check"></i><b>9.8</b> Making your own palette</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="basic-graphics.html"><a href="basic-graphics.html"><i class="fa fa-check"></i><b>10</b> – Basic Graphics</a>
<ul>
<li class="chapter" data-level="10.1" data-path="basic-graphics.html"><a href="basic-graphics.html#what-would-you-like-to-show"><i class="fa fa-check"></i><b>10.1</b> What would you like to show?</a></li>
<li class="chapter" data-level="10.2" data-path="basic-graphics.html"><a href="basic-graphics.html#bar-charts"><i class="fa fa-check"></i><b>10.2</b> Bar charts</a></li>
<li class="chapter" data-level="10.3" data-path="basic-graphics.html"><a href="basic-graphics.html#stacked-bar-charts"><i class="fa fa-check"></i><b>10.3</b> Stacked bar charts</a></li>
<li class="chapter" data-level="10.4" data-path="basic-graphics.html"><a href="basic-graphics.html#histograms"><i class="fa fa-check"></i><b>10.4</b> Histograms</a></li>
<li class="chapter" data-level="10.5" data-path="basic-graphics.html"><a href="basic-graphics.html#multiple-histograms"><i class="fa fa-check"></i><b>10.5</b> Multiple histograms</a></li>
<li class="chapter" data-level="10.6" data-path="basic-graphics.html"><a href="basic-graphics.html#density-plots"><i class="fa fa-check"></i><b>10.6</b> Density plots</a></li>
<li class="chapter" data-level="10.7" data-path="basic-graphics.html"><a href="basic-graphics.html#box-plots"><i class="fa fa-check"></i><b>10.7</b> Box plots</a></li>
<li class="chapter" data-level="10.8" data-path="basic-graphics.html"><a href="basic-graphics.html#violin-plots"><i class="fa fa-check"></i><b>10.8</b> Violin plots</a></li>
<li class="chapter" data-level="10.9" data-path="basic-graphics.html"><a href="basic-graphics.html#scatterplots"><i class="fa fa-check"></i><b>10.9</b> Scatterplots</a></li>
<li class="chapter" data-level="10.10" data-path="basic-graphics.html"><a href="basic-graphics.html#lines-and-scatterplots"><i class="fa fa-check"></i><b>10.10</b> Lines and Scatterplots</a></li>
<li class="chapter" data-level="10.11" data-path="basic-graphics.html"><a href="basic-graphics.html#save-your-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>10.11</b> Save your plots as PDF or PNG</a></li>
<li class="chapter" data-level="10.12" data-path="basic-graphics.html"><a href="basic-graphics.html#ggplot2-add-on-packages-treemapify"><i class="fa fa-check"></i><b>10.12</b> ggplot2 add-on packages: <code>treemapify</code></a></li>
<li class="chapter" data-level="10.13" data-path="basic-graphics.html"><a href="basic-graphics.html#other-resources"><i class="fa fa-check"></i><b>10.13</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="sequence-alignments.html"><a href="sequence-alignments.html"><i class="fa fa-check"></i><b>11</b> – Sequence Alignments</a>
<ul>
<li class="chapter" data-level="11.1" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-using-rentrez"><i class="fa fa-check"></i><b>11.1</b> Retrieving sequences using ‘rentrez’</a></li>
<li class="chapter" data-level="11.2" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-manually-from-ncbi"><i class="fa fa-check"></i><b>11.2</b> Retrieving sequences manually from NCBI</a></li>
<li class="chapter" data-level="11.3" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-accession-numbers"><i class="fa fa-check"></i><b>11.3</b> Retrieving accession numbers</a></li>
<li class="chapter" data-level="11.4" data-path="sequence-alignments.html"><a href="sequence-alignments.html#alignment-with-decipher"><i class="fa fa-check"></i><b>11.4</b> Alignment with <code>DECIPHER</code></a></li>
<li class="chapter" data-level="11.5" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualize-alignment"><i class="fa fa-check"></i><b>11.5</b> Visualize alignment</a></li>
<li class="chapter" data-level="11.6" data-path="sequence-alignments.html"><a href="sequence-alignments.html#save-alignment-to-file"><i class="fa fa-check"></i><b>11.6</b> Save alignment to file</a></li>
<li class="chapter" data-level="11.7" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-and-align-several-markers"><i class="fa fa-check"></i><b>11.7</b> Concatenate and align several markers</a></li>
<li class="chapter" data-level="11.8" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-data.table"><i class="fa fa-check"></i><b>11.8</b> Concatenate with <code>data.table</code></a></li>
<li class="chapter" data-level="11.9" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-biostrings"><i class="fa fa-check"></i><b>11.9</b> Concatenate with <code>Biostrings</code></a></li>
<li class="chapter" data-level="11.10" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualizing-alignments-in-r"><i class="fa fa-check"></i><b>11.10</b> Visualizing alignments in R</a></li>
<li class="chapter" data-level="11.11" data-path="sequence-alignments.html"><a href="sequence-alignments.html#other-sequence-alignment-software"><i class="fa fa-check"></i><b>11.11</b> Other sequence alignment software</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html"><i class="fa fa-check"></i><b>12</b> – Estimating Phylogenetic Trees</a>
<ul>
<li class="chapter" data-level="12.1" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#phylogeny-inference-with-iqtree2"><i class="fa fa-check"></i><b>12.1</b> Phylogeny inference with <code>iqtree2</code></a></li>
<li class="chapter" data-level="12.2" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-iqtree2-in-macos"><i class="fa fa-check"></i><b>12.2</b> Installing <code>iqtree2</code> in macOS</a></li>
<li class="chapter" data-level="12.3" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-and-running-iqtree2-in-a-pc"><i class="fa fa-check"></i><b>12.3</b> Installing and running <code>iqtree2</code> in a PC</a></li>
<li class="chapter" data-level="12.4" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#loading-the-iqtree_runner_r-function"><i class="fa fa-check"></i><b>12.4</b> Loading the <code>iqtree_runner_R</code> function</a></li>
<li class="chapter" data-level="12.5" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#running-iqtree2-from-r"><i class="fa fa-check"></i><b>12.5</b> Running <code>iqtree2</code> from R</a></li>
<li class="chapter" data-level="12.6" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-no-partitions"><i class="fa fa-check"></i><b>12.6</b> One marker and no partitions</a></li>
<li class="chapter" data-level="12.7" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-with-partitions"><i class="fa fa-check"></i><b>12.7</b> One marker and with partitions</a></li>
<li class="chapter" data-level="12.8" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#bootstrap-support"><i class="fa fa-check"></i><b>12.8</b> Bootstrap support</a></li>
<li class="chapter" data-level="12.9" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#many-markers-and-with-partitions"><i class="fa fa-check"></i><b>12.9</b> Many markers and with partitions</a></li>
<li class="chapter" data-level="12.10" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#other-sequence-phylogenetic-tree-estimation-software"><i class="fa fa-check"></i><b>12.10</b> Other sequence phylogenetic tree estimation software</a></li>
<li class="chapter" data-level="12.11" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#drawing-phylogenetic-trees-with-ggtree"><i class="fa fa-check"></i><b>12.11</b> Drawing phylogenetic trees with <code>ggtree</code></a></li>
<li class="chapter" data-level="12.12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#save-ggtree-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>12.12</b> Save ggtree plots as PDF or PNG</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html"><i class="fa fa-check"></i><b>13</b> – Mining from a reference dataset</a>
<ul>
<li class="chapter" data-level="13.1" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#a-worked-example-using-the-mtcars-dataset"><i class="fa fa-check"></i><b>13.1</b> A worked example using the <code>mtcars</code> dataset</a></li>
<li class="chapter" data-level="13.2" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-reference-dataset"><i class="fa fa-check"></i><b>13.2</b> Build a reference dataset</a></li>
<li class="chapter" data-level="13.3" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-sampling-dataset"><i class="fa fa-check"></i><b>13.3</b> Build a sampling dataset</a></li>
<li class="chapter" data-level="13.4" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#extracting-from-reference-dataset"><i class="fa fa-check"></i><b>13.4</b> Extracting from reference dataset</a></li>
<li class="chapter" data-level="13.5" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#plots-of-the-extracted-data"><i class="fa fa-check"></i><b>13.5</b> Plots of the extracted data</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="introduction-to-python.html"><a href="introduction-to-python.html"><i class="fa fa-check"></i><b>14</b> – Introduction to Python</a>
<ul>
<li class="chapter" data-level="14.1" data-path="introduction-to-python.html"><a href="introduction-to-python.html#welcome-to-python-why-python"><i class="fa fa-check"></i><b>14.1</b> Welcome to Python! Why Python?</a></li>
<li class="chapter" data-level="14.2" data-path="introduction-to-python.html"><a href="introduction-to-python.html#which-python-should-i-install"><i class="fa fa-check"></i><b>14.2</b> Which Python should I install?</a></li>
<li class="chapter" data-level="14.3" data-path="introduction-to-python.html"><a href="introduction-to-python.html#mac-os-x-python-installations"><i class="fa fa-check"></i><b>14.3</b> Mac OS X Python Installations</a></li>
<li class="chapter" data-level="14.4" data-path="introduction-to-python.html"><a href="introduction-to-python.html#windows-python-installations"><i class="fa fa-check"></i><b>14.4</b> Windows Python Installations</a></li>
<li class="chapter" data-level="14.5" data-path="introduction-to-python.html"><a href="introduction-to-python.html#install-an-ide-or-integrated-development-environment"><i class="fa fa-check"></i><b>14.5</b> Install an IDE, or integrated development environment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html"><i class="fa fa-check"></i><b>15</b> – Python: Keywords and Identifiers</a>
<ul>
<li class="chapter" data-level="15.1" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#keywords-in-python"><i class="fa fa-check"></i><b>15.1</b> Keywords in Python</a></li>
<li class="chapter" data-level="15.2" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#identifiers-in-python"><i class="fa fa-check"></i><b>15.2</b> Identifiers in Python</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html"><i class="fa fa-check"></i><b>16</b> – Python: Statements and Expressions</a>
<ul>
<li class="chapter" data-level="16.1" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#what-are-statements"><i class="fa fa-check"></i><b>16.1</b> What are statements?</a></li>
<li class="chapter" data-level="16.2" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#expressions-in-python"><i class="fa fa-check"></i><b>16.2</b> Expressions in Python</a></li>
<li class="chapter" data-level="16.3" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#simple-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.3</b> Simple Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.4" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#augmented-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.4</b> Augmented Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.5" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#explicit-and-implicit-line-continuation"><i class="fa fa-check"></i><b>16.5</b> Explicit and Implicit Line Continuation</a></li>
<li class="chapter" data-level="16.6" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#indentation-in-python-is-crucial-to-proper-code-function"><i class="fa fa-check"></i><b>16.6</b> Indentation in Python is Crucial to Proper Code function</a></li>
<li class="chapter" data-level="16.7" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#comments"><i class="fa fa-check"></i><b>16.7</b> Comments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="python-data-types.html"><a href="python-data-types.html"><i class="fa fa-check"></i><b>17</b> – Python: Data Types</a>
<ul>
<li class="chapter" data-level="17.1" data-path="python-data-types.html"><a href="python-data-types.html#booleans"><i class="fa fa-check"></i><b>17.1</b> Booleans</a></li>
<li class="chapter" data-level="17.2" data-path="python-data-types.html"><a href="python-data-types.html#numbers-int-float-and-complex"><i class="fa fa-check"></i><b>17.2</b> Numbers: int, float and complex</a></li>
<li class="chapter" data-level="17.3" data-path="python-data-types.html"><a href="python-data-types.html#strings"><i class="fa fa-check"></i><b>17.3</b> Strings</a></li>
<li class="chapter" data-level="17.4" data-path="python-data-types.html"><a href="python-data-types.html#bytes"><i class="fa fa-check"></i><b>17.4</b> Bytes</a></li>
<li class="chapter" data-level="17.5" data-path="python-data-types.html"><a href="python-data-types.html#lists-1"><i class="fa fa-check"></i><b>17.5</b> Lists</a></li>
<li class="chapter" data-level="17.6" data-path="python-data-types.html"><a href="python-data-types.html#tuples"><i class="fa fa-check"></i><b>17.6</b> Tuples</a></li>
<li class="chapter" data-level="17.7" data-path="python-data-types.html"><a href="python-data-types.html#sets"><i class="fa fa-check"></i><b>17.7</b> Sets</a></li>
<li class="chapter" data-level="17.8" data-path="python-data-types.html"><a href="python-data-types.html#dictionary"><i class="fa fa-check"></i><b>17.8</b> Dictionary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="python-strings.html"><a href="python-strings.html"><i class="fa fa-check"></i><b>18</b> – Python: Strings</a>
<ul>
<li class="chapter" data-level="18.1" data-path="python-strings.html"><a href="python-strings.html#string-operators"><i class="fa fa-check"></i><b>18.1</b> String Operators</a></li>
<li class="chapter" data-level="18.2" data-path="python-strings.html"><a href="python-strings.html#other-string-modifiers"><i class="fa fa-check"></i><b>18.2</b> Other String Modifiers</a></li>
<li class="chapter" data-level="18.3" data-path="python-strings.html"><a href="python-strings.html#format-characters-in-strings"><i class="fa fa-check"></i><b>18.3</b> Format-Characters in Strings</a></li>
<li class="chapter" data-level="18.4" data-path="python-strings.html"><a href="python-strings.html#comparisons-with-strings"><i class="fa fa-check"></i><b>18.4</b> Comparisons with Strings</a></li>
<li class="chapter" data-level="18.5" data-path="python-strings.html"><a href="python-strings.html#padding-strings"><i class="fa fa-check"></i><b>18.5</b> Padding strings</a></li>
<li class="chapter" data-level="18.6" data-path="python-strings.html"><a href="python-strings.html#search-for-words-in-string-and-output-location"><i class="fa fa-check"></i><b>18.6</b> Search for words in string and output location</a></li>
<li class="chapter" data-level="18.7" data-path="python-strings.html"><a href="python-strings.html#seek-and-replace-in-strings"><i class="fa fa-check"></i><b>18.7</b> Seek and replace in strings</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="python-operators.html"><a href="python-operators.html"><i class="fa fa-check"></i><b>19</b> – Python: Operators</a>
<ul>
<li class="chapter" data-level="19.1" data-path="python-operators.html"><a href="python-operators.html#many-types-of-operators"><i class="fa fa-check"></i><b>19.1</b> Many types of operators</a></li>
<li class="chapter" data-level="19.2" data-path="python-operators.html"><a href="python-operators.html#arithmetic-operators"><i class="fa fa-check"></i><b>19.2</b> Arithmetic operators</a></li>
<li class="chapter" data-level="19.3" data-path="python-operators.html"><a href="python-operators.html#comparison-operators"><i class="fa fa-check"></i><b>19.3</b> Comparison operators</a></li>
<li class="chapter" data-level="19.4" data-path="python-operators.html"><a href="python-operators.html#logical-operators"><i class="fa fa-check"></i><b>19.4</b> Logical operators</a></li>
<li class="chapter" data-level="19.5" data-path="python-operators.html"><a href="python-operators.html#bitwise-operators"><i class="fa fa-check"></i><b>19.5</b> Bitwise operators</a></li>
<li class="chapter" data-level="19.6" data-path="python-operators.html"><a href="python-operators.html#identity-operators"><i class="fa fa-check"></i><b>19.6</b> Identity Operators</a></li>
<li class="chapter" data-level="19.7" data-path="python-operators.html"><a href="python-operators.html#membership-operators"><i class="fa fa-check"></i><b>19.7</b> Membership operators</a></li>
<li class="chapter" data-level="19.8" data-path="python-operators.html"><a href="python-operators.html#what-is-the-order-of-operations-with-operators"><i class="fa fa-check"></i><b>19.8</b> What is the order of operations with operators?</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="biopython.html"><a href="biopython.html"><i class="fa fa-check"></i><b>20</b> – Biopython</a>
<ul>
<li class="chapter" data-level="20.1" data-path="biopython.html"><a href="biopython.html#introduction-to-biopython"><i class="fa fa-check"></i><b>20.1</b> Introduction to Biopython</a></li>
<li class="chapter" data-level="20.2" data-path="biopython.html"><a href="biopython.html#biopython-installation"><i class="fa fa-check"></i><b>20.2</b> Biopython Installation</a></li>
<li class="chapter" data-level="20.3" data-path="biopython.html"><a href="biopython.html#getting-started"><i class="fa fa-check"></i><b>20.3</b> Getting Started</a></li>
<li class="chapter" data-level="20.4" data-path="biopython.html"><a href="biopython.html#parsing-my-fasta"><i class="fa fa-check"></i><b>20.4</b> Parsing my FASTA</a></li>
<li class="chapter" data-level="20.5" data-path="biopython.html"><a href="biopython.html#parsing-my-genbank-or-genpept"><i class="fa fa-check"></i><b>20.5</b> Parsing my GenBank or GenPept</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="biopython-sequences.html"><a href="biopython-sequences.html"><i class="fa fa-check"></i><b>21</b> – Biopython: Sequences</a>
<ul>
<li class="chapter" data-level="21.1" data-path="biopython-sequences.html"><a href="biopython-sequences.html#sequence-objects"><i class="fa fa-check"></i><b>21.1</b> Sequence Objects</a></li>
<li class="chapter" data-level="21.2" data-path="biopython-sequences.html"><a href="biopython-sequences.html#concatenation-with-sequences"><i class="fa fa-check"></i><b>21.2</b> Concatenation with Sequences</a></li>
<li class="chapter" data-level="21.3" data-path="biopython-sequences.html"><a href="biopython-sequences.html#transcription-dna-to-mrna"><i class="fa fa-check"></i><b>21.3</b> Transcription: DNA to mRNA</a></li>
<li class="chapter" data-level="21.4" data-path="biopython-sequences.html"><a href="biopython-sequences.html#translation-rna-to-prot"><i class="fa fa-check"></i><b>21.4</b> Translation: RNA to PROT</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html"><i class="fa fa-check"></i><b>22</b> – Biopython: Mulitple Sequence Alignments</a>
<ul>
<li class="chapter" data-level="22.1" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwise2"><i class="fa fa-check"></i><b>22.1</b> pairwise2</a></li>
<li class="chapter" data-level="22.2" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwisealigner"><i class="fa fa-check"></i><b>22.2</b> PairwiseAligner</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html"><i class="fa fa-check"></i><b>23</b> – Biopython: GenomeDiagram</a>
<ul>
<li class="chapter" data-level="23.1" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#linearcircular-gene-diagrams"><i class="fa fa-check"></i><b>23.1</b> Linear/Circular Gene Diagrams</a></li>
<li class="chapter" data-level="23.2" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#diagram-readability"><i class="fa fa-check"></i><b>23.2</b> Diagram Readability</a></li>
<li class="chapter" data-level="23.3" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#box-shape"><i class="fa fa-check"></i><b>23.3</b> Box Shape</a></li>
<li class="chapter" data-level="23.4" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#edit-box-shape-arrows"><i class="fa fa-check"></i><b>23.4</b> Edit Box Shape: Arrows</a></li>
<li class="chapter" data-level="23.5" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#restriction-map"><i class="fa fa-check"></i><b>23.5</b> Restriction Map</a></li>
<li class="chapter" data-level="23.6" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#tracks-on-tracks"><i class="fa fa-check"></i><b>23.6</b> Tracks on Tracks</a></li>
<li class="chapter" data-level="23.7" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#cross-link-data"><i class="fa fa-check"></i><b>23.7</b> Cross-link Data</a></li>
<li class="chapter" data-level="23.8" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#clean-cross-link"><i class="fa fa-check"></i><b>23.8</b> Clean Cross-Link</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html"><i class="fa fa-check"></i><b>24</b> – Biopython: Chromosome Modeling</a>
<ul>
<li class="chapter" data-level="24.1" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html#gene-feature-modeling"><i class="fa fa-check"></i><b>24.1</b> Gene Feature Modeling</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html"><i class="fa fa-check"></i><b>25</b> – Biopython: Sequence Parsing with Plots</a>
<ul>
<li class="chapter" data-level="25.1" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#histograms-1"><i class="fa fa-check"></i><b>25.1</b> Histograms</a></li>
<li class="chapter" data-level="25.2" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#gc-content"><i class="fa fa-check"></i><b>25.2</b> GC Content</a></li>
<li class="chapter" data-level="25.3" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#nt-alignment-dot-plot"><i class="fa fa-check"></i><b>25.3</b> Nt Alignment Dot Plot</a></li>
<li class="chapter" data-level="25.4" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#bigger-dots-for-the-dot-plot"><i class="fa fa-check"></i><b>25.4</b> Bigger Dots for the Dot Plot</a></li>
<li class="chapter" data-level="25.5" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#fastq-we-did-not-forget-you"><i class="fa fa-check"></i><b>25.5</b> FASTQ we did not forget you</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html"><i class="fa fa-check"></i><b>26</b> – Python: Web Scraping w/ BeautifulSoup</a>
<ul>
<li class="chapter" data-level="26.1" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#introduction"><i class="fa fa-check"></i><b>26.1</b> Introduction</a></li>
<li class="chapter" data-level="26.2" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#installation"><i class="fa fa-check"></i><b>26.2</b> Installation</a></li>
<li class="chapter" data-level="26.3" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#getting-started-1"><i class="fa fa-check"></i><b>26.3</b> Getting started</a></li>
<li class="chapter" data-level="26.4" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-all-matching-tags"><i class="fa fa-check"></i><b>26.4</b> Mining all matching tags</a></li>
<li class="chapter" data-level="26.5" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-tags-by-class-or-id"><i class="fa fa-check"></i><b>26.5</b> Mining tags by class or id</a></li>
<li class="chapter" data-level="26.6" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#use-css-selectors-for-positional-mining"><i class="fa fa-check"></i><b>26.6</b> Use CSS Selectors for positional mining</a></li>
<li class="chapter" data-level="26.7" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#weather-page-example"><i class="fa fa-check"></i><b>26.7</b> Weather Page Example</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="python-subprocess.html"><a href="python-subprocess.html"><i class="fa fa-check"></i><b>27</b> – Python Subprocess</a>
<ul>
<li class="chapter" data-level="27.1" data-path="python-subprocess.html"><a href="python-subprocess.html#introduction-and-installation"><i class="fa fa-check"></i><b>27.1</b> Introduction and Installation</a></li>
<li class="chapter" data-level="27.2" data-path="python-subprocess.html"><a href="python-subprocess.html#creating-your-first-python-script-w-bbtools"><i class="fa fa-check"></i><b>27.2</b> Creating your first python script w/ BBTools</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html"><i class="fa fa-check"></i><b>28</b> – Control Structures &amp; Loops in R</a>
<ul>
<li class="chapter" data-level="28.1" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#controling-the-flow"><i class="fa fa-check"></i><b>28.1</b> Controling the flow</a></li>
<li class="chapter" data-level="28.2" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if"><i class="fa fa-check"></i><b>28.2</b> Conditional <code>if</code></a></li>
<li class="chapter" data-level="28.3" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if-and-else"><i class="fa fa-check"></i><b>28.3</b> Conditional <code>if</code> and <code>else</code></a></li>
<li class="chapter" data-level="28.4" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#nested-conditionals-if-else-and-if-else"><i class="fa fa-check"></i><b>28.4</b> Nested conditionals <code>if</code>, <code>else</code>, and <code>if else</code></a></li>
<li class="chapter" data-level="28.5" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-ifelse"><i class="fa fa-check"></i><b>28.5</b> Conditional <code>ifelse</code></a></li>
<li class="chapter" data-level="28.6" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#loops-that-control-iterations-and-repeations"><i class="fa fa-check"></i><b>28.6</b> Loops that control iterations and repeations</a></li>
<li class="chapter" data-level="28.7" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-for-loop"><i class="fa fa-check"></i><b>28.7</b> The <code>for</code> Loop</a></li>
<li class="chapter" data-level="28.8" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-while-loop"><i class="fa fa-check"></i><b>28.8</b> The <code>while</code> Loop</a></li>
<li class="chapter" data-level="28.9" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-repeat-loop"><i class="fa fa-check"></i><b>28.9</b> The <code>repeat</code> Loop</a></li>
<li class="chapter" data-level="28.10" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-next-and-break-statements"><i class="fa fa-check"></i><b>28.10</b> The <code>next</code> and <code>break</code> statements</a></li>
<li class="chapter" data-level="28.11" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functions"><i class="fa fa-check"></i><b>28.11</b> Other loop functions</a></li>
<li class="chapter" data-level="28.12" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#lapply-functions"><i class="fa fa-check"></i><b>28.12</b> <code>lapply</code> functions</a></li>
<li class="chapter" data-level="28.13" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsapply"><i class="fa fa-check"></i><b>28.13</b> Other loop functions:<code>apply</code></a></li>
<li class="chapter" data-level="28.14" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsmapply"><i class="fa fa-check"></i><b>28.14</b> Other loop functions:<code>mapply</code></a></li>
<li class="chapter" data-level="28.15" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionstapply"><i class="fa fa-check"></i><b>28.15</b> Other loop functions:<code>tapply</code></a></li>
<li class="chapter" data-level="28.16" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionssplit"><i class="fa fa-check"></i><b>28.16</b> Other loop functions:<code>split</code></a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html"><i class="fa fa-check"></i><b>29</b> – Bioinformatics Literature</a>
<ul>
<li class="chapter" data-level="29.1" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#peer-reviewed-bioinformatics-journals"><i class="fa fa-check"></i><b>29.1</b> Peer-reviewed bioinformatics journals</a></li>
<li class="chapter" data-level="29.2" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#access-to-journals-articles"><i class="fa fa-check"></i><b>29.2</b> Access to journals articles</a></li>
<li class="chapter" data-level="29.3" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#informal-access-to-code-preprints-and-vignettes"><i class="fa fa-check"></i><b>29.3</b> Informal access to code: Preprints and vignettes</a></li>
<li class="chapter" data-level="29.4" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#online-bioinformatics-courses"><i class="fa fa-check"></i><b>29.4</b> Online bioinformatics courses</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html"><i class="fa fa-check"></i><b>30</b> – Importing/Exporting Data into R</a>
<ul>
<li class="chapter" data-level="30.1" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#typing-directly"><i class="fa fa-check"></i><b>30.1</b> Typing directly</a></li>
<li class="chapter" data-level="30.2" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#summary-versus-raw-data"><i class="fa fa-check"></i><b>30.2</b> Summary versus raw data</a></li>
<li class="chapter" data-level="30.3" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#downloading-data-files-github"><i class="fa fa-check"></i><b>30.3</b> Downloading data files (GitHub)</a></li>
<li class="chapter" data-level="30.4" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#importing-data-from-a-file"><i class="fa fa-check"></i><b>30.4</b> Importing data from a file</a></li>
<li class="chapter" data-level="30.5" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-to-a-file"><i class="fa fa-check"></i><b>30.5</b> Saving data to a file</a></li>
<li class="chapter" data-level="30.6" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-as-rdata-.rdata-.rda"><i class="fa fa-check"></i><b>30.6</b> Saving data as Rdata (.rdata, .rda)</a></li>
<li class="chapter" data-level="30.7" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#import-excel-formated-data"><i class="fa fa-check"></i><b>30.7</b> Import Excel formated data</a></li>
<li class="chapter" data-level="30.8" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#save-data-as-excel-formated-file"><i class="fa fa-check"></i><b>30.8</b> Save data as Excel formated file</a></li>
<li class="chapter" data-level="30.9" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#fixing-problems-while-importing-datasets"><i class="fa fa-check"></i><b>30.9</b> Fixing problems while importing datasets</a></li>
<li class="chapter" data-level="30.10" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#simplifying-embellished-excel-datasets"><i class="fa fa-check"></i><b>30.10</b> Simplifying embellished excel datasets</a></li>
<li class="chapter" data-level="30.11" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#accessing-online-data"><i class="fa fa-check"></i><b>30.11</b> Accessing online data</a></li>
<li class="chapter" data-level="30.12" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#data-from-clinicaltrials.gov"><i class="fa fa-check"></i><b>30.12</b> Data from clinicaltrials.gov</a></li>
<li class="chapter" data-level="30.13" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#data-scrapping-from-a-website"><i class="fa fa-check"></i><b>30.13</b> Data scrapping from a website</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="working-with-proteins.html"><a href="working-with-proteins.html"><i class="fa fa-check"></i><b>31</b> – Working with Proteins</a>
<ul>
<li class="chapter" data-level="31.1" data-path="working-with-proteins.html"><a href="working-with-proteins.html#sequences-in-fasta-format"><i class="fa fa-check"></i><b>31.1</b> Sequences in FASTA format</a></li>
<li class="chapter" data-level="31.2" data-path="working-with-proteins.html"><a href="working-with-proteins.html#fetching-aa-sequences-from-ncbi"><i class="fa fa-check"></i><b>31.2</b> Fetching AA sequences from NCBI</a></li>
<li class="chapter" data-level="31.3" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-aa-sequences-in-r"><i class="fa fa-check"></i><b>31.3</b> Importing AA sequences in R</a></li>
<li class="chapter" data-level="31.4" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-nucleotide-sequences-in-r"><i class="fa fa-check"></i><b>31.4</b> Importing nucleotide sequences in R</a></li>
<li class="chapter" data-level="31.5" data-path="working-with-proteins.html"><a href="working-with-proteins.html#translating-a-nucleotides-to-aa"><i class="fa fa-check"></i><b>31.5</b> Translating a nucleotides to AA</a></li>
<li class="chapter" data-level="31.6" data-path="working-with-proteins.html"><a href="working-with-proteins.html#finding-orfs-in-a-nucleotide-sequence"><i class="fa fa-check"></i><b>31.6</b> Finding ORFs in a nucleotide sequence</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="miscellaneous.html"><a href="miscellaneous.html"><i class="fa fa-check"></i><b>32</b> – Miscellaneous</a>
<ul>
<li class="chapter" data-level="32.1" data-path="miscellaneous.html"><a href="miscellaneous.html#sampling-from-a-vector-of-names"><i class="fa fa-check"></i><b>32.1</b> Sampling from a vector of names</a></li>
<li class="chapter" data-level="32.2" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-an-article-pdf"><i class="fa fa-check"></i><b>32.2</b> Read a text from an article pdf</a></li>
<li class="chapter" data-level="32.3" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-website"><i class="fa fa-check"></i><b>32.3</b> Read a text from a website</a></li>
<li class="chapter" data-level="32.4" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-file"><i class="fa fa-check"></i><b>32.4</b> Read a text from a file</a></li>
<li class="chapter" data-level="32.5" data-path="miscellaneous.html"><a href="miscellaneous.html#generate-the-word-cloud"><i class="fa fa-check"></i><b>32.5</b> Generate the word cloud</a></li>
<li class="chapter" data-level="32.6" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-pdf"><i class="fa fa-check"></i><b>32.6</b> worldcloud – PDF</a></li>
<li class="chapter" data-level="32.7" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-website"><i class="fa fa-check"></i><b>32.7</b> worldcloud – website</a></li>
<li class="chapter" data-level="32.8" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-text"><i class="fa fa-check"></i><b>32.8</b> worldcloud – text</a></li>
<li class="chapter" data-level="32.9" data-path="miscellaneous.html"><a href="miscellaneous.html#miscellaneous-1"><i class="fa fa-check"></i><b>32.9</b> Miscellaneous</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics Workshop Gitbook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="getting-biological-data-from-public-repositories" class="section level1" number="5">
<h1><span class="header-section-number">Session 5</span> – Getting Biological Data from Public Repositories</h1>
<div id="ncbi-databases-genbank-and-others" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> NCBI databases: GenBank and others</h2>
<p><strong>1)</strong> The <a href="https://www.ncbi.nlm.nih.gov/" target="_blank"><strong>NCBI</strong></a> (National Center for Biotechnology Information) is likely the ultimate source of public information that includes nucleotide sequences (including genomes, transcriptomes) and protein data. The NCBI goal, as is stated, is to advance science and health by providing access to biomedical and genomic information. Therefore, at some point during any your research (if that involves DNA, RNA or proteins), you will visit NCBI to retrieve or deposit molecular data. Likewise, you can access through the NCBI website an extensive literature repository named <a href="https://pubmed.ncbi.nlm.nih.gov/" target="_blank"><strong>PubMed</strong></a>. This archive comprises more than 30 million citations for biomedical literature from MEDLINE, life science journals, and online books. Citations may include links to full-text content from PubMed Central and publisher web sites.
During this class, we will focus on only <strong>few</strong> aspects of the NCBI and mostly relevant to retrieve sample data (or your own project data).</p>
<p><strong>2)</strong> We can start by exploring data deposited from a poison frog <a href="https://amphibiaweb.org/species/1575" target="_blank"><em>Allobates kingsburyi</em></a>. For this purpose, we type (or copy-and-paste) the name of this species on the search tab.</p>
<p><img src="figures/session_5_figs/1_NCBI_search_bar.png" width="100%" /></p>
<p>For this taxon, NCBI will return results found in 6 databases (1 error). In the <strong>Literature</strong> database, it will return 3 results in the <strong>PubMed Central</strong> and it will provide link to the <strong>PMC</strong> database that contains 3 publications that include this species name. In the <strong>Genes</strong> database, it will return 12 results in the <strong>PopSet</strong> and it will provide link to the <strong>PopSet</strong> database that contains a collection sequences associated to this species name.
In the <strong>Proteins</strong> database, it will return 11 results in both the <strong>Identical Protein Groups</strong> and <strong>Protein</strong>. The latter will provide link to the <strong>Protein</strong> database that contains a collection 11 amino acid sequences associated to this species name. In the <strong>Genomes</strong> database, it will return 17 results in the <strong>Nucleotide</strong> and it will provide link to the <strong>Nucleotide</strong> database that contains a collection 17 nucleotide sequences associated to this species name. Likewise, <strong>Genomes</strong> database will also return 1 result in the <strong>Taxonomy</strong> and it will provide link to the <strong>Taxonomy</strong> database that contains the taxonomic link to the species that we are searching <em>Allobates kingsburyi</em>.</p>
<p><strong>3)</strong> For a model system like <a href="https://amphibiaweb.org/species/5255" target="_blank"><em>Xenopus laevis</em></a>, we will get far more results on our NCBI search. In this case, you will get an access to the genome assembly of this species (i.e., Xenopus_laevis_v2). Some other highlights include that this taxon has more than 43,660 associated publications, gene expression omnibus (GEO) has 184K entries, 149K protein entries, ~1.5M nucleotide entries, 2,249 SRA (NGS: sequence read archive) entries, and much more. This example provides the extent of the NCBI as central and extensive archive of biological information.</p>
<p><strong>4)</strong> We will explore NCBI nucleotide database to understand how the R-package <a href="https://cran.r-project.org/web/packages/rentrez/vignettes/rentrez_tutorial.html" target="_blank"><strong>rentrez</strong></a> can retrieve data as data frames in the R environment.</p>
<p>We will collect some acetylcholine receptor <strong>CHRNB2</strong> data of poison frogs. For this purpose, we need to retrieve the NCBI accession numbers. Each of these is a unique identifier for a sequence record and it applies to the complete record. These ID numbers are usually a combination of a letter(s) and numbers. Versions on these accession numbers are indicated by a number after a period, e.g., <code>MF580102.1</code> will be version one of this entry.</p>
<p>For our example taxon, we first search <em>Allobates kingsburyi</em> on the NCBI search tab.</p>
<p><img src="figures/session_5_figs/2_NCBI_allobates_kingsburyi.png" width="100%" /></p>
<p>Next, we click on <strong>PopSet</strong> under the <strong>Genes</strong> menu.</p>
<p><img src="figures/session_5_figs/3_NCBI_popset.png" width="40%" /></p>
<p>This will take us to a collection of studies that include sequences of our target species <em>Allobates kingsburyi</em>. We are interested on <strong>Hyloidea nicotinic acetylcholine receptor beta-2 (chrnb2) gene, complete cds</strong>.</p>
<p><img src="figures/session_5_figs/4_NCBI_chrnb2.png" width="100%" /></p>
<p>This is a phylogenetic study that includes 30 sequences of mostly frogs that include sequences of <em>Allobates kingsburyi</em>. We will click on the link to get the individual accession numbers and information on the associated publication.</p>
<p><img src="figures/session_5_figs/5_NCBI_summary_chrnb2.png" width="100%" /></p>
<p>Notice that the <code>PopSet</code> number of our focal study is <code>1248341763</code>. However, we can copy single accession numbers (e.g., <code>MF580109.1</code>, <code>MF580108.1</code>, <code>MF580107.1</code>, etc) and can be used to download individually when we use the R-package <strong>rentrez</strong>. Likewise, you can get the fasta sequences by clickling on the FASTA link.</p>
<p><img src="figures/session_5_figs/6_FASTA_seqs.png" width="100%" /></p>
<p>If you selected to see on the browser the FASTA sequences. You can copy and paste the output text in a file to import these sequences into R at a later time.</p>
<p><img src="figures/session_5_figs/7_FASTA_seq_to_copy.png" width="100%" /></p>
</div>
<div id="accessing-ncbi-within-r-rentrez" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Accessing NCBI within R: <code>rentrez</code></h2>
<p><strong>5)</strong> The above approach could be tedious and depended on lots of copy-and-paste actions. However, this can be streamlined using <strong>rentrez</strong> as long as you know what accession numbers you want to download.</p>
<p>We will proceed to get all the sequences associated to the corresponding poison frog PopSet <code>1248341763</code>. For this purpose, we start by installing (if necesseray) or load <strong>rentrez</strong> in to the R environment.</p>
<div class="sourceCode" id="cb223"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb223-1"><a href="getting-biological-data-from-public-repositories.html#cb223-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We have downloaded this R-package before, so you just need to load it in the R environment.</span></span>
<span id="cb223-2"><a href="getting-biological-data-from-public-repositories.html#cb223-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rentrez)</span>
<span id="cb223-3"><a href="getting-biological-data-from-public-repositories.html#cb223-3" aria-hidden="true" tabindex="-1"></a><span class="do">## We define our working folder, where we can download our retrieved sequences -- THIS IS EXCLUSIVE FOR YOUR COMPUTER AND IT IS NOT THE PATH SHOWN BELOW</span></span>
<span id="cb223-4"><a href="getting-biological-data-from-public-repositories.html#cb223-4" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/my_working_directory&quot;</span>)</span></code></pre></div>
<p>Second, we can get an idea of the databases that <strong>rentrez</strong> can access from the NCBI using the EUtils API. As mentioned on <strong>1)</strong>, the NCBI provides extensive documentation for each of their databases. The information about these databases can be easily accessed using the function <code>entrez_dbs()</code> of <strong>rentrez</strong>.</p>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb224-1"><a href="getting-biological-data-from-public-repositories.html#cb224-1" aria-hidden="true" tabindex="-1"></a><span class="fu">entrez_dbs</span>()</span>
<span id="cb224-2"><a href="getting-biological-data-from-public-repositories.html#cb224-2" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;pubmed&quot;          &quot;protein&quot;         &quot;nuccore&quot;         &quot;ipg&quot;             &quot;nucleotide&quot;      &quot;structure&quot;      </span></span>
<span id="cb224-3"><a href="getting-biological-data-from-public-repositories.html#cb224-3" aria-hidden="true" tabindex="-1"></a><span class="co"># [7] &quot;genome&quot;          &quot;annotinfo&quot;       &quot;assembly&quot;        &quot;bioproject&quot;      &quot;biosample&quot;       &quot;blastdbinfo&quot;    </span></span>
<span id="cb224-4"><a href="getting-biological-data-from-public-repositories.html#cb224-4" aria-hidden="true" tabindex="-1"></a><span class="co">#[13] &quot;books&quot;           &quot;cdd&quot;             &quot;clinvar&quot;         &quot;gap&quot;             &quot;gapplus&quot;         &quot;grasp&quot;          </span></span>
<span id="cb224-5"><a href="getting-biological-data-from-public-repositories.html#cb224-5" aria-hidden="true" tabindex="-1"></a><span class="co">#[19] &quot;dbvar&quot;           &quot;gene&quot;            &quot;gds&quot;             &quot;geoprofiles&quot;     &quot;homologene&quot;      &quot;medgen&quot;         </span></span>
<span id="cb224-6"><a href="getting-biological-data-from-public-repositories.html#cb224-6" aria-hidden="true" tabindex="-1"></a><span class="co">#[25] &quot;mesh&quot;            &quot;ncbisearch&quot;      &quot;nlmcatalog&quot;      &quot;omim&quot;            &quot;orgtrack&quot;        &quot;pmc&quot;            </span></span>
<span id="cb224-7"><a href="getting-biological-data-from-public-repositories.html#cb224-7" aria-hidden="true" tabindex="-1"></a><span class="co">#[31] &quot;popset&quot;          &quot;proteinclusters&quot; &quot;pcassay&quot;         &quot;protfam&quot;         &quot;biosystems&quot;      &quot;pccompound&quot;     </span></span>
<span id="cb224-8"><a href="getting-biological-data-from-public-repositories.html#cb224-8" aria-hidden="true" tabindex="-1"></a><span class="co">#[37] &quot;pcsubstance&quot;     &quot;seqannot&quot;        &quot;snp&quot;             &quot;sra&quot;             &quot;taxonomy&quot;        &quot;biocollections&quot; </span></span>
<span id="cb224-9"><a href="getting-biological-data-from-public-repositories.html#cb224-9" aria-hidden="true" tabindex="-1"></a><span class="co">#[43] &quot;gtr&quot;</span></span></code></pre></div>
<p>We can access at least 43 databases. To download the acetylcholine receptor dataset, we know that this is a <strong>PopSet</strong> with a accession number <code>1248341763</code>. We notice that this database is present as <code>"popset"</code> is an output of <code>entrez_dbs()</code> . Moreover, we can explore this database with three functions of <strong>rentrez:</strong> <code>entrez_db_summary()</code>, <code>entrez_db_searchable()</code> and <code>entrez_db_links()</code>.</p>
<div class="sourceCode" id="cb225"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb225-1"><a href="getting-biological-data-from-public-repositories.html#cb225-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Brief description of what the database is</span></span>
<span id="cb225-2"><a href="getting-biological-data-from-public-repositories.html#cb225-2" aria-hidden="true" tabindex="-1"></a><span class="fu">entrez_db_summary</span>(<span class="st">&quot;popset&quot;</span>)</span>
<span id="cb225-3"><a href="getting-biological-data-from-public-repositories.html#cb225-3" aria-hidden="true" tabindex="-1"></a><span class="co">#DbName: popset</span></span>
<span id="cb225-4"><a href="getting-biological-data-from-public-repositories.html#cb225-4" aria-hidden="true" tabindex="-1"></a><span class="co">#MenuName: PopSet</span></span>
<span id="cb225-5"><a href="getting-biological-data-from-public-repositories.html#cb225-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Description: PopSet sequence record</span></span>
<span id="cb225-6"><a href="getting-biological-data-from-public-repositories.html#cb225-6" aria-hidden="true" tabindex="-1"></a><span class="co">#DbBuild: Build210116-1232m.1</span></span>
<span id="cb225-7"><a href="getting-biological-data-from-public-repositories.html#cb225-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Count: 356706</span></span>
<span id="cb225-8"><a href="getting-biological-data-from-public-repositories.html#cb225-8" aria-hidden="true" tabindex="-1"></a><span class="co">#LastUpdate: 2021/01/16 14:57 </span></span>
<span id="cb225-9"><a href="getting-biological-data-from-public-repositories.html#cb225-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb225-10"><a href="getting-biological-data-from-public-repositories.html#cb225-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Set of search terms that can used with this database</span></span>
<span id="cb225-11"><a href="getting-biological-data-from-public-repositories.html#cb225-11" aria-hidden="true" tabindex="-1"></a><span class="fu">entrez_db_searchable</span>(<span class="st">&quot;popset&quot;</span>)</span>
<span id="cb225-12"><a href="getting-biological-data-from-public-repositories.html#cb225-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Searchable fields for database &#39;popset&#39;</span></span>
<span id="cb225-13"><a href="getting-biological-data-from-public-repositories.html#cb225-13" aria-hidden="true" tabindex="-1"></a><span class="co"># ALL    All terms from all searchable fields </span></span>
<span id="cb225-14"><a href="getting-biological-data-from-public-repositories.html#cb225-14" aria-hidden="true" tabindex="-1"></a><span class="co"># UID    Unique number assigned to each sequence </span></span>
<span id="cb225-15"><a href="getting-biological-data-from-public-repositories.html#cb225-15" aria-hidden="true" tabindex="-1"></a><span class="co"># FILT   Limits the records </span></span>
<span id="cb225-16"><a href="getting-biological-data-from-public-repositories.html#cb225-16" aria-hidden="true" tabindex="-1"></a><span class="co"># WORD   Free text associated with record </span></span>
<span id="cb225-17"><a href="getting-biological-data-from-public-repositories.html#cb225-17" aria-hidden="true" tabindex="-1"></a><span class="co"># TITL   Words in definition line </span></span>
<span id="cb225-18"><a href="getting-biological-data-from-public-repositories.html#cb225-18" aria-hidden="true" tabindex="-1"></a><span class="co"># KYWD   Nonstandardized terms provided by submitter </span></span>
<span id="cb225-19"><a href="getting-biological-data-from-public-repositories.html#cb225-19" aria-hidden="true" tabindex="-1"></a><span class="co"># AUTH   Author(s) of publication </span></span>
<span id="cb225-20"><a href="getting-biological-data-from-public-repositories.html#cb225-20" aria-hidden="true" tabindex="-1"></a><span class="co"># JOUR   Journal abbreviation of publication </span></span>
<span id="cb225-21"><a href="getting-biological-data-from-public-repositories.html#cb225-21" aria-hidden="true" tabindex="-1"></a><span class="co"># VOL    Volume number of publication </span></span>
<span id="cb225-22"><a href="getting-biological-data-from-public-repositories.html#cb225-22" aria-hidden="true" tabindex="-1"></a><span class="co"># ISS    Issue number of publication </span></span>
<span id="cb225-23"><a href="getting-biological-data-from-public-repositories.html#cb225-23" aria-hidden="true" tabindex="-1"></a><span class="co"># PAGE   Page number(s) of publication </span></span>
<span id="cb225-24"><a href="getting-biological-data-from-public-repositories.html#cb225-24" aria-hidden="true" tabindex="-1"></a><span class="co"># ORGN   Scientific and common names of organism, and all higher levels of taxonomy </span></span>
<span id="cb225-25"><a href="getting-biological-data-from-public-repositories.html#cb225-25" aria-hidden="true" tabindex="-1"></a><span class="co"># ACCN   Accession number of sequence </span></span>
<span id="cb225-26"><a href="getting-biological-data-from-public-repositories.html#cb225-26" aria-hidden="true" tabindex="-1"></a><span class="co"># PACC   Does not include retired secondary accessions </span></span>
<span id="cb225-27"><a href="getting-biological-data-from-public-repositories.html#cb225-27" aria-hidden="true" tabindex="-1"></a><span class="co"># GENE   Name of gene associated with sequence </span></span>
<span id="cb225-28"><a href="getting-biological-data-from-public-repositories.html#cb225-28" aria-hidden="true" tabindex="-1"></a><span class="co"># PROT   Name of protein associated with sequence </span></span>
<span id="cb225-29"><a href="getting-biological-data-from-public-repositories.html#cb225-29" aria-hidden="true" tabindex="-1"></a><span class="co"># ECNO   EC number for enzyme or CAS registry number </span></span>
<span id="cb225-30"><a href="getting-biological-data-from-public-repositories.html#cb225-30" aria-hidden="true" tabindex="-1"></a><span class="co"># PDAT   Date sequence added to GenBank </span></span>
<span id="cb225-31"><a href="getting-biological-data-from-public-repositories.html#cb225-31" aria-hidden="true" tabindex="-1"></a><span class="co"># MDAT   Date of last update </span></span>
<span id="cb225-32"><a href="getting-biological-data-from-public-repositories.html#cb225-32" aria-hidden="true" tabindex="-1"></a><span class="co"># SUBS   CAS chemical name or MEDLINE Substance Name </span></span>
<span id="cb225-33"><a href="getting-biological-data-from-public-repositories.html#cb225-33" aria-hidden="true" tabindex="-1"></a><span class="co"># PROP   Classification by source qualifiers and molecule type </span></span>
<span id="cb225-34"><a href="getting-biological-data-from-public-repositories.html#cb225-34" aria-hidden="true" tabindex="-1"></a><span class="co"># SQID   String identifier for sequence </span></span>
<span id="cb225-35"><a href="getting-biological-data-from-public-repositories.html#cb225-35" aria-hidden="true" tabindex="-1"></a><span class="co"># GPRJ   BioProject </span></span>
<span id="cb225-36"><a href="getting-biological-data-from-public-repositories.html#cb225-36" aria-hidden="true" tabindex="-1"></a><span class="co"># FKEY   Feature annotated on sequence </span></span>
<span id="cb225-37"><a href="getting-biological-data-from-public-repositories.html#cb225-37" aria-hidden="true" tabindex="-1"></a><span class="co"># PCNT   Number of proteins in the set </span></span>
<span id="cb225-38"><a href="getting-biological-data-from-public-repositories.html#cb225-38" aria-hidden="true" tabindex="-1"></a><span class="co"># NCNT   Number of nucleotides in the set </span></span>
<span id="cb225-39"><a href="getting-biological-data-from-public-repositories.html#cb225-39" aria-hidden="true" tabindex="-1"></a><span class="co"># STRN   Strain </span></span>
<span id="cb225-40"><a href="getting-biological-data-from-public-repositories.html#cb225-40" aria-hidden="true" tabindex="-1"></a><span class="co"># ISOL   Isolate </span></span>
<span id="cb225-41"><a href="getting-biological-data-from-public-repositories.html#cb225-41" aria-hidden="true" tabindex="-1"></a><span class="co"># CULT   Cultivar </span></span>
<span id="cb225-42"><a href="getting-biological-data-from-public-repositories.html#cb225-42" aria-hidden="true" tabindex="-1"></a><span class="co"># BRD    Breed </span></span>
<span id="cb225-43"><a href="getting-biological-data-from-public-repositories.html#cb225-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb225-44"><a href="getting-biological-data-from-public-repositories.html#cb225-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Set of databases that might contain linked records    </span></span>
<span id="cb225-45"><a href="getting-biological-data-from-public-repositories.html#cb225-45" aria-hidden="true" tabindex="-1"></a><span class="fu">entrez_db_links</span>(<span class="st">&quot;popset&quot;</span>)   </span>
<span id="cb225-46"><a href="getting-biological-data-from-public-repositories.html#cb225-46" aria-hidden="true" tabindex="-1"></a><span class="co">#Databases with linked records for database &#39;popset&#39;</span></span>
<span id="cb225-47"><a href="getting-biological-data-from-public-repositories.html#cb225-47" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] biocollections bioproject     nuccore        pmc            popset         protein        pubmed        </span></span>
<span id="cb225-48"><a href="getting-biological-data-from-public-repositories.html#cb225-48" aria-hidden="true" tabindex="-1"></a><span class="co">#[8] taxonomy      </span></span></code></pre></div>
<p>We have the required information to fetch the PopSet <code>1248341763</code> in the fasta format. For that purpose we use the function <code>entrez_fetch()</code>.</p>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb226-1"><a href="getting-biological-data-from-public-repositories.html#cb226-1" aria-hidden="true" tabindex="-1"></a>acetylcholine_receptor_data <span class="ot">&lt;-</span> <span class="fu">entrez_fetch</span>(<span class="at">db =</span> <span class="st">&quot;popset&quot;</span>,</span>
<span id="cb226-2"><a href="getting-biological-data-from-public-repositories.html#cb226-2" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">id =</span> <span class="dv">1248341763</span>,</span>
<span id="cb226-3"><a href="getting-biological-data-from-public-repositories.html#cb226-3" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">rettype =</span> <span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb226-4"><a href="getting-biological-data-from-public-repositories.html#cb226-4" aria-hidden="true" tabindex="-1"></a>acetylcholine_receptor_data</span>
<span id="cb226-5"><a href="getting-biological-data-from-public-repositories.html#cb226-5" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;&gt;MF580080.1 Espadarana callistomma nicotinic acetylcholine receptor beta-2 (chrnb2) gene, complete cds\nATGACGGTTCTCCTC...&quot;</span></span></code></pre></div>
<p>Finally, we can save this as a character vector in a text file that will exactly as we did in <strong>4)</strong> using the function <code>write()</code>.</p>
<div class="sourceCode" id="cb227"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb227-1"><a href="getting-biological-data-from-public-repositories.html#cb227-1" aria-hidden="true" tabindex="-1"></a>name_of_file <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;acetylcholine_receptor_popset_1248341763.txt&quot;</span>)</span>
<span id="cb227-2"><a href="getting-biological-data-from-public-repositories.html#cb227-2" aria-hidden="true" tabindex="-1"></a>name_of_file</span>
<span id="cb227-3"><a href="getting-biological-data-from-public-repositories.html#cb227-3" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;acetylcholine_receptor_popset_1248341763.txt&quot;</span></span>
<span id="cb227-4"><a href="getting-biological-data-from-public-repositories.html#cb227-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(acetylcholine_receptor_data, <span class="at">file =</span> name_of_file)</span></code></pre></div>
<p><strong>6)</strong> It is important to reiterate that <strong>rentrez</strong> can get us more information associated to our focal dataset: PopSet <code>1248341763</code>. Here are some interesting explorations on this accession number.</p>
<p>We can get the associated data link to the PopSet <code>1248341763</code>. Using the function <code>entrez_link()</code></p>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb228-1"><a href="getting-biological-data-from-public-repositories.html#cb228-1" aria-hidden="true" tabindex="-1"></a>my_popset_links <span class="ot">&lt;-</span> <span class="fu">entrez_link</span>(<span class="at">dbfrom=</span><span class="st">&#39;popset&#39;</span>, <span class="at">id=</span><span class="dv">1248341763</span>, <span class="at">db=</span><span class="st">&#39;all&#39;</span>)</span>
<span id="cb228-2"><a href="getting-biological-data-from-public-repositories.html#cb228-2" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(my_popset_links)</span>
<span id="cb228-3"><a href="getting-biological-data-from-public-repositories.html#cb228-3" aria-hidden="true" tabindex="-1"></a><span class="co">#List of 2</span></span>
<span id="cb228-4"><a href="getting-biological-data-from-public-repositories.html#cb228-4" aria-hidden="true" tabindex="-1"></a><span class="co">#$ links:List of 7</span></span>
<span id="cb228-5"><a href="getting-biological-data-from-public-repositories.html#cb228-5" aria-hidden="true" tabindex="-1"></a><span class="co">#  ..$ popset_nuccore      : chr [1:30] &quot;1248341821&quot; &quot;1248341819&quot; &quot;1248341817&quot; &quot;1248341815&quot; ...</span></span>
<span id="cb228-6"><a href="getting-biological-data-from-public-repositories.html#cb228-6" aria-hidden="true" tabindex="-1"></a><span class="co"># ..$ popset_pmc          : chr &quot;5834227&quot;</span></span>
<span id="cb228-7"><a href="getting-biological-data-from-public-repositories.html#cb228-7" aria-hidden="true" tabindex="-1"></a><span class="co">#  ..$ popset_popset       : chr [1:3] &quot;1248759970&quot; &quot;1248759962&quot; &quot;1248341823&quot;</span></span>
<span id="cb228-8"><a href="getting-biological-data-from-public-repositories.html#cb228-8" aria-hidden="true" tabindex="-1"></a><span class="co"># ..$ popset_protein      : chr [1:30] &quot;1248341822&quot; &quot;1248341820&quot; &quot;1248341818&quot; &quot;1248341816&quot; ...</span></span>
<span id="cb228-9"><a href="getting-biological-data-from-public-repositories.html#cb228-9" aria-hidden="true" tabindex="-1"></a><span class="co"># ..$ popset_pubmed       : chr &quot;28935799&quot;</span></span>
<span id="cb228-10"><a href="getting-biological-data-from-public-repositories.html#cb228-10" aria-hidden="true" tabindex="-1"></a><span class="co"># ..$ popset_taxonomy     : chr [1:63] &quot;2648769&quot; &quot;2483034&quot; &quot;2023954&quot; &quot;1004470&quot; ...</span></span>
<span id="cb228-11"><a href="getting-biological-data-from-public-repositories.html#cb228-11" aria-hidden="true" tabindex="-1"></a><span class="co"># ..$ popset_taxonomy_tree: chr [1:63] &quot;2648769&quot; &quot;2483034&quot; &quot;2023954&quot; &quot;1004470&quot; ...</span></span>
<span id="cb228-12"><a href="getting-biological-data-from-public-repositories.html#cb228-12" aria-hidden="true" tabindex="-1"></a><span class="co"># ..- attr(*, &quot;class&quot;)= chr [1:2] &quot;elink_classic&quot; &quot;list&quot;</span></span>
<span id="cb228-13"><a href="getting-biological-data-from-public-repositories.html#cb228-13" aria-hidden="true" tabindex="-1"></a><span class="co">#$ file :Classes &#39;XMLInternalElementNode&#39;, &#39;XMLInternalNode&#39;, &#39;XMLAbstractNode&#39; &lt;externalptr&gt; </span></span>
<span id="cb228-14"><a href="getting-biological-data-from-public-repositories.html#cb228-14" aria-hidden="true" tabindex="-1"></a><span class="co">#- attr(*, &quot;content&quot;)= chr &quot; $links: IDs for linked records from NCBI\n &quot;</span></span>
<span id="cb228-15"><a href="getting-biological-data-from-public-repositories.html#cb228-15" aria-hidden="true" tabindex="-1"></a><span class="co">#- attr(*, &quot;class&quot;)= chr [1:2] &quot;elink&quot; &quot;list&quot;</span></span></code></pre></div>
<p>We see that the literature database <code>popset_pmc</code> has one entry and this is likely the publication associated with PopSet <code>1248341763</code>. We can retrieve this information with the function <code>entrez_fetch()</code>. In this case, we will use a complex notation to retrieve a pmc dataset as <code>my_popset_links$links$popset_pmc</code>. This indicates that with the list named <code>my_popset_links</code>, we want to retrieve the list named <code>links</code> and the element named <code>popset_pmc</code>. When ready, we will use the function <code>entrez_fetch()</code> with the return format as <code>rettype="native"</code>.</p>
<div class="sourceCode" id="cb229"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb229-1"><a href="getting-biological-data-from-public-repositories.html#cb229-1" aria-hidden="true" tabindex="-1"></a>my_popset_links<span class="sc">$</span>links<span class="sc">$</span>popset_pmc</span>
<span id="cb229-2"><a href="getting-biological-data-from-public-repositories.html#cb229-2" aria-hidden="true" tabindex="-1"></a>pmc_ids <span class="ot">&lt;-</span> my_popset_links<span class="sc">$</span>links<span class="sc">$</span>popset_pmc</span>
<span id="cb229-3"><a href="getting-biological-data-from-public-repositories.html#cb229-3" aria-hidden="true" tabindex="-1"></a>pmc_ids</span>
<span id="cb229-4"><a href="getting-biological-data-from-public-repositories.html#cb229-4" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;5834227&quot;</span></span>
<span id="cb229-5"><a href="getting-biological-data-from-public-repositories.html#cb229-5" aria-hidden="true" tabindex="-1"></a>my_literature <span class="ot">&lt;-</span> <span class="fu">entrez_fetch</span>(<span class="at">db =</span> <span class="st">&quot;pmc&quot;</span>, <span class="at">id =</span> pmc_ids, <span class="at">rettype=</span><span class="st">&quot;native&quot;</span>)</span>
<span id="cb229-6"><a href="getting-biological-data-from-public-repositories.html#cb229-6" aria-hidden="true" tabindex="-1"></a>my_literature</span>
<span id="cb229-7"><a href="getting-biological-data-from-public-repositories.html#cb229-7" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;1:  Interacting Amino Acid Replacements Allow Poison Frogs to Evolve Epibatidine Resistance\nRebecca D. Tarvin, Cecilia M. Borghese, Wiebke Sachs, Juan C. Santos, Ying Lu, Lauren A. O’Connell, David C. Cannatella, R. Adron Harris, Harold H. Zakon\nScience. Author manuscript; available in PMC 2018 Mar 22.Published in final edited form as: Science. 2017 Sep 22; 357(6357): 1261–1266.  doi: 10.1126/science.aan5061\nPMCID: PMC5834227\n\n&quot;</span></span>
<span id="cb229-8"><a href="getting-biological-data-from-public-repositories.html#cb229-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(my_literature)</span>
<span id="cb229-9"><a href="getting-biological-data-from-public-repositories.html#cb229-9" aria-hidden="true" tabindex="-1"></a><span class="co">#1:  Interacting Amino Acid Replacements Allow Poison Frogs to Evolve Epibatidine Resistance</span></span>
<span id="cb229-10"><a href="getting-biological-data-from-public-repositories.html#cb229-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Rebecca D. Tarvin, Cecilia M. Borghese, Wiebke Sachs, Juan C. Santos, Ying Lu, Lauren A. O’Connell, David C. Cannatella, R. Adron Harris, Harold H. Zakon</span></span>
<span id="cb229-11"><a href="getting-biological-data-from-public-repositories.html#cb229-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Science. Author manuscript; available in PMC 2018 Mar 22.Published in final edited form as: Science. 2017 Sep 22; 357(6357): 1261–1266.  doi: 10.1126/science.aan5061</span></span>
<span id="cb229-12"><a href="getting-biological-data-from-public-repositories.html#cb229-12" aria-hidden="true" tabindex="-1"></a><span class="co">#PMCID: PMC5834227</span></span></code></pre></div>
<p>We can also retrieve the amino acid sequence associated with the sequences in our PopSet <code>1248341763</code> and save it in a text file using fasta format.</p>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb230-1"><a href="getting-biological-data-from-public-repositories.html#cb230-1" aria-hidden="true" tabindex="-1"></a>my_popset_links<span class="sc">$</span>links<span class="sc">$</span>popset_protein</span>
<span id="cb230-2"><a href="getting-biological-data-from-public-repositories.html#cb230-2" aria-hidden="true" tabindex="-1"></a>protein_ids <span class="ot">&lt;-</span> my_popset_links<span class="sc">$</span>links<span class="sc">$</span>popset_protein</span>
<span id="cb230-3"><a href="getting-biological-data-from-public-repositories.html#cb230-3" aria-hidden="true" tabindex="-1"></a>protein_ids</span>
<span id="cb230-4"><a href="getting-biological-data-from-public-repositories.html#cb230-4" aria-hidden="true" tabindex="-1"></a><span class="co">#[[1] &quot;1248341822&quot; &quot;1248341820&quot; &quot;1248341818&quot; &quot;1248341816&quot; &quot;1248341814&quot; &quot;1248341812&quot; &quot;1248341810&quot; &quot;1248341808&quot;</span></span>
<span id="cb230-5"><a href="getting-biological-data-from-public-repositories.html#cb230-5" aria-hidden="true" tabindex="-1"></a><span class="co"># [9] &quot;1248341806&quot; &quot;1248341804&quot; &quot;1248341802&quot; &quot;1248341800&quot; &quot;1248341798&quot; &quot;1248341796&quot; &quot;1248341794&quot; &quot;1248341792&quot;</span></span>
<span id="cb230-6"><a href="getting-biological-data-from-public-repositories.html#cb230-6" aria-hidden="true" tabindex="-1"></a><span class="co">#[17] &quot;1248341790&quot; &quot;1248341788&quot; &quot;1248341786&quot; &quot;1248341784&quot; &quot;1248341782&quot; &quot;1248341780&quot; &quot;1248341778&quot; &quot;1248341776&quot;</span></span>
<span id="cb230-7"><a href="getting-biological-data-from-public-repositories.html#cb230-7" aria-hidden="true" tabindex="-1"></a><span class="co">#[25] &quot;1248341774&quot; &quot;1248341772&quot; &quot;1248341770&quot; &quot;1248341768&quot; &quot;1248341766&quot; &quot;1248341764&quot;</span></span>
<span id="cb230-8"><a href="getting-biological-data-from-public-repositories.html#cb230-8" aria-hidden="true" tabindex="-1"></a>my_popset_protein <span class="ot">&lt;-</span> <span class="fu">entrez_fetch</span>(<span class="at">db =</span> <span class="st">&quot;protein&quot;</span>, <span class="at">id =</span> protein_ids, <span class="at">rettype=</span><span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb230-9"><a href="getting-biological-data-from-public-repositories.html#cb230-9" aria-hidden="true" tabindex="-1"></a>my_popset_protein</span>
<span id="cb230-10"><a href="getting-biological-data-from-public-repositories.html#cb230-10" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;&gt;ATG31811.1 nicotinic acetylcholine receptor beta-2, partial [Phyllobates aurotaenia]\nADGMYEVSFYSNAVVSHDGSIFWLPP...&quot; </span></span>
<span id="cb230-11"><a href="getting-biological-data-from-public-repositories.html#cb230-11" aria-hidden="true" tabindex="-1"></a>name_of_file <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;acetylcholine_receptor_popset_1248341763_protein.txt&quot;</span>)</span>
<span id="cb230-12"><a href="getting-biological-data-from-public-repositories.html#cb230-12" aria-hidden="true" tabindex="-1"></a>name_of_file</span>
<span id="cb230-13"><a href="getting-biological-data-from-public-repositories.html#cb230-13" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;acetylcholine_receptor_popset_1248341763_protein.txt&quot;</span></span>
<span id="cb230-14"><a href="getting-biological-data-from-public-repositories.html#cb230-14" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(my_popset_protein, <span class="at">file =</span> name_of_file)</span></code></pre></div>
<p><strong>7)</strong> To import these data back to R , we need the path to the files that contain the data using the R-package <a href="https://bioconductor.org/packages/release/bioc/html/Biostrings.html" target="_blank"><strong>Biostrings</strong></a>.</p>
<p>We can import the nucleotide sequences as DNAstrings.</p>
<div class="sourceCode" id="cb231"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb231-1"><a href="getting-biological-data-from-public-repositories.html#cb231-1" aria-hidden="true" tabindex="-1"></a><span class="do">## you might have already installed the library Biostrings</span></span>
<span id="cb231-2"><a href="getting-biological-data-from-public-repositories.html#cb231-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span>
<span id="cb231-3"><a href="getting-biological-data-from-public-repositories.html#cb231-3" aria-hidden="true" tabindex="-1"></a><span class="do">## get the path to the file with the nucleotide sequences -- THIS IS EXCLUSIVE FOR YOUR COMPUTER AND IT IS NOT THE PATH SHOWN BELOW</span></span>
<span id="cb231-4"><a href="getting-biological-data-from-public-repositories.html#cb231-4" aria-hidden="true" tabindex="-1"></a>my_popset_as_dna_stringset <span class="ot">&lt;-</span> <span class="fu">readDNAStringSet</span>(<span class="at">filepath =</span> <span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/my_working_directory/acetylcholine_receptor_popset_1248341763.txt&quot;</span>, </span>
<span id="cb231-5"><a href="getting-biological-data-from-public-repositories.html#cb231-5" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">format =</span> <span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb231-6"><a href="getting-biological-data-from-public-repositories.html#cb231-6" aria-hidden="true" tabindex="-1"></a>my_popset_as_dna_stringset</span>
<span id="cb231-7"><a href="getting-biological-data-from-public-repositories.html#cb231-7" aria-hidden="true" tabindex="-1"></a><span class="co"># A DNAStringSet instance of length 30</span></span>
<span id="cb231-8"><a href="getting-biological-data-from-public-repositories.html#cb231-8" aria-hidden="true" tabindex="-1"></a><span class="co">#    width seq                                                                                      names               </span></span>
<span id="cb231-9"><a href="getting-biological-data-from-public-repositories.html#cb231-9" aria-hidden="true" tabindex="-1"></a><span class="co"># [1]  1506 ATGACGGTTCTCCTCNNNNNNNNNNNNCTCGGTCTGCTCGGCC...ACGTTATTACAGTTGAACCATGCAGCTCCCGCCTCCAACTAA MF580080.1 Espada...</span></span>
<span id="cb231-10"><a href="getting-biological-data-from-public-repositories.html#cb231-10" aria-hidden="true" tabindex="-1"></a><span class="co"># [2]  1506 ATGACGGCTCTCCTCCTCGTCCTGCACCTCAGCCTGATCGGCC...ACGTTACTACAGCTCAACCACGCGATTCCTGCCTCCAACTAG MF580081.1 Boana ...</span></span>
<span id="cb231-11"><a href="getting-biological-data-from-public-repositories.html#cb231-11" aria-hidden="true" tabindex="-1"></a><span class="co"># [3]  1506 ATGACGGTTCTCCTCCTCGTCCTGCACCTGAGCCTGCTCGGCC...ACGTTACTGCAGCTCAACCACGCGGCTCCCGCCTCCAACTGA MF580082.1 Incili...</span></span>
<span id="cb231-12"><a href="getting-biological-data-from-public-repositories.html#cb231-12" aria-hidden="true" tabindex="-1"></a><span class="co"># [4]  1515 ATGACGGTTCTCCTCCTCGTCCTGCACCTCAGCCTGCTCGGCC...CAGCTGAACCACGGGGCTCCCGCCTCCAACTAAAGGGGCGCC MF580083.1 Hyloxa...</span></span>
<span id="cb231-13"><a href="getting-biological-data-from-public-repositories.html#cb231-13" aria-hidden="true" tabindex="-1"></a><span class="co"># [5]  1506 ATGACGGTTCTCCTCCTCGTCCTGCACCTGAGCCTGCTCGGCC...ACGTTACTGCAGCTCAACCACGCAGCTCCCGCCTCCAACTGA MF580084.1 Atelop...</span></span>
<span id="cb231-14"><a href="getting-biological-data-from-public-repositories.html#cb231-14" aria-hidden="true" tabindex="-1"></a><span class="co"># ...   ... ...</span></span>
<span id="cb231-15"><a href="getting-biological-data-from-public-repositories.html#cb231-15" aria-hidden="true" tabindex="-1"></a><span class="co">#[26]  1506 ATGGCGGCTCTCCTCNNNCTCCTACACCTCGGCCTGCTCGGCA...ACGTTACTGCAGCTGAACCCCGCAGCTCCCGCCTCTAAGTGA MF580105.1 Gastro...</span></span>
<span id="cb231-16"><a href="getting-biological-data-from-public-repositories.html#cb231-16" aria-hidden="true" tabindex="-1"></a><span class="co">#[27]   134 GCTGATGGGATGTACGAGGTCTCCTTCTACTCCAACGCGGTGG...CGCCTGTAAGATCGAGGTGAAGCACTTTCCGTTCGACCAGCA MF580106.1 Rheoba...</span></span>
<span id="cb231-17"><a href="getting-biological-data-from-public-repositories.html#cb231-17" aria-hidden="true" tabindex="-1"></a><span class="co">#[28]   203 GCTGATGGGATGTATGAGGTCTCCTTCTACTCTAACGCGGTGG...GACTTATGACCGCACTGAGCTGGACCTGGTGCTGAAGAGTGA MF580107.1 Ranito...</span></span>
<span id="cb231-18"><a href="getting-biological-data-from-public-repositories.html#cb231-18" aria-hidden="true" tabindex="-1"></a><span class="co">#[29]  1506 ATGACGGCTCTCCTCCTCGTCCTGCACCTCAGCCTGCTCGGCC...ACGTTAATACAGCTGAACCATGGGACCCCCGCCTCCAACTAA MF580108.1 Phyllo...</span></span>
<span id="cb231-19"><a href="getting-biological-data-from-public-repositories.html#cb231-19" aria-hidden="true" tabindex="-1"></a><span class="co">#[30]   195 GCTGATGGGATGTATGAGGTCTCCTTCTACTCTAACGCGGTGG...CGCTCGTGGACTTATGACCGCACCGAGCTGGACCTGGTGCTG MF580109.1 Phyllo...</span></span></code></pre></div>
<p>We can also import the amino acid sequences as AAstrings.</p>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="getting-biological-data-from-public-repositories.html#cb232-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get the path to the file with the amino acid sequences -- THIS IS EXCLUSIVE FOR YOUR COMPUTER AND IT IS NOT THE PATH SHOWN BELOW</span></span>
<span id="cb232-2"><a href="getting-biological-data-from-public-repositories.html#cb232-2" aria-hidden="true" tabindex="-1"></a>my_popset_as_aa_stringset <span class="ot">&lt;-</span> <span class="fu">readAAStringSet</span>(<span class="at">filepath =</span> <span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/my_working_directory/acetylcholine_receptor_popset_1248341763_protein.txt&quot;</span>, </span>
<span id="cb232-3"><a href="getting-biological-data-from-public-repositories.html#cb232-3" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">format =</span> <span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb232-4"><a href="getting-biological-data-from-public-repositories.html#cb232-4" aria-hidden="true" tabindex="-1"></a>my_popset_as_aa_stringset</span>
<span id="cb232-5"><a href="getting-biological-data-from-public-repositories.html#cb232-5" aria-hidden="true" tabindex="-1"></a><span class="co"># A AAStringSet instance of length 30</span></span>
<span id="cb232-6"><a href="getting-biological-data-from-public-repositories.html#cb232-6" aria-hidden="true" tabindex="-1"></a><span class="co">#    width seq                                                                                      names               </span></span>
<span id="cb232-7"><a href="getting-biological-data-from-public-repositories.html#cb232-7" aria-hidden="true" tabindex="-1"></a><span class="co"># [1]    65 ADGMYEVSFYSNAVVSHDGSIFWLPPAIYKSACKIEVKHFPFDQQNCTMKFRSWTYDRTELDLVL                        ATG31811.1 nicoti...</span></span>
<span id="cb232-8"><a href="getting-biological-data-from-public-repositories.html#cb232-8" aria-hidden="true" tabindex="-1"></a><span class="co"># [2]   501 MTALLLVLHLSLLGLVTRSMGTDTEERLVEFLLDPSRYNKLIR...LFLWIFVFVCVFGTIGMFLQPLFQNYTSNTLIQLNHGTPASN ATG31810.1 nicoti...</span></span>
<span id="cb232-9"><a href="getting-biological-data-from-public-repositories.html#cb232-9" aria-hidden="true" tabindex="-1"></a><span class="co"># [3]    67 ADGMYEVSFYSNAVVSHDGSIFWLPPAIYKSACKIEVKHFPFDQQNCTMKFRSWTYDRTELDLVLKS                      ATG31809.1 nicoti...</span></span>
<span id="cb232-10"><a href="getting-biological-data-from-public-repositories.html#cb232-10" aria-hidden="true" tabindex="-1"></a><span class="co"># [4]    44 ADGMYEVSFYSNAVVSHDGSIFWLPPAIYKSACKIEVKHFPFDQ                                             ATG31808.1 nicoti...</span></span>
<span id="cb232-11"><a href="getting-biological-data-from-public-repositories.html#cb232-11" aria-hidden="true" tabindex="-1"></a><span class="co"># [5]   501 MAALLXLLHLGLLGIVSRCLCTDTEERLVEFLLDPSRYNKLIR...LFLWIFIFVCVFGTIGMFLQPLFQNYTTNTLLQLNPAAPASK ATG31807.1 nicoti...</span></span>
<span id="cb232-12"><a href="getting-biological-data-from-public-repositories.html#cb232-12" aria-hidden="true" tabindex="-1"></a><span class="co"># ...   ... ...</span></span>
<span id="cb232-13"><a href="getting-biological-data-from-public-repositories.html#cb232-13" aria-hidden="true" tabindex="-1"></a><span class="co">#[26]   501 MTVLLLVLHLSLLGLVTRSLGTDTEERLVEFLLDPSRYNKLIR...LFLWVFVFVCVFGTIGMFLQPLFQNYTSNTLLQLNHAAPASN ATG31786.1 nicoti...</span></span>
<span id="cb232-14"><a href="getting-biological-data-from-public-repositories.html#cb232-14" aria-hidden="true" tabindex="-1"></a><span class="co">#[27]   501 MTVLLLVLHLSLLGLVTRSMGTDTEERLVEFLLDPSRYNKLIR...LFLWIFVFVCVFGTIGMFLQPLFQNYTTNTLLQLNHGAPASN ATG31785.1 nicoti...</span></span>
<span id="cb232-15"><a href="getting-biological-data-from-public-repositories.html#cb232-15" aria-hidden="true" tabindex="-1"></a><span class="co">#[28]   501 MTVLLLVLHLSLLGLVTRSLGTDTEERLVEFLLDPSRYNKLIR...LFLWVFVFVCVFGTIGMFLQPLFQNYTSNTLLQLNHAAPASN ATG31784.1 nicoti...</span></span>
<span id="cb232-16"><a href="getting-biological-data-from-public-repositories.html#cb232-16" aria-hidden="true" tabindex="-1"></a><span class="co">#[29]   501 MTALLLVLHLSLIGLVTRILGTDTEERLVEFLLDSSRYNKLIR...LFLWIFVFVCVFGTIGMFLQPLFQNYTTNTLLQLNHAIPASN ATG31783.1 nicoti...</span></span>
<span id="cb232-17"><a href="getting-biological-data-from-public-repositories.html#cb232-17" aria-hidden="true" tabindex="-1"></a><span class="co">#[30]   501 MTVLLXXXXLGLLGLVTRCLXTDTEERLVEFLLDSSRYNKLIR...LFLWIFVFVCVFGTIGMFLQPLFQNYTTNTLLQLNHAAPASN ATG31782.1 nicoti...</span></span></code></pre></div>
</div>
<div id="retrieving-pubmed-references" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Retrieving PubMed references</h2>
<p>We can use <code>rentrez</code> to retrive publication infomration relevant to a term (i.e., text argument) that you can use for citations or references. The main source of such information is <a href="https://pubmed.ncbi.nlm.nih.gov/" target="_blank"><strong>PubMed</strong></a> and it claims to include &gt;30 million citations for biomedical literature from MEDLINE, life science journals, and online books. Likewise, such citations may include links to full text content from PubMed Central and publisher web sites. However, from R, you will be able to access the citation, abstract and other identifiers that you can use later on <a href="https://pubmed.ncbi.nlm.nih.gov/" target="_blank"><strong>PubMed</strong></a> website.</p>
<p><strong>8)</strong> we can use publications relevant to a term of interest, for example, an genus of frogs <em>Ameerega</em> using the following procedure.</p>
<p>We can retrieve the results in the <a href="https://www.nlm.nih.gov/bsd/mms/medlineelements.html" target="_blank"><strong>Medline</strong></a> format as follows.</p>
<div class="sourceCode" id="cb233"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb233-1"><a href="getting-biological-data-from-public-repositories.html#cb233-1" aria-hidden="true" tabindex="-1"></a><span class="do">## make sure that &#39;rentrez&#39; is loaded</span></span>
<span id="cb233-2"><a href="getting-biological-data-from-public-repositories.html#cb233-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(rentrez)</span>
<span id="cb233-3"><a href="getting-biological-data-from-public-repositories.html#cb233-3" aria-hidden="true" tabindex="-1"></a><span class="do">## retrieve IDs from NCBI</span></span>
<span id="cb233-4"><a href="getting-biological-data-from-public-repositories.html#cb233-4" aria-hidden="true" tabindex="-1"></a>genus_publications_IDs <span class="ot">&lt;-</span> <span class="fu">entrez_search</span>(<span class="at">db=</span><span class="st">&quot;pubmed&quot;</span>, <span class="at">term=</span><span class="st">&quot;Ameerega&quot;</span>)</span>
<span id="cb233-5"><a href="getting-biological-data-from-public-repositories.html#cb233-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(genus_publications_IDs)</span>
<span id="cb233-6"><a href="getting-biological-data-from-public-repositories.html#cb233-6" aria-hidden="true" tabindex="-1"></a><span class="co">#List of 5</span></span>
<span id="cb233-7"><a href="getting-biological-data-from-public-repositories.html#cb233-7" aria-hidden="true" tabindex="-1"></a><span class="co"># $ ids             : chr [1:17] &quot;32230685&quot; &quot;31716726&quot; &quot;31586688&quot; &quot;31125660&quot; ...</span></span>
<span id="cb233-8"><a href="getting-biological-data-from-public-repositories.html#cb233-8" aria-hidden="true" tabindex="-1"></a><span class="co"># $ count           : int 17</span></span>
<span id="cb233-9"><a href="getting-biological-data-from-public-repositories.html#cb233-9" aria-hidden="true" tabindex="-1"></a><span class="co"># $ retmax          : int 17</span></span>
<span id="cb233-10"><a href="getting-biological-data-from-public-repositories.html#cb233-10" aria-hidden="true" tabindex="-1"></a><span class="co"># $ QueryTranslation: chr &quot;Ameerega[All Fields]&quot;</span></span>
<span id="cb233-11"><a href="getting-biological-data-from-public-repositories.html#cb233-11" aria-hidden="true" tabindex="-1"></a><span class="co"># $ file            :Classes &#39;XMLInternalDocument&#39;, &#39;XMLAbstractDocument&#39; &lt;externalptr&gt; </span></span>
<span id="cb233-12"><a href="getting-biological-data-from-public-repositories.html#cb233-12" aria-hidden="true" tabindex="-1"></a><span class="co"># - attr(*, &quot;class&quot;)= chr [1:2] &quot;esearch&quot; &quot;list&quot;</span></span>
<span id="cb233-13"><a href="getting-biological-data-from-public-repositories.html#cb233-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb233-14"><a href="getting-biological-data-from-public-repositories.html#cb233-14" aria-hidden="true" tabindex="-1"></a>genus_publications_medline <span class="ot">&lt;-</span> <span class="fu">entrez_fetch</span>(<span class="at">db=</span><span class="st">&quot;pubmed&quot;</span>, <span class="at">id=</span>genus_publications_IDs<span class="sc">$</span>ids, <span class="at">rettype=</span><span class="st">&quot;medline&quot;</span>)</span>
<span id="cb233-15"><a href="getting-biological-data-from-public-repositories.html#cb233-15" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(genus_publications_medline)</span>
<span id="cb233-16"><a href="getting-biological-data-from-public-repositories.html#cb233-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb233-17"><a href="getting-biological-data-from-public-repositories.html#cb233-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb233-18"><a href="getting-biological-data-from-public-repositories.html#cb233-18" aria-hidden="true" tabindex="-1"></a><span class="co">#PMID- 37889647</span></span>
<span id="cb233-19"><a href="getting-biological-data-from-public-repositories.html#cb233-19" aria-hidden="true" tabindex="-1"></a><span class="co">#OWN - NLM</span></span>
<span id="cb233-20"><a href="getting-biological-data-from-public-repositories.html#cb233-20" aria-hidden="true" tabindex="-1"></a><span class="co">#STAT- MEDLINE</span></span>
<span id="cb233-21"><a href="getting-biological-data-from-public-repositories.html#cb233-21" aria-hidden="true" tabindex="-1"></a><span class="co">#DCOM- 20231129</span></span>
<span id="cb233-22"><a href="getting-biological-data-from-public-repositories.html#cb233-22" aria-hidden="true" tabindex="-1"></a><span class="co">#LR  - 20231227</span></span>
<span id="cb233-23"><a href="getting-biological-data-from-public-repositories.html#cb233-23" aria-hidden="true" tabindex="-1"></a><span class="co">#IS  - 1521-6950 (Electronic)</span></span>
<span id="cb233-24"><a href="getting-biological-data-from-public-repositories.html#cb233-24" aria-hidden="true" tabindex="-1"></a><span class="co">#IS  - 1093-7404 (Linking)</span></span>
<span id="cb233-25"><a href="getting-biological-data-from-public-repositories.html#cb233-25" aria-hidden="true" tabindex="-1"></a><span class="co">#VI  - 27</span></span>
<span id="cb233-26"><a href="getting-biological-data-from-public-repositories.html#cb233-26" aria-hidden="true" tabindex="-1"></a><span class="co">#IP  - 1</span></span>
<span id="cb233-27"><a href="getting-biological-data-from-public-repositories.html#cb233-27" aria-hidden="true" tabindex="-1"></a><span class="co">#DP  - 2024 Jan 2</span></span>
<span id="cb233-28"><a href="getting-biological-data-from-public-repositories.html#cb233-28" aria-hidden="true" tabindex="-1"></a><span class="co">#TI  - Bioactive alkaloids from the venom of Dendrobatoidea Cope, 1865: a scoping </span></span>
<span id="cb233-29"><a href="getting-biological-data-from-public-repositories.html#cb233-29" aria-hidden="true" tabindex="-1"></a><span class="co">#      review.</span></span>
<span id="cb233-30"><a href="getting-biological-data-from-public-repositories.html#cb233-30" aria-hidden="true" tabindex="-1"></a><span class="co">#PG  - 1-20</span></span>
<span id="cb233-31"><a href="getting-biological-data-from-public-repositories.html#cb233-31" aria-hidden="true" tabindex="-1"></a><span class="co">#LID - 10.1080/10937404.2023.2270408 [doi]</span></span>
<span id="cb233-32"><a href="getting-biological-data-from-public-repositories.html#cb233-32" aria-hidden="true" tabindex="-1"></a><span class="co">#AB  - Bioactive compounds derived from secondary metabolism in animals have refined </span></span>
<span id="cb233-33"><a href="getting-biological-data-from-public-repositories.html#cb233-33" aria-hidden="true" tabindex="-1"></a><span class="co">#      selectivity and potency for certain biological targets. The superfamily </span></span>
<span id="cb233-34"><a href="getting-biological-data-from-public-repositories.html#cb233-34" aria-hidden="true" tabindex="-1"></a><span class="co">#      Dendrobatoidea is adapted to the dietary sequestration and secretion of toxic </span></span>
<span id="cb233-35"><a href="getting-biological-data-from-public-repositories.html#cb233-35" aria-hidden="true" tabindex="-1"></a><span class="co">#      alkaloids, which play a role in several biological activities, and thus serve as </span></span>
<span id="cb233-36"><a href="getting-biological-data-from-public-repositories.html#cb233-36" aria-hidden="true" tabindex="-1"></a><span class="co">#      a potential source for pharmacological and biotechnological applications. This </span></span>
<span id="cb233-37"><a href="getting-biological-data-from-public-repositories.html#cb233-37" aria-hidden="true" tabindex="-1"></a><span class="co">#      article constitutes a scoping review to understand the trends in experimental </span></span>
<span id="cb233-38"><a href="getting-biological-data-from-public-repositories.html#cb233-38" aria-hidden="true" tabindex="-1"></a><span class="co">#      research involving bioactive alkaloids derived from Dendrobatoidea based upon </span></span>
<span id="cb233-39"><a href="getting-biological-data-from-public-repositories.html#cb233-39" aria-hidden="true" tabindex="-1"></a><span class="co">#      scientometric approaches. Forty-eight (48) publications were found in 30 journals </span></span>
<span id="cb233-40"><a href="getting-biological-data-from-public-repositories.html#cb233-40" aria-hidden="true" tabindex="-1"></a><span class="co">#      in the period of 60 years, between 1962 and 2022. More than 23 structural classes </span></span>
<span id="cb233-41"><a href="getting-biological-data-from-public-repositories.html#cb233-41" aria-hidden="true" tabindex="-1"></a><span class="co">#      of alkaloids were cited, with 27.63% for batrachotoxins, 13.64% for pyridinics, </span></span>
<span id="cb233-42"><a href="getting-biological-data-from-public-repositories.html#cb233-42" aria-hidden="true" tabindex="-1"></a><span class="co">#      with an emphasis on epibatidine, 16.36% for pumiliotoxins, and 11.82% for </span></span>
<span id="cb233-43"><a href="getting-biological-data-from-public-repositories.html#cb233-43" aria-hidden="true" tabindex="-1"></a><span class="co">#      histrionicotoxins. These tests included in vivo (54.9%), in vitro (39.4%), and in </span></span>
<span id="cb233-44"><a href="getting-biological-data-from-public-repositories.html#cb233-44" aria-hidden="true" tabindex="-1"></a><span class="co">#      silico simulations (5.6%). Most compounds (54.8%) were isolated from skin </span></span>
<span id="cb233-45"><a href="getting-biological-data-from-public-repositories.html#cb233-45" aria-hidden="true" tabindex="-1"></a><span class="co">#      extracts, whereas the remainder were obtained through molecular synthesis. </span></span>
<span id="cb233-46"><a href="getting-biological-data-from-public-repositories.html#cb233-46" aria-hidden="true" tabindex="-1"></a><span class="co">#      Thirteen main biological activities were identified, including </span></span>
<span id="cb233-47"><a href="getting-biological-data-from-public-repositories.html#cb233-47" aria-hidden="true" tabindex="-1"></a><span class="co">#      acetylcholinesterase inhibitors (27.59%), sodium channel inhibitors (12.07%), </span></span>
<span id="cb233-48"><a href="getting-biological-data-from-public-repositories.html#cb233-48" aria-hidden="true" tabindex="-1"></a><span class="co">#      cardiac (12.07%), analgesic (8.62%), and neuromuscular effects (8.62%). The </span></span>
<span id="cb233-49"><a href="getting-biological-data-from-public-repositories.html#cb233-49" aria-hidden="true" tabindex="-1"></a><span class="co">#      substances were cited as being of natural origin in the &quot;Dendrobatidae&quot; family, </span></span>
<span id="cb233-50"><a href="getting-biological-data-from-public-repositories.html#cb233-50" aria-hidden="true" tabindex="-1"></a><span class="co">#      genus &quot;Phyllobates,&quot; &quot;Dendrobates,&quot; and seven species: Epipedobates tricolor, </span></span>
<span id="cb233-51"><a href="getting-biological-data-from-public-repositories.html#cb233-51" aria-hidden="true" tabindex="-1"></a><span class="co">#      Phyllobates aurotaenia, Oophaga histrionica, Oophaga pumilio, Phyllobates </span></span>
<span id="cb233-52"><a href="getting-biological-data-from-public-repositories.html#cb233-52" aria-hidden="true" tabindex="-1"></a><span class="co">#      terribilis, Epipedobates anthonyi, and Ameerega flavopicta. To date, only a few </span></span>
<span id="cb233-53"><a href="getting-biological-data-from-public-repositories.html#cb233-53" aria-hidden="true" tabindex="-1"></a><span class="co">#      biological activities have been experimentally tested; hence, further studies on </span></span>
<span id="cb233-54"><a href="getting-biological-data-from-public-repositories.html#cb233-54" aria-hidden="true" tabindex="-1"></a><span class="co">#      the bioprospecting of animal compounds and ecological approaches are needed.</span></span>
<span id="cb233-55"><a href="getting-biological-data-from-public-repositories.html#cb233-55" aria-hidden="true" tabindex="-1"></a><span class="co">#FAU - Dos Santos Arraes, Debora Regina</span></span>
<span id="cb233-56"><a href="getting-biological-data-from-public-repositories.html#cb233-56" aria-hidden="true" tabindex="-1"></a><span class="co">#AU  - Dos Santos Arraes DR</span></span>
<span id="cb233-57"><a href="getting-biological-data-from-public-repositories.html#cb233-57" aria-hidden="true" tabindex="-1"></a><span class="co">#AD  - Zoology Laboratory, Amapa State University, Macapa, Amapa, Brazil.</span></span>
<span id="cb233-58"><a href="getting-biological-data-from-public-repositories.html#cb233-58" aria-hidden="true" tabindex="-1"></a><span class="co">#FAU - Rodrigues, Alex Bruno Lobato</span></span>
<span id="cb233-59"><a href="getting-biological-data-from-public-repositories.html#cb233-59" aria-hidden="true" tabindex="-1"></a><span class="co">#AU  - Rodrigues ABL</span></span>
<span id="cb233-60"><a href="getting-biological-data-from-public-repositories.html#cb233-60" aria-hidden="true" tabindex="-1"></a><span class="co">#AD  - Laboratory of Analytical Chemistry, Federal University of Amapa, Macapa, Amapa, </span></span>
<span id="cb233-61"><a href="getting-biological-data-from-public-repositories.html#cb233-61" aria-hidden="true" tabindex="-1"></a><span class="co">#      Brazil.</span></span>
<span id="cb233-62"><a href="getting-biological-data-from-public-repositories.html#cb233-62" aria-hidden="true" tabindex="-1"></a><span class="co">#FAU - Sanches, Patrick Ribeiro</span></span>
<span id="cb233-63"><a href="getting-biological-data-from-public-repositories.html#cb233-63" aria-hidden="true" tabindex="-1"></a><span class="co">#AU  - Sanches PR</span></span>
<span id="cb233-64"><a href="getting-biological-data-from-public-repositories.html#cb233-64" aria-hidden="true" tabindex="-1"></a><span class="co">#AD  - Herpetology Laboratory, Federal University of Amapa, Macapa, Amapa, Brazil.</span></span>
<span id="cb233-65"><a href="getting-biological-data-from-public-repositories.html#cb233-65" aria-hidden="true" tabindex="-1"></a><span class="co">#FAU - Costa Campos, Carlos Eduardo</span></span>
<span id="cb233-66"><a href="getting-biological-data-from-public-repositories.html#cb233-66" aria-hidden="true" tabindex="-1"></a><span class="co">#AU  - Costa Campos CE</span></span>
<span id="cb233-67"><a href="getting-biological-data-from-public-repositories.html#cb233-67" aria-hidden="true" tabindex="-1"></a><span class="co">#AD  - Herpetology Laboratory, Federal University of Amapa, Macapa, Amapa, Brazil.</span></span>
<span id="cb233-68"><a href="getting-biological-data-from-public-repositories.html#cb233-68" aria-hidden="true" tabindex="-1"></a><span class="co">#FAU - Moreira da Silva de Almeida, Sheylla Susan</span></span>
<span id="cb233-69"><a href="getting-biological-data-from-public-repositories.html#cb233-69" aria-hidden="true" tabindex="-1"></a><span class="co">#AU  - Moreira da Silva de Almeida SS</span></span>
<span id="cb233-70"><a href="getting-biological-data-from-public-repositories.html#cb233-70" aria-hidden="true" tabindex="-1"></a><span class="co">#AD  - Laboratory of Pharmacognosy and Phytochemistry, Federal University of Amapa, </span></span>
<span id="cb233-71"><a href="getting-biological-data-from-public-repositories.html#cb233-71" aria-hidden="true" tabindex="-1"></a><span class="co">#      Macapa, Amapa, Brazil.</span></span>
<span id="cb233-72"><a href="getting-biological-data-from-public-repositories.html#cb233-72" aria-hidden="true" tabindex="-1"></a><span class="co">#FAU - Reis Ferreira Lima, Janaina</span></span>
<span id="cb233-73"><a href="getting-biological-data-from-public-repositories.html#cb233-73" aria-hidden="true" tabindex="-1"></a><span class="co">#AU  - Reis Ferreira Lima J</span></span>
<span id="cb233-74"><a href="getting-biological-data-from-public-repositories.html#cb233-74" aria-hidden="true" tabindex="-1"></a><span class="co">#AD  - Herpetology Laboratory, Institute of Scientific and Technological Research of the </span></span>
<span id="cb233-75"><a href="getting-biological-data-from-public-repositories.html#cb233-75" aria-hidden="true" tabindex="-1"></a><span class="co">#      State of Amapa, Macapa, Amapa, Brazil.</span></span>
<span id="cb233-76"><a href="getting-biological-data-from-public-repositories.html#cb233-76" aria-hidden="true" tabindex="-1"></a><span class="co">#FAU - Dias Lima, Jucivaldo</span></span>
<span id="cb233-77"><a href="getting-biological-data-from-public-repositories.html#cb233-77" aria-hidden="true" tabindex="-1"></a><span class="co">#AU  - Dias Lima J</span></span>
<span id="cb233-78"><a href="getting-biological-data-from-public-repositories.html#cb233-78" aria-hidden="true" tabindex="-1"></a><span class="co">#AD  - Herpetology Laboratory, Institute of Scientific and Technological Research of the </span></span>
<span id="cb233-79"><a href="getting-biological-data-from-public-repositories.html#cb233-79" aria-hidden="true" tabindex="-1"></a><span class="co">#      State of Amapa, Macapa, Amapa, Brazil.</span></span>
<span id="cb233-80"><a href="getting-biological-data-from-public-repositories.html#cb233-80" aria-hidden="true" tabindex="-1"></a><span class="co">#FAU - da Silva, Gabriel Araujo</span></span>
<span id="cb233-81"><a href="getting-biological-data-from-public-repositories.html#cb233-81" aria-hidden="true" tabindex="-1"></a><span class="co">#AU  - da Silva GA</span></span>
<span id="cb233-82"><a href="getting-biological-data-from-public-repositories.html#cb233-82" aria-hidden="true" tabindex="-1"></a><span class="co">#AD  - Laboratory of Organic Chemistry, Amapa State University, Macapa, Amapa, Brazil.</span></span>
<span id="cb233-83"><a href="getting-biological-data-from-public-repositories.html#cb233-83" aria-hidden="true" tabindex="-1"></a><span class="co">#LA  - eng</span></span>
<span id="cb233-84"><a href="getting-biological-data-from-public-repositories.html#cb233-84" aria-hidden="true" tabindex="-1"></a><span class="co">#PT  - Journal Article</span></span>
<span id="cb233-85"><a href="getting-biological-data-from-public-repositories.html#cb233-85" aria-hidden="true" tabindex="-1"></a><span class="co">#PT  - Research Support, Non-U.S. Gov&#39;t</span></span>
<span id="cb233-86"><a href="getting-biological-data-from-public-repositories.html#cb233-86" aria-hidden="true" tabindex="-1"></a><span class="co">#PT  - Review</span></span>
<span id="cb233-87"><a href="getting-biological-data-from-public-repositories.html#cb233-87" aria-hidden="true" tabindex="-1"></a><span class="co">#DEP - 20231128</span></span>
<span id="cb233-88"><a href="getting-biological-data-from-public-repositories.html#cb233-88" aria-hidden="true" tabindex="-1"></a><span class="co">#PL  - England</span></span>
<span id="cb233-89"><a href="getting-biological-data-from-public-repositories.html#cb233-89" aria-hidden="true" tabindex="-1"></a><span class="co">#TA  - J Toxicol Environ Health B Crit Rev</span></span>
<span id="cb233-90"><a href="getting-biological-data-from-public-repositories.html#cb233-90" aria-hidden="true" tabindex="-1"></a><span class="co">#JT  - Journal of toxicology and environmental health. Part B, Critical reviews</span></span>
<span id="cb233-91"><a href="getting-biological-data-from-public-repositories.html#cb233-91" aria-hidden="true" tabindex="-1"></a><span class="co">#JID - 9802627</span></span>
<span id="cb233-92"><a href="getting-biological-data-from-public-repositories.html#cb233-92" aria-hidden="true" tabindex="-1"></a><span class="co">#RN  - 0 (Venoms)</span></span>
<span id="cb233-93"><a href="getting-biological-data-from-public-repositories.html#cb233-93" aria-hidden="true" tabindex="-1"></a><span class="co">#RN  - EC 3.1.1.7 (Acetylcholinesterase)</span></span>
<span id="cb233-94"><a href="getting-biological-data-from-public-repositories.html#cb233-94" aria-hidden="true" tabindex="-1"></a><span class="co">#RN  - 0 (Batrachotoxins)</span></span>
<span id="cb233-95"><a href="getting-biological-data-from-public-repositories.html#cb233-95" aria-hidden="true" tabindex="-1"></a><span class="co">#RN  - 0 (Alkaloids)</span></span>
<span id="cb233-96"><a href="getting-biological-data-from-public-repositories.html#cb233-96" aria-hidden="true" tabindex="-1"></a><span class="co">#SB  - IM</span></span>
<span id="cb233-97"><a href="getting-biological-data-from-public-repositories.html#cb233-97" aria-hidden="true" tabindex="-1"></a><span class="co">#MH  - Animals</span></span>
<span id="cb233-98"><a href="getting-biological-data-from-public-repositories.html#cb233-98" aria-hidden="true" tabindex="-1"></a><span class="co">#MH  - *Venoms</span></span>
<span id="cb233-99"><a href="getting-biological-data-from-public-repositories.html#cb233-99" aria-hidden="true" tabindex="-1"></a><span class="co">#MH  - Acetylcholinesterase</span></span>
<span id="cb233-100"><a href="getting-biological-data-from-public-repositories.html#cb233-100" aria-hidden="true" tabindex="-1"></a><span class="co">#MH  - Anura/metabolism</span></span>
<span id="cb233-101"><a href="getting-biological-data-from-public-repositories.html#cb233-101" aria-hidden="true" tabindex="-1"></a><span class="co">#MH  - Batrachotoxins/chemistry</span></span>
<span id="cb233-102"><a href="getting-biological-data-from-public-repositories.html#cb233-102" aria-hidden="true" tabindex="-1"></a><span class="co">#MH  - *Alkaloids/chemistry/metabolism</span></span>
<span id="cb233-103"><a href="getting-biological-data-from-public-repositories.html#cb233-103" aria-hidden="true" tabindex="-1"></a><span class="co">#OTO - NOTNLM</span></span>
<span id="cb233-104"><a href="getting-biological-data-from-public-repositories.html#cb233-104" aria-hidden="true" tabindex="-1"></a><span class="co">#OT  - Biodiversity</span></span>
<span id="cb233-105"><a href="getting-biological-data-from-public-repositories.html#cb233-105" aria-hidden="true" tabindex="-1"></a><span class="co">#OT  - biotechnological</span></span>
<span id="cb233-106"><a href="getting-biological-data-from-public-repositories.html#cb233-106" aria-hidden="true" tabindex="-1"></a><span class="co">#OT  - poison frogs</span></span>
<span id="cb233-107"><a href="getting-biological-data-from-public-repositories.html#cb233-107" aria-hidden="true" tabindex="-1"></a><span class="co">#OT  - secondary metabolites</span></span>
<span id="cb233-108"><a href="getting-biological-data-from-public-repositories.html#cb233-108" aria-hidden="true" tabindex="-1"></a><span class="co">#OT  - toxins</span></span>
<span id="cb233-109"><a href="getting-biological-data-from-public-repositories.html#cb233-109" aria-hidden="true" tabindex="-1"></a><span class="co">#EDAT- 2023/10/27 18:42</span></span>
<span id="cb233-110"><a href="getting-biological-data-from-public-repositories.html#cb233-110" aria-hidden="true" tabindex="-1"></a><span class="co">#MHDA- 2023/11/29 06:42</span></span>
<span id="cb233-111"><a href="getting-biological-data-from-public-repositories.html#cb233-111" aria-hidden="true" tabindex="-1"></a><span class="co">#CRDT- 2023/10/27 12:24</span></span>
<span id="cb233-112"><a href="getting-biological-data-from-public-repositories.html#cb233-112" aria-hidden="true" tabindex="-1"></a><span class="co">#PHST- 2023/11/29 06:42 [medline]</span></span>
<span id="cb233-113"><a href="getting-biological-data-from-public-repositories.html#cb233-113" aria-hidden="true" tabindex="-1"></a><span class="co">#PHST- 2023/10/27 18:42 [pubmed]</span></span>
<span id="cb233-114"><a href="getting-biological-data-from-public-repositories.html#cb233-114" aria-hidden="true" tabindex="-1"></a><span class="co">#PHST- 2023/10/27 12:24 [entrez]</span></span>
<span id="cb233-115"><a href="getting-biological-data-from-public-repositories.html#cb233-115" aria-hidden="true" tabindex="-1"></a><span class="co">#AID - 10.1080/10937404.2023.2270408 [doi]</span></span>
<span id="cb233-116"><a href="getting-biological-data-from-public-repositories.html#cb233-116" aria-hidden="true" tabindex="-1"></a><span class="co">#PST - ppublish</span></span>
<span id="cb233-117"><a href="getting-biological-data-from-public-repositories.html#cb233-117" aria-hidden="true" tabindex="-1"></a><span class="co">#SO  - J Toxicol Environ Health B Crit Rev. 2024 Jan 2;27(1):1-20. doi: </span></span>
<span id="cb233-118"><a href="getting-biological-data-from-public-repositories.html#cb233-118" aria-hidden="true" tabindex="-1"></a><span class="co">#      10.1080/10937404.2023.2270408. Epub 2023 Nov 28.</span></span>
<span id="cb233-119"><a href="getting-biological-data-from-public-repositories.html#cb233-119" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span></code></pre></div>
<p>We can save these publication refereces into a text file in your working directory.</p>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="getting-biological-data-from-public-repositories.html#cb234-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this is exclusive to your OWN COMPUTER change it accordingly</span></span>
<span id="cb234-2"><a href="getting-biological-data-from-public-repositories.html#cb234-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/my_working_directory&quot;</span>)</span>
<span id="cb234-3"><a href="getting-biological-data-from-public-repositories.html#cb234-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(genus_publications_medline, <span class="st">&quot;my_genus_publications_as_medline_format.txt&quot;</span>)</span></code></pre></div>
<p>We can retrieve the results in the <a href="https://www.nlm.nih.gov/bsd/mms/medlineelements.html" target="_blank"><strong>abstract</strong></a> format as follows.</p>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb235-1"><a href="getting-biological-data-from-public-repositories.html#cb235-1" aria-hidden="true" tabindex="-1"></a><span class="do">## make sure that &#39;rentrez&#39; is loaded</span></span>
<span id="cb235-2"><a href="getting-biological-data-from-public-repositories.html#cb235-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(rentrez)</span>
<span id="cb235-3"><a href="getting-biological-data-from-public-repositories.html#cb235-3" aria-hidden="true" tabindex="-1"></a><span class="do">## retrieve IDs from NCBI</span></span>
<span id="cb235-4"><a href="getting-biological-data-from-public-repositories.html#cb235-4" aria-hidden="true" tabindex="-1"></a>genus_publications_IDs <span class="ot">&lt;-</span> <span class="fu">entrez_search</span>(<span class="at">db=</span><span class="st">&quot;pubmed&quot;</span>, <span class="at">term=</span><span class="st">&quot;Ameerega&quot;</span>)</span>
<span id="cb235-5"><a href="getting-biological-data-from-public-repositories.html#cb235-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(genus_publications_IDs)</span>
<span id="cb235-6"><a href="getting-biological-data-from-public-repositories.html#cb235-6" aria-hidden="true" tabindex="-1"></a><span class="co">#List of 5</span></span>
<span id="cb235-7"><a href="getting-biological-data-from-public-repositories.html#cb235-7" aria-hidden="true" tabindex="-1"></a><span class="co"># $ ids             : chr [1:17] &quot;32230685&quot; &quot;31716726&quot; &quot;31586688&quot; &quot;31125660&quot; ...</span></span>
<span id="cb235-8"><a href="getting-biological-data-from-public-repositories.html#cb235-8" aria-hidden="true" tabindex="-1"></a><span class="co"># $ count           : int 17</span></span>
<span id="cb235-9"><a href="getting-biological-data-from-public-repositories.html#cb235-9" aria-hidden="true" tabindex="-1"></a><span class="co"># $ retmax          : int 17</span></span>
<span id="cb235-10"><a href="getting-biological-data-from-public-repositories.html#cb235-10" aria-hidden="true" tabindex="-1"></a><span class="co"># $ QueryTranslation: chr &quot;Ameerega[All Fields]&quot;</span></span>
<span id="cb235-11"><a href="getting-biological-data-from-public-repositories.html#cb235-11" aria-hidden="true" tabindex="-1"></a><span class="co"># $ file            :Classes &#39;XMLInternalDocument&#39;, &#39;XMLAbstractDocument&#39; &lt;externalptr&gt; </span></span>
<span id="cb235-12"><a href="getting-biological-data-from-public-repositories.html#cb235-12" aria-hidden="true" tabindex="-1"></a><span class="co"># - attr(*, &quot;class&quot;)= chr [1:2] &quot;esearch&quot; &quot;list&quot;</span></span>
<span id="cb235-13"><a href="getting-biological-data-from-public-repositories.html#cb235-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb235-14"><a href="getting-biological-data-from-public-repositories.html#cb235-14" aria-hidden="true" tabindex="-1"></a>genus_publications_abstract <span class="ot">&lt;-</span> <span class="fu">entrez_fetch</span>(<span class="at">db=</span><span class="st">&quot;pubmed&quot;</span>, <span class="at">id=</span>genus_publications_IDs<span class="sc">$</span>ids, <span class="at">rettype=</span><span class="st">&quot;abstract&quot;</span>)</span>
<span id="cb235-15"><a href="getting-biological-data-from-public-repositories.html#cb235-15" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(genus_publications_abstract)</span>
<span id="cb235-16"><a href="getting-biological-data-from-public-repositories.html#cb235-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb235-17"><a href="getting-biological-data-from-public-repositories.html#cb235-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb235-18"><a href="getting-biological-data-from-public-repositories.html#cb235-18" aria-hidden="true" tabindex="-1"></a><span class="co">#J Toxicol Environ Health B Crit Rev. 2024 Jan 2;27(1):1-20. doi: </span></span>
<span id="cb235-19"><a href="getting-biological-data-from-public-repositories.html#cb235-19" aria-hidden="true" tabindex="-1"></a><span class="co">#10.1080/10937404.2023.2270408. Epub 2023 Nov 28.</span></span>
<span id="cb235-20"><a href="getting-biological-data-from-public-repositories.html#cb235-20" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb235-21"><a href="getting-biological-data-from-public-repositories.html#cb235-21" aria-hidden="true" tabindex="-1"></a><span class="co">#Bioactive alkaloids from the venom of Dendrobatoidea Cope, 1865: a scoping </span></span>
<span id="cb235-22"><a href="getting-biological-data-from-public-repositories.html#cb235-22" aria-hidden="true" tabindex="-1"></a><span class="co">#review.</span></span>
<span id="cb235-23"><a href="getting-biological-data-from-public-repositories.html#cb235-23" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb235-24"><a href="getting-biological-data-from-public-repositories.html#cb235-24" aria-hidden="true" tabindex="-1"></a><span class="co">#Dos Santos Arraes DR(1), Rodrigues ABL(2), Sanches PR(3), Costa Campos CE(3), </span></span>
<span id="cb235-25"><a href="getting-biological-data-from-public-repositories.html#cb235-25" aria-hidden="true" tabindex="-1"></a><span class="co">#Moreira da Silva de Almeida SS(4), Reis Ferreira Lima J(5), Dias Lima J(5), da </span></span>
<span id="cb235-26"><a href="getting-biological-data-from-public-repositories.html#cb235-26" aria-hidden="true" tabindex="-1"></a><span class="co">#Silva GA(6).</span></span>
<span id="cb235-27"><a href="getting-biological-data-from-public-repositories.html#cb235-27" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb235-28"><a href="getting-biological-data-from-public-repositories.html#cb235-28" aria-hidden="true" tabindex="-1"></a><span class="co">#Author information:</span></span>
<span id="cb235-29"><a href="getting-biological-data-from-public-repositories.html#cb235-29" aria-hidden="true" tabindex="-1"></a><span class="co">#(1)Zoology Laboratory, Amapá State University, Macapá, Amapá, Brazil.</span></span>
<span id="cb235-30"><a href="getting-biological-data-from-public-repositories.html#cb235-30" aria-hidden="true" tabindex="-1"></a><span class="co">#(2)Laboratory of Analytical Chemistry, Federal University of Amapá, Macapá, </span></span>
<span id="cb235-31"><a href="getting-biological-data-from-public-repositories.html#cb235-31" aria-hidden="true" tabindex="-1"></a><span class="co">#Amapá, Brazil.</span></span>
<span id="cb235-32"><a href="getting-biological-data-from-public-repositories.html#cb235-32" aria-hidden="true" tabindex="-1"></a><span class="co">#(3)Herpetology Laboratory, Federal University of Amapá, Macapá, Amapá, Brazil.</span></span>
<span id="cb235-33"><a href="getting-biological-data-from-public-repositories.html#cb235-33" aria-hidden="true" tabindex="-1"></a><span class="co">#(4)Laboratory of Pharmacognosy and Phytochemistry, Federal University of Amapá, </span></span>
<span id="cb235-34"><a href="getting-biological-data-from-public-repositories.html#cb235-34" aria-hidden="true" tabindex="-1"></a><span class="co">#Macapá, Amapá, Brazil.</span></span>
<span id="cb235-35"><a href="getting-biological-data-from-public-repositories.html#cb235-35" aria-hidden="true" tabindex="-1"></a><span class="co">#(5)Herpetology Laboratory, Institute of Scientific and Technological Research of </span></span>
<span id="cb235-36"><a href="getting-biological-data-from-public-repositories.html#cb235-36" aria-hidden="true" tabindex="-1"></a><span class="co">#the State of Amapá, Macapá, Amapá, Brazil.</span></span>
<span id="cb235-37"><a href="getting-biological-data-from-public-repositories.html#cb235-37" aria-hidden="true" tabindex="-1"></a><span class="co">#(6)Laboratory of Organic Chemistry, Amapá State University, Macapá, Amapá, </span></span>
<span id="cb235-38"><a href="getting-biological-data-from-public-repositories.html#cb235-38" aria-hidden="true" tabindex="-1"></a><span class="co">#Brazil.</span></span>
<span id="cb235-39"><a href="getting-biological-data-from-public-repositories.html#cb235-39" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb235-40"><a href="getting-biological-data-from-public-repositories.html#cb235-40" aria-hidden="true" tabindex="-1"></a><span class="co">#Bioactive compounds derived from secondary metabolism in animals have refined </span></span>
<span id="cb235-41"><a href="getting-biological-data-from-public-repositories.html#cb235-41" aria-hidden="true" tabindex="-1"></a><span class="co">#selectivity and potency for certain biological targets. The superfamily </span></span>
<span id="cb235-42"><a href="getting-biological-data-from-public-repositories.html#cb235-42" aria-hidden="true" tabindex="-1"></a><span class="co">#Dendrobatoidea is adapted to the dietary sequestration and secretion of toxic </span></span>
<span id="cb235-43"><a href="getting-biological-data-from-public-repositories.html#cb235-43" aria-hidden="true" tabindex="-1"></a><span class="co">#alkaloids, which play a role in several biological activities, and thus serve as </span></span>
<span id="cb235-44"><a href="getting-biological-data-from-public-repositories.html#cb235-44" aria-hidden="true" tabindex="-1"></a><span class="co">#a potential source for pharmacological and biotechnological applications. This </span></span>
<span id="cb235-45"><a href="getting-biological-data-from-public-repositories.html#cb235-45" aria-hidden="true" tabindex="-1"></a><span class="co">#article constitutes a scoping review to understand the trends in experimental </span></span>
<span id="cb235-46"><a href="getting-biological-data-from-public-repositories.html#cb235-46" aria-hidden="true" tabindex="-1"></a><span class="co">#research involving bioactive alkaloids derived from Dendrobatoidea based upon </span></span>
<span id="cb235-47"><a href="getting-biological-data-from-public-repositories.html#cb235-47" aria-hidden="true" tabindex="-1"></a><span class="co">#scientometric approaches. Forty-eight (48) publications were found in 30 </span></span>
<span id="cb235-48"><a href="getting-biological-data-from-public-repositories.html#cb235-48" aria-hidden="true" tabindex="-1"></a><span class="co">#journals in the period of 60 years, between 1962 and 2022. More than 23 </span></span>
<span id="cb235-49"><a href="getting-biological-data-from-public-repositories.html#cb235-49" aria-hidden="true" tabindex="-1"></a><span class="co">#structural classes of alkaloids were cited, with 27.63% for batrachotoxins, </span></span>
<span id="cb235-50"><a href="getting-biological-data-from-public-repositories.html#cb235-50" aria-hidden="true" tabindex="-1"></a><span class="co">#13.64% for pyridinics, with an emphasis on epibatidine, 16.36% for </span></span>
<span id="cb235-51"><a href="getting-biological-data-from-public-repositories.html#cb235-51" aria-hidden="true" tabindex="-1"></a><span class="co">#pumiliotoxins, and 11.82% for histrionicotoxins. These tests included in vivo </span></span>
<span id="cb235-52"><a href="getting-biological-data-from-public-repositories.html#cb235-52" aria-hidden="true" tabindex="-1"></a><span class="co">#(54.9%), in vitro (39.4%), and in silico simulations (5.6%). Most compounds </span></span>
<span id="cb235-53"><a href="getting-biological-data-from-public-repositories.html#cb235-53" aria-hidden="true" tabindex="-1"></a><span class="co">#(54.8%) were isolated from skin extracts, whereas the remainder were obtained </span></span>
<span id="cb235-54"><a href="getting-biological-data-from-public-repositories.html#cb235-54" aria-hidden="true" tabindex="-1"></a><span class="co">#through molecular synthesis. Thirteen main biological activities were </span></span>
<span id="cb235-55"><a href="getting-biological-data-from-public-repositories.html#cb235-55" aria-hidden="true" tabindex="-1"></a><span class="co">#identified, including acetylcholinesterase inhibitors (27.59%), sodium channel </span></span>
<span id="cb235-56"><a href="getting-biological-data-from-public-repositories.html#cb235-56" aria-hidden="true" tabindex="-1"></a><span class="co">#inhibitors (12.07%), cardiac (12.07%), analgesic (8.62%), and neuromuscular </span></span>
<span id="cb235-57"><a href="getting-biological-data-from-public-repositories.html#cb235-57" aria-hidden="true" tabindex="-1"></a><span class="co">#effects (8.62%). The substances were cited as being of natural origin in the </span></span>
<span id="cb235-58"><a href="getting-biological-data-from-public-repositories.html#cb235-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&quot;Dendrobatidae&quot; family, genus &quot;Phyllobates,&quot; &quot;Dendrobates,&quot; and seven species: </span></span>
<span id="cb235-59"><a href="getting-biological-data-from-public-repositories.html#cb235-59" aria-hidden="true" tabindex="-1"></a><span class="co">#Epipedobates tricolor, Phyllobates aurotaenia, Oophaga histrionica, Oophaga </span></span>
<span id="cb235-60"><a href="getting-biological-data-from-public-repositories.html#cb235-60" aria-hidden="true" tabindex="-1"></a><span class="co">#pumilio, Phyllobates terribilis, Epipedobates anthonyi, and Ameerega flavopicta. </span></span>
<span id="cb235-61"><a href="getting-biological-data-from-public-repositories.html#cb235-61" aria-hidden="true" tabindex="-1"></a><span class="co">#To date, only a few biological activities have been experimentally tested; </span></span>
<span id="cb235-62"><a href="getting-biological-data-from-public-repositories.html#cb235-62" aria-hidden="true" tabindex="-1"></a><span class="co">#hence, further studies on the bioprospecting of animal compounds and ecological </span></span>
<span id="cb235-63"><a href="getting-biological-data-from-public-repositories.html#cb235-63" aria-hidden="true" tabindex="-1"></a><span class="co">#approaches are needed.</span></span>
<span id="cb235-64"><a href="getting-biological-data-from-public-repositories.html#cb235-64" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb235-65"><a href="getting-biological-data-from-public-repositories.html#cb235-65" aria-hidden="true" tabindex="-1"></a><span class="co">#DOI: 10.1080/10937404.2023.2270408</span></span>
<span id="cb235-66"><a href="getting-biological-data-from-public-repositories.html#cb235-66" aria-hidden="true" tabindex="-1"></a><span class="co">#PMID: 37889647 [Indexed for MEDLINE]</span></span>
<span id="cb235-67"><a href="getting-biological-data-from-public-repositories.html#cb235-67" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span></code></pre></div>
<p>We can save these publication refereces into a text file in your working directory.</p>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="getting-biological-data-from-public-repositories.html#cb236-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this is exclusive to your OWN COMPUTER change it accordingly</span></span>
<span id="cb236-2"><a href="getting-biological-data-from-public-repositories.html#cb236-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/my_working_directory&quot;</span>)</span>
<span id="cb236-3"><a href="getting-biological-data-from-public-repositories.html#cb236-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(genus_publications_abstract, <span class="st">&quot;my_genus_publications_as_abstract_format.txt&quot;</span>)</span></code></pre></div>
</div>
<div id="comparing-sequences-blast" class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> Comparing sequences: BLAST</h2>
<p>One of the most important tools of the NCBI is the Basic Local Alignment Search Tool (BLAST). This tool finds regions of local similarity between sequences by comparing nucleotide or protein sequences to sequence databases and calculates the statistical significance of matches. <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" target="_blank"><strong>BLAST</strong></a> can be accessed form the NCBI website or link provided. It also has a friendly <a href="ftp://ftp.ncbi.nlm.nih.gov/pub/factsheets/HowTo_BLASTGuide.pdf" target="_blank"><strong>guide</strong></a>.</p>
<p><img src="figures/session_5_figs/10_NCBI_blast_intro.png" width="100%" /></p>
<p>Here is a table from the reference manual that provide you with key features of the BLAST search pages in the “Basic BLAST” category <a href="ftp://ftp.ncbi.nlm.nih.gov/pub/factsheets/HowTo_BLASTGuide.pdf" target="_blank"><strong>guide</strong></a>.</p>
<table>
<colgroup>
<col width="13%" />
<col width="19%" />
<col width="10%" />
<col width="56%" />
</colgroup>
<thead>
<tr class="header">
<th>Search page</th>
<th>Query &amp; database</th>
<th>Alignment</th>
<th>Programs &amp; Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>nucleotide blast</td>
<td>nucleotide vs nucleotide</td>
<td>nucleotide</td>
<td><strong>megablast:</strong> for sequence identification, intra-species comparison, <strong>discontiguous megablast:</strong> for cross-species comparison, searching with coding sequences; <strong>blastn:</strong> for searching with shorter queries, cross-species comparison</td>
</tr>
<tr class="even">
<td>protein blast</td>
<td>protein vs protein</td>
<td>protein</td>
<td><strong>blastp:</strong> general sequence identification and similarity searches; <strong>Quick BLASTP:</strong> with a kmer match to accelerate search speed for very similar proteins; <strong>DELTA-BLAST[3]:</strong> protein similarity search with higher sensitivity than blastp; <strong>PSI-BLAST:</strong> iterative search for position-specific score matrix (PSSM) to identify distant relatives for a protein family; <strong>PHI-BLAST:</strong> protein alignment with input pattern as anchor_constraint</td>
</tr>
<tr class="odd">
<td>blastx</td>
<td>nucleotide (translated) vs protein</td>
<td>protein</td>
<td><strong>blastx:</strong> for identifying potential protein products encoded by a nucleotide query</td>
</tr>
<tr class="even">
<td>tblastn</td>
<td>protein vs nucleotide (translated)</td>
<td>protein</td>
<td><strong>tblastn:</strong> for identifying database sequences encoding proteins similar to the query</td>
</tr>
<tr class="odd">
<td>tblastx</td>
<td>nucleotide (translated) vs nucleotide (translated)</td>
<td>protein</td>
<td><strong>tblastx:</strong> for identifying nucleotide sequences similar to the query based on their coding potential</td>
</tr>
</tbody>
</table>
<p>We will focus on nucleotide blast.</p>
<p><strong>9)</strong> If you are interested in comparing a nucleotide query against the nucleotide reference collection. You can use [<strong>blastn</strong>] or standard nucleotide BLAST.</p>
<p>For a simple query, you usually start by copying and pasting the nucleotide sequence that you want to <code>BLAST</code>. We will use a short sequence of the <code>MF580102.1</code> NCBI accession number of the nicotinic acetylcholine receptor beta-2 (chrnb2) from <em>Allobates kingsburyi</em>.</p>
<div class="sourceCode" id="cb237"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb237-1"><a href="getting-biological-data-from-public-repositories.html#cb237-1" aria-hidden="true" tabindex="-1"></a><span class="sc">&gt;</span>MF580102<span class="fl">.1</span> Allobates kingsburyi nicotinic acetylcholine receptor beta<span class="dv">-2</span> (chrnb2) gene, partial cds</span>
<span id="cb237-2"><a href="getting-biological-data-from-public-repositories.html#cb237-2" aria-hidden="true" tabindex="-1"></a>ATGACGGTTCTCCTCCTCCTCCTGCACCTCAGCCTGTTCGGCCTGGTCACCAGGAGTATGGGCACGGACA</span>
<span id="cb237-3"><a href="getting-biological-data-from-public-repositories.html#cb237-3" aria-hidden="true" tabindex="-1"></a>CCGAGGAGCGGCTCGTGGAATTCCTGCTGGACCCGTCCCAGTACAACAAGCTGATCCGGCCCGCCACCAA</span>
<span id="cb237-4"><a href="getting-biological-data-from-public-repositories.html#cb237-4" aria-hidden="true" tabindex="-1"></a>TGGATCCGAGCAGGTCACCGTCCAGCTGATGGTATCTCTGGCCCAGCTGATCAGCGTGCACGAGCGGGAG</span>
<span id="cb237-5"><a href="getting-biological-data-from-public-repositories.html#cb237-5" aria-hidden="true" tabindex="-1"></a>CAGATAATGACGACGAATGTCTGGCTCACTCAGGAATGGGANNNNNNNNNNNNNNNNNNNNNNNNNNNNN</span>
<span id="cb237-6"><a href="getting-biological-data-from-public-repositories.html#cb237-6" aria-hidden="true" tabindex="-1"></a>NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNCTGGCTGCCAGACGTCGTCCTGTACAA</span>
<span id="cb237-7"><a href="getting-biological-data-from-public-repositories.html#cb237-7" aria-hidden="true" tabindex="-1"></a>CAACGCTGATGGGATGTACGAGGTCTCATTCTACTCGAATGCGGTGGTCTCGCATGACGGCAGCATCTTC</span>
<span id="cb237-8"><a href="getting-biological-data-from-public-repositories.html#cb237-8" aria-hidden="true" tabindex="-1"></a>TGGCTACCCCCCGCCATCTATAAGAGTGCCTGTAAGATTGAGGTGAAGCATTTCCCCTTTGATCAGCAGA</span>
<span id="cb237-9"><a href="getting-biological-data-from-public-repositories.html#cb237-9" aria-hidden="true" tabindex="-1"></a>ACTGCACCATGAAGTTTCGCTCATGGACKTACGACCGCACGGAGCTGGACCTGGTGCTGAAGAGCGACGT</span>
<span id="cb237-10"><a href="getting-biological-data-from-public-repositories.html#cb237-10" aria-hidden="true" tabindex="-1"></a>GGCCAGTTTGGATGACTTCACCCCCAGCGGGGAGTGGGACATCATCGCCCTGCCGGGACGCCGGAATGAG</span>
<span id="cb237-11"><a href="getting-biological-data-from-public-repositories.html#cb237-11" aria-hidden="true" tabindex="-1"></a>AACCCCGAGGACTCCACCTACGTTGACATCACTTACGATTTTATCATCCGCCGCAAGCCGCTGTTCTACA</span>
<span id="cb237-12"><a href="getting-biological-data-from-public-repositories.html#cb237-12" aria-hidden="true" tabindex="-1"></a>CCATCAACTTGATCATCCCCTGCATCCTCATCACCTCACTGGCAATTCTGGTCTTCTACCTGCCCTCGGA</span>
<span id="cb237-13"><a href="getting-biological-data-from-public-repositories.html#cb237-13" aria-hidden="true" tabindex="-1"></a>CTGTGGCGAGAAGATGACGCTCTGCATCTCCGTGCTGCTGGCGCTCACCGTCTTCTTACTGCTCATCTCC</span>
<span id="cb237-14"><a href="getting-biological-data-from-public-repositories.html#cb237-14" aria-hidden="true" tabindex="-1"></a>AAGATTGTGCCCCCGACGTCCCTGGACGTCCCGCTCGTCGGCAAGTACCTGATGTTCAC</span></code></pre></div>
<p>Notice that you can paste multiple sequences in fasta format.
<img src="figures/session_5_figs/11_NCBI_blastn_input.png" width="100%" /></p>
<p><strong>10)</strong> You have several options under <code>Choose Search Set</code>.
The first set is to choose a database and the default is <code>Nucleotide collection (nr/nt)</code>. However, you have other 15 databases to choose. In most cases, the <code>Nucleotide collection (nr/nt)</code> is the best option. However, the others are useful if looking in reference collections (e.g., refseq_select, refseq_rna, refseq_representative_genomes, etc.) Other dataset are more ameneable for genomic/transcriptomic datasets (e.g., wgs, SRA, TSA) and in most cases they lack annotation (i.e., gene soruce is unknown).
<img src="figures/session_5_figs/12_NCBI_blastn_databases.png" width="100%" />
Next, you can select the organism taxonomic level that you want to restrict you search. If left blank, it will all references in that database. The lower the taxonomic level (e.g., genus versus family), the less sequences to compare the faster the BLAST will run. In this example, we will select the order Anura (all frogs.)
<img src="figures/session_5_figs/13_NCBI_blastn_taxonomy.png" width="100%" /></p>
<p>In the program selection, the default is <code>highly similar sequences (megablast)</code>. This is the strictest, this approach is intended for comparing a query to closely related sequences and works best if the target percent identity is 95% or more but is very fast. The option <code>somewhat similar sequences (blastn)</code> is slow but allows a word-size down to seven bases (you can recover more matches). Likewise, you can modify the <code>General Parameters</code> from its default values. In most cases before running BLAST, you leave them in their default. For our example, we will select <code>blastn</code> with other parameters in default.
<img src="figures/session_5_figs/14_NCBI_blastn_options.png" width="100%" /></p>
<p>Then, we click <strong>BLAST</strong>.</p>
<p><strong>11)</strong> After you submit your query, you might need to wait for seconds to minutes. This time is not consistent and depend on the NCBI servers. Here is the output of our search.</p>
<p><img src="figures/session_5_figs/15_NCBI_blastn_output.png" width="100%" /></p>
<p>Some key information is provided here. The <strong><code>Descriptions</code></strong> will provided the name of the submission that matched your search. In this case, <code>Allobates kingsburyi nicotinic acetylcholine receptor beta-2 (chrnb2) gene, partial cds</code> is a perfect match because we searched this exact sequence. The matched are presented in order from the top as closets matches to the bottom as the worst. The three most important numbers that guide you in your reading of the results are: <code>Query Cover</code> – numbers close to 100% are preferred and 91% indicates that this percentage of the sequence in the query covered the sequence in description; <code>E value</code> – numbers close to 0 are preferred and in our case was 0 so it is an extremely close match and <code>Per. Inden</code> – numbers close to 100% are preferred and in our case is 100% or a perfect match.</p>
<p>Other relevant information is <code>Accession</code> or the number in the NCBI that you can retrieve using from R such as using <code>rentrez</code>.</p>
<p><strong>12)</strong> If you click on any of the matches or the <strong><code>Alignments</code></strong> tab, the browser will take you to the alignment. In there, you can find nucleotide sites that matched (i.e., <code>Identities</code>) or not.
The reading frame of the query as a <code>Strand</code> (e.g., Plus/Plus is standard; if Plus/Minus one needs to reverse and complement)</p>
<p><img src="figures/session_5_figs/16_NCBI_blastn_alignment.png" width="100%" /></p>
<p>You can also click on the <code>GenBank</code> link to take you to the accession entry of the reference. In this case, <code>MF580103.1</code> that is the sequence of <em>Allobates zaparo</em>.</p>
<p><strong>13)</strong> From each alignment block, you can get a summary of how well the subject sequence matches your query. This includes the name of species, gene identity, and other information such if the subject sequence is complete (e.g., complete cds). It will also include the sequence ID or NCBI accession number and its length. On the alignment top row, you see the max score (‘Score’) and expect value (‘Expect’), a high score and very low expected values suggest a strong similarity to the query. The ‘Identities’ indicate the number of identical bases between the query and the subject sequences. The ‘Gaps’ indicate the number of gaps and their % in the alignment over the entire length of the matching section of the query versus the subject sequence alignment. The ‘Strand’ provides the orientation of the query sequence relative to the subject sequence. Such as Plus/Plus (both 3 -&gt; 5), Minus/Minus (both 5 -&gt; 3), and variations between these. Sometimes, you will see lowercase letters in the alignment (e.g., ‘ttttttttt’) these indicate low complexity sequences with lots of the same type of bases in the query sequence.</p>
<p><img src="figures/session_5_figs/17_NCBI_blastn_sequence_alignment_info.png" width="100%" /></p>
<p><strong>14)</strong> If you click on <strong><code>Graphic Summary</code></strong> this will show you the sequence alignments as colored lines that correspond to NCBI database matches to our query sequence. The light blue bar will indicate the nucleotide/amino acid position from 1 to the end. The matching sequences will present different colors based on the alignment scores to identify local regions of sequence similarity with the query sequence. Higher values (e.g., &gt;= 200) would indicate higher similarity. You can notice that some aligned regions can be separated and connected by a thin grey line these are multiple alignment blocks. You can mouse over an alignment sequence to reveal the sequence title as in the NCBI database. Clicking on a line (i.e., alignment) will displays a box with more details about such alignment and a link to the sequence itself in the <strong><code>Alignments</code></strong> section of the report.</p>
<p><img src="figures/session_5_figs/18_NCBI_blastn_sequence_summary.png" width="100%" /></p>
<p><strong>15)</strong> If you click on <strong><code>Taxonomy</code></strong> you will get a summary of the species source of the subject sequences (i.e., BLAST matches), including their higher taxonomic classification (e.g., family, class, order, etc).</p>
<p><img src="figures/session_5_figs/19_NCBI_blastn_sequence_taxonomy.png" width="100%" /></p>
</div>
<div id="other-databases-uniprot" class="section level2" number="5.5">
<h2><span class="header-section-number">5.5</span> Other databases: UniProt</h2>
<p><strong>16)</strong> Other source of important sequence information is the <a href="https://www.uniprot.org/" target="_blank"><strong>UniProt</strong></a> database, which provides high-quality and freely accessible protein sequences and functional information.</p>
<p><img src="figures/session_5_figs/8_UniProt_front.png" width="100%" />
These sequences can be retrieved by their entry number using the R-package <a href="https://cran.r-project.org/web/packages/UniprotR/index.html" target="_blank"><strong>UniprotR</strong></a>.</p>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="getting-biological-data-from-public-repositories.html#cb238-1" aria-hidden="true" tabindex="-1"></a><span class="co">#install and load the package</span></span>
<span id="cb238-2"><a href="getting-biological-data-from-public-repositories.html#cb238-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;UniprotR&quot;</span>)</span>
<span id="cb238-3"><a href="getting-biological-data-from-public-repositories.html#cb238-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(UniprotR)</span></code></pre></div>
<p>We can retrieve UniProt sequences of the same gene that presented before the acetylcholine receptor CHRNB2. For this purpose, we need to retrieve the list of entry numbers in the UniProt databases. For example, we will retrieve CHRNB2 from the following taxa: human <code>P17787</code>, mouse <code>Q9ERK7</code>, chicken <code>P09484</code>, <em>Bos taurus</em> <code>A0A3Q1MJN8</code>, <em>Danio rerio</em> <code>E7F4S7</code>, Coelacanth <code>H3B5V5</code>, <em>Anolis</em> <code>H9GMJ0</code>, <em>Xenopus tropicalis</em> <code>A4IIS6</code>.
We will use the function <code>GetSequences()</code>.</p>
<div class="sourceCode" id="cb239"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb239-1"><a href="getting-biological-data-from-public-repositories.html#cb239-1" aria-hidden="true" tabindex="-1"></a>other_CHRNB2_ids <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;P17787&quot;</span>, <span class="st">&quot;Q9ERK7&quot;</span>, <span class="st">&quot;P09484&quot;</span>, <span class="st">&quot;A0A3Q1MJN8&quot;</span>, <span class="st">&quot;E7F4S7&quot;</span>, <span class="st">&quot;H3B5V5&quot;</span>, <span class="st">&quot;H9GMJ0&quot;</span>, <span class="st">&quot;A4IIS6&quot;</span>)</span>
<span id="cb239-2"><a href="getting-biological-data-from-public-repositories.html#cb239-2" aria-hidden="true" tabindex="-1"></a>other_CHRNB2_raw <span class="ot">&lt;-</span> UniprotR<span class="sc">::</span><span class="fu">GetSequences</span>(other_CHRNB2_ids)</span>
<span id="cb239-3"><a href="getting-biological-data-from-public-repositories.html#cb239-3" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(other_CHRNB2_raw)</span>
<span id="cb239-4"><a href="getting-biological-data-from-public-repositories.html#cb239-4" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;data.frame&quot;</span></span>
<span id="cb239-5"><a href="getting-biological-data-from-public-repositories.html#cb239-5" aria-hidden="true" tabindex="-1"></a><span class="co"># this is a complex data structure and we need the sequences that can be later process as AAStringSet</span></span>
<span id="cb239-6"><a href="getting-biological-data-from-public-repositories.html#cb239-6" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(other_CHRNB2_raw)</span>
<span id="cb239-7"><a href="getting-biological-data-from-public-repositories.html#cb239-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  8 obs. of  23 variables:</span></span>
<span id="cb239-8"><a href="getting-biological-data-from-public-repositories.html#cb239-8" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Fragment                       : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-9"><a href="getting-biological-data-from-public-repositories.html#cb239-9" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Gene.encoded.by                : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-10"><a href="getting-biological-data-from-public-repositories.html#cb239-10" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Alternative.products..isoforms.: logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-11"><a href="getting-biological-data-from-public-repositories.html#cb239-11" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Erroneous.gene.model.prediction: logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-12"><a href="getting-biological-data-from-public-repositories.html#cb239-12" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Erroneous.initiation           : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-13"><a href="getting-biological-data-from-public-repositories.html#cb239-13" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Erroneous.termination          : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-14"><a href="getting-biological-data-from-public-repositories.html#cb239-14" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Erroneous.translation          : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-15"><a href="getting-biological-data-from-public-repositories.html#cb239-15" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Frameshift                     : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-16"><a href="getting-biological-data-from-public-repositories.html#cb239-16" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Mass.spectrometry              : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-17"><a href="getting-biological-data-from-public-repositories.html#cb239-17" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Polymorphism                   : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-18"><a href="getting-biological-data-from-public-repositories.html#cb239-18" aria-hidden="true" tabindex="-1"></a><span class="co">#$ RNA.editing                    : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-19"><a href="getting-biological-data-from-public-repositories.html#cb239-19" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Sequence.caution               : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-20"><a href="getting-biological-data-from-public-repositories.html#cb239-20" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Length                         : int  502 501 491 509 543 478 509 499</span></span>
<span id="cb239-21"><a href="getting-biological-data-from-public-repositories.html#cb239-21" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Mass                           : Factor w/ 8 levels &quot;57,019&quot;,&quot;57,113&quot;,..: 1 2 3 4 5 6 7 8</span></span>
<span id="cb239-22"><a href="getting-biological-data-from-public-repositories.html#cb239-22" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Sequence                       : Factor w/ 8 levels &quot;MARRCGPVALLLGFGLLRLCSGVWGTDTEERLVEHLLDPSRYNKLIRPATNGSELVTVQLMVSLAQLISVHEREQIMTTNVWLTQEWEDYRLTWKPEEFDNMKKVRLPSKH&quot;| __truncated__,..: 1 2 3 4 5 6 7 8</span></span>
<span id="cb239-23"><a href="getting-biological-data-from-public-repositories.html#cb239-23" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Alternative.sequence           : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-24"><a href="getting-biological-data-from-public-repositories.html#cb239-24" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Natural.variant                : Factor w/ 1 level &quot;VARIANT 287;  /note=V -&gt; L (in ENFL3; dbSNP:rs74315291);  /evidence=ECO:0000269|PubMed:11062464;  /id=VAR_01271&quot;| __truncated__: 1 NA NA NA NA NA NA NA</span></span>
<span id="cb239-25"><a href="getting-biological-data-from-public-repositories.html#cb239-25" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Non.adjacent.residues          : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-26"><a href="getting-biological-data-from-public-repositories.html#cb239-26" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Non.standard.residue           : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-27"><a href="getting-biological-data-from-public-repositories.html#cb239-27" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Non.terminal.residue           : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-28"><a href="getting-biological-data-from-public-repositories.html#cb239-28" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Sequence.conflict              : Factor w/ 2 levels &quot;CONFLICT 26;  /note=T -&gt; A (in Ref. 4; CAA05108);  /evidence=ECO:0000305; CONFLICT 426;  /note=E -&gt; A (in Ref. &quot;| __truncated__,..: 1 2 NA NA NA NA NA NA</span></span>
<span id="cb239-29"><a href="getting-biological-data-from-public-repositories.html#cb239-29" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Sequence.uncertainty           : logi  NA NA NA NA NA NA ...</span></span>
<span id="cb239-30"><a href="getting-biological-data-from-public-repositories.html#cb239-30" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Version..sequence.             : int  1 2 1 1 1 1 1 1</span></span></code></pre></div>
<p>We notice that from the above data frame, we only need the <code>$ Sequence</code> slot. However, this is a factor element and has to be transformed into vector. We will proceed as follows.</p>
<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb240-1"><a href="getting-biological-data-from-public-repositories.html#cb240-1" aria-hidden="true" tabindex="-1"></a>other_CHRNB2_aa <span class="ot">&lt;-</span> other_CHRNB2_raw<span class="sc">$</span>Sequence</span>
<span id="cb240-2"><a href="getting-biological-data-from-public-repositories.html#cb240-2" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(other_CHRNB2_aa)</span>
<span id="cb240-3"><a href="getting-biological-data-from-public-repositories.html#cb240-3" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;factor&quot;</span></span>
<span id="cb240-4"><a href="getting-biological-data-from-public-repositories.html#cb240-4" aria-hidden="true" tabindex="-1"></a><span class="co"># we transform to a character vector</span></span>
<span id="cb240-5"><a href="getting-biological-data-from-public-repositories.html#cb240-5" aria-hidden="true" tabindex="-1"></a>other_CHRNB2_aa <span class="ot">&lt;-</span> <span class="fu">as.character</span>(other_CHRNB2_aa)</span>
<span id="cb240-6"><a href="getting-biological-data-from-public-repositories.html#cb240-6" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(other_CHRNB2_aa)</span>
<span id="cb240-7"><a href="getting-biological-data-from-public-repositories.html#cb240-7" aria-hidden="true" tabindex="-1"></a>other_CHRNB2_aa</span>
<span id="cb240-8"><a href="getting-biological-data-from-public-repositories.html#cb240-8" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;MARRCGPVALLLGFGLLRLCSGVWGTDTEERLVEHLLDPSRYNKLIRPATNGSELVTVQLMVSLAQLISVHEREQIMTTNVWLTQEWEDYRLTWKPEEFDNMKKVRL...&quot;                                         </span></span>
<span id="cb240-9"><a href="getting-biological-data-from-public-repositories.html#cb240-9" aria-hidden="true" tabindex="-1"></a><span class="co">#[2] &quot;MARCSNSMALLFSFGLLWLCSGVLGTDTEERLVEHLLDPSRYNKLIRPATNGSELVTVQLMVSLAQLISVHEREQIMTTNVWLTQEWEDYRLTWKPEDFDNMKKVRL...&quot;                                          </span></span>
<span id="cb240-10"><a href="getting-biological-data-from-public-repositories.html#cb240-10" aria-hidden="true" tabindex="-1"></a><span class="co">#[3] &quot;MALLRVLCLLAALRRSLCTDTEERLVEYLLDPTRYNKLIRPATNGSQLVTVQLMVSLAQLISVHEREQIMTTNVWLTQEWEDYRLTWKPEDFDNMKKVRLPSKHIWL...&quot; </span></span></code></pre></div>
<p>We notice that the and UniProt entries are not associated to the sequences and can get these names form the row names from the <code>other_CHRNB2_raw</code> data frame. By assigning such names, we can improve our ability to identify the source of the sequence by adding organism and the starting fasta character <code>&gt;</code>. We can also prepare this object to build a fasta output by adding the corresponding new line character <code>\n</code>.</p>
<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb241-1"><a href="getting-biological-data-from-public-repositories.html#cb241-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get a names character vector</span></span>
<span id="cb241-2"><a href="getting-biological-data-from-public-repositories.html#cb241-2" aria-hidden="true" tabindex="-1"></a>names_of_other_CHRNB2_aa <span class="ot">&lt;-</span> <span class="fu">rownames</span>(other_CHRNB2_raw)</span>
<span id="cb241-3"><a href="getting-biological-data-from-public-repositories.html#cb241-3" aria-hidden="true" tabindex="-1"></a>names_of_other_CHRNB2_aa</span>
<span id="cb241-4"><a href="getting-biological-data-from-public-repositories.html#cb241-4" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;P17787&quot;     &quot;Q9ERK7&quot;     &quot;P09484&quot;     &quot;A0A3Q1MJN8&quot; &quot;E7F4S7&quot;     &quot;H3B5V5&quot;     &quot;H9GMJ0&quot;     &quot;A4IIS6&quot;</span></span>
<span id="cb241-5"><a href="getting-biological-data-from-public-repositories.html#cb241-5" aria-hidden="true" tabindex="-1"></a>names_of_organism <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;human&quot;</span>, <span class="st">&quot;mouse&quot;</span>, <span class="st">&quot;chicken&quot;</span>, <span class="st">&quot;Bos_taurus&quot;</span>, <span class="st">&quot;Danio_rerio&quot;</span>, <span class="st">&quot;coelacanth&quot;</span>, <span class="st">&quot;Anolis&quot;</span>, <span class="st">&quot;Xenopus_tropicalis&quot;</span>)</span>
<span id="cb241-6"><a href="getting-biological-data-from-public-repositories.html#cb241-6" aria-hidden="true" tabindex="-1"></a><span class="co"># we can combine names</span></span>
<span id="cb241-7"><a href="getting-biological-data-from-public-repositories.html#cb241-7" aria-hidden="true" tabindex="-1"></a>final_names <span class="ot">&lt;-</span> <span class="fu">character</span>()</span>
<span id="cb241-8"><a href="getting-biological-data-from-public-repositories.html#cb241-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(names_of_organism)) { final_names[i] <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;&gt;&quot;</span>,names_of_organism[i],<span class="st">&quot;_&quot;</span>,names_of_other_CHRNB2_aa[i],<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)}</span>
<span id="cb241-9"><a href="getting-biological-data-from-public-repositories.html#cb241-9" aria-hidden="true" tabindex="-1"></a>final_names</span>
<span id="cb241-10"><a href="getting-biological-data-from-public-repositories.html#cb241-10" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;&gt;human_P17787\n&quot;              &quot;&gt;mouse_Q9ERK7\n&quot;              &quot;&gt;chicken_P09484\n&quot;           </span></span>
<span id="cb241-11"><a href="getting-biological-data-from-public-repositories.html#cb241-11" aria-hidden="true" tabindex="-1"></a><span class="co">#[4] &quot;&gt;Bos_taurus_A0A3Q1MJN8\n&quot;     &quot;&gt;Danio_rerio_E7F4S7\n&quot;        &quot;&gt;coelacanth_H3B5V5\n&quot;        </span></span>
<span id="cb241-12"><a href="getting-biological-data-from-public-repositories.html#cb241-12" aria-hidden="true" tabindex="-1"></a><span class="co">#[7] &quot;&gt;Anolis_H9GMJ0\n&quot;             &quot;&gt;Xenopus_tropicalis_A4IIS6\n&quot;</span></span></code></pre></div>
<p>We can build the final FASTA output to be written in a text file. By collapsing both vectors with a new line character <code>\n</code>.</p>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="getting-biological-data-from-public-repositories.html#cb242-1" aria-hidden="true" tabindex="-1"></a>end_uniprot_fasta <span class="ot">&lt;-</span> <span class="fu">paste0</span>(final_names,other_CHRNB2_aa, <span class="at">collapse =</span> <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb242-2"><a href="getting-biological-data-from-public-repositories.html#cb242-2" aria-hidden="true" tabindex="-1"></a>end_uniprot_fasta</span>
<span id="cb242-3"><a href="getting-biological-data-from-public-repositories.html#cb242-3" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;&gt;human_P17787\nMARRCGPVALLLGFGLLRLCSGVWGTDTEERLVEHLLDPSRYNKLIRPATNGSELVTVQLMVSLAQLISVHEREQIMTTNVWLTQEWEDYRLTWKPEEFDNMKKVRL..</span></span>
<span id="cb242-4"><a href="getting-biological-data-from-public-repositories.html#cb242-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(end_uniprot_fasta)</span>
<span id="cb242-5"><a href="getting-biological-data-from-public-repositories.html#cb242-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;human_P17787</span></span>
<span id="cb242-6"><a href="getting-biological-data-from-public-repositories.html#cb242-6" aria-hidden="true" tabindex="-1"></a><span class="co">#MARRCGPVALLLGFGLLRLCSGVWGTDTEERLVEHLLDPSRYNKLIRPATNGSELVTVQLMVSLAQLISVHEREQIMTTNVWLTQEWEDYRLTWKPEEFDNMKKVRLPSKHIWLPDVVLYNNADGMY...</span></span>
<span id="cb242-7"><a href="getting-biological-data-from-public-repositories.html#cb242-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;mouse_Q9ERK7</span></span>
<span id="cb242-8"><a href="getting-biological-data-from-public-repositories.html#cb242-8" aria-hidden="true" tabindex="-1"></a><span class="co">#MARCSNSMALLFSFGLLWLCSGVLGTDTEERLVEHLLDPSRYNKLIRPATNGSELVTVQLMVSLAQLISVHEREQIMTTNVWLTQEWEDYRLTWKPEDFDNMKKVRLPSKHIWLPDVVLYNNADGMY...</span></span>
<span id="cb242-9"><a href="getting-biological-data-from-public-repositories.html#cb242-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;chicken_P09484</span></span>
<span id="cb242-10"><a href="getting-biological-data-from-public-repositories.html#cb242-10" aria-hidden="true" tabindex="-1"></a><span class="co">#MALLRVLCLLAALRRSLCTDTEERLVEYLLDPTRYNKLIRPATNGSQLVTVQLMVSLAQLISVHEREQIMTTNVWLTQEWEDYRLTWKPEDFDNMKKVRLPSKHIWLPDVVLYNNADGMYEVSFYSN...</span></span>
<span id="cb242-11"><a href="getting-biological-data-from-public-repositories.html#cb242-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb242-12"><a href="getting-biological-data-from-public-repositories.html#cb242-12" aria-hidden="true" tabindex="-1"></a><span class="co"># we write the output to a text file -- REMEMBER TO SET A PATH TO THE FOLDER THAT YOU WANT IN YOUR COMPUTER</span></span>
<span id="cb242-13"><a href="getting-biological-data-from-public-repositories.html#cb242-13" aria-hidden="true" tabindex="-1"></a>name_of_file <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;acetylcholine_receptor_Uniprot.txt&quot;</span>)</span>
<span id="cb242-14"><a href="getting-biological-data-from-public-repositories.html#cb242-14" aria-hidden="true" tabindex="-1"></a>name_of_file</span>
<span id="cb242-15"><a href="getting-biological-data-from-public-repositories.html#cb242-15" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;acetylcholine_receptor_popset_1248341763.txt&quot;</span></span>
<span id="cb242-16"><a href="getting-biological-data-from-public-repositories.html#cb242-16" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(end_uniprot_fasta, <span class="at">file =</span> name_of_file)</span></code></pre></div>
<p>Finally, we can import this updated file into R using <strong>Biostrings</strong>.</p>
<div class="sourceCode" id="cb243"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb243-1"><a href="getting-biological-data-from-public-repositories.html#cb243-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get the path to the uniprot file with the amino acid sequences -- THIS IS EXCLUSIVE FOR YOUR COMPUTER AND IT IS NOT THE PATH SHOWN BELOW</span></span>
<span id="cb243-2"><a href="getting-biological-data-from-public-repositories.html#cb243-2" aria-hidden="true" tabindex="-1"></a>my_uniprot_as_aa_stringset <span class="ot">&lt;-</span> <span class="fu">readAAStringSet</span>(<span class="at">filepath =</span> <span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/my_working_directory/acetylcholine_receptor_Uniprot.txt&quot;</span>, </span>
<span id="cb243-3"><a href="getting-biological-data-from-public-repositories.html#cb243-3" aria-hidden="true" tabindex="-1"></a>                                               <span class="at">format =</span> <span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb243-4"><a href="getting-biological-data-from-public-repositories.html#cb243-4" aria-hidden="true" tabindex="-1"></a>my_uniprot_as_aa_stringset</span>
<span id="cb243-5"><a href="getting-biological-data-from-public-repositories.html#cb243-5" aria-hidden="true" tabindex="-1"></a><span class="co">#  A AAStringSet instance of length 8</span></span>
<span id="cb243-6"><a href="getting-biological-data-from-public-repositories.html#cb243-6" aria-hidden="true" tabindex="-1"></a><span class="co">#   width seq                                                                                       names               </span></span>
<span id="cb243-7"><a href="getting-biological-data-from-public-repositories.html#cb243-7" aria-hidden="true" tabindex="-1"></a><span class="co">#[1]   502 MARRCGPVALLLGFGLLRLCSGVWGTDTEERLVEHLLDPSRYN...RLFLWIFVFVCVFGTIGMFLQPLFQNYTTTTFLHSDHSAPSSK human_P17787</span></span>
<span id="cb243-8"><a href="getting-biological-data-from-public-repositories.html#cb243-8" aria-hidden="true" tabindex="-1"></a><span class="co">#[2]   501 MARCSNSMALLFSFGLLWLCSGVLGTDTEERLVEHLLDPSRYN...RLFLWIFVFVCVFGTIGMFLQPLFQNYTATTFLHSDHSAPSSK mouse_Q9ERK7</span></span>
<span id="cb243-9"><a href="getting-biological-data-from-public-repositories.html#cb243-9" aria-hidden="true" tabindex="-1"></a><span class="co">#[3]   491 MALLRVLCLLAALRRSLCTDTEERLVEYLLDPTRYNKLIRPAT...RLFLWIFVFVCVFGTVGMFLQPLFQNYATNSLLQLGQGTPTSK chicken_P09484</span></span>
<span id="cb243-10"><a href="getting-biological-data-from-public-repositories.html#cb243-10" aria-hidden="true" tabindex="-1"></a><span class="co">#[4]   509 MAWLSGPKAMLLSFGLLGLCSGVWGTDTEERLVEHLLDPSRYN...FVCVFGTIGMFLQPLFQNYATATFLHADHSAPSSKCVLSPPEI Bos_taurus_A0A3Q1...</span></span>
<span id="cb243-11"><a href="getting-biological-data-from-public-repositories.html#cb243-11" aria-hidden="true" tabindex="-1"></a><span class="co">#[5]   543 MMALWTLFCILAIVKSGYGADTEERLVEHLLNPAHYNKLIRPA...VAMVIDRLFLWIFVFVCVFGTIGMFLQPLFQNYTAKTITHTPG Danio_rerio_E7F4S7</span></span>
<span id="cb243-12"><a href="getting-biological-data-from-public-repositories.html#cb243-12" aria-hidden="true" tabindex="-1"></a><span class="co">#[6]   478 SLAMDTEERLVGHLLNPAHYNKLIRPATNRSEVVTVQLMVSLA...MVIDRLFLWIFVFVCIFGTLGMFLQPVFQNSSFDSLPQKTNAA coelacanth_H3B5V5</span></span>
<span id="cb243-13"><a href="getting-biological-data-from-public-repositories.html#cb243-13" aria-hidden="true" tabindex="-1"></a><span class="co">#[7]   509 CAASSHFAACPASGSLLGLSAGVLGTDTEERLVEHLLDPLRYN...RLFLWIFVFVCVFGTIGMFLQPLFQNYATNSLLQIHQGAPGSK Anolis_H9GMJ0</span></span>
<span id="cb243-14"><a href="getting-biological-data-from-public-repositories.html#cb243-14" aria-hidden="true" tabindex="-1"></a><span class="co">#[8]   499 MIRTGMAPLLAALYLLLGLLPGCLGTDTEERLVEHLLDPSRYN...VIDRLFLWIFVFVCVFGTIGMFLQPLFQNYTTNALVHMNHAAN Xenopus_tropicali...</span></span></code></pre></div>
<p><strong>17)</strong> We can also append both AAStringSets: <code>my_popset_as_aa_stringset</code> and <code>my_uniprot_as_aa_stringset</code> for further downstream analyses with the function <code>c()</code>.</p>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="getting-biological-data-from-public-repositories.html#cb244-1" aria-hidden="true" tabindex="-1"></a>all_aa_stringset <span class="ot">&lt;-</span> <span class="fu">c</span>(my_popset_as_aa_stringset, my_uniprot_as_aa_stringset)</span>
<span id="cb244-2"><a href="getting-biological-data-from-public-repositories.html#cb244-2" aria-hidden="true" tabindex="-1"></a>all_aa_stringset</span>
<span id="cb244-3"><a href="getting-biological-data-from-public-repositories.html#cb244-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  A AAStringSet instance of length 38</span></span>
<span id="cb244-4"><a href="getting-biological-data-from-public-repositories.html#cb244-4" aria-hidden="true" tabindex="-1"></a><span class="co">#     width seq                                                                                      names               </span></span>
<span id="cb244-5"><a href="getting-biological-data-from-public-repositories.html#cb244-5" aria-hidden="true" tabindex="-1"></a><span class="co"># [1]    65 ADGMYEVSFYSNAVVSHDGSIFWLPPAIYKSACKIEVKHFPFDQQNCTMKFRSWTYDRTELDLVL                        ATG31811.1 nicoti...</span></span>
<span id="cb244-6"><a href="getting-biological-data-from-public-repositories.html#cb244-6" aria-hidden="true" tabindex="-1"></a><span class="co"># [2]   501 MTALLLVLHLSLLGLVTRSMGTDTEERLVEFLLDPSRYNKLIR...LFLWIFVFVCVFGTIGMFLQPLFQNYTSNTLIQLNHGTPASN ATG31810.1 nicoti...</span></span>
<span id="cb244-7"><a href="getting-biological-data-from-public-repositories.html#cb244-7" aria-hidden="true" tabindex="-1"></a><span class="co"># [3]    67 ADGMYEVSFYSNAVVSHDGSIFWLPPAIYKSACKIEVKHFPFDQQNCTMKFRSWTYDRTELDLVLKS                      ATG31809.1 nicoti...</span></span>
<span id="cb244-8"><a href="getting-biological-data-from-public-repositories.html#cb244-8" aria-hidden="true" tabindex="-1"></a><span class="co"># [4]    44 ADGMYEVSFYSNAVVSHDGSIFWLPPAIYKSACKIEVKHFPFDQ                                             ATG31808.1 nicoti...</span></span>
<span id="cb244-9"><a href="getting-biological-data-from-public-repositories.html#cb244-9" aria-hidden="true" tabindex="-1"></a><span class="co"># [5]   501 MAALLXLLHLGLLGIVSRCLCTDTEERLVEFLLDPSRYNKLIR...LFLWIFIFVCVFGTIGMFLQPLFQNYTTNTLLQLNPAAPASK ATG31807.1 nicoti...</span></span>
<span id="cb244-10"><a href="getting-biological-data-from-public-repositories.html#cb244-10" aria-hidden="true" tabindex="-1"></a><span class="co"># ...   ... ...</span></span>
<span id="cb244-11"><a href="getting-biological-data-from-public-repositories.html#cb244-11" aria-hidden="true" tabindex="-1"></a><span class="co">#[34]   509 MAWLSGPKAMLLSFGLLGLCSGVWGTDTEERLVEHLLDPSRYN...VCVFGTIGMFLQPLFQNYATATFLHADHSAPSSKCVLSPPEI Bos_taurus_A0A3Q1...</span></span>
<span id="cb244-12"><a href="getting-biological-data-from-public-repositories.html#cb244-12" aria-hidden="true" tabindex="-1"></a><span class="co">#[35]   543 MMALWTLFCILAIVKSGYGADTEERLVEHLLNPAHYNKLIRPA...AMVIDRLFLWIFVFVCVFGTIGMFLQPLFQNYTAKTITHTPG Danio_rerio_E7F4S7</span></span>
<span id="cb244-13"><a href="getting-biological-data-from-public-repositories.html#cb244-13" aria-hidden="true" tabindex="-1"></a><span class="co">#[36]   478 SLAMDTEERLVGHLLNPAHYNKLIRPATNRSEVVTVQLMVSLA...VIDRLFLWIFVFVCIFGTLGMFLQPVFQNSSFDSLPQKTNAA coelacanth_H3B5V5</span></span>
<span id="cb244-14"><a href="getting-biological-data-from-public-repositories.html#cb244-14" aria-hidden="true" tabindex="-1"></a><span class="co">#[37]   509 CAASSHFAACPASGSLLGLSAGVLGTDTEERLVEHLLDPLRYN...LFLWIFVFVCVFGTIGMFLQPLFQNYATNSLLQIHQGAPGSK Anolis_H9GMJ0</span></span>
<span id="cb244-15"><a href="getting-biological-data-from-public-repositories.html#cb244-15" aria-hidden="true" tabindex="-1"></a><span class="co">#[38]   499 MIRTGMAPLLAALYLLLGLLPGCLGTDTEERLVEHLLDPSRYN...IDRLFLWIFVFVCVFGTIGMFLQPLFQNYTTNALVHMNHAAN Xenopus_tropicali...</span></span></code></pre></div>
</div>
<div id="other-databases-ensembl" class="section level2" number="5.6">
<h2><span class="header-section-number">5.6</span> Other databases: Ensembl</h2>
<p><a href="https://useast.ensembl.org/index.html" target="_blank"><strong>Ensembl</strong></a> is an online archive for vertebrate genomes that are used in comparative genomics, evolution, sequence variation and transcriptional regulation. Ensembl is quite extensive and convoluted to access and export data.</p>
<p><img src="figures/session_5_figs/9_Ensembl_from.png" width="100%" /></p>
<p>The R-package <a href="https://bioconductor.org/packages/release/bioc/html/biomaRt.html" target="_blank"><strong>biomaRt</strong></a> provides a somehow easy access to datasets present or crosslinked with Ensembl.</p>
<p><strong>18)</strong> We will illustrate how to access the acetylcholine receptor CHRNB2 sequence in the genome of <em>Xenopus tropicalis</em> frog model system.</p>
<p>First, you might need to install <code>biomaRt</code> from Bioconductor.</p>
<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb245-1"><a href="getting-biological-data-from-public-repositories.html#cb245-1" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;biomaRt&quot;</span>)</span>
<span id="cb245-2"><a href="getting-biological-data-from-public-repositories.html#cb245-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span></code></pre></div>
<p>We can get a list to which biomaRt can connect using the function <code>listMarts()</code>.</p>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="getting-biological-data-from-public-repositories.html#cb246-1" aria-hidden="true" tabindex="-1"></a><span class="fu">listMarts</span>()</span>
<span id="cb246-2"><a href="getting-biological-data-from-public-repositories.html#cb246-2" aria-hidden="true" tabindex="-1"></a><span class="co">#               biomart                version</span></span>
<span id="cb246-3"><a href="getting-biological-data-from-public-repositories.html#cb246-3" aria-hidden="true" tabindex="-1"></a><span class="co">#1 ENSEMBL_MART_ENSEMBL      Ensembl Genes 102</span></span>
<span id="cb246-4"><a href="getting-biological-data-from-public-repositories.html#cb246-4" aria-hidden="true" tabindex="-1"></a><span class="co">#2   ENSEMBL_MART_MOUSE      Mouse strains 102</span></span>
<span id="cb246-5"><a href="getting-biological-data-from-public-repositories.html#cb246-5" aria-hidden="true" tabindex="-1"></a><span class="co">#3     ENSEMBL_MART_SNP  Ensembl Variation 102</span></span>
<span id="cb246-6"><a href="getting-biological-data-from-public-repositories.html#cb246-6" aria-hidden="true" tabindex="-1"></a><span class="co">#4 ENSEMBL_MART_FUNCGEN Ensembl Regulation 102</span></span></code></pre></div>
<p>We can also type <code>ensembl</code> using the function <code>useMart()</code> to access all accessible databases.</p>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="getting-biological-data-from-public-repositories.html#cb247-1" aria-hidden="true" tabindex="-1"></a>mart <span class="ot">&lt;-</span> <span class="fu">useMart</span>(<span class="st">&#39;ensembl&#39;</span>)</span>
<span id="cb247-2"><a href="getting-biological-data-from-public-repositories.html#cb247-2" aria-hidden="true" tabindex="-1"></a><span class="co"># this will give you at least 203 dataset to search</span></span>
<span id="cb247-3"><a href="getting-biological-data-from-public-repositories.html#cb247-3" aria-hidden="true" tabindex="-1"></a><span class="fu">listDatasets</span>(mart)</span>
<span id="cb247-4"><a href="getting-biological-data-from-public-repositories.html#cb247-4" aria-hidden="true" tabindex="-1"></a><span class="co">#                           dataset                                                  description                           version</span></span>
<span id="cb247-5"><a href="getting-biological-data-from-public-repositories.html#cb247-5" aria-hidden="true" tabindex="-1"></a><span class="co">#1         acalliptera_gene_ensembl                             Eastern happy genes (fAstCal1.2)                        fAstCal1.2</span></span>
<span id="cb247-6"><a href="getting-biological-data-from-public-repositories.html#cb247-6" aria-hidden="true" tabindex="-1"></a><span class="co">#2       acarolinensis_gene_ensembl                               Anole lizard genes (AnoCar2.0)                         AnoCar2.0</span></span>
<span id="cb247-7"><a href="getting-biological-data-from-public-repositories.html#cb247-7" aria-hidden="true" tabindex="-1"></a><span class="co">#3        acchrysaetos_gene_ensembl                              Golden eagle genes (bAquChr1.2)                        bAquChr1.2</span></span>
<span id="cb247-8"><a href="getting-biological-data-from-public-repositories.html#cb247-8" aria-hidden="true" tabindex="-1"></a><span class="co">#4        acitrinellus_gene_ensembl                               Midas cichlid genes (Midas_v5)                          Midas_v5</span></span>
<span id="cb247-9"><a href="getting-biological-data-from-public-repositories.html#cb247-9" aria-hidden="true" tabindex="-1"></a><span class="co">#5        amelanoleuca_gene_ensembl                                        Panda genes (ailMel1)                           ailMel1</span></span>
<span id="cb247-10"><a href="getting-biological-data-from-public-repositories.html#cb247-10" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span></code></pre></div>
<p>For our example, we will use the <em>Xenopus tropicalis</em> dataset named in the list above as <code>"xtropicalis_gene_ensembl"</code>. However, we need to be more specific how we want to search that database. For that purpose, we can use the function <code>listFilters()</code> to list the filters available in our selected dataset.</p>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="getting-biological-data-from-public-repositories.html#cb248-1" aria-hidden="true" tabindex="-1"></a>xenopus_mart <span class="ot">&lt;-</span> <span class="fu">useMart</span>(<span class="st">&quot;ensembl&quot;</span>, <span class="at">dataset=</span><span class="st">&quot;xtropicalis_gene_ensembl&quot;</span>)</span>
<span id="cb248-2"><a href="getting-biological-data-from-public-repositories.html#cb248-2" aria-hidden="true" tabindex="-1"></a><span class="fu">listFilters</span>(xenopus_mart)</span>
<span id="cb248-3"><a href="getting-biological-data-from-public-repositories.html#cb248-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                              name                                                      description</span></span>
<span id="cb248-4"><a href="getting-biological-data-from-public-repositories.html#cb248-4" aria-hidden="true" tabindex="-1"></a><span class="co">#1                  chromosome_name                                         Chromosome/scaffold name</span></span>
<span id="cb248-5"><a href="getting-biological-data-from-public-repositories.html#cb248-5" aria-hidden="true" tabindex="-1"></a><span class="co">#2                            start                                                            Start</span></span>
<span id="cb248-6"><a href="getting-biological-data-from-public-repositories.html#cb248-6" aria-hidden="true" tabindex="-1"></a><span class="co">#3                              end                                                              End</span></span>
<span id="cb248-7"><a href="getting-biological-data-from-public-repositories.html#cb248-7" aria-hidden="true" tabindex="-1"></a><span class="co">#4                           strand                                                           Strand</span></span>
<span id="cb248-8"><a href="getting-biological-data-from-public-repositories.html#cb248-8" aria-hidden="true" tabindex="-1"></a><span class="co">#5               chromosomal_region                           e.g. 1:100:10000:-1, 1:100000:200000:1</span></span>
<span id="cb248-9"><a href="getting-biological-data-from-public-repositories.html#cb248-9" aria-hidden="true" tabindex="-1"></a><span class="co">#6       with_entrezgene_trans_name                            With EntrezGene transcript name ID(s)</span></span>
<span id="cb248-10"><a href="getting-biological-data-from-public-repositories.html#cb248-10" aria-hidden="true" tabindex="-1"></a><span class="co">#7                        with_embl                           With European Nucleotide Archive ID(s)</span></span>
<span id="cb248-11"><a href="getting-biological-data-from-public-repositories.html#cb248-11" aria-hidden="true" tabindex="-1"></a><span class="co">#8                with_arrayexpress                                      With Expression Atlas ID(s)</span></span>
<span id="cb248-12"><a href="getting-biological-data-from-public-repositories.html#cb248-12" aria-hidden="true" tabindex="-1"></a><span class="co">#9                          with_go                                                    With GO ID(s)</span></span>
<span id="cb248-13"><a href="getting-biological-data-from-public-repositories.html#cb248-13" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span></code></pre></div>
<p>We will retrieve based on the name of gene in UniProt (i.e., CHRNB2) so two filter seem to be useful based on the description column <code>hgnc_symbol</code> and <code>uniprot_gn_symbol</code>. We will try to retrieve such gene sequence using the function <code>getSequence()</code>.</p>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="getting-biological-data-from-public-repositories.html#cb249-1" aria-hidden="true" tabindex="-1"></a>xenopus_CHRNB2_seq_hgnc_symbol <span class="ot">&lt;-</span> <span class="fu">getSequence</span>(<span class="at">id =</span> <span class="st">&quot;chrnb2&quot;</span>, </span>
<span id="cb249-2"><a href="getting-biological-data-from-public-repositories.html#cb249-2" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">type =</span> <span class="st">&quot;uniprot_gn_symbol&quot;</span>, </span>
<span id="cb249-3"><a href="getting-biological-data-from-public-repositories.html#cb249-3" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">seqType =</span> <span class="st">&quot;peptide&quot;</span>, </span>
<span id="cb249-4"><a href="getting-biological-data-from-public-repositories.html#cb249-4" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">mart =</span> xenopus_mart)</span>
<span id="cb249-5"><a href="getting-biological-data-from-public-repositories.html#cb249-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb249-6"><a href="getting-biological-data-from-public-repositories.html#cb249-6" aria-hidden="true" tabindex="-1"></a>xenopus_CHRNB2_seq_uniprot_id <span class="ot">&lt;-</span> <span class="fu">getSequence</span>(<span class="at">id =</span> <span class="st">&quot;A4IIS6&quot;</span>, </span>
<span id="cb249-7"><a href="getting-biological-data-from-public-repositories.html#cb249-7" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">type =</span> <span class="st">&quot;uniprot_gn_id&quot;</span>, </span>
<span id="cb249-8"><a href="getting-biological-data-from-public-repositories.html#cb249-8" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">seqType =</span> <span class="st">&quot;peptide&quot;</span>, </span>
<span id="cb249-9"><a href="getting-biological-data-from-public-repositories.html#cb249-9" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">mart =</span> xenopus_mart, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb249-10"><a href="getting-biological-data-from-public-repositories.html#cb249-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb249-11"><a href="getting-biological-data-from-public-repositories.html#cb249-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb249-12"><a href="getting-biological-data-from-public-repositories.html#cb249-12" aria-hidden="true" tabindex="-1"></a><span class="fu">show</span>(xenopus_CHRNB2_seq_uniprot_id)</span></code></pre></div>
</div>
<div id="other-databases-taxonomy-phylogenetic-trees-and-species-distributions" class="section level2" number="5.7">
<h2><span class="header-section-number">5.7</span> Other databases: Taxonomy, phylogenetic trees and species distributions</h2>
<p>Accessing other biological repositories through R is in some cases a hit-and-miss. Some with stand and be reliable, but the depend on the source database.</p>
<p>Here is a list of some R-packages and the databases that you can access. Most of these will have vignettes for you to follow:</p>
<ol style="list-style-type: decimal">
<li><a href="https://cran.r-project.org/web/packages/taxize/index.html" target="_blank"><strong>taxize</strong></a> – Taxonomic Information from Around the Web.</li>
<li><a href="https://cran.r-project.org/web/packages/BIEN/index.html" target="_blank"><strong>BIEN</strong></a> – Tools for Accessing the Botanical Information and Ecology Network Database.</li>
<li><a href="https://cran.r-project.org/web/packages/rotl/index.html" target="_blank"><strong>rotl</strong></a> – Interface to the ‘Open Tree of Life’ API.</li>
<li><a href="https://cran.r-project.org/web/packages/rgbif/index.html" target="_blank"><strong>rgbif</strong></a> – Interface to the Global ‘Biodiversity’ Information Facility API.</li>
<li><a href="https://cran.r-project.org/web/packages/rvertnet/index.html" target="_blank"><strong>rvertnet</strong></a> – Search ‘Vertnet’, a ‘Database’ of Vertebrate Specimen Records.</li>
<li><a href="https://cran.r-project.org/web/packages/spocc/index.html" target="_blank"><strong>spocc</strong></a> – Interface to Species Occurrence Data Sources.</li>
</ol>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="text-editing-and-data-transformations.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="basic-statistics-in-r.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/05-Getting_biological_data_v13.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
