<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Session 9 – Graphics in R – ggplot2 | Bioinformatics Workshop Gitbook</title>
  <meta name="description" content="none" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Session 9 – Graphics in R – ggplot2 | Bioinformatics Workshop Gitbook" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="none" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Session 9 – Graphics in R – ggplot2 | Bioinformatics Workshop Gitbook" />
  
  <meta name="twitter:description" content="none" />
  

<meta name="author" content="Juan C. Santos" />


<meta name="date" content="2021-02-01" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="exploring-multidimensional-data-in-r.html"/>

<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics Workshop Gitbook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html"><i class="fa fa-check"></i><b>1</b> – Getting Started with R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#why-we-use-r-in-this-workshop"><i class="fa fa-check"></i><b>1.1</b> Why we use R in this workshop?</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#get-a-good-text-editor"><i class="fa fa-check"></i><b>1.2</b> Get a good text editor</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-r"><i class="fa fa-check"></i><b>1.4</b> Installing R</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#annotating-code-in-r"><i class="fa fa-check"></i><b>1.5</b> Annotating code in R</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#defining-a-working-directory-in-r"><i class="fa fa-check"></i><b>1.6</b> Defining a working directory in R</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="your-first-r-session-installing-packages-and-importingexporting-data.html"><a href="your-first-r-session-installing-packages-and-importingexporting-data.html"><i class="fa fa-check"></i><b>2</b> – Your First R Session, Installing Packages and Importing/Exporting Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="your-first-r-session-installing-packages-and-importingexporting-data.html"><a href="your-first-r-session-installing-packages-and-importingexporting-data.html#your-first-session-with-r"><i class="fa fa-check"></i><b>2.1</b> Your first session with R</a></li>
<li class="chapter" data-level="2.2" data-path="your-first-r-session-installing-packages-and-importingexporting-data.html"><a href="your-first-r-session-installing-packages-and-importingexporting-data.html#doing-some-base-functions-and-getting-help"><i class="fa fa-check"></i><b>2.2</b> Doing some base functions and getting help</a></li>
<li class="chapter" data-level="2.3" data-path="your-first-r-session-installing-packages-and-importingexporting-data.html"><a href="your-first-r-session-installing-packages-and-importingexporting-data.html#about-r-packages"><i class="fa fa-check"></i><b>2.3</b> About R-packages</a></li>
<li class="chapter" data-level="2.4" data-path="your-first-r-session-installing-packages-and-importingexporting-data.html"><a href="your-first-r-session-installing-packages-and-importingexporting-data.html#installing-packages-in-r"><i class="fa fa-check"></i><b>2.4</b> Installing packages in R</a></li>
<li class="chapter" data-level="2.5" data-path="your-first-r-session-installing-packages-and-importingexporting-data.html"><a href="your-first-r-session-installing-packages-and-importingexporting-data.html#importing-and-exporting-data-into-r"><i class="fa fa-check"></i><b>2.5</b> Importing and exporting data into R</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html"><i class="fa fa-check"></i><b>3</b> – Basic Data Structures in R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#scalars-and-vectors"><i class="fa fa-check"></i><b>3.1</b> Scalars and vectors</a></li>
<li class="chapter" data-level="3.2" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-frames"><i class="fa fa-check"></i><b>3.3</b> Data Frames</a></li>
<li class="chapter" data-level="3.4" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tables"><i class="fa fa-check"></i><b>3.4</b> Tables</a></li>
<li class="chapter" data-level="3.5" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#lists"><i class="fa fa-check"></i><b>3.5</b> Lists</a></li>
<li class="chapter" data-level="3.6" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-tables"><i class="fa fa-check"></i><b>3.6</b> Data Tables</a></li>
<li class="chapter" data-level="3.7" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tibbles"><i class="fa fa-check"></i><b>3.7</b> Tibbles</a></li>
<li class="chapter" data-level="3.8" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#dnastringsets-rnastringsets-and-aastringsets"><i class="fa fa-check"></i><b>3.8</b> DNAStringSets, RNAStringSets and AAStringSets</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="text-editing-applying-transformations-and-functions.html"><a href="text-editing-applying-transformations-and-functions.html"><i class="fa fa-check"></i><b>4</b> – Text Editing, Applying Transformations and Functions</a>
<ul>
<li class="chapter" data-level="4.1" data-path="text-editing-applying-transformations-and-functions.html"><a href="text-editing-applying-transformations-and-functions.html#working-with-character-strings"><i class="fa fa-check"></i><b>4.1</b> Working with character strings</a></li>
<li class="chapter" data-level="4.2" data-path="text-editing-applying-transformations-and-functions.html"><a href="text-editing-applying-transformations-and-functions.html#regular-expressions-regex"><i class="fa fa-check"></i><b>4.2</b> Regular expressions (Regex)</a></li>
<li class="chapter" data-level="4.3" data-path="text-editing-applying-transformations-and-functions.html"><a href="text-editing-applying-transformations-and-functions.html#comparing-ordering-and-simplifying-vectors"><i class="fa fa-check"></i><b>4.3</b> Comparing, ordering and simplifying vectors</a></li>
<li class="chapter" data-level="4.4" data-path="text-editing-applying-transformations-and-functions.html"><a href="text-editing-applying-transformations-and-functions.html#some-useful-functions-of-the-r-package-stringr"><i class="fa fa-check"></i><b>4.4</b> Some useful functions of the R-package ‘stringr’</a></li>
<li class="chapter" data-level="4.5" data-path="text-editing-applying-transformations-and-functions.html"><a href="text-editing-applying-transformations-and-functions.html#data-transformations"><i class="fa fa-check"></i><b>4.5</b> Data transformations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html"><i class="fa fa-check"></i><b>5</b> – Getting Biological Data from Public Repositories</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#ncbi-databases-genbank-and-uniprot"><i class="fa fa-check"></i><b>5.1</b> NCBI databases: GenBank and UniProt</a></li>
<li class="chapter" data-level="5.2" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-sequence-databases-uniprot"><i class="fa fa-check"></i><b>5.2</b> Other sequence databases: UniProt</a></li>
<li class="chapter" data-level="5.3" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-sequence-databases-ensembl"><i class="fa fa-check"></i><b>5.3</b> Other sequence databases: Ensembl</a></li>
<li class="chapter" data-level="5.4" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-biological-databases-taxonomy-phylogenetic-trees-and-species-distributions"><i class="fa fa-check"></i><b>5.4</b> Other biological databases: Taxonomy, phylogenetic trees and species distributions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html"><i class="fa fa-check"></i><b>6</b> – Basic Statistics in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#exploring-your-data"><i class="fa fa-check"></i><b>6.1</b> Exploring your data</a></li>
<li class="chapter" data-level="6.2" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#normalizing-your-data"><i class="fa fa-check"></i><b>6.2</b> Normalizing your data</a></li>
<li class="chapter" data-level="6.3" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-means"><i class="fa fa-check"></i><b>6.3</b> Comparing means</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html"><i class="fa fa-check"></i><b>7</b> – Regressions and ANOVA in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#linear-regressions-in-r"><i class="fa fa-check"></i><b>7.1</b> Linear regressions in R</a></li>
<li class="chapter" data-level="7.2" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#selecting-predictors-for-regressions-in-r"><i class="fa fa-check"></i><b>7.2</b> Selecting predictors for regressions in R</a></li>
<li class="chapter" data-level="7.3" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions"><i class="fa fa-check"></i><b>7.3</b> Logistic regressions</a></li>
<li class="chapter" data-level="7.4" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#non-linear-regressions"><i class="fa fa-check"></i><b>7.4</b> Non-linear regressions</a></li>
<li class="chapter" data-level="7.5" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova"><i class="fa fa-check"></i><b>7.5</b> ANOVA</a></li>
<li class="chapter" data-level="7.6" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#one-way-anova"><i class="fa fa-check"></i><b>7.6</b> One-way ANOVA</a></li>
<li class="chapter" data-level="7.7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#two-way-anova"><i class="fa fa-check"></i><b>7.7</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="7.8" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#post-hoc-tests"><i class="fa fa-check"></i><b>7.8</b> Post-hoc tests</a></li>
<li class="chapter" data-level="7.9" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#alternative-to-anova-kruskal-wallis-test"><i class="fa fa-check"></i><b>7.9</b> Alternative to ANOVA: Kruskal-Wallis Test</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html"><i class="fa fa-check"></i><b>8</b> – Exploring Multidimensional Data in R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>8.1</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="8.2" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html#multidimensional-scaling-mds"><i class="fa fa-check"></i><b>8.2</b> Multidimensional scaling (MDS)</a></li>
<li class="chapter" data-level="8.3" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html#cluster-analysis-using-dfa"><i class="fa fa-check"></i><b>8.3</b> Cluster analysis using DFA</a></li>
<li class="chapter" data-level="8.4" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html#cluster-analysis-using-k-means-unsupervised"><i class="fa fa-check"></i><b>8.4</b> Cluster analysis using <em>k</em>-means (unsupervised)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="graphics-in-r-ggplot2.html"><a href="graphics-in-r-ggplot2.html"><i class="fa fa-check"></i><b>9</b> – Graphics in R – ggplot2</a>
<ul>
<li class="chapter" data-level="9.1" data-path="graphics-in-r-ggplot2.html"><a href="graphics-in-r-ggplot2.html#preparing-dataset-for-plot"><i class="fa fa-check"></i><b>9.1</b> Preparing dataset for plot</a></li>
<li class="chapter" data-level="9.2" data-path="graphics-in-r-ggplot2.html"><a href="graphics-in-r-ggplot2.html#color-r-packages"><i class="fa fa-check"></i><b>9.2</b> Color R-packages</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics Workshop Gitbook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="graphics-in-r-ggplot2" class="section level1" number="9">
<h1><span class="header-section-number">Session 9</span> – Graphics in R – ggplot2</h1>
<p>Plots, graphs and other visual representations are central for any visualization of data, many statistical results and simplification of complex data. The preloaded <code>graphics</code> package of R helps the easy access plots, bar charts, boxplots and other. However, with the development of the <a href="https://ggplot2.tidyverse.org/" target="_blank"><strong>ggplot2</strong></a> package that is part of the tidyverse collection. This package and its growing collection of add-ons, which include at least <a href="https://exts.ggplot2.tidyverse.org/gallery/" target="_blank"><strong>81</strong></a>, has made <a href="https://ggplot2.tidyverse.org/" target="_blank"><strong>ggplot2</strong></a> a dominant, flexible, and beautiful R graphics package. Therefore, we will explore graphics in R using this package.</p>
<p>Like most visualizations, it would be ideal to start with installing <strong>ggplot2</strong> and datasets that will serve as examples.</p>
<p><strong>1)</strong> If you have not installed <a href="https://ggplot2.tidyverse.org/" target="_blank"><strong>ggplot2</strong></a>, you should follow these instructions before we explore graphics and plots.</p>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb334-1"><a href="graphics-in-r-ggplot2.html#cb334-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb334-2"><a href="graphics-in-r-ggplot2.html#cb334-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code></pre></div>
<div id="preparing-dataset-for-plot" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> Preparing dataset for plot</h2>
<p><strong>2)</strong> We also need some reference datasets to use and plot for our graphics. We will use <code>cars</code> and the <code>airway_scaledcounts.csv</code> datasets.</p>
<div class="sourceCode" id="cb335"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb335-1"><a href="graphics-in-r-ggplot2.html#cb335-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we will load &#39;ade4&#39; package to get the bacteria dataset</span></span>
<span id="cb335-2"><a href="graphics-in-r-ggplot2.html#cb335-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb335-3"><a href="graphics-in-r-ggplot2.html#cb335-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(bacteria)</span>
<span id="cb335-4"><a href="graphics-in-r-ggplot2.html#cb335-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb335-5"><a href="graphics-in-r-ggplot2.html#cb335-5" aria-hidden="true" tabindex="-1"></a><span class="do">## we get the espaa dataframe with number of the 21 amino acids present in 43 species of bacteria</span></span>
<span id="cb335-6"><a href="graphics-in-r-ggplot2.html#cb335-6" aria-hidden="true" tabindex="-1"></a>bacteria_espaa <span class="ot">&lt;-</span> bacteria<span class="sc">$</span>espaa</span></code></pre></div>
<p>We then scale this dataset using the function <code>modify()</code> from the package <code>purrr</code>.</p>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb336-1"><a href="graphics-in-r-ggplot2.html#cb336-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load purrr package and scale the values</span></span>
<span id="cb336-2"><a href="graphics-in-r-ggplot2.html#cb336-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(purrr)</span>
<span id="cb336-3"><a href="graphics-in-r-ggplot2.html#cb336-3" aria-hidden="true" tabindex="-1"></a>bacteria_espaa_scaled <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">modify</span>(bacteria_espaa, scale)</span>
<span id="cb336-4"><a href="graphics-in-r-ggplot2.html#cb336-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(bacteria_espaa_scaled)</span>
<span id="cb336-5"><a href="graphics-in-r-ggplot2.html#cb336-5" aria-hidden="true" tabindex="-1"></a><span class="co">#               Ala        Arg         Asn        Asp        Cys        Gln        Glu        Gly        His         Ile         Leu</span></span>
<span id="cb336-6"><a href="graphics-in-r-ggplot2.html#cb336-6" aria-hidden="true" tabindex="-1"></a><span class="co">#AERPECG  0.3285668  0.7317790 -0.57189998 -0.1194444  0.1648422 -0.3942439  0.2871196  0.4766228  0.1461376 -0.04383607  0.39172834</span></span>
<span id="cb336-7"><a href="graphics-in-r-ggplot2.html#cb336-7" aria-hidden="true" tabindex="-1"></a><span class="co">#AQUAECG -0.4008694 -0.2034641 -0.22356528 -0.3025407 -0.3950725 -0.5262848  0.4634441 -0.1820602 -0.3947798  0.06415029 -0.09928685</span></span>
<span id="cb336-8"><a href="graphics-in-r-ggplot2.html#cb336-8" aria-hidden="true" tabindex="-1"></a><span class="co">#ARCFUCG  0.1073331  0.2833748  0.04401802  0.2066524  0.5498823 -0.4297732  0.8885965  0.2463444 -0.1330163  0.57892747  0.12987680</span></span>
<span id="cb336-9"><a href="graphics-in-r-ggplot2.html#cb336-9" aria-hidden="true" tabindex="-1"></a><span class="co">#BACHDCG  0.9012986  0.9875656  1.72105862  1.5171379  0.8224654  1.6598743  2.2634710  1.2984007  1.9400648  2.16751418  1.40433248</span></span>
<span id="cb336-10"><a href="graphics-in-r-ggplot2.html#cb336-10" aria-hidden="true" tabindex="-1"></a><span class="co">#BACSUCG  1.0314722  0.7271505  2.09337260  1.6457327  1.0718724  1.5394241  2.0612524  1.3252195  1.8173099  2.49733951  1.38291228</span></span>
<span id="cb336-11"><a href="graphics-in-r-ggplot2.html#cb336-11" aria-hidden="true" tabindex="-1"></a><span class="co">#BORBUCG -0.7597170 -0.7465478  0.01654507 -0.6016028 -0.9081081 -0.7293261 -0.6491864 -0.7234675 -0.8524967 -0.18035597 -0.60867882</span></span>
<span id="cb336-12"><a href="graphics-in-r-ggplot2.html#cb336-12" aria-hidden="true" tabindex="-1"></a><span class="co">#                Lys         Met        Phe         Pro        Ser        Stp          Thr         Trp        Tyr         Val</span></span>
<span id="cb336-13"><a href="graphics-in-r-ggplot2.html#cb336-13" aria-hidden="true" tabindex="-1"></a><span class="co">#AERPECG -0.36413421  0.09017282 -0.3275198  0.89114307  0.7319805  0.7113529  0.152094962  0.38406553  0.4369524  0.61086466</span></span>
<span id="cb336-14"><a href="graphics-in-r-ggplot2.html#cb336-14" aria-hidden="true" tabindex="-1"></a><span class="co">#AQUAECG  0.88477099 -0.28951403  0.1851469 -0.17796712 -0.4145196 -0.2123879 -0.323609531 -0.26598780  0.3076040  0.01812197</span></span>
<span id="cb336-15"><a href="graphics-in-r-ggplot2.html#cb336-15" aria-hidden="true" tabindex="-1"></a><span class="co">#ARCFUCG  0.79679827  0.48462587  0.5156315  0.08558657  0.1583118  0.2768920  0.008319353  0.09800132  0.6381195  0.61451548</span></span>
<span id="cb336-16"><a href="graphics-in-r-ggplot2.html#cb336-16" aria-hidden="true" tabindex="-1"></a><span class="co">#BACHDCG  2.08829038  2.07351272  2.0439218  1.06116153  1.5330792  1.4796369  1.887894271  1.20439921  2.0999898  1.67959246</span></span>
<span id="cb336-17"><a href="graphics-in-r-ggplot2.html#cb336-17" aria-hidden="true" tabindex="-1"></a><span class="co">#BACSUCG  2.95243158  2.18339064  2.1725251  1.03849240  1.9769142  1.5369105  1.887693467  1.04854940  2.2918176  1.46336710</span></span>
<span id="cb336-18"><a href="graphics-in-r-ggplot2.html#cb336-18" aria-hidden="true" tabindex="-1"></a><span class="co">#BORBUCG -0.02590849 -0.72066544 -0.3193467 -0.80668435 -0.5156512 -0.7990328 -0.785609156 -0.79075009 -0.4463524 -0.78199455</span></span></code></pre></div>
<p>We can also create a more easy to map plot with using the function <code>melt()</code> of the package <a href="https://cran.r-project.org/web/packages/reshape2/index.html" target="_blank"><strong>reshape2</strong></a>.</p>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb337-1"><a href="graphics-in-r-ggplot2.html#cb337-1" aria-hidden="true" tabindex="-1"></a><span class="do">## load reshape</span></span>
<span id="cb337-2"><a href="graphics-in-r-ggplot2.html#cb337-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb337-3"><a href="graphics-in-r-ggplot2.html#cb337-3" aria-hidden="true" tabindex="-1"></a><span class="co"># use melt() to transform</span></span>
<span id="cb337-4"><a href="graphics-in-r-ggplot2.html#cb337-4" aria-hidden="true" tabindex="-1"></a>bacteria_espaa_for_heatmap <span class="ot">&lt;-</span> <span class="fu">melt</span>(<span class="fu">as.matrix</span>(bacteria_espaa_scaled))</span>
<span id="cb337-5"><a href="graphics-in-r-ggplot2.html#cb337-5" aria-hidden="true" tabindex="-1"></a><span class="co"># transform factor variables to character</span></span>
<span id="cb337-6"><a href="graphics-in-r-ggplot2.html#cb337-6" aria-hidden="true" tabindex="-1"></a>factor_columns <span class="ot">&lt;-</span> <span class="fu">sapply</span>(bacteria_espaa_for_heatmap, is.factor)</span>
<span id="cb337-7"><a href="graphics-in-r-ggplot2.html#cb337-7" aria-hidden="true" tabindex="-1"></a>bacteria_espaa_for_heatmap[factor_columns] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(bacteria_espaa_for_heatmap[factor_columns], as.character)</span>
<span id="cb337-8"><a href="graphics-in-r-ggplot2.html#cb337-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb337-9"><a href="graphics-in-r-ggplot2.html#cb337-9" aria-hidden="true" tabindex="-1"></a><span class="do">## we can rename the columns in this data frame</span></span>
<span id="cb337-10"><a href="graphics-in-r-ggplot2.html#cb337-10" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(bacteria_espaa_for_heatmap) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;bacteria_sp&quot;</span>, <span class="st">&quot;Amino_acid&quot;</span>, <span class="st">&quot;scaled_value&quot;</span>)</span>
<span id="cb337-11"><a href="graphics-in-r-ggplot2.html#cb337-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(bacteria_espaa_for_heatmap)</span>
<span id="cb337-12"><a href="graphics-in-r-ggplot2.html#cb337-12" aria-hidden="true" tabindex="-1"></a><span class="co">#  bacteria_sp Amino_acid scaled_value</span></span>
<span id="cb337-13"><a href="graphics-in-r-ggplot2.html#cb337-13" aria-hidden="true" tabindex="-1"></a><span class="co">#1     AERPECG        Ala    0.3285668</span></span>
<span id="cb337-14"><a href="graphics-in-r-ggplot2.html#cb337-14" aria-hidden="true" tabindex="-1"></a><span class="co">#2     AQUAECG        Ala   -0.4008694</span></span>
<span id="cb337-15"><a href="graphics-in-r-ggplot2.html#cb337-15" aria-hidden="true" tabindex="-1"></a><span class="co">#3     ARCFUCG        Ala    0.1073331</span></span>
<span id="cb337-16"><a href="graphics-in-r-ggplot2.html#cb337-16" aria-hidden="true" tabindex="-1"></a><span class="co">#4     BACHDCG        Ala    0.9012986</span></span>
<span id="cb337-17"><a href="graphics-in-r-ggplot2.html#cb337-17" aria-hidden="true" tabindex="-1"></a><span class="co">#5     BACSUCG        Ala    1.0314722</span></span>
<span id="cb337-18"><a href="graphics-in-r-ggplot2.html#cb337-18" aria-hidden="true" tabindex="-1"></a><span class="co">#6     BORBUCG        Ala   -0.7597170</span></span></code></pre></div>
<p><strong>3)</strong> A more complex dataset for color schemes, we will use the <code>airway_scaledcounts.csv</code> that has expression analysis data for several RNAseq experiments. This dataset is in our class <a href="https://github.com/corytophanes/BIO_BIT_Bioinformatics_209/tree/main/ref_files_pdfs" target="_blank"><strong>GitHub repository</strong></a>.</p>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="graphics-in-r-ggplot2.html#cb338-1" aria-hidden="true" tabindex="-1"></a><span class="do">## remeber to update the path to file with the datset where you downloaded in your computer -- THIS IS EXCLUSIVE FOR YOUR COMPUTER</span></span>
<span id="cb338-2"><a href="graphics-in-r-ggplot2.html#cb338-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb338-3"><a href="graphics-in-r-ggplot2.html#cb338-3" aria-hidden="true" tabindex="-1"></a><span class="do">## load get &#39;airway_scaledcounts.csv&#39; dataset </span></span>
<span id="cb338-4"><a href="graphics-in-r-ggplot2.html#cb338-4" aria-hidden="true" tabindex="-1"></a>airway_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/my_working_directory/airway_scaledcounts.csv&quot;</span>,</span>
<span id="cb338-5"><a href="graphics-in-r-ggplot2.html#cb338-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb338-6"><a href="graphics-in-r-ggplot2.html#cb338-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data)</span>
<span id="cb338-7"><a href="graphics-in-r-ggplot2.html#cb338-7" aria-hidden="true" tabindex="-1"></a><span class="co">#          ensgene SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb338-8"><a href="graphics-in-r-ggplot2.html#cb338-8" aria-hidden="true" tabindex="-1"></a><span class="co">#1 ENSG00000000003        723        486        904        445       1170       1097        806        604</span></span>
<span id="cb338-9"><a href="graphics-in-r-ggplot2.html#cb338-9" aria-hidden="true" tabindex="-1"></a><span class="co">#2 ENSG00000000005          0          0          0          0          0          0          0          0</span></span>
<span id="cb338-10"><a href="graphics-in-r-ggplot2.html#cb338-10" aria-hidden="true" tabindex="-1"></a><span class="co">#3 ENSG00000000419        467        523        616        371        582        781        417        509</span></span>
<span id="cb338-11"><a href="graphics-in-r-ggplot2.html#cb338-11" aria-hidden="true" tabindex="-1"></a><span class="co">#4 ENSG00000000457        347        258        364        237        318        447        330        324</span></span>
<span id="cb338-12"><a href="graphics-in-r-ggplot2.html#cb338-12" aria-hidden="true" tabindex="-1"></a><span class="co">#5 ENSG00000000460         96         81         73         66        118         94        102         74</span></span>
<span id="cb338-13"><a href="graphics-in-r-ggplot2.html#cb338-13" aria-hidden="true" tabindex="-1"></a><span class="co">#6 ENSG00000000938          0          0          1          0          2          0          0          0</span></span>
<span id="cb338-14"><a href="graphics-in-r-ggplot2.html#cb338-14" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(airway_data)</span>
<span id="cb338-15"><a href="graphics-in-r-ggplot2.html#cb338-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  38694 obs. of  9 variables:</span></span>
<span id="cb338-16"><a href="graphics-in-r-ggplot2.html#cb338-16" aria-hidden="true" tabindex="-1"></a><span class="co"># $ ensgene   : chr  &quot;ENSG00000000003&quot; &quot;ENSG00000000005&quot; &quot;ENSG00000000419&quot; &quot;ENSG00000000457&quot; ...</span></span>
<span id="cb338-17"><a href="graphics-in-r-ggplot2.html#cb338-17" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SRR1039508: num  723 0 467 347 96 ...</span></span>
<span id="cb338-18"><a href="graphics-in-r-ggplot2.html#cb338-18" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SRR1039509: num  486 0 523 258 81 ...</span></span>
<span id="cb338-19"><a href="graphics-in-r-ggplot2.html#cb338-19" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SRR1039512: num  904 0 616 364 73 1 6000 2640 692 531 ...</span></span>
<span id="cb338-20"><a href="graphics-in-r-ggplot2.html#cb338-20" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SRR1039513: num  445 0 371 237 66 ...</span></span>
<span id="cb338-21"><a href="graphics-in-r-ggplot2.html#cb338-21" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SRR1039516: num  1170 0 582 318 118 ...</span></span>
<span id="cb338-22"><a href="graphics-in-r-ggplot2.html#cb338-22" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SRR1039517: num  1097 0 781 447 94 ...</span></span>
<span id="cb338-23"><a href="graphics-in-r-ggplot2.html#cb338-23" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SRR1039520: num  806 0 417 330 102 ...</span></span>
<span id="cb338-24"><a href="graphics-in-r-ggplot2.html#cb338-24" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SRR1039521: num  604 0 509 324 74 ...</span></span></code></pre></div>
<p>We then scale this dataset using the function <code>modify()</code> from the package <code>purrr</code>. We needed to remove and add again the names of the genes in column <code>ensgene</code>.</p>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb339-1"><a href="graphics-in-r-ggplot2.html#cb339-1" aria-hidden="true" tabindex="-1"></a>ensgene <span class="ot">&lt;-</span> airway_data[,<span class="dv">1</span>]</span>
<span id="cb339-2"><a href="graphics-in-r-ggplot2.html#cb339-2" aria-hidden="true" tabindex="-1"></a>airway_data_numeric <span class="ot">&lt;-</span> airway_data[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb339-3"><a href="graphics-in-r-ggplot2.html#cb339-3" aria-hidden="true" tabindex="-1"></a><span class="co"># load purrr package</span></span>
<span id="cb339-4"><a href="graphics-in-r-ggplot2.html#cb339-4" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(purrr)</span>
<span id="cb339-5"><a href="graphics-in-r-ggplot2.html#cb339-5" aria-hidden="true" tabindex="-1"></a>airway_data_scaled <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">modify</span>(airway_data_numeric, scale)</span>
<span id="cb339-6"><a href="graphics-in-r-ggplot2.html#cb339-6" aria-hidden="true" tabindex="-1"></a><span class="co"># cbind the name of the genes from vector &#39;ensgene&#39;</span></span>
<span id="cb339-7"><a href="graphics-in-r-ggplot2.html#cb339-7" aria-hidden="true" tabindex="-1"></a>airway_data_scaled <span class="ot">&lt;-</span> <span class="fu">cbind</span>(ensgene, airway_data_scaled, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb339-8"><a href="graphics-in-r-ggplot2.html#cb339-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data_scaled)</span>
<span id="cb339-9"><a href="graphics-in-r-ggplot2.html#cb339-9" aria-hidden="true" tabindex="-1"></a><span class="co">#          ensgene  SRR1039508   SRR1039509   SRR1039512   SRR1039513  SRR1039516   SRR1039517  SRR1039520   SRR1039521</span></span>
<span id="cb339-10"><a href="graphics-in-r-ggplot2.html#cb339-10" aria-hidden="true" tabindex="-1"></a><span class="co">#1 ENSG00000000003  0.03933056 -0.003924270  0.036555187  0.011021618  0.09101424  0.041361372  0.06313763  0.007896859</span></span>
<span id="cb339-11"><a href="graphics-in-r-ggplot2.html#cb339-11" aria-hidden="true" tabindex="-1"></a><span class="co">#2 ENSG00000000005 -0.12176875 -0.130284028 -0.106887175 -0.112547703 -0.11339060 -0.123910392 -0.10822243 -0.116658738</span></span>
<span id="cb339-12"><a href="graphics-in-r-ggplot2.html#cb339-12" aria-hidden="true" tabindex="-1"></a><span class="co">#3 ENSG00000000419 -0.01771152  0.005695712 -0.009143265 -0.009526988 -0.01171230 -0.006246538 -0.01956592 -0.011693839</span></span>
<span id="cb339-13"><a href="graphics-in-r-ggplot2.html#cb339-13" aria-hidden="true" tabindex="-1"></a><span class="co">#4 ENSG00000000457 -0.04444999 -0.063204156 -0.049129410 -0.046736626 -0.05783441 -0.056566291 -0.03806261 -0.049844146</span></span>
<span id="cb339-14"><a href="graphics-in-r-ggplot2.html#cb339-14" aria-hidden="true" tabindex="-1"></a><span class="co">#5 ENSG00000000460 -0.10037797 -0.109224069 -0.095303887 -0.094220568 -0.09277541 -0.109748545 -0.08653667 -0.101398615</span></span>
<span id="cb339-15"><a href="graphics-in-r-ggplot2.html#cb339-15" aria-hidden="true" tabindex="-1"></a><span class="co">#6 ENSG00000000938 -0.12176875 -0.130284028 -0.106728500 -0.112547703 -0.11304119 -0.123910392 -0.10822243 -0.116658738</span></span></code></pre></div>
<p>We can also create a more easy to map plot with using the function <code>melt()</code> of the package <a href="https://cran.r-project.org/web/packages/reshape2/index.html" target="_blank"><strong>reshape2</strong></a>.</p>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="graphics-in-r-ggplot2.html#cb340-1" aria-hidden="true" tabindex="-1"></a><span class="do">## load reshape</span></span>
<span id="cb340-2"><a href="graphics-in-r-ggplot2.html#cb340-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb340-3"><a href="graphics-in-r-ggplot2.html#cb340-3" aria-hidden="true" tabindex="-1"></a><span class="co"># prepare other data frame</span></span>
<span id="cb340-4"><a href="graphics-in-r-ggplot2.html#cb340-4" aria-hidden="true" tabindex="-1"></a>airway_data_scaled2 <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">modify</span>(airway_data_numeric, scale)</span>
<span id="cb340-5"><a href="graphics-in-r-ggplot2.html#cb340-5" aria-hidden="true" tabindex="-1"></a><span class="co"># add ensgene as rownames</span></span>
<span id="cb340-6"><a href="graphics-in-r-ggplot2.html#cb340-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(airway_data_scaled2) <span class="ot">&lt;-</span> ensgene</span>
<span id="cb340-7"><a href="graphics-in-r-ggplot2.html#cb340-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data_scaled2)</span>
<span id="cb340-8"><a href="graphics-in-r-ggplot2.html#cb340-8" aria-hidden="true" tabindex="-1"></a><span class="co">#                 SRR1039508   SRR1039509   SRR1039512   SRR1039513  SRR1039516   SRR1039517  SRR1039520   SRR1039521</span></span>
<span id="cb340-9"><a href="graphics-in-r-ggplot2.html#cb340-9" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000003  0.03933056 -0.003924270  0.036555187  0.011021618  0.09101424  0.041361372  0.06313763  0.007896859</span></span>
<span id="cb340-10"><a href="graphics-in-r-ggplot2.html#cb340-10" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000005 -0.12176875 -0.130284028 -0.106887175 -0.112547703 -0.11339060 -0.123910392 -0.10822243 -0.116658738</span></span>
<span id="cb340-11"><a href="graphics-in-r-ggplot2.html#cb340-11" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000419 -0.01771152  0.005695712 -0.009143265 -0.009526988 -0.01171230 -0.006246538 -0.01956592 -0.011693839</span></span>
<span id="cb340-12"><a href="graphics-in-r-ggplot2.html#cb340-12" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000457 -0.04444999 -0.063204156 -0.049129410 -0.046736626 -0.05783441 -0.056566291 -0.03806261 -0.049844146</span></span>
<span id="cb340-13"><a href="graphics-in-r-ggplot2.html#cb340-13" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000460 -0.10037797 -0.109224069 -0.095303887 -0.094220568 -0.09277541 -0.109748545 -0.08653667 -0.101398615</span></span>
<span id="cb340-14"><a href="graphics-in-r-ggplot2.html#cb340-14" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000938 -0.12176875 -0.130284028 -0.106728500 -0.112547703 -0.11304119 -0.123910392 -0.10822243 -0.116658738</span></span>
<span id="cb340-15"><a href="graphics-in-r-ggplot2.html#cb340-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb340-16"><a href="graphics-in-r-ggplot2.html#cb340-16" aria-hidden="true" tabindex="-1"></a><span class="do">## get matrix dimensions</span></span>
<span id="cb340-17"><a href="graphics-in-r-ggplot2.html#cb340-17" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(airway_data_scaled2)</span>
<span id="cb340-18"><a href="graphics-in-r-ggplot2.html#cb340-18" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 38694     8</span></span></code></pre></div>
<p>As indicated by the output of the function <code>dim()</code> this is a very large matrix (38,694 rows). This is a huge table, however most genes will have similar levels of expression across SRR samples. We will search for the ensgenes that have the most dispersion around the mean expression by estimating the coefficient of variation <a href="https://en.wikipedia.org/wiki/Coefficient_of_variation" target="_blank"><strong>CV</strong></a>. This statistic represents the ratio of the standard deviation to the mean. Then, we will extract those ensgenes with the 100 highest values.</p>
<div class="sourceCode" id="cb341"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb341-1"><a href="graphics-in-r-ggplot2.html#cb341-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Create a function to valculate coefficient of variation </span></span>
<span id="cb341-2"><a href="graphics-in-r-ggplot2.html#cb341-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb341-3"><a href="graphics-in-r-ggplot2.html#cb341-3" aria-hidden="true" tabindex="-1"></a>co.var <span class="ot">&lt;-</span> <span class="cf">function</span>(x) ( <span class="dv">100</span><span class="sc">*</span><span class="fu">sd</span>(x)<span class="sc">/</span><span class="fu">mean</span>(x) )</span>
<span id="cb341-4"><a href="graphics-in-r-ggplot2.html#cb341-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb341-5"><a href="graphics-in-r-ggplot2.html#cb341-5" aria-hidden="true" tabindex="-1"></a><span class="do">##  we estimate row-wise coefficient of variation </span></span>
<span id="cb341-6"><a href="graphics-in-r-ggplot2.html#cb341-6" aria-hidden="true" tabindex="-1"></a>airway_data_scaled2<span class="sc">$</span>coVar<span class="ot">&lt;-</span><span class="fu">apply</span>(airway_data_scaled2[,<span class="fu">names</span>(airway_data_scaled2)],<span class="dv">1</span>,co.var)</span>
<span id="cb341-7"><a href="graphics-in-r-ggplot2.html#cb341-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data_scaled2)</span>
<span id="cb341-8"><a href="graphics-in-r-ggplot2.html#cb341-8" aria-hidden="true" tabindex="-1"></a><span class="co">#                 SRR1039508   SRR1039509   SRR1039512   SRR1039513  SRR1039516   SRR1039517  SRR1039520   SRR1039521      coVar</span></span>
<span id="cb341-9"><a href="graphics-in-r-ggplot2.html#cb341-9" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000003  0.03933056 -0.003924270  0.036555187  0.011021618  0.09101424  0.041361372  0.06313763  0.007896859  87.129966</span></span>
<span id="cb341-10"><a href="graphics-in-r-ggplot2.html#cb341-10" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000005 -0.12176875 -0.130284028 -0.106887175 -0.112547703 -0.11339060 -0.123910392 -0.10822243 -0.116658738  -6.935013</span></span>
<span id="cb341-11"><a href="graphics-in-r-ggplot2.html#cb341-11" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000419 -0.01771152  0.005695712 -0.009143265 -0.009526988 -0.01171230 -0.006246538 -0.01956592 -0.011693839 -77.383502</span></span>
<span id="cb341-12"><a href="graphics-in-r-ggplot2.html#cb341-12" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000457 -0.04444999 -0.063204156 -0.049129410 -0.046736626 -0.05783441 -0.056566291 -0.03806261 -0.049844146 -15.968492</span></span>
<span id="cb341-13"><a href="graphics-in-r-ggplot2.html#cb341-13" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000460 -0.10037797 -0.109224069 -0.095303887 -0.094220568 -0.09277541 -0.109748545 -0.08653667 -0.101398615  -8.191639</span></span>
<span id="cb341-14"><a href="graphics-in-r-ggplot2.html#cb341-14" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000938 -0.12176875 -0.130284028 -0.106728500 -0.112547703 -0.11304119 -0.123910392 -0.10822243 -0.116658738  -6.980658</span></span>
<span id="cb341-15"><a href="graphics-in-r-ggplot2.html#cb341-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb341-16"><a href="graphics-in-r-ggplot2.html#cb341-16" aria-hidden="true" tabindex="-1"></a><span class="do">## descresing order by coVar</span></span>
<span id="cb341-17"><a href="graphics-in-r-ggplot2.html#cb341-17" aria-hidden="true" tabindex="-1"></a>airway_data_scaled2_order <span class="ot">&lt;-</span> airway_data_scaled2[<span class="fu">order</span>(airway_data_scaled2<span class="sc">$</span>coVar,  <span class="at">decreasing =</span> <span class="cn">TRUE</span>),]</span>
<span id="cb341-18"><a href="graphics-in-r-ggplot2.html#cb341-18" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data_scaled2_order)</span>
<span id="cb341-19"><a href="graphics-in-r-ggplot2.html#cb341-19" aria-hidden="true" tabindex="-1"></a><span class="co">#                   SRR1039508    SRR1039509   SRR1039512   SRR1039513   SRR1039516   SRR1039517   SRR1039520   SRR1039521    coVar</span></span>
<span id="cb341-20"><a href="graphics-in-r-ggplot2.html#cb341-20" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000126709  0.0313090216 -0.0005442761 -0.006445787  0.049897360 -0.012760526 -0.030050493 -0.019778529 -0.011075185 39115.82</span></span>
<span id="cb341-21"><a href="graphics-in-r-ggplot2.html#cb341-21" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000278270  0.1077364958  0.1148955034 -0.015648947 -0.035906955 -0.042285668 -0.052950501 -0.036574367 -0.037471074 30921.30</span></span>
<span id="cb341-22"><a href="graphics-in-r-ggplot2.html#cb341-22" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000041802 -0.0043422784  0.0150556941 -0.007239162 -0.015080665  0.012746403  0.017406759 -0.008723041 -0.009425442 25810.60</span></span>
<span id="cb341-23"><a href="graphics-in-r-ggplot2.html#cb341-23" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000086712 -0.0116953590  0.0098557040 -0.021678604 -0.008138569  0.004535268 -0.034570232  0.038475388  0.023981854 25111.84</span></span>
<span id="cb341-24"><a href="graphics-in-r-ggplot2.html#cb341-24" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000188603 -0.0001086865  0.0116757006 -0.020567877  0.002413418 -0.009441131 -0.004739959 -0.004258325  0.025631597 18384.05</span></span>
<span id="cb341-25"><a href="graphics-in-r-ggplot2.html#cb341-25" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000125398  0.0698569896 -0.0096442587  0.058928387 -0.045348207 -0.017652265 -0.093326829  0.090138534 -0.049637928 15830.54</span></span>
<span id="cb341-26"><a href="graphics-in-r-ggplot2.html#cb341-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb341-27"><a href="graphics-in-r-ggplot2.html#cb341-27" aria-hidden="true" tabindex="-1"></a><span class="do">## retain the first 100 genes</span></span>
<span id="cb341-28"><a href="graphics-in-r-ggplot2.html#cb341-28" aria-hidden="true" tabindex="-1"></a>airway_data_scaled_red <span class="ot">&lt;-</span> airway_data_scaled2_order[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>,]</span>
<span id="cb341-29"><a href="graphics-in-r-ggplot2.html#cb341-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb341-30"><a href="graphics-in-r-ggplot2.html#cb341-30" aria-hidden="true" tabindex="-1"></a><span class="do">## remove sd column</span></span>
<span id="cb341-31"><a href="graphics-in-r-ggplot2.html#cb341-31" aria-hidden="true" tabindex="-1"></a>airway_data_scaled_red <span class="ot">&lt;-</span> airway_data_scaled_red[,<span class="sc">-</span><span class="fu">length</span>(airway_data_scaled_red)]</span>
<span id="cb341-32"><a href="graphics-in-r-ggplot2.html#cb341-32" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data_scaled_red)</span>
<span id="cb341-33"><a href="graphics-in-r-ggplot2.html#cb341-33" aria-hidden="true" tabindex="-1"></a><span class="co">#                   SRR1039508    SRR1039509   SRR1039512   SRR1039513   SRR1039516   SRR1039517   SRR1039520   SRR1039521</span></span>
<span id="cb341-34"><a href="graphics-in-r-ggplot2.html#cb341-34" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000126709  0.0313090216 -0.0005442761 -0.006445787  0.049897360 -0.012760526 -0.030050493 -0.019778529 -0.011075185</span></span>
<span id="cb341-35"><a href="graphics-in-r-ggplot2.html#cb341-35" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000278270  0.1077364958  0.1148955034 -0.015648947 -0.035906955 -0.042285668 -0.052950501 -0.036574367 -0.037471074</span></span>
<span id="cb341-36"><a href="graphics-in-r-ggplot2.html#cb341-36" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000041802 -0.0043422784  0.0150556941 -0.007239162 -0.015080665  0.012746403  0.017406759 -0.008723041 -0.009425442</span></span>
<span id="cb341-37"><a href="graphics-in-r-ggplot2.html#cb341-37" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000086712 -0.0116953590  0.0098557040 -0.021678604 -0.008138569  0.004535268 -0.034570232  0.038475388  0.023981854</span></span>
<span id="cb341-38"><a href="graphics-in-r-ggplot2.html#cb341-38" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000188603 -0.0001086865  0.0116757006 -0.020567877  0.002413418 -0.009441131 -0.004739959 -0.004258325  0.025631597</span></span>
<span id="cb341-39"><a href="graphics-in-r-ggplot2.html#cb341-39" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000125398  0.0698569896 -0.0096442587  0.058928387 -0.045348207 -0.017652265 -0.093326829  0.090138534 -0.049637928</span></span>
<span id="cb341-40"><a href="graphics-in-r-ggplot2.html#cb341-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb341-41"><a href="graphics-in-r-ggplot2.html#cb341-41" aria-hidden="true" tabindex="-1"></a><span class="do">## get dimensions</span></span>
<span id="cb341-42"><a href="graphics-in-r-ggplot2.html#cb341-42" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(airway_data_scaled_red)</span>
<span id="cb341-43"><a href="graphics-in-r-ggplot2.html#cb341-43" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 100     8</span></span></code></pre></div>
<p>With this reduced matrix, we can use melt() to creae a heatmap amenable data frame.</p>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="graphics-in-r-ggplot2.html#cb342-1" aria-hidden="true" tabindex="-1"></a>airway_data_for_heatmap <span class="ot">&lt;-</span> <span class="fu">melt</span>(<span class="fu">as.matrix</span>(airway_data_scaled_red))</span>
<span id="cb342-2"><a href="graphics-in-r-ggplot2.html#cb342-2" aria-hidden="true" tabindex="-1"></a><span class="co"># transform factor variables to character</span></span>
<span id="cb342-3"><a href="graphics-in-r-ggplot2.html#cb342-3" aria-hidden="true" tabindex="-1"></a>factor_columns <span class="ot">&lt;-</span> <span class="fu">sapply</span>(airway_data_for_heatmap, is.factor)</span>
<span id="cb342-4"><a href="graphics-in-r-ggplot2.html#cb342-4" aria-hidden="true" tabindex="-1"></a>airway_data_for_heatmap[factor_columns] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(airway_data_for_heatmap[factor_columns], as.character)</span>
<span id="cb342-5"><a href="graphics-in-r-ggplot2.html#cb342-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(airway_data_for_heatmap)</span>
<span id="cb342-6"><a href="graphics-in-r-ggplot2.html#cb342-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  800 obs. of  3 variables:</span></span>
<span id="cb342-7"><a href="graphics-in-r-ggplot2.html#cb342-7" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Var1 : chr  &quot;ENSG00000126709&quot; &quot;ENSG00000278270&quot; &quot;ENSG00000041802&quot; &quot;ENSG00000086712&quot; ...</span></span>
<span id="cb342-8"><a href="graphics-in-r-ggplot2.html#cb342-8" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Var2 : chr  &quot;SRR1039508&quot; &quot;SRR1039508&quot; &quot;SRR1039508&quot; &quot;SRR1039508&quot; ...</span></span>
<span id="cb342-9"><a href="graphics-in-r-ggplot2.html#cb342-9" aria-hidden="true" tabindex="-1"></a><span class="co"># $ value: num  0.031309 0.107736 -0.004342 -0.011695 -0.000109 ...</span></span>
<span id="cb342-10"><a href="graphics-in-r-ggplot2.html#cb342-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb342-11"><a href="graphics-in-r-ggplot2.html#cb342-11" aria-hidden="true" tabindex="-1"></a><span class="do">## we can rename the columns in this data frame</span></span>
<span id="cb342-12"><a href="graphics-in-r-ggplot2.html#cb342-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(airway_data_for_heatmap) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ensgene&quot;</span>, <span class="st">&quot;SRR&quot;</span>, <span class="st">&quot;expression&quot;</span>)</span>
<span id="cb342-13"><a href="graphics-in-r-ggplot2.html#cb342-13" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data_for_heatmap)</span>
<span id="cb342-14"><a href="graphics-in-r-ggplot2.html#cb342-14" aria-hidden="true" tabindex="-1"></a><span class="co">#          ensgene        SRR    expression</span></span>
<span id="cb342-15"><a href="graphics-in-r-ggplot2.html#cb342-15" aria-hidden="true" tabindex="-1"></a><span class="co">#1 ENSG00000126709 SRR1039508  0.0313090216</span></span>
<span id="cb342-16"><a href="graphics-in-r-ggplot2.html#cb342-16" aria-hidden="true" tabindex="-1"></a><span class="co">#2 ENSG00000278270 SRR1039508  0.1077364958</span></span>
<span id="cb342-17"><a href="graphics-in-r-ggplot2.html#cb342-17" aria-hidden="true" tabindex="-1"></a><span class="co">#3 ENSG00000041802 SRR1039508 -0.0043422784</span></span>
<span id="cb342-18"><a href="graphics-in-r-ggplot2.html#cb342-18" aria-hidden="true" tabindex="-1"></a><span class="co">#4 ENSG00000086712 SRR1039508 -0.0116953590</span></span>
<span id="cb342-19"><a href="graphics-in-r-ggplot2.html#cb342-19" aria-hidden="true" tabindex="-1"></a><span class="co">#5 ENSG00000188603 SRR1039508 -0.0001086865</span></span>
<span id="cb342-20"><a href="graphics-in-r-ggplot2.html#cb342-20" aria-hidden="true" tabindex="-1"></a><span class="co">#6 ENSG00000125398 SRR1039508  0.0698569896</span></span></code></pre></div>
</div>
<div id="color-r-packages" class="section level2" number="9.2">
<h2><span class="header-section-number">9.2</span> Color R-packages</h2>
<p><strong>4)</strong> One of the “hardest” parts before or after doing graphics is choosing colors. Many colors can be called by name (e.g., <code>firebrick4</code>) and I have added a pdf for R-colors in this class <a href="https://github.com/corytophanes/BIO_BIT_Bioinformatics_209/tree/main/ref_files_pdfs" target="_blank"><strong>GitHub repository</strong></a>. Any color also can be included by their hex color codes (e.g., <code>#005f3f</code>). These codes and the colors can be found in many online color palette websites like following:</p>
<ol style="list-style-type: lower-alpha">
<li><a href="https://www.color-hex.com/" target="_blank"><strong>https://www.color-hex.com/</strong></a></li>
<li><a href="https://www.colourlovers.com/" target="_blank"><strong>https://www.colourlovers.com/</strong></a></li>
<li><a href="https://colorbrewer2.org/" target="_blank"><strong>https://colorbrewer2.org/</strong></a></li>
</ol>
<p>Thankfully, several R-packages can also help you in choosing colors. I will list some of those that I found useful.</p>
<p><strong>5)</strong> The package <a href="https://cran.r-project.org/web/packages/viridis/index.html" target="_blank"><strong>viridis</strong></a> can create color scales including palettes for those persons with colorblindness. This package has a <a href="https://cran.r-project.org/web/packages/viridis/vignettes/intro-to-viridis.html" target="_blank">vignette</a> that help you with more examples.</p>
<div class="sourceCode" id="cb343"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb343-1"><a href="graphics-in-r-ggplot2.html#cb343-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We need to install R-package &#39;viridis&#39; </span></span>
<span id="cb343-2"><a href="graphics-in-r-ggplot2.html#cb343-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;viridis&quot;</span>)</span>
<span id="cb343-3"><a href="graphics-in-r-ggplot2.html#cb343-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span></code></pre></div>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="graphics-in-r-ggplot2.html#cb344-1" aria-hidden="true" tabindex="-1"></a><span class="do">## checking if ggplot2 is loaded</span></span>
<span id="cb344-2"><a href="graphics-in-r-ggplot2.html#cb344-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggplot2)</span>
<span id="cb344-3"><a href="graphics-in-r-ggplot2.html#cb344-3" aria-hidden="true" tabindex="-1"></a><span class="do">## we can plot a heatmap with the expression values</span></span>
<span id="cb344-4"><a href="graphics-in-r-ggplot2.html#cb344-4" aria-hidden="true" tabindex="-1"></a>heatmap_viridis_bac_magma <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(bacteria_espaa_for_heatmap,<span class="fu">aes</span>(<span class="at">x=</span>Amino_acid,<span class="at">y=</span>bacteria_sp,<span class="at">fill=</span>scaled_value))<span class="sc">+</span></span>
<span id="cb344-5"><a href="graphics-in-r-ggplot2.html#cb344-5" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">geom_tile</span>()<span class="sc">+</span></span>
<span id="cb344-6"><a href="graphics-in-r-ggplot2.html#cb344-6" aria-hidden="true" tabindex="-1"></a>          <span class="fu">scale_fill_viridis</span>(<span class="at">option=</span><span class="st">&quot;A&quot;</span>)<span class="sc">+</span> </span>
<span id="cb344-7"><a href="graphics-in-r-ggplot2.html#cb344-7" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb344-8"><a href="graphics-in-r-ggplot2.html#cb344-8" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">0.5</span>)),</span>
<span id="cb344-9"><a href="graphics-in-r-ggplot2.html#cb344-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">0.7</span>),<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust=</span><span class="dv">1</span>))</span>
<span id="cb344-10"><a href="graphics-in-r-ggplot2.html#cb344-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb344-11"><a href="graphics-in-r-ggplot2.html#cb344-11" aria-hidden="true" tabindex="-1"></a>heatmap_viridis_bac_magma</span>
<span id="cb344-12"><a href="graphics-in-r-ggplot2.html#cb344-12" aria-hidden="true" tabindex="-1"></a><span class="do">## </span><span class="al">NOTE</span><span class="do">: you might need to run the plot &#39;heatmap_viridis_bac_magma&#39;, if it gives you an error</span></span></code></pre></div>
<p><img src="figures/session_9_figs/1_bacteria_heatmap.png" width="100%" /></p>
<p>We can also run this color example with <code>airway_data_for_heatmap</code>.</p>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="graphics-in-r-ggplot2.html#cb345-1" aria-hidden="true" tabindex="-1"></a><span class="do">## checking if ggplot2 is loaded</span></span>
<span id="cb345-2"><a href="graphics-in-r-ggplot2.html#cb345-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggplot2)</span>
<span id="cb345-3"><a href="graphics-in-r-ggplot2.html#cb345-3" aria-hidden="true" tabindex="-1"></a><span class="do">## we can plot a heatmap with the expression values</span></span>
<span id="cb345-4"><a href="graphics-in-r-ggplot2.html#cb345-4" aria-hidden="true" tabindex="-1"></a>heatmap_viridis <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(airway_data_for_heatmap,<span class="fu">aes</span>(<span class="at">x=</span>SRR,<span class="at">y=</span>ensgene,<span class="at">fill=</span>expression))<span class="sc">+</span></span>
<span id="cb345-5"><a href="graphics-in-r-ggplot2.html#cb345-5" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">geom_tile</span>()<span class="sc">+</span></span>
<span id="cb345-6"><a href="graphics-in-r-ggplot2.html#cb345-6" aria-hidden="true" tabindex="-1"></a>          <span class="fu">scale_fill_viridis</span>()<span class="sc">+</span> </span>
<span id="cb345-7"><a href="graphics-in-r-ggplot2.html#cb345-7" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb345-8"><a href="graphics-in-r-ggplot2.html#cb345-8" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">0.5</span>)),</span>
<span id="cb345-9"><a href="graphics-in-r-ggplot2.html#cb345-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">0.7</span>),<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust=</span><span class="dv">1</span>))</span>
<span id="cb345-10"><a href="graphics-in-r-ggplot2.html#cb345-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb345-11"><a href="graphics-in-r-ggplot2.html#cb345-11" aria-hidden="true" tabindex="-1"></a><span class="do">## </span><span class="al">NOTE</span><span class="do">: this will require time to load (if it is huge) -- you might not want to run this plot</span></span>
<span id="cb345-12"><a href="graphics-in-r-ggplot2.html#cb345-12" aria-hidden="true" tabindex="-1"></a>heatmap_viridis</span></code></pre></div>
<p><img src="figures/session_9_figs/2_ensgene_heatmap.png" width="100%" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="exploring-multidimensional-data-in-r.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/09-Graphics_up_v6.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
