<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Session 9 – Graphics and Colors | Bioinformatics Workshop Gitbook</title>
  <meta name="description" content="none" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Session 9 – Graphics and Colors | Bioinformatics Workshop Gitbook" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="none" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Session 9 – Graphics and Colors | Bioinformatics Workshop Gitbook" />
  
  <meta name="twitter:description" content="none" />
  

<meta name="author" content="Juan C. Santos &amp; Randy Ortiz" />


<meta name="date" content="2024-03-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="multidimensional-data-in-r.html"/>
<link rel="next" href="basic-graphics.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics Workshop Gitbook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html"><i class="fa fa-check"></i><b>1</b> – Getting Started with R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#why-do-we-use-r"><i class="fa fa-check"></i><b>1.1</b> Why do we use R?</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#get-a-good-text-editor"><i class="fa fa-check"></i><b>1.2</b> Get a good text editor</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-r"><i class="fa fa-check"></i><b>1.4</b> Installing R</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#annotating-code"><i class="fa fa-check"></i><b>1.5</b> Annotating code</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#defining-a-working-directory-in-r"><i class="fa fa-check"></i><b>1.6</b> Defining a working directory in R</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html"><i class="fa fa-check"></i><b>2</b> – Using R, Installing Packages and Importing/Exporting Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#your-first-session"><i class="fa fa-check"></i><b>2.1</b> Your first session</a></li>
<li class="chapter" data-level="2.2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#doing-some-base-functions-and-getting-help"><i class="fa fa-check"></i><b>2.2</b> Doing some base functions and getting help</a></li>
<li class="chapter" data-level="2.3" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#functions-between-two-or-more-vectors"><i class="fa fa-check"></i><b>2.3</b> Functions between two or more vectors</a></li>
<li class="chapter" data-level="2.4" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#about-r-packages"><i class="fa fa-check"></i><b>2.4</b> About R-packages</a></li>
<li class="chapter" data-level="2.5" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#cran-task-views-expert-opinion-on-useful-packages"><i class="fa fa-check"></i><b>2.5</b> CRAN Task Views – Expert opinion on useful packages</a></li>
<li class="chapter" data-level="2.6" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#installing-packages"><i class="fa fa-check"></i><b>2.6</b> Installing packages</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html"><i class="fa fa-check"></i><b>3</b> – Basic Data Structures in R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#scalars-and-vectors"><i class="fa fa-check"></i><b>3.1</b> Scalars and vectors</a></li>
<li class="chapter" data-level="3.2" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#arrays"><i class="fa fa-check"></i><b>3.3</b> Arrays</a></li>
<li class="chapter" data-level="3.4" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a></li>
<li class="chapter" data-level="3.5" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tables"><i class="fa fa-check"></i><b>3.5</b> Tables</a></li>
<li class="chapter" data-level="3.6" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#lists"><i class="fa fa-check"></i><b>3.6</b> Lists</a></li>
<li class="chapter" data-level="3.7" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-tables"><i class="fa fa-check"></i><b>3.7</b> Data Tables</a></li>
<li class="chapter" data-level="3.8" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tibbles"><i class="fa fa-check"></i><b>3.8</b> Tibbles</a></li>
<li class="chapter" data-level="3.9" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#dnastringsets-rnastringsets-and-aastringsets"><i class="fa fa-check"></i><b>3.9</b> DNAStringSets, RNAStringSets and AAStringSets</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html"><i class="fa fa-check"></i><b>4</b> – Text Editing and Data Transformations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#working-with-character-strings"><i class="fa fa-check"></i><b>4.1</b> Working with character strings</a></li>
<li class="chapter" data-level="4.2" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#regular-expressions-regex"><i class="fa fa-check"></i><b>4.2</b> Regular expressions (Regex)</a></li>
<li class="chapter" data-level="4.3" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#comparing-ordering-and-simplifying-vectors"><i class="fa fa-check"></i><b>4.3</b> Comparing, ordering and simplifying vectors</a></li>
<li class="chapter" data-level="4.4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#useful-functions-of-stringr"><i class="fa fa-check"></i><b>4.4</b> Useful functions of <code>stringr</code></a></li>
<li class="chapter" data-level="4.5" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#data-transformations"><i class="fa fa-check"></i><b>4.5</b> Data transformations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html"><i class="fa fa-check"></i><b>5</b> – Getting Biological Data from Public Repositories</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#ncbi-databases-genbank-and-others"><i class="fa fa-check"></i><b>5.1</b> NCBI databases: GenBank and others</a></li>
<li class="chapter" data-level="5.2" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#accessing-ncbi-within-r-rentrez"><i class="fa fa-check"></i><b>5.2</b> Accessing NCBI within R: <code>rentrez</code></a></li>
<li class="chapter" data-level="5.3" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#retrieving-pubmed-references"><i class="fa fa-check"></i><b>5.3</b> Retrieving PubMed references</a></li>
<li class="chapter" data-level="5.4" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#comparing-sequences-blast"><i class="fa fa-check"></i><b>5.4</b> Comparing sequences: BLAST</a></li>
<li class="chapter" data-level="5.5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-uniprot"><i class="fa fa-check"></i><b>5.5</b> Other databases: UniProt</a></li>
<li class="chapter" data-level="5.6" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-ensembl"><i class="fa fa-check"></i><b>5.6</b> Other databases: Ensembl</a></li>
<li class="chapter" data-level="5.7" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-taxonomy-phylogenetic-trees-and-species-distributions"><i class="fa fa-check"></i><b>5.7</b> Other databases: Taxonomy, phylogenetic trees and species distributions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html"><i class="fa fa-check"></i><b>6</b> – Basic Statistics in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#exploring-your-data"><i class="fa fa-check"></i><b>6.1</b> Exploring your data</a></li>
<li class="chapter" data-level="6.2" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#sub-setting-your-dataset"><i class="fa fa-check"></i><b>6.2</b> Sub-setting your dataset</a></li>
<li class="chapter" data-level="6.3" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#transposing-rotating-your-dataset"><i class="fa fa-check"></i><b>6.3</b> Transposing (rotating) your dataset</a></li>
<li class="chapter" data-level="6.4" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#dealing-with-missing-na-values"><i class="fa fa-check"></i><b>6.4</b> Dealing with missing (NA) values</a></li>
<li class="chapter" data-level="6.5" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#about-p-values"><i class="fa fa-check"></i><b>6.5</b> About p-values</a></li>
<li class="chapter" data-level="6.6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#how-to-read-p-values"><i class="fa fa-check"></i><b>6.6</b> How to read p-values</a></li>
<li class="chapter" data-level="6.7" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#multiple-testing-corrections"><i class="fa fa-check"></i><b>6.7</b> Multiple testing corrections</a></li>
<li class="chapter" data-level="6.8" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#normalizing-your-data"><i class="fa fa-check"></i><b>6.8</b> Normalizing your data</a></li>
<li class="chapter" data-level="6.9" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#what-test-should-i-use"><i class="fa fa-check"></i><b>6.9</b> What test should I use?</a></li>
<li class="chapter" data-level="6.10" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#type-of-variables-in-dataset"><i class="fa fa-check"></i><b>6.10</b> Type of variables in dataset</a></li>
<li class="chapter" data-level="6.11" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-means-t-test"><i class="fa fa-check"></i><b>6.11</b> Comparing means: t-test</a></li>
<li class="chapter" data-level="6.12" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#two-sample-t-test"><i class="fa fa-check"></i><b>6.12</b> Two sample t-test</a></li>
<li class="chapter" data-level="6.13" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-wilcoxon-signed-rank-test"><i class="fa fa-check"></i><b>6.13</b> Beyond t-test: Wilcoxon signed rank test</a></li>
<li class="chapter" data-level="6.14" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#tabulation-and-cross-tabulation"><i class="fa fa-check"></i><b>6.14</b> Tabulation and Cross-Tabulation</a></li>
<li class="chapter" data-level="6.15" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-binomial-test"><i class="fa fa-check"></i><b>6.15</b> Beyond t-test: Binomial test</a></li>
<li class="chapter" data-level="6.16" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-chi-square-goodness-of-fit"><i class="fa fa-check"></i><b>6.16</b> Beyond t-test: Chi-square goodness of fit</a></li>
<li class="chapter" data-level="6.17" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-mcnemars-test"><i class="fa fa-check"></i><b>6.17</b> Beyond t-test: McNemar’s test</a></li>
<li class="chapter" data-level="6.18" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-fishers-exact-test"><i class="fa fa-check"></i><b>6.18</b> Beyond t-test: Fisher’s exact test</a></li>
<li class="chapter" data-level="6.19" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#correlations"><i class="fa fa-check"></i><b>6.19</b> Correlations</a></li>
<li class="chapter" data-level="6.20" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-distributions"><i class="fa fa-check"></i><b>6.20</b> Comparing distributions</a></li>
<li class="chapter" data-level="6.21" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#bootstraps"><i class="fa fa-check"></i><b>6.21</b> Bootstraps</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html"><i class="fa fa-check"></i><b>7</b> – Regressions and ANOVA in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#linear-regressions"><i class="fa fa-check"></i><b>7.1</b> Linear regressions</a></li>
<li class="chapter" data-level="7.2" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anatomy-of-a-linear-regression"><i class="fa fa-check"></i><b>7.2</b> Anatomy of a linear regression</a></li>
<li class="chapter" data-level="7.3" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#multiple-linear-regressions"><i class="fa fa-check"></i><b>7.3</b> Multiple linear regressions</a></li>
<li class="chapter" data-level="7.4" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#diagnostic-plots-for-regressions"><i class="fa fa-check"></i><b>7.4</b> Diagnostic plots for regressions</a></li>
<li class="chapter" data-level="7.5" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-influential-measurements"><i class="fa fa-check"></i><b>7.5</b> Regression influential measurements</a></li>
<li class="chapter" data-level="7.6" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-anova-coefficients-ci"><i class="fa fa-check"></i><b>7.6</b> Regression (ANOVA, coefficients, CI)</a></li>
<li class="chapter" data-level="7.7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#selecting-predictors"><i class="fa fa-check"></i><b>7.7</b> Selecting predictors</a></li>
<li class="chapter" data-level="7.8" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-predictions"><i class="fa fa-check"></i><b>7.8</b> Regression predictions</a></li>
<li class="chapter" data-level="7.9" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions"><i class="fa fa-check"></i><b>7.9</b> Logistic regressions</a></li>
<li class="chapter" data-level="7.10" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions-requirements"><i class="fa fa-check"></i><b>7.10</b> Logistic regressions requirements</a></li>
<li class="chapter" data-level="7.11" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#non-linear-regressions"><i class="fa fa-check"></i><b>7.11</b> Non-linear regressions</a></li>
<li class="chapter" data-level="7.12" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-regressions"><i class="fa fa-check"></i><b>7.12</b> Toy datasets for regressions</a></li>
<li class="chapter" data-level="7.13" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova"><i class="fa fa-check"></i><b>7.13</b> ANOVA</a></li>
<li class="chapter" data-level="7.14" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#one-way-anova"><i class="fa fa-check"></i><b>7.14</b> One-way ANOVA</a></li>
<li class="chapter" data-level="7.15" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#two-way-anova"><i class="fa fa-check"></i><b>7.15</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="7.16" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova-post-hoc-tests"><i class="fa fa-check"></i><b>7.16</b> ANOVA Post-hoc tests</a></li>
<li class="chapter" data-level="7.17" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#beyond-anova-kruskal-wallis-test"><i class="fa fa-check"></i><b>7.17</b> Beyond ANOVA: Kruskal-Wallis Test</a></li>
<li class="chapter" data-level="7.18" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-anova"><i class="fa fa-check"></i><b>7.18</b> Toy datasets for ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html"><i class="fa fa-check"></i><b>8</b> – Multidimensional Data in R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#principal-component-analysis"><i class="fa fa-check"></i><b>8.1</b> Principal Component Analysis</a></li>
<li class="chapter" data-level="8.2" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#multidimensional-scaling"><i class="fa fa-check"></i><b>8.2</b> Multidimensional scaling</a></li>
<li class="chapter" data-level="8.3" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-dfa"><i class="fa fa-check"></i><b>8.3</b> Cluster analysis: DFA</a></li>
<li class="chapter" data-level="8.4" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-k-means-unsupervised"><i class="fa fa-check"></i><b>8.4</b> Cluster analysis: <em>k</em>-means (unsupervised)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html"><i class="fa fa-check"></i><b>9</b> – Graphics and Colors</a>
<ul>
<li class="chapter" data-level="9.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#avoid-bad-graphics"><i class="fa fa-check"></i><b>9.1</b> Avoid bad graphics</a></li>
<li class="chapter" data-level="9.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#graphics-in-r-ggplot2"><i class="fa fa-check"></i><b>9.2</b> Graphics in R – ggplot2</a></li>
<li class="chapter" data-level="9.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#preparing-your-dataset"><i class="fa fa-check"></i><b>9.3</b> Preparing your dataset</a></li>
<li class="chapter" data-level="9.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#color-palettes-online"><i class="fa fa-check"></i><b>9.4</b> Color palettes online</a></li>
<li class="chapter" data-level="9.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggplot2-default-palette"><i class="fa fa-check"></i><b>9.5</b> <code>ggplot2</code> default palette</a></li>
<li class="chapter" data-level="9.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#heatmaps-or-tiled-charts"><i class="fa fa-check"></i><b>9.6</b> Heatmaps or tiled charts</a></li>
<li class="chapter" data-level="9.7" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#r-packages-about-color"><i class="fa fa-check"></i><b>9.7</b> R-packages about color</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#viridis-package"><i class="fa fa-check"></i><b>9.7.1</b> <code>viridis</code> package</a></li>
<li class="chapter" data-level="9.7.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcolorbrewer-package"><i class="fa fa-check"></i><b>9.7.2</b> <code>RColorBrewer</code> package</a></li>
<li class="chapter" data-level="9.7.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggsci-package"><i class="fa fa-check"></i><b>9.7.3</b> <code>ggsci</code> package</a></li>
<li class="chapter" data-level="9.7.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#wesanderson-package"><i class="fa fa-check"></i><b>9.7.4</b> <code>wesanderson</code> package</a></li>
<li class="chapter" data-level="9.7.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcartocolor-package"><i class="fa fa-check"></i><b>9.7.5</b> <code>rcartocolor</code> package</a></li>
<li class="chapter" data-level="9.7.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#colorspace-package"><i class="fa fa-check"></i><b>9.7.6</b> <code>colorspace</code> package</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#making-your-own-palette"><i class="fa fa-check"></i><b>9.8</b> Making your own palette</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="basic-graphics.html"><a href="basic-graphics.html"><i class="fa fa-check"></i><b>10</b> – Basic Graphics</a>
<ul>
<li class="chapter" data-level="10.1" data-path="basic-graphics.html"><a href="basic-graphics.html#what-would-you-like-to-show"><i class="fa fa-check"></i><b>10.1</b> What would you like to show?</a></li>
<li class="chapter" data-level="10.2" data-path="basic-graphics.html"><a href="basic-graphics.html#bar-charts"><i class="fa fa-check"></i><b>10.2</b> Bar charts</a></li>
<li class="chapter" data-level="10.3" data-path="basic-graphics.html"><a href="basic-graphics.html#stacked-bar-charts"><i class="fa fa-check"></i><b>10.3</b> Stacked bar charts</a></li>
<li class="chapter" data-level="10.4" data-path="basic-graphics.html"><a href="basic-graphics.html#histograms"><i class="fa fa-check"></i><b>10.4</b> Histograms</a></li>
<li class="chapter" data-level="10.5" data-path="basic-graphics.html"><a href="basic-graphics.html#multiple-histograms"><i class="fa fa-check"></i><b>10.5</b> Multiple histograms</a></li>
<li class="chapter" data-level="10.6" data-path="basic-graphics.html"><a href="basic-graphics.html#density-plots"><i class="fa fa-check"></i><b>10.6</b> Density plots</a></li>
<li class="chapter" data-level="10.7" data-path="basic-graphics.html"><a href="basic-graphics.html#box-plots"><i class="fa fa-check"></i><b>10.7</b> Box plots</a></li>
<li class="chapter" data-level="10.8" data-path="basic-graphics.html"><a href="basic-graphics.html#violin-plots"><i class="fa fa-check"></i><b>10.8</b> Violin plots</a></li>
<li class="chapter" data-level="10.9" data-path="basic-graphics.html"><a href="basic-graphics.html#scatterplots"><i class="fa fa-check"></i><b>10.9</b> Scatterplots</a></li>
<li class="chapter" data-level="10.10" data-path="basic-graphics.html"><a href="basic-graphics.html#lines-and-scatterplots"><i class="fa fa-check"></i><b>10.10</b> Lines and Scatterplots</a></li>
<li class="chapter" data-level="10.11" data-path="basic-graphics.html"><a href="basic-graphics.html#save-your-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>10.11</b> Save your plots as PDF or PNG</a></li>
<li class="chapter" data-level="10.12" data-path="basic-graphics.html"><a href="basic-graphics.html#ggplot2-add-on-packages-treemapify"><i class="fa fa-check"></i><b>10.12</b> ggplot2 add-on packages: <code>treemapify</code></a></li>
<li class="chapter" data-level="10.13" data-path="basic-graphics.html"><a href="basic-graphics.html#other-resources"><i class="fa fa-check"></i><b>10.13</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="sequence-alignments.html"><a href="sequence-alignments.html"><i class="fa fa-check"></i><b>11</b> – Sequence Alignments</a>
<ul>
<li class="chapter" data-level="11.1" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-using-rentrez"><i class="fa fa-check"></i><b>11.1</b> Retrieving sequences using ‘rentrez’</a></li>
<li class="chapter" data-level="11.2" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-manually-from-ncbi"><i class="fa fa-check"></i><b>11.2</b> Retrieving sequences manually from NCBI</a></li>
<li class="chapter" data-level="11.3" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-accession-numbers"><i class="fa fa-check"></i><b>11.3</b> Retrieving accession numbers</a></li>
<li class="chapter" data-level="11.4" data-path="sequence-alignments.html"><a href="sequence-alignments.html#alignment-with-decipher"><i class="fa fa-check"></i><b>11.4</b> Alignment with <code>DECIPHER</code></a></li>
<li class="chapter" data-level="11.5" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualize-alignment"><i class="fa fa-check"></i><b>11.5</b> Visualize alignment</a></li>
<li class="chapter" data-level="11.6" data-path="sequence-alignments.html"><a href="sequence-alignments.html#save-alignment-to-file"><i class="fa fa-check"></i><b>11.6</b> Save alignment to file</a></li>
<li class="chapter" data-level="11.7" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-and-align-several-markers"><i class="fa fa-check"></i><b>11.7</b> Concatenate and align several markers</a></li>
<li class="chapter" data-level="11.8" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-data.table"><i class="fa fa-check"></i><b>11.8</b> Concatenate with <code>data.table</code></a></li>
<li class="chapter" data-level="11.9" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-biostrings"><i class="fa fa-check"></i><b>11.9</b> Concatenate with <code>Biostrings</code></a></li>
<li class="chapter" data-level="11.10" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualizing-alignments-in-r"><i class="fa fa-check"></i><b>11.10</b> Visualizing alignments in R</a></li>
<li class="chapter" data-level="11.11" data-path="sequence-alignments.html"><a href="sequence-alignments.html#other-sequence-alignment-software"><i class="fa fa-check"></i><b>11.11</b> Other sequence alignment software</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html"><i class="fa fa-check"></i><b>12</b> – Estimating Phylogenetic Trees</a>
<ul>
<li class="chapter" data-level="12.1" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#phylogeny-inference-with-iqtree2"><i class="fa fa-check"></i><b>12.1</b> Phylogeny inference with <code>iqtree2</code></a></li>
<li class="chapter" data-level="12.2" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-iqtree2-in-macos"><i class="fa fa-check"></i><b>12.2</b> Installing <code>iqtree2</code> in macOS</a></li>
<li class="chapter" data-level="12.3" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-and-running-iqtree2-in-a-pc"><i class="fa fa-check"></i><b>12.3</b> Installing and running <code>iqtree2</code> in a PC</a></li>
<li class="chapter" data-level="12.4" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#loading-the-iqtree_runner_r-function"><i class="fa fa-check"></i><b>12.4</b> Loading the <code>iqtree_runner_R</code> function</a></li>
<li class="chapter" data-level="12.5" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#running-iqtree2-from-r"><i class="fa fa-check"></i><b>12.5</b> Running <code>iqtree2</code> from R</a></li>
<li class="chapter" data-level="12.6" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-no-partitions"><i class="fa fa-check"></i><b>12.6</b> One marker and no partitions</a></li>
<li class="chapter" data-level="12.7" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-with-partitions"><i class="fa fa-check"></i><b>12.7</b> One marker and with partitions</a></li>
<li class="chapter" data-level="12.8" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#bootstrap-support"><i class="fa fa-check"></i><b>12.8</b> Bootstrap support</a></li>
<li class="chapter" data-level="12.9" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#many-markers-and-with-partitions"><i class="fa fa-check"></i><b>12.9</b> Many markers and with partitions</a></li>
<li class="chapter" data-level="12.10" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#other-sequence-phylogenetic-tree-estimation-software"><i class="fa fa-check"></i><b>12.10</b> Other sequence phylogenetic tree estimation software</a></li>
<li class="chapter" data-level="12.11" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#drawing-phylogenetic-trees-with-ggtree"><i class="fa fa-check"></i><b>12.11</b> Drawing phylogenetic trees with <code>ggtree</code></a></li>
<li class="chapter" data-level="12.12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#save-ggtree-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>12.12</b> Save ggtree plots as PDF or PNG</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html"><i class="fa fa-check"></i><b>13</b> – Mining from a reference dataset</a>
<ul>
<li class="chapter" data-level="13.1" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#a-worked-example-using-the-mtcars-dataset"><i class="fa fa-check"></i><b>13.1</b> A worked example using the <code>mtcars</code> dataset</a></li>
<li class="chapter" data-level="13.2" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-reference-dataset"><i class="fa fa-check"></i><b>13.2</b> Build a reference dataset</a></li>
<li class="chapter" data-level="13.3" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-sampling-dataset"><i class="fa fa-check"></i><b>13.3</b> Build a sampling dataset</a></li>
<li class="chapter" data-level="13.4" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#extracting-from-reference-dataset"><i class="fa fa-check"></i><b>13.4</b> Extracting from reference dataset</a></li>
<li class="chapter" data-level="13.5" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#plots-of-the-extracted-data"><i class="fa fa-check"></i><b>13.5</b> Plots of the extracted data</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="introduction-to-python.html"><a href="introduction-to-python.html"><i class="fa fa-check"></i><b>14</b> – Introduction to Python</a>
<ul>
<li class="chapter" data-level="14.1" data-path="introduction-to-python.html"><a href="introduction-to-python.html#welcome-to-python-why-python"><i class="fa fa-check"></i><b>14.1</b> Welcome to Python! Why Python?</a></li>
<li class="chapter" data-level="14.2" data-path="introduction-to-python.html"><a href="introduction-to-python.html#which-python-should-i-install"><i class="fa fa-check"></i><b>14.2</b> Which Python should I install?</a></li>
<li class="chapter" data-level="14.3" data-path="introduction-to-python.html"><a href="introduction-to-python.html#mac-os-x-python-installations"><i class="fa fa-check"></i><b>14.3</b> Mac OS X Python Installations</a></li>
<li class="chapter" data-level="14.4" data-path="introduction-to-python.html"><a href="introduction-to-python.html#windows-python-installations"><i class="fa fa-check"></i><b>14.4</b> Windows Python Installations</a></li>
<li class="chapter" data-level="14.5" data-path="introduction-to-python.html"><a href="introduction-to-python.html#install-an-ide-or-integrated-development-environment"><i class="fa fa-check"></i><b>14.5</b> Install an IDE, or integrated development environment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html"><i class="fa fa-check"></i><b>15</b> – Python: Keywords and Identifiers</a>
<ul>
<li class="chapter" data-level="15.1" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#keywords-in-python"><i class="fa fa-check"></i><b>15.1</b> Keywords in Python</a></li>
<li class="chapter" data-level="15.2" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#identifiers-in-python"><i class="fa fa-check"></i><b>15.2</b> Identifiers in Python</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html"><i class="fa fa-check"></i><b>16</b> – Python: Statements and Expressions</a>
<ul>
<li class="chapter" data-level="16.1" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#what-are-statements"><i class="fa fa-check"></i><b>16.1</b> What are statements?</a></li>
<li class="chapter" data-level="16.2" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#expressions-in-python"><i class="fa fa-check"></i><b>16.2</b> Expressions in Python</a></li>
<li class="chapter" data-level="16.3" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#simple-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.3</b> Simple Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.4" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#augmented-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.4</b> Augmented Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.5" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#explicit-and-implicit-line-continuation"><i class="fa fa-check"></i><b>16.5</b> Explicit and Implicit Line Continuation</a></li>
<li class="chapter" data-level="16.6" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#indentation-in-python-is-crucial-to-proper-code-function"><i class="fa fa-check"></i><b>16.6</b> Indentation in Python is Crucial to Proper Code function</a></li>
<li class="chapter" data-level="16.7" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#comments"><i class="fa fa-check"></i><b>16.7</b> Comments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="python-data-types.html"><a href="python-data-types.html"><i class="fa fa-check"></i><b>17</b> – Python: Data Types</a>
<ul>
<li class="chapter" data-level="17.1" data-path="python-data-types.html"><a href="python-data-types.html#booleans"><i class="fa fa-check"></i><b>17.1</b> Booleans</a></li>
<li class="chapter" data-level="17.2" data-path="python-data-types.html"><a href="python-data-types.html#numbers-int-float-and-complex"><i class="fa fa-check"></i><b>17.2</b> Numbers: int, float and complex</a></li>
<li class="chapter" data-level="17.3" data-path="python-data-types.html"><a href="python-data-types.html#strings"><i class="fa fa-check"></i><b>17.3</b> Strings</a></li>
<li class="chapter" data-level="17.4" data-path="python-data-types.html"><a href="python-data-types.html#bytes"><i class="fa fa-check"></i><b>17.4</b> Bytes</a></li>
<li class="chapter" data-level="17.5" data-path="python-data-types.html"><a href="python-data-types.html#lists-1"><i class="fa fa-check"></i><b>17.5</b> Lists</a></li>
<li class="chapter" data-level="17.6" data-path="python-data-types.html"><a href="python-data-types.html#tuples"><i class="fa fa-check"></i><b>17.6</b> Tuples</a></li>
<li class="chapter" data-level="17.7" data-path="python-data-types.html"><a href="python-data-types.html#sets"><i class="fa fa-check"></i><b>17.7</b> Sets</a></li>
<li class="chapter" data-level="17.8" data-path="python-data-types.html"><a href="python-data-types.html#dictionary"><i class="fa fa-check"></i><b>17.8</b> Dictionary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="python-strings.html"><a href="python-strings.html"><i class="fa fa-check"></i><b>18</b> – Python: Strings</a>
<ul>
<li class="chapter" data-level="18.1" data-path="python-strings.html"><a href="python-strings.html#string-operators"><i class="fa fa-check"></i><b>18.1</b> String Operators</a></li>
<li class="chapter" data-level="18.2" data-path="python-strings.html"><a href="python-strings.html#other-string-modifiers"><i class="fa fa-check"></i><b>18.2</b> Other String Modifiers</a></li>
<li class="chapter" data-level="18.3" data-path="python-strings.html"><a href="python-strings.html#format-characters-in-strings"><i class="fa fa-check"></i><b>18.3</b> Format-Characters in Strings</a></li>
<li class="chapter" data-level="18.4" data-path="python-strings.html"><a href="python-strings.html#comparisons-with-strings"><i class="fa fa-check"></i><b>18.4</b> Comparisons with Strings</a></li>
<li class="chapter" data-level="18.5" data-path="python-strings.html"><a href="python-strings.html#padding-strings"><i class="fa fa-check"></i><b>18.5</b> Padding strings</a></li>
<li class="chapter" data-level="18.6" data-path="python-strings.html"><a href="python-strings.html#search-for-words-in-string-and-output-location"><i class="fa fa-check"></i><b>18.6</b> Search for words in string and output location</a></li>
<li class="chapter" data-level="18.7" data-path="python-strings.html"><a href="python-strings.html#seek-and-replace-in-strings"><i class="fa fa-check"></i><b>18.7</b> Seek and replace in strings</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="python-operators.html"><a href="python-operators.html"><i class="fa fa-check"></i><b>19</b> – Python: Operators</a>
<ul>
<li class="chapter" data-level="19.1" data-path="python-operators.html"><a href="python-operators.html#many-types-of-operators"><i class="fa fa-check"></i><b>19.1</b> Many types of operators</a></li>
<li class="chapter" data-level="19.2" data-path="python-operators.html"><a href="python-operators.html#arithmetic-operators"><i class="fa fa-check"></i><b>19.2</b> Arithmetic operators</a></li>
<li class="chapter" data-level="19.3" data-path="python-operators.html"><a href="python-operators.html#comparison-operators"><i class="fa fa-check"></i><b>19.3</b> Comparison operators</a></li>
<li class="chapter" data-level="19.4" data-path="python-operators.html"><a href="python-operators.html#logical-operators"><i class="fa fa-check"></i><b>19.4</b> Logical operators</a></li>
<li class="chapter" data-level="19.5" data-path="python-operators.html"><a href="python-operators.html#bitwise-operators"><i class="fa fa-check"></i><b>19.5</b> Bitwise operators</a></li>
<li class="chapter" data-level="19.6" data-path="python-operators.html"><a href="python-operators.html#identity-operators"><i class="fa fa-check"></i><b>19.6</b> Identity Operators</a></li>
<li class="chapter" data-level="19.7" data-path="python-operators.html"><a href="python-operators.html#membership-operators"><i class="fa fa-check"></i><b>19.7</b> Membership operators</a></li>
<li class="chapter" data-level="19.8" data-path="python-operators.html"><a href="python-operators.html#what-is-the-order-of-operations-with-operators"><i class="fa fa-check"></i><b>19.8</b> What is the order of operations with operators?</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="biopython.html"><a href="biopython.html"><i class="fa fa-check"></i><b>20</b> – Biopython</a>
<ul>
<li class="chapter" data-level="20.1" data-path="biopython.html"><a href="biopython.html#introduction-to-biopython"><i class="fa fa-check"></i><b>20.1</b> Introduction to Biopython</a></li>
<li class="chapter" data-level="20.2" data-path="biopython.html"><a href="biopython.html#biopython-installation"><i class="fa fa-check"></i><b>20.2</b> Biopython Installation</a></li>
<li class="chapter" data-level="20.3" data-path="biopython.html"><a href="biopython.html#getting-started"><i class="fa fa-check"></i><b>20.3</b> Getting Started</a></li>
<li class="chapter" data-level="20.4" data-path="biopython.html"><a href="biopython.html#parsing-my-fasta"><i class="fa fa-check"></i><b>20.4</b> Parsing my FASTA</a></li>
<li class="chapter" data-level="20.5" data-path="biopython.html"><a href="biopython.html#parsing-my-genbank-or-genpept"><i class="fa fa-check"></i><b>20.5</b> Parsing my GenBank or GenPept</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="biopython-sequences.html"><a href="biopython-sequences.html"><i class="fa fa-check"></i><b>21</b> – Biopython: Sequences</a>
<ul>
<li class="chapter" data-level="21.1" data-path="biopython-sequences.html"><a href="biopython-sequences.html#sequence-objects"><i class="fa fa-check"></i><b>21.1</b> Sequence Objects</a></li>
<li class="chapter" data-level="21.2" data-path="biopython-sequences.html"><a href="biopython-sequences.html#concatenation-with-sequences"><i class="fa fa-check"></i><b>21.2</b> Concatenation with Sequences</a></li>
<li class="chapter" data-level="21.3" data-path="biopython-sequences.html"><a href="biopython-sequences.html#transcription-dna-to-mrna"><i class="fa fa-check"></i><b>21.3</b> Transcription: DNA to mRNA</a></li>
<li class="chapter" data-level="21.4" data-path="biopython-sequences.html"><a href="biopython-sequences.html#translation-rna-to-prot"><i class="fa fa-check"></i><b>21.4</b> Translation: RNA to PROT</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html"><i class="fa fa-check"></i><b>22</b> – Biopython: Mulitple Sequence Alignments</a>
<ul>
<li class="chapter" data-level="22.1" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwise2"><i class="fa fa-check"></i><b>22.1</b> pairwise2</a></li>
<li class="chapter" data-level="22.2" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwisealigner"><i class="fa fa-check"></i><b>22.2</b> PairwiseAligner</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html"><i class="fa fa-check"></i><b>23</b> – Biopython: GenomeDiagram</a>
<ul>
<li class="chapter" data-level="23.1" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#linearcircular-gene-diagrams"><i class="fa fa-check"></i><b>23.1</b> Linear/Circular Gene Diagrams</a></li>
<li class="chapter" data-level="23.2" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#diagram-readability"><i class="fa fa-check"></i><b>23.2</b> Diagram Readability</a></li>
<li class="chapter" data-level="23.3" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#box-shape"><i class="fa fa-check"></i><b>23.3</b> Box Shape</a></li>
<li class="chapter" data-level="23.4" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#edit-box-shape-arrows"><i class="fa fa-check"></i><b>23.4</b> Edit Box Shape: Arrows</a></li>
<li class="chapter" data-level="23.5" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#restriction-map"><i class="fa fa-check"></i><b>23.5</b> Restriction Map</a></li>
<li class="chapter" data-level="23.6" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#tracks-on-tracks"><i class="fa fa-check"></i><b>23.6</b> Tracks on Tracks</a></li>
<li class="chapter" data-level="23.7" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#cross-link-data"><i class="fa fa-check"></i><b>23.7</b> Cross-link Data</a></li>
<li class="chapter" data-level="23.8" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#clean-cross-link"><i class="fa fa-check"></i><b>23.8</b> Clean Cross-Link</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html"><i class="fa fa-check"></i><b>24</b> – Biopython: Chromosome Modeling</a>
<ul>
<li class="chapter" data-level="24.1" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html#gene-feature-modeling"><i class="fa fa-check"></i><b>24.1</b> Gene Feature Modeling</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html"><i class="fa fa-check"></i><b>25</b> – Biopython: Sequence Parsing with Plots</a>
<ul>
<li class="chapter" data-level="25.1" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#histograms-1"><i class="fa fa-check"></i><b>25.1</b> Histograms</a></li>
<li class="chapter" data-level="25.2" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#gc-content"><i class="fa fa-check"></i><b>25.2</b> GC Content</a></li>
<li class="chapter" data-level="25.3" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#nt-alignment-dot-plot"><i class="fa fa-check"></i><b>25.3</b> Nt Alignment Dot Plot</a></li>
<li class="chapter" data-level="25.4" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#bigger-dots-for-the-dot-plot"><i class="fa fa-check"></i><b>25.4</b> Bigger Dots for the Dot Plot</a></li>
<li class="chapter" data-level="25.5" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#fastq-we-did-not-forget-you"><i class="fa fa-check"></i><b>25.5</b> FASTQ we did not forget you</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html"><i class="fa fa-check"></i><b>26</b> – Python: Web Scraping w/ BeautifulSoup</a>
<ul>
<li class="chapter" data-level="26.1" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#introduction"><i class="fa fa-check"></i><b>26.1</b> Introduction</a></li>
<li class="chapter" data-level="26.2" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#installation"><i class="fa fa-check"></i><b>26.2</b> Installation</a></li>
<li class="chapter" data-level="26.3" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#getting-started-1"><i class="fa fa-check"></i><b>26.3</b> Getting started</a></li>
<li class="chapter" data-level="26.4" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-all-matching-tags"><i class="fa fa-check"></i><b>26.4</b> Mining all matching tags</a></li>
<li class="chapter" data-level="26.5" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-tags-by-class-or-id"><i class="fa fa-check"></i><b>26.5</b> Mining tags by class or id</a></li>
<li class="chapter" data-level="26.6" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#use-css-selectors-for-positional-mining"><i class="fa fa-check"></i><b>26.6</b> Use CSS Selectors for positional mining</a></li>
<li class="chapter" data-level="26.7" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#weather-page-example"><i class="fa fa-check"></i><b>26.7</b> Weather Page Example</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="python-subprocess.html"><a href="python-subprocess.html"><i class="fa fa-check"></i><b>27</b> – Python Subprocess</a>
<ul>
<li class="chapter" data-level="27.1" data-path="python-subprocess.html"><a href="python-subprocess.html#introduction-and-installation"><i class="fa fa-check"></i><b>27.1</b> Introduction and Installation</a></li>
<li class="chapter" data-level="27.2" data-path="python-subprocess.html"><a href="python-subprocess.html#creating-your-first-python-script-w-bbtools"><i class="fa fa-check"></i><b>27.2</b> Creating your first python script w/ BBTools</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html"><i class="fa fa-check"></i><b>28</b> – Control Structures &amp; Loops in R</a>
<ul>
<li class="chapter" data-level="28.1" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#controling-the-flow"><i class="fa fa-check"></i><b>28.1</b> Controling the flow</a></li>
<li class="chapter" data-level="28.2" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if"><i class="fa fa-check"></i><b>28.2</b> Conditional <code>if</code></a></li>
<li class="chapter" data-level="28.3" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if-and-else"><i class="fa fa-check"></i><b>28.3</b> Conditional <code>if</code> and <code>else</code></a></li>
<li class="chapter" data-level="28.4" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#nested-conditionals-if-else-and-if-else"><i class="fa fa-check"></i><b>28.4</b> Nested conditionals <code>if</code>, <code>else</code>, and <code>if else</code></a></li>
<li class="chapter" data-level="28.5" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-ifelse"><i class="fa fa-check"></i><b>28.5</b> Conditional <code>ifelse</code></a></li>
<li class="chapter" data-level="28.6" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#loops-that-control-iterations-and-repeations"><i class="fa fa-check"></i><b>28.6</b> Loops that control iterations and repeations</a></li>
<li class="chapter" data-level="28.7" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-for-loop"><i class="fa fa-check"></i><b>28.7</b> The <code>for</code> Loop</a></li>
<li class="chapter" data-level="28.8" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-while-loop"><i class="fa fa-check"></i><b>28.8</b> The <code>while</code> Loop</a></li>
<li class="chapter" data-level="28.9" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-repeat-loop"><i class="fa fa-check"></i><b>28.9</b> The <code>repeat</code> Loop</a></li>
<li class="chapter" data-level="28.10" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-next-and-break-statements"><i class="fa fa-check"></i><b>28.10</b> The <code>next</code> and <code>break</code> statements</a></li>
<li class="chapter" data-level="28.11" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functions"><i class="fa fa-check"></i><b>28.11</b> Other loop functions</a></li>
<li class="chapter" data-level="28.12" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#lapply-functions"><i class="fa fa-check"></i><b>28.12</b> <code>lapply</code> functions</a></li>
<li class="chapter" data-level="28.13" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsapply"><i class="fa fa-check"></i><b>28.13</b> Other loop functions:<code>apply</code></a></li>
<li class="chapter" data-level="28.14" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsmapply"><i class="fa fa-check"></i><b>28.14</b> Other loop functions:<code>mapply</code></a></li>
<li class="chapter" data-level="28.15" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionstapply"><i class="fa fa-check"></i><b>28.15</b> Other loop functions:<code>tapply</code></a></li>
<li class="chapter" data-level="28.16" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionssplit"><i class="fa fa-check"></i><b>28.16</b> Other loop functions:<code>split</code></a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html"><i class="fa fa-check"></i><b>29</b> – Bioinformatics Literature</a>
<ul>
<li class="chapter" data-level="29.1" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#peer-reviewed-bioinformatics-journals"><i class="fa fa-check"></i><b>29.1</b> Peer-reviewed bioinformatics journals</a></li>
<li class="chapter" data-level="29.2" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#access-to-journals-articles"><i class="fa fa-check"></i><b>29.2</b> Access to journals articles</a></li>
<li class="chapter" data-level="29.3" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#informal-access-to-code-preprints-and-vignettes"><i class="fa fa-check"></i><b>29.3</b> Informal access to code: Preprints and vignettes</a></li>
<li class="chapter" data-level="29.4" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#online-bioinformatics-courses"><i class="fa fa-check"></i><b>29.4</b> Online bioinformatics courses</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html"><i class="fa fa-check"></i><b>30</b> – Importing/Exporting Data into R</a>
<ul>
<li class="chapter" data-level="30.1" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#typing-directly"><i class="fa fa-check"></i><b>30.1</b> Typing directly</a></li>
<li class="chapter" data-level="30.2" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#summary-versus-raw-data"><i class="fa fa-check"></i><b>30.2</b> Summary versus raw data</a></li>
<li class="chapter" data-level="30.3" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#downloading-data-files-github"><i class="fa fa-check"></i><b>30.3</b> Downloading data files (GitHub)</a></li>
<li class="chapter" data-level="30.4" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#importing-data-from-a-file"><i class="fa fa-check"></i><b>30.4</b> Importing data from a file</a></li>
<li class="chapter" data-level="30.5" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-to-a-file"><i class="fa fa-check"></i><b>30.5</b> Saving data to a file</a></li>
<li class="chapter" data-level="30.6" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-as-rdata-.rdata-.rda"><i class="fa fa-check"></i><b>30.6</b> Saving data as Rdata (.rdata, .rda)</a></li>
<li class="chapter" data-level="30.7" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#import-excel-formated-data"><i class="fa fa-check"></i><b>30.7</b> Import Excel formated data</a></li>
<li class="chapter" data-level="30.8" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#save-data-as-excel-formated-file"><i class="fa fa-check"></i><b>30.8</b> Save data as Excel formated file</a></li>
<li class="chapter" data-level="30.9" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#fixing-problems-while-importing-datasets"><i class="fa fa-check"></i><b>30.9</b> Fixing problems while importing datasets</a></li>
<li class="chapter" data-level="30.10" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#simplifying-embellished-excel-datasets"><i class="fa fa-check"></i><b>30.10</b> Simplifying embellished excel datasets</a></li>
<li class="chapter" data-level="30.11" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#accessing-online-data"><i class="fa fa-check"></i><b>30.11</b> Accessing online data</a></li>
<li class="chapter" data-level="30.12" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#data-from-clinicaltrials.gov"><i class="fa fa-check"></i><b>30.12</b> Data from clinicaltrials.gov</a></li>
<li class="chapter" data-level="30.13" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#data-scrapping-from-a-website"><i class="fa fa-check"></i><b>30.13</b> Data scrapping from a website</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="working-with-proteins.html"><a href="working-with-proteins.html"><i class="fa fa-check"></i><b>31</b> – Working with Proteins</a>
<ul>
<li class="chapter" data-level="31.1" data-path="working-with-proteins.html"><a href="working-with-proteins.html#sequences-in-fasta-format"><i class="fa fa-check"></i><b>31.1</b> Sequences in FASTA format</a></li>
<li class="chapter" data-level="31.2" data-path="working-with-proteins.html"><a href="working-with-proteins.html#fetching-aa-sequences-from-ncbi"><i class="fa fa-check"></i><b>31.2</b> Fetching AA sequences from NCBI</a></li>
<li class="chapter" data-level="31.3" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-aa-sequences-in-r"><i class="fa fa-check"></i><b>31.3</b> Importing AA sequences in R</a></li>
<li class="chapter" data-level="31.4" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-nucleotide-sequences-in-r"><i class="fa fa-check"></i><b>31.4</b> Importing nucleotide sequences in R</a></li>
<li class="chapter" data-level="31.5" data-path="working-with-proteins.html"><a href="working-with-proteins.html#translating-a-nucleotides-to-aa"><i class="fa fa-check"></i><b>31.5</b> Translating a nucleotides to AA</a></li>
<li class="chapter" data-level="31.6" data-path="working-with-proteins.html"><a href="working-with-proteins.html#finding-orfs-in-a-nucleotide-sequence"><i class="fa fa-check"></i><b>31.6</b> Finding ORFs in a nucleotide sequence</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="miscellaneous.html"><a href="miscellaneous.html"><i class="fa fa-check"></i><b>32</b> – Miscellaneous</a>
<ul>
<li class="chapter" data-level="32.1" data-path="miscellaneous.html"><a href="miscellaneous.html#sampling-from-a-vector-of-names"><i class="fa fa-check"></i><b>32.1</b> Sampling from a vector of names</a></li>
<li class="chapter" data-level="32.2" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-an-article-pdf"><i class="fa fa-check"></i><b>32.2</b> Read a text from an article pdf</a></li>
<li class="chapter" data-level="32.3" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-website"><i class="fa fa-check"></i><b>32.3</b> Read a text from a website</a></li>
<li class="chapter" data-level="32.4" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-file"><i class="fa fa-check"></i><b>32.4</b> Read a text from a file</a></li>
<li class="chapter" data-level="32.5" data-path="miscellaneous.html"><a href="miscellaneous.html#generate-the-word-cloud"><i class="fa fa-check"></i><b>32.5</b> Generate the word cloud</a></li>
<li class="chapter" data-level="32.6" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-pdf"><i class="fa fa-check"></i><b>32.6</b> worldcloud – PDF</a></li>
<li class="chapter" data-level="32.7" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-website"><i class="fa fa-check"></i><b>32.7</b> worldcloud – website</a></li>
<li class="chapter" data-level="32.8" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-text"><i class="fa fa-check"></i><b>32.8</b> worldcloud – text</a></li>
<li class="chapter" data-level="32.9" data-path="miscellaneous.html"><a href="miscellaneous.html#miscellaneous-1"><i class="fa fa-check"></i><b>32.9</b> Miscellaneous</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics Workshop Gitbook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="graphics-and-colors" class="section level1" number="9">
<h1><span class="header-section-number">Session 9</span> – Graphics and Colors</h1>
<p>Plots, graphs and other visual representations of data and central to bioinformatics. Such illustrations summarize year/months of data collection and complex statistical analyzes. The main goal of such visual representations is to improve the observer, reader or researcher his/her understanding of the data and his/her abilities to compare with other datasets and reveal trends in the data collected, compiled or analyzed. However, you might have seen good, bad and ugly graphs. Yet, an effective graph is easier to decode or understand than others, by virtue of this, it is likely to be beautiful.</p>
<div id="avoid-bad-graphics" class="section level2" number="9.1">
<h2><span class="header-section-number">9.1</span> Avoid bad graphics</h2>
<p>I think it is a good idea to review about effective graphics. Here is a list of suggestions, mostly summarized, from the <a href="https://stat545.com/effective-graphs.html" target="_blank"><strong>graphics section</strong></a> by Jenny Bryan STAT 545 class at UBC.</p>
<p><strong>A) No 3D or overdone plots:</strong> Our current format to visualize data is paper or screen (a 2D surfaces) by adding another dimension makes information difficult to visualize. Likewise, eliminating clutter and de-emphasizing supporting elements make reduce the attention exhaustion of the reader (i.e., every aspect of a figure should add if needed). <strong>Suggestions:</strong> Try to keep graphics simple and make most of them as 2D, no shadows or excess ornamentation.</p>
<p>Here is a visual animation that transforms an overdone bar chart into one that is just right with the information that it wants to communicate.
<img src="figures/session_9_figs/10_bad_chart_bar.gif" width="70%" />
<strong>Source</strong> <a href="https://www.darkhorseanalytics.com/blog/data-looks-better-naked" target="_blank">https://www.darkhorseanalytics.com</a>.</p>
<p><strong>B) No Pie Charts:</strong> This type of infographic is probably the least effective when encoding quantitative data (e.g., percentages or fractions) because pie charts use angles and areas, which are very hard for visualize or compare. <strong>Suggestions:</strong> Use bar charts and dotplot for proportions and percentages. These graphs make comparisons easier by positioning objects along a common scale.
<img src="figures/session_9_figs/11_bad_pie.gif" width="70%" />
<strong>Source</strong> <a href="https://www.darkhorseanalytics.com/blog/salvaging-the-pie" target="_blank">https://www.darkhorseanalytics.com</a>.</p>
<p><strong>C) Careful about scale of things:</strong> Some data need to be scaled or transformed (e.g., logarithms) before plotting to allow the reader to visualize the data distribution. For example, some variables show a multiplicative or exponential scales that jumps from tens to hundreds or thousands. ** Suggestions:** Explore you data first and determine if it needs transformation.</p>
<div class="sourceCode" id="cb393"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb393-1"><a href="graphics-and-colors.html#cb393-1" aria-hidden="true" tabindex="-1"></a><span class="do">## load mammal dataset from &#39;MASS&#39; and transform them with modify() from &#39;purrr&#39;</span></span>
<span id="cb393-2"><a href="graphics-and-colors.html#cb393-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb393-3"><a href="graphics-and-colors.html#cb393-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb393-4"><a href="graphics-and-colors.html#cb393-4" aria-hidden="true" tabindex="-1"></a>mammals_raw <span class="ot">&lt;-</span> mammals</span>
<span id="cb393-5"><a href="graphics-and-colors.html#cb393-5" aria-hidden="true" tabindex="-1"></a>mammals_log <span class="ot">&lt;-</span> <span class="fu">modify</span>(mammals_raw,log10)</span>
<span id="cb393-6"><a href="graphics-and-colors.html#cb393-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb393-7"><a href="graphics-and-colors.html#cb393-7" aria-hidden="true" tabindex="-1"></a><span class="do">## plot them side to side into a 1*2 array</span></span>
<span id="cb393-8"><a href="graphics-and-colors.html#cb393-8" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))   </span>
<span id="cb393-9"><a href="graphics-and-colors.html#cb393-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb393-10"><a href="graphics-and-colors.html#cb393-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="at">x =</span> mammals_raw<span class="sc">$</span>body, <span class="at">y =</span> mammals_raw<span class="sc">$</span>brain)</span>
<span id="cb393-11"><a href="graphics-and-colors.html#cb393-11" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="at">side=</span><span class="dv">3</span>, <span class="at">line=</span><span class="fl">0.5</span>, <span class="at">adj=</span><span class="dv">0</span>, <span class="st">&quot;raw data from mammal -- body vs brain&quot;</span>)</span>
<span id="cb393-12"><a href="graphics-and-colors.html#cb393-12" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="fu">lm</span>(brain <span class="sc">~</span> body, <span class="at">data =</span> mammals_raw), <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb393-13"><a href="graphics-and-colors.html#cb393-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb393-14"><a href="graphics-and-colors.html#cb393-14" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="at">x =</span> mammals_log<span class="sc">$</span>body, <span class="at">y =</span> mammals_log<span class="sc">$</span>brain)</span>
<span id="cb393-15"><a href="graphics-and-colors.html#cb393-15" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="at">side=</span><span class="dv">3</span>, <span class="at">line=</span><span class="fl">0.5</span>, <span class="at">adj=</span><span class="dv">0</span>, <span class="st">&quot;Log10 data from mammal -- body vs brain&quot;</span>)</span>
<span id="cb393-16"><a href="graphics-and-colors.html#cb393-16" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="fu">lm</span>(brain <span class="sc">~</span> body, <span class="at">data =</span> mammals_log), <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>)</span></code></pre></div>
<p><img src="figures/session_9_figs/12_plot_transformed.png" width="100%" /></p>
<p><strong>D) Avoid Microsoft Excel default plots:</strong> Most people use Excel at some point to store their data, then comes the temptation to use their reduced offer of plots and charts. In general, these tend to be plain and less customizable than those derived from R graphics. ** Suggestions:** You already are taking this workshop.</p>
<p><strong>E) Careful with the figure legend:</strong> Most graphics need a legend when they are in paper, reports or other divulgation source as these images should stand alone from the main text of the article. Then, the accompanying legend should not contradict what the graphic is showing (i.e., self-contradiction). This is a common mistake and can be used against your paper and thesis by reviewers that will consider that your work is careless and sloppy. ** Suggestions:** Always proofread your manuscripts.</p>
<p><strong>F) Make simple tables:</strong> Tables also should be easy to read with simple formats (use minimal or no grid lines). The information should be visible and no more three decimal places. Remember that p-values cannot be zero (i.e., probability never is equal to 0), instead use p &lt; 0.001 if a value is really small or use scientific notation if such tiny p-value is of relevance.</p>
<p><img src="figures/session_9_figs/56_ClearOffTheTableMd.gif" width="70%" /></p>
<p><strong>Source</strong> <a href="https://www.darkhorseanalytics.com/blog/clear-off-the-table">https://www.darkhorseanalytics.com</a></p>
<p><strong>G) Other Plots to avoid:</strong> Radar charts that make comparisons of data points very difficult as they are placed in a circle, the middle portion stacks too many contrasts and there is a disproportionate emphasis of high numbers (i.e., middle or low values are less visible). Use stack bars or bar charts.</p>
</div>
<div id="graphics-in-r-ggplot2" class="section level2" number="9.2">
<h2><span class="header-section-number">9.2</span> Graphics in R – ggplot2</h2>
<p>The preloaded <code>graphics</code> package of R helps the easy access plots, bar charts, boxplots and other. However, with the development of the <a href="https://ggplot2.tidyverse.org/" target="_blank"><strong>ggplot2</strong></a> package that is part of the tidyverse collection. This package and its growing collection of add-ons, which include at least <a href="https://exts.ggplot2.tidyverse.org/gallery/" target="_blank"><strong>81</strong></a>, has made <a href="https://ggplot2.tidyverse.org/" target="_blank"><strong>ggplot2</strong></a> a dominant, flexible, and beautiful R graphics package. Therefore, we will explore graphics in R using this package. Like most visualizations, it would be ideal to start with installing <strong>ggplot2</strong> and datasets that will serve as examples.</p>
<p><strong>1)</strong> If you have not installed <a href="https://ggplot2.tidyverse.org/" target="_blank"><strong>ggplot2</strong></a>, you should follow these instructions before we explore graphics and plots.</p>
<div class="sourceCode" id="cb394"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb394-1"><a href="graphics-and-colors.html#cb394-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb394-2"><a href="graphics-and-colors.html#cb394-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code></pre></div>
</div>
<div id="preparing-your-dataset" class="section level2" number="9.3">
<h2><span class="header-section-number">9.3</span> Preparing your dataset</h2>
<p><strong>2)</strong> We also need some reference datasets to use and plot for our graphics. We will use <code>cars</code> and the <code>airway_scaledcounts.csv</code> datasets.</p>
<div class="sourceCode" id="cb395"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb395-1"><a href="graphics-and-colors.html#cb395-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we will load &#39;ade4&#39; package to get the bacteria dataset</span></span>
<span id="cb395-2"><a href="graphics-and-colors.html#cb395-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb395-3"><a href="graphics-and-colors.html#cb395-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(bacteria,  <span class="at">package =</span> <span class="st">&quot;ade4&quot;</span>)</span>
<span id="cb395-4"><a href="graphics-and-colors.html#cb395-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb395-5"><a href="graphics-and-colors.html#cb395-5" aria-hidden="true" tabindex="-1"></a><span class="do">## we get the espaa dataframe with number of the 21 amino acids present in 43 species of bacteria</span></span>
<span id="cb395-6"><a href="graphics-and-colors.html#cb395-6" aria-hidden="true" tabindex="-1"></a>bacteria_espaa <span class="ot">&lt;-</span> bacteria<span class="sc">$</span>espaa</span></code></pre></div>
<p>We then scale this dataset using the function <code>modify()</code> from the package <code>purrr</code>.</p>
<div class="sourceCode" id="cb396"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb396-1"><a href="graphics-and-colors.html#cb396-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load purrr package and scale the values</span></span>
<span id="cb396-2"><a href="graphics-and-colors.html#cb396-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(purrr)</span>
<span id="cb396-3"><a href="graphics-and-colors.html#cb396-3" aria-hidden="true" tabindex="-1"></a>bacteria_espaa_scaled <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">modify</span>(bacteria_espaa, scale)</span>
<span id="cb396-4"><a href="graphics-and-colors.html#cb396-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(bacteria_espaa_scaled)</span>
<span id="cb396-5"><a href="graphics-and-colors.html#cb396-5" aria-hidden="true" tabindex="-1"></a><span class="co">#               Ala        Arg         Asn        Asp        Cys        Gln        Glu        Gly        His         Ile         Leu</span></span>
<span id="cb396-6"><a href="graphics-and-colors.html#cb396-6" aria-hidden="true" tabindex="-1"></a><span class="co">#AERPECG  0.3285668  0.7317790 -0.57189998 -0.1194444  0.1648422 -0.3942439  0.2871196  0.4766228  0.1461376 -0.04383607  0.39172834</span></span>
<span id="cb396-7"><a href="graphics-and-colors.html#cb396-7" aria-hidden="true" tabindex="-1"></a><span class="co">#AQUAECG -0.4008694 -0.2034641 -0.22356528 -0.3025407 -0.3950725 -0.5262848  0.4634441 -0.1820602 -0.3947798  0.06415029 -0.09928685</span></span>
<span id="cb396-8"><a href="graphics-and-colors.html#cb396-8" aria-hidden="true" tabindex="-1"></a><span class="co">#ARCFUCG  0.1073331  0.2833748  0.04401802  0.2066524  0.5498823 -0.4297732  0.8885965  0.2463444 -0.1330163  0.57892747  0.12987680</span></span>
<span id="cb396-9"><a href="graphics-and-colors.html#cb396-9" aria-hidden="true" tabindex="-1"></a><span class="co">#BACHDCG  0.9012986  0.9875656  1.72105862  1.5171379  0.8224654  1.6598743  2.2634710  1.2984007  1.9400648  2.16751418  1.40433248</span></span>
<span id="cb396-10"><a href="graphics-and-colors.html#cb396-10" aria-hidden="true" tabindex="-1"></a><span class="co">#BACSUCG  1.0314722  0.7271505  2.09337260  1.6457327  1.0718724  1.5394241  2.0612524  1.3252195  1.8173099  2.49733951  1.38291228</span></span>
<span id="cb396-11"><a href="graphics-and-colors.html#cb396-11" aria-hidden="true" tabindex="-1"></a><span class="co">#BORBUCG -0.7597170 -0.7465478  0.01654507 -0.6016028 -0.9081081 -0.7293261 -0.6491864 -0.7234675 -0.8524967 -0.18035597 -0.60867882</span></span>
<span id="cb396-12"><a href="graphics-and-colors.html#cb396-12" aria-hidden="true" tabindex="-1"></a><span class="co">#                Lys         Met        Phe         Pro        Ser        Stp          Thr         Trp        Tyr         Val</span></span>
<span id="cb396-13"><a href="graphics-and-colors.html#cb396-13" aria-hidden="true" tabindex="-1"></a><span class="co">#AERPECG -0.36413421  0.09017282 -0.3275198  0.89114307  0.7319805  0.7113529  0.152094962  0.38406553  0.4369524  0.61086466</span></span>
<span id="cb396-14"><a href="graphics-and-colors.html#cb396-14" aria-hidden="true" tabindex="-1"></a><span class="co">#AQUAECG  0.88477099 -0.28951403  0.1851469 -0.17796712 -0.4145196 -0.2123879 -0.323609531 -0.26598780  0.3076040  0.01812197</span></span>
<span id="cb396-15"><a href="graphics-and-colors.html#cb396-15" aria-hidden="true" tabindex="-1"></a><span class="co">#ARCFUCG  0.79679827  0.48462587  0.5156315  0.08558657  0.1583118  0.2768920  0.008319353  0.09800132  0.6381195  0.61451548</span></span>
<span id="cb396-16"><a href="graphics-and-colors.html#cb396-16" aria-hidden="true" tabindex="-1"></a><span class="co">#BACHDCG  2.08829038  2.07351272  2.0439218  1.06116153  1.5330792  1.4796369  1.887894271  1.20439921  2.0999898  1.67959246</span></span>
<span id="cb396-17"><a href="graphics-and-colors.html#cb396-17" aria-hidden="true" tabindex="-1"></a><span class="co">#BACSUCG  2.95243158  2.18339064  2.1725251  1.03849240  1.9769142  1.5369105  1.887693467  1.04854940  2.2918176  1.46336710</span></span>
<span id="cb396-18"><a href="graphics-and-colors.html#cb396-18" aria-hidden="true" tabindex="-1"></a><span class="co">#BORBUCG -0.02590849 -0.72066544 -0.3193467 -0.80668435 -0.5156512 -0.7990328 -0.785609156 -0.79075009 -0.4463524 -0.78199455</span></span></code></pre></div>
<p>We can also create a more easy to map plot with using the function <code>melt()</code> of the package <a href="https://cran.r-project.org/web/packages/reshape2/index.html" target="_blank"><strong>reshape2</strong></a>.</p>
<div class="sourceCode" id="cb397"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb397-1"><a href="graphics-and-colors.html#cb397-1" aria-hidden="true" tabindex="-1"></a><span class="do">## load reshape</span></span>
<span id="cb397-2"><a href="graphics-and-colors.html#cb397-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb397-3"><a href="graphics-and-colors.html#cb397-3" aria-hidden="true" tabindex="-1"></a><span class="co"># use melt() to transform</span></span>
<span id="cb397-4"><a href="graphics-and-colors.html#cb397-4" aria-hidden="true" tabindex="-1"></a>bacteria_espaa_for_heatmap <span class="ot">&lt;-</span> <span class="fu">melt</span>(<span class="fu">as.matrix</span>(bacteria_espaa_scaled))</span>
<span id="cb397-5"><a href="graphics-and-colors.html#cb397-5" aria-hidden="true" tabindex="-1"></a><span class="co"># transform factor variables to character</span></span>
<span id="cb397-6"><a href="graphics-and-colors.html#cb397-6" aria-hidden="true" tabindex="-1"></a>factor_columns <span class="ot">&lt;-</span> <span class="fu">sapply</span>(bacteria_espaa_for_heatmap, is.factor)</span>
<span id="cb397-7"><a href="graphics-and-colors.html#cb397-7" aria-hidden="true" tabindex="-1"></a>bacteria_espaa_for_heatmap[factor_columns] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(bacteria_espaa_for_heatmap[factor_columns], as.character)</span>
<span id="cb397-8"><a href="graphics-and-colors.html#cb397-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb397-9"><a href="graphics-and-colors.html#cb397-9" aria-hidden="true" tabindex="-1"></a><span class="do">## we can rename the columns in this data frame</span></span>
<span id="cb397-10"><a href="graphics-and-colors.html#cb397-10" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(bacteria_espaa_for_heatmap) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;bacteria_sp&quot;</span>, <span class="st">&quot;Amino_acid&quot;</span>, <span class="st">&quot;scaled_value&quot;</span>)</span>
<span id="cb397-11"><a href="graphics-and-colors.html#cb397-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(bacteria_espaa_for_heatmap)</span>
<span id="cb397-12"><a href="graphics-and-colors.html#cb397-12" aria-hidden="true" tabindex="-1"></a><span class="co">#  bacteria_sp Amino_acid scaled_value</span></span>
<span id="cb397-13"><a href="graphics-and-colors.html#cb397-13" aria-hidden="true" tabindex="-1"></a><span class="co">#1     AERPECG        Ala    0.3285668</span></span>
<span id="cb397-14"><a href="graphics-and-colors.html#cb397-14" aria-hidden="true" tabindex="-1"></a><span class="co">#2     AQUAECG        Ala   -0.4008694</span></span>
<span id="cb397-15"><a href="graphics-and-colors.html#cb397-15" aria-hidden="true" tabindex="-1"></a><span class="co">#3     ARCFUCG        Ala    0.1073331</span></span>
<span id="cb397-16"><a href="graphics-and-colors.html#cb397-16" aria-hidden="true" tabindex="-1"></a><span class="co">#4     BACHDCG        Ala    0.9012986</span></span>
<span id="cb397-17"><a href="graphics-and-colors.html#cb397-17" aria-hidden="true" tabindex="-1"></a><span class="co">#5     BACSUCG        Ala    1.0314722</span></span>
<span id="cb397-18"><a href="graphics-and-colors.html#cb397-18" aria-hidden="true" tabindex="-1"></a><span class="co">#6     BORBUCG        Ala   -0.7597170</span></span></code></pre></div>
<p><strong>3)</strong> A more complex dataset for color schemes, we will use the <code>airway_scaledcounts.csv</code> that has expression analysis data for several RNAseq experiments. This dataset is in our class <a href="https://github.com/corytophanes/BIO_BIT_Bioinformatics_209/tree/main/ref_files_pdfs" target="_blank"><strong>GitHub repository</strong></a>.</p>
<div class="sourceCode" id="cb398"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb398-1"><a href="graphics-and-colors.html#cb398-1" aria-hidden="true" tabindex="-1"></a><span class="do">## </span><span class="al">NOTE</span><span class="do">: remember to update the path to file with the dataset where you downloaded in your computer -- THIS IS EXCLUSIVE TO YOUR COMPUTER AND IT IS NOT THE PATH SHOWN BELOW</span></span>
<span id="cb398-2"><a href="graphics-and-colors.html#cb398-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb398-3"><a href="graphics-and-colors.html#cb398-3" aria-hidden="true" tabindex="-1"></a><span class="do">## load get &#39;airway_scaledcounts.csv&#39; dataset </span></span>
<span id="cb398-4"><a href="graphics-and-colors.html#cb398-4" aria-hidden="true" tabindex="-1"></a>airway_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/my_working_directory/airway_scaledcounts.csv&quot;</span>,</span>
<span id="cb398-5"><a href="graphics-and-colors.html#cb398-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb398-6"><a href="graphics-and-colors.html#cb398-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data)</span>
<span id="cb398-7"><a href="graphics-and-colors.html#cb398-7" aria-hidden="true" tabindex="-1"></a><span class="co">#          ensgene SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb398-8"><a href="graphics-and-colors.html#cb398-8" aria-hidden="true" tabindex="-1"></a><span class="co">#1 ENSG00000000003        723        486        904        445       1170       1097        806        604</span></span>
<span id="cb398-9"><a href="graphics-and-colors.html#cb398-9" aria-hidden="true" tabindex="-1"></a><span class="co">#2 ENSG00000000005          0          0          0          0          0          0          0          0</span></span>
<span id="cb398-10"><a href="graphics-and-colors.html#cb398-10" aria-hidden="true" tabindex="-1"></a><span class="co">#3 ENSG00000000419        467        523        616        371        582        781        417        509</span></span>
<span id="cb398-11"><a href="graphics-and-colors.html#cb398-11" aria-hidden="true" tabindex="-1"></a><span class="co">#4 ENSG00000000457        347        258        364        237        318        447        330        324</span></span>
<span id="cb398-12"><a href="graphics-and-colors.html#cb398-12" aria-hidden="true" tabindex="-1"></a><span class="co">#5 ENSG00000000460         96         81         73         66        118         94        102         74</span></span>
<span id="cb398-13"><a href="graphics-and-colors.html#cb398-13" aria-hidden="true" tabindex="-1"></a><span class="co">#6 ENSG00000000938          0          0          1          0          2          0          0          0</span></span>
<span id="cb398-14"><a href="graphics-and-colors.html#cb398-14" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(airway_data)</span>
<span id="cb398-15"><a href="graphics-and-colors.html#cb398-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  38694 obs. of  9 variables:</span></span>
<span id="cb398-16"><a href="graphics-and-colors.html#cb398-16" aria-hidden="true" tabindex="-1"></a><span class="co"># $ ensgene   : chr  &quot;ENSG00000000003&quot; &quot;ENSG00000000005&quot; &quot;ENSG00000000419&quot; &quot;ENSG00000000457&quot; ...</span></span>
<span id="cb398-17"><a href="graphics-and-colors.html#cb398-17" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SRR1039508: num  723 0 467 347 96 ...</span></span>
<span id="cb398-18"><a href="graphics-and-colors.html#cb398-18" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SRR1039509: num  486 0 523 258 81 ...</span></span>
<span id="cb398-19"><a href="graphics-and-colors.html#cb398-19" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SRR1039512: num  904 0 616 364 73 1 6000 2640 692 531 ...</span></span>
<span id="cb398-20"><a href="graphics-and-colors.html#cb398-20" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SRR1039513: num  445 0 371 237 66 ...</span></span>
<span id="cb398-21"><a href="graphics-and-colors.html#cb398-21" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SRR1039516: num  1170 0 582 318 118 ...</span></span>
<span id="cb398-22"><a href="graphics-and-colors.html#cb398-22" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SRR1039517: num  1097 0 781 447 94 ...</span></span>
<span id="cb398-23"><a href="graphics-and-colors.html#cb398-23" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SRR1039520: num  806 0 417 330 102 ...</span></span>
<span id="cb398-24"><a href="graphics-and-colors.html#cb398-24" aria-hidden="true" tabindex="-1"></a><span class="co"># $ SRR1039521: num  604 0 509 324 74 ...</span></span></code></pre></div>
<p>We then scale this dataset using the function <code>modify()</code> from the package <code>purrr</code>. We needed to remove and add again the names of the genes in column <code>ensgene</code>.</p>
<div class="sourceCode" id="cb399"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb399-1"><a href="graphics-and-colors.html#cb399-1" aria-hidden="true" tabindex="-1"></a>ensgene <span class="ot">&lt;-</span> airway_data[,<span class="dv">1</span>]</span>
<span id="cb399-2"><a href="graphics-and-colors.html#cb399-2" aria-hidden="true" tabindex="-1"></a>airway_data_numeric <span class="ot">&lt;-</span> airway_data[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb399-3"><a href="graphics-and-colors.html#cb399-3" aria-hidden="true" tabindex="-1"></a><span class="co"># load purrr package</span></span>
<span id="cb399-4"><a href="graphics-and-colors.html#cb399-4" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(purrr)</span>
<span id="cb399-5"><a href="graphics-and-colors.html#cb399-5" aria-hidden="true" tabindex="-1"></a>airway_data_scaled <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">modify</span>(airway_data_numeric, scale)</span>
<span id="cb399-6"><a href="graphics-and-colors.html#cb399-6" aria-hidden="true" tabindex="-1"></a><span class="co"># cbind the name of the genes from vector &#39;ensgene&#39;</span></span>
<span id="cb399-7"><a href="graphics-and-colors.html#cb399-7" aria-hidden="true" tabindex="-1"></a>airway_data_scaled <span class="ot">&lt;-</span> <span class="fu">cbind</span>(ensgene, airway_data_scaled, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb399-8"><a href="graphics-and-colors.html#cb399-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data_scaled)</span>
<span id="cb399-9"><a href="graphics-and-colors.html#cb399-9" aria-hidden="true" tabindex="-1"></a><span class="co">#          ensgene  SRR1039508   SRR1039509   SRR1039512   SRR1039513  SRR1039516   SRR1039517  SRR1039520   SRR1039521</span></span>
<span id="cb399-10"><a href="graphics-and-colors.html#cb399-10" aria-hidden="true" tabindex="-1"></a><span class="co">#1 ENSG00000000003  0.03933056 -0.003924270  0.036555187  0.011021618  0.09101424  0.041361372  0.06313763  0.007896859</span></span>
<span id="cb399-11"><a href="graphics-and-colors.html#cb399-11" aria-hidden="true" tabindex="-1"></a><span class="co">#2 ENSG00000000005 -0.12176875 -0.130284028 -0.106887175 -0.112547703 -0.11339060 -0.123910392 -0.10822243 -0.116658738</span></span>
<span id="cb399-12"><a href="graphics-and-colors.html#cb399-12" aria-hidden="true" tabindex="-1"></a><span class="co">#3 ENSG00000000419 -0.01771152  0.005695712 -0.009143265 -0.009526988 -0.01171230 -0.006246538 -0.01956592 -0.011693839</span></span>
<span id="cb399-13"><a href="graphics-and-colors.html#cb399-13" aria-hidden="true" tabindex="-1"></a><span class="co">#4 ENSG00000000457 -0.04444999 -0.063204156 -0.049129410 -0.046736626 -0.05783441 -0.056566291 -0.03806261 -0.049844146</span></span>
<span id="cb399-14"><a href="graphics-and-colors.html#cb399-14" aria-hidden="true" tabindex="-1"></a><span class="co">#5 ENSG00000000460 -0.10037797 -0.109224069 -0.095303887 -0.094220568 -0.09277541 -0.109748545 -0.08653667 -0.101398615</span></span>
<span id="cb399-15"><a href="graphics-and-colors.html#cb399-15" aria-hidden="true" tabindex="-1"></a><span class="co">#6 ENSG00000000938 -0.12176875 -0.130284028 -0.106728500 -0.112547703 -0.11304119 -0.123910392 -0.10822243 -0.116658738</span></span></code></pre></div>
<p>We can also create a more easy to map plot with using the function <code>melt()</code> of the package <a href="https://cran.r-project.org/web/packages/reshape2/index.html" target="_blank"><strong>reshape2</strong></a>.</p>
<div class="sourceCode" id="cb400"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb400-1"><a href="graphics-and-colors.html#cb400-1" aria-hidden="true" tabindex="-1"></a><span class="do">## load reshape</span></span>
<span id="cb400-2"><a href="graphics-and-colors.html#cb400-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb400-3"><a href="graphics-and-colors.html#cb400-3" aria-hidden="true" tabindex="-1"></a><span class="co"># prepare other data frame</span></span>
<span id="cb400-4"><a href="graphics-and-colors.html#cb400-4" aria-hidden="true" tabindex="-1"></a>airway_data_scaled2 <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">modify</span>(airway_data_numeric, scale)</span>
<span id="cb400-5"><a href="graphics-and-colors.html#cb400-5" aria-hidden="true" tabindex="-1"></a><span class="co"># add ensgene as rownames</span></span>
<span id="cb400-6"><a href="graphics-and-colors.html#cb400-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(airway_data_scaled2) <span class="ot">&lt;-</span> ensgene</span>
<span id="cb400-7"><a href="graphics-and-colors.html#cb400-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data_scaled2)</span>
<span id="cb400-8"><a href="graphics-and-colors.html#cb400-8" aria-hidden="true" tabindex="-1"></a><span class="co">#                 SRR1039508   SRR1039509   SRR1039512   SRR1039513  SRR1039516   SRR1039517  SRR1039520   SRR1039521</span></span>
<span id="cb400-9"><a href="graphics-and-colors.html#cb400-9" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000003  0.03933056 -0.003924270  0.036555187  0.011021618  0.09101424  0.041361372  0.06313763  0.007896859</span></span>
<span id="cb400-10"><a href="graphics-and-colors.html#cb400-10" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000005 -0.12176875 -0.130284028 -0.106887175 -0.112547703 -0.11339060 -0.123910392 -0.10822243 -0.116658738</span></span>
<span id="cb400-11"><a href="graphics-and-colors.html#cb400-11" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000419 -0.01771152  0.005695712 -0.009143265 -0.009526988 -0.01171230 -0.006246538 -0.01956592 -0.011693839</span></span>
<span id="cb400-12"><a href="graphics-and-colors.html#cb400-12" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000457 -0.04444999 -0.063204156 -0.049129410 -0.046736626 -0.05783441 -0.056566291 -0.03806261 -0.049844146</span></span>
<span id="cb400-13"><a href="graphics-and-colors.html#cb400-13" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000460 -0.10037797 -0.109224069 -0.095303887 -0.094220568 -0.09277541 -0.109748545 -0.08653667 -0.101398615</span></span>
<span id="cb400-14"><a href="graphics-and-colors.html#cb400-14" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000938 -0.12176875 -0.130284028 -0.106728500 -0.112547703 -0.11304119 -0.123910392 -0.10822243 -0.116658738</span></span>
<span id="cb400-15"><a href="graphics-and-colors.html#cb400-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb400-16"><a href="graphics-and-colors.html#cb400-16" aria-hidden="true" tabindex="-1"></a><span class="do">## get matrix dimensions</span></span>
<span id="cb400-17"><a href="graphics-and-colors.html#cb400-17" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(airway_data_scaled2)</span>
<span id="cb400-18"><a href="graphics-and-colors.html#cb400-18" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 38694     8</span></span></code></pre></div>
<p>As indicated by the output of the function <code>dim()</code> this is a very large matrix (38,694 rows). This is a huge table; however, most genes will have similar levels of expression across SRR samples. We will search for the ensgenes that have the most dispersion around the mean expression by estimating the coefficient of variation <a href="https://en.wikipedia.org/wiki/Coefficient_of_variation" target="_blank"><strong>CV</strong></a>. This statistic represents the ratio of the standard deviation to the mean. Then, we will extract those ensgenes with the 100 highest values.</p>
<div class="sourceCode" id="cb401"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb401-1"><a href="graphics-and-colors.html#cb401-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Create a function to valculate coefficient of variation </span></span>
<span id="cb401-2"><a href="graphics-and-colors.html#cb401-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb401-3"><a href="graphics-and-colors.html#cb401-3" aria-hidden="true" tabindex="-1"></a>co.var <span class="ot">&lt;-</span> <span class="cf">function</span>(x) ( <span class="dv">100</span><span class="sc">*</span><span class="fu">sd</span>(x)<span class="sc">/</span><span class="fu">mean</span>(x) )</span>
<span id="cb401-4"><a href="graphics-and-colors.html#cb401-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb401-5"><a href="graphics-and-colors.html#cb401-5" aria-hidden="true" tabindex="-1"></a><span class="do">##  we estimate row-wise coefficient of variation </span></span>
<span id="cb401-6"><a href="graphics-and-colors.html#cb401-6" aria-hidden="true" tabindex="-1"></a>airway_data_scaled2<span class="sc">$</span>coVar<span class="ot">&lt;-</span><span class="fu">apply</span>(airway_data_scaled2[,<span class="fu">names</span>(airway_data_scaled2)],<span class="dv">1</span>,co.var)</span>
<span id="cb401-7"><a href="graphics-and-colors.html#cb401-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data_scaled2)</span>
<span id="cb401-8"><a href="graphics-and-colors.html#cb401-8" aria-hidden="true" tabindex="-1"></a><span class="co">#                 SRR1039508   SRR1039509   SRR1039512   SRR1039513  SRR1039516   SRR1039517  SRR1039520   SRR1039521      coVar</span></span>
<span id="cb401-9"><a href="graphics-and-colors.html#cb401-9" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000003  0.03933056 -0.003924270  0.036555187  0.011021618  0.09101424  0.041361372  0.06313763  0.007896859  87.129966</span></span>
<span id="cb401-10"><a href="graphics-and-colors.html#cb401-10" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000005 -0.12176875 -0.130284028 -0.106887175 -0.112547703 -0.11339060 -0.123910392 -0.10822243 -0.116658738  -6.935013</span></span>
<span id="cb401-11"><a href="graphics-and-colors.html#cb401-11" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000419 -0.01771152  0.005695712 -0.009143265 -0.009526988 -0.01171230 -0.006246538 -0.01956592 -0.011693839 -77.383502</span></span>
<span id="cb401-12"><a href="graphics-and-colors.html#cb401-12" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000457 -0.04444999 -0.063204156 -0.049129410 -0.046736626 -0.05783441 -0.056566291 -0.03806261 -0.049844146 -15.968492</span></span>
<span id="cb401-13"><a href="graphics-and-colors.html#cb401-13" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000460 -0.10037797 -0.109224069 -0.095303887 -0.094220568 -0.09277541 -0.109748545 -0.08653667 -0.101398615  -8.191639</span></span>
<span id="cb401-14"><a href="graphics-and-colors.html#cb401-14" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000000938 -0.12176875 -0.130284028 -0.106728500 -0.112547703 -0.11304119 -0.123910392 -0.10822243 -0.116658738  -6.980658</span></span>
<span id="cb401-15"><a href="graphics-and-colors.html#cb401-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb401-16"><a href="graphics-and-colors.html#cb401-16" aria-hidden="true" tabindex="-1"></a><span class="do">## descresing order by coVar</span></span>
<span id="cb401-17"><a href="graphics-and-colors.html#cb401-17" aria-hidden="true" tabindex="-1"></a>airway_data_scaled2_order <span class="ot">&lt;-</span> airway_data_scaled2[<span class="fu">order</span>(airway_data_scaled2<span class="sc">$</span>coVar,  <span class="at">decreasing =</span> <span class="cn">TRUE</span>),]</span>
<span id="cb401-18"><a href="graphics-and-colors.html#cb401-18" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data_scaled2_order)</span>
<span id="cb401-19"><a href="graphics-and-colors.html#cb401-19" aria-hidden="true" tabindex="-1"></a><span class="co">#                   SRR1039508    SRR1039509   SRR1039512   SRR1039513   SRR1039516   SRR1039517   SRR1039520   SRR1039521    coVar</span></span>
<span id="cb401-20"><a href="graphics-and-colors.html#cb401-20" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000126709  0.0313090216 -0.0005442761 -0.006445787  0.049897360 -0.012760526 -0.030050493 -0.019778529 -0.011075185 39115.82</span></span>
<span id="cb401-21"><a href="graphics-and-colors.html#cb401-21" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000278270  0.1077364958  0.1148955034 -0.015648947 -0.035906955 -0.042285668 -0.052950501 -0.036574367 -0.037471074 30921.30</span></span>
<span id="cb401-22"><a href="graphics-and-colors.html#cb401-22" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000041802 -0.0043422784  0.0150556941 -0.007239162 -0.015080665  0.012746403  0.017406759 -0.008723041 -0.009425442 25810.60</span></span>
<span id="cb401-23"><a href="graphics-and-colors.html#cb401-23" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000086712 -0.0116953590  0.0098557040 -0.021678604 -0.008138569  0.004535268 -0.034570232  0.038475388  0.023981854 25111.84</span></span>
<span id="cb401-24"><a href="graphics-and-colors.html#cb401-24" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000188603 -0.0001086865  0.0116757006 -0.020567877  0.002413418 -0.009441131 -0.004739959 -0.004258325  0.025631597 18384.05</span></span>
<span id="cb401-25"><a href="graphics-and-colors.html#cb401-25" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000125398  0.0698569896 -0.0096442587  0.058928387 -0.045348207 -0.017652265 -0.093326829  0.090138534 -0.049637928 15830.54</span></span>
<span id="cb401-26"><a href="graphics-and-colors.html#cb401-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb401-27"><a href="graphics-and-colors.html#cb401-27" aria-hidden="true" tabindex="-1"></a><span class="do">## retain the first 100 genes</span></span>
<span id="cb401-28"><a href="graphics-and-colors.html#cb401-28" aria-hidden="true" tabindex="-1"></a>airway_data_scaled_red <span class="ot">&lt;-</span> airway_data_scaled2_order[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>,]</span>
<span id="cb401-29"><a href="graphics-and-colors.html#cb401-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb401-30"><a href="graphics-and-colors.html#cb401-30" aria-hidden="true" tabindex="-1"></a><span class="do">## remove sd column</span></span>
<span id="cb401-31"><a href="graphics-and-colors.html#cb401-31" aria-hidden="true" tabindex="-1"></a>airway_data_scaled_red <span class="ot">&lt;-</span> airway_data_scaled_red[,<span class="sc">-</span><span class="fu">length</span>(airway_data_scaled_red)]</span>
<span id="cb401-32"><a href="graphics-and-colors.html#cb401-32" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data_scaled_red)</span>
<span id="cb401-33"><a href="graphics-and-colors.html#cb401-33" aria-hidden="true" tabindex="-1"></a><span class="co">#                   SRR1039508    SRR1039509   SRR1039512   SRR1039513   SRR1039516   SRR1039517   SRR1039520   SRR1039521</span></span>
<span id="cb401-34"><a href="graphics-and-colors.html#cb401-34" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000126709  0.0313090216 -0.0005442761 -0.006445787  0.049897360 -0.012760526 -0.030050493 -0.019778529 -0.011075185</span></span>
<span id="cb401-35"><a href="graphics-and-colors.html#cb401-35" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000278270  0.1077364958  0.1148955034 -0.015648947 -0.035906955 -0.042285668 -0.052950501 -0.036574367 -0.037471074</span></span>
<span id="cb401-36"><a href="graphics-and-colors.html#cb401-36" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000041802 -0.0043422784  0.0150556941 -0.007239162 -0.015080665  0.012746403  0.017406759 -0.008723041 -0.009425442</span></span>
<span id="cb401-37"><a href="graphics-and-colors.html#cb401-37" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000086712 -0.0116953590  0.0098557040 -0.021678604 -0.008138569  0.004535268 -0.034570232  0.038475388  0.023981854</span></span>
<span id="cb401-38"><a href="graphics-and-colors.html#cb401-38" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000188603 -0.0001086865  0.0116757006 -0.020567877  0.002413418 -0.009441131 -0.004739959 -0.004258325  0.025631597</span></span>
<span id="cb401-39"><a href="graphics-and-colors.html#cb401-39" aria-hidden="true" tabindex="-1"></a><span class="co">#ENSG00000125398  0.0698569896 -0.0096442587  0.058928387 -0.045348207 -0.017652265 -0.093326829  0.090138534 -0.049637928</span></span>
<span id="cb401-40"><a href="graphics-and-colors.html#cb401-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb401-41"><a href="graphics-and-colors.html#cb401-41" aria-hidden="true" tabindex="-1"></a><span class="do">## get dimensions</span></span>
<span id="cb401-42"><a href="graphics-and-colors.html#cb401-42" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(airway_data_scaled_red)</span>
<span id="cb401-43"><a href="graphics-and-colors.html#cb401-43" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 100     8</span></span></code></pre></div>
<p>With this reduced matrix, we can use melt() to create a heatmap amenable data frame.</p>
<div class="sourceCode" id="cb402"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb402-1"><a href="graphics-and-colors.html#cb402-1" aria-hidden="true" tabindex="-1"></a>airway_data_for_heatmap <span class="ot">&lt;-</span> <span class="fu">melt</span>(<span class="fu">as.matrix</span>(airway_data_scaled_red))</span>
<span id="cb402-2"><a href="graphics-and-colors.html#cb402-2" aria-hidden="true" tabindex="-1"></a><span class="co"># transform factor variables to character</span></span>
<span id="cb402-3"><a href="graphics-and-colors.html#cb402-3" aria-hidden="true" tabindex="-1"></a>factor_columns <span class="ot">&lt;-</span> <span class="fu">sapply</span>(airway_data_for_heatmap, is.factor)</span>
<span id="cb402-4"><a href="graphics-and-colors.html#cb402-4" aria-hidden="true" tabindex="-1"></a>airway_data_for_heatmap[factor_columns] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(airway_data_for_heatmap[factor_columns], as.character)</span>
<span id="cb402-5"><a href="graphics-and-colors.html#cb402-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(airway_data_for_heatmap)</span>
<span id="cb402-6"><a href="graphics-and-colors.html#cb402-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  800 obs. of  3 variables:</span></span>
<span id="cb402-7"><a href="graphics-and-colors.html#cb402-7" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Var1 : chr  &quot;ENSG00000126709&quot; &quot;ENSG00000278270&quot; &quot;ENSG00000041802&quot; &quot;ENSG00000086712&quot; ...</span></span>
<span id="cb402-8"><a href="graphics-and-colors.html#cb402-8" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Var2 : chr  &quot;SRR1039508&quot; &quot;SRR1039508&quot; &quot;SRR1039508&quot; &quot;SRR1039508&quot; ...</span></span>
<span id="cb402-9"><a href="graphics-and-colors.html#cb402-9" aria-hidden="true" tabindex="-1"></a><span class="co"># $ value: num  0.031309 0.107736 -0.004342 -0.011695 -0.000109 ...</span></span>
<span id="cb402-10"><a href="graphics-and-colors.html#cb402-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb402-11"><a href="graphics-and-colors.html#cb402-11" aria-hidden="true" tabindex="-1"></a><span class="do">## we can rename the columns in this data frame</span></span>
<span id="cb402-12"><a href="graphics-and-colors.html#cb402-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(airway_data_for_heatmap) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;ensgene&quot;</span>, <span class="st">&quot;SRR&quot;</span>, <span class="st">&quot;expression&quot;</span>)</span>
<span id="cb402-13"><a href="graphics-and-colors.html#cb402-13" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(airway_data_for_heatmap)</span>
<span id="cb402-14"><a href="graphics-and-colors.html#cb402-14" aria-hidden="true" tabindex="-1"></a><span class="co">#          ensgene        SRR    expression</span></span>
<span id="cb402-15"><a href="graphics-and-colors.html#cb402-15" aria-hidden="true" tabindex="-1"></a><span class="co">#1 ENSG00000126709 SRR1039508  0.0313090216</span></span>
<span id="cb402-16"><a href="graphics-and-colors.html#cb402-16" aria-hidden="true" tabindex="-1"></a><span class="co">#2 ENSG00000278270 SRR1039508  0.1077364958</span></span>
<span id="cb402-17"><a href="graphics-and-colors.html#cb402-17" aria-hidden="true" tabindex="-1"></a><span class="co">#3 ENSG00000041802 SRR1039508 -0.0043422784</span></span>
<span id="cb402-18"><a href="graphics-and-colors.html#cb402-18" aria-hidden="true" tabindex="-1"></a><span class="co">#4 ENSG00000086712 SRR1039508 -0.0116953590</span></span>
<span id="cb402-19"><a href="graphics-and-colors.html#cb402-19" aria-hidden="true" tabindex="-1"></a><span class="co">#5 ENSG00000188603 SRR1039508 -0.0001086865</span></span>
<span id="cb402-20"><a href="graphics-and-colors.html#cb402-20" aria-hidden="true" tabindex="-1"></a><span class="co">#6 ENSG00000125398 SRR1039508  0.0698569896</span></span></code></pre></div>
</div>
<div id="color-palettes-online" class="section level2" number="9.4">
<h2><span class="header-section-number">9.4</span> Color palettes online</h2>
<p><strong>4)</strong> One of the “hardest” parts before or after doing graphics is choosing colors. Many colors can be called by name (e.g., <code>firebrick4</code>) and I have added a file named <code>Rcolor.pdf</code> in this class <a href="https://github.com/corytophanes/BIO_BIT_Bioinformatics_209/tree/main/ref_files_pdfs" target="_blank"><strong>GitHub repository</strong></a>. Any color also can be included by their hexadecimal color codes (e.g., <code>#005f3f</code>). These codes and the colors can be found in many online color palette websites like following:</p>
<ol style="list-style-type: lower-alpha">
<li><a href="https://www.color-hex.com/" target="_blank"><strong>https://www.color-hex.com</strong></a></li>
<li><a href="https://www.colourlovers.com/" target="_blank"><strong>https://www.colourlovers.com</strong></a></li>
<li><a href="https://colorbrewer2.org/" target="_blank"><strong>https://colorbrewer2.org</strong></a></li>
<li><a href="http://www.cookbook-r.com/Graphs/Colors_(ggplot2)/" target="_blank"><strong>http://www.cookbook-r.com/Graphs/Colors_(ggplot2)/</strong></a></li>
<li><a href="https://carto.com/carto-colors/" target="_blank"><strong>https://carto.com/carto-colors/</strong></a></li>
</ol>
<p>Thankfully, there are some R base color palettes (rainbow, heat.colors, cm.color). These are called as functions, e.g., <code>rainbow()</code> and it will return in hexadecimal color code the number of colors that you want.</p>
<div class="sourceCode" id="cb403"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb403-1"><a href="graphics-and-colors.html#cb403-1" aria-hidden="true" tabindex="-1"></a><span class="do">## get 5 colors of rainbow()</span></span>
<span id="cb403-2"><a href="graphics-and-colors.html#cb403-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rainbow</span>(<span class="dv">5</span>)</span>
<span id="cb403-3"><a href="graphics-and-colors.html#cb403-3" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;#FF0000FF&quot; &quot;#CCFF00FF&quot; &quot;#00FF66FF&quot; &quot;#0066FFFF&quot; &quot;#CC00FFFF&quot;</span></span></code></pre></div>
<p>Yet, there are several R-packages can also help you in choosing colors. I will list some of those that I found useful.</p>
</div>
<div id="ggplot2-default-palette" class="section level2" number="9.5">
<h2><span class="header-section-number">9.5</span> <code>ggplot2</code> default palette</h2>
<p><strong>5)</strong> The package <strong>ggplot2</strong> has its own base palette. These can be called when the function <code>ggplot()</code> is called.</p>
<p>We can visualize the default colors of <strong>ggplot2</strong> usin the package <a href="https://cran.r-project.org/web/packages/scales/index.html" target="_blank"><strong>scales</strong></a> that provide scale functions for visualization. Using function <code>hue_pal()</code> to generate a hue palette and <code>show_col()</code> to generate a plot to show colours.</p>
<div class="sourceCode" id="cb404"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb404-1"><a href="graphics-and-colors.html#cb404-1" aria-hidden="true" tabindex="-1"></a><span class="do">## insall and load &#39;scales&#39; R package -- you will get an error but do worry</span></span>
<span id="cb404-2"><a href="graphics-and-colors.html#cb404-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;scales&quot;</span>)                              </span>
<span id="cb404-3"><a href="graphics-and-colors.html#cb404-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb404-4"><a href="graphics-and-colors.html#cb404-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Error in value[[3L]](cond) : </span></span>
<span id="cb404-5"><a href="graphics-and-colors.html#cb404-5" aria-hidden="true" tabindex="-1"></a><span class="co">#  Package ‘scales’ version 1.1.0 cannot be unloaded:</span></span>
<span id="cb404-6"><a href="graphics-and-colors.html#cb404-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Error in unloadNamespace(package) : namespace ‘scales’ is imported by ‘ggplot2’, ‘ggsci’ so cannot be unloaded</span></span>
<span id="cb404-7"><a href="graphics-and-colors.html#cb404-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb404-8"><a href="graphics-and-colors.html#cb404-8" aria-hidden="true" tabindex="-1"></a><span class="do">## load ggplot2</span></span>
<span id="cb404-9"><a href="graphics-and-colors.html#cb404-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb404-10"><a href="graphics-and-colors.html#cb404-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb404-11"><a href="graphics-and-colors.html#cb404-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Identify hex codes in ggplot2 defult for 10 colors</span></span>
<span id="cb404-12"><a href="graphics-and-colors.html#cb404-12" aria-hidden="true" tabindex="-1"></a>hex_codes1 <span class="ot">&lt;-</span> scales<span class="sc">::</span><span class="fu">hue_pal</span>()(<span class="dv">10</span>)                             </span>
<span id="cb404-13"><a href="graphics-and-colors.html#cb404-13" aria-hidden="true" tabindex="-1"></a>hex_codes1                                              </span>
<span id="cb404-14"><a href="graphics-and-colors.html#cb404-14" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;#F8766D&quot; &quot;#D89000&quot; &quot;#A3A500&quot; &quot;#39B600&quot; &quot;#00BF7D&quot; &quot;#00BFC4&quot; &quot;#00B0F6&quot; &quot;#9590FF&quot; &quot;#E76BF3&quot; &quot;#FF62BC&quot;</span></span>
<span id="cb404-15"><a href="graphics-and-colors.html#cb404-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb404-16"><a href="graphics-and-colors.html#cb404-16" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot hex codes/colors</span></span>
<span id="cb404-17"><a href="graphics-and-colors.html#cb404-17" aria-hidden="true" tabindex="-1"></a>scales<span class="sc">::</span><span class="fu">show_col</span>(hex_codes1)</span></code></pre></div>
<p><img src="figures/session_9_figs/32_show_color.png" width="70%" /></p>
</div>
<div id="heatmaps-or-tiled-charts" class="section level2" number="9.6">
<h2><span class="header-section-number">9.6</span> Heatmaps or tiled charts</h2>
<p><strong>6)</strong> I will illustrate the diversity of colors using tile plots. These are color matrices (x and y coordinates) that can be plot and they are also heatmaps (tiled charts) where each tile is a continuous value that can be trnaslated to a color scale. To get these plots, we use the function <code>geom_tile()</code>.</p>
<p>We make sure that we have a dataset <code>bacteria</code> from package <code>ade4</code> is in the correc format and ready to use for color exploration.</p>
<div class="sourceCode" id="cb405"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb405-1"><a href="graphics-and-colors.html#cb405-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we will load packages </span></span>
<span id="cb405-2"><a href="graphics-and-colors.html#cb405-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb405-3"><a href="graphics-and-colors.html#cb405-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb405-4"><a href="graphics-and-colors.html#cb405-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb405-5"><a href="graphics-and-colors.html#cb405-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb405-6"><a href="graphics-and-colors.html#cb405-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb405-7"><a href="graphics-and-colors.html#cb405-7" aria-hidden="true" tabindex="-1"></a><span class="do">## load data and assign to data frame</span></span>
<span id="cb405-8"><a href="graphics-and-colors.html#cb405-8" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(bacteria,  <span class="at">package =</span> <span class="st">&quot;ade4&quot;</span>)</span>
<span id="cb405-9"><a href="graphics-and-colors.html#cb405-9" aria-hidden="true" tabindex="-1"></a>bacteria_espaa <span class="ot">&lt;-</span> bacteria<span class="sc">$</span>espaa</span>
<span id="cb405-10"><a href="graphics-and-colors.html#cb405-10" aria-hidden="true" tabindex="-1"></a>bacteria_espaa_scaled <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">modify</span>(bacteria_espaa, scale)</span>
<span id="cb405-11"><a href="graphics-and-colors.html#cb405-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb405-12"><a href="graphics-and-colors.html#cb405-12" aria-hidden="true" tabindex="-1"></a><span class="do">## reshape dataframe</span></span>
<span id="cb405-13"><a href="graphics-and-colors.html#cb405-13" aria-hidden="true" tabindex="-1"></a>bacteria_espaa_for_heatmap <span class="ot">&lt;-</span> <span class="fu">melt</span>(<span class="fu">as.matrix</span>(bacteria_espaa_scaled))</span>
<span id="cb405-14"><a href="graphics-and-colors.html#cb405-14" aria-hidden="true" tabindex="-1"></a>factor_columns <span class="ot">&lt;-</span> <span class="fu">sapply</span>(bacteria_espaa_for_heatmap, is.factor)</span>
<span id="cb405-15"><a href="graphics-and-colors.html#cb405-15" aria-hidden="true" tabindex="-1"></a>bacteria_espaa_for_heatmap[factor_columns] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(bacteria_espaa_for_heatmap[factor_columns], as.character)</span>
<span id="cb405-16"><a href="graphics-and-colors.html#cb405-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb405-17"><a href="graphics-and-colors.html#cb405-17" aria-hidden="true" tabindex="-1"></a><span class="do">## we can rename the columns in this data frame</span></span>
<span id="cb405-18"><a href="graphics-and-colors.html#cb405-18" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(bacteria_espaa_for_heatmap) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;bacteria_sp&quot;</span>, <span class="st">&quot;Amino_acid&quot;</span>, <span class="st">&quot;scaled_value&quot;</span>)</span>
<span id="cb405-19"><a href="graphics-and-colors.html#cb405-19" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(bacteria_espaa_for_heatmap)</span>
<span id="cb405-20"><a href="graphics-and-colors.html#cb405-20" aria-hidden="true" tabindex="-1"></a><span class="co">#  bacteria_sp Amino_acid scaled_value</span></span>
<span id="cb405-21"><a href="graphics-and-colors.html#cb405-21" aria-hidden="true" tabindex="-1"></a><span class="co">#1     AERPECG        Ala    0.3285668</span></span>
<span id="cb405-22"><a href="graphics-and-colors.html#cb405-22" aria-hidden="true" tabindex="-1"></a><span class="co">#2     AQUAECG        Ala   -0.4008694</span></span>
<span id="cb405-23"><a href="graphics-and-colors.html#cb405-23" aria-hidden="true" tabindex="-1"></a><span class="co">#3     ARCFUCG        Ala    0.1073331</span></span>
<span id="cb405-24"><a href="graphics-and-colors.html#cb405-24" aria-hidden="true" tabindex="-1"></a><span class="co">#4     BACHDCG        Ala    0.9012986</span></span>
<span id="cb405-25"><a href="graphics-and-colors.html#cb405-25" aria-hidden="true" tabindex="-1"></a><span class="co">#5     BACSUCG        Ala    1.0314722</span></span>
<span id="cb405-26"><a href="graphics-and-colors.html#cb405-26" aria-hidden="true" tabindex="-1"></a><span class="co">#6     BORBUCG        Ala   -0.7597170</span></span></code></pre></div>
<p>We can explore some of the colors in <strong>ggplot2</strong>.</p>
<div class="sourceCode" id="cb406"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb406-1"><a href="graphics-and-colors.html#cb406-1" aria-hidden="true" tabindex="-1"></a><span class="do">## checking if ggplot2 is loaded</span></span>
<span id="cb406-2"><a href="graphics-and-colors.html#cb406-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggplot2)</span>
<span id="cb406-3"><a href="graphics-and-colors.html#cb406-3" aria-hidden="true" tabindex="-1"></a><span class="do">## we can plot a heatmap with the expression values</span></span>
<span id="cb406-4"><a href="graphics-and-colors.html#cb406-4" aria-hidden="true" tabindex="-1"></a>heatmap_bacteria_base <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(bacteria_espaa_for_heatmap,<span class="fu">aes</span>(<span class="at">x=</span>Amino_acid,<span class="at">y=</span>bacteria_sp,<span class="at">fill=</span>scaled_value))<span class="sc">+</span></span>
<span id="cb406-5"><a href="graphics-and-colors.html#cb406-5" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">geom_tile</span>()<span class="sc">+</span></span>
<span id="cb406-6"><a href="graphics-and-colors.html#cb406-6" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb406-7"><a href="graphics-and-colors.html#cb406-7" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">ggtitle</span>(<span class="st">&quot;My First Heatmap -- Base Color&quot;</span>) <span class="sc">+</span></span>
<span id="cb406-8"><a href="graphics-and-colors.html#cb406-8" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">0.5</span>)),</span>
<span id="cb406-9"><a href="graphics-and-colors.html#cb406-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">0.7</span>),<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust=</span><span class="dv">1</span>),</span>
<span id="cb406-10"><a href="graphics-and-colors.html#cb406-10" aria-hidden="true" tabindex="-1"></a>                          <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">7</span>),</span>
<span id="cb406-11"><a href="graphics-and-colors.html#cb406-11" aria-hidden="true" tabindex="-1"></a>                           <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">7</span>))</span>
<span id="cb406-12"><a href="graphics-and-colors.html#cb406-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb406-13"><a href="graphics-and-colors.html#cb406-13" aria-hidden="true" tabindex="-1"></a>heatmap_bacteria_base</span></code></pre></div>
<p>Here is the base colors for this tile plot.</p>
<p><img src="figures/session_9_figs/33_tileplot_base.png" width="100%" /></p>
</div>
<div id="r-packages-about-color" class="section level2" number="9.7">
<h2><span class="header-section-number">9.7</span> R-packages about color</h2>
<div id="viridis-package" class="section level3" number="9.7.1">
<h3><span class="header-section-number">9.7.1</span> <code>viridis</code> package</h3>
<p><strong>6)</strong> The package <a href="https://cran.r-project.org/web/packages/viridis/index.html" target="_blank"><strong>viridis</strong></a> can create color scales including palettes for those persons with colorblindness. This package has a <a href="https://cran.r-project.org/web/packages/viridis/vignettes/intro-to-viridis.html" target="_blank"><strong>vignette</strong></a> that help you with more examples.</p>
<div class="sourceCode" id="cb407"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb407-1"><a href="graphics-and-colors.html#cb407-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We need to install R-package &#39;viridis&#39; </span></span>
<span id="cb407-2"><a href="graphics-and-colors.html#cb407-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;viridis&quot;</span>)</span>
<span id="cb407-3"><a href="graphics-and-colors.html#cb407-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span>
<span id="cb407-4"><a href="graphics-and-colors.html#cb407-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Loading required package: viridisLite</span></span></code></pre></div>
<div class="sourceCode" id="cb408"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb408-1"><a href="graphics-and-colors.html#cb408-1" aria-hidden="true" tabindex="-1"></a><span class="do">## checking if ggplot2 is loaded</span></span>
<span id="cb408-2"><a href="graphics-and-colors.html#cb408-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggplot2)</span>
<span id="cb408-3"><a href="graphics-and-colors.html#cb408-3" aria-hidden="true" tabindex="-1"></a><span class="do">## we can plot a heatmap with the expression values</span></span>
<span id="cb408-4"><a href="graphics-and-colors.html#cb408-4" aria-hidden="true" tabindex="-1"></a>heatmap_viridis_bac_magma <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(bacteria_espaa_for_heatmap,<span class="fu">aes</span>(<span class="at">x=</span>Amino_acid,<span class="at">y=</span>bacteria_sp,<span class="at">fill=</span>scaled_value))<span class="sc">+</span></span>
<span id="cb408-5"><a href="graphics-and-colors.html#cb408-5" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">geom_tile</span>()<span class="sc">+</span></span>
<span id="cb408-6"><a href="graphics-and-colors.html#cb408-6" aria-hidden="true" tabindex="-1"></a>          <span class="fu">scale_fill_viridis</span>(<span class="at">option=</span><span class="st">&quot;A&quot;</span>)<span class="sc">+</span> </span>
<span id="cb408-7"><a href="graphics-and-colors.html#cb408-7" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb408-8"><a href="graphics-and-colors.html#cb408-8" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">0.5</span>)),</span>
<span id="cb408-9"><a href="graphics-and-colors.html#cb408-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">0.7</span>),<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust=</span><span class="dv">1</span>))</span>
<span id="cb408-10"><a href="graphics-and-colors.html#cb408-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb408-11"><a href="graphics-and-colors.html#cb408-11" aria-hidden="true" tabindex="-1"></a>heatmap_viridis_bac_magma</span>
<span id="cb408-12"><a href="graphics-and-colors.html#cb408-12" aria-hidden="true" tabindex="-1"></a><span class="do">## </span><span class="al">NOTE</span><span class="do">: you might need to run the plot &#39;heatmap_viridis_bac_magma&#39;, if it gives you an error</span></span></code></pre></div>
<p><img src="figures/session_9_figs/1_bacteria_heatmap.png" width="100%" /></p>
<p>We can also run this color example with <code>airway_data_for_heatmap</code>.</p>
<div class="sourceCode" id="cb409"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb409-1"><a href="graphics-and-colors.html#cb409-1" aria-hidden="true" tabindex="-1"></a><span class="do">## checking if ggplot2 is loaded</span></span>
<span id="cb409-2"><a href="graphics-and-colors.html#cb409-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggplot2)</span>
<span id="cb409-3"><a href="graphics-and-colors.html#cb409-3" aria-hidden="true" tabindex="-1"></a><span class="do">## we can plot a heatmap with the expression values</span></span>
<span id="cb409-4"><a href="graphics-and-colors.html#cb409-4" aria-hidden="true" tabindex="-1"></a>heatmap_viridis <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(airway_data_for_heatmap,<span class="fu">aes</span>(<span class="at">x=</span>SRR,<span class="at">y=</span>ensgene,<span class="at">fill=</span>expression))<span class="sc">+</span></span>
<span id="cb409-5"><a href="graphics-and-colors.html#cb409-5" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">geom_tile</span>()<span class="sc">+</span></span>
<span id="cb409-6"><a href="graphics-and-colors.html#cb409-6" aria-hidden="true" tabindex="-1"></a>          <span class="fu">scale_fill_viridis</span>()<span class="sc">+</span> </span>
<span id="cb409-7"><a href="graphics-and-colors.html#cb409-7" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb409-8"><a href="graphics-and-colors.html#cb409-8" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">theme</span>(<span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">0.5</span>)),</span>
<span id="cb409-9"><a href="graphics-and-colors.html#cb409-9" aria-hidden="true" tabindex="-1"></a>                          <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="fu">rel</span>(<span class="fl">0.7</span>),<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>, <span class="at">hjust=</span><span class="dv">1</span>))</span>
<span id="cb409-10"><a href="graphics-and-colors.html#cb409-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb409-11"><a href="graphics-and-colors.html#cb409-11" aria-hidden="true" tabindex="-1"></a><span class="do">## </span><span class="al">NOTE</span><span class="do">: this will require time to load (if it is huge) -- you might not want to run this plot</span></span>
<span id="cb409-12"><a href="graphics-and-colors.html#cb409-12" aria-hidden="true" tabindex="-1"></a>heatmap_viridis</span></code></pre></div>
<p><img src="figures/session_9_figs/2_ensgene_heatmap.png" width="100%" /></p>
</div>
<div id="rcolorbrewer-package" class="section level3" number="9.7.2">
<h3><span class="header-section-number">9.7.2</span> <code>RColorBrewer</code> package</h3>
<p><strong>7)</strong> The package <a href="https://cran.r-project.org/web/packages/RColorBrewer/index.html" target="_blank"><strong>RColorBrewer</strong></a>. This package is a collection of color palettes that can be easily incorporated in plots.</p>
<div class="sourceCode" id="cb410"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb410-1"><a href="graphics-and-colors.html#cb410-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We need to install R-package &#39;RColorBrewer&#39; </span></span>
<span id="cb410-2"><a href="graphics-and-colors.html#cb410-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;RColorBrewer&quot;</span>)</span>
<span id="cb410-3"><a href="graphics-and-colors.html#cb410-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span></code></pre></div>
<p>We can display available color palettes. Notice that names on the row can be call when using ggplot2.</p>
<div class="sourceCode" id="cb411"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb411-1"><a href="graphics-and-colors.html#cb411-1" aria-hidden="true" tabindex="-1"></a><span class="fu">display.brewer.all</span>()</span></code></pre></div>
<p><img src="figures/session_9_figs/3_colorbrewer_all_palettes.png" width="100%" /></p>
<p>This image provides three types of palettes: diverging, qualitative, and sequential color. The top group are colors for gradient <strong>quantitative</strong> and <strong>sequential</strong> data that progresses from low to higher values (e.g., temperature measurements). The middle group are colors for <strong>discrete</strong> or <strong>categorical</strong> data that do not imply magnitude differences between values or states in the data (e.g., types of fruits). The lower group are colors for <strong>quantitative</strong> and <strong>diverting</strong> data that increases in magnitude toward opposites or extremes (e.g., above and below sea level).</p>
<p><strong>8)</strong> The package [<strong>RColorBrewer</strong>] also provides colorblind-friendly palettes and these can be call with the argument <code>colorblindFriendly = TRUE</code>.</p>
<div class="sourceCode" id="cb412"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb412-1"><a href="graphics-and-colors.html#cb412-1" aria-hidden="true" tabindex="-1"></a><span class="fu">display.brewer.all</span>(<span class="at">colorblindFriendly =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="figures/session_9_figs/4_colorbrewer_all_colorblind.png" width="100%" /></p>
<p><strong>9)</strong> For most practical uses, you will need only few colors based on your requirements and <strong>RColorBrewer</strong> allows to select specific palettes and the number of colors that you want.</p>
<div class="sourceCode" id="cb413"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb413-1"><a href="graphics-and-colors.html#cb413-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we will load &#39;ade4&#39; and the ‘ichtyo’ dataset</span></span>
<span id="cb413-2"><a href="graphics-and-colors.html#cb413-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb413-3"><a href="graphics-and-colors.html#cb413-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(ichtyo)</span>
<span id="cb413-4"><a href="graphics-and-colors.html#cb413-4" aria-hidden="true" tabindex="-1"></a>ichtyo_data <span class="ot">&lt;-</span> ichtyo<span class="sc">$</span>tab</span>
<span id="cb413-5"><a href="graphics-and-colors.html#cb413-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ichtyo_data)</span>
<span id="cb413-6"><a href="graphics-and-colors.html#cb413-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  HOT VAN CHE SPI GOU BAF GAR ABL PER</span></span>
<span id="cb413-7"><a href="graphics-and-colors.html#cb413-7" aria-hidden="true" tabindex="-1"></a><span class="co">#1  57  47  60  11  27  17  10   9  21</span></span>
<span id="cb413-8"><a href="graphics-and-colors.html#cb413-8" aria-hidden="true" tabindex="-1"></a><span class="co">#2   9  25  19   6   2   5   0   4   6</span></span>
<span id="cb413-9"><a href="graphics-and-colors.html#cb413-9" aria-hidden="true" tabindex="-1"></a><span class="co">#3  48  60  52  16  25  24  17   8  12</span></span>
<span id="cb413-10"><a href="graphics-and-colors.html#cb413-10" aria-hidden="true" tabindex="-1"></a><span class="co">#4  26  50  32   8   6  11   2  10   5</span></span>
<span id="cb413-11"><a href="graphics-and-colors.html#cb413-11" aria-hidden="true" tabindex="-1"></a><span class="co">#5  26  43  31  14   9  18  13  14   2</span></span>
<span id="cb413-12"><a href="graphics-and-colors.html#cb413-12" aria-hidden="true" tabindex="-1"></a><span class="co">#6  62  43  47  16  11   9  15  16   3</span></span>
<span id="cb413-13"><a href="graphics-and-colors.html#cb413-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb413-14"><a href="graphics-and-colors.html#cb413-14" aria-hidden="true" tabindex="-1"></a><span class="do">## we use the function &#39;melt()&#39;&#39; of the package &#39;reshape2&#39; to arrange on the &#39;ggplot2&#39; format</span></span>
<span id="cb413-15"><a href="graphics-and-colors.html#cb413-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb413-16"><a href="graphics-and-colors.html#cb413-16" aria-hidden="true" tabindex="-1"></a>ichtyo_data_ggplot <span class="ot">&lt;-</span> <span class="fu">melt</span>(ichtyo_data)</span>
<span id="cb413-17"><a href="graphics-and-colors.html#cb413-17" aria-hidden="true" tabindex="-1"></a><span class="co">#No id variables; using all as measure variables</span></span>
<span id="cb413-18"><a href="graphics-and-colors.html#cb413-18" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(ichtyo_data_ggplot) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;faunistic_array&quot;</span>, <span class="st">&quot;counts&quot;</span>)</span>
<span id="cb413-19"><a href="graphics-and-colors.html#cb413-19" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(ichtyo_data_ggplot)</span>
<span id="cb413-20"><a href="graphics-and-colors.html#cb413-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  288 obs. of  2 variables:</span></span>
<span id="cb413-21"><a href="graphics-and-colors.html#cb413-21" aria-hidden="true" tabindex="-1"></a><span class="co"># $ faunistic_array: Factor w/ 9 levels &quot;HOT&quot;,&quot;VAN&quot;,&quot;CHE&quot;,..: 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb413-22"><a href="graphics-and-colors.html#cb413-22" aria-hidden="true" tabindex="-1"></a><span class="co"># $ counts         : num  57 9 48 26 26 62 17 18 9 12 ...</span></span>
<span id="cb413-23"><a href="graphics-and-colors.html#cb413-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb413-24"><a href="graphics-and-colors.html#cb413-24" aria-hidden="true" tabindex="-1"></a><span class="do">## View a single RColorBrewer palette by specifying its name (e.g., &#39;Paired&#39;) and 9 colors for each faunistic_array</span></span>
<span id="cb413-25"><a href="graphics-and-colors.html#cb413-25" aria-hidden="true" tabindex="-1"></a><span class="fu">display.brewer.pal</span>(<span class="at">n =</span> <span class="dv">9</span>, <span class="at">name =</span> <span class="st">&#39;Paired&#39;</span>)</span>
<span id="cb413-26"><a href="graphics-and-colors.html#cb413-26" aria-hidden="true" tabindex="-1"></a><span class="co"># you can also request the hexadecimal color code</span></span>
<span id="cb413-27"><a href="graphics-and-colors.html#cb413-27" aria-hidden="true" tabindex="-1"></a><span class="fu">brewer.pal</span>(<span class="at">n =</span> <span class="dv">9</span>, <span class="at">name =</span> <span class="st">&#39;Paired&#39;</span>)</span>
<span id="cb413-28"><a href="graphics-and-colors.html#cb413-28" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;#A6CEE3&quot; &quot;#1F78B4&quot; &quot;#B2DF8A&quot; &quot;#33A02C&quot; &quot;#FB9A99&quot; &quot;#E31A1C&quot; &quot;#FDBF6F&quot; &quot;#FF7F00&quot; &quot;#CAB2D6&quot;</span></span>
<span id="cb413-29"><a href="graphics-and-colors.html#cb413-29" aria-hidden="true" tabindex="-1"></a>my_paired_palette <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="at">n =</span> <span class="dv">9</span>, <span class="at">name =</span> <span class="st">&#39;Paired&#39;</span>)</span></code></pre></div>
<p><img src="figures/session_9_figs/5_colorbrewer_paired_9colors.png" width="100%" />
<strong>10)</strong> To implement the color palette selected, we then can call if from the ggplot2 function <code>ggplot()</code>. However, <strong>ggplot2</strong> has an standard palette so we will start with that for comparison.</p>
<p>This is the boxplot with <strong>ggplot2</strong> with its default colors.</p>
<div class="sourceCode" id="cb414"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb414-1"><a href="graphics-and-colors.html#cb414-1" aria-hidden="true" tabindex="-1"></a><span class="do">## check that ggplot2 has been loaded</span></span>
<span id="cb414-2"><a href="graphics-and-colors.html#cb414-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb414-3"><a href="graphics-and-colors.html#cb414-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Box plot with counts on y-axis and faunistic_array on x-axis and color by faunistic_array</span></span>
<span id="cb414-4"><a href="graphics-and-colors.html#cb414-4" aria-hidden="true" tabindex="-1"></a>my_ichtyo_boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> ichtyo_data_ggplot, <span class="at">mapping =</span> <span class="fu">aes</span>(faunistic_array, counts)) <span class="sc">+</span> </span>
<span id="cb414-5"><a href="graphics-and-colors.html#cb414-5" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> faunistic_array)) <span class="sc">+</span></span>
<span id="cb414-6"><a href="graphics-and-colors.html#cb414-6" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb414-7"><a href="graphics-and-colors.html#cb414-7" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb414-8"><a href="graphics-and-colors.html#cb414-8" aria-hidden="true" tabindex="-1"></a>my_ichtyo_boxplot</span></code></pre></div>
<p><img src="figures/session_9_figs/6_colorbrew_paired_ggplot2_std.png" width="100%" /></p>
<p>This is the boxplot with <strong>ggplot2</strong> with our RColorBrewer <code>Paired</code> color palette using the function <code>scale_fill_brewer()</code>.</p>
<div class="sourceCode" id="cb415"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb415-1"><a href="graphics-and-colors.html#cb415-1" aria-hidden="true" tabindex="-1"></a>my_ichtyo_boxplot2 <span class="ot">&lt;-</span> my_ichtyo_boxplot <span class="sc">+</span> <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">&#39;Paired&#39;</span>)</span>
<span id="cb415-2"><a href="graphics-and-colors.html#cb415-2" aria-hidden="true" tabindex="-1"></a>my_ichtyo_boxplot2</span></code></pre></div>
<p><img src="figures/session_9_figs/7_colorbrew_paired_ggplot2_Paired.png" width="100%" /></p>
<p>This is a scatter plot with <strong>ggplot2</strong> with its default colors.</p>
<div class="sourceCode" id="cb416"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb416-1"><a href="graphics-and-colors.html#cb416-1" aria-hidden="true" tabindex="-1"></a><span class="do">## check that ggplot2 has been loaded</span></span>
<span id="cb416-2"><a href="graphics-and-colors.html#cb416-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb416-3"><a href="graphics-and-colors.html#cb416-3" aria-hidden="true" tabindex="-1"></a><span class="do">## Scatter plot with counts on y-axis and faunistic_array on x-axis and color by faunistic_array</span></span>
<span id="cb416-4"><a href="graphics-and-colors.html#cb416-4" aria-hidden="true" tabindex="-1"></a>my_ichtyo_scatter <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> ichtyo_data_ggplot, <span class="at">mapping =</span> <span class="fu">aes</span>(faunistic_array, counts)) <span class="sc">+</span> </span>
<span id="cb416-5"><a href="graphics-and-colors.html#cb416-5" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">geom_violin</span>() <span class="sc">+</span></span>
<span id="cb416-6"><a href="graphics-and-colors.html#cb416-6" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> faunistic_array)) <span class="sc">+</span></span>
<span id="cb416-7"><a href="graphics-and-colors.html#cb416-7" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb416-8"><a href="graphics-and-colors.html#cb416-8" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb416-9"><a href="graphics-and-colors.html#cb416-9" aria-hidden="true" tabindex="-1"></a>my_ichtyo_scatter</span>
<span id="cb416-10"><a href="graphics-and-colors.html#cb416-10" aria-hidden="true" tabindex="-1"></a><span class="do">## Notice the violin shapes under the scatter plot.</span></span>
<span id="cb416-11"><a href="graphics-and-colors.html#cb416-11" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(my_ichtyo_scatter)</span>
<span id="cb416-12"><a href="graphics-and-colors.html#cb416-12" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;gg&quot;     &quot;ggplot&quot;</span></span></code></pre></div>
<p><img src="figures/session_9_figs/8_colorbrew_scatter_std.png" width="100%" /></p>
<p>This is the scatter plot with <strong>ggplot2</strong> with 9 colors form RColorBrewer <code>Paired</code> and using <code>scale_colour_manual()</code>.</p>
<div class="sourceCode" id="cb417"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb417-1"><a href="graphics-and-colors.html#cb417-1" aria-hidden="true" tabindex="-1"></a>my_paired_palette <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="at">n =</span> <span class="dv">9</span>, <span class="at">name =</span> <span class="st">&#39;Paired&#39;</span>)</span>
<span id="cb417-2"><a href="graphics-and-colors.html#cb417-2" aria-hidden="true" tabindex="-1"></a>my_paired_palette</span>
<span id="cb417-3"><a href="graphics-and-colors.html#cb417-3" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;#A6CEE3&quot; &quot;#1F78B4&quot; &quot;#B2DF8A&quot; &quot;#33A02C&quot; &quot;#FB9A99&quot; &quot;#E31A1C&quot; &quot;#FDBF6F&quot; &quot;#FF7F00&quot; &quot;#CAB2D6&quot;</span></span>
<span id="cb417-4"><a href="graphics-and-colors.html#cb417-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb417-5"><a href="graphics-and-colors.html#cb417-5" aria-hidden="true" tabindex="-1"></a>my_ichtyo_scatter2 <span class="ot">&lt;-</span> my_ichtyo_scatter <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> my_paired_palette)</span>
<span id="cb417-6"><a href="graphics-and-colors.html#cb417-6" aria-hidden="true" tabindex="-1"></a>my_ichtyo_scatter2</span>
<span id="cb417-7"><a href="graphics-and-colors.html#cb417-7" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(my_ichtyo_scatter2)</span>
<span id="cb417-8"><a href="graphics-and-colors.html#cb417-8" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;gg&quot;     &quot;ggplot&quot;</span></span></code></pre></div>
<p><img src="figures/session_9_figs/9_colorbrew_scatter_Paired.png" width="100%" />
<strong>Notice</strong> how easy is to change colors from existing <strong>ggplot</strong> object, in this case <code>my_ichtyo_scatter</code>. We will explore this further as we progress on this session.</p>
</div>
<div id="ggsci-package" class="section level3" number="9.7.3">
<h3><span class="header-section-number">9.7.3</span> <code>ggsci</code> package</h3>
<p><strong>11)</strong> The package <a href="https://cran.r-project.org/web/packages/ggsci/index.html" target="_blank"><strong>ggsci</strong></a> is general collection of scientific journal and sci-fi themed color palettes for <strong>ggplot2</strong>.</p>
<div class="sourceCode" id="cb418"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb418-1"><a href="graphics-and-colors.html#cb418-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We need to install R-package &#39;ggsci&#39; </span></span>
<span id="cb418-2"><a href="graphics-and-colors.html#cb418-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggsci&quot;</span>)</span>
<span id="cb418-3"><a href="graphics-and-colors.html#cb418-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggsci)</span></code></pre></div>
<p>The <strong>ggsci</strong> has its own <a href="https://cran.r-project.org/web/packages/ggsci/vignettes/ggsci.html" target="_blank"><strong>vignette</strong></a>. The palettes in this package are diverse and here is a list from their vignette.</p>
<table>
<colgroup>
<col width="8%" />
<col width="18%" />
<col width="36%" />
<col width="36%" />
</colgroup>
<thead>
<tr class="header">
<th>Name</th>
<th>Scales</th>
<th>Palette Types</th>
<th>Palette Generator</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>NPG</td>
<td>scale_color_npg() scale_fill_npg()</td>
<td>“nrc”</td>
<td>pal_npg()</td>
</tr>
<tr class="even">
<td>AAAS</td>
<td>scale_color_aaas() scale_fill_aaas()</td>
<td>“default”</td>
<td>pal_aaas()</td>
</tr>
<tr class="odd">
<td>NEJM</td>
<td>scale_color_nejm() scale_fill_nejm()</td>
<td>“default”</td>
<td>pal_nejm()</td>
</tr>
<tr class="even">
<td>Lancet</td>
<td>scale_color_lancet() scale_fill_lancet()</td>
<td>“lanonc”</td>
<td>pal_lancet()</td>
</tr>
<tr class="odd">
<td>JAMA</td>
<td>scale_color_jama() scale_fill_jama()</td>
<td>“default”</td>
<td>pal_jama()</td>
</tr>
<tr class="even">
<td>JCO</td>
<td>scale_color_jco() scale_fill_jco()</td>
<td>“default”</td>
<td>pal_jco()</td>
</tr>
<tr class="odd">
<td>UCSCGB</td>
<td>scale_color_ucscgb() scale_fill_ucscgb()</td>
<td>“default”</td>
<td>pal_ucscgb()</td>
</tr>
<tr class="even">
<td>D3</td>
<td>scale_color_d3() scale_fill_d3()</td>
<td>“category10” “category20” “category20b” “category20c”</td>
<td>pal_d3()</td>
</tr>
<tr class="odd">
<td>LocusZoom</td>
<td>scale_color_locuszoom() scale_fill_locuszoom()</td>
<td>“default”</td>
<td>pal_locuszoom()</td>
</tr>
<tr class="even">
<td>IGV</td>
<td>scale_color_igv() scale_fill_igv()</td>
<td>“default”“alternating”</td>
<td>pal_igv()</td>
</tr>
<tr class="odd">
<td>UChicago</td>
<td>scale_color_uchicago() scale_fill_uchicago()</td>
<td>“default”“light” “dark”</td>
<td>pal_uchicago()</td>
</tr>
<tr class="even">
<td>Star Trek</td>
<td>scale_color_startrek() scale_fill_startrek()</td>
<td>“uniform”</td>
<td>pal_startrek()</td>
</tr>
<tr class="odd">
<td>Tron Legacy</td>
<td>scale_color_tron() scale_fill_tron()</td>
<td>“legacy”</td>
<td>pal_tron()</td>
</tr>
<tr class="even">
<td>Futurama</td>
<td>scale_color_futurama() scale_fill_futurama()</td>
<td>“planetexpress”</td>
<td>pal_futurama()</td>
</tr>
<tr class="odd">
<td>Rick and Morty</td>
<td>scale_color_rickandmorty() scale_fill_rickandmorty()</td>
<td>“schwifty”</td>
<td>pal_rickandmorty()</td>
</tr>
<tr class="even">
<td>The Simpsons</td>
<td>scale_color_simpsons() scale_fill_simpsons()</td>
<td>“springfield”</td>
<td>pal_simpsons()</td>
</tr>
<tr class="odd">
<td>GSEA</td>
<td>scale_color_gsea() scale_fill_gsea()</td>
<td>“default”</td>
<td>pal_gsea()</td>
</tr>
<tr class="even">
<td>Material Design</td>
<td>scale_color_material() scale_fill_material()</td>
<td>“red” “pink” “purple” “deep-purple” “indigo” “blue” “light-blue” “cyan” “teal” “green” “light-green” “lime” “yellow” “amber” “orange” “deep-orange” “brown” “grey” “blue-grey”</td>
<td>pal_material()</td>
</tr>
</tbody>
</table>
<p><strong>12)</strong> You apply this palettes as we did in <strong>10)</strong> using <code>scale_color_XXX()</code> or <code>scale_fill_XXX()</code>.</p>
<p>For example, we will use <code>my_ichtyo_boxplot</code> the <code>aaas</code> palette.</p>
<div class="sourceCode" id="cb419"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb419-1"><a href="graphics-and-colors.html#cb419-1" aria-hidden="true" tabindex="-1"></a>my_ichtyo_ggsci_aaas <span class="ot">&lt;-</span> my_ichtyo_boxplot <span class="sc">+</span> <span class="fu">scale_fill_aaas</span>()</span>
<span id="cb419-2"><a href="graphics-and-colors.html#cb419-2" aria-hidden="true" tabindex="-1"></a>my_ichtyo_ggsci_aaas</span></code></pre></div>
<p><img src="figures/session_9_figs/13_plot_aaas.png" width="100%" /></p>
<p>We can use <code>my_ichtyo_scatter</code> the <code>uchicago</code> palette.</p>
<div class="sourceCode" id="cb420"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb420-1"><a href="graphics-and-colors.html#cb420-1" aria-hidden="true" tabindex="-1"></a>my_ichtyo_ggsci_uchicago <span class="ot">&lt;-</span> my_ichtyo_scatter <span class="sc">+</span> <span class="fu">scale_fill_uchicago</span>()</span>
<span id="cb420-2"><a href="graphics-and-colors.html#cb420-2" aria-hidden="true" tabindex="-1"></a>my_ichtyo_ggsci_uchicago</span></code></pre></div>
<p><img src="figures/session_9_figs/14_plot_uchicago.png" width="100%" /></p>
</div>
<div id="wesanderson-package" class="section level3" number="9.7.4">
<h3><span class="header-section-number">9.7.4</span> <code>wesanderson</code> package</h3>
<p><strong>13)</strong> The package <a href="https://cran.r-project.org/web/packages/wesanderson/index.html" target="_blank"><strong>wesanderson</strong></a> is a collection of palettes generated mostly from ‘Wes Anderson’ movies that can be incorporated in <strong>ggplot2</strong>.</p>
<div class="sourceCode" id="cb421"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb421-1"><a href="graphics-and-colors.html#cb421-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We need to install R-package &#39;wesanderson&#39; </span></span>
<span id="cb421-2"><a href="graphics-and-colors.html#cb421-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;wesanderson&quot;</span>)</span>
<span id="cb421-3"><a href="graphics-and-colors.html#cb421-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(wesanderson)</span></code></pre></div>
<p>The list of the palettes present include: BottleRocket1,BottleRocket2,Rushmore1,Royal1,Royal2,Zissou1,Darjeeling1,Darjeeling2,Chevalier1,FantasticFox1,Moonrise1,Moonrise2,Moonrise3,Cavalcanti1,GrandBudapest1,GrandBudapest2,IsleofDogs1,IsleofDogs2.</p>
<div class="sourceCode" id="cb422"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb422-1"><a href="graphics-and-colors.html#cb422-1" aria-hidden="true" tabindex="-1"></a><span class="co"># I will need 9 colors in a discrete scale for &#39;my_ichtyo_boxplot&#39;</span></span>
<span id="cb422-2"><a href="graphics-and-colors.html#cb422-2" aria-hidden="true" tabindex="-1"></a>my_wes_palette <span class="ot">&lt;-</span> <span class="fu">wes_palette</span>(<span class="st">&quot;IsleofDogs2&quot;</span>, <span class="dv">9</span>, <span class="at">type =</span> <span class="st">&quot;continuous&quot;</span>)</span>
<span id="cb422-3"><a href="graphics-and-colors.html#cb422-3" aria-hidden="true" tabindex="-1"></a>my_wes_palette</span>
<span id="cb422-4"><a href="graphics-and-colors.html#cb422-4" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(my_wes_palette)</span>
<span id="cb422-5"><a href="graphics-and-colors.html#cb422-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;palette&#39; chr [1:9] &quot;#EAD3BF&quot; &quot;#CAB3A2&quot; &quot;#AA9486&quot; &quot;#B08C69&quot; &quot;#B6854D&quot; &quot;#775B3E&quot; &quot;#39312F&quot; &quot;#2A2423&quot; &quot;#1C1718&quot;</span></span>
<span id="cb422-6"><a href="graphics-and-colors.html#cb422-6" aria-hidden="true" tabindex="-1"></a><span class="co"># - attr(*, &quot;name&quot;)= chr &quot;IsleofDogs2&quot;</span></span></code></pre></div>
<p><img src="figures/session_9_figs/15_plot_wes_palette.png" width="100%" /></p>
<p>We can apply the <code>my_ichtyo_scatter</code> with <strong>ggplot2</strong> with these 9 colors using <code>scale_colour_manual()</code>.</p>
<div class="sourceCode" id="cb423"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb423-1"><a href="graphics-and-colors.html#cb423-1" aria-hidden="true" tabindex="-1"></a>my_ichtyo_scatter_wes <span class="ot">&lt;-</span> my_ichtyo_scatter <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> my_wes_palette)</span>
<span id="cb423-2"><a href="graphics-and-colors.html#cb423-2" aria-hidden="true" tabindex="-1"></a>my_ichtyo_scatter_wes</span></code></pre></div>
<p><img src="figures/session_9_figs/16_plot_wes_scatter.png" width="100%" /></p>
</div>
<div id="rcartocolor-package" class="section level3" number="9.7.5">
<h3><span class="header-section-number">9.7.5</span> <code>rcartocolor</code> package</h3>
<p><strong>14)</strong> The package <a href="https://cran.r-project.org/web/packages/rcartocolor/index.html" target="_blank"><strong>rcartocolor</strong></a> is a collection of color schemes for maps and other graphicsdesigned by ‘CARTO’ as described at <a href="https://carto.com/carto-colors/" target="_blank"><strong>https://carto.com/carto-colors/</strong></a>. It includes four types of palettes: aggregation, diverging, qualitative,and quantitative that can be incorporated in <strong>ggplot2</strong>.</p>
<div class="sourceCode" id="cb424"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb424-1"><a href="graphics-and-colors.html#cb424-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We need to install R-package &#39;rcartocolor&#39; </span></span>
<span id="cb424-2"><a href="graphics-and-colors.html#cb424-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;rcartocolor&quot;</span>)</span>
<span id="cb424-3"><a href="graphics-and-colors.html#cb424-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rcartocolor)</span>
<span id="cb424-4"><a href="graphics-and-colors.html#cb424-4" aria-hidden="true" tabindex="-1"></a><span class="do">## diplay all colors</span></span>
<span id="cb424-5"><a href="graphics-and-colors.html#cb424-5" aria-hidden="true" tabindex="-1"></a><span class="fu">display_carto_all</span>()</span></code></pre></div>
<p><img src="figures/session_9_figs/34_rcartocolor_palette.png" width="100%" /></p>
<p>The list of 34 palettes presented can be select for a number specific of colors and if you want them to be <code>colorblind_friendly = TRUE</code>.</p>
<div class="sourceCode" id="cb425"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb425-1"><a href="graphics-and-colors.html#cb425-1" aria-hidden="true" tabindex="-1"></a><span class="do">## check information about how to change colors in palette</span></span>
<span id="cb425-2"><a href="graphics-and-colors.html#cb425-2" aria-hidden="true" tabindex="-1"></a>?scale_fill_carto_d</span>
<span id="cb425-3"><a href="graphics-and-colors.html#cb425-3" aria-hidden="true" tabindex="-1"></a><span class="do">## color the boxplot</span></span>
<span id="cb425-4"><a href="graphics-and-colors.html#cb425-4" aria-hidden="true" tabindex="-1"></a>my_ichtyo_boxplot_carto <span class="ot">&lt;-</span> my_ichtyo_boxplot <span class="sc">+</span> <span class="fu">scale_fill_carto_d</span>(<span class="at">type =</span> <span class="st">&quot;qualitative&quot;</span>, <span class="at">palette =</span> <span class="st">&quot;Vivid&quot;</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb425-5"><a href="graphics-and-colors.html#cb425-5" aria-hidden="true" tabindex="-1"></a>my_ichtyo_boxplot_carto</span></code></pre></div>
<p><img src="figures/session_9_figs/35_rcartcolor_boxplot.png" width="100%" /></p>
</div>
<div id="colorspace-package" class="section level3" number="9.7.6">
<h3><span class="header-section-number">9.7.6</span> <code>colorspace</code> package</h3>
<p><strong>15)</strong> Exploring and selecting your own colors and palettes can be performed with <a href="https://cran.r-project.org/web/packages/colorspace/index.html" target="_blank"><strong>colorspace</strong></a> that includes an extensive <a href="https://cran.r-project.org/web/packages/colorspace/vignettes/colorspace.html" target="_blank"><strong>manual</strong></a>.</p>
<p>To use <strong>colorspace</strong>, we can do as follow.</p>
<div class="sourceCode" id="cb426"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb426-1"><a href="graphics-and-colors.html#cb426-1" aria-hidden="true" tabindex="-1"></a><span class="do">## install and load package </span></span>
<span id="cb426-2"><a href="graphics-and-colors.html#cb426-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;colorspace&quot;</span>)</span>
<span id="cb426-3"><a href="graphics-and-colors.html#cb426-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(colorspace)</span>
<span id="cb426-4"><a href="graphics-and-colors.html#cb426-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb426-5"><a href="graphics-and-colors.html#cb426-5" aria-hidden="true" tabindex="-1"></a><span class="do">## palettes in this package, type and their names</span></span>
<span id="cb426-6"><a href="graphics-and-colors.html#cb426-6" aria-hidden="true" tabindex="-1"></a><span class="fu">hcl_palettes</span>()</span>
<span id="cb426-7"><a href="graphics-and-colors.html#cb426-7" aria-hidden="true" tabindex="-1"></a><span class="co">#HCL palettes</span></span>
<span id="cb426-8"><a href="graphics-and-colors.html#cb426-8" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb426-9"><a href="graphics-and-colors.html#cb426-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Type:  Qualitative </span></span>
<span id="cb426-10"><a href="graphics-and-colors.html#cb426-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Names: Pastel 1, Dark 2, Dark 3, Set 2, Set 3, Warm, Cold, Harmonic, Dynamic</span></span>
<span id="cb426-11"><a href="graphics-and-colors.html#cb426-11" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb426-12"><a href="graphics-and-colors.html#cb426-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Type:  Sequential (single-hue) </span></span>
<span id="cb426-13"><a href="graphics-and-colors.html#cb426-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Names: Grays, Light Grays, Blues 2, Blues 3, Purples 2, Purples 3, Reds 2, Reds 3, Greens 2, Greens 3, Oslo</span></span>
<span id="cb426-14"><a href="graphics-and-colors.html#cb426-14" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb426-15"><a href="graphics-and-colors.html#cb426-15" aria-hidden="true" tabindex="-1"></a><span class="co">#Type:  Sequential (multi-hue) </span></span>
<span id="cb426-16"><a href="graphics-and-colors.html#cb426-16" aria-hidden="true" tabindex="-1"></a><span class="co">#Names: Purple-Blue, Red-Purple, Red-Blue, Purple-Orange, Purple-Yellow, Blue-Yellow, Green-Yellow, Red-Yellow, Heat, Heat 2, Terrain,</span></span>
<span id="cb426-17"><a href="graphics-and-colors.html#cb426-17" aria-hidden="true" tabindex="-1"></a><span class="co">#       Terrain 2, Viridis, Plasma, Inferno, Dark Mint, Mint, BluGrn, Teal, TealGrn, Emrld, BluYl, ag_GrnYl, Peach, PinkYl,</span></span>
<span id="cb426-18"><a href="graphics-and-colors.html#cb426-18" aria-hidden="true" tabindex="-1"></a><span class="co">#       Burg, BurgYl, RedOr, OrYel, Purp, PurpOr, Sunset, Magenta, SunsetDark, ag_Sunset, BrwnYl, YlOrRd, YlOrBr, OrRd, Oranges,</span></span>
<span id="cb426-19"><a href="graphics-and-colors.html#cb426-19" aria-hidden="true" tabindex="-1"></a><span class="co">#       YlGn, YlGnBu, Reds, RdPu, PuRd, Purples, PuBuGn, PuBu, Greens, BuGn, GnBu, BuPu, Blues, Lajolla, Turku, Hawaii, Batlow</span></span>
<span id="cb426-20"><a href="graphics-and-colors.html#cb426-20" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb426-21"><a href="graphics-and-colors.html#cb426-21" aria-hidden="true" tabindex="-1"></a><span class="co">#Type:  Diverging </span></span>
<span id="cb426-22"><a href="graphics-and-colors.html#cb426-22" aria-hidden="true" tabindex="-1"></a><span class="co">#Names: Blue-Red, Blue-Red 2, Blue-Red 3, Red-Green, Purple-Green, Purple-Brown, Green-Brown, Blue-Yellow 2, Blue-Yellow 3,</span></span>
<span id="cb426-23"><a href="graphics-and-colors.html#cb426-23" aria-hidden="true" tabindex="-1"></a><span class="co">#       Green-Orange, Cyan-Magenta, Tropic, Broc, Cork, Vik, Berlin, Lisbon, Tofino</span></span>
<span id="cb426-24"><a href="graphics-and-colors.html#cb426-24" aria-hidden="true" tabindex="-1"></a><span class="fu">hcl_palettes</span>(<span class="at">plot =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="figures/session_9_figs/36_colorspace_palettes.png" width="100%" /></p>
<p>You then can select the palette that you want and request the number of colors for your plot. These color will be returned as hexadecimal codes.</p>
<div class="sourceCode" id="cb427"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb427-1"><a href="graphics-and-colors.html#cb427-1" aria-hidden="true" tabindex="-1"></a><span class="do">## you want 5 qualitative colors from palette </span></span>
<span id="cb427-2"><a href="graphics-and-colors.html#cb427-2" aria-hidden="true" tabindex="-1"></a>my_qualitative <span class="ot">&lt;-</span> <span class="fu">qualitative_hcl</span>(<span class="dv">5</span>, <span class="at">palette =</span> <span class="st">&quot;Harmonic&quot;</span>)</span>
<span id="cb427-3"><a href="graphics-and-colors.html#cb427-3" aria-hidden="true" tabindex="-1"></a>my_qualitative</span>
<span id="cb427-4"><a href="graphics-and-colors.html#cb427-4" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;#C7A76C&quot; &quot;#99B56B&quot; &quot;#5CBD92&quot; &quot;#3BBCBF&quot; &quot;#7DB0DD&quot;</span></span>
<span id="cb427-5"><a href="graphics-and-colors.html#cb427-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb427-6"><a href="graphics-and-colors.html#cb427-6" aria-hidden="true" tabindex="-1"></a><span class="do">## you want 5 sequential colors from palette </span></span>
<span id="cb427-7"><a href="graphics-and-colors.html#cb427-7" aria-hidden="true" tabindex="-1"></a>my_sequential <span class="ot">&lt;-</span> <span class="fu">sequential_hcl</span>(<span class="dv">5</span>, <span class="at">palette =</span> <span class="st">&quot;Batlow&quot;</span>)</span>
<span id="cb427-8"><a href="graphics-and-colors.html#cb427-8" aria-hidden="true" tabindex="-1"></a>my_sequential</span>
<span id="cb427-9"><a href="graphics-and-colors.html#cb427-9" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;#201158&quot; &quot;#005E5E&quot; &quot;#578B21&quot; &quot;#E89E6B&quot; &quot;#FFCEF4&quot;</span></span>
<span id="cb427-10"><a href="graphics-and-colors.html#cb427-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb427-11"><a href="graphics-and-colors.html#cb427-11" aria-hidden="true" tabindex="-1"></a><span class="do">## you want 5 diverging colors from palette </span></span>
<span id="cb427-12"><a href="graphics-and-colors.html#cb427-12" aria-hidden="true" tabindex="-1"></a>my_diverging <span class="ot">&lt;-</span> <span class="fu">diverging_hcl</span>(<span class="dv">5</span>, <span class="at">palette =</span> <span class="st">&quot;Berlin&quot;</span>)</span>
<span id="cb427-13"><a href="graphics-and-colors.html#cb427-13" aria-hidden="true" tabindex="-1"></a>my_diverging</span>
<span id="cb427-14"><a href="graphics-and-colors.html#cb427-14" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;#7FBFF5&quot; &quot;#004B73&quot; &quot;#111111&quot; &quot;#713430&quot; &quot;#F8A29E&quot;</span></span></code></pre></div>
<p>This R packate provides with some demos to visualize how it will look like.</p>
<div class="sourceCode" id="cb428"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb428-1"><a href="graphics-and-colors.html#cb428-1" aria-hidden="true" tabindex="-1"></a><span class="do">## my_qualitative visualization</span></span>
<span id="cb428-2"><a href="graphics-and-colors.html#cb428-2" aria-hidden="true" tabindex="-1"></a><span class="fu">specplot</span>(my_qualitative, <span class="at">type =</span> <span class="st">&quot;o&quot;</span>)</span></code></pre></div>
<p><img src="figures/session_9_figs/37_colorspace_plot1.png" width="100%" /></p>
<div class="sourceCode" id="cb429"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb429-1"><a href="graphics-and-colors.html#cb429-1" aria-hidden="true" tabindex="-1"></a><span class="do">## my_sequential visualization</span></span>
<span id="cb429-2"><a href="graphics-and-colors.html#cb429-2" aria-hidden="true" tabindex="-1"></a><span class="fu">specplot</span>(my_sequential, <span class="at">type =</span> <span class="st">&quot;o&quot;</span>)</span></code></pre></div>
<p><img src="figures/session_9_figs/38_colorspace_plot2.png" width="100%" /></p>
<div class="sourceCode" id="cb430"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb430-1"><a href="graphics-and-colors.html#cb430-1" aria-hidden="true" tabindex="-1"></a><span class="do">## my_diverging visualization</span></span>
<span id="cb430-2"><a href="graphics-and-colors.html#cb430-2" aria-hidden="true" tabindex="-1"></a><span class="fu">specplot</span>(my_diverging, <span class="at">type =</span> <span class="st">&quot;o&quot;</span>)</span></code></pre></div>
<p><img src="figures/session_9_figs/39_colorspace_plot3.png" width="100%" /></p>
</div>
</div>
<div id="making-your-own-palette" class="section level2" number="9.8">
<h2><span class="header-section-number">9.8</span> Making your own palette</h2>
<p><strong>16)</strong> In many instances, you want your own palette and this can be done with the base function <code>colorRampPalette()</code>.</p>
<p>You have to pick your colors by name as in the <a href="https://github.com/corytophanes/BIO_BIT_Bioinformatics_209/tree/main/ref_files_pdfs" target="_blank"><strong>Rcolor.pdf</strong></a>. Alternatively, you can get the names by <code>colors()</code>. We also want to be able to the colors, so we use the function <code>show_col()</code> from the R package <a href="https://cran.r-project.org/web/packages/scales/index.html" target="_blank"><strong>scales</strong></a>.</p>
<div class="sourceCode" id="cb431"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb431-1"><a href="graphics-and-colors.html#cb431-1" aria-hidden="true" tabindex="-1"></a><span class="do">## get a vector of all colors by name</span></span>
<span id="cb431-2"><a href="graphics-and-colors.html#cb431-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colors</span>()</span>
<span id="cb431-3"><a href="graphics-and-colors.html#cb431-3" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;white&quot;                &quot;aliceblue&quot;            &quot;antiquewhite&quot;         &quot;antiquewhite1&quot;        &quot;antiquewhite2&quot;        &quot;antiquewhite3&quot;       </span></span>
<span id="cb431-4"><a href="graphics-and-colors.html#cb431-4" aria-hidden="true" tabindex="-1"></a><span class="co"># [7] &quot;antiquewhite4&quot;        &quot;aquamarine&quot;           &quot;aquamarine1&quot;          &quot;aquamarine2&quot;          &quot;aquamarine3&quot;          &quot;aquamarine4&quot;         </span></span>
<span id="cb431-5"><a href="graphics-and-colors.html#cb431-5" aria-hidden="true" tabindex="-1"></a><span class="co">#[13] &quot;azure&quot;                &quot;azure1&quot;               &quot;azure2&quot;               &quot;azure3&quot;               &quot;azure4&quot;               &quot;beige&quot;               </span></span>
<span id="cb431-6"><a href="graphics-and-colors.html#cb431-6" aria-hidden="true" tabindex="-1"></a><span class="co">#[19] &quot;bisque&quot;               &quot;bisque1&quot;              &quot;bisque2&quot;              &quot;bisque3&quot;              &quot;bisque4&quot;              &quot;black&quot;               </span></span>
<span id="cb431-7"><a href="graphics-and-colors.html#cb431-7" aria-hidden="true" tabindex="-1"></a><span class="co"># ...</span></span>
<span id="cb431-8"><a href="graphics-and-colors.html#cb431-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb431-9"><a href="graphics-and-colors.html#cb431-9" aria-hidden="true" tabindex="-1"></a><span class="do">## the total number of colors by name</span></span>
<span id="cb431-10"><a href="graphics-and-colors.html#cb431-10" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">colors</span>())</span>
<span id="cb431-11"><a href="graphics-and-colors.html#cb431-11" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 657</span></span>
<span id="cb431-12"><a href="graphics-and-colors.html#cb431-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb431-13"><a href="graphics-and-colors.html#cb431-13" aria-hidden="true" tabindex="-1"></a><span class="do">## for our example, we will get three colors by name in a vector</span></span>
<span id="cb431-14"><a href="graphics-and-colors.html#cb431-14" aria-hidden="true" tabindex="-1"></a>my_colors_by_name <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;firebrick4&quot;</span>, <span class="st">&quot;gray89&quot;</span>, <span class="st">&quot;navyblue&quot;</span>)</span>
<span id="cb431-15"><a href="graphics-and-colors.html#cb431-15" aria-hidden="true" tabindex="-1"></a>scales<span class="sc">::</span><span class="fu">show_col</span>(my_colors_by_name)</span></code></pre></div>
<p><img src="figures/session_9_figs/52_my_pal_1.png" width="100%" /></p>
<p>Likewise, you can get some hexadecimal colors codes for those that you want, e.g., <code>#EEBD91</code>, <code>#F5A2A2</code>, <code>#40CB63</code>, <code>#4682B4</code> and <code>#862185</code>.</p>
<div class="sourceCode" id="cb432"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb432-1"><a href="graphics-and-colors.html#cb432-1" aria-hidden="true" tabindex="-1"></a><span class="do">## for our example, we will get these 5 hex colors in a vector</span></span>
<span id="cb432-2"><a href="graphics-and-colors.html#cb432-2" aria-hidden="true" tabindex="-1"></a>my_colors_by_hex <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;#EEBD91&quot;</span>, <span class="st">&quot;#F5A2A2&quot;</span>, <span class="st">&quot;#40CB63&quot;</span>, <span class="st">&quot;#4682B4&quot;</span>, <span class="st">&quot;#862185&quot;</span>)</span>
<span id="cb432-3"><a href="graphics-and-colors.html#cb432-3" aria-hidden="true" tabindex="-1"></a>scales<span class="sc">::</span><span class="fu">show_col</span>(my_colors_by_hex)</span></code></pre></div>
<p><img src="figures/session_9_figs/53_my_pal_2.png" width="100%" /></p>
<p>To generate your palette, you use the function <code>colorRampPalette()</code>.</p>
<div class="sourceCode" id="cb433"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb433-1"><a href="graphics-and-colors.html#cb433-1" aria-hidden="true" tabindex="-1"></a><span class="do">##NOTE: the vector of colors that you want to generate in this case 9 and this number is in parenthesis, so it is (9)</span></span>
<span id="cb433-2"><a href="graphics-and-colors.html#cb433-2" aria-hidden="true" tabindex="-1"></a>my_colors_by_name_ramp <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(my_colors_by_name)(<span class="dv">9</span>)</span>
<span id="cb433-3"><a href="graphics-and-colors.html#cb433-3" aria-hidden="true" tabindex="-1"></a>my_colors_by_name_ramp</span>
<span id="cb433-4"><a href="graphics-and-colors.html#cb433-4" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;#8B1A1A&quot; &quot;#A14C4C&quot; &quot;#B67E7E&quot; &quot;#CCB0B0&quot; &quot;#E3E3E3&quot; &quot;#AAAACA&quot; &quot;#7171B1&quot; &quot;#383898&quot; &quot;#000080&quot;</span></span>
<span id="cb433-5"><a href="graphics-and-colors.html#cb433-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb433-6"><a href="graphics-and-colors.html#cb433-6" aria-hidden="true" tabindex="-1"></a>my_colors_by_hex_ramp <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(my_colors_by_hex)(<span class="dv">9</span>)</span>
<span id="cb433-7"><a href="graphics-and-colors.html#cb433-7" aria-hidden="true" tabindex="-1"></a>my_colors_by_hex_ramp</span>
<span id="cb433-8"><a href="graphics-and-colors.html#cb433-8" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;#EEBD91&quot; &quot;#F1AF99&quot; &quot;#F5A2A2&quot; &quot;#9AB682&quot; &quot;#40CB63&quot; &quot;#43A68B&quot; &quot;#4682B4&quot; &quot;#66519C&quot; &quot;#862185&quot;</span></span></code></pre></div>
<p>Finally, we can apply these color ramps to <code>my_ichtyo_scatter</code> with <strong>ggplot2</strong> using <code>scale_colour_manual()</code>.</p>
<div class="sourceCode" id="cb434"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb434-1"><a href="graphics-and-colors.html#cb434-1" aria-hidden="true" tabindex="-1"></a>my_ichtyo_scatter_by_name <span class="ot">&lt;-</span> my_ichtyo_scatter <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> my_colors_by_name_ramp)</span>
<span id="cb434-2"><a href="graphics-and-colors.html#cb434-2" aria-hidden="true" tabindex="-1"></a>my_ichtyo_scatter_by_name</span></code></pre></div>
<p><img src="figures/session_9_figs/17_plot_ramp_3.png" width="100%" /></p>
<div class="sourceCode" id="cb435"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb435-1"><a href="graphics-and-colors.html#cb435-1" aria-hidden="true" tabindex="-1"></a>my_ichtyo_scatter_by_hex <span class="ot">&lt;-</span> my_ichtyo_scatter <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> my_colors_by_hex_ramp)</span>
<span id="cb435-2"><a href="graphics-and-colors.html#cb435-2" aria-hidden="true" tabindex="-1"></a>my_ichtyo_scatter_by_hex</span></code></pre></div>
<p><img src="figures/session_9_figs/18_plot_ramp_5.png" width="100%" /></p>
<p><strong>17)</strong> You can extract colors for an image that you like (jpg) from an url (website) or a photo using a clever combination of functions of the R packages:<a href="https://cran.r-project.org/web/packages/magick/index.html" target="_blank"><strong>magick</strong></a>, <a href="https://cran.r-project.org/web/packages/imager/index.html" target="_blank"><strong>imager</strong></a>, <a href="https://cran.r-project.org/web/packages/dplyr/index.html" target="_blank"><strong>dplyr</strong></a>. Other packages like <strong>scales</strong> and <strong>ggplot2</strong> were previously installes.</p>
<div class="sourceCode" id="cb436"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb436-1"><a href="graphics-and-colors.html#cb436-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Load up packages I&#39;m going to use first.</span></span>
<span id="cb436-2"><a href="graphics-and-colors.html#cb436-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;magick&quot;</span>)</span>
<span id="cb436-3"><a href="graphics-and-colors.html#cb436-3" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;imager&quot;</span>)</span>
<span id="cb436-4"><a href="graphics-and-colors.html#cb436-4" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;dplyr&quot;</span>)</span>
<span id="cb436-5"><a href="graphics-and-colors.html#cb436-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb436-6"><a href="graphics-and-colors.html#cb436-6" aria-hidden="true" tabindex="-1"></a><span class="do">## required libraries</span></span>
<span id="cb436-7"><a href="graphics-and-colors.html#cb436-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb436-8"><a href="graphics-and-colors.html#cb436-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb436-9"><a href="graphics-and-colors.html#cb436-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(magick) </span>
<span id="cb436-10"><a href="graphics-and-colors.html#cb436-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales) </span>
<span id="cb436-11"><a href="graphics-and-colors.html#cb436-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(imager)</span>
<span id="cb436-12"><a href="graphics-and-colors.html#cb436-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb436-13"><a href="graphics-and-colors.html#cb436-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb436-14"><a href="graphics-and-colors.html#cb436-14" aria-hidden="true" tabindex="-1"></a><span class="do">## read image to extract color </span></span>
<span id="cb436-15"><a href="graphics-and-colors.html#cb436-15" aria-hidden="true" tabindex="-1"></a><span class="do">## </span><span class="al">NOTE</span><span class="do">: this has to be your own image, I will not post this jpg.</span></span>
<span id="cb436-16"><a href="graphics-and-colors.html#cb436-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb436-17"><a href="graphics-and-colors.html#cb436-17" aria-hidden="true" tabindex="-1"></a>my_image <span class="ot">&lt;-</span> <span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/figures_emer/samurai_champloo.jpg&quot;</span></span>
<span id="cb436-18"><a href="graphics-and-colors.html#cb436-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb436-19"><a href="graphics-and-colors.html#cb436-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Function to get your image processed, n number of colors out of your image, and cs as colour space.</span></span>
<span id="cb436-20"><a href="graphics-and-colors.html#cb436-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb436-21"><a href="graphics-and-colors.html#cb436-21" aria-hidden="true" tabindex="-1"></a>get_colorPal_form_image <span class="ot">&lt;-</span> <span class="cf">function</span>(image_file_input,</span>
<span id="cb436-22"><a href="graphics-and-colors.html#cb436-22" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">n=</span><span class="dv">8</span>, </span>
<span id="cb436-23"><a href="graphics-and-colors.html#cb436-23" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">cs=</span><span class="st">&quot;RGB&quot;</span>,</span>
<span id="cb436-24"><a href="graphics-and-colors.html#cb436-24" aria-hidden="true" tabindex="-1"></a>                        <span class="at">show_image_row_user =</span> <span class="cn">TRUE</span>){</span>
<span id="cb436-25"><a href="graphics-and-colors.html#cb436-25" aria-hidden="true" tabindex="-1"></a><span class="co">#require libraries</span></span>
<span id="cb436-26"><a href="graphics-and-colors.html#cb436-26" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggplot2)</span>
<span id="cb436-27"><a href="graphics-and-colors.html#cb436-27" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(magick) </span>
<span id="cb436-28"><a href="graphics-and-colors.html#cb436-28" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(scales) </span>
<span id="cb436-29"><a href="graphics-and-colors.html#cb436-29" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(imager)</span>
<span id="cb436-30"><a href="graphics-and-colors.html#cb436-30" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(dplyr)</span>
<span id="cb436-31"><a href="graphics-and-colors.html#cb436-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb436-32"><a href="graphics-and-colors.html#cb436-32" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st"> *** code modified from: https://www.r-bloggers.com/2019/01/extracting-colours-from-your-images-with-image-quantization/ *** </span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb436-33"><a href="graphics-and-colors.html#cb436-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb436-34"><a href="graphics-and-colors.html#cb436-34" aria-hidden="true" tabindex="-1"></a>  image_file_input_pros <span class="ot">&lt;-</span> image_file_input</span>
<span id="cb436-35"><a href="graphics-and-colors.html#cb436-35" aria-hidden="true" tabindex="-1"></a>  show_image_row <span class="ot">&lt;-</span> show_image_row_user</span>
<span id="cb436-36"><a href="graphics-and-colors.html#cb436-36" aria-hidden="true" tabindex="-1"></a>  raw_image <span class="ot">&lt;-</span> magick<span class="sc">::</span><span class="fu">image_read</span>(image_file_input_pros)</span>
<span id="cb436-37"><a href="graphics-and-colors.html#cb436-37" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> magick<span class="sc">::</span><span class="fu">image_resize</span>(<span class="at">image =</span> raw_image ,  <span class="st">&quot;100&quot;</span>) </span>
<span id="cb436-38"><a href="graphics-and-colors.html#cb436-38" aria-hidden="true" tabindex="-1"></a>  <span class="co"># reducing colours! different colorspace gives you different result</span></span>
<span id="cb436-39"><a href="graphics-and-colors.html#cb436-39" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> magick<span class="sc">::</span><span class="fu">image_quantize</span>(tmp, <span class="at">max=</span>n, <span class="at">colorspace=</span>cs)</span>
<span id="cb436-40"><a href="graphics-and-colors.html#cb436-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># converting, becauase I want to use as.data.frame function in imager package.</span></span>
<span id="cb436-41"><a href="graphics-and-colors.html#cb436-41" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> imager<span class="sc">::</span><span class="fu">magick2cimg</span>(tmp) </span>
<span id="cb436-42"><a href="graphics-and-colors.html#cb436-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># sorting colour by hue rather than RGB (red green blue)</span></span>
<span id="cb436-43"><a href="graphics-and-colors.html#cb436-43" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> imager<span class="sc">::</span><span class="fu">RGBtoHSV</span>(tmp)</span>
<span id="cb436-44"><a href="graphics-and-colors.html#cb436-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># making it wide makes it easier to output hex colour</span></span>
<span id="cb436-45"><a href="graphics-and-colors.html#cb436-45" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(tmp, <span class="at">wide=</span><span class="st">&quot;c&quot;</span>)</span>
<span id="cb436-46"><a href="graphics-and-colors.html#cb436-46" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(tmp, <span class="at">hex =</span> <span class="fu">hsv</span>(<span class="fu">rescale</span>(c<span class="fl">.1</span>, <span class="at">from=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">360</span>)),c<span class="fl">.2</span>,c<span class="fl">.3</span>),</span>
<span id="cb436-47"><a href="graphics-and-colors.html#cb436-47" aria-hidden="true" tabindex="-1"></a>                            <span class="at">hue =</span> c<span class="fl">.1</span>,</span>
<span id="cb436-48"><a href="graphics-and-colors.html#cb436-48" aria-hidden="true" tabindex="-1"></a>                            <span class="at">sat =</span> c<span class="fl">.2</span>,</span>
<span id="cb436-49"><a href="graphics-and-colors.html#cb436-49" aria-hidden="true" tabindex="-1"></a>                          <span class="at">value =</span> c<span class="fl">.3</span>)</span>
<span id="cb436-50"><a href="graphics-and-colors.html#cb436-50" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">count</span>(tmp, hex, hue, sat,value, <span class="at">sort=</span>T) </span>
<span id="cb436-51"><a href="graphics-and-colors.html#cb436-51" aria-hidden="true" tabindex="-1"></a>  tmp <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">mutate</span>(tmp, <span class="at">colorspace =</span> cs)</span>
<span id="cb436-52"><a href="graphics-and-colors.html#cb436-52" aria-hidden="true" tabindex="-1"></a>  <span class="co"># out colors palette</span></span>
<span id="cb436-53"><a href="graphics-and-colors.html#cb436-53" aria-hidden="true" tabindex="-1"></a>  color_hex <span class="ot">&lt;-</span> tmp<span class="sc">$</span>hex</span>
<span id="cb436-54"><a href="graphics-and-colors.html#cb436-54" aria-hidden="true" tabindex="-1"></a>   color_df <span class="ot">&lt;-</span> tmp</span>
<span id="cb436-55"><a href="graphics-and-colors.html#cb436-55" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb436-56"><a href="graphics-and-colors.html#cb436-56" aria-hidden="true" tabindex="-1"></a>  <span class="co"># plot colors</span></span>
<span id="cb436-57"><a href="graphics-and-colors.html#cb436-57" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(show_image_row) {magick<span class="sc">::</span><span class="fu">image_browse</span>(raw_image)}</span>
<span id="cb436-58"><a href="graphics-and-colors.html#cb436-58" aria-hidden="true" tabindex="-1"></a>  scales<span class="sc">::</span><span class="fu">show_col</span>(color_hex)</span>
<span id="cb436-59"><a href="graphics-and-colors.html#cb436-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb436-60"><a href="graphics-and-colors.html#cb436-60" aria-hidden="true" tabindex="-1"></a>  <span class="co"># return colors</span></span>
<span id="cb436-61"><a href="graphics-and-colors.html#cb436-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(color_hex, color_df))</span>
<span id="cb436-62"><a href="graphics-and-colors.html#cb436-62" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb436-63"><a href="graphics-and-colors.html#cb436-63" aria-hidden="true" tabindex="-1"></a>                                           }</span>
<span id="cb436-64"><a href="graphics-and-colors.html#cb436-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb436-65"><a href="graphics-and-colors.html#cb436-65" aria-hidden="true" tabindex="-1"></a><span class="do">## running the function and asking for 8 colors</span></span>
<span id="cb436-66"><a href="graphics-and-colors.html#cb436-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb436-67"><a href="graphics-and-colors.html#cb436-67" aria-hidden="true" tabindex="-1"></a>samuari_champloo_color_hex <span class="ot">&lt;-</span> <span class="fu">get_colorPal_form_image</span>(my_image, <span class="at">n=</span> <span class="dv">8</span>)</span>
<span id="cb436-68"><a href="graphics-and-colors.html#cb436-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb436-69"><a href="graphics-and-colors.html#cb436-69" aria-hidden="true" tabindex="-1"></a><span class="co"># hex codes</span></span>
<span id="cb436-70"><a href="graphics-and-colors.html#cb436-70" aria-hidden="true" tabindex="-1"></a>samuari_champloo_color_hex[[<span class="dv">1</span>]]</span>
<span id="cb436-71"><a href="graphics-and-colors.html#cb436-71" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;#FFE39D&quot; &quot;#D93907&quot; &quot;#2F1E16&quot; &quot;#F86413&quot; &quot;#A56343&quot; &quot;#F8D894&quot; &quot;#E9A264&quot;</span></span>
<span id="cb436-72"><a href="graphics-and-colors.html#cb436-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb436-73"><a href="graphics-and-colors.html#cb436-73" aria-hidden="true" tabindex="-1"></a>scales<span class="sc">::</span><span class="fu">show_col</span>(samuari_champloo_color_hex[[<span class="dv">1</span>]])</span></code></pre></div>
<p><img src="figures/session_9_figs/54_samurai_champloo_pal.png" width="100%" /></p>
<p>Here is an image that is free in its URL to apply the function <code>get_colorPal_form_image()</code>.</p>
<div class="sourceCode" id="cb437"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb437-1"><a href="graphics-and-colors.html#cb437-1" aria-hidden="true" tabindex="-1"></a><span class="do">## from URL from jpg.</span></span>
<span id="cb437-2"><a href="graphics-and-colors.html#cb437-2" aria-hidden="true" tabindex="-1"></a>my_image <span class="ot">&lt;-</span> <span class="st">&quot;https://upload.wikimedia.org/wikipedia/commons/a/a4/Piet_Mondriaan%2C_1930_-_Mondrian_Composition_II_in_Red%2C_Blue%2C_and_Yellow.jpg&quot;</span></span>
<span id="cb437-3"><a href="graphics-and-colors.html#cb437-3" aria-hidden="true" tabindex="-1"></a>Mondriaan_color_hex <span class="ot">&lt;-</span> <span class="fu">get_colorPal_form_image</span>(my_image, <span class="at">n=</span> <span class="dv">10</span>)</span>
<span id="cb437-4"><a href="graphics-and-colors.html#cb437-4" aria-hidden="true" tabindex="-1"></a>Mondriaan_color_hex </span>
<span id="cb437-5"><a href="graphics-and-colors.html#cb437-5" aria-hidden="true" tabindex="-1"></a><span class="co">#[[1]]</span></span>
<span id="cb437-6"><a href="graphics-and-colors.html#cb437-6" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;#DB2920&quot; &quot;#E3312C&quot; &quot;#EAE9E9&quot; &quot;#E5E4DE&quot; &quot;#2E2B2F&quot; &quot;#617098&quot; &quot;#DCD6CD&quot; &quot;#DFCCA4&quot; &quot;#E3DDD8&quot; &quot;#E2AB9C&quot;</span></span>
<span id="cb437-7"><a href="graphics-and-colors.html#cb437-7" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb437-8"><a href="graphics-and-colors.html#cb437-8" aria-hidden="true" tabindex="-1"></a><span class="co">#[[2]]</span></span>
<span id="cb437-9"><a href="graphics-and-colors.html#cb437-9" aria-hidden="true" tabindex="-1"></a><span class="co">#       hex        hue         sat     value    n colorspace</span></span>
<span id="cb437-10"><a href="graphics-and-colors.html#cb437-10" aria-hidden="true" tabindex="-1"></a><span class="co">#1  #DB2920   2.887701 0.853881279 0.8588235 2669        RGB</span></span>
<span id="cb437-11"><a href="graphics-and-colors.html#cb437-11" aria-hidden="true" tabindex="-1"></a><span class="co">#2  #E3312C   1.639344 0.806167401 0.8901961 2286        RGB</span></span>
<span id="cb437-12"><a href="graphics-and-colors.html#cb437-12" aria-hidden="true" tabindex="-1"></a><span class="co">#3  #EAE9E9   0.000000 0.004273504 0.9176471 1636        RGB</span></span>
<span id="cb437-13"><a href="graphics-and-colors.html#cb437-13" aria-hidden="true" tabindex="-1"></a><span class="co">#4  #E5E4DE  51.428571 0.030567686 0.8980392 1197        RGB</span></span>
<span id="cb437-14"><a href="graphics-and-colors.html#cb437-14" aria-hidden="true" tabindex="-1"></a><span class="co">#5  #2E2B2F 285.000000 0.085106383 0.1843137  724        RGB</span></span>
<span id="cb437-15"><a href="graphics-and-colors.html#cb437-15" aria-hidden="true" tabindex="-1"></a><span class="co">#6  #617098 223.636364 0.361842105 0.5960784  670        RGB</span></span>
<span id="cb437-16"><a href="graphics-and-colors.html#cb437-16" aria-hidden="true" tabindex="-1"></a><span class="co">#7  #DCD6CD  36.000000 0.068181818 0.8627451  412        RGB</span></span>
<span id="cb437-17"><a href="graphics-and-colors.html#cb437-17" aria-hidden="true" tabindex="-1"></a><span class="co">#8  #DFCCA4  40.677966 0.264573991 0.8745098  203        RGB</span></span>
<span id="cb437-18"><a href="graphics-and-colors.html#cb437-18" aria-hidden="true" tabindex="-1"></a><span class="co">#9  #E3DDD8  27.272727 0.048458150 0.8901961  161        RGB</span></span>
<span id="cb437-19"><a href="graphics-and-colors.html#cb437-19" aria-hidden="true" tabindex="-1"></a><span class="co">#10 #E2AB9C  12.857143 0.309734513 0.8862745  142        RGB</span></span></code></pre></div>
<p><img src="figures/session_9_figs/55_mondriaan_pal.png" width="100%" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="multidimensional-data-in-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="basic-graphics.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/09-Graphics_Colors_v19.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
