<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Session 30 – Importing/Exporting Data into R | Bioinformatics Workshop Gitbook</title>
  <meta name="description" content="none" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Session 30 – Importing/Exporting Data into R | Bioinformatics Workshop Gitbook" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="none" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Session 30 – Importing/Exporting Data into R | Bioinformatics Workshop Gitbook" />
  
  <meta name="twitter:description" content="none" />
  

<meta name="author" content="Juan C. Santos, Randy Ortiz &amp; Genrietta Yagudayeva" />


<meta name="date" content="2025-04-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="bioinformatics-literature.html"/>
<link rel="next" href="working-with-proteins.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics Workshop Gitbook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html"><i class="fa fa-check"></i><b>1</b> – Getting Started with R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#why-do-we-use-r"><i class="fa fa-check"></i><b>1.1</b> Why do we use R?</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#get-a-good-text-editor"><i class="fa fa-check"></i><b>1.2</b> Get a good text editor</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-r"><i class="fa fa-check"></i><b>1.4</b> Installing R</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#annotating-code"><i class="fa fa-check"></i><b>1.5</b> Annotating code</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#defining-a-working-directory-in-r"><i class="fa fa-check"></i><b>1.6</b> Defining a working directory in R</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html"><i class="fa fa-check"></i><b>2</b> – Using R, Installing Packages and Importing/Exporting Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#your-first-session"><i class="fa fa-check"></i><b>2.1</b> Your first session</a></li>
<li class="chapter" data-level="2.2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#doing-some-base-functions-and-getting-help"><i class="fa fa-check"></i><b>2.2</b> Doing some base functions and getting help</a></li>
<li class="chapter" data-level="2.3" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#functions-between-two-or-more-vectors"><i class="fa fa-check"></i><b>2.3</b> Functions between two or more vectors</a></li>
<li class="chapter" data-level="2.4" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#about-r-packages"><i class="fa fa-check"></i><b>2.4</b> About R-packages</a></li>
<li class="chapter" data-level="2.5" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#cran-task-views-expert-opinion-on-useful-packages"><i class="fa fa-check"></i><b>2.5</b> CRAN Task Views – Expert opinion on useful packages</a></li>
<li class="chapter" data-level="2.6" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#installing-packages"><i class="fa fa-check"></i><b>2.6</b> Installing packages</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html"><i class="fa fa-check"></i><b>3</b> – Basic Data Structures in R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#scalars-and-vectors"><i class="fa fa-check"></i><b>3.1</b> Scalars and vectors</a></li>
<li class="chapter" data-level="3.2" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#arrays"><i class="fa fa-check"></i><b>3.3</b> Arrays</a></li>
<li class="chapter" data-level="3.4" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a></li>
<li class="chapter" data-level="3.5" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tables"><i class="fa fa-check"></i><b>3.5</b> Tables</a></li>
<li class="chapter" data-level="3.6" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#lists"><i class="fa fa-check"></i><b>3.6</b> Lists</a></li>
<li class="chapter" data-level="3.7" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-tables"><i class="fa fa-check"></i><b>3.7</b> Data Tables</a></li>
<li class="chapter" data-level="3.8" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tibbles"><i class="fa fa-check"></i><b>3.8</b> Tibbles</a></li>
<li class="chapter" data-level="3.9" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#dnastringsets-rnastringsets-and-aastringsets"><i class="fa fa-check"></i><b>3.9</b> DNAStringSets, RNAStringSets and AAStringSets</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html"><i class="fa fa-check"></i><b>4</b> – Text Editing and Data Transformations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#working-with-character-strings"><i class="fa fa-check"></i><b>4.1</b> Working with character strings</a></li>
<li class="chapter" data-level="4.2" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#regular-expressions-regex"><i class="fa fa-check"></i><b>4.2</b> Regular expressions (Regex)</a></li>
<li class="chapter" data-level="4.3" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#comparing-ordering-and-simplifying-vectors"><i class="fa fa-check"></i><b>4.3</b> Comparing, ordering and simplifying vectors</a></li>
<li class="chapter" data-level="4.4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#useful-functions-of-stringr"><i class="fa fa-check"></i><b>4.4</b> Useful functions of <code>stringr</code></a></li>
<li class="chapter" data-level="4.5" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#data-transformations"><i class="fa fa-check"></i><b>4.5</b> Data transformations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html"><i class="fa fa-check"></i><b>5</b> – Getting Biological Data from Public Repositories</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#ncbi-databases-genbank-and-others"><i class="fa fa-check"></i><b>5.1</b> NCBI databases: GenBank and others</a></li>
<li class="chapter" data-level="5.2" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#accessing-ncbi-within-r-rentrez"><i class="fa fa-check"></i><b>5.2</b> Accessing NCBI within R: <code>rentrez</code></a></li>
<li class="chapter" data-level="5.3" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#retrieving-pubmed-references"><i class="fa fa-check"></i><b>5.3</b> Retrieving PubMed references</a></li>
<li class="chapter" data-level="5.4" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#comparing-sequences-blast"><i class="fa fa-check"></i><b>5.4</b> Comparing sequences: BLAST</a></li>
<li class="chapter" data-level="5.5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-uniprot"><i class="fa fa-check"></i><b>5.5</b> Other databases: UniProt</a></li>
<li class="chapter" data-level="5.6" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-ensembl"><i class="fa fa-check"></i><b>5.6</b> Other databases: Ensembl</a></li>
<li class="chapter" data-level="5.7" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-taxonomy-phylogenetic-trees-and-species-distributions"><i class="fa fa-check"></i><b>5.7</b> Other databases: Taxonomy, phylogenetic trees and species distributions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html"><i class="fa fa-check"></i><b>6</b> – Basic Statistics in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#exploring-your-data"><i class="fa fa-check"></i><b>6.1</b> Exploring your data</a></li>
<li class="chapter" data-level="6.2" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#sub-setting-your-dataset"><i class="fa fa-check"></i><b>6.2</b> Sub-setting your dataset</a></li>
<li class="chapter" data-level="6.3" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#transposing-rotating-your-dataset"><i class="fa fa-check"></i><b>6.3</b> Transposing (rotating) your dataset</a></li>
<li class="chapter" data-level="6.4" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#dealing-with-missing-na-values"><i class="fa fa-check"></i><b>6.4</b> Dealing with missing (NA) values</a></li>
<li class="chapter" data-level="6.5" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#about-p-values"><i class="fa fa-check"></i><b>6.5</b> About p-values</a></li>
<li class="chapter" data-level="6.6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#how-to-read-p-values"><i class="fa fa-check"></i><b>6.6</b> How to read p-values</a></li>
<li class="chapter" data-level="6.7" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#multiple-testing-corrections"><i class="fa fa-check"></i><b>6.7</b> Multiple testing corrections</a></li>
<li class="chapter" data-level="6.8" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#normalizing-your-data"><i class="fa fa-check"></i><b>6.8</b> Normalizing your data</a></li>
<li class="chapter" data-level="6.9" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#what-test-should-i-use"><i class="fa fa-check"></i><b>6.9</b> What test should I use?</a></li>
<li class="chapter" data-level="6.10" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#type-of-variables-in-dataset"><i class="fa fa-check"></i><b>6.10</b> Type of variables in dataset</a></li>
<li class="chapter" data-level="6.11" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-means-t-test"><i class="fa fa-check"></i><b>6.11</b> Comparing means: t-test</a></li>
<li class="chapter" data-level="6.12" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#two-sample-t-test"><i class="fa fa-check"></i><b>6.12</b> Two sample t-test</a></li>
<li class="chapter" data-level="6.13" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-wilcoxon-signed-rank-test"><i class="fa fa-check"></i><b>6.13</b> Beyond t-test: Wilcoxon signed rank test</a></li>
<li class="chapter" data-level="6.14" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#tabulation-and-cross-tabulation"><i class="fa fa-check"></i><b>6.14</b> Tabulation and Cross-Tabulation</a></li>
<li class="chapter" data-level="6.15" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-binomial-test"><i class="fa fa-check"></i><b>6.15</b> Beyond t-test: Binomial test</a></li>
<li class="chapter" data-level="6.16" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-chi-square-goodness-of-fit"><i class="fa fa-check"></i><b>6.16</b> Beyond t-test: Chi-square goodness of fit</a></li>
<li class="chapter" data-level="6.17" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-mcnemars-test"><i class="fa fa-check"></i><b>6.17</b> Beyond t-test: McNemar’s test</a></li>
<li class="chapter" data-level="6.18" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-fishers-exact-test"><i class="fa fa-check"></i><b>6.18</b> Beyond t-test: Fisher’s exact test</a></li>
<li class="chapter" data-level="6.19" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#correlations"><i class="fa fa-check"></i><b>6.19</b> Correlations</a></li>
<li class="chapter" data-level="6.20" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#the-magic-criteria"><i class="fa fa-check"></i><b>6.20</b> The MAGIC criteria</a></li>
<li class="chapter" data-level="6.21" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#effect-size"><i class="fa fa-check"></i><b>6.21</b> Effect size</a></li>
<li class="chapter" data-level="6.22" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-distributions"><i class="fa fa-check"></i><b>6.22</b> Comparing distributions</a></li>
<li class="chapter" data-level="6.23" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#bootstraps"><i class="fa fa-check"></i><b>6.23</b> Bootstraps</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html"><i class="fa fa-check"></i><b>7</b> – Regressions and ANOVA in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#linear-regressions"><i class="fa fa-check"></i><b>7.1</b> Linear regressions</a></li>
<li class="chapter" data-level="7.2" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anatomy-of-a-linear-regression"><i class="fa fa-check"></i><b>7.2</b> Anatomy of a linear regression</a></li>
<li class="chapter" data-level="7.3" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#multiple-linear-regressions"><i class="fa fa-check"></i><b>7.3</b> Multiple linear regressions</a></li>
<li class="chapter" data-level="7.4" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#diagnostic-plots-for-regressions"><i class="fa fa-check"></i><b>7.4</b> Diagnostic plots for regressions</a></li>
<li class="chapter" data-level="7.5" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-influential-measurements"><i class="fa fa-check"></i><b>7.5</b> Regression influential measurements</a></li>
<li class="chapter" data-level="7.6" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-coefficients-and-their-95ci"><i class="fa fa-check"></i><b>7.6</b> Regression coefficients and their 95%CI</a></li>
<li class="chapter" data-level="7.7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#selecting-predictors"><i class="fa fa-check"></i><b>7.7</b> Selecting predictors</a></li>
<li class="chapter" data-level="7.8" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-predictions"><i class="fa fa-check"></i><b>7.8</b> Regression predictions</a></li>
<li class="chapter" data-level="7.9" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions"><i class="fa fa-check"></i><b>7.9</b> Logistic regressions</a></li>
<li class="chapter" data-level="7.10" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions-requirements"><i class="fa fa-check"></i><b>7.10</b> Logistic regressions requirements</a></li>
<li class="chapter" data-level="7.11" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#non-linear-regressions"><i class="fa fa-check"></i><b>7.11</b> Non-linear regressions</a></li>
<li class="chapter" data-level="7.12" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-regressions"><i class="fa fa-check"></i><b>7.12</b> Toy datasets for regressions</a></li>
<li class="chapter" data-level="7.13" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova"><i class="fa fa-check"></i><b>7.13</b> ANOVA</a></li>
<li class="chapter" data-level="7.14" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#one-way-anova"><i class="fa fa-check"></i><b>7.14</b> One-way ANOVA</a></li>
<li class="chapter" data-level="7.15" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#two-way-anova"><i class="fa fa-check"></i><b>7.15</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="7.16" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova-post-hoc-tests"><i class="fa fa-check"></i><b>7.16</b> ANOVA Post-hoc tests</a></li>
<li class="chapter" data-level="7.17" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#beyond-anova-kruskal-wallis-test"><i class="fa fa-check"></i><b>7.17</b> Beyond ANOVA: Kruskal-Wallis Test</a></li>
<li class="chapter" data-level="7.18" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-anova"><i class="fa fa-check"></i><b>7.18</b> Toy datasets for ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html"><i class="fa fa-check"></i><b>8</b> – Multidimensional Data in R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#principal-component-analysis"><i class="fa fa-check"></i><b>8.1</b> Principal Component Analysis</a></li>
<li class="chapter" data-level="8.2" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#multidimensional-scaling"><i class="fa fa-check"></i><b>8.2</b> Multidimensional scaling</a></li>
<li class="chapter" data-level="8.3" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-dfa"><i class="fa fa-check"></i><b>8.3</b> Cluster analysis: DFA</a></li>
<li class="chapter" data-level="8.4" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-k-means-unsupervised"><i class="fa fa-check"></i><b>8.4</b> Cluster analysis: <em>k</em>-means (unsupervised)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html"><i class="fa fa-check"></i><b>9</b> – Graphics and Colors</a>
<ul>
<li class="chapter" data-level="9.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#avoid-bad-graphics"><i class="fa fa-check"></i><b>9.1</b> Avoid bad graphics</a></li>
<li class="chapter" data-level="9.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#graphics-in-r-ggplot2"><i class="fa fa-check"></i><b>9.2</b> Graphics in R – ggplot2</a></li>
<li class="chapter" data-level="9.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#preparing-your-dataset"><i class="fa fa-check"></i><b>9.3</b> Preparing your dataset</a></li>
<li class="chapter" data-level="9.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#color-palettes-online"><i class="fa fa-check"></i><b>9.4</b> Color palettes online</a></li>
<li class="chapter" data-level="9.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggplot2-default-palette"><i class="fa fa-check"></i><b>9.5</b> <code>ggplot2</code> default palette</a></li>
<li class="chapter" data-level="9.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#heatmaps-or-tiled-charts"><i class="fa fa-check"></i><b>9.6</b> Heatmaps or tiled charts</a></li>
<li class="chapter" data-level="9.7" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#r-packages-about-color"><i class="fa fa-check"></i><b>9.7</b> R-packages about color</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#viridis-package"><i class="fa fa-check"></i><b>9.7.1</b> <code>viridis</code> package</a></li>
<li class="chapter" data-level="9.7.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcolorbrewer-package"><i class="fa fa-check"></i><b>9.7.2</b> <code>RColorBrewer</code> package</a></li>
<li class="chapter" data-level="9.7.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggsci-package"><i class="fa fa-check"></i><b>9.7.3</b> <code>ggsci</code> package</a></li>
<li class="chapter" data-level="9.7.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#wesanderson-package"><i class="fa fa-check"></i><b>9.7.4</b> <code>wesanderson</code> package</a></li>
<li class="chapter" data-level="9.7.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcartocolor-package"><i class="fa fa-check"></i><b>9.7.5</b> <code>rcartocolor</code> package</a></li>
<li class="chapter" data-level="9.7.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#colorspace-package"><i class="fa fa-check"></i><b>9.7.6</b> <code>colorspace</code> package</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#making-your-own-palette"><i class="fa fa-check"></i><b>9.8</b> Making your own palette</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="basic-graphics.html"><a href="basic-graphics.html"><i class="fa fa-check"></i><b>10</b> – Basic Graphics</a>
<ul>
<li class="chapter" data-level="10.1" data-path="basic-graphics.html"><a href="basic-graphics.html#what-would-you-like-to-show"><i class="fa fa-check"></i><b>10.1</b> What would you like to show?</a></li>
<li class="chapter" data-level="10.2" data-path="basic-graphics.html"><a href="basic-graphics.html#bar-charts"><i class="fa fa-check"></i><b>10.2</b> Bar charts</a></li>
<li class="chapter" data-level="10.3" data-path="basic-graphics.html"><a href="basic-graphics.html#stacked-bar-charts"><i class="fa fa-check"></i><b>10.3</b> Stacked bar charts</a></li>
<li class="chapter" data-level="10.4" data-path="basic-graphics.html"><a href="basic-graphics.html#histograms"><i class="fa fa-check"></i><b>10.4</b> Histograms</a></li>
<li class="chapter" data-level="10.5" data-path="basic-graphics.html"><a href="basic-graphics.html#multiple-histograms"><i class="fa fa-check"></i><b>10.5</b> Multiple histograms</a></li>
<li class="chapter" data-level="10.6" data-path="basic-graphics.html"><a href="basic-graphics.html#density-plots"><i class="fa fa-check"></i><b>10.6</b> Density plots</a></li>
<li class="chapter" data-level="10.7" data-path="basic-graphics.html"><a href="basic-graphics.html#box-plots"><i class="fa fa-check"></i><b>10.7</b> Box plots</a></li>
<li class="chapter" data-level="10.8" data-path="basic-graphics.html"><a href="basic-graphics.html#violin-plots"><i class="fa fa-check"></i><b>10.8</b> Violin plots</a></li>
<li class="chapter" data-level="10.9" data-path="basic-graphics.html"><a href="basic-graphics.html#scatterplots"><i class="fa fa-check"></i><b>10.9</b> Scatterplots</a></li>
<li class="chapter" data-level="10.10" data-path="basic-graphics.html"><a href="basic-graphics.html#lines-and-scatterplots"><i class="fa fa-check"></i><b>10.10</b> Lines and Scatterplots</a></li>
<li class="chapter" data-level="10.11" data-path="basic-graphics.html"><a href="basic-graphics.html#save-your-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>10.11</b> Save your plots as PDF or PNG</a></li>
<li class="chapter" data-level="10.12" data-path="basic-graphics.html"><a href="basic-graphics.html#ggplot2-add-on-packages-treemapify"><i class="fa fa-check"></i><b>10.12</b> ggplot2 add-on packages: <code>treemapify</code></a></li>
<li class="chapter" data-level="10.13" data-path="basic-graphics.html"><a href="basic-graphics.html#other-resources"><i class="fa fa-check"></i><b>10.13</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="sequence-alignments.html"><a href="sequence-alignments.html"><i class="fa fa-check"></i><b>11</b> – Sequence Alignments</a>
<ul>
<li class="chapter" data-level="11.1" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-using-rentrez"><i class="fa fa-check"></i><b>11.1</b> Retrieving sequences using ‘rentrez’</a></li>
<li class="chapter" data-level="11.2" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-manually-from-ncbi"><i class="fa fa-check"></i><b>11.2</b> Retrieving sequences manually from NCBI</a></li>
<li class="chapter" data-level="11.3" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-accession-numbers"><i class="fa fa-check"></i><b>11.3</b> Retrieving accession numbers</a></li>
<li class="chapter" data-level="11.4" data-path="sequence-alignments.html"><a href="sequence-alignments.html#alignment-with-decipher"><i class="fa fa-check"></i><b>11.4</b> Alignment with <code>DECIPHER</code></a></li>
<li class="chapter" data-level="11.5" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualize-alignment"><i class="fa fa-check"></i><b>11.5</b> Visualize alignment</a></li>
<li class="chapter" data-level="11.6" data-path="sequence-alignments.html"><a href="sequence-alignments.html#save-alignment-to-file"><i class="fa fa-check"></i><b>11.6</b> Save alignment to file</a></li>
<li class="chapter" data-level="11.7" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-and-align-several-markers"><i class="fa fa-check"></i><b>11.7</b> Concatenate and align several markers</a></li>
<li class="chapter" data-level="11.8" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-data.table"><i class="fa fa-check"></i><b>11.8</b> Concatenate with <code>data.table</code></a></li>
<li class="chapter" data-level="11.9" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-biostrings"><i class="fa fa-check"></i><b>11.9</b> Concatenate with <code>Biostrings</code></a></li>
<li class="chapter" data-level="11.10" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualizing-alignments-in-r"><i class="fa fa-check"></i><b>11.10</b> Visualizing alignments in R</a></li>
<li class="chapter" data-level="11.11" data-path="sequence-alignments.html"><a href="sequence-alignments.html#other-sequence-alignment-software"><i class="fa fa-check"></i><b>11.11</b> Other sequence alignment software</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html"><i class="fa fa-check"></i><b>12</b> – Estimating Phylogenetic Trees</a>
<ul>
<li class="chapter" data-level="12.1" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#phylogeny-inference-with-iqtree2"><i class="fa fa-check"></i><b>12.1</b> Phylogeny inference with <code>iqtree2</code></a></li>
<li class="chapter" data-level="12.2" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-iqtree2-in-macos"><i class="fa fa-check"></i><b>12.2</b> Installing <code>iqtree2</code> in macOS</a></li>
<li class="chapter" data-level="12.3" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-and-running-iqtree2-in-a-pc"><i class="fa fa-check"></i><b>12.3</b> Installing and running <code>iqtree2</code> in a PC</a></li>
<li class="chapter" data-level="12.4" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#loading-the-iqtree_runner_r-function"><i class="fa fa-check"></i><b>12.4</b> Loading the <code>iqtree_runner_R</code> function</a></li>
<li class="chapter" data-level="12.5" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#running-iqtree2-from-r"><i class="fa fa-check"></i><b>12.5</b> Running <code>iqtree2</code> from R</a></li>
<li class="chapter" data-level="12.6" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-no-partitions"><i class="fa fa-check"></i><b>12.6</b> One marker and no partitions</a></li>
<li class="chapter" data-level="12.7" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-with-partitions"><i class="fa fa-check"></i><b>12.7</b> One marker and with partitions</a></li>
<li class="chapter" data-level="12.8" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#bootstrap-support"><i class="fa fa-check"></i><b>12.8</b> Bootstrap support</a></li>
<li class="chapter" data-level="12.9" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#many-markers-and-with-partitions"><i class="fa fa-check"></i><b>12.9</b> Many markers and with partitions</a></li>
<li class="chapter" data-level="12.10" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#other-sequence-phylogenetic-tree-estimation-software"><i class="fa fa-check"></i><b>12.10</b> Other sequence phylogenetic tree estimation software</a></li>
<li class="chapter" data-level="12.11" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#drawing-phylogenetic-trees-with-ggtree"><i class="fa fa-check"></i><b>12.11</b> Drawing phylogenetic trees with <code>ggtree</code></a></li>
<li class="chapter" data-level="12.12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#save-ggtree-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>12.12</b> Save ggtree plots as PDF or PNG</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html"><i class="fa fa-check"></i><b>13</b> – Mining from a reference dataset</a>
<ul>
<li class="chapter" data-level="13.1" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#a-worked-example-using-the-mtcars-dataset"><i class="fa fa-check"></i><b>13.1</b> A worked example using the <code>mtcars</code> dataset</a></li>
<li class="chapter" data-level="13.2" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-reference-dataset"><i class="fa fa-check"></i><b>13.2</b> Build a reference dataset</a></li>
<li class="chapter" data-level="13.3" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-sampling-dataset"><i class="fa fa-check"></i><b>13.3</b> Build a sampling dataset</a></li>
<li class="chapter" data-level="13.4" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#extracting-from-reference-dataset"><i class="fa fa-check"></i><b>13.4</b> Extracting from reference dataset</a></li>
<li class="chapter" data-level="13.5" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#plots-of-the-extracted-data"><i class="fa fa-check"></i><b>13.5</b> Plots of the extracted data</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="introduction-to-python.html"><a href="introduction-to-python.html"><i class="fa fa-check"></i><b>14</b> – Introduction to Python</a>
<ul>
<li class="chapter" data-level="14.1" data-path="introduction-to-python.html"><a href="introduction-to-python.html#welcome-to-python-why-python"><i class="fa fa-check"></i><b>14.1</b> Welcome to Python! Why Python?</a></li>
<li class="chapter" data-level="14.2" data-path="introduction-to-python.html"><a href="introduction-to-python.html#which-python-should-i-install"><i class="fa fa-check"></i><b>14.2</b> Which Python should I install?</a></li>
<li class="chapter" data-level="14.3" data-path="introduction-to-python.html"><a href="introduction-to-python.html#mac-os-x-python-installations"><i class="fa fa-check"></i><b>14.3</b> Mac OS X Python Installations</a></li>
<li class="chapter" data-level="14.4" data-path="introduction-to-python.html"><a href="introduction-to-python.html#windows-python-installations"><i class="fa fa-check"></i><b>14.4</b> Windows Python Installations</a></li>
<li class="chapter" data-level="14.5" data-path="introduction-to-python.html"><a href="introduction-to-python.html#install-an-ide-or-integrated-development-environment"><i class="fa fa-check"></i><b>14.5</b> Install an IDE, or integrated development environment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html"><i class="fa fa-check"></i><b>15</b> – Python: Keywords and Identifiers</a>
<ul>
<li class="chapter" data-level="15.1" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#keywords-in-python"><i class="fa fa-check"></i><b>15.1</b> Keywords in Python</a></li>
<li class="chapter" data-level="15.2" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#identifiers-in-python"><i class="fa fa-check"></i><b>15.2</b> Identifiers in Python</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html"><i class="fa fa-check"></i><b>16</b> – Python: Statements and Expressions</a>
<ul>
<li class="chapter" data-level="16.1" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#what-are-statements"><i class="fa fa-check"></i><b>16.1</b> What are statements?</a></li>
<li class="chapter" data-level="16.2" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#expressions-in-python"><i class="fa fa-check"></i><b>16.2</b> Expressions in Python</a></li>
<li class="chapter" data-level="16.3" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#simple-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.3</b> Simple Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.4" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#augmented-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.4</b> Augmented Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.5" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#explicit-and-implicit-line-continuation"><i class="fa fa-check"></i><b>16.5</b> Explicit and Implicit Line Continuation</a></li>
<li class="chapter" data-level="16.6" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#indentation-in-python-is-crucial-to-proper-code-function"><i class="fa fa-check"></i><b>16.6</b> Indentation in Python is Crucial to Proper Code function</a></li>
<li class="chapter" data-level="16.7" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#comments"><i class="fa fa-check"></i><b>16.7</b> Comments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="python-data-types.html"><a href="python-data-types.html"><i class="fa fa-check"></i><b>17</b> – Python: Data Types</a>
<ul>
<li class="chapter" data-level="17.1" data-path="python-data-types.html"><a href="python-data-types.html#booleans"><i class="fa fa-check"></i><b>17.1</b> Booleans</a></li>
<li class="chapter" data-level="17.2" data-path="python-data-types.html"><a href="python-data-types.html#numbers-int-float-and-complex"><i class="fa fa-check"></i><b>17.2</b> Numbers: int, float and complex</a></li>
<li class="chapter" data-level="17.3" data-path="python-data-types.html"><a href="python-data-types.html#strings"><i class="fa fa-check"></i><b>17.3</b> Strings</a></li>
<li class="chapter" data-level="17.4" data-path="python-data-types.html"><a href="python-data-types.html#bytes"><i class="fa fa-check"></i><b>17.4</b> Bytes</a></li>
<li class="chapter" data-level="17.5" data-path="python-data-types.html"><a href="python-data-types.html#lists-1"><i class="fa fa-check"></i><b>17.5</b> Lists</a></li>
<li class="chapter" data-level="17.6" data-path="python-data-types.html"><a href="python-data-types.html#tuples"><i class="fa fa-check"></i><b>17.6</b> Tuples</a></li>
<li class="chapter" data-level="17.7" data-path="python-data-types.html"><a href="python-data-types.html#sets"><i class="fa fa-check"></i><b>17.7</b> Sets</a></li>
<li class="chapter" data-level="17.8" data-path="python-data-types.html"><a href="python-data-types.html#dictionary"><i class="fa fa-check"></i><b>17.8</b> Dictionary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="python-strings.html"><a href="python-strings.html"><i class="fa fa-check"></i><b>18</b> – Python: Strings</a>
<ul>
<li class="chapter" data-level="18.1" data-path="python-strings.html"><a href="python-strings.html#string-operators"><i class="fa fa-check"></i><b>18.1</b> String Operators</a></li>
<li class="chapter" data-level="18.2" data-path="python-strings.html"><a href="python-strings.html#other-string-modifiers"><i class="fa fa-check"></i><b>18.2</b> Other String Modifiers</a></li>
<li class="chapter" data-level="18.3" data-path="python-strings.html"><a href="python-strings.html#format-characters-in-strings"><i class="fa fa-check"></i><b>18.3</b> Format-Characters in Strings</a></li>
<li class="chapter" data-level="18.4" data-path="python-strings.html"><a href="python-strings.html#comparisons-with-strings"><i class="fa fa-check"></i><b>18.4</b> Comparisons with Strings</a></li>
<li class="chapter" data-level="18.5" data-path="python-strings.html"><a href="python-strings.html#padding-strings"><i class="fa fa-check"></i><b>18.5</b> Padding strings</a></li>
<li class="chapter" data-level="18.6" data-path="python-strings.html"><a href="python-strings.html#search-for-words-in-string-and-output-location"><i class="fa fa-check"></i><b>18.6</b> Search for words in string and output location</a></li>
<li class="chapter" data-level="18.7" data-path="python-strings.html"><a href="python-strings.html#seek-and-replace-in-strings"><i class="fa fa-check"></i><b>18.7</b> Seek and replace in strings</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="python-operators.html"><a href="python-operators.html"><i class="fa fa-check"></i><b>19</b> – Python: Operators</a>
<ul>
<li class="chapter" data-level="19.1" data-path="python-operators.html"><a href="python-operators.html#many-types-of-operators"><i class="fa fa-check"></i><b>19.1</b> Many types of operators</a></li>
<li class="chapter" data-level="19.2" data-path="python-operators.html"><a href="python-operators.html#arithmetic-operators"><i class="fa fa-check"></i><b>19.2</b> Arithmetic operators</a></li>
<li class="chapter" data-level="19.3" data-path="python-operators.html"><a href="python-operators.html#comparison-operators"><i class="fa fa-check"></i><b>19.3</b> Comparison operators</a></li>
<li class="chapter" data-level="19.4" data-path="python-operators.html"><a href="python-operators.html#logical-operators"><i class="fa fa-check"></i><b>19.4</b> Logical operators</a></li>
<li class="chapter" data-level="19.5" data-path="python-operators.html"><a href="python-operators.html#bitwise-operators"><i class="fa fa-check"></i><b>19.5</b> Bitwise operators</a></li>
<li class="chapter" data-level="19.6" data-path="python-operators.html"><a href="python-operators.html#identity-operators"><i class="fa fa-check"></i><b>19.6</b> Identity Operators</a></li>
<li class="chapter" data-level="19.7" data-path="python-operators.html"><a href="python-operators.html#membership-operators"><i class="fa fa-check"></i><b>19.7</b> Membership operators</a></li>
<li class="chapter" data-level="19.8" data-path="python-operators.html"><a href="python-operators.html#what-is-the-order-of-operations-with-operators"><i class="fa fa-check"></i><b>19.8</b> What is the order of operations with operators?</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="biopython.html"><a href="biopython.html"><i class="fa fa-check"></i><b>20</b> – Biopython</a>
<ul>
<li class="chapter" data-level="20.1" data-path="biopython.html"><a href="biopython.html#introduction-to-biopython"><i class="fa fa-check"></i><b>20.1</b> Introduction to Biopython</a></li>
<li class="chapter" data-level="20.2" data-path="biopython.html"><a href="biopython.html#biopython-installation"><i class="fa fa-check"></i><b>20.2</b> Biopython Installation</a></li>
<li class="chapter" data-level="20.3" data-path="biopython.html"><a href="biopython.html#getting-started"><i class="fa fa-check"></i><b>20.3</b> Getting Started</a></li>
<li class="chapter" data-level="20.4" data-path="biopython.html"><a href="biopython.html#parsing-my-fasta"><i class="fa fa-check"></i><b>20.4</b> Parsing my FASTA</a></li>
<li class="chapter" data-level="20.5" data-path="biopython.html"><a href="biopython.html#parsing-my-genbank-or-genpept"><i class="fa fa-check"></i><b>20.5</b> Parsing my GenBank or GenPept</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="biopython-sequences.html"><a href="biopython-sequences.html"><i class="fa fa-check"></i><b>21</b> – Biopython: Sequences</a>
<ul>
<li class="chapter" data-level="21.1" data-path="biopython-sequences.html"><a href="biopython-sequences.html#sequence-objects"><i class="fa fa-check"></i><b>21.1</b> Sequence Objects</a></li>
<li class="chapter" data-level="21.2" data-path="biopython-sequences.html"><a href="biopython-sequences.html#concatenation-with-sequences"><i class="fa fa-check"></i><b>21.2</b> Concatenation with Sequences</a></li>
<li class="chapter" data-level="21.3" data-path="biopython-sequences.html"><a href="biopython-sequences.html#transcription-dna-to-mrna"><i class="fa fa-check"></i><b>21.3</b> Transcription: DNA to mRNA</a></li>
<li class="chapter" data-level="21.4" data-path="biopython-sequences.html"><a href="biopython-sequences.html#translation-rna-to-prot"><i class="fa fa-check"></i><b>21.4</b> Translation: RNA to PROT</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html"><i class="fa fa-check"></i><b>22</b> – Biopython: Mulitple Sequence Alignments</a>
<ul>
<li class="chapter" data-level="22.1" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwise2"><i class="fa fa-check"></i><b>22.1</b> pairwise2</a></li>
<li class="chapter" data-level="22.2" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwisealigner"><i class="fa fa-check"></i><b>22.2</b> PairwiseAligner</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html"><i class="fa fa-check"></i><b>23</b> – Biopython: GenomeDiagram</a>
<ul>
<li class="chapter" data-level="23.1" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#linearcircular-gene-diagrams"><i class="fa fa-check"></i><b>23.1</b> Linear/Circular Gene Diagrams</a></li>
<li class="chapter" data-level="23.2" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#diagram-readability"><i class="fa fa-check"></i><b>23.2</b> Diagram Readability</a></li>
<li class="chapter" data-level="23.3" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#box-shape"><i class="fa fa-check"></i><b>23.3</b> Box Shape</a></li>
<li class="chapter" data-level="23.4" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#edit-box-shape-arrows"><i class="fa fa-check"></i><b>23.4</b> Edit Box Shape: Arrows</a></li>
<li class="chapter" data-level="23.5" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#restriction-map"><i class="fa fa-check"></i><b>23.5</b> Restriction Map</a></li>
<li class="chapter" data-level="23.6" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#tracks-on-tracks"><i class="fa fa-check"></i><b>23.6</b> Tracks on Tracks</a></li>
<li class="chapter" data-level="23.7" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#cross-link-data"><i class="fa fa-check"></i><b>23.7</b> Cross-link Data</a></li>
<li class="chapter" data-level="23.8" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#clean-cross-link"><i class="fa fa-check"></i><b>23.8</b> Clean Cross-Link</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html"><i class="fa fa-check"></i><b>24</b> – Biopython: Chromosome Modeling</a>
<ul>
<li class="chapter" data-level="24.1" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html#gene-feature-modeling"><i class="fa fa-check"></i><b>24.1</b> Gene Feature Modeling</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html"><i class="fa fa-check"></i><b>25</b> – Biopython: Sequence Parsing with Plots</a>
<ul>
<li class="chapter" data-level="25.1" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#histograms-1"><i class="fa fa-check"></i><b>25.1</b> Histograms</a></li>
<li class="chapter" data-level="25.2" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#gc-content"><i class="fa fa-check"></i><b>25.2</b> GC Content</a></li>
<li class="chapter" data-level="25.3" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#nt-alignment-dot-plot"><i class="fa fa-check"></i><b>25.3</b> Nt Alignment Dot Plot</a></li>
<li class="chapter" data-level="25.4" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#bigger-dots-for-the-dot-plot"><i class="fa fa-check"></i><b>25.4</b> Bigger Dots for the Dot Plot</a></li>
<li class="chapter" data-level="25.5" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#fastq-we-did-not-forget-you"><i class="fa fa-check"></i><b>25.5</b> FASTQ we did not forget you</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html"><i class="fa fa-check"></i><b>26</b> – Python: Web Scraping w/ BeautifulSoup</a>
<ul>
<li class="chapter" data-level="26.1" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#introduction"><i class="fa fa-check"></i><b>26.1</b> Introduction</a></li>
<li class="chapter" data-level="26.2" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#installation"><i class="fa fa-check"></i><b>26.2</b> Installation</a></li>
<li class="chapter" data-level="26.3" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#getting-started-1"><i class="fa fa-check"></i><b>26.3</b> Getting started</a></li>
<li class="chapter" data-level="26.4" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-all-matching-tags"><i class="fa fa-check"></i><b>26.4</b> Mining all matching tags</a></li>
<li class="chapter" data-level="26.5" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-tags-by-class-or-id"><i class="fa fa-check"></i><b>26.5</b> Mining tags by class or id</a></li>
<li class="chapter" data-level="26.6" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#use-css-selectors-for-positional-mining"><i class="fa fa-check"></i><b>26.6</b> Use CSS Selectors for positional mining</a></li>
<li class="chapter" data-level="26.7" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#weather-page-example"><i class="fa fa-check"></i><b>26.7</b> Weather Page Example</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="python-subprocess.html"><a href="python-subprocess.html"><i class="fa fa-check"></i><b>27</b> – Python Subprocess</a>
<ul>
<li class="chapter" data-level="27.1" data-path="python-subprocess.html"><a href="python-subprocess.html#introduction-and-installation"><i class="fa fa-check"></i><b>27.1</b> Introduction and Installation</a></li>
<li class="chapter" data-level="27.2" data-path="python-subprocess.html"><a href="python-subprocess.html#creating-your-first-python-script-w-bbtools"><i class="fa fa-check"></i><b>27.2</b> Creating your first python script w/ BBTools</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html"><i class="fa fa-check"></i><b>28</b> – Control Structures &amp; Loops in R</a>
<ul>
<li class="chapter" data-level="28.1" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#controling-the-flow"><i class="fa fa-check"></i><b>28.1</b> Controling the flow</a></li>
<li class="chapter" data-level="28.2" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if"><i class="fa fa-check"></i><b>28.2</b> Conditional <code>if</code></a></li>
<li class="chapter" data-level="28.3" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if-and-else"><i class="fa fa-check"></i><b>28.3</b> Conditional <code>if</code> and <code>else</code></a></li>
<li class="chapter" data-level="28.4" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#nested-conditionals-if-else-and-if-else"><i class="fa fa-check"></i><b>28.4</b> Nested conditionals <code>if</code>, <code>else</code>, and <code>if else</code></a></li>
<li class="chapter" data-level="28.5" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-ifelse"><i class="fa fa-check"></i><b>28.5</b> Conditional <code>ifelse</code></a></li>
<li class="chapter" data-level="28.6" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#loops-that-control-iterations-and-repeations"><i class="fa fa-check"></i><b>28.6</b> Loops that control iterations and repeations</a></li>
<li class="chapter" data-level="28.7" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-for-loop"><i class="fa fa-check"></i><b>28.7</b> The <code>for</code> Loop</a></li>
<li class="chapter" data-level="28.8" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-while-loop"><i class="fa fa-check"></i><b>28.8</b> The <code>while</code> Loop</a></li>
<li class="chapter" data-level="28.9" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-repeat-loop"><i class="fa fa-check"></i><b>28.9</b> The <code>repeat</code> Loop</a></li>
<li class="chapter" data-level="28.10" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-next-and-break-statements"><i class="fa fa-check"></i><b>28.10</b> The <code>next</code> and <code>break</code> statements</a></li>
<li class="chapter" data-level="28.11" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functions"><i class="fa fa-check"></i><b>28.11</b> Other loop functions</a></li>
<li class="chapter" data-level="28.12" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#lapply-functions"><i class="fa fa-check"></i><b>28.12</b> <code>lapply</code> functions</a></li>
<li class="chapter" data-level="28.13" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsapply"><i class="fa fa-check"></i><b>28.13</b> Other loop functions:<code>apply</code></a></li>
<li class="chapter" data-level="28.14" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsmapply"><i class="fa fa-check"></i><b>28.14</b> Other loop functions:<code>mapply</code></a></li>
<li class="chapter" data-level="28.15" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionstapply"><i class="fa fa-check"></i><b>28.15</b> Other loop functions:<code>tapply</code></a></li>
<li class="chapter" data-level="28.16" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionssplit"><i class="fa fa-check"></i><b>28.16</b> Other loop functions:<code>split</code></a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html"><i class="fa fa-check"></i><b>29</b> – Bioinformatics Literature</a>
<ul>
<li class="chapter" data-level="29.1" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#peer-reviewed-bioinformatics-journals"><i class="fa fa-check"></i><b>29.1</b> Peer-reviewed bioinformatics journals</a></li>
<li class="chapter" data-level="29.2" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#access-to-journals-articles"><i class="fa fa-check"></i><b>29.2</b> Access to journals articles</a></li>
<li class="chapter" data-level="29.3" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#informal-access-to-code-preprints-and-vignettes"><i class="fa fa-check"></i><b>29.3</b> Informal access to code: Preprints and vignettes</a></li>
<li class="chapter" data-level="29.4" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#online-bioinformatics-courses"><i class="fa fa-check"></i><b>29.4</b> Online bioinformatics courses</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html"><i class="fa fa-check"></i><b>30</b> – Importing/Exporting Data into R</a>
<ul>
<li class="chapter" data-level="30.1" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#typing-directly"><i class="fa fa-check"></i><b>30.1</b> Typing directly</a></li>
<li class="chapter" data-level="30.2" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#summary-versus-raw-data"><i class="fa fa-check"></i><b>30.2</b> Summary versus raw data</a></li>
<li class="chapter" data-level="30.3" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#downloading-data-files-github"><i class="fa fa-check"></i><b>30.3</b> Downloading data files (GitHub)</a></li>
<li class="chapter" data-level="30.4" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#importing-data-from-a-file"><i class="fa fa-check"></i><b>30.4</b> Importing data from a file</a></li>
<li class="chapter" data-level="30.5" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-to-a-file"><i class="fa fa-check"></i><b>30.5</b> Saving data to a file</a></li>
<li class="chapter" data-level="30.6" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-as-rdata-.rdata-.rda"><i class="fa fa-check"></i><b>30.6</b> Saving data as Rdata (.rdata, .rda)</a></li>
<li class="chapter" data-level="30.7" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#import-excel-formated-data"><i class="fa fa-check"></i><b>30.7</b> Import Excel formated data</a></li>
<li class="chapter" data-level="30.8" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#save-data-as-excel-formated-file"><i class="fa fa-check"></i><b>30.8</b> Save data as Excel formated file</a></li>
<li class="chapter" data-level="30.9" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#fixing-problems-while-importing-datasets"><i class="fa fa-check"></i><b>30.9</b> Fixing problems while importing datasets</a></li>
<li class="chapter" data-level="30.10" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#simplifying-embellished-excel-datasets"><i class="fa fa-check"></i><b>30.10</b> Simplifying embellished excel datasets</a></li>
<li class="chapter" data-level="30.11" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#accessing-online-data-dryad"><i class="fa fa-check"></i><b>30.11</b> Accessing online data: DRYAD</a></li>
<li class="chapter" data-level="30.12" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#accessing-online-data-gene-expression-omnibus-ncbi-geo"><i class="fa fa-check"></i><b>30.12</b> Accessing online data: Gene Expression Omnibus (NCBI GEO)</a></li>
<li class="chapter" data-level="30.13" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#accessing-online-data-clinicaltrials.gov"><i class="fa fa-check"></i><b>30.13</b> Accessing online data: clinicaltrials.gov</a></li>
<li class="chapter" data-level="30.14" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#accessing-online-data-data-scrapping-from-a-website"><i class="fa fa-check"></i><b>30.14</b> Accessing online data: Data scrapping from a website</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="working-with-proteins.html"><a href="working-with-proteins.html"><i class="fa fa-check"></i><b>31</b> – Working with Proteins</a>
<ul>
<li class="chapter" data-level="31.1" data-path="working-with-proteins.html"><a href="working-with-proteins.html#sequences-in-fasta-format"><i class="fa fa-check"></i><b>31.1</b> Sequences in FASTA format</a></li>
<li class="chapter" data-level="31.2" data-path="working-with-proteins.html"><a href="working-with-proteins.html#fetching-aa-sequences-from-ncbi"><i class="fa fa-check"></i><b>31.2</b> Fetching AA sequences from NCBI</a></li>
<li class="chapter" data-level="31.3" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-aa-sequences-in-r"><i class="fa fa-check"></i><b>31.3</b> Importing AA sequences in R</a></li>
<li class="chapter" data-level="31.4" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-nucleotide-sequences-in-r"><i class="fa fa-check"></i><b>31.4</b> Importing nucleotide sequences in R</a></li>
<li class="chapter" data-level="31.5" data-path="working-with-proteins.html"><a href="working-with-proteins.html#translating-a-nucleotides-to-aa"><i class="fa fa-check"></i><b>31.5</b> Translating a nucleotides to AA</a></li>
<li class="chapter" data-level="31.6" data-path="working-with-proteins.html"><a href="working-with-proteins.html#finding-orfs-in-a-nucleotide-sequence"><i class="fa fa-check"></i><b>31.6</b> Finding ORFs in a nucleotide sequence</a></li>
<li class="chapter" data-level="31.7" data-path="working-with-proteins.html"><a href="working-with-proteins.html#physochemical-properties-of-an-amino-acid-sequence"><i class="fa fa-check"></i><b>31.7</b> Physochemical properties of an amino acid sequence</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="miscellaneous.html"><a href="miscellaneous.html"><i class="fa fa-check"></i><b>32</b> – Miscellaneous</a>
<ul>
<li class="chapter" data-level="32.1" data-path="miscellaneous.html"><a href="miscellaneous.html#sampling-from-a-vector-of-names"><i class="fa fa-check"></i><b>32.1</b> Sampling from a vector of names</a></li>
<li class="chapter" data-level="32.2" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-an-article-pdf"><i class="fa fa-check"></i><b>32.2</b> Read a text from an article pdf</a></li>
<li class="chapter" data-level="32.3" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-website"><i class="fa fa-check"></i><b>32.3</b> Read a text from a website</a></li>
<li class="chapter" data-level="32.4" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-file"><i class="fa fa-check"></i><b>32.4</b> Read a text from a file</a></li>
<li class="chapter" data-level="32.5" data-path="miscellaneous.html"><a href="miscellaneous.html#generate-the-word-cloud"><i class="fa fa-check"></i><b>32.5</b> Generate the word cloud</a></li>
<li class="chapter" data-level="32.6" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-pdf"><i class="fa fa-check"></i><b>32.6</b> worldcloud – PDF</a></li>
<li class="chapter" data-level="32.7" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-website"><i class="fa fa-check"></i><b>32.7</b> Worldcloud – website</a></li>
<li class="chapter" data-level="32.8" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-text"><i class="fa fa-check"></i><b>32.8</b> Worldcloud – text</a></li>
<li class="chapter" data-level="32.9" data-path="miscellaneous.html"><a href="miscellaneous.html#more-miscellaneous"><i class="fa fa-check"></i><b>32.9</b> More miscellaneous</a></li>
</ul></li>
<li class="chapter" data-level="33" data-path="galaxy-getting-started-part-i.html"><a href="galaxy-getting-started-part-i.html"><i class="fa fa-check"></i><b>33</b> – Galaxy: Getting Started (Part I)</a>
<ul>
<li class="chapter" data-level="33.1" data-path="galaxy-getting-started-part-i.html"><a href="galaxy-getting-started-part-i.html#introduction-to-galaxy"><i class="fa fa-check"></i><b>33.1</b> Introduction to Galaxy</a></li>
<li class="chapter" data-level="33.2" data-path="galaxy-getting-started-part-i.html"><a href="galaxy-getting-started-part-i.html#hands-on-activity-lets-navigate-galaxy"><i class="fa fa-check"></i><b>33.2</b> Hands-on Activity: Let’s navigate Galaxy!</a></li>
<li class="chapter" data-level="33.3" data-path="galaxy-getting-started-part-i.html"><a href="galaxy-getting-started-part-i.html#galaxy-training-why-should-we-all-do-it"><i class="fa fa-check"></i><b>33.3</b> Galaxy Training: Why should we all do it?</a></li>
<li class="chapter" data-level="33.4" data-path="galaxy-getting-started-part-i.html"><a href="galaxy-getting-started-part-i.html#hands-on-activity-uploading-files-understanding-metadata-and-running-a-simple-workflow"><i class="fa fa-check"></i><b>33.4</b> Hands-on Activity: Uploading files, understanding metadata, and running a simple workflow</a></li>
</ul></li>
<li class="chapter" data-level="34" data-path="galaxy-workflows-part-ii.html"><a href="galaxy-workflows-part-ii.html"><i class="fa fa-check"></i><b>34</b> – Galaxy: Workflows (Part II)</a>
<ul>
<li class="chapter" data-level="34.1" data-path="galaxy-workflows-part-ii.html"><a href="galaxy-workflows-part-ii.html#introduction-to-workflows"><i class="fa fa-check"></i><b>34.1</b> Introduction to Workflows</a></li>
<li class="chapter" data-level="34.2" data-path="galaxy-workflows-part-ii.html"><a href="galaxy-workflows-part-ii.html#export-existing-workflows"><i class="fa fa-check"></i><b>34.2</b> Export existing workflows</a></li>
<li class="chapter" data-level="34.3" data-path="galaxy-workflows-part-ii.html"><a href="galaxy-workflows-part-ii.html#build-a-complete-history-made-genomic-aligning-workflow"><i class="fa fa-check"></i><b>34.3</b> Build a complete history-made genomic aligning workflow</a></li>
<li class="chapter" data-level="34.4" data-path="galaxy-workflows-part-ii.html"><a href="galaxy-workflows-part-ii.html#citing-and-sharing-workflows"><i class="fa fa-check"></i><b>34.4</b> Citing and sharing workflows</a></li>
<li class="chapter" data-level="34.5" data-path="galaxy-workflows-part-ii.html"><a href="galaxy-workflows-part-ii.html#pros-and-cons-of-galaxy"><i class="fa fa-check"></i><b>34.5</b> Pros and cons of Galaxy</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics Workshop Gitbook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="importingexporting-data-into-r" class="section level1 hasAnchor" number="30">
<h1><span class="header-section-number">Session 30</span> – Importing/Exporting Data into R<a href="importingexporting-data-into-r.html#importingexporting-data-into-r" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Importing data is crucial for analyses in R as most of these are produced, retrieved or generated by other software. For example, measurements might be collected and annotated in excel sheets, sequences might be encoded in fasta/fastq forma in text files, images are stored as .tiff, .jpeg or other formats, sound might be encoded in .wav, etc. A great advantage of R is that the same base software or R-packages can serve to bring those datasets into the R environment. However, most of these are imported as standard R data structures as data frames (i.e., similar to matrices with columns and rows that contain a single value). Next, we will survey the most common forms to data input</p>
<div id="typing-directly" class="section level2 hasAnchor" number="30.1">
<h2><span class="header-section-number">30.1</span> Typing directly<a href="importingexporting-data-into-r.html#typing-directly" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The simples way to input data in R by typing data in the R console. We explain this in more detail when we describe data structures. However, this should intuitive as follows.</p>
<div class="sourceCode" id="cb644"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb644-1"><a href="importingexporting-data-into-r.html#cb644-1" aria-hidden="true" tabindex="-1"></a><span class="do">## create a vector with some values named &#39;my_vector&#39;</span></span>
<span id="cb644-2"><a href="importingexporting-data-into-r.html#cb644-2" aria-hidden="true" tabindex="-1"></a>my_vector <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">2</span><span class="sc">:</span><span class="dv">20</span>,<span class="dv">99</span>)</span>
<span id="cb644-3"><a href="importingexporting-data-into-r.html#cb644-3" aria-hidden="true" tabindex="-1"></a>my_vector</span>
<span id="cb644-4"><a href="importingexporting-data-into-r.html#cb644-4" aria-hidden="true" tabindex="-1"></a><span class="co">#[1]  1  4  5  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 99</span></span></code></pre></div>
</div>
<div id="summary-versus-raw-data" class="section level2 hasAnchor" number="30.2">
<h2><span class="header-section-number">30.2</span> Summary versus raw data<a href="importingexporting-data-into-r.html#summary-versus-raw-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Data summaries are usually presented as tables, graphs, or diagrams with a list of descriptive (summary) statistics to represent or summarize the tendencies on the raw dataset. You can identify such summary tables if they present counts and statistics such as number of individuals in a treatment and control group (sample size), centrality statistics or averages (e.g., mean, median, and mode); measures of variability (e.g., standard deviation, standard error, confidence intervals, quartiles, minimum, and maximum values); bias on the distribution of the raw data (e.g., skewness, tendencies); and statistical dependence (if more than one variable was captured in the dataset). In most cases, such data summaries do not make inferences about the data and its ability to prove or disprove a research question.
More information can be found <a href="https://toolkit.ncats.nih.gov/glossary/data-summary/" target="_blank"><strong>here</strong></a>.</p>
<p>Here is an example of a summary table from this <a href="https://pubmed.ncbi.nlm.nih.gov/24005131/" target="_blank"><strong>article</strong></a>:</p>
<p><img src="figures/session_14_figs/16_summary_table.png" width="100%" /></p>
<p>Raw data is some cases is hard to find and are not usually presented in the research article per se, but as supplementary document or as link to a dedicated data archive such as
<a href="https://datadryad.org/stash" target="_blank"><strong>DRYAD</strong></a>,
<a href="https://zenodo.org/" target="_blank"><strong>Zenodo</strong></a>,
<a href="https://figshare.com/" target="_blank"><strong>Figshare</strong></a>, and
<a href="https://www.ncbi.nlm.nih.gov/" target="_blank"><strong>NCBI</strong></a>.</p>
<p>However, most scientists agree to increase the openness, integrity, and reproducibility of all scientific research. For this purpose, all used data to derive a research results and conclusions should be publicly accessible to anyone that wants to reproduce the results of a particular study. Therefore, raw data (i.e., sequences, individual measurements, metadata, etc) should be archived and its contents described for posterity. This is a relatively new tendency and emerged on early 2010’s after the reproducibility project aimed to reproduce the results of published from three leading psychology journals. In the original study, researchers volunteered to replicate a study of their choosing from these journals. Not surprisingly, the results were not astonishing and published in 2015 (see <a href="https://pubmed.ncbi.nlm.nih.gov/26315443/" target="_blank"><strong>Abstract</strong></a>) and showed
that the “…Replication effects were half the magnitude of original effects, representing a substantial decline. Ninety-seven percent of original studies had statistically significant results. Thirty-six percent of replications had statistically significant results; 47% of original effect sizes were in the 95% confidence interval of the replication effect size; 39% of effects were subjectively rated to have replicated the original result; and if no bias in original results is assumed, combining original and replication results left 68% with statistically significant effects…”
Therefore, there is substantial decline on the reproductively of older studies (see <a href="https://pubmed.ncbi.nlm.nih.gov/26315412/" target="_blank"><strong>commentary</strong></a>) and a new tendency has started to require that most data should be made public if an author desire his/her research published.</p>
<p>Here is an example of the raw table from the same <a href="https://pubmed.ncbi.nlm.nih.gov/24005131/" target="_blank"><strong>article</strong></a> in the <a href="https://datadryad.org/stash/dataset/doi:10.5061%2Fdryad.8bv8p" target="_blank"><strong>DRYAD repository</strong></a>:</p>
<p><img src="figures/session_14_figs/17_raw_data_table.png" width="100%" /></p>
</div>
<div id="downloading-data-files-github" class="section level2 hasAnchor" number="30.3">
<h2><span class="header-section-number">30.3</span> Downloading data files (GitHub)<a href="importingexporting-data-into-r.html#downloading-data-files-github" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>IMPORTANT</strong> I uploaded some files of this class, you can access those and download them from this <a href="https://github.com/corytophanes/BIO_BIT_Bioinformatics_209/tree/main/ref_files_pdfs" target="_blank"><strong>LINK TO FILES of BIO/BIT 209</strong></a> to a folder of your choosing.</p>
<p>You can <strong>download individual files</strong> based in their extension. In the case of text files (e.g., <strong>.csv</strong>, <strong>.fasta</strong>, <strong>.txt</strong>), you can follow these steps:</p>
<p><strong>1)</strong> Go to the file you want to download. Click on the file that you want to download. You will see the file contents.</p>
<p><img src="figures/session_2_figs/2_file_to_donwload_txt2.png" width="100%" /></p>
<p><strong>2)</strong> In the top right, you will see a tab that says <code>Raw</code> and click on this button.</p>
<p><img src="figures/session_2_figs/3_file_to_donwload_raw.png" width="100%" /></p>
<p><strong>3)</strong> A new tab will appear and you can save this file as <code>your_file.txt</code> (if this is fasta or csv, use the appropriate extension). In Firefox (MACs), you will click on <strong>File &gt; Save page as…</strong> and can name this file as <code>your_file.txt</code></p>
<p><img src="figures/session_2_figs/4_file_to_donwload_newpage.png" width="100%" />
<strong>4)</strong> You probably want to have a working directory where you will store these files. You should make these folders with easy and accessible paths (e.g., in your desktop).</p>
<p>In case of <strong>other files </strong>like PDFs, images, word, and excel files (e.g., <strong>.pdf</strong>, <strong>.jpg</strong>, <strong>.png</strong>, <strong>.xlxs</strong>, <strong>.docx</strong>), you can download those individually with the following steps:</p>
<p><strong>4.1)</strong> Go to the file you want to download.</p>
<p><img src="figures/session_2_figs/5_file_to_donwload_PDF2.png" width="100%" /></p>
<p><strong>4.2)</strong> Click on the file that you want to download. You will see a tab that says <code>Download</code> and click on this button.</p>
<p><img src="figures/session_2_figs/6_file_to_donwload_PDF_download.png" width="100%" /></p>
<p><strong>4.3)</strong> A new tab will appear and you can save this file as <code>your_file</code> and select the extension that best suits this file. In Firefox (MACs), you will click on <strong>File &gt; Save page as…</strong> and can name this file as <code>your_file.txt</code></p>
<p><strong>4.4)</strong> You probably want to have a working directory where you will store these files. You should make these folders with easy and accessible paths (e.g., in your desktop).</p>
<p>You can <strong>download an entire GitHub repository</strong>. You can download these archives from the main repository by clicking on <code>Code</code>. A submenu will appear, and you can click on <code>Download ZIP</code> and you can place this uncompressed folder in your desktop or other working directory.</p>
<p><img src="figures/session_2_figs/7_download_repository.png" width="100%" /></p>
</div>
<div id="importing-data-from-a-file" class="section level2 hasAnchor" number="30.4">
<h2><span class="header-section-number">30.4</span> Importing data from a file<a href="importingexporting-data-into-r.html#importing-data-from-a-file" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>IMPORTANT</strong> I uploaded some files of this class, you can access those and download them from this <a href="https://github.com/corytophanes/BIO_BIT_Bioinformatics_209/tree/main/ref_files_pdfs" target="_blank"><strong>LINK TO FILES of BIO/BIT 209</strong></a> to a folder of your choosing.</p>
<p><strong>1)</strong> In most cases, you already have your data file in some folder in your computer or plan to import it from some online repository. In the case of importing data from file from a known folder in your computer, this can be done as easily in R.</p>
<p>First, I will describe to import data from a text file with a tab delimited columns with numeric and character values.</p>
<p><strong>macOS:</strong> You need to get the path (folder location in your hard drive) so you can indicate R where to retrieve such information. For example, the file that you want is known folder. You just <strong>drag &amp; drop</strong> the folder in the R console and the path information will appear on it.</p>
<div class="sourceCode" id="cb645"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb645-1"><a href="importingexporting-data-into-r.html#cb645-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ~/Desktop/Teach_R/class_datasets</span></span>
<span id="cb645-2"><a href="importingexporting-data-into-r.html#cb645-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Next, we can explore what files are inside such folder. </span></span>
<span id="cb645-3"><a href="importingexporting-data-into-r.html#cb645-3" aria-hidden="true" tabindex="-1"></a><span class="co"># </span><span class="al">NOTE</span><span class="co"> THAT THIS PATH IS SPECIFIC FOR YOUR COMPUTER</span></span>
<span id="cb645-4"><a href="importingexporting-data-into-r.html#cb645-4" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/class_datasets&quot;</span>)</span>
<span id="cb645-5"><a href="importingexporting-data-into-r.html#cb645-5" aria-hidden="true" tabindex="-1"></a>path_to_my_datafiles <span class="ot">&lt;-</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets&quot;</span></span></code></pre></div>
<p>Note: You can get the path directly to the file by <strong>drag &amp; drop</strong> the file into the R console.</p>
<p><strong>Windows:</strong> For PCs, to get the path (folder location in your hard drive) so you can indicate R where to retrieve such information. For example, the file that you want is known folder. You need to find the path to this working directory. For this you need to highlight this folder, hold down the <strong>Shift</strong> key and <strong>right-click</strong> the file.</p>
<p>Then, you click on <strong>Properties</strong> and copy the names (text) of <strong>Location</strong> that includes the folder and file names. For example, if you want to change your working directory to the folder icon <code>class_datasets</code> (this is a made-up folder, you can name it as you wish or might one specific one).</p>
<p>Find its location: <code>C:\Users\myPC\Desktop</code></p>
<p>You will get its path name as: <code>C:\Users\myPC\Desktop\my_working_directory\class_datasets</code></p>
<p>Then use the <code>setwd()</code> function as follows</p>
<div class="sourceCode" id="cb646"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb646-1"><a href="importingexporting-data-into-r.html#cb646-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;C:\Users\myPC\Desktop\my_working_directory\class_datasets&quot;</span>)</span></code></pre></div>
<p>In some PCs, this will change the working directory to its desired path. However, in most, you will get the following error</p>
<div class="sourceCode" id="cb647"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb647-1"><a href="importingexporting-data-into-r.html#cb647-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;C:\Users\myPC\Desktop\my_working_directory\class_datasets&quot;</span>)</span>
<span id="cb647-2"><a href="importingexporting-data-into-r.html#cb647-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Error: &#39;\U&#39; used without hex digits in character string starting &quot;&quot;C:\U&quot;</span></span></code></pre></div>
<p>This indicates that the R is recognizing the <code>\</code> (backslashes) as part of regular expressions. To make it work, you will need to change such <code>\</code> character to <code>/</code> (forward slashes).</p>
<div class="sourceCode" id="cb648"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb648-1"><a href="importingexporting-data-into-r.html#cb648-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;C:/Users/myPC/Desktop/my_working_directory/class_datasets&quot;</span>)</span>
<span id="cb648-2"><a href="importingexporting-data-into-r.html#cb648-2" aria-hidden="true" tabindex="-1"></a><span class="fu">getwd</span>()</span>
<span id="cb648-3"><a href="importingexporting-data-into-r.html#cb648-3" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;C:/Users/myPC/Desktop/my_working_directory/class_datasets&quot;</span></span></code></pre></div>
<p>Other have reported is to use double backslashes <code>\\</code></p>
<div class="sourceCode" id="cb649"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb649-1"><a href="importingexporting-data-into-r.html#cb649-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;C:</span><span class="sc">\\</span><span class="st">Users</span><span class="sc">\\</span><span class="st">myPC</span><span class="sc">\\</span><span class="st">Desktop</span><span class="sc">\\</span><span class="st">my_working_directory</span><span class="sc">\\</span><span class="st">class_datasets&quot;</span>)</span>
<span id="cb649-2"><a href="importingexporting-data-into-r.html#cb649-2" aria-hidden="true" tabindex="-1"></a><span class="fu">getwd</span>()</span>
<span id="cb649-3"><a href="importingexporting-data-into-r.html#cb649-3" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;C:/Users/myPC/Desktop/my_working_directory/class_datasets &quot;</span></span></code></pre></div>
<p>Note: Like in macOS, you can apply the same protocol for a file rather than a folder.</p>
<p><strong>2)</strong> You can get a list of files inside that working directory using <code>file.list()</code>.</p>
<div class="sourceCode" id="cb650"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb650-1"><a href="importingexporting-data-into-r.html#cb650-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/class_datasets&quot;</span>)</span>
<span id="cb650-2"><a href="importingexporting-data-into-r.html#cb650-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb650-3"><a href="importingexporting-data-into-r.html#cb650-3" aria-hidden="true" tabindex="-1"></a>path_to_my_datafiles <span class="ot">&lt;-</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets&quot;</span></span>
<span id="cb650-4"><a href="importingexporting-data-into-r.html#cb650-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb650-5"><a href="importingexporting-data-into-r.html#cb650-5" aria-hidden="true" tabindex="-1"></a><span class="fu">list.files</span>(<span class="at">path =</span> path_to_my_datafiles, </span>
<span id="cb650-6"><a href="importingexporting-data-into-r.html#cb650-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">pattern =</span> <span class="cn">NULL</span>,</span>
<span id="cb650-7"><a href="importingexporting-data-into-r.html#cb650-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb650-8"><a href="importingexporting-data-into-r.html#cb650-8" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;/Users/juansantos/Desktop/Teach_R/class_datasets/acetylcholine_receptor_popset_1248341763_protein.txt&quot;</span></span>
<span id="cb650-9"><a href="importingexporting-data-into-r.html#cb650-9" aria-hidden="true" tabindex="-1"></a><span class="co">#[2] &quot;/Users/juansantos/Desktop/Teach_R/class_datasets/acetylcholine_receptor_popset_1248341763.txt&quot;        </span></span>
<span id="cb650-10"><a href="importingexporting-data-into-r.html#cb650-10" aria-hidden="true" tabindex="-1"></a><span class="co">#[3] &quot;/Users/juansantos/Desktop/Teach_R/class_datasets/acetylcholine_receptor_Uniprot.txt&quot;                  </span></span>
<span id="cb650-11"><a href="importingexporting-data-into-r.html#cb650-11" aria-hidden="true" tabindex="-1"></a><span class="co">#[4] &quot;/Users/juansantos/Desktop/Teach_R/class_datasets/airway_scaledcounts.csv&quot;  </span></span>
<span id="cb650-12"><a href="importingexporting-data-into-r.html#cb650-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb650-13"><a href="importingexporting-data-into-r.html#cb650-13" aria-hidden="true" tabindex="-1"></a>my_vector_of_paths <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> path_to_my_datafiles, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>You can select the type of text pattern or extension (e.g., .txt, .csv, .xlsx) to reduce the list of files.</p>
<div class="sourceCode" id="cb651"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb651-1"><a href="importingexporting-data-into-r.html#cb651-1" aria-hidden="true" tabindex="-1"></a><span class="co">#we could select only those that have a .csv (comma delimited) files</span></span>
<span id="cb651-2"><a href="importingexporting-data-into-r.html#cb651-2" aria-hidden="true" tabindex="-1"></a><span class="fu">list.files</span>(<span class="at">path =</span> path_to_my_datafiles, </span>
<span id="cb651-3"><a href="importingexporting-data-into-r.html#cb651-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">pattern =</span> <span class="st">&quot;*.csv&quot;</span>,</span>
<span id="cb651-4"><a href="importingexporting-data-into-r.html#cb651-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb651-5"><a href="importingexporting-data-into-r.html#cb651-5" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;/Users/santosj/Desktop/Teach_R/class_datasets/airway_scaledcounts.csv&quot;</span></span>
<span id="cb651-6"><a href="importingexporting-data-into-r.html#cb651-6" aria-hidden="true" tabindex="-1"></a><span class="co"># [2] &quot;/Users/santosj/Desktop/Teach_R/class_datasets/mtcars2_csv.csv&quot;        </span></span></code></pre></div>
<p><strong>3)</strong> We can now import the data of tab and comma delimited files as data frames (a data structure).</p>
<div class="sourceCode" id="cb652"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb652-1"><a href="importingexporting-data-into-r.html#cb652-1" aria-hidden="true" tabindex="-1"></a><span class="co"># for tab delimited files while preserving headers</span></span>
<span id="cb652-2"><a href="importingexporting-data-into-r.html#cb652-2" aria-hidden="true" tabindex="-1"></a>my_mtcars2_file_df <span class="ot">&lt;-</span> <span class="fu">read.table</span> (<span class="at">file =</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets/mtcars2_file_tab.txt&quot;</span>, </span>
<span id="cb652-3"><a href="importingexporting-data-into-r.html#cb652-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb652-4"><a href="importingexporting-data-into-r.html#cb652-4" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb652-5"><a href="importingexporting-data-into-r.html#cb652-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb652-6"><a href="importingexporting-data-into-r.html#cb652-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(my_mtcars2_file_df)</span>
<span id="cb652-7"><a href="importingexporting-data-into-r.html#cb652-7" aria-hidden="true" tabindex="-1"></a><span class="co">#               cars  mpg cyl disp  hp drat    wt  qsec vs am gear carb</span></span>
<span id="cb652-8"><a href="importingexporting-data-into-r.html#cb652-8" aria-hidden="true" tabindex="-1"></a><span class="co">#1         Mazda_RX4 21.0   6  160 110 3.90 2.620 16.46  0  1    4    4</span></span>
<span id="cb652-9"><a href="importingexporting-data-into-r.html#cb652-9" aria-hidden="true" tabindex="-1"></a><span class="co">#2     Mazda_RX4_Wag 21.0   6  160 110 3.90 2.875 17.02  0  1    4    4</span></span>
<span id="cb652-10"><a href="importingexporting-data-into-r.html#cb652-10" aria-hidden="true" tabindex="-1"></a><span class="co">#3        Datsun_710 22.8   4  108  93 3.85 2.320 18.61  1  1    4    1</span></span>
<span id="cb652-11"><a href="importingexporting-data-into-r.html#cb652-11" aria-hidden="true" tabindex="-1"></a><span class="co">#4    Hornet_4_Drive 21.4   6  258 110 3.08 3.215 19.44  1  0    3    1</span></span>
<span id="cb652-12"><a href="importingexporting-data-into-r.html#cb652-12" aria-hidden="true" tabindex="-1"></a><span class="co">#5 Hornet_Sportabout 18.7   8  360 175 3.15 3.440 17.02  0  0    3    2</span></span>
<span id="cb652-13"><a href="importingexporting-data-into-r.html#cb652-13" aria-hidden="true" tabindex="-1"></a><span class="co">#6           Valiant 18.1   6  225 105 2.76 3.460 20.22  1  0    3    1</span></span></code></pre></div>
<div class="sourceCode" id="cb653"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb653-1"><a href="importingexporting-data-into-r.html#cb653-1" aria-hidden="true" tabindex="-1"></a><span class="co"># for comma delimited files while preserving headers</span></span>
<span id="cb653-2"><a href="importingexporting-data-into-r.html#cb653-2" aria-hidden="true" tabindex="-1"></a>my_airway_file_df <span class="ot">&lt;-</span> <span class="fu">read.table</span> (<span class="at">file =</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets/airway_scaledcounts.csv&quot;</span>, </span>
<span id="cb653-3"><a href="importingexporting-data-into-r.html#cb653-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb653-4"><a href="importingexporting-data-into-r.html#cb653-4" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>,</span>
<span id="cb653-5"><a href="importingexporting-data-into-r.html#cb653-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb653-6"><a href="importingexporting-data-into-r.html#cb653-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(my_airway_file_df)</span>
<span id="cb653-7"><a href="importingexporting-data-into-r.html#cb653-7" aria-hidden="true" tabindex="-1"></a><span class="co">#          ensgene SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb653-8"><a href="importingexporting-data-into-r.html#cb653-8" aria-hidden="true" tabindex="-1"></a><span class="co">#1 ENSG00000000003        723        486        904        445       1170       1097        806        604</span></span>
<span id="cb653-9"><a href="importingexporting-data-into-r.html#cb653-9" aria-hidden="true" tabindex="-1"></a><span class="co">#2 ENSG00000000005          0          0          0          0          0          0          0          0</span></span>
<span id="cb653-10"><a href="importingexporting-data-into-r.html#cb653-10" aria-hidden="true" tabindex="-1"></a><span class="co">#3 ENSG00000000419        467        523        616        371        582        781        417        509</span></span>
<span id="cb653-11"><a href="importingexporting-data-into-r.html#cb653-11" aria-hidden="true" tabindex="-1"></a><span class="co">#4 ENSG00000000457        347        258        364        237        318        447        330        324</span></span>
<span id="cb653-12"><a href="importingexporting-data-into-r.html#cb653-12" aria-hidden="true" tabindex="-1"></a><span class="co">#5 ENSG00000000460         96         81         73         66        118         94        102         74</span></span>
<span id="cb653-13"><a href="importingexporting-data-into-r.html#cb653-13" aria-hidden="true" tabindex="-1"></a><span class="co">#6 ENSG00000000938          0          0          1          0          2          0          0          0</span></span></code></pre></div>
</div>
<div id="saving-data-to-a-file" class="section level2 hasAnchor" number="30.5">
<h2><span class="header-section-number">30.5</span> Saving data to a file<a href="importingexporting-data-into-r.html#saving-data-to-a-file" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You can save data frames as tab or comma delimited files using the function <code>write.table()</code>. However, you should make sure that the folder where you will like your files to be saved is defined.</p>
<div class="sourceCode" id="cb654"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb654-1"><a href="importingexporting-data-into-r.html#cb654-1" aria-hidden="true" tabindex="-1"></a><span class="co">#define the the output folder</span></span>
<span id="cb654-2"><a href="importingexporting-data-into-r.html#cb654-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/class_datasets&quot;</span>)</span>
<span id="cb654-3"><a href="importingexporting-data-into-r.html#cb654-3" aria-hidden="true" tabindex="-1"></a><span class="co">#confirm that the output directory is correct</span></span>
<span id="cb654-4"><a href="importingexporting-data-into-r.html#cb654-4" aria-hidden="true" tabindex="-1"></a><span class="fu">getwd</span>()</span>
<span id="cb654-5"><a href="importingexporting-data-into-r.html#cb654-5" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;/Users/juansantos/Desktop/Teach_R/class_datasets&quot;&quot;</span></span>
<span id="cb654-6"><a href="importingexporting-data-into-r.html#cb654-6" aria-hidden="true" tabindex="-1"></a><span class="co"># to determine the separation between columns is defined by the argument &#39;sep =&#39;. For tabs, you use sep =”\t” and for commas you use sep =”,”</span></span>
<span id="cb654-7"><a href="importingexporting-data-into-r.html#cb654-7" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(my_mtcars2_file_df, </span>
<span id="cb654-8"><a href="importingexporting-data-into-r.html#cb654-8" aria-hidden="true" tabindex="-1"></a>      <span class="at">file =</span> <span class="st">&quot;mtcars2_file_tab.txt&quot;</span>, </span>
<span id="cb654-9"><a href="importingexporting-data-into-r.html#cb654-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb654-10"><a href="importingexporting-data-into-r.html#cb654-10" aria-hidden="true" tabindex="-1"></a> <span class="at">col.names =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="saving-data-as-rdata-.rdata-.rda" class="section level2 hasAnchor" number="30.6">
<h2><span class="header-section-number">30.6</span> Saving data as Rdata (.rdata, .rda)<a href="importingexporting-data-into-r.html#saving-data-as-rdata-.rdata-.rda" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You might feel compelled to save your R session. This action will store your complete R workspace or selected “objects” from a workspace in a <a href="https://cran.r-project.org/doc/manuals/r-release/R-data.pdf" target="_blank"><strong>R-Data</strong></a> file with extension .rdata and .rda that can be loaded back by R. In most cases, you will do this to save you time in your next session. I do not usually save my R sessions, as the <em>.rdata and </em>.rda files are not human readable, and in most cases my interest are in the functionality of the scripts that call data and run functions within R rather than going back a preloaded session.</p>
</div>
<div id="import-excel-formated-data" class="section level2 hasAnchor" number="30.7">
<h2><span class="header-section-number">30.7</span> Import Excel formated data<a href="importingexporting-data-into-r.html#import-excel-formated-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can now import the data from Excel spreadsheet files as data frames (a data structure). We could use R-package <a href="https://cran.r-project.org/web/packages/openxlsx/index.html" target="_blank"><strong>openxlsx</strong></a> and its function <code>read.xlsx()</code>.</p>
<div class="sourceCode" id="cb655"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb655-1"><a href="importingexporting-data-into-r.html#cb655-1" aria-hidden="true" tabindex="-1"></a><span class="co">#install package if not present in your library</span></span>
<span id="cb655-2"><a href="importingexporting-data-into-r.html#cb655-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;openxlsx&quot;</span>)</span>
<span id="cb655-3"><a href="importingexporting-data-into-r.html#cb655-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span></code></pre></div>
<p>You can load the FIRST sheet to a data frame that has headers on the first row.</p>
<div class="sourceCode" id="cb656"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb656-1"><a href="importingexporting-data-into-r.html#cb656-1" aria-hidden="true" tabindex="-1"></a>my_mtcars2_xlsx_df <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(<span class="at">xlsxFile =</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets/mixed_datasets.xlsx&quot;</span>,</span>
<span id="cb656-2"><a href="importingexporting-data-into-r.html#cb656-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">sheet =</span> <span class="dv">1</span>,</span>
<span id="cb656-3"><a href="importingexporting-data-into-r.html#cb656-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">startRow =</span> <span class="dv">1</span>,</span>
<span id="cb656-4"><a href="importingexporting-data-into-r.html#cb656-4" aria-hidden="true" tabindex="-1"></a>                                <span class="at">colNames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb656-5"><a href="importingexporting-data-into-r.html#cb656-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(my_mtcars2_xlsx_df)</span>
<span id="cb656-6"><a href="importingexporting-data-into-r.html#cb656-6" aria-hidden="true" tabindex="-1"></a><span class="co">#               cars  mpg cyl disp  hp drat    wt  qsec vs am gear carb</span></span>
<span id="cb656-7"><a href="importingexporting-data-into-r.html#cb656-7" aria-hidden="true" tabindex="-1"></a><span class="co">#1         Mazda_RX4 21.0   6  160 110 3.90 2.620 16.46  0  1    4    4</span></span>
<span id="cb656-8"><a href="importingexporting-data-into-r.html#cb656-8" aria-hidden="true" tabindex="-1"></a><span class="co">#2     Mazda_RX4_Wag 21.0   6  160 110 3.90 2.875 17.02  0  1    4    4</span></span>
<span id="cb656-9"><a href="importingexporting-data-into-r.html#cb656-9" aria-hidden="true" tabindex="-1"></a><span class="co">#3        Datsun_710 22.8   4  108  93 3.85 2.320 18.61  1  1    4    1</span></span>
<span id="cb656-10"><a href="importingexporting-data-into-r.html#cb656-10" aria-hidden="true" tabindex="-1"></a><span class="co">#4    Hornet_4_Drive 21.4   6  258 110 3.08 3.215 19.44  1  0    3    1</span></span>
<span id="cb656-11"><a href="importingexporting-data-into-r.html#cb656-11" aria-hidden="true" tabindex="-1"></a><span class="co">#5 Hornet_Sportabout 18.7   8  360 175 3.15 3.440 17.02  0  0    3    2</span></span>
<span id="cb656-12"><a href="importingexporting-data-into-r.html#cb656-12" aria-hidden="true" tabindex="-1"></a><span class="co">#6           Valiant 18.1   6  225 105 2.76 3.460 20.22  1  0    3    1</span></span></code></pre></div>
<p>Now, you can load the SECOND sheet to a data frame that has headers on the first row.</p>
<div class="sourceCode" id="cb657"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb657-1"><a href="importingexporting-data-into-r.html#cb657-1" aria-hidden="true" tabindex="-1"></a>my_airway_xlsx_df <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(<span class="at">xlsxFile =</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets/mixed_datasets.xlsx&quot;</span>,</span>
<span id="cb657-2"><a href="importingexporting-data-into-r.html#cb657-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">sheet =</span> <span class="dv">2</span>,</span>
<span id="cb657-3"><a href="importingexporting-data-into-r.html#cb657-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">startRow =</span> <span class="dv">1</span>,</span>
<span id="cb657-4"><a href="importingexporting-data-into-r.html#cb657-4" aria-hidden="true" tabindex="-1"></a>                                <span class="at">colNames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb657-5"><a href="importingexporting-data-into-r.html#cb657-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(my_airway_xlsx_df)</span>
<span id="cb657-6"><a href="importingexporting-data-into-r.html#cb657-6" aria-hidden="true" tabindex="-1"></a><span class="co">#          ensgene SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb657-7"><a href="importingexporting-data-into-r.html#cb657-7" aria-hidden="true" tabindex="-1"></a><span class="co">#1 ENSG00000000003        723        486        904        445       1170       1097        806        604</span></span>
<span id="cb657-8"><a href="importingexporting-data-into-r.html#cb657-8" aria-hidden="true" tabindex="-1"></a><span class="co">#2 ENSG00000000005          0          0          0          0          0          0          0          0</span></span>
<span id="cb657-9"><a href="importingexporting-data-into-r.html#cb657-9" aria-hidden="true" tabindex="-1"></a><span class="co">#3 ENSG00000000419        467        523        616        371        582        781        417        509</span></span>
<span id="cb657-10"><a href="importingexporting-data-into-r.html#cb657-10" aria-hidden="true" tabindex="-1"></a><span class="co">#4 ENSG00000000457        347        258        364        237        318        447        330        324</span></span>
<span id="cb657-11"><a href="importingexporting-data-into-r.html#cb657-11" aria-hidden="true" tabindex="-1"></a><span class="co">#5 ENSG00000000460         96         81         73         66        118         94        102         74</span></span>
<span id="cb657-12"><a href="importingexporting-data-into-r.html#cb657-12" aria-hidden="true" tabindex="-1"></a><span class="co">#6 ENSG00000000938          0          0          1          0          2          0          0          0</span></span></code></pre></div>
</div>
<div id="save-data-as-excel-formated-file" class="section level2 hasAnchor" number="30.8">
<h2><span class="header-section-number">30.8</span> Save data as Excel formated file<a href="importingexporting-data-into-r.html#save-data-as-excel-formated-file" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can save a data frame as an Excel spreadsheet with R-package <a href="https://cran.r-project.org/web/packages/openxlsx/index.html" target="_blank"><strong>openxlsx</strong></a> and its function <code>write.xlsx()</code>.</p>
<div class="sourceCode" id="cb658"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb658-1"><a href="importingexporting-data-into-r.html#cb658-1" aria-hidden="true" tabindex="-1"></a><span class="co">#define the the output folder</span></span>
<span id="cb658-2"><a href="importingexporting-data-into-r.html#cb658-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/class_datasets&quot;</span>)</span>
<span id="cb658-3"><a href="importingexporting-data-into-r.html#cb658-3" aria-hidden="true" tabindex="-1"></a><span class="co">#confirm that the output directory is correct</span></span>
<span id="cb658-4"><a href="importingexporting-data-into-r.html#cb658-4" aria-hidden="true" tabindex="-1"></a><span class="fu">getwd</span>()</span>
<span id="cb658-5"><a href="importingexporting-data-into-r.html#cb658-5" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;/Users/juansantos/Desktop/Teach_R/class_datasets&quot;&quot;</span></span>
<span id="cb658-6"><a href="importingexporting-data-into-r.html#cb658-6" aria-hidden="true" tabindex="-1"></a>my_airway_xlsx_df <span class="ot">&lt;-</span> <span class="fu">write.xlsx</span>(my_mtcars2_file_df,</span>
<span id="cb658-7"><a href="importingexporting-data-into-r.html#cb658-7" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">sheetName =</span> <span class="st">&quot;my_mtcars_xlsx_sheet&quot;</span>,</span>
<span id="cb658-8"><a href="importingexporting-data-into-r.html#cb658-8" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">file =</span> <span class="st">&quot;mtcars_xlsx.xlsx&quot;</span>)</span></code></pre></div>
</div>
<div id="fixing-problems-while-importing-datasets" class="section level2 hasAnchor" number="30.9">
<h2><span class="header-section-number">30.9</span> Fixing problems while importing datasets<a href="importingexporting-data-into-r.html#fixing-problems-while-importing-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>It is not unusual to you might have problems while importing datasets from DRYAD or any other source into R. These usually must do column titles, unusual characters, uneven number of entries (i.e., some rows have more than the others). These can easily be preprocess directly in the text or excel files before you import and save you time.</p>
<p><strong>1)</strong> You can start by determine if R is unable to open or input data from your file. We can test with a file with errors <code>mtcars2_file_tab_with_errors.txt</code></p>
<div class="sourceCode" id="cb659"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb659-1"><a href="importingexporting-data-into-r.html#cb659-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We try to import this file and it has error</span></span>
<span id="cb659-2"><a href="importingexporting-data-into-r.html#cb659-2" aria-hidden="true" tabindex="-1"></a>my_mtcars2_file_df <span class="ot">&lt;-</span> <span class="fu">read.table</span> (<span class="at">file =</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets/mtcars2_file_tab_with_errors.txt&quot;</span>, </span>
<span id="cb659-3"><a href="importingexporting-data-into-r.html#cb659-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb659-4"><a href="importingexporting-data-into-r.html#cb659-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb659-5"><a href="importingexporting-data-into-r.html#cb659-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb659-6"><a href="importingexporting-data-into-r.html#cb659-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb659-7"><a href="importingexporting-data-into-r.html#cb659-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb659-8"><a href="importingexporting-data-into-r.html#cb659-8" aria-hidden="true" tabindex="-1"></a><span class="do">## some examples of errors will be</span></span>
<span id="cb659-9"><a href="importingexporting-data-into-r.html#cb659-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb659-10"><a href="importingexporting-data-into-r.html#cb659-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Error in read.table(file = &quot;~/Desktop/Teach_R/class_datasets/mtcars2_file_tab_with_errors.txt&quot;,  : </span></span>
<span id="cb659-11"><a href="importingexporting-data-into-r.html#cb659-11" aria-hidden="true" tabindex="-1"></a><span class="co">#  more columns than column names</span></span>
<span id="cb659-12"><a href="importingexporting-data-into-r.html#cb659-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb659-13"><a href="importingexporting-data-into-r.html#cb659-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Error in scan(file = file, what = what, sep = sep, quote = quote, dec = dec,  : </span></span>
<span id="cb659-14"><a href="importingexporting-data-into-r.html#cb659-14" aria-hidden="true" tabindex="-1"></a><span class="co"># line 3 did not have 15 elements</span></span>
<span id="cb659-15"><a href="importingexporting-data-into-r.html#cb659-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb659-16"><a href="importingexporting-data-into-r.html#cb659-16" aria-hidden="true" tabindex="-1"></a><span class="do">## if you try to see if you have loaded that dataset, you get an error</span></span>
<span id="cb659-17"><a href="importingexporting-data-into-r.html#cb659-17" aria-hidden="true" tabindex="-1"></a>my_mtcars2_file_df</span>
<span id="cb659-18"><a href="importingexporting-data-into-r.html#cb659-18" aria-hidden="true" tabindex="-1"></a><span class="co">#Error: object &#39;my_mtcars2_file_df&#39; not found</span></span></code></pre></div>
<p>The output indicates that the number of column names does not match the number of entries in a row of the dataset.</p>
<p><strong>2)</strong> You can start by opening the file in excel and notice if the column names (i.e., variable names) do match the number of columns. If they do not, you need to add a missing name(s). Likewise, no rows can have more values than those of the column names, unless you are planning to import without column names. It is not uncommon that some values might have incorrectly entered and you have to correct those before you import your dataset into R.</p>
<p><img src="figures/session_14_figs/7_entry_data_error_1.png" width="100%" /></p>
<p>We notice that rows <code>2</code>, <code>3</code>,<code>5</code> and <code>6</code> have misplaced values, that need to be corrected. Some might need to be removed (e.g., as in rows <code>2</code> and <code>6</code>) or the text that should be in one column
has moved to the next column (e.g., as in rows <code>3</code> and <code>5</code>). Here is this table after correction:</p>
<p><img src="figures/session_14_figs/8_entry_data_error_2.png" width="100%" /></p>
<p>We probably will be able to import this dataset, but we need to clean it up a bit more</p>
<div class="sourceCode" id="cb660"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb660-1"><a href="importingexporting-data-into-r.html#cb660-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We try to import this file again</span></span>
<span id="cb660-2"><a href="importingexporting-data-into-r.html#cb660-2" aria-hidden="true" tabindex="-1"></a>my_mtcars2_file_df <span class="ot">&lt;-</span> <span class="fu">read.table</span> (<span class="at">file =</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets/mtcars2_file_tab_with_errors2.txt&quot;</span>, </span>
<span id="cb660-3"><a href="importingexporting-data-into-r.html#cb660-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb660-4"><a href="importingexporting-data-into-r.html#cb660-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb660-5"><a href="importingexporting-data-into-r.html#cb660-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb660-6"><a href="importingexporting-data-into-r.html#cb660-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(my_mtcars2_file_df)</span>
<span id="cb660-7"><a href="importingexporting-data-into-r.html#cb660-7" aria-hidden="true" tabindex="-1"></a><span class="co">#  cars.XX.            mpg.XX cyl.XX This.9s.a...very.complex.......Name  hp drat   wt  qsec    vs am gear carb X</span></span>
<span id="cb660-8"><a href="importingexporting-data-into-r.html#cb660-8" aria-hidden="true" tabindex="-1"></a><span class="co">#1        1         Mazda_RX4   21.0                                   6 160  110 3.90 2.620 16.46  0    1    4 4</span></span>
<span id="cb660-9"><a href="importingexporting-data-into-r.html#cb660-9" aria-hidden="true" tabindex="-1"></a><span class="co">#2        2     Mazda_RX4_Wag   21.0                                   6 160  110 3.90 2.875 17.02  0    1    4 4</span></span>
<span id="cb660-10"><a href="importingexporting-data-into-r.html#cb660-10" aria-hidden="true" tabindex="-1"></a><span class="co">#3        3        Datsun_710   22.8                                   4 108   93 3.85 2.320 18.61  1    1    4 1</span></span>
<span id="cb660-11"><a href="importingexporting-data-into-r.html#cb660-11" aria-hidden="true" tabindex="-1"></a><span class="co">#4        4    Hornet_4_Drive   21.4                                   6 258  110 3.08 3.215 19.44  1    0    3 1</span></span>
<span id="cb660-12"><a href="importingexporting-data-into-r.html#cb660-12" aria-hidden="true" tabindex="-1"></a><span class="co">#5        5 Hornet_Sportabout   18.7                                   8 360  175 3.15 3.440 17.02  0    0    3 2</span></span>
<span id="cb660-13"><a href="importingexporting-data-into-r.html#cb660-13" aria-hidden="true" tabindex="-1"></a><span class="co">#6        6           Valiant   18.1                                   6 225  105 2.76 3.460 20.22  1    0    3 1</span></span>
<span id="cb660-14"><a href="importingexporting-data-into-r.html#cb660-14" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(my_mtcars2_file_df)</span>
<span id="cb660-15"><a href="importingexporting-data-into-r.html#cb660-15" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;cars.XX.&quot;                            &quot;mpg.XX&quot;                              &quot;cyl.XX&quot;                              &quot;This.9s.a...very.complex.......Name&quot; &quot;hp&quot;                                 </span></span>
<span id="cb660-16"><a href="importingexporting-data-into-r.html#cb660-16" aria-hidden="true" tabindex="-1"></a><span class="co">#[6] &quot;drat&quot;                                &quot;wt&quot;                                  &quot;qsec&quot;                                &quot;vs&quot;                                  &quot;am&quot;                                 </span></span>
<span id="cb660-17"><a href="importingexporting-data-into-r.html#cb660-17" aria-hidden="true" tabindex="-1"></a><span class="co">#[11] &quot;gear&quot;                                &quot;carb&quot;                                &quot;X&quot; </span></span></code></pre></div>
<p><strong>3)</strong> We notice that the names for the last column has been assigned a title of a generic <code>X</code> and the column 1 has only the row increasing numbers (i.e., is a useless column). We can move the names of the columns by one cell and delete the column 1.</p>
<p><img src="figures/session_14_figs/9_entry_data_error_3.png" width="100%" /></p>
<p>We probably will be able to import this dataset again, but we need to clean it up a bit more</p>
<div class="sourceCode" id="cb661"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb661-1"><a href="importingexporting-data-into-r.html#cb661-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We try to import this file again</span></span>
<span id="cb661-2"><a href="importingexporting-data-into-r.html#cb661-2" aria-hidden="true" tabindex="-1"></a>my_mtcars2_file_df <span class="ot">&lt;-</span> <span class="fu">read.table</span> (<span class="at">file =</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets/mtcars2_file_tab_with_errors3.txt&quot;</span>, </span>
<span id="cb661-3"><a href="importingexporting-data-into-r.html#cb661-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb661-4"><a href="importingexporting-data-into-r.html#cb661-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb661-5"><a href="importingexporting-data-into-r.html#cb661-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb661-6"><a href="importingexporting-data-into-r.html#cb661-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(my_mtcars2_file_df)</span>
<span id="cb661-7"><a href="importingexporting-data-into-r.html#cb661-7" aria-hidden="true" tabindex="-1"></a><span class="co">#             cars.XX. mpg.XX cyl.XX This.9s.a...very.complex.......Name  hp drat    wt  qsec vs am gear carb</span></span>
<span id="cb661-8"><a href="importingexporting-data-into-r.html#cb661-8" aria-hidden="true" tabindex="-1"></a><span class="co">#1         Mazda_RX4   21.0      6                                 160 110 3.90 2.620 16.46  0  1    4    4</span></span>
<span id="cb661-9"><a href="importingexporting-data-into-r.html#cb661-9" aria-hidden="true" tabindex="-1"></a><span class="co">#2     Mazda_RX4_Wag   21.0      6                                 160 110 3.90 2.875 17.02  0  1    4    4</span></span>
<span id="cb661-10"><a href="importingexporting-data-into-r.html#cb661-10" aria-hidden="true" tabindex="-1"></a><span class="co">#3        Datsun_710   22.8      4                                 108  93 3.85 2.320 18.61  1  1    4    1</span></span>
<span id="cb661-11"><a href="importingexporting-data-into-r.html#cb661-11" aria-hidden="true" tabindex="-1"></a><span class="co">#4    Hornet_4_Drive   21.4      6                                 258 110 3.08 3.215 19.44  1  0    3    1</span></span>
<span id="cb661-12"><a href="importingexporting-data-into-r.html#cb661-12" aria-hidden="true" tabindex="-1"></a><span class="co">#5 Hornet_Sportabout   18.7      8                                 360 175 3.15 3.440 17.02  0  0    3    2</span></span>
<span id="cb661-13"><a href="importingexporting-data-into-r.html#cb661-13" aria-hidden="true" tabindex="-1"></a><span class="co">#6           Valiant   18.1      6                                 225 105 2.76 3.460 20.22  1  0    3    1names(my_mtcars2_file_df)</span></span>
<span id="cb661-14"><a href="importingexporting-data-into-r.html#cb661-14" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(my_mtcars2_file_df)</span>
<span id="cb661-15"><a href="importingexporting-data-into-r.html#cb661-15" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;cars.XX.&quot;                            &quot;mpg.XX&quot;                              &quot;cyl.XX&quot;                              &quot;This.9s.a...very.complex.......Name&quot; &quot;hp&quot;                                 </span></span>
<span id="cb661-16"><a href="importingexporting-data-into-r.html#cb661-16" aria-hidden="true" tabindex="-1"></a><span class="co"># [6] &quot;drat&quot;                                &quot;wt&quot;                                  &quot;qsec&quot;                                &quot;vs&quot;                                  &quot;am&quot;                                 </span></span>
<span id="cb661-17"><a href="importingexporting-data-into-r.html#cb661-17" aria-hidden="true" tabindex="-1"></a><span class="co">#[11] &quot;gear&quot;                                &quot;carb&quot;</span></span></code></pre></div>
<p><strong>4)</strong> We can simplify the column names and texts if you have character columns by replacing spaces, <code>.</code>, <code>;</code>, <code>,</code>, <code>( )</code>, <code>[ ]</code>, <code>{ }</code>, <code>=</code>, <code>"</code>, <code>/</code>, <code>\</code>, <code>+</code>, <code>?</code>, <code>$</code>, <code>%</code>, <code>*</code>, <code>#</code> or any other strange symbols with <code>_</code>. Likewise, this will make the column names short and easy to interpret.</p>
<p><img src="figures/session_14_figs/10_entry_data_error_4.png" width="100%" /></p>
<p>We probably will be able to import this dataset again and this might be good for your analyses.</p>
<div class="sourceCode" id="cb662"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb662-1"><a href="importingexporting-data-into-r.html#cb662-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We try to import this file again</span></span>
<span id="cb662-2"><a href="importingexporting-data-into-r.html#cb662-2" aria-hidden="true" tabindex="-1"></a>my_mtcars2_file_df <span class="ot">&lt;-</span> <span class="fu">read.table</span> (<span class="at">file =</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets/mtcars2_file_tab.txt&quot;</span>, </span>
<span id="cb662-3"><a href="importingexporting-data-into-r.html#cb662-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb662-4"><a href="importingexporting-data-into-r.html#cb662-4" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb662-5"><a href="importingexporting-data-into-r.html#cb662-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb662-6"><a href="importingexporting-data-into-r.html#cb662-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(my_mtcars2_file_df)</span>
<span id="cb662-7"><a href="importingexporting-data-into-r.html#cb662-7" aria-hidden="true" tabindex="-1"></a><span class="co">#               cars  mpg cyl disp  hp drat    wt  qsec vs am gear carb</span></span>
<span id="cb662-8"><a href="importingexporting-data-into-r.html#cb662-8" aria-hidden="true" tabindex="-1"></a><span class="co">#1         Mazda_RX4 21.0   6  160 110 3.90 2.620 16.46  0  1    4    4</span></span>
<span id="cb662-9"><a href="importingexporting-data-into-r.html#cb662-9" aria-hidden="true" tabindex="-1"></a><span class="co">#2     Mazda_RX4_Wag 21.0   6  160 110 3.90 2.875 17.02  0  1    4    4</span></span>
<span id="cb662-10"><a href="importingexporting-data-into-r.html#cb662-10" aria-hidden="true" tabindex="-1"></a><span class="co">#3        Datsun_710 22.8   4  108  93 3.85 2.320 18.61  1  1    4    1</span></span>
<span id="cb662-11"><a href="importingexporting-data-into-r.html#cb662-11" aria-hidden="true" tabindex="-1"></a><span class="co">#4    Hornet_4_Drive 21.4   6  258 110 3.08 3.215 19.44  1  0    3    1</span></span>
<span id="cb662-12"><a href="importingexporting-data-into-r.html#cb662-12" aria-hidden="true" tabindex="-1"></a><span class="co">#5 Hornet_Sportabout 18.7   8  360 175 3.15 3.440 17.02  0  0    3    2</span></span>
<span id="cb662-13"><a href="importingexporting-data-into-r.html#cb662-13" aria-hidden="true" tabindex="-1"></a><span class="co">#6           Valiant 18.1   6  225 105 2.76 3.460 20.22  1  0    3    1</span></span>
<span id="cb662-14"><a href="importingexporting-data-into-r.html#cb662-14" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(my_mtcars2_file_df)</span>
<span id="cb662-15"><a href="importingexporting-data-into-r.html#cb662-15" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;cars&quot; &quot;mpg&quot;  &quot;cyl&quot;  &quot;disp&quot; &quot;hp&quot;   &quot;drat&quot; &quot;wt&quot;   &quot;qsec&quot; &quot;vs&quot;   &quot;am&quot;   &quot;gear&quot; &quot;carb&quot;  </span></span></code></pre></div>
</div>
<div id="simplifying-embellished-excel-datasets" class="section level2 hasAnchor" number="30.10">
<h2><span class="header-section-number">30.10</span> Simplifying embellished excel datasets<a href="importingexporting-data-into-r.html#simplifying-embellished-excel-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Many datasets might be heavily <strong>embellished</strong> to appeal readers, yet these might make importing such datasets to R very hard. For this purpose, we should simplify such datasets to make them simpler to import into R. Here is an example of an adorned excel file <code>mtcars_xlsx_to_clean.xlsx</code>, you can find it at <a href="https://github.com/corytophanes/BIO_BIT_Bioinformatics_209/tree/main/ref_files_pdfs" target="_blank"><strong>LINK TO FILES of BIO/BIT 209</strong></a> and download it to a folder of your choosing.</p>
<p>Here is how the first excel worksheet looks in such program.</p>
<p><img src="figures/session_14_figs/18_embelished_excel.png" width="100%" /></p>
<p>If we try to import the first excel worksheet into R, we get a data frame with incorrect column names.</p>
<div class="sourceCode" id="cb663"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb663-1"><a href="importingexporting-data-into-r.html#cb663-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb663-2"><a href="importingexporting-data-into-r.html#cb663-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb663-3"><a href="importingexporting-data-into-r.html#cb663-3" aria-hidden="true" tabindex="-1"></a>incorrect_mtcars_xlsx_df <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(<span class="at">xlsxFile =</span> <span class="st">&quot;~/Desktop/Teach_R/my_working_directory/mtcars_xlsx_to_clean.xlsx&quot;</span>,</span>
<span id="cb663-4"><a href="importingexporting-data-into-r.html#cb663-4" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">sheet =</span> <span class="dv">1</span>,</span>
<span id="cb663-5"><a href="importingexporting-data-into-r.html#cb663-5" aria-hidden="true" tabindex="-1"></a>                                <span class="at">startRow =</span> <span class="dv">1</span>,</span>
<span id="cb663-6"><a href="importingexporting-data-into-r.html#cb663-6" aria-hidden="true" tabindex="-1"></a>                                <span class="at">colNames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb663-7"><a href="importingexporting-data-into-r.html#cb663-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(incorrect_mtcars_xlsx_df)</span>
<span id="cb663-8"><a href="importingexporting-data-into-r.html#cb663-8" aria-hidden="true" tabindex="-1"></a><span class="co">#                X1 Group.1  X3   X4  X5   X6            Group.2                 X8                 X9 X10 X11  X12  X13</span></span>
<span id="cb663-9"><a href="importingexporting-data-into-r.html#cb663-9" aria-hidden="true" tabindex="-1"></a><span class="co">#1              cars     mpg cyl disp  hp drat                mpg                 wt               qsec  vs  am gear carb</span></span>
<span id="cb663-10"><a href="importingexporting-data-into-r.html#cb663-10" aria-hidden="true" tabindex="-1"></a><span class="co">#2         Mazda_RX4      21   6  160 110  3.9                 20               2.62              16.46   0   1    4    4</span></span>
<span id="cb663-11"><a href="importingexporting-data-into-r.html#cb663-11" aria-hidden="true" tabindex="-1"></a><span class="co">#3     Mazda_RX4_Wag      21   6  160 110  3.9                 20              2.875              17.02   0   1    4    4</span></span>
<span id="cb663-12"><a href="importingexporting-data-into-r.html#cb663-12" aria-hidden="true" tabindex="-1"></a><span class="co">#4        Datsun_710    22.8   4  108  93 3.85               21.8 2.3199999999999998              18.61   1   1    4    1</span></span>
<span id="cb663-13"><a href="importingexporting-data-into-r.html#cb663-13" aria-hidden="true" tabindex="-1"></a><span class="co">#5    Hornet_4_Drive    21.4   6  258 110 3.08 20.399999999999999 3.2149999999999999 19.440000000000001   1   0    3    1</span></span>
<span id="cb663-14"><a href="importingexporting-data-into-r.html#cb663-14" aria-hidden="true" tabindex="-1"></a><span class="co">#6 Hornet_Sportabout    18.7   8  360 175 3.15               17.7               3.44              17.02   0   0    3    2</span></span>
<span id="cb663-15"><a href="importingexporting-data-into-r.html#cb663-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb663-16"><a href="importingexporting-data-into-r.html#cb663-16" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(incorrect_mtcars_xlsx_df)</span>
<span id="cb663-17"><a href="importingexporting-data-into-r.html#cb663-17" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] &quot;X1&quot;      &quot;Group.1&quot; &quot;X3&quot;      &quot;X4&quot;      &quot;X5&quot;      &quot;X6&quot;      &quot;Group.2&quot; &quot;X8&quot;      &quot;X9&quot;      &quot;X10&quot;     &quot;X11&quot;     &quot;X12&quot;     &quot;X13&quot; </span></span></code></pre></div>
<p>We notice that all column names are incorrect and have a <code>X-number</code> code and there is a mixture of text and numbers in the columns, as if the column names have become measurements.</p>
<p>To correct this, it requires to remove the <strong>row 1</strong> in the excel sheet where the names <strong>Group 1</strong> and <strong>Group 2</strong> are located and incorporate such group names into the column names located in the second row.</p>
<p>It is important that each column has a <strong>unique name</strong> and it is informative enough to avoid R giving each column a <code>X-number</code> code to make these column names unique.</p>
<p>An example of the same excel sheet with <strong>simplified names</strong> with enough and descriptive information present, but we incorporated the name of the groups into the column names and removed the useless mean and SD rows as well as the empty rows in the middle of the spreadsheet.</p>
<p>Here is how the simplified and corrected excel worksheet looks in such program.</p>
<p><img src="figures/session_14_figs/19_simple_excel.png" width="100%" /></p>
<p>Notice that this Excel sheet has in its first row the simplified variable names and the first column has the names of each type of car (i.e., it corresponds to individual IDs in this dataset). If we try to import this updated excel worksheet into R, we get a better result. The variables names are recognized, and the data is better organized for further analyses.</p>
<div class="sourceCode" id="cb664"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb664-1"><a href="importingexporting-data-into-r.html#cb664-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb664-2"><a href="importingexporting-data-into-r.html#cb664-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb664-3"><a href="importingexporting-data-into-r.html#cb664-3" aria-hidden="true" tabindex="-1"></a>correct_mtcars_xlsx_df <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(<span class="at">xlsxFile =</span> <span class="st">&quot;~/Desktop/Teach_R/my_working_directory/mtcars_xlsx_cleaned.xlsx&quot;</span>,</span>
<span id="cb664-4"><a href="importingexporting-data-into-r.html#cb664-4" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">sheet =</span> <span class="dv">1</span>,</span>
<span id="cb664-5"><a href="importingexporting-data-into-r.html#cb664-5" aria-hidden="true" tabindex="-1"></a>                                <span class="at">startRow =</span> <span class="dv">1</span>,</span>
<span id="cb664-6"><a href="importingexporting-data-into-r.html#cb664-6" aria-hidden="true" tabindex="-1"></a>                                <span class="at">colNames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb664-7"><a href="importingexporting-data-into-r.html#cb664-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(correct_mtcars_xlsx_df)</span>
<span id="cb664-8"><a href="importingexporting-data-into-r.html#cb664-8" aria-hidden="true" tabindex="-1"></a><span class="co">#               cars Group_1_mpg Group_1_cyl Group_1_disp Group_1_hp Group_1_drat Group_2_mpg Group_2_wt Group_2_qsec Group_2_vs Group_2_am Group_2_gear Group_2_carb</span></span>
<span id="cb664-9"><a href="importingexporting-data-into-r.html#cb664-9" aria-hidden="true" tabindex="-1"></a><span class="co">#1         Mazda_RX4        21.0           6          160        110         3.90        20.0      2.620        16.46          0          1            4            4</span></span>
<span id="cb664-10"><a href="importingexporting-data-into-r.html#cb664-10" aria-hidden="true" tabindex="-1"></a><span class="co">#2     Mazda_RX4_Wag        21.0           6          160        110         3.90        20.0      2.875        17.02          0          1            4            4</span></span>
<span id="cb664-11"><a href="importingexporting-data-into-r.html#cb664-11" aria-hidden="true" tabindex="-1"></a><span class="co">#3        Datsun_710        22.8           4          108         93         3.85        21.8      2.320        18.61          1          1            4            1</span></span>
<span id="cb664-12"><a href="importingexporting-data-into-r.html#cb664-12" aria-hidden="true" tabindex="-1"></a><span class="co">#4    Hornet_4_Drive        21.4           6          258        110         3.08        20.4      3.215        19.44          1          0            3            1</span></span>
<span id="cb664-13"><a href="importingexporting-data-into-r.html#cb664-13" aria-hidden="true" tabindex="-1"></a><span class="co">#5 Hornet_Sportabout        18.7           8          360        175         3.15        17.7      3.440        17.02          0          0            3            2</span></span>
<span id="cb664-14"><a href="importingexporting-data-into-r.html#cb664-14" aria-hidden="true" tabindex="-1"></a><span class="co">#6           Valiant        18.1           6          225        105         2.76        17.1      3.460        20.22          1          0            3            1</span></span>
<span id="cb664-15"><a href="importingexporting-data-into-r.html#cb664-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb664-16"><a href="importingexporting-data-into-r.html#cb664-16" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(correct_mtcars_xlsx_df)</span>
<span id="cb664-17"><a href="importingexporting-data-into-r.html#cb664-17" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;cars&quot;         &quot;Group_1_mpg&quot;  &quot;Group_1_cyl&quot;  &quot;Group_1_disp&quot; &quot;Group_1_hp&quot;   &quot;Group_1_drat&quot; &quot;Group_2_mpg&quot;  &quot;Group_2_wt&quot;   &quot;Group_2_qsec&quot; &quot;Group_2_vs&quot;   &quot;Group_2_am&quot;   &quot;Group_2_gear&quot;</span></span>
<span id="cb664-18"><a href="importingexporting-data-into-r.html#cb664-18" aria-hidden="true" tabindex="-1"></a><span class="co">#[13] &quot;Group_2_carb&quot;</span></span></code></pre></div>
</div>
<div id="accessing-online-data-dryad" class="section level2 hasAnchor" number="30.11">
<h2><span class="header-section-number">30.11</span> Accessing online data: DRYAD<a href="importingexporting-data-into-r.html#accessing-online-data-dryad" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You might want to access data for your own research, data comparisons with previous published studies, and/or complement your own datasets and projects. This can be done using and accessing several online databases. One of the most important is <a href="https://datadryad.org/stash" target="_blank"><strong>DRYAD</strong></a>. From their website, Dryad Digital Repository mission is to serve as an indefinity repository of curated (i.e., revised) digital resources that makes published data discoverable, freely reusable, and citable. Therefore, these data can then be re-used to create new knowledge. <a href="https://datadryad.org/stash" target="_blank"><strong>DRYAD</strong></a> is an initiative &gt;300 journals and scientific societies that adopted a joint data archiving policy (JDAP) for their publications, and their recognition that open data, easy-to-use, not-for-profit, community-governed data infrastructure is fundamental to maintain such digital resources. Dryad stores mostly biological and medical data used in peer-reviewed journals.</p>
<p>To access <a href="https://datadryad.org/stash" target="_blank"><strong>DRYAD</strong></a>, you can first determine if the article of interest has archived data in this repository. We will use this article that you can access from <a href="https://pubmed.ncbi.nlm.nih.gov" target="_blank"><strong>PubMed</strong></a>:</p>
<p><strong>Allen, R. C., Angst, D. C., &amp; Hall, A. R. (2019). Resistance Gene Carriage Predicts Growth of Natural and Clinical Escherichia coli Isolates in the Absence of Antibiotics. Applied and environmental microbiology, 85(4), e02111-18.</strong> <a href="https://doi.org/10.1128/AEM.02111-18" target="_blank">https://doi.org/10.1128/AEM.02111-18</a></p>
<p><strong>1)</strong> For this example, we find the digital article by searching <a href="https://pubmed.ncbi.nlm.nih.gov" target="_blank"><strong>PubMed</strong></a>. The results show that it can be accessed by a <a href="https://pubmed.ncbi.nlm.nih.gov/30530714/" target="_blank"><strong>PubMed link</strong></a> or by a <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6365833/" target="_blank"><strong>PMC link</strong></a>.</p>
<p><img src="figures/session_14_figs/1_article_in_PMC.png" width="100%" /></p>
<p><strong>2)</strong> Most times, the authors will report that they have a supplementary material in their methods or in special section at the end of the article (e.g., footnotes, supplementary material, etc). In this case, it provides only a PDF and a link to journal digital article.</p>
<p><img src="figures/session_14_figs/2_article_SM_PMC.png" width="100%" /></p>
<p><strong>3)</strong> Like in many cases, there is no mention of a Dryad submission or DOI in the text of the paper. However, you can try to determine if such digital repository exists by searching the title of the article directly in <a href="https://datadryad.org/stash" target="_blank"><strong>DRYAD</strong></a>. I suggest doing this and not use the Dryad DOI (sometimes this is a broken link). To search in the Dryad website, you will paste the title of the article on the <strong>Search tab</strong> in the top right side.
<img src="figures/session_14_figs/3_dryad_intro.png" width="100%" /></p>
<p><strong>4)</strong> If associated data files exist for this paper, you will see how many entries are associated to the paper title. In this case, only one exists and you can click on such link.</p>
<p><img src="figures/session_14_figs/4_dryad_example_1.png" width="100%" /></p>
<p><strong>5)</strong> In the webpage associated with the data, you will see a title like <strong>Data from:…</strong>, a summary/abstract of the paper, and a link to the <strong>Data Files</strong> with a tab to <strong>Download dataset</strong>. You can click on the latter to get the associated zip file. For reference, the link for this datset in the DRYAD for this paper is in <a href="https://datadryad.org/stash/dataset/doi:10.5061/dryad.d0q331t"><strong>here</strong></a>.</p>
<p><img src="figures/session_14_figs/5_dryad_example_2.png" width="100%" /></p>
<p><strong>6)</strong> The associated data is in a compressed file called <code>doi_10.5061_dryad.d0q331t__v1.zip</code>. After uncompressing this file, you can identify several archives and folders that include a <code>Readme.txt</code> with explanation of the contents of each data file with a <code>*.csv</code> extension.</p>
<p><img src="figures/session_14_figs/6_dryad_example_3.png" width="100%" /></p>
<p>I added these files to the <a href="https://github.com/corytophanes/BIO_BIT_Bioinformatics_209/tree/main/ref_files_pdfs" target="_blank"><strong>LINK TO FILES of BIO/BIT 209</strong></a>.</p>
<p><strong>7)</strong> After downloading, you can explore those files in a text editor and they can be easily imported into R or excel. For reference, we can import two of these datasets (<code>ARG_Raw.csv</code> and <code>Growth_Curves.csv</code>) as explained above.</p>
<p>We can start with <code>ARG_Raw.csv</code></p>
<div class="sourceCode" id="cb665"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb665-1"><a href="importingexporting-data-into-r.html#cb665-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We start by readinf the metadata about such file in the text of README_for_Data_Files.txt. </span></span>
<span id="cb665-2"><a href="importingexporting-data-into-r.html#cb665-2" aria-hidden="true" tabindex="-1"></a><span class="do">## We found the following:</span></span>
<span id="cb665-3"><a href="importingexporting-data-into-r.html#cb665-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-4"><a href="importingexporting-data-into-r.html#cb665-4" aria-hidden="true" tabindex="-1"></a>ARG_Raw.csv</span>
<span id="cb665-5"><a href="importingexporting-data-into-r.html#cb665-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-6"><a href="importingexporting-data-into-r.html#cb665-6" aria-hidden="true" tabindex="-1"></a>This is a data table of the output of running BLASTn with the contigs of the draft </span>
<span id="cb665-7"><a href="importingexporting-data-into-r.html#cb665-7" aria-hidden="true" tabindex="-1"></a>genome assemblies of the isolates against the ResFinder <span class="fu">database</span> (nucleotide). The </span>
<span id="cb665-8"><a href="importingexporting-data-into-r.html#cb665-8" aria-hidden="true" tabindex="-1"></a>column headings mostly correspond to standard blast outputs but are outlined below<span class="sc">:</span></span>
<span id="cb665-9"><a href="importingexporting-data-into-r.html#cb665-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-10"><a href="importingexporting-data-into-r.html#cb665-10" aria-hidden="true" tabindex="-1"></a>   Strain<span class="sc">-</span>Which strain genome the hit was.</span>
<span id="cb665-11"><a href="importingexporting-data-into-r.html#cb665-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-12"><a href="importingexporting-data-into-r.html#cb665-12" aria-hidden="true" tabindex="-1"></a>   Contig<span class="sc">-</span>Most genomes are draft assemblies so there are multiple <span class="fu">segments</span> (here </span>
<span id="cb665-13"><a href="importingexporting-data-into-r.html#cb665-13" aria-hidden="true" tabindex="-1"></a>   called nodes), this shows which node the hit occurred in.</span>
<span id="cb665-14"><a href="importingexporting-data-into-r.html#cb665-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-15"><a href="importingexporting-data-into-r.html#cb665-15" aria-hidden="true" tabindex="-1"></a>   ARG<span class="sc">-</span> which antibiotic resistance gene <span class="cf">in</span> the query matched the sequence.</span>
<span id="cb665-16"><a href="importingexporting-data-into-r.html#cb665-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-17"><a href="importingexporting-data-into-r.html#cb665-17" aria-hidden="true" tabindex="-1"></a>   Per.ID<span class="sc">-</span> Percentage identity.</span>
<span id="cb665-18"><a href="importingexporting-data-into-r.html#cb665-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-19"><a href="importingexporting-data-into-r.html#cb665-19" aria-hidden="true" tabindex="-1"></a>   Length<span class="sc">-</span> Length of the matching region</span>
<span id="cb665-20"><a href="importingexporting-data-into-r.html#cb665-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-21"><a href="importingexporting-data-into-r.html#cb665-21" aria-hidden="true" tabindex="-1"></a>   MiMa<span class="sc">-</span> Number of mismatches </span>
<span id="cb665-22"><a href="importingexporting-data-into-r.html#cb665-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-23"><a href="importingexporting-data-into-r.html#cb665-23" aria-hidden="true" tabindex="-1"></a>   Gap<span class="sc">-</span> Total length of gaps</span>
<span id="cb665-24"><a href="importingexporting-data-into-r.html#cb665-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-25"><a href="importingexporting-data-into-r.html#cb665-25" aria-hidden="true" tabindex="-1"></a>   C.Start <span class="sc">&amp;</span> C.End<span class="sc">-</span> Start and end of the matching region on the contig</span>
<span id="cb665-26"><a href="importingexporting-data-into-r.html#cb665-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-27"><a href="importingexporting-data-into-r.html#cb665-27" aria-hidden="true" tabindex="-1"></a>   A.Start <span class="sc">&amp;</span> A.End<span class="sc">-</span> Start and end of the matching region on the query</span>
<span id="cb665-28"><a href="importingexporting-data-into-r.html#cb665-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-29"><a href="importingexporting-data-into-r.html#cb665-29" aria-hidden="true" tabindex="-1"></a>   E. value<span class="sc">-</span> Significance value <span class="cf">for</span> the hit, calculated within strain.</span>
<span id="cb665-30"><a href="importingexporting-data-into-r.html#cb665-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-31"><a href="importingexporting-data-into-r.html#cb665-31" aria-hidden="true" tabindex="-1"></a>   Bit.Score<span class="sc">-</span> overall score <span class="cf">for</span> the strength of the Hit. For overlapping</span>
<span id="cb665-32"><a href="importingexporting-data-into-r.html#cb665-32" aria-hidden="true" tabindex="-1"></a>   hits, only the highest bit score was retained.</span>
<span id="cb665-33"><a href="importingexporting-data-into-r.html#cb665-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-34"><a href="importingexporting-data-into-r.html#cb665-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-35"><a href="importingexporting-data-into-r.html#cb665-35" aria-hidden="true" tabindex="-1"></a><span class="do">## We need to find the path to the corresponding file by &quot;drag-and-drop&quot; in </span></span>
<span id="cb665-36"><a href="importingexporting-data-into-r.html#cb665-36" aria-hidden="true" tabindex="-1"></a><span class="do">## MACs or finding file properties in PCs. </span></span>
<span id="cb665-37"><a href="importingexporting-data-into-r.html#cb665-37" aria-hidden="true" tabindex="-1"></a><span class="do">## In my computer (MAC), they look like this:</span></span>
<span id="cb665-38"><a href="importingexporting-data-into-r.html#cb665-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-39"><a href="importingexporting-data-into-r.html#cb665-39" aria-hidden="true" tabindex="-1"></a><span class="sc">~</span><span class="er">/</span>Desktop<span class="sc">/</span>Teach_R<span class="sc">/</span>class_datasets<span class="sc">/</span>Dryad_example_for_Data_Files<span class="sc">/</span>ARG_Raw.csv</span>
<span id="cb665-40"><a href="importingexporting-data-into-r.html#cb665-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-41"><a href="importingexporting-data-into-r.html#cb665-41" aria-hidden="true" tabindex="-1"></a><span class="do">## Both files are *.csv (i.e., comma delimited files), so they can be easily </span></span>
<span id="cb665-42"><a href="importingexporting-data-into-r.html#cb665-42" aria-hidden="true" tabindex="-1"></a><span class="do">## imported with the read.table function. </span></span>
<span id="cb665-43"><a href="importingexporting-data-into-r.html#cb665-43" aria-hidden="true" tabindex="-1"></a><span class="do">## These files have data as comma delimited and we want to preserve headers</span></span>
<span id="cb665-44"><a href="importingexporting-data-into-r.html#cb665-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-45"><a href="importingexporting-data-into-r.html#cb665-45" aria-hidden="true" tabindex="-1"></a>ARG_Raw_Allen_etal_2019_df <span class="ot">&lt;-</span> <span class="fu">read.table</span> (<span class="at">file =</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets/Dryad_example_for_Data_Files/ARG_Raw.csv&quot;</span>, </span>
<span id="cb665-46"><a href="importingexporting-data-into-r.html#cb665-46" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb665-47"><a href="importingexporting-data-into-r.html#cb665-47" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>,</span>
<span id="cb665-48"><a href="importingexporting-data-into-r.html#cb665-48" aria-hidden="true" tabindex="-1"></a>                          <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb665-49"><a href="importingexporting-data-into-r.html#cb665-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-50"><a href="importingexporting-data-into-r.html#cb665-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-51"><a href="importingexporting-data-into-r.html#cb665-51" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ARG_Raw_Allen_etal_2019_df)</span>
<span id="cb665-52"><a href="importingexporting-data-into-r.html#cb665-52" aria-hidden="true" tabindex="-1"></a><span class="co">#    X Strain  Contig                  ARG  Per.ID Length MiMa Gap C.Start  C.End A.Start A.End E.val Bit.Score</span></span>
<span id="cb665-53"><a href="importingexporting-data-into-r.html#cb665-53" aria-hidden="true" tabindex="-1"></a><span class="co">#1  2 705963  NODE_6      mdf(A)_1_Y08743  97.891   1233   26   0   40970  42202       1  1233     0      2134</span></span>
<span id="cb665-54"><a href="importingexporting-data-into-r.html#cb665-54" aria-hidden="true" tabindex="-1"></a><span class="co">#2  1 706090  NODE_4      mdf(A)_1_Y08743  97.891   1233   26   0   19449  20681       1  1233     0      2134</span></span>
<span id="cb665-55"><a href="importingexporting-data-into-r.html#cb665-55" aria-hidden="true" tabindex="-1"></a><span class="co">#3 50 707404 NODE_12      mdf(A)_1_Y08743  97.567   1233   30   0  113267 114499    1233     1     0      2111</span></span>
<span id="cb665-56"><a href="importingexporting-data-into-r.html#cb665-56" aria-hidden="true" tabindex="-1"></a><span class="co">#4  6 707404 NODE_23      sul2_2_AY034138 100.000    816    0   0    8928   9743       1   816     0      1507</span></span>
<span id="cb665-57"><a href="importingexporting-data-into-r.html#cb665-57" aria-hidden="true" tabindex="-1"></a><span class="co">#5 28 707404 NODE_23 aph(3)-Ib_5_AF321551 100.000    529    0   0    9804  10332       1   529     0       977</span></span>
<span id="cb665-58"><a href="importingexporting-data-into-r.html#cb665-58" aria-hidden="true" tabindex="-1"></a><span class="co">#6 38 707404 NODE_23    dfrA14_5_DQ388123  99.793    483    1   0   10343  10825       1   483     0       887</span></span>
<span id="cb665-59"><a href="importingexporting-data-into-r.html#cb665-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-60"><a href="importingexporting-data-into-r.html#cb665-60" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(ARG_Raw_Allen_etal_2019_df)</span>
<span id="cb665-61"><a href="importingexporting-data-into-r.html#cb665-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  36 obs. of  14 variables:</span></span>
<span id="cb665-62"><a href="importingexporting-data-into-r.html#cb665-62" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X        : int  2 1 50 6 28 38 11 5 21 61 ...</span></span>
<span id="cb665-63"><a href="importingexporting-data-into-r.html#cb665-63" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Strain   : chr  &quot;705963&quot; &quot;706090&quot; &quot;707404&quot; &quot;707404&quot; ...</span></span>
<span id="cb665-64"><a href="importingexporting-data-into-r.html#cb665-64" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Contig   : chr  &quot;NODE_6&quot; &quot;NODE_4&quot; &quot;NODE_12&quot; &quot;NODE_23&quot; ...</span></span>
<span id="cb665-65"><a href="importingexporting-data-into-r.html#cb665-65" aria-hidden="true" tabindex="-1"></a><span class="co"># $ ARG      : chr  &quot;mdf(A)_1_Y08743&quot; &quot;mdf(A)_1_Y08743&quot; &quot;mdf(A)_1_Y08743&quot; &quot;sul2_2_AY034138&quot; ...</span></span>
<span id="cb665-66"><a href="importingexporting-data-into-r.html#cb665-66" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Per.ID   : num  97.9 97.9 97.6 100 100 ...</span></span>
<span id="cb665-67"><a href="importingexporting-data-into-r.html#cb665-67" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Length   : int  1233 1233 1233 816 529 483 837 1233 1206 861 ...</span></span>
<span id="cb665-68"><a href="importingexporting-data-into-r.html#cb665-68" aria-hidden="true" tabindex="-1"></a><span class="co"># $ MiMa     : int  26 26 30 0 0 1 0 26 0 0 ...</span></span>
<span id="cb665-69"><a href="importingexporting-data-into-r.html#cb665-69" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Gap      : int  0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb665-70"><a href="importingexporting-data-into-r.html#cb665-70" aria-hidden="true" tabindex="-1"></a><span class="co"># $ C.Start  : int  40970 19449 113267 8928 9804 10343 11175 28147 2128 4849 ...</span></span>
<span id="cb665-71"><a href="importingexporting-data-into-r.html#cb665-71" aria-hidden="true" tabindex="-1"></a><span class="co"># $ C.End    : int  42202 20681 114499 9743 10332 10825 12011 29379 3333 5709 ...</span></span>
<span id="cb665-72"><a href="importingexporting-data-into-r.html#cb665-72" aria-hidden="true" tabindex="-1"></a><span class="co"># $ A.Start  : int  1 1 1233 1 1 1 1 1233 1 1 ...</span></span>
<span id="cb665-73"><a href="importingexporting-data-into-r.html#cb665-73" aria-hidden="true" tabindex="-1"></a><span class="co"># $ A.End    : int  1233 1233 1 816 529 483 837 1 1206 861 ...</span></span>
<span id="cb665-74"><a href="importingexporting-data-into-r.html#cb665-74" aria-hidden="true" tabindex="-1"></a><span class="co"># $ E.val    : num  0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb665-75"><a href="importingexporting-data-into-r.html#cb665-75" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Bit.Score: int  2134 2134 2111 1507 977 887 1546 2134 2228 1591 ...</span></span>
<span id="cb665-76"><a href="importingexporting-data-into-r.html#cb665-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-77"><a href="importingexporting-data-into-r.html#cb665-77" aria-hidden="true" tabindex="-1"></a><span class="do">## We notice this data frame has LESS rows (only 36) than the original dataset </span></span>
<span id="cb665-78"><a href="importingexporting-data-into-r.html#cb665-78" aria-hidden="true" tabindex="-1"></a><span class="do">## that has 230 rows (you can check this by opening the ARG_Raw.csv file in a text editor). </span></span>
<span id="cb665-79"><a href="importingexporting-data-into-r.html#cb665-79" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ARG_Raw_Allen_etal_2019_df)</span>
<span id="cb665-80"><a href="importingexporting-data-into-r.html#cb665-80" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 36 14</span></span>
<span id="cb665-81"><a href="importingexporting-data-into-r.html#cb665-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-82"><a href="importingexporting-data-into-r.html#cb665-82" aria-hidden="true" tabindex="-1"></a><span class="do">## If we explore the original file: ARG_Raw.csv, this has special characters (&#39;). </span></span>
<span id="cb665-83"><a href="importingexporting-data-into-r.html#cb665-83" aria-hidden="true" tabindex="-1"></a><span class="do">## This character can be problematic, so we find a replace it  with &quot;_&quot; or removing </span></span>
<span id="cb665-84"><a href="importingexporting-data-into-r.html#cb665-84" aria-hidden="true" tabindex="-1"></a><span class="do">## them completely using a text editor. After removing this special character, we </span></span>
<span id="cb665-85"><a href="importingexporting-data-into-r.html#cb665-85" aria-hidden="true" tabindex="-1"></a><span class="do">## save the new modified file as ARG_Raw_modified.csv and we can import it to R.</span></span>
<span id="cb665-86"><a href="importingexporting-data-into-r.html#cb665-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-87"><a href="importingexporting-data-into-r.html#cb665-87" aria-hidden="true" tabindex="-1"></a>ARG_Raw_modified_Allen_etal_2019_df <span class="ot">&lt;-</span> <span class="fu">read.table</span> (<span class="at">file =</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets/Dryad_example_for_Data_Files/ARG_Raw_modified.csv&quot;</span>, </span>
<span id="cb665-88"><a href="importingexporting-data-into-r.html#cb665-88" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb665-89"><a href="importingexporting-data-into-r.html#cb665-89" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>,</span>
<span id="cb665-90"><a href="importingexporting-data-into-r.html#cb665-90" aria-hidden="true" tabindex="-1"></a>                          <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb665-91"><a href="importingexporting-data-into-r.html#cb665-91" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(ARG_Raw_modified_Allen_etal_2019_df)</span>
<span id="cb665-92"><a href="importingexporting-data-into-r.html#cb665-92" aria-hidden="true" tabindex="-1"></a><span class="co">#    X Strain  Contig                    ARG  Per.ID Length MiMa Gap C.Start  C.End A.Start A.End E.val Bit.Score</span></span>
<span id="cb665-93"><a href="importingexporting-data-into-r.html#cb665-93" aria-hidden="true" tabindex="-1"></a><span class="co">#1  2 705963  NODE_6        mdf(A)_1_Y08743  97.891   1233   26   0   40970  42202       1  1233     0      2134</span></span>
<span id="cb665-94"><a href="importingexporting-data-into-r.html#cb665-94" aria-hidden="true" tabindex="-1"></a><span class="co">#2  1 706090  NODE_4        mdf(A)_1_Y08743  97.891   1233   26   0   19449  20681       1  1233     0      2134</span></span>
<span id="cb665-95"><a href="importingexporting-data-into-r.html#cb665-95" aria-hidden="true" tabindex="-1"></a><span class="co">#3 50 707404 NODE_12        mdf(A)_1_Y08743  97.567   1233   30   0  113267 114499    1233     1     0      2111</span></span>
<span id="cb665-96"><a href="importingexporting-data-into-r.html#cb665-96" aria-hidden="true" tabindex="-1"></a><span class="co">#4  6 707404 NODE_23        sul2_2_AY034138 100.000    816    0   0    8928   9743       1   816     0      1507</span></span>
<span id="cb665-97"><a href="importingexporting-data-into-r.html#cb665-97" aria-hidden="true" tabindex="-1"></a><span class="co">#5 28 707404 NODE_23 aph(3__)-Ib_5_AF321551 100.000    529    0   0    9804  10332       1   529     0       977</span></span>
<span id="cb665-98"><a href="importingexporting-data-into-r.html#cb665-98" aria-hidden="true" tabindex="-1"></a><span class="co">#6 38 707404 NODE_23      dfrA14_5_DQ388123  99.793    483    1   0   10343  10825       1   483     0       887</span></span>
<span id="cb665-99"><a href="importingexporting-data-into-r.html#cb665-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-100"><a href="importingexporting-data-into-r.html#cb665-100" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(ARG_Raw_modified_Allen_etal_2019_df)</span>
<span id="cb665-101"><a href="importingexporting-data-into-r.html#cb665-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;&#39;data.frame&#39;: 230 obs. of  14 variables:</span></span>
<span id="cb665-102"><a href="importingexporting-data-into-r.html#cb665-102" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X        : int  2 1 50 6 28 38 11 5 21 61 ...</span></span>
<span id="cb665-103"><a href="importingexporting-data-into-r.html#cb665-103" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Strain   : chr  &quot;705963&quot; &quot;706090&quot; &quot;707404&quot; &quot;707404&quot; ...</span></span>
<span id="cb665-104"><a href="importingexporting-data-into-r.html#cb665-104" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Contig   : chr  &quot;NODE_6&quot; &quot;NODE_4&quot; &quot;NODE_12&quot; &quot;NODE_23&quot; ...</span></span>
<span id="cb665-105"><a href="importingexporting-data-into-r.html#cb665-105" aria-hidden="true" tabindex="-1"></a><span class="co"># $ ARG      : chr  &quot;mdf(A)_1_Y08743&quot; &quot;mdf(A)_1_Y08743&quot; &quot;mdf(A)_1_Y08743&quot; &quot;sul2_2_AY034138&quot; ...</span></span>
<span id="cb665-106"><a href="importingexporting-data-into-r.html#cb665-106" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Per.ID   : num  97.9 97.9 97.6 100 100 ...</span></span>
<span id="cb665-107"><a href="importingexporting-data-into-r.html#cb665-107" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Length   : int  1233 1233 1233 816 529 483 837 1233 1206 861 ...</span></span>
<span id="cb665-108"><a href="importingexporting-data-into-r.html#cb665-108" aria-hidden="true" tabindex="-1"></a><span class="co"># $ MiMa     : int  26 26 30 0 0 1 0 26 0 0 ...</span></span>
<span id="cb665-109"><a href="importingexporting-data-into-r.html#cb665-109" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Gap      : int  0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb665-110"><a href="importingexporting-data-into-r.html#cb665-110" aria-hidden="true" tabindex="-1"></a><span class="co"># $ C.Start  : int  40970 19449 113267 8928 9804 10343 11175 28147 2128 4849 ...</span></span>
<span id="cb665-111"><a href="importingexporting-data-into-r.html#cb665-111" aria-hidden="true" tabindex="-1"></a><span class="co"># $ C.End    : int  42202 20681 114499 9743 10332 10825 12011 29379 3333 5709 ...</span></span>
<span id="cb665-112"><a href="importingexporting-data-into-r.html#cb665-112" aria-hidden="true" tabindex="-1"></a><span class="co"># $ A.Start  : int  1 1 1233 1 1 1 1 1233 1 1 ...</span></span>
<span id="cb665-113"><a href="importingexporting-data-into-r.html#cb665-113" aria-hidden="true" tabindex="-1"></a><span class="co"># $ A.End    : int  1233 1233 1 816 529 483 837 1 1206 861 ...</span></span>
<span id="cb665-114"><a href="importingexporting-data-into-r.html#cb665-114" aria-hidden="true" tabindex="-1"></a><span class="co"># $ E.val    : num  0 0 0 0 0 0 0 0 0 0 ...</span></span>
<span id="cb665-115"><a href="importingexporting-data-into-r.html#cb665-115" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Bit.Score: int  2134 2134 2111 1507 977 887 1546 2134 2228 1591 ...</span></span>
<span id="cb665-116"><a href="importingexporting-data-into-r.html#cb665-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-117"><a href="importingexporting-data-into-r.html#cb665-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb665-118"><a href="importingexporting-data-into-r.html#cb665-118" aria-hidden="true" tabindex="-1"></a><span class="do">## We notice that have imported dataset has the corrected number of rows (230) </span></span>
<span id="cb665-119"><a href="importingexporting-data-into-r.html#cb665-119" aria-hidden="true" tabindex="-1"></a><span class="do">## and it is ready for our use in your analyses</span></span>
<span id="cb665-120"><a href="importingexporting-data-into-r.html#cb665-120" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(ARG_Raw_modified_Allen_etal_2019_df)</span>
<span id="cb665-121"><a href="importingexporting-data-into-r.html#cb665-121" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 230  14</span></span></code></pre></div>
<p>Now, we import the next file <code>Growth_Curves.csv</code>. We follow the similar procedure as explained above.</p>
<div class="sourceCode" id="cb666"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb666-1"><a href="importingexporting-data-into-r.html#cb666-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We start by readinf the metadata about such file in the text of README_for_Data_Files.txt. </span></span>
<span id="cb666-2"><a href="importingexporting-data-into-r.html#cb666-2" aria-hidden="true" tabindex="-1"></a><span class="do">## We found the following:</span></span>
<span id="cb666-3"><a href="importingexporting-data-into-r.html#cb666-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-4"><a href="importingexporting-data-into-r.html#cb666-4" aria-hidden="true" tabindex="-1"></a>Growth_Curves.csv</span>
<span id="cb666-5"><a href="importingexporting-data-into-r.html#cb666-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-6"><a href="importingexporting-data-into-r.html#cb666-6" aria-hidden="true" tabindex="-1"></a>This file contains the raw growth data which was used to fit the Gompertz growth curves. </span>
<span id="cb666-7"><a href="importingexporting-data-into-r.html#cb666-7" aria-hidden="true" tabindex="-1"></a>The column headings are<span class="sc">:</span></span>
<span id="cb666-8"><a href="importingexporting-data-into-r.html#cb666-8" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb666-9"><a href="importingexporting-data-into-r.html#cb666-9" aria-hidden="true" tabindex="-1"></a>   Master<span class="sc">-</span>Which of the two frozen master plates was used to inoculate the </span>
<span id="cb666-10"><a href="importingexporting-data-into-r.html#cb666-10" aria-hidden="true" tabindex="-1"></a>   overnight culture which inoculated this assay plate.</span>
<span id="cb666-11"><a href="importingexporting-data-into-r.html#cb666-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-12"><a href="importingexporting-data-into-r.html#cb666-12" aria-hidden="true" tabindex="-1"></a>   Sub.plate<span class="sc">-</span>Which of the <span class="dv">2</span> plates inoculated by the master plate this is. Thus </span>
<span id="cb666-13"><a href="importingexporting-data-into-r.html#cb666-13" aria-hidden="true" tabindex="-1"></a>   there are <span class="dv">4</span> replicates.</span>
<span id="cb666-14"><a href="importingexporting-data-into-r.html#cb666-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-15"><a href="importingexporting-data-into-r.html#cb666-15" aria-hidden="true" tabindex="-1"></a>   Treatment<span class="sc">-</span>Which of the <span class="dv">3</span> environmental conditions this growth was for.</span>
<span id="cb666-16"><a href="importingexporting-data-into-r.html#cb666-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-17"><a href="importingexporting-data-into-r.html#cb666-17" aria-hidden="true" tabindex="-1"></a>   Strain<span class="sc">-</span>Which strain the measurement came from. Those marked as blank had </span>
<span id="cb666-18"><a href="importingexporting-data-into-r.html#cb666-18" aria-hidden="true" tabindex="-1"></a>   no strain inoculated. Blanks have not been subtracted,there was a single </span>
<span id="cb666-19"><a href="importingexporting-data-into-r.html#cb666-19" aria-hidden="true" tabindex="-1"></a>   blank on each plate.</span>
<span id="cb666-20"><a href="importingexporting-data-into-r.html#cb666-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-21"><a href="importingexporting-data-into-r.html#cb666-21" aria-hidden="true" tabindex="-1"></a>   OD<span class="sc">-</span>Optical density measured at 595nm.</span>
<span id="cb666-22"><a href="importingexporting-data-into-r.html#cb666-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-23"><a href="importingexporting-data-into-r.html#cb666-23" aria-hidden="true" tabindex="-1"></a>   Read<span class="sc">-</span>An integer value <span class="cf">for</span> which read this is, takes values of <span class="dv">1</span><span class="fl">-41.</span></span>
<span id="cb666-24"><a href="importingexporting-data-into-r.html#cb666-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-25"><a href="importingexporting-data-into-r.html#cb666-25" aria-hidden="true" tabindex="-1"></a>   Time<span class="sc">-</span>The exact <span class="fu">time</span> (relative to the start of the growth experiment) </span>
<span id="cb666-26"><a href="importingexporting-data-into-r.html#cb666-26" aria-hidden="true" tabindex="-1"></a>   that the read was taken <span class="cf">in</span> hours, shown as a <span class="fu">fraction</span> (e.g. <span class="fl">4.5</span> is 4Hrs30Mins).</span>
<span id="cb666-27"><a href="importingexporting-data-into-r.html#cb666-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-28"><a href="importingexporting-data-into-r.html#cb666-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-29"><a href="importingexporting-data-into-r.html#cb666-29" aria-hidden="true" tabindex="-1"></a><span class="do">## We need to find the path to the corresponding file by &quot;drag-and_drop&quot; in MACs </span></span>
<span id="cb666-30"><a href="importingexporting-data-into-r.html#cb666-30" aria-hidden="true" tabindex="-1"></a><span class="do">## or finding file properties in PCs. </span></span>
<span id="cb666-31"><a href="importingexporting-data-into-r.html#cb666-31" aria-hidden="true" tabindex="-1"></a><span class="do">## In my computer (MAC), they look like this:</span></span>
<span id="cb666-32"><a href="importingexporting-data-into-r.html#cb666-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-33"><a href="importingexporting-data-into-r.html#cb666-33" aria-hidden="true" tabindex="-1"></a><span class="sc">~</span><span class="er">/</span>Desktop<span class="sc">/</span>Teach_R<span class="sc">/</span>class_datasets<span class="sc">/</span>Dryad_example_for_Data_Files<span class="sc">/</span>Growth_Curves.csv</span>
<span id="cb666-34"><a href="importingexporting-data-into-r.html#cb666-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-35"><a href="importingexporting-data-into-r.html#cb666-35" aria-hidden="true" tabindex="-1"></a><span class="do">## Both files are *.csv (i.e., comma delimited files), so they can be easily </span></span>
<span id="cb666-36"><a href="importingexporting-data-into-r.html#cb666-36" aria-hidden="true" tabindex="-1"></a><span class="do">## imported with the read.table function. These files have data as comma </span></span>
<span id="cb666-37"><a href="importingexporting-data-into-r.html#cb666-37" aria-hidden="true" tabindex="-1"></a><span class="do">## delimited and we want to preserve headers</span></span>
<span id="cb666-38"><a href="importingexporting-data-into-r.html#cb666-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-39"><a href="importingexporting-data-into-r.html#cb666-39" aria-hidden="true" tabindex="-1"></a>Growth_Curves_Allen_etal_2019_df <span class="ot">&lt;-</span> <span class="fu">read.table</span> (<span class="at">file =</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets/Dryad_example_for_Data_Files/Growth_Curves.csv&quot;</span>, </span>
<span id="cb666-40"><a href="importingexporting-data-into-r.html#cb666-40" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb666-41"><a href="importingexporting-data-into-r.html#cb666-41" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>,</span>
<span id="cb666-42"><a href="importingexporting-data-into-r.html#cb666-42" aria-hidden="true" tabindex="-1"></a>                          <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb666-43"><a href="importingexporting-data-into-r.html#cb666-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-44"><a href="importingexporting-data-into-r.html#cb666-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-45"><a href="importingexporting-data-into-r.html#cb666-45" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Growth_Curves_Allen_etal_2019_df)</span>
<span id="cb666-46"><a href="importingexporting-data-into-r.html#cb666-46" aria-hidden="true" tabindex="-1"></a><span class="co">#      X Master Sub.plate Treatment Strain     OD Read      Time</span></span>
<span id="cb666-47"><a href="importingexporting-data-into-r.html#cb666-47" aria-hidden="true" tabindex="-1"></a><span class="co">#1 1      2         1     Basal ECOR31 0.3877   38 44.488156</span></span>
<span id="cb666-48"><a href="importingexporting-data-into-r.html#cb666-48" aria-hidden="true" tabindex="-1"></a><span class="co">#2 2      2         1     Basal ECOR31 0.5019   12 13.760846</span></span>
<span id="cb666-49"><a href="importingexporting-data-into-r.html#cb666-49" aria-hidden="true" tabindex="-1"></a><span class="co">#3 3      2         1     Basal ECOR31 0.4839   11 12.588198</span></span>
<span id="cb666-50"><a href="importingexporting-data-into-r.html#cb666-50" aria-hidden="true" tabindex="-1"></a><span class="co">#4 4      2         1     Basal ECOR31 0.5245   20 23.211779</span></span>
<span id="cb666-51"><a href="importingexporting-data-into-r.html#cb666-51" aria-hidden="true" tabindex="-1"></a><span class="co">#5 5      2         1     Basal ECOR31 0.4581   29 33.816143</span></span>
<span id="cb666-52"><a href="importingexporting-data-into-r.html#cb666-52" aria-hidden="true" tabindex="-1"></a><span class="co">#6 6      2         1     Basal ECOR31 0.0590    2  1.977896</span></span>
<span id="cb666-53"><a href="importingexporting-data-into-r.html#cb666-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-54"><a href="importingexporting-data-into-r.html#cb666-54" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(Growth_Curves_Allen_etal_2019_df)</span>
<span id="cb666-55"><a href="importingexporting-data-into-r.html#cb666-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;&#39;data.frame&#39;: 45756 obs. of  8 variables:</span></span>
<span id="cb666-56"><a href="importingexporting-data-into-r.html#cb666-56" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X        : int  1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span id="cb666-57"><a href="importingexporting-data-into-r.html#cb666-57" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Master   : int  2 2 2 2 2 2 2 2 2 2 ...</span></span>
<span id="cb666-58"><a href="importingexporting-data-into-r.html#cb666-58" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Sub.plate: int  1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb666-59"><a href="importingexporting-data-into-r.html#cb666-59" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Treatment: chr  &quot;Basal&quot; &quot;Basal&quot; &quot;Basal&quot; &quot;Basal&quot; ...</span></span>
<span id="cb666-60"><a href="importingexporting-data-into-r.html#cb666-60" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Strain   : chr  &quot;ECOR31&quot; &quot;ECOR31&quot; &quot;ECOR31&quot; &quot;ECOR31&quot; ...</span></span>
<span id="cb666-61"><a href="importingexporting-data-into-r.html#cb666-61" aria-hidden="true" tabindex="-1"></a><span class="co"># $ OD       : num  0.388 0.502 0.484 0.524 0.458 ...</span></span>
<span id="cb666-62"><a href="importingexporting-data-into-r.html#cb666-62" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Read     : int  38 12 11 20 29 2 21 39 18 4 ...</span></span>
<span id="cb666-63"><a href="importingexporting-data-into-r.html#cb666-63" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Time     : num  44.5 13.8 12.6 23.2 33.8 ...</span></span>
<span id="cb666-64"><a href="importingexporting-data-into-r.html#cb666-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb666-65"><a href="importingexporting-data-into-r.html#cb666-65" aria-hidden="true" tabindex="-1"></a><span class="do">## We notice this data frame, this has EXACT number of rows than the original </span></span>
<span id="cb666-66"><a href="importingexporting-data-into-r.html#cb666-66" aria-hidden="true" tabindex="-1"></a><span class="do">## that the original files with 45756 rows. </span></span>
<span id="cb666-67"><a href="importingexporting-data-into-r.html#cb666-67" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(Growth_Curves_Allen_etal_2019_df)</span>
<span id="cb666-68"><a href="importingexporting-data-into-r.html#cb666-68" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 45756     8</span></span></code></pre></div>
</div>
<div id="accessing-online-data-gene-expression-omnibus-ncbi-geo" class="section level2 hasAnchor" number="30.12">
<h2><span class="header-section-number">30.12</span> Accessing online data: Gene Expression Omnibus (NCBI GEO)<a href="importingexporting-data-into-r.html#accessing-online-data-gene-expression-omnibus-ncbi-geo" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The Gene Expression Omnibus (GEO) <a href="https://www.ncbi.nlm.nih.gov/geo/" target="_blank"><strong>NCBI GEO</strong></a> is a public database of gene expression and functional genomics data that is relevant for biomedical research and drug discovery. The data relevant to functional genomics data such as gene expression profiles can be accessed in the <a href="https://www.ncbi.nlm.nih.gov/gds/" target="_blank"><strong>NCBI GEO DataSets</strong></a>. These public datasets include a diversity of data collections from microarrays, transcriptomics, epigenomics, and proteomics experiments. Many dataset are standard text files (tab-delimited, or comma-delimited) that you can download and import in R as described above.</p>
<p><img src="figures/session_14_figs/20_ncbi_geo.png" width="100%" /></p>
<p>You can search <a href="https://www.ncbi.nlm.nih.gov/gds/" target="_blank"><strong>NCBI GEO DataSets</strong></a> results based on keywords such as species, gene names, conditions, study types, or diverse filtering schemes. For more information on these see <a href="https://www.ncbi.nlm.nih.gov/geo/info/qqtutorial.html" target="_blank"><strong>GEO DataSets Query Examples</strong></a>.</p>
<p>For this example, I searched <strong>regeneration AND salamander</strong> and resulted in 2148 studies.</p>
<p><img src="figures/session_14_figs/21_ncbi_geo_res1.png" width="100%" /></p>
<p>I explored the first entry corresponding to this <a href="https://pubmed.ncbi.nlm.nih.gov/39575453/" target="_blank"><strong>study</strong></a>. The NCBI GEO included a detailed description of expression profiling experiments in the axolotl salamander <a href="https://en.wikipedia.org/wiki/Axolotl" target="_blank"><strong>Ambystoma mexicanum</strong></a>.
Including <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE237864" target="_blank"><strong>GEO accession</strong></a>, title, organism, experiment type, summary, overall design,</p>
<p><img src="figures/session_14_figs/22_ncbi_geo_res2.png" width="100%" /></p>
<p>contributors, citation, a submission descriptive, relationships with other NCBI submissions,</p>
<p><img src="figures/session_14_figs/23_ncbi_geo_res3.png" width="100%" /></p>
<p>and the files that can be download from the <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE237864" target="_blank"><strong>GEO accession</strong></a> submission such as the <code>GSE237864_count_matrix.txt.gz</code> that contain the NGS read counts used for differential gene expression analysis.</p>
<p><img src="figures/session_14_figs/24_ncbi_geo_res4.png" width="100%" /></p>
<p>I added this file uncompressed to the <a href="https://github.com/corytophanes/BIO_BIT_Bioinformatics_209/tree/main/ref_files_pdfs" target="_blank"><strong>LINK TO FILES of BIO/BIT 209</strong></a>. For reference, we can import this dataset as explained above.</p>
<div class="sourceCode" id="cb667"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb667-1"><a href="importingexporting-data-into-r.html#cb667-1" aria-hidden="true" tabindex="-1"></a>Salamander_Moral_Morales_etal_2024_df <span class="ot">&lt;-</span> <span class="fu">read.table</span> (<span class="at">file =</span> <span class="st">&quot;~/Desktop/Teach_R/class_datasets/GSE237864_count_matrix.txt&quot;</span>, </span>
<span id="cb667-2"><a href="importingexporting-data-into-r.html#cb667-2" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">header =</span> <span class="cn">TRUE</span>, </span>
<span id="cb667-3"><a href="importingexporting-data-into-r.html#cb667-3" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb667-4"><a href="importingexporting-data-into-r.html#cb667-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb667-5"><a href="importingexporting-data-into-r.html#cb667-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb667-6"><a href="importingexporting-data-into-r.html#cb667-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb667-7"><a href="importingexporting-data-into-r.html#cb667-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Salamander_Moral_Morales_etal_2024_df)</span>
<span id="cb667-8"><a href="importingexporting-data-into-r.html#cb667-8" aria-hidden="true" tabindex="-1"></a><span class="co">#   ensembl_gene_id_version X18134R.126.01 X18134R.126.02 X18134R.126.03 X18134R.126.04 X18134R.126.05 X18134R.126.06 X18134R.126.07</span></span>
<span id="cb667-9"><a href="importingexporting-data-into-r.html#cb667-9" aria-hidden="true" tabindex="-1"></a><span class="co">#1          AMEX60DD000001              2              0              8              1              0              3              0</span></span>
<span id="cb667-10"><a href="importingexporting-data-into-r.html#cb667-10" aria-hidden="true" tabindex="-1"></a><span class="co">#2          AMEX60DD000002              8             14              2              4              2              3              5</span></span>
<span id="cb667-11"><a href="importingexporting-data-into-r.html#cb667-11" aria-hidden="true" tabindex="-1"></a><span class="co">#3          AMEX60DD000003             88            123             99             53             59            176            121</span></span>
<span id="cb667-12"><a href="importingexporting-data-into-r.html#cb667-12" aria-hidden="true" tabindex="-1"></a><span class="co">#4          AMEX60DD000004             22             80             27             34             28             45             65</span></span>
<span id="cb667-13"><a href="importingexporting-data-into-r.html#cb667-13" aria-hidden="true" tabindex="-1"></a><span class="co">#5          AMEX60DD000005             11              2              0             11              1              4             10</span></span>
<span id="cb667-14"><a href="importingexporting-data-into-r.html#cb667-14" aria-hidden="true" tabindex="-1"></a><span class="co">#6          AMEX60DD000006              2              8              2              3              3              8              2</span></span>
<span id="cb667-15"><a href="importingexporting-data-into-r.html#cb667-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb667-16"><a href="importingexporting-data-into-r.html#cb667-16" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(Salamander_Moral_Morales_etal_2024_df)</span>
<span id="cb667-17"><a href="importingexporting-data-into-r.html#cb667-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;&#39;data.frame&#39;: 99088 obs. of  13 variables:</span></span>
<span id="cb667-18"><a href="importingexporting-data-into-r.html#cb667-18" aria-hidden="true" tabindex="-1"></a><span class="co"># $ ensembl_gene_id_version: chr  &quot;AMEX60DD000001&quot; &quot;AMEX60DD000002&quot; &quot;AMEX60DD000003&quot; &quot;AMEX60DD000004&quot; ...</span></span>
<span id="cb667-19"><a href="importingexporting-data-into-r.html#cb667-19" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X18134R.126.01         : int  2 8 88 22 11 2 0 213 0 0 ...</span></span>
<span id="cb667-20"><a href="importingexporting-data-into-r.html#cb667-20" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X18134R.126.02         : int  0 14 123 80 2 8 0 156 0 0 ...</span></span>
<span id="cb667-21"><a href="importingexporting-data-into-r.html#cb667-21" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X18134R.126.03         : int  8 2 99 27 0 2 0 276 0 0 ...</span></span>
<span id="cb667-22"><a href="importingexporting-data-into-r.html#cb667-22" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X18134R.126.04         : int  1 4 53 34 11 3 0 63 1 0 ...</span></span>
<span id="cb667-23"><a href="importingexporting-data-into-r.html#cb667-23" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X18134R.126.05         : int  0 2 59 28 1 3 0 63 0 0 ...</span></span>
<span id="cb667-24"><a href="importingexporting-data-into-r.html#cb667-24" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X18134R.126.06         : int  3 3 176 45 4 8 0 116 0 0 ...</span></span>
<span id="cb667-25"><a href="importingexporting-data-into-r.html#cb667-25" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X18134R.126.07         : int  0 5 121 65 10 2 0 75 0 0 ...</span></span>
<span id="cb667-26"><a href="importingexporting-data-into-r.html#cb667-26" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X18134R.126.08         : int  1 4 85 54 7 0 0 122 0 0 ...</span></span>
<span id="cb667-27"><a href="importingexporting-data-into-r.html#cb667-27" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X18134R.126.09         : int  4 26 164 65 12 6 0 165 0 0 ...</span></span>
<span id="cb667-28"><a href="importingexporting-data-into-r.html#cb667-28" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X18134R.126.10         : int  6 5 127 41 0 0 0 133 0 0 ...</span></span>
<span id="cb667-29"><a href="importingexporting-data-into-r.html#cb667-29" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X18134R.126.11         : int  2 20 176 63 9 7 0 195 0 0 ...</span></span>
<span id="cb667-30"><a href="importingexporting-data-into-r.html#cb667-30" aria-hidden="true" tabindex="-1"></a><span class="co"># $ X18134R.126.12         : int  3 3 73 46 5 4 0 164 0 0 ...</span></span>
<span id="cb667-31"><a href="importingexporting-data-into-r.html#cb667-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb667-32"><a href="importingexporting-data-into-r.html#cb667-32" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(Salamander_Moral_Morales_etal_2024_df)</span>
<span id="cb667-33"><a href="importingexporting-data-into-r.html#cb667-33" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 99088    13</span></span></code></pre></div>
</div>
<div id="accessing-online-data-clinicaltrials.gov" class="section level2 hasAnchor" number="30.13">
<h2><span class="header-section-number">30.13</span> Accessing online data: clinicaltrials.gov<a href="importingexporting-data-into-r.html#accessing-online-data-clinicaltrials.gov" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Sometimes you have data that is published in a website and you want to extract some for your analyses. For example,
<a href="https://clinicaltrials.gov/" target="_blank"><strong>NCBI Clinical Trials</strong></a> is an online registry and repository database
of research studies conducted in the United States and around the world about treatments, diseases, and others.</p>
<p><img src="figures/session_14_figs/11_clinical_trials.png" width="100%" /></p>
<p>You can search results of a clinical trial by key terms. For example, we could do a search for studies for asthma in Jamaica, NY, USA.</p>
<p><img src="figures/session_14_figs/12_asthma_in_jamaica.png" width="100%" /></p>
<p>As of today (02/01/2023) this search resulted in two completed studies on asthma that include people in Jamaica, NY, USA.</p>
<p><img src="figures/session_14_figs/13_asthma_page2.png" width="100%" /></p>
<p>If you click on the first study, you will get to the results for this locality.</p>
<p><img src="figures/session_14_figs/14_asthma_page3.png" width="100%" /></p>
<p>On this page, you can click on the <strong>Study Results</strong> tab and you can access the tabular results of this study. You could click on Beta.ClinicalTrials.gov tab for an alternative format view of these results.</p>
<p><img src="figures/session_14_figs/15_asthma_page4.png" width="100%" /></p>
<p><strong>Note:</strong> I could not find a easy way to get the results as tables other than copy such summary tables in text editor. More in the future.</p>
</div>
<div id="accessing-online-data-data-scrapping-from-a-website" class="section level2 hasAnchor" number="30.14">
<h2><span class="header-section-number">30.14</span> Accessing online data: Data scrapping from a website<a href="importingexporting-data-into-r.html#accessing-online-data-data-scrapping-from-a-website" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You can force text scraping from a website (e.g., the clinical trial results), but will require that you do the cleaning and selection of the data.
One useful R-package is <a href="https://cran.r-project.org/web/packages/rvest/index.html" target="_blank"><strong>rvest</strong></a> for an easy harvest (scrape) of web pages.
You can read the <a href="https://cran.r-project.org/web/packages/rvest/vignettes/rvest.html" target="_blank"><strong>rvest vignette</strong></a>.</p>
<p>A simple scrapping for a not a very complicated webpage.</p>
<div class="sourceCode" id="cb668"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb668-1"><a href="importingexporting-data-into-r.html#cb668-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Install and load rvest</span></span>
<span id="cb668-2"><a href="importingexporting-data-into-r.html#cb668-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;rvest&quot;</span>)</span>
<span id="cb668-3"><a href="importingexporting-data-into-r.html#cb668-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span> (rvest)</span>
<span id="cb668-4"><a href="importingexporting-data-into-r.html#cb668-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb668-5"><a href="importingexporting-data-into-r.html#cb668-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Change to working directory</span></span>
<span id="cb668-6"><a href="importingexporting-data-into-r.html#cb668-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb668-7"><a href="importingexporting-data-into-r.html#cb668-7" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/Some_directoty&quot;</span>)</span>
<span id="cb668-8"><a href="importingexporting-data-into-r.html#cb668-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb668-9"><a href="importingexporting-data-into-r.html#cb668-9" aria-hidden="true" tabindex="-1"></a><span class="co"># create an vector with web address </span></span>
<span id="cb668-10"><a href="importingexporting-data-into-r.html#cb668-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb668-11"><a href="importingexporting-data-into-r.html#cb668-11" aria-hidden="true" tabindex="-1"></a>html <span class="ot">&lt;-</span> <span class="fu">read_html</span>(<span class="st">&quot;https://www.stjohns.edu/academics/office-registrar/academic-calendar/final-exam-schedule&quot;</span>)</span>
<span id="cb668-12"><a href="importingexporting-data-into-r.html#cb668-12" aria-hidden="true" tabindex="-1"></a>html</span>
<span id="cb668-13"><a href="importingexporting-data-into-r.html#cb668-13" aria-hidden="true" tabindex="-1"></a><span class="co">#{{html_document}</span></span>
<span id="cb668-14"><a href="importingexporting-data-into-r.html#cb668-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&lt;html lang=&quot;en&quot; dir=&quot;ltr&quot; prefix=&quot;content: http://purl.org/rss/1.0/modules/content/  dc: http://purl.org/dc/terms/  foaf: http://xmlns.com/foaf/0.1/  og: http://ogp.me/ns#  rdfs: http://www.w3.org/2000/01/rdf-schema#  schema: http://schema.org/  sioc: http://rdfs.org/sioc/ns#  sioct: http://rdfs.org/sioc/types#  skos: http://www.w3.org/2004/02/skos/core#  xsd: http://www.w3.org/2001/XMLSchema# &quot;&gt;</span></span>
<span id="cb668-15"><a href="importingexporting-data-into-r.html#cb668-15" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &lt;head&gt;\n&lt;meta http-equiv=&quot;Content-Type&quot; content=&quot;text/html; charset=UTF-8&quot;&gt;\n&lt;meta charset=&quot;utf-8&quot;&gt;\n&lt;script type=&quot;text/javascript&quot;&gt;(window.NREUM|| ...</span></span>
<span id="cb668-16"><a href="importingexporting-data-into-r.html#cb668-16" aria-hidden="true" tabindex="-1"></a><span class="co">#[2] &lt;body&gt;\n    &lt;!-- Google Tag Manager --&gt;\n    &lt;noscript&gt;&lt;iframe src=&quot;//www.googletagmanager.com/ns.html?id=GTM-5STJNB&quot; height=&quot;0&quot; width=&quot;0&quot; style=&quot;d ...</span></span>
<span id="cb668-17"><a href="importingexporting-data-into-r.html#cb668-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb668-18"><a href="importingexporting-data-into-r.html#cb668-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Get dates as table</span></span>
<span id="cb668-19"><a href="importingexporting-data-into-r.html#cb668-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb668-20"><a href="importingexporting-data-into-r.html#cb668-20" aria-hidden="true" tabindex="-1"></a>exams_table <span class="ot">&lt;-</span> <span class="fu">html_node</span>(html , <span class="st">&quot;table&quot;</span>)</span>
<span id="cb668-21"><a href="importingexporting-data-into-r.html#cb668-21" aria-hidden="true" tabindex="-1"></a>exams_table_tibble <span class="ot">&lt;-</span> <span class="fu">html_table</span>(exams_table)</span>
<span id="cb668-22"><a href="importingexporting-data-into-r.html#cb668-22" aria-hidden="true" tabindex="-1"></a>exams_table_tibble</span>
<span id="cb668-23"><a href="importingexporting-data-into-r.html#cb668-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb668-24"><a href="importingexporting-data-into-r.html#cb668-24" aria-hidden="true" tabindex="-1"></a><span class="co"># A tibble: 31 × 4</span></span>
<span id="cb668-25"><a href="importingexporting-data-into-r.html#cb668-25" aria-hidden="true" tabindex="-1"></a><span class="co">#   `Class Meeting` Time                   `Final Exam/Final Class` Time                  </span></span>
<span id="cb668-26"><a href="importingexporting-data-into-r.html#cb668-26" aria-hidden="true" tabindex="-1"></a><span class="co">#   &lt;chr&gt;           &lt;chr&gt;                  &lt;chr&gt;                    &lt;chr&gt;                 </span></span>
<span id="cb668-27"><a href="importingexporting-data-into-r.html#cb668-27" aria-hidden="true" tabindex="-1"></a><span class="co"># 1 &quot;Days&quot;          &quot;Time&quot;                 &quot;Date&quot;                   &quot;Time&quot;                </span></span>
<span id="cb668-28"><a href="importingexporting-data-into-r.html#cb668-28" aria-hidden="true" tabindex="-1"></a><span class="co"># 2 &quot;M-R&quot;           &quot;9:05 a.m.-10:30 a.m.&quot; &quot;May 4 (R)&quot;              &quot;9:05 a.m.-11:05 a.m.&quot;</span></span>
<span id="cb668-29"><a href="importingexporting-data-into-r.html#cb668-29" aria-hidden="true" tabindex="-1"></a><span class="co"># 3 &quot;M-R&quot;           &quot;12:15 p.m.-1:40 p.m.&quot; &quot;May 4 (R)&quot;              &quot;12:15 p.m.-2:15 p.m.&quot;</span></span>
<span id="cb668-30"><a href="importingexporting-data-into-r.html#cb668-30" aria-hidden="true" tabindex="-1"></a><span class="co"># 4 &quot;M-R&quot;           &quot;5:00 p.m.-6:25 p.m.&quot;  &quot;May 4 (R)&quot;              &quot;5:00 p.m.-7:00 p.m.&quot; </span></span>
<span id="cb668-31"><a href="importingexporting-data-into-r.html#cb668-31" aria-hidden="true" tabindex="-1"></a><span class="co"># 5 &quot;M-R&quot;           &quot;8:45 p.m.-10:10 p.m.&quot; &quot;May 4 (R)&quot;              &quot;8:45 p.m.-10:45 p.m.&quot;</span></span>
<span id="cb668-32"><a href="importingexporting-data-into-r.html#cb668-32" aria-hidden="true" tabindex="-1"></a><span class="co"># 6 &quot;&quot;              &quot;&quot;                     &quot;&quot;                       &quot;&quot;                    </span></span>
<span id="cb668-33"><a href="importingexporting-data-into-r.html#cb668-33" aria-hidden="true" tabindex="-1"></a><span class="co"># 7 &quot;T-F&quot;           &quot;9:05 a.m.-10:30 a.m.&quot; &quot;May 5 (F)&quot;              &quot;9:05 a.m.-11:05 a.m.&quot;</span></span>
<span id="cb668-34"><a href="importingexporting-data-into-r.html#cb668-34" aria-hidden="true" tabindex="-1"></a><span class="co"># 8 &quot;T-F&quot;           &quot;12:15 p.m.-1:40 p.m.&quot; &quot;May 5 (F)&quot;              &quot;12:15 p.m.-2:15 p.m.&quot;</span></span>
<span id="cb668-35"><a href="importingexporting-data-into-r.html#cb668-35" aria-hidden="true" tabindex="-1"></a><span class="co"># 9 &quot;T-F&quot;           &quot;3:25 p.m.-4:50 p.m.&quot;  &quot;May 5 (F)&quot;              &quot;3:25 p.m.-5:25 p.m.&quot; </span></span>
<span id="cb668-36"><a href="importingexporting-data-into-r.html#cb668-36" aria-hidden="true" tabindex="-1"></a><span class="co">#10 &quot;T-F&quot;           &quot;7:10 p.m.-8:35 p.m.&quot;  &quot;May 5 (F)&quot;              &quot;7:10 p.m.-9:10 p.m.&quot; </span></span>
<span id="cb668-37"><a href="importingexporting-data-into-r.html#cb668-37" aria-hidden="true" tabindex="-1"></a><span class="co"># … with 21 more rows</span></span>
<span id="cb668-38"><a href="importingexporting-data-into-r.html#cb668-38" aria-hidden="true" tabindex="-1"></a><span class="co"># ℹ Use `print(n = ...)` to see more rows</span></span>
<span id="cb668-39"><a href="importingexporting-data-into-r.html#cb668-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb668-40"><a href="importingexporting-data-into-r.html#cb668-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb668-41"><a href="importingexporting-data-into-r.html#cb668-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Write such table in a text file in your working directory</span></span>
<span id="cb668-42"><a href="importingexporting-data-into-r.html#cb668-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb668-43"><a href="importingexporting-data-into-r.html#cb668-43" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(exams_table_tibble, <span class="st">&quot;sju_academic_calendar_scrap.csv&quot;</span>)</span>
<span id="cb668-44"><a href="importingexporting-data-into-r.html#cb668-44" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(exams_table_tibble, <span class="st">&quot;sju_academic_calendar_scrap.txt&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span></code></pre></div>
<p>A scrapping from more complex web pages will require you to work on the text data derived from it.</p>
<div class="sourceCode" id="cb669"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb669-1"><a href="importingexporting-data-into-r.html#cb669-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Change to working directory</span></span>
<span id="cb669-2"><a href="importingexporting-data-into-r.html#cb669-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb669-3"><a href="importingexporting-data-into-r.html#cb669-3" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/Some_directoty&quot;</span>)</span>
<span id="cb669-4"><a href="importingexporting-data-into-r.html#cb669-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb669-5"><a href="importingexporting-data-into-r.html#cb669-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Create an vector with web address (for example a clinical trial results)</span></span>
<span id="cb669-6"><a href="importingexporting-data-into-r.html#cb669-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb669-7"><a href="importingexporting-data-into-r.html#cb669-7" aria-hidden="true" tabindex="-1"></a>html <span class="ot">&lt;-</span> <span class="fu">read_html</span>(<span class="st">&quot;https://clinicaltrials.gov/ct2/show/results/NCT02414854?cond=Asthma&amp;cntry=US&amp;state=US%3ANY&amp;city=Jamaica&amp;draw=2&amp;rank=1&quot;</span>)</span>
<span id="cb669-8"><a href="importingexporting-data-into-r.html#cb669-8" aria-hidden="true" tabindex="-1"></a>html</span>
<span id="cb669-9"><a href="importingexporting-data-into-r.html#cb669-9" aria-hidden="true" tabindex="-1"></a><span class="co">#{html_document}</span></span>
<span id="cb669-10"><a href="importingexporting-data-into-r.html#cb669-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&lt;html xmlns=&quot;https://www.w3.org/1999/xhtml&quot; lang=&quot;en&quot;&gt;</span></span>
<span id="cb669-11"><a href="importingexporting-data-into-r.html#cb669-11" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &lt;head&gt;\n&lt;meta http-equiv=&quot;X-UA-Compatible&quot; content=&quot;IE=edge&quot;&gt;\n&lt;meta name=&quot;viewport&quot; content=&quot;width=device-width, initial-scale=1&quot;&gt;\n&lt;meta http-equ ...</span></span>
<span id="cb669-12"><a href="importingexporting-data-into-r.html#cb669-12" aria-hidden="true" tabindex="-1"></a><span class="co">#[2] &lt;body&gt; \n&lt;script type=&quot;text/javascript&quot;&gt;\n    jQuery.getScript(&quot;https://www.ncbi.nlm.nih.gov/core/alerts/alerts.js&quot;, function () {\n      galert([&#39; ...</span></span>
<span id="cb669-13"><a href="importingexporting-data-into-r.html#cb669-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb669-14"><a href="importingexporting-data-into-r.html#cb669-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Get all the text in a vector</span></span>
<span id="cb669-15"><a href="importingexporting-data-into-r.html#cb669-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb669-16"><a href="importingexporting-data-into-r.html#cb669-16" aria-hidden="true" tabindex="-1"></a>text_on_webpage <span class="ot">&lt;-</span> <span class="fu">html_text</span>(html)</span>
<span id="cb669-17"><a href="importingexporting-data-into-r.html#cb669-17" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(text_on_webpage)</span>
<span id="cb669-18"><a href="importingexporting-data-into-r.html#cb669-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb669-19"><a href="importingexporting-data-into-r.html#cb669-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Write such vector in a text file in your working directory</span></span>
<span id="cb669-20"><a href="importingexporting-data-into-r.html#cb669-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb669-21"><a href="importingexporting-data-into-r.html#cb669-21" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(text_on_webpage, <span class="st">&quot;text_on_webapge.txt&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="bioinformatics-literature.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="working-with-proteins.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/30-Import_data_in_R_v15.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
