<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Session 13 – Mining from a reference dataset | Bioinformatics Workshop Gitbook</title>
  <meta name="description" content="none" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Session 13 – Mining from a reference dataset | Bioinformatics Workshop Gitbook" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="none" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Session 13 – Mining from a reference dataset | Bioinformatics Workshop Gitbook" />
  
  <meta name="twitter:description" content="none" />
  

<meta name="author" content="Juan C. Santos &amp; Randy Ortiz" />


<meta name="date" content="2024-02-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="estimating-phylogenetic-trees.html"/>
<link rel="next" href="introduction-to-python.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics Workshop Gitbook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html"><i class="fa fa-check"></i><b>1</b> – Getting Started with R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#why-do-we-use-r"><i class="fa fa-check"></i><b>1.1</b> Why do we use R?</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#get-a-good-text-editor"><i class="fa fa-check"></i><b>1.2</b> Get a good text editor</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-r"><i class="fa fa-check"></i><b>1.4</b> Installing R</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#annotating-code"><i class="fa fa-check"></i><b>1.5</b> Annotating code</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#defining-a-working-directory-in-r"><i class="fa fa-check"></i><b>1.6</b> Defining a working directory in R</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html"><i class="fa fa-check"></i><b>2</b> – Using R, Installing Packages and Importing/Exporting Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#your-first-session"><i class="fa fa-check"></i><b>2.1</b> Your first session</a></li>
<li class="chapter" data-level="2.2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#doing-some-base-functions-and-getting-help"><i class="fa fa-check"></i><b>2.2</b> Doing some base functions and getting help</a></li>
<li class="chapter" data-level="2.3" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#functions-between-two-or-more-vectors"><i class="fa fa-check"></i><b>2.3</b> Functions between two or more vectors</a></li>
<li class="chapter" data-level="2.4" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#about-r-packages"><i class="fa fa-check"></i><b>2.4</b> About R-packages</a></li>
<li class="chapter" data-level="2.5" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#cran-task-views-expert-opinion-on-useful-packages"><i class="fa fa-check"></i><b>2.5</b> CRAN Task Views – Expert opinion on useful packages</a></li>
<li class="chapter" data-level="2.6" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#installing-packages"><i class="fa fa-check"></i><b>2.6</b> Installing packages</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html"><i class="fa fa-check"></i><b>3</b> – Basic Data Structures in R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#scalars-and-vectors"><i class="fa fa-check"></i><b>3.1</b> Scalars and vectors</a></li>
<li class="chapter" data-level="3.2" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#arrays"><i class="fa fa-check"></i><b>3.3</b> Arrays</a></li>
<li class="chapter" data-level="3.4" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a></li>
<li class="chapter" data-level="3.5" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tables"><i class="fa fa-check"></i><b>3.5</b> Tables</a></li>
<li class="chapter" data-level="3.6" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#lists"><i class="fa fa-check"></i><b>3.6</b> Lists</a></li>
<li class="chapter" data-level="3.7" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-tables"><i class="fa fa-check"></i><b>3.7</b> Data Tables</a></li>
<li class="chapter" data-level="3.8" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tibbles"><i class="fa fa-check"></i><b>3.8</b> Tibbles</a></li>
<li class="chapter" data-level="3.9" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#dnastringsets-rnastringsets-and-aastringsets"><i class="fa fa-check"></i><b>3.9</b> DNAStringSets, RNAStringSets and AAStringSets</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html"><i class="fa fa-check"></i><b>4</b> – Text Editing and Data Transformations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#working-with-character-strings"><i class="fa fa-check"></i><b>4.1</b> Working with character strings</a></li>
<li class="chapter" data-level="4.2" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#regular-expressions-regex"><i class="fa fa-check"></i><b>4.2</b> Regular expressions (Regex)</a></li>
<li class="chapter" data-level="4.3" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#comparing-ordering-and-simplifying-vectors"><i class="fa fa-check"></i><b>4.3</b> Comparing, ordering and simplifying vectors</a></li>
<li class="chapter" data-level="4.4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#useful-functions-of-stringr"><i class="fa fa-check"></i><b>4.4</b> Useful functions of <code>stringr</code></a></li>
<li class="chapter" data-level="4.5" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#data-transformations"><i class="fa fa-check"></i><b>4.5</b> Data transformations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html"><i class="fa fa-check"></i><b>5</b> – Getting Biological Data from Public Repositories</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#ncbi-databases-genbank-and-others"><i class="fa fa-check"></i><b>5.1</b> NCBI databases: GenBank and others</a></li>
<li class="chapter" data-level="5.2" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#accessing-ncbi-within-r-rentrez"><i class="fa fa-check"></i><b>5.2</b> Accessing NCBI within R: <code>rentrez</code></a></li>
<li class="chapter" data-level="5.3" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#retrieving-pubmed-references"><i class="fa fa-check"></i><b>5.3</b> Retrieving PubMed references</a></li>
<li class="chapter" data-level="5.4" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#comparing-sequences-blast"><i class="fa fa-check"></i><b>5.4</b> Comparing sequences: BLAST</a></li>
<li class="chapter" data-level="5.5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-uniprot"><i class="fa fa-check"></i><b>5.5</b> Other databases: UniProt</a></li>
<li class="chapter" data-level="5.6" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-ensembl"><i class="fa fa-check"></i><b>5.6</b> Other databases: Ensembl</a></li>
<li class="chapter" data-level="5.7" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-taxonomy-phylogenetic-trees-and-species-distributions"><i class="fa fa-check"></i><b>5.7</b> Other databases: Taxonomy, phylogenetic trees and species distributions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html"><i class="fa fa-check"></i><b>6</b> – Basic Statistics in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#exploring-your-data"><i class="fa fa-check"></i><b>6.1</b> Exploring your data</a></li>
<li class="chapter" data-level="6.2" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#sub-setting-your-dataset"><i class="fa fa-check"></i><b>6.2</b> Sub-setting your dataset</a></li>
<li class="chapter" data-level="6.3" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#transposing-rotating-your-dataset"><i class="fa fa-check"></i><b>6.3</b> Transposing (rotating) your dataset</a></li>
<li class="chapter" data-level="6.4" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#dealing-with-missing-na-values"><i class="fa fa-check"></i><b>6.4</b> Dealing with missing (NA) values</a></li>
<li class="chapter" data-level="6.5" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#about-p-values"><i class="fa fa-check"></i><b>6.5</b> About p-values</a></li>
<li class="chapter" data-level="6.6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#how-to-read-p-values"><i class="fa fa-check"></i><b>6.6</b> How to read p-values</a></li>
<li class="chapter" data-level="6.7" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#multiple-testing-corrections"><i class="fa fa-check"></i><b>6.7</b> Multiple testing corrections</a></li>
<li class="chapter" data-level="6.8" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#normalizing-your-data"><i class="fa fa-check"></i><b>6.8</b> Normalizing your data</a></li>
<li class="chapter" data-level="6.9" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#what-test-should-i-use"><i class="fa fa-check"></i><b>6.9</b> What test should I use?</a></li>
<li class="chapter" data-level="6.10" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#type-of-variables-in-dataset"><i class="fa fa-check"></i><b>6.10</b> Type of variables in dataset</a></li>
<li class="chapter" data-level="6.11" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-means-t-test"><i class="fa fa-check"></i><b>6.11</b> Comparing means: t-test</a></li>
<li class="chapter" data-level="6.12" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#two-sample-t-test"><i class="fa fa-check"></i><b>6.12</b> Two sample t-test</a></li>
<li class="chapter" data-level="6.13" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-wilcoxon-signed-rank-test"><i class="fa fa-check"></i><b>6.13</b> Beyond t-test: Wilcoxon signed rank test</a></li>
<li class="chapter" data-level="6.14" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#tabulation-and-cross-tabulation"><i class="fa fa-check"></i><b>6.14</b> Tabulation and Cross-Tabulation</a></li>
<li class="chapter" data-level="6.15" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-binomial-test"><i class="fa fa-check"></i><b>6.15</b> Beyond t-test: Binomial test</a></li>
<li class="chapter" data-level="6.16" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-chi-square-goodness-of-fit"><i class="fa fa-check"></i><b>6.16</b> Beyond t-test: Chi-square goodness of fit</a></li>
<li class="chapter" data-level="6.17" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-mcnemars-test"><i class="fa fa-check"></i><b>6.17</b> Beyond t-test: McNemar’s test</a></li>
<li class="chapter" data-level="6.18" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-fishers-exact-test"><i class="fa fa-check"></i><b>6.18</b> Beyond t-test: Fisher’s exact test</a></li>
<li class="chapter" data-level="6.19" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#correlations"><i class="fa fa-check"></i><b>6.19</b> Correlations</a></li>
<li class="chapter" data-level="6.20" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-distributions"><i class="fa fa-check"></i><b>6.20</b> Comparing distributions</a></li>
<li class="chapter" data-level="6.21" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#bootstraps"><i class="fa fa-check"></i><b>6.21</b> Bootstraps</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html"><i class="fa fa-check"></i><b>7</b> – Regressions and ANOVA in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#linear-regressions"><i class="fa fa-check"></i><b>7.1</b> Linear regressions</a></li>
<li class="chapter" data-level="7.2" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anatomy-of-a-linear-regression"><i class="fa fa-check"></i><b>7.2</b> Anatomy of a linear regression</a></li>
<li class="chapter" data-level="7.3" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#multiple-linear-regressions"><i class="fa fa-check"></i><b>7.3</b> Multiple linear regressions</a></li>
<li class="chapter" data-level="7.4" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#diagnostic-plots-for-regressions"><i class="fa fa-check"></i><b>7.4</b> Diagnostic plots for regressions</a></li>
<li class="chapter" data-level="7.5" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-influential-measurements"><i class="fa fa-check"></i><b>7.5</b> Regression influential measurements</a></li>
<li class="chapter" data-level="7.6" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-anova-coefficients-ci"><i class="fa fa-check"></i><b>7.6</b> Regression (ANOVA, coefficients, CI)</a></li>
<li class="chapter" data-level="7.7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#selecting-predictors"><i class="fa fa-check"></i><b>7.7</b> Selecting predictors</a></li>
<li class="chapter" data-level="7.8" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-predictions"><i class="fa fa-check"></i><b>7.8</b> Regression predictions</a></li>
<li class="chapter" data-level="7.9" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions"><i class="fa fa-check"></i><b>7.9</b> Logistic regressions</a></li>
<li class="chapter" data-level="7.10" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions-requirements"><i class="fa fa-check"></i><b>7.10</b> Logistic regressions requirements</a></li>
<li class="chapter" data-level="7.11" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#non-linear-regressions"><i class="fa fa-check"></i><b>7.11</b> Non-linear regressions</a></li>
<li class="chapter" data-level="7.12" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-regressions"><i class="fa fa-check"></i><b>7.12</b> Toy datasets for regressions</a></li>
<li class="chapter" data-level="7.13" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova"><i class="fa fa-check"></i><b>7.13</b> ANOVA</a></li>
<li class="chapter" data-level="7.14" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#one-way-anova"><i class="fa fa-check"></i><b>7.14</b> One-way ANOVA</a></li>
<li class="chapter" data-level="7.15" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#two-way-anova"><i class="fa fa-check"></i><b>7.15</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="7.16" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova-post-hoc-tests"><i class="fa fa-check"></i><b>7.16</b> ANOVA Post-hoc tests</a></li>
<li class="chapter" data-level="7.17" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#beyond-anova-kruskal-wallis-test"><i class="fa fa-check"></i><b>7.17</b> Beyond ANOVA: Kruskal-Wallis Test</a></li>
<li class="chapter" data-level="7.18" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-anova"><i class="fa fa-check"></i><b>7.18</b> Toy datasets for ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html"><i class="fa fa-check"></i><b>8</b> – Multidimensional Data in R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#principal-component-analysis"><i class="fa fa-check"></i><b>8.1</b> Principal Component Analysis</a></li>
<li class="chapter" data-level="8.2" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#multidimensional-scaling"><i class="fa fa-check"></i><b>8.2</b> Multidimensional scaling</a></li>
<li class="chapter" data-level="8.3" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-dfa"><i class="fa fa-check"></i><b>8.3</b> Cluster analysis: DFA</a></li>
<li class="chapter" data-level="8.4" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-k-means-unsupervised"><i class="fa fa-check"></i><b>8.4</b> Cluster analysis: <em>k</em>-means (unsupervised)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html"><i class="fa fa-check"></i><b>9</b> – Graphics and Colors</a>
<ul>
<li class="chapter" data-level="9.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#avoid-bad-graphics"><i class="fa fa-check"></i><b>9.1</b> Avoid bad graphics</a></li>
<li class="chapter" data-level="9.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#graphics-in-r-ggplot2"><i class="fa fa-check"></i><b>9.2</b> Graphics in R – ggplot2</a></li>
<li class="chapter" data-level="9.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#preparing-your-dataset"><i class="fa fa-check"></i><b>9.3</b> Preparing your dataset</a></li>
<li class="chapter" data-level="9.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#color-palettes-online"><i class="fa fa-check"></i><b>9.4</b> Color palettes online</a></li>
<li class="chapter" data-level="9.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggplot2-default-palette"><i class="fa fa-check"></i><b>9.5</b> <code>ggplot2</code> default palette</a></li>
<li class="chapter" data-level="9.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#heatmaps-or-tiled-charts"><i class="fa fa-check"></i><b>9.6</b> Heatmaps or tiled charts</a></li>
<li class="chapter" data-level="9.7" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#r-packages-about-color"><i class="fa fa-check"></i><b>9.7</b> R-packages about color</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#viridis-package"><i class="fa fa-check"></i><b>9.7.1</b> <code>viridis</code> package</a></li>
<li class="chapter" data-level="9.7.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcolorbrewer-package"><i class="fa fa-check"></i><b>9.7.2</b> <code>RColorBrewer</code> package</a></li>
<li class="chapter" data-level="9.7.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggsci-package"><i class="fa fa-check"></i><b>9.7.3</b> <code>ggsci</code> package</a></li>
<li class="chapter" data-level="9.7.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#wesanderson-package"><i class="fa fa-check"></i><b>9.7.4</b> <code>wesanderson</code> package</a></li>
<li class="chapter" data-level="9.7.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcartocolor-package"><i class="fa fa-check"></i><b>9.7.5</b> <code>rcartocolor</code> package</a></li>
<li class="chapter" data-level="9.7.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#colorspace-package"><i class="fa fa-check"></i><b>9.7.6</b> <code>colorspace</code> package</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#making-your-own-palette"><i class="fa fa-check"></i><b>9.8</b> Making your own palette</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="basic-graphics.html"><a href="basic-graphics.html"><i class="fa fa-check"></i><b>10</b> – Basic Graphics</a>
<ul>
<li class="chapter" data-level="10.1" data-path="basic-graphics.html"><a href="basic-graphics.html#what-would-you-like-to-show"><i class="fa fa-check"></i><b>10.1</b> What would you like to show?</a></li>
<li class="chapter" data-level="10.2" data-path="basic-graphics.html"><a href="basic-graphics.html#bar-charts"><i class="fa fa-check"></i><b>10.2</b> Bar charts</a></li>
<li class="chapter" data-level="10.3" data-path="basic-graphics.html"><a href="basic-graphics.html#stacked-bar-charts"><i class="fa fa-check"></i><b>10.3</b> Stacked bar charts</a></li>
<li class="chapter" data-level="10.4" data-path="basic-graphics.html"><a href="basic-graphics.html#histograms"><i class="fa fa-check"></i><b>10.4</b> Histograms</a></li>
<li class="chapter" data-level="10.5" data-path="basic-graphics.html"><a href="basic-graphics.html#multiple-histograms"><i class="fa fa-check"></i><b>10.5</b> Multiple histograms</a></li>
<li class="chapter" data-level="10.6" data-path="basic-graphics.html"><a href="basic-graphics.html#density-plots"><i class="fa fa-check"></i><b>10.6</b> Density plots</a></li>
<li class="chapter" data-level="10.7" data-path="basic-graphics.html"><a href="basic-graphics.html#box-plots"><i class="fa fa-check"></i><b>10.7</b> Box plots</a></li>
<li class="chapter" data-level="10.8" data-path="basic-graphics.html"><a href="basic-graphics.html#violin-plots"><i class="fa fa-check"></i><b>10.8</b> Violin plots</a></li>
<li class="chapter" data-level="10.9" data-path="basic-graphics.html"><a href="basic-graphics.html#scatterplots"><i class="fa fa-check"></i><b>10.9</b> Scatterplots</a></li>
<li class="chapter" data-level="10.10" data-path="basic-graphics.html"><a href="basic-graphics.html#lines-and-scatterplots"><i class="fa fa-check"></i><b>10.10</b> Lines and Scatterplots</a></li>
<li class="chapter" data-level="10.11" data-path="basic-graphics.html"><a href="basic-graphics.html#save-your-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>10.11</b> Save your plots as PDF or PNG</a></li>
<li class="chapter" data-level="10.12" data-path="basic-graphics.html"><a href="basic-graphics.html#ggplot2-add-on-packages-treemapify"><i class="fa fa-check"></i><b>10.12</b> ggplot2 add-on packages: <code>treemapify</code></a></li>
<li class="chapter" data-level="10.13" data-path="basic-graphics.html"><a href="basic-graphics.html#other-resources"><i class="fa fa-check"></i><b>10.13</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="sequence-alignments.html"><a href="sequence-alignments.html"><i class="fa fa-check"></i><b>11</b> – Sequence Alignments</a>
<ul>
<li class="chapter" data-level="11.1" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-using-rentrez"><i class="fa fa-check"></i><b>11.1</b> Retrieving sequences using ‘rentrez’</a></li>
<li class="chapter" data-level="11.2" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-manually-from-ncbi"><i class="fa fa-check"></i><b>11.2</b> Retrieving sequences manually from NCBI</a></li>
<li class="chapter" data-level="11.3" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-accession-numbers"><i class="fa fa-check"></i><b>11.3</b> Retrieving accession numbers</a></li>
<li class="chapter" data-level="11.4" data-path="sequence-alignments.html"><a href="sequence-alignments.html#alignment-with-decipher"><i class="fa fa-check"></i><b>11.4</b> Alignment with <code>DECIPHER</code></a></li>
<li class="chapter" data-level="11.5" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualize-alignment"><i class="fa fa-check"></i><b>11.5</b> Visualize alignment</a></li>
<li class="chapter" data-level="11.6" data-path="sequence-alignments.html"><a href="sequence-alignments.html#save-alignment-to-file"><i class="fa fa-check"></i><b>11.6</b> Save alignment to file</a></li>
<li class="chapter" data-level="11.7" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-and-align-several-markers"><i class="fa fa-check"></i><b>11.7</b> Concatenate and align several markers</a></li>
<li class="chapter" data-level="11.8" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-data.table"><i class="fa fa-check"></i><b>11.8</b> Concatenate with <code>data.table</code></a></li>
<li class="chapter" data-level="11.9" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-biostrings"><i class="fa fa-check"></i><b>11.9</b> Concatenate with <code>Biostrings</code></a></li>
<li class="chapter" data-level="11.10" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualizing-alignments-in-r"><i class="fa fa-check"></i><b>11.10</b> Visualizing alignments in R</a></li>
<li class="chapter" data-level="11.11" data-path="sequence-alignments.html"><a href="sequence-alignments.html#other-sequence-alignment-software"><i class="fa fa-check"></i><b>11.11</b> Other sequence alignment software</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html"><i class="fa fa-check"></i><b>12</b> – Estimating Phylogenetic Trees</a>
<ul>
<li class="chapter" data-level="12.1" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#phylogeny-inference-with-iqtree2"><i class="fa fa-check"></i><b>12.1</b> Phylogeny inference with <code>iqtree2</code></a></li>
<li class="chapter" data-level="12.2" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-iqtree2-in-macos"><i class="fa fa-check"></i><b>12.2</b> Installing <code>iqtree2</code> in macOS</a></li>
<li class="chapter" data-level="12.3" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-and-running-iqtree2-in-a-pc"><i class="fa fa-check"></i><b>12.3</b> Installing and running <code>iqtree2</code> in a PC</a></li>
<li class="chapter" data-level="12.4" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#loading-the-iqtree_runner_r-function"><i class="fa fa-check"></i><b>12.4</b> Loading the <code>iqtree_runner_R</code> function</a></li>
<li class="chapter" data-level="12.5" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#running-iqtree2-from-r"><i class="fa fa-check"></i><b>12.5</b> Running <code>iqtree2</code> from R</a></li>
<li class="chapter" data-level="12.6" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-no-partitions"><i class="fa fa-check"></i><b>12.6</b> One marker and no partitions</a></li>
<li class="chapter" data-level="12.7" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-with-partitions"><i class="fa fa-check"></i><b>12.7</b> One marker and with partitions</a></li>
<li class="chapter" data-level="12.8" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#bootstrap-support"><i class="fa fa-check"></i><b>12.8</b> Bootstrap support</a></li>
<li class="chapter" data-level="12.9" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#many-markers-and-with-partitions"><i class="fa fa-check"></i><b>12.9</b> Many markers and with partitions</a></li>
<li class="chapter" data-level="12.10" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#other-sequence-phylogenetic-tree-estimation-software"><i class="fa fa-check"></i><b>12.10</b> Other sequence phylogenetic tree estimation software</a></li>
<li class="chapter" data-level="12.11" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#drawing-phylogenetic-trees-with-ggtree"><i class="fa fa-check"></i><b>12.11</b> Drawing phylogenetic trees with <code>ggtree</code></a></li>
<li class="chapter" data-level="12.12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#save-ggtree-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>12.12</b> Save ggtree plots as PDF or PNG</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html"><i class="fa fa-check"></i><b>13</b> – Mining from a reference dataset</a>
<ul>
<li class="chapter" data-level="13.1" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#a-worked-example-using-the-mtcars-dataset"><i class="fa fa-check"></i><b>13.1</b> A worked example using the <code>mtcars</code> dataset</a></li>
<li class="chapter" data-level="13.2" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-reference-dataset"><i class="fa fa-check"></i><b>13.2</b> Build a reference dataset</a></li>
<li class="chapter" data-level="13.3" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-sampling-dataset"><i class="fa fa-check"></i><b>13.3</b> Build a sampling dataset</a></li>
<li class="chapter" data-level="13.4" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#extracting-from-reference-dataset"><i class="fa fa-check"></i><b>13.4</b> Extracting from reference dataset</a></li>
<li class="chapter" data-level="13.5" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#plots-of-the-extracted-data"><i class="fa fa-check"></i><b>13.5</b> Plots of the extracted data</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="introduction-to-python.html"><a href="introduction-to-python.html"><i class="fa fa-check"></i><b>14</b> – Introduction to Python</a>
<ul>
<li class="chapter" data-level="14.1" data-path="introduction-to-python.html"><a href="introduction-to-python.html#welcome-to-python-why-python"><i class="fa fa-check"></i><b>14.1</b> Welcome to Python! Why Python?</a></li>
<li class="chapter" data-level="14.2" data-path="introduction-to-python.html"><a href="introduction-to-python.html#which-python-should-i-install"><i class="fa fa-check"></i><b>14.2</b> Which Python should I install?</a></li>
<li class="chapter" data-level="14.3" data-path="introduction-to-python.html"><a href="introduction-to-python.html#mac-os-x-python-installations"><i class="fa fa-check"></i><b>14.3</b> Mac OS X Python Installations</a></li>
<li class="chapter" data-level="14.4" data-path="introduction-to-python.html"><a href="introduction-to-python.html#windows-python-installations"><i class="fa fa-check"></i><b>14.4</b> Windows Python Installations</a></li>
<li class="chapter" data-level="14.5" data-path="introduction-to-python.html"><a href="introduction-to-python.html#install-an-ide-or-integrated-development-environment"><i class="fa fa-check"></i><b>14.5</b> Install an IDE, or integrated development environment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html"><i class="fa fa-check"></i><b>15</b> – Python: Keywords and Identifiers</a>
<ul>
<li class="chapter" data-level="15.1" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#keywords-in-python"><i class="fa fa-check"></i><b>15.1</b> Keywords in Python</a></li>
<li class="chapter" data-level="15.2" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#identifiers-in-python"><i class="fa fa-check"></i><b>15.2</b> Identifiers in Python</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html"><i class="fa fa-check"></i><b>16</b> – Python: Statements and Expressions</a>
<ul>
<li class="chapter" data-level="16.1" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#what-are-statements"><i class="fa fa-check"></i><b>16.1</b> What are statements?</a></li>
<li class="chapter" data-level="16.2" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#expressions-in-python"><i class="fa fa-check"></i><b>16.2</b> Expressions in Python</a></li>
<li class="chapter" data-level="16.3" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#simple-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.3</b> Simple Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.4" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#augmented-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.4</b> Augmented Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.5" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#explicit-and-implicit-line-continuation"><i class="fa fa-check"></i><b>16.5</b> Explicit and Implicit Line Continuation</a></li>
<li class="chapter" data-level="16.6" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#indentation-in-python-is-crucial-to-proper-code-function"><i class="fa fa-check"></i><b>16.6</b> Indentation in Python is Crucial to Proper Code function</a></li>
<li class="chapter" data-level="16.7" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#comments"><i class="fa fa-check"></i><b>16.7</b> Comments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="python-data-types.html"><a href="python-data-types.html"><i class="fa fa-check"></i><b>17</b> – Python: Data Types</a>
<ul>
<li class="chapter" data-level="17.1" data-path="python-data-types.html"><a href="python-data-types.html#booleans"><i class="fa fa-check"></i><b>17.1</b> Booleans</a></li>
<li class="chapter" data-level="17.2" data-path="python-data-types.html"><a href="python-data-types.html#numbers-int-float-and-complex"><i class="fa fa-check"></i><b>17.2</b> Numbers: int, float and complex</a></li>
<li class="chapter" data-level="17.3" data-path="python-data-types.html"><a href="python-data-types.html#strings"><i class="fa fa-check"></i><b>17.3</b> Strings</a></li>
<li class="chapter" data-level="17.4" data-path="python-data-types.html"><a href="python-data-types.html#bytes"><i class="fa fa-check"></i><b>17.4</b> Bytes</a></li>
<li class="chapter" data-level="17.5" data-path="python-data-types.html"><a href="python-data-types.html#lists-1"><i class="fa fa-check"></i><b>17.5</b> Lists</a></li>
<li class="chapter" data-level="17.6" data-path="python-data-types.html"><a href="python-data-types.html#tuples"><i class="fa fa-check"></i><b>17.6</b> Tuples</a></li>
<li class="chapter" data-level="17.7" data-path="python-data-types.html"><a href="python-data-types.html#sets"><i class="fa fa-check"></i><b>17.7</b> Sets</a></li>
<li class="chapter" data-level="17.8" data-path="python-data-types.html"><a href="python-data-types.html#dictionary"><i class="fa fa-check"></i><b>17.8</b> Dictionary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="python-strings.html"><a href="python-strings.html"><i class="fa fa-check"></i><b>18</b> – Python: Strings</a>
<ul>
<li class="chapter" data-level="18.1" data-path="python-strings.html"><a href="python-strings.html#string-operators"><i class="fa fa-check"></i><b>18.1</b> String Operators</a></li>
<li class="chapter" data-level="18.2" data-path="python-strings.html"><a href="python-strings.html#other-string-modifiers"><i class="fa fa-check"></i><b>18.2</b> Other String Modifiers</a></li>
<li class="chapter" data-level="18.3" data-path="python-strings.html"><a href="python-strings.html#format-characters-in-strings"><i class="fa fa-check"></i><b>18.3</b> Format-Characters in Strings</a></li>
<li class="chapter" data-level="18.4" data-path="python-strings.html"><a href="python-strings.html#comparisons-with-strings"><i class="fa fa-check"></i><b>18.4</b> Comparisons with Strings</a></li>
<li class="chapter" data-level="18.5" data-path="python-strings.html"><a href="python-strings.html#padding-strings"><i class="fa fa-check"></i><b>18.5</b> Padding strings</a></li>
<li class="chapter" data-level="18.6" data-path="python-strings.html"><a href="python-strings.html#search-for-words-in-string-and-output-location"><i class="fa fa-check"></i><b>18.6</b> Search for words in string and output location</a></li>
<li class="chapter" data-level="18.7" data-path="python-strings.html"><a href="python-strings.html#seek-and-replace-in-strings"><i class="fa fa-check"></i><b>18.7</b> Seek and replace in strings</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="python-operators.html"><a href="python-operators.html"><i class="fa fa-check"></i><b>19</b> – Python: Operators</a>
<ul>
<li class="chapter" data-level="19.1" data-path="python-operators.html"><a href="python-operators.html#many-types-of-operators"><i class="fa fa-check"></i><b>19.1</b> Many types of operators</a></li>
<li class="chapter" data-level="19.2" data-path="python-operators.html"><a href="python-operators.html#arithmetic-operators"><i class="fa fa-check"></i><b>19.2</b> Arithmetic operators</a></li>
<li class="chapter" data-level="19.3" data-path="python-operators.html"><a href="python-operators.html#comparison-operators"><i class="fa fa-check"></i><b>19.3</b> Comparison operators</a></li>
<li class="chapter" data-level="19.4" data-path="python-operators.html"><a href="python-operators.html#logical-operators"><i class="fa fa-check"></i><b>19.4</b> Logical operators</a></li>
<li class="chapter" data-level="19.5" data-path="python-operators.html"><a href="python-operators.html#bitwise-operators"><i class="fa fa-check"></i><b>19.5</b> Bitwise operators</a></li>
<li class="chapter" data-level="19.6" data-path="python-operators.html"><a href="python-operators.html#identity-operators"><i class="fa fa-check"></i><b>19.6</b> Identity Operators</a></li>
<li class="chapter" data-level="19.7" data-path="python-operators.html"><a href="python-operators.html#membership-operators"><i class="fa fa-check"></i><b>19.7</b> Membership operators</a></li>
<li class="chapter" data-level="19.8" data-path="python-operators.html"><a href="python-operators.html#what-is-the-order-of-operations-with-operators"><i class="fa fa-check"></i><b>19.8</b> What is the order of operations with operators?</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="biopython.html"><a href="biopython.html"><i class="fa fa-check"></i><b>20</b> – Biopython</a>
<ul>
<li class="chapter" data-level="20.1" data-path="biopython.html"><a href="biopython.html#introduction-to-biopython"><i class="fa fa-check"></i><b>20.1</b> Introduction to Biopython</a></li>
<li class="chapter" data-level="20.2" data-path="biopython.html"><a href="biopython.html#biopython-installation"><i class="fa fa-check"></i><b>20.2</b> Biopython Installation</a></li>
<li class="chapter" data-level="20.3" data-path="biopython.html"><a href="biopython.html#getting-started"><i class="fa fa-check"></i><b>20.3</b> Getting Started</a></li>
<li class="chapter" data-level="20.4" data-path="biopython.html"><a href="biopython.html#parsing-my-fasta"><i class="fa fa-check"></i><b>20.4</b> Parsing my FASTA</a></li>
<li class="chapter" data-level="20.5" data-path="biopython.html"><a href="biopython.html#parsing-my-genbank-or-genpept"><i class="fa fa-check"></i><b>20.5</b> Parsing my GenBank or GenPept</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="biopython-sequences.html"><a href="biopython-sequences.html"><i class="fa fa-check"></i><b>21</b> – Biopython: Sequences</a>
<ul>
<li class="chapter" data-level="21.1" data-path="biopython-sequences.html"><a href="biopython-sequences.html#sequence-objects"><i class="fa fa-check"></i><b>21.1</b> Sequence Objects</a></li>
<li class="chapter" data-level="21.2" data-path="biopython-sequences.html"><a href="biopython-sequences.html#concatenation-with-sequences"><i class="fa fa-check"></i><b>21.2</b> Concatenation with Sequences</a></li>
<li class="chapter" data-level="21.3" data-path="biopython-sequences.html"><a href="biopython-sequences.html#transcription-dna-to-mrna"><i class="fa fa-check"></i><b>21.3</b> Transcription: DNA to mRNA</a></li>
<li class="chapter" data-level="21.4" data-path="biopython-sequences.html"><a href="biopython-sequences.html#translation-rna-to-prot"><i class="fa fa-check"></i><b>21.4</b> Translation: RNA to PROT</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html"><i class="fa fa-check"></i><b>22</b> – Biopython: Mulitple Sequence Alignments</a>
<ul>
<li class="chapter" data-level="22.1" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwise2"><i class="fa fa-check"></i><b>22.1</b> pairwise2</a></li>
<li class="chapter" data-level="22.2" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwisealigner"><i class="fa fa-check"></i><b>22.2</b> PairwiseAligner</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html"><i class="fa fa-check"></i><b>23</b> – Biopython: GenomeDiagram</a>
<ul>
<li class="chapter" data-level="23.1" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#linearcircular-gene-diagrams"><i class="fa fa-check"></i><b>23.1</b> Linear/Circular Gene Diagrams</a></li>
<li class="chapter" data-level="23.2" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#diagram-readability"><i class="fa fa-check"></i><b>23.2</b> Diagram Readability</a></li>
<li class="chapter" data-level="23.3" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#box-shape"><i class="fa fa-check"></i><b>23.3</b> Box Shape</a></li>
<li class="chapter" data-level="23.4" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#edit-box-shape-arrows"><i class="fa fa-check"></i><b>23.4</b> Edit Box Shape: Arrows</a></li>
<li class="chapter" data-level="23.5" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#restriction-map"><i class="fa fa-check"></i><b>23.5</b> Restriction Map</a></li>
<li class="chapter" data-level="23.6" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#tracks-on-tracks"><i class="fa fa-check"></i><b>23.6</b> Tracks on Tracks</a></li>
<li class="chapter" data-level="23.7" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#cross-link-data"><i class="fa fa-check"></i><b>23.7</b> Cross-link Data</a></li>
<li class="chapter" data-level="23.8" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#clean-cross-link"><i class="fa fa-check"></i><b>23.8</b> Clean Cross-Link</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html"><i class="fa fa-check"></i><b>24</b> – Biopython: Chromosome Modeling</a>
<ul>
<li class="chapter" data-level="24.1" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html#gene-feature-modeling"><i class="fa fa-check"></i><b>24.1</b> Gene Feature Modeling</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html"><i class="fa fa-check"></i><b>25</b> – Biopython: Sequence Parsing with Plots</a>
<ul>
<li class="chapter" data-level="25.1" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#histograms-1"><i class="fa fa-check"></i><b>25.1</b> Histograms</a></li>
<li class="chapter" data-level="25.2" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#gc-content"><i class="fa fa-check"></i><b>25.2</b> GC Content</a></li>
<li class="chapter" data-level="25.3" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#nt-alignment-dot-plot"><i class="fa fa-check"></i><b>25.3</b> Nt Alignment Dot Plot</a></li>
<li class="chapter" data-level="25.4" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#bigger-dots-for-the-dot-plot"><i class="fa fa-check"></i><b>25.4</b> Bigger Dots for the Dot Plot</a></li>
<li class="chapter" data-level="25.5" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#fastq-we-did-not-forget-you"><i class="fa fa-check"></i><b>25.5</b> FASTQ we did not forget you</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html"><i class="fa fa-check"></i><b>26</b> – Python: Web Scraping w/ BeautifulSoup</a>
<ul>
<li class="chapter" data-level="26.1" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#introduction"><i class="fa fa-check"></i><b>26.1</b> Introduction</a></li>
<li class="chapter" data-level="26.2" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#installation"><i class="fa fa-check"></i><b>26.2</b> Installation</a></li>
<li class="chapter" data-level="26.3" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#getting-started-1"><i class="fa fa-check"></i><b>26.3</b> Getting started</a></li>
<li class="chapter" data-level="26.4" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-all-matching-tags"><i class="fa fa-check"></i><b>26.4</b> Mining all matching tags</a></li>
<li class="chapter" data-level="26.5" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-tags-by-class-or-id"><i class="fa fa-check"></i><b>26.5</b> Mining tags by class or id</a></li>
<li class="chapter" data-level="26.6" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#use-css-selectors-for-positional-mining"><i class="fa fa-check"></i><b>26.6</b> Use CSS Selectors for positional mining</a></li>
<li class="chapter" data-level="26.7" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#weather-page-example"><i class="fa fa-check"></i><b>26.7</b> Weather Page Example</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="python-subprocess.html"><a href="python-subprocess.html"><i class="fa fa-check"></i><b>27</b> – Python Subprocess</a>
<ul>
<li class="chapter" data-level="27.1" data-path="python-subprocess.html"><a href="python-subprocess.html#introduction-and-installation"><i class="fa fa-check"></i><b>27.1</b> Introduction and Installation</a></li>
<li class="chapter" data-level="27.2" data-path="python-subprocess.html"><a href="python-subprocess.html#creating-your-first-python-script-w-bbtools"><i class="fa fa-check"></i><b>27.2</b> Creating your first python script w/ BBTools</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html"><i class="fa fa-check"></i><b>28</b> – Control Structures &amp; Loops in R</a>
<ul>
<li class="chapter" data-level="28.1" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#controling-the-flow"><i class="fa fa-check"></i><b>28.1</b> Controling the flow</a></li>
<li class="chapter" data-level="28.2" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if"><i class="fa fa-check"></i><b>28.2</b> Conditional <code>if</code></a></li>
<li class="chapter" data-level="28.3" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if-and-else"><i class="fa fa-check"></i><b>28.3</b> Conditional <code>if</code> and <code>else</code></a></li>
<li class="chapter" data-level="28.4" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#nested-conditionals-if-else-and-if-else"><i class="fa fa-check"></i><b>28.4</b> Nested conditionals <code>if</code>, <code>else</code>, and <code>if else</code></a></li>
<li class="chapter" data-level="28.5" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-ifelse"><i class="fa fa-check"></i><b>28.5</b> Conditional <code>ifelse</code></a></li>
<li class="chapter" data-level="28.6" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#loops-that-control-iterations-and-repeations"><i class="fa fa-check"></i><b>28.6</b> Loops that control iterations and repeations</a></li>
<li class="chapter" data-level="28.7" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-for-loop"><i class="fa fa-check"></i><b>28.7</b> The <code>for</code> Loop</a></li>
<li class="chapter" data-level="28.8" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-while-loop"><i class="fa fa-check"></i><b>28.8</b> The <code>while</code> Loop</a></li>
<li class="chapter" data-level="28.9" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-repeat-loop"><i class="fa fa-check"></i><b>28.9</b> The <code>repeat</code> Loop</a></li>
<li class="chapter" data-level="28.10" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-next-and-break-statements"><i class="fa fa-check"></i><b>28.10</b> The <code>next</code> and <code>break</code> statements</a></li>
<li class="chapter" data-level="28.11" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functions"><i class="fa fa-check"></i><b>28.11</b> Other loop functions</a></li>
<li class="chapter" data-level="28.12" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#lapply-functions"><i class="fa fa-check"></i><b>28.12</b> <code>lapply</code> functions</a></li>
<li class="chapter" data-level="28.13" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsapply"><i class="fa fa-check"></i><b>28.13</b> Other loop functions:<code>apply</code></a></li>
<li class="chapter" data-level="28.14" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsmapply"><i class="fa fa-check"></i><b>28.14</b> Other loop functions:<code>mapply</code></a></li>
<li class="chapter" data-level="28.15" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionstapply"><i class="fa fa-check"></i><b>28.15</b> Other loop functions:<code>tapply</code></a></li>
<li class="chapter" data-level="28.16" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionssplit"><i class="fa fa-check"></i><b>28.16</b> Other loop functions:<code>split</code></a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html"><i class="fa fa-check"></i><b>29</b> – Bioinformatics Literature</a>
<ul>
<li class="chapter" data-level="29.1" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#peer-reviewed-bioinformatics-journals"><i class="fa fa-check"></i><b>29.1</b> Peer-reviewed bioinformatics journals</a></li>
<li class="chapter" data-level="29.2" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#access-to-journals-articles"><i class="fa fa-check"></i><b>29.2</b> Access to journals articles</a></li>
<li class="chapter" data-level="29.3" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#informal-access-to-code-preprints-and-vignettes"><i class="fa fa-check"></i><b>29.3</b> Informal access to code: Preprints and vignettes</a></li>
<li class="chapter" data-level="29.4" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#online-bioinformatics-courses"><i class="fa fa-check"></i><b>29.4</b> Online bioinformatics courses</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html"><i class="fa fa-check"></i><b>30</b> – Importing/Exporting Data into R</a>
<ul>
<li class="chapter" data-level="30.1" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#typing-directly"><i class="fa fa-check"></i><b>30.1</b> Typing directly</a></li>
<li class="chapter" data-level="30.2" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#summary-versus-raw-data"><i class="fa fa-check"></i><b>30.2</b> Summary versus raw data</a></li>
<li class="chapter" data-level="30.3" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#downloading-data-files-github"><i class="fa fa-check"></i><b>30.3</b> Downloading data files (GitHub)</a></li>
<li class="chapter" data-level="30.4" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#importing-data-from-a-file"><i class="fa fa-check"></i><b>30.4</b> Importing data from a file</a></li>
<li class="chapter" data-level="30.5" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-to-a-file"><i class="fa fa-check"></i><b>30.5</b> Saving data to a file</a></li>
<li class="chapter" data-level="30.6" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-as-rdata-.rdata-.rda"><i class="fa fa-check"></i><b>30.6</b> Saving data as Rdata (.rdata, .rda)</a></li>
<li class="chapter" data-level="30.7" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#import-excel-formated-data"><i class="fa fa-check"></i><b>30.7</b> Import Excel formated data</a></li>
<li class="chapter" data-level="30.8" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#save-data-as-excel-formated-file"><i class="fa fa-check"></i><b>30.8</b> Save data as Excel formated file</a></li>
<li class="chapter" data-level="30.9" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#fixing-problems-while-importing-datasets"><i class="fa fa-check"></i><b>30.9</b> Fixing problems while importing datasets</a></li>
<li class="chapter" data-level="30.10" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#simplifying-embellished-excel-datasets"><i class="fa fa-check"></i><b>30.10</b> Simplifying embellished excel datasets</a></li>
<li class="chapter" data-level="30.11" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#accessing-online-data"><i class="fa fa-check"></i><b>30.11</b> Accessing online data</a></li>
<li class="chapter" data-level="30.12" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#data-from-clinicaltrials.gov"><i class="fa fa-check"></i><b>30.12</b> Data from clinicaltrials.gov</a></li>
<li class="chapter" data-level="30.13" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#data-scrapping-from-a-website"><i class="fa fa-check"></i><b>30.13</b> Data scrapping from a website</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="working-with-proteins.html"><a href="working-with-proteins.html"><i class="fa fa-check"></i><b>31</b> – Working with Proteins</a>
<ul>
<li class="chapter" data-level="31.1" data-path="working-with-proteins.html"><a href="working-with-proteins.html#sequences-in-fasta-format"><i class="fa fa-check"></i><b>31.1</b> Sequences in FASTA format</a></li>
<li class="chapter" data-level="31.2" data-path="working-with-proteins.html"><a href="working-with-proteins.html#fetching-aa-sequences-from-ncbi"><i class="fa fa-check"></i><b>31.2</b> Fetching AA sequences from NCBI</a></li>
<li class="chapter" data-level="31.3" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-aa-sequences-in-r"><i class="fa fa-check"></i><b>31.3</b> Importing AA sequences in R</a></li>
<li class="chapter" data-level="31.4" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-nucleotide-sequences-in-r"><i class="fa fa-check"></i><b>31.4</b> Importing nucleotide sequences in R</a></li>
<li class="chapter" data-level="31.5" data-path="working-with-proteins.html"><a href="working-with-proteins.html#translating-a-nucleotides-to-aa"><i class="fa fa-check"></i><b>31.5</b> Translating a nucleotides to AA</a></li>
<li class="chapter" data-level="31.6" data-path="working-with-proteins.html"><a href="working-with-proteins.html#finding-orfs-in-a-nucleotide-sequence"><i class="fa fa-check"></i><b>31.6</b> Finding ORFs in a nucleotide sequence</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="miscellaneous.html"><a href="miscellaneous.html"><i class="fa fa-check"></i><b>32</b> – Miscellaneous</a>
<ul>
<li class="chapter" data-level="32.1" data-path="miscellaneous.html"><a href="miscellaneous.html#sampling-from-a-vector-of-names"><i class="fa fa-check"></i><b>32.1</b> Sampling from a vector of names</a></li>
<li class="chapter" data-level="32.2" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-an-article-pdf"><i class="fa fa-check"></i><b>32.2</b> Read a text from an article pdf</a></li>
<li class="chapter" data-level="32.3" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-website"><i class="fa fa-check"></i><b>32.3</b> Read a text from a website</a></li>
<li class="chapter" data-level="32.4" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-file"><i class="fa fa-check"></i><b>32.4</b> Read a text from a file</a></li>
<li class="chapter" data-level="32.5" data-path="miscellaneous.html"><a href="miscellaneous.html#generate-the-word-cloud"><i class="fa fa-check"></i><b>32.5</b> Generate the word cloud</a></li>
<li class="chapter" data-level="32.6" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-pdf"><i class="fa fa-check"></i><b>32.6</b> worldcloud – PDF</a></li>
<li class="chapter" data-level="32.7" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-website"><i class="fa fa-check"></i><b>32.7</b> worldcloud – website</a></li>
<li class="chapter" data-level="32.8" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-text"><i class="fa fa-check"></i><b>32.8</b> worldcloud – text</a></li>
<li class="chapter" data-level="32.9" data-path="miscellaneous.html"><a href="miscellaneous.html#miscellaneous-1"><i class="fa fa-check"></i><b>32.9</b> Miscellaneous</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics Workshop Gitbook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mining-from-a-reference-dataset" class="section level1" number="13">
<h1><span class="header-section-number">Session 13</span> – Mining from a reference dataset</h1>
<p>Sometimes you have a list of individuals with some indices and you want to retrieve values associated from a large reference database or data frame. Using these indices will help you retrieve data from these references without going to the painstaking work to copy-and-pasting such value one by one.</p>
<div id="a-worked-example-using-the-mtcars-dataset" class="section level2" number="13.1">
<h2><span class="header-section-number">13.1</span> A worked example using the <code>mtcars</code> dataset</h2>
<p>We will assume that you have four friends (Joe, Moe, Larry, Fred) and each one has a collection of cars. They want to compare the overall collection based on statistics provided by <code>mtcars</code> dataset.</p>
</div>
<div id="build-a-reference-dataset" class="section level2" number="13.2">
<h2><span class="header-section-number">13.2</span> Build a reference dataset</h2>
<p><strong>1)</strong> We will create the reference database as <strong>mtcars_data</strong> from the preloaded <code>mtcars</code> dataset. This dataset exists in the R environment. You can import a data frame (e.g., a txt or excel file) as data frame, but make sure that you include the indexing variable: <code>my_index</code> as demonstrated below.</p>
<div class="sourceCode" id="cb532"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb532-1"><a href="mining-from-a-reference-dataset.html#cb532-1" aria-hidden="true" tabindex="-1"></a><span class="do">## this is an example to create a reference index</span></span>
<span id="cb532-2"><a href="mining-from-a-reference-dataset.html#cb532-2" aria-hidden="true" tabindex="-1"></a>mtcars_data <span class="ot">&lt;-</span> mtcars</span>
<span id="cb532-3"><a href="mining-from-a-reference-dataset.html#cb532-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb532-4"><a href="mining-from-a-reference-dataset.html#cb532-4" aria-hidden="true" tabindex="-1"></a><span class="do">## we need to create a indexing variable, so will use car names</span></span>
<span id="cb532-5"><a href="mining-from-a-reference-dataset.html#cb532-5" aria-hidden="true" tabindex="-1"></a>mtcars_names_vector <span class="ot">&lt;-</span> <span class="fu">rownames</span>(mtcars_data)</span>
<span id="cb532-6"><a href="mining-from-a-reference-dataset.html#cb532-6" aria-hidden="true" tabindex="-1"></a>mtcars_names_vector</span>
<span id="cb532-7"><a href="mining-from-a-reference-dataset.html#cb532-7" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;Mazda RX4&quot;           &quot;Mazda RX4 Wag&quot;       &quot;Datsun 710&quot;          &quot;Hornet 4 Drive&quot;      &quot;Hornet Sportabout&quot;   &quot;Valiant&quot;            </span></span>
<span id="cb532-8"><a href="mining-from-a-reference-dataset.html#cb532-8" aria-hidden="true" tabindex="-1"></a><span class="co"># [7] &quot;Duster 360&quot;          &quot;Merc 240D&quot;           &quot;Merc 230&quot;            &quot;Merc 280&quot;            &quot;Merc 280C&quot;           &quot;Merc 450SE&quot;         </span></span>
<span id="cb532-9"><a href="mining-from-a-reference-dataset.html#cb532-9" aria-hidden="true" tabindex="-1"></a><span class="co">#[13] &quot;Merc 450SL&quot;          &quot;Merc 450SLC&quot;         &quot;Cadillac Fleetwood&quot;  &quot;Lincoln Continental&quot; &quot;Chrysler Imperial&quot;   &quot;Fiat 128&quot;           </span></span>
<span id="cb532-10"><a href="mining-from-a-reference-dataset.html#cb532-10" aria-hidden="true" tabindex="-1"></a><span class="co">#[19] &quot;Honda Civic&quot;         &quot;Toyota Corolla&quot;      &quot;Toyota Corona&quot;       &quot;Dodge Challenger&quot;    &quot;AMC Javelin&quot;         &quot;Camaro Z28&quot;         </span></span>
<span id="cb532-11"><a href="mining-from-a-reference-dataset.html#cb532-11" aria-hidden="true" tabindex="-1"></a><span class="co">#[25] &quot;Pontiac Firebird&quot;    &quot;Fiat X1-9&quot;           &quot;Porsche 914-2&quot;       &quot;Lotus Europa&quot;        &quot;Ford Pantera L&quot;      &quot;Ferrari Dino&quot;       </span></span>
<span id="cb532-12"><a href="mining-from-a-reference-dataset.html#cb532-12" aria-hidden="true" tabindex="-1"></a><span class="co">#[31] &quot;Maserati Bora&quot;       &quot;Volvo 142E&quot;         </span></span>
<span id="cb532-13"><a href="mining-from-a-reference-dataset.html#cb532-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb532-14"><a href="mining-from-a-reference-dataset.html#cb532-14" aria-hidden="true" tabindex="-1"></a><span class="do">## this vector has spaces (&quot; &quot;) and dashes (-), these might cause problems with some analyses and functions, we will replace them with underscores &quot;_&quot;</span></span>
<span id="cb532-15"><a href="mining-from-a-reference-dataset.html#cb532-15" aria-hidden="true" tabindex="-1"></a>mtcars_names_vector <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, mtcars_names_vector)</span>
<span id="cb532-16"><a href="mining-from-a-reference-dataset.html#cb532-16" aria-hidden="true" tabindex="-1"></a>mtcars_names_vector <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;-&quot;</span>, <span class="st">&quot;_&quot;</span>, mtcars_names_vector)</span>
<span id="cb532-17"><a href="mining-from-a-reference-dataset.html#cb532-17" aria-hidden="true" tabindex="-1"></a>mtcars_names_vector</span>
<span id="cb532-18"><a href="mining-from-a-reference-dataset.html#cb532-18" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;Mazda_RX4&quot;           &quot;Mazda_RX4_Wag&quot;       &quot;Datsun_710&quot;          &quot;Hornet_4_Drive&quot;      &quot;Hornet_Sportabout&quot;   &quot;Valiant&quot;            </span></span>
<span id="cb532-19"><a href="mining-from-a-reference-dataset.html#cb532-19" aria-hidden="true" tabindex="-1"></a><span class="co"># [7] &quot;Duster_360&quot;          &quot;Merc_240D&quot;           &quot;Merc_230&quot;            &quot;Merc_280&quot;            &quot;Merc_280C&quot;           &quot;Merc_450SE&quot;         </span></span>
<span id="cb532-20"><a href="mining-from-a-reference-dataset.html#cb532-20" aria-hidden="true" tabindex="-1"></a><span class="co">#[13] &quot;Merc_450SL&quot;          &quot;Merc_450SLC&quot;         &quot;Cadillac_Fleetwood&quot;  &quot;Lincoln_Continental&quot; &quot;Chrysler_Imperial&quot;   &quot;Fiat_128&quot;           </span></span>
<span id="cb532-21"><a href="mining-from-a-reference-dataset.html#cb532-21" aria-hidden="true" tabindex="-1"></a><span class="co">#[19] &quot;Honda_Civic&quot;         &quot;Toyota_Corolla&quot;      &quot;Toyota_Corona&quot;       &quot;Dodge_Challenger&quot;    &quot;AMC_Javelin&quot;         &quot;Camaro_Z28&quot;         </span></span>
<span id="cb532-22"><a href="mining-from-a-reference-dataset.html#cb532-22" aria-hidden="true" tabindex="-1"></a><span class="co">#[25] &quot;Pontiac_Firebird&quot;    &quot;Fiat_X1_9&quot;           &quot;Porsche_914_2&quot;       &quot;Lotus_Europa&quot;        &quot;Ford_Pantera_L&quot;      &quot;Ferrari_Dino&quot;       </span></span>
<span id="cb532-23"><a href="mining-from-a-reference-dataset.html#cb532-23" aria-hidden="true" tabindex="-1"></a><span class="co">#[31] &quot;Maserati_Bora&quot;       &quot;Volvo_142E&quot;         </span></span>
<span id="cb532-24"><a href="mining-from-a-reference-dataset.html#cb532-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb532-25"><a href="mining-from-a-reference-dataset.html#cb532-25" aria-hidden="true" tabindex="-1"></a><span class="do">## we can now add this vector as our indexing variable</span></span>
<span id="cb532-26"><a href="mining-from-a-reference-dataset.html#cb532-26" aria-hidden="true" tabindex="-1"></a>mtcars_data<span class="sc">$</span>my_index <span class="ot">&lt;-</span> mtcars_names_vector</span>
<span id="cb532-27"><a href="mining-from-a-reference-dataset.html#cb532-27" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mtcars_data)</span>
<span id="cb532-28"><a href="mining-from-a-reference-dataset.html#cb532-28" aria-hidden="true" tabindex="-1"></a><span class="co">#                   mpg cyl disp  hp drat    wt  qsec vs am gear carb          my_index</span></span>
<span id="cb532-29"><a href="mining-from-a-reference-dataset.html#cb532-29" aria-hidden="true" tabindex="-1"></a><span class="co">#Mazda RX4         21.0   6  160 110 3.90 2.620 16.46  0  1    4    4         Mazda_RX4</span></span>
<span id="cb532-30"><a href="mining-from-a-reference-dataset.html#cb532-30" aria-hidden="true" tabindex="-1"></a><span class="co">#Mazda RX4 Wag     21.0   6  160 110 3.90 2.875 17.02  0  1    4    4     Mazda_RX4_Wag</span></span>
<span id="cb532-31"><a href="mining-from-a-reference-dataset.html#cb532-31" aria-hidden="true" tabindex="-1"></a><span class="co">#Datsun 710        22.8   4  108  93 3.85 2.320 18.61  1  1    4    1        Datsun_710</span></span>
<span id="cb532-32"><a href="mining-from-a-reference-dataset.html#cb532-32" aria-hidden="true" tabindex="-1"></a><span class="co">#Hornet 4 Drive    21.4   6  258 110 3.08 3.215 19.44  1  0    3    1    Hornet_4_Drive</span></span>
<span id="cb532-33"><a href="mining-from-a-reference-dataset.html#cb532-33" aria-hidden="true" tabindex="-1"></a><span class="co">#Hornet Sportabout 18.7   8  360 175 3.15 3.440 17.02  0  0    3    2 Hornet_Sportabout</span></span>
<span id="cb532-34"><a href="mining-from-a-reference-dataset.html#cb532-34" aria-hidden="true" tabindex="-1"></a><span class="co">#Valiant           18.1   6  225 105 2.76 3.460 20.22  1  0    3    1           Valiant</span></span></code></pre></div>
</div>
<div id="build-a-sampling-dataset" class="section level2" number="13.3">
<h2><span class="header-section-number">13.3</span> Build a sampling dataset</h2>
<p><strong>2)</strong> Now, we will create a data frame to be used to retrive the information for the <strong>friends_data</strong>. Our indexing varaible will be <code>my_index</code>, so the information on <strong>mtcars_data</strong> can be combined, merged or subset to match the <strong>friends_data</strong>. A similar procedure can be performed by importing a data frame (e.g., a txt or excel file) with the same data frame, make sure that <code>my_index</code> variable is present.</p>
<p>This is an illustration of how to make a data frame with data that can be indexed.</p>
<div class="sourceCode" id="cb533"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb533-1"><a href="mining-from-a-reference-dataset.html#cb533-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we will start with four random vectors of the my_index variable to serve as what cars these friends have</span></span>
<span id="cb533-2"><a href="mining-from-a-reference-dataset.html#cb533-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-3"><a href="mining-from-a-reference-dataset.html#cb533-3" aria-hidden="true" tabindex="-1"></a>Joe_10_cars_vector <span class="ot">&lt;-</span> <span class="fu">sample</span>(mtcars_data<span class="sc">$</span>my_index, <span class="dv">10</span>, <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb533-4"><a href="mining-from-a-reference-dataset.html#cb533-4" aria-hidden="true" tabindex="-1"></a>Moe_15_cars_vector <span class="ot">&lt;-</span> <span class="fu">sample</span>(mtcars_data<span class="sc">$</span>my_index, <span class="dv">15</span>, <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb533-5"><a href="mining-from-a-reference-dataset.html#cb533-5" aria-hidden="true" tabindex="-1"></a>Larry_12_cars_vector <span class="ot">&lt;-</span> <span class="fu">sample</span>(mtcars_data<span class="sc">$</span>my_index, <span class="dv">12</span>, <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb533-6"><a href="mining-from-a-reference-dataset.html#cb533-6" aria-hidden="true" tabindex="-1"></a>Fred_20_cars_vector <span class="ot">&lt;-</span> <span class="fu">sample</span>(mtcars_data<span class="sc">$</span>my_index, <span class="dv">20</span>, <span class="at">replace =</span> <span class="cn">TRUE</span>)</span>
<span id="cb533-7"><a href="mining-from-a-reference-dataset.html#cb533-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-8"><a href="mining-from-a-reference-dataset.html#cb533-8" aria-hidden="true" tabindex="-1"></a><span class="do">## </span><span class="al">NOTICE</span><span class="do">: that this are random samples and each of you will get a different set.</span></span>
<span id="cb533-9"><a href="mining-from-a-reference-dataset.html#cb533-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-10"><a href="mining-from-a-reference-dataset.html#cb533-10" aria-hidden="true" tabindex="-1"></a><span class="do">## we build a two-column data frame as follows by making a list of data frames and collect each one</span></span>
<span id="cb533-11"><a href="mining-from-a-reference-dataset.html#cb533-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-12"><a href="mining-from-a-reference-dataset.html#cb533-12" aria-hidden="true" tabindex="-1"></a>friends_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb533-13"><a href="mining-from-a-reference-dataset.html#cb533-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-14"><a href="mining-from-a-reference-dataset.html#cb533-14" aria-hidden="true" tabindex="-1"></a><span class="do">## Joe</span></span>
<span id="cb533-15"><a href="mining-from-a-reference-dataset.html#cb533-15" aria-hidden="true" tabindex="-1"></a>Joe_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">my_index =</span> Joe_10_cars_vector, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb533-16"><a href="mining-from-a-reference-dataset.html#cb533-16" aria-hidden="true" tabindex="-1"></a>Joe_df<span class="sc">$</span>friend <span class="ot">&lt;-</span> <span class="st">&quot;Joe&quot;</span></span>
<span id="cb533-17"><a href="mining-from-a-reference-dataset.html#cb533-17" aria-hidden="true" tabindex="-1"></a>friends_list[[<span class="dv">1</span>]] <span class="ot">&lt;-</span> Joe_df</span>
<span id="cb533-18"><a href="mining-from-a-reference-dataset.html#cb533-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-19"><a href="mining-from-a-reference-dataset.html#cb533-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Moe</span></span>
<span id="cb533-20"><a href="mining-from-a-reference-dataset.html#cb533-20" aria-hidden="true" tabindex="-1"></a>Moe_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">my_index =</span> Moe_15_cars_vector, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb533-21"><a href="mining-from-a-reference-dataset.html#cb533-21" aria-hidden="true" tabindex="-1"></a>Moe_df<span class="sc">$</span>friend <span class="ot">&lt;-</span> <span class="st">&quot;Moe&quot;</span></span>
<span id="cb533-22"><a href="mining-from-a-reference-dataset.html#cb533-22" aria-hidden="true" tabindex="-1"></a>friends_list[[<span class="dv">2</span>]] <span class="ot">&lt;-</span> Moe_df</span>
<span id="cb533-23"><a href="mining-from-a-reference-dataset.html#cb533-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-24"><a href="mining-from-a-reference-dataset.html#cb533-24" aria-hidden="true" tabindex="-1"></a><span class="do">## Larry</span></span>
<span id="cb533-25"><a href="mining-from-a-reference-dataset.html#cb533-25" aria-hidden="true" tabindex="-1"></a>Larry_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">my_index =</span> Larry_12_cars_vector, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb533-26"><a href="mining-from-a-reference-dataset.html#cb533-26" aria-hidden="true" tabindex="-1"></a>Larry_df<span class="sc">$</span>friend <span class="ot">&lt;-</span> <span class="st">&quot;Larry&quot;</span></span>
<span id="cb533-27"><a href="mining-from-a-reference-dataset.html#cb533-27" aria-hidden="true" tabindex="-1"></a>friends_list[[<span class="dv">3</span>]] <span class="ot">&lt;-</span> Larry_df</span>
<span id="cb533-28"><a href="mining-from-a-reference-dataset.html#cb533-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-29"><a href="mining-from-a-reference-dataset.html#cb533-29" aria-hidden="true" tabindex="-1"></a><span class="do">## Fred</span></span>
<span id="cb533-30"><a href="mining-from-a-reference-dataset.html#cb533-30" aria-hidden="true" tabindex="-1"></a>Fred_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">my_index =</span> Fred_20_cars_vector, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb533-31"><a href="mining-from-a-reference-dataset.html#cb533-31" aria-hidden="true" tabindex="-1"></a>Fred_df<span class="sc">$</span>friend <span class="ot">&lt;-</span> <span class="st">&quot;Fred&quot;</span></span>
<span id="cb533-32"><a href="mining-from-a-reference-dataset.html#cb533-32" aria-hidden="true" tabindex="-1"></a>friends_list[[<span class="dv">4</span>]] <span class="ot">&lt;-</span> Fred_df</span>
<span id="cb533-33"><a href="mining-from-a-reference-dataset.html#cb533-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-34"><a href="mining-from-a-reference-dataset.html#cb533-34" aria-hidden="true" tabindex="-1"></a><span class="do">## we rbind these data frames into one</span></span>
<span id="cb533-35"><a href="mining-from-a-reference-dataset.html#cb533-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb533-36"><a href="mining-from-a-reference-dataset.html#cb533-36" aria-hidden="true" tabindex="-1"></a>friends_data <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind,friends_list)</span>
<span id="cb533-37"><a href="mining-from-a-reference-dataset.html#cb533-37" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(friends_data)</span>
<span id="cb533-38"><a href="mining-from-a-reference-dataset.html#cb533-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;: 57 obs. of  2 variables:</span></span>
<span id="cb533-39"><a href="mining-from-a-reference-dataset.html#cb533-39" aria-hidden="true" tabindex="-1"></a><span class="co"># $ my_index: chr  &quot;Porsche_914_2&quot; &quot;Dodge_Challenger&quot; &quot;Merc_450SE&quot; &quot;Fiat_X1_9&quot; ...</span></span>
<span id="cb533-40"><a href="mining-from-a-reference-dataset.html#cb533-40" aria-hidden="true" tabindex="-1"></a><span class="co"># $ friend  : chr  &quot;Joe&quot; &quot;Joe&quot; &quot;Joe&quot; &quot;Joe&quot; ...</span></span>
<span id="cb533-41"><a href="mining-from-a-reference-dataset.html#cb533-41" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(friends_data)</span>
<span id="cb533-42"><a href="mining-from-a-reference-dataset.html#cb533-42" aria-hidden="true" tabindex="-1"></a><span class="co">#          my_index friend</span></span>
<span id="cb533-43"><a href="mining-from-a-reference-dataset.html#cb533-43" aria-hidden="true" tabindex="-1"></a><span class="co">#1    Porsche_914_2    Joe</span></span>
<span id="cb533-44"><a href="mining-from-a-reference-dataset.html#cb533-44" aria-hidden="true" tabindex="-1"></a><span class="co">#2 Dodge_Challenger    Joe</span></span>
<span id="cb533-45"><a href="mining-from-a-reference-dataset.html#cb533-45" aria-hidden="true" tabindex="-1"></a><span class="co">#3       Merc_450SE    Joe</span></span>
<span id="cb533-46"><a href="mining-from-a-reference-dataset.html#cb533-46" aria-hidden="true" tabindex="-1"></a><span class="co">#4        Fiat_X1_9    Joe</span></span>
<span id="cb533-47"><a href="mining-from-a-reference-dataset.html#cb533-47" aria-hidden="true" tabindex="-1"></a><span class="co">#5        Merc_280C    Joe</span></span>
<span id="cb533-48"><a href="mining-from-a-reference-dataset.html#cb533-48" aria-hidden="true" tabindex="-1"></a><span class="co">#6         Merc_230    Joe</span></span></code></pre></div>
</div>
<div id="extracting-from-reference-dataset" class="section level2" number="13.4">
<h2><span class="header-section-number">13.4</span> Extracting from reference dataset</h2>
<p><strong>3)</strong> We can subset the <strong>mtcars_data</strong> to match the <strong>friends_data</strong> using the function <code>merge</code> and the argument <code>by.x = "my_index"</code>. This will subset and merge from the <strong>mtcars_data</strong> the information about the cars based only on values of <code>my_index</code> in <strong>friends_data</strong>.</p>
<div class="sourceCode" id="cb534"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb534-1"><a href="mining-from-a-reference-dataset.html#cb534-1" aria-hidden="true" tabindex="-1"></a><span class="do">## with reference data frame</span></span>
<span id="cb534-2"><a href="mining-from-a-reference-dataset.html#cb534-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb534-3"><a href="mining-from-a-reference-dataset.html#cb534-3" aria-hidden="true" tabindex="-1"></a>friends_data_with_mtcars_data <span class="ot">&lt;-</span> <span class="fu">merge</span>(friends_data, mtcars_data, <span class="at">by.x =</span> <span class="st">&quot;my_index&quot;</span>)</span>
<span id="cb534-4"><a href="mining-from-a-reference-dataset.html#cb534-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(friends_data_with_mtcars_data)</span>
<span id="cb534-5"><a href="mining-from-a-reference-dataset.html#cb534-5" aria-hidden="true" tabindex="-1"></a><span class="co">#            my_index friend  mpg cyl disp  hp drat    wt  qsec vs am gear carb</span></span>
<span id="cb534-6"><a href="mining-from-a-reference-dataset.html#cb534-6" aria-hidden="true" tabindex="-1"></a><span class="co">#1        AMC_Javelin   Fred 15.2   8  304 150 3.15 3.435 17.30  0  0    3    2</span></span>
<span id="cb534-7"><a href="mining-from-a-reference-dataset.html#cb534-7" aria-hidden="true" tabindex="-1"></a><span class="co">#2        AMC_Javelin   Fred 15.2   8  304 150 3.15 3.435 17.30  0  0    3    2</span></span>
<span id="cb534-8"><a href="mining-from-a-reference-dataset.html#cb534-8" aria-hidden="true" tabindex="-1"></a><span class="co">#3 Cadillac_Fleetwood  Larry 10.4   8  472 205 2.93 5.250 17.98  0  0    3    4</span></span>
<span id="cb534-9"><a href="mining-from-a-reference-dataset.html#cb534-9" aria-hidden="true" tabindex="-1"></a><span class="co">#4 Cadillac_Fleetwood  Larry 10.4   8  472 205 2.93 5.250 17.98  0  0    3    4</span></span>
<span id="cb534-10"><a href="mining-from-a-reference-dataset.html#cb534-10" aria-hidden="true" tabindex="-1"></a><span class="co">#5  Chrysler_Imperial  Larry 14.7   8  440 230 3.23 5.345 17.42  0  0    3    4</span></span>
<span id="cb534-11"><a href="mining-from-a-reference-dataset.html#cb534-11" aria-hidden="true" tabindex="-1"></a><span class="co">#6         Datsun_710    Moe 22.8   4  108  93 3.85 2.320 18.61  1  1    4    1</span></span>
<span id="cb534-12"><a href="mining-from-a-reference-dataset.html#cb534-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb534-13"><a href="mining-from-a-reference-dataset.html#cb534-13" aria-hidden="true" tabindex="-1"></a><span class="do">## we can order by friend name</span></span>
<span id="cb534-14"><a href="mining-from-a-reference-dataset.html#cb534-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb534-15"><a href="mining-from-a-reference-dataset.html#cb534-15" aria-hidden="true" tabindex="-1"></a>friends_data_with_mtcars_data_order <span class="ot">&lt;-</span> friends_data_with_mtcars_data[<span class="fu">order</span>(friends_data_with_mtcars_data<span class="sc">$</span>friend),]</span>
<span id="cb534-16"><a href="mining-from-a-reference-dataset.html#cb534-16" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(friends_data_with_mtcars_data_order)</span>
<span id="cb534-17"><a href="mining-from-a-reference-dataset.html#cb534-17" aria-hidden="true" tabindex="-1"></a><span class="co">#       my_index friend  mpg cyl disp  hp drat    wt  qsec vs am gear carb</span></span>
<span id="cb534-18"><a href="mining-from-a-reference-dataset.html#cb534-18" aria-hidden="true" tabindex="-1"></a><span class="co">#1   AMC_Javelin   Fred 15.2   8  304 150 3.15 3.435 17.30  0  0    3    2</span></span>
<span id="cb534-19"><a href="mining-from-a-reference-dataset.html#cb534-19" aria-hidden="true" tabindex="-1"></a><span class="co">#2   AMC_Javelin   Fred 15.2   8  304 150 3.15 3.435 17.30  0  0    3    2</span></span>
<span id="cb534-20"><a href="mining-from-a-reference-dataset.html#cb534-20" aria-hidden="true" tabindex="-1"></a><span class="co">#7    Datsun_710   Fred 22.8   4  108  93 3.85 2.320 18.61  1  1    4    1</span></span>
<span id="cb534-21"><a href="mining-from-a-reference-dataset.html#cb534-21" aria-hidden="true" tabindex="-1"></a><span class="co">#8    Datsun_710   Fred 22.8   4  108  93 3.85 2.320 18.61  1  1    4    1</span></span>
<span id="cb534-22"><a href="mining-from-a-reference-dataset.html#cb534-22" aria-hidden="true" tabindex="-1"></a><span class="co">#12   Duster_360   Fred 14.3   8  360 245 3.21 3.570 15.84  0  0    3    4</span></span>
<span id="cb534-23"><a href="mining-from-a-reference-dataset.html#cb534-23" aria-hidden="true" tabindex="-1"></a><span class="co">#13 Ferrari_Dino   Fred 19.7   6  145 175 3.62 2.770 15.50  0  1    5    6</span></span></code></pre></div>
<p><strong>4)</strong> You can determine different parameters of the car sets per each friend and compere between them</p>
<div class="sourceCode" id="cb535"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb535-1"><a href="mining-from-a-reference-dataset.html#cb535-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We will split the large data frame into each friend</span></span>
<span id="cb535-2"><a href="mining-from-a-reference-dataset.html#cb535-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb535-3"><a href="mining-from-a-reference-dataset.html#cb535-3" aria-hidden="true" tabindex="-1"></a>list_friends <span class="ot">&lt;-</span> <span class="fu">split</span>(friends_data_with_mtcars_data_order, friends_data_with_mtcars_data_order<span class="sc">$</span>friend)</span>
<span id="cb535-4"><a href="mining-from-a-reference-dataset.html#cb535-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb535-5"><a href="mining-from-a-reference-dataset.html#cb535-5" aria-hidden="true" tabindex="-1"></a><span class="do">## names on the lists</span></span>
<span id="cb535-6"><a href="mining-from-a-reference-dataset.html#cb535-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb535-7"><a href="mining-from-a-reference-dataset.html#cb535-7" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(list_friends)</span>
<span id="cb535-8"><a href="mining-from-a-reference-dataset.html#cb535-8" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;Fred&quot;  &quot;Joe&quot;   &quot;Larry&quot; &quot;Moe&quot;  </span></span>
<span id="cb535-9"><a href="mining-from-a-reference-dataset.html#cb535-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb535-10"><a href="mining-from-a-reference-dataset.html#cb535-10" aria-hidden="true" tabindex="-1"></a><span class="do">## fred</span></span>
<span id="cb535-11"><a href="mining-from-a-reference-dataset.html#cb535-11" aria-hidden="true" tabindex="-1"></a>fred_df <span class="ot">&lt;-</span> list_friends[[<span class="dv">1</span>]]</span>
<span id="cb535-12"><a href="mining-from-a-reference-dataset.html#cb535-12" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(fred_df)</span>
<span id="cb535-13"><a href="mining-from-a-reference-dataset.html#cb535-13" aria-hidden="true" tabindex="-1"></a><span class="co">#       my_index friend  mpg cyl disp  hp drat    wt  qsec vs am gear carb</span></span>
<span id="cb535-14"><a href="mining-from-a-reference-dataset.html#cb535-14" aria-hidden="true" tabindex="-1"></a><span class="co">#1   AMC_Javelin   Fred 15.2   8  304 150 3.15 3.435 17.30  0  0    3    2</span></span>
<span id="cb535-15"><a href="mining-from-a-reference-dataset.html#cb535-15" aria-hidden="true" tabindex="-1"></a><span class="co">#2   AMC_Javelin   Fred 15.2   8  304 150 3.15 3.435 17.30  0  0    3    2</span></span>
<span id="cb535-16"><a href="mining-from-a-reference-dataset.html#cb535-16" aria-hidden="true" tabindex="-1"></a><span class="co">#7    Datsun_710   Fred 22.8   4  108  93 3.85 2.320 18.61  1  1    4    1</span></span>
<span id="cb535-17"><a href="mining-from-a-reference-dataset.html#cb535-17" aria-hidden="true" tabindex="-1"></a><span class="co">#8    Datsun_710   Fred 22.8   4  108  93 3.85 2.320 18.61  1  1    4    1</span></span>
<span id="cb535-18"><a href="mining-from-a-reference-dataset.html#cb535-18" aria-hidden="true" tabindex="-1"></a><span class="co">#12   Duster_360   Fred 14.3   8  360 245 3.21 3.570 15.84  0  0    3    4</span></span>
<span id="cb535-19"><a href="mining-from-a-reference-dataset.html#cb535-19" aria-hidden="true" tabindex="-1"></a><span class="co">#13 Ferrari_Dino   Fred 19.7   6  145 175 3.62 2.770 15.50  0  1    5    6</span></span>
<span id="cb535-20"><a href="mining-from-a-reference-dataset.html#cb535-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb535-21"><a href="mining-from-a-reference-dataset.html#cb535-21" aria-hidden="true" tabindex="-1"></a><span class="do">## you can do the same as: fred_df &lt;- subset(friends_data_with_mtcars_data_order, friend == &quot;Fred&quot;)</span></span>
<span id="cb535-22"><a href="mining-from-a-reference-dataset.html#cb535-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb535-23"><a href="mining-from-a-reference-dataset.html#cb535-23" aria-hidden="true" tabindex="-1"></a>joe_df <span class="ot">&lt;-</span> list_friends[[<span class="dv">2</span>]]</span>
<span id="cb535-24"><a href="mining-from-a-reference-dataset.html#cb535-24" aria-hidden="true" tabindex="-1"></a>larry_df <span class="ot">&lt;-</span> list_friends[[<span class="dv">3</span>]]</span>
<span id="cb535-25"><a href="mining-from-a-reference-dataset.html#cb535-25" aria-hidden="true" tabindex="-1"></a>moe_df <span class="ot">&lt;-</span> list_friends[[<span class="dv">4</span>]]</span></code></pre></div>
<p>We can ask which friend has more weight in their car set?</p>
<div class="sourceCode" id="cb536"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb536-1"><a href="mining-from-a-reference-dataset.html#cb536-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(fred_df<span class="sc">$</span>wt)</span>
<span id="cb536-2"><a href="mining-from-a-reference-dataset.html#cb536-2" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 58.893</span></span>
<span id="cb536-3"><a href="mining-from-a-reference-dataset.html#cb536-3" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(joe_df<span class="sc">$</span>wt)</span>
<span id="cb536-4"><a href="mining-from-a-reference-dataset.html#cb536-4" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 30.205</span></span>
<span id="cb536-5"><a href="mining-from-a-reference-dataset.html#cb536-5" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(larry_df<span class="sc">$</span>wt)</span>
<span id="cb536-6"><a href="mining-from-a-reference-dataset.html#cb536-6" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 44.445</span></span>
<span id="cb536-7"><a href="mining-from-a-reference-dataset.html#cb536-7" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(moe_df<span class="sc">$</span>wt)</span>
<span id="cb536-8"><a href="mining-from-a-reference-dataset.html#cb536-8" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 44.445</span></span></code></pre></div>
<p>In this case, Fred clearly has won. We can ask which friend has more horsepower in their car set?</p>
<div class="sourceCode" id="cb537"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb537-1"><a href="mining-from-a-reference-dataset.html#cb537-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(fred_df<span class="sc">$</span>hp)</span>
<span id="cb537-2"><a href="mining-from-a-reference-dataset.html#cb537-2" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 2756</span></span>
<span id="cb537-3"><a href="mining-from-a-reference-dataset.html#cb537-3" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(joe_df<span class="sc">$</span>hp)</span>
<span id="cb537-4"><a href="mining-from-a-reference-dataset.html#cb537-4" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 1166</span></span>
<span id="cb537-5"><a href="mining-from-a-reference-dataset.html#cb537-5" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(larry_df<span class="sc">$</span>hp)</span>
<span id="cb537-6"><a href="mining-from-a-reference-dataset.html#cb537-6" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 1816</span></span>
<span id="cb537-7"><a href="mining-from-a-reference-dataset.html#cb537-7" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(moe_df<span class="sc">$</span>hp)</span>
<span id="cb537-8"><a href="mining-from-a-reference-dataset.html#cb537-8" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 1816</span></span></code></pre></div>
</div>
<div id="plots-of-the-extracted-data" class="section level2" number="13.5">
<h2><span class="header-section-number">13.5</span> Plots of the extracted data</h2>
<p><strong>5)</strong> We can do some plots to exemplify some of these observations.</p>
<p>We can plot boxplots for hp.</p>
<div class="sourceCode" id="cb538"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb538-1"><a href="mining-from-a-reference-dataset.html#cb538-1" aria-hidden="true" tabindex="-1"></a><span class="do">## check ig ggplot2 is loaded</span></span>
<span id="cb538-2"><a href="mining-from-a-reference-dataset.html#cb538-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggplot2)</span>
<span id="cb538-3"><a href="mining-from-a-reference-dataset.html#cb538-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb538-4"><a href="mining-from-a-reference-dataset.html#cb538-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Boxplots with standard color</span></span>
<span id="cb538-5"><a href="mining-from-a-reference-dataset.html#cb538-5" aria-hidden="true" tabindex="-1"></a>friends_hp_boxplot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(friends_data_with_mtcars_data_order, <span class="fu">aes</span>(<span class="at">x=</span>friend, <span class="at">y=</span>hp, <span class="at">fill=</span>friend)) <span class="sc">+</span> </span>
<span id="cb538-6"><a href="mining-from-a-reference-dataset.html#cb538-6" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">geom_boxplot</span>(<span class="at">width=</span><span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb538-7"><a href="mining-from-a-reference-dataset.html#cb538-7" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;Friends car collection -- HP boxplots&quot;</span>, <span class="at">x=</span><span class="st">&quot;Friends&quot;</span>, <span class="at">y=</span><span class="st">&quot;hp&quot;</span>) <span class="sc">+</span></span>
<span id="cb538-8"><a href="mining-from-a-reference-dataset.html#cb538-8" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">guides</span>(<span class="at">fill=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb538-9"><a href="mining-from-a-reference-dataset.html#cb538-9" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme_minimal</span>()</span>
<span id="cb538-10"><a href="mining-from-a-reference-dataset.html#cb538-10" aria-hidden="true" tabindex="-1"></a>friends_hp_boxplot</span></code></pre></div>
<p><img src="figures/session_13_figs/1_hp_box.png" width="90%" /></p>
<p>We can plot density for wt.</p>
<div class="sourceCode" id="cb539"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb539-1"><a href="mining-from-a-reference-dataset.html#cb539-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Overlaid densities</span></span>
<span id="cb539-2"><a href="mining-from-a-reference-dataset.html#cb539-2" aria-hidden="true" tabindex="-1"></a>friends_wt_plot_den <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(friends_data_with_mtcars_data_order, <span class="fu">aes</span>(<span class="at">x=</span>wt, <span class="at">fill=</span>friend)) <span class="sc">+</span></span>
<span id="cb539-3"><a href="mining-from-a-reference-dataset.html#cb539-3" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">geom_density</span>(<span class="at">alpha=</span>.<span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb539-4"><a href="mining-from-a-reference-dataset.html#cb539-4" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">ggtitle</span>(<span class="st">&quot;Friends car collection -- WT density plots&quot;</span>) <span class="sc">+</span></span>
<span id="cb539-5"><a href="mining-from-a-reference-dataset.html#cb539-5" aria-hidden="true" tabindex="-1"></a>                     <span class="fu">theme_minimal</span>()</span>
<span id="cb539-6"><a href="mining-from-a-reference-dataset.html#cb539-6" aria-hidden="true" tabindex="-1"></a>friends_wt_plot_den</span></code></pre></div>
<p><img src="figures/session_13_figs/2_wt_density.png" width="90%" /></p>
<p>We can have a scatterplot for mpg by hp.</p>
<div class="sourceCode" id="cb540"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb540-1"><a href="mining-from-a-reference-dataset.html#cb540-1" aria-hidden="true" tabindex="-1"></a><span class="do">## scatterplot</span></span>
<span id="cb540-2"><a href="mining-from-a-reference-dataset.html#cb540-2" aria-hidden="true" tabindex="-1"></a>friends_scatterplot_mpg_hp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(friends_data_with_mtcars_data_order, <span class="fu">aes</span>(<span class="at">x=</span>mpg,  <span class="at">y =</span> hp, <span class="at">color=</span>friend)) <span class="sc">+</span></span>
<span id="cb540-3"><a href="mining-from-a-reference-dataset.html#cb540-3" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb540-4"><a href="mining-from-a-reference-dataset.html#cb540-4" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">ggtitle</span>(<span class="st">&quot;Friends car collection -- mpg versus hp&quot;</span>) <span class="sc">+</span></span>
<span id="cb540-5"><a href="mining-from-a-reference-dataset.html#cb540-5" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">theme_minimal</span>()</span>
<span id="cb540-6"><a href="mining-from-a-reference-dataset.html#cb540-6" aria-hidden="true" tabindex="-1"></a>friends_scatterplot_mpg_hp</span></code></pre></div>
<p><img src="figures/session_13_figs/3_mpg_hp_scatterplot.png" width="90%" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="estimating-phylogenetic-trees.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="introduction-to-python.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/13-Mining_datasets_v2.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
