<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Session 8 – Multidimensional Data in R | Bioinformatics Workshop Gitbook</title>
  <meta name="description" content="none" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Session 8 – Multidimensional Data in R | Bioinformatics Workshop Gitbook" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="none" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Session 8 – Multidimensional Data in R | Bioinformatics Workshop Gitbook" />
  
  <meta name="twitter:description" content="none" />
  

<meta name="author" content="Juan C. Santos &amp; Randy Ortiz" />


<meta name="date" content="2024-01-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="regressions-and-anova-in-r.html"/>
<link rel="next" href="graphics-and-colors.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics Workshop Gitbook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html"><i class="fa fa-check"></i><b>1</b> – Getting Started with R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#why-do-we-use-r"><i class="fa fa-check"></i><b>1.1</b> Why do we use R?</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#get-a-good-text-editor"><i class="fa fa-check"></i><b>1.2</b> Get a good text editor</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-r"><i class="fa fa-check"></i><b>1.4</b> Installing R</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#annotating-code"><i class="fa fa-check"></i><b>1.5</b> Annotating code</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#defining-a-working-directory-in-r"><i class="fa fa-check"></i><b>1.6</b> Defining a working directory in R</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html"><i class="fa fa-check"></i><b>2</b> – Using R, Installing Packages and Importing/Exporting Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#your-first-session"><i class="fa fa-check"></i><b>2.1</b> Your first session</a></li>
<li class="chapter" data-level="2.2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#doing-some-base-functions-and-getting-help"><i class="fa fa-check"></i><b>2.2</b> Doing some base functions and getting help</a></li>
<li class="chapter" data-level="2.3" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#functions-between-two-or-more-vectors"><i class="fa fa-check"></i><b>2.3</b> Functions between two or more vectors</a></li>
<li class="chapter" data-level="2.4" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#about-r-packages"><i class="fa fa-check"></i><b>2.4</b> About R-packages</a></li>
<li class="chapter" data-level="2.5" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#cran-task-views-expert-opinion-on-useful-packages"><i class="fa fa-check"></i><b>2.5</b> CRAN Task Views – Expert opinion on useful packages</a></li>
<li class="chapter" data-level="2.6" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#installing-packages"><i class="fa fa-check"></i><b>2.6</b> Installing packages</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html"><i class="fa fa-check"></i><b>3</b> – Basic Data Structures in R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#scalars-and-vectors"><i class="fa fa-check"></i><b>3.1</b> Scalars and vectors</a></li>
<li class="chapter" data-level="3.2" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#arrays"><i class="fa fa-check"></i><b>3.3</b> Arrays</a></li>
<li class="chapter" data-level="3.4" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a></li>
<li class="chapter" data-level="3.5" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tables"><i class="fa fa-check"></i><b>3.5</b> Tables</a></li>
<li class="chapter" data-level="3.6" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#lists"><i class="fa fa-check"></i><b>3.6</b> Lists</a></li>
<li class="chapter" data-level="3.7" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-tables"><i class="fa fa-check"></i><b>3.7</b> Data Tables</a></li>
<li class="chapter" data-level="3.8" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tibbles"><i class="fa fa-check"></i><b>3.8</b> Tibbles</a></li>
<li class="chapter" data-level="3.9" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#dnastringsets-rnastringsets-and-aastringsets"><i class="fa fa-check"></i><b>3.9</b> DNAStringSets, RNAStringSets and AAStringSets</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html"><i class="fa fa-check"></i><b>4</b> – Text Editing and Data Transformations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#working-with-character-strings"><i class="fa fa-check"></i><b>4.1</b> Working with character strings</a></li>
<li class="chapter" data-level="4.2" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#regular-expressions-regex"><i class="fa fa-check"></i><b>4.2</b> Regular expressions (Regex)</a></li>
<li class="chapter" data-level="4.3" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#comparing-ordering-and-simplifying-vectors"><i class="fa fa-check"></i><b>4.3</b> Comparing, ordering and simplifying vectors</a></li>
<li class="chapter" data-level="4.4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#useful-functions-of-stringr"><i class="fa fa-check"></i><b>4.4</b> Useful functions of <code>stringr</code></a></li>
<li class="chapter" data-level="4.5" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#data-transformations"><i class="fa fa-check"></i><b>4.5</b> Data transformations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html"><i class="fa fa-check"></i><b>5</b> – Getting Biological Data from Public Repositories</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#ncbi-databases-genbank-and-others"><i class="fa fa-check"></i><b>5.1</b> NCBI databases: GenBank and others</a></li>
<li class="chapter" data-level="5.2" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#accessing-ncbi-within-r-rentrez"><i class="fa fa-check"></i><b>5.2</b> Accessing NCBI within R: <code>rentrez</code></a></li>
<li class="chapter" data-level="5.3" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#retrieving-pubmed-references"><i class="fa fa-check"></i><b>5.3</b> Retrieving PubMed references</a></li>
<li class="chapter" data-level="5.4" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#comparing-sequences-blast"><i class="fa fa-check"></i><b>5.4</b> Comparing sequences: BLAST</a></li>
<li class="chapter" data-level="5.5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-uniprot"><i class="fa fa-check"></i><b>5.5</b> Other databases: UniProt</a></li>
<li class="chapter" data-level="5.6" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-ensembl"><i class="fa fa-check"></i><b>5.6</b> Other databases: Ensembl</a></li>
<li class="chapter" data-level="5.7" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-taxonomy-phylogenetic-trees-and-species-distributions"><i class="fa fa-check"></i><b>5.7</b> Other databases: Taxonomy, phylogenetic trees and species distributions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html"><i class="fa fa-check"></i><b>6</b> – Basic Statistics in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#exploring-your-data"><i class="fa fa-check"></i><b>6.1</b> Exploring your data</a></li>
<li class="chapter" data-level="6.2" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#sub-setting-your-dataset"><i class="fa fa-check"></i><b>6.2</b> Sub-setting your dataset</a></li>
<li class="chapter" data-level="6.3" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#transposing-rotating-your-dataset"><i class="fa fa-check"></i><b>6.3</b> Transposing (rotating) your dataset</a></li>
<li class="chapter" data-level="6.4" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#dealing-with-missing-na-values"><i class="fa fa-check"></i><b>6.4</b> Dealing with missing (NA) values</a></li>
<li class="chapter" data-level="6.5" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#about-p-values"><i class="fa fa-check"></i><b>6.5</b> About p-values</a></li>
<li class="chapter" data-level="6.6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#how-to-read-p-values"><i class="fa fa-check"></i><b>6.6</b> How to read p-values</a></li>
<li class="chapter" data-level="6.7" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#multiple-testing-corrections"><i class="fa fa-check"></i><b>6.7</b> Multiple testing corrections</a></li>
<li class="chapter" data-level="6.8" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#normalizing-your-data"><i class="fa fa-check"></i><b>6.8</b> Normalizing your data</a></li>
<li class="chapter" data-level="6.9" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#what-test-should-i-use"><i class="fa fa-check"></i><b>6.9</b> What test should I use?</a></li>
<li class="chapter" data-level="6.10" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#type-of-variables-in-dataset"><i class="fa fa-check"></i><b>6.10</b> Type of variables in dataset</a></li>
<li class="chapter" data-level="6.11" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-means-t-test"><i class="fa fa-check"></i><b>6.11</b> Comparing means: t-test</a></li>
<li class="chapter" data-level="6.12" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#two-sample-t-test"><i class="fa fa-check"></i><b>6.12</b> Two sample t-test</a></li>
<li class="chapter" data-level="6.13" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-wilcoxon-signed-rank-test"><i class="fa fa-check"></i><b>6.13</b> Beyond t-test: Wilcoxon signed rank test</a></li>
<li class="chapter" data-level="6.14" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#tabulation-and-cross-tabulation"><i class="fa fa-check"></i><b>6.14</b> Tabulation and Cross-Tabulation</a></li>
<li class="chapter" data-level="6.15" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-binomial-test"><i class="fa fa-check"></i><b>6.15</b> Beyond t-test: Binomial test</a></li>
<li class="chapter" data-level="6.16" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-chi-square-goodness-of-fit"><i class="fa fa-check"></i><b>6.16</b> Beyond t-test: Chi-square goodness of fit</a></li>
<li class="chapter" data-level="6.17" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-mcnemars-test"><i class="fa fa-check"></i><b>6.17</b> Beyond t-test: McNemar’s test</a></li>
<li class="chapter" data-level="6.18" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-fishers-exact-test"><i class="fa fa-check"></i><b>6.18</b> Beyond t-test: Fisher’s exact test</a></li>
<li class="chapter" data-level="6.19" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#correlations"><i class="fa fa-check"></i><b>6.19</b> Correlations</a></li>
<li class="chapter" data-level="6.20" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-distributions"><i class="fa fa-check"></i><b>6.20</b> Comparing distributions</a></li>
<li class="chapter" data-level="6.21" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#bootstraps"><i class="fa fa-check"></i><b>6.21</b> Bootstraps</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html"><i class="fa fa-check"></i><b>7</b> – Regressions and ANOVA in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#linear-regressions"><i class="fa fa-check"></i><b>7.1</b> Linear regressions</a></li>
<li class="chapter" data-level="7.2" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anatomy-of-a-linear-regression"><i class="fa fa-check"></i><b>7.2</b> Anatomy of a linear regression</a></li>
<li class="chapter" data-level="7.3" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#multiple-linear-regressions"><i class="fa fa-check"></i><b>7.3</b> Multiple linear regressions</a></li>
<li class="chapter" data-level="7.4" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#diagnostic-plots-for-regressions"><i class="fa fa-check"></i><b>7.4</b> Diagnostic plots for regressions</a></li>
<li class="chapter" data-level="7.5" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-influential-measurements"><i class="fa fa-check"></i><b>7.5</b> Regression influential measurements</a></li>
<li class="chapter" data-level="7.6" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-anova-coefficients-ci"><i class="fa fa-check"></i><b>7.6</b> Regression (ANOVA, coefficients, CI)</a></li>
<li class="chapter" data-level="7.7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#selecting-predictors"><i class="fa fa-check"></i><b>7.7</b> Selecting predictors</a></li>
<li class="chapter" data-level="7.8" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-predictions"><i class="fa fa-check"></i><b>7.8</b> Regression predictions</a></li>
<li class="chapter" data-level="7.9" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions"><i class="fa fa-check"></i><b>7.9</b> Logistic regressions</a></li>
<li class="chapter" data-level="7.10" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions-requirements"><i class="fa fa-check"></i><b>7.10</b> Logistic regressions requirements</a></li>
<li class="chapter" data-level="7.11" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#non-linear-regressions"><i class="fa fa-check"></i><b>7.11</b> Non-linear regressions</a></li>
<li class="chapter" data-level="7.12" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-regressions"><i class="fa fa-check"></i><b>7.12</b> Toy datasets for regressions</a></li>
<li class="chapter" data-level="7.13" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova"><i class="fa fa-check"></i><b>7.13</b> ANOVA</a></li>
<li class="chapter" data-level="7.14" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#one-way-anova"><i class="fa fa-check"></i><b>7.14</b> One-way ANOVA</a></li>
<li class="chapter" data-level="7.15" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#two-way-anova"><i class="fa fa-check"></i><b>7.15</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="7.16" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova-post-hoc-tests"><i class="fa fa-check"></i><b>7.16</b> ANOVA Post-hoc tests</a></li>
<li class="chapter" data-level="7.17" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#beyond-anova-kruskal-wallis-test"><i class="fa fa-check"></i><b>7.17</b> Beyond ANOVA: Kruskal-Wallis Test</a></li>
<li class="chapter" data-level="7.18" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-anova"><i class="fa fa-check"></i><b>7.18</b> Toy datasets for ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html"><i class="fa fa-check"></i><b>8</b> – Multidimensional Data in R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#principal-component-analysis"><i class="fa fa-check"></i><b>8.1</b> Principal Component Analysis</a></li>
<li class="chapter" data-level="8.2" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#multidimensional-scaling"><i class="fa fa-check"></i><b>8.2</b> Multidimensional scaling</a></li>
<li class="chapter" data-level="8.3" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-dfa"><i class="fa fa-check"></i><b>8.3</b> Cluster analysis: DFA</a></li>
<li class="chapter" data-level="8.4" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-k-means-unsupervised"><i class="fa fa-check"></i><b>8.4</b> Cluster analysis: <em>k</em>-means (unsupervised)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html"><i class="fa fa-check"></i><b>9</b> – Graphics and Colors</a>
<ul>
<li class="chapter" data-level="9.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#avoid-bad-graphics"><i class="fa fa-check"></i><b>9.1</b> Avoid bad graphics</a></li>
<li class="chapter" data-level="9.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#graphics-in-r-ggplot2"><i class="fa fa-check"></i><b>9.2</b> Graphics in R – ggplot2</a></li>
<li class="chapter" data-level="9.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#preparing-your-dataset"><i class="fa fa-check"></i><b>9.3</b> Preparing your dataset</a></li>
<li class="chapter" data-level="9.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#color-palettes-online"><i class="fa fa-check"></i><b>9.4</b> Color palettes online</a></li>
<li class="chapter" data-level="9.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggplot2-default-palette"><i class="fa fa-check"></i><b>9.5</b> <code>ggplot2</code> default palette</a></li>
<li class="chapter" data-level="9.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#heatmaps-or-tiled-charts"><i class="fa fa-check"></i><b>9.6</b> Heatmaps or tiled charts</a></li>
<li class="chapter" data-level="9.7" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#r-packages-about-color"><i class="fa fa-check"></i><b>9.7</b> R-packages about color</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#viridis-package"><i class="fa fa-check"></i><b>9.7.1</b> <code>viridis</code> package</a></li>
<li class="chapter" data-level="9.7.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcolorbrewer-package"><i class="fa fa-check"></i><b>9.7.2</b> <code>RColorBrewer</code> package</a></li>
<li class="chapter" data-level="9.7.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggsci-package"><i class="fa fa-check"></i><b>9.7.3</b> <code>ggsci</code> package</a></li>
<li class="chapter" data-level="9.7.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#wesanderson-package"><i class="fa fa-check"></i><b>9.7.4</b> <code>wesanderson</code> package</a></li>
<li class="chapter" data-level="9.7.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcartocolor-package"><i class="fa fa-check"></i><b>9.7.5</b> <code>rcartocolor</code> package</a></li>
<li class="chapter" data-level="9.7.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#colorspace-package"><i class="fa fa-check"></i><b>9.7.6</b> <code>colorspace</code> package</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#making-your-own-palette"><i class="fa fa-check"></i><b>9.8</b> Making your own palette</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="basic-graphics.html"><a href="basic-graphics.html"><i class="fa fa-check"></i><b>10</b> – Basic Graphics</a>
<ul>
<li class="chapter" data-level="10.1" data-path="basic-graphics.html"><a href="basic-graphics.html#what-would-you-like-to-show"><i class="fa fa-check"></i><b>10.1</b> What would you like to show?</a></li>
<li class="chapter" data-level="10.2" data-path="basic-graphics.html"><a href="basic-graphics.html#bar-charts"><i class="fa fa-check"></i><b>10.2</b> Bar charts</a></li>
<li class="chapter" data-level="10.3" data-path="basic-graphics.html"><a href="basic-graphics.html#stacked-bar-charts"><i class="fa fa-check"></i><b>10.3</b> Stacked bar charts</a></li>
<li class="chapter" data-level="10.4" data-path="basic-graphics.html"><a href="basic-graphics.html#histograms"><i class="fa fa-check"></i><b>10.4</b> Histograms</a></li>
<li class="chapter" data-level="10.5" data-path="basic-graphics.html"><a href="basic-graphics.html#multiple-histograms"><i class="fa fa-check"></i><b>10.5</b> Multiple histograms</a></li>
<li class="chapter" data-level="10.6" data-path="basic-graphics.html"><a href="basic-graphics.html#density-plots"><i class="fa fa-check"></i><b>10.6</b> Density plots</a></li>
<li class="chapter" data-level="10.7" data-path="basic-graphics.html"><a href="basic-graphics.html#box-plots"><i class="fa fa-check"></i><b>10.7</b> Box plots</a></li>
<li class="chapter" data-level="10.8" data-path="basic-graphics.html"><a href="basic-graphics.html#violin-plots"><i class="fa fa-check"></i><b>10.8</b> Violin plots</a></li>
<li class="chapter" data-level="10.9" data-path="basic-graphics.html"><a href="basic-graphics.html#scatterplots"><i class="fa fa-check"></i><b>10.9</b> Scatterplots</a></li>
<li class="chapter" data-level="10.10" data-path="basic-graphics.html"><a href="basic-graphics.html#lines-and-scatterplots"><i class="fa fa-check"></i><b>10.10</b> Lines and Scatterplots</a></li>
<li class="chapter" data-level="10.11" data-path="basic-graphics.html"><a href="basic-graphics.html#save-your-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>10.11</b> Save your plots as PDF or PNG</a></li>
<li class="chapter" data-level="10.12" data-path="basic-graphics.html"><a href="basic-graphics.html#ggplot2-add-on-packages-treemapify"><i class="fa fa-check"></i><b>10.12</b> ggplot2 add-on packages: <code>treemapify</code></a></li>
<li class="chapter" data-level="10.13" data-path="basic-graphics.html"><a href="basic-graphics.html#other-resources"><i class="fa fa-check"></i><b>10.13</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="sequence-alignments.html"><a href="sequence-alignments.html"><i class="fa fa-check"></i><b>11</b> – Sequence Alignments</a>
<ul>
<li class="chapter" data-level="11.1" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-using-rentrez"><i class="fa fa-check"></i><b>11.1</b> Retrieving sequences using ‘rentrez’</a></li>
<li class="chapter" data-level="11.2" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-manually-from-ncbi"><i class="fa fa-check"></i><b>11.2</b> Retrieving sequences manually from NCBI</a></li>
<li class="chapter" data-level="11.3" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-accession-numbers"><i class="fa fa-check"></i><b>11.3</b> Retrieving accession numbers</a></li>
<li class="chapter" data-level="11.4" data-path="sequence-alignments.html"><a href="sequence-alignments.html#alignment-with-decipher"><i class="fa fa-check"></i><b>11.4</b> Alignment with <code>DECIPHER</code></a></li>
<li class="chapter" data-level="11.5" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualize-alignment"><i class="fa fa-check"></i><b>11.5</b> Visualize alignment</a></li>
<li class="chapter" data-level="11.6" data-path="sequence-alignments.html"><a href="sequence-alignments.html#save-alignment-to-file"><i class="fa fa-check"></i><b>11.6</b> Save alignment to file</a></li>
<li class="chapter" data-level="11.7" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-and-align-several-markers"><i class="fa fa-check"></i><b>11.7</b> Concatenate and align several markers</a></li>
<li class="chapter" data-level="11.8" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-data.table"><i class="fa fa-check"></i><b>11.8</b> Concatenate with <code>data.table</code></a></li>
<li class="chapter" data-level="11.9" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-biostrings"><i class="fa fa-check"></i><b>11.9</b> Concatenate with <code>Biostrings</code></a></li>
<li class="chapter" data-level="11.10" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualizing-alignments-in-r"><i class="fa fa-check"></i><b>11.10</b> Visualizing alignments in R</a></li>
<li class="chapter" data-level="11.11" data-path="sequence-alignments.html"><a href="sequence-alignments.html#other-sequence-alignment-software"><i class="fa fa-check"></i><b>11.11</b> Other sequence alignment software</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html"><i class="fa fa-check"></i><b>12</b> – Estimating Phylogenetic Trees</a>
<ul>
<li class="chapter" data-level="12.1" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#phylogeny-inference-with-iqtree2"><i class="fa fa-check"></i><b>12.1</b> Phylogeny inference with <code>iqtree2</code></a></li>
<li class="chapter" data-level="12.2" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-iqtree2-in-macos"><i class="fa fa-check"></i><b>12.2</b> Installing <code>iqtree2</code> in macOS</a></li>
<li class="chapter" data-level="12.3" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-and-running-iqtree2-in-a-pc"><i class="fa fa-check"></i><b>12.3</b> Installing and running <code>iqtree2</code> in a PC</a></li>
<li class="chapter" data-level="12.4" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#loading-the-iqtree_runner_r-function"><i class="fa fa-check"></i><b>12.4</b> Loading the <code>iqtree_runner_R</code> function</a></li>
<li class="chapter" data-level="12.5" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#running-iqtree2-from-r"><i class="fa fa-check"></i><b>12.5</b> Running <code>iqtree2</code> from R</a></li>
<li class="chapter" data-level="12.6" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-no-partitions"><i class="fa fa-check"></i><b>12.6</b> One marker and no partitions</a></li>
<li class="chapter" data-level="12.7" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-with-partitions"><i class="fa fa-check"></i><b>12.7</b> One marker and with partitions</a></li>
<li class="chapter" data-level="12.8" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#bootstrap-support"><i class="fa fa-check"></i><b>12.8</b> Bootstrap support</a></li>
<li class="chapter" data-level="12.9" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#many-markers-and-with-partitions"><i class="fa fa-check"></i><b>12.9</b> Many markers and with partitions</a></li>
<li class="chapter" data-level="12.10" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#other-sequence-phylogenetic-tree-estimation-software"><i class="fa fa-check"></i><b>12.10</b> Other sequence phylogenetic tree estimation software</a></li>
<li class="chapter" data-level="12.11" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#drawing-phylogenetic-trees-with-ggtree"><i class="fa fa-check"></i><b>12.11</b> Drawing phylogenetic trees with <code>ggtree</code></a></li>
<li class="chapter" data-level="12.12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#save-ggtree-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>12.12</b> Save ggtree plots as PDF or PNG</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html"><i class="fa fa-check"></i><b>13</b> – Mining from a reference dataset</a>
<ul>
<li class="chapter" data-level="13.1" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#a-worked-example-using-the-mtcars-dataset"><i class="fa fa-check"></i><b>13.1</b> A worked example using the <code>mtcars</code> dataset</a></li>
<li class="chapter" data-level="13.2" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-reference-dataset"><i class="fa fa-check"></i><b>13.2</b> Build a reference dataset</a></li>
<li class="chapter" data-level="13.3" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-sampling-dataset"><i class="fa fa-check"></i><b>13.3</b> Build a sampling dataset</a></li>
<li class="chapter" data-level="13.4" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#extracting-from-reference-dataset"><i class="fa fa-check"></i><b>13.4</b> Extracting from reference dataset</a></li>
<li class="chapter" data-level="13.5" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#plots-of-the-extracted-data"><i class="fa fa-check"></i><b>13.5</b> Plots of the extracted data</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="introduction-to-python.html"><a href="introduction-to-python.html"><i class="fa fa-check"></i><b>14</b> – Introduction to Python</a>
<ul>
<li class="chapter" data-level="14.1" data-path="introduction-to-python.html"><a href="introduction-to-python.html#welcome-to-python-why-python"><i class="fa fa-check"></i><b>14.1</b> Welcome to Python! Why Python?</a></li>
<li class="chapter" data-level="14.2" data-path="introduction-to-python.html"><a href="introduction-to-python.html#which-python-should-i-install"><i class="fa fa-check"></i><b>14.2</b> Which Python should I install?</a></li>
<li class="chapter" data-level="14.3" data-path="introduction-to-python.html"><a href="introduction-to-python.html#mac-os-x-python-installations"><i class="fa fa-check"></i><b>14.3</b> Mac OS X Python Installations</a></li>
<li class="chapter" data-level="14.4" data-path="introduction-to-python.html"><a href="introduction-to-python.html#windows-python-installations"><i class="fa fa-check"></i><b>14.4</b> Windows Python Installations</a></li>
<li class="chapter" data-level="14.5" data-path="introduction-to-python.html"><a href="introduction-to-python.html#install-an-ide-or-integrated-development-environment"><i class="fa fa-check"></i><b>14.5</b> Install an IDE, or integrated development environment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html"><i class="fa fa-check"></i><b>15</b> – Python: Keywords and Identifiers</a>
<ul>
<li class="chapter" data-level="15.1" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#keywords-in-python"><i class="fa fa-check"></i><b>15.1</b> Keywords in Python</a></li>
<li class="chapter" data-level="15.2" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#identifiers-in-python"><i class="fa fa-check"></i><b>15.2</b> Identifiers in Python</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html"><i class="fa fa-check"></i><b>16</b> – Python: Statements and Expressions</a>
<ul>
<li class="chapter" data-level="16.1" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#what-are-statements"><i class="fa fa-check"></i><b>16.1</b> What are statements?</a></li>
<li class="chapter" data-level="16.2" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#expressions-in-python"><i class="fa fa-check"></i><b>16.2</b> Expressions in Python</a></li>
<li class="chapter" data-level="16.3" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#simple-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.3</b> Simple Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.4" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#augmented-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.4</b> Augmented Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.5" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#explicit-and-implicit-line-continuation"><i class="fa fa-check"></i><b>16.5</b> Explicit and Implicit Line Continuation</a></li>
<li class="chapter" data-level="16.6" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#indentation-in-python-is-crucial-to-proper-code-function"><i class="fa fa-check"></i><b>16.6</b> Indentation in Python is Crucial to Proper Code function</a></li>
<li class="chapter" data-level="16.7" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#comments"><i class="fa fa-check"></i><b>16.7</b> Comments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="python-data-types.html"><a href="python-data-types.html"><i class="fa fa-check"></i><b>17</b> – Python: Data Types</a>
<ul>
<li class="chapter" data-level="17.1" data-path="python-data-types.html"><a href="python-data-types.html#booleans"><i class="fa fa-check"></i><b>17.1</b> Booleans</a></li>
<li class="chapter" data-level="17.2" data-path="python-data-types.html"><a href="python-data-types.html#numbers-int-float-and-complex"><i class="fa fa-check"></i><b>17.2</b> Numbers: int, float and complex</a></li>
<li class="chapter" data-level="17.3" data-path="python-data-types.html"><a href="python-data-types.html#strings"><i class="fa fa-check"></i><b>17.3</b> Strings</a></li>
<li class="chapter" data-level="17.4" data-path="python-data-types.html"><a href="python-data-types.html#bytes"><i class="fa fa-check"></i><b>17.4</b> Bytes</a></li>
<li class="chapter" data-level="17.5" data-path="python-data-types.html"><a href="python-data-types.html#lists-1"><i class="fa fa-check"></i><b>17.5</b> Lists</a></li>
<li class="chapter" data-level="17.6" data-path="python-data-types.html"><a href="python-data-types.html#tuples"><i class="fa fa-check"></i><b>17.6</b> Tuples</a></li>
<li class="chapter" data-level="17.7" data-path="python-data-types.html"><a href="python-data-types.html#sets"><i class="fa fa-check"></i><b>17.7</b> Sets</a></li>
<li class="chapter" data-level="17.8" data-path="python-data-types.html"><a href="python-data-types.html#dictionary"><i class="fa fa-check"></i><b>17.8</b> Dictionary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="python-strings.html"><a href="python-strings.html"><i class="fa fa-check"></i><b>18</b> – Python: Strings</a>
<ul>
<li class="chapter" data-level="18.1" data-path="python-strings.html"><a href="python-strings.html#string-operators"><i class="fa fa-check"></i><b>18.1</b> String Operators</a></li>
<li class="chapter" data-level="18.2" data-path="python-strings.html"><a href="python-strings.html#other-string-modifiers"><i class="fa fa-check"></i><b>18.2</b> Other String Modifiers</a></li>
<li class="chapter" data-level="18.3" data-path="python-strings.html"><a href="python-strings.html#format-characters-in-strings"><i class="fa fa-check"></i><b>18.3</b> Format-Characters in Strings</a></li>
<li class="chapter" data-level="18.4" data-path="python-strings.html"><a href="python-strings.html#comparisons-with-strings"><i class="fa fa-check"></i><b>18.4</b> Comparisons with Strings</a></li>
<li class="chapter" data-level="18.5" data-path="python-strings.html"><a href="python-strings.html#padding-strings"><i class="fa fa-check"></i><b>18.5</b> Padding strings</a></li>
<li class="chapter" data-level="18.6" data-path="python-strings.html"><a href="python-strings.html#search-for-words-in-string-and-output-location"><i class="fa fa-check"></i><b>18.6</b> Search for words in string and output location</a></li>
<li class="chapter" data-level="18.7" data-path="python-strings.html"><a href="python-strings.html#seek-and-replace-in-strings"><i class="fa fa-check"></i><b>18.7</b> Seek and replace in strings</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="python-operators.html"><a href="python-operators.html"><i class="fa fa-check"></i><b>19</b> – Python: Operators</a>
<ul>
<li class="chapter" data-level="19.1" data-path="python-operators.html"><a href="python-operators.html#many-types-of-operators"><i class="fa fa-check"></i><b>19.1</b> Many types of operators</a></li>
<li class="chapter" data-level="19.2" data-path="python-operators.html"><a href="python-operators.html#arithmetic-operators"><i class="fa fa-check"></i><b>19.2</b> Arithmetic operators</a></li>
<li class="chapter" data-level="19.3" data-path="python-operators.html"><a href="python-operators.html#comparison-operators"><i class="fa fa-check"></i><b>19.3</b> Comparison operators</a></li>
<li class="chapter" data-level="19.4" data-path="python-operators.html"><a href="python-operators.html#logical-operators"><i class="fa fa-check"></i><b>19.4</b> Logical operators</a></li>
<li class="chapter" data-level="19.5" data-path="python-operators.html"><a href="python-operators.html#bitwise-operators"><i class="fa fa-check"></i><b>19.5</b> Bitwise operators</a></li>
<li class="chapter" data-level="19.6" data-path="python-operators.html"><a href="python-operators.html#identity-operators"><i class="fa fa-check"></i><b>19.6</b> Identity Operators</a></li>
<li class="chapter" data-level="19.7" data-path="python-operators.html"><a href="python-operators.html#membership-operators"><i class="fa fa-check"></i><b>19.7</b> Membership operators</a></li>
<li class="chapter" data-level="19.8" data-path="python-operators.html"><a href="python-operators.html#what-is-the-order-of-operations-with-operators"><i class="fa fa-check"></i><b>19.8</b> What is the order of operations with operators?</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="biopython.html"><a href="biopython.html"><i class="fa fa-check"></i><b>20</b> – Biopython</a>
<ul>
<li class="chapter" data-level="20.1" data-path="biopython.html"><a href="biopython.html#introduction-to-biopython"><i class="fa fa-check"></i><b>20.1</b> Introduction to Biopython</a></li>
<li class="chapter" data-level="20.2" data-path="biopython.html"><a href="biopython.html#biopython-installation"><i class="fa fa-check"></i><b>20.2</b> Biopython Installation</a></li>
<li class="chapter" data-level="20.3" data-path="biopython.html"><a href="biopython.html#getting-started"><i class="fa fa-check"></i><b>20.3</b> Getting Started</a></li>
<li class="chapter" data-level="20.4" data-path="biopython.html"><a href="biopython.html#parsing-my-fasta"><i class="fa fa-check"></i><b>20.4</b> Parsing my FASTA</a></li>
<li class="chapter" data-level="20.5" data-path="biopython.html"><a href="biopython.html#parsing-my-genbank-or-genpept"><i class="fa fa-check"></i><b>20.5</b> Parsing my GenBank or GenPept</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="biopython-sequences.html"><a href="biopython-sequences.html"><i class="fa fa-check"></i><b>21</b> – Biopython: Sequences</a>
<ul>
<li class="chapter" data-level="21.1" data-path="biopython-sequences.html"><a href="biopython-sequences.html#sequence-objects"><i class="fa fa-check"></i><b>21.1</b> Sequence Objects</a></li>
<li class="chapter" data-level="21.2" data-path="biopython-sequences.html"><a href="biopython-sequences.html#concatenation-with-sequences"><i class="fa fa-check"></i><b>21.2</b> Concatenation with Sequences</a></li>
<li class="chapter" data-level="21.3" data-path="biopython-sequences.html"><a href="biopython-sequences.html#transcription-dna-to-mrna"><i class="fa fa-check"></i><b>21.3</b> Transcription: DNA to mRNA</a></li>
<li class="chapter" data-level="21.4" data-path="biopython-sequences.html"><a href="biopython-sequences.html#translation-rna-to-prot"><i class="fa fa-check"></i><b>21.4</b> Translation: RNA to PROT</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html"><i class="fa fa-check"></i><b>22</b> – Biopython: Mulitple Sequence Alignments</a>
<ul>
<li class="chapter" data-level="22.1" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwise2"><i class="fa fa-check"></i><b>22.1</b> pairwise2</a></li>
<li class="chapter" data-level="22.2" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwisealigner"><i class="fa fa-check"></i><b>22.2</b> PairwiseAligner</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html"><i class="fa fa-check"></i><b>23</b> – Biopython: GenomeDiagram</a>
<ul>
<li class="chapter" data-level="23.1" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#linearcircular-gene-diagrams"><i class="fa fa-check"></i><b>23.1</b> Linear/Circular Gene Diagrams</a></li>
<li class="chapter" data-level="23.2" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#diagram-readability"><i class="fa fa-check"></i><b>23.2</b> Diagram Readability</a></li>
<li class="chapter" data-level="23.3" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#box-shape"><i class="fa fa-check"></i><b>23.3</b> Box Shape</a></li>
<li class="chapter" data-level="23.4" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#edit-box-shape-arrows"><i class="fa fa-check"></i><b>23.4</b> Edit Box Shape: Arrows</a></li>
<li class="chapter" data-level="23.5" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#restriction-map"><i class="fa fa-check"></i><b>23.5</b> Restriction Map</a></li>
<li class="chapter" data-level="23.6" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#tracks-on-tracks"><i class="fa fa-check"></i><b>23.6</b> Tracks on Tracks</a></li>
<li class="chapter" data-level="23.7" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#cross-link-data"><i class="fa fa-check"></i><b>23.7</b> Cross-link Data</a></li>
<li class="chapter" data-level="23.8" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#clean-cross-link"><i class="fa fa-check"></i><b>23.8</b> Clean Cross-Link</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html"><i class="fa fa-check"></i><b>24</b> – Biopython: Chromosome Modeling</a>
<ul>
<li class="chapter" data-level="24.1" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html#gene-feature-modeling"><i class="fa fa-check"></i><b>24.1</b> Gene Feature Modeling</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html"><i class="fa fa-check"></i><b>25</b> – Biopython: Sequence Parsing with Plots</a>
<ul>
<li class="chapter" data-level="25.1" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#histograms-1"><i class="fa fa-check"></i><b>25.1</b> Histograms</a></li>
<li class="chapter" data-level="25.2" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#gc-content"><i class="fa fa-check"></i><b>25.2</b> GC Content</a></li>
<li class="chapter" data-level="25.3" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#nt-alignment-dot-plot"><i class="fa fa-check"></i><b>25.3</b> Nt Alignment Dot Plot</a></li>
<li class="chapter" data-level="25.4" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#bigger-dots-for-the-dot-plot"><i class="fa fa-check"></i><b>25.4</b> Bigger Dots for the Dot Plot</a></li>
<li class="chapter" data-level="25.5" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#fastq-we-did-not-forget-you"><i class="fa fa-check"></i><b>25.5</b> FASTQ we did not forget you</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html"><i class="fa fa-check"></i><b>26</b> – Python: Web Scraping w/ BeautifulSoup</a>
<ul>
<li class="chapter" data-level="26.1" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#introduction"><i class="fa fa-check"></i><b>26.1</b> Introduction</a></li>
<li class="chapter" data-level="26.2" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#installation"><i class="fa fa-check"></i><b>26.2</b> Installation</a></li>
<li class="chapter" data-level="26.3" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#getting-started-1"><i class="fa fa-check"></i><b>26.3</b> Getting started</a></li>
<li class="chapter" data-level="26.4" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-all-matching-tags"><i class="fa fa-check"></i><b>26.4</b> Mining all matching tags</a></li>
<li class="chapter" data-level="26.5" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-tags-by-class-or-id"><i class="fa fa-check"></i><b>26.5</b> Mining tags by class or id</a></li>
<li class="chapter" data-level="26.6" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#use-css-selectors-for-positional-mining"><i class="fa fa-check"></i><b>26.6</b> Use CSS Selectors for positional mining</a></li>
<li class="chapter" data-level="26.7" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#weather-page-example"><i class="fa fa-check"></i><b>26.7</b> Weather Page Example</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="python-subprocess.html"><a href="python-subprocess.html"><i class="fa fa-check"></i><b>27</b> – Python Subprocess</a>
<ul>
<li class="chapter" data-level="27.1" data-path="python-subprocess.html"><a href="python-subprocess.html#introduction-and-installation"><i class="fa fa-check"></i><b>27.1</b> Introduction and Installation</a></li>
<li class="chapter" data-level="27.2" data-path="python-subprocess.html"><a href="python-subprocess.html#creating-your-first-python-script-w-bbtools"><i class="fa fa-check"></i><b>27.2</b> Creating your first python script w/ BBTools</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html"><i class="fa fa-check"></i><b>28</b> – Control Structures &amp; Loops in R</a>
<ul>
<li class="chapter" data-level="28.1" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#controling-the-flow"><i class="fa fa-check"></i><b>28.1</b> Controling the flow</a></li>
<li class="chapter" data-level="28.2" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if"><i class="fa fa-check"></i><b>28.2</b> Conditional <code>if</code></a></li>
<li class="chapter" data-level="28.3" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if-and-else"><i class="fa fa-check"></i><b>28.3</b> Conditional <code>if</code> and <code>else</code></a></li>
<li class="chapter" data-level="28.4" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#nested-conditionals-if-else-and-if-else"><i class="fa fa-check"></i><b>28.4</b> Nested conditionals <code>if</code>, <code>else</code>, and <code>if else</code></a></li>
<li class="chapter" data-level="28.5" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-ifelse"><i class="fa fa-check"></i><b>28.5</b> Conditional <code>ifelse</code></a></li>
<li class="chapter" data-level="28.6" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#loops-that-control-iterations-and-repeations"><i class="fa fa-check"></i><b>28.6</b> Loops that control iterations and repeations</a></li>
<li class="chapter" data-level="28.7" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-for-loop"><i class="fa fa-check"></i><b>28.7</b> The <code>for</code> Loop</a></li>
<li class="chapter" data-level="28.8" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-while-loop"><i class="fa fa-check"></i><b>28.8</b> The <code>while</code> Loop</a></li>
<li class="chapter" data-level="28.9" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-repeat-loop"><i class="fa fa-check"></i><b>28.9</b> The <code>repeat</code> Loop</a></li>
<li class="chapter" data-level="28.10" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-next-and-break-statements"><i class="fa fa-check"></i><b>28.10</b> The <code>next</code> and <code>break</code> statements</a></li>
<li class="chapter" data-level="28.11" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functions"><i class="fa fa-check"></i><b>28.11</b> Other loop functions</a></li>
<li class="chapter" data-level="28.12" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#lapply-functions"><i class="fa fa-check"></i><b>28.12</b> <code>lapply</code> functions</a></li>
<li class="chapter" data-level="28.13" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsapply"><i class="fa fa-check"></i><b>28.13</b> Other loop functions:<code>apply</code></a></li>
<li class="chapter" data-level="28.14" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsmapply"><i class="fa fa-check"></i><b>28.14</b> Other loop functions:<code>mapply</code></a></li>
<li class="chapter" data-level="28.15" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionstapply"><i class="fa fa-check"></i><b>28.15</b> Other loop functions:<code>tapply</code></a></li>
<li class="chapter" data-level="28.16" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionssplit"><i class="fa fa-check"></i><b>28.16</b> Other loop functions:<code>split</code></a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html"><i class="fa fa-check"></i><b>29</b> – Bioinformatics Literature</a>
<ul>
<li class="chapter" data-level="29.1" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#peer-reviewed-bioinformatics-journals"><i class="fa fa-check"></i><b>29.1</b> Peer-reviewed bioinformatics journals</a></li>
<li class="chapter" data-level="29.2" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#access-to-journals-articles"><i class="fa fa-check"></i><b>29.2</b> Access to journals articles</a></li>
<li class="chapter" data-level="29.3" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#informal-access-to-code-preprints-and-vignettes"><i class="fa fa-check"></i><b>29.3</b> Informal access to code: Preprints and vignettes</a></li>
<li class="chapter" data-level="29.4" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#online-bioinformatics-courses"><i class="fa fa-check"></i><b>29.4</b> Online bioinformatics courses</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html"><i class="fa fa-check"></i><b>30</b> – Importing/Exporting Data into R</a>
<ul>
<li class="chapter" data-level="30.1" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#typing-directly"><i class="fa fa-check"></i><b>30.1</b> Typing directly</a></li>
<li class="chapter" data-level="30.2" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#summary-versus-raw-data"><i class="fa fa-check"></i><b>30.2</b> Summary versus raw data</a></li>
<li class="chapter" data-level="30.3" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#importing-data-from-a-file"><i class="fa fa-check"></i><b>30.3</b> Importing data from a file</a></li>
<li class="chapter" data-level="30.4" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-to-a-file"><i class="fa fa-check"></i><b>30.4</b> Saving data to a file</a></li>
<li class="chapter" data-level="30.5" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-as-rdata-.rdata-.rda"><i class="fa fa-check"></i><b>30.5</b> Saving data as Rdata (.rdata, .rda)</a></li>
<li class="chapter" data-level="30.6" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#import-excel-formated-data"><i class="fa fa-check"></i><b>30.6</b> Import Excel formated data</a></li>
<li class="chapter" data-level="30.7" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#save-data-as-excel-formated-file"><i class="fa fa-check"></i><b>30.7</b> Save data as Excel formated file</a></li>
<li class="chapter" data-level="30.8" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#downloading-data-files-github"><i class="fa fa-check"></i><b>30.8</b> Downloading data files (GitHub)</a></li>
<li class="chapter" data-level="30.9" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#accessing-online-data"><i class="fa fa-check"></i><b>30.9</b> Accessing online data</a></li>
<li class="chapter" data-level="30.10" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#fixing-problems-while-importing-datasets"><i class="fa fa-check"></i><b>30.10</b> Fixing problems while importing datasets</a></li>
<li class="chapter" data-level="30.11" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#simplifying-embellished-excel-datasets"><i class="fa fa-check"></i><b>30.11</b> Simplifying embellished excel datasets</a></li>
<li class="chapter" data-level="30.12" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#data-from-clinicaltrials.gov"><i class="fa fa-check"></i><b>30.12</b> Data from clinicaltrials.gov</a></li>
<li class="chapter" data-level="30.13" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#data-scrapping-from-a-website"><i class="fa fa-check"></i><b>30.13</b> Data scrapping from a website</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="working-with-proteins.html"><a href="working-with-proteins.html"><i class="fa fa-check"></i><b>31</b> – Working with Proteins</a>
<ul>
<li class="chapter" data-level="31.1" data-path="working-with-proteins.html"><a href="working-with-proteins.html#sequences-in-fasta-format"><i class="fa fa-check"></i><b>31.1</b> Sequences in FASTA format</a></li>
<li class="chapter" data-level="31.2" data-path="working-with-proteins.html"><a href="working-with-proteins.html#fetching-aa-sequences-from-ncbi"><i class="fa fa-check"></i><b>31.2</b> Fetching AA sequences from NCBI</a></li>
<li class="chapter" data-level="31.3" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-aa-sequences-in-r"><i class="fa fa-check"></i><b>31.3</b> Importing AA sequences in R</a></li>
<li class="chapter" data-level="31.4" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-nucleotide-sequences-in-r"><i class="fa fa-check"></i><b>31.4</b> Importing nucleotide sequences in R</a></li>
<li class="chapter" data-level="31.5" data-path="working-with-proteins.html"><a href="working-with-proteins.html#translating-a-nucleotides-to-aa"><i class="fa fa-check"></i><b>31.5</b> Translating a nucleotides to AA</a></li>
<li class="chapter" data-level="31.6" data-path="working-with-proteins.html"><a href="working-with-proteins.html#finding-orfs-in-a-nucleotide-sequence"><i class="fa fa-check"></i><b>31.6</b> Finding ORFs in a nucleotide sequence</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="miscellaneous.html"><a href="miscellaneous.html"><i class="fa fa-check"></i><b>32</b> – Miscellaneous</a>
<ul>
<li class="chapter" data-level="32.1" data-path="miscellaneous.html"><a href="miscellaneous.html#sampling-from-a-vector-of-names"><i class="fa fa-check"></i><b>32.1</b> Sampling from a vector of names</a></li>
<li class="chapter" data-level="32.2" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-an-article-pdf"><i class="fa fa-check"></i><b>32.2</b> Read a text from an article pdf</a></li>
<li class="chapter" data-level="32.3" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-website"><i class="fa fa-check"></i><b>32.3</b> Read a text from a website</a></li>
<li class="chapter" data-level="32.4" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-file"><i class="fa fa-check"></i><b>32.4</b> Read a text from a file</a></li>
<li class="chapter" data-level="32.5" data-path="miscellaneous.html"><a href="miscellaneous.html#generate-the-word-cloud"><i class="fa fa-check"></i><b>32.5</b> Generate the word cloud</a></li>
<li class="chapter" data-level="32.6" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-pdf"><i class="fa fa-check"></i><b>32.6</b> worldcloud – PDF</a></li>
<li class="chapter" data-level="32.7" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-website"><i class="fa fa-check"></i><b>32.7</b> worldcloud – website</a></li>
<li class="chapter" data-level="32.8" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-text"><i class="fa fa-check"></i><b>32.8</b> worldcloud – text</a></li>
<li class="chapter" data-level="32.9" data-path="miscellaneous.html"><a href="miscellaneous.html#miscellaneous-1"><i class="fa fa-check"></i><b>32.9</b> Miscellaneous</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics Workshop Gitbook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="multidimensional-data-in-r" class="section level1" number="8">
<h1><span class="header-section-number">Session 8</span> – Multidimensional Data in R</h1>
<p>Many datasets derive from multiple variables measured from hundreds of individuals and many of such parameters are obtained simultaneously (i.e., multidimensional) and together are often difficult to interpret. Most researchers aim to reduce such complexity, identify groups of interest or increase data interpretability using diverse methods like principal component analysis (PCA), multidimensional scaling (MDS), cluster analyses among others.
We will explore some of these methods.</p>
<div id="principal-component-analysis" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Principal Component Analysis</h2>
<p>PCA is a common approach to reduce dimensionality in a dataset.
For instance, PCA reduce the data complexity from many variables into few summary components, while preserving much of the statistical information. During the analysis, new variables (named components) are found that are linear functions of those in the original dataset, that successively maximize variance and that are uncorrelated with each other, see <a href="https://royalsocietypublishing.org/doi/10.1098/rsta.2015.0202" target="_blank">Jolliffe and Cadima 2016</a>.</p>
<p>PCA is recommended when you have large dataset of variables (e.g., &gt; 8) and many individuals (&gt;20 or better in the hundreds) with measurements of such variables. For this example, we will use the <code>env</code> data frame of the <code>doubs</code> dataset from the R-package <a href="https://cran.r-project.org/web/packages/ade4/index.html" target="_blank"><strong>ade4</strong></a>. The <code>env</code> is a data frame with 30 rows (sites) and 11 environmental variables.</p>
<div class="sourceCode" id="cb363"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb363-1"><a href="multidimensional-data-in-r.html#cb363-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we can install ade4 and load the dataset for this example</span></span>
<span id="cb363-2"><a href="multidimensional-data-in-r.html#cb363-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ade4&quot;</span>)</span>
<span id="cb363-3"><a href="multidimensional-data-in-r.html#cb363-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb363-4"><a href="multidimensional-data-in-r.html#cb363-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb363-5"><a href="multidimensional-data-in-r.html#cb363-5" aria-hidden="true" tabindex="-1"></a><span class="do">##load the doubs dataset into the R environment</span></span>
<span id="cb363-6"><a href="multidimensional-data-in-r.html#cb363-6" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(doubs)</span>
<span id="cb363-7"><a href="multidimensional-data-in-r.html#cb363-7" aria-hidden="true" tabindex="-1"></a>env_raw <span class="ot">&lt;-</span> doubs<span class="sc">$</span>env</span>
<span id="cb363-8"><a href="multidimensional-data-in-r.html#cb363-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(env_raw)</span>
<span id="cb363-9"><a href="multidimensional-data-in-r.html#cb363-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  dfs alt   slo flo pH har pho nit amm oxy bdo</span></span>
<span id="cb363-10"><a href="multidimensional-data-in-r.html#cb363-10" aria-hidden="true" tabindex="-1"></a><span class="co">#1   3 934 6.176  84 79  45   1  20   0 122  27</span></span>
<span id="cb363-11"><a href="multidimensional-data-in-r.html#cb363-11" aria-hidden="true" tabindex="-1"></a><span class="co">#2  22 932 3.434 100 80  40   2  20  10 103  19</span></span>
<span id="cb363-12"><a href="multidimensional-data-in-r.html#cb363-12" aria-hidden="true" tabindex="-1"></a><span class="co">#3 102 914 3.638 180 83  52   5  22   5 105  35</span></span>
<span id="cb363-13"><a href="multidimensional-data-in-r.html#cb363-13" aria-hidden="true" tabindex="-1"></a><span class="co">#4 185 854 3.497 253 80  72  10  21   0 110  13</span></span>
<span id="cb363-14"><a href="multidimensional-data-in-r.html#cb363-14" aria-hidden="true" tabindex="-1"></a><span class="co">#5 215 849 3.178 264 81  84  38  52  20  80  62</span></span>
<span id="cb363-15"><a href="multidimensional-data-in-r.html#cb363-15" aria-hidden="true" tabindex="-1"></a><span class="co">#6 324 846 3.497 286 79  60  20  15   0 102  53</span></span></code></pre></div>
<p>We can start by getting a pairwise bivariate plots of these 11 variables.</p>
<div class="sourceCode" id="cb364"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb364-1"><a href="multidimensional-data-in-r.html#cb364-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we can install ade4 and load the dataset for this example</span></span>
<span id="cb364-2"><a href="multidimensional-data-in-r.html#cb364-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pairs</span>(env_raw, <span class="at">pch =</span> <span class="dv">19</span>, <span class="at">cex=</span><span class="fl">0.5</span>)</span></code></pre></div>
<p><img src="figures/session_8_figs/1_pairwise_env_raw.png" width="100%" /></p>
<p>We notice a relative variability in these pairwise plots. Some show a linear relationship, others do not. Likewise, some scales differences are evident (some are in the hundreds, others much less).</p>
<div class="sourceCode" id="cb365"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb365-1"><a href="multidimensional-data-in-r.html#cb365-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we can use the function modify() form the R-package **purrr** to standardize this dataset using log</span></span>
<span id="cb365-2"><a href="multidimensional-data-in-r.html#cb365-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb365-3"><a href="multidimensional-data-in-r.html#cb365-3" aria-hidden="true" tabindex="-1"></a>env_log <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">modify</span>(env_raw, log)</span>
<span id="cb365-4"><a href="multidimensional-data-in-r.html#cb365-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(env_log)</span>
<span id="cb365-5"><a href="multidimensional-data-in-r.html#cb365-5" aria-hidden="true" tabindex="-1"></a><span class="co">#       dfs      alt      slo      flo       pH      har       pho      nit      amm      oxy      bdo</span></span>
<span id="cb365-6"><a href="multidimensional-data-in-r.html#cb365-6" aria-hidden="true" tabindex="-1"></a><span class="co">#1 1.098612 6.839476 1.820671 4.430817 4.369448 3.806662 0.0000000 2.995732     -Inf 4.804021 3.295837</span></span>
<span id="cb365-7"><a href="multidimensional-data-in-r.html#cb365-7" aria-hidden="true" tabindex="-1"></a><span class="co">#2 3.091042 6.837333 1.233726 4.605170 4.382027 3.688879 0.6931472 2.995732 2.302585 4.634729 2.944439</span></span>
<span id="cb365-8"><a href="multidimensional-data-in-r.html#cb365-8" aria-hidden="true" tabindex="-1"></a><span class="co">#3 4.624973 6.817831 1.291434 5.192957 4.418841 3.951244 1.6094379 3.091042 1.609438 4.653960 3.555348</span></span>
<span id="cb365-9"><a href="multidimensional-data-in-r.html#cb365-9" aria-hidden="true" tabindex="-1"></a><span class="co">#4 5.220356 6.749931 1.251905 5.533389 4.382027 4.276666 2.3025851 3.044522     -Inf 4.700480 2.564949</span></span>
<span id="cb365-10"><a href="multidimensional-data-in-r.html#cb365-10" aria-hidden="true" tabindex="-1"></a><span class="co">#5 5.370638 6.744059 1.156252 5.575949 4.394449 4.430817 3.6375862 3.951244 2.995732 4.382027 4.127134</span></span>
<span id="cb365-11"><a href="multidimensional-data-in-r.html#cb365-11" aria-hidden="true" tabindex="-1"></a><span class="co">#6 5.780744 6.740519 1.251905 5.655992 4.369448 4.094345 2.9957323 2.708050     -Inf 4.624973 3.970292</span></span>
<span id="cb365-12"><a href="multidimensional-data-in-r.html#cb365-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb365-13"><a href="multidimensional-data-in-r.html#cb365-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Notice some -Inf value, we will remove that column 9 that corresponds to ‘amm’</span></span>
<span id="cb365-14"><a href="multidimensional-data-in-r.html#cb365-14" aria-hidden="true" tabindex="-1"></a>env_log_red <span class="ot">&lt;-</span> env_log[,<span class="sc">-</span><span class="dv">9</span>]</span>
<span id="cb365-15"><a href="multidimensional-data-in-r.html#cb365-15" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(env_log_red)</span>
<span id="cb365-16"><a href="multidimensional-data-in-r.html#cb365-16" aria-hidden="true" tabindex="-1"></a><span class="co">#       dfs      alt      slo      flo       pH      har       pho      nit      oxy      bdo</span></span>
<span id="cb365-17"><a href="multidimensional-data-in-r.html#cb365-17" aria-hidden="true" tabindex="-1"></a><span class="co">#1 1.098612 6.839476 1.820671 4.430817 4.369448 3.806662 0.0000000 2.995732 4.804021 3.295837</span></span>
<span id="cb365-18"><a href="multidimensional-data-in-r.html#cb365-18" aria-hidden="true" tabindex="-1"></a><span class="co">#2 3.091042 6.837333 1.233726 4.605170 4.382027 3.688879 0.6931472 2.995732 4.634729 2.944439</span></span>
<span id="cb365-19"><a href="multidimensional-data-in-r.html#cb365-19" aria-hidden="true" tabindex="-1"></a><span class="co">#3 4.624973 6.817831 1.291434 5.192957 4.418841 3.951244 1.6094379 3.091042 4.653960 3.555348</span></span>
<span id="cb365-20"><a href="multidimensional-data-in-r.html#cb365-20" aria-hidden="true" tabindex="-1"></a><span class="co">#4 5.220356 6.749931 1.251905 5.533389 4.382027 4.276666 2.3025851 3.044522 4.700480 2.564949</span></span>
<span id="cb365-21"><a href="multidimensional-data-in-r.html#cb365-21" aria-hidden="true" tabindex="-1"></a><span class="co">#5 5.370638 6.744059 1.156252 5.575949 4.394449 4.430817 3.6375862 3.951244 4.382027 4.127134</span></span>
<span id="cb365-22"><a href="multidimensional-data-in-r.html#cb365-22" aria-hidden="true" tabindex="-1"></a><span class="co">#6 5.780744 6.740519 1.251905 5.655992 4.369448 4.094345 2.9957323 2.708050 4.624973 3.970292</span></span>
<span id="cb365-23"><a href="multidimensional-data-in-r.html#cb365-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb365-24"><a href="multidimensional-data-in-r.html#cb365-24" aria-hidden="true" tabindex="-1"></a><span class="do">## we can install ade4 and load the modified dataset for this example</span></span>
<span id="cb365-25"><a href="multidimensional-data-in-r.html#cb365-25" aria-hidden="true" tabindex="-1"></a><span class="fu">pairs</span>(env_log_red, <span class="at">pch =</span> <span class="dv">19</span>, <span class="at">cex=</span><span class="fl">0.5</span>)</span></code></pre></div>
<p>We can run PCA using the preloaded function in R or with several specialized packages. We will start with the functions <code>prcomp()</code> and <code>summary()</code>.</p>
<div class="sourceCode" id="cb366"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb366-1"><a href="multidimensional-data-in-r.html#cb366-1" aria-hidden="true" tabindex="-1"></a><span class="do">## apply PCA - scale. = TRUE is highly advisable, but default is FALSE. </span></span>
<span id="cb366-2"><a href="multidimensional-data-in-r.html#cb366-2" aria-hidden="true" tabindex="-1"></a>env_prcomp <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(env_log_red, <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">scale. =</span> <span class="cn">TRUE</span>) </span>
<span id="cb366-3"><a href="multidimensional-data-in-r.html#cb366-3" aria-hidden="true" tabindex="-1"></a>env_prcomp</span>
<span id="cb366-4"><a href="multidimensional-data-in-r.html#cb366-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Standard deviations (1, .., p=10):</span></span>
<span id="cb366-5"><a href="multidimensional-data-in-r.html#cb366-5" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] 2.5031260 1.2651443 0.9875811 0.6860815 0.5095091 0.3946341 0.3212448 0.2824013 0.2580574 0.1512203</span></span>
<span id="cb366-6"><a href="multidimensional-data-in-r.html#cb366-6" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb366-7"><a href="multidimensional-data-in-r.html#cb366-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Rotation (n x k) = (10 x 10):</span></span>
<span id="cb366-8"><a href="multidimensional-data-in-r.html#cb366-8" aria-hidden="true" tabindex="-1"></a><span class="co">#            PC1         PC2        PC3         PC4          PC5         PC6          PC7         PC8         PC9         PC10</span></span>
<span id="cb366-9"><a href="multidimensional-data-in-r.html#cb366-9" aria-hidden="true" tabindex="-1"></a><span class="co">#dfs  0.36486258 -0.18617302  0.1482270 -0.27425018  0.070856291 -0.06822936  0.375972370  0.04774989 -0.73277048  0.211581601</span></span>
<span id="cb366-10"><a href="multidimensional-data-in-r.html#cb366-10" aria-hidden="true" tabindex="-1"></a><span class="co">#alt -0.36451449  0.12764399 -0.1690422 -0.36872511  0.165563399  0.30103670  0.159545574  0.45954368 -0.16927250 -0.550436218</span></span>
<span id="cb366-11"><a href="multidimensional-data-in-r.html#cb366-11" aria-hidden="true" tabindex="-1"></a><span class="co">#slo -0.33426066  0.15461463  0.2062596 -0.41599181 -0.694152941  0.08059586 -0.008314283 -0.38539136 -0.10748520  0.016710642</span></span>
<span id="cb366-12"><a href="multidimensional-data-in-r.html#cb366-12" aria-hidden="true" tabindex="-1"></a><span class="co">#flo  0.35536702 -0.26645120  0.2469480  0.07962110 -0.140470255 -0.21095156  0.148040510 -0.22735329  0.11111159 -0.764991160</span></span>
<span id="cb366-13"><a href="multidimensional-data-in-r.html#cb366-13" aria-hidden="true" tabindex="-1"></a><span class="co">#pH   0.02686455 -0.43046956 -0.8301590 -0.07333534 -0.270021833 -0.14147519 -0.099820286 -0.07667608 -0.09705901 -0.035234427</span></span>
<span id="cb366-14"><a href="multidimensional-data-in-r.html#cb366-14" aria-hidden="true" tabindex="-1"></a><span class="co">#har  0.32775459 -0.15043755  0.1025784 -0.71869778  0.192865111 -0.02065425 -0.450542106  0.06200716  0.30538254  0.071982806</span></span>
<span id="cb366-15"><a href="multidimensional-data-in-r.html#cb366-15" aria-hidden="true" tabindex="-1"></a><span class="co">#pho  0.35978479  0.15025127 -0.2384279 -0.09780973  0.004998182  0.59283974  0.493290882 -0.25929815  0.33522306  0.084468872</span></span>
<span id="cb366-16"><a href="multidimensional-data-in-r.html#cb366-16" aria-hidden="true" tabindex="-1"></a><span class="co">#nit  0.36509191 -0.03139547  0.1053406  0.23840435 -0.511528294  0.41078684 -0.313263865  0.51006660 -0.09509725 -0.006021563</span></span>
<span id="cb366-17"><a href="multidimensional-data-in-r.html#cb366-17" aria-hidden="true" tabindex="-1"></a><span class="co">#oxy -0.25192228 -0.56073462  0.2173434 -0.06473107 -0.174757541 -0.08405657  0.438140938  0.36832297  0.39110186  0.229102408</span></span>
<span id="cb366-18"><a href="multidimensional-data-in-r.html#cb366-18" aria-hidden="true" tabindex="-1"></a><span class="co">#bdo  0.24827761  0.55517993 -0.1762335 -0.13016095 -0.252295130 -0.55318468  0.254587830  0.33694829  0.18286003  0.026082994</span></span>
<span id="cb366-19"><a href="multidimensional-data-in-r.html#cb366-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb366-20"><a href="multidimensional-data-in-r.html#cb366-20" aria-hidden="true" tabindex="-1"></a><span class="do">## see its structure</span></span>
<span id="cb366-21"><a href="multidimensional-data-in-r.html#cb366-21" aria-hidden="true" tabindex="-1"></a><span class="co">#str(env_prcomp)</span></span>
<span id="cb366-22"><a href="multidimensional-data-in-r.html#cb366-22" aria-hidden="true" tabindex="-1"></a><span class="co">#List of 5</span></span>
<span id="cb366-23"><a href="multidimensional-data-in-r.html#cb366-23" aria-hidden="true" tabindex="-1"></a><span class="co"># $ sdev    : num [1:10] 2.503 1.265 0.988 0.686 0.51 ...</span></span>
<span id="cb366-24"><a href="multidimensional-data-in-r.html#cb366-24" aria-hidden="true" tabindex="-1"></a><span class="co">#$ rotation: num [1:10, 1:10] 0.3649 -0.3645 -0.3343 0.3554 0.0269 ...</span></span>
<span id="cb366-25"><a href="multidimensional-data-in-r.html#cb366-25" aria-hidden="true" tabindex="-1"></a><span class="co">#  ..- attr(*, &quot;dimnames&quot;)=List of 2</span></span>
<span id="cb366-26"><a href="multidimensional-data-in-r.html#cb366-26" aria-hidden="true" tabindex="-1"></a><span class="co"># .. ..$ : chr [1:10] &quot;dfs&quot; &quot;alt&quot; &quot;slo&quot; &quot;flo&quot; ...</span></span>
<span id="cb366-27"><a href="multidimensional-data-in-r.html#cb366-27" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. ..$ : chr [1:10] &quot;PC1&quot; &quot;PC2&quot; &quot;PC3&quot; &quot;PC4&quot; ...</span></span>
<span id="cb366-28"><a href="multidimensional-data-in-r.html#cb366-28" aria-hidden="true" tabindex="-1"></a><span class="co">#$ center  : Named num [1:10] 6.887 6.019 0.943 7.153 4.388 ...</span></span>
<span id="cb366-29"><a href="multidimensional-data-in-r.html#cb366-29" aria-hidden="true" tabindex="-1"></a><span class="co"># ..- attr(*, &quot;names&quot;)= chr [1:10] &quot;dfs&quot; &quot;alt&quot; &quot;slo&quot; &quot;flo&quot; ...</span></span>
<span id="cb366-30"><a href="multidimensional-data-in-r.html#cb366-30" aria-hidden="true" tabindex="-1"></a><span class="co">#$ scale   : Named num [1:10] 1.6786 0.5747 0.3863 1.3041 0.0214 ...</span></span>
<span id="cb366-31"><a href="multidimensional-data-in-r.html#cb366-31" aria-hidden="true" tabindex="-1"></a><span class="co"># ..- attr(*, &quot;names&quot;)= chr [1:10] &quot;dfs&quot; &quot;alt&quot; &quot;slo&quot; &quot;flo&quot; ...</span></span>
<span id="cb366-32"><a href="multidimensional-data-in-r.html#cb366-32" aria-hidden="true" tabindex="-1"></a><span class="co">#$ x       : num [1:30, 1:10] -6 -4.96 -3.6 -3.18 -1.24 ...</span></span>
<span id="cb366-33"><a href="multidimensional-data-in-r.html#cb366-33" aria-hidden="true" tabindex="-1"></a><span class="co">#  ..- attr(*, &quot;dimnames&quot;)=List of 2</span></span>
<span id="cb366-34"><a href="multidimensional-data-in-r.html#cb366-34" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. ..$ : chr [1:30] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; ...</span></span>
<span id="cb366-35"><a href="multidimensional-data-in-r.html#cb366-35" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. ..$ : chr [1:10] &quot;PC1&quot; &quot;PC2&quot; &quot;PC3&quot; &quot;PC4&quot; ...</span></span>
<span id="cb366-36"><a href="multidimensional-data-in-r.html#cb366-36" aria-hidden="true" tabindex="-1"></a><span class="co"># - attr(*, &quot;class&quot;)= chr &quot;prcomp&quot;</span></span>
<span id="cb366-37"><a href="multidimensional-data-in-r.html#cb366-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb366-38"><a href="multidimensional-data-in-r.html#cb366-38" aria-hidden="true" tabindex="-1"></a><span class="do">## we can get a summary</span></span>
<span id="cb366-39"><a href="multidimensional-data-in-r.html#cb366-39" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(env_prcomp)</span>
<span id="cb366-40"><a href="multidimensional-data-in-r.html#cb366-40" aria-hidden="true" tabindex="-1"></a><span class="co">#Importance of components:</span></span>
<span id="cb366-41"><a href="multidimensional-data-in-r.html#cb366-41" aria-hidden="true" tabindex="-1"></a><span class="co">#                          PC1    PC2     PC3     PC4     PC5     PC6     PC7     PC8     PC9    PC10</span></span>
<span id="cb366-42"><a href="multidimensional-data-in-r.html#cb366-42" aria-hidden="true" tabindex="-1"></a><span class="co">#Standard deviation     2.5031 1.2651 0.98758 0.68608 0.50951 0.39463 0.32124 0.28240 0.25806 0.15122</span></span>
<span id="cb366-43"><a href="multidimensional-data-in-r.html#cb366-43" aria-hidden="true" tabindex="-1"></a><span class="co">#Proportion of Variance 0.6266 0.1601 0.09753 0.04707 0.02596 0.01557 0.01032 0.00798 0.00666 0.00229</span></span>
<span id="cb366-44"><a href="multidimensional-data-in-r.html#cb366-44" aria-hidden="true" tabindex="-1"></a><span class="co">#Cumulative Proportion  0.6266 0.7866 0.88415 0.93123 0.95719 0.97276 0.98308 0.99105 0.99771 1.00000</span></span></code></pre></div>
<p>The output of <code>env_prcomp</code> includes the standard deviation of each of the 10 principal components (PCs). You should expect as many PCs as variables in your dataset. The values under <code>Rotation (n x k)</code> are the rotation (or loadings) of each of variables in the original dataset that have been maximize its relevance on each components. The higher the absolute value of the loading, the more that variable loads (i.e., is better represented) on each specific PC.
The<code>summary(env_prcomp)</code> output describe the importance of each PC. The proportion of variance indicates the proportion of the total variance in the data explained by each PC. We can notice the values of the second row are progressively added resulting in a cumulative proportion of explained variance. In most cases, the first PC accounts for more most of the variance explain (i.e., 62.66%). If we consider the two firsts PCs, then the total variance explained by these 2 PCs is 78.66%. The rest of PCs are progressively less and less relevant to explain the variance of the data.</p>
<p>For most purposes, the first two PCs are used for plots and visualizing dimensionality reduction. However, we can determine how many PCs to retain using a scree plot and other references using the R-package <a href="https://cran.r-project.org/web/packages/factoextra/index.html" target="_blank"><strong>factoextra</strong></a>.</p>
<p>We can get an scree plot with the function <code>fviz_eig()</code>.</p>
<div class="sourceCode" id="cb367"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb367-1"><a href="multidimensional-data-in-r.html#cb367-1" aria-hidden="true" tabindex="-1"></a><span class="do">## install package, if not present. It also requires the R-package ggplot2.</span></span>
<span id="cb367-2"><a href="multidimensional-data-in-r.html#cb367-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;factoextra&quot;</span>)</span>
<span id="cb367-3"><a href="multidimensional-data-in-r.html#cb367-3" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb367-4"><a href="multidimensional-data-in-r.html#cb367-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb367-5"><a href="multidimensional-data-in-r.html#cb367-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Loading required package: ggplot2</span></span>
<span id="cb367-6"><a href="multidimensional-data-in-r.html#cb367-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Welcome! Want to learn more? See two factoextra-related books at https://goo.gl/ve3WBa</span></span>
<span id="cb367-7"><a href="multidimensional-data-in-r.html#cb367-7" aria-hidden="true" tabindex="-1"></a>factoextra<span class="sc">::</span><span class="fu">fviz_eig</span>(env_prcomp)</span></code></pre></div>
<p><img src="figures/session_8_figs/3_scree_plot.png" width="100%" /></p>
<p>This <strong>scree plot</strong> shows the eigenvalues that provide information on the percentage of variances explained by each PC and support our intuition that the first two PCs are the most useful to summarize the variability of our dataset. For instance, we see a marked drop in the percentage of explained after PC2, and we can ignore those PCs (3-10) that explain relatively little variation. Thus, we should focus primarily on PC1 and PC2.</p>
<p>We can graph our samples (i.e., individuals on our dataset) with their PC1 and PC2 coordinates using the function <code>fviz_pca_ind()</code>.</p>
<div class="sourceCode" id="cb368"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb368-1"><a href="multidimensional-data-in-r.html#cb368-1" aria-hidden="true" tabindex="-1"></a>factoextra<span class="sc">::</span><span class="fu">fviz_pca_ind</span>(env_prcomp,</span>
<span id="cb368-2"><a href="multidimensional-data-in-r.html#cb368-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">col.ind =</span> <span class="st">&quot;cos2&quot;</span>, </span>
<span id="cb368-3"><a href="multidimensional-data-in-r.html#cb368-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">gradient.cols =</span> <span class="fu">c</span>(<span class="st">&quot;firebrick4&quot;</span>, <span class="st">&quot;gray&quot;</span>, <span class="st">&quot;dodgerblue4&quot;</span>),</span>
<span id="cb368-4"><a href="multidimensional-data-in-r.html#cb368-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">repel =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="figures/session_8_figs/4_individual_plot.png" width="100%" /></p>
<p>Individuals with a similar profile tend to group together and color <code>cos2</code> refers to the quality of representation. Notice that you can choose the color gradient that you want with the argument <code>gradient.cols</code>, it might need 3 colors.</p>
<div class="sourceCode" id="cb369"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb369-1"><a href="multidimensional-data-in-r.html#cb369-1" aria-hidden="true" tabindex="-1"></a><span class="do">### </span><span class="al">NOTE</span><span class="do">: Sometimes you get this error. </span></span>
<span id="cb369-2"><a href="multidimensional-data-in-r.html#cb369-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Error in UseMethod(&quot;depth&quot;) : </span></span>
<span id="cb369-3"><a href="multidimensional-data-in-r.html#cb369-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  no applicable method for &#39;depth&#39; applied to an object of class &quot;NULL&quot;</span></span>
<span id="cb369-4"><a href="multidimensional-data-in-r.html#cb369-4" aria-hidden="true" tabindex="-1"></a><span class="do">### Try to resubmit the fviz_pca_ind() after closing the graph window.</span></span></code></pre></div>
<p>We can also graph of variables that we used in our PCA using the function <code>fviz_pca_var()</code>.</p>
<div class="sourceCode" id="cb370"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb370-1"><a href="multidimensional-data-in-r.html#cb370-1" aria-hidden="true" tabindex="-1"></a>factoextra<span class="sc">::</span><span class="fu">fviz_pca_var</span>(env_prcomp,</span>
<span id="cb370-2"><a href="multidimensional-data-in-r.html#cb370-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">col.var =</span> <span class="st">&quot;contrib&quot;</span>, </span>
<span id="cb370-3"><a href="multidimensional-data-in-r.html#cb370-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">gradient.cols =</span> <span class="fu">c</span>(<span class="st">&quot;dodgerblue4&quot;</span>, <span class="st">&quot;#E7B800&quot;</span>, <span class="st">&quot;firebrick4&quot;</span>),</span>
<span id="cb370-4"><a href="multidimensional-data-in-r.html#cb370-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">repel =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="figures/session_8_figs/5_variable_plot.png" width="100%" /></p>
<p>We can see that positive correlated variables point to the same side of the plot, while negative correlated variables point to opposite sides of the graph. Notice that you can choose the color gradient that you want with the argument <code>gradient.cols</code>, it might need 3 colors.</p>
<div class="sourceCode" id="cb371"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb371-1"><a href="multidimensional-data-in-r.html#cb371-1" aria-hidden="true" tabindex="-1"></a><span class="do">### </span><span class="al">NOTE</span><span class="do">: Sometimes you get this error. </span></span>
<span id="cb371-2"><a href="multidimensional-data-in-r.html#cb371-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Error in UseMethod(&quot;depth&quot;) : </span></span>
<span id="cb371-3"><a href="multidimensional-data-in-r.html#cb371-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  no applicable method for &#39;depth&#39; applied to an object of class &quot;NULL&quot;</span></span>
<span id="cb371-4"><a href="multidimensional-data-in-r.html#cb371-4" aria-hidden="true" tabindex="-1"></a><span class="do">### Try to resubmit the fviz_pca_ind() after closing the graph window.</span></span></code></pre></div>
<p>We can also graph plot both individuals and variables at the same time using the function <code>fviz_pca_biplot()</code>.</p>
<div class="sourceCode" id="cb372"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb372-1"><a href="multidimensional-data-in-r.html#cb372-1" aria-hidden="true" tabindex="-1"></a>factoextra<span class="sc">::</span><span class="fu">fviz_pca_biplot</span>(env_prcomp,</span>
<span id="cb372-2"><a href="multidimensional-data-in-r.html#cb372-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">col.var =</span> <span class="st">&quot;gray&quot;</span>, <span class="co"># Variables color</span></span>
<span id="cb372-3"><a href="multidimensional-data-in-r.html#cb372-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">col.ind =</span> <span class="st">&quot;dodgerblue4&quot;</span>,  <span class="co"># Individuals color</span></span>
<span id="cb372-4"><a href="multidimensional-data-in-r.html#cb372-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">repel =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="figures/session_8_figs/6_biplot_individual_variable.png" width="100%" /></p>
<p>We can extract other metrics of our PCA with several functions of <strong>factoextra</strong>.</p>
<p>The eigenvalues can be printed with the function <code>get_eigenvalue()</code>describe the importance of each PC. Similar to <code>summary(env_prcomp)</code>.</p>
<div class="sourceCode" id="cb373"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb373-1"><a href="multidimensional-data-in-r.html#cb373-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Eigenvalues</span></span>
<span id="cb373-2"><a href="multidimensional-data-in-r.html#cb373-2" aria-hidden="true" tabindex="-1"></a>eig_val_env_prcomp <span class="ot">&lt;-</span> factoextra<span class="sc">::</span><span class="fu">get_eigenvalue</span>(env_prcomp)</span>
<span id="cb373-3"><a href="multidimensional-data-in-r.html#cb373-3" aria-hidden="true" tabindex="-1"></a>eig_val_env_prcomp</span>
<span id="cb373-4"><a href="multidimensional-data-in-r.html#cb373-4" aria-hidden="true" tabindex="-1"></a><span class="co">#       eigenvalue variance.percent cumulative.variance.percent</span></span>
<span id="cb373-5"><a href="multidimensional-data-in-r.html#cb373-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Dim.1  6.26563999       62.6563999                    62.65640</span></span>
<span id="cb373-6"><a href="multidimensional-data-in-r.html#cb373-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Dim.2  1.60059015       16.0059015                    78.66230</span></span>
<span id="cb373-7"><a href="multidimensional-data-in-r.html#cb373-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Dim.3  0.97531642        9.7531642                    88.41547</span></span>
<span id="cb373-8"><a href="multidimensional-data-in-r.html#cb373-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Dim.4  0.47070789        4.7070789                    93.12254</span></span>
<span id="cb373-9"><a href="multidimensional-data-in-r.html#cb373-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Dim.5  0.25959955        2.5959955                    95.71854</span></span>
<span id="cb373-10"><a href="multidimensional-data-in-r.html#cb373-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Dim.6  0.15573609        1.5573609                    97.27590</span></span>
<span id="cb373-11"><a href="multidimensional-data-in-r.html#cb373-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Dim.7  0.10319822        1.0319822                    98.30788</span></span>
<span id="cb373-12"><a href="multidimensional-data-in-r.html#cb373-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Dim.8  0.07975052        0.7975052                    99.10539</span></span>
<span id="cb373-13"><a href="multidimensional-data-in-r.html#cb373-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Dim.9  0.06659361        0.6659361                    99.77132</span></span>
<span id="cb373-14"><a href="multidimensional-data-in-r.html#cb373-14" aria-hidden="true" tabindex="-1"></a><span class="co">#Dim.10 0.02286757        0.2286757                   100.00000</span></span></code></pre></div>
<p>The variable results can be printed with the function <code>get_pca_var()</code>.</p>
<div class="sourceCode" id="cb374"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb374-1"><a href="multidimensional-data-in-r.html#cb374-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Results for Variables</span></span>
<span id="cb374-2"><a href="multidimensional-data-in-r.html#cb374-2" aria-hidden="true" tabindex="-1"></a>res_var_env_prcomp <span class="ot">&lt;-</span> factoextra<span class="sc">::</span><span class="fu">get_pca_var</span>(env_prcomp)</span>
<span id="cb374-3"><a href="multidimensional-data-in-r.html#cb374-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Coordinates</span></span>
<span id="cb374-4"><a href="multidimensional-data-in-r.html#cb374-4" aria-hidden="true" tabindex="-1"></a>res_var_env_prcomp<span class="sc">$</span>coord </span>
<span id="cb374-5"><a href="multidimensional-data-in-r.html#cb374-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Contributions to the PCs     </span></span>
<span id="cb374-6"><a href="multidimensional-data-in-r.html#cb374-6" aria-hidden="true" tabindex="-1"></a>res_var_env_prcomp<span class="sc">$</span>contrib </span>
<span id="cb374-7"><a href="multidimensional-data-in-r.html#cb374-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Quality of representation </span></span>
<span id="cb374-8"><a href="multidimensional-data-in-r.html#cb374-8" aria-hidden="true" tabindex="-1"></a>res_var_env_prcomp<span class="sc">$</span>cos2           </span></code></pre></div>
<p>The individual (i.e., samples) associated results can be printed with the function <code>get_pca_ind()</code>.</p>
<div class="sourceCode" id="cb375"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb375-1"><a href="multidimensional-data-in-r.html#cb375-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Results for individuals</span></span>
<span id="cb375-2"><a href="multidimensional-data-in-r.html#cb375-2" aria-hidden="true" tabindex="-1"></a>res_ind_env_prcomp <span class="ot">&lt;-</span> factoextra<span class="sc">::</span><span class="fu">get_pca_ind</span>(env_prcomp)</span>
<span id="cb375-3"><a href="multidimensional-data-in-r.html#cb375-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Coordinates -- this is same as env_prcomp$x</span></span>
<span id="cb375-4"><a href="multidimensional-data-in-r.html#cb375-4" aria-hidden="true" tabindex="-1"></a>res_ind_env_prcomp<span class="sc">$</span>coord   </span>
<span id="cb375-5"><a href="multidimensional-data-in-r.html#cb375-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Contributions to the PCs       </span></span>
<span id="cb375-6"><a href="multidimensional-data-in-r.html#cb375-6" aria-hidden="true" tabindex="-1"></a>res_ind_env_prcomp<span class="sc">$</span>contrib</span>
<span id="cb375-7"><a href="multidimensional-data-in-r.html#cb375-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Quality of representation         </span></span>
<span id="cb375-8"><a href="multidimensional-data-in-r.html#cb375-8" aria-hidden="true" tabindex="-1"></a>res_ind_env_prcomp<span class="sc">$</span>cos2           </span></code></pre></div>
</div>
<div id="multidimensional-scaling" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> Multidimensional scaling</h2>
<p>We use multidimensional scaling helps to <strong>visualize similarities</strong> of individual cases or samples of a dataset. In this analysis, we reduce dimensions to be able to visualize or represent data points in a more friendly manner. We start with a set of samples or individuals measured on several numeric variables. After the analysis, we get a representation of the distances among these samples or individuals in a lower dimensional space (e.g., 2 dimensions), so we can infer the relative relationship (distance) between them. In other words, individuals or samples that are closer are more similar.</p>
<p>We will use our example `env_log_red, but we need to add names to the sites for visualization.</p>
<div class="sourceCode" id="cb376"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb376-1"><a href="multidimensional-data-in-r.html#cb376-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We can name our samples by assigned a unique row name.</span></span>
<span id="cb376-2"><a href="multidimensional-data-in-r.html#cb376-2" aria-hidden="true" tabindex="-1"></a>env_log_red_MDS <span class="ot">&lt;-</span> env_log_red</span>
<span id="cb376-3"><a href="multidimensional-data-in-r.html#cb376-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(env_log_red_MDS) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;site_&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(env_log_red_MDS))</span>
<span id="cb376-4"><a href="multidimensional-data-in-r.html#cb376-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(env_log_red_MDS)</span>
<span id="cb376-5"><a href="multidimensional-data-in-r.html#cb376-5" aria-hidden="true" tabindex="-1"></a><span class="co">#            dfs      alt      slo      flo       pH      har       pho      nit      oxy      bdo</span></span>
<span id="cb376-6"><a href="multidimensional-data-in-r.html#cb376-6" aria-hidden="true" tabindex="-1"></a><span class="co">#site_1 1.098612 6.839476 1.820671 4.430817 4.369448 3.806662 0.0000000 2.995732 4.804021 3.295837</span></span>
<span id="cb376-7"><a href="multidimensional-data-in-r.html#cb376-7" aria-hidden="true" tabindex="-1"></a><span class="co">#site_2 3.091042 6.837333 1.233726 4.605170 4.382027 3.688879 0.6931472 2.995732 4.634729 2.944439</span></span>
<span id="cb376-8"><a href="multidimensional-data-in-r.html#cb376-8" aria-hidden="true" tabindex="-1"></a><span class="co">#site_3 4.624973 6.817831 1.291434 5.192957 4.418841 3.951244 1.6094379 3.091042 4.653960 3.555348</span></span>
<span id="cb376-9"><a href="multidimensional-data-in-r.html#cb376-9" aria-hidden="true" tabindex="-1"></a><span class="co">#site_4 5.220356 6.749931 1.251905 5.533389 4.382027 4.276666 2.3025851 3.044522 4.700480 2.564949</span></span>
<span id="cb376-10"><a href="multidimensional-data-in-r.html#cb376-10" aria-hidden="true" tabindex="-1"></a><span class="co">#site_5 5.370638 6.744059 1.156252 5.575949 4.394449 4.430817 3.6375862 3.951244 4.382027 4.127134</span></span>
<span id="cb376-11"><a href="multidimensional-data-in-r.html#cb376-11" aria-hidden="true" tabindex="-1"></a><span class="co">#site_6 5.780744 6.740519 1.251905 5.655992 4.369448 4.094345 2.9957323 2.708050 4.624973 3.970292</span></span></code></pre></div>
<p>We can perform MDS using the function <code>dis()</code> and <code>cmdscale()</code>for metric or classical MDS with two dimensions <code>k = 2</code>.</p>
<div class="sourceCode" id="cb377"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb377-1"><a href="multidimensional-data-in-r.html#cb377-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Classical MDS: N rows (objects) x p columns (variables)</span></span>
<span id="cb377-2"><a href="multidimensional-data-in-r.html#cb377-2" aria-hidden="true" tabindex="-1"></a><span class="co"># get Euclidean distances between the rows</span></span>
<span id="cb377-3"><a href="multidimensional-data-in-r.html#cb377-3" aria-hidden="true" tabindex="-1"></a>env_log_red_MDS_d <span class="ot">&lt;-</span> <span class="fu">dist</span>(env_log_red_MDS) </span>
<span id="cb377-4"><a href="multidimensional-data-in-r.html#cb377-4" aria-hidden="true" tabindex="-1"></a><span class="co"># k is the number of dimensions</span></span>
<span id="cb377-5"><a href="multidimensional-data-in-r.html#cb377-5" aria-hidden="true" tabindex="-1"></a>env_log_red_MDS_fit <span class="ot">&lt;-</span> <span class="fu">cmdscale</span>(env_log_red_MDS_d,<span class="at">eig=</span><span class="cn">TRUE</span>, <span class="at">k=</span><span class="dv">2</span>) </span>
<span id="cb377-6"><a href="multidimensional-data-in-r.html#cb377-6" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(env_log_red_MDS_fit)</span>
<span id="cb377-7"><a href="multidimensional-data-in-r.html#cb377-7" aria-hidden="true" tabindex="-1"></a><span class="co">#List of 5</span></span>
<span id="cb377-8"><a href="multidimensional-data-in-r.html#cb377-8" aria-hidden="true" tabindex="-1"></a><span class="co"># $ points: num [1:30, 1:2] -7.21 -5.58 -3.84 -3.11 -1.83 ...</span></span>
<span id="cb377-9"><a href="multidimensional-data-in-r.html#cb377-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  ..- attr(*, &quot;dimnames&quot;)=List of 2</span></span>
<span id="cb377-10"><a href="multidimensional-data-in-r.html#cb377-10" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. ..$ : chr [1:30] &quot;site_1&quot; &quot;site_2&quot; &quot;site_3&quot; &quot;site_4&quot; ...</span></span>
<span id="cb377-11"><a href="multidimensional-data-in-r.html#cb377-11" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. ..$ : NULL</span></span>
<span id="cb377-12"><a href="multidimensional-data-in-r.html#cb377-12" aria-hidden="true" tabindex="-1"></a><span class="co">#$ eig   : num [1:30] 203.13 25.14 9.43 4.54 3.22 ...</span></span>
<span id="cb377-13"><a href="multidimensional-data-in-r.html#cb377-13" aria-hidden="true" tabindex="-1"></a><span class="co">#$ x     : NULL</span></span>
<span id="cb377-14"><a href="multidimensional-data-in-r.html#cb377-14" aria-hidden="true" tabindex="-1"></a><span class="co">#$ ac    : num 0</span></span>
<span id="cb377-15"><a href="multidimensional-data-in-r.html#cb377-15" aria-hidden="true" tabindex="-1"></a><span class="co">#$ GOF   : num [1:2] 0.92 0.92</span></span></code></pre></div>
<p>We can plot this samples using the x-y coordinates in <code>env_log_red_MDS_fit$points</code>.</p>
<div class="sourceCode" id="cb378"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb378-1"><a href="multidimensional-data-in-r.html#cb378-1" aria-hidden="true" tabindex="-1"></a><span class="do">## plot solution</span></span>
<span id="cb378-2"><a href="multidimensional-data-in-r.html#cb378-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> env_log_red_MDS_fit<span class="sc">$</span>points[,<span class="dv">1</span>]</span>
<span id="cb378-3"><a href="multidimensional-data-in-r.html#cb378-3" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> env_log_red_MDS_fit<span class="sc">$</span>points[,<span class="dv">2</span>]</span>
<span id="cb378-4"><a href="multidimensional-data-in-r.html#cb378-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(x, y, <span class="at">xlab=</span> <span class="st">&quot;Coordinate 1 - MDS&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;Coordinate 2 - MDS&quot;</span>, <span class="at">main=</span><span class="st">&quot;Metric MDS for env_log_red&quot;</span>, <span class="at">type=</span><span class="st">&quot;n&quot;</span>)</span>
<span id="cb378-5"><a href="multidimensional-data-in-r.html#cb378-5" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(x, y, <span class="at">labels =</span> <span class="fu">row.names</span>(env_log_red_MDS), <span class="at">cex=</span>.<span class="dv">7</span>) </span></code></pre></div>
<p><img src="figures/session_8_figs/7_MDS_classic.png" width="100%" /></p>
<p>We use nonmetric multidimensional scaling to visualizing the function <code>isoMDS()</code> of the R-package <a href="https://cran.r-project.org/web/packages/MASS/index.html" target="_blank"><strong>MASS</strong></a>.</p>
<div class="sourceCode" id="cb379"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb379-1"><a href="multidimensional-data-in-r.html#cb379-1" aria-hidden="true" tabindex="-1"></a><span class="do">## For nonmetric MDS, we load MASS (we installed this package before)</span></span>
<span id="cb379-2"><a href="multidimensional-data-in-r.html#cb379-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb379-3"><a href="multidimensional-data-in-r.html#cb379-3" aria-hidden="true" tabindex="-1"></a>env_log_red_MDS_d <span class="ot">&lt;-</span> <span class="fu">dist</span>(env_log_red_MDS) </span>
<span id="cb379-4"><a href="multidimensional-data-in-r.html#cb379-4" aria-hidden="true" tabindex="-1"></a>env_log_red_NMDS <span class="ot">&lt;-</span> MASS<span class="sc">::</span><span class="fu">isoMDS</span>(env_log_red_MDS_d, <span class="at">k=</span><span class="dv">2</span>)</span>
<span id="cb379-5"><a href="multidimensional-data-in-r.html#cb379-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(env_log_red_NMDS)</span>
<span id="cb379-6"><a href="multidimensional-data-in-r.html#cb379-6" aria-hidden="true" tabindex="-1"></a><span class="co">#List of 2</span></span>
<span id="cb379-7"><a href="multidimensional-data-in-r.html#cb379-7" aria-hidden="true" tabindex="-1"></a><span class="co"># $ points: num [1:30, 1:2] -7.64 -5.6 -3.74 -3.08 -1.72 ...</span></span>
<span id="cb379-8"><a href="multidimensional-data-in-r.html#cb379-8" aria-hidden="true" tabindex="-1"></a><span class="co">#  ..- attr(*, &quot;dimnames&quot;)=List of 2</span></span>
<span id="cb379-9"><a href="multidimensional-data-in-r.html#cb379-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. ..$ : chr [1:30] &quot;site_1&quot; &quot;site_2&quot; &quot;site_3&quot; &quot;site_4&quot; ...</span></span>
<span id="cb379-10"><a href="multidimensional-data-in-r.html#cb379-10" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. ..$ : NULL</span></span>
<span id="cb379-11"><a href="multidimensional-data-in-r.html#cb379-11" aria-hidden="true" tabindex="-1"></a><span class="co"># $ stress: num 2.49</span></span>
<span id="cb379-12"><a href="multidimensional-data-in-r.html#cb379-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb379-13"><a href="multidimensional-data-in-r.html#cb379-13" aria-hidden="true" tabindex="-1"></a><span class="do">## plot solution NMDS solution</span></span>
<span id="cb379-14"><a href="multidimensional-data-in-r.html#cb379-14" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> env_log_red_NMDS<span class="sc">$</span>points[,<span class="dv">1</span>]</span>
<span id="cb379-15"><a href="multidimensional-data-in-r.html#cb379-15" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> env_log_red_NMDS<span class="sc">$</span>points[,<span class="dv">2</span>]</span>
<span id="cb379-16"><a href="multidimensional-data-in-r.html#cb379-16" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(x, y, <span class="at">xlab=</span> <span class="st">&quot;Coordinate 1 - NMDS&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;Coordinate 2 - NMDS&quot;</span>, <span class="at">main=</span><span class="st">&quot;Nonmetric MDS for env_log_red&quot;</span>, <span class="at">type=</span><span class="st">&quot;n&quot;</span>)</span>
<span id="cb379-17"><a href="multidimensional-data-in-r.html#cb379-17" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(x, y, <span class="at">labels =</span> <span class="fu">row.names</span>(env_log_red_MDS), <span class="at">cex=</span>.<span class="dv">7</span>) </span></code></pre></div>
<p><img src="figures/session_8_figs/8_NMDS_plot.png" width="100%" /></p>
</div>
<div id="cluster-analysis-dfa" class="section level2" number="8.3">
<h2><span class="header-section-number">8.3</span> Cluster analysis: DFA</h2>
<p>Many biological datasets might have an underlying structure were groups or clusters that can be outcome of a treatment. Finding those set of objects that are more similar might help to identify groups or clusters that can used to classification of other samples.</p>
<p>We use a classic example known as the <a href="https://en.wikipedia.org/wiki/Iris_flower_data_set" target="_blank"><strong>Fisher’s or Anderson’s Iris data set</strong></a>, which is a dataset of multiple measurements introduced by Ronald Fisher to illustrate taxonomic problems as an example of linear discriminant analysis. This dataset consists of 50 samples of three species of flowers: <em>Iris setosa</em>, <em>Iris virginica</em> and <em>Iris versicolor</em>. The dataset includes four features of the flower measured in centimeters.
This dataset can be loaded from the R-package <a href="https://stat.ethz.ch/R-manual/R-devel/library/datasets/html/00Index.html" target="_blank"><strong>datasets</strong></a>.</p>
<div class="sourceCode" id="cb380"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb380-1"><a href="multidimensional-data-in-r.html#cb380-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;datasets&quot;</span>)</span>
<span id="cb380-2"><a href="multidimensional-data-in-r.html#cb380-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(datasets)</span>
<span id="cb380-3"><a href="multidimensional-data-in-r.html#cb380-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(iris)</span>
<span id="cb380-4"><a href="multidimensional-data-in-r.html#cb380-4" aria-hidden="true" tabindex="-1"></a>iris_dataset <span class="ot">&lt;-</span> iris</span>
<span id="cb380-5"><a href="multidimensional-data-in-r.html#cb380-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(iris_dataset)</span>
<span id="cb380-6"><a href="multidimensional-data-in-r.html#cb380-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  Sepal.Length    Sepal.Width     Petal.Length    Petal.Width          Species  </span></span>
<span id="cb380-7"><a href="multidimensional-data-in-r.html#cb380-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Min.   :4.300   Min.   :2.000   Min.   :1.000   Min.   :0.100   setosa    :50  </span></span>
<span id="cb380-8"><a href="multidimensional-data-in-r.html#cb380-8" aria-hidden="true" tabindex="-1"></a><span class="co">#1st Qu.:5.100   1st Qu.:2.800   1st Qu.:1.600   1st Qu.:0.300   versicolor:50  </span></span>
<span id="cb380-9"><a href="multidimensional-data-in-r.html#cb380-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Median :5.800   Median :3.000   Median :4.350   Median :1.300   virginica :50  </span></span>
<span id="cb380-10"><a href="multidimensional-data-in-r.html#cb380-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Mean   :5.843   Mean   :3.057   Mean   :3.758   Mean   :1.199                  </span></span>
<span id="cb380-11"><a href="multidimensional-data-in-r.html#cb380-11" aria-hidden="true" tabindex="-1"></a><span class="co">#3rd Qu.:6.400   3rd Qu.:3.300   3rd Qu.:5.100   3rd Qu.:1.800                  </span></span>
<span id="cb380-12"><a href="multidimensional-data-in-r.html#cb380-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Max.   :7.900   Max.   :4.400   Max.   :6.900   Max.   :2.500                  </span></span>
<span id="cb380-13"><a href="multidimensional-data-in-r.html#cb380-13" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(iris_dataset)</span>
<span id="cb380-14"><a href="multidimensional-data-in-r.html#cb380-14" aria-hidden="true" tabindex="-1"></a><span class="co">#  Sepal.Length Sepal.Width Petal.Length Petal.Width Species</span></span>
<span id="cb380-15"><a href="multidimensional-data-in-r.html#cb380-15" aria-hidden="true" tabindex="-1"></a><span class="co">#1          5.1         3.5          1.4         0.2  setosa</span></span>
<span id="cb380-16"><a href="multidimensional-data-in-r.html#cb380-16" aria-hidden="true" tabindex="-1"></a><span class="co">#2          4.9         3.0          1.4         0.2  setosa</span></span>
<span id="cb380-17"><a href="multidimensional-data-in-r.html#cb380-17" aria-hidden="true" tabindex="-1"></a><span class="co">#3          4.7         3.2          1.3         0.2  setosa</span></span>
<span id="cb380-18"><a href="multidimensional-data-in-r.html#cb380-18" aria-hidden="true" tabindex="-1"></a><span class="co">#4          4.6         3.1          1.5         0.2  setosa</span></span>
<span id="cb380-19"><a href="multidimensional-data-in-r.html#cb380-19" aria-hidden="true" tabindex="-1"></a><span class="co">#5          5.0         3.6          1.4         0.2  setosa</span></span>
<span id="cb380-20"><a href="multidimensional-data-in-r.html#cb380-20" aria-hidden="true" tabindex="-1"></a><span class="co">#6          5.4         3.9          1.7         0.4  setosa</span></span></code></pre></div>
<p>We can perform <a href="https://en.wikipedia.org/wiki/Linear_discriminant_analysis" target="_blank"><strong>discriminant function analysis (DFA) or Linear Discriminant Analysis (LDA)</strong></a> to find a function (e.g., linear classifier) to separate samples in to two or more classes. For this analysis, we will the function <code>lda()</code> from the package <strong>MASS</strong>.</p>
<div class="sourceCode" id="cb381"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb381-1"><a href="multidimensional-data-in-r.html#cb381-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we have used the packages MASS and ggplot2.</span></span>
<span id="cb381-2"><a href="multidimensional-data-in-r.html#cb381-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb381-3"><a href="multidimensional-data-in-r.html#cb381-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb381-4"><a href="multidimensional-data-in-r.html#cb381-4" aria-hidden="true" tabindex="-1"></a><span class="do">## we get the names of predictors (i.e., measurements) and target group</span></span>
<span id="cb381-5"><a href="multidimensional-data-in-r.html#cb381-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(iris_dataset)</span>
<span id="cb381-6"><a href="multidimensional-data-in-r.html#cb381-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  150 obs. of  5 variables:</span></span>
<span id="cb381-7"><a href="multidimensional-data-in-r.html#cb381-7" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Sepal.Length: num  5.1 4.9 4.7 4.6 5 5.4 4.6 5 4.4 4.9 ...</span></span>
<span id="cb381-8"><a href="multidimensional-data-in-r.html#cb381-8" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Sepal.Width : num  3.5 3 3.2 3.1 3.6 3.9 3.4 3.4 2.9 3.1 ...</span></span>
<span id="cb381-9"><a href="multidimensional-data-in-r.html#cb381-9" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Petal.Length: num  1.4 1.4 1.3 1.5 1.4 1.7 1.4 1.5 1.4 1.5 ...</span></span>
<span id="cb381-10"><a href="multidimensional-data-in-r.html#cb381-10" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Petal.Width : num  0.2 0.2 0.2 0.2 0.2 0.4 0.3 0.2 0.2 0.1 ...</span></span>
<span id="cb381-11"><a href="multidimensional-data-in-r.html#cb381-11" aria-hidden="true" tabindex="-1"></a><span class="co">#$ Species     : Factor w/ 3 levels &quot;setosa&quot;,&quot;versicolor&quot;,..: 1 1 1 1 1 1 1 1 1 1 ...</span></span></code></pre></div>
<p>We see that the grouping variable is <code>Species</code> with the three species and four continuous measurements associated with each individual in the dataset. In this case, we input a formula <code>Species ~ Sepal.Length + Sepal.Width + Petal.Length + Petal.Width</code>.</p>
<div class="sourceCode" id="cb382"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb382-1"><a href="multidimensional-data-in-r.html#cb382-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we fit the lda function</span></span>
<span id="cb382-2"><a href="multidimensional-data-in-r.html#cb382-2" aria-hidden="true" tabindex="-1"></a>iris_lda_model <span class="ot">&lt;-</span> <span class="fu">lda</span>(<span class="at">formula =</span> Species <span class="sc">~</span> Sepal.Length <span class="sc">+</span> Sepal.Width <span class="sc">+</span> Petal.Length <span class="sc">+</span> Petal.Width, <span class="at">data =</span> iris_dataset)</span>
<span id="cb382-3"><a href="multidimensional-data-in-r.html#cb382-3" aria-hidden="true" tabindex="-1"></a>iris_lda_model</span>
<span id="cb382-4"><a href="multidimensional-data-in-r.html#cb382-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Call:</span></span>
<span id="cb382-5"><a href="multidimensional-data-in-r.html#cb382-5" aria-hidden="true" tabindex="-1"></a><span class="co">#lda(Species ~ Sepal.Length + Sepal.Width + Petal.Length + Petal.Width, </span></span>
<span id="cb382-6"><a href="multidimensional-data-in-r.html#cb382-6" aria-hidden="true" tabindex="-1"></a><span class="co">#    data = iris_dataset)</span></span>
<span id="cb382-7"><a href="multidimensional-data-in-r.html#cb382-7" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb382-8"><a href="multidimensional-data-in-r.html#cb382-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Prior probabilities of groups:</span></span>
<span id="cb382-9"><a href="multidimensional-data-in-r.html#cb382-9" aria-hidden="true" tabindex="-1"></a><span class="co">#    setosa versicolor  virginica </span></span>
<span id="cb382-10"><a href="multidimensional-data-in-r.html#cb382-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 0.3333333  0.3333333  0.3333333 </span></span>
<span id="cb382-11"><a href="multidimensional-data-in-r.html#cb382-11" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb382-12"><a href="multidimensional-data-in-r.html#cb382-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Group means:</span></span>
<span id="cb382-13"><a href="multidimensional-data-in-r.html#cb382-13" aria-hidden="true" tabindex="-1"></a><span class="co">#           Sepal.Length Sepal.Width Petal.Length Petal.Width</span></span>
<span id="cb382-14"><a href="multidimensional-data-in-r.html#cb382-14" aria-hidden="true" tabindex="-1"></a><span class="co">#setosa            5.006       3.428        1.462       0.246</span></span>
<span id="cb382-15"><a href="multidimensional-data-in-r.html#cb382-15" aria-hidden="true" tabindex="-1"></a><span class="co">#versicolor        5.936       2.770        4.260       1.326</span></span>
<span id="cb382-16"><a href="multidimensional-data-in-r.html#cb382-16" aria-hidden="true" tabindex="-1"></a><span class="co">#virginica         6.588       2.974        5.552       2.026</span></span>
<span id="cb382-17"><a href="multidimensional-data-in-r.html#cb382-17" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb382-18"><a href="multidimensional-data-in-r.html#cb382-18" aria-hidden="true" tabindex="-1"></a><span class="co">#Coefficients of linear discriminants:</span></span>
<span id="cb382-19"><a href="multidimensional-data-in-r.html#cb382-19" aria-hidden="true" tabindex="-1"></a><span class="co">#                    LD1         LD2</span></span>
<span id="cb382-20"><a href="multidimensional-data-in-r.html#cb382-20" aria-hidden="true" tabindex="-1"></a><span class="co">#Sepal.Length  0.8293776  0.02410215</span></span>
<span id="cb382-21"><a href="multidimensional-data-in-r.html#cb382-21" aria-hidden="true" tabindex="-1"></a><span class="co">#Sepal.Width   1.5344731  2.16452123</span></span>
<span id="cb382-22"><a href="multidimensional-data-in-r.html#cb382-22" aria-hidden="true" tabindex="-1"></a><span class="co">#Petal.Length -2.2012117 -0.93192121</span></span>
<span id="cb382-23"><a href="multidimensional-data-in-r.html#cb382-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Petal.Width  -2.8104603  2.83918785</span></span>
<span id="cb382-24"><a href="multidimensional-data-in-r.html#cb382-24" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb382-25"><a href="multidimensional-data-in-r.html#cb382-25" aria-hidden="true" tabindex="-1"></a><span class="co">#Proportion of trace:</span></span>
<span id="cb382-26"><a href="multidimensional-data-in-r.html#cb382-26" aria-hidden="true" tabindex="-1"></a><span class="co">#   LD1    LD2 </span></span>
<span id="cb382-27"><a href="multidimensional-data-in-r.html#cb382-27" aria-hidden="true" tabindex="-1"></a><span class="co">#0.9912 0.0088 </span></span></code></pre></div>
<p>We can now evaluate how good LDA can discriminate our dataset. We need to install two R-packages <a href="http://topepo.github.io/caret/index.html" target="_blank"><strong>caret</strong></a> and <a href="https://cran.r-project.org/web/packages/e1071/index.html" target="_blank"><strong>e1071</strong></a>.</p>
<div class="sourceCode" id="cb383"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb383-1"><a href="multidimensional-data-in-r.html#cb383-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we install these packages</span></span>
<span id="cb383-2"><a href="multidimensional-data-in-r.html#cb383-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;caret&quot;</span>)</span>
<span id="cb383-3"><a href="multidimensional-data-in-r.html#cb383-3" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;e1071&quot;</span>)</span>
<span id="cb383-4"><a href="multidimensional-data-in-r.html#cb383-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span>
<span id="cb383-5"><a href="multidimensional-data-in-r.html#cb383-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(e1071)</span></code></pre></div>
<p>We need to fit predictive model over our data using the function <code>train()</code> and obtain the confusion matrix <code>confusionMatrix()</code> that is table that determines how good was the classification using LDA.</p>
<div class="sourceCode" id="cb384"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb384-1"><a href="multidimensional-data-in-r.html#cb384-1" aria-hidden="true" tabindex="-1"></a>iris_lda_predict <span class="ot">&lt;-</span> caret<span class="sc">::</span><span class="fu">train</span>(Species <span class="sc">~</span> Sepal.Length <span class="sc">+</span> Sepal.Width <span class="sc">+</span> Petal.Length <span class="sc">+</span> Petal.Width, <span class="at">method =</span> <span class="st">&quot;lda&quot;</span>, <span class="at">data =</span> iris_dataset)</span>
<span id="cb384-2"><a href="multidimensional-data-in-r.html#cb384-2" aria-hidden="true" tabindex="-1"></a>iris_lda_predict</span>
<span id="cb384-3"><a href="multidimensional-data-in-r.html#cb384-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Linear Discriminant Analysis </span></span>
<span id="cb384-4"><a href="multidimensional-data-in-r.html#cb384-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb384-5"><a href="multidimensional-data-in-r.html#cb384-5" aria-hidden="true" tabindex="-1"></a><span class="co">#150 samples</span></span>
<span id="cb384-6"><a href="multidimensional-data-in-r.html#cb384-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  4 predictor</span></span>
<span id="cb384-7"><a href="multidimensional-data-in-r.html#cb384-7" aria-hidden="true" tabindex="-1"></a><span class="co">#  3 classes: &#39;setosa&#39;, &#39;versicolor&#39;, &#39;virginica&#39; </span></span>
<span id="cb384-8"><a href="multidimensional-data-in-r.html#cb384-8" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb384-9"><a href="multidimensional-data-in-r.html#cb384-9" aria-hidden="true" tabindex="-1"></a><span class="co">#No pre-processing</span></span>
<span id="cb384-10"><a href="multidimensional-data-in-r.html#cb384-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Resampling: Bootstrapped (25 reps) </span></span>
<span id="cb384-11"><a href="multidimensional-data-in-r.html#cb384-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Summary of sample sizes: 150, 150, 150, 150, 150, 150, ... </span></span>
<span id="cb384-12"><a href="multidimensional-data-in-r.html#cb384-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Resampling results:</span></span>
<span id="cb384-13"><a href="multidimensional-data-in-r.html#cb384-13" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb384-14"><a href="multidimensional-data-in-r.html#cb384-14" aria-hidden="true" tabindex="-1"></a><span class="co">#  Accuracy   Kappa    </span></span>
<span id="cb384-15"><a href="multidimensional-data-in-r.html#cb384-15" aria-hidden="true" tabindex="-1"></a><span class="co">#  0.9697408  0.9542207</span></span>
<span id="cb384-16"><a href="multidimensional-data-in-r.html#cb384-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb384-17"><a href="multidimensional-data-in-r.html#cb384-17" aria-hidden="true" tabindex="-1"></a>iris_lda_confusion_matrix <span class="ot">&lt;-</span> <span class="fu">confusionMatrix</span>(iris_dataset<span class="sc">$</span>Species, <span class="fu">predict</span>(iris_lda_predict, iris_dataset))</span>
<span id="cb384-18"><a href="multidimensional-data-in-r.html#cb384-18" aria-hidden="true" tabindex="-1"></a>iris_lda_confusion_matrix</span>
<span id="cb384-19"><a href="multidimensional-data-in-r.html#cb384-19" aria-hidden="true" tabindex="-1"></a><span class="co">#Confusion Matrix and Statistics</span></span>
<span id="cb384-20"><a href="multidimensional-data-in-r.html#cb384-20" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb384-21"><a href="multidimensional-data-in-r.html#cb384-21" aria-hidden="true" tabindex="-1"></a><span class="co">#            Reference</span></span>
<span id="cb384-22"><a href="multidimensional-data-in-r.html#cb384-22" aria-hidden="true" tabindex="-1"></a><span class="co">#Prediction   setosa versicolor virginica</span></span>
<span id="cb384-23"><a href="multidimensional-data-in-r.html#cb384-23" aria-hidden="true" tabindex="-1"></a><span class="co">#  setosa         50          0         0</span></span>
<span id="cb384-24"><a href="multidimensional-data-in-r.html#cb384-24" aria-hidden="true" tabindex="-1"></a><span class="co">#  versicolor      0         48         2</span></span>
<span id="cb384-25"><a href="multidimensional-data-in-r.html#cb384-25" aria-hidden="true" tabindex="-1"></a><span class="co">#  virginica       0          1        49</span></span>
<span id="cb384-26"><a href="multidimensional-data-in-r.html#cb384-26" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb384-27"><a href="multidimensional-data-in-r.html#cb384-27" aria-hidden="true" tabindex="-1"></a><span class="co">#Overall Statistics</span></span>
<span id="cb384-28"><a href="multidimensional-data-in-r.html#cb384-28" aria-hidden="true" tabindex="-1"></a><span class="co">#                                          </span></span>
<span id="cb384-29"><a href="multidimensional-data-in-r.html#cb384-29" aria-hidden="true" tabindex="-1"></a><span class="co">#               Accuracy : 0.98            </span></span>
<span id="cb384-30"><a href="multidimensional-data-in-r.html#cb384-30" aria-hidden="true" tabindex="-1"></a><span class="co">#                 95% CI : (0.9427, 0.9959)</span></span>
<span id="cb384-31"><a href="multidimensional-data-in-r.html#cb384-31" aria-hidden="true" tabindex="-1"></a><span class="co">#    No Information Rate : 0.34            </span></span>
<span id="cb384-32"><a href="multidimensional-data-in-r.html#cb384-32" aria-hidden="true" tabindex="-1"></a><span class="co">#    P-Value [Acc &gt; NIR] : &lt; 2.2e-16       </span></span>
<span id="cb384-33"><a href="multidimensional-data-in-r.html#cb384-33" aria-hidden="true" tabindex="-1"></a><span class="co">#                                          </span></span>
<span id="cb384-34"><a href="multidimensional-data-in-r.html#cb384-34" aria-hidden="true" tabindex="-1"></a><span class="co">#                  Kappa : 0.97            </span></span>
<span id="cb384-35"><a href="multidimensional-data-in-r.html#cb384-35" aria-hidden="true" tabindex="-1"></a><span class="co">#                                          </span></span>
<span id="cb384-36"><a href="multidimensional-data-in-r.html#cb384-36" aria-hidden="true" tabindex="-1"></a><span class="co"># Mcnemar&#39;s Test P-Value : NA              </span></span>
<span id="cb384-37"><a href="multidimensional-data-in-r.html#cb384-37" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb384-38"><a href="multidimensional-data-in-r.html#cb384-38" aria-hidden="true" tabindex="-1"></a><span class="co">#Statistics by Class:</span></span>
<span id="cb384-39"><a href="multidimensional-data-in-r.html#cb384-39" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb384-40"><a href="multidimensional-data-in-r.html#cb384-40" aria-hidden="true" tabindex="-1"></a><span class="co">#                     Class: setosa Class: versicolor Class: virginica</span></span>
<span id="cb384-41"><a href="multidimensional-data-in-r.html#cb384-41" aria-hidden="true" tabindex="-1"></a><span class="co">#Sensitivity                 1.0000            0.9796           0.9608</span></span>
<span id="cb384-42"><a href="multidimensional-data-in-r.html#cb384-42" aria-hidden="true" tabindex="-1"></a><span class="co">#Specificity                 1.0000            0.9802           0.9899</span></span>
<span id="cb384-43"><a href="multidimensional-data-in-r.html#cb384-43" aria-hidden="true" tabindex="-1"></a><span class="co">#Pos Pred Value              1.0000            0.9600           0.9800</span></span>
<span id="cb384-44"><a href="multidimensional-data-in-r.html#cb384-44" aria-hidden="true" tabindex="-1"></a><span class="co">#Neg Pred Value              1.0000            0.9900           0.9800</span></span>
<span id="cb384-45"><a href="multidimensional-data-in-r.html#cb384-45" aria-hidden="true" tabindex="-1"></a><span class="co">#Prevalence                  0.3333            0.3267           0.3400</span></span>
<span id="cb384-46"><a href="multidimensional-data-in-r.html#cb384-46" aria-hidden="true" tabindex="-1"></a><span class="co">#Detection Rate              0.3333            0.3200           0.3267</span></span>
<span id="cb384-47"><a href="multidimensional-data-in-r.html#cb384-47" aria-hidden="true" tabindex="-1"></a><span class="co">#Detection Prevalence        0.3333            0.3333           0.3333</span></span>
<span id="cb384-48"><a href="multidimensional-data-in-r.html#cb384-48" aria-hidden="true" tabindex="-1"></a><span class="co">#Balanced Accuracy           1.0000            0.9799           0.9753</span></span></code></pre></div>
<p>We can see the confusion matrix and values that not consider between Reference and Prediction are discrepancies between what we assigned and what LDA determined as a part of a group. We can add such predictions to the <code>iris_dataset</code> using the function <code>predict()</code>.</p>
<div class="sourceCode" id="cb385"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb385-1"><a href="multidimensional-data-in-r.html#cb385-1" aria-hidden="true" tabindex="-1"></a>iris_lda_predicton <span class="ot">&lt;-</span> <span class="fu">predict</span> (iris_lda_predict, iris_dataset)</span>
<span id="cb385-2"><a href="multidimensional-data-in-r.html#cb385-2" aria-hidden="true" tabindex="-1"></a>iris_dataset<span class="sc">$</span>lda_species_prediction <span class="ot">&lt;-</span> iris_lda_predicton</span>
<span id="cb385-3"><a href="multidimensional-data-in-r.html#cb385-3" aria-hidden="true" tabindex="-1"></a>iris_dataset </span>
<span id="cb385-4"><a href="multidimensional-data-in-r.html#cb385-4" aria-hidden="true" tabindex="-1"></a><span class="co">#    Sepal.Length Sepal.Width Petal.Length Petal.Width    Species lda_species_prediction</span></span>
<span id="cb385-5"><a href="multidimensional-data-in-r.html#cb385-5" aria-hidden="true" tabindex="-1"></a><span class="co">#1            5.1         3.5          1.4         0.2     setosa                 setosa</span></span>
<span id="cb385-6"><a href="multidimensional-data-in-r.html#cb385-6" aria-hidden="true" tabindex="-1"></a><span class="co">#2            4.9         3.0          1.4         0.2     setosa                 setosa</span></span>
<span id="cb385-7"><a href="multidimensional-data-in-r.html#cb385-7" aria-hidden="true" tabindex="-1"></a><span class="co">#3            4.7         3.2          1.3         0.2     setosa                 setosa</span></span>
<span id="cb385-8"><a href="multidimensional-data-in-r.html#cb385-8" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span>
<span id="cb385-9"><a href="multidimensional-data-in-r.html#cb385-9" aria-hidden="true" tabindex="-1"></a><span class="co">#69           6.2         2.2          4.5         1.5 versicolor             versicolor</span></span>
<span id="cb385-10"><a href="multidimensional-data-in-r.html#cb385-10" aria-hidden="true" tabindex="-1"></a><span class="co">#70           5.6         2.5          3.9         1.1 versicolor             versicolor</span></span>
<span id="cb385-11"><a href="multidimensional-data-in-r.html#cb385-11" aria-hidden="true" tabindex="-1"></a><span class="co">#71           5.9         3.2          4.8         1.8 versicolor              virginica</span></span>
<span id="cb385-12"><a href="multidimensional-data-in-r.html#cb385-12" aria-hidden="true" tabindex="-1"></a><span class="co">#72           6.1         2.8          4.0         1.3 versicolor             versicolor</span></span>
<span id="cb385-13"><a href="multidimensional-data-in-r.html#cb385-13" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span>
<span id="cb385-14"><a href="multidimensional-data-in-r.html#cb385-14" aria-hidden="true" tabindex="-1"></a><span class="co">#83           5.8         2.7          3.9         1.2 versicolor             versicolor</span></span>
<span id="cb385-15"><a href="multidimensional-data-in-r.html#cb385-15" aria-hidden="true" tabindex="-1"></a><span class="co">#84           6.0         2.7          5.1         1.6 versicolor              virginica</span></span>
<span id="cb385-16"><a href="multidimensional-data-in-r.html#cb385-16" aria-hidden="true" tabindex="-1"></a><span class="co">#85           5.4         3.0          4.5         1.5 versicolor             versicolor</span></span>
<span id="cb385-17"><a href="multidimensional-data-in-r.html#cb385-17" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span>
<span id="cb385-18"><a href="multidimensional-data-in-r.html#cb385-18" aria-hidden="true" tabindex="-1"></a><span class="co">#132          7.9         3.8          6.4         2.0  virginica              virginica</span></span>
<span id="cb385-19"><a href="multidimensional-data-in-r.html#cb385-19" aria-hidden="true" tabindex="-1"></a><span class="co">#133          6.4         2.8          5.6         2.2  virginica              virginica</span></span>
<span id="cb385-20"><a href="multidimensional-data-in-r.html#cb385-20" aria-hidden="true" tabindex="-1"></a><span class="co">#134          6.3         2.8          5.1         1.5  virginica             versicolor</span></span>
<span id="cb385-21"><a href="multidimensional-data-in-r.html#cb385-21" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span></code></pre></div>
<p>You can see above that discrepancies in the classification assigned using LDA, these discrepancies should agree with the confusion table.</p>
<div class="sourceCode" id="cb386"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb386-1"><a href="multidimensional-data-in-r.html#cb386-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we can add the LD coordinates</span></span>
<span id="cb386-2"><a href="multidimensional-data-in-r.html#cb386-2" aria-hidden="true" tabindex="-1"></a>iris_dataset <span class="ot">&lt;-</span> <span class="fu">cbind</span>(iris_dataset, <span class="fu">predict</span>(iris_lda_model)<span class="sc">$</span>x)</span>
<span id="cb386-3"><a href="multidimensional-data-in-r.html#cb386-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(iris_dataset)</span>
<span id="cb386-4"><a href="multidimensional-data-in-r.html#cb386-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  Sepal.Length Sepal.Width Petal.Length Petal.Width Species lda_species_prediction      LD1        LD2</span></span>
<span id="cb386-5"><a href="multidimensional-data-in-r.html#cb386-5" aria-hidden="true" tabindex="-1"></a><span class="co">#1          5.1         3.5          1.4         0.2  setosa                 setosa 8.061800  0.3004206</span></span>
<span id="cb386-6"><a href="multidimensional-data-in-r.html#cb386-6" aria-hidden="true" tabindex="-1"></a><span class="co">#2          4.9         3.0          1.4         0.2  setosa                 setosa 7.128688 -0.7866604</span></span>
<span id="cb386-7"><a href="multidimensional-data-in-r.html#cb386-7" aria-hidden="true" tabindex="-1"></a><span class="co">#3          4.7         3.2          1.3         0.2  setosa                 setosa 7.489828 -0.2653845</span></span>
<span id="cb386-8"><a href="multidimensional-data-in-r.html#cb386-8" aria-hidden="true" tabindex="-1"></a><span class="co">#4          4.6         3.1          1.5         0.2  setosa                 setosa 6.813201 -0.6706311</span></span>
<span id="cb386-9"><a href="multidimensional-data-in-r.html#cb386-9" aria-hidden="true" tabindex="-1"></a><span class="co">#5          5.0         3.6          1.4         0.2  setosa                 setosa 8.132309  0.5144625</span></span>
<span id="cb386-10"><a href="multidimensional-data-in-r.html#cb386-10" aria-hidden="true" tabindex="-1"></a><span class="co">#6          5.4         3.9          1.7         0.4  setosa                 setosa 7.701947  1.4617210</span></span>
<span id="cb386-11"><a href="multidimensional-data-in-r.html#cb386-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb386-12"><a href="multidimensional-data-in-r.html#cb386-12" aria-hidden="true" tabindex="-1"></a><span class="do">## we subset the iris_dataset to the original classification</span></span>
<span id="cb386-13"><a href="multidimensional-data-in-r.html#cb386-13" aria-hidden="true" tabindex="-1"></a>iris_dataset_original <span class="ot">&lt;-</span> <span class="fu">subset</span>(iris_dataset, <span class="at">select =</span> <span class="fu">c</span>(Species, LD1,LD2))</span>
<span id="cb386-14"><a href="multidimensional-data-in-r.html#cb386-14" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(iris_dataset_original)</span>
<span id="cb386-15"><a href="multidimensional-data-in-r.html#cb386-15" aria-hidden="true" tabindex="-1"></a><span class="co">#  Species      LD1        LD2</span></span>
<span id="cb386-16"><a href="multidimensional-data-in-r.html#cb386-16" aria-hidden="true" tabindex="-1"></a><span class="co">#1  setosa 8.061800  0.3004206</span></span>
<span id="cb386-17"><a href="multidimensional-data-in-r.html#cb386-17" aria-hidden="true" tabindex="-1"></a><span class="co">#2  setosa 7.128688 -0.7866604</span></span>
<span id="cb386-18"><a href="multidimensional-data-in-r.html#cb386-18" aria-hidden="true" tabindex="-1"></a><span class="co">#3  setosa 7.489828 -0.2653845</span></span>
<span id="cb386-19"><a href="multidimensional-data-in-r.html#cb386-19" aria-hidden="true" tabindex="-1"></a><span class="co">#4  setosa 6.813201 -0.6706311</span></span>
<span id="cb386-20"><a href="multidimensional-data-in-r.html#cb386-20" aria-hidden="true" tabindex="-1"></a><span class="co">#5  setosa 8.132309  0.5144625</span></span>
<span id="cb386-21"><a href="multidimensional-data-in-r.html#cb386-21" aria-hidden="true" tabindex="-1"></a><span class="co">#6  setosa 7.701947  1.4617210</span></span>
<span id="cb386-22"><a href="multidimensional-data-in-r.html#cb386-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb386-23"><a href="multidimensional-data-in-r.html#cb386-23" aria-hidden="true" tabindex="-1"></a><span class="do">## show original groups</span></span>
<span id="cb386-24"><a href="multidimensional-data-in-r.html#cb386-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb386-25"><a href="multidimensional-data-in-r.html#cb386-25" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(iris_dataset, <span class="fu">aes</span>(LD1, LD2)) <span class="sc">+</span></span>
<span id="cb386-26"><a href="multidimensional-data-in-r.html#cb386-26" aria-hidden="true" tabindex="-1"></a>       <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> Species)) <span class="sc">+</span></span>
<span id="cb386-27"><a href="multidimensional-data-in-r.html#cb386-27" aria-hidden="true" tabindex="-1"></a>           <span class="fu">ggtitle</span>(<span class="st">&quot;Iris Original Groups&quot;</span>)</span>
<span id="cb386-28"><a href="multidimensional-data-in-r.html#cb386-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb386-29"><a href="multidimensional-data-in-r.html#cb386-29" aria-hidden="true" tabindex="-1"></a><span class="do">## show LDA predicted groups</span></span>
<span id="cb386-30"><a href="multidimensional-data-in-r.html#cb386-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb386-31"><a href="multidimensional-data-in-r.html#cb386-31" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(iris_dataset, <span class="fu">aes</span>(LD1, LD2)) <span class="sc">+</span></span>
<span id="cb386-32"><a href="multidimensional-data-in-r.html#cb386-32" aria-hidden="true" tabindex="-1"></a>       <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> lda_species_prediction)) <span class="sc">+</span></span>
<span id="cb386-33"><a href="multidimensional-data-in-r.html#cb386-33" aria-hidden="true" tabindex="-1"></a>           <span class="fu">ggtitle</span>(<span class="st">&quot;Iris LDA Predicted Groups&quot;</span>)</span></code></pre></div>
<p><strong>NOTE</strong> sometimes the <code>ggplot()</code> might need to run twice without closing the plot window to produce a correct plot</p>
<p><img src="figures/session_8_figs/9_Iris_original_groups.png" width="100%" /></p>
<p><img src="figures/session_8_figs/10_Iris_LDA_groups.png" width="100%" /></p>
<p>Can you find the changes on the classification between original and LDA predicted?</p>
</div>
<div id="cluster-analysis-k-means-unsupervised" class="section level2" number="8.4">
<h2><span class="header-section-number">8.4</span> Cluster analysis: <em>k</em>-means (unsupervised)</h2>
<p>We can use our own data without defining groups and ask if an algorithm used to find homogeneous subgroups in our data. One of such approaches is k-means clustering and we can use the preloaded function <code>kmeans()</code>.</p>
<p>We will use the <code>iris_dataset</code> and assume that there are 3 groups (as we know about such dataset and its three putative species).</p>
<div class="sourceCode" id="cb387"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb387-1"><a href="multidimensional-data-in-r.html#cb387-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Let&#39;s check again this dataset</span></span>
<span id="cb387-2"><a href="multidimensional-data-in-r.html#cb387-2" aria-hidden="true" tabindex="-1"></a>iris_dataset <span class="ot">&lt;-</span> iris</span>
<span id="cb387-3"><a href="multidimensional-data-in-r.html#cb387-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(iris_dataset)</span>
<span id="cb387-4"><a href="multidimensional-data-in-r.html#cb387-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  Sepal.Length Sepal.Width Petal.Length Petal.Width Species</span></span>
<span id="cb387-5"><a href="multidimensional-data-in-r.html#cb387-5" aria-hidden="true" tabindex="-1"></a><span class="co">#1          5.1         3.5          1.4         0.2  setosa</span></span>
<span id="cb387-6"><a href="multidimensional-data-in-r.html#cb387-6" aria-hidden="true" tabindex="-1"></a><span class="co">#2          4.9         3.0          1.4         0.2  setosa</span></span>
<span id="cb387-7"><a href="multidimensional-data-in-r.html#cb387-7" aria-hidden="true" tabindex="-1"></a><span class="co">#3          4.7         3.2          1.3         0.2  setosa</span></span>
<span id="cb387-8"><a href="multidimensional-data-in-r.html#cb387-8" aria-hidden="true" tabindex="-1"></a><span class="co">#4          4.6         3.1          1.5         0.2  setosa</span></span>
<span id="cb387-9"><a href="multidimensional-data-in-r.html#cb387-9" aria-hidden="true" tabindex="-1"></a><span class="co">#5          5.0         3.6          1.4         0.2  setosa</span></span>
<span id="cb387-10"><a href="multidimensional-data-in-r.html#cb387-10" aria-hidden="true" tabindex="-1"></a><span class="co">#6          5.4         3.9          1.7         0.4  setosa</span></span>
<span id="cb387-11"><a href="multidimensional-data-in-r.html#cb387-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb387-12"><a href="multidimensional-data-in-r.html#cb387-12" aria-hidden="true" tabindex="-1"></a><span class="do">## We need to remove the column with species assignments</span></span>
<span id="cb387-13"><a href="multidimensional-data-in-r.html#cb387-13" aria-hidden="true" tabindex="-1"></a>iris_dataset_num <span class="ot">&lt;-</span> iris_dataset[,<span class="sc">-</span><span class="fu">which</span>(<span class="fu">names</span>(iris_dataset) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;Species&quot;</span>))]</span>
<span id="cb387-14"><a href="multidimensional-data-in-r.html#cb387-14" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(iris_dataset_num)</span>
<span id="cb387-15"><a href="multidimensional-data-in-r.html#cb387-15" aria-hidden="true" tabindex="-1"></a><span class="co">#  Sepal.Length Sepal.Width Petal.Length Petal.Width</span></span>
<span id="cb387-16"><a href="multidimensional-data-in-r.html#cb387-16" aria-hidden="true" tabindex="-1"></a><span class="co">#1          5.1         3.5          1.4         0.2</span></span>
<span id="cb387-17"><a href="multidimensional-data-in-r.html#cb387-17" aria-hidden="true" tabindex="-1"></a><span class="co">#2          4.9         3.0          1.4         0.2</span></span>
<span id="cb387-18"><a href="multidimensional-data-in-r.html#cb387-18" aria-hidden="true" tabindex="-1"></a><span class="co">#3          4.7         3.2          1.3         0.2</span></span>
<span id="cb387-19"><a href="multidimensional-data-in-r.html#cb387-19" aria-hidden="true" tabindex="-1"></a><span class="co">#4          4.6         3.1          1.5         0.2</span></span>
<span id="cb387-20"><a href="multidimensional-data-in-r.html#cb387-20" aria-hidden="true" tabindex="-1"></a><span class="co">#5          5.0         3.6          1.4         0.2</span></span>
<span id="cb387-21"><a href="multidimensional-data-in-r.html#cb387-21" aria-hidden="true" tabindex="-1"></a><span class="co">#6          5.4         3.9          1.7         0.4</span></span></code></pre></div>
<p>Notice how I removed the “Species” column, this is one of many forms to remove columns by name. Now we determine the 3 groups with <code>centers = 3</code> by repeating 30 times using the argument <code>nstart = 30</code> with the function <code>kmeans()</code>.</p>
<div class="sourceCode" id="cb388"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb388-1"><a href="multidimensional-data-in-r.html#cb388-1" aria-hidden="true" tabindex="-1"></a>iris_dataset_kmeans <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(iris_dataset_num, <span class="at">centers =</span> <span class="dv">3</span>, <span class="at">nstart =</span> <span class="dv">30</span>)</span>
<span id="cb388-2"><a href="multidimensional-data-in-r.html#cb388-2" aria-hidden="true" tabindex="-1"></a><span class="co">#K-means clustering with 3 clusters of sizes 50, 62, 38</span></span>
<span id="cb388-3"><a href="multidimensional-data-in-r.html#cb388-3" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb388-4"><a href="multidimensional-data-in-r.html#cb388-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Cluster means:</span></span>
<span id="cb388-5"><a href="multidimensional-data-in-r.html#cb388-5" aria-hidden="true" tabindex="-1"></a><span class="co">#  Sepal.Length Sepal.Width Petal.Length Petal.Width</span></span>
<span id="cb388-6"><a href="multidimensional-data-in-r.html#cb388-6" aria-hidden="true" tabindex="-1"></a><span class="co">#1     5.006000    3.428000     1.462000    0.246000</span></span>
<span id="cb388-7"><a href="multidimensional-data-in-r.html#cb388-7" aria-hidden="true" tabindex="-1"></a><span class="co">#2     5.901613    2.748387     4.393548    1.433871</span></span>
<span id="cb388-8"><a href="multidimensional-data-in-r.html#cb388-8" aria-hidden="true" tabindex="-1"></a><span class="co">#3     6.850000    3.073684     5.742105    2.071053</span></span>
<span id="cb388-9"><a href="multidimensional-data-in-r.html#cb388-9" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb388-10"><a href="multidimensional-data-in-r.html#cb388-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Clustering vector:</span></span>
<span id="cb388-11"><a href="multidimensional-data-in-r.html#cb388-11" aria-hidden="true" tabindex="-1"></a><span class="co">#  [1] 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 2 2 3 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2</span></span>
<span id="cb388-12"><a href="multidimensional-data-in-r.html#cb388-12" aria-hidden="true" tabindex="-1"></a><span class="co"># [70] 2 2 2 2 2 2 2 2 3 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 3 2 3 3 3 3 2 3 3 3 3 3 3 2 2 3 3 3 3 2 3 2 3 2 3 3 2 2 3 3 3 3 3 2 3 3 3 3</span></span>
<span id="cb388-13"><a href="multidimensional-data-in-r.html#cb388-13" aria-hidden="true" tabindex="-1"></a><span class="co">#[139] 2 3 3 3 2 3 3 3 2 3 3 2</span></span>
<span id="cb388-14"><a href="multidimensional-data-in-r.html#cb388-14" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb388-15"><a href="multidimensional-data-in-r.html#cb388-15" aria-hidden="true" tabindex="-1"></a><span class="co">#Within cluster sum of squares by cluster:</span></span>
<span id="cb388-16"><a href="multidimensional-data-in-r.html#cb388-16" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 15.15100 39.82097 23.87947</span></span>
<span id="cb388-17"><a href="multidimensional-data-in-r.html#cb388-17" aria-hidden="true" tabindex="-1"></a><span class="co"># (between_SS / total_SS =  88.4 %)</span></span>
<span id="cb388-18"><a href="multidimensional-data-in-r.html#cb388-18" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb388-19"><a href="multidimensional-data-in-r.html#cb388-19" aria-hidden="true" tabindex="-1"></a><span class="co">#Available components:</span></span>
<span id="cb388-20"><a href="multidimensional-data-in-r.html#cb388-20" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb388-21"><a href="multidimensional-data-in-r.html#cb388-21" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;cluster&quot;      &quot;centers&quot;      &quot;totss&quot;        &quot;withinss&quot;     &quot;tot.withinss&quot; &quot;betweenss&quot;    &quot;size&quot;         &quot;iter&quot;         &quot;ifault&quot;      </span></span></code></pre></div>
<p>We can add this kmean groups and compare those with the original species names.</p>
<div class="sourceCode" id="cb389"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb389-1"><a href="multidimensional-data-in-r.html#cb389-1" aria-hidden="true" tabindex="-1"></a><span class="do">## add the cluster vector to iris_dataset</span></span>
<span id="cb389-2"><a href="multidimensional-data-in-r.html#cb389-2" aria-hidden="true" tabindex="-1"></a>iris_dataset_kmeans<span class="sc">$</span>cluster</span>
<span id="cb389-3"><a href="multidimensional-data-in-r.html#cb389-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  [1] 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 2 2 3 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2</span></span>
<span id="cb389-4"><a href="multidimensional-data-in-r.html#cb389-4" aria-hidden="true" tabindex="-1"></a><span class="co"># [70] 2 2 2 2 2 2 2 2 3 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 3 2 3 3 3 3 2 3 3 3 3 3 3 2 2 3 3 3 3 2 3 2 3 2 3 3 2 2 3 3 3 3 3 2 3 3 3 3</span></span>
<span id="cb389-5"><a href="multidimensional-data-in-r.html#cb389-5" aria-hidden="true" tabindex="-1"></a><span class="co">#[139] 2 3 3 3 2 3 3 3 2 3 3 2</span></span>
<span id="cb389-6"><a href="multidimensional-data-in-r.html#cb389-6" aria-hidden="true" tabindex="-1"></a>iris_dataset_original<span class="sc">$</span>kmeans_cluster <span class="ot">&lt;-</span> <span class="fu">factor</span>(iris_dataset_kmeans<span class="sc">$</span>cluster)</span>
<span id="cb389-7"><a href="multidimensional-data-in-r.html#cb389-7" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(iris_dataset_original)</span>
<span id="cb389-8"><a href="multidimensional-data-in-r.html#cb389-8" aria-hidden="true" tabindex="-1"></a><span class="co">#    Species      LD1        LD2 kmeans_cluster</span></span>
<span id="cb389-9"><a href="multidimensional-data-in-r.html#cb389-9" aria-hidden="true" tabindex="-1"></a><span class="co">#1  setosa 8.061800  0.3004206              1</span></span>
<span id="cb389-10"><a href="multidimensional-data-in-r.html#cb389-10" aria-hidden="true" tabindex="-1"></a><span class="co">#2  setosa 7.128688 -0.7866604              1</span></span>
<span id="cb389-11"><a href="multidimensional-data-in-r.html#cb389-11" aria-hidden="true" tabindex="-1"></a><span class="co">#3  setosa 7.489828 -0.2653845              1</span></span>
<span id="cb389-12"><a href="multidimensional-data-in-r.html#cb389-12" aria-hidden="true" tabindex="-1"></a><span class="co">#4  setosa 6.813201 -0.6706311              1</span></span>
<span id="cb389-13"><a href="multidimensional-data-in-r.html#cb389-13" aria-hidden="true" tabindex="-1"></a><span class="co">#5  setosa 8.132309  0.5144625              1</span></span>
<span id="cb389-14"><a href="multidimensional-data-in-r.html#cb389-14" aria-hidden="true" tabindex="-1"></a><span class="co">#6  setosa 7.701947  1.4617210              1</span></span>
<span id="cb389-15"><a href="multidimensional-data-in-r.html#cb389-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb389-16"><a href="multidimensional-data-in-r.html#cb389-16" aria-hidden="true" tabindex="-1"></a><span class="do">## we can plot these groups using the lda coordinates to visualize.</span></span>
<span id="cb389-17"><a href="multidimensional-data-in-r.html#cb389-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb389-18"><a href="multidimensional-data-in-r.html#cb389-18" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(iris_dataset_original, <span class="fu">aes</span>(LD1, LD2)) <span class="sc">+</span></span>
<span id="cb389-19"><a href="multidimensional-data-in-r.html#cb389-19" aria-hidden="true" tabindex="-1"></a>       <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> kmeans_cluster)) <span class="sc">+</span></span>
<span id="cb389-20"><a href="multidimensional-data-in-r.html#cb389-20" aria-hidden="true" tabindex="-1"></a>           <span class="fu">ggtitle</span>(<span class="st">&quot;Iris kmeans with 3 Groups&quot;</span>)</span></code></pre></div>
<p><img src="figures/session_8_figs/11_Iris_kmeans_groups.png" width="100%" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="regressions-and-anova-in-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="graphics-and-colors.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/08-PCA_DFA_exploratory_v6.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
