<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Session 7 – Regressions and ANOVA in R | Bioinformatics Workshop Gitbook</title>
  <meta name="description" content="none" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Session 7 – Regressions and ANOVA in R | Bioinformatics Workshop Gitbook" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="none" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Session 7 – Regressions and ANOVA in R | Bioinformatics Workshop Gitbook" />
  
  <meta name="twitter:description" content="none" />
  

<meta name="author" content="Juan C. Santos &amp; Randy Ortiz" />


<meta name="date" content="2021-04-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="basic-statistics-in-r.html"/>
<link rel="next" href="exploring-multidimensional-data-in-r.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics Workshop Gitbook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html"><i class="fa fa-check"></i><b>1</b> – Getting Started with R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#why-do-we-use-r"><i class="fa fa-check"></i><b>1.1</b> Why do we use R?</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#get-a-good-text-editor"><i class="fa fa-check"></i><b>1.2</b> Get a good text editor</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-r"><i class="fa fa-check"></i><b>1.4</b> Installing R</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#annotating-code"><i class="fa fa-check"></i><b>1.5</b> Annotating code</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#defining-a-working-directory-in-r"><i class="fa fa-check"></i><b>1.6</b> Defining a working directory in R</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html"><i class="fa fa-check"></i><b>2</b> – Using R, Installing Packages and Importing/Exporting Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#your-first-session"><i class="fa fa-check"></i><b>2.1</b> Your first session</a></li>
<li class="chapter" data-level="2.2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#doing-some-base-functions-and-getting-help"><i class="fa fa-check"></i><b>2.2</b> Doing some base functions and getting help</a></li>
<li class="chapter" data-level="2.3" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#about-r-packages"><i class="fa fa-check"></i><b>2.3</b> About R-packages</a></li>
<li class="chapter" data-level="2.4" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#installing-packages"><i class="fa fa-check"></i><b>2.4</b> Installing packages</a></li>
<li class="chapter" data-level="2.5" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#importing-and-exporting-data"><i class="fa fa-check"></i><b>2.5</b> Importing and exporting data</a></li>
<li class="chapter" data-level="2.6" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#saving-data-as-rdata-.rdata-.rda"><i class="fa fa-check"></i><b>2.6</b> Saving data as Rdata (.rdata, .rda)</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html"><i class="fa fa-check"></i><b>3</b> – Basic Data Structures in R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#scalars-and-vectors"><i class="fa fa-check"></i><b>3.1</b> Scalars and vectors</a></li>
<li class="chapter" data-level="3.2" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#arrays"><i class="fa fa-check"></i><b>3.3</b> Arrays</a></li>
<li class="chapter" data-level="3.4" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a></li>
<li class="chapter" data-level="3.5" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tables"><i class="fa fa-check"></i><b>3.5</b> Tables</a></li>
<li class="chapter" data-level="3.6" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#lists"><i class="fa fa-check"></i><b>3.6</b> Lists</a></li>
<li class="chapter" data-level="3.7" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-tables"><i class="fa fa-check"></i><b>3.7</b> Data Tables</a></li>
<li class="chapter" data-level="3.8" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tibbles"><i class="fa fa-check"></i><b>3.8</b> Tibbles</a></li>
<li class="chapter" data-level="3.9" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#dnastringsets-rnastringsets-and-aastringsets"><i class="fa fa-check"></i><b>3.9</b> DNAStringSets, RNAStringSets and AAStringSets</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html"><i class="fa fa-check"></i><b>4</b> – Text Editing and Data Transformations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#working-with-character-strings"><i class="fa fa-check"></i><b>4.1</b> Working with character strings</a></li>
<li class="chapter" data-level="4.2" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#regular-expressions-regex"><i class="fa fa-check"></i><b>4.2</b> Regular expressions (Regex)</a></li>
<li class="chapter" data-level="4.3" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#comparing-ordering-and-simplifying-vectors"><i class="fa fa-check"></i><b>4.3</b> Comparing, ordering and simplifying vectors</a></li>
<li class="chapter" data-level="4.4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#useful-functions-of-stringr"><i class="fa fa-check"></i><b>4.4</b> Useful functions of <code>stringr</code></a></li>
<li class="chapter" data-level="4.5" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#data-transformations"><i class="fa fa-check"></i><b>4.5</b> Data transformations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html"><i class="fa fa-check"></i><b>5</b> – Getting Biological Data from Public Repositories</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#ncbi-databases-genbank-and-others"><i class="fa fa-check"></i><b>5.1</b> NCBI databases: GenBank and others</a></li>
<li class="chapter" data-level="5.2" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#accessing-ncbi-within-r-rentrez"><i class="fa fa-check"></i><b>5.2</b> Accessing NCBI within R: <code>rentrez</code></a></li>
<li class="chapter" data-level="5.3" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#retrieving-pubmed-references"><i class="fa fa-check"></i><b>5.3</b> Retrieving PubMed references</a></li>
<li class="chapter" data-level="5.4" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#comparing-sequences-blast"><i class="fa fa-check"></i><b>5.4</b> Comparing sequences: BLAST</a></li>
<li class="chapter" data-level="5.5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-uniprot"><i class="fa fa-check"></i><b>5.5</b> Other databases: UniProt</a></li>
<li class="chapter" data-level="5.6" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-ensembl"><i class="fa fa-check"></i><b>5.6</b> Other databases: Ensembl</a></li>
<li class="chapter" data-level="5.7" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-taxonomy-phylogenetic-trees-and-species-distributions"><i class="fa fa-check"></i><b>5.7</b> Other databases: Taxonomy, phylogenetic trees and species distributions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html"><i class="fa fa-check"></i><b>6</b> – Basic Statistics in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#exploring-your-data"><i class="fa fa-check"></i><b>6.1</b> Exploring your data</a></li>
<li class="chapter" data-level="6.2" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#normalizing-your-data"><i class="fa fa-check"></i><b>6.2</b> Normalizing your data</a></li>
<li class="chapter" data-level="6.3" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#what-test-should-i-use"><i class="fa fa-check"></i><b>6.3</b> What test should I use?</a></li>
<li class="chapter" data-level="6.4" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-means-t-test"><i class="fa fa-check"></i><b>6.4</b> Comparing means: t-test</a></li>
<li class="chapter" data-level="6.5" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-wilcoxon-signed-rank-test"><i class="fa fa-check"></i><b>6.5</b> Beyond t-test: Wilcoxon signed rank test</a></li>
<li class="chapter" data-level="6.6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-binomial-test"><i class="fa fa-check"></i><b>6.6</b> Beyond t-test: Binomial test</a></li>
<li class="chapter" data-level="6.7" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-chi-square-goodness-of-fit"><i class="fa fa-check"></i><b>6.7</b> Beyond t-test: Chi-square goodness of fit</a></li>
<li class="chapter" data-level="6.8" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-mcnemars-test"><i class="fa fa-check"></i><b>6.8</b> Beyond t-test: McNemar’s test</a></li>
<li class="chapter" data-level="6.9" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-fishers-exact-test"><i class="fa fa-check"></i><b>6.9</b> Beyond t-test: Fisher’s exact test</a></li>
<li class="chapter" data-level="6.10" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#correlations"><i class="fa fa-check"></i><b>6.10</b> Correlations</a></li>
<li class="chapter" data-level="6.11" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-distributions"><i class="fa fa-check"></i><b>6.11</b> Comparing distributions</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html"><i class="fa fa-check"></i><b>7</b> – Regressions and ANOVA in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#linear-regressions"><i class="fa fa-check"></i><b>7.1</b> Linear regressions</a></li>
<li class="chapter" data-level="7.2" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#selecting-predictors"><i class="fa fa-check"></i><b>7.2</b> Selecting predictors</a></li>
<li class="chapter" data-level="7.3" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions"><i class="fa fa-check"></i><b>7.3</b> Logistic regressions</a></li>
<li class="chapter" data-level="7.4" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#non-linear-regressions"><i class="fa fa-check"></i><b>7.4</b> Non-linear regressions</a></li>
<li class="chapter" data-level="7.5" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-regressions"><i class="fa fa-check"></i><b>7.5</b> Toy datasets for regressions</a></li>
<li class="chapter" data-level="7.6" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova"><i class="fa fa-check"></i><b>7.6</b> ANOVA</a></li>
<li class="chapter" data-level="7.7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#one-way-anova"><i class="fa fa-check"></i><b>7.7</b> One-way ANOVA</a></li>
<li class="chapter" data-level="7.8" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#two-way-anova"><i class="fa fa-check"></i><b>7.8</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="7.9" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#post-hoc-tests"><i class="fa fa-check"></i><b>7.9</b> Post-hoc tests</a></li>
<li class="chapter" data-level="7.10" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#beyond-anova-kruskal-wallis-test"><i class="fa fa-check"></i><b>7.10</b> Beyond ANOVA: Kruskal-Wallis Test</a></li>
<li class="chapter" data-level="7.11" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-anova"><i class="fa fa-check"></i><b>7.11</b> Toy datasets for ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html"><i class="fa fa-check"></i><b>8</b> – Exploring Multidimensional Data in R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html#principal-component-analysis"><i class="fa fa-check"></i><b>8.1</b> Principal Component Analysis</a></li>
<li class="chapter" data-level="8.2" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html#multidimensional-scaling"><i class="fa fa-check"></i><b>8.2</b> Multidimensional scaling</a></li>
<li class="chapter" data-level="8.3" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html#cluster-analysis-dfa"><i class="fa fa-check"></i><b>8.3</b> Cluster analysis: DFA</a></li>
<li class="chapter" data-level="8.4" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html#cluster-analysis-k-means-unsupervised"><i class="fa fa-check"></i><b>8.4</b> Cluster analysis: <em>k</em>-means (unsupervised)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html"><i class="fa fa-check"></i><b>9</b> – Graphics and Colors</a>
<ul>
<li class="chapter" data-level="9.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#avoid-bad-graphics"><i class="fa fa-check"></i><b>9.1</b> Avoid bad graphics</a></li>
<li class="chapter" data-level="9.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#graphics-in-r-ggplot2"><i class="fa fa-check"></i><b>9.2</b> Graphics in R – ggplot2</a></li>
<li class="chapter" data-level="9.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#preparing-your-dataset"><i class="fa fa-check"></i><b>9.3</b> Preparing your dataset</a></li>
<li class="chapter" data-level="9.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#color-palettes-online"><i class="fa fa-check"></i><b>9.4</b> Color palettes online</a></li>
<li class="chapter" data-level="9.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggplot2-default-palette"><i class="fa fa-check"></i><b>9.5</b> <code>ggplot2</code> default palette</a></li>
<li class="chapter" data-level="9.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#heatmaps-or-tiled-charts"><i class="fa fa-check"></i><b>9.6</b> Heatmaps or tiled charts</a></li>
<li class="chapter" data-level="9.7" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#r-packages-about-color"><i class="fa fa-check"></i><b>9.7</b> R-packages about color</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#viridis-package"><i class="fa fa-check"></i><b>9.7.1</b> <code>viridis</code> package</a></li>
<li class="chapter" data-level="9.7.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcolorbrewer-package"><i class="fa fa-check"></i><b>9.7.2</b> <code>RColorBrewer</code> package</a></li>
<li class="chapter" data-level="9.7.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggsci-package"><i class="fa fa-check"></i><b>9.7.3</b> <code>ggsci</code> package</a></li>
<li class="chapter" data-level="9.7.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#wesanderson-package"><i class="fa fa-check"></i><b>9.7.4</b> <code>wesanderson</code> package</a></li>
<li class="chapter" data-level="9.7.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcartocolor-package"><i class="fa fa-check"></i><b>9.7.5</b> <code>rcartocolor</code> package</a></li>
<li class="chapter" data-level="9.7.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#colorspace-package"><i class="fa fa-check"></i><b>9.7.6</b> <code>colorspace</code> package</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#making-your-own-palette"><i class="fa fa-check"></i><b>9.8</b> Making your own palette</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="basic-graphics.html"><a href="basic-graphics.html"><i class="fa fa-check"></i><b>10</b> – Basic Graphics</a>
<ul>
<li class="chapter" data-level="10.1" data-path="basic-graphics.html"><a href="basic-graphics.html#bar-charts"><i class="fa fa-check"></i><b>10.1</b> Bar charts</a></li>
<li class="chapter" data-level="10.2" data-path="basic-graphics.html"><a href="basic-graphics.html#stacked-bar-charts"><i class="fa fa-check"></i><b>10.2</b> Stacked bar charts</a></li>
<li class="chapter" data-level="10.3" data-path="basic-graphics.html"><a href="basic-graphics.html#histograms"><i class="fa fa-check"></i><b>10.3</b> Histograms</a></li>
<li class="chapter" data-level="10.4" data-path="basic-graphics.html"><a href="basic-graphics.html#multiple-histograms"><i class="fa fa-check"></i><b>10.4</b> Multiple histograms</a></li>
<li class="chapter" data-level="10.5" data-path="basic-graphics.html"><a href="basic-graphics.html#density-plots"><i class="fa fa-check"></i><b>10.5</b> Density plots</a></li>
<li class="chapter" data-level="10.6" data-path="basic-graphics.html"><a href="basic-graphics.html#box-plots"><i class="fa fa-check"></i><b>10.6</b> Box plots</a></li>
<li class="chapter" data-level="10.7" data-path="basic-graphics.html"><a href="basic-graphics.html#violin-plots"><i class="fa fa-check"></i><b>10.7</b> Violin plots</a></li>
<li class="chapter" data-level="10.8" data-path="basic-graphics.html"><a href="basic-graphics.html#scatterplots"><i class="fa fa-check"></i><b>10.8</b> Scatterplots</a></li>
<li class="chapter" data-level="10.9" data-path="basic-graphics.html"><a href="basic-graphics.html#lines-and-scatterplots"><i class="fa fa-check"></i><b>10.9</b> Lines and Scatterplots</a></li>
<li class="chapter" data-level="10.10" data-path="basic-graphics.html"><a href="basic-graphics.html#multiple-graphs-at-once-with-cowplot"><i class="fa fa-check"></i><b>10.10</b> Multiple graphs at once with <code>cowplot</code></a></li>
<li class="chapter" data-level="10.11" data-path="basic-graphics.html"><a href="basic-graphics.html#save-your-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>10.11</b> Save your plots as PDF or PNG</a></li>
<li class="chapter" data-level="10.12" data-path="basic-graphics.html"><a href="basic-graphics.html#other-resources"><i class="fa fa-check"></i><b>10.12</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="sequence-alignments.html"><a href="sequence-alignments.html"><i class="fa fa-check"></i><b>11</b> – Sequence Alignments</a>
<ul>
<li class="chapter" data-level="11.1" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-using-rentrez"><i class="fa fa-check"></i><b>11.1</b> Retrieving sequences using ‘rentrez’</a></li>
<li class="chapter" data-level="11.2" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-manually-from-ncbi"><i class="fa fa-check"></i><b>11.2</b> Retrieving sequences manually from NCBI</a></li>
<li class="chapter" data-level="11.3" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-accession-numbers"><i class="fa fa-check"></i><b>11.3</b> Retrieving accession numbers</a></li>
<li class="chapter" data-level="11.4" data-path="sequence-alignments.html"><a href="sequence-alignments.html#alignment-with-decipher"><i class="fa fa-check"></i><b>11.4</b> Alignment with <code>DECIPHER</code></a></li>
<li class="chapter" data-level="11.5" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualize-alignment"><i class="fa fa-check"></i><b>11.5</b> Visualize alignment</a></li>
<li class="chapter" data-level="11.6" data-path="sequence-alignments.html"><a href="sequence-alignments.html#save-alignment-to-file"><i class="fa fa-check"></i><b>11.6</b> Save alignment to file</a></li>
<li class="chapter" data-level="11.7" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-and-align-several-markers"><i class="fa fa-check"></i><b>11.7</b> Concatenate and align several markers</a></li>
<li class="chapter" data-level="11.8" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-data.table"><i class="fa fa-check"></i><b>11.8</b> Concatenate with <code>data.table</code></a></li>
<li class="chapter" data-level="11.9" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-biostrings"><i class="fa fa-check"></i><b>11.9</b> Concatenate with <code>Biostrings</code></a></li>
<li class="chapter" data-level="11.10" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualizing-alignments-in-r"><i class="fa fa-check"></i><b>11.10</b> Visualizing alignments in R</a></li>
<li class="chapter" data-level="11.11" data-path="sequence-alignments.html"><a href="sequence-alignments.html#other-sequence-alignment-software"><i class="fa fa-check"></i><b>11.11</b> Other sequence alignment software</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html"><i class="fa fa-check"></i><b>12</b> – Estimating Phylogenetic Trees</a>
<ul>
<li class="chapter" data-level="12.1" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#phylogeny-inference-with-iqtree2"><i class="fa fa-check"></i><b>12.1</b> Phylogeny inference with <code>iqtree2</code></a></li>
<li class="chapter" data-level="12.2" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-iqtree2-in-macos"><i class="fa fa-check"></i><b>12.2</b> Installing <code>iqtree2</code> in macOS</a></li>
<li class="chapter" data-level="12.3" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-and-running-iqtree2-in-a-pc"><i class="fa fa-check"></i><b>12.3</b> Installing and running <code>iqtree2</code> in a PC</a></li>
<li class="chapter" data-level="12.4" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#loading-the-iqtree_runner_r-function"><i class="fa fa-check"></i><b>12.4</b> Loading the <code>iqtree_runner_R</code> function</a></li>
<li class="chapter" data-level="12.5" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#running-iqtree2-from-r"><i class="fa fa-check"></i><b>12.5</b> Running <code>iqtree2</code> from R</a></li>
<li class="chapter" data-level="12.6" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-no-partitions"><i class="fa fa-check"></i><b>12.6</b> One marker and no partitions</a></li>
<li class="chapter" data-level="12.7" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-with-partitions"><i class="fa fa-check"></i><b>12.7</b> One marker and with partitions</a></li>
<li class="chapter" data-level="12.8" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#bootstrap-support"><i class="fa fa-check"></i><b>12.8</b> Bootstrap support</a></li>
<li class="chapter" data-level="12.9" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#many-markers-and-with-partitions"><i class="fa fa-check"></i><b>12.9</b> Many markers and with partitions</a></li>
<li class="chapter" data-level="12.10" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#other-sequence-phylogenetic-tree-estimation-software"><i class="fa fa-check"></i><b>12.10</b> Other sequence phylogenetic tree estimation software</a></li>
<li class="chapter" data-level="12.11" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#drawing-phylogenetic-trees-with-ggtree"><i class="fa fa-check"></i><b>12.11</b> Drawing phylogenetic trees with <code>ggtree</code></a></li>
<li class="chapter" data-level="12.12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#save-ggtree-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>12.12</b> Save ggtree plots as PDF or PNG</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html"><i class="fa fa-check"></i><b>13</b> – Mining from a reference dataset</a>
<ul>
<li class="chapter" data-level="13.1" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#a-worked-example-using-the-mtcars-dataset"><i class="fa fa-check"></i><b>13.1</b> A worked example using the <code>mtcars</code> dataset</a></li>
<li class="chapter" data-level="13.2" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-reference-dataset"><i class="fa fa-check"></i><b>13.2</b> Build a reference dataset</a></li>
<li class="chapter" data-level="13.3" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-sampling-dataset"><i class="fa fa-check"></i><b>13.3</b> Build a sampling dataset</a></li>
<li class="chapter" data-level="13.4" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#extracting-from-reference-dataset"><i class="fa fa-check"></i><b>13.4</b> Extracting from reference dataset</a></li>
<li class="chapter" data-level="13.5" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#plots-of-the-extracted-data"><i class="fa fa-check"></i><b>13.5</b> Plots of the extracted data</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="introduction-to-python.html"><a href="introduction-to-python.html"><i class="fa fa-check"></i><b>14</b> – Introduction to Python</a>
<ul>
<li class="chapter" data-level="14.1" data-path="introduction-to-python.html"><a href="introduction-to-python.html#welcome-to-python-why-python"><i class="fa fa-check"></i><b>14.1</b> Welcome to Python! Why Python?</a></li>
<li class="chapter" data-level="14.2" data-path="introduction-to-python.html"><a href="introduction-to-python.html#which-python-should-i-install"><i class="fa fa-check"></i><b>14.2</b> Which Python should I install?</a></li>
<li class="chapter" data-level="14.3" data-path="introduction-to-python.html"><a href="introduction-to-python.html#mac-os-x-python-installations"><i class="fa fa-check"></i><b>14.3</b> Mac OS X Python Installations</a></li>
<li class="chapter" data-level="14.4" data-path="introduction-to-python.html"><a href="introduction-to-python.html#windows-python-installations"><i class="fa fa-check"></i><b>14.4</b> Windows Python Installations</a></li>
<li class="chapter" data-level="14.5" data-path="introduction-to-python.html"><a href="introduction-to-python.html#install-an-ide-or-integrated-development-environment"><i class="fa fa-check"></i><b>14.5</b> Install an IDE, or integrated development environment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html"><i class="fa fa-check"></i><b>15</b> – Python: Keywords and Identifiers</a>
<ul>
<li class="chapter" data-level="15.1" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#keywords-in-python"><i class="fa fa-check"></i><b>15.1</b> Keywords in Python</a></li>
<li class="chapter" data-level="15.2" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#identifiers-in-python"><i class="fa fa-check"></i><b>15.2</b> Identifiers in Python</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html"><i class="fa fa-check"></i><b>16</b> – Python: Statements and Expressions</a>
<ul>
<li class="chapter" data-level="16.1" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#what-are-statements"><i class="fa fa-check"></i><b>16.1</b> What are statements?</a></li>
<li class="chapter" data-level="16.2" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#expressions-in-python"><i class="fa fa-check"></i><b>16.2</b> Expressions in Python</a></li>
<li class="chapter" data-level="16.3" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#simple-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.3</b> Simple Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.4" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#augmented-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.4</b> Augmented Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.5" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#explicit-and-implicit-line-continuation"><i class="fa fa-check"></i><b>16.5</b> Explicit and Implicit Line Continuation</a></li>
<li class="chapter" data-level="16.6" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#comments"><i class="fa fa-check"></i><b>16.6</b> Comments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="python-data-types.html"><a href="python-data-types.html"><i class="fa fa-check"></i><b>17</b> – Python: Data Types</a>
<ul>
<li class="chapter" data-level="17.1" data-path="python-data-types.html"><a href="python-data-types.html#booleans"><i class="fa fa-check"></i><b>17.1</b> Booleans</a></li>
<li class="chapter" data-level="17.2" data-path="python-data-types.html"><a href="python-data-types.html#numbers-int-float-and-complex"><i class="fa fa-check"></i><b>17.2</b> Numbers: int, float and complex</a></li>
<li class="chapter" data-level="17.3" data-path="python-data-types.html"><a href="python-data-types.html#strings"><i class="fa fa-check"></i><b>17.3</b> Strings</a></li>
<li class="chapter" data-level="17.4" data-path="python-data-types.html"><a href="python-data-types.html#bytes"><i class="fa fa-check"></i><b>17.4</b> Bytes</a></li>
<li class="chapter" data-level="17.5" data-path="python-data-types.html"><a href="python-data-types.html#lists-1"><i class="fa fa-check"></i><b>17.5</b> Lists</a></li>
<li class="chapter" data-level="17.6" data-path="python-data-types.html"><a href="python-data-types.html#tuples"><i class="fa fa-check"></i><b>17.6</b> Tuples</a></li>
<li class="chapter" data-level="17.7" data-path="python-data-types.html"><a href="python-data-types.html#sets"><i class="fa fa-check"></i><b>17.7</b> Sets</a></li>
<li class="chapter" data-level="17.8" data-path="python-data-types.html"><a href="python-data-types.html#dictionary"><i class="fa fa-check"></i><b>17.8</b> Dictionary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="python-strings.html"><a href="python-strings.html"><i class="fa fa-check"></i><b>18</b> – Python: Strings</a>
<ul>
<li class="chapter" data-level="18.1" data-path="python-strings.html"><a href="python-strings.html#string-operators"><i class="fa fa-check"></i><b>18.1</b> String Operators</a></li>
<li class="chapter" data-level="18.2" data-path="python-strings.html"><a href="python-strings.html#other-string-modifiers"><i class="fa fa-check"></i><b>18.2</b> Other String Modifiers</a></li>
<li class="chapter" data-level="18.3" data-path="python-strings.html"><a href="python-strings.html#comparisons-with-strings"><i class="fa fa-check"></i><b>18.3</b> Comparisons with Strings</a></li>
<li class="chapter" data-level="18.4" data-path="python-strings.html"><a href="python-strings.html#padding-strings"><i class="fa fa-check"></i><b>18.4</b> Padding strings</a></li>
<li class="chapter" data-level="18.5" data-path="python-strings.html"><a href="python-strings.html#search-for-words-in-string-and-output-location"><i class="fa fa-check"></i><b>18.5</b> Search for words in string and output location</a></li>
<li class="chapter" data-level="18.6" data-path="python-strings.html"><a href="python-strings.html#seek-and-replace-in-strings"><i class="fa fa-check"></i><b>18.6</b> Seek and replace in strings</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="python-operators.html"><a href="python-operators.html"><i class="fa fa-check"></i><b>19</b> – Python: Operators</a>
<ul>
<li class="chapter" data-level="19.1" data-path="python-operators.html"><a href="python-operators.html#many-types-of-operators"><i class="fa fa-check"></i><b>19.1</b> Many types of operators</a></li>
<li class="chapter" data-level="19.2" data-path="python-operators.html"><a href="python-operators.html#arithmetic-operators"><i class="fa fa-check"></i><b>19.2</b> Arithmetic operators</a></li>
<li class="chapter" data-level="19.3" data-path="python-operators.html"><a href="python-operators.html#comparison-operators"><i class="fa fa-check"></i><b>19.3</b> Comparison operators</a></li>
<li class="chapter" data-level="19.4" data-path="python-operators.html"><a href="python-operators.html#logical-operators"><i class="fa fa-check"></i><b>19.4</b> Logical operators</a></li>
<li class="chapter" data-level="19.5" data-path="python-operators.html"><a href="python-operators.html#bitwise-operators"><i class="fa fa-check"></i><b>19.5</b> Bitwise operators</a></li>
<li class="chapter" data-level="19.6" data-path="python-operators.html"><a href="python-operators.html#identity-operators"><i class="fa fa-check"></i><b>19.6</b> Identity Operators</a></li>
<li class="chapter" data-level="19.7" data-path="python-operators.html"><a href="python-operators.html#membership-operators"><i class="fa fa-check"></i><b>19.7</b> Membership operators</a></li>
<li class="chapter" data-level="19.8" data-path="python-operators.html"><a href="python-operators.html#what-is-the-order-of-operations-with-operators"><i class="fa fa-check"></i><b>19.8</b> What is the order of operations with operators?</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics Workshop Gitbook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="regressions-and-anova-in-r" class="section level1" number="7">
<h1><span class="header-section-number">Session 7</span> – Regressions and ANOVA in R</h1>
<p>An important aspect of any statistical analyses is determining the relationship between variables and developing predictive models. A simple linear regression is a model of the relationship between a dependent (response) with a predictor variable and some error associated with it. This model can be expanded to include more than one predictor variable, which is known as a multivariate linear regression.
These can be easily implemented in R.</p>
<p>As mentioned, this workshop is not a biostatistics class and you might need to consult your notes for biostatistics. However, as mentioned by Long and Teetor (2019) here are some important considerations when you evaluate the usefulness of your regression models. This is provided by the function <code>summary()</code> on your model.</p>
<p><strong>a) Is the model statistically significant?</strong>
Check the <em>F</em> statistic at the bottom of the summary.</p>
<p><strong>b) Are the regression coefficients significant?</strong>
Check the coefficient <em>t</em> statistic and the <em>p-value</em> associate with such statistic.</p>
<p><strong>c) Is the model useful?</strong>
Check the <em>R</em><sup>2</sup> and its significance.</p>
<p><strong>d) Does the model fit the provided data well?</strong>
Plot the residuals and check regression diagnostics.</p>
<p><strong>e) Does the data satisfy the assumptions behind linear regression?</strong>
Check whether the diagnostics confirm that a linear model is reasonable for your data.</p>
<p>As happens in many cases, biological data are more complex and not necessarily follow a linear model. Many alternative and non-linear models might be more useful when dealing with such complexity. We will revise some non-parametric models.
This can be visualized in the <a href="https://en.wikipedia.org/wiki/Anscombe%27s_quartet" target="_blank"><strong>Anscombes quartet</strong></a>.</p>
<p><img src="figures/session_7_figs/1_Anscombes_quartet_3.png" width="70%" /></p>
<p>Jan Vanhove <a href="https://janhove.github.io/teaching/2016/11/21/what-correlations-look-like" target="_blank"><strong>What data patterns can lie behind a correlation coefficient?</strong></a>.</p>
<p><img src="figures/session_7_figs/2_Vanhove_plots.png" width="70%" /></p>
<div id="linear-regressions" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> Linear regressions</h2>
<p><strong>1)</strong> The most basic form is a simple linear regression that can be performed directly in R using the function <code>lm()</code>. In this case, we have to two vectors with paired responses where a measurement of one predictor usually represented as <strong>x</strong> is corresponded with one dependent response represented as <strong>y</strong>.</p>
<p>For this example, we will use the R-package <a href="https://cran.r-project.org/web/packages/MASS/index.html" target="_blank"><strong>MASS</strong></a> and its dataset <code>mammals</code> that contains brain and body weights for 62 species of land mammals.</p>
<div class="sourceCode" id="cb285"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb285-1"><a href="regressions-and-anova-in-r.html#cb285-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we can install MASS and load the dataset for this example</span></span>
<span id="cb285-2"><a href="regressions-and-anova-in-r.html#cb285-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;MASS&quot;</span>)</span>
<span id="cb285-3"><a href="regressions-and-anova-in-r.html#cb285-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb285-4"><a href="regressions-and-anova-in-r.html#cb285-4" aria-hidden="true" tabindex="-1"></a><span class="co"># assign mammals dataset to an object</span></span>
<span id="cb285-5"><a href="regressions-and-anova-in-r.html#cb285-5" aria-hidden="true" tabindex="-1"></a>mammals_raw <span class="ot">&lt;-</span> mammals</span>
<span id="cb285-6"><a href="regressions-and-anova-in-r.html#cb285-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mammals_raw)</span>
<span id="cb285-7"><a href="regressions-and-anova-in-r.html#cb285-7" aria-hidden="true" tabindex="-1"></a><span class="co">#                   body brain</span></span>
<span id="cb285-8"><a href="regressions-and-anova-in-r.html#cb285-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Arctic fox        3.385  44.5</span></span>
<span id="cb285-9"><a href="regressions-and-anova-in-r.html#cb285-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Owl monkey        0.480  15.5</span></span>
<span id="cb285-10"><a href="regressions-and-anova-in-r.html#cb285-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Mountain beaver   1.350   8.1</span></span>
<span id="cb285-11"><a href="regressions-and-anova-in-r.html#cb285-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Cow             465.000 423.0</span></span>
<span id="cb285-12"><a href="regressions-and-anova-in-r.html#cb285-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Grey wolf        36.330 119.5</span></span>
<span id="cb285-13"><a href="regressions-and-anova-in-r.html#cb285-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Goat             27.660 115.0</span></span></code></pre></div>
<p>We can explore the relationship of <code>body</code>as predictor of <code>brain</code> by plotting their one-to-one relationship. Notice the formula <code>brain ~ body</code> inside the regression function <code>lm()</code> that correspond <code>brain</code> (y-axis) and <code>body</code> (x-axis).</p>
<div class="sourceCode" id="cb286"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb286-1"><a href="regressions-and-anova-in-r.html#cb286-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="at">x =</span> mammals_raw<span class="sc">$</span>body, <span class="at">y =</span> mammals_raw<span class="sc">$</span>brain)</span>
<span id="cb286-2"><a href="regressions-and-anova-in-r.html#cb286-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="at">side=</span><span class="dv">3</span>, <span class="at">line=</span><span class="fl">0.5</span>, <span class="at">adj=</span><span class="dv">0</span>, <span class="st">&quot;raw data from mammal -- body vs brain&quot;</span>)</span>
<span id="cb286-3"><a href="regressions-and-anova-in-r.html#cb286-3" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="fu">lm</span>(brain <span class="sc">~</span> body, <span class="at">data =</span> mammals_raw), <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb286-4"><a href="regressions-and-anova-in-r.html#cb286-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb286-5"><a href="regressions-and-anova-in-r.html#cb286-5" aria-hidden="true" tabindex="-1"></a><span class="co"># let&#39;s save this model to an object</span></span>
<span id="cb286-6"><a href="regressions-and-anova-in-r.html#cb286-6" aria-hidden="true" tabindex="-1"></a>raw_brain_body <span class="ot">&lt;-</span> <span class="fu">lm</span>(brain <span class="sc">~</span> body, <span class="at">data =</span> mammals_raw)</span></code></pre></div>
<p><img src="figures/session_7_figs/3_body_vs_brain_raw.png" width="70%" /></p>
<p>This plot suggest that this might not be quite linear relationship. The reason is that well-known logistic relationship. For this transformation, we can use the function <code>modify()</code> of the package <strong>purrr</strong></p>
<div class="sourceCode" id="cb287"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb287-1"><a href="regressions-and-anova-in-r.html#cb287-1" aria-hidden="true" tabindex="-1"></a><span class="co"># we have already downloaded this function before</span></span>
<span id="cb287-2"><a href="regressions-and-anova-in-r.html#cb287-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb287-3"><a href="regressions-and-anova-in-r.html#cb287-3" aria-hidden="true" tabindex="-1"></a>mammals_log <span class="ot">&lt;-</span> <span class="fu">modify</span>(mammals_raw,log10)</span>
<span id="cb287-4"><a href="regressions-and-anova-in-r.html#cb287-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mammals_log)</span>
<span id="cb287-5"><a href="regressions-and-anova-in-r.html#cb287-5" aria-hidden="true" tabindex="-1"></a><span class="co">#                     body    brain</span></span>
<span id="cb287-6"><a href="regressions-and-anova-in-r.html#cb287-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Arctic fox       0.5295587 1.648360</span></span>
<span id="cb287-7"><a href="regressions-and-anova-in-r.html#cb287-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Owl monkey      -0.3187588 1.190332</span></span>
<span id="cb287-8"><a href="regressions-and-anova-in-r.html#cb287-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Mountain beaver  0.1303338 0.908485</span></span>
<span id="cb287-9"><a href="regressions-and-anova-in-r.html#cb287-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Cow              2.6674530 2.626340</span></span>
<span id="cb287-10"><a href="regressions-and-anova-in-r.html#cb287-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Grey wolf        1.5602654 2.077368</span></span>
<span id="cb287-11"><a href="regressions-and-anova-in-r.html#cb287-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Goat             1.4418522 2.060698</span></span>
<span id="cb287-12"><a href="regressions-and-anova-in-r.html#cb287-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb287-13"><a href="regressions-and-anova-in-r.html#cb287-13" aria-hidden="true" tabindex="-1"></a><span class="co"># now we can plot this transformed data</span></span>
<span id="cb287-14"><a href="regressions-and-anova-in-r.html#cb287-14" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="at">x =</span> mammals_log<span class="sc">$</span>body, <span class="at">y =</span> mammals_log<span class="sc">$</span>brain)</span>
<span id="cb287-15"><a href="regressions-and-anova-in-r.html#cb287-15" aria-hidden="true" tabindex="-1"></a><span class="fu">mtext</span>(<span class="at">side=</span><span class="dv">3</span>, <span class="at">line=</span><span class="fl">0.5</span>, <span class="at">adj=</span><span class="dv">0</span>, <span class="st">&quot;Log10 data from mammal -- body vs brain&quot;</span>)</span>
<span id="cb287-16"><a href="regressions-and-anova-in-r.html#cb287-16" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="fu">lm</span>(brain <span class="sc">~</span> body, <span class="at">data =</span> mammals_log), <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>)</span>
<span id="cb287-17"><a href="regressions-and-anova-in-r.html#cb287-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb287-18"><a href="regressions-and-anova-in-r.html#cb287-18" aria-hidden="true" tabindex="-1"></a><span class="co"># let&#39;s save this model to an object</span></span>
<span id="cb287-19"><a href="regressions-and-anova-in-r.html#cb287-19" aria-hidden="true" tabindex="-1"></a>log_brain_body <span class="ot">&lt;-</span> <span class="fu">lm</span>(brain <span class="sc">~</span> body, <span class="at">data =</span> mammals_log)</span>
<span id="cb287-20"><a href="regressions-and-anova-in-r.html#cb287-20" aria-hidden="true" tabindex="-1"></a><span class="co"># class of this regression object is &quot;lm&quot;.</span></span>
<span id="cb287-21"><a href="regressions-and-anova-in-r.html#cb287-21" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(log_brain_body)</span>
<span id="cb287-22"><a href="regressions-and-anova-in-r.html#cb287-22" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;lm&quot;</span></span></code></pre></div>
<p><img src="figures/session_7_figs/4_body_vs_brain_log10.png" width="70%" /></p>
<p>This is a better plot and it is easier to visualize the linear trend in the log10 transformed data for body and brain size.</p>
<p><strong>2)</strong> We can summarize the results of these simple linear regressions with the function <code>summary()</code>.</p>
<p>First, we need to understand significance values and symbols (i.e., the asterisks <code>*</code>) next to sections of the output</p>
<table>
<colgroup>
<col width="21%" />
<col width="37%" />
<col width="40%" />
</colgroup>
<thead>
<tr class="header">
<th>Symbol</th>
<th>p-value range</th>
<th>Interpretation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>***</td>
<td>less than 0.001</td>
<td>strongly significant and very likely to reject null (e.g., coefficient different than 0)</td>
</tr>
<tr class="even">
<td>**</td>
<td>0.001 &lt; p-value &lt; 0.01</td>
<td>significant and most likely to reject null (e.g., coefficient different than 0)</td>
</tr>
<tr class="odd">
<td>*</td>
<td>0.05 &lt; p-value &lt; 0.1</td>
<td>barely significant and likely to reject null (e.g., coefficient different than 0)</td>
</tr>
<tr class="even">
<td>.</td>
<td>p-value &gt; 0.05</td>
<td>likely not significant and might not reject null (e.g., coefficient different than 0)</td>
</tr>
<tr class="odd">
<td>no- asterisks</td>
<td>p-value &gt; 0.05</td>
<td>does not reject null (e.g., coefficient is not different than 0)</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb288-1"><a href="regressions-and-anova-in-r.html#cb288-1" aria-hidden="true" tabindex="-1"></a><span class="do">## summary from untransformed data</span></span>
<span id="cb288-2"><a href="regressions-and-anova-in-r.html#cb288-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(raw_brain_body)</span>
<span id="cb288-3"><a href="regressions-and-anova-in-r.html#cb288-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Call:</span></span>
<span id="cb288-4"><a href="regressions-and-anova-in-r.html#cb288-4" aria-hidden="true" tabindex="-1"></a><span class="co">#lm(formula = brain ~ body, data = mammals_raw)</span></span>
<span id="cb288-5"><a href="regressions-and-anova-in-r.html#cb288-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb288-6"><a href="regressions-and-anova-in-r.html#cb288-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Residuals:</span></span>
<span id="cb288-7"><a href="regressions-and-anova-in-r.html#cb288-7" aria-hidden="true" tabindex="-1"></a><span class="co">#    Min      1Q  Median      3Q     Max </span></span>
<span id="cb288-8"><a href="regressions-and-anova-in-r.html#cb288-8" aria-hidden="true" tabindex="-1"></a><span class="co">#-810.07  -88.52  -79.64  -13.02 2050.33 </span></span>
<span id="cb288-9"><a href="regressions-and-anova-in-r.html#cb288-9" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb288-10"><a href="regressions-and-anova-in-r.html#cb288-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Coefficients:</span></span>
<span id="cb288-11"><a href="regressions-and-anova-in-r.html#cb288-11" aria-hidden="true" tabindex="-1"></a><span class="co">#            Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span id="cb288-12"><a href="regressions-and-anova-in-r.html#cb288-12" aria-hidden="true" tabindex="-1"></a><span class="co">#(Intercept) 91.00440   43.55258    2.09   0.0409 *  </span></span>
<span id="cb288-13"><a href="regressions-and-anova-in-r.html#cb288-13" aria-hidden="true" tabindex="-1"></a><span class="co">#body         0.96650    0.04766   20.28   &lt;2e-16 ***</span></span>
<span id="cb288-14"><a href="regressions-and-anova-in-r.html#cb288-14" aria-hidden="true" tabindex="-1"></a><span class="co">#---</span></span>
<span id="cb288-15"><a href="regressions-and-anova-in-r.html#cb288-15" aria-hidden="true" tabindex="-1"></a><span class="co">#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1</span></span>
<span id="cb288-16"><a href="regressions-and-anova-in-r.html#cb288-16" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb288-17"><a href="regressions-and-anova-in-r.html#cb288-17" aria-hidden="true" tabindex="-1"></a><span class="co">#Residual standard error: 334.7 on 60 degrees of freedom</span></span>
<span id="cb288-18"><a href="regressions-and-anova-in-r.html#cb288-18" aria-hidden="true" tabindex="-1"></a><span class="co">#Multiple R-squared:  0.8727,   Adjusted R-squared:  0.8705 </span></span>
<span id="cb288-19"><a href="regressions-and-anova-in-r.html#cb288-19" aria-hidden="true" tabindex="-1"></a><span class="co">#F-statistic: 411.2 on 1 and 60 DF,  p-value: &lt; 2.2e-16</span></span>
<span id="cb288-20"><a href="regressions-and-anova-in-r.html#cb288-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb288-21"><a href="regressions-and-anova-in-r.html#cb288-21" aria-hidden="true" tabindex="-1"></a><span class="do">## summary from log10 transformed data</span></span>
<span id="cb288-22"><a href="regressions-and-anova-in-r.html#cb288-22" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(log_brain_body)</span>
<span id="cb288-23"><a href="regressions-and-anova-in-r.html#cb288-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Call:</span></span>
<span id="cb288-24"><a href="regressions-and-anova-in-r.html#cb288-24" aria-hidden="true" tabindex="-1"></a><span class="co">#lm(formula = brain ~ body, data = mammals_log)</span></span>
<span id="cb288-25"><a href="regressions-and-anova-in-r.html#cb288-25" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb288-26"><a href="regressions-and-anova-in-r.html#cb288-26" aria-hidden="true" tabindex="-1"></a><span class="co">#Residuals:</span></span>
<span id="cb288-27"><a href="regressions-and-anova-in-r.html#cb288-27" aria-hidden="true" tabindex="-1"></a><span class="co">#     Min       1Q   Median       3Q      Max </span></span>
<span id="cb288-28"><a href="regressions-and-anova-in-r.html#cb288-28" aria-hidden="true" tabindex="-1"></a><span class="co">#-0.74503 -0.21380 -0.02676  0.18934  0.84613 </span></span>
<span id="cb288-29"><a href="regressions-and-anova-in-r.html#cb288-29" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb288-30"><a href="regressions-and-anova-in-r.html#cb288-30" aria-hidden="true" tabindex="-1"></a><span class="co">#Coefficients:</span></span>
<span id="cb288-31"><a href="regressions-and-anova-in-r.html#cb288-31" aria-hidden="true" tabindex="-1"></a><span class="co">#            Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span id="cb288-32"><a href="regressions-and-anova-in-r.html#cb288-32" aria-hidden="true" tabindex="-1"></a><span class="co">#(Intercept)  0.92713    0.04171   22.23   &lt;2e-16 ***</span></span>
<span id="cb288-33"><a href="regressions-and-anova-in-r.html#cb288-33" aria-hidden="true" tabindex="-1"></a><span class="co">#body         0.75169    0.02846   26.41   &lt;2e-16 ***</span></span>
<span id="cb288-34"><a href="regressions-and-anova-in-r.html#cb288-34" aria-hidden="true" tabindex="-1"></a><span class="co">#---</span></span>
<span id="cb288-35"><a href="regressions-and-anova-in-r.html#cb288-35" aria-hidden="true" tabindex="-1"></a><span class="co">#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1</span></span>
<span id="cb288-36"><a href="regressions-and-anova-in-r.html#cb288-36" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb288-37"><a href="regressions-and-anova-in-r.html#cb288-37" aria-hidden="true" tabindex="-1"></a><span class="co">#Residual standard error: 0.3015 on 60 degrees of freedom</span></span>
<span id="cb288-38"><a href="regressions-and-anova-in-r.html#cb288-38" aria-hidden="true" tabindex="-1"></a><span class="co">#Multiple R-squared:  0.9208,   Adjusted R-squared:  0.9195 </span></span>
<span id="cb288-39"><a href="regressions-and-anova-in-r.html#cb288-39" aria-hidden="true" tabindex="-1"></a><span class="co">#F-statistic: 697.4 on 1 and 60 DF,  p-value: &lt; 2.2e-16</span></span></code></pre></div>
<p>Some of the highlights of the <code>raw_brain_body</code> output can be as follows.</p>
<table>
<colgroup>
<col width="18%" />
<col width="21%" />
<col width="19%" />
<col width="17%" />
<col width="8%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th>Coefficient name</th>
<th>Coefficient value</th>
<th>Statistic name</th>
<th>Statistic value</th>
<th>p-value</th>
<th>Interpretation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td></td>
<td></td>
<td><em>F</em></td>
<td>411.2</td>
<td>&lt; 2.2e-16 ***</td>
<td>the model is strongly significant</td>
</tr>
<tr class="even">
<td>Intercept</td>
<td>91.00440</td>
<td><em>t</em>-value</td>
<td>2.09</td>
<td>0.0409 *</td>
<td>Intercept coefficient is barely significant</td>
</tr>
<tr class="odd">
<td>Body</td>
<td>0.96650</td>
<td><em>t</em>-value</td>
<td>20.28</td>
<td>&lt; 2.2e-16 ***</td>
<td>Body coefficient is strongly significant</td>
</tr>
<tr class="even">
<td>R-squared</td>
<td>0.8727</td>
<td></td>
<td></td>
<td></td>
<td>the value is close to 1.0 and model is useful</td>
</tr>
</tbody>
</table>
<p>Some of the highlights of the <code>log_brain_body</code> output can be as follows.</p>
<table>
<colgroup>
<col width="18%" />
<col width="21%" />
<col width="19%" />
<col width="17%" />
<col width="8%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th>Coefficient name</th>
<th>Coefficient value</th>
<th>Statistic name</th>
<th>Statistic value</th>
<th>p-value</th>
<th>Interpretation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td></td>
<td></td>
<td><em>F</em></td>
<td>697.4</td>
<td>&lt; 2.2e-16 ***</td>
<td>the model is strongly significant</td>
</tr>
<tr class="even">
<td>Intercept</td>
<td>0.92713</td>
<td><em>t</em>-value</td>
<td>22.23</td>
<td>&lt; 2.2e-16 ***</td>
<td>Intercept coefficient is strongly significant</td>
</tr>
<tr class="odd">
<td>Body</td>
<td>0.75169</td>
<td><em>t</em>-value</td>
<td>26.41</td>
<td>&lt; 2.2e-16 ***</td>
<td>body coefficient is strongly significant</td>
</tr>
<tr class="even">
<td>R-squared</td>
<td>0.9208</td>
<td></td>
<td></td>
<td></td>
<td>the value is close to 1.0 and model is useful</td>
</tr>
</tbody>
</table>
<p>We can conclude that both models are significant and useful, but the log10 transformed data seems to fit better a linear model (e.g., see intercept row).</p>
<p><strong>3)</strong> We can expand our regression models to more than one predictor (i.e., multiple linear regression) that can be performed again using the function <code>lm()</code>. In this case, we have to multiple vectors: one response and many predictors.</p>
<p>For this example, we will use the dataset <code>crabs</code> that has morphological measurements on <em>Leptograpsus</em> crabs from R-package <strong>MASS</strong>. The variables that included: <strong>sp</strong> is species - “B” or “O” for blue or orange, <strong>sex</strong>, <strong>index</strong> is index 1:50 within each of the four groups, <strong>FL</strong> is frontal lobe size (mm), <strong>RW</strong> is rear width (mm), <strong>CL</strong> is carapace length (mm), <strong>CW</strong> is carapace width (mm) and <strong>BD</strong> is body depth (mm).</p>
<div class="sourceCode" id="cb289"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb289-1"><a href="regressions-and-anova-in-r.html#cb289-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we have already downloaded MASS and we assign the crabs dataset to an object</span></span>
<span id="cb289-2"><a href="regressions-and-anova-in-r.html#cb289-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb289-3"><a href="regressions-and-anova-in-r.html#cb289-3" aria-hidden="true" tabindex="-1"></a>crab_dataset <span class="ot">&lt;-</span> crabs</span>
<span id="cb289-4"><a href="regressions-and-anova-in-r.html#cb289-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(crab_dataset)</span>
<span id="cb289-5"><a href="regressions-and-anova-in-r.html#cb289-5" aria-hidden="true" tabindex="-1"></a><span class="co">#  sp sex index   FL  RW   CL   CW  BD</span></span>
<span id="cb289-6"><a href="regressions-and-anova-in-r.html#cb289-6" aria-hidden="true" tabindex="-1"></a><span class="co">#1  B   M     1  8.1 6.7 16.1 19.0 7.0</span></span>
<span id="cb289-7"><a href="regressions-and-anova-in-r.html#cb289-7" aria-hidden="true" tabindex="-1"></a><span class="co">#2  B   M     2  8.8 7.7 18.1 20.8 7.4</span></span>
<span id="cb289-8"><a href="regressions-and-anova-in-r.html#cb289-8" aria-hidden="true" tabindex="-1"></a><span class="co">#3  B   M     3  9.2 7.8 19.0 22.4 7.7</span></span>
<span id="cb289-9"><a href="regressions-and-anova-in-r.html#cb289-9" aria-hidden="true" tabindex="-1"></a><span class="co">#4  B   M     4  9.6 7.9 20.1 23.1 8.2</span></span>
<span id="cb289-10"><a href="regressions-and-anova-in-r.html#cb289-10" aria-hidden="true" tabindex="-1"></a><span class="co">#5  B   M     5  9.8 8.0 20.3 23.0 8.2</span></span>
<span id="cb289-11"><a href="regressions-and-anova-in-r.html#cb289-11" aria-hidden="true" tabindex="-1"></a><span class="co">#6  B   M     6 10.8 9.0 23.0 26.5 9.8</span></span>
<span id="cb289-12"><a href="regressions-and-anova-in-r.html#cb289-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb289-13"><a href="regressions-and-anova-in-r.html#cb289-13" aria-hidden="true" tabindex="-1"></a><span class="do">## we will use BD as dependent and FL, RW, CL and CW as predictors.</span></span>
<span id="cb289-14"><a href="regressions-and-anova-in-r.html#cb289-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb289-15"><a href="regressions-and-anova-in-r.html#cb289-15" aria-hidden="true" tabindex="-1"></a>crab_multiple_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(<span class="at">formula =</span> BD <span class="sc">~</span> FL<span class="sc">+</span>RW<span class="sc">+</span>CL<span class="sc">+</span>CW, <span class="at">data =</span> crab_dataset)</span>
<span id="cb289-16"><a href="regressions-and-anova-in-r.html#cb289-16" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(crab_multiple_lm)</span>
<span id="cb289-17"><a href="regressions-and-anova-in-r.html#cb289-17" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;lm&quot;</span></span>
<span id="cb289-18"><a href="regressions-and-anova-in-r.html#cb289-18" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(crab_multiple_lm)</span>
<span id="cb289-19"><a href="regressions-and-anova-in-r.html#cb289-19" aria-hidden="true" tabindex="-1"></a><span class="co">#Call:</span></span>
<span id="cb289-20"><a href="regressions-and-anova-in-r.html#cb289-20" aria-hidden="true" tabindex="-1"></a><span class="co">#lm(formula = BD ~ FL + RW + CL + CW, data = crab_dataset)</span></span>
<span id="cb289-21"><a href="regressions-and-anova-in-r.html#cb289-21" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb289-22"><a href="regressions-and-anova-in-r.html#cb289-22" aria-hidden="true" tabindex="-1"></a><span class="co">#Residuals:</span></span>
<span id="cb289-23"><a href="regressions-and-anova-in-r.html#cb289-23" aria-hidden="true" tabindex="-1"></a><span class="co">#    Min      1Q  Median      3Q     Max </span></span>
<span id="cb289-24"><a href="regressions-and-anova-in-r.html#cb289-24" aria-hidden="true" tabindex="-1"></a><span class="co">#-1.3518 -0.2291  0.0112  0.2664  1.0965 </span></span>
<span id="cb289-25"><a href="regressions-and-anova-in-r.html#cb289-25" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb289-26"><a href="regressions-and-anova-in-r.html#cb289-26" aria-hidden="true" tabindex="-1"></a><span class="co">#Coefficients:</span></span>
<span id="cb289-27"><a href="regressions-and-anova-in-r.html#cb289-27" aria-hidden="true" tabindex="-1"></a><span class="co">#            Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span id="cb289-28"><a href="regressions-and-anova-in-r.html#cb289-28" aria-hidden="true" tabindex="-1"></a><span class="co">#(Intercept) -0.97616    0.15296  -6.382 1.25e-09 ***</span></span>
<span id="cb289-29"><a href="regressions-and-anova-in-r.html#cb289-29" aria-hidden="true" tabindex="-1"></a><span class="co">#FL           0.42022    0.05585   7.524 1.91e-12 ***</span></span>
<span id="cb289-30"><a href="regressions-and-anova-in-r.html#cb289-30" aria-hidden="true" tabindex="-1"></a><span class="co">#RW           0.03768    0.03209   1.174    0.242    </span></span>
<span id="cb289-31"><a href="regressions-and-anova-in-r.html#cb289-31" aria-hidden="true" tabindex="-1"></a><span class="co">#CL           0.59511    0.06825   8.720 1.22e-15 ***</span></span>
<span id="cb289-32"><a href="regressions-and-anova-in-r.html#cb289-32" aria-hidden="true" tabindex="-1"></a><span class="co">#CW          -0.30560    0.04935  -6.193 3.43e-09 ***</span></span>
<span id="cb289-33"><a href="regressions-and-anova-in-r.html#cb289-33" aria-hidden="true" tabindex="-1"></a><span class="co">#---</span></span>
<span id="cb289-34"><a href="regressions-and-anova-in-r.html#cb289-34" aria-hidden="true" tabindex="-1"></a><span class="co">#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1</span></span>
<span id="cb289-35"><a href="regressions-and-anova-in-r.html#cb289-35" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb289-36"><a href="regressions-and-anova-in-r.html#cb289-36" aria-hidden="true" tabindex="-1"></a><span class="co">#Residual standard error: 0.4208 on 195 degrees of freedom</span></span>
<span id="cb289-37"><a href="regressions-and-anova-in-r.html#cb289-37" aria-hidden="true" tabindex="-1"></a><span class="co">#Multiple R-squared:  0.9852,   Adjusted R-squared:  0.9849 </span></span>
<span id="cb289-38"><a href="regressions-and-anova-in-r.html#cb289-38" aria-hidden="true" tabindex="-1"></a><span class="co">#F-statistic:  3247 on 4 and 195 DF,  p-value: &lt; 2.2e-16</span></span></code></pre></div>
<p>Based on these statistics, multiple regression is significant (<em>F</em>-statistic), the model seems to be useful (Multiple R-squared is very close to 1), all coefficients with the exception for <strong>CL</strong> (carapace length) were significant.</p>
<p><strong>4)</strong> We can perform some diagnostic plots to checks for heteroscedasticity, normality, and influential observerations.</p>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="regressions-and-anova-in-r.html#cb290-1" aria-hidden="true" tabindex="-1"></a><span class="co"># diagnostic plots</span></span>
<span id="cb290-2"><a href="regressions-and-anova-in-r.html#cb290-2" aria-hidden="true" tabindex="-1"></a><span class="fu">layout</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>),<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb290-3"><a href="regressions-and-anova-in-r.html#cb290-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(crab_multiple_lm)</span></code></pre></div>
<p><img src="figures/session_7_figs/5_diagnostic_plots.png" width="70%" /></p>
<p>To understand these plots, we need to look for indications that data might not follow normality or has bias. These include: (a) the <code>Residual vs Fitted</code> should have a random distribution without any pattern (our plot shows that); (b) the <code>Normal Q-Q</code> show fall into oblique dashed line that supports relative multivariate normality (i.e., our multiple linear method is valid) and (c) the plots <code>Scale-Location</code> and <code>Residuals vs Leverage</code> should fall close to the center (i.e., these plots are more or less OK)</p>
<p><strong>5)</strong> You can also get a list of influential measurements using functions <code>influence.measures()</code> and <code>influence()</code>. These measurements might be outliers or errors in data input (i.e., miss typing, incorrect instrument calibration, contamination, etc.)</p>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb291-1"><a href="regressions-and-anova-in-r.html#cb291-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Notice values with an asterisk (*)</span></span>
<span id="cb291-2"><a href="regressions-and-anova-in-r.html#cb291-2" aria-hidden="true" tabindex="-1"></a>influence_values <span class="ot">&lt;-</span> <span class="fu">influence.measures</span>(crab_multiple_lm)</span>
<span id="cb291-3"><a href="regressions-and-anova-in-r.html#cb291-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Influence measures of</span></span>
<span id="cb291-4"><a href="regressions-and-anova-in-r.html#cb291-4" aria-hidden="true" tabindex="-1"></a><span class="co">#    lm(formula = BD ~ FL + RW + CL + CW, data = crab_dataset) :</span></span>
<span id="cb291-5"><a href="regressions-and-anova-in-r.html#cb291-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb291-6"><a href="regressions-and-anova-in-r.html#cb291-6" aria-hidden="true" tabindex="-1"></a><span class="co">#       dfb.1_    dfb.FL    dfb.RW    dfb.CL    dfb.CW     dffit cov.r   cook.d     hat inf</span></span>
<span id="cb291-7"><a href="regressions-and-anova-in-r.html#cb291-7" aria-hidden="true" tabindex="-1"></a><span class="co">#1    0.230383  1.06e-01 -1.27e-01 -1.15e-01  9.97e-02  0.272172 1.022 1.48e-02 0.04052    </span></span>
<span id="cb291-8"><a href="regressions-and-anova-in-r.html#cb291-8" aria-hidden="true" tabindex="-1"></a><span class="co">#2   -0.009954 -4.12e-05  1.55e-03 -6.81e-05  5.21e-04 -0.010550 1.053 2.24e-05 0.02556    </span></span>
<span id="cb291-9"><a href="regressions-and-anova-in-r.html#cb291-9" aria-hidden="true" tabindex="-1"></a><span class="co">#3    0.018286  6.27e-03 -9.58e-03 -9.30e-03  9.08e-03  0.022461 1.056 1.01e-04 0.02866    </span></span>
<span id="cb291-10"><a href="regressions-and-anova-in-r.html#cb291-10" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span>
<span id="cb291-11"><a href="regressions-and-anova-in-r.html#cb291-11" aria-hidden="true" tabindex="-1"></a><span class="co">#46   0.076014 -8.84e-02  2.99e-01  1.25e-01 -1.87e-01 -0.426499 0.914 3.55e-02 0.03051   *</span></span>
<span id="cb291-12"><a href="regressions-and-anova-in-r.html#cb291-12" aria-hidden="true" tabindex="-1"></a><span class="co">#47   0.037663 -1.85e-02  3.83e-02  4.76e-02 -6.15e-02 -0.088701 1.057 1.58e-03 0.03509    </span></span>
<span id="cb291-13"><a href="regressions-and-anova-in-r.html#cb291-13" aria-hidden="true" tabindex="-1"></a><span class="co">#48  -0.027395  8.70e-03 -7.23e-02 -3.70e-02  6.03e-02  0.115458 1.077 2.68e-03 0.05319   *</span></span>
<span id="cb291-14"><a href="regressions-and-anova-in-r.html#cb291-14" aria-hidden="true" tabindex="-1"></a><span class="co">#49  -0.017943  1.88e-02 -4.78e-02 -3.53e-02  4.67e-02  0.071147 1.077 1.02e-03 0.04985   *</span></span>
<span id="cb291-15"><a href="regressions-and-anova-in-r.html#cb291-15" aria-hidden="true" tabindex="-1"></a><span class="co">#50  -0.023874  7.47e-05 -3.10e-02 -2.11e-02  3.51e-02  0.063645 1.110 8.14e-04 0.07675   *</span></span>
<span id="cb291-16"><a href="regressions-and-anova-in-r.html#cb291-16" aria-hidden="true" tabindex="-1"></a><span class="co">#51   0.127156  8.62e-03 -2.53e-02 -8.99e-03  2.53e-03  0.133549 1.052 3.58e-03 0.03666    </span></span>
<span id="cb291-17"><a href="regressions-and-anova-in-r.html#cb291-17" aria-hidden="true" tabindex="-1"></a><span class="co">#32  -0.020620 -6.29e-02 -1.96e-02 -3.45e-03  4.20e-02  0.145671 1.028 4.25e-03 0.02396    </span></span>
<span id="cb291-18"><a href="regressions-and-anova-in-r.html#cb291-18" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span>
<span id="cb291-19"><a href="regressions-and-anova-in-r.html#cb291-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb291-20"><a href="regressions-and-anova-in-r.html#cb291-20" aria-hidden="true" tabindex="-1"></a><span class="do">## now we can get the identity of these values organized in a vector by focusing in column &#39;hat&#39;</span></span>
<span id="cb291-21"><a href="regressions-and-anova-in-r.html#cb291-21" aria-hidden="true" tabindex="-1"></a>influence_df <span class="ot">&lt;-</span> <span class="fu">influence</span>(crab_multiple_lm)</span>
<span id="cb291-22"><a href="regressions-and-anova-in-r.html#cb291-22" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(influence_df)</span>
<span id="cb291-23"><a href="regressions-and-anova-in-r.html#cb291-23" aria-hidden="true" tabindex="-1"></a><span class="co"># we can the 10 most influential value</span></span>
<span id="cb291-24"><a href="regressions-and-anova-in-r.html#cb291-24" aria-hidden="true" tabindex="-1"></a><span class="fu">order</span>(influence_df<span class="sc">$</span>hat, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span>
<span id="cb291-25"><a href="regressions-and-anova-in-r.html#cb291-25" aria-hidden="true" tabindex="-1"></a><span class="co">#[1]  50 145 199 195  98 200  99 144 186  97</span></span></code></pre></div>
<p><strong>6)</strong> We can further explore our simple and multiple regression examples by calling different regression statistics. We have two regression objects <code>log_brain_body</code> and <code>crab_multiple_lm</code>.</p>
<p>ANOVA table, we will explore this later in this session. We use the function <code>anova()</code>.</p>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="regressions-and-anova-in-r.html#cb292-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(log_brain_body)</span>
<span id="cb292-2"><a href="regressions-and-anova-in-r.html#cb292-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Analysis of Variance Table</span></span>
<span id="cb292-3"><a href="regressions-and-anova-in-r.html#cb292-3" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb292-4"><a href="regressions-and-anova-in-r.html#cb292-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Response: brain</span></span>
<span id="cb292-5"><a href="regressions-and-anova-in-r.html#cb292-5" aria-hidden="true" tabindex="-1"></a><span class="co">#          Df Sum Sq Mean Sq F value    Pr(&gt;F)    </span></span>
<span id="cb292-6"><a href="regressions-and-anova-in-r.html#cb292-6" aria-hidden="true" tabindex="-1"></a><span class="co">#body       1 63.409  63.409  697.42 &lt; 2.2e-16 ***</span></span>
<span id="cb292-7"><a href="regressions-and-anova-in-r.html#cb292-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Residuals 60  5.455   0.091                      </span></span>
<span id="cb292-8"><a href="regressions-and-anova-in-r.html#cb292-8" aria-hidden="true" tabindex="-1"></a><span class="co">#---</span></span>
<span id="cb292-9"><a href="regressions-and-anova-in-r.html#cb292-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1</span></span>
<span id="cb292-10"><a href="regressions-and-anova-in-r.html#cb292-10" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(crab_multiple_lm)</span>
<span id="cb292-11"><a href="regressions-and-anova-in-r.html#cb292-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Analysis of Variance Table</span></span>
<span id="cb292-12"><a href="regressions-and-anova-in-r.html#cb292-12" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb292-13"><a href="regressions-and-anova-in-r.html#cb292-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Response: BD</span></span>
<span id="cb292-14"><a href="regressions-and-anova-in-r.html#cb292-14" aria-hidden="true" tabindex="-1"></a><span class="co">#           Df  Sum Sq Mean Sq    F value    Pr(&gt;F)    </span></span>
<span id="cb292-15"><a href="regressions-and-anova-in-r.html#cb292-15" aria-hidden="true" tabindex="-1"></a><span class="co">#FL          1 2276.68 2276.68 12858.0018 &lt; 2.2e-16 ***</span></span>
<span id="cb292-16"><a href="regressions-and-anova-in-r.html#cb292-16" aria-hidden="true" tabindex="-1"></a><span class="co">#RW          1    0.57    0.57     3.1984   0.07526 .  </span></span>
<span id="cb292-17"><a href="regressions-and-anova-in-r.html#cb292-17" aria-hidden="true" tabindex="-1"></a><span class="co">#CL          1   15.52   15.52    87.6332 &lt; 2.2e-16 ***</span></span>
<span id="cb292-18"><a href="regressions-and-anova-in-r.html#cb292-18" aria-hidden="true" tabindex="-1"></a><span class="co">#CW          1    6.79    6.79    38.3516  3.43e-09 ***</span></span>
<span id="cb292-19"><a href="regressions-and-anova-in-r.html#cb292-19" aria-hidden="true" tabindex="-1"></a><span class="co">#Residuals 195   34.53    0.18                         </span></span>
<span id="cb292-20"><a href="regressions-and-anova-in-r.html#cb292-20" aria-hidden="true" tabindex="-1"></a><span class="co">#---</span></span>
<span id="cb292-21"><a href="regressions-and-anova-in-r.html#cb292-21" aria-hidden="true" tabindex="-1"></a><span class="co">#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1</span></span></code></pre></div>
<p>Get the model coefficients using the function <code>coefficients()</code> or <code>coef()</code>.</p>
<div class="sourceCode" id="cb293"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb293-1"><a href="regressions-and-anova-in-r.html#cb293-1" aria-hidden="true" tabindex="-1"></a><span class="fu">coefficients</span>(log_brain_body)</span>
<span id="cb293-2"><a href="regressions-and-anova-in-r.html#cb293-2" aria-hidden="true" tabindex="-1"></a><span class="co">#(Intercept)        body </span></span>
<span id="cb293-3"><a href="regressions-and-anova-in-r.html#cb293-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  0.9271269   0.7516859 </span></span>
<span id="cb293-4"><a href="regressions-and-anova-in-r.html#cb293-4" aria-hidden="true" tabindex="-1"></a><span class="fu">coef</span>(crab_multiple_lm)</span>
<span id="cb293-5"><a href="regressions-and-anova-in-r.html#cb293-5" aria-hidden="true" tabindex="-1"></a><span class="co">#(Intercept)          FL          RW          CL          CW </span></span>
<span id="cb293-6"><a href="regressions-and-anova-in-r.html#cb293-6" aria-hidden="true" tabindex="-1"></a><span class="co">#-0.97615668  0.42022057  0.03768281  0.59511478 -0.30559587 </span></span></code></pre></div>
<p>Get the model confidence intervals for regression coefficients (i.e., range that contains the estimate regression coefficients) using the function <code>confint()</code>.</p>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="regressions-and-anova-in-r.html#cb294-1" aria-hidden="true" tabindex="-1"></a><span class="fu">confint</span>(log_brain_body)</span>
<span id="cb294-2"><a href="regressions-and-anova-in-r.html#cb294-2" aria-hidden="true" tabindex="-1"></a>                <span class="fl">2.5</span> <span class="sc">%    97.5 %</span></span>
<span id="cb294-3"><a href="regressions-and-anova-in-r.html#cb294-3" aria-hidden="true" tabindex="-1"></a><span class="co">#(Intercept) 0.8436923 1.0105616</span></span>
<span id="cb294-4"><a href="regressions-and-anova-in-r.html#cb294-4" aria-hidden="true" tabindex="-1"></a><span class="co">#body        0.6947503 0.8086215</span></span>
<span id="cb294-5"><a href="regressions-and-anova-in-r.html#cb294-5" aria-hidden="true" tabindex="-1"></a><span class="fu">confint</span>(crab_multiple_lm)</span>
<span id="cb294-6"><a href="regressions-and-anova-in-r.html#cb294-6" aria-hidden="true" tabindex="-1"></a><span class="co">#                  2.5 %     97.5 %</span></span>
<span id="cb294-7"><a href="regressions-and-anova-in-r.html#cb294-7" aria-hidden="true" tabindex="-1"></a><span class="co">#(Intercept) -1.27783361 -0.6744798</span></span>
<span id="cb294-8"><a href="regressions-and-anova-in-r.html#cb294-8" aria-hidden="true" tabindex="-1"></a><span class="co">#FL           0.31007030  0.5303708</span></span>
<span id="cb294-9"><a href="regressions-and-anova-in-r.html#cb294-9" aria-hidden="true" tabindex="-1"></a><span class="co">#RW          -0.02560597  0.1009716</span></span>
<span id="cb294-10"><a href="regressions-and-anova-in-r.html#cb294-10" aria-hidden="true" tabindex="-1"></a><span class="co">#CL           0.46051803  0.7297115</span></span>
<span id="cb294-11"><a href="regressions-and-anova-in-r.html#cb294-11" aria-hidden="true" tabindex="-1"></a><span class="co">#CW          -0.40291710 -0.2082746</span></span></code></pre></div>
<p>Notice the confidence interval for <code>RW</code> is (-0.02560597, 0.1009716) and this range contains <code>0.0</code> (i.e., zero). This is consistent with the <code>summary(crab_multiple_lm)</code> where <code>RW</code> was not significant, see <strong>3)</strong>.</p>
</div>
<div id="selecting-predictors" class="section level2" number="7.2">
<h2><span class="header-section-number">7.2</span> Selecting predictors</h2>
<p><strong>7)</strong> We can further improve our regression by selecting the best predictors and removing those that do not contribute to the regression. The function <code>step()</code> can help you to determine what predictors to keep .</p>
<p>We will illustrate this using our model <code>crab_multiple_lm</code>.</p>
<div class="sourceCode" id="cb295"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb295-1"><a href="regressions-and-anova-in-r.html#cb295-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we will assign this as our full model</span></span>
<span id="cb295-2"><a href="regressions-and-anova-in-r.html#cb295-2" aria-hidden="true" tabindex="-1"></a>full_model <span class="ot">&lt;-</span> crab_multiple_lm</span>
<span id="cb295-3"><a href="regressions-and-anova-in-r.html#cb295-3" aria-hidden="true" tabindex="-1"></a>full_model</span>
<span id="cb295-4"><a href="regressions-and-anova-in-r.html#cb295-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Call:</span></span>
<span id="cb295-5"><a href="regressions-and-anova-in-r.html#cb295-5" aria-hidden="true" tabindex="-1"></a><span class="co">#lm(formula = BD ~ FL + RW + CL + CW, data = crab_dataset)</span></span>
<span id="cb295-6"><a href="regressions-and-anova-in-r.html#cb295-6" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb295-7"><a href="regressions-and-anova-in-r.html#cb295-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Coefficients:</span></span>
<span id="cb295-8"><a href="regressions-and-anova-in-r.html#cb295-8" aria-hidden="true" tabindex="-1"></a><span class="co">#(Intercept)           FL           RW           CL           CW  </span></span>
<span id="cb295-9"><a href="regressions-and-anova-in-r.html#cb295-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  -0.97616      0.42022      0.03768      0.59511     -0.30560  </span></span></code></pre></div>
<p>We can do this using backward or both approaches (they usually reach the same conclusion).</p>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="regressions-and-anova-in-r.html#cb296-1" aria-hidden="true" tabindex="-1"></a>reduced_model <span class="ot">&lt;-</span> <span class="fu">step</span>(full_model, <span class="at">direction =</span> <span class="st">&quot;backward&quot;</span>)</span>
<span id="cb296-2"><a href="regressions-and-anova-in-r.html#cb296-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Start:  AIC=-341.31</span></span>
<span id="cb296-3"><a href="regressions-and-anova-in-r.html#cb296-3" aria-hidden="true" tabindex="-1"></a><span class="co">#BD ~ FL + RW + CL + CW</span></span>
<span id="cb296-4"><a href="regressions-and-anova-in-r.html#cb296-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb296-5"><a href="regressions-and-anova-in-r.html#cb296-5" aria-hidden="true" tabindex="-1"></a><span class="co">#       Df Sum of Sq    RSS     AIC</span></span>
<span id="cb296-6"><a href="regressions-and-anova-in-r.html#cb296-6" aria-hidden="true" tabindex="-1"></a><span class="co">#- RW    1    0.2442 34.772 -341.90</span></span>
<span id="cb296-7"><a href="regressions-and-anova-in-r.html#cb296-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&lt;none&gt;              34.527 -341.31</span></span>
<span id="cb296-8"><a href="regressions-and-anova-in-r.html#cb296-8" aria-hidden="true" tabindex="-1"></a><span class="co">#- CW    1    6.7907 41.318 -307.40</span></span>
<span id="cb296-9"><a href="regressions-and-anova-in-r.html#cb296-9" aria-hidden="true" tabindex="-1"></a><span class="co">#- FL    1   10.0234 44.551 -292.34</span></span>
<span id="cb296-10"><a href="regressions-and-anova-in-r.html#cb296-10" aria-hidden="true" tabindex="-1"></a><span class="co">#- CL    1   13.4637 47.991 -277.46</span></span>
<span id="cb296-11"><a href="regressions-and-anova-in-r.html#cb296-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb296-12"><a href="regressions-and-anova-in-r.html#cb296-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Step:  AIC=-341.9</span></span>
<span id="cb296-13"><a href="regressions-and-anova-in-r.html#cb296-13" aria-hidden="true" tabindex="-1"></a><span class="co">#BD ~ FL + CL + CW</span></span>
<span id="cb296-14"><a href="regressions-and-anova-in-r.html#cb296-14" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb296-15"><a href="regressions-and-anova-in-r.html#cb296-15" aria-hidden="true" tabindex="-1"></a><span class="co">#       Df Sum of Sq    RSS     AIC</span></span>
<span id="cb296-16"><a href="regressions-and-anova-in-r.html#cb296-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&lt;none&gt;              34.772 -341.90</span></span>
<span id="cb296-17"><a href="regressions-and-anova-in-r.html#cb296-17" aria-hidden="true" tabindex="-1"></a><span class="co">#- CW    1    7.5018 42.273 -304.83</span></span>
<span id="cb296-18"><a href="regressions-and-anova-in-r.html#cb296-18" aria-hidden="true" tabindex="-1"></a><span class="co">#- CL    1   15.2099 49.981 -271.33</span></span>
<span id="cb296-19"><a href="regressions-and-anova-in-r.html#cb296-19" aria-hidden="true" tabindex="-1"></a><span class="co">#- FL    1   17.0716 51.843 -264.02</span></span>
<span id="cb296-20"><a href="regressions-and-anova-in-r.html#cb296-20" aria-hidden="true" tabindex="-1"></a>reduced_model</span>
<span id="cb296-21"><a href="regressions-and-anova-in-r.html#cb296-21" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb296-22"><a href="regressions-and-anova-in-r.html#cb296-22" aria-hidden="true" tabindex="-1"></a><span class="co">#Call:</span></span>
<span id="cb296-23"><a href="regressions-and-anova-in-r.html#cb296-23" aria-hidden="true" tabindex="-1"></a><span class="co">#lm(formula = BD ~ FL + CL + CW, data = crab_dataset)</span></span>
<span id="cb296-24"><a href="regressions-and-anova-in-r.html#cb296-24" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb296-25"><a href="regressions-and-anova-in-r.html#cb296-25" aria-hidden="true" tabindex="-1"></a><span class="co">#Coefficients:</span></span>
<span id="cb296-26"><a href="regressions-and-anova-in-r.html#cb296-26" aria-hidden="true" tabindex="-1"></a><span class="co">#(Intercept)           FL           CL           CW  </span></span></code></pre></div>
<p>We can notice that the predictor <code>RW</code> has been dropped from the reduced model and we can get some statistics on this <strong>improved</strong> model.</p>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="regressions-and-anova-in-r.html#cb297-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(reduced_model)</span>
<span id="cb297-2"><a href="regressions-and-anova-in-r.html#cb297-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Call:</span></span>
<span id="cb297-3"><a href="regressions-and-anova-in-r.html#cb297-3" aria-hidden="true" tabindex="-1"></a><span class="co">#lm(formula = BD ~ FL + CL + CW, data = crab_dataset)</span></span>
<span id="cb297-4"><a href="regressions-and-anova-in-r.html#cb297-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb297-5"><a href="regressions-and-anova-in-r.html#cb297-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Residuals:</span></span>
<span id="cb297-6"><a href="regressions-and-anova-in-r.html#cb297-6" aria-hidden="true" tabindex="-1"></a><span class="co">#     Min       1Q   Median       3Q      Max </span></span>
<span id="cb297-7"><a href="regressions-and-anova-in-r.html#cb297-7" aria-hidden="true" tabindex="-1"></a><span class="co">#-1.34264 -0.23109  0.00893  0.26523  1.05895 </span></span>
<span id="cb297-8"><a href="regressions-and-anova-in-r.html#cb297-8" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb297-9"><a href="regressions-and-anova-in-r.html#cb297-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Coefficients:</span></span>
<span id="cb297-10"><a href="regressions-and-anova-in-r.html#cb297-10" aria-hidden="true" tabindex="-1"></a><span class="co">#            Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span id="cb297-11"><a href="regressions-and-anova-in-r.html#cb297-11" aria-hidden="true" tabindex="-1"></a><span class="co">#(Intercept) -0.92444    0.14663  -6.305 1.87e-09 ***</span></span>
<span id="cb297-12"><a href="regressions-and-anova-in-r.html#cb297-12" aria-hidden="true" tabindex="-1"></a><span class="co">#FL           0.45656    0.04654   9.810  &lt; 2e-16 ***</span></span>
<span id="cb297-13"><a href="regressions-and-anova-in-r.html#cb297-13" aria-hidden="true" tabindex="-1"></a><span class="co">#CL           0.55718    0.06018   9.259  &lt; 2e-16 ***</span></span>
<span id="cb297-14"><a href="regressions-and-anova-in-r.html#cb297-14" aria-hidden="true" tabindex="-1"></a><span class="co">#CW          -0.27594    0.04243  -6.503 6.41e-10 ***</span></span>
<span id="cb297-15"><a href="regressions-and-anova-in-r.html#cb297-15" aria-hidden="true" tabindex="-1"></a><span class="co">#---</span></span>
<span id="cb297-16"><a href="regressions-and-anova-in-r.html#cb297-16" aria-hidden="true" tabindex="-1"></a><span class="co">#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1</span></span>
<span id="cb297-17"><a href="regressions-and-anova-in-r.html#cb297-17" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb297-18"><a href="regressions-and-anova-in-r.html#cb297-18" aria-hidden="true" tabindex="-1"></a><span class="co">#Residual standard error: 0.4212 on 196 degrees of freedom</span></span>
<span id="cb297-19"><a href="regressions-and-anova-in-r.html#cb297-19" aria-hidden="true" tabindex="-1"></a><span class="co">#Multiple R-squared:  0.9851,   Adjusted R-squared:  0.9849 </span></span>
<span id="cb297-20"><a href="regressions-and-anova-in-r.html#cb297-20" aria-hidden="true" tabindex="-1"></a><span class="co">#F-statistic:  4320 on 3 and 196 DF,  p-value: &lt; 2.2e-16</span></span></code></pre></div>
<p><strong>8)</strong> One of the most important uses of regressions is predictions of the dependent variable. For example, given a set of measurements on your predictors, can you predict an outcome. This can be done using the function <code>predict()</code>.</p>
<p>As an example, we will predict brain size of some mammals that are not in the <code>mammals_raw</code> dataset.</p>
<div class="sourceCode" id="cb298"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb298-1"><a href="regressions-and-anova-in-r.html#cb298-1" aria-hidden="true" tabindex="-1"></a><span class="do">## recall the regression and the predictor name for the model: log_brain_body</span></span>
<span id="cb298-2"><a href="regressions-and-anova-in-r.html#cb298-2" aria-hidden="true" tabindex="-1"></a>log_brain_body</span>
<span id="cb298-3"><a href="regressions-and-anova-in-r.html#cb298-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Call:</span></span>
<span id="cb298-4"><a href="regressions-and-anova-in-r.html#cb298-4" aria-hidden="true" tabindex="-1"></a><span class="co">#lm(formula = brain ~ body, data = mammals_log)</span></span>
<span id="cb298-5"><a href="regressions-and-anova-in-r.html#cb298-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb298-6"><a href="regressions-and-anova-in-r.html#cb298-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Coefficients:</span></span>
<span id="cb298-7"><a href="regressions-and-anova-in-r.html#cb298-7" aria-hidden="true" tabindex="-1"></a><span class="co">#(Intercept)         body  </span></span>
<span id="cb298-8"><a href="regressions-and-anova-in-r.html#cb298-8" aria-hidden="true" tabindex="-1"></a><span class="co">#     0.9271       0.7517  </span></span></code></pre></div>
<p>We cab revise species in <code>mammals_raw</code>.</p>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="regressions-and-anova-in-r.html#cb299-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mammals_raw)</span>
<span id="cb299-2"><a href="regressions-and-anova-in-r.html#cb299-2" aria-hidden="true" tabindex="-1"></a><span class="co">#                   body brain</span></span>
<span id="cb299-3"><a href="regressions-and-anova-in-r.html#cb299-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Arctic fox        3.385  44.5</span></span>
<span id="cb299-4"><a href="regressions-and-anova-in-r.html#cb299-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Owl monkey        0.480  15.5</span></span>
<span id="cb299-5"><a href="regressions-and-anova-in-r.html#cb299-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Mountain beaver   1.350   8.1</span></span>
<span id="cb299-6"><a href="regressions-and-anova-in-r.html#cb299-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Cow             465.000 423.0</span></span>
<span id="cb299-7"><a href="regressions-and-anova-in-r.html#cb299-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Grey wolf        36.330 119.5</span></span>
<span id="cb299-8"><a href="regressions-and-anova-in-r.html#cb299-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Goat             27.660 115.0</span></span>
<span id="cb299-9"><a href="regressions-and-anova-in-r.html#cb299-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(mammals_raw)</span>
<span id="cb299-10"><a href="regressions-and-anova-in-r.html#cb299-10" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;Arctic fox&quot;                &quot;Owl monkey&quot;                &quot;Mountain beaver&quot;           &quot;Cow&quot;                      </span></span>
<span id="cb299-11"><a href="regressions-and-anova-in-r.html#cb299-11" aria-hidden="true" tabindex="-1"></a><span class="co"># [5] &quot;Grey wolf&quot;                 &quot;Goat&quot;                      &quot;Roe deer&quot;                  &quot;Guinea pig&quot;               </span></span>
<span id="cb299-12"><a href="regressions-and-anova-in-r.html#cb299-12" aria-hidden="true" tabindex="-1"></a><span class="co"># [9] &quot;Verbet&quot;                    &quot;Chinchilla&quot;                &quot;Ground squirrel&quot;           &quot;Arctic ground squirrel&quot;   </span></span>
<span id="cb299-13"><a href="regressions-and-anova-in-r.html#cb299-13" aria-hidden="true" tabindex="-1"></a><span class="co">#[13] &quot;African giant pouched rat&quot; &quot;Lesser short-tailed shrew&quot; &quot;Star-nosed mole&quot;           &quot;Nine-banded armadillo&quot;    </span></span>
<span id="cb299-14"><a href="regressions-and-anova-in-r.html#cb299-14" aria-hidden="true" tabindex="-1"></a><span class="co">#[17] &quot;Tree hyrax&quot;                &quot;N.A. opossum&quot;              &quot;Asian elephant&quot;            &quot;Big brown bat&quot;            </span></span>
<span id="cb299-15"><a href="regressions-and-anova-in-r.html#cb299-15" aria-hidden="true" tabindex="-1"></a><span class="co">#[21] &quot;Donkey&quot;                    &quot;Horse&quot;                     &quot;European hedgehog&quot;         &quot;Patas monkey&quot;             </span></span>
<span id="cb299-16"><a href="regressions-and-anova-in-r.html#cb299-16" aria-hidden="true" tabindex="-1"></a><span class="co">#[25] &quot;Cat&quot;                       &quot;Galago&quot;                    &quot;Genet&quot;                     &quot;Giraffe&quot;                  </span></span>
<span id="cb299-17"><a href="regressions-and-anova-in-r.html#cb299-17" aria-hidden="true" tabindex="-1"></a><span class="co">#[29] &quot;Gorilla&quot;                   &quot;Grey seal&quot;                 &quot;Rock hyrax-a&quot;              &quot;Human&quot;                    </span></span>
<span id="cb299-18"><a href="regressions-and-anova-in-r.html#cb299-18" aria-hidden="true" tabindex="-1"></a><span class="co">#[33] &quot;African elephant&quot;          &quot;Water opossum&quot;             &quot;Rhesus monkey&quot;             &quot;Kangaroo&quot;                 </span></span>
<span id="cb299-19"><a href="regressions-and-anova-in-r.html#cb299-19" aria-hidden="true" tabindex="-1"></a><span class="co">#[37] &quot;Yellow-bellied marmot&quot;     &quot;Golden hamster&quot;            &quot;Mouse&quot;                     &quot;Little brown bat&quot;         </span></span>
<span id="cb299-20"><a href="regressions-and-anova-in-r.html#cb299-20" aria-hidden="true" tabindex="-1"></a><span class="co">#[41] &quot;Slow loris&quot;                &quot;Okapi&quot;                     &quot;Rabbit&quot;                    &quot;Sheep&quot;                    </span></span>
<span id="cb299-21"><a href="regressions-and-anova-in-r.html#cb299-21" aria-hidden="true" tabindex="-1"></a><span class="co">#[45] &quot;Jaguar&quot;                    &quot;Chimpanzee&quot;                &quot;Baboon&quot;                    &quot;Desert hedgehog&quot;          </span></span>
<span id="cb299-22"><a href="regressions-and-anova-in-r.html#cb299-22" aria-hidden="true" tabindex="-1"></a><span class="co">#[49] &quot;Giant armadillo&quot;           &quot;Rock hyrax-b&quot;              &quot;Raccoon&quot;                   &quot;Rat&quot;                      </span></span>
<span id="cb299-23"><a href="regressions-and-anova-in-r.html#cb299-23" aria-hidden="true" tabindex="-1"></a><span class="co">#[53] &quot;E. American mole&quot;          &quot;Mole rat&quot;                  &quot;Musk shrew&quot;                &quot;Pig&quot;                      </span></span>
<span id="cb299-24"><a href="regressions-and-anova-in-r.html#cb299-24" aria-hidden="true" tabindex="-1"></a><span class="co">#[57] &quot;Echidna&quot;                   &quot;Brazilian tapir&quot;           &quot;Tenrec&quot;                    &quot;Phalanger&quot;                </span></span>
<span id="cb299-25"><a href="regressions-and-anova-in-r.html#cb299-25" aria-hidden="true" tabindex="-1"></a><span class="co">#[61] &quot;Tree shrew&quot;                &quot;Red fox&quot;</span></span></code></pre></div>
<p>We will add three more mammals body weight in kg and estimate their brain weight in g by create a new data frame with the new body sizes.</p>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="regressions-and-anova-in-r.html#cb300-1" aria-hidden="true" tabindex="-1"></a>new_body <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">body =</span> <span class="fu">c</span>(<span class="dv">272</span>, <span class="dv">495</span>, <span class="dv">160</span>, <span class="fl">3.13</span>), <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb300-2"><a href="regressions-and-anova-in-r.html#cb300-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(new_body) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;grizzly_bear&quot;</span>, <span class="st">&quot;bactrian_camel&quot;</span>, <span class="st">&quot;llama&quot;</span>, <span class="st">&quot;capuchin_monkey&quot;</span>)</span>
<span id="cb300-3"><a href="regressions-and-anova-in-r.html#cb300-3" aria-hidden="true" tabindex="-1"></a>new_body</span>
<span id="cb300-4"><a href="regressions-and-anova-in-r.html#cb300-4" aria-hidden="true" tabindex="-1"></a><span class="co">#                  body</span></span>
<span id="cb300-5"><a href="regressions-and-anova-in-r.html#cb300-5" aria-hidden="true" tabindex="-1"></a><span class="co">#grizzly_bear    272.00</span></span>
<span id="cb300-6"><a href="regressions-and-anova-in-r.html#cb300-6" aria-hidden="true" tabindex="-1"></a><span class="co">#bactrian_camel  495.00</span></span>
<span id="cb300-7"><a href="regressions-and-anova-in-r.html#cb300-7" aria-hidden="true" tabindex="-1"></a><span class="co">#llama           160.00</span></span>
<span id="cb300-8"><a href="regressions-and-anova-in-r.html#cb300-8" aria-hidden="true" tabindex="-1"></a><span class="co">#capuchin_monkey   3.13</span></span></code></pre></div>
<p>Remember that the model is for log10 transformed data, so we need to transform accordingly.</p>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="regressions-and-anova-in-r.html#cb301-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb301-2"><a href="regressions-and-anova-in-r.html#cb301-2" aria-hidden="true" tabindex="-1"></a>new_body_log <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">modify</span>(new_body,log10)</span>
<span id="cb301-3"><a href="regressions-and-anova-in-r.html#cb301-3" aria-hidden="true" tabindex="-1"></a>new_body_log</span>
<span id="cb301-4"><a href="regressions-and-anova-in-r.html#cb301-4" aria-hidden="true" tabindex="-1"></a><span class="co">#                     body</span></span>
<span id="cb301-5"><a href="regressions-and-anova-in-r.html#cb301-5" aria-hidden="true" tabindex="-1"></a><span class="co">#grizzly_bear    2.4345689</span></span>
<span id="cb301-6"><a href="regressions-and-anova-in-r.html#cb301-6" aria-hidden="true" tabindex="-1"></a><span class="co">#bactrian_camel  2.6946052</span></span>
<span id="cb301-7"><a href="regressions-and-anova-in-r.html#cb301-7" aria-hidden="true" tabindex="-1"></a><span class="co">#llama           2.2041200</span></span>
<span id="cb301-8"><a href="regressions-and-anova-in-r.html#cb301-8" aria-hidden="true" tabindex="-1"></a><span class="co">#capuchin_monkey 0.4955443</span></span></code></pre></div>
<p>We can estimate their brain weight in g of these mammals using the function <code>predict()</code>.</p>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="regressions-and-anova-in-r.html#cb302-1" aria-hidden="true" tabindex="-1"></a>new_brain_log <span class="ot">&lt;-</span> <span class="fu">predict</span>(log_brain_body, new_body_log)</span>
<span id="cb302-2"><a href="regressions-and-anova-in-r.html#cb302-2" aria-hidden="true" tabindex="-1"></a>new_brain_log</span>
<span id="cb302-3"><a href="regressions-and-anova-in-r.html#cb302-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   grizzly_bear  bactrian_camel           llama capuchin_monkey </span></span>
<span id="cb302-4"><a href="regressions-and-anova-in-r.html#cb302-4" aria-hidden="true" tabindex="-1"></a><span class="co">#      2.757158        2.952624        2.583933        1.299621 </span></span></code></pre></div>
<p>We can transform this log10 weight brains back by exponentiation with base 10.</p>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="regressions-and-anova-in-r.html#cb303-1" aria-hidden="true" tabindex="-1"></a>new_brain <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">^</span>new_brain_log</span>
<span id="cb303-2"><a href="regressions-and-anova-in-r.html#cb303-2" aria-hidden="true" tabindex="-1"></a>new_brain</span>
<span id="cb303-3"><a href="regressions-and-anova-in-r.html#cb303-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   grizzly_bear  bactrian_camel           llama capuchin_monkey </span></span>
<span id="cb303-4"><a href="regressions-and-anova-in-r.html#cb303-4" aria-hidden="true" tabindex="-1"></a><span class="co">#       571.6868        896.6517        383.6480         19.9352 </span></span></code></pre></div>
</div>
<div id="logistic-regressions" class="section level2" number="7.3">
<h2><span class="header-section-number">7.3</span> Logistic regressions</h2>
<p><strong>9)</strong> We can use a logistic regression when our dependent variable is binary (i.e., 1 or 0; yes or no; A or B; male or female) from a set of continuous predictor variables that determine such outcome. We can implement such regressions using the function <code>glm()</code> for a generalized linear model (GLM).</p>
<p>We will dataset <code>cats</code>that has anatomical data from domestic cats including sex and two continuous variables (body and heart weight). This dataset is part of the R-package <strong>MASS</strong>.</p>
<div class="sourceCode" id="cb304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb304-1"><a href="regressions-and-anova-in-r.html#cb304-1" aria-hidden="true" tabindex="-1"></a><span class="do">## load MASS package and ‘cats’ dataset</span></span>
<span id="cb304-2"><a href="regressions-and-anova-in-r.html#cb304-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb304-3"><a href="regressions-and-anova-in-r.html#cb304-3" aria-hidden="true" tabindex="-1"></a>cats_data <span class="ot">&lt;-</span> cats</span>
<span id="cb304-4"><a href="regressions-and-anova-in-r.html#cb304-4" aria-hidden="true" tabindex="-1"></a><span class="co"># check structure</span></span>
<span id="cb304-5"><a href="regressions-and-anova-in-r.html#cb304-5" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(cats_data)</span>
<span id="cb304-6"><a href="regressions-and-anova-in-r.html#cb304-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&#39;data.frame&#39;:  144 obs. of  3 variables:</span></span>
<span id="cb304-7"><a href="regressions-and-anova-in-r.html#cb304-7" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Sex: Factor w/ 2 levels &quot;F&quot;,&quot;M&quot;: 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb304-8"><a href="regressions-and-anova-in-r.html#cb304-8" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Bwt: num  2 2 2 2.1 2.1 2.1 2.1 2.1 2.1 2.1 ...</span></span>
<span id="cb304-9"><a href="regressions-and-anova-in-r.html#cb304-9" aria-hidden="true" tabindex="-1"></a><span class="co"># $ Hwt: num  7 7.4 9.5 7.2 7.3 7.6 8.1 8.2 8.3 8.5 ...</span></span></code></pre></div>
<p>We can now run the model with the formula with <code>Sex</code> as a binary dependent and the two continuous as predictors. Notice the argument for type or family of distribution <code>family = binomial()</code>.</p>
<div class="sourceCode" id="cb305"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb305-1"><a href="regressions-and-anova-in-r.html#cb305-1" aria-hidden="true" tabindex="-1"></a>cats_data_logistic_fit <span class="ot">&lt;-</span> <span class="fu">glm</span>(Sex <span class="sc">~</span> Bwt <span class="sc">+</span> Hwt, <span class="at">data =</span> cats_data, <span class="at">family =</span> <span class="fu">binomial</span>())</span>
<span id="cb305-2"><a href="regressions-and-anova-in-r.html#cb305-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(cats_data_logistic_fit)</span>
<span id="cb305-3"><a href="regressions-and-anova-in-r.html#cb305-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Call:</span></span>
<span id="cb305-4"><a href="regressions-and-anova-in-r.html#cb305-4" aria-hidden="true" tabindex="-1"></a><span class="co">#glm(formula = Sex ~ Bwt + Hwt, family = binomial(), data = cats_data)</span></span>
<span id="cb305-5"><a href="regressions-and-anova-in-r.html#cb305-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb305-6"><a href="regressions-and-anova-in-r.html#cb305-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Deviance Residuals: </span></span>
<span id="cb305-7"><a href="regressions-and-anova-in-r.html#cb305-7" aria-hidden="true" tabindex="-1"></a><span class="co">#    Min       1Q   Median       3Q      Max  </span></span>
<span id="cb305-8"><a href="regressions-and-anova-in-r.html#cb305-8" aria-hidden="true" tabindex="-1"></a><span class="co">#-2.1727  -0.7815   0.3141   0.7476   1.8189  </span></span>
<span id="cb305-9"><a href="regressions-and-anova-in-r.html#cb305-9" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb305-10"><a href="regressions-and-anova-in-r.html#cb305-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Coefficients:</span></span>
<span id="cb305-11"><a href="regressions-and-anova-in-r.html#cb305-11" aria-hidden="true" tabindex="-1"></a><span class="co">#            Estimate Std. Error z value Pr(&gt;|z|)    </span></span>
<span id="cb305-12"><a href="regressions-and-anova-in-r.html#cb305-12" aria-hidden="true" tabindex="-1"></a><span class="co">#(Intercept) -8.69896    1.69982  -5.118 3.09e-07 ***</span></span>
<span id="cb305-13"><a href="regressions-and-anova-in-r.html#cb305-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Bwt          3.55391    0.88208   4.029 5.60e-05 ***</span></span>
<span id="cb305-14"><a href="regressions-and-anova-in-r.html#cb305-14" aria-hidden="true" tabindex="-1"></a><span class="co">#Hwt          0.02295    0.16146   0.142    0.887    </span></span>
<span id="cb305-15"><a href="regressions-and-anova-in-r.html#cb305-15" aria-hidden="true" tabindex="-1"></a><span class="co">#---</span></span>
<span id="cb305-16"><a href="regressions-and-anova-in-r.html#cb305-16" aria-hidden="true" tabindex="-1"></a><span class="co">#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1</span></span>
<span id="cb305-17"><a href="regressions-and-anova-in-r.html#cb305-17" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb305-18"><a href="regressions-and-anova-in-r.html#cb305-18" aria-hidden="true" tabindex="-1"></a><span class="co">#(Dispersion parameter for binomial family taken to be 1)</span></span>
<span id="cb305-19"><a href="regressions-and-anova-in-r.html#cb305-19" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb305-20"><a href="regressions-and-anova-in-r.html#cb305-20" aria-hidden="true" tabindex="-1"></a><span class="co">#    Null deviance: 181.90  on 143  degrees of freedom</span></span>
<span id="cb305-21"><a href="regressions-and-anova-in-r.html#cb305-21" aria-hidden="true" tabindex="-1"></a><span class="co">#Residual deviance: 132.24  on 141  degrees of freedom</span></span>
<span id="cb305-22"><a href="regressions-and-anova-in-r.html#cb305-22" aria-hidden="true" tabindex="-1"></a><span class="co">#AIC: 138.24</span></span>
<span id="cb305-23"><a href="regressions-and-anova-in-r.html#cb305-23" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb305-24"><a href="regressions-and-anova-in-r.html#cb305-24" aria-hidden="true" tabindex="-1"></a><span class="co">#Number of Fisher Scoring iterations: 5</span></span></code></pre></div>
<p>For the summary, we can that the intercept and body weight are strongly significant, while heart weight is not. This suggest that the sex of the cat has a strong association of the weight of the cat, while not its heart weight. For body weight, its positive coefficient (3.55391) suggests that heavier cats are most likely males while controlling for heart weight.
Now, we can plot this logistic regression.</p>
<div class="sourceCode" id="cb306"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb306-1"><a href="regressions-and-anova-in-r.html#cb306-1" aria-hidden="true" tabindex="-1"></a>cats_data_newdata <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">Bwt =</span> <span class="fu">pretty</span>(cats_data<span class="sc">$</span>Bwt, <span class="dv">20</span>), <span class="at">Hwt =</span> <span class="fu">pretty</span>(cats_data<span class="sc">$</span>Hwt, <span class="dv">5</span>))</span>
<span id="cb306-2"><a href="regressions-and-anova-in-r.html#cb306-2" aria-hidden="true" tabindex="-1"></a>cats_data_newdata<span class="sc">$</span>Sex <span class="ot">&lt;-</span> <span class="fu">predict</span>(cats_data_logistic_fit, <span class="at">newdata =</span> cats_data_newdata, <span class="at">type =</span> <span class="st">&quot;response&quot;</span>)</span>
<span id="cb306-3"><a href="regressions-and-anova-in-r.html#cb306-3" aria-hidden="true" tabindex="-1"></a><span class="do">## we require library ggplot2</span></span>
<span id="cb306-4"><a href="regressions-and-anova-in-r.html#cb306-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb306-5"><a href="regressions-and-anova-in-r.html#cb306-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(cats_data_newdata, <span class="fu">aes</span>(<span class="at">x =</span> Bwt, <span class="at">y =</span> Sex)) <span class="sc">+</span> <span class="fu">geom_line</span>() <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>Hwt)</span></code></pre></div>
<p><img src="figures/session_7_figs/6_1_logistic_plot.png" width="90%" /></p>
</div>
<div id="non-linear-regressions" class="section level2" number="7.4">
<h2><span class="header-section-number">7.4</span> Non-linear regressions</h2>
<p><strong>10)</strong> We can have a dataset that is not adjust to a linear model (as in many cases in biological experiment outcomes), so we have to use non-parametric models.
Several methods can perform such complex regression include those of the R-packages <a href="https://cran.r-project.org/web/packages/drc/index.html" target="_blank"><strong>drc</strong></a>.
We will illustrate some of these.</p>
<div class="sourceCode" id="cb307"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb307-1"><a href="regressions-and-anova-in-r.html#cb307-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we need to install the &#39;drc&#39; package</span></span>
<span id="cb307-2"><a href="regressions-and-anova-in-r.html#cb307-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;drc&quot;</span>)</span>
<span id="cb307-3"><a href="regressions-and-anova-in-r.html#cb307-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(drc)</span>
<span id="cb307-4"><a href="regressions-and-anova-in-r.html#cb307-4" aria-hidden="true" tabindex="-1"></a><span class="do">## we install a library with examples of exponential growth</span></span>
<span id="cb307-5"><a href="regressions-and-anova-in-r.html#cb307-5" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;growthrates&quot;</span>)</span>
<span id="cb307-6"><a href="regressions-and-anova-in-r.html#cb307-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(growthrates)</span></code></pre></div>
<p>We can determine estimate a regression that fit an exponential curve like bacterial growth</p>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="regressions-and-anova-in-r.html#cb308-1" aria-hidden="true" tabindex="-1"></a><span class="do">## load an example</span></span>
<span id="cb308-2"><a href="regressions-and-anova-in-r.html#cb308-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(bactgrowth)</span>
<span id="cb308-3"><a href="regressions-and-anova-in-r.html#cb308-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(bactgrowth)</span>
<span id="cb308-4"><a href="regressions-and-anova-in-r.html#cb308-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  strain replicate conc time value</span></span>
<span id="cb308-5"><a href="regressions-and-anova-in-r.html#cb308-5" aria-hidden="true" tabindex="-1"></a><span class="co">#1      T         2    0    0 0.013</span></span>
<span id="cb308-6"><a href="regressions-and-anova-in-r.html#cb308-6" aria-hidden="true" tabindex="-1"></a><span class="co">#2      T         2    0    1 0.014</span></span>
<span id="cb308-7"><a href="regressions-and-anova-in-r.html#cb308-7" aria-hidden="true" tabindex="-1"></a><span class="co">#3      T         2    0    2 0.017</span></span>
<span id="cb308-8"><a href="regressions-and-anova-in-r.html#cb308-8" aria-hidden="true" tabindex="-1"></a><span class="co">#4      T         2    0    3 0.022</span></span>
<span id="cb308-9"><a href="regressions-and-anova-in-r.html#cb308-9" aria-hidden="true" tabindex="-1"></a><span class="co">#5      T         2    0    4 0.030</span></span>
<span id="cb308-10"><a href="regressions-and-anova-in-r.html#cb308-10" aria-hidden="true" tabindex="-1"></a><span class="co">#6      T         2    0    5 0.039</span></span></code></pre></div>
<p>Now, we subset this data for strain <code>T</code> and replicate <code>2</code> (i.e., two conditions) by using <code>&amp;</code>.</p>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="regressions-and-anova-in-r.html#cb309-1" aria-hidden="true" tabindex="-1"></a>bactgrowth_strain_T <span class="ot">&lt;-</span> <span class="fu">subset</span>(bactgrowth, strain <span class="sc">==</span> <span class="st">&quot;T&quot;</span> <span class="sc">&amp;</span> replicate <span class="sc">==</span> <span class="dv">2</span> <span class="sc">&amp;</span> conc <span class="sc">==</span> <span class="fl">31.25</span>)</span>
<span id="cb309-2"><a href="regressions-and-anova-in-r.html#cb309-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(bactgrowth_strain_T)</span>
<span id="cb309-3"><a href="regressions-and-anova-in-r.html#cb309-3" aria-hidden="true" tabindex="-1"></a><span class="co">#          strain replicate  conc time value</span></span>
<span id="cb309-4"><a href="regressions-and-anova-in-r.html#cb309-4" aria-hidden="true" tabindex="-1"></a><span class="co">#249      T         2 31.25    0 0.011</span></span>
<span id="cb309-5"><a href="regressions-and-anova-in-r.html#cb309-5" aria-hidden="true" tabindex="-1"></a><span class="co">#250      T         2 31.25    1 0.010</span></span>
<span id="cb309-6"><a href="regressions-and-anova-in-r.html#cb309-6" aria-hidden="true" tabindex="-1"></a><span class="co">#251      T         2 31.25    2 0.010</span></span>
<span id="cb309-7"><a href="regressions-and-anova-in-r.html#cb309-7" aria-hidden="true" tabindex="-1"></a><span class="co">#252      T         2 31.25    3 0.010</span></span>
<span id="cb309-8"><a href="regressions-and-anova-in-r.html#cb309-8" aria-hidden="true" tabindex="-1"></a><span class="co">#253      T         2 31.25    4 0.010</span></span>
<span id="cb309-9"><a href="regressions-and-anova-in-r.html#cb309-9" aria-hidden="true" tabindex="-1"></a><span class="co">#254      T         2 31.25    5 0.011</span></span>
<span id="cb309-10"><a href="regressions-and-anova-in-r.html#cb309-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb309-11"><a href="regressions-and-anova-in-r.html#cb309-11" aria-hidden="true" tabindex="-1"></a><span class="co"># visualize this graph</span></span>
<span id="cb309-12"><a href="regressions-and-anova-in-r.html#cb309-12" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(value<span class="sc">~</span>time, <span class="at">data =</span> bactgrowth_strain_T)</span></code></pre></div>
<p><img src="figures/session_7_figs/6_exponential_growth.png" width="70%" /></p>
<p>We can fit an exponential curve using the function <code>drm()</code> of the package <code>drc</code>. Notice that the type of curve will depend on its parametrization as indicated by the argument <code>fct =</code>. In this case is <code>fct = G.4()</code> or , for more information check <code>drc</code> manual and <a href="https://www.statforbiology.com/nonlinearregression/usefulequations#introduction" target="_blank"><strong>this page</strong></a>.</p>
<div class="sourceCode" id="cb310"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb310-1"><a href="regressions-and-anova-in-r.html#cb310-1" aria-hidden="true" tabindex="-1"></a><span class="do">## fct for exponential growth </span></span>
<span id="cb310-2"><a href="regressions-and-anova-in-r.html#cb310-2" aria-hidden="true" tabindex="-1"></a>model_bactgrowth_strain_T <span class="ot">&lt;-</span> drc<span class="sc">::</span><span class="fu">drm</span>(value <span class="sc">~</span> time, <span class="at">fct =</span> <span class="fu">G.4</span>(), <span class="at">data =</span> bactgrowth_strain_T)</span>
<span id="cb310-3"><a href="regressions-and-anova-in-r.html#cb310-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model_bactgrowth_strain_T)</span>
<span id="cb310-4"><a href="regressions-and-anova-in-r.html#cb310-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Model fitted: Gompertz (4 parms)</span></span>
<span id="cb310-5"><a href="regressions-and-anova-in-r.html#cb310-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb310-6"><a href="regressions-and-anova-in-r.html#cb310-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Parameter estimates:</span></span>
<span id="cb310-7"><a href="regressions-and-anova-in-r.html#cb310-7" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb310-8"><a href="regressions-and-anova-in-r.html#cb310-8" aria-hidden="true" tabindex="-1"></a><span class="co">#                Estimate Std. Error t-value   p-value    </span></span>
<span id="cb310-9"><a href="regressions-and-anova-in-r.html#cb310-9" aria-hidden="true" tabindex="-1"></a><span class="co">#b:(Intercept) -0.1799145  0.0061157 -29.419 &lt; 2.2e-16 ***</span></span>
<span id="cb310-10"><a href="regressions-and-anova-in-r.html#cb310-10" aria-hidden="true" tabindex="-1"></a><span class="co">#c:(Intercept)  0.0105954  0.0003694  28.683 &lt; 2.2e-16 ***</span></span>
<span id="cb310-11"><a href="regressions-and-anova-in-r.html#cb310-11" aria-hidden="true" tabindex="-1"></a><span class="co">#d:(Intercept)  0.1008913  0.0013430  75.122 &lt; 2.2e-16 ***</span></span>
<span id="cb310-12"><a href="regressions-and-anova-in-r.html#cb310-12" aria-hidden="true" tabindex="-1"></a><span class="co">#e:(Intercept) 16.6860271  0.1201873 138.833 &lt; 2.2e-16 ***</span></span>
<span id="cb310-13"><a href="regressions-and-anova-in-r.html#cb310-13" aria-hidden="true" tabindex="-1"></a><span class="co">#---</span></span>
<span id="cb310-14"><a href="regressions-and-anova-in-r.html#cb310-14" aria-hidden="true" tabindex="-1"></a><span class="co">#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1</span></span>
<span id="cb310-15"><a href="regressions-and-anova-in-r.html#cb310-15" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb310-16"><a href="regressions-and-anova-in-r.html#cb310-16" aria-hidden="true" tabindex="-1"></a><span class="co">#Residual standard error:</span></span>
<span id="cb310-17"><a href="regressions-and-anova-in-r.html#cb310-17" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb310-18"><a href="regressions-and-anova-in-r.html#cb310-18" aria-hidden="true" tabindex="-1"></a><span class="co">#0.001105757 (27 degrees of freedom)</span></span>
<span id="cb310-19"><a href="regressions-and-anova-in-r.html#cb310-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb310-20"><a href="regressions-and-anova-in-r.html#cb310-20" aria-hidden="true" tabindex="-1"></a><span class="do">## we can plot the fit of this function</span></span>
<span id="cb310-21"><a href="regressions-and-anova-in-r.html#cb310-21" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(model_bactgrowth_strain_T)</span></code></pre></div>
<p><img src="figures/session_7_figs/7_nonlinear_plot.png" width="70%" /></p>
<p>We can also estimate new values of growth of the bacteria strain T (dependent) for new time (predictor) using the function <code>predict()</code>.</p>
<div class="sourceCode" id="cb311"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb311-1"><a href="regressions-and-anova-in-r.html#cb311-1" aria-hidden="true" tabindex="-1"></a>new_time <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">time =</span> <span class="fu">c</span>(<span class="dv">31</span>,<span class="dv">50</span>,<span class="dv">100</span>,<span class="dv">1000</span>))</span>
<span id="cb311-2"><a href="regressions-and-anova-in-r.html#cb311-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(new_time) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;time_31&quot;</span>, <span class="st">&quot;time_50&quot;</span>, <span class="st">&quot;time_100&quot;</span>, <span class="st">&quot;time_1000&quot;</span>)</span>
<span id="cb311-3"><a href="regressions-and-anova-in-r.html#cb311-3" aria-hidden="true" tabindex="-1"></a>new_values_growth <span class="ot">&lt;-</span> <span class="fu">predict</span>(model_bactgrowth_strain_T, new_time)</span>
<span id="cb311-4"><a href="regressions-and-anova-in-r.html#cb311-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(new_values_growth) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;time_31&quot;</span>, <span class="st">&quot;time_50&quot;</span>, <span class="st">&quot;time_100&quot;</span>, <span class="st">&quot;time_1000&quot;</span>)</span>
<span id="cb311-5"><a href="regressions-and-anova-in-r.html#cb311-5" aria-hidden="true" tabindex="-1"></a>new_values_growth</span>
<span id="cb311-6"><a href="regressions-and-anova-in-r.html#cb311-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  time_31   time_50  time_100 time_1000 </span></span>
<span id="cb311-7"><a href="regressions-and-anova-in-r.html#cb311-7" aria-hidden="true" tabindex="-1"></a><span class="co">#0.0942721 0.1006664 0.1008913 0.1008913 </span></span></code></pre></div>
<p><strong>11)</strong> Like most non-linear models, it requires some parametrization determined by the curve that you observe by plotting your data. Another useful function is <code>nls()</code>.</p>
<p>We can fit a exponential growth to our <code>bactgrowth_strain_T</code> using <code>nls()</code>.</p>
<div class="sourceCode" id="cb312"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb312-1"><a href="regressions-and-anova-in-r.html#cb312-1" aria-hidden="true" tabindex="-1"></a>model_bactgrowth_strain_T_nls <span class="ot">&lt;-</span> <span class="fu">nls</span>(value <span class="sc">~</span> a<span class="sc">*</span><span class="fu">exp</span>(r<span class="sc">*</span>time), <span class="at">data =</span>  bactgrowth_strain_T,<span class="at">start =</span> <span class="fu">list</span>(<span class="at">a =</span> <span class="fl">0.5</span>, <span class="at">r =</span> <span class="fl">0.2</span>))</span>
<span id="cb312-2"><a href="regressions-and-anova-in-r.html#cb312-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model_bactgrowth_strain_T_nls)</span>
<span id="cb312-3"><a href="regressions-and-anova-in-r.html#cb312-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Formula: value ~ a * exp(r * time)</span></span>
<span id="cb312-4"><a href="regressions-and-anova-in-r.html#cb312-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb312-5"><a href="regressions-and-anova-in-r.html#cb312-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Parameters:</span></span>
<span id="cb312-6"><a href="regressions-and-anova-in-r.html#cb312-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span id="cb312-7"><a href="regressions-and-anova-in-r.html#cb312-7" aria-hidden="true" tabindex="-1"></a><span class="co">#a 0.011133   0.001356   8.213 4.69e-09 ***</span></span>
<span id="cb312-8"><a href="regressions-and-anova-in-r.html#cb312-8" aria-hidden="true" tabindex="-1"></a><span class="co">#r 0.076416   0.004887  15.636 1.15e-15 ***</span></span>
<span id="cb312-9"><a href="regressions-and-anova-in-r.html#cb312-9" aria-hidden="true" tabindex="-1"></a><span class="co">#---</span></span>
<span id="cb312-10"><a href="regressions-and-anova-in-r.html#cb312-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1</span></span>
<span id="cb312-11"><a href="regressions-and-anova-in-r.html#cb312-11" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb312-12"><a href="regressions-and-anova-in-r.html#cb312-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Residual standard error: 0.008326 on 29 degrees of freedom</span></span>
<span id="cb312-13"><a href="regressions-and-anova-in-r.html#cb312-13" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb312-14"><a href="regressions-and-anova-in-r.html#cb312-14" aria-hidden="true" tabindex="-1"></a><span class="co">#Number of iterations to convergence: 11 </span></span>
<span id="cb312-15"><a href="regressions-and-anova-in-r.html#cb312-15" aria-hidden="true" tabindex="-1"></a><span class="co">#Achieved convergence tolerance: 3.249e-06</span></span>
<span id="cb312-16"><a href="regressions-and-anova-in-r.html#cb312-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb312-17"><a href="regressions-and-anova-in-r.html#cb312-17" aria-hidden="true" tabindex="-1"></a><span class="fu">coef</span>(model_bactgrowth_strain_T_nls)</span>
<span id="cb312-18"><a href="regressions-and-anova-in-r.html#cb312-18" aria-hidden="true" tabindex="-1"></a><span class="co">#         a          r </span></span>
<span id="cb312-19"><a href="regressions-and-anova-in-r.html#cb312-19" aria-hidden="true" tabindex="-1"></a><span class="co">#0.01113342 0.07641632</span></span>
<span id="cb312-20"><a href="regressions-and-anova-in-r.html#cb312-20" aria-hidden="true" tabindex="-1"></a>model_coeff <span class="ot">&lt;-</span> <span class="fu">coef</span>(model_bactgrowth_strain_T_nls)</span>
<span id="cb312-21"><a href="regressions-and-anova-in-r.html#cb312-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb312-22"><a href="regressions-and-anova-in-r.html#cb312-22" aria-hidden="true" tabindex="-1"></a><span class="co"># plot this graph</span></span>
<span id="cb312-23"><a href="regressions-and-anova-in-r.html#cb312-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb312-24"><a href="regressions-and-anova-in-r.html#cb312-24" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(value <span class="sc">~</span> time, <span class="at">data =</span> bactgrowth_strain_T, <span class="at">main =</span> <span class="st">&quot;bactgrowth_strain_T&quot;</span>)</span>
<span id="cb312-25"><a href="regressions-and-anova-in-r.html#cb312-25" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(bactgrowth_strain_T<span class="sc">$</span>time, model_coeff[<span class="dv">1</span>]<span class="sc">*</span><span class="fu">exp</span>(model_coeff[<span class="dv">2</span>]<span class="sc">*</span>bactgrowth_strain_T<span class="sc">$</span>time), <span class="at">col =</span> <span class="st">&quot;orange2&quot;</span>, <span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb312-26"><a href="regressions-and-anova-in-r.html#cb312-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb312-27"><a href="regressions-and-anova-in-r.html#cb312-27" aria-hidden="true" tabindex="-1"></a><span class="co"># this is the same graph</span></span>
<span id="cb312-28"><a href="regressions-and-anova-in-r.html#cb312-28" aria-hidden="true" tabindex="-1"></a><span class="co">#plot(value ~ time, data = bactgrowth_strain_T, main = &quot;bactgrowth_strain_T&quot;)</span></span>
<span id="cb312-29"><a href="regressions-and-anova-in-r.html#cb312-29" aria-hidden="true" tabindex="-1"></a><span class="co">#lines(bactgrowth_strain_T$time, 0.01113342*exp(0.07641632*bactgrowth_strain_T$time), col = &quot;orange2&quot;, lwd = 2)</span></span></code></pre></div>
<p><img src="figures/session_7_figs/8_nonlinear_plot_nls.png" width="70%" />
We can also estimate new values of growth of the bacteria strain T (dependent) for new time (predictor) using the function <code>predict()</code>.</p>
<div class="sourceCode" id="cb313"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb313-1"><a href="regressions-and-anova-in-r.html#cb313-1" aria-hidden="true" tabindex="-1"></a>new_time <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">time =</span> <span class="fu">c</span>(<span class="dv">31</span>,<span class="dv">50</span>,<span class="dv">100</span>,<span class="dv">1000</span>))</span>
<span id="cb313-2"><a href="regressions-and-anova-in-r.html#cb313-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(new_time) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;time_31&quot;</span>, <span class="st">&quot;time_50&quot;</span>, <span class="st">&quot;time_100&quot;</span>, <span class="st">&quot;time_1000&quot;</span>)</span>
<span id="cb313-3"><a href="regressions-and-anova-in-r.html#cb313-3" aria-hidden="true" tabindex="-1"></a>new_values_growth_nls <span class="ot">&lt;-</span> <span class="fu">predict</span>(model_bactgrowth_strain_T_nls, new_time)</span>
<span id="cb313-4"><a href="regressions-and-anova-in-r.html#cb313-4" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(new_values_growth_nls) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;time_31&quot;</span>, <span class="st">&quot;time_50&quot;</span>, <span class="st">&quot;time_100&quot;</span>, <span class="st">&quot;time_1000&quot;</span>)</span>
<span id="cb313-5"><a href="regressions-and-anova-in-r.html#cb313-5" aria-hidden="true" tabindex="-1"></a>new_values_growth_nls</span>
<span id="cb313-6"><a href="regressions-and-anova-in-r.html#cb313-6" aria-hidden="true" tabindex="-1"></a><span class="co">#       time_31      time_50     time_100    time_1000 </span></span>
<span id="cb313-7"><a href="regressions-and-anova-in-r.html#cb313-7" aria-hidden="true" tabindex="-1"></a><span class="co">#1.189684e-01 5.081454e-01 2.319249e+01 1.713226e+31 </span></span></code></pre></div>
</div>
<div id="toy-datasets-for-regressions" class="section level2" number="7.5">
<h2><span class="header-section-number">7.5</span> Toy datasets for regressions</h2>
<p>Now that you have an illustrated example of regressions, you can choose among the following examples to do one of each exercise (e.g., linear regression, selecting predictors, logistic regressions, and non-linear regressions):</p>
<p><strong>a)</strong> R-package <code>ade4</code> dataset <code>aravo</code> for distribution of Alpine plants in Aravo (Valloire, France).</p>
<div class="sourceCode" id="cb314"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb314-1"><a href="regressions-and-anova-in-r.html#cb314-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ade4&quot;</span>)</span>
<span id="cb314-2"><a href="regressions-and-anova-in-r.html#cb314-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb314-3"><a href="regressions-and-anova-in-r.html#cb314-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(aravo)</span>
<span id="cb314-4"><a href="regressions-and-anova-in-r.html#cb314-4" aria-hidden="true" tabindex="-1"></a>aravo_traits <span class="ot">&lt;-</span> aravo<span class="sc">$</span>traits</span>
<span id="cb314-5"><a href="regressions-and-anova-in-r.html#cb314-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(aravo_traits)</span>
<span id="cb314-6"><a href="regressions-and-anova-in-r.html#cb314-6" aria-hidden="true" tabindex="-1"></a><span class="co">#          Height Spread Angle  Area Thick  SLA N_mass Seed</span></span>
<span id="cb314-7"><a href="regressions-and-anova-in-r.html#cb314-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Agro.rupe      6     10    80  60.0  0.12  8.1 218.70 0.08</span></span>
<span id="cb314-8"><a href="regressions-and-anova-in-r.html#cb314-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Alop.alpi      5     20    20 190.9  0.20 15.1 203.85 0.21</span></span>
<span id="cb314-9"><a href="regressions-and-anova-in-r.html#cb314-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Anth.nipp     15      5    50 280.0  0.08 18.0 219.60 0.54</span></span>
<span id="cb314-10"><a href="regressions-and-anova-in-r.html#cb314-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Heli.sede      0     30    80 600.0  0.20 10.6 233.20 1.72</span></span>
<span id="cb314-11"><a href="regressions-and-anova-in-r.html#cb314-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Aven.vers     12     30    60 420.0  0.14 12.5 156.25 1.17</span></span>
<span id="cb314-12"><a href="regressions-and-anova-in-r.html#cb314-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Care.rosa     30     20    80 180.0  0.40  6.5 208.65 1.68</span></span></code></pre></div>
<p><strong>b)</strong> R-package <code>ade4</code> dataset <code>bacteria</code> for total number of the 21 amino acids in 43 species of bacteria.</p>
<div class="sourceCode" id="cb315"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb315-1"><a href="regressions-and-anova-in-r.html#cb315-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ade4&quot;</span>)</span>
<span id="cb315-2"><a href="regressions-and-anova-in-r.html#cb315-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb315-3"><a href="regressions-and-anova-in-r.html#cb315-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(bacteria)</span>
<span id="cb315-4"><a href="regressions-and-anova-in-r.html#cb315-4" aria-hidden="true" tabindex="-1"></a>bacteria_espaa <span class="ot">&lt;-</span> bacteria<span class="sc">$</span>espaa</span>
<span id="cb315-5"><a href="regressions-and-anova-in-r.html#cb315-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(bacteria_espaa)</span>
<span id="cb315-6"><a href="regressions-and-anova-in-r.html#cb315-6" aria-hidden="true" tabindex="-1"></a><span class="co">#          Ala   Arg   Asn   Asp  Cys   Gln   Glu   Gly   His   Ile    Leu   Lys   Met   Phe   Pro   Ser  Stp   Thr   Trp   Tyr   Val</span></span>
<span id="cb315-7"><a href="regressions-and-anova-in-r.html#cb315-7" aria-hidden="true" tabindex="-1"></a><span class="co">#AERPECG 60384 48823 12907 24567 5896 12058 41892 54161 12189 32902  72081 22335 12476 17397 40897 47682 2619 29697  8324 21272 55421</span></span>
<span id="cb315-8"><a href="regressions-and-anova-in-r.html#cb315-8" aria-hidden="true" tabindex="-1"></a><span class="co">#AQUAECG 28326 23565 17294 20754 3770  9791 46216 32425  7430 35203  50946 45092  8979 24736 19580 23036 1490 20221  4495 19887 38211</span></span>
<span id="cb315-9"><a href="regressions-and-anova-in-r.html#cb315-9" aria-hidden="true" tabindex="-1"></a><span class="co">#ARCFUCG 50661 36713 20664 31358 7358 11448 56642 46562  9733 46172  60810 43489 16109 29467 24835 35350 2088 26833  6639 23426 55527</span></span>
<span id="cb315-10"><a href="regressions-and-anova-in-r.html#cb315-10" aria-hidden="true" tabindex="-1"></a><span class="co">#BACHDCG 85555 55731 41785 58649 8393 47325 90358 81279 27972 80022 115667 67022 30743 51345 44287 64903 3558 64274 13156 39079 86451</span></span>
<span id="cb315-11"><a href="regressions-and-anova-in-r.html#cb315-11" aria-hidden="true" tabindex="-1"></a><span class="co">#BACSUCG 91276 48698 46474 61327 9340 45257 85399 82164 26892 87050 114745 82768 31755 53186 43835 74444 3628 64270 12238 41133 80173</span></span>
<span id="cb315-12"><a href="regressions-and-anova-in-r.html#cb315-12" aria-hidden="true" tabindex="-1"></a><span class="co">#BORBUCG 12555  8898 20318 14526 1822  6305 18931 14559  3403 29993  29020 28498  5008 17514  7044 20862  773 11018  1404 11814 14980</span></span></code></pre></div>
<p><strong>c)</strong> R-package <code>ade4</code> dataset <code>carni19</code> for carnivores body size and range.</p>
<div class="sourceCode" id="cb316"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb316-1"><a href="regressions-and-anova-in-r.html#cb316-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ade4&quot;</span>)</span>
<span id="cb316-2"><a href="regressions-and-anova-in-r.html#cb316-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb316-3"><a href="regressions-and-anova-in-r.html#cb316-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(carni70)</span>
<span id="cb316-4"><a href="regressions-and-anova-in-r.html#cb316-4" aria-hidden="true" tabindex="-1"></a>carni70_size_range <span class="ot">&lt;-</span> carni70<span class="sc">$</span>tab</span>
<span id="cb316-5"><a href="regressions-and-anova-in-r.html#cb316-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(carni70_size_range)</span>
<span id="cb316-6"><a href="regressions-and-anova-in-r.html#cb316-6" aria-hidden="true" tabindex="-1"></a><span class="co">#                         size range</span></span>
<span id="cb316-7"><a href="regressions-and-anova-in-r.html#cb316-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Puma_concolor           37.01 36.00</span></span>
<span id="cb316-8"><a href="regressions-and-anova-in-r.html#cb316-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Herpailurus_yaguaroundi  2.59 24.00</span></span>
<span id="cb316-9"><a href="regressions-and-anova-in-r.html#cb316-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Leopardus_wiedii         3.20 21.75</span></span>
<span id="cb316-10"><a href="regressions-and-anova-in-r.html#cb316-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Leopardus_pardalis       7.90 24.50</span></span>
<span id="cb316-11"><a href="regressions-and-anova-in-r.html#cb316-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Oreailurus_jacobita      3.99  1.75</span></span>
<span id="cb316-12"><a href="regressions-and-anova-in-r.html#cb316-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Oncifelis_colocolo       2.99  6.25</span></span></code></pre></div>
<p><strong>d)</strong> R-package <code>ade4</code> dataset <code>cnc2003</code> for a data frame about cinema attendanace in France. This data frame contains the following variables, <strong>popuis</strong> the population department in million inhabitants. <strong>entris</strong> the number of movie theater visitors in million. <strong>receis</strong> the takings from ticket offices. <strong>seanis</strong> the number of proposed shows in thousands. <strong>commis</strong> the number of equipped communes in movie theaters (units). <strong>etabis</strong> the number of active movie theaters (units). <strong>salleis</strong> the number of active screens. <strong>fautis</strong> the number of proposed seats. <strong>artesis</strong> the number of movie theaters offering <em>Art and Essay</em> movies. <strong>multiis</strong> the number of active multiplexes. <strong>departis</strong> the name of the department. <strong>regis</strong> the administrative region of the department.</p>
<div class="sourceCode" id="cb317"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb317-1"><a href="regressions-and-anova-in-r.html#cb317-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ade4&quot;</span>)</span>
<span id="cb317-2"><a href="regressions-and-anova-in-r.html#cb317-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ade4)</span>
<span id="cb317-3"><a href="regressions-and-anova-in-r.html#cb317-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(cnc2003)</span>
<span id="cb317-4"><a href="regressions-and-anova-in-r.html#cb317-4" aria-hidden="true" tabindex="-1"></a>cnc2003_data <span class="ot">&lt;-</span> cnc2003</span>
<span id="cb317-5"><a href="regressions-and-anova-in-r.html#cb317-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cnc2003_data)</span>
<span id="cb317-6"><a href="regressions-and-anova-in-r.html#cb317-6" aria-hidden="true" tabindex="-1"></a><span class="co">#     popu  entr   rece sean comm etab salle  faut artes multi                   depart                       reg</span></span>
<span id="cb317-7"><a href="regressions-and-anova-in-r.html#cb317-7" aria-hidden="true" tabindex="-1"></a><span class="co">#D1 0.515 0.769  4.065   26   16   20    35  6288    12     0                      Ain               Rhone-Alpes</span></span>
<span id="cb317-8"><a href="regressions-and-anova-in-r.html#cb317-8" aria-hidden="true" tabindex="-1"></a><span class="co">#D2 0.536 0.731  3.942   28   14   15    38  7403     8     0                    Aisne                  Picardie</span></span>
<span id="cb317-9"><a href="regressions-and-anova-in-r.html#cb317-9" aria-hidden="true" tabindex="-1"></a><span class="co">#D3 0.345 0.499  2.789   17    7   11    28  3956     4     0                   Allier                  Auvergne</span></span>
<span id="cb317-10"><a href="regressions-and-anova-in-r.html#cb317-10" aria-hidden="true" tabindex="-1"></a><span class="co">#D4 0.140 0.453  2.262   17   13   15    23  3480     7     0 Alpes de Hautes Provence Provence-Alpes-Coted&#39;Azur</span></span>
<span id="cb317-11"><a href="regressions-and-anova-in-r.html#cb317-11" aria-hidden="true" tabindex="-1"></a><span class="co">#D5 0.121 0.522  2.908   21   19   23    35  6053     5     0             Hautes Alpes Provence-Alpes-Coted&#39;Azur</span></span>
<span id="cb317-12"><a href="regressions-and-anova-in-r.html#cb317-12" aria-hidden="true" tabindex="-1"></a><span class="co">#D6 1.011 3.520 21.731  111   23   42    94 16764     8     1          Alpes Maritimes Provence-Alpes-Coted&#39;Azur</span></span></code></pre></div>
<p><strong>e)</strong> R-package <code>mlbench</code> dataset <code>PimaIndiansDiabetes2</code> for a data frame with 768 observations on 10 variables for the Pima Indians Diabetes Database: <strong>pregnant</strong> Number of times pregnant; <strong>glucose</strong> Plasma glucose concentration (glucose tolerance test); <strong>pressure</strong> Diastolic blood pressure (mm Hg); <strong>triceps</strong> Triceps skin fold thickness (mm); <strong>insulin</strong> 2-Hour serum insulin (mu U/ml); <strong>mass</strong> Body mass index (weight in kg/(height in m)^2); <strong>pedigree</strong> Diabetes pedigree function; <strong>age</strong> Age (years); <strong>diabetes</strong> Class variable (test for diabetes); <strong>prob_diabetes</strong> probability of diabetes (1 = yes, 0 = no).</p>
<div class="sourceCode" id="cb318"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb318-1"><a href="regressions-and-anova-in-r.html#cb318-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;mlbench&quot;</span>)</span>
<span id="cb318-2"><a href="regressions-and-anova-in-r.html#cb318-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mlbench)</span>
<span id="cb318-3"><a href="regressions-and-anova-in-r.html#cb318-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(PimaIndiansDiabetes2)</span>
<span id="cb318-4"><a href="regressions-and-anova-in-r.html#cb318-4" aria-hidden="true" tabindex="-1"></a>PimaIndiansDiabetes2 <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(PimaIndiansDiabetes2)</span>
<span id="cb318-5"><a href="regressions-and-anova-in-r.html#cb318-5" aria-hidden="true" tabindex="-1"></a>PimaIndiansDiabetes2<span class="sc">$</span>prob_diabetes <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(PimaIndiansDiabetes2<span class="sc">$</span>diabetes <span class="sc">==</span> <span class="st">&quot;pos&quot;</span>, <span class="dv">1</span>, <span class="dv">0</span>)</span>
<span id="cb318-6"><a href="regressions-and-anova-in-r.html#cb318-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(PimaIndiansDiabetes2)</span>
<span id="cb318-7"><a href="regressions-and-anova-in-r.html#cb318-7" aria-hidden="true" tabindex="-1"></a><span class="co">#    pregnant glucose pressure triceps insulin mass pedigree age diabetes prob_diabetes</span></span>
<span id="cb318-8"><a href="regressions-and-anova-in-r.html#cb318-8" aria-hidden="true" tabindex="-1"></a><span class="co">#4         1      89       66      23      94 28.1    0.167  21      neg             0</span></span>
<span id="cb318-9"><a href="regressions-and-anova-in-r.html#cb318-9" aria-hidden="true" tabindex="-1"></a><span class="co">#5         0     137       40      35     168 43.1    2.288  33      pos             1</span></span>
<span id="cb318-10"><a href="regressions-and-anova-in-r.html#cb318-10" aria-hidden="true" tabindex="-1"></a><span class="co">#7         3      78       50      32      88 31.0    0.248  26      pos             1</span></span>
<span id="cb318-11"><a href="regressions-and-anova-in-r.html#cb318-11" aria-hidden="true" tabindex="-1"></a><span class="co">#9         2     197       70      45     543 30.5    0.158  53      pos             1</span></span>
<span id="cb318-12"><a href="regressions-and-anova-in-r.html#cb318-12" aria-hidden="true" tabindex="-1"></a><span class="co">#14        1     189       60      23     846 30.1    0.398  59      pos             1</span></span>
<span id="cb318-13"><a href="regressions-and-anova-in-r.html#cb318-13" aria-hidden="true" tabindex="-1"></a><span class="co">#15        5     166       72      19     175 25.8    0.587  51      pos             1</span></span></code></pre></div>
<p><strong>f)</strong> R-package <code>mlbench</code> dataset <code>BostonHousing</code> for Housing data for 506 census tracts of Boston from the 1970 census. Variables include <strong>crim </strong> per capita crime rate by town; <strong>zn</strong> proportion of residential land zoned for lots over 25,000 sq.ft; <strong>indus</strong> proportion of non-retail business acres per town; <strong>chas</strong> Charles River dummy variable (= 1 if tract bounds river; 0 otherwise); <strong>nox</strong> nitric oxides concentration (parts per 10 million)rmaverage number of rooms per dwelling; <strong>age</strong> proportion of owner-occupied units built prior to 1940; <strong>dis</strong> weighted distances to five Boston employment centres; <strong>rad</strong> index of accessibility to radial highways; <strong>tax</strong> full-value property-tax rate per USD 10,000; <strong>ptratio</strong> pupil-teacher ratio by town; <strong>b</strong> 1000(B−0.63)2whereBis the proportion of blacks by town; <strong>lstat</strong> percentage of lower status of the population; <strong>medv</strong> median value of owner-occupied homes in USD 1000’s.</p>
<div class="sourceCode" id="cb319"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb319-1"><a href="regressions-and-anova-in-r.html#cb319-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;mlbench&quot;</span>)</span>
<span id="cb319-2"><a href="regressions-and-anova-in-r.html#cb319-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mlbench)</span>
<span id="cb319-3"><a href="regressions-and-anova-in-r.html#cb319-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(BostonHousing)</span>
<span id="cb319-4"><a href="regressions-and-anova-in-r.html#cb319-4" aria-hidden="true" tabindex="-1"></a>BostonHousing_data <span class="ot">&lt;-</span> BostonHousing</span>
<span id="cb319-5"><a href="regressions-and-anova-in-r.html#cb319-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(BostonHousing_data)</span>
<span id="cb319-6"><a href="regressions-and-anova-in-r.html#cb319-6" aria-hidden="true" tabindex="-1"></a><span class="co">#     crim zn indus chas   nox    rm  age    dis rad tax ptratio      b lstat medv</span></span>
<span id="cb319-7"><a href="regressions-and-anova-in-r.html#cb319-7" aria-hidden="true" tabindex="-1"></a><span class="co">#1 0.00632 18  2.31    0 0.538 6.575 65.2 4.0900   1 296    15.3 396.90  4.98 24.0</span></span>
<span id="cb319-8"><a href="regressions-and-anova-in-r.html#cb319-8" aria-hidden="true" tabindex="-1"></a><span class="co">#2 0.02731  0  7.07    0 0.469 6.421 78.9 4.9671   2 242    17.8 396.90  9.14 21.6</span></span>
<span id="cb319-9"><a href="regressions-and-anova-in-r.html#cb319-9" aria-hidden="true" tabindex="-1"></a><span class="co">#3 0.02729  0  7.07    0 0.469 7.185 61.1 4.9671   2 242    17.8 392.83  4.03 34.7</span></span>
<span id="cb319-10"><a href="regressions-and-anova-in-r.html#cb319-10" aria-hidden="true" tabindex="-1"></a><span class="co">#4 0.03237  0  2.18    0 0.458 6.998 45.8 6.0622   3 222    18.7 394.63  2.94 33.4</span></span>
<span id="cb319-11"><a href="regressions-and-anova-in-r.html#cb319-11" aria-hidden="true" tabindex="-1"></a><span class="co">#5 0.06905  0  2.18    0 0.458 7.147 54.2 6.0622   3 222    18.7 396.90  5.33 36.2</span></span>
<span id="cb319-12"><a href="regressions-and-anova-in-r.html#cb319-12" aria-hidden="true" tabindex="-1"></a><span class="co">#6 0.02985  0  2.18    0 0.458 6.430 58.7 6.0622   3 222    18.7 394.12  5.21 28.7</span></span></code></pre></div>
</div>
<div id="anova" class="section level2" number="7.6">
<h2><span class="header-section-number">7.6</span> ANOVA</h2>
<p>An analysis of variance (ANOVA) will help you to determine if a dependent variable change according to the levels of one or more categorical independent variables. As part of this test, ANOVA will determine if a particular group has a different mean from the overall mean value of the data. This is determined by checking the variance of each of these individual group against the overall variance of all the data. A significant result is reported if one or more groups falls outside the range of variation predicted by the null hypothesis (all group means are equal). ANOVA will test a null hypothesis (H<sub>0</sub>) that indicates a no difference in means versus the alternative (H<sub>a</sub>) that indicates that the means are different between groups.</p>
</div>
<div id="one-way-anova" class="section level2" number="7.7">
<h2><span class="header-section-number">7.7</span> One-way ANOVA</h2>
<p><strong>12)</strong> The most common types of ANOVA include a one-way ANOVA with one independent variable and a two-way ANOVA with two independent variables. In order to perform an ANOVA, we use the function <code>aov()</code>.</p>
<p>For an example of one-way ANOVA, we will continue using the <code>crab_dataset</code></p>
<div class="sourceCode" id="cb320"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb320-1"><a href="regressions-and-anova-in-r.html#cb320-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we have already downloaded MASS and we assign the crabs dataset to an object</span></span>
<span id="cb320-2"><a href="regressions-and-anova-in-r.html#cb320-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb320-3"><a href="regressions-and-anova-in-r.html#cb320-3" aria-hidden="true" tabindex="-1"></a>crab_dataset <span class="ot">&lt;-</span> crabs</span>
<span id="cb320-4"><a href="regressions-and-anova-in-r.html#cb320-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(crab_dataset)</span>
<span id="cb320-5"><a href="regressions-and-anova-in-r.html#cb320-5" aria-hidden="true" tabindex="-1"></a><span class="co">#  sp sex index   FL  RW   CL   CW  BD</span></span>
<span id="cb320-6"><a href="regressions-and-anova-in-r.html#cb320-6" aria-hidden="true" tabindex="-1"></a><span class="co">#1  B   M     1  8.1 6.7 16.1 19.0 7.0</span></span>
<span id="cb320-7"><a href="regressions-and-anova-in-r.html#cb320-7" aria-hidden="true" tabindex="-1"></a><span class="co">#2  B   M     2  8.8 7.7 18.1 20.8 7.4</span></span>
<span id="cb320-8"><a href="regressions-and-anova-in-r.html#cb320-8" aria-hidden="true" tabindex="-1"></a><span class="co">#3  B   M     3  9.2 7.8 19.0 22.4 7.7</span></span>
<span id="cb320-9"><a href="regressions-and-anova-in-r.html#cb320-9" aria-hidden="true" tabindex="-1"></a><span class="co">#4  B   M     4  9.6 7.9 20.1 23.1 8.2</span></span>
<span id="cb320-10"><a href="regressions-and-anova-in-r.html#cb320-10" aria-hidden="true" tabindex="-1"></a><span class="co">#5  B   M     5  9.8 8.0 20.3 23.0 8.2</span></span>
<span id="cb320-11"><a href="regressions-and-anova-in-r.html#cb320-11" aria-hidden="true" tabindex="-1"></a><span class="co">#6  B   M     6 10.8 9.0 23.0 26.5 9.8</span></span></code></pre></div>
<p>We notice two categorical variables <code>sp</code> for species as “B” or “O” for blue or orange and <code>sex</code> for “M” males and “F” females.
For one-way ANOVA, we will determine if species variable <code>sp</code> that includes two categories or groups “B” or “O” have different means for the continuous variable <code>CL</code> or the carapace length (mm). We use functions <code>aov()</code> and <code>summary()</code></p>
<div class="sourceCode" id="cb321"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb321-1"><a href="regressions-and-anova-in-r.html#cb321-1" aria-hidden="true" tabindex="-1"></a>CL_anova_one_way <span class="ot">&lt;-</span> <span class="fu">aov</span>(CL <span class="sc">~</span> sp, <span class="at">data =</span> crab_dataset)</span>
<span id="cb321-2"><a href="regressions-and-anova-in-r.html#cb321-2" aria-hidden="true" tabindex="-1"></a>CL_anova_one_way</span>
<span id="cb321-3"><a href="regressions-and-anova-in-r.html#cb321-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Call:</span></span>
<span id="cb321-4"><a href="regressions-and-anova-in-r.html#cb321-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   aov(formula = CL ~ sp, data = crab_dataset)</span></span>
<span id="cb321-5"><a href="regressions-and-anova-in-r.html#cb321-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb321-6"><a href="regressions-and-anova-in-r.html#cb321-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Terms:</span></span>
<span id="cb321-7"><a href="regressions-and-anova-in-r.html#cb321-7" aria-hidden="true" tabindex="-1"></a><span class="co">#                      sp Residuals</span></span>
<span id="cb321-8"><a href="regressions-and-anova-in-r.html#cb321-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Sum of Squares   838.451  9246.853</span></span>
<span id="cb321-9"><a href="regressions-and-anova-in-r.html#cb321-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Deg. of Freedom        1       198</span></span>
<span id="cb321-10"><a href="regressions-and-anova-in-r.html#cb321-10" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb321-11"><a href="regressions-and-anova-in-r.html#cb321-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Residual standard error: 6.833833</span></span>
<span id="cb321-12"><a href="regressions-and-anova-in-r.html#cb321-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Estimated effects may be unbalanced</span></span>
<span id="cb321-13"><a href="regressions-and-anova-in-r.html#cb321-13" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(CL_anova_one_way)</span>
<span id="cb321-14"><a href="regressions-and-anova-in-r.html#cb321-14" aria-hidden="true" tabindex="-1"></a><span class="co">#             Df Sum Sq Mean Sq F value   Pr(&gt;F)    </span></span>
<span id="cb321-15"><a href="regressions-and-anova-in-r.html#cb321-15" aria-hidden="true" tabindex="-1"></a><span class="co">#sp            1    838   838.5   17.95 3.47e-05 ***</span></span>
<span id="cb321-16"><a href="regressions-and-anova-in-r.html#cb321-16" aria-hidden="true" tabindex="-1"></a><span class="co">#Residuals   198   9247    46.7                     </span></span>
<span id="cb321-17"><a href="regressions-and-anova-in-r.html#cb321-17" aria-hidden="true" tabindex="-1"></a><span class="co">#---</span></span>
<span id="cb321-18"><a href="regressions-and-anova-in-r.html#cb321-18" aria-hidden="true" tabindex="-1"></a><span class="co">#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1</span></span></code></pre></div>
<p>Overall, the <code>Pr(&gt;F)</code> column provides the p-value (significance) of the F-statistic, which is very low (p &lt; 0.001) and suggest that the we find litter support for the null hypothesis of no difference among group means of <code>sp</code> (i.e., species “B” or “O”) based on <code>CL</code> or the carapace length (mm). Therefore, the measurements of carapace length seem to be different between the two species of crab.</p>
</div>
<div id="two-way-anova" class="section level2" number="7.8">
<h2><span class="header-section-number">7.8</span> Two-way ANOVA</h2>
<p><strong>13)</strong> We can perform a two-way ANOVA by including two independent variables.</p>
<p>For a two-way ANOVA, we will both categorical variables <code>sp</code> for species as “B” or “O” for blue or orange and <code>sex</code> for “M” males and “F” females. As a dependent, we will use again the continuous variable <code>CL</code> or the carapace length (mm). We use functions <code>aov()</code> and <code>summary()</code></p>
<div class="sourceCode" id="cb322"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb322-1"><a href="regressions-and-anova-in-r.html#cb322-1" aria-hidden="true" tabindex="-1"></a>CL_anova_two_way <span class="ot">&lt;-</span> <span class="fu">aov</span>(CL <span class="sc">~</span> sp <span class="sc">+</span> sex, <span class="at">data =</span> crab_dataset)</span>
<span id="cb322-2"><a href="regressions-and-anova-in-r.html#cb322-2" aria-hidden="true" tabindex="-1"></a>CL_anova_two_way</span>
<span id="cb322-3"><a href="regressions-and-anova-in-r.html#cb322-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Call:</span></span>
<span id="cb322-4"><a href="regressions-and-anova-in-r.html#cb322-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   aov(formula = CL ~ sp + sex, data = crab_dataset)</span></span>
<span id="cb322-5"><a href="regressions-and-anova-in-r.html#cb322-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb322-6"><a href="regressions-and-anova-in-r.html#cb322-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Terms:</span></span>
<span id="cb322-7"><a href="regressions-and-anova-in-r.html#cb322-7" aria-hidden="true" tabindex="-1"></a><span class="co">#                      sp      sex Residuals</span></span>
<span id="cb322-8"><a href="regressions-and-anova-in-r.html#cb322-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Sum of Squares   838.451  111.154  9135.699</span></span>
<span id="cb322-9"><a href="regressions-and-anova-in-r.html#cb322-9" aria-hidden="true" tabindex="-1"></a><span class="co">#Deg. of Freedom        1        1       197</span></span>
<span id="cb322-10"><a href="regressions-and-anova-in-r.html#cb322-10" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb322-11"><a href="regressions-and-anova-in-r.html#cb322-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Residual standard error: 6.809854</span></span>
<span id="cb322-12"><a href="regressions-and-anova-in-r.html#cb322-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Estimated effects may be unbalanced</span></span>
<span id="cb322-13"><a href="regressions-and-anova-in-r.html#cb322-13" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(CL_anova_two_way)</span>
<span id="cb322-14"><a href="regressions-and-anova-in-r.html#cb322-14" aria-hidden="true" tabindex="-1"></a><span class="co">#             Df Sum Sq Mean Sq F value   Pr(&gt;F)    </span></span>
<span id="cb322-15"><a href="regressions-and-anova-in-r.html#cb322-15" aria-hidden="true" tabindex="-1"></a><span class="co">#sp            1    838   838.5  18.080 3.27e-05 ***</span></span>
<span id="cb322-16"><a href="regressions-and-anova-in-r.html#cb322-16" aria-hidden="true" tabindex="-1"></a><span class="co">#sex           1    111   111.2   2.397    0.123    </span></span>
<span id="cb322-17"><a href="regressions-and-anova-in-r.html#cb322-17" aria-hidden="true" tabindex="-1"></a><span class="co">#Residuals   197   9136    46.4                     </span></span>
<span id="cb322-18"><a href="regressions-and-anova-in-r.html#cb322-18" aria-hidden="true" tabindex="-1"></a><span class="co">#---</span></span>
<span id="cb322-19"><a href="regressions-and-anova-in-r.html#cb322-19" aria-hidden="true" tabindex="-1"></a><span class="co">#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1</span></span></code></pre></div>
<p>Overall, the <code>Pr(&gt;F)</code> column provides two p-values for the corresponding F-statistic and, again, only the species type shows evidence of difference among group means of their <code>CL</code> or the carapace length (mm). In contrast, sex does not explain differences in the measurements of carapace length.</p>
</div>
<div id="post-hoc-tests" class="section level2" number="7.9">
<h2><span class="header-section-number">7.9</span> Post-hoc tests</h2>
<p><strong>14)</strong> We can perform a post-hoc test to find the differences among group means. One of such tests is Tukey’s Honestly Significant Difference (Tukey’s HSD) for pairwise comparisons.</p>
<p>We use the function <code>TukeyHSD()</code> on the resulting ANOVA object.</p>
<div class="sourceCode" id="cb323"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb323-1"><a href="regressions-and-anova-in-r.html#cb323-1" aria-hidden="true" tabindex="-1"></a><span class="do">## one-way ANOVA</span></span>
<span id="cb323-2"><a href="regressions-and-anova-in-r.html#cb323-2" aria-hidden="true" tabindex="-1"></a><span class="fu">TukeyHSD</span>(CL_anova_one_way)</span>
<span id="cb323-3"><a href="regressions-and-anova-in-r.html#cb323-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  Tukey multiple comparisons of means</span></span>
<span id="cb323-4"><a href="regressions-and-anova-in-r.html#cb323-4" aria-hidden="true" tabindex="-1"></a><span class="co">#    95% family-wise confidence level</span></span>
<span id="cb323-5"><a href="regressions-and-anova-in-r.html#cb323-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb323-6"><a href="regressions-and-anova-in-r.html#cb323-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Fit: aov(formula = CL ~ sp, data = crab_dataset)</span></span>
<span id="cb323-7"><a href="regressions-and-anova-in-r.html#cb323-7" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb323-8"><a href="regressions-and-anova-in-r.html#cb323-8" aria-hidden="true" tabindex="-1"></a><span class="co">#$sp</span></span>
<span id="cb323-9"><a href="regressions-and-anova-in-r.html#cb323-9" aria-hidden="true" tabindex="-1"></a><span class="co">#     diff      lwr      upr    p adj</span></span>
<span id="cb323-10"><a href="regressions-and-anova-in-r.html#cb323-10" aria-hidden="true" tabindex="-1"></a><span class="co">#O-B 4.095 2.189144 6.000856 3.47e-05</span></span>
<span id="cb323-11"><a href="regressions-and-anova-in-r.html#cb323-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb323-12"><a href="regressions-and-anova-in-r.html#cb323-12" aria-hidden="true" tabindex="-1"></a><span class="do">## two-way ANOVA</span></span>
<span id="cb323-13"><a href="regressions-and-anova-in-r.html#cb323-13" aria-hidden="true" tabindex="-1"></a><span class="fu">TukeyHSD</span>(CL_anova_two_way)</span>
<span id="cb323-14"><a href="regressions-and-anova-in-r.html#cb323-14" aria-hidden="true" tabindex="-1"></a><span class="co">#  Tukey multiple comparisons of means</span></span>
<span id="cb323-15"><a href="regressions-and-anova-in-r.html#cb323-15" aria-hidden="true" tabindex="-1"></a><span class="co">#    95% family-wise confidence level</span></span>
<span id="cb323-16"><a href="regressions-and-anova-in-r.html#cb323-16" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb323-17"><a href="regressions-and-anova-in-r.html#cb323-17" aria-hidden="true" tabindex="-1"></a><span class="co">#Fit: aov(formula = CL ~ sp + sex, data = crab_dataset)</span></span>
<span id="cb323-18"><a href="regressions-and-anova-in-r.html#cb323-18" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb323-19"><a href="regressions-and-anova-in-r.html#cb323-19" aria-hidden="true" tabindex="-1"></a><span class="co">#$sp</span></span>
<span id="cb323-20"><a href="regressions-and-anova-in-r.html#cb323-20" aria-hidden="true" tabindex="-1"></a><span class="co">#     diff      lwr      upr    p adj</span></span>
<span id="cb323-21"><a href="regressions-and-anova-in-r.html#cb323-21" aria-hidden="true" tabindex="-1"></a><span class="co">#O-B 4.095 2.195772 5.994228 3.27e-05</span></span>
<span id="cb323-22"><a href="regressions-and-anova-in-r.html#cb323-22" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb323-23"><a href="regressions-and-anova-in-r.html#cb323-23" aria-hidden="true" tabindex="-1"></a><span class="co">#$sex</span></span>
<span id="cb323-24"><a href="regressions-and-anova-in-r.html#cb323-24" aria-hidden="true" tabindex="-1"></a><span class="co">#     diff        lwr      upr    p adj</span></span>
<span id="cb323-25"><a href="regressions-and-anova-in-r.html#cb323-25" aria-hidden="true" tabindex="-1"></a><span class="co">#M-F 1.491 -0.4082279 3.390228 0.123181</span></span></code></pre></div>
<p>Again, these results support that <code>sp</code> for species as “B” or “O” is important when comparing means of <code>CL</code> or the carapace length (mm) of these crabs.</p>
</div>
<div id="beyond-anova-kruskal-wallis-test" class="section level2" number="7.10">
<h2><span class="header-section-number">7.10</span> Beyond ANOVA: Kruskal-Wallis Test</h2>
<p><strong>15)</strong> Some samples that you want to compare might have a non-normal or unknown distributions. In this case, you can use non-parametric methods like Kruskal-Wallis Test for to determine if samples have different medians. The null hypothesis is that the medians of all groups are all equal, while the alternative hypothesis is that at least one sample median is different from the median of at least one other group or sample.</p>
<p>We can reuse the <code>crabs</code> dataset from the package ‘MASS’ – this test does not assume normality.</p>
<div class="sourceCode" id="cb324"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb324-1"><a href="regressions-and-anova-in-r.html#cb324-1" aria-hidden="true" tabindex="-1"></a><span class="do">## we can get crabs dataset</span></span>
<span id="cb324-2"><a href="regressions-and-anova-in-r.html#cb324-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(MASS)</span>
<span id="cb324-3"><a href="regressions-and-anova-in-r.html#cb324-3" aria-hidden="true" tabindex="-1"></a>crab_dataset <span class="ot">&lt;-</span> crabs</span>
<span id="cb324-4"><a href="regressions-and-anova-in-r.html#cb324-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(crab_dataset)</span>
<span id="cb324-5"><a href="regressions-and-anova-in-r.html#cb324-5" aria-hidden="true" tabindex="-1"></a><span class="co">#  sp sex index   FL  RW   CL   CW  BD</span></span>
<span id="cb324-6"><a href="regressions-and-anova-in-r.html#cb324-6" aria-hidden="true" tabindex="-1"></a><span class="co">#1  B   M     1  8.1 6.7 16.1 19.0 7.0</span></span>
<span id="cb324-7"><a href="regressions-and-anova-in-r.html#cb324-7" aria-hidden="true" tabindex="-1"></a><span class="co">#2  B   M     2  8.8 7.7 18.1 20.8 7.4</span></span>
<span id="cb324-8"><a href="regressions-and-anova-in-r.html#cb324-8" aria-hidden="true" tabindex="-1"></a><span class="co">#3  B   M     3  9.2 7.8 19.0 22.4 7.7</span></span>
<span id="cb324-9"><a href="regressions-and-anova-in-r.html#cb324-9" aria-hidden="true" tabindex="-1"></a><span class="co">#4  B   M     4  9.6 7.9 20.1 23.1 8.2</span></span>
<span id="cb324-10"><a href="regressions-and-anova-in-r.html#cb324-10" aria-hidden="true" tabindex="-1"></a><span class="co">#5  B   M     5  9.8 8.0 20.3 23.0 8.2</span></span>
<span id="cb324-11"><a href="regressions-and-anova-in-r.html#cb324-11" aria-hidden="true" tabindex="-1"></a><span class="co">#6  B   M     6 10.8 9.0 23.0 26.5 9.8</span></span></code></pre></div>
<p>We can implement this Kruskal-Wallis Test using the function <code>kruskal.test()</code> for <code>CL</code> carapace length (mm) as dependent and grouping by <code>sex</code> or <code>species</code>.</p>
<div class="sourceCode" id="cb325"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb325-1"><a href="regressions-and-anova-in-r.html#cb325-1" aria-hidden="true" tabindex="-1"></a><span class="do">## for sex as grouping</span></span>
<span id="cb325-2"><a href="regressions-and-anova-in-r.html#cb325-2" aria-hidden="true" tabindex="-1"></a><span class="fu">kruskal.test</span>(CL <span class="sc">~</span> sex, <span class="at">data =</span> crab_dataset)</span>
<span id="cb325-3"><a href="regressions-and-anova-in-r.html#cb325-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   Kruskal-Wallis rank sum test</span></span>
<span id="cb325-4"><a href="regressions-and-anova-in-r.html#cb325-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb325-5"><a href="regressions-and-anova-in-r.html#cb325-5" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  CL by sex</span></span>
<span id="cb325-6"><a href="regressions-and-anova-in-r.html#cb325-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Kruskal-Wallis chi-squared = 1.8424, df = 1, p-value = 0.1747</span></span>
<span id="cb325-7"><a href="regressions-and-anova-in-r.html#cb325-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb325-8"><a href="regressions-and-anova-in-r.html#cb325-8" aria-hidden="true" tabindex="-1"></a><span class="do">## for species as grouping</span></span>
<span id="cb325-9"><a href="regressions-and-anova-in-r.html#cb325-9" aria-hidden="true" tabindex="-1"></a><span class="fu">kruskal.test</span>(CL <span class="sc">~</span> sp, <span class="at">data =</span> crab_dataset)</span>
<span id="cb325-10"><a href="regressions-and-anova-in-r.html#cb325-10" aria-hidden="true" tabindex="-1"></a><span class="co">#   Kruskal-Wallis rank sum test</span></span>
<span id="cb325-11"><a href="regressions-and-anova-in-r.html#cb325-11" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb325-12"><a href="regressions-and-anova-in-r.html#cb325-12" aria-hidden="true" tabindex="-1"></a><span class="co">#data:  CL by sp</span></span>
<span id="cb325-13"><a href="regressions-and-anova-in-r.html#cb325-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Kruskal-Wallis chi-squared = 15.698, df = 1, p-value = 7.431e-05</span></span></code></pre></div>
<p>These results agree with the one-way ANOVA results indicated on <strong>12)</strong>.</p>
</div>
<div id="toy-datasets-for-anova" class="section level2" number="7.11">
<h2><span class="header-section-number">7.11</span> Toy datasets for ANOVA</h2>
<p>Now that you have examples of ANOVA, you can choose among the following examples for further examples</p>
<p><strong>a)</strong> R-package <code>agridat</code> dataset <code>fisher.barley</code> for results of trials of 5 varieties of barley were conducted at 6 stations in Minnesota during the years 1931-1932. This result in a data frame with 60 observations on the following 4 variables: <strong>yield</strong> yield, bu/ac; <strong>gen</strong> genotype/variety, 5 levels; <strong>env</strong> environment/location, 2 levels; <strong>year</strong> year, 1931/1932.</p>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb326-1"><a href="regressions-and-anova-in-r.html#cb326-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;agridat&quot;</span>)</span>
<span id="cb326-2"><a href="regressions-and-anova-in-r.html#cb326-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(agridat)</span>
<span id="cb326-3"><a href="regressions-and-anova-in-r.html#cb326-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(fisher.barley)</span>
<span id="cb326-4"><a href="regressions-and-anova-in-r.html#cb326-4" aria-hidden="true" tabindex="-1"></a>fisher_barley <span class="ot">&lt;-</span> fisher.barley</span>
<span id="cb326-5"><a href="regressions-and-anova-in-r.html#cb326-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(fisher_barley)</span>
<span id="cb326-6"><a href="regressions-and-anova-in-r.html#cb326-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  yield       gen            env year</span></span>
<span id="cb326-7"><a href="regressions-and-anova-in-r.html#cb326-7" aria-hidden="true" tabindex="-1"></a><span class="co">#1  81.0 Manchuria UniversityFarm 1931</span></span>
<span id="cb326-8"><a href="regressions-and-anova-in-r.html#cb326-8" aria-hidden="true" tabindex="-1"></a><span class="co">#2  80.7 Manchuria UniversityFarm 1932</span></span>
<span id="cb326-9"><a href="regressions-and-anova-in-r.html#cb326-9" aria-hidden="true" tabindex="-1"></a><span class="co">#3 146.6 Manchuria         Waseca 1931</span></span>
<span id="cb326-10"><a href="regressions-and-anova-in-r.html#cb326-10" aria-hidden="true" tabindex="-1"></a><span class="co">#4 100.4 Manchuria         Waseca 1932</span></span>
<span id="cb326-11"><a href="regressions-and-anova-in-r.html#cb326-11" aria-hidden="true" tabindex="-1"></a><span class="co">#5  82.3 Manchuria         Morris 1931</span></span>
<span id="cb326-12"><a href="regressions-and-anova-in-r.html#cb326-12" aria-hidden="true" tabindex="-1"></a><span class="co">#6 103.1 Manchuria         Morris 1932</span></span></code></pre></div>
<p><strong>b)</strong> R-package <code>agridat</code> dataset <code>gumpertz.pepper</code> for results with 800 observations on the following 6 variables: <strong>field</strong> field factor, 2 levels; <strong>row</strong> x ordinate; <strong>quadrat</strong> y ordinate; <strong>disease</strong> presence (Y) or absence (N) of disease; <strong>water</strong> soil moisture percent; <strong>leaf</strong> leaf assay count. Each field is 20 rows by 20 quadrates, with 2 to 3 bell pepper plants per plot. If any plant was wilted, dead, or had lesions, the <em>Phytophthora</em> disease was considered to be present in the plot. The soil pathogen load was assayed as the number of leaf disks colonized out of five. In field 2, the pattern of disease presence appears to follow soil water content. In field 1, no obvious trends were present.</p>
<div class="sourceCode" id="cb327"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb327-1"><a href="regressions-and-anova-in-r.html#cb327-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;agridat&quot;</span>)</span>
<span id="cb327-2"><a href="regressions-and-anova-in-r.html#cb327-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(agridat)</span>
<span id="cb327-3"><a href="regressions-and-anova-in-r.html#cb327-3" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(gumpertz.pepper)</span>
<span id="cb327-4"><a href="regressions-and-anova-in-r.html#cb327-4" aria-hidden="true" tabindex="-1"></a>gumpertz_pepper <span class="ot">&lt;-</span> gumpertz.pepper</span>
<span id="cb327-5"><a href="regressions-and-anova-in-r.html#cb327-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gumpertz_pepper)</span>
<span id="cb327-6"><a href="regressions-and-anova-in-r.html#cb327-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  field row quadrat disease water leaf</span></span>
<span id="cb327-7"><a href="regressions-and-anova-in-r.html#cb327-7" aria-hidden="true" tabindex="-1"></a><span class="co">#1    F1   1       1       N 15.05    5</span></span>
<span id="cb327-8"><a href="regressions-and-anova-in-r.html#cb327-8" aria-hidden="true" tabindex="-1"></a><span class="co">#2    F1   1       2       N 14.32    2</span></span>
<span id="cb327-9"><a href="regressions-and-anova-in-r.html#cb327-9" aria-hidden="true" tabindex="-1"></a><span class="co">#3    F1   1       3       N 13.99    3</span></span>
<span id="cb327-10"><a href="regressions-and-anova-in-r.html#cb327-10" aria-hidden="true" tabindex="-1"></a><span class="co">#4    F1   1       4       N 13.85    0</span></span>
<span id="cb327-11"><a href="regressions-and-anova-in-r.html#cb327-11" aria-hidden="true" tabindex="-1"></a><span class="co">#5    F1   1       5       N 13.16    5</span></span>
<span id="cb327-12"><a href="regressions-and-anova-in-r.html#cb327-12" aria-hidden="true" tabindex="-1"></a><span class="co">#6    F1   1       6       N 11.81    5</span></span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="basic-statistics-in-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="exploring-multidimensional-data-in-r.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/07-Regression-ANOVA_v10.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
