<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Session 11 – Sequence Alignments | Bioinformatics Workshop Gitbook</title>
  <meta name="description" content="none" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Session 11 – Sequence Alignments | Bioinformatics Workshop Gitbook" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="none" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Session 11 – Sequence Alignments | Bioinformatics Workshop Gitbook" />
  
  <meta name="twitter:description" content="none" />
  

<meta name="author" content="Juan C. Santos &amp; Randy Ortiz" />


<meta name="date" content="2021-04-14" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="basic-graphics.html"/>
<link rel="next" href="estimating-phylogenetic-trees.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics Workshop Gitbook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html"><i class="fa fa-check"></i><b>1</b> – Getting Started with R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#why-do-we-use-r"><i class="fa fa-check"></i><b>1.1</b> Why do we use R?</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#get-a-good-text-editor"><i class="fa fa-check"></i><b>1.2</b> Get a good text editor</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-r"><i class="fa fa-check"></i><b>1.4</b> Installing R</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#annotating-code"><i class="fa fa-check"></i><b>1.5</b> Annotating code</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#defining-a-working-directory-in-r"><i class="fa fa-check"></i><b>1.6</b> Defining a working directory in R</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html"><i class="fa fa-check"></i><b>2</b> – Using R, Installing Packages and Importing/Exporting Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#your-first-session"><i class="fa fa-check"></i><b>2.1</b> Your first session</a></li>
<li class="chapter" data-level="2.2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#doing-some-base-functions-and-getting-help"><i class="fa fa-check"></i><b>2.2</b> Doing some base functions and getting help</a></li>
<li class="chapter" data-level="2.3" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#about-r-packages"><i class="fa fa-check"></i><b>2.3</b> About R-packages</a></li>
<li class="chapter" data-level="2.4" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#installing-packages"><i class="fa fa-check"></i><b>2.4</b> Installing packages</a></li>
<li class="chapter" data-level="2.5" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#importing-and-exporting-data"><i class="fa fa-check"></i><b>2.5</b> Importing and exporting data</a></li>
<li class="chapter" data-level="2.6" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#saving-data-as-rdata-.rdata-.rda"><i class="fa fa-check"></i><b>2.6</b> Saving data as Rdata (.rdata, .rda)</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html"><i class="fa fa-check"></i><b>3</b> – Basic Data Structures in R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#scalars-and-vectors"><i class="fa fa-check"></i><b>3.1</b> Scalars and vectors</a></li>
<li class="chapter" data-level="3.2" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#arrays"><i class="fa fa-check"></i><b>3.3</b> Arrays</a></li>
<li class="chapter" data-level="3.4" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a></li>
<li class="chapter" data-level="3.5" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tables"><i class="fa fa-check"></i><b>3.5</b> Tables</a></li>
<li class="chapter" data-level="3.6" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#lists"><i class="fa fa-check"></i><b>3.6</b> Lists</a></li>
<li class="chapter" data-level="3.7" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-tables"><i class="fa fa-check"></i><b>3.7</b> Data Tables</a></li>
<li class="chapter" data-level="3.8" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tibbles"><i class="fa fa-check"></i><b>3.8</b> Tibbles</a></li>
<li class="chapter" data-level="3.9" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#dnastringsets-rnastringsets-and-aastringsets"><i class="fa fa-check"></i><b>3.9</b> DNAStringSets, RNAStringSets and AAStringSets</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html"><i class="fa fa-check"></i><b>4</b> – Text Editing and Data Transformations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#working-with-character-strings"><i class="fa fa-check"></i><b>4.1</b> Working with character strings</a></li>
<li class="chapter" data-level="4.2" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#regular-expressions-regex"><i class="fa fa-check"></i><b>4.2</b> Regular expressions (Regex)</a></li>
<li class="chapter" data-level="4.3" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#comparing-ordering-and-simplifying-vectors"><i class="fa fa-check"></i><b>4.3</b> Comparing, ordering and simplifying vectors</a></li>
<li class="chapter" data-level="4.4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#useful-functions-of-stringr"><i class="fa fa-check"></i><b>4.4</b> Useful functions of <code>stringr</code></a></li>
<li class="chapter" data-level="4.5" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#data-transformations"><i class="fa fa-check"></i><b>4.5</b> Data transformations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html"><i class="fa fa-check"></i><b>5</b> – Getting Biological Data from Public Repositories</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#ncbi-databases-genbank-and-others"><i class="fa fa-check"></i><b>5.1</b> NCBI databases: GenBank and others</a></li>
<li class="chapter" data-level="5.2" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#accessing-ncbi-within-r-rentrez"><i class="fa fa-check"></i><b>5.2</b> Accessing NCBI within R: <code>rentrez</code></a></li>
<li class="chapter" data-level="5.3" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#retrieving-pubmed-references"><i class="fa fa-check"></i><b>5.3</b> Retrieving PubMed references</a></li>
<li class="chapter" data-level="5.4" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#comparing-sequences-blast"><i class="fa fa-check"></i><b>5.4</b> Comparing sequences: BLAST</a></li>
<li class="chapter" data-level="5.5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-uniprot"><i class="fa fa-check"></i><b>5.5</b> Other databases: UniProt</a></li>
<li class="chapter" data-level="5.6" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-ensembl"><i class="fa fa-check"></i><b>5.6</b> Other databases: Ensembl</a></li>
<li class="chapter" data-level="5.7" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-taxonomy-phylogenetic-trees-and-species-distributions"><i class="fa fa-check"></i><b>5.7</b> Other databases: Taxonomy, phylogenetic trees and species distributions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html"><i class="fa fa-check"></i><b>6</b> – Basic Statistics in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#exploring-your-data"><i class="fa fa-check"></i><b>6.1</b> Exploring your data</a></li>
<li class="chapter" data-level="6.2" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#normalizing-your-data"><i class="fa fa-check"></i><b>6.2</b> Normalizing your data</a></li>
<li class="chapter" data-level="6.3" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#what-test-should-i-use"><i class="fa fa-check"></i><b>6.3</b> What test should I use?</a></li>
<li class="chapter" data-level="6.4" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-means-t-test"><i class="fa fa-check"></i><b>6.4</b> Comparing means: t-test</a></li>
<li class="chapter" data-level="6.5" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-wilcoxon-signed-rank-test"><i class="fa fa-check"></i><b>6.5</b> Beyond t-test: Wilcoxon signed rank test</a></li>
<li class="chapter" data-level="6.6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-binomial-test"><i class="fa fa-check"></i><b>6.6</b> Beyond t-test: Binomial test</a></li>
<li class="chapter" data-level="6.7" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-chi-square-goodness-of-fit"><i class="fa fa-check"></i><b>6.7</b> Beyond t-test: Chi-square goodness of fit</a></li>
<li class="chapter" data-level="6.8" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-mcnemars-test"><i class="fa fa-check"></i><b>6.8</b> Beyond t-test: McNemar’s test</a></li>
<li class="chapter" data-level="6.9" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-fishers-exact-test"><i class="fa fa-check"></i><b>6.9</b> Beyond t-test: Fisher’s exact test</a></li>
<li class="chapter" data-level="6.10" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#correlations"><i class="fa fa-check"></i><b>6.10</b> Correlations</a></li>
<li class="chapter" data-level="6.11" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-distributions"><i class="fa fa-check"></i><b>6.11</b> Comparing distributions</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html"><i class="fa fa-check"></i><b>7</b> – Regressions and ANOVA in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#linear-regressions"><i class="fa fa-check"></i><b>7.1</b> Linear regressions</a></li>
<li class="chapter" data-level="7.2" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#selecting-predictors"><i class="fa fa-check"></i><b>7.2</b> Selecting predictors</a></li>
<li class="chapter" data-level="7.3" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions"><i class="fa fa-check"></i><b>7.3</b> Logistic regressions</a></li>
<li class="chapter" data-level="7.4" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#non-linear-regressions"><i class="fa fa-check"></i><b>7.4</b> Non-linear regressions</a></li>
<li class="chapter" data-level="7.5" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-regressions"><i class="fa fa-check"></i><b>7.5</b> Toy datasets for regressions</a></li>
<li class="chapter" data-level="7.6" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova"><i class="fa fa-check"></i><b>7.6</b> ANOVA</a></li>
<li class="chapter" data-level="7.7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#one-way-anova"><i class="fa fa-check"></i><b>7.7</b> One-way ANOVA</a></li>
<li class="chapter" data-level="7.8" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#two-way-anova"><i class="fa fa-check"></i><b>7.8</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="7.9" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#post-hoc-tests"><i class="fa fa-check"></i><b>7.9</b> Post-hoc tests</a></li>
<li class="chapter" data-level="7.10" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#beyond-anova-kruskal-wallis-test"><i class="fa fa-check"></i><b>7.10</b> Beyond ANOVA: Kruskal-Wallis Test</a></li>
<li class="chapter" data-level="7.11" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-anova"><i class="fa fa-check"></i><b>7.11</b> Toy datasets for ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html"><i class="fa fa-check"></i><b>8</b> – Exploring Multidimensional Data in R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html#principal-component-analysis"><i class="fa fa-check"></i><b>8.1</b> Principal Component Analysis</a></li>
<li class="chapter" data-level="8.2" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html#multidimensional-scaling"><i class="fa fa-check"></i><b>8.2</b> Multidimensional scaling</a></li>
<li class="chapter" data-level="8.3" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html#cluster-analysis-dfa"><i class="fa fa-check"></i><b>8.3</b> Cluster analysis: DFA</a></li>
<li class="chapter" data-level="8.4" data-path="exploring-multidimensional-data-in-r.html"><a href="exploring-multidimensional-data-in-r.html#cluster-analysis-k-means-unsupervised"><i class="fa fa-check"></i><b>8.4</b> Cluster analysis: <em>k</em>-means (unsupervised)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html"><i class="fa fa-check"></i><b>9</b> – Graphics and Colors</a>
<ul>
<li class="chapter" data-level="9.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#avoid-bad-graphics"><i class="fa fa-check"></i><b>9.1</b> Avoid bad graphics</a></li>
<li class="chapter" data-level="9.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#graphics-in-r-ggplot2"><i class="fa fa-check"></i><b>9.2</b> Graphics in R – ggplot2</a></li>
<li class="chapter" data-level="9.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#preparing-your-dataset"><i class="fa fa-check"></i><b>9.3</b> Preparing your dataset</a></li>
<li class="chapter" data-level="9.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#color-palettes-online"><i class="fa fa-check"></i><b>9.4</b> Color palettes online</a></li>
<li class="chapter" data-level="9.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggplot2-default-palette"><i class="fa fa-check"></i><b>9.5</b> <code>ggplot2</code> default palette</a></li>
<li class="chapter" data-level="9.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#heatmaps-or-tiled-charts"><i class="fa fa-check"></i><b>9.6</b> Heatmaps or tiled charts</a></li>
<li class="chapter" data-level="9.7" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#r-packages-about-color"><i class="fa fa-check"></i><b>9.7</b> R-packages about color</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#viridis-package"><i class="fa fa-check"></i><b>9.7.1</b> <code>viridis</code> package</a></li>
<li class="chapter" data-level="9.7.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcolorbrewer-package"><i class="fa fa-check"></i><b>9.7.2</b> <code>RColorBrewer</code> package</a></li>
<li class="chapter" data-level="9.7.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggsci-package"><i class="fa fa-check"></i><b>9.7.3</b> <code>ggsci</code> package</a></li>
<li class="chapter" data-level="9.7.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#wesanderson-package"><i class="fa fa-check"></i><b>9.7.4</b> <code>wesanderson</code> package</a></li>
<li class="chapter" data-level="9.7.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcartocolor-package"><i class="fa fa-check"></i><b>9.7.5</b> <code>rcartocolor</code> package</a></li>
<li class="chapter" data-level="9.7.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#colorspace-package"><i class="fa fa-check"></i><b>9.7.6</b> <code>colorspace</code> package</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#making-your-own-palette"><i class="fa fa-check"></i><b>9.8</b> Making your own palette</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="basic-graphics.html"><a href="basic-graphics.html"><i class="fa fa-check"></i><b>10</b> – Basic Graphics</a>
<ul>
<li class="chapter" data-level="10.1" data-path="basic-graphics.html"><a href="basic-graphics.html#bar-charts"><i class="fa fa-check"></i><b>10.1</b> Bar charts</a></li>
<li class="chapter" data-level="10.2" data-path="basic-graphics.html"><a href="basic-graphics.html#stacked-bar-charts"><i class="fa fa-check"></i><b>10.2</b> Stacked bar charts</a></li>
<li class="chapter" data-level="10.3" data-path="basic-graphics.html"><a href="basic-graphics.html#histograms"><i class="fa fa-check"></i><b>10.3</b> Histograms</a></li>
<li class="chapter" data-level="10.4" data-path="basic-graphics.html"><a href="basic-graphics.html#multiple-histograms"><i class="fa fa-check"></i><b>10.4</b> Multiple histograms</a></li>
<li class="chapter" data-level="10.5" data-path="basic-graphics.html"><a href="basic-graphics.html#density-plots"><i class="fa fa-check"></i><b>10.5</b> Density plots</a></li>
<li class="chapter" data-level="10.6" data-path="basic-graphics.html"><a href="basic-graphics.html#box-plots"><i class="fa fa-check"></i><b>10.6</b> Box plots</a></li>
<li class="chapter" data-level="10.7" data-path="basic-graphics.html"><a href="basic-graphics.html#violin-plots"><i class="fa fa-check"></i><b>10.7</b> Violin plots</a></li>
<li class="chapter" data-level="10.8" data-path="basic-graphics.html"><a href="basic-graphics.html#scatterplots"><i class="fa fa-check"></i><b>10.8</b> Scatterplots</a></li>
<li class="chapter" data-level="10.9" data-path="basic-graphics.html"><a href="basic-graphics.html#lines-and-scatterplots"><i class="fa fa-check"></i><b>10.9</b> Lines and Scatterplots</a></li>
<li class="chapter" data-level="10.10" data-path="basic-graphics.html"><a href="basic-graphics.html#multiple-graphs-at-once-with-cowplot"><i class="fa fa-check"></i><b>10.10</b> Multiple graphs at once with <code>cowplot</code></a></li>
<li class="chapter" data-level="10.11" data-path="basic-graphics.html"><a href="basic-graphics.html#save-your-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>10.11</b> Save your plots as PDF or PNG</a></li>
<li class="chapter" data-level="10.12" data-path="basic-graphics.html"><a href="basic-graphics.html#other-resources"><i class="fa fa-check"></i><b>10.12</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="sequence-alignments.html"><a href="sequence-alignments.html"><i class="fa fa-check"></i><b>11</b> – Sequence Alignments</a>
<ul>
<li class="chapter" data-level="11.1" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-using-rentrez"><i class="fa fa-check"></i><b>11.1</b> Retrieving sequences using ‘rentrez’</a></li>
<li class="chapter" data-level="11.2" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-manually-from-ncbi"><i class="fa fa-check"></i><b>11.2</b> Retrieving sequences manually from NCBI</a></li>
<li class="chapter" data-level="11.3" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-accession-numbers"><i class="fa fa-check"></i><b>11.3</b> Retrieving accession numbers</a></li>
<li class="chapter" data-level="11.4" data-path="sequence-alignments.html"><a href="sequence-alignments.html#alignment-with-decipher"><i class="fa fa-check"></i><b>11.4</b> Alignment with <code>DECIPHER</code></a></li>
<li class="chapter" data-level="11.5" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualize-alignment"><i class="fa fa-check"></i><b>11.5</b> Visualize alignment</a></li>
<li class="chapter" data-level="11.6" data-path="sequence-alignments.html"><a href="sequence-alignments.html#save-alignment-to-file"><i class="fa fa-check"></i><b>11.6</b> Save alignment to file</a></li>
<li class="chapter" data-level="11.7" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-and-align-several-markers"><i class="fa fa-check"></i><b>11.7</b> Concatenate and align several markers</a></li>
<li class="chapter" data-level="11.8" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-data.table"><i class="fa fa-check"></i><b>11.8</b> Concatenate with <code>data.table</code></a></li>
<li class="chapter" data-level="11.9" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-biostrings"><i class="fa fa-check"></i><b>11.9</b> Concatenate with <code>Biostrings</code></a></li>
<li class="chapter" data-level="11.10" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualizing-alignments-in-r"><i class="fa fa-check"></i><b>11.10</b> Visualizing alignments in R</a></li>
<li class="chapter" data-level="11.11" data-path="sequence-alignments.html"><a href="sequence-alignments.html#other-sequence-alignment-software"><i class="fa fa-check"></i><b>11.11</b> Other sequence alignment software</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html"><i class="fa fa-check"></i><b>12</b> – Estimating Phylogenetic Trees</a>
<ul>
<li class="chapter" data-level="12.1" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#phylogeny-inference-with-iqtree2"><i class="fa fa-check"></i><b>12.1</b> Phylogeny inference with <code>iqtree2</code></a></li>
<li class="chapter" data-level="12.2" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-iqtree2-in-macos"><i class="fa fa-check"></i><b>12.2</b> Installing <code>iqtree2</code> in macOS</a></li>
<li class="chapter" data-level="12.3" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-and-running-iqtree2-in-a-pc"><i class="fa fa-check"></i><b>12.3</b> Installing and running <code>iqtree2</code> in a PC</a></li>
<li class="chapter" data-level="12.4" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#loading-the-iqtree_runner_r-function"><i class="fa fa-check"></i><b>12.4</b> Loading the <code>iqtree_runner_R</code> function</a></li>
<li class="chapter" data-level="12.5" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#running-iqtree2-from-r"><i class="fa fa-check"></i><b>12.5</b> Running <code>iqtree2</code> from R</a></li>
<li class="chapter" data-level="12.6" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-no-partitions"><i class="fa fa-check"></i><b>12.6</b> One marker and no partitions</a></li>
<li class="chapter" data-level="12.7" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-with-partitions"><i class="fa fa-check"></i><b>12.7</b> One marker and with partitions</a></li>
<li class="chapter" data-level="12.8" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#bootstrap-support"><i class="fa fa-check"></i><b>12.8</b> Bootstrap support</a></li>
<li class="chapter" data-level="12.9" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#many-markers-and-with-partitions"><i class="fa fa-check"></i><b>12.9</b> Many markers and with partitions</a></li>
<li class="chapter" data-level="12.10" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#other-sequence-phylogenetic-tree-estimation-software"><i class="fa fa-check"></i><b>12.10</b> Other sequence phylogenetic tree estimation software</a></li>
<li class="chapter" data-level="12.11" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#drawing-phylogenetic-trees-with-ggtree"><i class="fa fa-check"></i><b>12.11</b> Drawing phylogenetic trees with <code>ggtree</code></a></li>
<li class="chapter" data-level="12.12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#save-ggtree-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>12.12</b> Save ggtree plots as PDF or PNG</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html"><i class="fa fa-check"></i><b>13</b> – Mining from a reference dataset</a>
<ul>
<li class="chapter" data-level="13.1" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#a-worked-example-using-the-mtcars-dataset"><i class="fa fa-check"></i><b>13.1</b> A worked example using the <code>mtcars</code> dataset</a></li>
<li class="chapter" data-level="13.2" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-reference-dataset"><i class="fa fa-check"></i><b>13.2</b> Build a reference dataset</a></li>
<li class="chapter" data-level="13.3" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-sampling-dataset"><i class="fa fa-check"></i><b>13.3</b> Build a sampling dataset</a></li>
<li class="chapter" data-level="13.4" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#extracting-from-reference-dataset"><i class="fa fa-check"></i><b>13.4</b> Extracting from reference dataset</a></li>
<li class="chapter" data-level="13.5" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#plots-of-the-extracted-data"><i class="fa fa-check"></i><b>13.5</b> Plots of the extracted data</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="introduction-to-python.html"><a href="introduction-to-python.html"><i class="fa fa-check"></i><b>14</b> – Introduction to Python</a>
<ul>
<li class="chapter" data-level="14.1" data-path="introduction-to-python.html"><a href="introduction-to-python.html#welcome-to-python-why-python"><i class="fa fa-check"></i><b>14.1</b> Welcome to Python! Why Python?</a></li>
<li class="chapter" data-level="14.2" data-path="introduction-to-python.html"><a href="introduction-to-python.html#which-python-should-i-install"><i class="fa fa-check"></i><b>14.2</b> Which Python should I install?</a></li>
<li class="chapter" data-level="14.3" data-path="introduction-to-python.html"><a href="introduction-to-python.html#mac-os-x-python-installations"><i class="fa fa-check"></i><b>14.3</b> Mac OS X Python Installations</a></li>
<li class="chapter" data-level="14.4" data-path="introduction-to-python.html"><a href="introduction-to-python.html#windows-python-installations"><i class="fa fa-check"></i><b>14.4</b> Windows Python Installations</a></li>
<li class="chapter" data-level="14.5" data-path="introduction-to-python.html"><a href="introduction-to-python.html#install-an-ide-or-integrated-development-environment"><i class="fa fa-check"></i><b>14.5</b> Install an IDE, or integrated development environment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html"><i class="fa fa-check"></i><b>15</b> – Python: Keywords and Identifiers</a>
<ul>
<li class="chapter" data-level="15.1" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#keywords-in-python"><i class="fa fa-check"></i><b>15.1</b> Keywords in Python</a></li>
<li class="chapter" data-level="15.2" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#identifiers-in-python"><i class="fa fa-check"></i><b>15.2</b> Identifiers in Python</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html"><i class="fa fa-check"></i><b>16</b> – Python: Statements and Expressions</a>
<ul>
<li class="chapter" data-level="16.1" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#what-are-statements"><i class="fa fa-check"></i><b>16.1</b> What are statements?</a></li>
<li class="chapter" data-level="16.2" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#expressions-in-python"><i class="fa fa-check"></i><b>16.2</b> Expressions in Python</a></li>
<li class="chapter" data-level="16.3" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#simple-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.3</b> Simple Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.4" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#augmented-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.4</b> Augmented Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.5" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#explicit-and-implicit-line-continuation"><i class="fa fa-check"></i><b>16.5</b> Explicit and Implicit Line Continuation</a></li>
<li class="chapter" data-level="16.6" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#indentation-in-python-is-crucial-to-proper-code-function"><i class="fa fa-check"></i><b>16.6</b> Indentation in Python is Crucial to Proper Code function</a></li>
<li class="chapter" data-level="16.7" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#comments"><i class="fa fa-check"></i><b>16.7</b> Comments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="python-data-types.html"><a href="python-data-types.html"><i class="fa fa-check"></i><b>17</b> – Python: Data Types</a>
<ul>
<li class="chapter" data-level="17.1" data-path="python-data-types.html"><a href="python-data-types.html#booleans"><i class="fa fa-check"></i><b>17.1</b> Booleans</a></li>
<li class="chapter" data-level="17.2" data-path="python-data-types.html"><a href="python-data-types.html#numbers-int-float-and-complex"><i class="fa fa-check"></i><b>17.2</b> Numbers: int, float and complex</a></li>
<li class="chapter" data-level="17.3" data-path="python-data-types.html"><a href="python-data-types.html#strings"><i class="fa fa-check"></i><b>17.3</b> Strings</a></li>
<li class="chapter" data-level="17.4" data-path="python-data-types.html"><a href="python-data-types.html#bytes"><i class="fa fa-check"></i><b>17.4</b> Bytes</a></li>
<li class="chapter" data-level="17.5" data-path="python-data-types.html"><a href="python-data-types.html#lists-1"><i class="fa fa-check"></i><b>17.5</b> Lists</a></li>
<li class="chapter" data-level="17.6" data-path="python-data-types.html"><a href="python-data-types.html#tuples"><i class="fa fa-check"></i><b>17.6</b> Tuples</a></li>
<li class="chapter" data-level="17.7" data-path="python-data-types.html"><a href="python-data-types.html#sets"><i class="fa fa-check"></i><b>17.7</b> Sets</a></li>
<li class="chapter" data-level="17.8" data-path="python-data-types.html"><a href="python-data-types.html#dictionary"><i class="fa fa-check"></i><b>17.8</b> Dictionary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="python-strings.html"><a href="python-strings.html"><i class="fa fa-check"></i><b>18</b> – Python: Strings</a>
<ul>
<li class="chapter" data-level="18.1" data-path="python-strings.html"><a href="python-strings.html#string-operators"><i class="fa fa-check"></i><b>18.1</b> String Operators</a></li>
<li class="chapter" data-level="18.2" data-path="python-strings.html"><a href="python-strings.html#other-string-modifiers"><i class="fa fa-check"></i><b>18.2</b> Other String Modifiers</a></li>
<li class="chapter" data-level="18.3" data-path="python-strings.html"><a href="python-strings.html#format-characters-in-strings"><i class="fa fa-check"></i><b>18.3</b> Format-Characters in Strings</a></li>
<li class="chapter" data-level="18.4" data-path="python-strings.html"><a href="python-strings.html#comparisons-with-strings"><i class="fa fa-check"></i><b>18.4</b> Comparisons with Strings</a></li>
<li class="chapter" data-level="18.5" data-path="python-strings.html"><a href="python-strings.html#padding-strings"><i class="fa fa-check"></i><b>18.5</b> Padding strings</a></li>
<li class="chapter" data-level="18.6" data-path="python-strings.html"><a href="python-strings.html#search-for-words-in-string-and-output-location"><i class="fa fa-check"></i><b>18.6</b> Search for words in string and output location</a></li>
<li class="chapter" data-level="18.7" data-path="python-strings.html"><a href="python-strings.html#seek-and-replace-in-strings"><i class="fa fa-check"></i><b>18.7</b> Seek and replace in strings</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="python-operators.html"><a href="python-operators.html"><i class="fa fa-check"></i><b>19</b> – Python: Operators</a>
<ul>
<li class="chapter" data-level="19.1" data-path="python-operators.html"><a href="python-operators.html#many-types-of-operators"><i class="fa fa-check"></i><b>19.1</b> Many types of operators</a></li>
<li class="chapter" data-level="19.2" data-path="python-operators.html"><a href="python-operators.html#arithmetic-operators"><i class="fa fa-check"></i><b>19.2</b> Arithmetic operators</a></li>
<li class="chapter" data-level="19.3" data-path="python-operators.html"><a href="python-operators.html#comparison-operators"><i class="fa fa-check"></i><b>19.3</b> Comparison operators</a></li>
<li class="chapter" data-level="19.4" data-path="python-operators.html"><a href="python-operators.html#logical-operators"><i class="fa fa-check"></i><b>19.4</b> Logical operators</a></li>
<li class="chapter" data-level="19.5" data-path="python-operators.html"><a href="python-operators.html#bitwise-operators"><i class="fa fa-check"></i><b>19.5</b> Bitwise operators</a></li>
<li class="chapter" data-level="19.6" data-path="python-operators.html"><a href="python-operators.html#identity-operators"><i class="fa fa-check"></i><b>19.6</b> Identity Operators</a></li>
<li class="chapter" data-level="19.7" data-path="python-operators.html"><a href="python-operators.html#membership-operators"><i class="fa fa-check"></i><b>19.7</b> Membership operators</a></li>
<li class="chapter" data-level="19.8" data-path="python-operators.html"><a href="python-operators.html#what-is-the-order-of-operations-with-operators"><i class="fa fa-check"></i><b>19.8</b> What is the order of operations with operators?</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="biopython.html"><a href="biopython.html"><i class="fa fa-check"></i><b>20</b> – Biopython</a>
<ul>
<li class="chapter" data-level="20.1" data-path="biopython.html"><a href="biopython.html#introduction-to-biopython"><i class="fa fa-check"></i><b>20.1</b> Introduction to Biopython</a></li>
<li class="chapter" data-level="20.2" data-path="biopython.html"><a href="biopython.html#biopython-installation"><i class="fa fa-check"></i><b>20.2</b> Biopython Installation</a></li>
<li class="chapter" data-level="20.3" data-path="biopython.html"><a href="biopython.html#getting-started"><i class="fa fa-check"></i><b>20.3</b> Getting Started</a></li>
<li class="chapter" data-level="20.4" data-path="biopython.html"><a href="biopython.html#parsing-my-fasta"><i class="fa fa-check"></i><b>20.4</b> Parsing my FASTA</a></li>
<li class="chapter" data-level="20.5" data-path="biopython.html"><a href="biopython.html#parsing-my-genbank-or-genpept"><i class="fa fa-check"></i><b>20.5</b> Parsing my GenBank or GenPept</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="biopython-sequences.html"><a href="biopython-sequences.html"><i class="fa fa-check"></i><b>21</b> – Biopython: Sequences</a>
<ul>
<li class="chapter" data-level="21.1" data-path="biopython-sequences.html"><a href="biopython-sequences.html#sequence-objects"><i class="fa fa-check"></i><b>21.1</b> Sequence Objects</a></li>
<li class="chapter" data-level="21.2" data-path="biopython-sequences.html"><a href="biopython-sequences.html#concatenation-with-sequences"><i class="fa fa-check"></i><b>21.2</b> Concatenation with Sequences</a></li>
<li class="chapter" data-level="21.3" data-path="biopython-sequences.html"><a href="biopython-sequences.html#transcription-dna-to-mrna"><i class="fa fa-check"></i><b>21.3</b> Transcription: DNA to mRNA</a></li>
<li class="chapter" data-level="21.4" data-path="biopython-sequences.html"><a href="biopython-sequences.html#translation-rna-to-prot"><i class="fa fa-check"></i><b>21.4</b> Translation: RNA to PROT</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html"><i class="fa fa-check"></i><b>22</b> – Biopython: Mulitple Sequence Alignments</a>
<ul>
<li class="chapter" data-level="22.1" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwise2"><i class="fa fa-check"></i><b>22.1</b> pairwise2</a></li>
<li class="chapter" data-level="22.2" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwisealigner"><i class="fa fa-check"></i><b>22.2</b> PairwiseAligner</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html"><i class="fa fa-check"></i><b>23</b> – BioPython: GenomeDiagram</a>
<ul>
<li class="chapter" data-level="23.1" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#linearcircular-gene-diagrams"><i class="fa fa-check"></i><b>23.1</b> Linear/Circular Gene Diagrams</a></li>
<li class="chapter" data-level="23.2" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#diagram-readability"><i class="fa fa-check"></i><b>23.2</b> Diagram Readability</a></li>
<li class="chapter" data-level="23.3" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#box-shape"><i class="fa fa-check"></i><b>23.3</b> Box Shape</a></li>
<li class="chapter" data-level="23.4" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#edit-box-shape-arrows"><i class="fa fa-check"></i><b>23.4</b> Edit Box Shape: Arrows</a></li>
<li class="chapter" data-level="23.5" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#restriction-map"><i class="fa fa-check"></i><b>23.5</b> Restriction Map</a></li>
<li class="chapter" data-level="23.6" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#tracks-on-tracks"><i class="fa fa-check"></i><b>23.6</b> Tracks on Tracks</a></li>
<li class="chapter" data-level="23.7" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#cross-link-data"><i class="fa fa-check"></i><b>23.7</b> Cross-link Data</a></li>
<li class="chapter" data-level="23.8" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#clean-cross-link"><i class="fa fa-check"></i><b>23.8</b> Clean Cross-Link</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html"><i class="fa fa-check"></i><b>24</b> – BioPython: Chromosome Modeling</a>
<ul>
<li class="chapter" data-level="24.1" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html#gene-feature-modeling"><i class="fa fa-check"></i><b>24.1</b> Gene Feature Modeling</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="biopython-sequence-parsing-w-plots.html"><a href="biopython-sequence-parsing-w-plots.html"><i class="fa fa-check"></i><b>25</b> – BioPython: Sequence Parsing w/ Plots</a>
<ul>
<li class="chapter" data-level="25.1" data-path="biopython-sequence-parsing-w-plots.html"><a href="biopython-sequence-parsing-w-plots.html#histograms-1"><i class="fa fa-check"></i><b>25.1</b> Histograms</a></li>
<li class="chapter" data-level="25.2" data-path="biopython-sequence-parsing-w-plots.html"><a href="biopython-sequence-parsing-w-plots.html#gc-content"><i class="fa fa-check"></i><b>25.2</b> GC Content</a></li>
<li class="chapter" data-level="25.3" data-path="biopython-sequence-parsing-w-plots.html"><a href="biopython-sequence-parsing-w-plots.html#nt-alignment-dot-plot"><i class="fa fa-check"></i><b>25.3</b> Nt Alignment Dot Plot</a></li>
<li class="chapter" data-level="25.4" data-path="biopython-sequence-parsing-w-plots.html"><a href="biopython-sequence-parsing-w-plots.html#bigger-dots-for-the-dot-plot"><i class="fa fa-check"></i><b>25.4</b> Bigger Dots for the Dot Plot</a></li>
<li class="chapter" data-level="25.5" data-path="biopython-sequence-parsing-w-plots.html"><a href="biopython-sequence-parsing-w-plots.html#fastq-we-did-not-forget-you"><i class="fa fa-check"></i><b>25.5</b> FASTQ we did not forget you</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html"><i class="fa fa-check"></i><b>26</b> – Python: Web Scraping w/ BeautifulSoup</a>
<ul>
<li class="chapter" data-level="26.1" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#introduction"><i class="fa fa-check"></i><b>26.1</b> Introduction</a></li>
<li class="chapter" data-level="26.2" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#installation"><i class="fa fa-check"></i><b>26.2</b> Installation</a></li>
<li class="chapter" data-level="26.3" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#getting-started-1"><i class="fa fa-check"></i><b>26.3</b> Getting started</a></li>
<li class="chapter" data-level="26.4" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-all-matching-tags"><i class="fa fa-check"></i><b>26.4</b> Mining all matching tags</a></li>
<li class="chapter" data-level="26.5" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-tags-by-class-or-id"><i class="fa fa-check"></i><b>26.5</b> Mining tags by class or id</a></li>
<li class="chapter" data-level="26.6" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#use-css-selectors-for-positional-mining"><i class="fa fa-check"></i><b>26.6</b> Use CSS Selectors for positional mining</a></li>
<li class="chapter" data-level="26.7" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#weather-page-example"><i class="fa fa-check"></i><b>26.7</b> Weather Page Example</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics Workshop Gitbook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="sequence-alignments" class="section level1" number="11">
<h1><span class="header-section-number">Session 11</span> – Sequence Alignments</h1>
<p>One central objective in many bioinformatics analyses is to understand the relationship between a set of biological sequences (i.e., DNA, RNA, proteins). To address this objective, we must organize these sequences in order to reflect their evolutionary, functional or structural relationships. Therefore, sequence aligning is just the insertion of gaps or spaces of varying length within the otherwise continuous sequences that we want to investigate. A set of diverse alignment programs achieve this by identifying homologous positions and add gaps or spaces in optimal sites to maximize the number of aligned homologous positions. In an evolutionary context, these gaps represent insertions and deletions (indels) within the genome that are hypothesized to have occurred during the evolutionary history of species or populations being studied. However, defining homologous sites is, at times complex, given the noise (i.e., non-homologous similarities in the sequences) resulting from convergent evolution.</p>
<div id="retrieving-sequences-using-rentrez" class="section level2" number="11.1">
<h2><span class="header-section-number">11.1</span> Retrieving sequences using ‘rentrez’</h2>
<p>For alignments, you must have a minimum of two sequences to perform an alignment.</p>
<p><strong>1)</strong> We will retrieve some sequences from GenBank using the ‘rentrez’ package and we will restrict to one genera of poison frogs: <em>Ameerega</em>. You can try any group of organisms that you want.</p>
<p>In our example, we can first retrieve by searching the genus.</p>
<div class="sourceCode" id="cb455"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb455-1"><a href="sequence-alignments.html#cb455-1" aria-hidden="true" tabindex="-1"></a><span class="do">## make sure that &#39;rentrez&#39; is loaded</span></span>
<span id="cb455-2"><a href="sequence-alignments.html#cb455-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(rentrez)</span>
<span id="cb455-3"><a href="sequence-alignments.html#cb455-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb455-4"><a href="sequence-alignments.html#cb455-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Download some nucleotide sequences from NCBI for Ameerega [Organism] that correspond to the gene COI</span></span>
<span id="cb455-5"><a href="sequence-alignments.html#cb455-5" aria-hidden="true" tabindex="-1"></a>Ameerega_name_COX1 <span class="ot">&lt;-</span> <span class="st">&quot;Ameerega[Organism] AND COI[Gene]&quot;</span></span>
<span id="cb455-6"><a href="sequence-alignments.html#cb455-6" aria-hidden="true" tabindex="-1"></a>Ameerega_seq_IDs <span class="ot">&lt;-</span> <span class="fu">entrez_search</span>(<span class="at">db=</span><span class="st">&quot;nuccore&quot;</span>, <span class="at">term=</span> Ameerega_name_COX1)</span>
<span id="cb455-7"><a href="sequence-alignments.html#cb455-7" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(Ameerega_seq_IDs)</span>
<span id="cb455-8"><a href="sequence-alignments.html#cb455-8" aria-hidden="true" tabindex="-1"></a><span class="co">#List of 5</span></span>
<span id="cb455-9"><a href="sequence-alignments.html#cb455-9" aria-hidden="true" tabindex="-1"></a><span class="co"># $ ids             : chr [1:20] &quot;1952638364&quot; &quot;1952638361&quot; &quot;1952638358&quot; &quot;1074806460&quot; ...</span></span>
<span id="cb455-10"><a href="sequence-alignments.html#cb455-10" aria-hidden="true" tabindex="-1"></a><span class="co"># $ count           : int 37</span></span>
<span id="cb455-11"><a href="sequence-alignments.html#cb455-11" aria-hidden="true" tabindex="-1"></a><span class="co"># $ retmax          : int 20</span></span>
<span id="cb455-12"><a href="sequence-alignments.html#cb455-12" aria-hidden="true" tabindex="-1"></a><span class="co"># $ QueryTranslation: chr &quot;\&quot;Ameerega\&quot;[Organism] AND COI[Gene]&quot;</span></span>
<span id="cb455-13"><a href="sequence-alignments.html#cb455-13" aria-hidden="true" tabindex="-1"></a><span class="co"># $ file            :Classes &#39;XMLInternalDocument&#39;, &#39;XMLAbstractDocument&#39; &lt;externalptr&gt; </span></span>
<span id="cb455-14"><a href="sequence-alignments.html#cb455-14" aria-hidden="true" tabindex="-1"></a><span class="co"># - attr(*, &quot;class&quot;)= chr [1:2] &quot;esearch&quot; &quot;list&quot;</span></span>
<span id="cb455-15"><a href="sequence-alignments.html#cb455-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb455-16"><a href="sequence-alignments.html#cb455-16" aria-hidden="true" tabindex="-1"></a><span class="do">## Note: the $ids indicate that 37 sequences exist in Ameerega for COX1 -- make sure to use Ameerega_seq_IDs$ids</span></span>
<span id="cb455-17"><a href="sequence-alignments.html#cb455-17" aria-hidden="true" tabindex="-1"></a>Ameerega_seqs_fasta <span class="ot">&lt;-</span> <span class="fu">entrez_fetch</span>(<span class="at">db=</span><span class="st">&quot;nuccore&quot;</span>, <span class="at">id=</span>Ameerega_seq_IDs<span class="sc">$</span>ids, <span class="at">rettype=</span><span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb455-18"><a href="sequence-alignments.html#cb455-18" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(Ameerega_seqs_fasta)</span></code></pre></div>
</div>
<div id="retrieving-sequences-manually-from-ncbi" class="section level2" number="11.2">
<h2><span class="header-section-number">11.2</span> Retrieving sequences manually from NCBI</h2>
<p><strong>2)</strong> In some cases, you might have already the NCBI accession numbers and you can use ‘rentrez’ to retrieve such specific sequences. However, you might want to explore if sequences are associated to your group of interest. In this example, we will exemplify this by retrieving some outgroups of <em>Ameerega</em> frogs in the genera <em>Leucostethus</em>, <em>Colostethus</em>, <em>Epipedobates</em> and <em>Silverstoneia</em>.</p>
<p>You can search those using a <strong>point-and-click</strong> approach as follows:</p>
<ol style="list-style-type: upper-alpha">
<li>Go to the <a href="https://www.ncbi.nlm.nih.gov/guide/taxonomy/" target="_blank"><strong>NCBI&gt;Taxonomy</strong></a>.</li>
</ol>
<p><img src="figures/session_11_figs/1_NCBI_taxonomy.png" width="90%" /></p>
<ol start="2" style="list-style-type: upper-alpha">
<li>Click on <a href="https://www.ncbi.nlm.nih.gov/taxonomy" target="_blank"><strong>Taxonomy</strong></a>. An type the family, genus or species of interest and click <strong>Search</strong>.</li>
</ol>
<p><img src="figures/session_11_figs/2_NCBI_Leucotethus.png" width="90%" /></p>
<ol start="3" style="list-style-type: upper-alpha">
<li>Click on the taxon of interest (e.g., <em>Leucostethus</em>). You will get a list of species associated with this genus.</li>
</ol>
<p><img src="figures/session_11_figs/3_NCBI_Leucostethus_2.png" width="90%" /></p>
<ol start="4" style="list-style-type: upper-alpha">
<li>For this example, we will click on <em>Leucostethus fugax</em>. You will get a list of sequences associated with this taxon.</li>
</ol>
<p><img src="figures/session_11_figs/4_NCBI_Leucostethus_3.png" width="90%" /></p>
<ol start="5" style="list-style-type: upper-alpha">
<li>For this example, we want the nucleotide sequences and their accession numbers. You will click on <strong>19</strong> next to Nucleotide.</li>
</ol>
<p><img src="figures/session_11_figs/4_NCBI_Leucostethus_3.png" width="90%" /></p>
<ol start="6" style="list-style-type: upper-alpha">
<li>For this example, we get the accession number <code>MW042037.1</code> that is associated with <strong>Leucostethus fugax voucher QCAZ16513 cytochrome oxidase subunit I (COI) gene </strong>.</li>
</ol>
<p><img src="figures/session_11_figs/5_NCBI_Leucostethus_4.png" width="90%" /></p>
</div>
<div id="retrieving-accession-numbers" class="section level2" number="11.3">
<h2><span class="header-section-number">11.3</span> Retrieving accession numbers</h2>
<p><strong>3)</strong> You can use ‘rentrez’ to explore each genus for such specific sequences rather than the tedious option <strong>2)</strong> as indicated above.</p>
<p>We will retrieve sequences for the other outgroups of <em>Ameerega</em>: <em>Colostethus</em>, <em>Epipedobates</em> and <em>Silverstoneia</em>.</p>
<div class="sourceCode" id="cb456"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb456-1"><a href="sequence-alignments.html#cb456-1" aria-hidden="true" tabindex="-1"></a><span class="do">## sequences for Colostethus and the corresponding COI gene</span></span>
<span id="cb456-2"><a href="sequence-alignments.html#cb456-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb456-3"><a href="sequence-alignments.html#cb456-3" aria-hidden="true" tabindex="-1"></a>Colostethus_name_COX1 <span class="ot">&lt;-</span> <span class="st">&quot;Colostethus[Organism] AND COI[Gene]&quot;</span></span>
<span id="cb456-4"><a href="sequence-alignments.html#cb456-4" aria-hidden="true" tabindex="-1"></a>Colostethus_seq_IDs <span class="ot">&lt;-</span> <span class="fu">entrez_search</span>(<span class="at">db=</span><span class="st">&quot;nuccore&quot;</span>, <span class="at">term=</span> Colostethus_name_COX1)</span>
<span id="cb456-5"><a href="sequence-alignments.html#cb456-5" aria-hidden="true" tabindex="-1"></a>Colostethus_seq_fasta <span class="ot">&lt;-</span> <span class="fu">entrez_fetch</span>(<span class="at">db=</span><span class="st">&quot;nuccore&quot;</span>, <span class="at">id=</span>Colostethus_seq_IDs<span class="sc">$</span>ids, <span class="at">rettype=</span><span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb456-6"><a href="sequence-alignments.html#cb456-6" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(Colostethus_seq_fasta)</span>
<span id="cb456-7"><a href="sequence-alignments.html#cb456-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb456-8"><a href="sequence-alignments.html#cb456-8" aria-hidden="true" tabindex="-1"></a><span class="co"># for example, we will retrieve “KR862889.1 Colostethus pratti voucher CH 6816 cytochrome oxidase subunit 1 (COI) gene, partial cds; mitochondria”</span></span>
<span id="cb456-9"><a href="sequence-alignments.html#cb456-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb456-10"><a href="sequence-alignments.html#cb456-10" aria-hidden="true" tabindex="-1"></a><span class="do">## sequences for Epipedobates and the corresponding COI gene</span></span>
<span id="cb456-11"><a href="sequence-alignments.html#cb456-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb456-12"><a href="sequence-alignments.html#cb456-12" aria-hidden="true" tabindex="-1"></a>Epipedobates_name_COX1 <span class="ot">&lt;-</span> <span class="st">&quot;Epipedobates[Organism] AND COI[Gene]&quot;</span></span>
<span id="cb456-13"><a href="sequence-alignments.html#cb456-13" aria-hidden="true" tabindex="-1"></a>Epipedobates_seq_IDs <span class="ot">&lt;-</span> <span class="fu">entrez_search</span>(<span class="at">db=</span><span class="st">&quot;nuccore&quot;</span>, <span class="at">term=</span> Epipedobates_name_COX1)</span>
<span id="cb456-14"><a href="sequence-alignments.html#cb456-14" aria-hidden="true" tabindex="-1"></a>Epipedobates_seq_fasta <span class="ot">&lt;-</span> <span class="fu">entrez_fetch</span>(<span class="at">db=</span><span class="st">&quot;nuccore&quot;</span>, <span class="at">id=</span>Epipedobates_seq_IDs<span class="sc">$</span>ids, <span class="at">rettype=</span><span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb456-15"><a href="sequence-alignments.html#cb456-15" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(Epipedobates_seq_fasta)</span>
<span id="cb456-16"><a href="sequence-alignments.html#cb456-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb456-17"><a href="sequence-alignments.html#cb456-17" aria-hidden="true" tabindex="-1"></a><span class="co"># for example, we will retrieve “MW042036.1 Epipedobates machalilla voucher QCAZ16527 cytochrome oxidase subunit I (COI) gene, partial cds; mitochondrial”</span></span>
<span id="cb456-18"><a href="sequence-alignments.html#cb456-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb456-19"><a href="sequence-alignments.html#cb456-19" aria-hidden="true" tabindex="-1"></a><span class="do">## sequences for Silverstoneia and the corresponding COI gene</span></span>
<span id="cb456-20"><a href="sequence-alignments.html#cb456-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb456-21"><a href="sequence-alignments.html#cb456-21" aria-hidden="true" tabindex="-1"></a>Silverstoneia_name_COX1 <span class="ot">&lt;-</span> <span class="st">&quot;Silverstoneia[Organism] AND COI[Gene]&quot;</span></span>
<span id="cb456-22"><a href="sequence-alignments.html#cb456-22" aria-hidden="true" tabindex="-1"></a>Silverstoneia_seq_IDs <span class="ot">&lt;-</span> <span class="fu">entrez_search</span>(<span class="at">db=</span><span class="st">&quot;nuccore&quot;</span>, <span class="at">term=</span> Silverstoneia_name_COX1)</span>
<span id="cb456-23"><a href="sequence-alignments.html#cb456-23" aria-hidden="true" tabindex="-1"></a>Silverstoneia_seq_fasta <span class="ot">&lt;-</span> <span class="fu">entrez_fetch</span>(<span class="at">db=</span><span class="st">&quot;nuccore&quot;</span>, <span class="at">id=</span>Silverstoneia_seq_IDs<span class="sc">$</span>ids, <span class="at">rettype=</span><span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb456-24"><a href="sequence-alignments.html#cb456-24" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(Silverstoneia_seq_fasta)</span>
<span id="cb456-25"><a href="sequence-alignments.html#cb456-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb456-26"><a href="sequence-alignments.html#cb456-26" aria-hidden="true" tabindex="-1"></a><span class="co"># for example, we will retrieve “MW042039.1 Silverstoneia flotator voucher TNHCFS4804 cytochrome oxidase subunit I (COI) gene, partial cds; mitochondrial”</span></span></code></pre></div>
<p><strong>4)</strong> With the NCBI accession numbers, you can use ‘rentrez’ to extract these sequences to vector and append to the <strong>Ameerega</strong> vector in <strong>1)</strong>.</p>
<p>These accession numbers were determined in <strong>2)</strong> and <strong>3)</strong>: <code>MW042037.1</code>, <code>KR862889.1</code>, <code>MW042036.1</code> and <code>MW042039.1</code>.</p>
<div class="sourceCode" id="cb457"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb457-1"><a href="sequence-alignments.html#cb457-1" aria-hidden="true" tabindex="-1"></a><span class="do">## sequences for outgroups of Ameerega</span></span>
<span id="cb457-2"><a href="sequence-alignments.html#cb457-2" aria-hidden="true" tabindex="-1"></a>outgroups_seq_fasta <span class="ot">&lt;-</span> <span class="fu">entrez_fetch</span>(<span class="at">db=</span><span class="st">&quot;nuccore&quot;</span>, </span>
<span id="cb457-3"><a href="sequence-alignments.html#cb457-3" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">id=</span><span class="fu">c</span>(<span class="st">&quot;MW042037.1&quot;</span>, <span class="st">&quot;KR862889.1&quot;</span>, <span class="st">&quot;MW042036.1&quot;</span>, <span class="st">&quot;MW042039.1&quot;</span>), </span>
<span id="cb457-4"><a href="sequence-alignments.html#cb457-4" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">rettype=</span><span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb457-5"><a href="sequence-alignments.html#cb457-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(outgroups_seq_fasta)</span>
<span id="cb457-6"><a href="sequence-alignments.html#cb457-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;MW042037.1 Leucostethus fugax voucher QCAZ16513 cytochrome oxidase subunit I (COI) gene, partial cds; mitochondrial</span></span>
<span id="cb457-7"><a href="sequence-alignments.html#cb457-7" aria-hidden="true" tabindex="-1"></a><span class="co">#GTGATAATTACCCGATGATTATTTTCCACAAACCACAAAGACATTGGAACCCTATACCTAGTATTTGGCG</span></span>
<span id="cb457-8"><a href="sequence-alignments.html#cb457-8" aria-hidden="true" tabindex="-1"></a><span class="co">#CTTGGGCAGGAATAGTGGGGACTGCCCTAAGTCTCCTCATTCGTGCAGAATTAAGCCAGCCTGGATCCTT</span></span>
<span id="cb457-9"><a href="sequence-alignments.html#cb457-9" aria-hidden="true" tabindex="-1"></a><span class="co">#ACTTGGTGACGACCAAATTTACAACGTAATCGTCACTGCCCACGCTTTCGTAATAATTTTTTTTATGGTT</span></span>
<span id="cb457-10"><a href="sequence-alignments.html#cb457-10" aria-hidden="true" tabindex="-1"></a><span class="co">#ATACCCATCTTAATTGGGGGATTTGGCAATTGATTAGTCCCATTAATAATTGGAGCACCAGACATAGCTT</span></span>
<span id="cb457-11"><a href="sequence-alignments.html#cb457-11" aria-hidden="true" tabindex="-1"></a><span class="co">#TTCCTCGAATAAACAATATAAGCTTCTGACTTTTACCCCCCTCTTTCCTTCTTCTATTAGCTTCTGCAGG</span></span>
<span id="cb457-12"><a href="sequence-alignments.html#cb457-12" aria-hidden="true" tabindex="-1"></a><span class="co">#...</span></span></code></pre></div>
<p>Now, we can append this vector to <code>Ameerega_seqs_fasta</code> and we can save these sequences in a text file in your working directory.</p>
<div class="sourceCode" id="cb458"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb458-1"><a href="sequence-alignments.html#cb458-1" aria-hidden="true" tabindex="-1"></a><span class="do">## append sequences in a single vector</span></span>
<span id="cb458-2"><a href="sequence-alignments.html#cb458-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb458-3"><a href="sequence-alignments.html#cb458-3" aria-hidden="true" tabindex="-1"></a>all_sequences <span class="ot">&lt;-</span> <span class="fu">c</span>(Ameerega_seqs_fasta, outgroups_seq_fasta)</span>
<span id="cb458-4"><a href="sequence-alignments.html#cb458-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb458-5"><a href="sequence-alignments.html#cb458-5" aria-hidden="true" tabindex="-1"></a><span class="do">## this is exclusive to your OWN COMPUTER change it accordingly</span></span>
<span id="cb458-6"><a href="sequence-alignments.html#cb458-6" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/my_working_directory&quot;</span>)</span>
<span id="cb458-7"><a href="sequence-alignments.html#cb458-7" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(all_sequences, <span class="st">&quot;Ameerega_COI_seqs_fasta.txt&quot;</span>)</span></code></pre></div>
<p>Now we can import the nucleotide sequences from this file that are in <code>fasta</code> format into R.</p>
<div class="sourceCode" id="cb459"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb459-1"><a href="sequence-alignments.html#cb459-1" aria-hidden="true" tabindex="-1"></a><span class="do">## make sure Biostrings is loaded</span></span>
<span id="cb459-2"><a href="sequence-alignments.html#cb459-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(Biostrings)</span>
<span id="cb459-3"><a href="sequence-alignments.html#cb459-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb459-4"><a href="sequence-alignments.html#cb459-4" aria-hidden="true" tabindex="-1"></a><span class="do">## load sequences</span></span>
<span id="cb459-5"><a href="sequence-alignments.html#cb459-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb459-6"><a href="sequence-alignments.html#cb459-6" aria-hidden="true" tabindex="-1"></a>Ameerega_Biostrings_set <span class="ot">&lt;-</span> <span class="fu">readDNAStringSet</span>(<span class="at">filepath =</span> <span class="st">&quot;~/Desktop/Teach_R/my_working_directory/Ameerega_COI_seqs_fasta.txt&quot;</span>, </span>
<span id="cb459-7"><a href="sequence-alignments.html#cb459-7" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">format =</span> <span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb459-8"><a href="sequence-alignments.html#cb459-8" aria-hidden="true" tabindex="-1"></a>Ameerega_Biostrings_set</span>
<span id="cb459-9"><a href="sequence-alignments.html#cb459-9" aria-hidden="true" tabindex="-1"></a><span class="co">#A DNAStringSet instance of length 24</span></span>
<span id="cb459-10"><a href="sequence-alignments.html#cb459-10" aria-hidden="true" tabindex="-1"></a><span class="co">#     width seq                                                                                                                  names               </span></span>
<span id="cb459-11"><a href="sequence-alignments.html#cb459-11" aria-hidden="true" tabindex="-1"></a><span class="co"># [1]  1539 GTGATAATTACTCGATGATTATTTTCCACCAACCATAAAGATATTGGAACCTTATAC...TATACGGGTCTCCCCCACCTTATCACACATTTGAGGAAGCCGTTTACTCCAAAATT MW042032.1 Ameere...</span></span>
<span id="cb459-12"><a href="sequence-alignments.html#cb459-12" aria-hidden="true" tabindex="-1"></a><span class="co"># [2]  1539 GTGATAATTACTCGATGATTATTTTCCACCAACCATAAAGACATTGGAACCTTATAT...TGTACGGATCCCCCCCTCCCTACCATACATTTGAAGAAGCCGTTTATTCCAAAATT MW042031.1 Ameere...</span></span>
<span id="cb459-13"><a href="sequence-alignments.html#cb459-13" aria-hidden="true" tabindex="-1"></a><span class="co"># [3]  1539 GTGATAATTACTCGATGATTATTTTCTACCAACCACAAAGACATCGGAACTTTATAC...TATACGGATCCCCCCCACCTTACCACACTTTTGAGGAAGCCGTTTACTCCAAAATT MW042030.1 Ameere...</span></span>
<span id="cb459-14"><a href="sequence-alignments.html#cb459-14" aria-hidden="true" tabindex="-1"></a><span class="co"># [4]   646 AACTTTATACCTAGTATTTGGGGCATGAGCGGGCATAGTCGGTACTGCTCTTAGCCT...GTAACCTAAATACCACTTTTTTTGACCCGGCAGGGGGAGGTGACCCTGTCCTATAC KU494334.1 Ameere...</span></span>
<span id="cb459-15"><a href="sequence-alignments.html#cb459-15" aria-hidden="true" tabindex="-1"></a><span class="co"># [5]   647 GAACTTTATACCTAGTGTTTGGGGCATGAGCAGGCATAGTCGGCACTGCTCTCAGCC...GTAACTTAAATACCACCTTCTTTGACCCAGCAGGGGGAGGTGACCCTGTTCTATAC KU494333.1 Ameere...</span></span>
<span id="cb459-16"><a href="sequence-alignments.html#cb459-16" aria-hidden="true" tabindex="-1"></a><span class="co"># ...   ... ...</span></span>
<span id="cb459-17"><a href="sequence-alignments.html#cb459-17" aria-hidden="true" tabindex="-1"></a><span class="co">#[20]   658 AACTTTATATCTAGTATTTGGGGCATGGGCAGGCATAGTCGGTACTGCTCTCAGCCT...CCACTTTCTTTGACCCAGCAGGAGGAGGCGACCCCGTCTTATACCAACACCTGTTC DQ502825.1 Epiped...</span></span>
<span id="cb459-18"><a href="sequence-alignments.html#cb459-18" aria-hidden="true" tabindex="-1"></a><span class="co">#[21]  1539 GTGATAATTACCCGATGATTATTTTCCACAAACCACAAAGACATTGGAACCCTATAC...TATACGGAACCCCTCCCCCTTATCACACATTTGAAGAAGCTGTTTACTCCAAAATT MW042037.1 Leucos...</span></span>
<span id="cb459-19"><a href="sequence-alignments.html#cb459-19" aria-hidden="true" tabindex="-1"></a><span class="co">#[22]   658 AACTTTATACNTTGTATTNGGGGCATGGGCNGGAATAGTNGGAACCGCCCTAAGCCT...CTACTTTCTTCGACCCGGCTGGTGGAGGAGACCCCGTTCTCTACCAACATCTCTTT KR862889.1 Colost...</span></span>
<span id="cb459-20"><a href="sequence-alignments.html#cb459-20" aria-hidden="true" tabindex="-1"></a><span class="co">#[23]  1539 GTGATAATTACCCGATGATTATTCTCCACAAACCATAAAGATATCGGAACCCTATAT...TATATGGCTCCCCTCCTCCTTACCACACATTCGAGGAAGCTGTTTATTCTAAAGTA MW042036.1 Epiped...</span></span>
<span id="cb459-21"><a href="sequence-alignments.html#cb459-21" aria-hidden="true" tabindex="-1"></a><span class="co">#[24]  1539 GTGATAATTACCCGATGATTATTTTCTACAAACCACAAAGACATTGGAACTTTATAT...TGTATGGATCACCACCACCATATCACACGTTTGAAGAAGCTGTTTATTCTAAAATT MW042039.1 Silver...</span></span></code></pre></div>
</div>
<div id="alignment-with-decipher" class="section level2" number="11.4">
<h2><span class="header-section-number">11.4</span> Alignment with <code>DECIPHER</code></h2>
<p>We will align our collected sequences with <a href="https://www.bioconductor.org/packages/release/bioc/html/DECIPHER.html" target="_blank"><strong>DECIPHER</strong></a>. This R package has several vignettes, but this <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/DECIPHER/inst/doc/ArtOfAlignmentInR.pdf" target="_blank"><strong>one</strong></a> is most useful for alignments.</p>
<p><strong>5)</strong> The <code>DECIPHER</code> aligner is very easy to use as long as you have already your sequences as a XStringSet from <code>Biostrings</code> (i.e., DNAStringSet, RNAStringSet or AAStringSet).</p>
<div class="sourceCode" id="cb460"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb460-1"><a href="sequence-alignments.html#cb460-1" aria-hidden="true" tabindex="-1"></a><span class="do">## install DECIPHER</span></span>
<span id="cb460-2"><a href="sequence-alignments.html#cb460-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb460-3"><a href="sequence-alignments.html#cb460-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb460-4"><a href="sequence-alignments.html#cb460-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb460-5"><a href="sequence-alignments.html#cb460-5" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;DECIPHER&quot;</span>)</span>
<span id="cb460-6"><a href="sequence-alignments.html#cb460-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb460-7"><a href="sequence-alignments.html#cb460-7" aria-hidden="true" tabindex="-1"></a><span class="do">## make sure that DECIPHER and Biostrings are loaded</span></span>
<span id="cb460-8"><a href="sequence-alignments.html#cb460-8" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(DECIPHER)</span>
<span id="cb460-9"><a href="sequence-alignments.html#cb460-9" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(Biostrings)</span>
<span id="cb460-10"><a href="sequence-alignments.html#cb460-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb460-11"><a href="sequence-alignments.html#cb460-11" aria-hidden="true" tabindex="-1"></a><span class="do">## load YOUR sequences or the example</span></span>
<span id="cb460-12"><a href="sequence-alignments.html#cb460-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb460-13"><a href="sequence-alignments.html#cb460-13" aria-hidden="true" tabindex="-1"></a>Ameerega_Biostrings_set <span class="ot">&lt;-</span> <span class="fu">readDNAStringSet</span>(<span class="at">filepath =</span> <span class="st">&quot;~/Desktop/Teach_R/my_working_directory/Ameerega_COI_seqs_fasta.txt&quot;</span>, </span>
<span id="cb460-14"><a href="sequence-alignments.html#cb460-14" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">format =</span> <span class="st">&quot;fasta&quot;</span>)</span></code></pre></div>
<p>We just can proceed to do the alignment with function <code>AlignSeqs()</code>.</p>
<div class="sourceCode" id="cb461"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb461-1"><a href="sequence-alignments.html#cb461-1" aria-hidden="true" tabindex="-1"></a><span class="do">## simple alignment procedure for DNA</span></span>
<span id="cb461-2"><a href="sequence-alignments.html#cb461-2" aria-hidden="true" tabindex="-1"></a>Ameerega_aligned <span class="ot">&lt;-</span> <span class="fu">AlignSeqs</span>(Ameerega_Biostrings_set)</span>
<span id="cb461-3"><a href="sequence-alignments.html#cb461-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Determining distance matrix based on shared 9-mers:</span></span>
<span id="cb461-4"><a href="sequence-alignments.html#cb461-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  |==========================================================================================================================================| 100%</span></span>
<span id="cb461-5"><a href="sequence-alignments.html#cb461-5" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb461-6"><a href="sequence-alignments.html#cb461-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Time difference of 0.08 secs</span></span>
<span id="cb461-7"><a href="sequence-alignments.html#cb461-7" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb461-8"><a href="sequence-alignments.html#cb461-8" aria-hidden="true" tabindex="-1"></a><span class="co">#Clustering into groups by similarity:</span></span>
<span id="cb461-9"><a href="sequence-alignments.html#cb461-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  |==========================================================================================================================================| 100%</span></span>
<span id="cb461-10"><a href="sequence-alignments.html#cb461-10" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb461-11"><a href="sequence-alignments.html#cb461-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Alignment converged - skipping remaining iteration.</span></span>
<span id="cb461-12"><a href="sequence-alignments.html#cb461-12" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb461-13"><a href="sequence-alignments.html#cb461-13" aria-hidden="true" tabindex="-1"></a><span class="co">#Refining the alignment:</span></span>
<span id="cb461-14"><a href="sequence-alignments.html#cb461-14" aria-hidden="true" tabindex="-1"></a><span class="co">#  |==========================================================================================================================================| 100%</span></span>
<span id="cb461-15"><a href="sequence-alignments.html#cb461-15" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb461-16"><a href="sequence-alignments.html#cb461-16" aria-hidden="true" tabindex="-1"></a><span class="co">#Time difference of 0.18 secs</span></span>
<span id="cb461-17"><a href="sequence-alignments.html#cb461-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb461-18"><a href="sequence-alignments.html#cb461-18" aria-hidden="true" tabindex="-1"></a><span class="do">## you can call DNAstringSet aligned</span></span>
<span id="cb461-19"><a href="sequence-alignments.html#cb461-19" aria-hidden="true" tabindex="-1"></a><span class="co">#Ameerega_aligned</span></span>
<span id="cb461-20"><a href="sequence-alignments.html#cb461-20" aria-hidden="true" tabindex="-1"></a><span class="co">#  A DNAStringSet instance of length 24</span></span>
<span id="cb461-21"><a href="sequence-alignments.html#cb461-21" aria-hidden="true" tabindex="-1"></a><span class="co">#     width seq                                                                                                                  names               </span></span>
<span id="cb461-22"><a href="sequence-alignments.html#cb461-22" aria-hidden="true" tabindex="-1"></a><span class="co"># [1]  1539 GTGATAATTACTCGATGATTATTTTCCACCAACCATAAAGATATTGGAACCTTATAC...TATACGGGTCTCCCCCACCTTATCACACATTTGAGGAAGCCGTTTACTCCAAAATT MW042032.1 Ameere...</span></span>
<span id="cb461-23"><a href="sequence-alignments.html#cb461-23" aria-hidden="true" tabindex="-1"></a><span class="co"># [2]  1539 GTGATAATTACTCGATGATTATTTTCCACCAACCATAAAGACATTGGAACCTTATAT...TGTACGGATCCCCCCCTCCCTACCATACATTTGAAGAAGCCGTTTATTCCAAAATT MW042031.1 Ameere...</span></span>
<span id="cb461-24"><a href="sequence-alignments.html#cb461-24" aria-hidden="true" tabindex="-1"></a><span class="co"># [3]  1539 GTGATAATTACTCGATGATTATTTTCTACCAACCACAAAGACATCGGAACTTTATAC...TATACGGATCCCCCCCACCTTACCACACTTTTGAGGAAGCCGTTTACTCCAAAATT MW042030.1 Ameere...</span></span>
<span id="cb461-25"><a href="sequence-alignments.html#cb461-25" aria-hidden="true" tabindex="-1"></a><span class="co"># [4]  1539 -----------------------------------------------AACTTTATAC...-------------------------------------------------------- KU494334.1 Ameere...</span></span>
<span id="cb461-26"><a href="sequence-alignments.html#cb461-26" aria-hidden="true" tabindex="-1"></a><span class="co"># [5]  1539 ----------------------------------------------GAACTTTATAC...-------------------------------------------------------- KU494333.1 Ameere...</span></span>
<span id="cb461-27"><a href="sequence-alignments.html#cb461-27" aria-hidden="true" tabindex="-1"></a><span class="co"># ...   ... ...</span></span>
<span id="cb461-28"><a href="sequence-alignments.html#cb461-28" aria-hidden="true" tabindex="-1"></a><span class="co">#[20]  1539 -----------------------------------------------AACTTTATAT...-------------------------------------------------------- DQ502825.1 Epiped...</span></span>
<span id="cb461-29"><a href="sequence-alignments.html#cb461-29" aria-hidden="true" tabindex="-1"></a><span class="co">#[21]  1539 GTGATAATTACCCGATGATTATTTTCCACAAACCACAAAGACATTGGAACCCTATAC...TATACGGAACCCCTCCCCCTTATCACACATTTGAAGAAGCTGTTTACTCCAAAATT MW042037.1 Leucos...</span></span>
<span id="cb461-30"><a href="sequence-alignments.html#cb461-30" aria-hidden="true" tabindex="-1"></a><span class="co">#[22]  1539 -----------------------------------------------AACTTTATAC...-------------------------------------------------------- KR862889.1 Colost...</span></span>
<span id="cb461-31"><a href="sequence-alignments.html#cb461-31" aria-hidden="true" tabindex="-1"></a><span class="co">#[23]  1539 GTGATAATTACCCGATGATTATTCTCCACAAACCATAAAGATATCGGAACCCTATAT...TATATGGCTCCCCTCCTCCTTACCACACATTCGAGGAAGCTGTTTATTCTAAAGTA MW042036.1 Epiped...</span></span>
<span id="cb461-32"><a href="sequence-alignments.html#cb461-32" aria-hidden="true" tabindex="-1"></a><span class="co">#[24]  1539 GTGATAATTACCCGATGATTATTTTCTACAAACCACAAAGACATTGGAACTTTATAT...TGTATGGATCACCACCACCATATCACACGTTTGAAGAAGCTGTTTATTCTAAAATT MW042039.1 Silver...</span></span></code></pre></div>
</div>
<div id="visualize-alignment" class="section level2" number="11.5">
<h2><span class="header-section-number">11.5</span> Visualize alignment</h2>
<p><strong>6)</strong> We can visualize the alignment with <code>BrowseSeqs()</code> and a new tab in your browser will appear with the alignment to visualize.</p>
<div class="sourceCode" id="cb462"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb462-1"><a href="sequence-alignments.html#cb462-1" aria-hidden="true" tabindex="-1"></a><span class="do">## simple visualization of alignment</span></span>
<span id="cb462-2"><a href="sequence-alignments.html#cb462-2" aria-hidden="true" tabindex="-1"></a><span class="fu">BrowseSeqs</span>(Ameerega_aligned)</span></code></pre></div>
<p><img src="figures/session_11_figs/6_DECIPHER_1.png" width="90%" /></p>
<p><strong>7)</strong> We can align the sequences based on their amino acid translation using <code>AlignTranslation()</code>. Notice that the genetic code needs to be specified, if this is not the standard.</p>
<div class="sourceCode" id="cb463"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb463-1"><a href="sequence-alignments.html#cb463-1" aria-hidden="true" tabindex="-1"></a><span class="do">## if the genetic code is standard, you do not need to define it. In this case, we have sequences from the mitochondria and this needs to be defined as follows.</span></span>
<span id="cb463-2"><a href="sequence-alignments.html#cb463-2" aria-hidden="true" tabindex="-1"></a>mt_vertebrate_code <span class="ot">&lt;-</span> <span class="fu">getGeneticCode</span>(<span class="st">&quot;SGC1&quot;</span>)</span>
<span id="cb463-3"><a href="sequence-alignments.html#cb463-3" aria-hidden="true" tabindex="-1"></a>mt_vertebrate_code </span>
<span id="cb463-4"><a href="sequence-alignments.html#cb463-4" aria-hidden="true" tabindex="-1"></a><span class="co">#TTT TTC TTA TTG TCT TCC TCA TCG TAT TAC TAA TAG TGT TGC TGA TGG CTT CTC CTA CTG CCT CCC CCA CCG CAT CAC CAA CAG CGT CGC CGA CGG ATT ATC ATA ATG ACT </span></span>
<span id="cb463-5"><a href="sequence-alignments.html#cb463-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&quot;F&quot; &quot;F&quot; &quot;L&quot; &quot;L&quot; &quot;S&quot; &quot;S&quot; &quot;S&quot; &quot;S&quot; &quot;Y&quot; &quot;Y&quot; &quot;*&quot; &quot;*&quot; &quot;C&quot; &quot;C&quot; &quot;W&quot; &quot;W&quot; &quot;L&quot; &quot;L&quot; &quot;L&quot; &quot;L&quot; &quot;P&quot; &quot;P&quot; &quot;P&quot; &quot;P&quot; &quot;H&quot; &quot;H&quot; &quot;Q&quot; &quot;Q&quot; &quot;R&quot; &quot;R&quot; &quot;R&quot; &quot;R&quot; &quot;I&quot; &quot;I&quot; &quot;M&quot; &quot;M&quot; &quot;T&quot; </span></span>
<span id="cb463-6"><a href="sequence-alignments.html#cb463-6" aria-hidden="true" tabindex="-1"></a><span class="co">#ACC ACA ACG AAT AAC AAA AAG AGT AGC AGA AGG GTT GTC GTA GTG GCT GCC GCA GCG GAT GAC GAA GAG GGT GGC GGA GGG </span></span>
<span id="cb463-7"><a href="sequence-alignments.html#cb463-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&quot;T&quot; &quot;T&quot; &quot;T&quot; &quot;N&quot; &quot;N&quot; &quot;K&quot; &quot;K&quot; &quot;S&quot; &quot;S&quot; &quot;*&quot; &quot;*&quot; &quot;V&quot; &quot;V&quot; &quot;V&quot; &quot;V&quot; &quot;A&quot; &quot;A&quot; &quot;A&quot; &quot;A&quot; &quot;D&quot; &quot;D&quot; &quot;E&quot; &quot;E&quot; &quot;G&quot; &quot;G&quot; &quot;G&quot; &quot;G&quot; </span></span>
<span id="cb463-8"><a href="sequence-alignments.html#cb463-8" aria-hidden="true" tabindex="-1"></a><span class="co">#attr(,&quot;alt_init_codons&quot;)</span></span>
<span id="cb463-9"><a href="sequence-alignments.html#cb463-9" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;ATT&quot; &quot;ATC&quot; &quot;GTG&quot;</span></span>
<span id="cb463-10"><a href="sequence-alignments.html#cb463-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb463-11"><a href="sequence-alignments.html#cb463-11" aria-hidden="true" tabindex="-1"></a><span class="do">## you translate and align your nucleotide sequence into amino acids (AA). </span></span>
<span id="cb463-12"><a href="sequence-alignments.html#cb463-12" aria-hidden="true" tabindex="-1"></a>Ameerega_AA_set <span class="ot">&lt;-</span> <span class="fu">AlignTranslation</span>(Ameerega_Biostrings_set, <span class="at">type=</span><span class="st">&quot;AAStringSet&quot;</span>, <span class="at">geneticCode =</span> mt_vertebrate_code)</span>
<span id="cb463-13"><a href="sequence-alignments.html#cb463-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb463-14"><a href="sequence-alignments.html#cb463-14" aria-hidden="true" tabindex="-1"></a><span class="do">## see the AAStringSet</span></span>
<span id="cb463-15"><a href="sequence-alignments.html#cb463-15" aria-hidden="true" tabindex="-1"></a>Ameerega_AA_set</span>
<span id="cb463-16"><a href="sequence-alignments.html#cb463-16" aria-hidden="true" tabindex="-1"></a><span class="co">#A AAStringSet instance of length 24</span></span>
<span id="cb463-17"><a href="sequence-alignments.html#cb463-17" aria-hidden="true" tabindex="-1"></a><span class="co">#     width seq                                                                                                                  names               </span></span>
<span id="cb463-18"><a href="sequence-alignments.html#cb463-18" aria-hidden="true" tabindex="-1"></a><span class="co"># [1]   555 MMITRWLFSTNHKDIGTLYLVFGAWAGMVGTALSLLIRAELSQPGSLLGDDQIYNVI...SLISLVAVIIMMFIIWEAFSSKRLPFPAEMTPTNVEWLYGSPPPYHTFEEAVYSKI MW042032.1 Ameere...</span></span>
<span id="cb463-19"><a href="sequence-alignments.html#cb463-19" aria-hidden="true" tabindex="-1"></a><span class="co"># [2]   555 MMITRWLFSTNHKDIGTLYLVFGAWAGMVGTALSLLIRAELSQPGSLLGDDQIYNVI...SLISLVAVIIMMFIIWEAFSSKRLPLPAEMTPTNVEWLYGSPPPYHTFEEAVYSKI MW042031.1 Ameere...</span></span>
<span id="cb463-20"><a href="sequence-alignments.html#cb463-20" aria-hidden="true" tabindex="-1"></a><span class="co"># [3]   555 MMITRWLFSTNHKDIGTLYLVFGAWAGMVGTALSLLIRAELSQPGSLLGDDQIYNVI...SLISLVAVIIMMFIIWEAFSSKRLPLPAEMTPTNVEWLYGSPPPYHTFEEAVYSKI MW042030.1 Ameere...</span></span>
<span id="cb463-21"><a href="sequence-alignments.html#cb463-21" aria-hidden="true" tabindex="-1"></a><span class="co"># [4]   555 ---------------+TLYLVFGAWAGMVGTALSLLIRAELSQPGSLLGDDQIYNVI...-------------------------------------------------------- KU494334.1 Ameere...</span></span>
<span id="cb463-22"><a href="sequence-alignments.html#cb463-22" aria-hidden="true" tabindex="-1"></a><span class="co"># [5]   555 ---------------+TLYLVFGAWAGMVGTALSLLIRAELSQPGSLLGDDQIYNVI...-------------------------------------------------------- KU494333.1 Ameere...</span></span>
<span id="cb463-23"><a href="sequence-alignments.html#cb463-23" aria-hidden="true" tabindex="-1"></a><span class="co"># ...   ... ...</span></span>
<span id="cb463-24"><a href="sequence-alignments.html#cb463-24" aria-hidden="true" tabindex="-1"></a><span class="co">#[20]   555 ---------------+TLYLVFGAWAGMVGTALSLLIRAELSQPGSLLGDDQIYNVI...-------------------------------------------------------- DQ502825.1 Epiped...</span></span>
<span id="cb463-25"><a href="sequence-alignments.html#cb463-25" aria-hidden="true" tabindex="-1"></a><span class="co">#[21]   555 MMITRWLFSTNHKDIGTLYLVFGAWAGMVGTALSLLIRAELSQPGSLLGDDQIYNVI...SLISLVAVILMMFIIWEAFSSKRLFLTAEMTSTNVEWLYGTPPPYHTFEEAVYSKI MW042037.1 Leucos...</span></span>
<span id="cb463-26"><a href="sequence-alignments.html#cb463-26" aria-hidden="true" tabindex="-1"></a><span class="co">#[22]   555 ---------------+TLYXVXGAWAGMVGTALSLLIRAELSQPGSLLGDDQIYNVI...-------------------------------------------------------- KR862889.1 Colost...</span></span>
<span id="cb463-27"><a href="sequence-alignments.html#cb463-27" aria-hidden="true" tabindex="-1"></a><span class="co">#[23]   555 MMITRWLFSTNHKDIGTLYLVFGAWAGMVGTALSLLIRAELSQPGSLLGDDQIYNVI...SLISLVAVIIMMFIIWEAFSSKRLFLNAEMTPTNVEWLYGSPPPYHTFEEAVYSKV MW042036.1 Epiped...</span></span>
<span id="cb463-28"><a href="sequence-alignments.html#cb463-28" aria-hidden="true" tabindex="-1"></a><span class="co">#[24]   555 MMITRWLFSTNHKDIGTLYLVFGAWAGMVGTALSLLIRAELSQPGSLLGDDQIYNVI...SLISLVAVILMMFIIWEAFSSKRLFLNAEMTPTNVEWLYGSPPPYHTFEEAVYSKI MW042039.1 Silver...</span></span>
<span id="cb463-29"><a href="sequence-alignments.html#cb463-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb463-30"><a href="sequence-alignments.html#cb463-30" aria-hidden="true" tabindex="-1"></a><span class="do">## simple visualization of AA alignment with references to sequence 1</span></span>
<span id="cb463-31"><a href="sequence-alignments.html#cb463-31" aria-hidden="true" tabindex="-1"></a><span class="fu">BrowseSeqs</span>(Ameerega_AA_set, <span class="at">highlight=</span><span class="dv">1</span>)</span></code></pre></div>
<p><img src="figures/session_11_figs/7_DECIPHER_2.png" width="90%" /></p>
<p><strong>8)</strong> You can further improve the alignment with the function <code>AdjustAlignment()</code>. This function makes small adjustments with the goal of removing artifacts of the progressive alignment process by shifting groups of gaps left and right to find their optimal positioning in a multiple sequence alignment. This function will efficiently correct most obvious inaccuracies that could be found by-eye and this is repeatable and not subjective (i.e., a common criticism of alignments done by-eye).</p>
<div class="sourceCode" id="cb464"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb464-1"><a href="sequence-alignments.html#cb464-1" aria-hidden="true" tabindex="-1"></a><span class="do">## some further adjustments if necessary</span></span>
<span id="cb464-2"><a href="sequence-alignments.html#cb464-2" aria-hidden="true" tabindex="-1"></a>Ameerega_aligned_end <span class="ot">&lt;-</span> <span class="fu">AdjustAlignment</span>(Ameerega_aligned)</span>
<span id="cb464-3"><a href="sequence-alignments.html#cb464-3" aria-hidden="true" tabindex="-1"></a>Ameerega_AA_set_end <span class="ot">&lt;-</span> <span class="fu">AdjustAlignment</span>(Ameerega_AA_set)</span>
<span id="cb464-4"><a href="sequence-alignments.html#cb464-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb464-5"><a href="sequence-alignments.html#cb464-5" aria-hidden="true" tabindex="-1"></a><span class="do">## simple visualization of alignments</span></span>
<span id="cb464-6"><a href="sequence-alignments.html#cb464-6" aria-hidden="true" tabindex="-1"></a><span class="fu">BrowseSeqs</span>(Ameerega_aligned_end)</span>
<span id="cb464-7"><a href="sequence-alignments.html#cb464-7" aria-hidden="true" tabindex="-1"></a><span class="fu">BrowseSeqs</span>(Ameerega_AA_set_end)</span></code></pre></div>
<p>These alignments look pretty good. However, in the <code>Ameerega_AA_set_end</code> one sequence looks problematic: <code>DQ502832.1 Epipedobates braccatus isolate 537 cytochrome oxidase subunit I-like (COI) gene, partial sequence; mitochondrial</code>.</p>
<p><img src="figures/session_11_figs/8_DECIPHER_3.png" width="90%" /></p>
<p>This might be the result of the translation, but we will eliminate this sequence in further analyses.</p>
<div class="sourceCode" id="cb465"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb465-1"><a href="sequence-alignments.html#cb465-1" aria-hidden="true" tabindex="-1"></a><span class="do">## to remove a sequence from DNAStringSet, AAStringSet, we need to make sure is the one that we want to eliminate</span></span>
<span id="cb465-2"><a href="sequence-alignments.html#cb465-2" aria-hidden="true" tabindex="-1"></a><span class="do">## in this case is [18]</span></span>
<span id="cb465-3"><a href="sequence-alignments.html#cb465-3" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(Ameerega_AA_set_end[<span class="dv">18</span>])</span>
<span id="cb465-4"><a href="sequence-alignments.html#cb465-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Formal class &#39;AAStringSet&#39; [package &quot;Biostrings&quot;] with 5 slots</span></span>
<span id="cb465-5"><a href="sequence-alignments.html#cb465-5" aria-hidden="true" tabindex="-1"></a><span class="co">#  ..@ pool           :Formal class &#39;SharedRaw_Pool&#39; [package &quot;XVector&quot;] with 2 slots</span></span>
<span id="cb465-6"><a href="sequence-alignments.html#cb465-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. .. ..@ xp_list                    :List of 1</span></span>
<span id="cb465-7"><a href="sequence-alignments.html#cb465-7" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. .. .. ..$ :&lt;externalptr&gt; </span></span>
<span id="cb465-8"><a href="sequence-alignments.html#cb465-8" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. .. ..@ .link_to_cached_object_list:List of 1</span></span>
<span id="cb465-9"><a href="sequence-alignments.html#cb465-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. .. .. ..$ :&lt;environment: 0x7fdf44ffa898&gt; </span></span>
<span id="cb465-10"><a href="sequence-alignments.html#cb465-10" aria-hidden="true" tabindex="-1"></a><span class="co">#  ..@ ranges         :Formal class &#39;GroupedIRanges&#39; [package &quot;XVector&quot;] with 7 slots</span></span>
<span id="cb465-11"><a href="sequence-alignments.html#cb465-11" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. .. ..@ group          : int 1</span></span>
<span id="cb465-12"><a href="sequence-alignments.html#cb465-12" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. .. ..@ start          : int 9436</span></span>
<span id="cb465-13"><a href="sequence-alignments.html#cb465-13" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. .. ..@ width          : int 555</span></span>
<span id="cb465-14"><a href="sequence-alignments.html#cb465-14" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. .. ..@ NAMES          : chr &quot;DQ502832.1 Epipedobates braccatus isolate 537 cytochrome oxidase subunit I-like (COI) gene, partial sequence; mitochondrial&quot;</span></span>
<span id="cb465-15"><a href="sequence-alignments.html#cb465-15" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. .. ..@ elementType    : chr &quot;ANY&quot;</span></span>
<span id="cb465-16"><a href="sequence-alignments.html#cb465-16" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. .. ..@ elementMetadata: NULL</span></span>
<span id="cb465-17"><a href="sequence-alignments.html#cb465-17" aria-hidden="true" tabindex="-1"></a><span class="co">#  .. .. ..@ metadata       : list()</span></span>
<span id="cb465-18"><a href="sequence-alignments.html#cb465-18" aria-hidden="true" tabindex="-1"></a><span class="co">#  ..@ elementType    : chr &quot;AAString&quot;</span></span>
<span id="cb465-19"><a href="sequence-alignments.html#cb465-19" aria-hidden="true" tabindex="-1"></a><span class="co">#  ..@ elementMetadata: NULL</span></span>
<span id="cb465-20"><a href="sequence-alignments.html#cb465-20" aria-hidden="true" tabindex="-1"></a><span class="co">#  ..@ metadata       : list()</span></span>
<span id="cb465-21"><a href="sequence-alignments.html#cb465-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb465-22"><a href="sequence-alignments.html#cb465-22" aria-hidden="true" tabindex="-1"></a><span class="do">## we remove AA seq 18 and adjust the alignment</span></span>
<span id="cb465-23"><a href="sequence-alignments.html#cb465-23" aria-hidden="true" tabindex="-1"></a>Ameerega_AA_set_clean <span class="ot">&lt;-</span> Ameerega_AA_set[<span class="sc">-</span><span class="dv">18</span>]</span>
<span id="cb465-24"><a href="sequence-alignments.html#cb465-24" aria-hidden="true" tabindex="-1"></a>Ameerega_AA_set_clean <span class="ot">&lt;-</span> <span class="fu">AdjustAlignment</span>(Ameerega_AA_set_clean)</span>
<span id="cb465-25"><a href="sequence-alignments.html#cb465-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb465-26"><a href="sequence-alignments.html#cb465-26" aria-hidden="true" tabindex="-1"></a><span class="do">## we make sure that it is not present in the alignment</span></span>
<span id="cb465-27"><a href="sequence-alignments.html#cb465-27" aria-hidden="true" tabindex="-1"></a><span class="fu">BrowseSeqs</span>(Ameerega_AA_set_clean)</span></code></pre></div>
<p>The sequence with errors is now removed from further analyses.</p>
<p><img src="figures/session_11_figs/9_DECIPHER_4.png" width="90%" /></p>
</div>
<div id="save-alignment-to-file" class="section level2" number="11.6">
<h2><span class="header-section-number">11.6</span> Save alignment to file</h2>
<p><strong>9)</strong> With our aligned sequences, we can export these as <code>fasta</code> and then they can be used to reconstruct phylogenetic trees, get protein properties and other analyses. We can save our sequences as fasta using the function <code>writeXStringSet()</code>.</p>
<div class="sourceCode" id="cb466"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb466-1"><a href="sequence-alignments.html#cb466-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this is exclusive to your OWN COMPUTER change it accordingly</span></span>
<span id="cb466-2"><a href="sequence-alignments.html#cb466-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/my_working_directory&quot;</span>)</span>
<span id="cb466-3"><a href="sequence-alignments.html#cb466-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb466-4"><a href="sequence-alignments.html#cb466-4" aria-hidden="true" tabindex="-1"></a><span class="do">## We save the nucleotide alignment</span></span>
<span id="cb466-5"><a href="sequence-alignments.html#cb466-5" aria-hidden="true" tabindex="-1"></a><span class="fu">writeXStringSet</span>(Ameerega_aligned_end, <span class="at">file=</span><span class="st">&quot;Ameerega_aligned_end.fasta&quot;</span>, <span class="at">format=</span><span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb466-6"><a href="sequence-alignments.html#cb466-6" aria-hidden="true" tabindex="-1"></a><span class="fu">writeXStringSet</span>(Ameerega_AA_set_clean, <span class="at">file=</span><span class="st">&quot;Ameerega_AA_aligned_end.fasta&quot;</span>, <span class="at">format=</span><span class="st">&quot;fasta&quot;</span>)</span></code></pre></div>
</div>
<div id="concatenate-and-align-several-markers" class="section level2" number="11.7">
<h2><span class="header-section-number">11.7</span> Concatenate and align several markers</h2>
<p>In many cases, you might combine several markers, segments, exons or other disconnected (not adjacent) sequences. We will follow a procedure in <code>DECIPHER</code>.</p>
<p><strong>10)</strong> We need to find a sequence that is not adjacent to our COI marker, in this case we will download cytochrome b (CYTB) and COI sequences of <em>Ameerega</em>, <em>Colostethus</em>, <em>Epipedobates</em> and <em>Silverstoneia</em>.</p>
<div class="sourceCode" id="cb467"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb467-1"><a href="sequence-alignments.html#cb467-1" aria-hidden="true" tabindex="-1"></a><span class="do">## make sure rentrez is loaded</span></span>
<span id="cb467-2"><a href="sequence-alignments.html#cb467-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(rentrez)</span>
<span id="cb467-3"><a href="sequence-alignments.html#cb467-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-4"><a href="sequence-alignments.html#cb467-4" aria-hidden="true" tabindex="-1"></a><span class="do">## sequences for of all using a loop and list to store each genus CYTB sequences</span></span>
<span id="cb467-5"><a href="sequence-alignments.html#cb467-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-6"><a href="sequence-alignments.html#cb467-6" aria-hidden="true" tabindex="-1"></a>my_genus <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Ameerega&quot;</span>, <span class="st">&quot;Colostethus&quot;</span>, <span class="st">&quot;Epipedobates&quot;</span>, <span class="st">&quot;Silverstoneia&quot;</span>, <span class="st">&quot;Leucostethus&quot;</span>)</span>
<span id="cb467-7"><a href="sequence-alignments.html#cb467-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-8"><a href="sequence-alignments.html#cb467-8" aria-hidden="true" tabindex="-1"></a><span class="do">## loop for CYTB</span></span>
<span id="cb467-9"><a href="sequence-alignments.html#cb467-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-10"><a href="sequence-alignments.html#cb467-10" aria-hidden="true" tabindex="-1"></a>collect_CYTB_list_seqs <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb467-11"><a href="sequence-alignments.html#cb467-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-12"><a href="sequence-alignments.html#cb467-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(my_genus)) {</span>
<span id="cb467-13"><a href="sequence-alignments.html#cb467-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-14"><a href="sequence-alignments.html#cb467-14" aria-hidden="true" tabindex="-1"></a>one_genus_CYTB <span class="ot">&lt;-</span> <span class="fu">paste0</span>(my_genus[i], <span class="st">&quot;[Organism] AND CYTB[Gene]&quot;</span>)</span>
<span id="cb467-15"><a href="sequence-alignments.html#cb467-15" aria-hidden="true" tabindex="-1"></a>one_genus_seq_IDs <span class="ot">&lt;-</span> <span class="fu">entrez_search</span>(<span class="at">db=</span><span class="st">&quot;nuccore&quot;</span>, <span class="at">term=</span> one_genus_CYTB )</span>
<span id="cb467-16"><a href="sequence-alignments.html#cb467-16" aria-hidden="true" tabindex="-1"></a>one_genus_seq_fasta <span class="ot">&lt;-</span> <span class="fu">entrez_fetch</span>(<span class="at">db=</span><span class="st">&quot;nuccore&quot;</span>, <span class="at">id=</span>one_genus_seq_IDs<span class="sc">$</span>ids, <span class="at">rettype=</span><span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb467-17"><a href="sequence-alignments.html#cb467-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-18"><a href="sequence-alignments.html#cb467-18" aria-hidden="true" tabindex="-1"></a><span class="do">## collect sequences in list</span></span>
<span id="cb467-19"><a href="sequence-alignments.html#cb467-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-20"><a href="sequence-alignments.html#cb467-20" aria-hidden="true" tabindex="-1"></a>collect_CYTB_list_seqs[[i]] <span class="ot">&lt;-</span> one_genus_seq_fasta</span>
<span id="cb467-21"><a href="sequence-alignments.html#cb467-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-22"><a href="sequence-alignments.html#cb467-22" aria-hidden="true" tabindex="-1"></a><span class="do">## visualize the collected sequences</span></span>
<span id="cb467-23"><a href="sequence-alignments.html#cb467-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-24"><a href="sequence-alignments.html#cb467-24" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;these are CYTB sequences from genus:&quot;</span>, my_genus[i], <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb467-25"><a href="sequence-alignments.html#cb467-25" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.sleep</span>(<span class="dv">1</span>)</span>
<span id="cb467-26"><a href="sequence-alignments.html#cb467-26" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(collect_CYTB_list_seqs[[i]])</span>
<span id="cb467-27"><a href="sequence-alignments.html#cb467-27" aria-hidden="true" tabindex="-1"></a>                              }</span>
<span id="cb467-28"><a href="sequence-alignments.html#cb467-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-29"><a href="sequence-alignments.html#cb467-29" aria-hidden="true" tabindex="-1"></a><span class="do">## loop for COI</span></span>
<span id="cb467-30"><a href="sequence-alignments.html#cb467-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-31"><a href="sequence-alignments.html#cb467-31" aria-hidden="true" tabindex="-1"></a>collect_COI_list_seqs <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb467-32"><a href="sequence-alignments.html#cb467-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-33"><a href="sequence-alignments.html#cb467-33" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(my_genus)) {</span>
<span id="cb467-34"><a href="sequence-alignments.html#cb467-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-35"><a href="sequence-alignments.html#cb467-35" aria-hidden="true" tabindex="-1"></a>one_genus_COI <span class="ot">&lt;-</span> <span class="fu">paste0</span>(my_genus[i], <span class="st">&quot;[Organism] AND COI[Gene]&quot;</span>)</span>
<span id="cb467-36"><a href="sequence-alignments.html#cb467-36" aria-hidden="true" tabindex="-1"></a>one_genus_seq_IDs <span class="ot">&lt;-</span> <span class="fu">entrez_search</span>(<span class="at">db=</span><span class="st">&quot;nuccore&quot;</span>, <span class="at">term=</span> one_genus_COI )</span>
<span id="cb467-37"><a href="sequence-alignments.html#cb467-37" aria-hidden="true" tabindex="-1"></a>one_genus_seq_fasta <span class="ot">&lt;-</span> <span class="fu">entrez_fetch</span>(<span class="at">db=</span><span class="st">&quot;nuccore&quot;</span>, <span class="at">id=</span>one_genus_seq_IDs<span class="sc">$</span>ids, <span class="at">rettype=</span><span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb467-38"><a href="sequence-alignments.html#cb467-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-39"><a href="sequence-alignments.html#cb467-39" aria-hidden="true" tabindex="-1"></a><span class="do">## collect sequences in list</span></span>
<span id="cb467-40"><a href="sequence-alignments.html#cb467-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-41"><a href="sequence-alignments.html#cb467-41" aria-hidden="true" tabindex="-1"></a>collect_COI_list_seqs[[i]] <span class="ot">&lt;-</span> one_genus_seq_fasta</span>
<span id="cb467-42"><a href="sequence-alignments.html#cb467-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-43"><a href="sequence-alignments.html#cb467-43" aria-hidden="true" tabindex="-1"></a><span class="do">## visualize the collected sequences</span></span>
<span id="cb467-44"><a href="sequence-alignments.html#cb467-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb467-45"><a href="sequence-alignments.html#cb467-45" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;these are COI sequences from genus:&quot;</span>, my_genus[i], <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span>
<span id="cb467-46"><a href="sequence-alignments.html#cb467-46" aria-hidden="true" tabindex="-1"></a><span class="fu">Sys.sleep</span>(<span class="dv">1</span>)</span>
<span id="cb467-47"><a href="sequence-alignments.html#cb467-47" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(collect_COI_list_seqs[[i]])</span>
<span id="cb467-48"><a href="sequence-alignments.html#cb467-48" aria-hidden="true" tabindex="-1"></a>                              }</span></code></pre></div>
<p><strong>11)</strong> Now, we need to inspect these sequences and choose the one sequence per species per maker (e.g., for <em>Ameerega bilinguis</em> one CYTB and one COI). We will record the accession number for these two sequences per species. You will do this for example, but in most cases, you will collect the accession numbers for the markers of interest in your project. You might also need to check the NCBI website.</p>
<div class="sourceCode" id="cb468"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb468-1"><a href="sequence-alignments.html#cb468-1" aria-hidden="true" tabindex="-1"></a><span class="do">## visualize sequences in console </span></span>
<span id="cb468-2"><a href="sequence-alignments.html#cb468-2" aria-hidden="true" tabindex="-1"></a><span class="co"># &quot;Ameerega&quot;</span></span>
<span id="cb468-3"><a href="sequence-alignments.html#cb468-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(collect_CYTB_list_seqs[[<span class="dv">1</span>]])</span>
<span id="cb468-4"><a href="sequence-alignments.html#cb468-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(collect_COI_list_seqs[[<span class="dv">1</span>]])</span>
<span id="cb468-5"><a href="sequence-alignments.html#cb468-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb468-6"><a href="sequence-alignments.html#cb468-6" aria-hidden="true" tabindex="-1"></a><span class="co"># &quot;Colostethus&quot;</span></span>
<span id="cb468-7"><a href="sequence-alignments.html#cb468-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(collect_CYTB_list_seqs[[<span class="dv">2</span>]])</span>
<span id="cb468-8"><a href="sequence-alignments.html#cb468-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(collect_COI_list_seqs[[<span class="dv">2</span>]])</span>
<span id="cb468-9"><a href="sequence-alignments.html#cb468-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb468-10"><a href="sequence-alignments.html#cb468-10" aria-hidden="true" tabindex="-1"></a><span class="co"># &quot;Epipedobates&quot;</span></span>
<span id="cb468-11"><a href="sequence-alignments.html#cb468-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(collect_CYTB_list_seqs[[<span class="dv">3</span>]])</span>
<span id="cb468-12"><a href="sequence-alignments.html#cb468-12" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(collect_COI_list_seqs[[<span class="dv">3</span>]])</span>
<span id="cb468-13"><a href="sequence-alignments.html#cb468-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb468-14"><a href="sequence-alignments.html#cb468-14" aria-hidden="true" tabindex="-1"></a><span class="co"># &quot;Silverstoneia&quot;</span></span>
<span id="cb468-15"><a href="sequence-alignments.html#cb468-15" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(collect_CYTB_list_seqs[[<span class="dv">4</span>]])</span>
<span id="cb468-16"><a href="sequence-alignments.html#cb468-16" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(collect_COI_list_seqs[[<span class="dv">4</span>]])</span>
<span id="cb468-17"><a href="sequence-alignments.html#cb468-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb468-18"><a href="sequence-alignments.html#cb468-18" aria-hidden="true" tabindex="-1"></a><span class="co"># &quot;Leucostethus&quot;</span></span>
<span id="cb468-19"><a href="sequence-alignments.html#cb468-19" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(collect_CYTB_list_seqs[[<span class="dv">5</span>]])</span>
<span id="cb468-20"><a href="sequence-alignments.html#cb468-20" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(collect_COI_list_seqs[[<span class="dv">5</span>]])</span>
<span id="cb468-21"><a href="sequence-alignments.html#cb468-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb468-22"><a href="sequence-alignments.html#cb468-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb468-23"><a href="sequence-alignments.html#cb468-23" aria-hidden="true" tabindex="-1"></a><span class="do">## build vectors before a dataframe</span></span>
<span id="cb468-24"><a href="sequence-alignments.html#cb468-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb468-25"><a href="sequence-alignments.html#cb468-25" aria-hidden="true" tabindex="-1"></a>collected_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb468-26"><a href="sequence-alignments.html#cb468-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">taxa =</span> <span class="st">&quot;Ameerega_bilinguis&quot;</span>, <span class="at">CYTB =</span> <span class="st">&quot;AF128559.1&quot;</span>, <span class="at">COX =</span> <span class="st">&quot;MW042030.1&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),</span>
<span id="cb468-27"><a href="sequence-alignments.html#cb468-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">taxa =</span> <span class="st">&quot;Ameerega_hahneli&quot;</span>, <span class="at">CYTB =</span> <span class="st">&quot;HQ290575&quot;</span>, <span class="at">COX =</span> <span class="st">&quot;MW042031.1&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),</span>
<span id="cb468-28"><a href="sequence-alignments.html#cb468-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">taxa =</span> <span class="st">&quot;Ameerega_parvula&quot;</span>, <span class="at">CYTB =</span> <span class="st">&quot;HQ290576.1&quot;</span>, <span class="at">COX =</span> <span class="st">&quot;MW042032.1&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),</span>
<span id="cb468-29"><a href="sequence-alignments.html#cb468-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">taxa =</span> <span class="st">&quot;Ameerega_braccata&quot;</span>, <span class="at">CYTB =</span> <span class="st">&quot;DQ502556.1&quot;</span>, <span class="at">COX =</span> <span class="st">&quot;KU494333.1&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),</span>
<span id="cb468-30"><a href="sequence-alignments.html#cb468-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">taxa =</span> <span class="st">&quot;Ameerega_trivittata&quot;</span>, <span class="at">CYTB =</span> <span class="st">&quot;HQ290579.1&quot;</span>, <span class="at">COX =</span> <span class="st">&quot;DQ502903.1&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),</span>
<span id="cb468-31"><a href="sequence-alignments.html#cb468-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">taxa =</span> <span class="st">&quot;Ameerega_silverstonei&quot;</span>, <span class="at">CYTB =</span> <span class="st">&quot;DQ523154.1&quot;</span>, <span class="at">COX =</span> <span class="st">&quot;DQ502851.1&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),</span>
<span id="cb468-32"><a href="sequence-alignments.html#cb468-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">taxa =</span> <span class="st">&quot;Colostethus_panamansis&quot;</span>, <span class="at">CYTB =</span> <span class="st">&quot;HQ290546.1&quot;</span>, <span class="at">COX =</span> <span class="st">&quot;KC129189.1&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),</span>
<span id="cb468-33"><a href="sequence-alignments.html#cb468-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">taxa =</span> <span class="st">&quot;Colostethus_pratti&quot;</span>, <span class="at">CYTB =</span> <span class="st">&quot;HQ290547.1&quot;</span>, <span class="at">COX =</span> <span class="st">&quot;KF807016.1&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),</span>
<span id="cb468-34"><a href="sequence-alignments.html#cb468-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">taxa =</span> <span class="st">&quot;Epipedobates_boulengeri&quot;</span>, <span class="at">CYTB =</span> <span class="st">&quot;HQ290574.1&quot;</span>, <span class="at">COX =</span> <span class="st">&quot;MW042034.1&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),</span>
<span id="cb468-35"><a href="sequence-alignments.html#cb468-35" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">taxa =</span> <span class="st">&quot;Epipedobates_machalilla&quot;</span>, <span class="at">CYTB =</span> <span class="st">&quot;HQ290542.1&quot;</span>, <span class="at">COX =</span> <span class="st">&quot;MW042036.1&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),</span>
<span id="cb468-36"><a href="sequence-alignments.html#cb468-36" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">taxa =</span> <span class="st">&quot;Silverstoneia_flotator&quot;</span>, <span class="at">CYTB =</span> <span class="st">&quot;HQ290537.1&quot;</span>, <span class="at">COX =</span> <span class="st">&quot;MW042039.1&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>),</span>
<span id="cb468-37"><a href="sequence-alignments.html#cb468-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">taxa =</span> <span class="st">&quot;Leucostethus_fugax&quot;</span>, <span class="at">CYTB =</span> <span class="cn">NA</span>, <span class="at">COX =</span> <span class="st">&quot;MW042037.1&quot;</span>, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>))</span>
<span id="cb468-38"><a href="sequence-alignments.html#cb468-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb468-39"><a href="sequence-alignments.html#cb468-39" aria-hidden="true" tabindex="-1"></a><span class="do">## build a dataframe</span></span>
<span id="cb468-40"><a href="sequence-alignments.html#cb468-40" aria-hidden="true" tabindex="-1"></a>collected_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, collected_list)</span>
<span id="cb468-41"><a href="sequence-alignments.html#cb468-41" aria-hidden="true" tabindex="-1"></a>collected_df</span>
<span id="cb468-42"><a href="sequence-alignments.html#cb468-42" aria-hidden="true" tabindex="-1"></a><span class="co">#                      taxa       CYTB        COX</span></span>
<span id="cb468-43"><a href="sequence-alignments.html#cb468-43" aria-hidden="true" tabindex="-1"></a><span class="co">#1       Ameerega_bilinguis AF128559.1 MW042030.1</span></span>
<span id="cb468-44"><a href="sequence-alignments.html#cb468-44" aria-hidden="true" tabindex="-1"></a><span class="co">#2         Ameerega_hahneli   HQ290575 MW042031.1</span></span>
<span id="cb468-45"><a href="sequence-alignments.html#cb468-45" aria-hidden="true" tabindex="-1"></a><span class="co">#3         Ameerega_parvula HQ290576.1 MW042032.1</span></span>
<span id="cb468-46"><a href="sequence-alignments.html#cb468-46" aria-hidden="true" tabindex="-1"></a><span class="co">#4        Ameerega_braccata DQ502556.1 KU494333.1</span></span>
<span id="cb468-47"><a href="sequence-alignments.html#cb468-47" aria-hidden="true" tabindex="-1"></a><span class="co">#5      Ameerega_trivittata HQ290579.1 DQ502903.1</span></span>
<span id="cb468-48"><a href="sequence-alignments.html#cb468-48" aria-hidden="true" tabindex="-1"></a><span class="co">#6    Ameerega_silverstonei DQ523154.1 DQ502851.1</span></span>
<span id="cb468-49"><a href="sequence-alignments.html#cb468-49" aria-hidden="true" tabindex="-1"></a><span class="co">#7   Colostethus_panamansis HQ290546.1 KC129189.1</span></span>
<span id="cb468-50"><a href="sequence-alignments.html#cb468-50" aria-hidden="true" tabindex="-1"></a><span class="co">#8       Colostethus_pratti HQ290547.1 KF807016.1</span></span>
<span id="cb468-51"><a href="sequence-alignments.html#cb468-51" aria-hidden="true" tabindex="-1"></a><span class="co">#9  Epipedobates_boulengeri HQ290574.1 MW042034.1</span></span>
<span id="cb468-52"><a href="sequence-alignments.html#cb468-52" aria-hidden="true" tabindex="-1"></a><span class="co">#10 Epipedobates_machalilla HQ290542.1 MW042036.1</span></span>
<span id="cb468-53"><a href="sequence-alignments.html#cb468-53" aria-hidden="true" tabindex="-1"></a><span class="co">#11  Silverstoneia_flotator HQ290537.1 MW042039.1</span></span>
<span id="cb468-54"><a href="sequence-alignments.html#cb468-54" aria-hidden="true" tabindex="-1"></a><span class="co">#12      Leucostethus_fugax       &lt;NA&gt; MW042037.1</span></span></code></pre></div>
<p><strong>12)</strong> We can now retrieve these sequences in a data frame and build a fasta file with commo taxon names.</p>
<div class="sourceCode" id="cb469"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb469-1"><a href="sequence-alignments.html#cb469-1" aria-hidden="true" tabindex="-1"></a><span class="do">## retrive sequences for each taxon</span></span>
<span id="cb469-2"><a href="sequence-alignments.html#cb469-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb469-3"><a href="sequence-alignments.html#cb469-3" aria-hidden="true" tabindex="-1"></a>collected_CYTB_seqs <span class="ot">&lt;-</span> <span class="fu">character</span>()</span>
<span id="cb469-4"><a href="sequence-alignments.html#cb469-4" aria-hidden="true" tabindex="-1"></a>collected_COI_seqs <span class="ot">&lt;-</span> <span class="fu">character</span>()</span>
<span id="cb469-5"><a href="sequence-alignments.html#cb469-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb469-6"><a href="sequence-alignments.html#cb469-6" aria-hidden="true" tabindex="-1"></a><span class="do">## start of loop</span></span>
<span id="cb469-7"><a href="sequence-alignments.html#cb469-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb469-8"><a href="sequence-alignments.html#cb469-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(collected_df)) {</span>
<span id="cb469-9"><a href="sequence-alignments.html#cb469-9" aria-hidden="true" tabindex="-1"></a><span class="co"># i &lt;- 12</span></span>
<span id="cb469-10"><a href="sequence-alignments.html#cb469-10" aria-hidden="true" tabindex="-1"></a>one_taxon_name <span class="ot">&lt;-</span> collected_df[i,<span class="st">&quot;taxa&quot;</span>]</span>
<span id="cb469-11"><a href="sequence-alignments.html#cb469-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb469-12"><a href="sequence-alignments.html#cb469-12" aria-hidden="true" tabindex="-1"></a><span class="co"># retrieve sequences</span></span>
<span id="cb469-13"><a href="sequence-alignments.html#cb469-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb469-14"><a href="sequence-alignments.html#cb469-14" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.na</span>(collected_df[i,<span class="st">&quot;CYTB&quot;</span>])) {</span>
<span id="cb469-15"><a href="sequence-alignments.html#cb469-15" aria-hidden="true" tabindex="-1"></a>one_taxon_CYTB <span class="ot">&lt;-</span> <span class="fu">entrez_fetch</span>(<span class="at">db=</span><span class="st">&quot;nuccore&quot;</span>, <span class="at">id=</span>collected_df[i,<span class="st">&quot;CYTB&quot;</span>], <span class="at">rettype=</span><span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb469-16"><a href="sequence-alignments.html#cb469-16" aria-hidden="true" tabindex="-1"></a>                                   } <span class="cf">else</span> { one_taxon_CYTB <span class="ot">&lt;-</span> <span class="st">&quot;--------&quot;</span>}</span>
<span id="cb469-17"><a href="sequence-alignments.html#cb469-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb469-18"><a href="sequence-alignments.html#cb469-18" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.na</span>(collected_df[i,<span class="st">&quot;COX&quot;</span>])) {</span>
<span id="cb469-19"><a href="sequence-alignments.html#cb469-19" aria-hidden="true" tabindex="-1"></a>one_taxon_COI <span class="ot">&lt;-</span> <span class="fu">entrez_fetch</span>(<span class="at">db=</span><span class="st">&quot;nuccore&quot;</span>, <span class="at">id=</span>collected_df[i,<span class="st">&quot;COX&quot;</span>], <span class="at">rettype=</span><span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb469-20"><a href="sequence-alignments.html#cb469-20" aria-hidden="true" tabindex="-1"></a>                                  } <span class="cf">else</span> { one_taxon_COI <span class="ot">&lt;-</span> <span class="st">&quot;--------&quot;</span>}</span>
<span id="cb469-21"><a href="sequence-alignments.html#cb469-21" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb469-22"><a href="sequence-alignments.html#cb469-22" aria-hidden="true" tabindex="-1"></a><span class="co"># create a common name, split sequence vecto by first \n</span></span>
<span id="cb469-23"><a href="sequence-alignments.html#cb469-23" aria-hidden="true" tabindex="-1"></a>one_taxon_CYTB_1 <span class="ot">&lt;-</span>  <span class="fu">unlist</span>(<span class="fu">strsplit</span>(one_taxon_CYTB, <span class="at">split=</span><span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="at">perl=</span><span class="cn">TRUE</span>))</span>
<span id="cb469-24"><a href="sequence-alignments.html#cb469-24" aria-hidden="true" tabindex="-1"></a>one_taxon_CYTB_1</span>
<span id="cb469-25"><a href="sequence-alignments.html#cb469-25" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;&gt;AF128559.1 Epipedobates bilinguis cytochrome b (cytb) gene, partial cds; mitochondrial gene for mitochondrial product&quot;</span></span>
<span id="cb469-26"><a href="sequence-alignments.html#cb469-26" aria-hidden="true" tabindex="-1"></a><span class="co">#[2] &quot;CTTCTAGGTCTATGTCTTATTGCCCAAATCGCTACAGGCCTTTTTCTTGCCATACACTATACTGCTGATA&quot;                                                </span></span>
<span id="cb469-27"><a href="sequence-alignments.html#cb469-27" aria-hidden="true" tabindex="-1"></a><span class="co">#[3] &quot;CTTCTATAGCTTTTTCCTCTCTAGCCCATATTTGCCGAGATGTCAACAATGGCTGACTTCTTCGTAATCT&quot;                                                </span></span>
<span id="cb469-28"><a href="sequence-alignments.html#cb469-28" aria-hidden="true" tabindex="-1"></a><span class="co">#[4] &quot;TCACGCTAACGGAGCCTCATTCTTCTTCATTTGTATTTATCTTCATATTGGCCGAGGAATATACTATGGA&quot;                                                </span></span>
<span id="cb469-29"><a href="sequence-alignments.html#cb469-29" aria-hidden="true" tabindex="-1"></a><span class="co">#[5] &quot;TCCTTTTTATTTAAAGAAACATGAAACATTGGAGTAATTTTATTATTTTTAGTTATAGCCAC&quot;</span></span>
<span id="cb469-30"><a href="sequence-alignments.html#cb469-30" aria-hidden="true" tabindex="-1"></a><span class="co">#[6] &quot;&quot;                                                         </span></span>
<span id="cb469-31"><a href="sequence-alignments.html#cb469-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb469-32"><a href="sequence-alignments.html#cb469-32" aria-hidden="true" tabindex="-1"></a><span class="co"># we want to create a common name, so we will replace the first line and concatenate all others</span></span>
<span id="cb469-33"><a href="sequence-alignments.html#cb469-33" aria-hidden="true" tabindex="-1"></a>one_taxon_CYTB_2 <span class="ot">&lt;-</span> <span class="fu">paste0</span>(one_taxon_CYTB_1[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(one_taxon_CYTB_1)], <span class="at">collapse =</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb469-34"><a href="sequence-alignments.html#cb469-34" aria-hidden="true" tabindex="-1"></a>one_taxon_CYTB_2</span>
<span id="cb469-35"><a href="sequence-alignments.html#cb469-35" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;CTTCTAGGTCTATGTCTTATTGCCCAAATCGCTACAGGCCTTTTTCTTGCCATACACTATACTGCTGATACTTCTATAGCTTTTTCCTCTCTAGCCCATATTTGCCGAGATGTCAACAATGGCTGACTTCTTCGTAATCTTCACGCTAACGGAGCCTCATTCTTCTTCATTTGTATTTATCTTCATATTGGCCGAGGAATATACTATGGATCCTTTTTATTTAAAGAAACATGAAACATTGGAGTAATTTTATTATTTTTAGTTATAGCCAC&quot;</span></span>
<span id="cb469-36"><a href="sequence-alignments.html#cb469-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb469-37"><a href="sequence-alignments.html#cb469-37" aria-hidden="true" tabindex="-1"></a><span class="co">#we will vector for sequences</span></span>
<span id="cb469-38"><a href="sequence-alignments.html#cb469-38" aria-hidden="true" tabindex="-1"></a>one_taxon_CYTB_3 <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;&gt;&quot;</span>,one_taxon_name,<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,one_taxon_CYTB_2,<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb469-39"><a href="sequence-alignments.html#cb469-39" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(one_taxon_CYTB_3)</span>
<span id="cb469-40"><a href="sequence-alignments.html#cb469-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;Ameerega_bilinguis</span></span>
<span id="cb469-41"><a href="sequence-alignments.html#cb469-41" aria-hidden="true" tabindex="-1"></a><span class="co">#CTTCTAGGTCTATGTCTTATTGCCCAAATCGCTACAGGCCTTTTTCTTGCCATACACTATACTGCTGATACTTCTATAGCTTTTTCCTCTCTAGCCCATATTTGCCGAGATGTCAACAATGGCTGACTTCTTCGTAATCTTCACGCTAACGGAGCCTCATTCTTCTTCATTTGTATTTATCTTCATATTGGCCGAGGAATATACTATGGATCCTTTTTATTTAAAGAAACATGAAACATTGGAGTAATTTTATTATTTTTAGTTATAGCCAC</span></span>
<span id="cb469-42"><a href="sequence-alignments.html#cb469-42" aria-hidden="true" tabindex="-1"></a>collected_CYTB_seqs[i] <span class="ot">&lt;-</span> one_taxon_CYTB_3</span>
<span id="cb469-43"><a href="sequence-alignments.html#cb469-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb469-44"><a href="sequence-alignments.html#cb469-44" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb469-45"><a href="sequence-alignments.html#cb469-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb469-46"><a href="sequence-alignments.html#cb469-46" aria-hidden="true" tabindex="-1"></a><span class="do">## for COI</span></span>
<span id="cb469-47"><a href="sequence-alignments.html#cb469-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb469-48"><a href="sequence-alignments.html#cb469-48" aria-hidden="true" tabindex="-1"></a>one_taxon_COI_1 <span class="ot">&lt;-</span>  <span class="fu">unlist</span>(<span class="fu">strsplit</span>(one_taxon_COI, <span class="at">split=</span><span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="at">perl=</span><span class="cn">TRUE</span>))</span>
<span id="cb469-49"><a href="sequence-alignments.html#cb469-49" aria-hidden="true" tabindex="-1"></a>one_taxon_COI_2 <span class="ot">&lt;-</span> <span class="fu">paste0</span>(one_taxon_COI_1[<span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(one_taxon_COI_1)], <span class="at">collapse =</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb469-50"><a href="sequence-alignments.html#cb469-50" aria-hidden="true" tabindex="-1"></a>one_taxon_COI_3 <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;&gt;&quot;</span>,one_taxon_name,<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>,one_taxon_COI_2,<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>, <span class="at">sep=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb469-51"><a href="sequence-alignments.html#cb469-51" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(one_taxon_COI_3)</span>
<span id="cb469-52"><a href="sequence-alignments.html#cb469-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;Ameerega_bilinguis</span></span>
<span id="cb469-53"><a href="sequence-alignments.html#cb469-53" aria-hidden="true" tabindex="-1"></a><span class="co">#GTGATAATTACTCGATGATTATTTTCTACCAACCACAAAGACATCGGAACTTTATACCTAGTGTTTGGGGCATGAGCAGGCATAGTCGGCACTGCTCTTAGCCTTTTAATTCGAGCCGAATTAAGCCAGCCCGGGTCCTTACTAGGCGATGACCAGATCTACAACGTTATTGTTACCGCCCATGCTTTCGTTATAATCTTTTTTATAGTAATGCCAATTCTAATCGGTGGCTTTGGGAATTGATTAGTGCCCCTAATAATTGGAGCCCCAGACATAGCTTTTCCCCGAATAAACAATATGAGCTTTTGGCTTCTTCCCCCCTCTTTCCTACTACTCCTAGCATCCGCAGGCGTTGAAGCAGGCGCCGGTACTGGCTGAACTGTGTACCCTCCCCTTGCAGGCAACCTAGCTCATGCTGGCCCATCAGTTGATTTAACTATTTTTTCACTTCATCTCGCCGGTGTTTCTTCTATTCTAGGGGCAATTAACTTTATTACAACAACCTTAAACATAAAACCCCCTTCATTAACACAATATCAAACCCCATTATTTGTCTGATCTGTATTAATTACTGCAGTCCTTCTTCTTCTCTCCCTCCCAGTTCTGGCTGCCGGAATCACTATACTCTTGACTGACCGAAACCTAAACACCACCTTCTTTGACCCAGCAGGTGGAGGCGACCCTGTCCTGTACCAACACCTGTTCTGATTCTTTGGTCACCCCGAAGTCTACATCCTTATCCTGCCTGGATTTGGTATCATCTCCCATGTTGTCACATTCTACTCTAGCAAAAAAGAACCCTTCGGCTATATAGGAATAGTCTGAGCTATAATATCGATTGGTCTCCTAGGTTTCATTGTTTGAGCTCACCACATATTCACAACAGACCTTAATGTAGACACTCGAGCCTACTTTACCTCAGCTACTATAATCATCGCTATCCCAACAGGTGTCAAAGTCTTTAGCTGACTTGCCACCATGCACGGAGGAATTATTAAATGAGACGCCGCCATATTATGGGCTCTCGGATTCATCTTTTTATTTACAGTTGGAGGACTAACTGGAATCGTTTTAGCCAACTCCTCTTTAGACATTGTTTTGCATGATACATATTATGTAGTAGCCCACTTTCACTACGTTCTTTCTATGGGGGCAGTATTTGCCATTATAGCCGGCTTCGTACACTGATTTCCTCTCTTTTCCGGATTTACCCTTCATGAAGCCTGAACAAAAATTCAATTTGGCGTCATATTTACCGGCGTAAATTTAACATTCTTCCCCCAGCATTTCTTAGGTCTCGCAGGCATGCCTCGACGTTATTCAGACTACCCTGACGCCTACACATTATGAAACACCGTTTCATCAATCGGCTCTTTAATCTCTCTAGTTGCAGTAATCATTATGATGTTTATCATTTGAGAAGCTTTCTCTTCCAAACGCCTACCTCTACCTGCAGAAATAACCCCAACTAATGTAGAATGATTATACGGATCCCCCCCACCTTACCACACTTTTGAGGAAGCCGTTTACTCCAAAATT</span></span>
<span id="cb469-54"><a href="sequence-alignments.html#cb469-54" aria-hidden="true" tabindex="-1"></a>collected_COI_seqs[i] <span class="ot">&lt;-</span> one_taxon_COI_3</span>
<span id="cb469-55"><a href="sequence-alignments.html#cb469-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb469-56"><a href="sequence-alignments.html#cb469-56" aria-hidden="true" tabindex="-1"></a>                           }</span>
<span id="cb469-57"><a href="sequence-alignments.html#cb469-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb469-58"><a href="sequence-alignments.html#cb469-58" aria-hidden="true" tabindex="-1"></a><span class="do">## end of loop</span></span>
<span id="cb469-59"><a href="sequence-alignments.html#cb469-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb469-60"><a href="sequence-alignments.html#cb469-60" aria-hidden="true" tabindex="-1"></a><span class="do">## collapse all collections before write a text file</span></span>
<span id="cb469-61"><a href="sequence-alignments.html#cb469-61" aria-hidden="true" tabindex="-1"></a>collected_CYTB_seqs_con <span class="ot">&lt;-</span> <span class="fu">paste0</span>(collected_CYTB_seqs, <span class="at">collapse=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb469-62"><a href="sequence-alignments.html#cb469-62" aria-hidden="true" tabindex="-1"></a>collected_COI_seqs_con <span class="ot">&lt;-</span> <span class="fu">paste0</span>(collected_COI_seqs, <span class="at">collapse=</span><span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p>We can save these sequences in a text file in your working directory.</p>
<div class="sourceCode" id="cb470"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb470-1"><a href="sequence-alignments.html#cb470-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this is exclusive to your OWN COMPUTER change it accordingly</span></span>
<span id="cb470-2"><a href="sequence-alignments.html#cb470-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/my_working_directory&quot;</span>)</span>
<span id="cb470-3"><a href="sequence-alignments.html#cb470-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(collected_CYTB_seqs_con, <span class="st">&quot;my_collected_CYTB_fasta.txt&quot;</span>)</span>
<span id="cb470-4"><a href="sequence-alignments.html#cb470-4" aria-hidden="true" tabindex="-1"></a><span class="fu">write</span>(collected_COI_seqs_con, <span class="st">&quot;my_collected_COI_fasta.txt&quot;</span>)</span></code></pre></div>
<p><strong>13)</strong> Now we can import the CYTB and COI nucleotide sequences from this file that are in <code>fasta</code> format into R and align each with <code>DECIPHER</code>.</p>
<div class="sourceCode" id="cb471"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb471-1"><a href="sequence-alignments.html#cb471-1" aria-hidden="true" tabindex="-1"></a><span class="do">## make sure that Biostrings is loaded</span></span>
<span id="cb471-2"><a href="sequence-alignments.html#cb471-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(Biostrings)</span>
<span id="cb471-3"><a href="sequence-alignments.html#cb471-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb471-4"><a href="sequence-alignments.html#cb471-4" aria-hidden="true" tabindex="-1"></a><span class="do">## load CYTB</span></span>
<span id="cb471-5"><a href="sequence-alignments.html#cb471-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb471-6"><a href="sequence-alignments.html#cb471-6" aria-hidden="true" tabindex="-1"></a>CYTB_set <span class="ot">&lt;-</span> <span class="fu">readDNAStringSet</span>(<span class="at">filepath =</span> <span class="st">&quot;~/Desktop/Teach_R/my_working_directory/my_collected_CYTB_fasta.txt&quot;</span>, </span>
<span id="cb471-7"><a href="sequence-alignments.html#cb471-7" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">format =</span> <span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb471-8"><a href="sequence-alignments.html#cb471-8" aria-hidden="true" tabindex="-1"></a>CYTB_set</span>
<span id="cb471-9"><a href="sequence-alignments.html#cb471-9" aria-hidden="true" tabindex="-1"></a><span class="co">#A DNAStringSet instance of length 12</span></span>
<span id="cb471-10"><a href="sequence-alignments.html#cb471-10" aria-hidden="true" tabindex="-1"></a><span class="co">#     width seq                                                                                                                  names               </span></span>
<span id="cb471-11"><a href="sequence-alignments.html#cb471-11" aria-hidden="true" tabindex="-1"></a><span class="co"># [1]   272 CTTCTAGGTCTATGTCTTATTGCCCAAATCGCTACAGGCCTTTTTCTTGCCATACAC...TTATTTAAAGAAACATGAAACATTGGAGTAATTTTATTATTTTTAGTTATAGCCAC Ameerega_bilinguis</span></span>
<span id="cb471-12"><a href="sequence-alignments.html#cb471-12" aria-hidden="true" tabindex="-1"></a><span class="co"># [2]   690 GGCCTATGTTTAATTGCCCAAATCATCACAGGTCTCTTCCTAGCCATACACTATACA...TTCTCGGAGATCCAGATAATTTTACCCCAGCTAACCCCCTAGTCACCCCCCCTCAT Ameerega_hahneli</span></span>
<span id="cb471-13"><a href="sequence-alignments.html#cb471-13" aria-hidden="true" tabindex="-1"></a><span class="co"># [3]   690 GGCTTATGTCTAATTGCTCAAATCATTACAGGTCTTTTTCTAGCTATACATTATACA...TTCTCGGAGATCCAGACAATTTCACCCCAGCTAACCCCCTAGTCACCCCCCCTCAC Ameerega_parvula</span></span>
<span id="cb471-14"><a href="sequence-alignments.html#cb471-14" aria-hidden="true" tabindex="-1"></a><span class="co"># [4]   385 AACACACCCCGCTCTAAAAATTATCAACAACTCATTCATTGACCTTCCATCCCCTGC...GCGTGATTCTTCTATTTTTAGTAATAGCCACCGCCTTCGTCGGCTATGTCCTCCCT Ameerega_braccata</span></span>
<span id="cb471-15"><a href="sequence-alignments.html#cb471-15" aria-hidden="true" tabindex="-1"></a><span class="co"># [5]   690 GGCCTCTGCCTAATCGCCCAGATCATCACAGGCCTCTTCTTAGCTATACACTACACA...TTCTCGGAGACCCAGACAATTTCACCCCCGCTAACCCCCTGGTCACCCCACCTCAC Ameerega_trivittata</span></span>
<span id="cb471-16"><a href="sequence-alignments.html#cb471-16" aria-hidden="true" tabindex="-1"></a><span class="co"># ...   ... ...</span></span>
<span id="cb471-17"><a href="sequence-alignments.html#cb471-17" aria-hidden="true" tabindex="-1"></a><span class="co"># [8]   690 GGCCTCTGCCTTATTATCCAGATTGTCACTGGCCTTTTCCTAGCTATGCACTATACA...TCTTGGGCGACCCAGACAACTTCACCCCAGCTAACCCCCTAGTTACTCCCCCCCAC Colostethus_pratti</span></span>
<span id="cb471-18"><a href="sequence-alignments.html#cb471-18" aria-hidden="true" tabindex="-1"></a><span class="co"># [9]   690 GGTCTATGTCTTATTGCCCAAATCGCTACAGGACTTTTTCTTGCTATACACTATACT...TTTTAGGAGACCCAGACAACTTCACCCCTGCTAACCCTCTAGTCACCCCTCCTCAT Epipedobates_boul...</span></span>
<span id="cb471-19"><a href="sequence-alignments.html#cb471-19" aria-hidden="true" tabindex="-1"></a><span class="co">#[10]   690 GGCCTATGTCTTATTGCCCAAATCGCTACAGGTCTTTTTCTTGCCATACACTATACT...TCCTAGGAGATCCAGACAACTTCACCCCTGCCAACCCTTTAGTCACCCCTCCTCAC Epipedobates_mach...</span></span>
<span id="cb471-20"><a href="sequence-alignments.html#cb471-20" aria-hidden="true" tabindex="-1"></a><span class="co">#[11]   690 GGCCTTTGCCTCATTGCCCAAATTGCCACAGGCCTTTTTTTAGCTATACACTACACC...TTCTAGGCGACCCAGACAACTTCACCCCAGCTAATCCCCTAGTAACCCCACCACAC Silverstoneia_flo...</span></span>
<span id="cb471-21"><a href="sequence-alignments.html#cb471-21" aria-hidden="true" tabindex="-1"></a><span class="co">#[12]    10 NA--------</span></span>
<span id="cb471-22"><a href="sequence-alignments.html#cb471-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb471-23"><a href="sequence-alignments.html#cb471-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb471-24"><a href="sequence-alignments.html#cb471-24" aria-hidden="true" tabindex="-1"></a><span class="do">## </span><span class="al">NOTICE</span><span class="do">: for alignment, we need to remove any gap of missing gaps. In this case sequence [12] needs to be removed.</span></span>
<span id="cb471-25"><a href="sequence-alignments.html#cb471-25" aria-hidden="true" tabindex="-1"></a>CYTB_set <span class="ot">&lt;-</span> CYTB_set[<span class="sc">-</span><span class="dv">12</span>]</span>
<span id="cb471-26"><a href="sequence-alignments.html#cb471-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb471-27"><a href="sequence-alignments.html#cb471-27" aria-hidden="true" tabindex="-1"></a><span class="do">## load COI</span></span>
<span id="cb471-28"><a href="sequence-alignments.html#cb471-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb471-29"><a href="sequence-alignments.html#cb471-29" aria-hidden="true" tabindex="-1"></a>COI_set <span class="ot">&lt;-</span> <span class="fu">readDNAStringSet</span>(<span class="at">filepath =</span> <span class="st">&quot;~/Desktop/Teach_R/my_working_directory/my_collected_COI_fasta.txt&quot;</span>, </span>
<span id="cb471-30"><a href="sequence-alignments.html#cb471-30" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">format =</span> <span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb471-31"><a href="sequence-alignments.html#cb471-31" aria-hidden="true" tabindex="-1"></a>COI_set</span>
<span id="cb471-32"><a href="sequence-alignments.html#cb471-32" aria-hidden="true" tabindex="-1"></a><span class="co">#A DNAStringSet instance of length 12</span></span>
<span id="cb471-33"><a href="sequence-alignments.html#cb471-33" aria-hidden="true" tabindex="-1"></a><span class="co">#     width seq                                                                                                                  names               </span></span>
<span id="cb471-34"><a href="sequence-alignments.html#cb471-34" aria-hidden="true" tabindex="-1"></a><span class="co"># [1]  1539 GTGATAATTACTCGATGATTATTTTCTACCAACCACAAAGACATCGGAACTTTATAC...TATACGGATCCCCCCCACCTTACCACACTTTTGAGGAAGCCGTTTACTCCAAAATT Ameerega_bilinguis</span></span>
<span id="cb471-35"><a href="sequence-alignments.html#cb471-35" aria-hidden="true" tabindex="-1"></a><span class="co"># [2]  1539 GTGATAATTACTCGATGATTATTTTCCACCAACCATAAAGACATTGGAACCTTATAT...TGTACGGATCCCCCCCTCCCTACCATACATTTGAAGAAGCCGTTTATTCCAAAATT Ameerega_hahneli</span></span>
<span id="cb471-36"><a href="sequence-alignments.html#cb471-36" aria-hidden="true" tabindex="-1"></a><span class="co"># [3]  1539 GTGATAATTACTCGATGATTATTTTCCACCAACCATAAAGATATTGGAACCTTATAC...TATACGGGTCTCCCCCACCTTATCACACATTTGAGGAAGCCGTTTACTCCAAAATT Ameerega_parvula</span></span>
<span id="cb471-37"><a href="sequence-alignments.html#cb471-37" aria-hidden="true" tabindex="-1"></a><span class="co"># [4]   647 GAACTTTATACCTAGTGTTTGGGGCATGAGCAGGCATAGTCGGCACTGCTCTCAGCC...GTAACTTAAATACCACCTTCTTTGACCCAGCAGGGGGAGGTGACCCTGTTCTATAC Ameerega_braccata</span></span>
<span id="cb471-38"><a href="sequence-alignments.html#cb471-38" aria-hidden="true" tabindex="-1"></a><span class="co"># [5]   658 AACTCTATACCTAGTGTTTGGGGCATGAGCAGGCATAGTCGGCACTGCTCTTAGTCT...CCACTTTCTTTGACCCAGCAGGGGGAGGCGACCCCGTCCTGTACCAACACCTGTTC Ameerega_trivittata</span></span>
<span id="cb471-39"><a href="sequence-alignments.html#cb471-39" aria-hidden="true" tabindex="-1"></a><span class="co"># ...   ... ...</span></span>
<span id="cb471-40"><a href="sequence-alignments.html#cb471-40" aria-hidden="true" tabindex="-1"></a><span class="co"># [8]   658 AACTTTATACCTTGTATTTGGGGCATGGGCCGGAATAGTCGGAACCGCCCTAAGCCT...CTACTTTCTTTGACCCGGCCGGCGGAGGTGACCCTGTTCTCTACCAACATCTCTTT Colostethus_pratti</span></span>
<span id="cb471-41"><a href="sequence-alignments.html#cb471-41" aria-hidden="true" tabindex="-1"></a><span class="co"># [9]  1539 GTGATAATTACCCGATGATTATTTTCCACAAACCATAAAGATATTGGAACCCTATAT...TATATGGCTCCCCTCCTCCTTACCACACATTTGAGGAAGCCGTTTATTCTAAAGTA Epipedobates_boul...</span></span>
<span id="cb471-42"><a href="sequence-alignments.html#cb471-42" aria-hidden="true" tabindex="-1"></a><span class="co">#[10]  1539 GTGATAATTACCCGATGATTATTCTCCACAAACCATAAAGATATCGGAACCCTATAT...TATATGGCTCCCCTCCTCCTTACCACACATTCGAGGAAGCTGTTTATTCTAAAGTA Epipedobates_mach...</span></span>
<span id="cb471-43"><a href="sequence-alignments.html#cb471-43" aria-hidden="true" tabindex="-1"></a><span class="co">#[11]  1539 GTGATAATTACCCGATGATTATTTTCTACAAACCACAAAGACATTGGAACTTTATAT...TGTATGGATCACCACCACCATATCACACGTTTGAAGAAGCTGTTTATTCTAAAATT Silverstoneia_flo...</span></span>
<span id="cb471-44"><a href="sequence-alignments.html#cb471-44" aria-hidden="true" tabindex="-1"></a><span class="co">#[12]  1539 GTGATAATTACCCGATGATTATTTTCCACAAACCACAAAGACATTGGAACCCTATAC...TATACGGAACCCCTCCCCCTTATCACACATTTGAAGAAGCTGTTTACTCCAAAATT Leucostethus_fugax</span></span></code></pre></div>
<p>We just can proceed to do the alignment with function <code>AlignSeqs()</code> and <code>AdjustAlignment()</code>.</p>
<div class="sourceCode" id="cb472"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb472-1"><a href="sequence-alignments.html#cb472-1" aria-hidden="true" tabindex="-1"></a><span class="do">## make sure that DECIPHER is loaded</span></span>
<span id="cb472-2"><a href="sequence-alignments.html#cb472-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(DECIPHER)</span>
<span id="cb472-3"><a href="sequence-alignments.html#cb472-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb472-4"><a href="sequence-alignments.html#cb472-4" aria-hidden="true" tabindex="-1"></a><span class="do">## CYTB alignment and further adjustments if necessary</span></span>
<span id="cb472-5"><a href="sequence-alignments.html#cb472-5" aria-hidden="true" tabindex="-1"></a>CYTB_aligned <span class="ot">&lt;-</span> <span class="fu">AlignSeqs</span>(CYTB_set)</span>
<span id="cb472-6"><a href="sequence-alignments.html#cb472-6" aria-hidden="true" tabindex="-1"></a>CYTB_aligned <span class="ot">&lt;-</span> <span class="fu">AdjustAlignment</span>(CYTB_aligned)</span>
<span id="cb472-7"><a href="sequence-alignments.html#cb472-7" aria-hidden="true" tabindex="-1"></a>CYTB_aligned</span>
<span id="cb472-8"><a href="sequence-alignments.html#cb472-8" aria-hidden="true" tabindex="-1"></a><span class="co">#A DNAStringSet instance of length 11</span></span>
<span id="cb472-9"><a href="sequence-alignments.html#cb472-9" aria-hidden="true" tabindex="-1"></a><span class="co">#     width seq                                                                                                                  names               </span></span>
<span id="cb472-10"><a href="sequence-alignments.html#cb472-10" aria-hidden="true" tabindex="-1"></a><span class="co"># [1]   944 ---------------------------------------------------------...-------------------------------------------------------- Ameerega_bilinguis</span></span>
<span id="cb472-11"><a href="sequence-alignments.html#cb472-11" aria-hidden="true" tabindex="-1"></a><span class="co"># [2]   944 ---------------------------------------------------------...-------------------------------------------------------- Ameerega_hahneli</span></span>
<span id="cb472-12"><a href="sequence-alignments.html#cb472-12" aria-hidden="true" tabindex="-1"></a><span class="co"># [3]   944 ---------------------------------------------------------...-------------------------------------------------------- Ameerega_parvula</span></span>
<span id="cb472-13"><a href="sequence-alignments.html#cb472-13" aria-hidden="true" tabindex="-1"></a><span class="co"># [4]   944 AACACACCCCGCTCTAAAAATTATCAACAACTCATTCATTGACCTTCCATCCCCTGC...-------------------------------------------------------- Ameerega_braccata</span></span>
<span id="cb472-14"><a href="sequence-alignments.html#cb472-14" aria-hidden="true" tabindex="-1"></a><span class="co"># [5]   944 ---------------------------------------------------------...-------------------------------------------------------- Ameerega_trivittata</span></span>
<span id="cb472-15"><a href="sequence-alignments.html#cb472-15" aria-hidden="true" tabindex="-1"></a><span class="co"># ...   ... ...</span></span>
<span id="cb472-16"><a href="sequence-alignments.html#cb472-16" aria-hidden="true" tabindex="-1"></a><span class="co"># [7]   944 ---------------------------------------------------------...-------------------------------------------------------- Colostethus_panam...</span></span>
<span id="cb472-17"><a href="sequence-alignments.html#cb472-17" aria-hidden="true" tabindex="-1"></a><span class="co"># [8]   944 ---------------------------------------------------------...-------------------------------------------------------- Colostethus_pratti</span></span>
<span id="cb472-18"><a href="sequence-alignments.html#cb472-18" aria-hidden="true" tabindex="-1"></a><span class="co"># [9]   944 ---------------------------------------------------------...-------------------------------------------------------- Epipedobates_boul...</span></span>
<span id="cb472-19"><a href="sequence-alignments.html#cb472-19" aria-hidden="true" tabindex="-1"></a><span class="co">#[10]   944 ---------------------------------------------------------...-------------------------------------------------------- Epipedobates_mach...</span></span>
<span id="cb472-20"><a href="sequence-alignments.html#cb472-20" aria-hidden="true" tabindex="-1"></a><span class="co">#[11]   944 ---------------------------------------------------------...-------------------------------------------------------- Silverstoneia_flo...</span></span>
<span id="cb472-21"><a href="sequence-alignments.html#cb472-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb472-22"><a href="sequence-alignments.html#cb472-22" aria-hidden="true" tabindex="-1"></a><span class="do">## check CYTB alignment</span></span>
<span id="cb472-23"><a href="sequence-alignments.html#cb472-23" aria-hidden="true" tabindex="-1"></a><span class="fu">BrowseSeqs</span>(CYTB_aligned)</span>
<span id="cb472-24"><a href="sequence-alignments.html#cb472-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb472-25"><a href="sequence-alignments.html#cb472-25" aria-hidden="true" tabindex="-1"></a><span class="do">## COI alignment and further adjustments if necessary</span></span>
<span id="cb472-26"><a href="sequence-alignments.html#cb472-26" aria-hidden="true" tabindex="-1"></a>COI_aligned <span class="ot">&lt;-</span> <span class="fu">AlignSeqs</span>(COI_set)</span>
<span id="cb472-27"><a href="sequence-alignments.html#cb472-27" aria-hidden="true" tabindex="-1"></a>COI_aligned <span class="ot">&lt;-</span> <span class="fu">AdjustAlignment</span>(COI_aligned)</span>
<span id="cb472-28"><a href="sequence-alignments.html#cb472-28" aria-hidden="true" tabindex="-1"></a>COI_aligned</span>
<span id="cb472-29"><a href="sequence-alignments.html#cb472-29" aria-hidden="true" tabindex="-1"></a><span class="co">#  A DNAStringSet instance of length 12</span></span>
<span id="cb472-30"><a href="sequence-alignments.html#cb472-30" aria-hidden="true" tabindex="-1"></a><span class="co">#     width seq                                                                                                                  names               </span></span>
<span id="cb472-31"><a href="sequence-alignments.html#cb472-31" aria-hidden="true" tabindex="-1"></a><span class="co"># [1]  1539 GTGATAATTACTCGATGATTATTTTCTACCAACCACAAAGACATCGGAACTTTATAC...TATACGGATCCCCCCCACCTTACCACACTTTTGAGGAAGCCGTTTACTCCAAAATT Ameerega_bilinguis</span></span>
<span id="cb472-32"><a href="sequence-alignments.html#cb472-32" aria-hidden="true" tabindex="-1"></a><span class="co"># [2]  1539 GTGATAATTACTCGATGATTATTTTCCACCAACCATAAAGACATTGGAACCTTATAT...TGTACGGATCCCCCCCTCCCTACCATACATTTGAAGAAGCCGTTTATTCCAAAATT Ameerega_hahneli</span></span>
<span id="cb472-33"><a href="sequence-alignments.html#cb472-33" aria-hidden="true" tabindex="-1"></a><span class="co"># [3]  1539 GTGATAATTACTCGATGATTATTTTCCACCAACCATAAAGATATTGGAACCTTATAC...TATACGGGTCTCCCCCACCTTATCACACATTTGAGGAAGCCGTTTACTCCAAAATT Ameerega_parvula</span></span>
<span id="cb472-34"><a href="sequence-alignments.html#cb472-34" aria-hidden="true" tabindex="-1"></a><span class="co"># [4]  1539 ----------------------------------------------GAACTTTATAC...-------------------------------------------------------- Ameerega_braccata</span></span>
<span id="cb472-35"><a href="sequence-alignments.html#cb472-35" aria-hidden="true" tabindex="-1"></a><span class="co"># [5]  1539 -----------------------------------------------AACTCTATAC...-------------------------------------------------------- Ameerega_trivittata</span></span>
<span id="cb472-36"><a href="sequence-alignments.html#cb472-36" aria-hidden="true" tabindex="-1"></a><span class="co"># ...   ... ...</span></span>
<span id="cb472-37"><a href="sequence-alignments.html#cb472-37" aria-hidden="true" tabindex="-1"></a><span class="co"># [8]  1539 -----------------------------------------------AACTTTATAC...-------------------------------------------------------- Colostethus_pratti</span></span>
<span id="cb472-38"><a href="sequence-alignments.html#cb472-38" aria-hidden="true" tabindex="-1"></a><span class="co"># [9]  1539 GTGATAATTACCCGATGATTATTTTCCACAAACCATAAAGATATTGGAACCCTATAT...TATATGGCTCCCCTCCTCCTTACCACACATTTGAGGAAGCCGTTTATTCTAAAGTA Epipedobates_boul...</span></span>
<span id="cb472-39"><a href="sequence-alignments.html#cb472-39" aria-hidden="true" tabindex="-1"></a><span class="co">#[10]  1539 GTGATAATTACCCGATGATTATTCTCCACAAACCATAAAGATATCGGAACCCTATAT...TATATGGCTCCCCTCCTCCTTACCACACATTCGAGGAAGCTGTTTATTCTAAAGTA Epipedobates_mach...</span></span>
<span id="cb472-40"><a href="sequence-alignments.html#cb472-40" aria-hidden="true" tabindex="-1"></a><span class="co">#[11]  1539 GTGATAATTACCCGATGATTATTTTCTACAAACCACAAAGACATTGGAACTTTATAT...TGTATGGATCACCACCACCATATCACACGTTTGAAGAAGCTGTTTATTCTAAAATT Silverstoneia_flo...</span></span>
<span id="cb472-41"><a href="sequence-alignments.html#cb472-41" aria-hidden="true" tabindex="-1"></a><span class="co">#[12]  1539 GTGATAATTACCCGATGATTATTTTCCACAAACCACAAAGACATTGGAACCCTATAC...TATACGGAACCCCTCCCCCTTATCACACATTTGAAGAAGCTGTTTACTCCAAAATT Leucostethus_fugax</span></span>
<span id="cb472-42"><a href="sequence-alignments.html#cb472-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb472-43"><a href="sequence-alignments.html#cb472-43" aria-hidden="true" tabindex="-1"></a><span class="do">## check COI alignment</span></span>
<span id="cb472-44"><a href="sequence-alignments.html#cb472-44" aria-hidden="true" tabindex="-1"></a><span class="fu">BrowseSeqs</span>(COI_aligned)</span></code></pre></div>
</div>
<div id="concatenate-with-data.table" class="section level2" number="11.8">
<h2><span class="header-section-number">11.8</span> Concatenate with <code>data.table</code></h2>
<p><strong>14)</strong> We can now concatenate all sequences before we proceed to phylogenetic tree estimation. Given that <code>CYTB_aligned</code> lacks one sequence, the process of concatenation is convoluted.</p>
<p>Here is procedure derived from a suggestion <a href="https://stackoverflow.com/questions/53549442/how-to-concatenate-merge-aastringsets-by-name" target="_blank"><strong>here</strong></a> using the R package <a href="https://cran.r-project.org/web/packages/data.table/index.html" target="_blank"><strong>data.table</strong></a>.</p>
<div class="sourceCode" id="cb473"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb473-1"><a href="sequence-alignments.html#cb473-1" aria-hidden="true" tabindex="-1"></a><span class="do">## make sure that data.table is loaded</span></span>
<span id="cb473-2"><a href="sequence-alignments.html#cb473-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(data.table)</span>
<span id="cb473-3"><a href="sequence-alignments.html#cb473-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb473-4"><a href="sequence-alignments.html#cb473-4" aria-hidden="true" tabindex="-1"></a><span class="do">## transform into matrices using data.table</span></span>
<span id="cb473-5"><a href="sequence-alignments.html#cb473-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb473-6"><a href="sequence-alignments.html#cb473-6" aria-hidden="true" tabindex="-1"></a>CYTB_aligned.dt<span class="ot">&lt;-</span><span class="fu">data.table</span>(<span class="fu">as.matrix</span>(CYTB_aligned), <span class="at">keep.rownames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb473-7"><a href="sequence-alignments.html#cb473-7" aria-hidden="true" tabindex="-1"></a>COI_aligned.dt<span class="ot">&lt;-</span><span class="fu">data.table</span>(<span class="fu">as.matrix</span>(COI_aligned), <span class="at">keep.rownames =</span> <span class="cn">TRUE</span>)</span>
<span id="cb473-8"><a href="sequence-alignments.html#cb473-8" aria-hidden="true" tabindex="-1"></a>COI_CYTB.dt   <span class="ot">&lt;-</span><span class="fu">merge</span>(COI_aligned.dt, CYTB_aligned.dt, <span class="at">by=</span><span class="st">&quot;rn&quot;</span>, <span class="at">all=</span><span class="cn">TRUE</span>)</span>
<span id="cb473-9"><a href="sequence-alignments.html#cb473-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb473-10"><a href="sequence-alignments.html#cb473-10" aria-hidden="true" tabindex="-1"></a><span class="do">## slightly modified from original, added arg &quot;x&quot;</span></span>
<span id="cb473-11"><a href="sequence-alignments.html#cb473-11" aria-hidden="true" tabindex="-1"></a>f_dowle <span class="ot">&lt;-</span> <span class="cf">function</span>(dt, x) {     </span>
<span id="cb473-12"><a href="sequence-alignments.html#cb473-12" aria-hidden="true" tabindex="-1"></a>                na.replace <span class="ot">=</span> <span class="cf">function</span>(v,<span class="at">value=</span>x) { v[<span class="fu">is.na</span>(v)] <span class="ot">=</span> value; v }</span>
<span id="cb473-13"><a href="sequence-alignments.html#cb473-13" aria-hidden="true" tabindex="-1"></a>                       <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">names</span>(dt))</span>
<span id="cb473-14"><a href="sequence-alignments.html#cb473-14" aria-hidden="true" tabindex="-1"></a>        <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text=</span><span class="fu">paste</span>(<span class="st">&quot;dt[,&quot;</span>,i,<span class="st">&quot;:=na.replace(&quot;</span>,i,<span class="st">&quot;)]&quot;</span>)))    }</span>
<span id="cb473-15"><a href="sequence-alignments.html#cb473-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb473-16"><a href="sequence-alignments.html#cb473-16" aria-hidden="true" tabindex="-1"></a><span class="do">## use f_dowle() to replace NA for &quot;-&quot;</span></span>
<span id="cb473-17"><a href="sequence-alignments.html#cb473-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb473-18"><a href="sequence-alignments.html#cb473-18" aria-hidden="true" tabindex="-1"></a><span class="fu">f_dowle</span>(COI_CYTB.dt, <span class="st">&quot;-&quot;</span>)</span>
<span id="cb473-19"><a href="sequence-alignments.html#cb473-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb473-20"><a href="sequence-alignments.html#cb473-20" aria-hidden="true" tabindex="-1"></a><span class="do">## return to matrix</span></span>
<span id="cb473-21"><a href="sequence-alignments.html#cb473-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb473-22"><a href="sequence-alignments.html#cb473-22" aria-hidden="true" tabindex="-1"></a>COI_CYTB <span class="ot">&lt;-</span> <span class="fu">apply</span>(COI_CYTB.dt[ ,<span class="sc">!</span><span class="st">&quot;rn&quot;</span>], <span class="dv">1</span>, paste, <span class="at">collapse=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb473-23"><a href="sequence-alignments.html#cb473-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb473-24"><a href="sequence-alignments.html#cb473-24" aria-hidden="true" tabindex="-1"></a><span class="do">## Convert back to DNAStringSet and add back names</span></span>
<span id="cb473-25"><a href="sequence-alignments.html#cb473-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb473-26"><a href="sequence-alignments.html#cb473-26" aria-hidden="true" tabindex="-1"></a>COI_CYTB_DT_path_set <span class="ot">&lt;-</span> <span class="fu">DNAStringSet</span>(COI_CYTB)</span>
<span id="cb473-27"><a href="sequence-alignments.html#cb473-27" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(COI_CYTB_DT_path_set) <span class="ot">&lt;-</span> COI_CYTB.dt<span class="sc">$</span>rn</span>
<span id="cb473-28"><a href="sequence-alignments.html#cb473-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb473-29"><a href="sequence-alignments.html#cb473-29" aria-hidden="true" tabindex="-1"></a><span class="do">## check COI_CYTB alignment</span></span>
<span id="cb473-30"><a href="sequence-alignments.html#cb473-30" aria-hidden="true" tabindex="-1"></a><span class="fu">BrowseSeqs</span>(COI_CYTB_DT_path_set)</span></code></pre></div>
</div>
<div id="concatenate-with-biostrings" class="section level2" number="11.9">
<h2><span class="header-section-number">11.9</span> Concatenate with <code>Biostrings</code></h2>
<p><strong>15)</strong> An alternative way to perform concatenation using Biostrings is as follows.</p>
<div class="sourceCode" id="cb474"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb474-1"><a href="sequence-alignments.html#cb474-1" aria-hidden="true" tabindex="-1"></a><span class="do">## make sure that Biostrings is loaded</span></span>
<span id="cb474-2"><a href="sequence-alignments.html#cb474-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(Biostrings)</span>
<span id="cb474-3"><a href="sequence-alignments.html#cb474-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-4"><a href="sequence-alignments.html#cb474-4" aria-hidden="true" tabindex="-1"></a><span class="do">## find what sequence (taxon) name is missing</span></span>
<span id="cb474-5"><a href="sequence-alignments.html#cb474-5" aria-hidden="true" tabindex="-1"></a>missing_COI_taxon <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">setdiff</span>(<span class="fu">names</span>(CYTB_aligned),<span class="fu">names</span>(COI_aligned))</span>
<span id="cb474-6"><a href="sequence-alignments.html#cb474-6" aria-hidden="true" tabindex="-1"></a>missing_COI_taxon</span>
<span id="cb474-7"><a href="sequence-alignments.html#cb474-7" aria-hidden="true" tabindex="-1"></a><span class="co">#character(0)</span></span>
<span id="cb474-8"><a href="sequence-alignments.html#cb474-8" aria-hidden="true" tabindex="-1"></a>missing_CYTB_taxon <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">setdiff</span>(<span class="fu">names</span>(COI_aligned),<span class="fu">names</span>(CYTB_aligned))</span>
<span id="cb474-9"><a href="sequence-alignments.html#cb474-9" aria-hidden="true" tabindex="-1"></a>missing_CYTB_taxon</span>
<span id="cb474-10"><a href="sequence-alignments.html#cb474-10" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;Leucostethus_fugax&quot;</span></span>
<span id="cb474-11"><a href="sequence-alignments.html#cb474-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-12"><a href="sequence-alignments.html#cb474-12" aria-hidden="true" tabindex="-1"></a><span class="do">## The sequence of &quot;Leucostethus_fugax&quot; is missing in CYTB matrix, we need add such sequence as gaps &quot;-&quot; to the aligned matrix.</span></span>
<span id="cb474-13"><a href="sequence-alignments.html#cb474-13" aria-hidden="true" tabindex="-1"></a>CYTB_aligned</span>
<span id="cb474-14"><a href="sequence-alignments.html#cb474-14" aria-hidden="true" tabindex="-1"></a><span class="co">#A DNAStringSet instance of length 11</span></span>
<span id="cb474-15"><a href="sequence-alignments.html#cb474-15" aria-hidden="true" tabindex="-1"></a><span class="co">#     width seq                                                                                                                  names               </span></span>
<span id="cb474-16"><a href="sequence-alignments.html#cb474-16" aria-hidden="true" tabindex="-1"></a><span class="co"># [1]   944 ---------------------------------------------------------...-------------------------------------------------------- Ameerega_bilinguis</span></span>
<span id="cb474-17"><a href="sequence-alignments.html#cb474-17" aria-hidden="true" tabindex="-1"></a><span class="co"># [2]   944 ---------------------------------------------------------...-------------------------------------------------------- Ameerega_hahneli</span></span>
<span id="cb474-18"><a href="sequence-alignments.html#cb474-18" aria-hidden="true" tabindex="-1"></a><span class="co"># [3]   944 ---------------------------------------------------------...-------------------------------------------------------- Ameerega_parvula</span></span>
<span id="cb474-19"><a href="sequence-alignments.html#cb474-19" aria-hidden="true" tabindex="-1"></a><span class="co"># [4]   944 AACACACCCCGCTCTAAAAATTATCAACAACTCATTCATTGACCTTCCATCCCCTGC...-------------------------------------------------------- Ameerega_braccata</span></span>
<span id="cb474-20"><a href="sequence-alignments.html#cb474-20" aria-hidden="true" tabindex="-1"></a><span class="co"># [5]   944 ---------------------------------------------------------...-------------------------------------------------------- Ameerega_trivittata</span></span>
<span id="cb474-21"><a href="sequence-alignments.html#cb474-21" aria-hidden="true" tabindex="-1"></a><span class="co"># ...   ... ...</span></span>
<span id="cb474-22"><a href="sequence-alignments.html#cb474-22" aria-hidden="true" tabindex="-1"></a><span class="co"># [7]   944 ---------------------------------------------------------...-------------------------------------------------------- Colostethus_panam...</span></span>
<span id="cb474-23"><a href="sequence-alignments.html#cb474-23" aria-hidden="true" tabindex="-1"></a><span class="co"># [8]   944 ---------------------------------------------------------...-------------------------------------------------------- Colostethus_pratti</span></span>
<span id="cb474-24"><a href="sequence-alignments.html#cb474-24" aria-hidden="true" tabindex="-1"></a><span class="co"># [9]   944 ---------------------------------------------------------...-------------------------------------------------------- Epipedobates_boul...</span></span>
<span id="cb474-25"><a href="sequence-alignments.html#cb474-25" aria-hidden="true" tabindex="-1"></a><span class="co">#[10]   944 ---------------------------------------------------------...-------------------------------------------------------- Epipedobates_mach...</span></span>
<span id="cb474-26"><a href="sequence-alignments.html#cb474-26" aria-hidden="true" tabindex="-1"></a><span class="co">#[11]   944 ---------------------------------------------------------...-------------------------------------------------------- Silverstoneia_flo...</span></span>
<span id="cb474-27"><a href="sequence-alignments.html#cb474-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-28"><a href="sequence-alignments.html#cb474-28" aria-hidden="true" tabindex="-1"></a><span class="do">## we notice that each aligned CYTB sequences has 944 characters, so we need to create a vector named Leucostethus_fugax with 944 &quot;-&quot;.</span></span>
<span id="cb474-29"><a href="sequence-alignments.html#cb474-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-30"><a href="sequence-alignments.html#cb474-30" aria-hidden="true" tabindex="-1"></a>missing_taxon_CYTB <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">rep</span>(<span class="st">&quot;-&quot;</span>,<span class="dv">944</span>), <span class="at">collapse =</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb474-31"><a href="sequence-alignments.html#cb474-31" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(missing_taxon_CYTB) <span class="ot">&lt;-</span> <span class="st">&quot;Leucostethus_fugax&quot;</span></span>
<span id="cb474-32"><a href="sequence-alignments.html#cb474-32" aria-hidden="true" tabindex="-1"></a>missing_taxon_CYTB_set <span class="ot">&lt;-</span> <span class="fu">DNAStringSet</span>(missing_taxon_CYTB)</span>
<span id="cb474-33"><a href="sequence-alignments.html#cb474-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-34"><a href="sequence-alignments.html#cb474-34" aria-hidden="true" tabindex="-1"></a><span class="do">## we add missing sequence</span></span>
<span id="cb474-35"><a href="sequence-alignments.html#cb474-35" aria-hidden="true" tabindex="-1"></a>CYTB_aligned_add <span class="ot">&lt;-</span> CYTB_aligned</span>
<span id="cb474-36"><a href="sequence-alignments.html#cb474-36" aria-hidden="true" tabindex="-1"></a>names_on_stringset <span class="ot">&lt;-</span> <span class="fu">names</span>(CYTB_aligned_add)</span>
<span id="cb474-37"><a href="sequence-alignments.html#cb474-37" aria-hidden="true" tabindex="-1"></a>names_on_stringset <span class="ot">&lt;-</span> <span class="fu">c</span>(names_on_stringset, <span class="st">&quot;Leucostethus_fugax&quot;</span>)</span>
<span id="cb474-38"><a href="sequence-alignments.html#cb474-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-39"><a href="sequence-alignments.html#cb474-39" aria-hidden="true" tabindex="-1"></a><span class="do">## append sequence at the [12] slots and add taxon name</span></span>
<span id="cb474-40"><a href="sequence-alignments.html#cb474-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-41"><a href="sequence-alignments.html#cb474-41" aria-hidden="true" tabindex="-1"></a>CYTB_aligned_add[<span class="dv">12</span>] <span class="ot">&lt;-</span> missing_taxon_CYTB_set</span>
<span id="cb474-42"><a href="sequence-alignments.html#cb474-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-43"><a href="sequence-alignments.html#cb474-43" aria-hidden="true" tabindex="-1"></a><span class="do">## change names</span></span>
<span id="cb474-44"><a href="sequence-alignments.html#cb474-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-45"><a href="sequence-alignments.html#cb474-45" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(CYTB_aligned_add) <span class="ot">&lt;-</span> names_on_stringset</span>
<span id="cb474-46"><a href="sequence-alignments.html#cb474-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-47"><a href="sequence-alignments.html#cb474-47" aria-hidden="true" tabindex="-1"></a><span class="do">## Confirm that we do not have any more missing taxa (i.e., character(0))</span></span>
<span id="cb474-48"><a href="sequence-alignments.html#cb474-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-49"><a href="sequence-alignments.html#cb474-49" aria-hidden="true" tabindex="-1"></a>missing_COI_taxon <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">setdiff</span>(<span class="fu">names</span>(CYTB_aligned_add),<span class="fu">names</span>(COI_aligned))</span>
<span id="cb474-50"><a href="sequence-alignments.html#cb474-50" aria-hidden="true" tabindex="-1"></a>missing_COI_taxon</span>
<span id="cb474-51"><a href="sequence-alignments.html#cb474-51" aria-hidden="true" tabindex="-1"></a><span class="co">#character(0)</span></span>
<span id="cb474-52"><a href="sequence-alignments.html#cb474-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-53"><a href="sequence-alignments.html#cb474-53" aria-hidden="true" tabindex="-1"></a>missing_CYTB_taxon <span class="ot">&lt;-</span> base<span class="sc">::</span><span class="fu">setdiff</span>(<span class="fu">names</span>(COI_aligned),<span class="fu">names</span>(CYTB_aligned_add))</span>
<span id="cb474-54"><a href="sequence-alignments.html#cb474-54" aria-hidden="true" tabindex="-1"></a>missing_CYTB_taxon</span>
<span id="cb474-55"><a href="sequence-alignments.html#cb474-55" aria-hidden="true" tabindex="-1"></a><span class="co">#character(0)</span></span>
<span id="cb474-56"><a href="sequence-alignments.html#cb474-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-57"><a href="sequence-alignments.html#cb474-57" aria-hidden="true" tabindex="-1"></a><span class="do">## IMPORTANT: to make concatenation, you have to make sure that the order of names in both COI and CYTB are the same</span></span>
<span id="cb474-58"><a href="sequence-alignments.html#cb474-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-59"><a href="sequence-alignments.html#cb474-59" aria-hidden="true" tabindex="-1"></a>COI_aligned_order <span class="ot">&lt;-</span> COI_aligned[<span class="fu">order</span>(<span class="fu">names</span>(COI_aligned), <span class="at">decreasing=</span><span class="cn">FALSE</span>)]</span>
<span id="cb474-60"><a href="sequence-alignments.html#cb474-60" aria-hidden="true" tabindex="-1"></a>CYTB_aligned_add_order <span class="ot">&lt;-</span> CYTB_aligned_add[<span class="fu">order</span>(<span class="fu">names</span>(CYTB_aligned_add), <span class="at">decreasing=</span><span class="cn">FALSE</span>)]</span>
<span id="cb474-61"><a href="sequence-alignments.html#cb474-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-62"><a href="sequence-alignments.html#cb474-62" aria-hidden="true" tabindex="-1"></a><span class="do">## Now, we can finally concatenate with xsact()</span></span>
<span id="cb474-63"><a href="sequence-alignments.html#cb474-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-64"><a href="sequence-alignments.html#cb474-64" aria-hidden="true" tabindex="-1"></a>COI_CYTB_String_path_set <span class="ot">&lt;-</span> <span class="fu">xscat</span>(COI_aligned_order,CYTB_aligned_add_order)</span>
<span id="cb474-65"><a href="sequence-alignments.html#cb474-65" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(COI_CYTB_String_path_set) <span class="ot">&lt;-</span> <span class="fu">names</span>(COI_aligned_order)</span>
<span id="cb474-66"><a href="sequence-alignments.html#cb474-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb474-67"><a href="sequence-alignments.html#cb474-67" aria-hidden="true" tabindex="-1"></a><span class="do">## check COI_CYTB alignment</span></span>
<span id="cb474-68"><a href="sequence-alignments.html#cb474-68" aria-hidden="true" tabindex="-1"></a><span class="fu">BrowseSeqs</span>(COI_CYTB_String_path_set )</span></code></pre></div>
<p><strong>16)</strong> We can finally save these concatenated sequences to a <code>fasta</code> file.</p>
<div class="sourceCode" id="cb475"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb475-1"><a href="sequence-alignments.html#cb475-1" aria-hidden="true" tabindex="-1"></a><span class="co"># this is exclusive to your OWN COMPUTER change it accordingly</span></span>
<span id="cb475-2"><a href="sequence-alignments.html#cb475-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;~/Desktop/Teach_R/my_working_directory&quot;</span>)</span>
<span id="cb475-3"><a href="sequence-alignments.html#cb475-3" aria-hidden="true" tabindex="-1"></a><span class="fu">writeXStringSet</span>(COI_CYTB_DT_path_set, <span class="at">file=</span><span class="st">&quot;my_collected_COI_CYTB_DT_path_fasta.txt&quot;</span>, <span class="at">format=</span><span class="st">&quot;fasta&quot;</span>)</span>
<span id="cb475-4"><a href="sequence-alignments.html#cb475-4" aria-hidden="true" tabindex="-1"></a><span class="fu">writeXStringSet</span>(COI_CYTB_String_path_set, <span class="at">file=</span><span class="st">&quot;my_collected_COI_CYTB_String_path_fasta.txt&quot;</span>, <span class="at">format=</span><span class="st">&quot;fasta&quot;</span>)</span></code></pre></div>
</div>
<div id="visualizing-alignments-in-r" class="section level2" number="11.10">
<h2><span class="header-section-number">11.10</span> Visualizing alignments in R</h2>
<p><strong>17)</strong> Visualizing your alignment in R is not yet optimal, yet there is at least one ggplot2 add-on that can do it <a href="https://cran.r-project.org/web/packages/ggmsa/vignettes/ggmsa.html#load-sample-data" target="_blank"><strong>ggmsa</strong></a>.
This package requires another package: <a href="https://cran.r-project.org/web/packages/seqmagick/vignettes/seqmagick.html" target="_blank"><strong>seqmagick</strong></a> for sequence manipulation utilities. The installation of both seem a bit convoluted, yet they would do the job.</p>
<div class="sourceCode" id="cb476"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb476-1"><a href="sequence-alignments.html#cb476-1" aria-hidden="true" tabindex="-1"></a><span class="do">## installing the packages</span></span>
<span id="cb476-2"><a href="sequence-alignments.html#cb476-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggmsa&quot;</span>)</span>
<span id="cb476-3"><a href="sequence-alignments.html#cb476-3" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;seqmagick&quot;</span>)</span>
<span id="cb476-4"><a href="sequence-alignments.html#cb476-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-5"><a href="sequence-alignments.html#cb476-5" aria-hidden="true" tabindex="-1"></a><span class="do">## load packages including cowplot</span></span>
<span id="cb476-6"><a href="sequence-alignments.html#cb476-6" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggmsa)</span>
<span id="cb476-7"><a href="sequence-alignments.html#cb476-7" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(seqmagick)</span>
<span id="cb476-8"><a href="sequence-alignments.html#cb476-8" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(cowplot)</span>
<span id="cb476-9"><a href="sequence-alignments.html#cb476-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-10"><a href="sequence-alignments.html#cb476-10" aria-hidden="true" tabindex="-1"></a><span class="do">## you need just to provide the fasta file that contains the alignment from DECIPHER (change path to your computer)</span></span>
<span id="cb476-11"><a href="sequence-alignments.html#cb476-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Notice: I have renamed the taxa to fit on the plot if called</span></span>
<span id="cb476-12"><a href="sequence-alignments.html#cb476-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-13"><a href="sequence-alignments.html#cb476-13" aria-hidden="true" tabindex="-1"></a>AA_alignment <span class="ot">&lt;-</span> <span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/my_working_directory/Ameerega_AA_aligned_end_renamed.fasta&quot;</span></span>
<span id="cb476-14"><a href="sequence-alignments.html#cb476-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-15"><a href="sequence-alignments.html#cb476-15" aria-hidden="true" tabindex="-1"></a><span class="do">## we can plot the amino acid sequence from 1 to 300 residues. You can select any section of sites in your alignment.</span></span>
<span id="cb476-16"><a href="sequence-alignments.html#cb476-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-17"><a href="sequence-alignments.html#cb476-17" aria-hidden="true" tabindex="-1"></a>my_aaplot_1 <span class="ot">&lt;-</span> <span class="fu">ggmsa</span>(<span class="at">msa =</span> AA_alignment,</span>
<span id="cb476-18"><a href="sequence-alignments.html#cb476-18" aria-hidden="true" tabindex="-1"></a>                 <span class="at">start =</span> <span class="dv">1</span>,</span>
<span id="cb476-19"><a href="sequence-alignments.html#cb476-19" aria-hidden="true" tabindex="-1"></a>                   <span class="at">end =</span> <span class="dv">100</span>,</span>
<span id="cb476-20"><a href="sequence-alignments.html#cb476-20" aria-hidden="true" tabindex="-1"></a>                  <span class="at">font =</span> <span class="st">&quot;helvetical&quot;</span>,</span>
<span id="cb476-21"><a href="sequence-alignments.html#cb476-21" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="st">&quot;Chemistry_AA&quot;</span>,</span>
<span id="cb476-22"><a href="sequence-alignments.html#cb476-22" aria-hidden="true" tabindex="-1"></a>            <span class="at">char_width =</span> <span class="fl">0.9</span>)</span>
<span id="cb476-23"><a href="sequence-alignments.html#cb476-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-24"><a href="sequence-alignments.html#cb476-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-25"><a href="sequence-alignments.html#cb476-25" aria-hidden="true" tabindex="-1"></a>my_aaplot_2 <span class="ot">&lt;-</span> <span class="fu">ggmsa</span>(<span class="at">msa =</span> AA_alignment,</span>
<span id="cb476-26"><a href="sequence-alignments.html#cb476-26" aria-hidden="true" tabindex="-1"></a>                 <span class="at">start =</span> <span class="dv">101</span>,</span>
<span id="cb476-27"><a href="sequence-alignments.html#cb476-27" aria-hidden="true" tabindex="-1"></a>                   <span class="at">end =</span> <span class="dv">200</span>,</span>
<span id="cb476-28"><a href="sequence-alignments.html#cb476-28" aria-hidden="true" tabindex="-1"></a>                  <span class="at">font =</span> <span class="st">&quot;helvetical&quot;</span>,</span>
<span id="cb476-29"><a href="sequence-alignments.html#cb476-29" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="st">&quot;Chemistry_AA&quot;</span>,</span>
<span id="cb476-30"><a href="sequence-alignments.html#cb476-30" aria-hidden="true" tabindex="-1"></a>            <span class="at">char_width =</span> <span class="fl">0.9</span>)</span>
<span id="cb476-31"><a href="sequence-alignments.html#cb476-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-32"><a href="sequence-alignments.html#cb476-32" aria-hidden="true" tabindex="-1"></a>my_aaplot_3 <span class="ot">&lt;-</span> <span class="fu">ggmsa</span>(<span class="at">msa =</span> AA_alignment,</span>
<span id="cb476-33"><a href="sequence-alignments.html#cb476-33" aria-hidden="true" tabindex="-1"></a>                 <span class="at">start =</span> <span class="dv">201</span>,</span>
<span id="cb476-34"><a href="sequence-alignments.html#cb476-34" aria-hidden="true" tabindex="-1"></a>                   <span class="at">end =</span> <span class="dv">300</span>,</span>
<span id="cb476-35"><a href="sequence-alignments.html#cb476-35" aria-hidden="true" tabindex="-1"></a>                  <span class="at">font =</span> <span class="st">&quot;helvetical&quot;</span>,</span>
<span id="cb476-36"><a href="sequence-alignments.html#cb476-36" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="st">&quot;Chemistry_AA&quot;</span>,</span>
<span id="cb476-37"><a href="sequence-alignments.html#cb476-37" aria-hidden="true" tabindex="-1"></a>            <span class="at">char_width =</span> <span class="fl">0.9</span>)</span>
<span id="cb476-38"><a href="sequence-alignments.html#cb476-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-39"><a href="sequence-alignments.html#cb476-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-40"><a href="sequence-alignments.html#cb476-40" aria-hidden="true" tabindex="-1"></a><span class="do">## we plot all using cowplot</span></span>
<span id="cb476-41"><a href="sequence-alignments.html#cb476-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb476-42"><a href="sequence-alignments.html#cb476-42" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(my_aaplot_1, </span>
<span id="cb476-43"><a href="sequence-alignments.html#cb476-43" aria-hidden="true" tabindex="-1"></a>          my_aaplot_2, </span>
<span id="cb476-44"><a href="sequence-alignments.html#cb476-44" aria-hidden="true" tabindex="-1"></a>          my_aaplot_3, </span>
<span id="cb476-45"><a href="sequence-alignments.html#cb476-45" aria-hidden="true" tabindex="-1"></a>          <span class="at">ncol =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>These plots are very demanding on computer memory and slow (not optimal). Here is the alignment visualized. The order of the sequences is the same one as is in the fasta input file.</p>
<p><img src="figures/session_11_figs/10_ggmsa_AA_plot.png" width="90%" /></p>
<p><strong>18)</strong> For nucleotide alignment visualization, we can proceed similarly.</p>
<div class="sourceCode" id="cb477"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb477-1"><a href="sequence-alignments.html#cb477-1" aria-hidden="true" tabindex="-1"></a><span class="do">## load packages including cowplot</span></span>
<span id="cb477-2"><a href="sequence-alignments.html#cb477-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggmsa)</span>
<span id="cb477-3"><a href="sequence-alignments.html#cb477-3" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(seqmagick)</span>
<span id="cb477-4"><a href="sequence-alignments.html#cb477-4" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(cowplot)</span>
<span id="cb477-5"><a href="sequence-alignments.html#cb477-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb477-6"><a href="sequence-alignments.html#cb477-6" aria-hidden="true" tabindex="-1"></a><span class="do">## you need just to provide the fasta file that contains the alignment from DECIPHER (change path to your computer)</span></span>
<span id="cb477-7"><a href="sequence-alignments.html#cb477-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb477-8"><a href="sequence-alignments.html#cb477-8" aria-hidden="true" tabindex="-1"></a>nt_alignment <span class="ot">&lt;-</span> <span class="st">&quot;~/Desktop/Teach_R/class_pages_reference/bioinformatics_gitbook_1/my_working_directory/Ameerega_aligned_end.fasta&quot;</span></span>
<span id="cb477-9"><a href="sequence-alignments.html#cb477-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb477-10"><a href="sequence-alignments.html#cb477-10" aria-hidden="true" tabindex="-1"></a><span class="do">## we can plot the amino acid sequence from 1 to 600 residues. You can select any section of sites in your alignment.</span></span>
<span id="cb477-11"><a href="sequence-alignments.html#cb477-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb477-12"><a href="sequence-alignments.html#cb477-12" aria-hidden="true" tabindex="-1"></a>my_aaplot_1 <span class="ot">&lt;-</span> <span class="fu">ggmsa</span>(<span class="at">msa =</span> nt_alignment,</span>
<span id="cb477-13"><a href="sequence-alignments.html#cb477-13" aria-hidden="true" tabindex="-1"></a>                 <span class="at">start =</span> <span class="dv">1</span>,</span>
<span id="cb477-14"><a href="sequence-alignments.html#cb477-14" aria-hidden="true" tabindex="-1"></a>                   <span class="at">end =</span> <span class="dv">200</span>,</span>
<span id="cb477-15"><a href="sequence-alignments.html#cb477-15" aria-hidden="true" tabindex="-1"></a>                  <span class="at">font =</span> <span class="st">&quot;helvetical&quot;</span>,</span>
<span id="cb477-16"><a href="sequence-alignments.html#cb477-16" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="st">&quot;Shapely_NT&quot;</span>,</span>
<span id="cb477-17"><a href="sequence-alignments.html#cb477-17" aria-hidden="true" tabindex="-1"></a>            <span class="at">char_width =</span> <span class="fl">0.3</span>)</span>
<span id="cb477-18"><a href="sequence-alignments.html#cb477-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb477-19"><a href="sequence-alignments.html#cb477-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb477-20"><a href="sequence-alignments.html#cb477-20" aria-hidden="true" tabindex="-1"></a>my_aaplot_2 <span class="ot">&lt;-</span> <span class="fu">ggmsa</span>(<span class="at">msa =</span> nt_alignment,</span>
<span id="cb477-21"><a href="sequence-alignments.html#cb477-21" aria-hidden="true" tabindex="-1"></a>                 <span class="at">start =</span> <span class="dv">201</span>,</span>
<span id="cb477-22"><a href="sequence-alignments.html#cb477-22" aria-hidden="true" tabindex="-1"></a>                   <span class="at">end =</span> <span class="dv">400</span>,</span>
<span id="cb477-23"><a href="sequence-alignments.html#cb477-23" aria-hidden="true" tabindex="-1"></a>                  <span class="at">font =</span> <span class="st">&quot;helvetical&quot;</span>,</span>
<span id="cb477-24"><a href="sequence-alignments.html#cb477-24" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="st">&quot;Shapely_NT&quot;</span>,</span>
<span id="cb477-25"><a href="sequence-alignments.html#cb477-25" aria-hidden="true" tabindex="-1"></a>            <span class="at">char_width =</span> <span class="fl">0.3</span>)</span>
<span id="cb477-26"><a href="sequence-alignments.html#cb477-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb477-27"><a href="sequence-alignments.html#cb477-27" aria-hidden="true" tabindex="-1"></a>my_aaplot_3 <span class="ot">&lt;-</span> <span class="fu">ggmsa</span>(<span class="at">msa =</span> nt_alignment,</span>
<span id="cb477-28"><a href="sequence-alignments.html#cb477-28" aria-hidden="true" tabindex="-1"></a>                 <span class="at">start =</span> <span class="dv">401</span>,</span>
<span id="cb477-29"><a href="sequence-alignments.html#cb477-29" aria-hidden="true" tabindex="-1"></a>                   <span class="at">end =</span> <span class="dv">600</span>,</span>
<span id="cb477-30"><a href="sequence-alignments.html#cb477-30" aria-hidden="true" tabindex="-1"></a>                  <span class="at">font =</span> <span class="st">&quot;helvetical&quot;</span>,</span>
<span id="cb477-31"><a href="sequence-alignments.html#cb477-31" aria-hidden="true" tabindex="-1"></a>                 <span class="at">color =</span> <span class="st">&quot;Shapely_NT&quot;</span>,</span>
<span id="cb477-32"><a href="sequence-alignments.html#cb477-32" aria-hidden="true" tabindex="-1"></a>            <span class="at">char_width =</span> <span class="fl">0.3</span>)</span>
<span id="cb477-33"><a href="sequence-alignments.html#cb477-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb477-34"><a href="sequence-alignments.html#cb477-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb477-35"><a href="sequence-alignments.html#cb477-35" aria-hidden="true" tabindex="-1"></a><span class="do">## we plot all using cowplot</span></span>
<span id="cb477-36"><a href="sequence-alignments.html#cb477-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb477-37"><a href="sequence-alignments.html#cb477-37" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(my_aaplot_1, </span>
<span id="cb477-38"><a href="sequence-alignments.html#cb477-38" aria-hidden="true" tabindex="-1"></a>          my_aaplot_2, </span>
<span id="cb477-39"><a href="sequence-alignments.html#cb477-39" aria-hidden="true" tabindex="-1"></a>          my_aaplot_3, </span>
<span id="cb477-40"><a href="sequence-alignments.html#cb477-40" aria-hidden="true" tabindex="-1"></a>          <span class="at">ncol =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>These plots are very demanding on computer memory and slow (not optimal). Here is the alignment visualized. The order of the sequences is the same one as is in the fasta input file.</p>
<p><img src="figures/session_11_figs/10_ggmsa_nucleotide_plot.png" width="90%" /></p>
</div>
<div id="other-sequence-alignment-software" class="section level2" number="11.11">
<h2><span class="header-section-number">11.11</span> Other sequence alignment software</h2>
<p><strong>19)</strong> Other widely used programs that can do alignments, but they are less efficient are listed below. We will not cover them here, but most require you to input your sequences as Ameerega_COI_seqs_fasta.txt</p>
<p><strong>a)</strong> <a href="https://www.ebi.ac.uk/Tools/msa/clustalo/" target="_blank"><strong>Clustal Omega</strong></a>. This tool can run online, and you need just to paste your sequences in <code>fasta</code> format.</p>
<p><strong>b)</strong> <a href="https://mafft.cbrc.jp/alignment/server/" target="_blank"><strong>MAFFT</strong></a>. This tool is a good as <strong>DECIPHER</strong>, it is best run on your computer. However, you can run MAFFT online and you need just to paste your sequences in <code>fasta</code> format.</p>
<p><strong>c)</strong> <a href="https://www.ebi.ac.uk/Tools/msa/muscle/" target="_blank"><strong>MUSCLE</strong></a>. This tool can run online, and you need just to paste your sequences in <code>fasta</code> format.</p>
<p><strong>d)</strong> <a href="http://prodata.swmed.edu/promals/promals.php" target="_blank"><strong>PROMALS</strong></a>. This tool can run online, and you need just to paste your sequences in <code>fasta</code> format.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="basic-graphics.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="estimating-phylogenetic-trees.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/11-Sequence_Alignments_v8.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
