<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Session 2 – Using R, Installing Packages and Importing/Exporting Data | Bioinformatics Workshop Gitbook</title>
  <meta name="description" content="none" />
  <meta name="generator" content="bookdown 0.42 and GitBook 2.6.7" />

  <meta property="og:title" content="Session 2 – Using R, Installing Packages and Importing/Exporting Data | Bioinformatics Workshop Gitbook" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="none" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Session 2 – Using R, Installing Packages and Importing/Exporting Data | Bioinformatics Workshop Gitbook" />
  
  <meta name="twitter:description" content="none" />
  

<meta name="author" content="Juan C. Santos &amp; Randy Ortiz" />


<meta name="date" content="2025-01-28" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="getting-started-with-r.html"/>
<link rel="next" href="basic-data-structures-in-r.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioinformatics Workshop Gitbook</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html"><i class="fa fa-check"></i><b>1</b> – Getting Started with R</a>
<ul>
<li class="chapter" data-level="1.1" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#why-do-we-use-r"><i class="fa fa-check"></i><b>1.1</b> Why do we use R?</a></li>
<li class="chapter" data-level="1.2" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#get-a-good-text-editor"><i class="fa fa-check"></i><b>1.2</b> Get a good text editor</a></li>
<li class="chapter" data-level="1.3" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#r-studio"><i class="fa fa-check"></i><b>1.3</b> R studio</a></li>
<li class="chapter" data-level="1.4" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#installing-r"><i class="fa fa-check"></i><b>1.4</b> Installing R</a></li>
<li class="chapter" data-level="1.5" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#annotating-code"><i class="fa fa-check"></i><b>1.5</b> Annotating code</a></li>
<li class="chapter" data-level="1.6" data-path="getting-started-with-r.html"><a href="getting-started-with-r.html#defining-a-working-directory-in-r"><i class="fa fa-check"></i><b>1.6</b> Defining a working directory in R</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html"><i class="fa fa-check"></i><b>2</b> – Using R, Installing Packages and Importing/Exporting Data</a>
<ul>
<li class="chapter" data-level="2.1" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#your-first-session"><i class="fa fa-check"></i><b>2.1</b> Your first session</a></li>
<li class="chapter" data-level="2.2" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#doing-some-base-functions-and-getting-help"><i class="fa fa-check"></i><b>2.2</b> Doing some base functions and getting help</a></li>
<li class="chapter" data-level="2.3" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#functions-between-two-or-more-vectors"><i class="fa fa-check"></i><b>2.3</b> Functions between two or more vectors</a></li>
<li class="chapter" data-level="2.4" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#about-r-packages"><i class="fa fa-check"></i><b>2.4</b> About R-packages</a></li>
<li class="chapter" data-level="2.5" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#cran-task-views-expert-opinion-on-useful-packages"><i class="fa fa-check"></i><b>2.5</b> CRAN Task Views – Expert opinion on useful packages</a></li>
<li class="chapter" data-level="2.6" data-path="using-r-installing-packages-and-importingexporting-data.html"><a href="using-r-installing-packages-and-importingexporting-data.html#installing-packages"><i class="fa fa-check"></i><b>2.6</b> Installing packages</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html"><i class="fa fa-check"></i><b>3</b> – Basic Data Structures in R</a>
<ul>
<li class="chapter" data-level="3.1" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#scalars-and-vectors"><i class="fa fa-check"></i><b>3.1</b> Scalars and vectors</a></li>
<li class="chapter" data-level="3.2" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#matrices"><i class="fa fa-check"></i><b>3.2</b> Matrices</a></li>
<li class="chapter" data-level="3.3" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#arrays"><i class="fa fa-check"></i><b>3.3</b> Arrays</a></li>
<li class="chapter" data-level="3.4" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-frames"><i class="fa fa-check"></i><b>3.4</b> Data Frames</a></li>
<li class="chapter" data-level="3.5" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tables"><i class="fa fa-check"></i><b>3.5</b> Tables</a></li>
<li class="chapter" data-level="3.6" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#lists"><i class="fa fa-check"></i><b>3.6</b> Lists</a></li>
<li class="chapter" data-level="3.7" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#data-tables"><i class="fa fa-check"></i><b>3.7</b> Data Tables</a></li>
<li class="chapter" data-level="3.8" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#tibbles"><i class="fa fa-check"></i><b>3.8</b> Tibbles</a></li>
<li class="chapter" data-level="3.9" data-path="basic-data-structures-in-r.html"><a href="basic-data-structures-in-r.html#dnastringsets-rnastringsets-and-aastringsets"><i class="fa fa-check"></i><b>3.9</b> DNAStringSets, RNAStringSets and AAStringSets</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html"><i class="fa fa-check"></i><b>4</b> – Text Editing and Data Transformations</a>
<ul>
<li class="chapter" data-level="4.1" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#working-with-character-strings"><i class="fa fa-check"></i><b>4.1</b> Working with character strings</a></li>
<li class="chapter" data-level="4.2" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#regular-expressions-regex"><i class="fa fa-check"></i><b>4.2</b> Regular expressions (Regex)</a></li>
<li class="chapter" data-level="4.3" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#comparing-ordering-and-simplifying-vectors"><i class="fa fa-check"></i><b>4.3</b> Comparing, ordering and simplifying vectors</a></li>
<li class="chapter" data-level="4.4" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#useful-functions-of-stringr"><i class="fa fa-check"></i><b>4.4</b> Useful functions of <code>stringr</code></a></li>
<li class="chapter" data-level="4.5" data-path="text-editing-and-data-transformations.html"><a href="text-editing-and-data-transformations.html#data-transformations"><i class="fa fa-check"></i><b>4.5</b> Data transformations</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html"><i class="fa fa-check"></i><b>5</b> – Getting Biological Data from Public Repositories</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#ncbi-databases-genbank-and-others"><i class="fa fa-check"></i><b>5.1</b> NCBI databases: GenBank and others</a></li>
<li class="chapter" data-level="5.2" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#accessing-ncbi-within-r-rentrez"><i class="fa fa-check"></i><b>5.2</b> Accessing NCBI within R: <code>rentrez</code></a></li>
<li class="chapter" data-level="5.3" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#retrieving-pubmed-references"><i class="fa fa-check"></i><b>5.3</b> Retrieving PubMed references</a></li>
<li class="chapter" data-level="5.4" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#comparing-sequences-blast"><i class="fa fa-check"></i><b>5.4</b> Comparing sequences: BLAST</a></li>
<li class="chapter" data-level="5.5" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-uniprot"><i class="fa fa-check"></i><b>5.5</b> Other databases: UniProt</a></li>
<li class="chapter" data-level="5.6" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-ensembl"><i class="fa fa-check"></i><b>5.6</b> Other databases: Ensembl</a></li>
<li class="chapter" data-level="5.7" data-path="getting-biological-data-from-public-repositories.html"><a href="getting-biological-data-from-public-repositories.html#other-databases-taxonomy-phylogenetic-trees-and-species-distributions"><i class="fa fa-check"></i><b>5.7</b> Other databases: Taxonomy, phylogenetic trees and species distributions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html"><i class="fa fa-check"></i><b>6</b> – Basic Statistics in R</a>
<ul>
<li class="chapter" data-level="6.1" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#exploring-your-data"><i class="fa fa-check"></i><b>6.1</b> Exploring your data</a></li>
<li class="chapter" data-level="6.2" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#sub-setting-your-dataset"><i class="fa fa-check"></i><b>6.2</b> Sub-setting your dataset</a></li>
<li class="chapter" data-level="6.3" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#transposing-rotating-your-dataset"><i class="fa fa-check"></i><b>6.3</b> Transposing (rotating) your dataset</a></li>
<li class="chapter" data-level="6.4" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#dealing-with-missing-na-values"><i class="fa fa-check"></i><b>6.4</b> Dealing with missing (NA) values</a></li>
<li class="chapter" data-level="6.5" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#about-p-values"><i class="fa fa-check"></i><b>6.5</b> About p-values</a></li>
<li class="chapter" data-level="6.6" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#how-to-read-p-values"><i class="fa fa-check"></i><b>6.6</b> How to read p-values</a></li>
<li class="chapter" data-level="6.7" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#multiple-testing-corrections"><i class="fa fa-check"></i><b>6.7</b> Multiple testing corrections</a></li>
<li class="chapter" data-level="6.8" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#normalizing-your-data"><i class="fa fa-check"></i><b>6.8</b> Normalizing your data</a></li>
<li class="chapter" data-level="6.9" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#what-test-should-i-use"><i class="fa fa-check"></i><b>6.9</b> What test should I use?</a></li>
<li class="chapter" data-level="6.10" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#type-of-variables-in-dataset"><i class="fa fa-check"></i><b>6.10</b> Type of variables in dataset</a></li>
<li class="chapter" data-level="6.11" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-means-t-test"><i class="fa fa-check"></i><b>6.11</b> Comparing means: t-test</a></li>
<li class="chapter" data-level="6.12" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#two-sample-t-test"><i class="fa fa-check"></i><b>6.12</b> Two sample t-test</a></li>
<li class="chapter" data-level="6.13" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-wilcoxon-signed-rank-test"><i class="fa fa-check"></i><b>6.13</b> Beyond t-test: Wilcoxon signed rank test</a></li>
<li class="chapter" data-level="6.14" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#tabulation-and-cross-tabulation"><i class="fa fa-check"></i><b>6.14</b> Tabulation and Cross-Tabulation</a></li>
<li class="chapter" data-level="6.15" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-binomial-test"><i class="fa fa-check"></i><b>6.15</b> Beyond t-test: Binomial test</a></li>
<li class="chapter" data-level="6.16" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-chi-square-goodness-of-fit"><i class="fa fa-check"></i><b>6.16</b> Beyond t-test: Chi-square goodness of fit</a></li>
<li class="chapter" data-level="6.17" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-mcnemars-test"><i class="fa fa-check"></i><b>6.17</b> Beyond t-test: McNemar’s test</a></li>
<li class="chapter" data-level="6.18" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#beyond-t-test-fishers-exact-test"><i class="fa fa-check"></i><b>6.18</b> Beyond t-test: Fisher’s exact test</a></li>
<li class="chapter" data-level="6.19" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#correlations"><i class="fa fa-check"></i><b>6.19</b> Correlations</a></li>
<li class="chapter" data-level="6.20" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#comparing-distributions"><i class="fa fa-check"></i><b>6.20</b> Comparing distributions</a></li>
<li class="chapter" data-level="6.21" data-path="basic-statistics-in-r.html"><a href="basic-statistics-in-r.html#bootstraps"><i class="fa fa-check"></i><b>6.21</b> Bootstraps</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html"><i class="fa fa-check"></i><b>7</b> – Regressions and ANOVA in R</a>
<ul>
<li class="chapter" data-level="7.1" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#linear-regressions"><i class="fa fa-check"></i><b>7.1</b> Linear regressions</a></li>
<li class="chapter" data-level="7.2" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anatomy-of-a-linear-regression"><i class="fa fa-check"></i><b>7.2</b> Anatomy of a linear regression</a></li>
<li class="chapter" data-level="7.3" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#multiple-linear-regressions"><i class="fa fa-check"></i><b>7.3</b> Multiple linear regressions</a></li>
<li class="chapter" data-level="7.4" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#diagnostic-plots-for-regressions"><i class="fa fa-check"></i><b>7.4</b> Diagnostic plots for regressions</a></li>
<li class="chapter" data-level="7.5" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-influential-measurements"><i class="fa fa-check"></i><b>7.5</b> Regression influential measurements</a></li>
<li class="chapter" data-level="7.6" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-coefficients-and-their-95ci"><i class="fa fa-check"></i><b>7.6</b> Regression coefficients and their 95%CI</a></li>
<li class="chapter" data-level="7.7" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#selecting-predictors"><i class="fa fa-check"></i><b>7.7</b> Selecting predictors</a></li>
<li class="chapter" data-level="7.8" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#regression-predictions"><i class="fa fa-check"></i><b>7.8</b> Regression predictions</a></li>
<li class="chapter" data-level="7.9" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions"><i class="fa fa-check"></i><b>7.9</b> Logistic regressions</a></li>
<li class="chapter" data-level="7.10" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#logistic-regressions-requirements"><i class="fa fa-check"></i><b>7.10</b> Logistic regressions requirements</a></li>
<li class="chapter" data-level="7.11" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#non-linear-regressions"><i class="fa fa-check"></i><b>7.11</b> Non-linear regressions</a></li>
<li class="chapter" data-level="7.12" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-regressions"><i class="fa fa-check"></i><b>7.12</b> Toy datasets for regressions</a></li>
<li class="chapter" data-level="7.13" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova"><i class="fa fa-check"></i><b>7.13</b> ANOVA</a></li>
<li class="chapter" data-level="7.14" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#one-way-anova"><i class="fa fa-check"></i><b>7.14</b> One-way ANOVA</a></li>
<li class="chapter" data-level="7.15" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#two-way-anova"><i class="fa fa-check"></i><b>7.15</b> Two-way ANOVA</a></li>
<li class="chapter" data-level="7.16" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#anova-post-hoc-tests"><i class="fa fa-check"></i><b>7.16</b> ANOVA Post-hoc tests</a></li>
<li class="chapter" data-level="7.17" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#beyond-anova-kruskal-wallis-test"><i class="fa fa-check"></i><b>7.17</b> Beyond ANOVA: Kruskal-Wallis Test</a></li>
<li class="chapter" data-level="7.18" data-path="regressions-and-anova-in-r.html"><a href="regressions-and-anova-in-r.html#toy-datasets-for-anova"><i class="fa fa-check"></i><b>7.18</b> Toy datasets for ANOVA</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html"><i class="fa fa-check"></i><b>8</b> – Multidimensional Data in R</a>
<ul>
<li class="chapter" data-level="8.1" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#principal-component-analysis"><i class="fa fa-check"></i><b>8.1</b> Principal Component Analysis</a></li>
<li class="chapter" data-level="8.2" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#multidimensional-scaling"><i class="fa fa-check"></i><b>8.2</b> Multidimensional scaling</a></li>
<li class="chapter" data-level="8.3" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-dfa"><i class="fa fa-check"></i><b>8.3</b> Cluster analysis: DFA</a></li>
<li class="chapter" data-level="8.4" data-path="multidimensional-data-in-r.html"><a href="multidimensional-data-in-r.html#cluster-analysis-k-means-unsupervised"><i class="fa fa-check"></i><b>8.4</b> Cluster analysis: <em>k</em>-means (unsupervised)</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html"><i class="fa fa-check"></i><b>9</b> – Graphics and Colors</a>
<ul>
<li class="chapter" data-level="9.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#avoid-bad-graphics"><i class="fa fa-check"></i><b>9.1</b> Avoid bad graphics</a></li>
<li class="chapter" data-level="9.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#graphics-in-r-ggplot2"><i class="fa fa-check"></i><b>9.2</b> Graphics in R – ggplot2</a></li>
<li class="chapter" data-level="9.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#preparing-your-dataset"><i class="fa fa-check"></i><b>9.3</b> Preparing your dataset</a></li>
<li class="chapter" data-level="9.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#color-palettes-online"><i class="fa fa-check"></i><b>9.4</b> Color palettes online</a></li>
<li class="chapter" data-level="9.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggplot2-default-palette"><i class="fa fa-check"></i><b>9.5</b> <code>ggplot2</code> default palette</a></li>
<li class="chapter" data-level="9.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#heatmaps-or-tiled-charts"><i class="fa fa-check"></i><b>9.6</b> Heatmaps or tiled charts</a></li>
<li class="chapter" data-level="9.7" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#r-packages-about-color"><i class="fa fa-check"></i><b>9.7</b> R-packages about color</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#viridis-package"><i class="fa fa-check"></i><b>9.7.1</b> <code>viridis</code> package</a></li>
<li class="chapter" data-level="9.7.2" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcolorbrewer-package"><i class="fa fa-check"></i><b>9.7.2</b> <code>RColorBrewer</code> package</a></li>
<li class="chapter" data-level="9.7.3" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#ggsci-package"><i class="fa fa-check"></i><b>9.7.3</b> <code>ggsci</code> package</a></li>
<li class="chapter" data-level="9.7.4" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#wesanderson-package"><i class="fa fa-check"></i><b>9.7.4</b> <code>wesanderson</code> package</a></li>
<li class="chapter" data-level="9.7.5" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#rcartocolor-package"><i class="fa fa-check"></i><b>9.7.5</b> <code>rcartocolor</code> package</a></li>
<li class="chapter" data-level="9.7.6" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#colorspace-package"><i class="fa fa-check"></i><b>9.7.6</b> <code>colorspace</code> package</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="graphics-and-colors.html"><a href="graphics-and-colors.html#making-your-own-palette"><i class="fa fa-check"></i><b>9.8</b> Making your own palette</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="basic-graphics.html"><a href="basic-graphics.html"><i class="fa fa-check"></i><b>10</b> – Basic Graphics</a>
<ul>
<li class="chapter" data-level="10.1" data-path="basic-graphics.html"><a href="basic-graphics.html#what-would-you-like-to-show"><i class="fa fa-check"></i><b>10.1</b> What would you like to show?</a></li>
<li class="chapter" data-level="10.2" data-path="basic-graphics.html"><a href="basic-graphics.html#bar-charts"><i class="fa fa-check"></i><b>10.2</b> Bar charts</a></li>
<li class="chapter" data-level="10.3" data-path="basic-graphics.html"><a href="basic-graphics.html#stacked-bar-charts"><i class="fa fa-check"></i><b>10.3</b> Stacked bar charts</a></li>
<li class="chapter" data-level="10.4" data-path="basic-graphics.html"><a href="basic-graphics.html#histograms"><i class="fa fa-check"></i><b>10.4</b> Histograms</a></li>
<li class="chapter" data-level="10.5" data-path="basic-graphics.html"><a href="basic-graphics.html#multiple-histograms"><i class="fa fa-check"></i><b>10.5</b> Multiple histograms</a></li>
<li class="chapter" data-level="10.6" data-path="basic-graphics.html"><a href="basic-graphics.html#density-plots"><i class="fa fa-check"></i><b>10.6</b> Density plots</a></li>
<li class="chapter" data-level="10.7" data-path="basic-graphics.html"><a href="basic-graphics.html#box-plots"><i class="fa fa-check"></i><b>10.7</b> Box plots</a></li>
<li class="chapter" data-level="10.8" data-path="basic-graphics.html"><a href="basic-graphics.html#violin-plots"><i class="fa fa-check"></i><b>10.8</b> Violin plots</a></li>
<li class="chapter" data-level="10.9" data-path="basic-graphics.html"><a href="basic-graphics.html#scatterplots"><i class="fa fa-check"></i><b>10.9</b> Scatterplots</a></li>
<li class="chapter" data-level="10.10" data-path="basic-graphics.html"><a href="basic-graphics.html#lines-and-scatterplots"><i class="fa fa-check"></i><b>10.10</b> Lines and Scatterplots</a></li>
<li class="chapter" data-level="10.11" data-path="basic-graphics.html"><a href="basic-graphics.html#save-your-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>10.11</b> Save your plots as PDF or PNG</a></li>
<li class="chapter" data-level="10.12" data-path="basic-graphics.html"><a href="basic-graphics.html#ggplot2-add-on-packages-treemapify"><i class="fa fa-check"></i><b>10.12</b> ggplot2 add-on packages: <code>treemapify</code></a></li>
<li class="chapter" data-level="10.13" data-path="basic-graphics.html"><a href="basic-graphics.html#other-resources"><i class="fa fa-check"></i><b>10.13</b> Other resources</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="sequence-alignments.html"><a href="sequence-alignments.html"><i class="fa fa-check"></i><b>11</b> – Sequence Alignments</a>
<ul>
<li class="chapter" data-level="11.1" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-using-rentrez"><i class="fa fa-check"></i><b>11.1</b> Retrieving sequences using ‘rentrez’</a></li>
<li class="chapter" data-level="11.2" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-sequences-manually-from-ncbi"><i class="fa fa-check"></i><b>11.2</b> Retrieving sequences manually from NCBI</a></li>
<li class="chapter" data-level="11.3" data-path="sequence-alignments.html"><a href="sequence-alignments.html#retrieving-accession-numbers"><i class="fa fa-check"></i><b>11.3</b> Retrieving accession numbers</a></li>
<li class="chapter" data-level="11.4" data-path="sequence-alignments.html"><a href="sequence-alignments.html#alignment-with-decipher"><i class="fa fa-check"></i><b>11.4</b> Alignment with <code>DECIPHER</code></a></li>
<li class="chapter" data-level="11.5" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualize-alignment"><i class="fa fa-check"></i><b>11.5</b> Visualize alignment</a></li>
<li class="chapter" data-level="11.6" data-path="sequence-alignments.html"><a href="sequence-alignments.html#save-alignment-to-file"><i class="fa fa-check"></i><b>11.6</b> Save alignment to file</a></li>
<li class="chapter" data-level="11.7" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-and-align-several-markers"><i class="fa fa-check"></i><b>11.7</b> Concatenate and align several markers</a></li>
<li class="chapter" data-level="11.8" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-data.table"><i class="fa fa-check"></i><b>11.8</b> Concatenate with <code>data.table</code></a></li>
<li class="chapter" data-level="11.9" data-path="sequence-alignments.html"><a href="sequence-alignments.html#concatenate-with-biostrings"><i class="fa fa-check"></i><b>11.9</b> Concatenate with <code>Biostrings</code></a></li>
<li class="chapter" data-level="11.10" data-path="sequence-alignments.html"><a href="sequence-alignments.html#visualizing-alignments-in-r"><i class="fa fa-check"></i><b>11.10</b> Visualizing alignments in R</a></li>
<li class="chapter" data-level="11.11" data-path="sequence-alignments.html"><a href="sequence-alignments.html#other-sequence-alignment-software"><i class="fa fa-check"></i><b>11.11</b> Other sequence alignment software</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html"><i class="fa fa-check"></i><b>12</b> – Estimating Phylogenetic Trees</a>
<ul>
<li class="chapter" data-level="12.1" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#phylogeny-inference-with-iqtree2"><i class="fa fa-check"></i><b>12.1</b> Phylogeny inference with <code>iqtree2</code></a></li>
<li class="chapter" data-level="12.2" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-iqtree2-in-macos"><i class="fa fa-check"></i><b>12.2</b> Installing <code>iqtree2</code> in macOS</a></li>
<li class="chapter" data-level="12.3" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#installing-and-running-iqtree2-in-a-pc"><i class="fa fa-check"></i><b>12.3</b> Installing and running <code>iqtree2</code> in a PC</a></li>
<li class="chapter" data-level="12.4" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#loading-the-iqtree_runner_r-function"><i class="fa fa-check"></i><b>12.4</b> Loading the <code>iqtree_runner_R</code> function</a></li>
<li class="chapter" data-level="12.5" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#running-iqtree2-from-r"><i class="fa fa-check"></i><b>12.5</b> Running <code>iqtree2</code> from R</a></li>
<li class="chapter" data-level="12.6" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-no-partitions"><i class="fa fa-check"></i><b>12.6</b> One marker and no partitions</a></li>
<li class="chapter" data-level="12.7" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#one-marker-and-with-partitions"><i class="fa fa-check"></i><b>12.7</b> One marker and with partitions</a></li>
<li class="chapter" data-level="12.8" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#bootstrap-support"><i class="fa fa-check"></i><b>12.8</b> Bootstrap support</a></li>
<li class="chapter" data-level="12.9" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#many-markers-and-with-partitions"><i class="fa fa-check"></i><b>12.9</b> Many markers and with partitions</a></li>
<li class="chapter" data-level="12.10" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#other-sequence-phylogenetic-tree-estimation-software"><i class="fa fa-check"></i><b>12.10</b> Other sequence phylogenetic tree estimation software</a></li>
<li class="chapter" data-level="12.11" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#drawing-phylogenetic-trees-with-ggtree"><i class="fa fa-check"></i><b>12.11</b> Drawing phylogenetic trees with <code>ggtree</code></a></li>
<li class="chapter" data-level="12.12" data-path="estimating-phylogenetic-trees.html"><a href="estimating-phylogenetic-trees.html#save-ggtree-plots-as-pdf-or-png"><i class="fa fa-check"></i><b>12.12</b> Save ggtree plots as PDF or PNG</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html"><i class="fa fa-check"></i><b>13</b> – Mining from a reference dataset</a>
<ul>
<li class="chapter" data-level="13.1" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#a-worked-example-using-the-mtcars-dataset"><i class="fa fa-check"></i><b>13.1</b> A worked example using the <code>mtcars</code> dataset</a></li>
<li class="chapter" data-level="13.2" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-reference-dataset"><i class="fa fa-check"></i><b>13.2</b> Build a reference dataset</a></li>
<li class="chapter" data-level="13.3" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#build-a-sampling-dataset"><i class="fa fa-check"></i><b>13.3</b> Build a sampling dataset</a></li>
<li class="chapter" data-level="13.4" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#extracting-from-reference-dataset"><i class="fa fa-check"></i><b>13.4</b> Extracting from reference dataset</a></li>
<li class="chapter" data-level="13.5" data-path="mining-from-a-reference-dataset.html"><a href="mining-from-a-reference-dataset.html#plots-of-the-extracted-data"><i class="fa fa-check"></i><b>13.5</b> Plots of the extracted data</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="introduction-to-python.html"><a href="introduction-to-python.html"><i class="fa fa-check"></i><b>14</b> – Introduction to Python</a>
<ul>
<li class="chapter" data-level="14.1" data-path="introduction-to-python.html"><a href="introduction-to-python.html#welcome-to-python-why-python"><i class="fa fa-check"></i><b>14.1</b> Welcome to Python! Why Python?</a></li>
<li class="chapter" data-level="14.2" data-path="introduction-to-python.html"><a href="introduction-to-python.html#which-python-should-i-install"><i class="fa fa-check"></i><b>14.2</b> Which Python should I install?</a></li>
<li class="chapter" data-level="14.3" data-path="introduction-to-python.html"><a href="introduction-to-python.html#mac-os-x-python-installations"><i class="fa fa-check"></i><b>14.3</b> Mac OS X Python Installations</a></li>
<li class="chapter" data-level="14.4" data-path="introduction-to-python.html"><a href="introduction-to-python.html#windows-python-installations"><i class="fa fa-check"></i><b>14.4</b> Windows Python Installations</a></li>
<li class="chapter" data-level="14.5" data-path="introduction-to-python.html"><a href="introduction-to-python.html#install-an-ide-or-integrated-development-environment"><i class="fa fa-check"></i><b>14.5</b> Install an IDE, or integrated development environment</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html"><i class="fa fa-check"></i><b>15</b> – Python: Keywords and Identifiers</a>
<ul>
<li class="chapter" data-level="15.1" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#keywords-in-python"><i class="fa fa-check"></i><b>15.1</b> Keywords in Python</a></li>
<li class="chapter" data-level="15.2" data-path="python-keywords-and-identifiers.html"><a href="python-keywords-and-identifiers.html#identifiers-in-python"><i class="fa fa-check"></i><b>15.2</b> Identifiers in Python</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html"><i class="fa fa-check"></i><b>16</b> – Python: Statements and Expressions</a>
<ul>
<li class="chapter" data-level="16.1" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#what-are-statements"><i class="fa fa-check"></i><b>16.1</b> What are statements?</a></li>
<li class="chapter" data-level="16.2" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#expressions-in-python"><i class="fa fa-check"></i><b>16.2</b> Expressions in Python</a></li>
<li class="chapter" data-level="16.3" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#simple-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.3</b> Simple Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.4" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#augmented-assignment-statements-in-python"><i class="fa fa-check"></i><b>16.4</b> Augmented Assignment Statements in Python</a></li>
<li class="chapter" data-level="16.5" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#explicit-and-implicit-line-continuation"><i class="fa fa-check"></i><b>16.5</b> Explicit and Implicit Line Continuation</a></li>
<li class="chapter" data-level="16.6" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#indentation-in-python-is-crucial-to-proper-code-function"><i class="fa fa-check"></i><b>16.6</b> Indentation in Python is Crucial to Proper Code function</a></li>
<li class="chapter" data-level="16.7" data-path="python-statements-and-expressions.html"><a href="python-statements-and-expressions.html#comments"><i class="fa fa-check"></i><b>16.7</b> Comments</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="python-data-types.html"><a href="python-data-types.html"><i class="fa fa-check"></i><b>17</b> – Python: Data Types</a>
<ul>
<li class="chapter" data-level="17.1" data-path="python-data-types.html"><a href="python-data-types.html#booleans"><i class="fa fa-check"></i><b>17.1</b> Booleans</a></li>
<li class="chapter" data-level="17.2" data-path="python-data-types.html"><a href="python-data-types.html#numbers-int-float-and-complex"><i class="fa fa-check"></i><b>17.2</b> Numbers: int, float and complex</a></li>
<li class="chapter" data-level="17.3" data-path="python-data-types.html"><a href="python-data-types.html#strings"><i class="fa fa-check"></i><b>17.3</b> Strings</a></li>
<li class="chapter" data-level="17.4" data-path="python-data-types.html"><a href="python-data-types.html#bytes"><i class="fa fa-check"></i><b>17.4</b> Bytes</a></li>
<li class="chapter" data-level="17.5" data-path="python-data-types.html"><a href="python-data-types.html#lists-1"><i class="fa fa-check"></i><b>17.5</b> Lists</a></li>
<li class="chapter" data-level="17.6" data-path="python-data-types.html"><a href="python-data-types.html#tuples"><i class="fa fa-check"></i><b>17.6</b> Tuples</a></li>
<li class="chapter" data-level="17.7" data-path="python-data-types.html"><a href="python-data-types.html#sets"><i class="fa fa-check"></i><b>17.7</b> Sets</a></li>
<li class="chapter" data-level="17.8" data-path="python-data-types.html"><a href="python-data-types.html#dictionary"><i class="fa fa-check"></i><b>17.8</b> Dictionary</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="python-strings.html"><a href="python-strings.html"><i class="fa fa-check"></i><b>18</b> – Python: Strings</a>
<ul>
<li class="chapter" data-level="18.1" data-path="python-strings.html"><a href="python-strings.html#string-operators"><i class="fa fa-check"></i><b>18.1</b> String Operators</a></li>
<li class="chapter" data-level="18.2" data-path="python-strings.html"><a href="python-strings.html#other-string-modifiers"><i class="fa fa-check"></i><b>18.2</b> Other String Modifiers</a></li>
<li class="chapter" data-level="18.3" data-path="python-strings.html"><a href="python-strings.html#format-characters-in-strings"><i class="fa fa-check"></i><b>18.3</b> Format-Characters in Strings</a></li>
<li class="chapter" data-level="18.4" data-path="python-strings.html"><a href="python-strings.html#comparisons-with-strings"><i class="fa fa-check"></i><b>18.4</b> Comparisons with Strings</a></li>
<li class="chapter" data-level="18.5" data-path="python-strings.html"><a href="python-strings.html#padding-strings"><i class="fa fa-check"></i><b>18.5</b> Padding strings</a></li>
<li class="chapter" data-level="18.6" data-path="python-strings.html"><a href="python-strings.html#search-for-words-in-string-and-output-location"><i class="fa fa-check"></i><b>18.6</b> Search for words in string and output location</a></li>
<li class="chapter" data-level="18.7" data-path="python-strings.html"><a href="python-strings.html#seek-and-replace-in-strings"><i class="fa fa-check"></i><b>18.7</b> Seek and replace in strings</a></li>
</ul></li>
<li class="chapter" data-level="19" data-path="python-operators.html"><a href="python-operators.html"><i class="fa fa-check"></i><b>19</b> – Python: Operators</a>
<ul>
<li class="chapter" data-level="19.1" data-path="python-operators.html"><a href="python-operators.html#many-types-of-operators"><i class="fa fa-check"></i><b>19.1</b> Many types of operators</a></li>
<li class="chapter" data-level="19.2" data-path="python-operators.html"><a href="python-operators.html#arithmetic-operators"><i class="fa fa-check"></i><b>19.2</b> Arithmetic operators</a></li>
<li class="chapter" data-level="19.3" data-path="python-operators.html"><a href="python-operators.html#comparison-operators"><i class="fa fa-check"></i><b>19.3</b> Comparison operators</a></li>
<li class="chapter" data-level="19.4" data-path="python-operators.html"><a href="python-operators.html#logical-operators"><i class="fa fa-check"></i><b>19.4</b> Logical operators</a></li>
<li class="chapter" data-level="19.5" data-path="python-operators.html"><a href="python-operators.html#bitwise-operators"><i class="fa fa-check"></i><b>19.5</b> Bitwise operators</a></li>
<li class="chapter" data-level="19.6" data-path="python-operators.html"><a href="python-operators.html#identity-operators"><i class="fa fa-check"></i><b>19.6</b> Identity Operators</a></li>
<li class="chapter" data-level="19.7" data-path="python-operators.html"><a href="python-operators.html#membership-operators"><i class="fa fa-check"></i><b>19.7</b> Membership operators</a></li>
<li class="chapter" data-level="19.8" data-path="python-operators.html"><a href="python-operators.html#what-is-the-order-of-operations-with-operators"><i class="fa fa-check"></i><b>19.8</b> What is the order of operations with operators?</a></li>
</ul></li>
<li class="chapter" data-level="20" data-path="biopython.html"><a href="biopython.html"><i class="fa fa-check"></i><b>20</b> – Biopython</a>
<ul>
<li class="chapter" data-level="20.1" data-path="biopython.html"><a href="biopython.html#introduction-to-biopython"><i class="fa fa-check"></i><b>20.1</b> Introduction to Biopython</a></li>
<li class="chapter" data-level="20.2" data-path="biopython.html"><a href="biopython.html#biopython-installation"><i class="fa fa-check"></i><b>20.2</b> Biopython Installation</a></li>
<li class="chapter" data-level="20.3" data-path="biopython.html"><a href="biopython.html#getting-started"><i class="fa fa-check"></i><b>20.3</b> Getting Started</a></li>
<li class="chapter" data-level="20.4" data-path="biopython.html"><a href="biopython.html#parsing-my-fasta"><i class="fa fa-check"></i><b>20.4</b> Parsing my FASTA</a></li>
<li class="chapter" data-level="20.5" data-path="biopython.html"><a href="biopython.html#parsing-my-genbank-or-genpept"><i class="fa fa-check"></i><b>20.5</b> Parsing my GenBank or GenPept</a></li>
</ul></li>
<li class="chapter" data-level="21" data-path="biopython-sequences.html"><a href="biopython-sequences.html"><i class="fa fa-check"></i><b>21</b> – Biopython: Sequences</a>
<ul>
<li class="chapter" data-level="21.1" data-path="biopython-sequences.html"><a href="biopython-sequences.html#sequence-objects"><i class="fa fa-check"></i><b>21.1</b> Sequence Objects</a></li>
<li class="chapter" data-level="21.2" data-path="biopython-sequences.html"><a href="biopython-sequences.html#concatenation-with-sequences"><i class="fa fa-check"></i><b>21.2</b> Concatenation with Sequences</a></li>
<li class="chapter" data-level="21.3" data-path="biopython-sequences.html"><a href="biopython-sequences.html#transcription-dna-to-mrna"><i class="fa fa-check"></i><b>21.3</b> Transcription: DNA to mRNA</a></li>
<li class="chapter" data-level="21.4" data-path="biopython-sequences.html"><a href="biopython-sequences.html#translation-rna-to-prot"><i class="fa fa-check"></i><b>21.4</b> Translation: RNA to PROT</a></li>
</ul></li>
<li class="chapter" data-level="22" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html"><i class="fa fa-check"></i><b>22</b> – Biopython: Mulitple Sequence Alignments</a>
<ul>
<li class="chapter" data-level="22.1" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwise2"><i class="fa fa-check"></i><b>22.1</b> pairwise2</a></li>
<li class="chapter" data-level="22.2" data-path="biopython-mulitple-sequence-alignments.html"><a href="biopython-mulitple-sequence-alignments.html#pairwisealigner"><i class="fa fa-check"></i><b>22.2</b> PairwiseAligner</a></li>
</ul></li>
<li class="chapter" data-level="23" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html"><i class="fa fa-check"></i><b>23</b> – Biopython: GenomeDiagram</a>
<ul>
<li class="chapter" data-level="23.1" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#linearcircular-gene-diagrams"><i class="fa fa-check"></i><b>23.1</b> Linear/Circular Gene Diagrams</a></li>
<li class="chapter" data-level="23.2" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#diagram-readability"><i class="fa fa-check"></i><b>23.2</b> Diagram Readability</a></li>
<li class="chapter" data-level="23.3" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#box-shape"><i class="fa fa-check"></i><b>23.3</b> Box Shape</a></li>
<li class="chapter" data-level="23.4" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#edit-box-shape-arrows"><i class="fa fa-check"></i><b>23.4</b> Edit Box Shape: Arrows</a></li>
<li class="chapter" data-level="23.5" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#restriction-map"><i class="fa fa-check"></i><b>23.5</b> Restriction Map</a></li>
<li class="chapter" data-level="23.6" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#tracks-on-tracks"><i class="fa fa-check"></i><b>23.6</b> Tracks on Tracks</a></li>
<li class="chapter" data-level="23.7" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#cross-link-data"><i class="fa fa-check"></i><b>23.7</b> Cross-link Data</a></li>
<li class="chapter" data-level="23.8" data-path="biopython-genomediagram.html"><a href="biopython-genomediagram.html#clean-cross-link"><i class="fa fa-check"></i><b>23.8</b> Clean Cross-Link</a></li>
</ul></li>
<li class="chapter" data-level="24" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html"><i class="fa fa-check"></i><b>24</b> – Biopython: Chromosome Modeling</a>
<ul>
<li class="chapter" data-level="24.1" data-path="biopython-chromosome-modeling.html"><a href="biopython-chromosome-modeling.html#gene-feature-modeling"><i class="fa fa-check"></i><b>24.1</b> Gene Feature Modeling</a></li>
</ul></li>
<li class="chapter" data-level="25" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html"><i class="fa fa-check"></i><b>25</b> – Biopython: Sequence Parsing with Plots</a>
<ul>
<li class="chapter" data-level="25.1" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#histograms-1"><i class="fa fa-check"></i><b>25.1</b> Histograms</a></li>
<li class="chapter" data-level="25.2" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#gc-content"><i class="fa fa-check"></i><b>25.2</b> GC Content</a></li>
<li class="chapter" data-level="25.3" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#nt-alignment-dot-plot"><i class="fa fa-check"></i><b>25.3</b> Nt Alignment Dot Plot</a></li>
<li class="chapter" data-level="25.4" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#bigger-dots-for-the-dot-plot"><i class="fa fa-check"></i><b>25.4</b> Bigger Dots for the Dot Plot</a></li>
<li class="chapter" data-level="25.5" data-path="biopython-sequence-parsing-with-plots.html"><a href="biopython-sequence-parsing-with-plots.html#fastq-we-did-not-forget-you"><i class="fa fa-check"></i><b>25.5</b> FASTQ we did not forget you</a></li>
</ul></li>
<li class="chapter" data-level="26" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html"><i class="fa fa-check"></i><b>26</b> – Python: Web Scraping w/ BeautifulSoup</a>
<ul>
<li class="chapter" data-level="26.1" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#introduction"><i class="fa fa-check"></i><b>26.1</b> Introduction</a></li>
<li class="chapter" data-level="26.2" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#installation"><i class="fa fa-check"></i><b>26.2</b> Installation</a></li>
<li class="chapter" data-level="26.3" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#getting-started-1"><i class="fa fa-check"></i><b>26.3</b> Getting started</a></li>
<li class="chapter" data-level="26.4" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-all-matching-tags"><i class="fa fa-check"></i><b>26.4</b> Mining all matching tags</a></li>
<li class="chapter" data-level="26.5" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#mining-tags-by-class-or-id"><i class="fa fa-check"></i><b>26.5</b> Mining tags by class or id</a></li>
<li class="chapter" data-level="26.6" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#use-css-selectors-for-positional-mining"><i class="fa fa-check"></i><b>26.6</b> Use CSS Selectors for positional mining</a></li>
<li class="chapter" data-level="26.7" data-path="python-web-scraping-w-beautifulsoup.html"><a href="python-web-scraping-w-beautifulsoup.html#weather-page-example"><i class="fa fa-check"></i><b>26.7</b> Weather Page Example</a></li>
</ul></li>
<li class="chapter" data-level="27" data-path="python-subprocess.html"><a href="python-subprocess.html"><i class="fa fa-check"></i><b>27</b> – Python Subprocess</a>
<ul>
<li class="chapter" data-level="27.1" data-path="python-subprocess.html"><a href="python-subprocess.html#introduction-and-installation"><i class="fa fa-check"></i><b>27.1</b> Introduction and Installation</a></li>
<li class="chapter" data-level="27.2" data-path="python-subprocess.html"><a href="python-subprocess.html#creating-your-first-python-script-w-bbtools"><i class="fa fa-check"></i><b>27.2</b> Creating your first python script w/ BBTools</a></li>
</ul></li>
<li class="chapter" data-level="28" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html"><i class="fa fa-check"></i><b>28</b> – Control Structures &amp; Loops in R</a>
<ul>
<li class="chapter" data-level="28.1" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#controling-the-flow"><i class="fa fa-check"></i><b>28.1</b> Controling the flow</a></li>
<li class="chapter" data-level="28.2" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if"><i class="fa fa-check"></i><b>28.2</b> Conditional <code>if</code></a></li>
<li class="chapter" data-level="28.3" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-if-and-else"><i class="fa fa-check"></i><b>28.3</b> Conditional <code>if</code> and <code>else</code></a></li>
<li class="chapter" data-level="28.4" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#nested-conditionals-if-else-and-if-else"><i class="fa fa-check"></i><b>28.4</b> Nested conditionals <code>if</code>, <code>else</code>, and <code>if else</code></a></li>
<li class="chapter" data-level="28.5" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#conditional-ifelse"><i class="fa fa-check"></i><b>28.5</b> Conditional <code>ifelse</code></a></li>
<li class="chapter" data-level="28.6" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#loops-that-control-iterations-and-repeations"><i class="fa fa-check"></i><b>28.6</b> Loops that control iterations and repeations</a></li>
<li class="chapter" data-level="28.7" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-for-loop"><i class="fa fa-check"></i><b>28.7</b> The <code>for</code> Loop</a></li>
<li class="chapter" data-level="28.8" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-while-loop"><i class="fa fa-check"></i><b>28.8</b> The <code>while</code> Loop</a></li>
<li class="chapter" data-level="28.9" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-repeat-loop"><i class="fa fa-check"></i><b>28.9</b> The <code>repeat</code> Loop</a></li>
<li class="chapter" data-level="28.10" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#the-next-and-break-statements"><i class="fa fa-check"></i><b>28.10</b> The <code>next</code> and <code>break</code> statements</a></li>
<li class="chapter" data-level="28.11" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functions"><i class="fa fa-check"></i><b>28.11</b> Other loop functions</a></li>
<li class="chapter" data-level="28.12" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#lapply-functions"><i class="fa fa-check"></i><b>28.12</b> <code>lapply</code> functions</a></li>
<li class="chapter" data-level="28.13" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsapply"><i class="fa fa-check"></i><b>28.13</b> Other loop functions:<code>apply</code></a></li>
<li class="chapter" data-level="28.14" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionsmapply"><i class="fa fa-check"></i><b>28.14</b> Other loop functions:<code>mapply</code></a></li>
<li class="chapter" data-level="28.15" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionstapply"><i class="fa fa-check"></i><b>28.15</b> Other loop functions:<code>tapply</code></a></li>
<li class="chapter" data-level="28.16" data-path="control-structures-loops-in-r.html"><a href="control-structures-loops-in-r.html#other-loop-functionssplit"><i class="fa fa-check"></i><b>28.16</b> Other loop functions:<code>split</code></a></li>
</ul></li>
<li class="chapter" data-level="29" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html"><i class="fa fa-check"></i><b>29</b> – Bioinformatics Literature</a>
<ul>
<li class="chapter" data-level="29.1" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#peer-reviewed-bioinformatics-journals"><i class="fa fa-check"></i><b>29.1</b> Peer-reviewed bioinformatics journals</a></li>
<li class="chapter" data-level="29.2" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#access-to-journals-articles"><i class="fa fa-check"></i><b>29.2</b> Access to journals articles</a></li>
<li class="chapter" data-level="29.3" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#informal-access-to-code-preprints-and-vignettes"><i class="fa fa-check"></i><b>29.3</b> Informal access to code: Preprints and vignettes</a></li>
<li class="chapter" data-level="29.4" data-path="bioinformatics-literature.html"><a href="bioinformatics-literature.html#online-bioinformatics-courses"><i class="fa fa-check"></i><b>29.4</b> Online bioinformatics courses</a></li>
</ul></li>
<li class="chapter" data-level="30" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html"><i class="fa fa-check"></i><b>30</b> – Importing/Exporting Data into R</a>
<ul>
<li class="chapter" data-level="30.1" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#typing-directly"><i class="fa fa-check"></i><b>30.1</b> Typing directly</a></li>
<li class="chapter" data-level="30.2" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#summary-versus-raw-data"><i class="fa fa-check"></i><b>30.2</b> Summary versus raw data</a></li>
<li class="chapter" data-level="30.3" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#downloading-data-files-github"><i class="fa fa-check"></i><b>30.3</b> Downloading data files (GitHub)</a></li>
<li class="chapter" data-level="30.4" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#importing-data-from-a-file"><i class="fa fa-check"></i><b>30.4</b> Importing data from a file</a></li>
<li class="chapter" data-level="30.5" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-to-a-file"><i class="fa fa-check"></i><b>30.5</b> Saving data to a file</a></li>
<li class="chapter" data-level="30.6" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#saving-data-as-rdata-.rdata-.rda"><i class="fa fa-check"></i><b>30.6</b> Saving data as Rdata (.rdata, .rda)</a></li>
<li class="chapter" data-level="30.7" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#import-excel-formated-data"><i class="fa fa-check"></i><b>30.7</b> Import Excel formated data</a></li>
<li class="chapter" data-level="30.8" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#save-data-as-excel-formated-file"><i class="fa fa-check"></i><b>30.8</b> Save data as Excel formated file</a></li>
<li class="chapter" data-level="30.9" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#fixing-problems-while-importing-datasets"><i class="fa fa-check"></i><b>30.9</b> Fixing problems while importing datasets</a></li>
<li class="chapter" data-level="30.10" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#simplifying-embellished-excel-datasets"><i class="fa fa-check"></i><b>30.10</b> Simplifying embellished excel datasets</a></li>
<li class="chapter" data-level="30.11" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#accessing-online-data-dryad"><i class="fa fa-check"></i><b>30.11</b> Accessing online data: DRYAD</a></li>
<li class="chapter" data-level="30.12" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#accessing-online-data-gene-expression-omnibus-ncbi-geo"><i class="fa fa-check"></i><b>30.12</b> Accessing online data: Gene Expression Omnibus (NCBI GEO)</a></li>
<li class="chapter" data-level="30.13" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#accessing-online-data-clinicaltrials.gov"><i class="fa fa-check"></i><b>30.13</b> Accessing online data: clinicaltrials.gov</a></li>
<li class="chapter" data-level="30.14" data-path="importingexporting-data-into-r.html"><a href="importingexporting-data-into-r.html#accessing-online-data-data-scrapping-from-a-website"><i class="fa fa-check"></i><b>30.14</b> Accessing online data: Data scrapping from a website</a></li>
</ul></li>
<li class="chapter" data-level="31" data-path="working-with-proteins.html"><a href="working-with-proteins.html"><i class="fa fa-check"></i><b>31</b> – Working with Proteins</a>
<ul>
<li class="chapter" data-level="31.1" data-path="working-with-proteins.html"><a href="working-with-proteins.html#sequences-in-fasta-format"><i class="fa fa-check"></i><b>31.1</b> Sequences in FASTA format</a></li>
<li class="chapter" data-level="31.2" data-path="working-with-proteins.html"><a href="working-with-proteins.html#fetching-aa-sequences-from-ncbi"><i class="fa fa-check"></i><b>31.2</b> Fetching AA sequences from NCBI</a></li>
<li class="chapter" data-level="31.3" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-aa-sequences-in-r"><i class="fa fa-check"></i><b>31.3</b> Importing AA sequences in R</a></li>
<li class="chapter" data-level="31.4" data-path="working-with-proteins.html"><a href="working-with-proteins.html#importing-nucleotide-sequences-in-r"><i class="fa fa-check"></i><b>31.4</b> Importing nucleotide sequences in R</a></li>
<li class="chapter" data-level="31.5" data-path="working-with-proteins.html"><a href="working-with-proteins.html#translating-a-nucleotides-to-aa"><i class="fa fa-check"></i><b>31.5</b> Translating a nucleotides to AA</a></li>
<li class="chapter" data-level="31.6" data-path="working-with-proteins.html"><a href="working-with-proteins.html#finding-orfs-in-a-nucleotide-sequence"><i class="fa fa-check"></i><b>31.6</b> Finding ORFs in a nucleotide sequence</a></li>
<li class="chapter" data-level="31.7" data-path="working-with-proteins.html"><a href="working-with-proteins.html#physochemical-properties-of-an-amino-acid-sequence"><i class="fa fa-check"></i><b>31.7</b> Physochemical properties of an amino acid sequence</a></li>
</ul></li>
<li class="chapter" data-level="32" data-path="miscellaneous.html"><a href="miscellaneous.html"><i class="fa fa-check"></i><b>32</b> – Miscellaneous</a>
<ul>
<li class="chapter" data-level="32.1" data-path="miscellaneous.html"><a href="miscellaneous.html#sampling-from-a-vector-of-names"><i class="fa fa-check"></i><b>32.1</b> Sampling from a vector of names</a></li>
<li class="chapter" data-level="32.2" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-an-article-pdf"><i class="fa fa-check"></i><b>32.2</b> Read a text from an article pdf</a></li>
<li class="chapter" data-level="32.3" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-website"><i class="fa fa-check"></i><b>32.3</b> Read a text from a website</a></li>
<li class="chapter" data-level="32.4" data-path="miscellaneous.html"><a href="miscellaneous.html#read-a-text-from-a-file"><i class="fa fa-check"></i><b>32.4</b> Read a text from a file</a></li>
<li class="chapter" data-level="32.5" data-path="miscellaneous.html"><a href="miscellaneous.html#generate-the-word-cloud"><i class="fa fa-check"></i><b>32.5</b> Generate the word cloud</a></li>
<li class="chapter" data-level="32.6" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-pdf"><i class="fa fa-check"></i><b>32.6</b> worldcloud – PDF</a></li>
<li class="chapter" data-level="32.7" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-website"><i class="fa fa-check"></i><b>32.7</b> worldcloud – website</a></li>
<li class="chapter" data-level="32.8" data-path="miscellaneous.html"><a href="miscellaneous.html#worldcloud-text"><i class="fa fa-check"></i><b>32.8</b> worldcloud – text</a></li>
<li class="chapter" data-level="32.9" data-path="miscellaneous.html"><a href="miscellaneous.html#miscellaneous-1"><i class="fa fa-check"></i><b>32.9</b> Miscellaneous</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Bioinformatics Workshop Gitbook</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="using-r-installing-packages-and-importingexporting-data" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Session 2</span> – Using R, Installing Packages and Importing/Exporting Data<a href="using-r-installing-packages-and-importingexporting-data.html#using-r-installing-packages-and-importingexporting-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="your-first-session" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Your first session<a href="using-r-installing-packages-and-importingexporting-data.html#your-first-session" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You will start by entering commands in the console.</p>
<p><strong>1)</strong> After clicking on the R icon you should see the R console and the computer will be ready to receive commands</p>
<p><img src="figures/session_2_figs/1_macOS_R_console.png" width="100%" /></p>
<p>This is the R console in a macOS computer.</p>
<p><strong>2)</strong> The blinking command will be next to R prompts which are indicated by <code>&gt;</code>. This basically indicates where you can start typing commands like a excel or a calculator. Let’s try a simple addition:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span> <span class="sc">+</span> <span class="dv">2</span></span>
<span id="cb15-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 3</span></span></code></pre></div>
<p>You can try more complicated calculations like the natural logarithm of 10:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">log</span>(<span class="dv">10</span>)</span>
<span id="cb16-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 2.302585</span></span></code></pre></div>
<p>The R console gives you a [1] that indicates that the result is a vector of one element or scalar (i.e., 3 or 2.302585). If more that 1, this is called a vector and R will number elements sequentially in a vector. For example, you can ask for an expression for a sequence of number form 1 to 100 by addition of 1 as <code>seq(1:100)</code>. This vector will have 100 elements.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">seq</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>)</span>
<span id="cb17-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="co">#[1]   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19</span></span>
<span id="cb17-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="co">#[20]  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38</span></span>
<span id="cb17-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co">#[39]  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57</span></span>
<span id="cb17-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="co">#[58]  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76</span></span>
<span id="cb17-6"><a href="using-r-installing-packages-and-importingexporting-data.html#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="co">#[77]  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95</span></span>
<span id="cb17-7"><a href="using-r-installing-packages-and-importingexporting-data.html#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co">#[96]  96  97  98  99 100</span></span></code></pre></div>
<p><strong>3)</strong> R provides easy command-line editing tools that will help you to recall previous commands. This help you change parameters in the expression or correct typos in it. For example,</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Same result as before </span></span>
<span id="cb18-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">seq</span>(<span class="at">from =</span> <span class="dv">1</span>, <span class="at">to =</span> <span class="dv">100</span>, <span class="at">by =</span><span class="dv">1</span>)</span>
<span id="cb18-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co"># [1]   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19</span></span>
<span id="cb18-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="co"># [20]  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38</span></span>
<span id="cb18-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb18-5" aria-hidden="true" tabindex="-1"></a><span class="co"># [39]  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57</span></span>
<span id="cb18-6"><a href="using-r-installing-packages-and-importingexporting-data.html#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="co"># [58]  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76</span></span>
<span id="cb18-7"><a href="using-r-installing-packages-and-importingexporting-data.html#cb18-7" aria-hidden="true" tabindex="-1"></a><span class="co"># [77]  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95</span></span>
<span id="cb18-8"><a href="using-r-installing-packages-and-importingexporting-data.html#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="co"># [96]  96  97  98  99 100</span></span></code></pre></div>
<p>You can press <strong>↑ (arrow up) </strong> or <strong>Ctrl-P</strong> to recall previous command and we can change the argument <code>by =</code> to get sequence by addition of 5 instead of 1.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb19-1" aria-hidden="true" tabindex="-1"></a> <span class="fu">seq</span>(<span class="at">from =</span> <span class="dv">1</span>, <span class="at">to =</span> <span class="dv">100</span>, <span class="at">by =</span><span class="dv">5</span>)</span>
<span id="cb19-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb19-2" aria-hidden="true" tabindex="-1"></a> <span class="co">#[1]  1  6 11 16 21 26 31 36 41 46 51 56 61 66 71 76 81 86 91 96</span></span></code></pre></div>
<p>Typing errors are easy to introduce and R will let you know about then. For example, I forgot to add the <code>=</code> after <code>to 100</code>. This is not valid, and an error is produced.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">seq</span>(<span class="at">from =</span> <span class="dv">1</span>, to  <span class="dv">100</span>, <span class="at">by =</span><span class="dv">5</span>)</span>
<span id="cb20-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Error: unexpected numeric constant in &quot;seq(from = 1, to  100&quot;</span></span></code></pre></div>
<p>Now, you can press arrow up to fix the error in the previous command and it can be fixed.</p>
<p><strong>4)</strong> Here are some basic functions with numeric vectors. Let’s get <strong>create a vector</strong> of numbers named <code>my_vector</code> by assigning seven random numbers to it using the arrow <strong><code>&lt;-</code></strong> known as the assign operator (the <code>=</code> also works in similar fashion) and the function <code>c()</code>.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb21-1" aria-hidden="true" tabindex="-1"></a>my_vector <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">4</span>,<span class="dv">1</span>,<span class="dv">12</span>,<span class="dv">30</span>,<span class="dv">5</span>,<span class="dv">6</span>)</span></code></pre></div>
<p>The <code>c()</code> is a base function to combine values into a vector or list.</p>
<p><strong>5)</strong> You can read or retrieve the contents of <code>my_vector</code> by typing its name.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb22-1" aria-hidden="true" tabindex="-1"></a>my_vector</span>
<span id="cb22-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="co">#[1]  2  4  1 12 30  5  6 </span></span></code></pre></div>
<p><strong>6)</strong> You can also get the structure of <code>my_vector</code> by using <code>str()</code>. This function provides a compactly display the structure of an arbitrary R object (e.g., a vector). This provides a diagnostic description and summary what is contains.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(my_vector)</span>
<span id="cb23-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="co">#num [1:7] 2 4 1 12 30 5 6 </span></span></code></pre></div>
<p>This indicates that the structure of <code>my_vector</code> is a numeric vector with 7 elements that are numbers.</p>
</div>
<div id="doing-some-base-functions-and-getting-help" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Doing some base functions and getting help<a href="using-r-installing-packages-and-importingexporting-data.html#doing-some-base-functions-and-getting-help" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>1)</strong> Let’s create a new <code>my_vector</code>. This will overwrite the previous one.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb24-1" aria-hidden="true" tabindex="-1"></a>my_vector <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="at">n =</span> <span class="dv">1000</span>, <span class="at">mean =</span> <span class="dv">50</span>, <span class="at">sd =</span> <span class="dv">25</span>)</span>
<span id="cb24-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="co"># I omitted the result because it has 1000 numbers</span></span>
<span id="cb24-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb24-3" aria-hidden="true" tabindex="-1"></a>my_vector</span></code></pre></div>
<p>The function <code>rnorm()</code> will draw n random numbers (in this case 1000) of a normal distribution with mean 50 and standard deviation of 25.</p>
<p><strong>2)</strong> We can get the structure of <code>my_vector</code>.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(my_vector)</span>
<span id="cb25-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="co">#num [1:1000] 50.3 60.3 20.4 44.1 43.8 ...</span></span></code></pre></div>
<p>The structure of the <code>my_vector</code> object indicates that is a numeric vector of 1000 numbers.</p>
<p><strong>3)</strong> We can confirm that the mean is 50 by using the function <code>mean()</code> and its standard deviation of 25 by using the function <code>sd()</code>.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(my_vector)</span>
<span id="cb26-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 50.29036</span></span>
<span id="cb26-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span>(my_vector)</span>
<span id="cb26-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 25.05207</span></span></code></pre></div>
<p>As you can see in this example, the mean and standard deviation are not exactly 50 and 25, respectively. However, it is very close as these parameters inputted in the <code>rnorm()</code> function.</p>
<p><strong>4)</strong> We can get some information about each function used with the following using function <code>?</code> and <code>help()</code>. You put the function that you want to get information after <code>?</code> or within the parenthesis of <code>help()</code>.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get help about function &#39;mean&#39;</span></span>
<span id="cb27-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb27-2" aria-hidden="true" tabindex="-1"></a>?mean</span>
<span id="cb27-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb27-3" aria-hidden="true" tabindex="-1"></a><span class="co"># another way to ask help</span></span>
<span id="cb27-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="fu">help</span>(mean)</span></code></pre></div>
<p>The functions <code>?</code> and <code>help()</code> provide a link into a new window with the corresponding manual of that function and how it is use and (in most cases) examples.</p>
<p>This a transcript of the popup window for the help of <code>mean</code>. The name within {} is the library or package that it is from in this case base.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-1" aria-hidden="true" tabindex="-1"></a>mean {base}</span>
<span id="cb28-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-3" aria-hidden="true" tabindex="-1"></a>The long name of the <span class="fu">mean</span>() <span class="cf">function</span>, <span class="cf">in</span> this case<span class="sc">:</span> Arithmetic Mean. </span>
<span id="cb28-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-5" aria-hidden="true" tabindex="-1"></a>A description of the <span class="fu">mean</span>() <span class="cf">function</span>, <span class="cf">in</span> this case<span class="sc">:</span> <span class="st">&quot;Generic function for the (trimmed) arithmetic mean&quot;</span></span>
<span id="cb28-6"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-7"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-7" aria-hidden="true" tabindex="-1"></a>A usage example<span class="sc">:</span></span>
<span id="cb28-8"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-9"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-9" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(x, ...)</span>
<span id="cb28-10"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-11"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-11" aria-hidden="true" tabindex="-1"></a><span class="do">## Default S3 method:</span></span>
<span id="cb28-12"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-12" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(x, <span class="at">trim =</span> <span class="dv">0</span>, <span class="at">na.rm =</span> <span class="cn">FALSE</span>, ...)</span>
<span id="cb28-13"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-14"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-14" aria-hidden="true" tabindex="-1"></a>An explanation of its arguments<span class="sc">:</span></span>
<span id="cb28-15"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-16"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-16" aria-hidden="true" tabindex="-1"></a>x <span class="sc">--</span> An R object. Currently there are methods <span class="cf">for</span> numeric<span class="sc">/</span>logical vectors and date, date<span class="sc">-</span>time and time </span>
<span id="cb28-17"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-17" aria-hidden="true" tabindex="-1"></a>     interval objects. Complex vectors are allowed <span class="cf">for</span> trim <span class="ot">=</span> <span class="dv">0</span>, only.</span>
<span id="cb28-18"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-19"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-19" aria-hidden="true" tabindex="-1"></a>trim <span class="sc">--</span> the <span class="fu">fraction</span> (<span class="dv">0</span> to <span class="fl">0.5</span>) of observations to be trimmed from each end of x before the mean is computed. </span>
<span id="cb28-20"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-20" aria-hidden="true" tabindex="-1"></a>        Values of trim outside that range are taken as the nearest endpoint.</span>
<span id="cb28-21"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-22"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-22" aria-hidden="true" tabindex="-1"></a>na.rm <span class="sc">--</span> a logical value indicating whether <span class="cn">NA</span> values should be stripped before the computation </span>
<span id="cb28-23"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-23" aria-hidden="true" tabindex="-1"></a>         proceeds.</span>
<span id="cb28-24"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-25"><a href="using-r-installing-packages-and-importingexporting-data.html#cb28-25" aria-hidden="true" tabindex="-1"></a>... <span class="sc">--</span> further arguments passed to or from other methods.</span></code></pre></div>
<p>Further information about the function <code>mean()</code> are provided in the same pop up window as value, references and related functions. Finally, examples of how to use <code>mean()</code> are also provided.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb29-1" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span><span class="sc">:</span><span class="dv">10</span>, <span class="dv">50</span>)</span>
<span id="cb29-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb29-2" aria-hidden="true" tabindex="-1"></a>xm <span class="ot">&lt;-</span> <span class="fu">mean</span>(x)</span>
<span id="cb29-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(xm, <span class="fu">mean</span>(x, <span class="at">trim =</span> <span class="fl">0.10</span>))</span></code></pre></div>
<p><strong>5)</strong> Some basic functions for numeric vectors.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Arithmetic Mean</span></span>
<span id="cb30-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(my_vector) </span>
<span id="cb30-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Standard Deviation</span></span>
<span id="cb30-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="fu">sd</span> (my_vector) </span>
<span id="cb30-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Variance</span></span>
<span id="cb30-6"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-6" aria-hidden="true" tabindex="-1"></a><span class="fu">var</span> (my_vector) </span>
<span id="cb30-7"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Median Value</span></span>
<span id="cb30-8"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-8" aria-hidden="true" tabindex="-1"></a><span class="fu">median</span> (my_vector) </span>
<span id="cb30-9"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Maximum Value</span></span>
<span id="cb30-10"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-10" aria-hidden="true" tabindex="-1"></a><span class="fu">max</span> (my_vector)</span>
<span id="cb30-11"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Minimum Value </span></span>
<span id="cb30-12"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-12" aria-hidden="true" tabindex="-1"></a><span class="fu">min</span> (my_vector)</span>
<span id="cb30-13"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Sums values in vector</span></span>
<span id="cb30-14"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-14" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(my_vector)</span>
<span id="cb30-15"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Provides the number of elements in the vector </span></span>
<span id="cb30-16"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-16" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(my_vector) </span>
<span id="cb30-17"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Round numbers to of elements in the vector to number of digits (in this case to two decimal points)</span></span>
<span id="cb30-18"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-18" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fl">3.1415</span>, <span class="at">digits =</span> <span class="dv">2</span>)</span>
<span id="cb30-19"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Absolute value</span></span>
<span id="cb30-20"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-20" aria-hidden="true" tabindex="-1"></a><span class="fu">abs</span>(my_vector)                              </span>
<span id="cb30-21"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Square root</span></span>
<span id="cb30-22"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-22" aria-hidden="true" tabindex="-1"></a><span class="fu">sqrt</span>(my_vector)</span>
<span id="cb30-23"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Natural logarithm                          </span></span>
<span id="cb30-24"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-24" aria-hidden="true" tabindex="-1"></a><span class="fu">log</span>(my_vector)</span>
<span id="cb30-25"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-25" aria-hidden="true" tabindex="-1"></a><span class="co"># Common (base 10) logarithm                              </span></span>
<span id="cb30-26"><a href="using-r-installing-packages-and-importingexporting-data.html#cb30-26" aria-hidden="true" tabindex="-1"></a><span class="fu">log10</span>(my_vector)</span></code></pre></div>
<p><strong>6)</strong> You can print what is in a vector (or object) or return the result of calculation, you just can type its name on the console and get what is contained on it. You can also use the function <code>print()</code>.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span><span class="sc">:</span><span class="dv">10</span></span>
<span id="cb31-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="co">#[1]  1  2  3  4  5  6  7  8  9 10</span></span>
<span id="cb31-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb31-3" aria-hidden="true" tabindex="-1"></a>my_numbers <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">10</span></span>
<span id="cb31-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb31-4" aria-hidden="true" tabindex="-1"></a>my_numbers</span>
<span id="cb31-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="co">#[1]  1  2  3  4  5  6  7  8  9 10</span></span>
<span id="cb31-6"><a href="using-r-installing-packages-and-importingexporting-data.html#cb31-6" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(my_numbers)</span>
<span id="cb31-7"><a href="using-r-installing-packages-and-importingexporting-data.html#cb31-7" aria-hidden="true" tabindex="-1"></a><span class="co">#[1]  1  2  3  4  5  6  7  8  9 10</span></span></code></pre></div>
<p><strong>7)</strong> We can construct a character vector <code>my_names</code> with text (i.e., string vector) that contains letter or numbers that are treated as text (i.e., not quantities).</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb32-1" aria-hidden="true" tabindex="-1"></a>my_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;juan&quot;</span>, <span class="st">&quot;c&quot;</span>, <span class="st">&quot;santos&quot;</span>, <span class="dv">123</span>)</span>
<span id="cb32-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb32-2" aria-hidden="true" tabindex="-1"></a>my_names</span>
<span id="cb32-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb32-3" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;juan&quot;   &quot;c&quot;      &quot;santos&quot; &quot;123&quot;</span></span>
<span id="cb32-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb32-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(my_names)</span>
<span id="cb32-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb32-5" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;juan&quot;   &quot;c&quot;      &quot;santos&quot; &quot;123&quot;</span></span></code></pre></div>
<p>You can revise its structure with <code>str()</code>. You will notice that the numbers are within quotations.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(my_names)</span>
<span id="cb33-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="co">#chr [1:4] &quot;juan&quot; &quot;c&quot; &quot;santos&quot; &quot;123&quot;</span></span></code></pre></div>
<p><strong>8)</strong> Some more complex R data structures like matrices, lists or data.frames can also be printed and visualized using <code>print()</code>.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb34-1" aria-hidden="true" tabindex="-1"></a>my_matrix <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>), <span class="dv">3</span>,<span class="dv">2</span>)</span>
<span id="cb34-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(my_matrix)</span>
<span id="cb34-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb34-3" aria-hidden="true" tabindex="-1"></a><span class="co">#     [,1] [,2]</span></span>
<span id="cb34-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="co">#[1,]    1    4</span></span>
<span id="cb34-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="co">#[2,]    2    5</span></span>
<span id="cb34-6"><a href="using-r-installing-packages-and-importingexporting-data.html#cb34-6" aria-hidden="true" tabindex="-1"></a><span class="co">#[3,]    3    6</span></span></code></pre></div>
<p>In this case, we are printing the object <code>my_matrix</code> that has the numbers in the sequence from 1 to 6 in 3 rows and 2 columns.</p>
<p><strong>9)</strong> However, <code>print()</code> can only return one object at a time.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(my_numbers, my_names, my_matrix)</span>
<span id="cb35-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Error in print.default(my_numbers, my_names, my_matrix) : </span></span>
<span id="cb35-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  invalid &#39;digits&#39; argument</span></span>
<span id="cb35-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="co">#In addition: Warning message:</span></span>
<span id="cb35-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb35-5" aria-hidden="true" tabindex="-1"></a><span class="co">#In print.default(my_numbers, my_names, my_matrix) :</span></span>
<span id="cb35-6"><a href="using-r-installing-packages-and-importingexporting-data.html#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  NAs introduced by coercion</span></span></code></pre></div>
<p>In this case cannot print the three objects (i.e., <code>my_numbers</code>, <code>my_names</code>, <code>my_matrix</code>) at the same time.</p>
<p><strong>10)</strong> We can use the function like <code>cat()</code> that allows you to concatenate and print several objects.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(my_numbers,my_names)</span>
<span id="cb36-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="co">#1 2 3 4 5 6 7 8 9 10 juan c santos 123</span></span></code></pre></div>
<p><strong>11)</strong> One common use of the function <code>cat()</code> is to provide users with screen reports (i.e., print on the screen) on the status of a process that the computer is doing while running iterations or loops that are recursive and time consuming.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">Print the two vectors my_numbers and my_names separed by *** &quot;</span>,</span>
<span id="cb37-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb37-2" aria-hidden="true" tabindex="-1"></a>      my_numbers,</span>
<span id="cb37-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb37-3" aria-hidden="true" tabindex="-1"></a>      <span class="st">&quot; *** &quot;</span>,</span>
<span id="cb37-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb37-4" aria-hidden="true" tabindex="-1"></a>      my_names)</span>
<span id="cb37-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Print the two vectors my_numbers and my_names separed by ***  1 2 3 4 5 6 7 8 9 10  ***  juan c santos 123</span></span></code></pre></div>
<p><strong>12)</strong> You can also list all object “in memory” that is to display the names of objects in your
workspace using function <code>ls()</code>.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span>()</span>
<span id="cb38-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;my_matrix&quot;  &quot;my_names&quot;   &quot;my_numbers&quot; &quot;my_vector&quot; </span></span></code></pre></div>
<p>You can also list objects and their structure with the function <code>ls.str()</code>.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls.str</span>()</span>
<span id="cb39-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="co">#my_matrix :  num [1:3, 1:2] 1 2 3 4 5 6</span></span>
<span id="cb39-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="co">#my_names :  chr [1:3] &quot;juan&quot; &quot;c&quot; &quot;santos&quot; &quot;123&quot;</span></span>
<span id="cb39-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="co">#my_numbers :  int [1:10] 1 2 3 4 5 6 7 8 9 10</span></span>
<span id="cb39-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="co">#my_vector :  num [1:1000] 50.3 60.3 20.4 44.1 43.8 ... </span></span></code></pre></div>
<p><strong>13)</strong> You can also remove objects from your workspace with the function <code>rm()</code>.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb40-1" aria-hidden="true" tabindex="-1"></a>my_names</span>
<span id="cb40-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;juan&quot;   &quot;c&quot;      &quot;santos&quot; &quot;123&quot;</span></span>
<span id="cb40-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="st">&quot;my_names&quot;</span>)</span>
<span id="cb40-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb40-4" aria-hidden="true" tabindex="-1"></a>my_names</span>
<span id="cb40-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb40-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Error: object &#39;my_names&#39; not found</span></span>
<span id="cb40-6"><a href="using-r-installing-packages-and-importingexporting-data.html#cb40-6" aria-hidden="true" tabindex="-1"></a><span class="co">#In this case the object &quot;my_names&quot; has been removed from the workspace</span></span></code></pre></div>
</div>
<div id="functions-between-two-or-more-vectors" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Functions between two or more vectors<a href="using-r-installing-packages-and-importingexporting-data.html#functions-between-two-or-more-vectors" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>1)</strong> You can compare the elements of two vector using ** Set Operations **. These vectors could be numeric and character.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># we create two vector</span></span>
<span id="cb41-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-2" aria-hidden="true" tabindex="-1"></a>my_vector_A <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">7</span>,<span class="dv">9</span>,<span class="dv">10</span>)</span>
<span id="cb41-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-3" aria-hidden="true" tabindex="-1"></a>my_vector_B <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">5</span>,<span class="dv">7</span>,<span class="dv">9</span>)</span>
<span id="cb41-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-5" aria-hidden="true" tabindex="-1"></a><span class="co"># get union of these two vectors (i.e., all values, not repeated, in these two vectors)</span></span>
<span id="cb41-6"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-6" aria-hidden="true" tabindex="-1"></a><span class="fu">union</span>(my_vector_A,my_vector_B)</span>
<span id="cb41-7"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-7" aria-hidden="true" tabindex="-1"></a><span class="co">#[1]  1  3  4  5  6  7  9 10</span></span>
<span id="cb41-8"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-9"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-9" aria-hidden="true" tabindex="-1"></a><span class="co"># get intersect of these two vectors (i.e., values in common between both vectors)</span></span>
<span id="cb41-10"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-10" aria-hidden="true" tabindex="-1"></a><span class="fu">intersect</span>(my_vector_A,my_vector_B)</span>
<span id="cb41-11"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-11" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] 1 3 5 7 9</span></span>
<span id="cb41-12"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-13"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-13" aria-hidden="true" tabindex="-1"></a><span class="co"># get values unique to one of the vectors.</span></span>
<span id="cb41-14"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-14" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(my_vector_A,my_vector_B)<span class="co"># unique to my_vector_A</span></span>
<span id="cb41-15"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-15" aria-hidden="true" tabindex="-1"></a><span class="co">#[1]  4  6 10</span></span>
<span id="cb41-16"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-16" aria-hidden="true" tabindex="-1"></a><span class="fu">setdiff</span>(my_vector_B,my_vector_A)<span class="co"># unique to my_vector_B</span></span>
<span id="cb41-17"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-17" aria-hidden="true" tabindex="-1"></a><span class="co">#numeric(0) -- in this case there are not unique elements for my_vector_B</span></span>
<span id="cb41-18"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-19"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Are the elements of these two vectors identical</span></span>
<span id="cb41-20"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-20" aria-hidden="true" tabindex="-1"></a><span class="fu">setequal</span>(my_vector_A,my_vector_B)</span>
<span id="cb41-21"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-21" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] FALSE</span></span>
<span id="cb41-22"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-23"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Is an element present in a vector?</span></span>
<span id="cb41-24"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-24" aria-hidden="true" tabindex="-1"></a><span class="fu">is.element</span>(<span class="dv">6</span>, my_vector_A) <span class="co"># 6 is present in my_vector_A</span></span>
<span id="cb41-25"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-25" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] TRUE</span></span>
<span id="cb41-26"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-26" aria-hidden="true" tabindex="-1"></a><span class="fu">is.element</span>(<span class="dv">6</span>, my_vector_B) <span class="co"># 6 is NOT present in my_vector_B</span></span>
<span id="cb41-27"><a href="using-r-installing-packages-and-importingexporting-data.html#cb41-27" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] FALSE</span></span></code></pre></div>
<p><strong>2)</strong> Some other basic functions between two numeric vectors.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb42-1" aria-hidden="true" tabindex="-1"></a>my_vector2 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>)</span>
<span id="cb42-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="co"># get the correlation between two vectors</span></span>
<span id="cb42-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(my_vector, my_vector2) <span class="co"># Correlation between two numeric vectors</span></span>
<span id="cb42-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb42-4" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] -0.001297859</span></span>
<span id="cb42-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb42-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-6"><a href="using-r-installing-packages-and-importingexporting-data.html#cb42-6" aria-hidden="true" tabindex="-1"></a><span class="co"># get the covariance between two vectors</span></span>
<span id="cb42-7"><a href="using-r-installing-packages-and-importingexporting-data.html#cb42-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cov</span>(my_vector, my_vector2) <span class="co"># Covariance between two numeric vectors</span></span>
<span id="cb42-8"><a href="using-r-installing-packages-and-importingexporting-data.html#cb42-8" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] -9.390692</span></span></code></pre></div>
</div>
<div id="about-r-packages" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> About R-packages<a href="using-r-installing-packages-and-importingexporting-data.html#about-r-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>What are R packages?</strong></p>
<p>A formal definition is that a R package is collection of code, scripts of functions, datasets, add-ons, and plug-ins that extend the base R statistical programming language and might include code written in other languages like C, Python, Fortran, etc. In most cases, packages contain R-code scripts, preformatted data, and extensive function documentation for help in a format that is standardized (i.e., easy to read and similar across packages). The large number and diversity of R-packages plus their easy installation have favored the widespread use of R in data science and bioinformatics.</p>
<p><strong>How are R packages loaded in the R environment?</strong></p>
<p>R-packages need to be installed first (see below) before they can be loaded and used in the R environment. Once a R-package is installed, it can be called and loaded using the function <code>library()</code>. The packages are stored under a <strong>library</strong> directory called in the R environment by <code>library()</code>. However, some R-packages have add-ons or depended on other packages which need to be installed (R usually will determine and try to install the required packages). However, some R-packages might require other software that is not R-based to be able to run and should preinstalled beforehand. For this workshop, we will install several packages based on data analyses, graphs and other computational needs.</p>
<p><strong>NOTE:</strong> you <strong>DO NOT need to install</strong> again and again the same R-package every time that you start a new session in R. Once you have installed a R-package, it stays in library folder and you just need to call it with <code>library()</code>.</p>
<p><strong>Before you install an R package</strong></p>
<p>One of the strengths of the R environment is the growing set of packages that can be installed and loaded to do specific analyses, plots, calculations and data management within this environment. To date (01/23/24), a total of <a href="https://cran.r-project.org/" target="_blank">CRAN</a> repository features <strong>a total of 20313 available packages</strong> written by the R community. To find all available packages, you can check the left menu under <em>Software</em> and click on <a href="https://cran.r-project.org/web/packages" target="_blank"><strong>Packages</strong></a>.</p>
<p>These are usually two list based on <strong>(1) sorted by date of publication</strong> and <strong>(2) sorted by name</strong>.</p>
<p>Some of the few disadvantages of such diversity of R-packages is that many of these packages might overlap in their functions, their function names, vary in the difficulty of their implementations, bugs and inherent errors in the scripts/functions (i.e., there is <strong>no warranty</strong> that they will work as described, the user must evaluate the output). Given the open nature of these R-packages, the user can access the source code and modify, improve, adapt, combine, repurpose, etc.</p>
<p>Some packages eventually become <strong>obsolete</strong> and disappear from CRAN if they are not maintained. However, those can be found in CRAN archives and might require a more convoluted installation (described below).</p>
</div>
<div id="cran-task-views-expert-opinion-on-useful-packages" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> CRAN Task Views – Expert opinion on useful packages<a href="using-r-installing-packages-and-importingexporting-data.html#cran-task-views-expert-opinion-on-useful-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Before you install a package, you could check which are useful to your intended goal. In most cases, you will already know what you want. However, if you are unsure, you could check the suggestions by R community posted in the <a href="http://cran.r-project.org/web/views" target="_blank"><strong>CRAN Task Views</strong></a>. Here are some of the communities relevant for our workshop and those based on our student background:</p>
<p><a href="https://cran.r-project.org/web/views/ClinicalTrials.html" target="_blank"><strong>CRAN Task View: Clinical Trial Design</strong></a></p>
<p><a href="https://cran.r-project.org/web/views/Databases.html" target="_blank"><strong>CRAN Task View: Databases</strong></a></p>
<p><a href="https://cran.r-project.org/web/views/Phylogenetics.html" target="_blank"><strong>CRAN Task View: Phylogenetics, Especially Comparative Methods</strong></a></p>
<p><a href="https://cran.r-project.org/web/views/TeachingStatistics.html" target="_blank"><strong>CRAN Task View: Teaching Statistics</strong></a></p>
<p><a href="https://cran.r-project.org/web/views/ExperimentalDesign.html" target="_blank"><strong>CRAN Task View: Design of Experiments</strong></a></p>
<p><a href="https://cran.r-project.org/web/views/Bayesian.html" target="_blank"><strong>CRAN Task View: Bayesian Inference</strong></a></p>
<p><a href="https://cran.r-project.org/web/views/Pharmacokinetics.html" target="_blank"><strong>CRAN Task View: Analysis of Pharmacokinetic Data</strong></a></p>
<p><a href="https://cran.r-project.org/web/views/Omics.html" target="_blank"><strong>CRAN Task View: Genomics, Proteomics, Metabolomics, Transcriptomics, and Other Omics</strong></a></p>
</div>
<div id="installing-packages" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> Installing packages<a href="using-r-installing-packages-and-importingexporting-data.html#installing-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>1)</strong> Check what packages are preloaded or were loaded in the R environment before installing new one using the function <code>search()</code>.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">search</span>()</span>
<span id="cb43-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb43-2" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] &quot;.GlobalEnv&quot;        &quot;tools:RGUI&quot;        &quot;package:stats&quot;     &quot;package:graphics&quot;  &quot;package:grDevices&quot; &quot;package:utils&quot;    </span></span>
<span id="cb43-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb43-3" aria-hidden="true" tabindex="-1"></a><span class="co">#[7] &quot;package:datasets&quot;  &quot;package:methods&quot;   &quot;Autoloads&quot;         &quot;package:base&quot; </span></span></code></pre></div>
<p><strong>2)</strong> To install a package that you want and you can use <code>install.packages()</code> as indicated below.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;reshape2&quot;</span>)</span></code></pre></div>
<p>You will be prompted to select a CRAN mirror. Usually take the one geographically close like: USA (OH) [https] or the cloud: 0-Cloud [http]. Once you have accepted it will proceed to install. Sometimes, some other packages are required and R will install those during the process. In some rare cases, you might need to install those on your own (i.e., not authomatic installation)</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;reshape2&quot;</span>)</span>
<span id="cb45-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="co">#--- Please select a CRAN mirror for use in this session ---</span></span>
<span id="cb45-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="co">#trying URL &#39;https://cloud.r-project.org/bin/macosx/el-capitan/contrib/3.6/reshape2_1.4.4.tgz&#39;</span></span>
<span id="cb45-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Content type &#39;application/x-gzip&#39; length 333148 bytes (325 KB)</span></span>
<span id="cb45-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb45-5" aria-hidden="true" tabindex="-1"></a><span class="co">#==================================================</span></span>
<span id="cb45-6"><a href="using-r-installing-packages-and-importingexporting-data.html#cb45-6" aria-hidden="true" tabindex="-1"></a><span class="co">#downloaded 325 KB</span></span>
<span id="cb45-7"><a href="using-r-installing-packages-and-importingexporting-data.html#cb45-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-8"><a href="using-r-installing-packages-and-importingexporting-data.html#cb45-8" aria-hidden="true" tabindex="-1"></a><span class="co">#The downloaded binary packages are in</span></span>
<span id="cb45-9"><a href="using-r-installing-packages-and-importingexporting-data.html#cb45-9" aria-hidden="true" tabindex="-1"></a><span class="co">#   /var/folders/vx/rjmrkyz566zf4khl9xjhkqyr0000gp/T//RtmpwxVTvt/downloaded_packages</span></span></code></pre></div>
<p><strong>3)</strong> For PC users, you might need to define a folder (R library) where you R-packages will be kept.</p>
<p><strong>4)</strong> To load a R-package already installed, call for its name using <code>library()</code>.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape)</span></code></pre></div>
<p>if the library is successfully loaded, you will get <strong>NO RESPONSE</strong> in most cases as above. Some packages will report some information about the package being loaded successfully.</p>
<p><strong>5)</strong> If errors occurs and the selected R-package was not installed, R will let know that there is a problem in some form.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="do">## if the package name is misspelled or does not exist in CRAN then:</span></span>
<span id="cb47-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;my_madeup_package&quot;</span>)</span>
<span id="cb47-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Warning message:</span></span>
<span id="cb47-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="co"># package ‘my_madeup_package’ is not available (for R version 3.6.2) </span></span></code></pre></div>
<p><strong>6)</strong> If did not installed the R-package beforehand, R will also let you know that the package is not installed in your library.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(my_madeup_package)</span>
<span id="cb48-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Error in library(my_madeup_package) : </span></span>
<span id="cb48-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb48-3" aria-hidden="true" tabindex="-1"></a><span class="co"># there is no package called ‘my_madeup_package’</span></span></code></pre></div>
<p><strong>7)</strong> To install packages from older versions or no longer maintained in CRAN, you need to install this R-package ‘devtools’.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb49-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span></code></pre></div>
<p><strong>8)</strong> For example, we will install an older version (i.e., ver 1.4.3) of the R-package ‘FlexParamCurve’, which is a statistics tool to fit complex parametric curves.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install_version</span>(<span class="st">&quot;FlexParamCurve&quot;</span>, <span class="at">version =</span> <span class="st">&quot;1.4.3&quot;</span>, <span class="at">repos =</span> <span class="st">&quot;http://cran.us.r-project.org&quot;</span>)</span>
<span id="cb50-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Downloading package from url: http://cran.us.r-project.org/src/contrib/Archive/FlexParamCurve/FlexParamCurve_1.4-3.tar.gz</span></span>
<span id="cb50-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="co">#* installing *source* package ‘FlexParamCurve’ ...</span></span>
<span id="cb50-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="co">#** package ‘FlexParamCurve’ successfully unpacked and MD5 sums checked</span></span>
<span id="cb50-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="co">#** using staged installation</span></span>
<span id="cb50-6"><a href="using-r-installing-packages-and-importingexporting-data.html#cb50-6" aria-hidden="true" tabindex="-1"></a><span class="co">#** R</span></span>
<span id="cb50-7"><a href="using-r-installing-packages-and-importingexporting-data.html#cb50-7" aria-hidden="true" tabindex="-1"></a><span class="co">#** data</span></span>
<span id="cb50-8"><a href="using-r-installing-packages-and-importingexporting-data.html#cb50-8" aria-hidden="true" tabindex="-1"></a><span class="co">#** byte-compile and prepare package for lazy loading</span></span>
<span id="cb50-9"><a href="using-r-installing-packages-and-importingexporting-data.html#cb50-9" aria-hidden="true" tabindex="-1"></a><span class="co">#** help</span></span>
<span id="cb50-10"><a href="using-r-installing-packages-and-importingexporting-data.html#cb50-10" aria-hidden="true" tabindex="-1"></a><span class="co">#*** installing help indices</span></span>
<span id="cb50-11"><a href="using-r-installing-packages-and-importingexporting-data.html#cb50-11" aria-hidden="true" tabindex="-1"></a><span class="co">#** building package indices</span></span>
<span id="cb50-12"><a href="using-r-installing-packages-and-importingexporting-data.html#cb50-12" aria-hidden="true" tabindex="-1"></a><span class="co">#** testing if installed package can be loaded from temporary location</span></span>
<span id="cb50-13"><a href="using-r-installing-packages-and-importingexporting-data.html#cb50-13" aria-hidden="true" tabindex="-1"></a><span class="co">#** testing if installed package can be loaded from final location</span></span>
<span id="cb50-14"><a href="using-r-installing-packages-and-importingexporting-data.html#cb50-14" aria-hidden="true" tabindex="-1"></a><span class="co">#** testing if installed package keeps a record of temporary installation path</span></span>
<span id="cb50-15"><a href="using-r-installing-packages-and-importingexporting-data.html#cb50-15" aria-hidden="true" tabindex="-1"></a><span class="co">#* DONE (FlexParamCurve)</span></span></code></pre></div>
<p><strong>9)</strong> You can check the version of the R-packages that just installed like ‘devtools’ or ‘FlexParamCurve’ with function <code>packageVersion()</code>.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb51-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="fu">packageVersion</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb51-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] ‘2.4.5’</span></span>
<span id="cb51-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb51-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(FlexParamCurve)</span>
<span id="cb51-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb51-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Loading required package: nlme</span></span>
<span id="cb51-6"><a href="using-r-installing-packages-and-importingexporting-data.html#cb51-6" aria-hidden="true" tabindex="-1"></a><span class="fu">packageVersion</span>(<span class="st">&quot;FlexParamCurve&quot;</span>)</span>
<span id="cb51-7"><a href="using-r-installing-packages-and-importingexporting-data.html#cb51-7" aria-hidden="true" tabindex="-1"></a><span class="co">#[1] ‘1.4.3’</span></span></code></pre></div>
<p>This information is useful if you what to know the current version of a R-package that has been loaded after using ‘library()’.</p>
<p><strong>10)</strong> More detailed information about any R-package can be obtained with the function ‘packageDescription()’.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">packageDescription</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb52-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="co">#Package: devtools</span></span>
<span id="cb52-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Title: Tools to Make Developing R Packages Easier</span></span>
<span id="cb52-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-4" aria-hidden="true" tabindex="-1"></a><span class="co">#Version: 2.4.5</span></span>
<span id="cb52-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-5" aria-hidden="true" tabindex="-1"></a><span class="co">#Authors@R: c( person(&quot;Hadley&quot;, &quot;Wickham&quot;, role = &quot;aut&quot;), person(&quot;Jim&quot;, &quot;Hester&quot;, role = &quot;aut&quot;), person(&quot;Winston&quot;, &quot;Chang&quot;, role = &quot;aut&quot;), person(&quot;Jennifer&quot;, &quot;Bryan&quot;,</span></span>
<span id="cb52-6"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-6" aria-hidden="true" tabindex="-1"></a><span class="co">#                  , &quot;jenny@rstudio.com&quot;, role = c(&quot;aut&quot;, &quot;cre&quot;), comment = c(ORCID = &quot;0000-0002-6983-2759&quot;)), person(&quot;RStudio&quot;, role = c(&quot;cph&quot;, &quot;fnd&quot;)) )</span></span>
<span id="cb52-7"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Description: Collection of package development tools.</span></span>
<span id="cb52-8"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-8" aria-hidden="true" tabindex="-1"></a><span class="co">#License: MIT + file LICENSE</span></span>
<span id="cb52-9"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-9" aria-hidden="true" tabindex="-1"></a><span class="co">#URL: https://devtools.r-lib.org/, https://github.com/r-lib/devtools</span></span>
<span id="cb52-10"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-10" aria-hidden="true" tabindex="-1"></a><span class="co">#BugReports: https://github.com/r-lib/devtools/issues</span></span>
<span id="cb52-11"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Depends: R (&gt;= 3.0.2), usethis (&gt;= 2.1.6)</span></span>
<span id="cb52-12"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-12" aria-hidden="true" tabindex="-1"></a><span class="co">#Imports: cli (&gt;= 3.3.0), desc (&gt;= 1.4.1), ellipsis (&gt;= 0.3.2), fs (&gt;= 1.5.2), lifecycle (&gt;= 1.0.1), memoise (&gt;= 2.0.1), miniUI (&gt;= 0.1.1.1), pkgbuild (&gt;= 1.3.1),</span></span>
<span id="cb52-13"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-13" aria-hidden="true" tabindex="-1"></a><span class="co">#                  pkgdown (&gt;= 2.0.6), pkgload (&gt;= 1.3.0), profvis (&gt;= 0.3.7), rcmdcheck (&gt;= 1.4.0), remotes (&gt;= 2.4.2), rlang (&gt;= 1.0.4), roxygen2 (&gt;= 7.2.1),</span></span>
<span id="cb52-14"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-14" aria-hidden="true" tabindex="-1"></a><span class="co">#                  rversions (&gt;= 2.1.1), sessioninfo (&gt;= 1.2.2), stats, testthat (&gt;= 3.1.5), tools, urlchecker (&gt;= 1.0.1), utils, withr (&gt;= 2.5.0)</span></span>
<span id="cb52-15"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-15" aria-hidden="true" tabindex="-1"></a><span class="co">#Suggests: BiocManager (&gt;= 1.30.18), callr (&gt;= 3.7.1), covr (&gt;= 3.5.1), curl (&gt;= 4.3.2), digest (&gt;= 0.6.29), DT (&gt;= 0.23), foghorn (&gt;= 1.4.2), gh (&gt;= 1.3.0), gmailr</span></span>
<span id="cb52-16"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-16" aria-hidden="true" tabindex="-1"></a><span class="co">#                  (&gt;= 1.0.1), httr (&gt;= 1.4.3), knitr (&gt;= 1.39), lintr (&gt;= 3.0.0), MASS, mockery (&gt;= 0.4.3), pingr (&gt;= 2.0.1), rhub (&gt;= 1.1.1), rmarkdown (&gt;= 2.14),</span></span>
<span id="cb52-17"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-17" aria-hidden="true" tabindex="-1"></a><span class="co">#                  rstudioapi (&gt;= 0.13), spelling (&gt;= 2.2)</span></span>
<span id="cb52-18"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-18" aria-hidden="true" tabindex="-1"></a><span class="co">#VignetteBuilder: knitr</span></span>
<span id="cb52-19"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-19" aria-hidden="true" tabindex="-1"></a><span class="co">#Config/Needs/website: tidyverse/tidytemplate</span></span>
<span id="cb52-20"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-20" aria-hidden="true" tabindex="-1"></a><span class="co">#Encoding: UTF-8</span></span>
<span id="cb52-21"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-21" aria-hidden="true" tabindex="-1"></a><span class="co">#Language: en-US</span></span>
<span id="cb52-22"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-22" aria-hidden="true" tabindex="-1"></a><span class="co">#RoxygenNote: 7.2.1</span></span>
<span id="cb52-23"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-23" aria-hidden="true" tabindex="-1"></a><span class="co">#Config/testthat/edition: 3</span></span>
<span id="cb52-24"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-24" aria-hidden="true" tabindex="-1"></a><span class="co">#NeedsCompilation: no</span></span>
<span id="cb52-25"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-25" aria-hidden="true" tabindex="-1"></a><span class="co">#Packaged: 2022-10-11 16:13:16 UTC; jenny</span></span>
<span id="cb52-26"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-26" aria-hidden="true" tabindex="-1"></a><span class="co">#Author: Hadley Wickham [aut], Jim Hester [aut], Winston Chang [aut], Jennifer Bryan [aut, cre] (&lt;https://orcid.org/0000-0002-6983-2759&gt;), RStudio [cph, fnd]</span></span>
<span id="cb52-27"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-27" aria-hidden="true" tabindex="-1"></a><span class="co">#Maintainer: Jennifer Bryan &lt;jenny@rstudio.com&gt;</span></span>
<span id="cb52-28"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-28" aria-hidden="true" tabindex="-1"></a><span class="co">#Repository: CRAN</span></span>
<span id="cb52-29"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-29" aria-hidden="true" tabindex="-1"></a><span class="co">#Date/Publication: 2022-10-11 17:12:36 UTC</span></span>
<span id="cb52-30"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-30" aria-hidden="true" tabindex="-1"></a><span class="co">#Built: R 4.1.2; ; 2022-10-12 11:12:20 UTC; unix</span></span>
<span id="cb52-31"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-31" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb52-32"><a href="using-r-installing-packages-and-importingexporting-data.html#cb52-32" aria-hidden="true" tabindex="-1"></a><span class="co">#-- File: /Library/Frameworks/R.framework/Versions/4.1/Resources/library/devtools/Meta/package.rds </span></span></code></pre></div>
<p><strong>11)</strong> You can also get information about a R-package for citation in an article or report using the function <code>citation()</code>.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb53-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="fu">citation</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb53-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-3" aria-hidden="true" tabindex="-1"></a><span class="co">#To cite package ‘devtools’ in publications use:</span></span>
<span id="cb53-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-4" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb53-5"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-5" aria-hidden="true" tabindex="-1"></a><span class="co">#  Hadley Wickham, Jim Hester, Winston Chang and Jennifer Bryan (2022). devtools: Tools to Make Developing R Packages Easier. R package version 2.4.5.</span></span>
<span id="cb53-6"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  https://CRAN.R-project.org/package=devtools</span></span>
<span id="cb53-7"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-7" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb53-8"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-8" aria-hidden="true" tabindex="-1"></a><span class="co">#A BibTeX entry for LaTeX users is</span></span>
<span id="cb53-9"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-9" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb53-10"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-10" aria-hidden="true" tabindex="-1"></a><span class="co">#  @Manual{,</span></span>
<span id="cb53-11"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-11" aria-hidden="true" tabindex="-1"></a><span class="co">#    title = {devtools: Tools to Make Developing R Packages Easier},</span></span>
<span id="cb53-12"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-12" aria-hidden="true" tabindex="-1"></a><span class="co">#    author = {Hadley Wickham and Jim Hester and Winston Chang and Jennifer Bryan},</span></span>
<span id="cb53-13"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-13" aria-hidden="true" tabindex="-1"></a><span class="co">#    year = {2022},</span></span>
<span id="cb53-14"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-14" aria-hidden="true" tabindex="-1"></a><span class="co">#    note = {R package version 2.4.5},</span></span>
<span id="cb53-15"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-15" aria-hidden="true" tabindex="-1"></a><span class="co">#    url = {https://CRAN.R-project.org/package=devtools},</span></span>
<span id="cb53-16"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-16" aria-hidden="true" tabindex="-1"></a><span class="co">#  }</span></span>
<span id="cb53-17"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-18"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(FlexParamCurve)</span>
<span id="cb53-19"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-19" aria-hidden="true" tabindex="-1"></a><span class="co">#Loading required package: nlme</span></span>
<span id="cb53-20"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-20" aria-hidden="true" tabindex="-1"></a><span class="fu">citation</span>(<span class="st">&quot;FlexParamCurve&quot;</span>)</span>
<span id="cb53-21"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-21" aria-hidden="true" tabindex="-1"></a><span class="co">#To cite package ‘FlexParamCurve’ in publications use:</span></span>
<span id="cb53-22"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-22" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb53-23"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-23" aria-hidden="true" tabindex="-1"></a><span class="co">#  Stephen Oswald (2012). FlexParamCurve: Tools to Fit Flexible Parametric Curves. R package version 1.4-3. https://CRAN.R-project.org/package=FlexParamCurve</span></span>
<span id="cb53-24"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-24" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb53-25"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-25" aria-hidden="true" tabindex="-1"></a><span class="co">#A BibTeX entry for LaTeX users is</span></span>
<span id="cb53-26"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-26" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb53-27"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-27" aria-hidden="true" tabindex="-1"></a><span class="co">#  @Manual{,</span></span>
<span id="cb53-28"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-28" aria-hidden="true" tabindex="-1"></a><span class="co">#    title = {FlexParamCurve: Tools to Fit Flexible Parametric Curves},</span></span>
<span id="cb53-29"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-29" aria-hidden="true" tabindex="-1"></a><span class="co">#    author = {Stephen Oswald},</span></span>
<span id="cb53-30"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-30" aria-hidden="true" tabindex="-1"></a><span class="co">#    year = {2012},</span></span>
<span id="cb53-31"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-31" aria-hidden="true" tabindex="-1"></a><span class="co">#    note = {R package version 1.4-3},</span></span>
<span id="cb53-32"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-32" aria-hidden="true" tabindex="-1"></a><span class="co">#    url = {https://CRAN.R-project.org/package=FlexParamCurve},</span></span>
<span id="cb53-33"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-33" aria-hidden="true" tabindex="-1"></a><span class="co">#  }</span></span>
<span id="cb53-34"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-35"><a href="using-r-installing-packages-and-importingexporting-data.html#cb53-35" aria-hidden="true" tabindex="-1"></a><span class="co">#ATTENTION: This citation information has been auto-generated from the package DESCRIPTION file and may need manual editing, see ‘help(&quot;citation&quot;)’.</span></span></code></pre></div>
<p><strong>12)</strong> Most genetic, genomic, transcriptomic R-packages are <a href="https://www.bioconductor.org/" target="_blank"><strong>Bioconductor</strong></a>. To date (1/23/24), this archive has <strong>a total of 3649 packages available packages</strong>. To install packages present on this archive, you need to follow the next instructions.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="do">## test if &#39;BiocManager&#39; is installed</span></span>
<span id="cb54-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb54-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {<span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)}</span></code></pre></div>
<p>Now, we can use the function <code>BiocManager::install("my_desired_bioconductor_package")</code>. For example, we can install an excellent sequence aligner that is the Bioconductor R-package DECIPHER.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb55-1" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;DECIPHER&quot;</span>)</span></code></pre></div>
<p>Note that R will <strong>ask you to update your older packages</strong>, in most cases you decline (i.e., type ‘n’).</p>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb56-1" aria-hidden="true" tabindex="-1"></a>Update all<span class="sc">/</span>some<span class="sc">/</span>none? [a<span class="sc">/</span>s<span class="sc">/</span>n]<span class="sc">:</span> n</span></code></pre></div>
<p><strong>13)</strong> Some experimental or not yet released R-packages are sometimes stored in the developers’ repositories like <a href="https://github.com/" target="_blank"><strong>GitHub</strong></a> (like this gitbook). This website has become one of the most popular store websites for free software for a diversity of fields. To install such packages you use again ‘devtools’</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="using-r-installing-packages-and-importingexporting-data.html#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb57-2"><a href="using-r-installing-packages-and-importingexporting-data.html#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="co"># install_github(&quot;github_handler/package_name&quot;</span></span>
<span id="cb57-3"><a href="using-r-installing-packages-and-importingexporting-data.html#cb57-3" aria-hidden="true" tabindex="-1"></a><span class="co"># installing httr for web API (application programming interface) like donwload of data from the web</span></span>
<span id="cb57-4"><a href="using-r-installing-packages-and-importingexporting-data.html#cb57-4" aria-hidden="true" tabindex="-1"></a><span class="fu">install_github</span>(<span class="st">&quot;r-lib/httr&quot;</span>)</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="getting-started-with-r.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="basic-data-structures-in-r.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/02-First_session_with_R_v11.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
